--- xloadimage.4.1/Imakefile.nobr	Thu Aug 13 08:17:34 1998
+++ xloadimage.4.1/Imakefile	Thu Aug 13 08:18:14 1998
@@ -48,8 +48,8 @@
 InstallManPage(uufilter,$(MANDIR))
 
 install:: $(SYSPATHFILE)
-	$(RM) $(BINDIR)/xview $(BINDIR)/xsetbg
-	cd $(BINDIR); $(LN) xloadimage xview; $(LN) xloadimage xsetbg
+	$(RM) $(DESTDIR)$(BINDIR)/xview $(DESTDIR)$(BINDIR)/xsetbg
+	cd $(DESTDIR)$(BINDIR); $(LN) xloadimage xview; $(LN) xloadimage xsetbg
 
 clean::
 	/bin/rm -f build.c
