--- xloadimage.4.1/Imakefile.orig	Tue Nov  9 17:26:58 1993
+++ xloadimage.4.1/Imakefile	Fri Oct 24 14:03:52 1997
@@ -22,15 +22,15 @@
 #define HAS_JPEG
 
 #ifdef HAS_TIFF
-TIFF_DEFS = -DHAS_TIFF -I/usr/local/include
- TIFF_LIB = -L/usr/local/lib/tiff -ltiff
+TIFF_DEFS = -DHAS_TIFF
+ TIFF_LIB = -ltiff
  TIFF_SRC = tiff.c
  TIFF_OBJ = tiff.o
 #endif
 
 #ifdef HAS_JPEG
-JPEG_DEFS = -DHAS_JPEG -I/usr/local/include/jpeg
- JPEG_LIB = -L/usr/local/lib/jpeg -ljpeg
+JPEG_DEFS = -DHAS_JPEG 
+ JPEG_LIB = jpeg/libjpeg.a
  JPEG_SRC = jpeg.c
  JPEG_OBJ = jpeg.o
 #endif
--- xloadimage.4.1/misc.c.orig	Thu Oct 21 17:28:39 1993
+++ xloadimage.4.1/misc.c	Fri Oct 24 13:55:34 1997
@@ -21,9 +21,9 @@
 extern char    *ProgramName;
 extern Display *Disp;
 extern int      Scrn;
-extern char    *BuildDate;
-extern char    *BuildUser;
-extern char    *BuildSystem;
+char    *BuildDate = __DATE__;
+char    *BuildUser = "Red Hat";
+char    *BuildSystem = "redhat.com";
 
 static char *signalName(sig)
      int sig;
--- xloadimage.4.1/jpeg.c.orig	Mon Nov  8 13:36:03 1993
+++ xloadimage.4.1/jpeg.c	Fri Oct 24 13:55:34 1997
@@ -7,7 +7,6 @@
  */
 
 #include "image.h"	/* xloadimage declarations */
-#include "jpeg.conf.h"  /* definitions used in jpeg directory */
 #include "jpeg/jinclude.h" /* IJG declarations */
 #include <setjmp.h>	/* need setjmp/longjmp */
 
