--- pybb/feeds.py.orig	2017-12-06 03:30:35.000000000 +0000
+++ pybb/feeds.py	2021-10-04 01:48:00.985607156 +0000
@@ -37,7 +37,7 @@
 
     def items(self, user):
         ids = [p.id for p in perms.filter_posts(user, Post.objects.only('id')).order_by('-created', '-id')[:15]]
-        return Post.objects.filter(id__in=ids).select_related('topic', 'topic__forum', 'user')
+        return Post.objects.filter(id__in=ids).select_related('topic', 'topic__forum', 'user').reverse()
 
 
 class LastTopics(PybbFeed):
