--- pybb/static/markitup/sets/markdown/set.js.orig	2015-08-18 12:52:42.000000000 +0000
+++ pybb/static/markitup/sets/markdown/set.js	2021-06-28 03:32:06.489728079 +0000
@@ -24,16 +24,16 @@
 		{name:'Bold', key:'B', openWith:'**', closeWith:'**'},
 		{name:'Italic', key:'I', openWith:'_', closeWith:'_'},
 		{separator:'---------------' },
-		{name:'Bulleted List', openWith:'- ' },
+		{name:'Bulleted List', openWith:'- ', multiline:true},
 		{name:'Numeric List', openWith:function(markItUp) {
 			return markItUp.line+'. ';
-		}},
+		}, multiline:true},
 		{separator:'---------------' },
 		{name:'Picture', key:'P', replaceWith:'![[![Alternative text]!]]([![Url:!:http://]!] "[![Title]!]")'},
 		{name:'Link', key:'L', openWith:'[', closeWith:']([![Url:!:http://]!] "[![Title]!]")', placeHolder:'Your text to link here...' },
 		{separator:'---------------'},	
-		{name:'Quotes', openWith:'> '},
-		{name:'Code Block / Code', openWith:'(!(\t|!|`)!)', closeWith:'(!(`)!)'},
+		{name:'Quotes', openWith:'> ', multiline:true},
+		{name:'Code Block / Code', openWith:'(!(\t|!|`)!)', closeWith:'(!(`)!)', multiline:true},
 		{separator:'---------------'},
 		{name:'Preview', call:'preview', className:"preview"}
 	]
