--- pybb/models.py.orig	2020-03-21 22:44:43.904089549 +0000
+++ pybb/models.py	2020-03-21 22:44:53.524090627 +0000
@@ -380,6 +380,9 @@
         verbose_name = _('Profile')
         verbose_name_plural = _('Profiles')
 
+    def __str__(self):
+        return self.get_display_name()
+
     def get_absolute_url(self):
         return reverse('pybb:user', kwargs={'username': getattr(self.user, get_username_field())})
 
