From 97774cb149c5b03d5ef82a5af3f19e2ce4d79d0b Mon Sep 17 00:00:00 2001
From: John (J5) Palmieri <johnp@redhat.com>
Date: Mon, 04 Oct 2010 16:43:31 +0000
Subject: return NULL instead of -1 which fixes crash when introspection is turned off

* see https://bugzilla.gnome.org/show_bug.cgi?id=631158
---
diff --git a/gi/pygi.h b/gi/pygi.h
index 551bb6f..da71d28 100644
--- a/gi/pygi.h
+++ b/gi/pygi.h
@@ -155,7 +155,7 @@ static inline PyObject *
 pygi_get_property_value (PyGObject *instance,
                          const gchar *attr_name)
 {
-    return -1;
+    return NULL;
 }
 
 static inline gint
--
cgit v0.8.3.1
