--- error.cc	2007/07/06 19:08:17	1.7
+++ error.cc	2009/04/11 04:11:24	1.8
@@ -32,6 +32,7 @@
 #include <iostream>
 #include <sstream>
 #include <iomanip>
+#include <cstdio>
 
 using std::ostringstream;
 using std::string;
--- smiltime.cc	2008/11/26 03:49:43	1.7
+++ smiltime.cc	2009/04/11 04:11:24	1.8
@@ -27,6 +27,7 @@
 #include <iostream>
 #include <iomanip>
 #include <sstream>
+#include <cstdio>
 
 #include <stdlib.h>
 #include <math.h>
