diff -uNr libgphoto2-2.4.7/camlibs/ptp2/library.c libgphoto2-2.4.7.cp/camlibs/ptp2/library.c
--- libgphoto2-2.4.7/camlibs/ptp2/library.c	2009-08-16 15:58:17.000000000 +0200
+++ libgphoto2-2.4.7.cp/camlibs/ptp2/library.c	2009-09-02 18:51:18.000000000 +0200
@@ -619,6 +619,8 @@
 	{"Nikon:Coolpix P50 (PTP mode)",  0x04b0, 0x0169, 0},
 	/*   http://bugs.debian.org/520752 */
 	{"Nikon:Coolpix S60 (PTP mode)",  0x04b0, 0x0171, 0},
+        /* Nikon:Coolpix P90: Treeve Jelbert 2009.08.31*/
+	{"Nikon:Coolpix P90 (PTP mode)",  0x04b0, 0x0173, PTP_CAP|PTP_NIKON_BROKEN_CAP},
 	{"Nikon:Coolpix SQ (PTP mode)",   0x04b0, 0x0202, 0},
 	/* lars marowski bree, 16.8.2004 */
 	{"Nikon:Coolpix 4200 (PTP mode)", 0x04b0, 0x0204, 0},
@@ -716,6 +718,8 @@
 	{"Casio:EX-S770",                 0x07cf, 0x1049, 0},
 	/* https://launchpad.net/bugs/64146 */
 	{"Casio:EX-Z700",                 0x07cf, 0x104c, 0},
+	/* Casio:EX-Z150 (PTP mode): Treeve Jelbert 2009/09/02 */
+	{"Casio:EX-Z150",                 0x07cf, 0x120f, 0},
 
 	/* (at least some) newer Canon cameras can be switched between
 	 * PTP and "normal" (i.e. Canon) mode
