diff -uNr libgphoto2-2.4.8/camlibs/ptp2/library.c libgphoto2-2.4.8.cp/camlibs/ptp2/library.c
--- libgphoto2-2.4.8/camlibs/ptp2/library.c	2010-01-22 20:20:57.000000000 +0100
+++ libgphoto2-2.4.8.cp/camlibs/ptp2/library.c	2010-01-26 13:44:05.000000000 +0100
@@ -647,6 +647,8 @@
 	{"Nikon:Coolpix P50 (PTP mode)",  0x04b0, 0x0169, 0},
 	/*   http://bugs.debian.org/520752 */
 	{"Nikon:Coolpix S60 (PTP mode)",  0x04b0, 0x0171, 0},
+        /* Nikon:Coolpix P90: Treeve Jelbert 2009.08.31*/
+	{"Nikon:Coolpix P90 (PTP mode)",  0x04b0, 0x0173, PTP_CAP|PTP_NIKON_BROKEN_CAP},
 	/* Christoph Muehlmann <c.muehlmann@nagnag.de> */
 	{"Nikon:Coolpix S220 (PTP mode)", 0x04b0, 0x0177, PTP_CAP|PTP_NIKON_BROKEN_CAP},
 	/* */
@@ -755,6 +757,8 @@
 	{"Casio:EX-S770",                 0x07cf, 0x1049, 0},
 	/* https://launchpad.net/bugs/64146 */
 	{"Casio:EX-Z700",                 0x07cf, 0x104c, 0},
+	/* Casio:EX-Z150 (PTP mode): Treeve Jelbert 2009/09/02 */
+	{"Casio:EX-Z150",                 0x07cf, 0x120f, 0},
 
 	/* (at least some) newer Canon cameras can be switched between
 	 * PTP and "normal" (i.e. Canon) mode
