diff -uNr libgphoto2-2.4.11.orig//camlibs/ptp2/library.c libgphoto2-2.4.11/camlibs/ptp2/library.c
--- libgphoto2-2.4.11.orig//camlibs/ptp2/library.c	2011-04-17 16:22:30.000000000 +0200
+++ libgphoto2-2.4.11/camlibs/ptp2/library.c	2011-05-01 21:47:04.000000000 +0200
@@ -843,6 +843,8 @@
 	{"Casio:EX-Z700",                 0x07cf, 0x104c, 0},
 	/* IRC Reporter */
 	{"Casio:EX-Z65",                  0x07cf, 0x104d, 0},
+        /* Casio:EX-Z150 (PTP mode): Treeve Jelbert 2009/09/02 */
+        {"Casio:EX-Z150",                 0x07cf, 0x120f, 0},
 
 	/* (at least some) newer Canon cameras can be switched between
 	 * PTP and "normal" (i.e. Canon) mode
