--- Makefile.in.old	2006-04-27 12:43:03.000000000 +0200
+++ Makefile.in	2006-04-27 12:43:25.000000000 +0200
@@ -295,7 +295,7 @@
 EXTRA_DIST = HACKING MAINTAINERS makefile.os2 os2mak.inc TESTERS CHANGES
 
 # Note: @subdirs@ lists all the directories from AC_CONFIG_SUBDIRS()
-SUBDIRS = @subdirs@ libgphoto2 camlibs tests po packaging doc
+SUBDIRS = @subdirs@ libgphoto2 camlibs tests po packaging
 doc_DATA = AUTHORS COPYING README CHANGES NEWS
 
 ########################################################################
