Only in planner.mod: config.log
diff -u -r planner-0.14.3/configure planner.mod/configure
--- planner-0.14.3/configure	2008-04-12 11:14:43.000000000 -0500
+++ planner.mod/configure	2008-04-19 18:41:45.000000000 -0500
@@ -25655,13 +25655,13 @@
 		libecal-\$EDS_PACKAGE >= \$EDS_REQUIRED
     		libedata-cal-\$EDS_PACKAGE >= \$EDS_REQUIRED
     		libedataserver-\$EDS_PACKAGE >= \$EDS_REQUIRED
-    		evolution-plugin-\$EVOLUTION_PACKAGE >= \$EVOLUTION_REQUIRED
+    		evolution-plugin >= \$EVOLUTION_REQUIRED
     \"") >&5
   ($PKG_CONFIG --exists --print-errors "
 		libecal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedata-cal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedataserver-$EDS_PACKAGE >= $EDS_REQUIRED
-    		evolution-plugin-$EVOLUTION_PACKAGE >= $EVOLUTION_REQUIRED
+    		evolution-plugin >= $EVOLUTION_REQUIRED
     ") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
@@ -25670,7 +25670,7 @@
 		libecal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedata-cal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedataserver-$EDS_PACKAGE >= $EDS_REQUIRED
-    		evolution-plugin-$EVOLUTION_PACKAGE >= $EVOLUTION_REQUIRED
+    		evolution-plugin >= $EVOLUTION_REQUIRED
     " 2>/dev/null`
 else
   pkg_failed=yes
@@ -25688,13 +25688,13 @@
 		libecal-\$EDS_PACKAGE >= \$EDS_REQUIRED
     		libedata-cal-\$EDS_PACKAGE >= \$EDS_REQUIRED
     		libedataserver-\$EDS_PACKAGE >= \$EDS_REQUIRED
-    		evolution-plugin-\$EVOLUTION_PACKAGE >= \$EVOLUTION_REQUIRED
+    		evolution-plugin >= \$EVOLUTION_REQUIRED
     \"") >&5
   ($PKG_CONFIG --exists --print-errors "
 		libecal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedata-cal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedataserver-$EDS_PACKAGE >= $EDS_REQUIRED
-    		evolution-plugin-$EVOLUTION_PACKAGE >= $EVOLUTION_REQUIRED
+    		evolution-plugin >= $EVOLUTION_REQUIRED
     ") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
@@ -25703,7 +25703,7 @@
 		libecal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedata-cal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedataserver-$EDS_PACKAGE >= $EDS_REQUIRED
-    		evolution-plugin-$EVOLUTION_PACKAGE >= $EVOLUTION_REQUIRED
+    		evolution-plugin >= $EVOLUTION_REQUIRED
     " 2>/dev/null`
 else
   pkg_failed=yes
@@ -25727,14 +25727,14 @@
 		libecal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedata-cal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedataserver-$EDS_PACKAGE >= $EDS_REQUIRED
-    		evolution-plugin-$EVOLUTION_PACKAGE >= $EVOLUTION_REQUIRED
+    		evolution-plugin >= $EVOLUTION_REQUIRED
     "`
         else
 	        EDS_BACKEND_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "
 		libecal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedata-cal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedataserver-$EDS_PACKAGE >= $EDS_REQUIRED
-    		evolution-plugin-$EVOLUTION_PACKAGE >= $EVOLUTION_REQUIRED
+    		evolution-plugin >= $EVOLUTION_REQUIRED
     "`
         fi
 	# Put the nasty error message in config.log where it belongs
@@ -25744,7 +25744,7 @@
 		libecal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedata-cal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedataserver-$EDS_PACKAGE >= $EDS_REQUIRED
-    		evolution-plugin-$EVOLUTION_PACKAGE >= $EVOLUTION_REQUIRED
+    		evolution-plugin >= $EVOLUTION_REQUIRED
     ) were not met:
 
 $EDS_BACKEND_PKG_ERRORS
@@ -25760,7 +25760,7 @@
 		libecal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedata-cal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedataserver-$EDS_PACKAGE >= $EDS_REQUIRED
-    		evolution-plugin-$EVOLUTION_PACKAGE >= $EVOLUTION_REQUIRED
+    		evolution-plugin >= $EVOLUTION_REQUIRED
     ) were not met:
 
 $EDS_BACKEND_PKG_ERRORS
@@ -25804,7 +25804,7 @@
 fi
     extensiondir=`$PKG_CONFIG --variable=extensiondir evolution-data-server-$EDS_PACKAGE`
 
-    plugindir=`$PKG_CONFIG --variable=plugindir evolution-plugin-$EVOLUTION_PACKAGE`
+    plugindir=`$PKG_CONFIG --variable=plugindir evolution-plugin`
 
     use_eds_backend=yes
 else
diff -u -r planner-0.14.3/configure.in planner.mod/configure.in
--- planner-0.14.3/configure.in	2008-04-12 07:49:23.000000000 -0500
+++ planner.mod/configure.in	2008-04-19 18:38:27.000000000 -0500
@@ -289,11 +289,11 @@
 		libecal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedata-cal-$EDS_PACKAGE >= $EDS_REQUIRED
     		libedataserver-$EDS_PACKAGE >= $EDS_REQUIRED
-    		evolution-plugin-$EVOLUTION_PACKAGE >= $EVOLUTION_REQUIRED
+    		evolution-plugin >= $EVOLUTION_REQUIRED
     ])
     extensiondir=`$PKG_CONFIG --variable=extensiondir evolution-data-server-$EDS_PACKAGE`
     AC_SUBST(extensiondir)
-    plugindir=`$PKG_CONFIG --variable=plugindir evolution-plugin-$EVOLUTION_PACKAGE`
+    plugindir=`$PKG_CONFIG --variable=plugindir evolution-plugin`
     AC_SUBST(plugindir)
     use_eds_backend=yes
 else
