diff -Nurp gltt-2.5.2.old/FTGlyphVectorizer.h gltt-2.5.2/FTGlyphVectorizer.h
--- gltt-2.5.2.old/FTGlyphVectorizer.h	2002-10-15 22:17:44.000000000 +0200
+++ gltt-2.5.2/FTGlyphVectorizer.h	2002-10-15 22:17:59.000000000 +0200
@@ -76,7 +76,7 @@ public:
     private:
       void add_point( double x, double y );
 
-    friend FTGlyphVectorizer;
+    friend class FTGlyphVectorizer;
     };
 
 protected:
