diff -uNr gltt-2.5.2.old/GLTTGlyphPolygonizerHandler.h gltt-2.5.2/GLTTGlyphPolygonizerHandler.h
--- gltt-2.5.2.old/GLTTGlyphPolygonizerHandler.h	1999-03-13 17:50:13.000000000 +0100
+++ gltt-2.5.2/GLTTGlyphPolygonizerHandler.h	2006-03-06 13:07:44.000000000 +0100
@@ -28,6 +28,7 @@
 #include "FTGlyphVectorizer.h"
 #endif
 
+#include "GLTTGlyphPolygonizer.h"
 /////////////////////////////////////////////////////////////////////////////
 
 // Mmh, this class name is a bit long... /SR
diff -uNr gltt-2.5.2.old/GLTTGlyphTriangulator.h gltt-2.5.2/GLTTGlyphTriangulator.h
--- gltt-2.5.2.old/GLTTGlyphTriangulator.h	1999-03-13 17:50:11.000000000 +0100
+++ gltt-2.5.2/GLTTGlyphTriangulator.h	2006-03-06 13:07:44.000000000 +0100
@@ -27,7 +27,7 @@
 #ifndef __GLTTGlyphPolygonizerHandler_h
 #include "GLTTGlyphPolygonizerHandler.h"
 #endif
-
+#include "GLTTGlyphPolygonizer.h"
 // class FTGlyphVectorizer;
 // struct FTGlyphVectorizer::POINT;
 class GLTTGlyphPolygonizer;
