diff -u -a -r ser-0.9.6.orig/Makefile.defs ser-0.9.6/Makefile.defs
--- ser-0.9.6.orig/Makefile.defs	2006-05-01 17:06:01.000000000 -0500
+++ ser-0.9.6/Makefile.defs	2006-05-01 17:16:24.000000000 -0500
@@ -98,7 +98,7 @@
 ifeq ($(OS), linux)
 	doc-dir = share/doc/ser/
 	man-dir = share/man/
-	LOCALBASE ?= /usr/local
+	LOCALBASE ?= /usr
 else
 ifeq ($(OS), freebsd)
 	doc-dir = share/doc/ser/
@@ -122,7 +122,7 @@
 else
 	doc-dir = doc/ser/
 	man-dir = man/
-	LOCALBASE ?= /usr/local
+	LOCALBASE ?= /usr
 endif
 endif
 endif
@@ -142,7 +142,7 @@
 BASEDIR ?=
 basedir = $(BASEDIR)
 # install prefixes for various stuff
-cfg-prefix = $(basedir)$(prefix)
+cfg-prefix = $(basedir)/
 bin-prefix = $(basedir)$(prefix)
 modules-prefix = $(basedir)$(prefix)
 doc-prefix = $(basedir)$(prefix)
@@ -151,7 +151,7 @@
 
 
 # target dirs for various stuff
-cfg-target = $(prefix)/$(cfg-dir)
+cfg-target = /$(cfg-dir)
 bin-target = $(prefix)/$(bin-dir)
 modules-target = $(prefix)/$(modules-dir)
 doc-target = $(prefix)/$(doc-dir)
