--- plugins/vfs_zip/Makefile.in.o	2012-04-09 22:06:09.071334093 -0400
+++ plugins/vfs_zip/Makefile.in	2012-04-09 22:00:05.525321540 -0400
@@ -105,7 +105,7 @@
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(vfs_zip_la_LDFLAGS) $(LDFLAGS) -o $@
 @HAVE_VFS_ZIP_TRUE@am_vfs_zip_la_rpath = -rpath $(pkglibdir)
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir) -I/usr/lib/libzip/include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 am__mv = mv -f
