From 22420d678afa4d4336a605ca8557f102e8a357e9 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Tue, 24 Jul 2018 11:22:37 +0200
Subject: [PATCH] add missing include

---
 src/qmapshack/helpers/CToolBarSetupDialog.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/qmapshack/helpers/CToolBarSetupDialog.cpp b/src/qmapshack/helpers/CToolBarSetupDialog.cpp
index 8ea3797..07e9517 100644
--- a/src/qmapshack/helpers/CToolBarSetupDialog.cpp
+++ b/src/qmapshack/helpers/CToolBarSetupDialog.cpp
@@ -19,6 +19,7 @@
 
 #include "helpers/CToolBarConfig.h"
 #include "helpers/CToolBarSetupDialog.h"
+#include <QAction>
 
 bool CToolBarSetupDialog::CItemFilter::shouldBeMoved(QListWidgetItem *item)
 {
-- 
2.18.0

