From 5d636ad60cde62d576e3b2e66e22620bad2c5e65 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 5 Oct 2019 18:12:35 +0200
Subject: [PATCH] fix incompatibility with recent proj

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 397dbdb..4d0aa15 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,6 +28,7 @@ project(QMapShack VERSION 1.11.1)
 set(VERSION_SUFFIX "")
 # !!! Do not forget to update versions in the sub-repos, too!!!
 
+add_definitions("-DACCEPT_USE_OF_DEPRECATED_PROJ_API_H")
 
 ###############################################################################################
 # Define paths and include other modules
-- 
2.23.0

