From 4611e43b6475c2d12fdaf4dd895f979aacfe401b Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Wed, 15 Sep 2021 00:03:15 +0200
Subject: [PATCH 2/4] Remove -O2 from CFLAGS

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 62e81ef74788..ad31de91cfa1 100644
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ PMPATH1  = /usr/lib/pm-utils/sleep.d
 PMPATH2  = /usr/lib64/pm-utils/sleep.d
 PMPATHD  = /usr/lib/systemd/system-sleep
 
-CFLAGS  += -O2 -I. -Wall -Wno-stringop-truncation # -DNOPERFEVENT   # -DHTTPSTATS
+CFLAGS  += -I. -Wall -Wno-stringop-truncation
 OBJMOD0  = version.o
 OBJMOD1  = various.o  deviate.o   procdbase.o
 OBJMOD2  = acctproc.o photoproc.o photosyst.o  rawlog.o ifprop.o parseable.o
-- 
2.33.0

