From 9d11df0b6b9213e49524496c793d3c58fbbe1537 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Wed, 15 Sep 2021 00:03:15 +0200
Subject: [PATCH 3/4] Replace vwprintw with vw_printw

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 showgeneric.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/showgeneric.c b/showgeneric.c
index 2fc63ca1e72e..380f98b28068 100644
--- a/showgeneric.c
+++ b/showgeneric.c
@@ -2682,7 +2682,7 @@ printg(const char *format, ...)
 	va_start(args, format);
 
 	if (screen)
-		vwprintw(stdscr, (char *) format, args);
+		vw_printw(stdscr, format, args);
 	else
 		vprintf(format, args);
 
-- 
2.33.0

