--- texk/texlive/Makefile.in.orig	2009-08-05 18:57:18.000000000 +0200
+++ texk/texlive/Makefile.in	2009-08-05 18:57:43.000000000 +0200
@@ -107,15 +107,10 @@
 	  target=$(scriptdir)/`basename $$s | sed 's,\.[^/]*$$,,' | tr '[A-Z]' '[a-z]'`; \
 	  target=$$target; \
 	  rm -f $$target; \
-	  case "$(scriptdir)" in \
-	    */bin)   d=..;; \
-	    */bin/*) d=../..;; \
-	    *) echo "$(scriptdir): strange directory for linked scripts" >&2;\
-	      exit 1;; \
-	  esac; \
-	  $(SHELL) $(srcdir)/../mkinstalldirs $(prefix)/`dirname $$s`; \
-	  $(INSTALL_SCRIPT) $(srcdir)/linked_scripts/`basename $$s` $(prefix)/$$s; \
-	  ln -s $$d/$$s $$target; \
+	  d=../share/; \
+	  $(SHELL) $(srcdir)/../mkinstalldirs $(datadir)/`dirname $$s`; \
+	  $(INSTALL_SCRIPT) $(srcdir)/linked_scripts/`basename $$s` $(datadir)/$$s; \
+	  ln -vs $$d/$$s $$target; \
 	done
 	rm -f $(scriptdir)/getnonfreefonts-sys
 	ln -s getnonfreefonts $(scriptdir)/getnonfreefonts-sys
