diff -ru texlive-20091107-source/texk/bibtex8/Makefile.in tl/texk/bibtex8/Makefile.in
--- texlive-20091107-source/texk/bibtex8/Makefile.in	2009-11-10 07:51:56.076903492 +0100
+++ tl/texk/bibtex8/Makefile.in	2009-11-10 08:11:55.197300678 +0100
@@ -263,7 +263,7 @@
 
 bibtex8_DEPENDENCIES = $(KPATHSEA_DEPEND)
 LDADD = $(KPATHSEA_LIBS)	
-csfdir = ${prefix}/texmf-dist/bibtex/csf/base
+csfdir = ${datadir}/texmf-dist/bibtex/csf/base
 dist_csf_DATA = \
 	csf/88591lat.csf \
 	csf/88591sca.csf \
@@ -274,7 +274,7 @@
 	csf/cp866rus.csf \
 	csf/csfile.txt
 
-btdocdir = ${prefix}/texmf/doc/bibtex8
+btdocdir = ${datadir}/texmf/doc/bibtex8
 dist_btdoc_DATA = \
 	00readme.txt \
 	HISTORY \
diff -ru texlive-20091107-source/texk/bibtexu/Makefile.in tl/texk/bibtexu/Makefile.in
--- texlive-20091107-source/texk/bibtexu/Makefile.in	2009-11-10 07:51:56.056903463 +0100
+++ tl/texk/bibtexu/Makefile.in	2009-11-10 09:35:40.604997976 +0100
@@ -289,7 +289,7 @@
 
 bibtexu_DEPENDENCIES = $(KPATHSEA_DEPEND) $(ICU_DEPEND)
 LDADD = $(KPATHSEA_LIBS) $(ICU_LIBS) -lm
-csfdir = ${prefix}/texmf-dist/bibtex/csf/base
+csfdir = ${datadir}/texmf-dist/bibtex/csf/base
 dist_csf_DATA = \
 	csf/88591lat.csf \
 	csf/88591sca.csf \
@@ -300,7 +300,7 @@
 	csf/cp866rus.csf \
 	csf/csfile.txt
 
-btdocdir = ${prefix}/texmf/doc/bibtexu
+btdocdir = ${datadir}/texmf/doc/bibtexu
 dist_btdoc_DATA = \
 	00readme.txt \
 	HISTORY \
diff -ru texlive-20091107-source/texk/dvipdfmx/data/Makefile.in tl/texk/dvipdfmx/data/Makefile.in
--- texlive-20091107-source/texk/dvipdfmx/data/Makefile.in	2009-11-10 07:51:56.128902281 +0100
+++ tl/texk/dvipdfmx/data/Makefile.in	2009-11-10 09:46:53.505445330 +0100
@@ -208,20 +208,20 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-configdatadir = ${prefix}/texmf/dvipdfmx
+configdatadir = ${datadir}/texmf/dvipdfmx
 dist_configdata_DATA = \
   $(srcdir)/dvipdfmx.cfg
 
-glyphlistdatadir = ${prefix}/texmf-dist/fonts/map/glyphlist
+glyphlistdatadir = ${datadir}/texmf-dist/fonts/map/glyphlist
 dist_glyphlistdata_DATA = \
   $(srcdir)/glyphlist.txt \
   $(srcdir)/pdfglyphlist.txt
 
-mapdatadir = ${prefix}/texmf/fonts/map/dvipdfm/dvipdfmx
+mapdatadir = ${datadir}/texmf/fonts/map/dvipdfm/dvipdfmx
 dist_mapdata_DATA = \
   $(srcdir)/cid-x.map
 
-cmapdatadir = ${prefix}/texmf/fonts/cmap/dvipdfmx
+cmapdatadir = ${datadir}/texmf/fonts/cmap/dvipdfmx
 dist_cmapdata_DATA = \
   $(srcdir)/EUC-UCS2
 
diff -ru texlive-20091107-source/texk/dvipsk/Makefile.in tl/texk/dvipsk/Makefile.in
--- texlive-20091107-source/texk/dvipsk/Makefile.in	2009-11-10 07:51:56.156903115 +0100
+++ tl/texk/dvipsk/Makefile.in	2009-11-10 09:46:58.166678965 +0100
@@ -376,7 +376,7 @@
 	tex.lpro \
 	texps.lpro
 
-prologdir = ${prefix}/texmf/dvips/base
+prologdir = ${datadir}/texmf/dvips/base
 prologues = $(dist_prologues:.lpro=.pro) texc.pro
 SUFFIXES = .pro .lpro
 EXTRA_DIST = $(dist_prologues) texc.script dvips.test pfbincl.test \
diff -ru texlive-20091107-source/texk/gsftopk/Makefile.in tl/texk/gsftopk/Makefile.in
--- texlive-20091107-source/texk/gsftopk/Makefile.in	2009-11-10 07:51:56.072902704 +0100
+++ tl/texk/gsftopk/Makefile.in	2009-11-10 09:35:44.064408320 +0100
@@ -256,7 +256,7 @@
 gsftopk_DEPENDENCIES = $(KPATHSEA_DEPEND)
 LDADD = $(KPATHSEA_LIBS)
 dist_gsftopkpsheader_DATA = render.ps
-gsftopkpsheaderdir = ${prefix}/texmf/dvips/gsftopk
+gsftopkpsheaderdir = ${datadir}/texmf/dvips/gsftopk
 nodist_man1_MANS = gsftopk.1
 DISTCLEANFILES = gsftopk.1 sedscript
 EXTRA_DIST = gsftopk1.sed mksedscript
diff -ru texlive-20091107-source/texk/kpathsea/Makefile.in tl/texk/kpathsea/Makefile.in
--- texlive-20091107-source/texk/kpathsea/Makefile.in	2009-11-10 07:51:56.100902493 +0100
+++ tl/texk/kpathsea/Makefile.in	2009-11-10 09:35:52.163678762 +0100
@@ -432,7 +432,7 @@
 kpseaccess_SOURCES = access.c
 kpsereadlink_SOURCES = readlink.c
 kpsewhich_LDADD = libkpathsea.la
-web2cdir = ${prefix}/texmf/web2c
+web2cdir = ${datadir}/texmf/web2c
 dist_web2c_SCRIPTS = mktexdir mktexnam mktexupd
 dist_web2c_DATA = mktex.opt mktexdir.opt mktexnam.opt
 dist_noinst_SCRIPTS = mktexlsr mktexmf mktexpk mktextfm
diff -ru texlive-20091107-source/texk/lcdf-typetools/Makefile.in tl/texk/lcdf-typetools/Makefile.in
--- texlive-20091107-source/texk/lcdf-typetools/Makefile.in	2009-11-10 07:51:56.116902013 +0100
+++ tl/texk/lcdf-typetools/Makefile.in	2009-11-10 09:36:02.871630882 +0100
@@ -275,9 +275,9 @@
 DISTCHECK_CONFIGURE_FLAGS = --without-kpathsea
 SUBDIRS = $(SELECTED_SUBDIRS)
 DIST_SUBDIRS = liblcdf libefont cfftot1 mmafm mmpfb otfinfo otftotfm t1dotlessj t1lint t1rawafm t1reencode t1testpage ttftotype42
-@texlive_build_TRUE@glyphlistdir = ${prefix}/texmf-dist/fonts/map/glyphlist
+@texlive_build_TRUE@glyphlistdir = ${datadir}/texmf-dist/fonts/map/glyphlist
 @texlive_build_TRUE@glyphlist_DATA = glyphlist.txt texglyphlist.txt
-@texlive_build_TRUE@encdir = ${prefix}/texmf-dist/fonts/enc/dvips/base
+@texlive_build_TRUE@encdir = ${datadir}/texmf-dist/fonts/enc/dvips/base
 @texlive_build_TRUE@enc_DATA = 7t.enc
 @texlive_build_FALSE@pkgdata_DATA = glyphlist.txt texglyphlist.txt 7t.enc
 EXTRA_DIST = ONEWS lcdf-typetools.spec include/lcdf/bezier.hh \
diff -ru texlive-20091107-source/texk/tetex/Makefile.in tl/texk/tetex/Makefile.in
--- texlive-20091107-source/texk/tetex/Makefile.in	2009-11-10 07:51:56.124903449 +0100
+++ tl/texk/tetex/Makefile.in	2009-11-10 09:58:37.661294317 +0100
@@ -254,14 +254,14 @@
 	fmtutil.cnf.5 \
 	updmap.cfg.5
 
-tetexdocdir = ${prefix}/texmf/doc/tetex
+tetexdocdir = ${datadir}/texmf/doc/tetex
 dist_tetexdoc_DATA = \
 	doc/TETEXDOC.dvi \
 	doc/TETEXDOC.pdf \
 	doc/TETEXDOC.ps \
 	doc/teTeX-FAQ
 
-texconfdir = ${prefix}
+texconfdir = ${datadir}
 nobase_dist_texconf_DATA = \
 	texmf/texconfig/README \
 	texmf/texconfig/g/generic \
@@ -269,10 +269,10 @@
 	texmf/texconfig/v/vt100 \
 	texmf/texconfig/x/xterm
 
-texconfigdir = ${prefix}/texmf/texconfig
+texconfigdir = ${datadir}/texmf/texconfig
 dist_texconfig_SCRIPTS = tcfmgr
 dist_texconfig_DATA = tcfmgr.map
-web2cdir = ${prefix}/texmf/web2c
+web2cdir = ${datadir}/texmf/web2c
 dist_web2c_DATA = fmtutil.cnf
 EXTRA_DIST = README.a2ping README.texdoctk doc/Makefile \
 	doc/TETEXDOC.bib doc/TETEXDOC.tex fontinst.bat pdfetex-pl.pool \
diff -ru texlive-20091107-source/texk/tex4htk/Makefile.in tl/texk/tex4htk/Makefile.in
--- texlive-20091107-source/texk/tex4htk/Makefile.in	2009-11-10 07:51:56.104901674 +0100
+++ tl/texk/tex4htk/Makefile.in	2009-11-10 10:09:16.655391593 +0100
@@ -273,7 +273,7 @@
 	xhlatex xhmcontext xhmex xhmlatex xhmmex xhmtex xhmtexi \
 	xhmxelatex xhmxetex xhtex xhtexi xhxelatex xhxetex xv4ht.java
 tex4ht_subdir = texmf-dist/scripts/tex4ht
-tex4htdir = ${prefix}/$(tex4ht_subdir)
+tex4htdir = ${datadir}/$(tex4ht_subdir)
 tex4ht_SCRIPTS = 
 bin_SCRIPTS = 
 all: c-auto.h
Nur in tl/texk/tex4htk: .Makefile.in.swp.
diff -ru texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in tl/texk/texlive/linked_scripts/Makefile.in
--- texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in	2009-11-10 07:51:55.952902523 +0100
+++ tl/texk/texlive/linked_scripts/Makefile.in	2009-11-10 09:35:30.862879772 +0100
@@ -156,7 +156,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 bin_SCRIPTS = 
-scriptxdir = ${prefix}/texmf/scripts
+scriptxdir = ${datadir}/texmf/scripts
 nobase_dist_scriptx_SCRIPTS = \
 	a2ping/a2ping.pl \
 	getnonfreefonts/getnonfreefonts.pl \
@@ -167,7 +167,7 @@
 	texlive/rungs.tlu \
 	texlive/tlmgr.pl
 
-scriptdir = ${prefix}/texmf-dist/scripts
+scriptdir = ${datadir}/texmf-dist/scripts
 nobase_dist_script_SCRIPTS = \
 	accfonts/mkt1font \
 	accfonts/vpl2ovp \
Nur in tl/texk/web2c/am: .texmf.am.swp.
Nur in tl/texk/web2c: .Makefile.in.swp.
diff -ru texlive-20091107-source/texk/xdvik/Makefile.in tl/texk/xdvik/Makefile.in
--- texlive-20091107-source/texk/xdvik/Makefile.in	2009-11-10 07:51:56.084902344 +0100
+++ tl/texk/xdvik/Makefile.in	2009-11-10 09:35:48.058678814 +0100
@@ -436,9 +436,9 @@
 	xdvi.icon xdvizilla xdvizilla.1
 CLEANFILES = psheader.c sedscript xdvi.1
 nodist_man1_MANS = xdvi.1
-xdvidir = ${prefix}/texmf/xdvi
+xdvidir = ${datadir}/texmf/xdvi
 dist_xdvi_DATA = texmf/XDvi
-pixmapdir = ${prefix}/texmf/xdvi/pixmap
+pixmapdir = ${datadir}/texmf/xdvi/pixmap
 dist_pixmap_DATA = pixmaps/toolbar.xpm pixmaps/toolbar2.xpm
 dist_noinst_DATA = texmf/xdvi.cfg
 all: c-auto.h
diff -ru texlive-20091107-source/utils/xindy/user-commands/Makefile.in tl/utils/xindy/user-commands/Makefile.in
--- texlive-20091107-source/utils/xindy/user-commands/Makefile.in	2009-11-10 07:51:55.580901938 +0100
+++ tl/utils/xindy/user-commands/Makefile.in	2009-11-10 08:09:09.322065690 +0100
@@ -501,16 +501,16 @@
 
 
 @TETEX_BUILD_TRUE@install-exec-hook: xindy
-@TETEX_BUILD_TRUE@	$(MKDIR_P) "$(DESTDIR)$(prefix)/texmf/scripts/xindy"
+@TETEX_BUILD_TRUE@	$(MKDIR_P) "$(DESTDIR)$(datadir)/texmf/scripts/xindy"
 @TETEX_BUILD_TRUE@	-rm -f $(DESTDIR)$(bindir)/xindy $(DESTDIR)$(bindir)/texindy
-@TETEX_BUILD_TRUE@	$(INSTALL_SCRIPT) xindy $(DESTDIR)$(prefix)/texmf/scripts/xindy/xindy.pl
-@TETEX_BUILD_TRUE@	$(INSTALL_SCRIPT) $(srcdir)/texindy $(DESTDIR)$(prefix)/texmf/scripts/xindy/texindy.pl
+@TETEX_BUILD_TRUE@	$(INSTALL_SCRIPT) xindy $(DESTDIR)$(datadir)/texmf/scripts/xindy/xindy.pl
+@TETEX_BUILD_TRUE@	$(INSTALL_SCRIPT) $(srcdir)/texindy $(DESTDIR)$(datadir)/texmf/scripts/xindy/texindy.pl
 @TETEX_BUILD_TRUE@	if test -d "$(DESTDIR)$(bindir)/../../texmf/scripts/xindy"; then \
-@TETEX_BUILD_TRUE@	  ln -s ../../texmf/scripts/xindy/xindy.pl $(DESTDIR)$(bindir)/xindy; \
-@TETEX_BUILD_TRUE@	  ln -s ../../texmf/scripts/xindy/texindy.pl $(DESTDIR)$(bindir)/texindy; \
+@TETEX_BUILD_TRUE@	  ln -s $(datadir)/texmf/scripts/xindy/xindy.pl $(DESTDIR)$(bindir)/xindy; \
+@TETEX_BUILD_TRUE@	  ln -s $(datadir)/texmf/scripts/xindy/texindy.pl $(DESTDIR)$(bindir)/texindy; \
 @TETEX_BUILD_TRUE@	elif test -d "$(DESTDIR)$(bindir)/../texmf/scripts/xindy"; then \
-@TETEX_BUILD_TRUE@	  ln -s ../texmf/scripts/xindy/xindy.pl $(DESTDIR)$(bindir)/xindy; \
-@TETEX_BUILD_TRUE@	  ln -s ../texmf/scripts/xindy/texindy.pl $(DESTDIR)$(bindir)/texindy; \
+@TETEX_BUILD_TRUE@	  ln -s $(datadir)/texmf/scripts/xindy/xindy.pl $(DESTDIR)$(bindir)/xindy; \
+@TETEX_BUILD_TRUE@	  ln -s $(datadir)/texmf/scripts/xindy/texindy.pl $(DESTDIR)$(bindir)/texindy; \
 @TETEX_BUILD_TRUE@	else \
 @TETEX_BUILD_TRUE@	  echo "Error: texmf/scripts/xindy directory not found" >&2; \
 @TETEX_BUILD_TRUE@	  exit 1; \
