diff -ur texlive-20091107-source.orig/texk/bibtex8/Makefile.in texlive-20091107-source/texk/bibtex8/Makefile.in
--- texlive-20091107-source.orig/texk/bibtex8/Makefile.in	2009-06-19 01:15:24.000000000 +0800
+++ texlive-20091107-source/texk/bibtex8/Makefile.in	2010-07-17 00:48:58.074062142 +0800
@@ -263,7 +263,7 @@
 
 bibtex8_DEPENDENCIES = $(KPATHSEA_DEPEND)
 LDADD = $(KPATHSEA_LIBS)	
-csfdir = ${prefix}/texmf-dist/bibtex/csf/base
+csfdir = ${prefix}/share/texmf-dist/bibtex/csf/base
 dist_csf_DATA = \
 	csf/88591lat.csf \
 	csf/88591sca.csf \
@@ -274,7 +274,7 @@
 	csf/cp866rus.csf \
 	csf/csfile.txt
 
-btdocdir = ${prefix}/texmf/doc/bibtex8
+btdocdir = ${prefix}/share/texmf/doc/bibtex8
 dist_btdoc_DATA = \
 	00readme.txt \
 	HISTORY \
diff -ur texlive-20091107-source.orig/texk/bibtexu/Makefile.in texlive-20091107-source/texk/bibtexu/Makefile.in
--- texlive-20091107-source.orig/texk/bibtexu/Makefile.in	2009-10-09 00:07:57.000000000 +0800
+++ texlive-20091107-source/texk/bibtexu/Makefile.in	2010-07-17 00:48:41.894061307 +0800
@@ -289,7 +289,7 @@
 
 bibtexu_DEPENDENCIES = $(KPATHSEA_DEPEND) $(ICU_DEPEND)
 LDADD = $(KPATHSEA_LIBS) $(ICU_LIBS) -lm
-csfdir = ${prefix}/texmf-dist/bibtex/csf/base
+csfdir = ${prefix}/share/texmf-dist/bibtex/csf/base
 dist_csf_DATA = \
 	csf/88591lat.csf \
 	csf/88591sca.csf \
@@ -300,7 +300,7 @@
 	csf/cp866rus.csf \
 	csf/csfile.txt
 
-btdocdir = ${prefix}/texmf/doc/bibtexu
+btdocdir = ${prefix}/share/texmf/doc/bibtexu
 dist_btdoc_DATA = \
 	00readme.txt \
 	HISTORY \
diff -ur texlive-20091107-source.orig/texk/dvipdfmx/data/Makefile.in texlive-20091107-source/texk/dvipdfmx/data/Makefile.in
--- texlive-20091107-source.orig/texk/dvipdfmx/data/Makefile.in	2009-06-15 23:14:49.000000000 +0800
+++ texlive-20091107-source/texk/dvipdfmx/data/Makefile.in	2010-07-17 01:15:07.504125133 +0800
@@ -208,20 +208,20 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-configdatadir = ${prefix}/texmf/dvipdfmx
+configdatadir = ${prefix}/share/texmf/dvipdfmx
 dist_configdata_DATA = \
   $(srcdir)/dvipdfmx.cfg
 
-glyphlistdatadir = ${prefix}/texmf-dist/fonts/map/glyphlist
+glyphlistdatadir = ${prefix}/share/texmf-dist/fonts/map/glyphlist
 dist_glyphlistdata_DATA = \
   $(srcdir)/glyphlist.txt \
   $(srcdir)/pdfglyphlist.txt
 
-mapdatadir = ${prefix}/texmf/fonts/map/dvipdfm/dvipdfmx
+mapdatadir = ${prefix}/share/texmf/fonts/map/dvipdfm/dvipdfmx
 dist_mapdata_DATA = \
   $(srcdir)/cid-x.map
 
-cmapdatadir = ${prefix}/texmf/fonts/cmap/dvipdfmx
+cmapdatadir = ${prefix}/share/texmf/fonts/cmap/dvipdfmx
 dist_cmapdata_DATA = \
   $(srcdir)/EUC-UCS2
 
diff -ur texlive-20091107-source.orig/texk/dvipsk/Makefile.in texlive-20091107-source/texk/dvipsk/Makefile.in
--- texlive-20091107-source.orig/texk/dvipsk/Makefile.in	2009-06-23 17:46:14.000000000 +0800
+++ texlive-20091107-source/texk/dvipsk/Makefile.in	2010-07-17 01:12:22.149312340 +0800
@@ -376,7 +376,7 @@
 	tex.lpro \
 	texps.lpro
 
-prologdir = ${prefix}/texmf/dvips/base
+prologdir = ${prefix}/share/texmf/dvips/base
 prologues = $(dist_prologues:.lpro=.pro) texc.pro
 SUFFIXES = .pro .lpro
 EXTRA_DIST = $(dist_prologues) texc.script dvips.test pfbincl.test \
diff -ur texlive-20091107-source.orig/texk/gsftopk/Makefile.in texlive-20091107-source/texk/gsftopk/Makefile.in
--- texlive-20091107-source.orig/texk/gsftopk/Makefile.in	2009-06-22 15:25:39.000000000 +0800
+++ texlive-20091107-source/texk/gsftopk/Makefile.in	2010-07-17 01:15:27.990063403 +0800
@@ -256,7 +256,7 @@
 gsftopk_DEPENDENCIES = $(KPATHSEA_DEPEND)
 LDADD = $(KPATHSEA_LIBS)
 dist_gsftopkpsheader_DATA = render.ps
-gsftopkpsheaderdir = ${prefix}/texmf/dvips/gsftopk
+gsftopkpsheaderdir = ${prefix}/share/texmf/dvips/gsftopk
 nodist_man1_MANS = gsftopk.1
 DISTCLEANFILES = gsftopk.1 sedscript
 EXTRA_DIST = gsftopk1.sed mksedscript
diff -ur texlive-20091107-source.orig/texk/kpathsea/Makefile.in texlive-20091107-source/texk/kpathsea/Makefile.in
--- texlive-20091107-source.orig/texk/kpathsea/Makefile.in	2009-07-02 21:04:27.000000000 +0800
+++ texlive-20091107-source/texk/kpathsea/Makefile.in	2010-07-17 01:15:51.607054071 +0800
@@ -432,7 +432,7 @@
 kpseaccess_SOURCES = access.c
 kpsereadlink_SOURCES = readlink.c
 kpsewhich_LDADD = libkpathsea.la
-web2cdir = ${prefix}/texmf/web2c
+web2cdir = ${prefix}/share/texmf/web2c
 dist_web2c_SCRIPTS = mktexdir mktexnam mktexupd
 dist_web2c_DATA = mktex.opt mktexdir.opt mktexnam.opt
 dist_noinst_SCRIPTS = mktexlsr mktexmf mktexpk mktextfm
diff -ur texlive-20091107-source.orig/texk/lcdf-typetools/Makefile.in texlive-20091107-source/texk/lcdf-typetools/Makefile.in
--- texlive-20091107-source.orig/texk/lcdf-typetools/Makefile.in	2009-09-04 04:09:39.000000000 +0800
+++ texlive-20091107-source/texk/lcdf-typetools/Makefile.in	2010-07-17 00:53:19.596049631 +0800
@@ -275,9 +275,9 @@
 DISTCHECK_CONFIGURE_FLAGS = --without-kpathsea
 SUBDIRS = $(SELECTED_SUBDIRS)
 DIST_SUBDIRS = liblcdf libefont cfftot1 mmafm mmpfb otfinfo otftotfm t1dotlessj t1lint t1rawafm t1reencode t1testpage ttftotype42
-@texlive_build_TRUE@glyphlistdir = ${prefix}/texmf-dist/fonts/map/glyphlist
+@texlive_build_TRUE@glyphlistdir = ${prefix}/share/texmf-dist/fonts/map/glyphlist
 @texlive_build_TRUE@glyphlist_DATA = glyphlist.txt texglyphlist.txt
-@texlive_build_TRUE@encdir = ${prefix}/texmf-dist/fonts/enc/dvips/base
+@texlive_build_TRUE@encdir = ${prefix}/share/texmf-dist/fonts/enc/dvips/base
 @texlive_build_TRUE@enc_DATA = 7t.enc
 @texlive_build_FALSE@pkgdata_DATA = glyphlist.txt texglyphlist.txt 7t.enc
 EXTRA_DIST = ONEWS lcdf-typetools.spec include/lcdf/bezier.hh \
diff -ur texlive-20091107-source.orig/texk/tetex/Makefile.in texlive-20091107-source/texk/tetex/Makefile.in
--- texlive-20091107-source.orig/texk/tetex/Makefile.in	2009-09-28 23:46:20.000000000 +0800
+++ texlive-20091107-source/texk/tetex/Makefile.in	2010-07-17 09:32:29.799063942 +0800
@@ -254,14 +254,14 @@
 	fmtutil.cnf.5 \
 	updmap.cfg.5
 
-tetexdocdir = ${prefix}/texmf/doc/tetex
+tetexdocdir = ${prefix}/share/texmf/doc/tetex
 dist_tetexdoc_DATA = \
 	doc/TETEXDOC.dvi \
 	doc/TETEXDOC.pdf \
 	doc/TETEXDOC.ps \
 	doc/teTeX-FAQ
 
-texconfdir = ${prefix}
+texconfdir = ${prefix}/share
 nobase_dist_texconf_DATA = \
 	texmf/texconfig/README \
 	texmf/texconfig/g/generic \
@@ -269,10 +269,10 @@
 	texmf/texconfig/v/vt100 \
 	texmf/texconfig/x/xterm
 
-texconfigdir = ${prefix}/texmf/texconfig
+texconfigdir = ${prefix}/share/texmf/texconfig
 dist_texconfig_SCRIPTS = tcfmgr
 dist_texconfig_DATA = tcfmgr.map
-web2cdir = ${prefix}/texmf/web2c
+web2cdir = ${prefix}/share/texmf/web2c
 dist_web2c_DATA = fmtutil.cnf
 EXTRA_DIST = README.a2ping README.texdoctk doc/Makefile \
 	doc/TETEXDOC.bib doc/TETEXDOC.tex fontinst.bat pdfetex-pl.pool \
diff -ur texlive-20091107-source.orig/texk/tex4htk/Makefile.in texlive-20091107-source/texk/tex4htk/Makefile.in
--- texlive-20091107-source.orig/texk/tex4htk/Makefile.in	2009-06-18 22:50:50.000000000 +0800
+++ texlive-20091107-source/texk/tex4htk/Makefile.in	2010-07-17 00:50:19.333067714 +0800
@@ -272,7 +272,7 @@
 	wcontext wlatex wmex wtex wtexi wxelatex wxetex xhcontext \
 	xhlatex xhmcontext xhmex xhmlatex xhmmex xhmtex xhmtexi \
 	xhmxelatex xhmxetex xhtex xhtexi xhxelatex xhxetex xv4ht.java
-tex4ht_subdir = texmf-dist/scripts/tex4ht
+tex4ht_subdir = share/texmf-dist/scripts/tex4ht
 tex4htdir = ${prefix}/$(tex4ht_subdir)
 tex4ht_SCRIPTS = 
 bin_SCRIPTS = 
diff -ur texlive-20091107-source.orig/texk/texlive/linked_scripts/Makefile.in texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in
--- texlive-20091107-source.orig/texk/texlive/linked_scripts/Makefile.in	2009-10-15 07:39:50.000000000 +0800
+++ texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in	2010-07-17 09:53:22.230064779 +0800
@@ -156,7 +156,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 bin_SCRIPTS = 
-scriptxdir = ${prefix}/texmf/scripts
+scriptxdir = ${prefix}/share/texmf/scripts
 nobase_dist_scriptx_SCRIPTS = \
 	a2ping/a2ping.pl \
 	getnonfreefonts/getnonfreefonts.pl \
@@ -167,7 +167,7 @@
 	texlive/rungs.tlu \
 	texlive/tlmgr.pl
 
-scriptdir = ${prefix}/texmf-dist/scripts
+scriptdir = ${prefix}/share/texmf-dist/scripts
 nobase_dist_script_SCRIPTS = \
 	accfonts/mkt1font \
 	accfonts/vpl2ovp \
@@ -554,16 +554,16 @@
 	  for s in $(nobase_dist_scriptx_SCRIPTS); do \
 	    target=`basename $$s | sed 's,\.[^/]*$$,,' | tr '[A-Z]' '[a-z]'`; \
 	    rm -f $$target; \
-	    echo "creating link '$$target' -> '$(REL)/texmf/scripts/$$s'"; \
-	    $(LN_S) $(REL)/texmf/scripts/$$s $$target; \
+	    echo "creating link '$$target' -> '$(REL)/share/texmf/scripts/$$s'"; \
+	    $(LN_S) $(REL)/share/texmf/scripts/$$s $$target; \
 	  done && \
 	  for s in $(nobase_dist_script_SCRIPTS); do \
 	    target=`basename $$s | tr '[A-Z]' '[a-z]'`; \
 	    echo "$$s" | grep listings-ext.sh >/dev/null \
 	    || target=`echo $$target | sed 's,\.[^/]*$$,,'`; \
 	    rm -f $$target; \
-	    echo "creating link '$$target' -> '$(REL)/texmf-dist/scripts/$$s'"; \
-	    $(LN_S) $(REL)/texmf-dist/scripts/$$s $$target; \
+	    echo "creating link '$$target' -> '$(REL)/share/texmf-dist/scripts/$$s'"; \
+	    $(LN_S) $(REL)/share/texmf-dist/scripts/$$s $$target; \
 	  done && \
 	  rm -f rpdfcrop && \
 	  echo "creating link 'rpdfcrop' -> 'pdfcrop'" && \
@@ -573,10 +573,7 @@
 	  $(LN_S) epstopdf repstopdf && \
 	  rm -f getnonfreefonts-sys && \
 	  echo "creating link 'getnonfreefonts-sys' -> 'getnonfreefonts'" && \
-	  $(LN_S) getnonfreefonts getnonfreefonts-sys && \
-	  rm -f man && \
-	  echo "creating link 'man' -> '$(REL)/texmf/doc/man'" && \
-	  (ln -s $(REL)/texmf/doc/man man || :)
+	  $(LN_S) getnonfreefonts getnonfreefonts-sys
 
 uninstall-hook:
 	@for s in $(nobase_dist_scriptx_SCRIPTS) $(nobase_dist_script_SCRIPTS); do \
diff -ur texlive-20091107-source.orig/texk/xdvik/Makefile.in texlive-20091107-source/texk/xdvik/Makefile.in
--- texlive-20091107-source.orig/texk/xdvik/Makefile.in	2009-06-16 20:01:38.000000000 +0800
+++ texlive-20091107-source/texk/xdvik/Makefile.in	2010-07-17 01:11:18.764319683 +0800
@@ -436,9 +436,9 @@
 	xdvi.icon xdvizilla xdvizilla.1
 CLEANFILES = psheader.c sedscript xdvi.1
 nodist_man1_MANS = xdvi.1
-xdvidir = ${prefix}/texmf/xdvi
+xdvidir = ${prefix}/share/texmf/xdvi
 dist_xdvi_DATA = texmf/XDvi
-pixmapdir = ${prefix}/texmf/xdvi/pixmap
+pixmapdir = ${prefix}/share/texmf/xdvi/pixmap
 dist_pixmap_DATA = pixmaps/toolbar.xpm pixmaps/toolbar2.xpm
 dist_noinst_DATA = texmf/xdvi.cfg
 all: c-auto.h
diff -ur texlive-20091107-source.orig/texk/xdvipdfmx/data/Makefile.in texlive-20091107-source/texk/xdvipdfmx/data/Makefile.in
--- texlive-20091107-source.orig/texk/xdvipdfmx/data/Makefile.in	2008-06-26 18:21:46.000000000 +0800
+++ texlive-20091107-source/texk/xdvipdfmx/data/Makefile.in	2010-07-17 01:14:02.343062640 +0800
@@ -62,7 +62,7 @@
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
-pkgdatadir = $(datadir)/texmf/dvipdfm
+pkgdatadir = $(datadir)/share/texmf/dvipdfm
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
 AR = @AR@
diff -ur texlive-20091107-source.orig/utils/xindy/user-commands/Makefile.in texlive-20091107-source/utils/xindy/user-commands/Makefile.in
--- texlive-20091107-source.orig/utils/xindy/user-commands/Makefile.in	2009-05-22 20:38:02.000000000 +0800
+++ texlive-20091107-source/utils/xindy/user-commands/Makefile.in	2010-07-17 01:09:15.327310949 +0800
@@ -501,16 +501,16 @@
 
 
 @TETEX_BUILD_TRUE@install-exec-hook: xindy
-@TETEX_BUILD_TRUE@	$(MKDIR_P) "$(DESTDIR)$(prefix)/texmf/scripts/xindy"
+@TETEX_BUILD_TRUE@	$(MKDIR_P) "$(DESTDIR)$(prefix)/share/texmf/scripts/xindy"
 @TETEX_BUILD_TRUE@	-rm -f $(DESTDIR)$(bindir)/xindy $(DESTDIR)$(bindir)/texindy
-@TETEX_BUILD_TRUE@	$(INSTALL_SCRIPT) xindy $(DESTDIR)$(prefix)/texmf/scripts/xindy/xindy.pl
-@TETEX_BUILD_TRUE@	$(INSTALL_SCRIPT) $(srcdir)/texindy $(DESTDIR)$(prefix)/texmf/scripts/xindy/texindy.pl
-@TETEX_BUILD_TRUE@	if test -d "$(DESTDIR)$(bindir)/../../texmf/scripts/xindy"; then \
-@TETEX_BUILD_TRUE@	  ln -s ../../texmf/scripts/xindy/xindy.pl $(DESTDIR)$(bindir)/xindy; \
-@TETEX_BUILD_TRUE@	  ln -s ../../texmf/scripts/xindy/texindy.pl $(DESTDIR)$(bindir)/texindy; \
-@TETEX_BUILD_TRUE@	elif test -d "$(DESTDIR)$(bindir)/../texmf/scripts/xindy"; then \
-@TETEX_BUILD_TRUE@	  ln -s ../texmf/scripts/xindy/xindy.pl $(DESTDIR)$(bindir)/xindy; \
-@TETEX_BUILD_TRUE@	  ln -s ../texmf/scripts/xindy/texindy.pl $(DESTDIR)$(bindir)/texindy; \
+@TETEX_BUILD_TRUE@	$(INSTALL_SCRIPT) xindy $(DESTDIR)$(prefix)/share/texmf/scripts/xindy/xindy.pl
+@TETEX_BUILD_TRUE@	$(INSTALL_SCRIPT) $(srcdir)/texindy $(DESTDIR)$(prefix)/share/texmf/scripts/xindy/texindy.pl
+@TETEX_BUILD_TRUE@	if test -d "$(DESTDIR)$(bindir)/../../share/texmf/scripts/xindy"; then \
+@TETEX_BUILD_TRUE@	  ln -s ../../share/texmf/scripts/xindy/xindy.pl $(DESTDIR)$(bindir)/xindy; \
+@TETEX_BUILD_TRUE@	  ln -s ../../share/texmf/scripts/xindy/texindy.pl $(DESTDIR)$(bindir)/texindy; \
+@TETEX_BUILD_TRUE@	elif test -d "$(DESTDIR)$(bindir)/../share/texmf/scripts/xindy"; then \
+@TETEX_BUILD_TRUE@	  ln -s ../share/texmf/scripts/xindy/xindy.pl $(DESTDIR)$(bindir)/xindy; \
+@TETEX_BUILD_TRUE@	  ln -s ../share/texmf/scripts/xindy/texindy.pl $(DESTDIR)$(bindir)/texindy; \
 @TETEX_BUILD_TRUE@	else \
 @TETEX_BUILD_TRUE@	  echo "Error: texmf/scripts/xindy directory not found" >&2; \
 @TETEX_BUILD_TRUE@	  exit 1; \
