--- a/texk/tetex/Makefile.in
+++ b/texk/tetex/Makefile.in
@@ -254,7 +254,7 @@ dist_man5_MANS = \
 	fmtutil.cnf.5 \
 	updmap.cfg.5
 
-texconfdir = ${prefix}
+texconfdir = ${datadir}
 nobase_dist_texconf_DATA = \
 	texmf/texconfig/README \
 	texmf/texconfig/g/generic \
--- a/texk/tex4htk/Makefile.in
+++ b/texk/tex4htk/Makefile.in
@@ -269,7 +269,7 @@ nodist_bin_SCRIPTS = $(am__append_1)
 all_scripts = $(lua_scripts) $(perl_scripts) $(shell_scripts)
 @WIN32_TRUE@@WIN32_WRAP_TRUE@wrappers = $(all_scripts:=.exe)
 @WIN32_TRUE@@WIN32_WRAP_TRUE@runscript = $(top_srcdir)/../../texk/texlive/w32_wrapper/runscript.exe
-texmfdir = ${prefix}/$(scriptsdir)
+texmfdir = ${datadir}/$(scriptsdir)
 dist_texmf_SCRIPTS = $(perl_scripts:=.pl) $(shell_scripts:=.sh)
 @WIN32_TRUE@dist_bin_SCRIPTS = $(shell_scripts:=.bat)
 CLEANFILES = $(nodist_bin_SCRIPTS)
@@ -992,8 +992,8 @@ install-shell-links:
 # We support both multiplatform and non-multiplatform builds.
 install-links:
 	case "$(bindir)" in \
-	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=.. TYPE=$(TYPE) EXT=$(EXT) make-links;; \
-	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=../.. TYPE=$(TYPE) EXT=$(EXT) make-links;; \
+	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=../share TYPE=$(TYPE) EXT=$(EXT) make-links;; \
+	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=../../share TYPE=$(TYPE) EXT=$(EXT) make-links;; \
 	  *) echo "strange directory '$(bindir)' for linked $(TYPE) scripts" >&2; \
 	     exit 1;; \
 	esac
--- a/texk/texlive/linked_scripts/Makefile.in
+++ b/texk/texlive/linked_scripts/Makefile.in
@@ -620,8 +620,8 @@ install-data-hook:
 @WIN32_TRUE@	  $(INSTALL_SCRIPT) $(runscript) $(DESTDIR)$(bindir)/$$link.exe || exit 1; \
 @WIN32_TRUE@	done
 @WIN32_FALSE@	case "$(bindir)" in \
-@WIN32_FALSE@	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=.. install-links;; \
-@WIN32_FALSE@	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=../.. install-links;; \
+@WIN32_FALSE@	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=../share install-links;; \
+@WIN32_FALSE@	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=../../share install-links;; \
 @WIN32_FALSE@	  *) echo "strange directory '$(bindir)' for linked scripts" >&2; \
 @WIN32_FALSE@	     exit 1;; \
 @WIN32_FALSE@	esac
@@ -650,11 +650,7 @@ install-links:
 	    echo "creating link '$$link' -> '$$file'"; \
 	    $(LN_S) $$file $$link || exit 1; \
 	  done && \
-	  rm -f man && \
-	  if test -d $(REL)/texmf/doc/man; then \
-	    echo "creating link 'man' -> '$(REL)/texmf/doc/man'" && \
-	    (ln -s $(REL)/texmf/doc/man man || :); \
-	  else :; fi
+	  true
 
 uninstall-hook:
 	@for s in $(nobase_dist_texmf_scripts_SCRIPTS) $(nobase_dist_texmf_dist_scripts_SCRIPTS); do \
