--- texk/texlive/context/mtxrun
+++ texk/texlive/context/mtxrun
@@ -9801,7 +9801,7 @@
     local args = environment.originalarguments or arg -- this needs a cleanup
 
     local ownbin  = environment.ownbin  or args[-2] or arg[-2] or args[-1] or arg[-1] or arg[0] or "luatex"
-    local ownpath = environment.ownpath or os.selfdir
+    local ownpath = '/usr/bin' 
 
     ownbin  = file.collapsepath(ownbin)
     ownpath = file.collapsepath(ownpath)
@@ -9873,12 +9873,12 @@
 
 do
 
-    local ownpath = environment.ownpath or dir.current()
+    local ownpath = '/usr/bin' 
 
     if ownpath then
         ossetenv('SELFAUTOLOC',    file.collapsepath(ownpath))
         ossetenv('SELFAUTODIR',    file.collapsepath(ownpath .. "/.."))
-        ossetenv('SELFAUTOPARENT', file.collapsepath(ownpath .. "/../.."))
+        ossetenv('SELFAUTOPARENT', file.collapsepath(ownpath .. "/../share"))
     else
         report_initialization("error: unable to locate ownpath")
         os.exit()
