--- texk/web2c/tangleboot.p.orig	2009-08-06 13:53:53.000000000 +0200
+++ texk/web2c/tangleboot.p	2009-08-06 13:54:25.000000000 +0200
@@ -577,7 +577,7 @@
 if not inputln(webfile)then begin begin writeln(stdout);
 write(stdout,'! WEB file ended during a change');error;end;
 inputhasended:=true;goto 10;end;if linesdontmatch then n:=n+1;end;
-10:end;{:132}{135:}procedure getline;label 20;
+10:end;{:132}{135:}procedure tex_getline;label 20;
 begin 20:if changing then{137:}begin line:=line+1;
 if not inputln(changefile)then begin begin writeln(stdout);
 write(stdout,'! Change file ended without @z');error;end;buffer[0]:=64;
@@ -604,13 +604,13 @@
 60:controlcode:=135;61:controlcode:=2;92:controlcode:=3;
 others:controlcode:=0 end;end;{:139}{140:}function skipahead:eightbits;
 label 30;var c:eightbits;
-begin while true do begin if loc>limit then begin getline;
+begin while true do begin if loc>limit then begin tex_getline;
 if inputhasended then begin c:=136;goto 30;end;end;buffer[limit+1]:=64;
 while buffer[loc]<>64 do loc:=loc+1;if loc<=limit then begin loc:=loc+2;
 c:=controlcode(buffer[loc-1]);if(c<>0)or(buffer[loc-1]=62)then goto 30;
 end;end;30:skipahead:=c;end;{:140}{141:}procedure skipcomment;label 10;
 var bal:eightbits;c:ASCIIcode;begin bal:=0;
-while true do begin if loc>limit then begin getline;
+while true do begin if loc>limit then begin tex_getline;
 if inputhasended then begin begin writeln(stdout);
 write(stdout,'! Input ended in mid-comment');error;end;goto 10;end;end;
 c:=buffer[loc];loc:=loc+1;{142:}if c=64 then begin c:=buffer[loc];
@@ -622,7 +622,7 @@
 then bal:=bal+1 else if c=125 then begin if bal=0 then goto 10;
 bal:=bal-1;end{:142};end;10:end;{:141}{145:}function getnext:eightbits;
 label 20,30,31;var c:eightbits;d:eightbits;j,k:0..longestname;
-begin 20:if loc>limit then begin getline;
+begin 20:if loc>limit then begin tex_getline;
 if inputhasended then begin c:=136;goto 31;end;end;c:=buffer[loc];
 loc:=loc+1;
 if scanninghex then{146:}if((c>=48)and(c<=57))or((c>=65)and(c<=70))then
@@ -647,7 +647,7 @@
 64:{150:}begin c:=controlcode(buffer[loc]);loc:=loc+1;
 if c=0 then goto 20 else if c=13 then scanninghex:=true else if c=135
 then{151:}begin{153:}k:=0;
-while true do begin if loc>limit then begin getline;
+while true do begin if loc>limit then begin tex_getline;
 if inputhasended then begin begin writeln(stdout);
 write(stdout,'! Input ended in section name');error;end;goto 30;end;end;
 d:=buffer[loc];{154:}if d=64 then begin d:=buffer[loc+1];
