--- a/texk/kpathsea/texmf.cnf
+++ b/texk/kpathsea/texmf.cnf
@@ -49,7 +49,7 @@
 % (http://tug.org/tds), or files may not be found.
 %
 % Redistributors will probably want $SELFAUTODIR/share, i.e., /usr/share.
-TEXMFROOT = $SELFAUTOPARENT
+TEXMFROOT = $SELFAUTOPARENT/share
 
 % The tree containing runtime files related to the specific
 % distribution and version.
@@ -59,10 +59,10 @@ TEXMFMAIN = $TEXMFROOT/texmf
 TEXMFDIST = $TEXMFROOT/texmf-dist
 
 % Local additions to the distribution trees.
-TEXMFLOCAL = $TEXMFROOT/../texmf-local
+TEXMFLOCAL = $TEXMFROOT/texmf-local
 
 % TEXMFSYSVAR, where *-sys store cached runtime data.
-TEXMFSYSVAR = $TEXMFROOT/texmf-var
+TEXMFSYSVAR = /var/tmp/texmf-var
 
 % TEXMFSYSCONFIG, where *-sys store configuration data.
 TEXMFSYSCONFIG = $TEXMFROOT/texmf-config
