diff -Naur a/js/src/old-configure.in b/js/src/old-configure.in
--- a/js/src/old-configure.in	2022-12-13 07:48:54.317336755 +0200
+++ b/js/src/old-configure.in	2022-12-13 07:49:43.722154247 +0200
@@ -255,15 +255,6 @@
                   AC_MSG_RESULT([no])
                   LDFLAGS=$_SAVE_LDFLAGS)
 
-    if test -z "$DEVELOPER_OPTIONS" -o "$OS_TARGET" = "Android"; then
-        AC_MSG_CHECKING([for --build-id=sha1 option to ld])
-        _SAVE_LDFLAGS=$LDFLAGS
-        LDFLAGS="$LDFLAGS -Wl,--build-id=sha1"
-        AC_TRY_LINK(,,AC_MSG_RESULT([yes]),
-                      AC_MSG_RESULT([no])
-                      LDFLAGS=$_SAVE_LDFLAGS)
-    fi
-
     _DEFINES_CFLAGS="-include $jsconfdefs -DMOZILLA_CLIENT"
 fi
 
diff -Naur a/old-configure.in b/old-configure.in
--- a/old-configure.in	2022-12-13 07:48:54.169334306 +0200
+++ b/old-configure.in	2022-12-13 07:50:01.182443153 +0200
@@ -269,15 +269,6 @@
                   AC_MSG_RESULT([no])
                   LDFLAGS=$_SAVE_LDFLAGS)
 
-    if test -z "$DEVELOPER_OPTIONS" -o "$OS_TARGET" = "Android"; then
-        AC_MSG_CHECKING([for --build-id=sha1 option to ld])
-        _SAVE_LDFLAGS=$LDFLAGS
-        LDFLAGS="$LDFLAGS -Wl,--build-id=sha1"
-        AC_TRY_LINK(,,AC_MSG_RESULT([yes]),
-                      AC_MSG_RESULT([no])
-                      LDFLAGS=$_SAVE_LDFLAGS)
-    fi
-
     AC_MSG_CHECKING([for --ignore-unresolved-symbol option to ld])
     HAVE_LINKER_SUPPORT_IGNORE_UNRESOLVED=
     _SAVE_LDFLAGS=$LDFLAGS
