diff -Naur a/js/src/old-configure.in b/js/src/old-configure.in
--- a/js/src/old-configure.in	2022-12-13 07:48:54.317336755 +0200
+++ b/js/src/old-configure.in	2022-12-13 07:49:43.722154247 +0200
@@ -239,12 +239,7 @@ if test "$GNU_CC"; then
                   AC_MSG_RESULT([no])
                   LDFLAGS=$_SAVE_LDFLAGS)
 
-    # While sha1 is deterministic, it is slower.
-    if test -z "$DEVELOPER_OPTIONS" -o "$OS_TARGET" = "Android"; then
-        build_id=sha1
-    else
-        build_id=uuid
-    fi
+    build_id=uuid
     AC_MSG_CHECKING([for --build-id=$build_id option to ld])
     _SAVE_LDFLAGS=$LDFLAGS
     LDFLAGS="$LDFLAGS -Wl,--build-id=$build_id"
diff -Naur a/old-configure.in b/old-configure.in
--- a/old-configure.in	2022-12-13 07:48:54.169334306 +0200
+++ b/old-configure.in	2022-12-13 07:50:01.182443153 +0200
@@ -253,12 +253,7 @@ if test "$GNU_CC"; then
                   AC_MSG_RESULT([no])
                   LDFLAGS=$_SAVE_LDFLAGS)
 
-    # While sha1 is deterministic, it is slower.
-    if test -z "$DEVELOPER_OPTIONS" -o "$OS_TARGET" = "Android"; then
-        build_id=sha1
-    else
-        build_id=uuid
-    fi
+    build_id=uuid
     AC_MSG_CHECKING([for --build-id=$build_id option to ld])
     _SAVE_LDFLAGS=$LDFLAGS
     LDFLAGS="$LDFLAGS -Wl,--build-id=$build_id"
