--- a/dom/canvas/WebGLStrongTypes.h
+++ b/dom/canvas/WebGLStrongTypes.h
@@ -7,6 +7,7 @@
 #define WEBGL_STRONG_TYPES_H_
 
 #include <algorithm>
+#include <array>
 
 #include "GLDefs.h"
 #include "mozilla/Assertions.h"
--- a/intl/components/src/TimeZone.cpp
+++ b/intl/components/src/TimeZone.cpp
@@ -4,6 +4,7 @@
 
 #include "mozilla/intl/TimeZone.h"
 
+#include "mozilla/Try.h"
 #include "mozilla/Vector.h"
 
 #include <algorithm>
--- a/toolkit/components/telemetry/pingsender/pingsender.h
+++ b/toolkit/components/telemetry/pingsender/pingsender.h
@@ -6,6 +6,7 @@
 #ifndef mozilla_telemetry_pingsender_h
 #define mozilla_telemetry_pingsender_h
 
+#include <cstdint>
 #include <string>
 
 #ifdef DEBUG
--- a/widget/gtk/WakeLockListener.cpp
+++ b/widget/gtk/WakeLockListener.cpp
@@ -14,6 +14,7 @@
 #include "nsIStringBundle.h"
 #include "nsReadableUtils.h"
 #include "nsContentUtils.h"
+#include "prenv.h"
 
 #ifdef MOZ_ENABLE_DBUS
 #  include <gio/gio.h>
