diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index c99b4d628c54..0b1b1614620a 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -703,9 +703,10 @@ set_define("MOZ_AV1", av1)
 option("--disable-jxl", help="Disable jxl image support")
 
 
-@depends("--disable-jxl", milestone.is_nightly)
-def jxl(value, is_nightly):
-    if is_nightly and value:
+@depends("--disable-jxl")
+def jxl(value):
+    enabled = bool(value)
+    if enabled:
         return True
 
 
-- 
2.38.1

