diff --git a/dom/media/platforms/moz.build b/dom/media/platforms/moz.build
--- a/dom/media/platforms/moz.build
+++ b/dom/media/platforms/moz.build
@@ -72,10 +72,12 @@
         "agnostic/DAV1DDecoder.cpp",
     ]
     if CONFIG["MOZ_SYSTEM_AV1"]:
         CXXFLAGS += CONFIG["MOZ_SYSTEM_LIBAOM_CFLAGS"]
         CXXFLAGS += CONFIG["MOZ_SYSTEM_LIBDAV1D_CFLAGS"]
+        OS_LIBS += CONFIG["MOZ_SYSTEM_LIBAOM_LIBS"]
+        OS_LIBS += CONFIG["MOZ_SYSTEM_LIBDAV1D_LIBS"]
 
 if CONFIG["MOZ_OMX"]:
     EXPORTS += [
         "omx/OmxCoreLibLinker.h",
     ]
diff --git a/media/ffvpx/libavcodec/moz.build b/media/ffvpx/libavcodec/moz.build
--- a/media/ffvpx/libavcodec/moz.build
+++ b/media/ffvpx/libavcodec/moz.build
@@ -120,11 +120,13 @@
         'vpx_rac.c',
     ]
 
     if CONFIG["MOZ_SYSTEM_AV1"]:
         CFLAGS += CONFIG['MOZ_SYSTEM_LIBDAV1D_CFLAGS']
+        CFLAGS += CONFIG['MOZ_SYSTEM_LIBAOM_CFLAGS']
         OS_LIBS += CONFIG['MOZ_SYSTEM_LIBDAV1D_LIBS']
+        OS_LIBS += CONFIG['MOZ_SYSTEM_LIBAOM_LIBS']
     else:
         USE_LIBS += [
             'dav1d',
             'media_libdav1d_asm',
         ]

