diff --git a/tools/profiler/rust-api/build.rs b/tools/profiler/rust-api/build.rs
index 70e6096..d415db5 100644
--- a/tools/profiler/rust-api/build.rs
+++ b/tools/profiler/rust-api/build.rs
@@ -84,6 +84,13 @@ fn generate_bindings() {
         .rustified_enum("mozilla::MarkerSchema_Location")
         .rustified_enum("mozilla::MarkerSchema_Format")
         .rustified_enum("mozilla::MarkerSchema_Searchable")
+        .blocklist_type(".*basic_string_view.*")
+        .opaque_type(".*basic_string_view.*")
+        .blocklist_type("std::.*basic_string_view.*")
+        .opaque_type("std::.*basic_string_view.*")
+        .blocklist_type("std::__1::basic_string_view.*")
+        .opaque_type("std::__1::basic_string_view.*")
+        .blocklist_type(".*basic_string___self_view.*")
         // Converting std::string to an opaque type makes some platforms build
         // successfully. Otherwise, it fails to build because MarkerSchema has
         // some std::strings as its fields.
