--- a/dom/media/flac/FlacDecoder.cpp  2023-11-23 07:59:17.466486168 -0500
+++ b/dom/media/flac/FlacDecoder.cpp  2023-11-23 08:00:49.048026379 -0500
@@ -18,7 +18,7 @@
 #elif defined(MOZ_FFMPEG)
   RefPtr<PDMFactory> platform = new PDMFactory();
   return StaticPrefs::media_flac_enabled() &&
-         (platform->SupportsMimeType("audio/flac"_ns) != media::DecodeSupport::Unsupported);
+         !(platform->SupportsMimeType("audio/flac"_ns).isEmpty());
 #else
   return false;
 #endif
