From abc720ed6f0451e3f1c5ff9ae074ae02684306f7 Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Sat, 29 Aug 2020 22:06:15 +0200
Subject: [PATCH 24/30] build: Disable -Werror

Bug: https://bugzilla.mozilla.org/822978
Signed-off-by: Thomas Deutschmann <whissi@gentoo.org>
---
 build/moz.configure/warnings.configure | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build/moz.configure/warnings.configure b/build/moz.configure/warnings.configure
index 0d925a9b80..2ac91626c4 100755
--- a/build/moz.configure/warnings.configure
+++ b/build/moz.configure/warnings.configure
@@ -216,7 +216,8 @@ with only_when(depends(target)(lambda t: t.kernel == "WINNT")):
 check_and_add_gcc_warning("-Wno-gnu-zero-variadic-macro-arguments")
 
 # Make it an error to be missing function declarations for C code.
-check_and_add_gcc_warning("-Werror=implicit-function-declaration", c_compiler)
+check_and_add_gcc_warning("-Werror=implicit-function-declaration",
+                         when="--enable-warnings-as-errors")
 
 # New in clang 11. We can't really do anything about this warning.
 check_and_add_gcc_warning("-Wno-psabi")
-- 
2.34.1

