diff --git a/build/pgo/profileserver.py b/build/pgo/profileserver.py
index 7354714..0385792 100755
--- a/build/pgo/profileserver.py
+++ b/build/pgo/profileserver.py
@@ -171,6 +171,9 @@ if __name__ == "__main__":
                 env["UPLOAD_PATH"], "profile-run-1.log"
             )
 
+        # Workaround for https://bugs.gentoo.org/929967
+        env["LD_LIBRARY_PATH"] = os.path.join(os.getcwd(), "dist", "bin")
+
         # Run Firefox a first time to initialize its profile
         runner = FirefoxRunner(
             profile=profile,
