diff --git a/config/system-headers.mozbuild b/config/system-headers.mozbuild
--- a/config/system-headers.mozbuild
+++ b/config/system-headers.mozbuild
@@ -855,11 +855,10 @@
     "sys/sem.h",
     "sys/sendfile.h",
     "sys/shm.h",
     "sys/siginfo.h",
     "sys/signal.h",
-    "sys/single_threaded.h",
     "sys/socket.h",
     "sys/sockio.h",
     "sys/sparc/frame.h",
     "sys/stack.h",
     "sys/statfs.h",
@@ -1030,10 +1029,15 @@
 if CONFIG["HAVE_EXECINFO"]:
     system_headers += [
         "execinfo.h",
     ]
 
+if CONFIG["HAVE_SINGLE_THREADED_H"]:
+    system_headers += [
+        "sys/single_threaded.h",
+    ]
+
 if CONFIG["MOZ_X11"]:
     system_headers += [
         "gdk/gdkx.h",
         "gtk/gtkx.h",
         "X11/cursorfont.h",
diff --git a/toolkit/moz.configure b/toolkit/moz.configure
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -1622,10 +1622,12 @@
 with only_when(compile_environment & depends(target.os)(lambda os: os != "WINNT")):
     set_define("HAVE_SYSIOCCOM_H", check_header("sys/ioccom.h"))
 
     set_config("HAVE_EXECINFO", check_header("execinfo.h"))
 
+    set_config("HAVE_SINGLE_THREADED_H", check_header("sys/single_threaded.h"))
+
 # Elfhack
 # ==============================================================
 with only_when("--enable-compile-environment"):
 
     @depends(host, target)


