--- a/ipc/glue/SharedMemoryHandle.h	2026-05-08 09:41:05.625846466 -0400
+++ b/ipc/glue/SharedMemoryHandle.h	2026-05-08 09:40:58.170035583 -0400
@@ -5,6 +5,7 @@
 #ifndef mozilla_ipc_SharedMemoryHandle_h
 #define mozilla_ipc_SharedMemoryHandle_h
 
+#include <cstdint>
 #include <utility>
 
 #include "mozilla/UniquePtrExtensions.h"
--- a/modules/woff2/include/woff2/output.h	2026-04-27 09:33:07.000000000 -0400
+++ b/modules/woff2/include/woff2/output.h	2026-05-08 20:55:09.770322838 -0400
@@ -10,6 +10,7 @@
 #define WOFF2_WOFF2_OUT_H_
 
 #include <algorithm>
+#include <cstdint>
 #include <cstring>
 #include <memory>
 #include <string>
 