From e56b9b14238305b82c32cc58c71d1263feab9710 Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Sat, 29 Aug 2020 22:06:15 +0200
Subject: [PATCH 27/37] build: Disable -Werror

Bug: https://bugzilla.mozilla.org/822978
Signed-off-by: Thomas Deutschmann <whissi@gentoo.org>
---
 build/moz.configure/warnings.configure | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build/moz.configure/warnings.configure b/build/moz.configure/warnings.configure
index d0db70f6d3..a2b1f78a5d 100755
--- a/build/moz.configure/warnings.configure
+++ b/build/moz.configure/warnings.configure
@@ -234,7 +234,8 @@ with only_when(depends(target)(lambda t: t.kernel == "WINNT")):
 check_and_add_gcc_warning("-Wno-gnu-zero-variadic-macro-arguments")
 
 # Make it an error to be missing function declarations for C code.
-check_and_add_gcc_warning("-Werror=implicit-function-declaration", c_compiler)
+check_and_add_gcc_warning("-Werror=implicit-function-declaration",
+                         when="--enable-warnings-as-errors")
 
 # New in clang 11. We can't really do anything about this warning.
 check_and_add_gcc_warning("-Wno-psabi")
-- 
2.31.1

