--- a/js/src/gc/Marking.cpp
+++ b/js/src/gc/Marking.cpp
@@ -909,6 +909,12 @@ void MarkingTracerT<opts>::onEdge(T** thingp, const char* name) {
       type * *thingp, const char* name);                              \
   template void                                                       \
   MarkingTracerT<MarkingOptions::MarkRootCompartments>::onEdge<type>( \
+      type * *thingp, const char* name);                              \
+  template void                                                       \
+  MarkingTracerT<MarkingOptions::AtomicMarking>::onEdge<type>(        \
+      type * *thingp, const char* name);                              \
+  template void                                                       \
+  MarkingTracerT<ConcurrentMarkingOptions>::onEdge<type>(             \
       type * *thingp, const char* name);
 JS_FOR_EACH_TRACEKIND(INSTANTIATE_ONEDGE_METHOD)
 #undef INSTANTIATE_ONEDGE_METHOD
