diff -Naur a/build/moz.configure/flags.configure b/build/moz.configure/flags.configure
--- a/build/moz.configure/flags.configure	2024-03-11 21:28:51.000000000 +0200
+++ b/build/moz.configure/flags.configure	2024-03-19 08:28:05.913736113 +0200
@@ -167,9 +167,6 @@
 
 check_and_add_flag("-pipe", when=is_gcc)
 
-check_and_add_linker_flag("-Wl,--build-id=uuid", when=check_build_id_uuid)
-check_and_add_linker_flag("-Wl,--build-id=sha1", when=check_build_id_sha1)
-
 check_and_add_asm_flag("-Wa,--noexecstack", when=is_gnu_cc)
 check_and_add_linker_flag("-Wl,-z,noexecstack", when=is_gnu_cc)
 check_and_add_linker_flag("-Wl,-z,text", when=is_gnu_cc)
