diff -Naur a/gfx/skia/generate_mozbuild.py b/gfx/skia/generate_mozbuild.py
--- a/gfx/skia/generate_mozbuild.py	2023-06-06 13:26:33.188693705 -0000
+++ b/gfx/skia/generate_mozbuild.py	2023-06-06 13:27:25.821463691 -0000
@@ -61,6 +61,8 @@
 elif CONFIG['CPU_ARCH'] == 'aarch64' and CONFIG['CC_TYPE'] in ('clang', 'gcc'):
     SOURCES['skia/src/opts/SkOpts_crc32.cpp'].flags += ['-march=armv8-a+crc']
 
+DEFINES['MOZ_SKIA'] = True
+
 DEFINES['SKIA_IMPLEMENTATION'] = 1
 
 DEFINES['SK_PDF_USE_HARFBUZZ_SUBSETTING'] = 1
diff -Naur a/gfx/skia/moz.build b/gfx/skia/moz.build
--- a/gfx/skia/moz.build	2023-06-06 13:26:33.189693720 -0000
+++ b/gfx/skia/moz.build	2023-06-06 13:27:48.805802442 -0000
@@ -572,6 +572,8 @@
 elif CONFIG['CPU_ARCH'] == 'aarch64' and CONFIG['CC_TYPE'] in ('clang', 'gcc'):
     SOURCES['skia/src/opts/SkOpts_crc32.cpp'].flags += ['-march=armv8-a+crc']
 
+DEFINES['MOZ_SKIA'] = True
+
 DEFINES['SKIA_IMPLEMENTATION'] = 1
 
 DEFINES['SK_PDF_USE_HARFBUZZ_SUBSETTING'] = 1
