diff --git a/security/sandbox/chromium/sandbox/linux/system_headers/x86_32_linux_syscalls.h b/security/sandbox/chromium/sandbox/linux/system_headers/x86_32_linux_syscalls.h
index dc846ee..0f6b98d 100644
--- a/security/sandbox/chromium/sandbox/linux/system_headers/x86_32_linux_syscalls.h
+++ b/security/sandbox/chromium/sandbox/linux/system_headers/x86_32_linux_syscalls.h
@@ -1279,6 +1279,10 @@
 #define __NR_epoll_pwait 319
 #endif
 
+#if !defined(__NR_epoll_pwait2)
+#define __NR_epoll_pwait 441
+#endif
+
 #if !defined(__NR_utimensat)
 #define __NR_utimensat 320
 #endif
diff --git a/security/sandbox/chromium/sandbox/linux/system_headers/x86_64_linux_syscalls.h b/security/sandbox/chromium/sandbox/linux/system_headers/x86_64_linux_syscalls.h
index ab51703..ff96dc4 100644
--- a/security/sandbox/chromium/sandbox/linux/system_headers/x86_64_linux_syscalls.h
+++ b/security/sandbox/chromium/sandbox/linux/system_headers/x86_64_linux_syscalls.h
@@ -1143,6 +1143,10 @@
 #define __NR_epoll_pwait 281
 #endif
 
+#if !defined(__NR_epoll_pwait2)
+#define __NR_epoll_pwait2 441
+#endif
+
 #if !defined(__NR_signalfd)
 #define __NR_signalfd 282
 #endif
diff --git a/security/sandbox/linux/SandboxFilterUtil.h b/security/sandbox/linux/SandboxFilterUtil.h
index 6e9180b..abdeb44 100644
--- a/security/sandbox/linux/SandboxFilterUtil.h
+++ b/security/sandbox/linux/SandboxFilterUtil.h
@@ -215,7 +215,16 @@ class SandboxPolicyBase : public sandbox::bpf_dsl::Policy {
 #  define CASES_FOR_epoll_create case __NR_epoll_create1
 #endif
 
-#ifdef __NR_epoll_wait
+#if defined(__NR_epoll_pwait2) && defined(__NR_epoll_wait)
+#  define CASES_FOR_epoll_wait \
+    case __NR_epoll_wait:      \
+    case __NR_epoll_pwait:     \
+    case __NR_epoll_pwait2
+#elif defined(__NR_epoll_pwait2)
+#  define CASES_FOR_epoll_wait \
+    case __NR_epoll_pwait:     \
+    case __NR_epoll_pwait2
+#elif defined(__NR_epoll_wait)
 #  define CASES_FOR_epoll_wait \
     case __NR_epoll_wait:      \
     case __NR_epoll_pwait
