diff --git a/toolkit/components/remote/moz.build b/toolkit/components/remote/moz.build
--- a/toolkit/components/remote/moz.build
+++ b/toolkit/components/remote/moz.build
@@ -23,10 +23,11 @@
         ]
         EXPORTS += [
             "nsUnixRemoteServer.h",
             "RemoteUtils.h",
         ]
+        CXXFLAGS += CONFIG["MOZ_DBUS_CFLAGS"]
     else:
         SOURCES += [
             "nsGTKRemoteServer.cpp",
             "nsXRemoteClient.cpp",
             "nsXRemoteServer.cpp",
diff --git a/widget/gtk/moz.build b/widget/gtk/moz.build
--- a/widget/gtk/moz.build
+++ b/widget/gtk/moz.build
@@ -172,7 +172,8 @@
         "AsyncDBus.h",
     ]
     UNIFIED_SOURCES += [
         "AsyncDBus.cpp",
     ]
+    CXXFLAGS += CONFIG["MOZ_DBUS_CFLAGS"]
 
 CXXFLAGS += ["-Werror=switch"]

