--- a/widget/gtk/nsAppShell.cpp
+++ b/widget/gtk/nsAppShell.cpp
@@ -296,8 +296,10 @@
         do_GetService(POWERMANAGERSERVICE_CONTRACTID);
 
     if (powerManagerService) {
+#ifdef MOZ_ENABLE_DBUS
       powerManagerService->AddWakeLockListener(
           WakeLockListener::GetSingleton());
+#endif
     } else {
       NS_WARNING(
           "Failed to retrieve PowerManagerService, wakelocks will be broken!");
