From 7d1ba8593c2a4f72b52b20035e22d56be4df7a91 Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Wed, 27 Mar 2024 17:47:14 +0530
Subject: [PATCH 1/4] bmo#1882942: add missing header for size_t

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 layout/style/PseudoStyleType.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/layout/style/PseudoStyleType.h b/layout/style/PseudoStyleType.h
index 6804b500c3..88eb4e4cf8 100644
--- a/layout/style/PseudoStyleType.h
+++ b/layout/style/PseudoStyleType.h
@@ -7,6 +7,7 @@
 #ifndef mozilla_PseudoStyleType_h
 #define mozilla_PseudoStyleType_h
 
+#include <cstddef>
 #include <cstdint>
 #include <iosfwd>
 
-- 
2.44.0

From 24d3b03711bcb152ecb30a05a25d33ce8688f960 Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Wed, 27 Mar 2024 17:50:20 +0530
Subject: [PATCH 2/2] bmo#1862601: add missing header for MOZ_TRY in
 non-unified builds and system-icu

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 intl/components/src/TimeZone.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/intl/components/src/TimeZone.h b/intl/components/src/TimeZone.h
index 9d0423ef13..89770839ae 100644
--- a/intl/components/src/TimeZone.h
+++ b/intl/components/src/TimeZone.h
@@ -33,6 +33,7 @@
 #include "mozilla/Maybe.h"
 #include "mozilla/Result.h"
 #include "mozilla/Span.h"
+#include "mozilla/Try.h"
 #include "mozilla/UniquePtr.h"
 
 namespace mozilla::intl {
-- 
2.44.0

From 6345da9269c0511c1f8c60fad5aaa14734c84e4d Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Wed, 27 Mar 2024 18:48:33 +0530
Subject: [PATCH 3/3] GtkCompositorWidget.h: add missing header for
 GdkIsWaylandDisplay in non-unified builds

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 widget/gtk/GtkCompositorWidget.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/widget/gtk/GtkCompositorWidget.h b/widget/gtk/GtkCompositorWidget.h
index 5bf89835d7..d5c279a75f 100644
--- a/widget/gtk/GtkCompositorWidget.h
+++ b/widget/gtk/GtkCompositorWidget.h
@@ -8,6 +8,7 @@
 
 #include "GLDefs.h"
 #include "mozilla/DataMutex.h"
+#include "mozilla/WidgetUtilsGtk.h"
 #include "mozilla/widget/CompositorWidget.h"
 #include "WindowSurfaceProvider.h"
 
-- 
2.44.0

From 9ba53ddea3ebaeb335ca17aca4d33dad04526f15 Mon Sep 17 00:00:00 2001
From: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
Date: Thu, 28 Mar 2024 17:31:11 +0530
Subject: [PATCH 4/4] 
 accessible/html/{HTMLSelectAccessible,HTMLTableAccessible}.cpp: add missing
 header DocAccessible-inl.h for non-unified builds

Signed-off-by: Amit Prakash Ambasta <amit.prakash.ambasta@gmail.com>
---
 accessible/html/HTMLSelectAccessible.cpp | 1 +
 accessible/html/HTMLTableAccessible.cpp  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/accessible/html/HTMLSelectAccessible.cpp b/accessible/html/HTMLSelectAccessible.cpp
index f8fb4180c7..02d449df36 100644
--- a/accessible/html/HTMLSelectAccessible.cpp
+++ b/accessible/html/HTMLSelectAccessible.cpp
@@ -6,6 +6,7 @@
 #include "HTMLSelectAccessible.h"
 
 #include "LocalAccessible-inl.h"
+#include "DocAccessible-inl.h"
 #include "nsAccessibilityService.h"
 #include "nsAccUtils.h"
 #include "DocAccessible.h"
diff --git a/accessible/html/HTMLTableAccessible.cpp b/accessible/html/HTMLTableAccessible.cpp
index 2c3dc6b82d..f4b8f9964a 100644
--- a/accessible/html/HTMLTableAccessible.cpp
+++ b/accessible/html/HTMLTableAccessible.cpp
@@ -12,6 +12,7 @@
 #include "ARIAMap.h"
 #include "CacheConstants.h"
 #include "LocalAccessible-inl.h"
+#include "DocAccessible-inl.h"
 #include "nsTextEquivUtils.h"
 #include "Relation.h"
 #include "mozilla/a11y/Role.h"
-- 
2.44.0


