diff -Naur a/build/moz.configure/util.configure b/build/moz.configure/util.configure
--- a/build/moz.configure/util.configure	2023-08-07 09:25:05.036606566 -0000
+++ b/build/moz.configure/util.configure	2023-08-07 11:16:45.797899734 -0000
@@ -217,9 +217,9 @@
 ):
     compiler_path = compiler[0]
     compiler = wrapper + compiler
-    use_cache = True
+    use_cache = configure_cache is not None
 
-    if compiler_path not in configure_cache.version_checked_compilers:
+    if use_cache and compiler_path not in configure_cache.version_checked_compilers:
         try:
             version_info = subprocess.check_output(
                 [compiler_path, "--version"],
diff -Naur a/toolkit/moz.configure b/toolkit/moz.configure
--- a/toolkit/moz.configure	2023-08-07 09:25:00.818403536 -0000
+++ b/toolkit/moz.configure	2023-08-07 11:17:44.794947991 -0000
@@ -1618,7 +1618,6 @@
         extra_toolchain_flags,
         linker_ldflags,
         readelf,
-        configure_cache,
         when=may_use_pack_relative_relocs,
     )
     @checking("for -z pack-relative-relocs option to ld", bool)
@@ -1631,7 +1630,6 @@
         extra_toolchain_flags,
         linker_ldflags,
         readelf,
-        configure_cache,
     ):
         try:
             fd, path = mkstemp(prefix="conftest.")
@@ -1640,7 +1638,7 @@
             pack_rel_relocs = ["-Wl,-z,pack-relative-relocs"]
             if (
                 try_invoke_compiler(
-                    configure_cache,
+                    None,
                     [c_compiler.compiler] + c_compiler.flags,
                     c_compiler.language,
                     "int main() { return 0; }",
