diff --git a/widget/gtk/nsWaylandDisplay.cpp b/widget/gtk/nsWaylandDisplay.cpp
--- a/widget/gtk/nsWaylandDisplay.cpp
+++ b/widget/gtk/nsWaylandDisplay.cpp
@@ -288,19 +288,19 @@
 
 bool nsWaylandDisplay::Matches(wl_display* aDisplay) {
   return mThreadId == PR_GetCurrentThread() && aDisplay == mDisplay;
 }
 
-static void WlCrashHandler(const char* format, va_list args) {
-  MOZ_CRASH_UNSAFE(g_strdup_vprintf(format, args));
+static void WlLogHandler(const char* format, va_list args) {
+  vfprintf(stderr, format, args);
 }
 
 nsWaylandDisplay::nsWaylandDisplay(wl_display* aDisplay)
     : mThreadId(PR_GetCurrentThread()), mDisplay(aDisplay) {
   // GTK sets the log handler on display creation, thus we overwrite it here
   // in a similar fashion
-  wl_log_set_handler_client(WlCrashHandler);
+  wl_log_set_handler_client(WlLogHandler);
 
   wl_registry* registry = wl_display_get_registry(mDisplay);
   wl_registry_add_listener(registry, &registry_listener, this);
   if (!NS_IsMainThread()) {
     mEventQueue = wl_display_create_queue(mDisplay);

