From eb2b14167139b890c45da22917429a3dd29f510c Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Wed, 16 Jun 2021 02:13:33 +0200
Subject: [PATCH 34/36] bmo#1646135: Disable HW-WR on Nvidia prop. drivers on
 Wayland

Signed-off-by: Thomas Deutschmann <whissi@gentoo.org>
---
 widget/gtk/GfxInfo.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/widget/gtk/GfxInfo.cpp b/widget/gtk/GfxInfo.cpp
index fd6e7509fd..d3f3baf311 100644
--- a/widget/gtk/GfxInfo.cpp
+++ b/widget/gtk/GfxInfo.cpp
@@ -682,6 +682,16 @@ const nsTArray<GfxDriverInfo>& GfxInfo::GetGfxDriverInfo() {
         nsIGfxInfo::FEATURE_BLOCKED_DEVICE, DRIVER_LESS_THAN, V(460, 32, 3, 0),
         "FEATURE_FAILURE_WEBRENDER_OLD_NVIDIA", "460.32.03");
 
+    // Disable Nvidia proprietary drivers on Wayland.
+    APPEND_TO_DRIVER_BLOCKLIST_EXT(
+        OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
+        DesktopEnvironment::All, WindowProtocol::Wayland,
+        DriverVendor::NonMesaAll, DeviceFamily::NvidiaAll,
+        nsIGfxInfo::FEATURE_WEBRENDER, nsIGfxInfo::FEATURE_BLOCKED_DEVICE,
+        DRIVER_COMPARISON_IGNORED, V(0, 0, 0, 0),
+        "FEATURE_FAILURE_WEBRENDER_NVIDIA_WAYLAND",
+        "https://bugzilla.mozilla.org/show_bug.cgi?id=1646135");
+
     // ATI Mesa baseline, chosen arbitrarily.
     APPEND_TO_DRIVER_BLOCKLIST_EXT(
         OperatingSystem::Linux, ScreenSizeStatus::All, BatteryStatus::All,
-- 
2.32.0

