From bb8fedf1391d759e787207f5b5f2a81b289b4772 Mon Sep 17 00:00:00 2001
From: Thomas Deutschmann <whissi@gentoo.org>
Date: Mon, 12 Jul 2021 18:03:43 +0200
Subject: [PATCH 35/36] bmo#1715254: Deny clone3 to force glibc fallback

Signed-off-by: Thomas Deutschmann <whissi@gentoo.org>
---
 security/sandbox/linux/SandboxFilter.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/security/sandbox/linux/SandboxFilter.cpp b/security/sandbox/linux/SandboxFilter.cpp
index 73a6e7fd72..f4e66fc148 100644
--- a/security/sandbox/linux/SandboxFilter.cpp
+++ b/security/sandbox/linux/SandboxFilter.cpp
@@ -835,6 +835,9 @@ class SandboxPolicyCommon : public SandboxPolicyBase {
       case __NR_clone:
         return ClonePolicy(InvalidSyscall());
 
+      case __NR_clone3:
+        return Error(ENOSYS);
+
         // More thread creation.
 #ifdef __NR_set_robust_list
       case __NR_set_robust_list:
@@ -1479,6 +1482,10 @@ class ContentSandboxPolicy : public SandboxPolicyCommon {
         // usually do something reasonable on error.
       case __NR_clone:
         return ClonePolicy(Error(EPERM));
+
+      case __NR_clone3:
+        return Error(ENOSYS);
+
 #  ifdef __NR_fork
       case __NR_fork:
         return Error(ENOSYS);
-- 
2.32.0

