diff -up firefox-102.0/widget/gtk/DMABufLibWrapper.cpp.vaapi firefox-102.0/widget/gtk/DMABufLibWrapper.cpp
--- firefox-102.0/widget/gtk/DMABufLibWrapper.cpp.vaapi	2022-07-01 09:49:50.215536872 +0200
+++ firefox-102.0/widget/gtk/DMABufLibWrapper.cpp	2022-07-01 09:50:43.036306610 +0200
@@ -283,10 +283,11 @@ bool nsDMABufDevice::IsDMABufTexturesEna
 bool nsDMABufDevice::IsDMABufVAAPIEnabled() {
   LOGDMABUF(
       ("nsDMABufDevice::IsDMABufVAAPIEnabled: EGL %d "
-       "media_ffmpeg_vaapi_enabled %d CanUseHardwareVideoDecoding %d "
+       "CanUseHardwareVideoDecoding %d "
        "XRE_IsRDDProcess %d\n",
-       gfx::gfxVars::UseEGL(), StaticPrefs::media_ffmpeg_vaapi_enabled(),
-       gfx::gfxVars::CanUseHardwareVideoDecoding(), XRE_IsRDDProcess()));
+       gfx::gfxVars::UseEGL(),
+       gfx::gfxVars::CanUseHardwareVideoDecoding(),
+       XRE_IsRDDProcess()));
   return gfx::gfxVars::UseVAAPI() && XRE_IsRDDProcess() &&
          gfx::gfxVars::CanUseHardwareVideoDecoding();
 }
