--- firefox-48.0/build/moz.configure/old.configure.orig	2016-08-02 18:54:01.628898690 -0400
+++ firefox-48.0/build/moz.configure/old.configure	2016-08-02 18:59:02.439491374 -0400
@@ -154,6 +154,16 @@
 
 
 @old_configure_options(
+    '--disable-installer',
+    '--mandir',
+    '--enable-mathml',
+    '--enable-pango',
+    '--enable-xft',
+    '--disable-freetype2',
+    '--enable-svg',
+    '--with-system-vpx',
+    '--enable-gstreamer',
+    '--enable-system-libpng',
     '--cache-file',
     '--enable-accessibility',
     '--enable-address-sanitizer',
