From f96b94d03126252506d9f3e7bed7cb09fab37954 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <Florian.Franzmann@fau.de>
Date: Wed, 5 Aug 2015 15:34:50 +0200
Subject: [PATCH] do not use dpkg-architecture

---
 defn2c.pl   | 2 +-
 defn2man.pl | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/defn2c.pl b/defn2c.pl
index c449de2..bf19dcb 100755
--- a/defn2c.pl
+++ b/defn2c.pl
@@ -2,7 +2,7 @@
 
 use strict;
 
-my $DEB_HOST_ARCH_OS = `dpkg-architecture -qDEB_HOST_ARCH_OS`;
+my $DEB_HOST_ARCH_OS = 'linux';
 
 $DEB_HOST_ARCH_OS =~ s/\n//;
 
diff --git a/defn2man.pl b/defn2man.pl
index 6ddcfdd..ae897a7 100755
--- a/defn2man.pl
+++ b/defn2man.pl
@@ -2,7 +2,7 @@
 
 use strict;
 
-my $DEB_HOST_ARCH_OS = `dpkg-architecture -qDEB_HOST_ARCH_OS`;
+my $DEB_HOST_ARCH_OS = 'linux';
 
 $DEB_HOST_ARCH_OS =~ s/\n//;
 
-- 
2.5.0

