From 39827381ab56ffbfce4acda04b362c1c73319477 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 30 Mar 2018 16:04:46 +0200
Subject: [PATCH 1/2] do not use dpkg architecture

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0ce2fa3..739aef2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION ?= 0.8
 CFLAGS ?= -Wall -W -Wno-unused-parameter -g -O2
-ARCH := $(shell dpkg-architecture -qDEB_HOST_ARCH_OS)
+ARCH := linux
 
 BASEDIR ?= $(DESTDIR)
 
-- 
2.16.3

