--- ifupdown/Makefile.orig	2016-09-26 20:25:32.050091106 -0400
+++ ifupdown/Makefile	2016-09-26 20:27:58.532721317 -0400
@@ -46,7 +46,7 @@
 ifquery: ifup
 	ln -sf ifup ifquery
 
-ARCH := $(shell dpkg-architecture -qDEB_HOST_ARCH_OS)
+ARCH := 'linux'
 check: ifup ifdown
 	@echo running ./tests/testbuild-$(ARCH)
 	@if ! exec ./tests/testbuild-$(ARCH); then \
diff -Naur ifupdown.orig/defn2c.pl ifupdown/defn2c.pl
--- ifupdown.orig/defn2c.pl	2015-09-26 09:58:35.000000000 -0700
+++ ifupdown/defn2c.pl	2016-03-08 13:45:40.085021187 -0800
@@ -2,7 +2,7 @@
 
 use strict;
 
-my $DEB_HOST_ARCH_OS = `dpkg-architecture -qDEB_HOST_ARCH_OS`;
+my $DEB_HOST_ARCH_OS = 'linux';
 
 $DEB_HOST_ARCH_OS =~ s/\n//;
 
diff -Naur ifupdown.orig/defn2man.pl ifupdown/defn2man.pl
--- ifupdown.orig/defn2man.pl	2015-06-02 00:32:27.000000000 -0700
+++ ifupdown/defn2man.pl	2016-03-08 13:45:40.087021162 -0800
@@ -2,7 +2,7 @@
 
 use strict;
 
-my $DEB_HOST_ARCH_OS = `dpkg-architecture -qDEB_HOST_ARCH_OS`;
+my $DEB_HOST_ARCH_OS = 'linux';
 
 $DEB_HOST_ARCH_OS =~ s/\n//;
 
