From 302af57feb0eab4556fd6a74207fb4f25059c382 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 30 Mar 2018 23:44:34 +0200
Subject: [PATCH] remove dpkg-architecture invokations from perl scripts

---
 defn2c.pl   | 2 +-
 defn2man.pl | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/defn2c.pl b/defn2c.pl
index fa7a02e..0524732 100755
--- a/defn2c.pl
+++ b/defn2c.pl
@@ -2,7 +2,7 @@
 
 use strict;
 
-my $DEB_HOST_ARCH_OS = `dpkg-architecture -qDEB_HOST_ARCH_OS`;
+my $DEB_HOST_ARCH_OS = "linux";
 
 $DEB_HOST_ARCH_OS =~ s/\n//;
 
diff --git a/defn2man.pl b/defn2man.pl
index 6ddcfdd..015e280 100755
--- a/defn2man.pl
+++ b/defn2man.pl
@@ -2,7 +2,7 @@
 
 use strict;
 
-my $DEB_HOST_ARCH_OS = `dpkg-architecture -qDEB_HOST_ARCH_OS`;
+my $DEB_HOST_ARCH_OS = "linux";
 
 $DEB_HOST_ARCH_OS =~ s/\n//;
 
-- 
2.16.3

