#
# From http://bugs.gentoo.org/show_bug.cgi?id=367363
#
diff -Ncr upower-0.9.10_orig/src/up-main.c upower-0.9.10/src/up-main.c
*** upower-0.9.10_orig/src/up-main.c	Fri Apr 29 02:20:10 2011
--- upower-0.9.10/src/up-main.c	Mon May 23 16:04:14 2011
***************
*** 41,47 ****
  #include "up-kbd-backlight.h"
  #include "up-wakeups.h"
  
! #if GLIB_CHECK_VERSION(2,28,7)
   #include <glib-unix.h>
  #endif
  
--- 41,47 ----
  #include "up-kbd-backlight.h"
  #include "up-wakeups.h"
  
! #if GLIB_CHECK_VERSION(2,29,4)
   #include <glib-unix.h>
  #endif
  
***************
*** 92,98 ****
  	return ret;
  }
  
! #if GLIB_CHECK_VERSION(2,28,7)
  
  /**
   * up_main_sigint_cb:
--- 92,98 ----
  	return ret;
  }
  
! #if GLIB_CHECK_VERSION(2,29,4)
  
  /**
   * up_main_sigint_cb:
***************
*** 201,207 ****
  		goto out;
  	}
  
! #if GLIB_CHECK_VERSION(2,28,7)
  	/* do stuff on ctrl-c */
  	g_unix_signal_add_watch_full (SIGINT,
  				      G_PRIORITY_DEFAULT,
--- 201,207 ----
  		goto out;
  	}
  
! #if GLIB_CHECK_VERSION(2,29,4)
  	/* do stuff on ctrl-c */
  	g_unix_signal_add_watch_full (SIGINT,
  				      G_PRIORITY_DEFAULT,
