From cce5ff737c67ed319b7d146c117ddeaa04969257 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Tue, 10 Jan 2017 10:35:52 +0100
Subject: [PATCH] build ping

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3a4198e..b7de60e 100644
--- a/Makefile
+++ b/Makefile
@@ -116,7 +116,7 @@ endif
 endif
 
 # -------------------------------------
-TARGETS=ping tracepath traceroute6 clockdiff rdisc arping tftpd rarpd
+TARGETS=ping
 
 LDLIBS=$(LDLIB) $(ADDLIB)
 
-- 
2.11.0

