diff --git a/Source/WTF/wtf/unicode/icu/UnicodeIcu.h b/Source/WTF/wtf/unicode/icu/UnicodeIcu.h
index 4867f65..6409a6d 100644
--- a/Source/WTF/wtf/unicode/icu/UnicodeIcu.h
+++ b/Source/WTF/wtf/unicode/icu/UnicodeIcu.h
@@ -31,6 +31,11 @@
 #include <unicode/ustring.h>
 #include <unicode/utf16.h>
 
+#include <unicode/uvernum.h>
+#if U_ICU_VERSION_MAJOR_NUM>=51
+#include <unicode/utf_old.h>
+#endif
+
 namespace WTF {
 
 namespace Unicode {

