From 0da7ecca9917449e28510ffefc721e1dc6f13be4 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 15 Jan 2017 00:02:07 +0100
Subject: [PATCH] add missing include of Xutil.h

---
 xbmc/windowing/X11/GLContextEGL.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/windowing/X11/GLContextEGL.h b/xbmc/windowing/X11/GLContextEGL.h
index 87b5755..9a8c35d 100644
--- a/xbmc/windowing/X11/GLContextEGL.h
+++ b/xbmc/windowing/X11/GLContextEGL.h
@@ -23,6 +23,7 @@
 #if defined(HAVE_X11)
 #include "GLContext.h"
 #include "EGL/egl.h"
+#include "X11/Xutil.h"
 
 class CGLContextEGL : public CGLContext
 {
-- 
2.11.0

