diff -Nuar xbmc-17.4-Krypton.orig/configure.ac xbmc-17.4-Krypton/configure.ac
--- xbmc-17.4-Krypton.orig/configure.ac	2017-08-24 12:59:10.390335315 -0400
+++ xbmc-17.4-Krypton/configure.ac	2017-08-24 13:02:57.363693138 -0400
@@ -1063,12 +1063,12 @@
 AC_MSG_CHECKING("for libcrossguid")
 AC_LANG_PUSH([C++])
 SAVED_LIBS="$LIBS"
-LIBS="$LIBS -lcrossguid $UUID_LIBS"
+LIBS="$LIBS -lxg $UUID_LIBS"
 AC_LINK_IFELSE(
   [AC_LANG_PROGRAM(
-    [#include <guid.h>],
-    [Guid dummy])],
-  [AC_MSG_RESULT(yes); SAVED_LIBS="$SAVED_LIBS $LDFLAGS -lcrossguid $UUID_LIBS"],
+    [#include <Guid.hpp>],
+    [xg::Guid dummy])],
+  [AC_MSG_RESULT(yes); SAVED_LIBS="$SAVED_LIBS $LDFLAGS -lxg $UUID_LIBS"],
   [AC_MSG_ERROR(crossguid not found)])
 LIBS="$SAVED_LIBS"
 AC_LANG_POP([C++])
diff -Nuar xbmc-17.4-Krypton.orig/xbmc/utils/StringUtils.cpp xbmc-17.4-Krypton/xbmc/utils/StringUtils.cpp
--- xbmc-17.4-Krypton.orig/xbmc/utils/StringUtils.cpp	2017-08-24 12:59:16.508345154 -0400
+++ xbmc-17.4-Krypton/xbmc/utils/StringUtils.cpp	2017-08-24 13:12:39.486505409 -0400
@@ -28,7 +28,7 @@
 //
 //------------------------------------------------------------------------
 
-#include <guid.h>
+#include <Guid.hpp>
 
 #include "StringUtils.h"
 #include "CharsetConverter.h"
@@ -1191,8 +1191,7 @@
 
 std::string StringUtils::CreateUUID()
 {
-  static GuidGenerator guidGenerator;
-  auto guid = guidGenerator.newGuid();
+  auto guid = xg::newGuid();
 
   std::stringstream strGuid; strGuid << guid;
   return strGuid.str();
