diff -Nuar xbmc-17.4-Krypton.orig/configure.ac xbmc-17.4-Krypton/configure.ac
--- xbmc-17.4-Krypton.orig/configure.ac	2017-08-24 12:59:10.390335315 -0400
+++ xbmc-17.4-Krypton/configure.ac	2017-08-24 13:02:57.363693138 -0400
@@ -1063,12 +1063,12 @@
 AC_MSG_CHECKING("for libcrossguid")
 AC_LANG_PUSH([C++])
 SAVED_LIBS="$LIBS"
-LIBS="$LIBS -lcrossguid $UUID_LIBS"
+LIBS="$LIBS -lxg $UUID_LIBS"
 AC_LINK_IFELSE(
   [AC_LANG_PROGRAM(
-    [#include <guid.h>],
-    [Guid dummy])],
-  [AC_MSG_RESULT(yes); SAVED_LIBS="$SAVED_LIBS $LDFLAGS -lcrossguid $UUID_LIBS"],
+    [#include <Guid.hpp>],
+    [xg::Guid dummy])],
+  [AC_MSG_RESULT(yes); SAVED_LIBS="$SAVED_LIBS $LDFLAGS -lxg $UUID_LIBS"],
   [AC_MSG_ERROR(crossguid not found)])
 LIBS="$SAVED_LIBS"
 AC_LANG_POP([C++])
