diff -urp barrier-2.4.0.orig/src/lib/base/Event.cpp barrier-2.4.0/src/lib/base/Event.cpp
--- barrier-2.4.0.orig/src/lib/base/Event.cpp	2021-11-01 15:46:43.000000000 -0500
+++ barrier-2.4.0/src/lib/base/Event.cpp	2022-12-12 22:24:16.204896613 -0600
@@ -19,6 +19,8 @@
 #include "base/Event.h"
 #include "base/EventQueue.h"
 
+#include <cstddef>
+
 //
 // Event
 //
diff -urp barrier-2.4.0.orig/src/lib/base/Event.h barrier-2.4.0/src/lib/base/Event.h
--- barrier-2.4.0.orig/src/lib/base/Event.h	2021-11-01 15:46:43.000000000 -0500
+++ barrier-2.4.0/src/lib/base/Event.h	2022-12-12 22:24:33.393895666 -0600
@@ -21,6 +21,8 @@
 #include "common/basic_types.h"
 #include "common/stdmap.h"
 
+#include <cstddef>
+
 class EventData {
 public:
     EventData() { }
