From 964c7c97e27b1e12623d819a68a5025844ee5bc4 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 21 Mar 2014 14:24:45 +0100
Subject: [PATCH] make PREFIX configurable from environment

---
 source/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/Makefile b/source/Makefile
index 1d10985..6029e95 100644
--- a/source/Makefile
+++ b/source/Makefile
@@ -18,7 +18,7 @@
 # ----------------------------------------------------------------------------
 
 
-PREFIX = /usr/local
+PREFIX ?= /usr/local
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 VERSION = 0.4.0
 BINDIR = $(PREFIX)/bin
-- 
1.9.0

