From ac9b3847405574dfd6022c1a9347a4449e8c1510 Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Sun, 23 Jun 2019 02:14:23 +0200
Subject: [PATCH 04/14] Use UTF-8 encoding and include bzip.css as link for
 HTML manual.

---
 bz-html.xsl | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/bz-html.xsl b/bz-html.xsl
index 1785fff..b6000d5 100644
--- a/bz-html.xsl
+++ b/bz-html.xsl
@@ -7,11 +7,14 @@
 <xsl:import href="http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl"/>
 <xsl:import href="bz-common.xsl"/>
 
-<!-- use 8859-1 encoding -->
-<xsl:output method="html" encoding="ISO-8859-1" indent="yes"/>
+<!-- use UTF-8 encoding -->
+<xsl:output method="html" encoding="UTF-8" indent="yes"/>
 
-<!-- we include the css directly when generating one large file -->
-<xsl:template name="user.head.content">  
+<!-- we include the css as link and directly when generating one large file -->
+<xsl:template name="user.head.content">
+  <xsl:text disable-output-escaping="yes">
+<![CDATA[<]]>link rel="stylesheet" type="text/css" href="bzip.css" />
+  </xsl:text>
   <style type="text/css" media="screen">
     <xsl:text>&bz-css;</xsl:text>
   </style>
-- 
2.22.0

