--- configure.ac	2005-10-18 03:59:29.000000000 +0500
+++ configure.ac.fixed	2005-11-16 01:36:00.000000000 +0500
@@ -53,13 +53,13 @@
 fi
 
 # Append ***Magick's lists of flags to the user's lists (if any)
-LDFLAGS="${LDFLAGS} `${MAGICKCONFIG} --ldflags`"
+LDFLAGS="${LDFLAGS} `${MAGICKCONFIG} --ldflags | sed 's:-L/usr/lib/graphviz /::'`"
 CPPFLAGS="${CPPFLAGS} `${MAGICKCONFIG} --cppflags`"
 
 AC_SUBST(LDFLAGS, ${LDFLAGS})
 AC_SUBST(CFLAGS, ${rm_saved_CFLAGS})
 AC_SUBST(CPPFLAGS, ${CPPFLAGS})
-AC_SUBST(LOCAL_LIBS, `${MAGICKCONFIG} --libs`)
+AC_SUBST(LOCAL_LIBS, `${MAGICKCONFIG} --libs | sed 's:-lgvc ::'`)
 
 AC_CHECK_LIB($MAGICKLIB, InitializeMagick, [],
     [AC_MSG_ERROR([Can't install RMagick. Can't find lib${MAGICKLIB} or one of the dependent libraries.
@@ -79,7 +79,7 @@
 AC_CHECK_FUNCS([snprintf])
 
 # Append ***Magick's lists of libs to the user's list (if any)
-LIBS="${LIBS} `${MAGICKCONFIG} --libs`"
+LIBS="${LIBS} `${MAGICKCONFIG} --libs | sed 's:-lgvc ::'`"
 
 # What ImageMagick/GraphicsMagick functions?
 AC_CHECK_FUNCS([AdaptiveThresholdImage AppendImageToList LevelImageChannel])
