diff -ur src/mozilla/xulrunner.c src/mozilla/xulrunner.c
--- src/mozilla/xulrunner.c	2008-11-13 07:44:29.731856409 -0500
+++ src/mozilla/xulrunner.c	2008-11-13 07:44:46.887913149 -0500
@@ -30,7 +30,7 @@
 
 static void xulrunner_init(void) {
 	
-#ifndef XPCOM_CLUE
+#ifndef XPCOM_GLUE
 	/* gtk_moz_embed_set_comp_path() must not be called
 	   without a component path for XulRunner. Passing
 	   NULL will crash Liferea. */

