--- configure~	2006-04-22 10:27:55.000000000 +0200
+++ configure	2006-04-24 14:39:35.000000000 +0200
@@ -21942,8 +21942,8 @@
    echo "$as_me:$LINENO: result: $gecko_provider: $enable_mozilla" >&5
 echo "${ECHO_T}$gecko_provider: $enable_mozilla" >&6
 fi
-if test "x$enable_mozilla" != "xyes" -a \( "x$enable_gecko" = "xyes" -o "x$enable_gecko" = "xfirefox" \) ; then
-   gecko_provider=firefox
+if test "x$enable_gecko" != "x"; then
+   gecko_provider=$enable_gecko
 
 pkg_failed=no
 echo "$as_me:$LINENO: checking for MOZILLA" >&5
--- src/liferea.in~	2006-04-24 14:39:35.000000000 +0200
+++ src/liferea.in	2006-04-24 14:39:54.000000000 +0200
@@ -27,8 +27,6 @@
 	do
 		path=`echo "$results" | tail -n 1`
 		results=`echo "$results" | grep -v $path`
-		path=`dirname $path` 	# strip comm.jar
-		path=`dirname $path`	# strip chrome
 		run_with_mozilla $path
 		
 	done
@@ -39,10 +37,6 @@
 	export DBUS_SESSION_BUS_ADDRESS
 fi
 
-if [ -n "$MOZILLA_FIVE_HOME" ]; then
-	run_with_mozilla $MOZILLA_FIVE_HOME
-fi
-
 @MOZILLA_LIB_DIR_SEARCH@
 look_for_mozilla "/usr/lib/mozilla*/chrome/comm.jar"
 look_for_mozilla "/usr/lib64/mozilla*/chrome/comm.jar"
