diff -ruN jpgraph-1.9/src/jpgraph.php jpgraph-1.9.new/src/jpgraph.php
--- jpgraph-1.9/src/jpgraph.php	2002-10-26 17:37:10.000000000 +0200
+++ jpgraph-1.9.new/src/jpgraph.php	2002-10-28 04:06:26.000000000 +0100
@@ -19,7 +19,7 @@
 // define (furter down) is false. If you enable the cache please note that
 // this directory MUST be readable and writable for the process running PHP. 
 // Must end with '/'
-DEFINE("CACHE_DIR","/tmp/jpgraph_cache/");
+DEFINE("CACHE_DIR","/tmp/jpgraph/");
 
 // Directory for jpGraph TTF fonts. Must end with '/'
 // Note: The fonts must follow the naming conventions as
@@ -126,7 +126,7 @@
 // Please note that the Apache user must be a member of the
 // specified group since otherwise it is impossible for Apache
 // to set the specified group.
-DEFINE("CACHE_FILE_GROUP","wwwadmin");
+DEFINE("CACHE_FILE_GROUP","nogroup");
 
 // What permissions should the cached file have
 // (Set to "" will give the default persmissions for the "PHP-user")
