diff -uNr dazukofs-3.1.3-rc1/dentry.c dazukofs-3.1.3-rc1.cp//dentry.c
--- dazukofs-3.1.3-rc1/dentry.c	2009-06-28 22:16:38.000000000 +0200
+++ dazukofs-3.1.3-rc1.cp//dentry.c	2010-03-09 19:26:29.000000000 +0100
@@ -94,7 +94,7 @@
 	/* update the inode, even if d_revalidate() != 1 */
 	if (dentry->d_inode) {
 		struct inode *lower_inode = get_lower_inode(dentry->d_inode);
-		fsstack_copy_attr_all(dentry->d_inode, lower_inode, NULL);
+		fsstack_copy_attr_all(dentry->d_inode, lower_inode);
 	}
 	return valid;
 }
diff -uNr dazukofs-3.1.3-rc1/inode.c dazukofs-3.1.3-rc1.cp//inode.c
--- dazukofs-3.1.3-rc1/inode.c	2009-06-28 22:17:09.000000000 +0200
+++ dazukofs-3.1.3-rc1.cp//inode.c	2010-03-09 19:21:40.000000000 +0100
@@ -129,7 +129,7 @@
 	else
 		d_instantiate(dentry, inode);
 
-	fsstack_copy_attr_all(inode, lower_inode, NULL);
+	fsstack_copy_attr_all(inode, lower_inode);
 	fsstack_copy_inode_size(inode, lower_inode);
 	return 0;
 }
@@ -487,7 +487,7 @@
 	err = notify_change(lower_dentry, ia);
 	mutex_unlock(&lower_inode->i_mutex);
 
-	fsstack_copy_attr_all(inode, lower_inode, NULL);
+	fsstack_copy_attr_all(inode, lower_inode);
 	fsstack_copy_inode_size(inode, lower_inode);
 	return err;
 }
@@ -520,7 +520,7 @@
 						 size, flags);
 	mutex_unlock(&lower_dentry_inode->i_mutex);
 
-	fsstack_copy_attr_all(dentry->d_inode, lower_dentry_inode, NULL);
+	fsstack_copy_attr_all(dentry->d_inode, lower_dentry_inode);
 	fsstack_copy_inode_size(dentry->d_inode, lower_dentry_inode);
 out:
 	return err;
@@ -737,10 +737,9 @@
 	if (err)
 		goto out;
 
-	fsstack_copy_attr_all(new_dir, lower_new_dentry_parent_inode, NULL);
+	fsstack_copy_attr_all(new_dir, lower_new_dentry_parent_inode);
 	if (new_dir != old_dir)
-		fsstack_copy_attr_all(old_dir, lower_old_dentry_parent_inode,
-				      NULL);
+		fsstack_copy_attr_all(old_dir, lower_old_dentry_parent_inode);
 out:
 	dput(lower_old_dentry_parent);
 	dput(lower_new_dentry_parent);
