diff -aur nautilus-sendto-0.4-orig/src/plugins/evolution.c nautilus-sendto-0.4/src/plugins/evolution.c
--- nautilus-sendto-0.4-orig/src/plugins/evolution.c	2005-07-31 14:34:52.000000000 -0700
+++ nautilus-sendto-0.4/src/plugins/evolution.c	2005-10-28 21:54:13.000000000 -0700
@@ -87,7 +87,7 @@
 			gchar *email = e->data;
 			
 			hash_value = g_strdup_printf ("mailto:%s",email);
-			if (strlen (family_name)==0){
+			if (family_name != NULL && strlen (family_name)==0){
 				/* Output : name <email> */
 				str = g_string_new("");				
 				g_string_printf (str, "%s <%s>", given_name, email);
