--- GNUmakefile.in.orig	2012-03-27 20:31:16.000000000 +0400
+++ GNUmakefile.in	2012-04-15 10:40:36.226272195 +0400
@@ -74045,7 +74045,7 @@
 @ENABLE_WEBKIT2_TRUE@	      $${installdir}/webkit2gtk-$(DOC_MODULE_VERSION).devhelp2; \
 @ENABLE_WEBKIT2_TRUE@	  fi; \
 @ENABLE_WEBKIT2_TRUE@	fi
-	@$(AM_V_GEN)$(PYTHON) $(srcdir)/Tools/gtk/generate-gtkdoc --rebase --virtual-root=$${DESTDIR}
+@ENABLE_GTK_DOC_TRUE@	@$(AM_V_GEN)$(PYTHON) $(srcdir)/Tools/gtk/generate-gtkdoc --rebase --virtual-root=$${DESTDIR}
 
 uninstall-local:
 	@DOC_MODULE_VERSION=`cat ./Documentation/webkitgtk/version.xml`; \
