--- Source/WebCore/platform/graphics/OpenGLShims.h.orig
+++ Source/WebCore/platform/graphics/OpenGLShims.h
@@ -56,6 +56,8 @@
 typedef void (*glBufferSubDataType) (GLenum, GLintptr, GLsizeiptr, const GLvoid*);
 typedef GLenum (*glCheckFramebufferStatusType) (GLenum);
 typedef void (*glCompileShaderType) (GLuint);
+typedef void (*glCompressedTexImage2DType) (GLenum, GLint, GLenum, GLsizei, GLsizei, GLint, GLsizei, const GLvoid *);
+typedef void (*glCompressedTexSubImage2DType) (GLenum, GLint, GLint, GLint, GLsizei, GLsizei, GLenum, GLsizei, const GLvoid *);
 typedef GLuint (*glCreateProgramType) ();
 typedef GLuint (*glCreateShaderType) (GLenum);
 typedef void (*glDeleteBuffersType) (GLsizei, const GLuint*);
@@ -152,6 +154,8 @@
     FUNCTION_TABLE_ENTRY(glBufferSubData);
     FUNCTION_TABLE_ENTRY(glCheckFramebufferStatus);
     FUNCTION_TABLE_ENTRY(glCompileShader);
+    FUNCTION_TABLE_ENTRY(glCompressedTexImage2D);
+    FUNCTION_TABLE_ENTRY(glCompressedTexSubImage2D);
     FUNCTION_TABLE_ENTRY(glCreateProgram);
     FUNCTION_TABLE_ENTRY(glCreateShader);
     FUNCTION_TABLE_ENTRY(glDeleteBuffers);
@@ -253,6 +257,8 @@
 #define glCheckFramebufferStatusEXT            glCheckFramebufferStatus
 #define glCheckFramebufferStatus               LOOKUP_GL_FUNCTION(glCheckFramebufferStatus)
 #define glCompileShader                        LOOKUP_GL_FUNCTION(glCompileShader)
+#define glCompressedTexImage2D                 LOOKUP_GL_FUNCTION(glCompressedTexImage2D)
+#define glCompressedTexSubImage2D              LOOKUP_GL_FUNCTION(glCompressedTexSubImage2D)
 #define glCreateProgram                        LOOKUP_GL_FUNCTION(glCreateProgram)
 #define glCreateShader                         LOOKUP_GL_FUNCTION(glCreateShader)
 #define glDeleteBuffers                        LOOKUP_GL_FUNCTION(glDeleteBuffers)
