--- Source/WebCore/make-hash-tools.pl.orig	2011-04-25 23:27:15.000000000 +0400
+++ Source/WebCore/make-hash-tools.pl	2011-05-29 12:18:23.501169110 +0400
@@ -20,7 +20,6 @@
 #   Boston, MA 02110-1301, USA.
 
 use strict;
-use Switch;
 use File::Basename;
 
 my $outdir = $ARGV[0];
@@ -28,9 +27,7 @@
 my $option = basename($ARGV[0],".gperf");
 
 
-switch ($option) {
-
-case "DocTypeStrings" {
+if ($option eq "DocTypeStrings") {
 
     my $docTypeStringsGenerated    = "$outdir/DocTypeStrings.cpp";
     my $docTypeStringsGperf        = $ARGV[0];
@@ -38,9 +35,7 @@
 
     system("gperf --key-positions=\"*\" -s 2 $docTypeStringsGperf > $docTypeStringsGenerated") == 0 || die "calling gperf failed: $?";
 
-} # case "DocTypeStrings"
-
-case "ColorData" {
+} elsif ($option eq "ColorData") {
 
     my $colorDataGenerated         = "$outdir/ColorData.cpp";
     my $colorDataGperf             = $ARGV[0];
@@ -48,6 +43,4 @@
 
     system("gperf --key-positions=\"*\" -D -s 2 $colorDataGperf > $colorDataGenerated") == 0 || die "calling gperf failed: $?";
 
-} # case "ColorData"
-
-} # switch ($option)
+}
