--- ./cmake/modules/FindSVN.cmake.orig	2008-09-26 13:35:20.000000000 +0200
+++ ./cmake/modules/FindSVN.cmake	2008-09-26 13:40:02.000000000 +0200
@@ -17,6 +17,9 @@
 FIND_PROGRAM(APRCONFIG_EXECUTABLE NAMES apr-1-config apr-config PATHS
    /usr/local/apr/bin
 )
+FIND_PROGRAM(APUCONFIG_EXECUTABLE NAMES apu-1-config apu-config PATHS
+   /usr/local/apr/bin
+)
 
 if(SVNCONFIG_EXECUTABLE)
 
@@ -72,6 +75,11 @@
          set(SVN_FOUND FALSE) # no apr == can't compile!
       endif(_INCLUDES)
    endif(APRCONFIG_EXECUTABLE)
+   if(APUCONFIG_EXECUTABLE)
+      EXEC_PROGRAM(${APUCONFIG_EXECUTABLE} ARGS --includes RETURN_VALUE _return_VALUE OUTPUT_VARIABLE _INCLUDES)
+      string(REPLACE "-I" "" _INCLUDES ${_INCLUDES})
+      set(SVN_INCLUDES ${SVN_INCLUDES} ${_INCLUDES})
+   endif(APUCONFIG_EXECUTABLE)
    FIND_LIBRARY(SVN_LIBRARIES NAMES svn_client-1)
    if(SVN_LIBRARIES)
       FIND_LIBRARY(_LIBRARIES NAMES svn_subr-1)

