--- Makefile-old	2004-02-09 15:18:25.000000000 +0100
+++ Makefile	2004-02-09 15:20:16.000000000 +0100
@@ -1,6 +1,5 @@
 CC = gcc
 NASM = nasm
-USE_MMX = 0
 USE_CSS = 1
 A52DIR := $(shell expr a52dec* )
 
@@ -16,6 +15,11 @@
 OBJDIR := $(shell uname --machine)
 
 
+ifneq ($(OBJDIR), alpha)
+  ifneq ($(OBJDIR), x86_64)
+    CFLAGS += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
+  endif
+endif
 
 
 ifeq ($(OBJDIR), alpha)
@@ -25,14 +29,6 @@
   endif
 endif
 
-ifeq ($(OBJDIR), i686)
-  USE_MMX = 1
-  ifneq ($(HAVE_CFLAGS), y)
-    CFLAGS := -O2 -fomit-frame-pointer -falign-loops=2 -falign-jumps=2 -falign-functions=2 -I/usr/local/include
-  endif
-  CFLAGS += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-endif
-
 ifeq ($(OBJDIR), x86_64)
   ifneq ($(HAVE_CFLAGS), y)
     CFLAGS := -O2 -fomit-frame-pointer -I/usr/local/include
@@ -61,6 +57,7 @@
 
 CFLAGS += \
 	-I. \
+	-I/usr/include/a52dec \
 	-I$(A52DIR)/include \
 	-I$(A52DIR)/liba52
 
@@ -127,18 +124,13 @@
 	$(OBJDIR)/audio \
 	$(OBJDIR)/video
 
-include Makefile.a52
-
-DIRS += $(A52DIRS)
-
-
-OUTPUT = $(OBJDIR)/libmpeg3.a
+OUTPUT = $(OBJDIR)/libmpeg3.so
 UTILS = $(OBJDIR)/mpeg3dump $(OBJDIR)/mpeg3toc  $(OBJDIR)/mpeg3cat
 
 #$(OBJDIR)/mpeg3split
 
 
-LIBS = -lm -lpthread
+LIBS = -lm -lpthread -L$(shell uname --machine)
 
 $(shell if ! test -d $(OBJDIR) \; then mkdir -p $(OBJDIR) \; fi )
 
@@ -151,18 +143,18 @@
 
 
 $(OUTPUT): $(OBJS) $(ASMOBJS) $(NASMOBJS) $(A52OBJS)
-	ar rcs $(OUTPUT) `cat $(OBJDIR)/objs`
+	ld -shared -o $(OUTPUT) `cat $(OBJDIR)/objs` $(LIBS) -la52
 
 
 
 $(OBJDIR)/mpeg3dump: $(OUTPUT) mpeg3dump.c
-	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3dump mpeg3dump.c $(OUTPUT) $(LIBS)
+	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3dump mpeg3dump.c -lmpeg3 $(LIBS)
 
 $(OBJDIR)/mpeg3toc: $(OUTPUT) mpeg3toc.c
-	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3toc mpeg3toc.c $(OUTPUT) $(LIBS)
+	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3toc mpeg3toc.c -lmpeg3 $(LIBS)
 
 $(OBJDIR)/mpeg3cat: $(OUTPUT) mpeg3cat.c
-	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3cat mpeg3cat.c $(OUTPUT) $(LIBS)
+	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3cat mpeg3cat.c -lmpeg3 $(LIBS)
 
 #$(OBJDIR)/mpeg3split: $(OUTPUT)
 #	$(CC) `cat $(OBJDIR)/c_flags` -o $(OBJDIR)/mpeg3split mpeg3split.c $(OUTPUT) $(LIBS)
@@ -179,7 +171,11 @@
 		-ldl
 
 install: 
+	mkdir -p /usr/include/libmpeg3
+	cp libmpeg3.h mpeg3private.h mpeg3protos.h /usr/include/libmpeg3
 	cp $(UTILS) /usr/bin
+	cp $(OUTPUT) /usr/lib
+	ldconfig
 
 clean:
 	rm -rf $(OBJDIR)
