diff --git a/src/webkittrunk.pri b/src/webkittrunk.pri
index 46c8e8f..7e3eb70 100644
--- a/src/webkittrunk.pri
+++ b/src/webkittrunk.pri
@@ -1,19 +1,13 @@
 #
 # export QT_WEBKIT=webkit_trunk
-# export WEBKITDIR=$HOME/dev/webkit
 #
 CONFIG += $$(QT_WEBKIT)
 webkit_trunk {
-    WEBKITDIR = $$(WEBKITDIR)
-    CONFIG(release):WEBKITBUILD = $$WEBKITDIR/WebKitBuild/Release/lib
-    CONFIG(debug):WEBKITBUILD = $$WEBKITDIR/WebKitBuild/Debug/lib
-    message(Using WebKit Trunk at $$WEBKITDIR)
-    message(Using WebKit Build at $$WEBKITBUILD)
     QT -= webkit
     DEFINES += WEBKIT_TRUNK
-    QMAKE_LIBDIR_FLAGS = -L$$WEBKITBUILD
+    QMAKE_LIBDIR_FLAGS = -L/usr/lib
     LIBS = -lQtWebKit
-    INCLUDEPATH = $$WEBKITDIR/WebKit/qt/Api $$INCLUDEPATH
-    QMAKE_RPATHDIR = $$WEBKITBUILD $$QMAKE_RPATHDIR
+    INCLUDEPATH = /usr/include $$INCLUDEPATH
+    QMAKE_RPATHDIR = /usr/lib $$QMAKE_RPATHDIR
 }
 
