--- configure.orig	2006-03-17 09:25:07.000000000 -0600
+++ configure	2006-03-17 09:33:43.000000000 -0600
@@ -42,29 +42,29 @@
 
 # in case of automatic build, default to gtk+-1.x...
 
-X=""
-echo "Build GTKWave for GTK+-1.x or 2.x? Please enter '1' or '2'"
-read X
-
-if [ "$X" = "" ]
-	then X="1"
-	fi
+#X=""
+#echo "Build GTKWave for GTK+-1.x or 2.x? Please enter '1' or '2'"
+#read X
+
+#if [ "$X" = "" ]
+#	then X="1"
+#	fi
 
-if [ "$X" = "1" ]
-	then echo "Building for GTK+-1.x..."
+#if [ "$X" = "1" ]; then
+	echo "Building for GTK+-1.x..."
 	ln -fs Makefile_GTK1.in src/Makefile.in
 	./configure2 "$@"
 	./configure3
 	exit
-	fi
+#	fi
 
-if [ "$X" = "2" ]
-	then echo "Building for GTK+-2.x..."
-	ln -fs Makefile_GTK2.in src/Makefile.in
-	./configure2 "$@"
-	./configure3
-	exit
-	fi
+#if [ "$X" = "2" ]; then 
+#	echo "Building for GTK+-2.x..."
+#	ln -fs Makefile_GTK2.in src/Makefile.in
+#	./configure2 "$@"
+#	./configure3
+#	exit
+#	fi
 
 done
 
