--- Makefile.old	1998-11-19 04:09:38.000000000 -0800
+++ Makefile	2005-06-09 15:53:11.000000000 -0700
@@ -1,22 +1,24 @@
-default: dos2unix
+all:	unix2dos dos2unix link 
 
-all:	dos2unix link install
 
+dos2unix: dos2unix.c
+	$(CC) $(LDFLAGS) -o $@ $^
 
-dos2unix:
-	gcc -O dos2unix.c -o dos2unix
-
+unix2dos: unix2dos.c
+	$(CC) $(LDFLAGS) -o $@ $^
 
 link:	
 	ln -s dos2unix mac2unix
 	ln -s dos2unix.1 mac2unix.1		
 
 clean:
-	rm dos2unix
+	$(RM) dos2unix unix2dos mac2unix mac2unix.1
 
 install:
-	install -m 755 dos2unix /usr/local/bin
-	install -m 644 dos2unix.1 /usr/local/man/man1
-	install -m 755 mac2unix /usr/local/bin
-	install -m 644 mac2unix.1 /usr/local/man/man1
+	install -m 755 dos2unix /usr/bin
+	install -m 644 dos2unix.1 /usr/man/man1
+	install -m 755 unix2dos /usr/bin
+	install -m 644 unix2dos.1 /usr/man/man1
+	install -m 755 mac2unix /usr/bin
+	install -m 644 mac2unix.1 /usr/man/man1
 
