Index: misc/mc.keymap.default
===================================================================
--- misc/mc.keymap.default	(revision 5f0cef1e6e8982cd0503d025d7c08a36f2fece74)
+++ misc/mc.keymap.default	(revision be145b595b23932570dd79d393fc3b68306f74b9)
@@ -162,4 +162,8 @@
 SelectCodepage = alt-e
 
+EditExtMode =
+
+[edit:xmap]
+
 [viewer]
 ViewHelp = f1
@@ -251,5 +255,5 @@
 CmdReverseSelection = kpasterisk
 CmdDialogList = alt-prime
-ExtMap1 =
+ExtMap1 = ctrl-x
 
 [main:xmap]
--- misc/mc.keymap	(revision 5f0cef1e6e8982cd0503d025d7c08a36f2fece74)
+++ misc/mc.keymap	(revision be145b595b23932570dd79d393fc3b68306f74b9)
@@ -162,4 +162,8 @@
 SelectCodepage = alt-e
 
+EditExtMode =
+
+[edit:xmap]
+
 [viewer]
 ViewHelp = f1
@@ -251,5 +255,5 @@
 CmdReverseSelection = kpasterisk
 CmdDialogList = alt-prime
-ExtMap1 =
+ExtMap1 = ctrl-x
 
 [main:xmap]
