--- a/style/subfigure.el
+++ b/style/subfigure.el
@@ -47,8 +47,9 @@
    ;; Install completion for labels:
    (setq TeX-complete-list
 	 (append
-	  '(("\\\\[Ss]ubref{\\([^{}\n\r\\%,]*\\)" 1 LaTeX-label-list "}")))
-	  TeX-complete-list)
+         '(("\\\\subref{\\([^{}\n\r\\%,]*\\)" 1 LaTeX-label-list "}")
+           ("\\\\Subref{\\([^{}\n\r\\%,]*\\)" 1 LaTeX-label-list "}"))
+	  TeX-complete-list))
 
    ;; Fontification
    (when (and (featurep 'font-latex)
