--- ice-9/boot-9.scm	2007/01/13 21:34:01	1.1
+++ ice-9/boot-9.scm	2007/01/13 21:34:22
@@ -1768,9 +1768,9 @@
 	     (let* ((interface-args (cadr kws))
 		    (interface (apply resolve-interface interface-args)))
 	       (and (eq? (car kws) #:use-syntax)
-		    (or (symbol? (car interface-args))
+		    (or (symbol? (caar interface-args))
 			(error "invalid module name for use-syntax"
-			       interface-args))
+			       (car interface-args)))
 		    (set-module-transformer!
 		     module
 		     (module-ref interface (car
