--- Makefile.old	2003-04-22 05:51:38.000000000 +1000
+++ Makefile	2005-12-17 14:09:37.000000000 +1000
@@ -19,23 +19,23 @@
 
 install:
 	if [ -d /usr/lib/gkrellm2/plugins/ ] ; then \
-		install -C -s -m 644 gkrellstock.so /usr/lib/gkrellm2/plugins/ ; \
+		install -s -m 644 gkrellstock.so /usr/lib/gkrellm2/plugins/ ; \
 	elif [ -d /usr/local/lib/gkrellm2/plugins/ ] ; then \
-		install -C -s -m 644 gkrellstock.so /usr/local/lib/gkrellm2/plugins/ ; \
+		install -s -m 644 gkrellstock.so /usr/local/lib/gkrellm2/plugins/ ; \
 	else \
-		install -D -C -s -m 644 gkrellstock.so /usr/lib/gkrellm2/plugins/gkrellstock.so ; \
+		install -D -s -m 644 gkrellstock.so /usr/lib/gkrellm2/plugins/gkrellstock.so ; \
 	fi
-	install -C -m 755 GetQuote2 /usr/X11R6/bin
+	install -m 755 GetQuote2 /usr/X11R6/bin
 userinstall:
 	if [ -d $(HOME)/.gkrellm2/plugins/ ] ; then \
-		install -C -s -m 644 gkrellstock.so $(HOME)/.gkrellm2/plugins/ ; \
+		install -s -m 644 gkrellstock.so $(HOME)/.gkrellm2/plugins/ ; \
 	else \
-		install -D -C -s -m 644 gkrellstock.so $(HOME)/.gkrellm2/plugins/gkrellstock.so ; \
+		install -D -s -m 644 gkrellstock.so $(HOME)/.gkrellm2/plugins/gkrellstock.so ; \
         fi
 	if [ -d $(HOME)/bin/ ] ; then \
-		install -C -m 755 GetQuote2 $(HOME)/bin/ ; \
+		install -m 755 GetQuote2 $(HOME)/bin/ ; \
 	else \
-		install -D -C -m 755 GetQuote2 $(HOME)/bin/GetQuote2 ; \
+		install -D -m 755 GetQuote2 $(HOME)/bin/GetQuote2 ; \
         fi
 
 uninstall:
