From 3f6879b463d1bdc352375d88a712965e01837cab Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 19 Mar 2014 15:59:31 +0000
Subject: [PATCH] honour external FLAGS

---
 src/makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/makefile b/src/makefile
index 8a0f795..6194e62 100644
--- a/src/makefile
+++ b/src/makefile
@@ -11,8 +11,8 @@ INSTALL_PLUGINS_DIR	=	/usr/lib/ladspa/
 # GENERAL
 #
 
-CFLAGS		=	$(INCLUDES) -Wall -O3 -fPIC
-CXXFLAGS	=	$(CFLAGS)
+CFLAGS		+=	$(INCLUDES)
+CXXFLAGS	+=	$(CFLAGS)
 PLUGIN_LIB	=	../plugins/cmt.so
 
 ###############################################################################
-- 
1.9.0

