--- transcode-1.0.2/import/import_mpeg3.c.orig	2006-09-09 23:39:08.598897708 +0200
+++ transcode-1.0.2/import/import_mpeg3.c	2006-09-09 23:40:36.455540042 +0200
@@ -68,7 +68,7 @@
 MOD_open
 {
 
-  int i;
+  int i,return_value;
 
   param->fd = NULL;
 
@@ -79,14 +79,14 @@
   if (param->flag == TC_VIDEO) {
       if (!file) {
 	  if (!file_a) {
-	      if((file = mpeg3_open(vob->video_in_file))==NULL) {
+	      if((file = mpeg3_open(vob->video_in_file, &return_value))==NULL) {
 		  fprintf(stderr, "open file failed\n");
 		  return(TC_IMPORT_ERROR);
 	      }
 	      if (verbose & TC_DEBUG)
                   printf("[%s] Opened video NO copy\n", MOD_NAME);
 	  } else if (file_a) {
-	      if((file = mpeg3_open_copy(vob->video_in_file, file_a))==NULL) {
+	      if((file = mpeg3_open_copy(vob->video_in_file, file_a, &return_value))==NULL) {
 		  fprintf(stderr, "open file failed\n");
 		  return(TC_IMPORT_ERROR);
 	      }
@@ -98,14 +98,14 @@
   if (param->flag == TC_AUDIO) {
       if (!file_a) {
 	  if (!file) {
-	      if((file_a = mpeg3_open(vob->audio_in_file))==NULL) {
+	      if((file_a = mpeg3_open(vob->audio_in_file, &return_value))==NULL) {
 		  fprintf(stderr, "open audio file failed\n");
 		  return(TC_IMPORT_ERROR);
 	      }
 	      if (verbose & TC_DEBUG)
                   printf("[%s] Opened audio NO copy\n", MOD_NAME);
 	  } else if (file) {
-	      if((file_a = mpeg3_open_copy(vob->audio_in_file, file))==NULL) {
+	      if((file_a = mpeg3_open_copy(vob->audio_in_file, file, &return_value))==NULL) {
 		  fprintf(stderr, "open_copy audio file failed\n");
 		  return(TC_IMPORT_ERROR);
 	      }
