--- CNetworkOfPoses.h.ori	2012-10-24 21:04:24.000000000 +0200
+++ CNetworkOfPoses.h	2012-10-24 21:56:48.000000000 +0200
@@ -49,8 +49,6 @@
 #include <mrpt/utils/traits_map.h>
 #include <mrpt/utils/stl_extensions.h>
 
-#include <mrpt/graphs/link_pragmas.h>
-
 // The main class in this file is CNetworkOfPoses, a generic basic template for
 //  predefined 2D/3D graphs of pose contraints.
 namespace mrpt
