ff -Naur vbetool-1.1.orig/Makefile.am vbetool-1.1/Makefile.am
--- vbetool-1.1.orig/Makefile.am        2008-05-16 14:52:33.000000000 +0200
+++ vbetool-1.1/Makefile.am     2009-03-12 15:00:36.000000000 +0100
@@ -2,7 +2,7 @@

 sbin_PROGRAMS = vbetool

-vbetool_LDADD = $(libdir)/libpci.a
+vbetool_LDADD = $(libdir)/libpci.so

 man_MANS = vbetool.1
 vbetool_SOURCES = vbetool.c $(x86)
diff -Naur vbetool-1.1.orig/Makefile.in vbetool-1.1/Makefile.in
--- vbetool-1.1.orig/Makefile.in        2008-05-16 14:52:38.000000000 +0200
+++ vbetool-1.1/Makefile.in     2009-03-12 15:00:27.000000000 +0100
@@ -50,7 +50,7 @@
 PROGRAMS = $(sbin_PROGRAMS)
 am_vbetool_OBJECTS = vbetool.$(OBJEXT)
 vbetool_OBJECTS = $(am_vbetool_OBJECTS)
-vbetool_DEPENDENCIES = $(libdir)/libpci.a
+vbetool_DEPENDENCIES = $(libdir)/libpci.so
 DEFAULT_INCLUDES = -I.@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -159,7 +159,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign
-vbetool_LDADD = $(libdir)/libpci.a
+vbetool_LDADD = $(libdir)/libpci.so
 man_MANS = vbetool.1
 vbetool_SOURCES = vbetool.c $(x86)
 noinst_HEADERS = vbetool.h
