--- unix/Xvnc/programs/Xserver/hw/vnc/rfbssl_openssl.c.orig	2019-05-22 11:23:51.000000000 -0700
+++ unix/Xvnc/programs/Xserver/hw/vnc/rfbssl_openssl.c	2019-05-23 18:59:11.599311213 -0700
@@ -52,7 +52,7 @@
 #else
 #define CONST
 #endif
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || !defined(LIBRESSL_VERSION)
 typedef struct ossl_init_settings_st OPENSSL_INIT_SETTINGS;
 #endif
 
