--- Makefile.PL~	2005-05-20 07:03:47.000000000 +0200
+++ Makefile.PL	2005-07-14 15:11:19.000000000 +0200
@@ -11,18 +11,6 @@
 
 my($old_modperl_version, $old_modperl_pm, $old_Apache2_pm);
 
-BEGIN {
-    eval {
-        my $old_mp2 = eval { require Apache2 };
-        require mod_perl;
-        if ($mod_perl::VERSION < 1.999_22 && $old_mp2) {
-            $old_modperl_version = $mod_perl::VERSION;
-            $old_modperl_pm = delete $INC{'mod_perl.pm'};
-            $old_Apache2_pm = delete $INC{'Apache2.pm'};
-        }
-    };
-}
-
 use lib qw(lib Apache-Test/lib);
 
 use Config;
