diff -uNr captive-1.1.7.old/src/libcaptive/Makefile.in captive-1.1.7/src/libcaptive/Makefile.in
--- captive-1.1.7.old/src/libcaptive/Makefile.in	2006-06-03 01:27:58.000000000 +0200
+++ captive-1.1.7/src/libcaptive/Makefile.in	2006-06-03 01:28:29.000000000 +0200
@@ -1051,12 +1051,7 @@
 @HAVE_POD2MAN_TRUE@	$(POD2MAN) --section=8 $< >$@
 
 install-exec-hook:
-	if test `id -u` -eq 0 -a -z "$(DESTDIR)";then \
-		if test x$(libdir) != x/lib -a x$(libdir) != x/usr/lib && ! grep -q '^$(libdir)$$' /etc/ld.so.conf;then \
-			echo -n "WARNING: Add '$(libdir)' to your '/etc/ld.so.conf' and run ldconfig(8)! Press ENTER...";read; \
-		fi; \
-		ldconfig || echo -n "WARNING: Failed to run ldconfig(8) - run manually! Press ENTER...";read; \
-	fi
+
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
