--- src/Makefile.am.old	2006-11-27 12:25:08.000000000 -0800
+++ src/Makefile.am	2006-11-27 12:24:47.000000000 -0800
@@ -13,8 +13,9 @@
 BUILT_SOURCES = notificationdaemon-dbus-glue.h
 
 notificationdaemon-dbus-glue.h: notificationdaemon.xml
-	dbus-binding-tool --mode=glib-server \
-		$(srcdir)/notificationdaemon.xml > notificationdaemon-dbus-glue.h
+	dbus-binding-tool --prefix= --mode=glib-server \
+		--output=$(srcdir)/notificationdaemon-dbus-glue.h \
+		$(srcdir)/notificationdaemon.xml
 
 INCLUDES = \
 	-I$(top_srcdir) \
