--- asio-1.4.1/include/asio/ssl/detail/openssl_context_service.hpp.orig	2010-04-23 20:57:39.935389413 +0200
+++ asio-1.4.1/include/asio/ssl/detail/openssl_context_service.hpp	2010-04-23 21:00:04.978889835 +0200
@@ -70,40 +70,40 @@
     switch (m)
     {
     case context_base::sslv2:
-      ssl_method = ::SSLv2_method();
+      ssl_method = const_cast<SSL_METHOD *>(::SSLv2_method());
       break;
     case context_base::sslv2_client:
-      ssl_method = ::SSLv2_client_method();
+      ssl_method = const_cast<SSL_METHOD *>(::SSLv2_client_method());
       break;
     case context_base::sslv2_server:
-      ssl_method = ::SSLv2_server_method();
+      ssl_method = const_cast<SSL_METHOD *>(::SSLv2_server_method());
       break;
     case context_base::sslv3:
-      ssl_method = ::SSLv3_method();
+      ssl_method = const_cast<SSL_METHOD *>(::SSLv3_method());
       break;
     case context_base::sslv3_client:
-      ssl_method = ::SSLv3_client_method();
+      ssl_method = const_cast<SSL_METHOD *>(::SSLv3_client_method());
       break;
     case context_base::sslv3_server:
-      ssl_method = ::SSLv3_server_method();
+      ssl_method = const_cast<SSL_METHOD *>(::SSLv3_server_method());
       break;
     case context_base::tlsv1:
-      ssl_method = ::TLSv1_method();
+      ssl_method = const_cast<SSL_METHOD *>(::TLSv1_method());
       break;
     case context_base::tlsv1_client:
-      ssl_method = ::TLSv1_client_method();
+      ssl_method = const_cast<SSL_METHOD *>(::TLSv1_client_method());
       break;
     case context_base::tlsv1_server:
-      ssl_method = ::TLSv1_server_method();
+      ssl_method = const_cast<SSL_METHOD *>(::TLSv1_server_method());
       break;
     case context_base::sslv23:
-      ssl_method = ::SSLv23_method();
+      ssl_method = const_cast<SSL_METHOD *>(::SSLv23_method());
       break;
     case context_base::sslv23_client:
-      ssl_method = ::SSLv23_client_method();
+      ssl_method = const_cast<SSL_METHOD *>(::SSLv23_client_method());
       break;
     case context_base::sslv23_server:
-      ssl_method = ::SSLv23_server_method();
+      ssl_method = const_cast<SSL_METHOD *>(::SSLv23_server_method());
       break;
     default:
       break;
