From 749b18a14e28a098e54790a6bdef0b102d0f4567 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 26 Jan 2018 21:22:39 +0100
Subject: [PATCH] fix dependency on entropy

---
 nonce.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nonce.cabal b/nonce.cabal
index cd244ef..dc1f4d9 100644
--- a/nonce.cabal
+++ b/nonce.cabal
@@ -33,7 +33,7 @@ library
     , bytestring        >= 0.9
     , text              >= 0.9
     , transformers      >= 0.2
-    , entropy           >= 0.3.7 && < 0.4
+    , entropy           >= 0.3.7 && < 0.5
     , unliftio
     , unliftio-core
   hs-source-dirs: src/
-- 
2.15.1

