--- dvb-mpegtools/Makefile.orig	2004-04-25 01:30:29.595445456 -0500
+++ dvb-mpegtools/Makefile	2004-04-25 01:34:10.012936904 -0500
@@ -25,7 +25,7 @@
 
 links: dvb-mpegtools_main
 	for f in $(LINKS); do\
-		ln -sf dvb-mpegtools $$f ;\
+		ln -sf dvb-mpegtools_main $$f ;\
 	done
 
 clean:
@@ -61,7 +61,7 @@
 	install -m 755 audiofilter $(PREFIX)/bin/
 	install -m 755 dvb-mpegtools_main $(PREFIX)/bin/
 	install -m 755 dvbaudio $(PREFIX)/bin/
-	install -m 755 dvb_show_image $(PREFIX)/bin/
+	install -m 755 show_dvb_image $(PREFIX)/bin/
 	install -m 755 dia $(PREFIX)/bin/
 	for f in $(LINKS); do\
 		ln -sf $(PREFIX)/bin/dvb-mpegtools_main $(PREFIX)/bin/$$f ;\
