From 0ee63912133943e0837b90e46bec274761d46598 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 9 Jun 2008 01:23:59 -0400
Subject: [PATCH] Pull in stdlib.h for NULL definition to fix #225425

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 lib/ec_hw.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/lib/ec_hw.c b/lib/ec_hw.c
index 825e501..558f26f 100644
--- lib/ec_hw.c
+++ lib/ec_hw.c
@@ -16,6 +16,7 @@
 
 #if HAVE_HWEC
 
+#include <stdlib.h>
 #include <net/if_arp.h>
 #include "net-support.h"
 
-- 
1.5.5.3

