diff -ur gvfs-orig/gconf/gapplookupgconf.h gvfs-mod/gconf/gapplookupgconf.h
--- gvfs-orig/gconf/gapplookupgconf.h	2011-03-11 15:48:20.674000008 -0600
+++ gvfs-mod/gconf/gapplookupgconf.h	2011-03-11 15:52:15.559000006 -0600
@@ -25,8 +25,14 @@
 
 #include <glib-object.h>
 #include <gio/gio.h>
-#include <gio/gdesktopappinfo.h>
 
+#ifdef G_DISABLE_DEPRECATED
+#undef G_DISABLE_DEPRECATED
+#include <gio/gdesktopappinfo.h>
+#define G_DISABLE_DEPRECATED
+#else
+#include <gio/gdesktopappinfo.h>
+#endif
 
 G_BEGIN_DECLS
 
