--- libdwfl/linux-kernel-modules.c.orig	2007-08-15 03:28:27.000000000 +0500
+++ libdwfl/linux-kernel-modules.c	2007-10-01 20:46:01.000000000 +0500
@@ -48,8 +48,6 @@
    <http://www.openinventionnetwork.com>.  */
 
 #include <config.h>
-#undef	_FILE_OFFSET_BITS	/* Doesn't jibe with fts.  */
-
 #include "libdwflP.h"
 #include <inttypes.h>
 #include <errno.h>
@@ -60,6 +58,9 @@
 #include <sys/utsname.h>
 #include <fcntl.h>
 #include <unistd.h>
+#undef  _FILE_OFFSET_BITS       /* Doesn't jibe with fts.  */
+#undef __USE_FILE_OFFSET64
+
 #include <fts.h>
 
 
