--- Makefile.orig	2004-03-19 17:47:49.000000000 +0100
+++ Makefile	2004-03-19 18:48:34.000000000 +0100
@@ -52,7 +52,6 @@
 
 $(RELEASE_DIR).tar.gz: clean
 	@chmod +x sbin/hotplug
-	@chmod +x etc/rc.d/init.d/hotplug
 	@chmod +x etc/hotplug/*.agent etc/hotplug/*.rc
 	@-rm -rf $(RELEASE_DIR)
 	@mkdir $(RELEASE_DIR)
@@ -88,10 +87,13 @@
 	for F in etc/hotplug/{*.{agent,rc},hotplug.functions} ; do \
 	    ${INSTALL_SCRIPT} $$F $(etcdir)/hotplug ; \
 	    done
-	for F in etc/hotplug/{blacklist,usb.{user,hand,dist}map} ; do \
+	cd etc/hotplug ; \
+	for F in {blacklist,usb.{user,hand,dist}map,pnp.{user,hand,dist}map} ; do \
+	    if [ ! -f $(etcdir)/hotplug/$$F ];then \
 	    ${INSTALL_DATA} $$F $(etcdir)/hotplug ; \
-	    done
-	${INSTALL_SCRIPT} -D etc/rc.d/init.d/hotplug $(etcdir)/init.d/hotplug
+	    fi \
+	    done ;\
+	cd ../..
 	$(INSTALL_DATA) -D hotplug.8  $(mandir)/man8/hotplug.8
 	$(INSTALL) -d $(prefix)/var/log/hotplug $(prefix)/var/run
 	$(INSTALL) -d -m 700 $(prefix)/var/run/usb
