diff -Naur faad2.orig/libfaad/common.h faad2/libfaad/common.h
--- faad2.orig/libfaad/common.h	2004-10-22 21:01:54.841786248 +0200
+++ faad2/libfaad/common.h	2004-10-22 21:05:19.927576554 +0200
@@ -80,8 +80,8 @@
 /* Allow decoding of scalable profiles */
 //#define SCALABLE_DEC
 /* Allow decoding of Digital Radio Mondiale (DRM) */
-//#define DRM
-//#define DRM_PS
+#define DRM
+#define DRM_PS
 
 /* LD can't do without LTP */
 #ifdef LD_DEC
