From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@sourcemage.org>
Date: Fri, 13 Jun 2025 15:38:35 +0200
Subject: [PATCH] Make lseek64 available

Origin: Source Mage
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 raw_ostream.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/Support/raw_ostream.cpp b/lib/Support/raw_ostream.cpp
index 651949ad5765..258ad2b79e3e 100644
--- a/lib/Support/raw_ostream.cpp
+++ b/lib/Support/raw_ostream.cpp
@@ -10,6 +10,8 @@
 //
 //===----------------------------------------------------------------------===//
 
+#define _LARGEFILE64_SOURCE
+
 #include "llvm/Support/raw_ostream.h"
 #include "llvm/ADT/STLArrayExtras.h"
 #include "llvm/ADT/StringExtras.h"
