From 83de529d6d22b2e8a890b3d88d7abcad74d7f301 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 14 Jul 2018 20:50:08 +0200
Subject: [PATCH] do not build designer plugin

---
 qwtconfig.pri | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qwtconfig.pri b/qwtconfig.pri
index 2dc3715..4aae30f 100644
--- a/qwtconfig.pri
+++ b/qwtconfig.pri
@@ -118,7 +118,7 @@ QWT_CONFIG     += QwtOpenGL
 # Otherwise you have to build it from the designer directory.
 ######################################################################
 
-QWT_CONFIG     += QwtDesigner
+#QWT_CONFIG     += QwtDesigner
 
 ######################################################################
 # Compile all Qwt classes into the designer plugin instead
-- 
2.18.0

