diff -Nur netkit-telnet-0.17-orig/telnet/externs.h netkit-telnet-0.17/telnet/externs.h
--- netkit-telnet-0.17-orig/telnet/externs.h	1999-08-19 11:34:15.000000000 +0200
+++ netkit-telnet-0.17/telnet/externs.h	2003-09-06 20:40:42.000000000 +0200
@@ -43,6 +43,8 @@
 #include <sys/ioctl.h>
 #include <features.h>
 #include <termios.h>
+#include <stdlib.h>
+#include <string.h>
 
 #if defined(NO_CC_T)
 typedef unsigned char cc_t;
diff -Nur netkit-telnet-0.17-orig/telnet/Makefile netkit-telnet-0.17/telnet/Makefile
--- netkit-telnet-0.17-orig/telnet/Makefile	1999-08-01 07:06:37.000000000 +0200
+++ netkit-telnet-0.17/telnet/Makefile	2003-09-06 20:40:27.000000000 +0200
@@ -7,7 +7,7 @@
 
 # -DAUTHENTICATE
 CXXFLAGS += -DUSE_TERMIO -DKLUDGELINEMODE
-LIBS += $(LIBTERMCAP)
+LIBS += $(LIBTERMCAP) -lstdc++
 
 SRCS = commands.cc main.cc network.cc ring.cc sys_bsd.cc telnet.cc \
 	terminal.cc tn3270.cc utilities.cc genget.cc environ.cc netlink.cc
diff -Nur netkit-telnet-0.17-orig/telnet/netlink.cc netkit-telnet-0.17/telnet/netlink.cc
--- netkit-telnet-0.17-orig/telnet/netlink.cc	2000-07-23 06:16:25.000000000 +0200
+++ netkit-telnet-0.17/telnet/netlink.cc	2003-09-06 20:42:03.000000000 +0200
@@ -8,6 +8,7 @@
 #include <sys/ioctl.h>
 #include <sys/time.h>
 #include <netdb.h>
+#include <string.h>
 #include "netlink.h"
 #include "proto.h"
 #include "ring.h"
diff -Nur netkit-telnet-0.17-orig/telnet/telnet.cc netkit-telnet-0.17/telnet/telnet.cc
--- netkit-telnet-0.17-orig/telnet/telnet.cc	2000-07-23 05:24:53.000000000 +0200
+++ netkit-telnet-0.17/telnet/telnet.cc	2003-09-06 20:40:55.000000000 +0200
@@ -61,7 +61,9 @@
  * Due to lossage in some linux distributions/kernel releases/libc versions
  * this must come *after* termios.h (which is included in externs.h)
  */
+extern "C" {
 #include <termcap.h>
+}
 
 #ifdef USE_NCURSES
 #include <term.h>
