--- Makefile.in.orig
+++ Makefile.in
@@ -38,15 +38,15 @@
 	done
 
 install: build
-	$(INSTALL) -m 750 -d $(DESTDIR)$(prefix)
-	$(INSTALL) -m 750 -d $(DESTDIR)$(bindir)
-	$(INSTALL) -m 750 -d $(DESTDIR)$(sbindir)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(prefix)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(bindir)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(sbindir)
 	$(INSTALL) -m 750 -d $(DESTDIR)$(sysconfdir)
-	$(INSTALL) -m 750 -d $(DESTDIR)$(includedir)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(includedir)
 	$(INSTALL) -m 750 -d $(DESTDIR)$(logdir)
 	$(INSTALL) -m 750 -d $(DESTDIR)$(rundir)
 	$(INSTALL) -m 750 -d $(DESTDIR)$(helpdir)
-	$(INSTALL) -m 750 -d $(DESTDIR)$(includedir)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(includedir)
 	$(INSTALL) -m 750 tools/ircd-shortcut.pl $(DESTDIR)$(bindir)
 	$(INSTALL) -m 750 tools/dbupgrade.pl $(DESTDIR)$(bindir)
 	$(INSTALL) -m 640 tools/definetolength.pl $(DESTDIR)$(bindir)
