Common subdirectories: minidlna-1.0.24/linux and minidlna-1.0.24-fixed/linux
diff -Nau minidlna-1.0.24/metadata.c minidlna-1.0.24-fixed/metadata.c
--- minidlna-1.0.24/metadata.c	2012-02-14 19:44:56.000000000 +0100
+++ minidlna-1.0.24-fixed/metadata.c	2012-06-19 17:11:39.150099226 +0200
@@ -710,7 +710,7 @@
 		DPRINTF(E_WARN, L_METADATA, "Opening %s failed!\n", path);
 		return 0;
 	}
-	av_find_stream_info(ctx);
+	avformat_find_stream_info(ctx, NULL);
 	//dump_format(ctx, 0, NULL, 0);
 	for( i=0; i<ctx->nb_streams; i++)
 	{
@@ -732,7 +732,7 @@
 	/* This must not be a video file. */
 	if( !vc )
 	{
-		av_close_input_file(ctx);
+		avformat_close_input(&ctx);
 		if( !is_audio(path) )
 			DPRINTF(E_DEBUG, L_METADATA, "File %s does not contain a video stream.\n", basename(path));
 		return 0;
@@ -1514,10 +1514,10 @@
 	{
 		if( ctx->metadata )
 		{
-			AVMetadataTag *tag = NULL;
+			AVDictionaryEntry *tag = NULL;
 
 			//DEBUG DPRINTF(E_DEBUG, L_METADATA, "Metadata:\n");
-			while( (tag = av_metadata_get(ctx->metadata, "", tag, AV_METADATA_IGNORE_SUFFIX)) )
+			while( (tag = av_dict_get(ctx->metadata, "", tag, AV_DICT_IGNORE_SUFFIX)) )
 			{
 				//DEBUG DPRINTF(E_DEBUG, L_METADATA, "  %-16s: %s\n", tag->key, tag->value);
 				if( strcmp(tag->key, "title") == 0 )
@@ -1534,7 +1534,7 @@
 	#endif
 	#endif
 video_no_dlna:
-	av_close_input_file(ctx);
+	avformat_close_input(&ctx);
 
 #ifdef TIVO_SUPPORT
 	if( ends_with(path, ".TiVo") && is_tivo_file(path) )
Common subdirectories: minidlna-1.0.24/po and minidlna-1.0.24-fixed/po
Common subdirectories: minidlna-1.0.24/tagutils and minidlna-1.0.24-fixed/tagutils
