--- configure	2006-07-24 18:15:49.000000000 +0300
+++ configure.new	2006-09-19 22:50:53.000000000 +0300
@@ -36607,6 +36607,12 @@
      LIBS="$kde_save_LIBS"
 
 
+  if test "x$with_akode" != xcheck && test "x$have_akode" != xyes; then
+    { { echo "$as_me:$LINENO: error: --with-akode was given, but test for aKode failed" >&5
+echo "$as_me: error: --with-akode was given, but test for aKode failed" >&2;}
+   { (exit 1); exit 1; }; }
+  fi
+fi
 
 
 if test x$have_akode_mpeg = xyes; then
@@ -36650,12 +36656,6 @@
 
 
 
-  if test "x$with_akode" != xcheck && test "x$have_akode" != xyes; then
-    { { echo "$as_me:$LINENO: error: --with-akode was given, but test for aKode failed" >&5
-echo "$as_me: error: --with-akode was given, but test for aKode failed" >&2;}
-   { (exit 1); exit 1; }; }
-  fi
-fi
 
 artsc_config_test_path=$prefix/bin:$exec_prefix/bin:$KDEDIR/bin:$PATH
 # Extract the first word of "artsc-config", so it can be a program name with args.
