--- mpeglib/lib/input/cdromAccess.cpp.org	2005-02-02 14:06:28.000000000 +0100
+++ mpeglib/lib/input/cdromAccess.cpp	2005-02-02 14:07:30.000000000 +0100
@@ -27,6 +27,9 @@
   #include <sys/types.h>
   #ifndef __u64
   #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,70)
+  #ifdef __x86_64__
+  typedef unsigned long long __u64;
+  #else
   #include <bits/wordsize.h>
   #if __WORDSIZE == 64
   typedef unsigned long __u64;
@@ -35,6 +38,7 @@
   #endif 
   #endif
   #endif
+  #endif
   #include "cdromAccess_Linux.cpp"
 #endif
 
