diff -Naur virt-manager-0.8.5.orig/src/virt-manager.py.in virt-manager-0.8.5/src/virt-manager.py.in
--- virt-manager-0.8.5.orig/src/virt-manager.py.in	2010-09-16 14:01:40.384000000 -0700
+++ virt-manager-0.8.5/src/virt-manager.py.in	2010-09-16 14:05:57.497999779 -0700
@@ -278,7 +278,8 @@
     # on as if all were fine. Ultimately bad stuff happens,
     # so lets catch it here & get the hell out...
     import warnings
-    warnings.filterwarnings('error', module='gtk')
+    #Don't segfault with Python 2.7
+    #warnings.filterwarnings('error', module='gtk')
     try:
         import gobject
 
