--- configure.in	2003-12-22 11:16:08.000000000 +0100
+++ configure.in.cp	2003-12-22 11:34:31.000000000 +0100
@@ -8,8 +8,8 @@
 
 dnl AC_CONFIG_SUBDIRS(dir ...)
 
-CFLAGS=""
-CXXFLAGS=""
+dnl CFLAGS=""
+dnl CXXFLAGS=""
 includedir="$includedir/dbstep"
 
 dnl $Format: "MAJOR_VERSION=$ReleaseMajorVersion: 0 $"
@@ -40,7 +40,7 @@
 
 dnl Add -Wall if gcc
 if test "$GCC" = "yes"; then
-   CFLAGS="-Wall"
+   CFLAGS="$CFLAGS -Wall"
 fi
 
 AC_PROG_INSTALL
@@ -73,7 +73,6 @@
 	case "$enable_debug" in
 	     no)
 		AC_MSG_RESULT(no)
-		CFLAGS="$CFLAGS -O2"
 		;;
 	     yes)
 	        AC_MSG_RESULT(yes)
@@ -84,9 +83,7 @@
 		CFLAGS="$CFLAGS -DDEBUG$with_debug"
 	        ;;
 	esac
- ],
- [ AC_MSG_RESULT(no)
-   CFLAGS="$CFLAGS -O2" ]
+ ]
 )
 
 AC_CXX_HAVE_SSTREAM
