diff -uNr icmake.orig/comp/callrss.c icmake/comp/callrss.c
--- icmake.orig/comp/callrss.c	1999-11-06 11:33:52.000000000 +0100
+++ icmake/comp/callrss.c	2006-09-16 11:21:58.000000000 +0200
@@ -68,7 +68,7 @@
             args = va_arg(marker, unsigned);
         break;
 
-        default:
+        default: ;
         /*
             default is entered in the switch to prevent a long compiler 
             warning
diff -uNr icmake.orig/comp/cast.c icmake/comp/cast.c
--- icmake.orig/comp/cast.c	1999-11-06 11:34:34.000000000 +0100
+++ icmake/comp/cast.c	2006-09-16 11:22:32.000000000 +0200
@@ -26,7 +26,7 @@
             lcast(e);
         break;
 
-        default:
+        default: ;
         /* 
             default is entered in the switch to prevent a long compiler
             warning
diff -uNr icmake.orig/comp/gencode.c icmake/comp/gencode.c
--- icmake.orig/comp/gencode.c	1999-11-06 11:40:51.000000000 +0100
+++ icmake/comp/gencode.c	2006-09-16 11:23:06.000000000 +0200
@@ -91,7 +91,7 @@
             ++dead[dead_sp];
         break;
 
-        default:
+        default: ;
 /*
         The default switch entry is inserted to prvent a long compiler
         warning about a not-handled enum value
diff -uNr icmake.orig/comp/negate.c icmake/comp/negate.c
--- icmake.orig/comp/negate.c	1994-03-14 14:46:46.000000000 +0100
+++ icmake/comp/negate.c	2006-09-16 11:25:28.000000000 +0200
@@ -13,7 +13,7 @@
     }
 
     if (e->type & e_const)                  /* immediate value */
-        (int)e->evalue = -(int)e->evalue;
+        e->evalue = -(int)e->evalue;
     else
     {
         etoc(e);                            /* convert to code */
diff -uNr icmake.orig/exec/icm-exec.c icmake/exec/icm-exec.c
--- icmake.orig/exec/icm-exec.c	1999-11-06 11:52:06.000000000 +0100
+++ icmake/exec/icm-exec.c	2006-09-16 11:31:05.000000000 +0200
@@ -34,7 +34,7 @@
 
 #include "icm-exec.h"
 
-void main (argc, argv, envp)
+int main (argc, argv, envp)
 int argc;
 char **argv;
 char **envp;
diff -uNr icmake.orig/pp/icm-pp.c icmake/pp/icm-pp.c
--- icmake.orig/pp/icm-pp.c	1995-07-06 14:39:48.000000000 +0200
+++ icmake/pp/icm-pp.c	2006-09-16 11:29:02.000000000 +0200
@@ -35,7 +35,7 @@
 
 #include "icm-pp.h"
 
-void main (int argc, char **argv)
+int main (int argc, char **argv)
 {
     register char
         *progname;
diff -uNr icmake.orig/un/icmun.c icmake/un/icmun.c
--- icmake.orig/un/icmun.c	1994-03-14 13:50:28.000000000 +0100
+++ icmake/un/icmun.c	2006-09-16 11:30:07.000000000 +0200
@@ -6,7 +6,7 @@
 
 #include "icmun.h"
 
-void main (argc, argv)
+int main (argc, argv)
 int argc;
 char **argv;
 {
