--- data/Makefile.in.old	2006-11-14 12:35:23.000000000 +0100
+++ data/Makefile.in	2006-11-14 12:35:34.000000000 +0100
@@ -550,13 +550,6 @@
 	$(LN_S) -f $(pixmapdir)/gdesklets.png \
 	$(DESTDIR)$(fakepixmapdir)/gdesklets.png;
 
-	if test -n "$(UPDATE_MIME_DATABASE)"; then \
-	    $(UPDATE_MIME_DATABASE) "$(DESTDIR)${sharedmimedir}"; \
-	fi;
-	if test -n "$(UPDATE_DESKTOP_DATABASE)"; then \
-	    $(UPDATE_DESKTOP_DATABASE) "$(DESTDIR)${desktopdir}"; \
-	fi;
-
 uninstall-local: uninstall-desktopDATA uninstall-sharedmimepackageDATA
 	if test -n "$(UPDATE_MIME_DATABASE)"; then \
 	    $(UPDATE_MIME_DATABASE) "$(DESTDIR)${sharedmimedir}"; \
