From d018c1945e61fb5a66c51bce6ee35304de0d7f74 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Fri, 23 Sep 2022 16:30:05 +0200
Subject: [PATCH 2/2] Fix duplicated definition of scopegroup

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 app/track-editor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/track-editor.c b/app/track-editor.c
index 12dff47abde9..2bbee6d59672 100644
--- a/app/track-editor.c
+++ b/app/track-editor.c
@@ -43,7 +43,7 @@
 Tracker *tracker;
 GtkWidget *trackersettings;
 GtkWidget *vscrollbar;
-ScopeGroup *scopegroup;
+extern ScopeGroup *scopegroup;
 
 static GtkWidget *hscrollbar;
 
-- 
2.37.3

