diff --git a/src/liblzma/Makefile.am b/src/liblzma/Makefile.am
+++ b/src/liblzma/Makefile.am
@@ -97,7 +97,7 @@
 
 
 ## pkg-config
-pkgconfigdir = $(libdir)/pkgconfig
+pkgconfigdir = $(prefix)/lib/pkgconfig
 pkgconfig_DATA = liblzma.pc
 EXTRA_DIST += liblzma.pc.in
 
diff --git a/src/liblzma/Makefile.in b/src/liblzma/Makefile.in
--- a/src/liblzma/Makefile.in
+++ b/src/liblzma/Makefile.in
@@ -752,7 +752,7 @@
 
 liblzma_la_LDFLAGS = -no-undefined -version-info 7:4:2 $(am__append_1) \
 	$(am__append_42)
-pkgconfigdir = $(libdir)/pkgconfig
+pkgconfigdir = $(prefix)/lib/pkgconfig
 pkgconfig_DATA = liblzma.pc
 pc_verbose = $(pc_verbose_@AM_V@)
 pc_verbose_ = $(pc_verbose_@AM_DEFAULT_V@)
