--- Makefile.in	2004-04-06 09:07:55.452930992 -0700
+++ Makefile.in	2004-04-06 09:09:16.704578864 -0700
@@ -40,15 +40,11 @@
 magic.c: magic.m4 mkmagic config.h
 	@srcdir@/mkmagic magic.m4
 
-cfmagic: file/magic
-	@MAKE_MAGIC@ magic
-
-commoninstall: textonly cfmagic
+commoninstall: textonly
 	@INSTALL_DIR@ $(prefix)/@exedir@
 	@INSTALL_DIR@ $(prefix)/@mandir@/man8
 	@INSTALL_DIR@ $(prefix)/@mandir@/man5
 	@INSTALL_DIR@ $(prefix)/@filterdir@
-	@INSTALL_MAGIC@ -m 444 file/magic $(prefix)/@MAGIC@
 	@INSTALL_PROGRAM@ -m 511 textonly $(prefix)/@filterdir@
 
 install:  @DO_WHAT@
