Author:  Ismael Luceno <ismael.luceno@sourcemage.org>
Subject: Fix missing declaration of wVBox at EV_UnixMenuBar::synthesizeMenuBar

--- a/src/af/ev/gtk/ev_UnixMenu.cpp	2013-04-07 10:53:03.000000000 -0300
+++ b/src/af/ev/gtk/ev_UnixMenu.cpp	2015-01-11 03:00:48.762458410 -0300
@@ -978,12 +978,14 @@
	GtkWidget * wWidget = static_cast<XAP_UnixFrameImpl *>(m_pFrame->getFrameImpl())->getTopLevelWindow();
	m_wMenuBar = gtk_menu_new ();
     hildon_window_set_menu (HILDON_WINDOW (wWidget), GTK_MENU (m_wMenuBar));
-#elif defined (ENABLE_MENUBUTTON)
-	m_wMenuBar = gtk_menu_new ();
 #else
	GtkWidget * wVBox = static_cast<XAP_UnixFrameImpl *>(m_pFrame->getFrameImpl())->getVBoxWidget();
+#if defined (ENABLE_MENUBUTTON)
+	m_wMenuBar = gtk_menu_new ();
+#else
	m_wMenuBar = gtk_menu_bar_new();
 #endif
+#endif

	synthesizeMenu(m_wMenuBar, false);
	gtk_widget_show_all(m_wMenuBar);
