diff -ur audacious-plugins-3.3.4/src/cdaudio-ng/cdaudio-ng.c audacious-plugins-mod/src/cdaudio-ng/cdaudio-ng.c
--- audacious-plugins-3.3.4/src/cdaudio-ng/cdaudio-ng.c	2013-02-03 14:09:33.000000000 -0600
+++ audacious-plugins-mod/src/cdaudio-ng/cdaudio-ng.c	2013-02-17 20:43:38.505087973 -0600
@@ -29,12 +29,7 @@
 #include <cdio/audio.h>
 #include <cdio/sector.h>
 #include <cdio/cd_types.h>
-
-#if LIBCDIO_VERSION_NUM >= 90
-#include <cdio/paranoia/cdda.h>
-#else
 #include <cdio/cdda.h>
-#endif
 
 /* libcdio's header files #define these */
 #undef PACKAGE
