From 6aa2974e6aba87154fa1b47935074b6b18eb431b Mon Sep 17 00:00:00 2001
From: "Chris \"Koying\" Browet" <cbro@semperpax.com>
Date: Fri, 15 Jun 2012 11:24:23 +0200
Subject: [PATCH 3/8] FIX : secondary road simple color

---
 src/Features/Way.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Features/Way.cpp b/src/Features/Way.cpp
index 7eea383..ece774b 100644
--- a/src/Features/Way.cpp
+++ b/src/Features/Way.cpp
@@ -93,7 +93,7 @@ void WayPrivate::CalculateWidth()
         SimpleColor = QColor(228, 109, 113);
     } else if (h == "secondary") {
         SimpleWidth =  2*LANEWIDTH; // 2 lanes
-        SimpleColor = QColor(253, 191, 252);
+        SimpleColor = QColor(253, 191, 111);
     } else if (h == "tertiary") {
         SimpleWidth =  1.5*LANEWIDTH; // shared middle lane
         SimpleColor = QColor(255, 229, 33);
-- 
1.8.4.2

