From 950109c2fe443b3b60c8e98a0be7914be4301d5d Mon Sep 17 00:00:00 2001
From: "Chris \"Koying\" Browet" <cbro@semperpax.com>
Date: Fri, 29 Jun 2012 13:24:27 +0200
Subject: [PATCH 6/8] FIX : French cadastre parsing issue (fixes #3334)

---
 .../background/MCadastreFranceBackground/qadastre/cadastrewrapper.cpp  | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/plugins/background/MCadastreFranceBackground/qadastre/cadastrewrapper.cpp b/plugins/background/MCadastreFranceBackground/qadastre/cadastrewrapper.cpp
index 6abcab6..30e87e2 100644
--- a/plugins/background/MCadastreFranceBackground/qadastre/cadastrewrapper.cpp
+++ b/plugins/background/MCadastreFranceBackground/qadastre/cadastrewrapper.cpp
@@ -219,6 +219,7 @@ void CadastreWrapper::networkFinished(QNetworkReply *reply)
         m_gotCookie = true;
     } else if (reply->url() == QUrl("http://www.cadastre.gouv.fr/scpc/rechercherPlan.do")) {
         QString pageData = reply->readAll();
+        qDebug() << pageData;
         QWebPage parsedPage(this);
         QWebFrame *frame = parsedPage.mainFrame();
         frame->setHtml(pageData);
@@ -238,7 +239,7 @@ void CadastreWrapper::networkFinished(QNetworkReply *reply)
         } else {
             // We may have been successfull, who knows ?
             QString name = frame->findFirstElement("#ville").attribute("value");
-            QWebElementCollection links = frame->findAllElements(".resultat > .parcelles .view a");
+            QWebElementCollection links = frame->findAllElements(".resultat > .parcelles a");
             QRegExp linkRE("c=(\\w+)");
             foreach (QWebElement link, links) {
                 QString js = link.attribute("onclick");
-- 
1.8.4.2

