From a95b212a29d3a4d624e5d14a90605a3dd4c9e476 Mon Sep 17 00:00:00 2001
From: "Chris \"Koying\" Browet" <cbro@semperpax.com>
Date: Fri, 29 Jun 2012 14:01:48 +0200
Subject: [PATCH 7/8] FIX : Virtual nodes not highlighted on hover (fixes
 #3333)

---
 src/Interactions/Interaction.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/Interactions/Interaction.cpp b/src/Interactions/Interaction.cpp
index 8d02324..0560f88 100644
--- a/src/Interactions/Interaction.cpp
+++ b/src/Interactions/Interaction.cpp
@@ -347,8 +347,7 @@ void FeatureSnapInteraction::paintEvent(QPaintEvent* anEvent, QPainter& thePaint
         theMain->properties()->highlighted(i)->drawHighlight(thePainter, view());
     }
 
-    //FIXME document()->exists necessary?
-    if (LastSnap && document()->exists(LastSnap)) {
+    if (LastSnap) {
         LastSnap->drawHover(thePainter, view());
         view()->setToolTip(LastSnap->toHtml());
     } else {
-- 
1.8.4.2

