From 2d51cab30924999634a43b78e954d1745d5dcd28 Mon Sep 17 00:00:00 2001
From: "Chris \"Koying\" Browet" <cbro@semperpax.com>
Date: Thu, 25 Apr 2013 20:10:27 +0200
Subject: [PATCH 8/8] FIX: GDAL crash

---
 src/ImportExport/ImportExportGdal.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/ImportExport/ImportExportGdal.cpp b/src/ImportExport/ImportExportGdal.cpp
index 7c829f4..2005d25 100644
--- a/src/ImportExport/ImportExportGdal.cpp
+++ b/src/ImportExport/ImportExportGdal.cpp
@@ -331,7 +331,8 @@ bool ImportExportGdal::importGDALDataset(OGRDataSource* poDS, Layer* aLayer, boo
         }
     }
 
-    theSrs->Release();
+    if (theSrs)
+        theSrs->Release();
     theSrs = new OGRSpatialReference();
     theSrs->importFromProj4(sPrj.toLatin1().data());
     toWGS84 = OGRCreateCoordinateTransformation(theSrs, &wgs84srs);
-- 
1.8.4.2

