--- nikto.pl	2004-09-03 00:09:14.964998890 +1200
+++ nikto.pl.old	2004-09-03 00:07:42.156666682 +1200
@@ -3,9 +3,9 @@
 Getopt::Long::Configure('no_ignore_case');
 
 # INSTALLED LW:
+use LW; 
-#use LW; 
 # LOCAL LW:
+#require "./plugins/LW.pm";
-require "./plugins/LW.pm";
 
 #######################################################################
 # last update: 05.30.2003
@@ -102,7 +102,7 @@
 # load config file
 sub load_configs
 {
+ my $configfile="/usr/share/nikto/config.txt";
- my $configfile="config.txt";
  my $noconfig=0;
    open(CONF,"<$configfile") || $noconfig++;
    my @CONFILE=<CONF>;
