From 99e4891bb79a484932e6b4e6a5221fb3ae05a011 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Fri, 8 Jan 2016 22:56:14 +0100
Subject: [PATCH] disable pulseaudio

---
 Makefile           | 4 +---
 src/print_volume.c | 2 +-
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index bb90b1a..cbaf416 100644
--- a/Makefile
+++ b/Makefile
@@ -70,10 +70,8 @@ CFLAGS += -idirafter yajl-fallback
 OBJS:=$(wildcard src/*.c *.c)
 OBJS:=$(OBJS:.c=.o)
 
-ifeq ($(OS),OpenBSD)
 OBJS:=$(filter-out src/pulse.o, $(OBJS))
-LIBS:=$(filter-out -lpulse, $(LIBS)) -lpthread
-endif
+LIBS:=$(filter-out -lpulse, $(LIBS))
 
 src/%.o: src/%.c include/i3status.h
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $<
diff --git a/src/print_volume.c b/src/print_volume.c
index 58cba2a..151f9b2 100644
--- a/src/print_volume.c
+++ b/src/print_volume.c
@@ -60,7 +60,7 @@ void print_volume(yajl_gen json_gen, char *buffer, const char *fmt, const char *
         free(instance);
     }
 
-#ifndef __OpenBSD__
+#if 0
     /* Try PulseAudio first */
 
     /* If the device name has the format "pulse[:N]" where N is the
-- 
2.7.0

