--- CLAM-0.91.0/src/Defines/Assert.hxx.old	2006-07-25 17:36:12.210981280 +0200
+++ CLAM-0.91.0/src/Defines/Assert.hxx	2006-07-25 17:36:35.945373104 +0200
@@ -50,6 +50,9 @@ namespace CLAM
 #elif defined (__GNUC__) && defined  (__i386__)
 #define CLAM_BREAKPOINT {__asm__ (" int $3 "); }
 
+#elif defined (__GNUC__) && defined  (__amd64__)
+#define CLAM_BREAKPOINT {__asm__ (" int $3 "); }
+
 /* g++ on powerpc linux */
 #elif defined (__GNUC__) && defined  (__powerpc__)
 #define CLAM_BREAKPOINT {__asm__ (" .long 0x7d821008 "); }
