--- configure.old	2005-03-18 13:59:01.441343594 +0100
+++ configure	2005-03-18 13:59:19.749010182 +0100
@@ -13063,7 +13063,7 @@
 
 #include <sys/asoundlib.h>
 #if defined(SND_LIB_MAJOR) && defined(SND_LIB_MINOR)
-#if SND_LIB_MAJOR>0 || (SND_LIB_MAJOR==0 && SND_LIB_MINOR==5)
+#if SND_LIB_MAJOR==0 && SND_LIB_MINOR==5
 AP_maGiC_VALUE
 #endif
 #endif
@@ -13824,7 +13824,7 @@
 i?86)
 	if test "x$GCC" = xyes; then
 	  O_CFLAGS="$O_CFLAGS -fomit-frame-pointer -funroll-loops"
-	  XCFLAGS="$CFLAGS -march=i586 -mcpu=pentiumpro"
+	  XCFLAGS="$CFLAGS "
 	fi
 	USE_MMX=yes
 	;;
--- libogleao/alsa_audio.c.old	2005-03-18 14:07:35.953766181 +0100
+++ libogleao/alsa_audio.c	2005-03-18 14:07:50.456917675 +0100
@@ -251,6 +251,7 @@
     snd_ctl_close(ctl_handle);
   
   __diga_end:
+    ;
   }
   
   return 0;
--- ac3/ac3dec_wrap.c~	2003-03-02 16:50:03.000000000 +0100
+++ ac3/ac3dec_wrap.c	2005-04-27 12:03:57.000000000 +0200
@@ -48,7 +48,7 @@
 static void handle_events(MsgEventQ_t *q, MsgEvent_t *ev);
 
 
-static char *program_name;
+char *program_name;
 
 static FILE *outfile;
 
