diff -Nur watchdog-5.2-orig/include/sundries.h watchdog-5.2/include/sundries.h
--- watchdog-5.2-orig/include/sundries.h	1998-09-10 17:19:11.000000000 +0200
+++ watchdog-5.2/include/sundries.h	2003-07-31 01:40:07.000000000 +0200
@@ -51,7 +51,7 @@
 #if defined(__GNUC_MINOR__) && __GNUC__ == 2 && __GNUC_MINOR__ >= 5
 void die (int errcode, const char *fmt, ...) __attribute__ ((noreturn));
 #else /* GNUC < 2.5 */
-void volatile die (int errcode, const char *fmt, ...);
+void die (int errcode, const char *fmt, ...);
 #endif /* GNUC < 2.5 */
 #else /* !__GNUC__ */
 void die (int errcode, const char *fmt, ...);
diff -Nur watchdog-5.2-orig/src/quotactl.c watchdog-5.2/src/quotactl.c
--- watchdog-5.2-orig/src/quotactl.c	1999-09-09 11:03:06.000000000 +0200
+++ watchdog-5.2/src/quotactl.c	2003-07-31 01:40:26.000000000 +0200
@@ -34,6 +34,7 @@
 	return syscall(__NR_quotactl, cmd, special, id, addr);
 }
 #else
+#include <errno.h>
 #include <sys/types.h>
 #define __LIBRARY__
 #include <linux/unistd.h>
