--- wu-ftpd-2.6.0/makefiles/Makefile.lnx	Sun Sep 19 23:21:11 1999
+++ wu-ftpd-2.6.0/makefiles/blah	Mon Apr 10 09:41:05 2000
@@ -47,36 +47,36 @@
 install: bin/ftpd bin/ftpcount bin/ftpshut bin/ftprestart bin/ftpwho bin/privatepw
 	@echo installing binaries.
 	@if [ ! -e ${BINDIR} ]; then \
-		${INSTALL} -o ${BINOWN} -g ${BINGRP} -m 755 -d ${BINDIR} ; \
+		${INSTALL}  -m 755 -d ${BINDIR} ; \
 	fi
 	@if [ ! -e ${SBINDIR} ]; then \
-		${INSTALL} -o ${SBINOWN} -g ${SBINGRP} -m 755 -d ${SBINDIR} ; \
+		${INSTALL}  -m 755 -d ${SBINDIR} ; \
 	fi
-	${INSTALL} -c -o ${SBINOWN} -g ${SBINGRP} -m 110 bin/ftpd           ${SBINDIR}/in.ftpd
-	@( cd ${SBINDIR} ; ln -s -f in.ftpd wu.ftpd   ; chown ${SBINOWN}.${SBINGRP} wu.ftpd   )
-	@( cd ${SBINDIR} ; ln -s -f in.ftpd in.wuftpd ; chown ${SBINOWN}.${SBINGRP} in.wuftpd )
-	${INSTALL} -c -o ${BINOWN}  -g ${BINGRP}  -m 111 bin/ftpshut        ${BINDIR}/ftpshut
-	${INSTALL} -c -o ${BINOWN}  -g ${BINGRP}  -m 111 bin/ftprestart     ${BINDIR}/ftprestart
-	${INSTALL} -c -o ${BINOWN}  -g ${BINGRP}  -m 111 bin/ftpcount       ${BINDIR}/ftpcount
-	${INSTALL} -c -o ${BINOWN}  -g ${BINGRP}  -m 111 bin/ftpwho         ${BINDIR}/ftpwho
-	${INSTALL} -c -o ${BINOWN}  -g ${BINGRP}  -m 111 bin/privatepw      ${BINDIR}/privatepw
+	${INSTALL} -c  -m 110 bin/ftpd           ${SBINDIR}/in.ftpd
+	@( cd ${SBINDIR} ; ln -s -f in.ftpd wu.ftpd   ; echo blah  )
+	@( cd ${SBINDIR} ; ln -s -f in.ftpd in.wuftpd ; echo blah )
+	${INSTALL} -c   -m 111 bin/ftpshut        ${BINDIR}/ftpshut
+	${INSTALL} -c   -m 111 bin/ftprestart     ${BINDIR}/ftprestart
+	${INSTALL} -c   -m 111 bin/ftpcount       ${BINDIR}/ftpcount
+	${INSTALL} -c   -m 111 bin/ftpwho         ${BINDIR}/ftpwho
+	${INSTALL} -c   -m 111 bin/privatepw      ${BINDIR}/privatepw
 	@echo installing manpages.
 	@if [ ! -e ${MANDIR}/man1 ]; then \
-		${INSTALL} -o ${MANOWN} -g ${MANGRP} -m 755 -d ${MANDIR}/man1 ; \
+		${INSTALL}  -m 755 -d ${MANDIR}/man1 ; \
 	fi
-	${INSTALL} -c -o ${MANOWN} -g ${MANGRP} -m 444 doc/ftpcount.1       ${MANDIR}/man1/ftpcount.1
-	${INSTALL} -c -o ${MANOWN} -g ${MANGRP} -m 444 doc/ftpwho.1         ${MANDIR}/man1/ftpwho.1
+	${INSTALL} -c  -m 444 doc/ftpcount.1       ${MANDIR}/man1/ftpcount.1
+	${INSTALL} -c  -m 444 doc/ftpwho.1         ${MANDIR}/man1/ftpwho.1
 	@if [ ! -e ${MANDIR}/man5 ]; then \
-		${INSTALL} -o ${MANOWN} -g ${MANGRP} -m 755 -d ${MANDIR}/man5 ; \
+		${INSTALL}  -m 755 -d ${MANDIR}/man5 ; \
 	fi
-	${INSTALL} -c -o ${MANOWN} -g ${MANGRP} -m 444 doc/ftpaccess.5      ${MANDIR}/man5/ftpaccess.5
-	${INSTALL} -c -o ${MANOWN} -g ${MANGRP} -m 444 doc/ftpconversions.5 ${MANDIR}/man5/ftpconversions.5
-	${INSTALL} -c -o ${MANOWN} -g ${MANGRP} -m 444 doc/ftphosts.5       ${MANDIR}/man5/ftphosts.5
-	${INSTALL} -c -o ${MANOWN} -g ${MANGRP} -m 444 doc/xferlog.5        ${MANDIR}/man5/xferlog.5
+	${INSTALL} -c  -m 444 doc/ftpaccess.5      ${MANDIR}/man5/ftpaccess.5
+	${INSTALL} -c  -m 444 doc/ftpconversions.5 ${MANDIR}/man5/ftpconversions.5
+	${INSTALL} -c  -m 444 doc/ftphosts.5       ${MANDIR}/man5/ftphosts.5
+	${INSTALL} -c  -m 444 doc/xferlog.5        ${MANDIR}/man5/xferlog.5
 	@if [ ! -e ${MANDIR}/man8 ]; then \
-		${INSTALL} -o ${MANOWN} -g ${MANGRP} -m 755 -d ${MANDIR}/man8 ; \
+		${INSTALL}  -m 755 -d ${MANDIR}/man8 ; \
 	fi
-	${INSTALL} -c -o ${MANOWN} -g ${MANGRP} -m 444 doc/ftpd.8           ${MANDIR}/man8/ftpd.8
-	${INSTALL} -c -o ${MANOWN} -g ${MANGRP} -m 444 doc/ftpshut.8        ${MANDIR}/man8/ftpshut.8
-	${INSTALL} -c -o ${MANOWN} -g ${MANGRP} -m 444 doc/ftprestart.8     ${MANDIR}/man8/ftprestart.8
-	${INSTALL} -c -o ${MANOWN} -g ${MANGRP} -m 444 util/privatepw/privatepw.8     ${MANDIR}/man8/privatepw.8
+	${INSTALL} -c  -m 444 doc/ftpd.8           ${MANDIR}/man8/ftpd.8
+	${INSTALL} -c  -m 444 doc/ftpshut.8        ${MANDIR}/man8/ftpshut.8
+	${INSTALL} -c  -m 444 doc/ftprestart.8     ${MANDIR}/man8/ftprestart.8
+	${INSTALL} -c  -m 444 util/privatepw/privatepw.8     ${MANDIR}/man8/privatepw.8
