diff -Naur file-4.13.orig/magic/magic.mime file-4.13/magic/magic.mime
--- file-4.13.orig/magic/magic.mime	2005-02-10 11:58:50.565522432 -0800
+++ file-4.13/magic/magic.mime	2005-02-10 11:58:52.513226336 -0800
@@ -901,6 +901,14 @@
 # http://www.djvuzone.org/
 0	string		AT&TFORM	image/x.djvu
 
+#------------------------------------------------------------------------------
+# zisofs: file(1) magic for zisofs/RockRidge compressed files
+#
+# from H. Peter Anvin <hpa@zytor.com> May 4, 2001
+#
+0	string	\x37\xE4\x53\x96\xC9\xDB\xD6\x07	zisofs/Rockridge compressed file
+>8	lelong	x	- %d bytes
+
 # Danny Milosavljevic <danny.milo@gmx.net>
 # this are adrift (adventure game standard) game files, extension .taf
 # depending on version magic continues with 0x93453E6139FA (V 4.0)
