diff -uNr hal-0.5.7/tools/hal-device.c hal-0.5.7.cp/tools/hal-device.c
--- hal-0.5.7/tools/hal-device.c	2005-11-02 16:38:14.000000000 +0100
+++ hal-0.5.7.cp/tools/hal-device.c	2006-07-25 09:07:13.000000000 +0200
@@ -155,7 +155,7 @@
 
 	libhal_ctx_shutdown(hal_ctx, &error);
 	libhal_ctx_free(hal_ctx);
-	dbus_connection_disconnect(conn);
+	dbus_connection_close(conn);
 	dbus_connection_unref(conn);
 	dbus_error_free(&error);
 
diff -uNr hal-0.5.7/tools/lshal.c hal-0.5.7.cp/tools/lshal.c
--- hal-0.5.7/tools/lshal.c	2006-02-13 15:35:31.000000000 +0100
+++ hal-0.5.7.cp/tools/lshal.c	2006-07-25 09:08:02.000000000 +0200
@@ -701,7 +701,7 @@
 	libhal_ctx_shutdown (hal_ctx, &error);
 	libhal_ctx_free (hal_ctx);
 
-	dbus_connection_disconnect (conn);
+	dbus_connection_close (conn);
 	dbus_connection_unref (conn);
 
 	if (show_device)
