--- fdi/policy/10osvendor/10-storage-policy.fdi.orig	2005-12-01 13:57:06.000000000 +0000
+++ fdi/policy/10osvendor/10-storage-policy.fdi	2005-12-01 13:57:29.000000000 +0000
@@ -12,6 +12,7 @@
       <merge key="storage.policy.default.mount_option.noauto" type="bool">true</merge>
       <merge key="storage.policy.default.mount_option.pamconsole" type="bool">true</merge>
       <merge key="storage.policy.default.mount_option.exec" type="bool">true</merge>
+      <merge key="storage.policy.default.mount_option.user" type="bool">true</merge>
     </match>
   </device>
 
