--- hald/linux/addons/addon-acpi-buttons-toshiba.c.orig
+++ hald/linux/addons/addon-acpi-buttons-toshiba.c
@@ -40,7 +40,7 @@
 #include <string.h>
 #include <unistd.h>
 
-#include <glib/gmain.h>
+#include <glib.h>
 
 #include "libhal/libhal.h"
 #include "../../logger.h"
--- hald/linux/addons/addon-dell-backlight.cpp.orig
+++ hald/linux/addons/addon-dell-backlight.cpp
@@ -29,7 +29,7 @@
 
 #include <string.h>
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hald/linux/addons/addon-generic-backlight.c.orig
+++ hald/linux/addons/addon-generic-backlight.c
@@ -35,7 +35,7 @@
 #include <sys/stat.h>
 #include <unistd.h> 
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hald/linux/addons/addon-imac-backlight.c.orig
+++ hald/linux/addons/addon-imac-backlight.c
@@ -29,7 +29,7 @@
 #include <stdlib.h>
 #include <sys/io.h>
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hald/linux/addons/addon-input.c.orig
+++ hald/linux/addons/addon-input.c
@@ -41,7 +41,7 @@
 #else
   #include <linux/input.h>
 #endif
-#include <glib/gmain.h>
+#include <glib.h>
 #include <glib/gprintf.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hald/linux/addons/addon-ipw-killswitch.c.orig
+++ hald/linux/addons/addon-ipw-killswitch.c
@@ -36,7 +36,6 @@
 #include <unistd.h> 
 
 #include <glib.h>
-#include <glib/gmain.h>
 #include <glib/gstdio.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
--- hald/linux/addons/addon-leds.c.orig
+++ hald/linux/addons/addon-leds.c
@@ -36,7 +36,6 @@
 #include <unistd.h> 
 
 #include <glib.h>
-#include <glib/gmain.h>
 #include <glib/gstdio.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
--- hald/linux/addons/addon-macbookpro-backlight.c.orig
+++ hald/linux/addons/addon-macbookpro-backlight.c
@@ -39,7 +39,7 @@
 #include <pci/pci.h>
 #include <unistd.h> 
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hald/linux/addons/addon-omap-backlight.c.orig
+++ hald/linux/addons/addon-omap-backlight.c
@@ -41,7 +41,7 @@
 #include <fcntl.h>
 #include <unistd.h> 
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hald/linux/addons/addon-rfkill-killswitch.c.orig
+++ hald/linux/addons/addon-rfkill-killswitch.c
@@ -36,7 +36,6 @@
 #include <unistd.h> 
 
 #include <glib.h>
-#include <glib/gmain.h>
 #include <glib/gstdio.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
--- hald/linux/addons/addon-storage.c.orig
+++ hald/linux/addons/addon-storage.c
@@ -39,7 +39,7 @@
 #include <string.h>
 #include <sys/ioctl.h>
 #include <unistd.h>
-#include <glib/gmain.h>
+#include <glib.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hald/linux/addons/addon-usb-csr.c.orig
+++ hald/linux/addons/addon-usb-csr.c
@@ -29,7 +29,7 @@
 #include <string.h>
 #include <usb.h>
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
