diff -Naupr hal-0.5.8.1.orig/tools/hal-device.c hal-0.5.8.1/tools/hal-device.c
--- hal-0.5.8.1.orig/tools/hal-device.c	2007-05-09 16:18:19.000000000 +0200
+++ hal-0.5.8.1/tools/hal-device.c	2007-05-09 16:40:06.000000000 +0200
@@ -157,7 +157,6 @@ int main(int argc, char **argv)
 
 	libhal_ctx_shutdown(hal_ctx, &error);
 	libhal_ctx_free(hal_ctx);
-	dbus_connection_close(conn);
 	dbus_connection_unref(conn);
 	dbus_error_free(&error);
 
