diff -Naupr hal-0.5.8.1.orig/tools/lshal.c hal-0.5.8.1/tools/lshal.c
--- hal-0.5.8.1.orig/tools/lshal.c	2007-05-09 16:18:19.000000000 +0200
+++ hal-0.5.8.1/tools/lshal.c	2007-05-09 16:20:44.000000000 +0200
@@ -722,7 +722,6 @@ main (int argc, char *argv[])
 		LIBHAL_FREE_DBUS_ERROR (&error);
 	libhal_ctx_free (hal_ctx);
 
-	dbus_connection_close (conn);
 	dbus_connection_unref (conn);
 
 	if (show_device)
