Index: hald/linux2/blockdev.c
===================================================================
RCS file: /cvs/hal/hal/hald/linux2/blockdev.c,v
retrieving revision 1.27
diff -u -3 -p -r1.27 blockdev.c
--- hald/linux2/blockdev.c	2 Nov 2005 15:38:13 -0000	1.27
+++ hald/linux2/blockdev.c	28 Nov 2005 23:04:24 -0000
@@ -827,7 +827,8 @@ hotplug_event_begin_add_blockdev (const 
 
 			/* These magic values are documented in the kernel source */
 			switch (type) {
-			case 0:	/* Disk */
+			case 0:	 /* Disk */
+			case 14: /* Simple Direct Access Device, set it to disk */
 				hal_device_property_set_string (d, "storage.drive_type", "disk");
 				break;
 
 
