diff -uNr hal-0.5.9.1/configure hal-0.5.9.1.cp/configure
--- hal-0.5.9.1/configure	2007-06-12 20:27:26.000000000 +0200
+++ hal-0.5.9.1.cp/configure	2007-06-21 10:04:22.000000000 +0200
@@ -22743,8 +22743,8 @@
    { (exit 1); exit 1; }; }
 fi
 
-   { echo "$as_me:$LINENO: checking for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6" >&5
-echo $ECHO_N "checking for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6... $ECHO_C" >&6; }
+   { echo "$as_me:$LINENO: checking for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6, 1.8.7" >&5
+echo $ECHO_N "checking for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6, 1.8.8... $ECHO_C" >&6; }
    LDFLAGS=-lparted
    if test "$cross_compiling" = yes; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
@@ -22774,7 +22774,8 @@
 	    (major == 1 && minor == 8 && micro == 0) ||
 	    (major == 1 && minor == 8 && micro == 1) ||
 	    (major == 1 && minor == 8 && micro == 2) ||
-	    (major == 1 && minor == 8 && micro == 6))
+	    (major == 1 && minor == 8 && micro == 6) ||
+	    (major == 1 && minor == 8 && micro == 8))
 	   return 0;
 
 	return 1;
@@ -22809,8 +22810,8 @@
 sed 's/^/| /' conftest.$ac_ext >&5
 
 ( exit $ac_status )
-{ { echo "$as_me:$LINENO: error: *** Requires libparted == 1.7.1/1.8.0/1.8.1/1.8.2 or 1.8.6" >&5
-echo "$as_me: error: *** Requires libparted == 1.7.1/1.8.0/1.8.1/1.8.2 or 1.8.6" >&2;}
+{ { echo "$as_me:$LINENO: error: *** Requires libparted == 1.7.1/1.8.0/1.8.1/1.8.2 or 1.8.6/7" >&5
+echo "$as_me: error: *** Requires libparted == 1.7.1/1.8.0/1.8.1/1.8.2 or 1.8.6/7/8" >&2;}
    { (exit 1); exit 1; }; }
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
