diff -uNr hal-0.5.10/configure hal-0.5.10.cp/configure
--- hal-0.5.10/configure	2007-09-25 21:08:31.000000000 +0200
+++ hal-0.5.10.cp/configure	2007-10-30 20:50:49.000000000 +0100
@@ -21845,8 +21845,8 @@
    { (exit 1); exit 1; }; }
 fi
 
-   { echo "$as_me:$LINENO: checking for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6, 1.8.7" >&5
-echo $ECHO_N "checking for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6, 1.8.7... $ECHO_C" >&6; }
+   { echo "$as_me:$LINENO: checking for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6, 1.8.7, 1.8.8" >&5
+echo $ECHO_N "checking for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6, 1.8.7, 1.8.8... $ECHO_C" >&6; }
    LDFLAGS=-lparted
    if test "$cross_compiling" = yes; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
@@ -21878,7 +21878,8 @@
 	    (major == 1 && minor == 8 && micro == 1) ||
 	    (major == 1 && minor == 8 && micro == 2) ||
 	    (major == 1 && minor == 8 && micro == 6) ||
-	    (major == 1 && minor == 8 && micro == 7))
+	    (major == 1 && minor == 8 && micro == 7) ||
+	    (major == 1 && minor == 8 && micro == 8))
 	   return 0;
 
 	return 1;
@@ -21914,7 +21915,7 @@
 
 ( exit $ac_status )
 { { echo "$as_me:$LINENO: error: *** Requires libparted == 1.7.1/1.8.0/1.8.1/1.8.2/1.8.6 or 1.8.7" >&5
-echo "$as_me: error: *** Requires libparted == 1.7.1/1.8.0/1.8.1/1.8.2/1.8.6 or 1.8.7" >&2;}
+echo "$as_me: error: *** Requires libparted == 1.7.1/1.8.0/1.8.1/1.8.2/1.8.6/1.8.7 or 1.8.8" >&2;}
    { (exit 1); exit 1; }; }
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
