--- configure.old	2006-09-19 22:24:41.000000000 +0200
+++ configure	2006-10-04 22:45:20.000000000 +0200
@@ -21658,7 +21658,7 @@
 echo "$as_me: error: *** dl library (libdl) not found" >&2;}
    { (exit 1); exit 1; }; }
 fi
-
+save_LDFLAGS="$LDFLAGS"
    echo "$as_me:$LINENO: checking for libparted == 1.7.1" >&5
 echo $ECHO_N "checking for libparted == 1.7.1... $ECHO_C" >&6
    LDFLAGS=-lparted
@@ -21717,7 +21717,7 @@
 fi
     PARTED_LIBS=-lparted
 
-    LDFLAGS=
+    LDFLAGS="$save_LDFLAGS"
 else
    USE_PARTED=no
 fi
