--- configure.old	2006-09-19 22:24:41.000000000 +0200
+++ configure	2006-12-02 10:34:32.000000000 +0100
@@ -21658,9 +21658,9 @@
 echo "$as_me: error: *** dl library (libdl) not found" >&2;}
    { (exit 1); exit 1; }; }
 fi
-
-   echo "$as_me:$LINENO: checking for libparted == 1.7.1" >&5
-echo $ECHO_N "checking for libparted == 1.7.1... $ECHO_C" >&6
+save_LDFLAGS="$LDFLAGS"
+   echo "$as_me:$LINENO: checking for libparted >= 1.7.1" >&5
+echo $ECHO_N "checking for libparted >= 1.7.1... $ECHO_C" >&6
    LDFLAGS=-lparted
    if test "$cross_compiling" = yes; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
@@ -21686,7 +21686,9 @@
 	if ( sscanf( ped_get_version(), "%d.%d.%d", &major, &minor, &micro ) == 3 )
 		printf( "Found libparted %s", ped_get_version() ) ;
 
-        return ! ( major == 1 && minor == 7 && micro == 1 ) ;
+        return ! ((major > 1) ||
+                 ((major == 1) && (minor > 7)) ||
+                 ((major == 1) && (minor == 7) && (micro >= 1))) ;
     }
 
 _ACEOF
@@ -21717,7 +21719,7 @@
 fi
     PARTED_LIBS=-lparted
 
-    LDFLAGS=
+    LDFLAGS="$save_LDFLAGS"
 else
    USE_PARTED=no
 fi
