diff -Naru hal-0.5.12rc1.old/hald/linux/probing/probe-serial.c hal-0.5.12rc1/hald/linux/probing/probe-serial.c
--- hal-0.5.12rc1.old/hald/linux/probing/probe-serial.c	2009-02-06 10:34:01.000000000 +0000
+++ hal-0.5.12rc1/hald/linux/probing/probe-serial.c	2009-02-06 10:27:47.000000000 +0000
@@ -29,6 +29,7 @@
 #endif
 
 #include <fcntl.h>
+#include <linux/types.h>
 #include <linux/serial.h>
 #include <stdint.h>
 #include <stdio.h>
