--- akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp~    2005-10-20 22:33:56.000000000 +0200
+++ akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp     2007-02-27 21:01:15.000000000 +0100
@@ -126,9 +126,6 @@
         case SAMPLE_FMT_FLT:
             config->sample_width = -32;
             break;
-        case SAMPLE_FMT_DBL:
-            config->sample_width = -64;
-            break;
         default:
             return false;;
      }
