diff -Naur xosview-1.8.3.orig/Xrm.cc xosview-1.8.3/Xrm.cc
--- xosview-1.8.3.orig/Xrm.cc	2007-12-01 10:37:19.247343664 -0800
+++ xosview-1.8.3/Xrm.cc	2007-12-01 10:43:33.736684594 -0800
@@ -17,7 +17,6 @@
 #include <iostream.h>
 #endif
 #include <unistd.h>  //  for access(), etc.  BCG
-#include "snprintf.h"
 #include "general.h"
 #ifndef NULL
 #define NULL 0
diff -Naur xosview-1.8.3.orig/bitfieldmeter.cc xosview-1.8.3/bitfieldmeter.cc
--- xosview-1.8.3.orig/bitfieldmeter.cc	2007-12-01 10:37:19.247343664 -0800
+++ xosview-1.8.3/bitfieldmeter.cc	2007-12-01 10:43:36.752856477 -0800
@@ -11,7 +11,6 @@
 #include <fstream.h>
 #endif
 #include <stdlib.h>
-#include "snprintf.h"
 #include "general.h"
 #include "bitfieldmeter.h"
 #include "xosview.h"
diff -Naur xosview-1.8.3.orig/config/Makefile.top.in xosview-1.8.3/config/Makefile.top.in
--- xosview-1.8.3.orig/config/Makefile.top.in	2007-12-01 10:37:19.247343664 -0800
+++ xosview-1.8.3/config/Makefile.top.in	2007-12-01 10:46:37.839175999 -0800
@@ -19,7 +19,6 @@
 bitfieldmeter.o \
 bitmeter.o \
 xosview.o \
-snprintf.o \
 main.o
 
 CFILES := $(OBJS:.o=.cc)
diff -Naur xosview-1.8.3.orig/fieldmeter.cc xosview-1.8.3/fieldmeter.cc
--- xosview-1.8.3.orig/fieldmeter.cc	2007-12-01 10:37:19.243343436 -0800
+++ xosview-1.8.3/fieldmeter.cc	2007-12-01 10:43:39.521014224 -0800
@@ -17,7 +17,6 @@
 #include <fstream.h>
 #endif
 #include <stdlib.h>
-#include "snprintf.h"
 #include "general.h"
 #include "fieldmeter.h"
 #include "xosview.h"
diff -Naur xosview-1.8.3.orig/snprintf.cc xosview-1.8.3/snprintf.cc
--- xosview-1.8.3.orig/snprintf.cc	2007-12-01 10:37:19.243343436 -0800
+++ xosview-1.8.3/snprintf.cc	1969-12-31 16:00:00.000000000 -0800
@@ -1,17 +0,0 @@
-#ifndef HAVE_SNPRINTF
-
-#include <stdarg.h>
-#include <stdio.h>
-
-extern "C" int snprintf ( char *str, int n, const char *format, ...)
-    {
-    /* punt the warning */
-    n++;
-    va_list ap;
-    va_start(ap, format);
-    int rval = vsprintf(str, format, ap);
-    va_end(ap);
-    return rval;
-    }
-
-#endif
diff -Naur xosview-1.8.3.orig/snprintf.h xosview-1.8.3/snprintf.h
--- xosview-1.8.3.orig/snprintf.h	2007-12-01 10:37:19.243343436 -0800
+++ xosview-1.8.3/snprintf.h	1969-12-31 16:00:00.000000000 -0800
@@ -1,10 +0,0 @@
-#ifndef snprintf_h
-#define snprintf_h
-
-#ifndef HAVE_SNPRINTF
-extern "C" int snprintf ( char *str, int n, const char *format, ...);
-#else
-#include <stdio.h>
-#endif
-
-#endif
diff -Naur xosview-1.8.3.orig/xosview.cc xosview-1.8.3/xosview.cc
--- xosview-1.8.3.orig/xosview.cc	2007-12-01 10:37:19.239343207 -0800
+++ xosview-1.8.3/xosview.cc	2007-12-01 10:43:42.205167189 -0800
@@ -9,7 +9,6 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <sys/time.h>
-#include "snprintf.h"
 #include "general.h"
 #include "xosview.h"
 #include "meter.h"
diff -Naur xosview-1.8.3.orig/xwin.cc xosview-1.8.3/xwin.cc
--- xosview-1.8.3.orig/xwin.cc	2007-12-01 10:37:19.239343207 -0800
+++ xosview-1.8.3/xwin.cc	2007-12-01 10:43:44.781313993 -0800
@@ -5,7 +5,6 @@
 #include <X11/Xatom.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include "snprintf.h"
 #include "general.h"
 #include "xwin.h"
 #include "Xrm.h"
