diff -r 15b32ce27a28 config/system-headers
--- config/system-headers	Thu Jul 21 20:50:16 2011 +1200
+++ config/system-headers	Thu Jul 21 21:28:49 2011 +1200
@@ -199,17 +199,16 @@ crypt.h
 cstddef
 cstdio
 cstdlib
 cstring
 ctime
 ctype.h
 curl/curl.h
 curl/easy.h
-curl/types.h
 curses.h
 cxxabi.h
 DateTimeUtils.h
 dbus/dbus.h
 dbus/dbus-glib.h
 dbus/dbus-glib-lowlevel.h
 ddeml.h
 Debug.h
diff -r 15b32ce27a28 js/src/config/system-headers
--- js/src/config/system-headers	Thu Jul 21 20:50:16 2011 +1200
+++ js/src/config/system-headers	Thu Jul 21 21:28:49 2011 +1200
@@ -199,17 +199,16 @@ crypt.h
 cstddef
 cstdio
 cstdlib
 cstring
 ctime
 ctype.h
 curl/curl.h
 curl/easy.h
-curl/types.h
 curses.h
 cxxabi.h
 DateTimeUtils.h
 dbus/dbus.h
 dbus/dbus-glib.h
 dbus/dbus-glib-lowlevel.h
 ddeml.h
 Debug.h
diff -r 15b32ce27a28 toolkit/crashreporter/google-breakpad/src/common/linux/http_upload.cc
--- toolkit/crashreporter/google-breakpad/src/common/linux/http_upload.cc	Thu Jul 21 20:50:16 2011 +1200
+++ toolkit/crashreporter/google-breakpad/src/common/linux/http_upload.cc	Thu Jul 21 21:28:49 2011 +1200
@@ -28,17 +28,16 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include "common/linux/http_upload.h"
 
 #include <assert.h>
 #include <dlfcn.h>
 #include <curl/curl.h>
 #include <curl/easy.h>
-#include <curl/types.h>
 
 namespace {
 
 // Callback to get the response data from server.
 static size_t WriteCallback(void *ptr, size_t size,
                             size_t nmemb, void *userp) {
   if (!userp)
     return 0;
diff -r 15b32ce27a28 toolkit/crashreporter/google-breakpad/src/common/linux/libcurl_wrapper.cc
--- toolkit/crashreporter/google-breakpad/src/common/linux/libcurl_wrapper.cc	Thu Jul 21 20:50:16 2011 +1200
+++ toolkit/crashreporter/google-breakpad/src/common/linux/libcurl_wrapper.cc	Thu Jul 21 21:28:49 2011 +1200
@@ -24,17 +24,16 @@
 // LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
 // DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
 // THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <curl/curl.h>
 #include <curl/easy.h>
-#include <curl/types.h>
 #include <dlfcn.h>
 
 #include <iostream>
 #include <string>
 
 #include "common/linux/libcurl_wrapper.h"
 
 using std::string;
