--- config/rules.mk
+++ config/rules.mk
@@ -57,16 +57,21 @@
 REPORT_BUILD = @echo $(notdir $<)
 
 ifeq ($(OS_ARCH),OS2)
 EXEC			=
 else
 EXEC			= exec
 endif
 
+# Don't copy xulrunner files at install time, when using system xulrunner
+ifdef LIBXUL_SDK
+  SKIP_COPY_XULRUNNER=1
+endif
+
 # ELOG prints out failed command when building silently (gmake -s).
 ifneq (,$(findstring -s,$(MAKEFLAGS)))
   ELOG := $(EXEC) sh $(BUILD_TOOLS)/print-failed-commands.sh
 else
   ELOG :=
 endif
 
 ifeq ($(MOZ_OS2_TOOLS),VACPP)
