--- config/rules.mk.orig	2009-06-30 09:00:33.000000000 -0500
+++ config/rules.mk	2009-06-30 09:00:57.000000000 -0500
@@ -67,6 +67,11 @@
   SKIP_COPY_XULRUNNER=1
 endif
 
+# Don't copy xulrunner files at install time, when using system xulrunner
+ifdef LIBXUL_SDK
+  SKIP_COPY_XULRUNNER=1
+endif
+
 # ELOG prints out failed command when building silently (gmake -s).
 ifneq (,$(findstring -s,$(MAKEFLAGS)))
   ELOG := $(EXEC) sh $(BUILD_TOOLS)/print-failed-commands.sh
