diff -up browser/base/Makefile.in.fu browser/base/Makefile.in
--- browser/base/Makefile.in.fu	2011-07-01 22:40:52.048773336 -0300
+++ browser/base/Makefile.in	2011-07-01 22:42:30.866213879 -0300
@@ -62,10 +62,9 @@ EXTRA_JS_MODULES = \
 
 include $(topsrcdir)/config/rules.mk
 
-PRE_RELEASE_SUFFIX := ""
+PRE_RELEASE_SUFFIX :=
 
 DEFINES += \
-	-DMOZ_APP_VERSION=$(MOZ_APP_VERSION) \
 	-DAPP_LICENSE_BLOCK=$(abs_srcdir)/content/overrides/app-license.html \
 	-DPRE_RELEASE_SUFFIX="$(PRE_RELEASE_SUFFIX)" \
 	$(NULL)
