From f08c2e802224e5824b6f251553aa46c9c23dbcfc Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Mon, 28 Apr 2014 22:34:23 +0200
Subject: [PATCH] fix install path

---
 config/baseconfig.mk | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/config/baseconfig.mk b/config/baseconfig.mk
index b4723a9..8417918 100644
--- a/config/baseconfig.mk
+++ b/config/baseconfig.mk
@@ -2,10 +2,10 @@
 # directly in python/mozbuild/mozbuild/base.py for gmake validation.
 # We thus use INCLUDED_AUTOCONF_MK to enable/disable some parts depending
 # whether a normal build is happening or whether the check is running.
-includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+includedir := $(includedir)/$(MOZ_APP_NAME)
+idldir = $(datadir)/idl/$(MOZ_APP_NAME)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
 ifndef TOP_DIST
 TOP_DIST = dist
 endif
-- 
1.9.1

