From a53a38ef122f25d6ee65e1a418c9de08a13eb81b Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 2 Jun 2012 11:20:29 +0200
Subject: [PATCH] removed manpage installation from setup.py to avoid install
 failure

---
 setup.py |    1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index e2786cc..fc7e96e 100644
--- a/setup.py
+++ b/setup.py
@@ -36,5 +36,4 @@ benchmark backup software than distributing very large test sets.
       license='GNU General Public License, version 3 or later',
       packages=['genbackupdatalib'],
       scripts=['genbackupdata'],
-      data_files=[('share/man/man1', ['genbackupdata.1'])],
      )
-- 
1.7.10.1

