--- installer/core_install.py.orig	2010-05-13 03:45:01.000000000 +0400
+++ installer/core_install.py	2010-07-09 13:50:43.374678083 +0400
@@ -1990,7 +1990,7 @@
         if mode == GUI_MODE:
             return os.system("sh %s --nox11 -- -u" % plugin_file) == 0
         else:
-            return os.system("sh %s --nox11 -- -i" % plugin_file) == 0
+            return os.system("sh %s --nox11" % plugin_file) == 0
 
 
     def delete_plugin(self):
