--- base/module.py.orig
+++ base/module.py
@@ -263,18 +263,17 @@
             params = ''.join([params, 'd:p:P:'])
             long_params.extend(['device=', 'device-uri=', 'printer=', 'printer-name'])
 
-        if self.num_valid_modes > 1:
-            if GUI_MODE in self.avail_modes and prop.gui_build:
-                params = ''.join([params, 'u'])
-                long_params.extend(['gui', 'ui'])
-
-            if INTERACTIVE_MODE in self.avail_modes:
-                params = ''.join([params, 'i'])
-                long_params.extend(['interactive', 'text'])
-
-            if NON_INTERACTIVE_MODE in self.avail_modes:
-                params = ''.join([params, 'n'])
-                long_params.extend(['noninteractive', 'non-interactive', 'batch'])
+        if GUI_MODE in self.avail_modes and prop.gui_build:
+            params = ''.join([params, 'u'])
+            long_params.extend(['gui', 'ui'])
+            
+        if INTERACTIVE_MODE in self.avail_modes:
+            params = ''.join([params, 'i'])
+            long_params.extend(['interactive', 'text'])
+    
+        if NON_INTERACTIVE_MODE in self.avail_modes:
+            params = ''.join([params, 'n'])
+            long_params.extend(['noninteractive', 'non-interactive', 'batch'])
 
         if self.supported_ui_toolkits is not None and \
             self.num_supported_ui_toolkits >= 1 and prop.gui_build and \
