From: Ismael Luceno <ismael@iodev.co.uk>
Subject: Update OpenEXR API

diff --git a/Source/FreeImage/PluginEXR.cpp b/Source/FreeImage/PluginEXR.cpp
index a70739cc0e67..decc3a01ae10 100644
--- a/Source/FreeImage/PluginEXR.cpp
+++ b/Source/FreeImage/PluginEXR.cpp
@@ -66,12 +66,12 @@ public:
 		return ((unsigned)n != _io->read_proc(c, 1, n, _handle));
 	}
 
-	virtual Imath::Int64 tellg() {
+	virtual uint64_t tellg() {
 		return _io->tell_proc(_handle);
 	}
 
-	virtual void seekg(Imath::Int64 pos) {
-		_io->seek_proc(_handle, (unsigned)pos, SEEK_SET);
+	virtual void seekg(uint64_t pos) {
+		_io->seek_proc(_handle, pos, SEEK_SET);
 	}
 
 	virtual void clear() {
@@ -100,11 +100,11 @@ public:
 		}
 	}
 
-	virtual Imath::Int64 tellp() {
+	virtual uint64_t tellp() {
 		return _io->tell_proc(_handle);
 	}
 
-	virtual void seekp(Imath::Int64 pos) {
+	virtual void seekp(uint64_t pos) {
 		_io->seek_proc(_handle, (unsigned)pos, SEEK_SET);
 	}
 };
