--- src/MainWindow.cpp.orig	2014-12-09 11:50:59.000000000 -0500
+++ src/MainWindow.cpp	2018-06-19 12:28:18.710651146 -0400
@@ -94,6 +94,7 @@
 {
     Config* config = Config::getInstance();
     gint xpos, ypos;
+    GtkRequisition req;
     widget = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 
     gtk_window_set_title(GTK_WINDOW(widget), "gsimplecal");
@@ -127,6 +128,9 @@
         timezones = new Timezones();
         timezones->updateTime();
         timezones->addToBox(children_box);
+
+        gtk_widget_size_request(children_box, &req);
+        gtk_widget_set_size_request(children_box, -1, req.height + 5);
     }
 
     gtk_container_add(GTK_CONTAINER(widget), children_box);
