diff -ur eflwebkit-svn/Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj eflwebkit-mod/Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj
--- eflwebkit-svn/Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj	2011-11-03 19:04:16.000000000 -0500
+++ eflwebkit-mod/Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj	2011-11-05 22:32:15.211587478 -0500
@@ -2955,7 +2955,7 @@
 				651122F814046A4C002B101D /* PBXTargetDependency */,
 			);
 			name = testRegExp;
-			productInstallPath = /usr/local/bin;
+			productInstallPath = /usr/bin;
 			productName = jsc;
 			productReference = 6511230514046A4C002B101D /* testRegExp */;
 			productType = "com.apple.product-type.tool";
@@ -2998,7 +2998,7 @@
 				14270B0C0A451DA40080EEEA /* PBXTargetDependency */,
 			);
 			name = jsc;
-			productInstallPath = /usr/local/bin;
+			productInstallPath = /usr/bin;
 			productName = jsc;
 			productReference = 932F5BE10822A1C700736975 /* jsc */;
 			productType = "com.apple.product-type.tool";
diff -ur eflwebkit-svn/Source/ThirdParty/ANGLE/Configurations/ANGLE.xcconfig eflwebkit-mod/Source/ThirdParty/ANGLE/Configurations/ANGLE.xcconfig
--- eflwebkit-svn/Source/ThirdParty/ANGLE/Configurations/ANGLE.xcconfig	2011-11-03 19:06:23.000000000 -0500
+++ eflwebkit-mod/Source/ThirdParty/ANGLE/Configurations/ANGLE.xcconfig	2011-11-06 06:50:45.983666773 -0600
@@ -1,5 +1,5 @@
 PRODUCT_NAME = ANGLE;
 
 HEADER_SEARCH_PATHS = include src;
-INSTALL_PATH = /usr/local/lib;
-PUBLIC_HEADERS_FOLDER_PATH = /usr/local/include/ANGLE;
+INSTALL_PATH = /usr/lib;
+PUBLIC_HEADERS_FOLDER_PATH = /usr/include/ANGLE;
diff -ur eflwebkit-svn/Source/ThirdParty/gtest/xcode/Samples/FrameworkSample/WidgetFramework.xcodeproj/project.pbxproj eflwebkit-mod/Source/ThirdParty/gtest/xcode/Samples/FrameworkSample/WidgetFramework.xcodeproj/project.pbxproj
--- eflwebkit-svn/Source/ThirdParty/gtest/xcode/Samples/FrameworkSample/WidgetFramework.xcodeproj/project.pbxproj	2011-11-03 19:06:24.000000000 -0500
+++ eflwebkit-mod/Source/ThirdParty/gtest/xcode/Samples/FrameworkSample/WidgetFramework.xcodeproj/project.pbxproj	2011-11-06 06:52:16.932661363 -0600
@@ -78,8 +78,8 @@
 		3B7EB1230E5AEE3500C7F239 /* widget.cc */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; path = widget.cc; sourceTree = "<group>"; };
 		3B7EB1240E5AEE3500C7F239 /* widget.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = widget.h; sourceTree = "<group>"; };
 		3B7EB1270E5AEE4600C7F239 /* widget_test.cc */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; path = widget_test.cc; sourceTree = "<group>"; };
-		4024D183113D7D5500C7059E /* libgtest_main.a */ = {isa = PBXFileReference; lastKnownFileType = archive.ar; name = libgtest_main.a; path = /usr/local/lib/libgtest_main.a; sourceTree = "<absolute>"; };
-		4024D185113D7D5500C7059E /* libgtest.a */ = {isa = PBXFileReference; lastKnownFileType = archive.ar; name = libgtest.a; path = /usr/local/lib/libgtest.a; sourceTree = "<absolute>"; };
+		4024D183113D7D5500C7059E /* libgtest_main.a */ = {isa = PBXFileReference; lastKnownFileType = archive.ar; name = libgtest_main.a; path = /usr/lib/libgtest_main.a; sourceTree = "<absolute>"; };
+		4024D185113D7D5500C7059E /* libgtest.a */ = {isa = PBXFileReference; lastKnownFileType = archive.ar; name = libgtest.a; path = /usr/lib/libgtest.a; sourceTree = "<absolute>"; };
 		4024D1E2113D838200C7059E /* runtests.sh */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.script.sh; path = runtests.sh; sourceTree = "<group>"; };
 		8D07F2C70486CC7A007CD1D0 /* Info.plist */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist; path = Info.plist; sourceTree = "<group>"; };
 		8D07F2C80486CC7A007CD1D0 /* Widget.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Widget.framework; sourceTree = BUILT_PRODUCTS_DIR; };
diff -ur eflwebkit-svn/Source/ThirdParty/gyp/pylib/gyp/generator/xcode.py eflwebkit-mod/Source/ThirdParty/gyp/pylib/gyp/generator/xcode.py
--- eflwebkit-svn/Source/ThirdParty/gyp/pylib/gyp/generator/xcode.py	2011-11-03 19:06:20.000000000 -0500
+++ eflwebkit-mod/Source/ThirdParty/gyp/pylib/gyp/generator/xcode.py	2011-11-06 07:58:15.230424139 -0600
@@ -86,7 +86,7 @@
 # in LIBRARY_SEARCH_PATHS. This list is not exhaustive, but that's okay.
 xcode_standard_library_dirs = frozenset([
   '$(SDKROOT)/usr/lib',
-  '$(SDKROOT)/usr/local/lib',
+  '$(SDKROOT)/usr/lib',
 ])
 
 def CreateXCConfigurationList(configuration_names):
diff -ur eflwebkit-svn/Source/WebCore/Configurations/Base.xcconfig eflwebkit-mod/Source/WebCore/Configurations/Base.xcconfig
--- eflwebkit-svn/Source/WebCore/Configurations/Base.xcconfig	2011-11-03 19:06:18.000000000 -0500
+++ eflwebkit-mod/Source/WebCore/Configurations/Base.xcconfig	2011-11-06 07:59:44.362418454 -0600
@@ -99,7 +99,7 @@
 
 SECTORDER_FLAGS = -Wl,-order_file,WebCore.order;
 
-WEBCORE_SQLITE3_HEADER_SEARCH_PATHS = $(NEXT_ROOT)/usr/local/include/WebCoreSQLite3;
+WEBCORE_SQLITE3_HEADER_SEARCH_PATHS = $(NEXT_ROOT)/usr/include/WebCoreSQLite3;
 SQLITE3_HEADER_SEARCH_PATHS = $(SQLITE3_HEADER_SEARCH_PATHS_$(REAL_PLATFORM_NAME));
 SQLITE3_HEADER_SEARCH_PATHS_macosx = $(SQLITE3_HEADER_SEARCH_PATHS_macosx_$(TARGET_MAC_OS_X_VERSION_MAJOR));
 SQLITE3_HEADER_SEARCH_PATHS_macosx_1050 = $(WEBCORE_SQLITE3_HEADER_SEARCH_PATHS);
diff -ur eflwebkit-svn/Source/WebCore/Configurations/WebCoreTestSupport.xcconfig eflwebkit-mod/Source/WebCore/Configurations/WebCoreTestSupport.xcconfig
--- eflwebkit-svn/Source/WebCore/Configurations/WebCoreTestSupport.xcconfig	2011-11-03 19:06:18.000000000 -0500
+++ eflwebkit-mod/Source/WebCore/Configurations/WebCoreTestSupport.xcconfig	2011-11-06 08:00:34.289415807 -0600
@@ -26,10 +26,10 @@
 PRIVATE_HEADERS_FOLDER_PATH = $(PRIVATE_HEADERS_FOLDER_PATH_$(CONFIGURATION));
 PRIVATE_HEADERS_FOLDER_PATH_Debug = WebCoreTestSupport;
 PRIVATE_HEADERS_FOLDER_PATH_Release = $(PRIVATE_HEADERS_FOLDER_PATH_Debug);
-PRIVATE_HEADERS_FOLDER_PATH_Production = /usr/local/include/WebCoreTestSupport;
+PRIVATE_HEADERS_FOLDER_PATH_Production = /usr/include/WebCoreTestSupport;
 
 INSTALL_PATH = $(INSTALL_PATH_$(CONFIGURATION));
-INSTALL_PATH_Production = /usr/local/lib;
+INSTALL_PATH_Production = /usr/lib;
 
 LD_DYLIB_INSTALL_NAME = $(LD_DYLIB_INSTALL_NAME_$(CONFIGURATION));
 LD_DYLIB_INSTALL_NAME_Debug = @rpath/$(EXECUTABLE_PATH);
diff -ur eflwebkit-svn/Source/WebCore/Configurations/WebCore.xcconfig eflwebkit-mod/Source/WebCore/Configurations/WebCore.xcconfig
--- eflwebkit-svn/Source/WebCore/Configurations/WebCore.xcconfig	2011-11-03 19:06:18.000000000 -0500
+++ eflwebkit-mod/Source/WebCore/Configurations/WebCore.xcconfig	2011-11-06 07:59:11.612420687 -0600
@@ -43,7 +43,7 @@
 FRAMEWORK_SEARCH_PATHS_iphonesimulator = $(FRAMEWORK_SEARCH_PATHS_iphoneos_$(CONFIGURATION));
 FRAMEWORK_SEARCH_PATHS_macosx = $(SYSTEM_LIBRARY_DIR)/Frameworks/Carbon.framework/Frameworks $(SYSTEM_LIBRARY_DIR)/Frameworks/ApplicationServices.framework/Frameworks $(SYSTEM_LIBRARY_DIR)/Frameworks/CoreServices.framework/Frameworks $(FRAMEWORK_SEARCH_PATHS);
 
-HEADER_SEARCH_PATHS = ForwardingHeaders icu /usr/include/libxslt /usr/include/libxml2 $(SQLITE3_HEADER_SEARCH_PATHS) "${BUILT_PRODUCTS_DIR}/DerivedSources/WebCore" "${BUILT_PRODUCTS_DIR}/usr/local/include" $(HEADER_SEARCH_PATHS);
+HEADER_SEARCH_PATHS = ForwardingHeaders icu /usr/include/libxslt /usr/include/libxml2 $(SQLITE3_HEADER_SEARCH_PATHS) "${BUILT_PRODUCTS_DIR}/DerivedSources/WebCore" "${BUILT_PRODUCTS_DIR}/usr/include" $(HEADER_SEARCH_PATHS);
 INFOPLIST_FILE = Info.plist;
 INSTALL_PATH = $(INSTALL_PATH_$(REAL_PLATFORM_NAME));
 INSTALL_PATH_macosx = $(SYSTEM_LIBRARY_DIR)/Frameworks/WebKit.framework/Versions/A/Frameworks;
diff -ur eflwebkit-svn/Source/WebCore/gyp/WebCore.gyp eflwebkit-mod/Source/WebCore/gyp/WebCore.gyp
--- eflwebkit-svn/Source/WebCore/gyp/WebCore.gyp	2011-11-03 19:05:15.000000000 -0500
+++ eflwebkit-mod/Source/WebCore/gyp/WebCore.gyp	2011-11-06 08:02:13.293409885 -0600
@@ -38,7 +38,7 @@
         '<(project_dir)',
         '<(project_dir)/icu',
         '<(project_dir)/ForwardingHeaders',
-        '<(PRODUCT_DIR)/usr/local/include',
+        '<(PRODUCT_DIR)/usr/include',
         '/usr/include/libxml2',
         '<(PRODUCT_DIR)/DerivedSources',
         '<(PRODUCT_DIR)/DerivedSources/WebCore',
@@ -254,7 +254,7 @@
         'Production': {
             'EXPORTED_SYMBOLS_FILE': '',
             'GCC_OPTIMIZATION_LEVEL': '0',
-            'INSTALL_PATH': '/usr/local/bin',
+            'INSTALL_PATH': '/usr/bin',
             'OTHER_LDFLAGS': '',
             'SKIP_INSTALL': 'YES',
         },
@@ -262,7 +262,7 @@
           'xcode_settings': {
             'EXPORTED_SYMBOLS_FILE': '',
             'GCC_OPTIMIZATION_LEVEL': '0',
-            'INSTALL_PATH': '/usr/local/bin',
+            'INSTALL_PATH': '/usr/bin',
             'OTHER_LDFLAGS': '',
             'SKIP_INSTALL': 'YES',
           },
@@ -271,7 +271,7 @@
           'xcode_settings': {
             'EXPORTED_SYMBOLS_FILE': '',
             'GCC_OPTIMIZATION_LEVEL': '0',
-            'INSTALL_PATH': '/usr/local/bin',
+            'INSTALL_PATH': '/usr/bin',
             'OTHER_LDFLAGS': '',
             'SKIP_INSTALL': 'YES',
           },
diff -ur eflwebkit-svn/Source/WebCore/plugins/PluginDatabase.cpp eflwebkit-mod/Source/WebCore/plugins/PluginDatabase.cpp
--- eflwebkit-svn/Source/WebCore/plugins/PluginDatabase.cpp	2011-11-03 19:05:12.000000000 -0500
+++ eflwebkit-mod/Source/WebCore/plugins/PluginDatabase.cpp	2011-11-06 08:03:24.219405715 -0600
@@ -376,12 +376,12 @@
     paths.append(userPluginPath);
 
     paths.append("/usr/lib/browser/plugins");
-    paths.append("/usr/local/lib/mozilla/plugins");
+    paths.append("/usr/lib/mozilla/plugins");
     paths.append("/usr/lib/firefox/plugins");
     paths.append("/usr/lib64/browser-plugins");
     paths.append("/usr/lib/browser-plugins");
     paths.append("/usr/lib/mozilla/plugins");
-    paths.append("/usr/local/netscape/plugins");
+    paths.append("/usr/netscape/plugins");
     paths.append("/opt/mozilla/plugins");
     paths.append("/opt/mozilla/lib/plugins");
     paths.append("/opt/netscape/plugins");
diff -ur eflwebkit-svn/Source/WebCore/WebCore.xcodeproj/project.pbxproj eflwebkit-mod/Source/WebCore/WebCore.xcodeproj/project.pbxproj
--- eflwebkit-svn/Source/WebCore/WebCore.xcodeproj/project.pbxproj	2011-11-05 21:43:07.000000000 -0500
+++ eflwebkit-mod/Source/WebCore/WebCore.xcodeproj/project.pbxproj	2011-11-06 08:01:24.275412808 -0600
@@ -27633,7 +27633,7 @@
 			buildSettings = {
 				EXPORTED_SYMBOLS_FILE = "";
 				GCC_OPTIMIZATION_LEVEL = 0;
-				INSTALL_PATH = /usr/local/bin;
+				INSTALL_PATH = /usr/bin;
 				OTHER_LDFLAGS = "";
 				PRODUCT_NAME = WebCoreExportFileGenerator;
 				SKIP_INSTALL = YES;
@@ -27645,7 +27645,7 @@
 			baseConfigurationReference = 1CDD45E50BA9C84600F90147 /* WebCore.xcconfig */;
 			buildSettings = {
 				EXPORTED_SYMBOLS_FILE = "";
-				INSTALL_PATH = /usr/local/bin;
+				INSTALL_PATH = /usr/bin;
 				OTHER_LDFLAGS = "";
 				PRODUCT_NAME = WebCoreExportFileGenerator;
 				SKIP_INSTALL = YES;
@@ -27657,7 +27657,7 @@
 			baseConfigurationReference = 1CDD45E50BA9C84600F90147 /* WebCore.xcconfig */;
 			buildSettings = {
 				EXPORTED_SYMBOLS_FILE = "";
-				INSTALL_PATH = /usr/local/bin;
+				INSTALL_PATH = /usr/bin;
 				OTHER_LDFLAGS = "";
 				PRODUCT_NAME = WebCoreExportFileGenerator;
 				SKIP_INSTALL = YES;
diff -ur eflwebkit-svn/Source/WebKit/mac/Configurations/WebKit.xcconfig eflwebkit-mod/Source/WebKit/mac/Configurations/WebKit.xcconfig
--- eflwebkit-svn/Source/WebKit/mac/Configurations/WebKit.xcconfig	2011-11-03 19:04:53.000000000 -0500
+++ eflwebkit-mod/Source/WebKit/mac/Configurations/WebKit.xcconfig	2011-11-06 08:04:50.679400454 -0600
@@ -87,6 +87,6 @@
 WEBKITSYSTEMINTERFACE_STATIC_LIBRARY_HEADERS_FOLDER_PATH = $(WEBKITSYSTEMINTERFACE_STATIC_LIBRARY_HEADERS_FOLDER_PATH_$(CONFIGURATION));
 WEBKITSYSTEMINTERFACE_STATIC_LIBRARY_HEADERS_FOLDER_PATH_Debug = $(WEBKITSYSTEMINTERFACE_STATIC_LIBRARY_HEADERS_FOLDER_PATH_engineering);
 WEBKITSYSTEMINTERFACE_STATIC_LIBRARY_HEADERS_FOLDER_PATH_Release = $(WEBKITSYSTEMINTERFACE_STATIC_LIBRARY_HEADERS_FOLDER_PATH_engineering);
-WEBKITSYSTEMINTERFACE_STATIC_LIBRARY_HEADERS_FOLDER_PATH_Production = $(PRODUCTION_ROOT)/usr/local/include;
-WEBKITSYSTEMINTERFACE_STATIC_LIBRARY_HEADERS_FOLDER_PATH_engineering = $(BUILT_PRODUCTS_DIR)/usr/local/include;
+WEBKITSYSTEMINTERFACE_STATIC_LIBRARY_HEADERS_FOLDER_PATH_Production = $(PRODUCTION_ROOT)/usr/include;
+WEBKITSYSTEMINTERFACE_STATIC_LIBRARY_HEADERS_FOLDER_PATH_engineering = $(BUILT_PRODUCTS_DIR)/usr/include;
 
diff -ur eflwebkit-svn/Source/WebKit/qt/docs/qtwebkit.qdoc eflwebkit-mod/Source/WebKit/qt/docs/qtwebkit.qdoc
--- eflwebkit-svn/Source/WebKit/qt/docs/qtwebkit.qdoc	2011-11-03 19:04:42.000000000 -0500
+++ eflwebkit-mod/Source/WebKit/qt/docs/qtwebkit.qdoc	2011-11-06 08:04:09.697402866 -0600
@@ -114,12 +114,12 @@
     \o System locations, such as
     \list
     \o \c{/usr/lib/browser/plugins}
-    \o \c{/usr/local/lib/mozilla/plugins}
+    \o \c{/usr/lib/mozilla/plugins}
     \o \c{/usr/lib/firefox/plugins}
     \o \c{/usr/lib64/browser-plugins}
     \o \c{/usr/lib/browser-plugins}
     \o \c{/usr/lib/mozilla/plugins}
-    \o \c{/usr/local/netscape/plugins}
+    \o \c{/usr/netscape/plugins}
     \o \c{/opt/mozilla/plugins}
     \o \c{/opt/mozilla/lib/plugins}
     \o \c{/opt/netscape/plugins}
diff -ur eflwebkit-svn/Source/WebKit2/Configurations/BaseTarget.xcconfig eflwebkit-mod/Source/WebKit2/Configurations/BaseTarget.xcconfig
--- eflwebkit-svn/Source/WebKit2/Configurations/BaseTarget.xcconfig	2011-11-03 19:06:31.000000000 -0500
+++ eflwebkit-mod/Source/WebKit2/Configurations/BaseTarget.xcconfig	2011-11-06 08:05:24.406398420 -0600
@@ -27,7 +27,7 @@
 FRAMEWORK_SEARCH_PATHS = $(UMBRELLA_FRAMEWORKS_DIR) $(SYSTEM_LIBRARY_DIR)/Frameworks/ApplicationServices.framework/Frameworks $(SYSTEM_LIBRARY_DIR)/Frameworks/Carbon.framework/Frameworks $(SYSTEM_LIBRARY_DIR)/Frameworks/Quartz.framework/Frameworks $(SYSTEM_LIBRARY_DIR)/Frameworks/CoreServices.framework/Frameworks $(FRAMEWORK_SEARCH_PATHS);
 GCC_PREFIX_HEADER = WebKit2Prefix.h;
 GCC_PREPROCESSOR_DEFINITIONS = $(DEBUG_DEFINES) $(FEATURE_DEFINES) FRAMEWORK_NAME=WebKit2
-HEADER_SEARCH_PATHS = $(WEBCORE_PRIVATE_HEADERS_DIR)/ForwardingHeaders $(WEBCORE_PRIVATE_HEADERS_DIR)/icu $(BUILT_PRODUCTS_DIR)/usr/local/include $(BUILT_PRODUCTS_DIR)/DerivedSources/WebKit2 $(HEADER_SEARCH_PATHS);
+HEADER_SEARCH_PATHS = $(WEBCORE_PRIVATE_HEADERS_DIR)/ForwardingHeaders $(WEBCORE_PRIVATE_HEADERS_DIR)/icu $(BUILT_PRODUCTS_DIR)/usr/include $(BUILT_PRODUCTS_DIR)/DerivedSources/WebKit2 $(HEADER_SEARCH_PATHS);
 
 UMBRELLA_FRAMEWORKS_DIR = $(UMBRELLA_FRAMEWORKS_DIR_$(REAL_PLATFORM_NAME)_$(CONFIGURATION));
 UMBRELLA_FRAMEWORKS_DIR_macosx_Release = $(UMBRELLA_FRAMEWORKS_DIR_engineering);
