--- util.c.orig	2004-02-08 17:42:33.000000000 +0400
+++ util.c	2012-07-24 19:32:31.725182191 +0400
@@ -19,7 +19,7 @@
 
 GtkWidget* unfocussable(GtkWidget *w)
 {
-	GTK_WIDGET_UNSET_FLAGS(w, GTK_CAN_FOCUS);
+	gtk_widget_set_can_focus(w, TRUE);
 	return w;
 }
 
--- theme_sel.c.orig	2004-02-09 06:20:28.000000000 +0400
+++ theme_sel.c	2012-07-24 19:37:24.000000000 +0400
@@ -110,7 +110,7 @@
 		GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
 	gtk_container_add(GTK_CONTAINER(sw), theme_list);
 
-	gtk_timeout_add(0, (GtkFunction)theme_list_focus, theme_list);
+	gtk_timeout_add(0, (GSourceFunc)theme_list_focus, theme_list);
 
 	return sw;
 }
