From b247ce0e8745104a383d124e37378eb6c705d14e Mon Sep 17 00:00:00 2001
From: Vlad Glagolev <stealth@sourcemage.org>
Date: Wed, 30 Jul 2008 23:49:53 +0400
Subject: [PATCH 06/23] Honour NEED_LIBCRYPT in sysvtools/Makefile

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 sysvtools/Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sysvtools/Makefile b/sysvtools/Makefile
index 64af14f74f11..53af34ce47f1 100644
--- a/sysvtools/Makefile
+++ b/sysvtools/Makefile
@@ -8,6 +8,8 @@
 # Version:	@(#)Makefile  2.85-13  23-Mar-2004  miquels@cistron.nl
 #
 
+include ../make_include
+
 CC	= gcc
 CFLAGS	= -Wall -O2 -fomit-frame-pointer -D_GNU_SOURCE
 LDFLAGS	= -s
@@ -36,7 +38,7 @@ INSTALL		= install -o $(BIN_OWNER) -g $(BIN_GROUP)
 MANDIR		= /usr/share/man
 
 # Additional libs for GNU libc.
-ifneq ($(wildcard /usr/lib/libcrypt.a),)
+ifeq "$(NEED_LIBCRYPT)" "yes"
 LCRYPT		= -lcrypt
 endif
 
-- 
2.44.0

