From 5f3637625bce6ee3b790ed7b86f798c06603e964 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Wed, 29 May 2019 07:02:21 +0200
Subject: [PATCH 10/23] Fix compiler flags setup for sysvtools/Makefile

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 sysvtools/Makefile | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/sysvtools/Makefile b/sysvtools/Makefile
index 53af34ce47f1..7546c7620303 100644
--- a/sysvtools/Makefile
+++ b/sysvtools/Makefile
@@ -10,10 +10,8 @@
 
 include ../make_include
 
-CC	= gcc
-CFLAGS	= -Wall -O2 -fomit-frame-pointer -D_GNU_SOURCE
-LDFLAGS	= -s
-STATIC	=
+CFLAGS	+= -D_GNU_SOURCE
+STATIC	 =
 
 # For some known distributions we do not build all programs, otherwise we do.
 BIN	=
-- 
2.44.0

