From 5bd831d439ace38e0c3a358e8bd3a1cc584ff9c1 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sat, 8 Jun 2024 20:57:28 +0200
Subject: [PATCH 14/23] Fix non-std fcntl.h include

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 sysvtools/last.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sysvtools/last.c b/sysvtools/last.c
index 70a142c034dc..7339d38989ca 100644
--- a/sysvtools/last.c
+++ b/sysvtools/last.c
@@ -19,7 +19,7 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #include <time.h>
 #include <stdio.h>
 #include <ctype.h>
-- 
2.44.0

