From 87b889f1fcb9d973d99279c2dc755a1d94d19acf Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sat, 8 Jun 2024 21:22:38 +0200
Subject: [PATCH 17/23] simpleinit: Avoid setting termenv var if unused

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 login-utils/simpleinit.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/login-utils/simpleinit.c b/login-utils/simpleinit.c
index 5c40a48f7a3f..cfd9f3b0564d 100644
--- a/login-utils/simpleinit.c
+++ b/login-utils/simpleinit.c
@@ -818,12 +818,11 @@ void read_inittab (void)
 #ifdef SPECIAL_CONSOLE_TERM
 	char tty[50];
 	struct stat stb;
-#endif
 	char *termenv;
-	
 	termenv = getenv("TERM");	/* set by kernel */
 	/* termenv = "vt100"; */
-			
+#endif
+
 	if(!(f = fopen(_path_inittab, "r"))) {
 		err(_("cannot open inittab\n"));
 		return;
-- 
2.44.0

