From 5b0247d66408a9507e3007ba0fb3dfb3402b587a Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sat, 8 Jun 2024 21:23:40 +0200
Subject: [PATCH 18/23] simpleinit: Fix off-by-one loop

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 login-utils/simpleinit.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/login-utils/simpleinit.c b/login-utils/simpleinit.c
index cfd9f3b0564d..5b3280a41638 100644
--- a/login-utils/simpleinit.c
+++ b/login-utils/simpleinit.c
@@ -832,9 +832,7 @@ void read_inittab (void)
 	i = 0;
 	while(!feof(f) && i < NUMCMD - 2) {
 		if(fgets(buf, CMDSIZ - 1, f) == 0) break;
-		buf[CMDSIZ-1] = 0;
-
-		for(k = 0; k < CMDSIZ && buf[k]; k++)
+		for(k = 0; k < CMDSIZ - 1 && buf[k]; k++)
 			if ((buf[k] == '#') || (buf[k] == '\n')) break;
 		buf[k]=0;
 		/*treat lines that start with whitespace as empty*/
-- 
2.44.0

