From 00b0ddfa46fc80e2d41fac2b59460308e9d646cd Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sat, 8 Jun 2024 21:56:49 +0200
Subject: [PATCH 21/23] shutdown: Fix buffer to accomodate actual message size

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 login-utils/shutdown.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/login-utils/shutdown.c b/login-utils/shutdown.c
index 2961d9174ba4..23fe24fca1e2 100644
--- a/login-utils/shutdown.c
+++ b/login-utils/shutdown.c
@@ -582,7 +582,7 @@ write_user(struct utmp *ut)
 	int fd;
 	int minutes, hours;
 	char term[40] = {'/','d','e','v','/',0};
-	char msg[100];
+	char msg[266];
 
 	minutes = timeout / 60;
 	hours = minutes / 60;
-- 
2.44.0

