--- simpleinit-msb-1.3/sysvtools/last.c.orig	2025-10-28 10:29:53.736016777 +0100
+++ simpleinit-msb-1.3/sysvtools/last.c	2025-10-28 10:30:31.958211757 +0100
@@ -268,7 +268,7 @@
 /*
  *	SIGINT handler
  */
-void int_handler()
+void int_handler(int)
 {
 	printf("Interrupted %s\n", showdate());
 	exit(1);
@@ -277,7 +277,7 @@
 /*
  *	SIGQUIT handler
  */
-void quit_handler()
+void quit_handler(int)
 {
 	printf("Interrupted %s\n", showdate());
 	signal(SIGQUIT, quit_handler);
--- simpleinit-msb-1.3/sysvtools/sulogin.c.orig	2025-10-28 10:28:52.854892279 +0100
+++ simpleinit-msb-1.3/sysvtools/sulogin.c	2025-10-28 10:29:29.089180807 +0100
@@ -87,7 +87,7 @@
 /*
  *	Called at timeout.
  */
-void alrm_handler()
+void alrm_handler(int)
 {
 }
 
