diff --git a/src/3rdparty/chromium/base/i18n/file_util_icu.cc b/src/3rdparty/chromium/base/i18n/file_util_icu.cc
index f7525adf816..7085a51fbe3 100644
--- a/src/3rdparty/chromium/base/i18n/file_util_icu.cc
+++ b/src/3rdparty/chromium/base/i18n/file_util_icu.cc
@@ -22,6 +22,7 @@
 #include "build/build_config.h"
 #include "third_party/icu/source/common/unicode/uniset.h"
 #include "third_party/icu/source/i18n/unicode/coll.h"
+#include "third_party/icu/source/common/unicode/utf8.h"
 
 namespace base {
 namespace i18n {
diff --git a/src/3rdparty/chromium/base/i18n/rtl.cc b/src/3rdparty/chromium/base/i18n/rtl.cc
index 62c2d3f5cdf..5efd76a783e 100644
--- a/src/3rdparty/chromium/base/i18n/rtl.cc
+++ b/src/3rdparty/chromium/base/i18n/rtl.cc
@@ -23,6 +23,7 @@
 #include "third_party/icu/source/common/unicode/uchar.h"
 #include "third_party/icu/source/common/unicode/uscript.h"
 #include "third_party/icu/source/i18n/unicode/coll.h"
+#include "third_party/icu/source/common/unicode/utf16.h"
 
 #if defined(OS_IOS)
 #include "base/debug/crash_logging.h"
diff --git a/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/wtf_string.h b/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/wtf_string.h
index d7a6b487484..8709861b1bf 100644
--- a/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/wtf_string.h
+++ b/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/wtf_string.h
@@ -35,6 +35,7 @@
 #include "third_party/blink/renderer/platform/wtf/text/string_view.h"
 #include "third_party/blink/renderer/platform/wtf/wtf_export.h"
 #include "third_party/blink/renderer/platform/wtf/wtf_size_t.h"
+#include "third_party/icu/source/common/unicode/utf.h"
 
 #ifdef __OBJC__
 #include <objc/objc.h>
diff --git a/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/string_view.h b/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/string_view.h
index 0178c7b0c59..5d67f67d5a0 100644
--- a/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/string_view.h
+++ b/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/string_view.h
@@ -15,6 +15,7 @@
 #include "third_party/blink/renderer/platform/wtf/std_lib_extras.h"
 #include "third_party/blink/renderer/platform/wtf/text/string_impl.h"
 #include "third_party/blink/renderer/platform/wtf/text/unicode.h"
+#include "third_party/icu/source/common/unicode/utf16.h"
 
 namespace WTF {
 
