diff -uNr qtwebengine-5.15.10/src/3rdparty/chromium/base/i18n/file_util_icu.cc qtw-mod/src/3rdparty/chromium/base/i18n/file_util_icu.cc
--- qtwebengine-5.15.10/src/3rdparty/chromium/base/i18n/file_util_icu.cc	2022-05-23 13:38:40.000000000 +0200
+++ qtw-mod/src/3rdparty/chromium/base/i18n/file_util_icu.cc	2022-09-09 13:42:19.527652774 +0200
@@ -22,6 +22,7 @@
 #include "build/build_config.h"
 #include "third_party/icu/source/common/unicode/uniset.h"
 #include "third_party/icu/source/i18n/unicode/coll.h"
+#include "third_party/icu/source/common/unicode/utf8.h"
 
 namespace base {
 namespace i18n {
diff -uNr qtwebengine-5.15.10/src/3rdparty/chromium/base/i18n/rtl.cc qtw-mod/src/3rdparty/chromium/base/i18n/rtl.cc
--- qtwebengine-5.15.10/src/3rdparty/chromium/base/i18n/rtl.cc	2022-05-23 13:38:40.000000000 +0200
+++ qtw-mod/src/3rdparty/chromium/base/i18n/rtl.cc	2022-09-09 13:42:19.527652774 +0200
@@ -23,6 +23,7 @@
 #include "third_party/icu/source/common/unicode/uchar.h"
 #include "third_party/icu/source/common/unicode/uscript.h"
 #include "third_party/icu/source/i18n/unicode/coll.h"
+#include "third_party/icu/source/common/unicode/utf16.h"
 
 #if defined(OS_IOS)
 #include "base/debug/crash_logging.h"
diff -uNr qtwebengine-5.15.10/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/string_view.h qtw-mod/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/string_view.h
--- qtwebengine-5.15.10/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/string_view.h	2022-05-23 13:38:40.000000000 +0200
+++ qtw-mod/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/string_view.h	2022-09-09 13:42:19.527652774 +0200
@@ -15,6 +15,7 @@
 #include "third_party/blink/renderer/platform/wtf/std_lib_extras.h"
 #include "third_party/blink/renderer/platform/wtf/text/string_impl.h"
 #include "third_party/blink/renderer/platform/wtf/text/unicode.h"
+#include "third_party/icu/source/common/unicode/utf16.h"
 
 namespace WTF {
 
diff -uNr qtwebengine-5.15.10/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/text_codec_utf8.h qtw-mod/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/text_codec_utf8.h
--- qtwebengine-5.15.10/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/text_codec_utf8.h	2022-05-23 13:38:40.000000000 +0200
+++ qtw-mod/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/text_codec_utf8.h	2022-09-09 13:50:27.928617067 +0200
@@ -27,6 +27,7 @@
 #define THIRD_PARTY_BLINK_RENDERER_PLATFORM_WTF_TEXT_TEXT_CODEC_UTF8_H_
 
 #include <memory>
+#include "third_party/icu/source/common/unicode/utf8.h"
 #include "third_party/blink/renderer/platform/wtf/text/text_codec.h"
 
 namespace WTF {
diff -uNr qtwebengine-5.15.10/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/wtf_string.h qtw-mod/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/wtf_string.h
--- qtwebengine-5.15.10/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/wtf_string.h	2022-05-23 13:38:40.000000000 +0200
+++ qtw-mod/src/3rdparty/chromium/third_party/blink/renderer/platform/wtf/text/wtf_string.h	2022-09-09 13:42:19.527652774 +0200
@@ -35,6 +35,7 @@
 #include "third_party/blink/renderer/platform/wtf/text/string_view.h"
 #include "third_party/blink/renderer/platform/wtf/wtf_export.h"
 #include "third_party/blink/renderer/platform/wtf/wtf_size_t.h"
+#include "third_party/icu/source/common/unicode/utf.h"
 
 #ifdef __OBJC__
 #include <objc/objc.h>
