diff --git a/src/core/content_browser_client_qt.cpp b/src/core/content_browser_client_qt.cpp
index 30ec43ca..1949823f 100644
--- a/src/core/content_browser_client_qt.cpp
+++ b/src/core/content_browser_client_qt.cpp
@@ -68,6 +68,7 @@
 #include "content/public/common/main_function_params.h"
 #include "content/public/common/service_names.mojom.h"
 #include "content/public/common/url_constants.h"
+#include "device/geolocation/public/cpp/location_provider.h"
 #include "mojo/public/cpp/bindings/binding.h"
 #include "mojo/public/cpp/bindings/binding_set.h"
 #include "printing/features/features.h"
@@ -123,6 +124,10 @@
 #include "renderer_host/pepper/pepper_host_factory_qt.h"
 #endif
 
+#if defined(QT_USE_POSITIONING)
+#include "location_provider_qt.h"
+#endif
+
 #include <QGuiApplication>
 #include <QLocale>
 #ifndef QT_NO_OPENGL
@@ -737,7 +742,11 @@ bool ContentBrowserClientQt::CanCreateWindow(
 
 std::unique_ptr<device::LocationProvider> ContentBrowserClientQt::OverrideSystemLocationProvider()
 {
+#if defined(QT_USE_POSITIONING)
     return base::WrapUnique(new LocationProviderQt());
+#else
+    return nullptr;
+#endif
 }
 
 scoped_refptr<net::URLRequestContextGetter> GetSystemRequestContextOnUIThread()

