diff -Naur gcc-4.3.0.orig/gcc/config/i386/crtfastmath.c gcc-4.3.0/gcc/config/i386/crtfastmath.c
--- gcc-4.3.0.orig/gcc/config/i386/crtfastmath.c	2008-03-15 19:59:26.925738030 -0700
+++ gcc-4.3.0/gcc/config/i386/crtfastmath.c	2008-03-15 19:59:32.823482534 -0700
@@ -40,6 +40,23 @@
 #include "cpuid.h"
 #endif
 
+struct
+{
+	unsigned short int cwd;
+	unsigned short int swd;
+	unsigned short int twd;
+	unsigned short int fop;
+	long int fip;
+	long int fcs;
+	long int foo;
+	long int fos;
+	long int mxcsr;
+	long int mxcsr_mask;
+	long int st_space[32];
+	long int xmm_space[32];
+	long int padding[56];
+} __attribute__ ((aligned (16))) fxsave;
+
 static void __attribute__((constructor))
 #ifndef __x86_64__
 /* The i386 ABI only requires 4-byte stack alignment, so this is necessary
@@ -64,22 +81,6 @@
       if (edx & bit_FXSAVE)
 	{
 	  /* Check if DAZ is available.  */
-	  struct
-	    {
-	      unsigned short int cwd;
-	      unsigned short int swd;
-	      unsigned short int twd;
-	      unsigned short int fop;
-	      long int fip;
-	      long int fcs;
-	      long int foo;
-	      long int fos;
-	      long int mxcsr;
-	      long int mxcsr_mask;
-	      long int st_space[32];
-	      long int xmm_space[32];
-	      long int padding[56];
-	    } __attribute__ ((aligned (16))) fxsave;
 
 	  __builtin_memset (&fxsave, 0, sizeof (fxsave));
 
