--- gcc/config/i386/linux-unwind.h.orig	2012-10-03 15:45:48.914460075 +0400
+++ gcc/config/i386/linux-unwind.h	2012-10-03 15:46:46.432460072 +0400
@@ -133,9 +133,9 @@
     {
       struct rt_sigframe {
 	int sig;
-	struct siginfo *pinfo;
+	siginfo_t *pinfo;
 	void *puc;
-	struct siginfo info;
+	siginfo_t info;
 	struct ucontext uc;
       } *rt_ = context->cfa;
       /* The void * cast is necessary to avoid an aliasing warning.
