--- libstdc++-v3/python/Makefile.in.orig	2011-03-08 03:04:05.000000000 +0300
+++ libstdc++-v3/python/Makefile.in	2011-09-10 18:09:36.692249020 +0400
@@ -493,7 +493,7 @@
 	    -e 's,@toolexeclibdir@,$(toolexeclibdir),' < $(srcdir)/hook.in > $@
 
 install-data-local: gdb.py
-	@$(mkdir_p) $(DESTDIR)$(toolexeclibdir)
+	@$(mkdir_p) $(DESTDIR)$(datadir)/gdb/auto-load
 	@here=`pwd`; cd $(DESTDIR)$(toolexeclibdir); \
 	  for file in libstdc++*; do \
 	    case $$file in \
@@ -506,8 +506,8 @@
 	    esac; \
 	  done; \
 	cd $$here; \
-	echo " $(INSTALL_DATA) gdb.py $(DESTDIR)$(toolexeclibdir)/$$libname-gdb.py"; \
-	$(INSTALL_DATA) gdb.py $(DESTDIR)$(toolexeclibdir)/$$libname-gdb.py
+	echo " $(INSTALL_DATA) gdb.py $(DESTDIR)$(datadir)/gdb/auto-load/$$libname-gdb.py"; \
+	$(INSTALL_DATA) gdb.py $(DESTDIR)$(datadir)/gdb/auto-load/$$libname-gdb.py
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
