--- gcc-4.6.2/Makefile.in.orig	2012-02-04 22:17:43.000000000 +0100
+++ gcc-4.6.2/Makefile.in	2012-02-04 22:19:31.000000000 +0100
@@ -1848,7 +1848,6 @@
 install-info-target: maybe-install-info-target-libgloss
 install-info-target: maybe-install-info-target-gperf
 install-info-target: maybe-install-info-target-examples
-install-info-target: maybe-install-info-target-libffi
 install-info-target: maybe-install-info-target-libjava
 install-info-target: maybe-install-info-target-zlib
 install-info-target: maybe-install-info-target-boehm-gc
@@ -1963,7 +1962,6 @@
 install-pdf-target: maybe-install-pdf-target-libgloss
 install-pdf-target: maybe-install-pdf-target-gperf
 install-pdf-target: maybe-install-pdf-target-examples
-install-pdf-target: maybe-install-pdf-target-libffi
 install-pdf-target: maybe-install-pdf-target-libjava
 install-pdf-target: maybe-install-pdf-target-zlib
 install-pdf-target: maybe-install-pdf-target-boehm-gc
@@ -2078,7 +2076,6 @@
 install-html-target: maybe-install-html-target-libgloss
 install-html-target: maybe-install-html-target-gperf
 install-html-target: maybe-install-html-target-examples
-install-html-target: maybe-install-html-target-libffi
 install-html-target: maybe-install-html-target-libjava
 install-html-target: maybe-install-html-target-zlib
 install-html-target: maybe-install-html-target-boehm-gc
@@ -2193,7 +2190,6 @@
 installcheck-target: maybe-installcheck-target-libgloss
 installcheck-target: maybe-installcheck-target-gperf
 installcheck-target: maybe-installcheck-target-examples
-installcheck-target: maybe-installcheck-target-libffi
 installcheck-target: maybe-installcheck-target-libjava
 installcheck-target: maybe-installcheck-target-zlib
 installcheck-target: maybe-installcheck-target-boehm-gc
@@ -3045,7 +3041,6 @@
     maybe-install-target-libgloss \
     maybe-install-target-gperf \
     maybe-install-target-examples \
-    maybe-install-target-libffi \
     maybe-install-target-libjava \
     maybe-install-target-zlib \
     maybe-install-target-boehm-gc \
@@ -3179,7 +3174,6 @@
     maybe-install-strip-target-libgloss \
     maybe-install-strip-target-gperf \
     maybe-install-strip-target-examples \
-    maybe-install-strip-target-libffi \
     maybe-install-strip-target-libjava \
     maybe-install-strip-target-zlib \
     maybe-install-strip-target-boehm-gc \
