From 48a08719022e22094a8d3b786128f61b9a19ca19 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 23 Sep 2017 11:41:24 +0200
Subject: [PATCH] fix dependency on base

---
 vector.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vector.cabal b/vector.cabal
index 38c957d..767e373 100644
--- a/vector.cabal
+++ b/vector.cabal
@@ -144,7 +144,7 @@ Library
   Install-Includes:
         vector.h
 
-  Build-Depends: base >= 4.5 && < 4.10
+  Build-Depends: base >= 4.5 && < 4.11
                , primitive >= 0.5.0.1 && < 0.7
                , ghc-prim >= 0.2 && < 0.6
                , deepseq >= 1.1 && < 1.5
-- 
2.14.1

