--- a/src/access_io.c
+++ b/src/access_io.c
@@ -47,7 +47,7 @@
 #include "parport.h"
 #include "ppdev.h"
 
-#ifdef HAVE_LINUX
+#if defined(HAVE_LINUX) && defined(__GLIBC__)
 
 #ifdef HAVE_SYS_IO_H
 #include <sys/io.h>
@@ -62,7 +62,7 @@ struct iopbuf {
   unsigned char port_value;
 };
 
-#elif defined(HAVE_CYGWIN_9X)
+#elif defined(HAVE_LINUX) && !defined(__GLIBC__) || defined(HAVE_CYGWIN_9X)
 
 #include "io.h"
 
