diff -u -a -b -B -N -r doc.orig/Makefile.am doc/Makefile.am
--- doc.orig/Makefile.am	2006-01-04 17:24:37.000000000 -0600
+++ doc/Makefile.am	2006-01-04 17:22:44.000000000 -0600
@@ -45,7 +45,7 @@
 	cp `find html -name '*.html' -o -name '*.gif' -o -name '*.png'` \
 	  $(DESTDIR)$(datadir)/doc/libcaca-dev/html
 	$(mkinstalldirs) $(DESTDIR)$(mandir)/man3
-	cp man/man3/*.3caca $(DESTDIR)$(mandir)/man3
+	cp man/man3caca/*.3caca $(DESTDIR)$(mandir)/man3
 	$(mkinstalldirs) $(DESTDIR)$(mandir)/man1
 	rm -f $(DESTDIR)$(mandir)/man1/cacaball.1
 	ln -s cacademo.1 $(DESTDIR)$(mandir)/man1/cacaball.1
diff -u -a -b -B -N -r doc.orig/Makefile.in doc/Makefile.in
--- doc.orig/Makefile.in	2006-01-04 17:27:38.000000000 -0600
+++ doc/Makefile.in	2006-01-04 17:23:57.000000000 -0600
@@ -405,7 +405,7 @@
 @BUILD_DOCUMENTATION_TRUE@	cp `find html -name '*.html' -o -name '*.gif' -o -name '*.png'` \
 @BUILD_DOCUMENTATION_TRUE@	  $(DESTDIR)$(datadir)/doc/libcaca-dev/html
 @BUILD_DOCUMENTATION_TRUE@	$(mkinstalldirs) $(DESTDIR)$(mandir)/man3
-@BUILD_DOCUMENTATION_TRUE@	cp man/man3/*.3caca $(DESTDIR)$(mandir)/man3
+@BUILD_DOCUMENTATION_TRUE@	cp man/man3caca/*.3caca $(DESTDIR)$(mandir)/man3
 @BUILD_DOCUMENTATION_TRUE@	$(mkinstalldirs) $(DESTDIR)$(mandir)/man1
 @BUILD_DOCUMENTATION_TRUE@	rm -f $(DESTDIR)$(mandir)/man1/cacaball.1
 @BUILD_DOCUMENTATION_TRUE@	ln -s cacademo.1 $(DESTDIR)$(mandir)/man1/cacaball.1
