From 90ec62eef51f8e356327d730c71c191906a5287a Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 30 May 2013 14:24:16 +0200
Subject: [PATCH] add missing libX11 to LIBS

---
 qmpdclient.pro | 1 +
 1 file changed, 1 insertion(+)

diff --git a/qmpdclient.pro b/qmpdclient.pro
index 6e7b64e..1edac71 100644
--- a/qmpdclient.pro
+++ b/qmpdclient.pro
@@ -198,6 +198,7 @@ win32 {
 
 # Installation in done through own installer on win32
 unix {
+    LIBS += -lX11
     !mac {
         SOURCES += src/qmpdclient_x11.cpp
 
-- 
1.8.3

