diff -Nur audacity-src-1.3.8-orig/src/AudacityApp.cpp audacity-src-1.3.8/src/AudacityApp.cpp
--- audacity-src-1.3.8-orig/src/AudacityApp.cpp	2009-07-16 05:27:35.000000000 +0200
+++ audacity-src-1.3.8/src/AudacityApp.cpp	2009-07-20 23:45:41.000000000 +0200
@@ -331,7 +331,9 @@
 ///////////////////////////////////////////////////////////////////////////////
 
 #include <dlfcn.h>
-#include <gtk/gtk.h>
+/*#include <gtk/gtk.h>*/
+#include <glib/gtypes.h>
+#include <glib-object.h>
 
 typedef struct _GnomeProgram GnomeProgram;
 typedef struct _GnomeModuleInfo GnomeModuleInfo;
