diff -Naur qingy-0.6.0.orig/src/DirectFB/directfb_mode.c qingy-0.6.0/src/DirectFB/directfb_mode.c
--- qingy-0.6.0.orig/src/DirectFB/directfb_mode.c	2005-06-11 09:45:28.087391184 -0700
+++ qingy-0.6.0/src/DirectFB/directfb_mode.c	2005-06-11 09:54:00.652469448 -0700
@@ -585,7 +585,7 @@
 			primary->DrawString (primary, "shutting down system...", -1, screen_width / 2, screen_height / 2, DSTF_CENTER);
 			primary->Flip (primary, NULL, 0);
 		}
-		execl ("/sbin/shutdown", "/sbin/shutdown", "-h", "now", (char*)NULL);
+		execl ("/sbin/shutdown", "/sbin/shutdown", "-q", "-h", "now", (char*)NULL);
 	}
   if (action == DO_REBOOT)
 	{
@@ -594,7 +594,7 @@
 			primary->DrawString (primary, "rebooting system...", -1, screen_width / 2, screen_height / 2, DSTF_CENTER);
 			primary->Flip (primary, NULL, 0);
 		}
-		execl ("/sbin/shutdown", "/sbin/shutdown", "-r", "now", (char*)NULL);
+		execl ("/sbin/shutdown", "/sbin/shutdown", "-q", "-r", "now", (char*)NULL);
 	}
   
   /* we should never get here unless call to /sbin/shutdown fails */
diff -Naur qingy-0.6.0.orig/src/main.c qingy-0.6.0/src/main.c
--- qingy-0.6.0.orig/src/main.c	2005-06-11 09:45:28.088391032 -0700
+++ qingy-0.6.0/src/main.c	2005-06-11 09:54:17.990833616 -0700
@@ -295,12 +295,12 @@
 		case EXIT_SHUTDOWN_R:
 			if (username) memset(username, '\0', sizeof(username));
       if (password) memset(password, '\0', sizeof(password));
-			execl ("/sbin/shutdown", "/sbin/shutdown", "-r", "now", (char*)NULL);
+			execl ("/sbin/shutdown", "/sbin/shutdown", "-q", "-r", "now", (char*)NULL);
 			break;
 		case EXIT_SHUTDOWN_H:
 			if (username) memset(username, '\0', sizeof(username));
       if (password) memset(password, '\0', sizeof(password));
-			execl ("/sbin/shutdown", "/sbin/shutdown", "-h", "now", (char*)NULL);
+			execl ("/sbin/shutdown", "/sbin/shutdown", "-q", "-h", "now", (char*)NULL);
 			break;
 		case EXIT_SLEEP:
 			if (username) memset(username, '\0', sizeof(username));
