diff -Naur qingy-0.9.1/src/main.c qingy-0.9.1-fixed/src/main.c
--- qingy-0.9.1/src/main.c	2006-08-04 07:52:17.000000000 +0200
+++ qingy-0.9.1-fixed/src/main.c	2006-10-06 11:09:06.000000000 +0200
@@ -390,12 +390,12 @@
 		case EXIT_SHUTDOWN_R:
 			if (username) memset(username, '\0', sizeof(username));
       if (password) memset(password, '\0', sizeof(password));
-			execl ("/sbin/shutdown", "/sbin/shutdown", "-r", "now", (char*)NULL);
+			execl ("/sbin/shutdown", "/sbin/shutdown", "-qr", "now", (char*)NULL);
 			break;
 		case EXIT_SHUTDOWN_H:
 			if (username) memset(username, '\0', sizeof(username));
       if (password) memset(password, '\0', sizeof(password));
-			execl ("/sbin/shutdown", "/sbin/shutdown", "-h", "now", (char*)NULL);
+			execl ("/sbin/shutdown", "/sbin/shutdown", "-qh", "now", (char*)NULL);
 			break;
 		case EXIT_SLEEP:
 			if (username) memset(username, '\0', sizeof(username));
