From Anthony Hillairet <anthony@hillairet.org>
Bug http://bugs.sourcemage.org/show_bug.cgi?id=14078
--
diff -Naur hydrogen-0.9.3.orig/src/lib/FLACFile.cpp hydrogen-0.9.3/src/lib/FLACFile.cpp
--- hydrogen-0.9.3.orig/src/lib/FLACFile.cpp	2008-04-13 10:49:49.000000000 +0000
+++ hydrogen-0.9.3/src/lib/FLACFile.cpp	2008-04-13 10:49:01.000000000 +0000
@@ -165,14 +165,13 @@
 	}
 
 	set_metadata_ignore_all();
-	set_filename( sFilename.c_str() );
 
-	State s=init();
-	if( s != FLAC__FILE_DECODER_OK ) {
+	FLAC__StreamDecoderInitStatus s = init(sFilename.c_str());
+	if( s != FLAC__STREAM_DECODER_INIT_STATUS_OK ) {
 		errorLog( "[load] Error in init()" );
 	}
 
-	if ( process_until_end_of_file() == false ) {
+	if ( process_until_end_of_stream() == false ) {
 		errorLog( "[load] Error in process_until_end_of_file()" );
 	}
 }
