diff -Naur hydrogen-0.9.3.orig/src/gui/DrumkitManager.cpp hydrogen-0.9.3/src/gui/DrumkitManager.cpp
--- hydrogen-0.9.3.orig/src/gui/DrumkitManager.cpp	2008-04-13 10:29:07.000000000 +0000
+++ hydrogen-0.9.3/src/gui/DrumkitManager.cpp	2008-04-13 10:36:33.000000000 +0000
@@ -22,6 +22,7 @@
  *
  */
 #include <qcursor.h>
+#include <cstdlib>
 
 #include "config.h"
 #include "Skin.h"
diff -Naur hydrogen-0.9.3.orig/src/gui/HydrogenApp.cpp hydrogen-0.9.3/src/gui/HydrogenApp.cpp
--- hydrogen-0.9.3.orig/src/gui/HydrogenApp.cpp	2008-04-13 10:29:07.000000000 +0000
+++ hydrogen-0.9.3/src/gui/HydrogenApp.cpp	2008-04-13 10:38:19.000000000 +0000
@@ -51,6 +51,8 @@
 #include <qhbox.h>
 #include <qdir.h>
 
+#include <cstdlib>
+
 HydrogenApp* HydrogenApp::m_pInstance = NULL;
 
 HydrogenApp::HydrogenApp( MainForm *pMainForm, Song *pFirstSong )
diff -Naur hydrogen-0.9.3.orig/src/gui/main.cpp hydrogen-0.9.3/src/gui/main.cpp
--- hydrogen-0.9.3.orig/src/gui/main.cpp	2008-04-13 10:29:07.000000000 +0000
+++ hydrogen-0.9.3/src/gui/main.cpp	2008-04-13 10:39:58.000000000 +0000
@@ -43,6 +43,8 @@
 #include "lib/Exception.h"
 
 #include <iostream>
+#include <cstdlib>
+
 using namespace std;
 
 void showInfo();
diff -Naur hydrogen-0.9.3.orig/src/lib/FLACFile.cpp hydrogen-0.9.3/src/lib/FLACFile.cpp
--- hydrogen-0.9.3.orig/src/lib/FLACFile.cpp	2008-04-13 10:29:07.000000000 +0000
+++ hydrogen-0.9.3/src/lib/FLACFile.cpp	2008-04-13 10:45:11.000000000 +0000
@@ -27,6 +27,7 @@
 
 #include <vector>
 #include <fstream>
+#include <cstring>
 
 #ifdef FLAC_SUPPORT
 
diff -Naur hydrogen-0.9.3.orig/src/lib/drivers/OssDriver.cpp hydrogen-0.9.3/src/lib/drivers/OssDriver.cpp
--- hydrogen-0.9.3.orig/src/lib/drivers/OssDriver.cpp	2008-04-13 10:29:07.000000000 +0000
+++ hydrogen-0.9.3/src/lib/drivers/OssDriver.cpp	2008-04-13 10:31:49.000000000 +0000
@@ -30,6 +30,7 @@
 #include "../Preferences.h"
 
 #include <pthread.h>
+#include <cstring>
 
 audioProcessCallback ossDriver_audioProcessCallback;
 bool ossDriver_running;
diff -Naur hydrogen-0.9.3.orig/src/tools/HydrogenPlayer.cpp hydrogen-0.9.3/src/tools/HydrogenPlayer.cpp
--- hydrogen-0.9.3.orig/src/tools/HydrogenPlayer.cpp	2008-04-13 10:29:07.000000000 +0000
+++ hydrogen-0.9.3/src/tools/HydrogenPlayer.cpp	2008-04-13 10:36:39.000000000 +0000
@@ -27,6 +27,7 @@
 
 #include <iostream>
 #include <stdio.h>
+#include <cstdlib>
 #include <string>
 
 #include "lib/Hydrogen.h"
