From 23442eafb9553fc68dac24695b98626df4a2a8b1 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 2 Nov 2018 12:18:05 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 haskell-src-meta.cabal | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/haskell-src-meta.cabal b/haskell-src-meta.cabal
index 25ea377..ca7c909 100644
--- a/haskell-src-meta.cabal
+++ b/haskell-src-meta.cabal
@@ -17,11 +17,11 @@ description:        The translation from haskell-src-exts abstract syntax
 extra-source-files: ChangeLog README.md examples/*.hs
 
 library
-  build-depends:   base >= 4.6 && < 4.12,
+  build-depends:   base >= 4.6,
                    haskell-src-exts >= 1.18 && < 1.21,
                    pretty >= 1.0 && < 1.2,
                    syb >= 0.1 && < 0.8,
-                   template-haskell >= 2.8 && < 2.14,
+                   template-haskell >= 2.8,
                    th-orphans >= 0.9.1 && < 0.14
 
   if impl(ghc < 7.8)
@@ -40,11 +40,11 @@ test-suite unit
 
   build-depends:
     HUnit                >= 1.2  && < 1.7,
-    base                 >= 4.5  && < 4.12,
+    base                 >= 4.5,
     haskell-src-exts     >= 1.17 && < 1.21,
     haskell-src-meta,
     pretty               >= 1.0  && < 1.2,
-    template-haskell     >= 2.7  && < 2.14,
+    template-haskell     >= 2.7,
     test-framework       >= 0.8  && < 0.9,
     test-framework-hunit >= 0.3  && < 0.4
 
-- 
2.19.1

