diff -Nuar ./AliasDomain.cpp ../xmail-1.22.patched/AliasDomain.cpp
--- ./AliasDomain.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/AliasDomain.cpp	2005-10-29 22:33:38.000000000 +0200
@@ -41,7 +41,7 @@
 #include "MailDomains.h"
 #include "AliasDomain.h"
 
-#define ADOMAIN_FILE                "aliasdomain.tab"
+#define ADOMAIN_FILE                "etc/xmail/aliasdomain.tab"
 #define WILD_ADOMAIN_HASH           0
 #define ADOMAIN_LINE_MAX            512
 
diff -Nuar ./CTRLSvr.cpp ../xmail-1.22.patched/CTRLSvr.cpp
--- ./CTRLSvr.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/CTRLSvr.cpp	2005-10-29 22:50:49.000000000 +0200
@@ -48,15 +48,15 @@
 #include "MailSvr.h"
 #include "CTRLSvr.h"
 
-#define CTRL_ACCOUNTS_FILE      "ctrlaccounts.tab"
+#define CTRL_ACCOUNTS_FILE      "etc/xmail/ctrlaccounts.tab"
 #define CTRL_ACCOUNTS_LINE_MAX  512
 #define CTRLSRV_ACCEPT_TIMEOUT  4
 #define CTRL_LISTEN_SIZE        8
 #define CTRL_WAIT_SLEEP         2
 #define MAX_CLIENTS_WAIT        300
 #define STD_CTRL_TIMEOUT        45
-#define CTRL_IPMAP_FILE         "ctrl.ipmap.tab"
-#define CTRL_LOG_FILE           "ctrl"
+#define CTRL_IPMAP_FILE         "etc/xmail/ctrl.ipmap.tab"
+#define CTRL_LOG_FILE           "var/log/xmail/ctrl"
 #define CTRL_MAX_LINE_SIZE      4096
 #define CTRL_QUIT_CMD_EXIT      1
 #define CTRL_LISTFOLLOW_RESULT  100
diff -Nuar ./DNS.cpp ../xmail-1.22.patched/DNS.cpp
--- ./DNS.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/DNS.cpp	2005-10-29 22:34:48.000000000 +0200
@@ -50,7 +50,7 @@
 
 #define DNS_LABEL_LEN_INVMASK   0xc0
 
-#define ROOTS_FILE              "dnsroots"
+#define ROOTS_FILE              "etc/xmail/dnsroots"
 
 struct DNSQuery {
 	DNS_HEADER DNSH;
diff -Nuar ./ExtAliases.cpp ../xmail-1.22.patched/ExtAliases.cpp
--- ./ExtAliases.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/ExtAliases.cpp	2005-10-29 22:35:08.000000000 +0200
@@ -37,7 +37,7 @@
 #include "TabIndex.h"
 #include "ExtAliases.h"
 
-#define SVR_EXT_ALIAS_FILE          "extaliases.tab"
+#define SVR_EXT_ALIAS_FILE          "etc/xmail/extaliases.tab"
 #define ALIAS_TABLE_LINE_MAX        1024
 
 enum ExtAliasFields {
diff -Nuar ./Filter.cpp ../xmail-1.22.patched/Filter.cpp
--- ./Filter.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/Filter.cpp	2005-10-29 22:51:18.000000000 +0200
@@ -44,8 +44,8 @@
 #include "AppDefines.h"
 #include "MailSvr.h"
 
-#define FILTER_LOG_FILE             "filters"
-#define FILTER_STORAGE_DIR          "filters"
+#define FILTER_LOG_FILE             "var/log/xmail/filters"
+#define FILTER_STORAGE_DIR          "etc/xmail/filters"
 #define FILTER_SELECT_MAX           128
 #define FILTER_DB_LINE_MAX          512
 #define FILTER_LINE_MAX             1024
diff -Nuar ./FINGSvr.cpp ../xmail-1.22.patched/FINGSvr.cpp
--- ./FINGSvr.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/FINGSvr.cpp	2005-10-29 22:51:55.000000000 +0200
@@ -44,8 +44,8 @@
 #define FING_LISTEN_SIZE        64
 #define FING_WAIT_SLEEP         2
 #define MAX_CLIENTS_WAIT        300
-#define FING_IPMAP_FILE         "finger.ipmap.tab"
-#define FING_LOG_FILE           "finger"
+#define FING_IPMAP_FILE         "etc/xmail/finger.ipmap.tab"
+#define FING_LOG_FILE           "var/log/xmail/finger"
 #define FING_SERVER_NAME        "[" APP_NAME_VERSION_STR " FINGER Server]"
 
 static int FINGCheckPeerIP(SYS_SOCKET SockFD);
diff -Nuar ./LMAILSvr.cpp ../xmail-1.22.patched/LMAILSvr.cpp
--- ./LMAILSvr.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/LMAILSvr.cpp	2005-10-29 22:44:57.000000000 +0200
@@ -43,8 +43,8 @@
 #include "MailSvr.h"
 
 #define LMAIL_SERVER_NAME           "[" APP_NAME_VERSION_STR " LMAIL Server]"
-#define LOCAL_SPOOL_DIR             "local"
-#define LMAIL_LOG_FILE              "lmail"
+#define LOCAL_SPOOL_DIR             "var/spool/xmail/local"
+#define LMAIL_LOG_FILE              "var/log/xmail/lmail"
 
 static LMAILConfig *LMAILGetConfigCopy(SHB_HANDLE hShbLMAIL);
 static int LMAILThreadCountAdd(long lCount, SHB_HANDLE hShbLMAIL, LMAILConfig * pLMAILCfg = NULL);
diff -Nuar ./MailDomains.cpp ../xmail-1.22.patched/MailDomains.cpp
--- ./MailDomains.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailDomains.cpp	2005-10-29 22:30:17.000000000 +0200
@@ -42,8 +42,8 @@
 #include "SMAILUtils.h"
 #include "MailDomains.h"
 
-#define MAIL_DOMAINS_DIR            "domains"
-#define MAIL_DOMAINS_FILE           "domains.tab"
+#define MAIL_DOMAINS_DIR            "etc/xmail/domains"
+#define MAIL_DOMAINS_FILE           "etc/xmail/domains.tab"
 #define MAIL_DOMAINS_LINE_MAX       512
 
 enum PopDomainFileds {
diff -Nuar ./MailRoot/aliases.tab ../xmail-1.22.patched/MailRoot/aliases.tab
--- ./MailRoot/aliases.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/aliases.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1,2 +0,0 @@
-"xmailserver.test"	"root"	"xmailuser"
-"xmailserver.test"	"postmaster"	"xmailuser"
diff -Nuar ./MailRoot/ctrl.ipmap.tab ../xmail-1.22.patched/MailRoot/ctrl.ipmap.tab
--- ./MailRoot/ctrl.ipmap.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/ctrl.ipmap.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1 +0,0 @@
-"0.0.0.0"	"0.0.0.0"	"ALLOW"	1
diff -Nuar ./MailRoot/dnsroots ../xmail-1.22.patched/MailRoot/dnsroots
--- ./MailRoot/dnsroots	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/dnsroots	1970-01-01 01:00:00.000000000 +0100
@@ -1,13 +0,0 @@
-a.root-servers.net.
-h.root-servers.net.
-c.root-servers.net.
-g.root-servers.net.
-f.root-servers.net.
-b.root-servers.net.
-j.root-servers.net.
-k.root-servers.net.
-l.root-servers.net.
-m.root-servers.net.
-i.root-servers.net.
-e.root-servers.net.
-d.root-servers.net.
diff -Nuar ./MailRoot/domains/xmailserver.test/xmailuser/mailproc.sample.tab ../xmail-1.22.patched/MailRoot/domains/xmailserver.test/xmailuser/mailproc.sample.tab
--- ./MailRoot/domains/xmailserver.test/xmailuser/mailproc.sample.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/domains/xmailserver.test/xmailuser/mailproc.sample.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-"mailbox"
-"redirect"	"foo-user@supermx.net"	"root@maindomain.org"
-"lredirect"	"foo-user@supermx2.org"	"root@foodomain.net"
-"external"	"0"	"0"	"mailprocess.sh"	"-s"	"@@FROM"	"-r"	"@@RCPT"	"-f"	"@@FILE"
-"wait"	"8"
-"external"	"0"	"0"	"mailprocess-delfile.sh"	"-s"	"@@FROM"	"-r"	"@@RCPT"	"-f"	"@@TMPFILE"
diff -Nuar ./MailRoot/domains/xmailserver.test/xmailuser/pop3.ipmap.sample.tab ../xmail-1.22.patched/MailRoot/domains/xmailserver.test/xmailuser/pop3.ipmap.sample.tab
--- ./MailRoot/domains/xmailserver.test/xmailuser/pop3.ipmap.sample.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/domains/xmailserver.test/xmailuser/pop3.ipmap.sample.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1,2 +0,0 @@
-"0.0.0.0"	"0.0.0.0"	"DENY"	1
-"192.4.1.0"	"255.255.255.0"	"ALLOW"	2
diff -Nuar ./MailRoot/domains/xmailserver.test/xmailuser/user.tab ../xmail-1.22.patched/MailRoot/domains/xmailserver.test/xmailuser/user.tab
--- ./MailRoot/domains/xmailserver.test/xmailuser/user.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/domains/xmailserver.test/xmailuser/user.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1,6 +0,0 @@
-"RealName"	"XMail sample user"
-"HomePage"	"http://www.xmailserver.org/davide.html"
-"MaxMBSize"	"30000"
-"MaxMessageSize"	"10000"
-"SmtpPerms"	"MRV"
-
diff -Nuar ./MailRoot/domains.tab ../xmail-1.22.patched/MailRoot/domains.tab
--- ./MailRoot/domains.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/domains.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1 +0,0 @@
-"xmailserver.test"
diff -Nuar ./MailRoot/etc/xmail/aliases.tab ../xmail-1.22.patched/MailRoot/etc/xmail/aliases.tab
--- ./MailRoot/etc/xmail/aliases.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/aliases.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1,2 @@
+"xmailserver.test"	"root"	"xmailuser"
+"xmailserver.test"	"postmaster"	"xmailuser"
diff -Nuar ./MailRoot/etc/xmail/ctrl.ipmap.tab ../xmail-1.22.patched/MailRoot/etc/xmail/ctrl.ipmap.tab
--- ./MailRoot/etc/xmail/ctrl.ipmap.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/ctrl.ipmap.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1 @@
+"0.0.0.0"	"0.0.0.0"	"ALLOW"	1
diff -Nuar ./MailRoot/etc/xmail/dnsroots ../xmail-1.22.patched/MailRoot/etc/xmail/dnsroots
--- ./MailRoot/etc/xmail/dnsroots	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/dnsroots	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1,13 @@
+a.root-servers.net.
+h.root-servers.net.
+c.root-servers.net.
+g.root-servers.net.
+f.root-servers.net.
+b.root-servers.net.
+j.root-servers.net.
+k.root-servers.net.
+l.root-servers.net.
+m.root-servers.net.
+i.root-servers.net.
+e.root-servers.net.
+d.root-servers.net.
diff -Nuar ./MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/mailproc.sample.tab ../xmail-1.22.patched/MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/mailproc.sample.tab
--- ./MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/mailproc.sample.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/mailproc.sample.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1,6 @@
+"mailbox"
+"redirect"	"foo-user@supermx.net"	"root@maindomain.org"
+"lredirect"	"foo-user@supermx2.org"	"root@foodomain.net"
+"external"	"0"	"0"	"mailprocess.sh"	"-s"	"@@FROM"	"-r"	"@@RCPT"	"-f"	"@@FILE"
+"wait"	"8"
+"external"	"0"	"0"	"mailprocess-delfile.sh"	"-s"	"@@FROM"	"-r"	"@@RCPT"	"-f"	"@@TMPFILE"
diff -Nuar ./MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/pop3.ipmap.sample.tab ../xmail-1.22.patched/MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/pop3.ipmap.sample.tab
--- ./MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/pop3.ipmap.sample.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/pop3.ipmap.sample.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1,2 @@
+"0.0.0.0"	"0.0.0.0"	"DENY"	1
+"192.4.1.0"	"255.255.255.0"	"ALLOW"	2
diff -Nuar ./MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/user.tab ../xmail-1.22.patched/MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/user.tab
--- ./MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/user.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/domains/xmailserver.test/xmailuser/user.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1,6 @@
+"RealName"	"XMail sample user"
+"HomePage"	"http://www.xmailserver.org/davide.html"
+"MaxMBSize"	"30000"
+"MaxMessageSize"	"10000"
+"SmtpPerms"	"MRV"
+
diff -Nuar ./MailRoot/etc/xmail/domains.tab ../xmail-1.22.patched/MailRoot/etc/xmail/domains.tab
--- ./MailRoot/etc/xmail/domains.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/domains.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1 @@
+"xmailserver.test"
diff -Nuar ./MailRoot/etc/xmail/finger.ipmap.tab ../xmail-1.22.patched/MailRoot/etc/xmail/finger.ipmap.tab
--- ./MailRoot/etc/xmail/finger.ipmap.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/finger.ipmap.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1 @@
+"0.0.0.0"	"0.0.0.0"	"ALLOW"	1
diff -Nuar ./MailRoot/etc/xmail/mailusers.tab ../xmail-1.22.patched/MailRoot/etc/xmail/mailusers.tab
--- ./MailRoot/etc/xmail/mailusers.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/mailusers.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1 @@
+"xmailserver.test"	"xmailuser"	"1d08040c09"	1	"xmailuser"	"U"
diff -Nuar ./MailRoot/etc/xmail/message.id ../xmail-1.22.patched/MailRoot/etc/xmail/message.id
--- ./MailRoot/etc/xmail/message.id	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/message.id	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1 @@
+1
diff -Nuar ./MailRoot/etc/xmail/pop3.ipmap.tab ../xmail-1.22.patched/MailRoot/etc/xmail/pop3.ipmap.tab
--- ./MailRoot/etc/xmail/pop3.ipmap.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/pop3.ipmap.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1 @@
+"0.0.0.0"	"0.0.0.0"	"ALLOW"	1
diff -Nuar ./MailRoot/etc/xmail/server.tab ../xmail-1.22.patched/MailRoot/etc/xmail/server.tab
--- ./MailRoot/etc/xmail/server.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/server.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1,45 @@
+#
+# Example configuration file.
+# Note : remember to use _REAL_ TABs and " to format this file
+#
+"RootDomain"	"xmailserver.test"
+"SmtpServerDomain"	"xmailserver.test"
+"POP3Domain"	"xmailserver.test"
+"HeloDomain"	"xmailserver.test"
+"PostMaster"	"root@xmailserver.test"
+"ErrorsAdmin"	"root@xmailserver.test"
+#"TempErrorsAdmin"	"send-failures@xmailserver.test"
+#"DefaultSMTPGateways"	"192.168.1.2,192.168.1.15"
+"RemoveSpoolErrors"	"0"
+#"NoSenderBounce"	"1"
+#"DisableEmitAuthUser"	"1"
+#"NotifyMsgLinesExtra"	"8"
+#"NotifySendLogToSender"	"0"
+#"NotifyTryPattern"	"1"
+"MaxMTAOps"	"16"
+"ReceivedHdrType"	"0"
+"FetchHdrTags"	"+X-Deliver-To,+Received,To,Cc"
+#"SmtpMsgIPBanSpammers"	"550 Denied due inclusion of your IP in our spam lists"
+#"SmtpMsgIPBanSpamAddress"	"550 Denied due inclusion of your email address in our spam lists"
+#"SmtpMsgIPBanMaps"	"550 Denied due inclusion of your IP in the following map"
+#"CustomSMTPMessage"	"Please open http://www.xmailserver.test/smtp_errors.html to get more information about this error"
+#"MaxMessageSize"	"20000"
+#"EnableAuthSMTP-POP3"	"0"
+#"Pop3SyncErrorAccount"	"psync-errors@xmailserver.test"
+#"AllowNullSender"	"1"
+#"AllowSmtpVRFY"	"1"
+#"AllowSmtpETRN"	"1"
+#"SMTP-MaxErrors"	"4"
+#"SmtpMinDiskSpace"	"100000"
+#"SmtpMinVirtMemSpace"	"64000"
+#"Pop3MinVirtMemSpace"	"64000"
+#"CustMapsList"	"list.dsbl.org.:1,blackholes.mail-abuse.org.:1,dialups.mail-abuse.org.:0"
+#"SMTP-RDNSCheck"	"1"
+#"CheckMailerDomain"	"1"
+#"SmartDNSHost"	"dns.home.bogus.net:tcp,192.168.1.1:udp"
+#"DynDnsSetup"	"www.dns4ever.com,80,/sys/u.cgi?d=MYDOMAIN&u=MYUSERNAME&p=MYPASSWORD&i=%s"
+#"DynDnsSetup"	"members.dyndns.org,80,/nic/dyndns?action=edit&started=1&hostname=YES&host_id=yourhost.ourdomain.ext&myip=%s&wildcard=OFF&mx=mail.exchanger.ext&backmx=NO,foouser,foopasswd"
+#"SmtpConfig"	"mail-auth"
+#"SmtpConfig-192.168.0.1"	"mail-auth"
+"DefaultSmtpPerms"	"MRVZ"
+
diff -Nuar ./MailRoot/etc/xmail/smtp.ipmap.tab ../xmail-1.22.patched/MailRoot/etc/xmail/smtp.ipmap.tab
--- ./MailRoot/etc/xmail/smtp.ipmap.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/smtp.ipmap.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1 @@
+"0.0.0.0"	"0.0.0.0"	"ALLOW"	1
diff -Nuar ./MailRoot/etc/xmail/smtprelay.tab ../xmail-1.22.patched/MailRoot/etc/xmail/smtprelay.tab
--- ./MailRoot/etc/xmail/smtprelay.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/smtprelay.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1,3 @@
+"10.0.0.0"	"255.0.0.0"
+"172.16.0.0"	"255.255.0.0"
+"192.168.0.0"	"255.255.0.0"
diff -Nuar ./MailRoot/etc/xmail/userdef.tab ../xmail-1.22.patched/MailRoot/etc/xmail/userdef.tab
--- ./MailRoot/etc/xmail/userdef.tab	1970-01-01 01:00:00.000000000 +0100
+++ ../xmail-1.22.patched/MailRoot/etc/xmail/userdef.tab	2005-10-12 23:35:22.000000000 +0200
@@ -0,0 +1,8 @@
+"RealName"	"??"
+"HomePage"	"??"
+"Address"	"??"
+"Telephone"	"??"
+"MaxMBSize"	"10000"
+"SmtpPerms"	"MR"
+"ReceiveEnable"	"1"
+"PopEnable"	"1"
diff -Nuar ./MailRoot/finger.ipmap.tab ../xmail-1.22.patched/MailRoot/finger.ipmap.tab
--- ./MailRoot/finger.ipmap.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/finger.ipmap.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1 +0,0 @@
-"0.0.0.0"	"0.0.0.0"	"ALLOW"	1
diff -Nuar ./MailRoot/mailusers.tab ../xmail-1.22.patched/MailRoot/mailusers.tab
--- ./MailRoot/mailusers.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/mailusers.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1 +0,0 @@
-"xmailserver.test"	"xmailuser"	"1d08040c09"	1	"xmailuser"	"U"
diff -Nuar ./MailRoot/message.id ../xmail-1.22.patched/MailRoot/message.id
--- ./MailRoot/message.id	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/message.id	1970-01-01 01:00:00.000000000 +0100
@@ -1 +0,0 @@
-1
diff -Nuar ./MailRoot/pop3.ipmap.tab ../xmail-1.22.patched/MailRoot/pop3.ipmap.tab
--- ./MailRoot/pop3.ipmap.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/pop3.ipmap.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1 +0,0 @@
-"0.0.0.0"	"0.0.0.0"	"ALLOW"	1
diff -Nuar ./MailRoot/server.tab ../xmail-1.22.patched/MailRoot/server.tab
--- ./MailRoot/server.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/server.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1,45 +0,0 @@
-#
-# Example configuration file.
-# Note : remember to use _REAL_ TABs and " to format this file
-#
-"RootDomain"	"xmailserver.test"
-"SmtpServerDomain"	"xmailserver.test"
-"POP3Domain"	"xmailserver.test"
-"HeloDomain"	"xmailserver.test"
-"PostMaster"	"root@xmailserver.test"
-"ErrorsAdmin"	"root@xmailserver.test"
-#"TempErrorsAdmin"	"send-failures@xmailserver.test"
-#"DefaultSMTPGateways"	"192.168.1.2,192.168.1.15"
-"RemoveSpoolErrors"	"0"
-#"NoSenderBounce"	"1"
-#"DisableEmitAuthUser"	"1"
-#"NotifyMsgLinesExtra"	"8"
-#"NotifySendLogToSender"	"0"
-#"NotifyTryPattern"	"1"
-"MaxMTAOps"	"16"
-"ReceivedHdrType"	"0"
-"FetchHdrTags"	"+X-Deliver-To,+Received,To,Cc"
-#"SmtpMsgIPBanSpammers"	"550 Denied due inclusion of your IP in our spam lists"
-#"SmtpMsgIPBanSpamAddress"	"550 Denied due inclusion of your email address in our spam lists"
-#"SmtpMsgIPBanMaps"	"550 Denied due inclusion of your IP in the following map"
-#"CustomSMTPMessage"	"Please open http://www.xmailserver.test/smtp_errors.html to get more information about this error"
-#"MaxMessageSize"	"20000"
-#"EnableAuthSMTP-POP3"	"0"
-#"Pop3SyncErrorAccount"	"psync-errors@xmailserver.test"
-#"AllowNullSender"	"1"
-#"AllowSmtpVRFY"	"1"
-#"AllowSmtpETRN"	"1"
-#"SMTP-MaxErrors"	"4"
-#"SmtpMinDiskSpace"	"100000"
-#"SmtpMinVirtMemSpace"	"64000"
-#"Pop3MinVirtMemSpace"	"64000"
-#"CustMapsList"	"list.dsbl.org.:1,blackholes.mail-abuse.org.:1,dialups.mail-abuse.org.:0"
-#"SMTP-RDNSCheck"	"1"
-#"CheckMailerDomain"	"1"
-#"SmartDNSHost"	"dns.home.bogus.net:tcp,192.168.1.1:udp"
-#"DynDnsSetup"	"www.dns4ever.com,80,/sys/u.cgi?d=MYDOMAIN&u=MYUSERNAME&p=MYPASSWORD&i=%s"
-#"DynDnsSetup"	"members.dyndns.org,80,/nic/dyndns?action=edit&started=1&hostname=YES&host_id=yourhost.ourdomain.ext&myip=%s&wildcard=OFF&mx=mail.exchanger.ext&backmx=NO,foouser,foopasswd"
-#"SmtpConfig"	"mail-auth"
-#"SmtpConfig-192.168.0.1"	"mail-auth"
-"DefaultSmtpPerms"	"MRVZ"
-
diff -Nuar ./MailRoot/smtp.ipmap.tab ../xmail-1.22.patched/MailRoot/smtp.ipmap.tab
--- ./MailRoot/smtp.ipmap.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/smtp.ipmap.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1 +0,0 @@
-"0.0.0.0"	"0.0.0.0"	"ALLOW"	1
diff -Nuar ./MailRoot/smtprelay.tab ../xmail-1.22.patched/MailRoot/smtprelay.tab
--- ./MailRoot/smtprelay.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/smtprelay.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1,3 +0,0 @@
-"10.0.0.0"	"255.0.0.0"
-"172.16.0.0"	"255.255.0.0"
-"192.168.0.0"	"255.255.0.0"
diff -Nuar ./MailRoot/userdef.tab ../xmail-1.22.patched/MailRoot/userdef.tab
--- ./MailRoot/userdef.tab	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailRoot/userdef.tab	1970-01-01 01:00:00.000000000 +0100
@@ -1,8 +0,0 @@
-"RealName"	"??"
-"HomePage"	"??"
-"Address"	"??"
-"Telephone"	"??"
-"MaxMBSize"	"10000"
-"SmtpPerms"	"MR"
-"ReceiveEnable"	"1"
-"PopEnable"	"1"
diff -Nuar ./MailSvr.cpp ../xmail-1.22.patched/MailSvr.cpp
--- ./MailSvr.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/MailSvr.cpp	2005-10-29 22:45:16.000000000 +0200
@@ -54,7 +54,7 @@
 
 #define ENV_MAIN_PATH               "MAIL_ROOT"
 #define ENV_CMD_LINE                "MAIL_CMD_LINE"
-#define SVR_SHUTDOWN_FILE           ".shutdown"
+#define SVR_SHUTDOWN_FILE           "var/run/xmail/.shutdown"
 #define STD_SMAIL_THREADS           16
 #define MAX_SMAIL_THREADS           256
 #define STD_SMAIL_RETRY_TIMEOUT     480
diff -Nuar ./Makefile.lnx ../xmail-1.22.patched/Makefile.lnx
--- ./Makefile.lnx	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/Makefile.lnx	2005-10-29 23:29:09.000000000 +0200
@@ -37,7 +37,7 @@
 
 LDFLAGS = -ldl -lpthread
 
-OUTDIR = bin
+OUTDIR = usr/sbin
 MKMACHDEPINC = SysMachine.h
 MKMACHDEP = ${OUTDIR}/MkMachDep
 MAILSVR = ${OUTDIR}/XMail
@@ -85,7 +85,7 @@
 all: ${OUTDIR} ${MKMACHDEPINC} ${MAILSVR} ${CRTLCLNT} ${XMCRYPT} ${MKUSERS} ${SENDMAIL}
 
 ${OUTDIR}:
-	@mkdir ${OUTDIR}
+	@mkdir -p ${OUTDIR}
 
 ${MKMACHDEPINC}: ${MKMACHDEP}
 	./${MKMACHDEP} > ${MKMACHDEPINC}
diff -Nuar ./POP3GwLink.cpp ../xmail-1.22.patched/POP3GwLink.cpp
--- ./POP3GwLink.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/POP3GwLink.cpp	2005-10-29 22:45:34.000000000 +0200
@@ -36,9 +36,9 @@
 #include "UsrUtils.h"
 #include "POP3GwLink.h"
 
-#define SVR_LINKS_FILE              "pop3links.tab"
-#define SVR_LINKS_ENABLE_DIR        "pop3links"
-#define SVR_POP3LOCKS_DIR           "pop3linklocks"
+#define SVR_LINKS_FILE              "etc/xmail/pop3links.tab"
+#define SVR_LINKS_ENABLE_DIR        "etc/xmail/pop3links"
+#define SVR_POP3LOCKS_DIR           "etc/xmail/pop3linklocks"
 #define LINKS_TABLE_LINE_MAX        2048
 
 enum LinksFileds {
diff -Nuar ./POP3Svr.cpp ../xmail-1.22.patched/POP3Svr.cpp
--- ./POP3Svr.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/POP3Svr.cpp	2005-10-29 22:45:47.000000000 +0200
@@ -46,8 +46,8 @@
 #define POP3_LISTEN_SIZE        64
 #define POP3_WAIT_SLEEP         2
 #define MAX_CLIENTS_WAIT        300
-#define POP3_IPMAP_FILE         "pop3.ipmap.tab"
-#define POP3_LOG_FILE           "pop3"
+#define POP3_IPMAP_FILE         "etc/xmail/pop3.ipmap.tab"
+#define POP3_LOG_FILE           "var/log/xmail/pop3"
 #define POP3_SERVER_NAME        "[" APP_NAME_VERSION_STR " POP3 Server]"
 
 enum POP3States {
diff -Nuar ./POP3Utils.cpp ../xmail-1.22.patched/POP3Utils.cpp
--- ./POP3Utils.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/POP3Utils.cpp	2005-10-29 22:46:03.000000000 +0200
@@ -42,8 +42,8 @@
 #include "SMTPUtils.h"
 #include "MailSvr.h"
 
-#define UPOP_IPMAP_FILE         "pop3.ipmap.tab"
-#define POP3_IP_LOGFILE         ".ipconn"
+#define UPOP_IPMAP_FILE         "etc/xmail/pop3.ipmap.tab"
+#define POP3_IP_LOGFILE         "var/log/xmail/.ipconn"
 
 #define POPF_MSG_DELETED        (1 << 0)
 #define POPF_MSG_SENT           (1 << 1)
diff -Nuar ./PSYNCSvr.cpp ../xmail-1.22.patched/PSYNCSvr.cpp
--- ./PSYNCSvr.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/PSYNCSvr.cpp	2005-10-29 22:46:20.000000000 +0200
@@ -39,8 +39,8 @@
 #include "POP3GwLink.h"
 #include "PSYNCSvr.h"
 
-#define PSYNC_LOG_FILE              "psync"
-#define PSYNC_TRIGGER_FILE          ".psync-trigger"
+#define PSYNC_LOG_FILE              "var/log/xmail/psync"
+#define PSYNC_TRIGGER_FILE          "var/run/xmail/.psync-trigger"
 #define PSYNC_WAIT_SLEEP            2
 #define MAX_CLIENTS_WAIT            300
 #define PSYNC_WAKEUP_TIME           2
diff -Nuar ./SMAILUtils.cpp ../xmail-1.22.patched/SMAILUtils.cpp
--- ./SMAILUtils.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/SMAILUtils.cpp	2005-10-29 22:46:49.000000000 +0200
@@ -47,10 +47,10 @@
 
 #define STD_TAG_BUFFER_LENGTH           1024
 #define CUSTOM_CMD_LINE_MAX             512
-#define SMAIL_DOMAIN_PROC_DIR           "custdomains"
-#define SMAIL_CMDALIAS_DIR              "cmdaliases"
+#define SMAIL_DOMAIN_PROC_DIR           "etc/xmail/custdomains"
+#define SMAIL_CMDALIAS_DIR              "etc/xmail/cmdaliases"
 #define SMAIL_DEFAULT_FILTER            ".tab"
-#define SMAIL_LOG_FILE                  "smail"
+#define SMAIL_LOG_FILE                  "var/log/xmail/smail"
 #define MAX_MTA_OPS                     16
 #define ADDRESS_TOKENIZER               ","
 #define SMAIL_EXTERNAL_EXIT_BREAK       16
diff -Nuar ./SMTPSvr.cpp ../xmail-1.22.patched/SMTPSvr.cpp
--- ./SMTPSvr.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/SMTPSvr.cpp	2005-10-29 22:47:32.000000000 +0200
@@ -52,9 +52,9 @@
 #define SMTP_LISTEN_SIZE        64
 #define SMTP_WAIT_SLEEP         2
 #define MAX_CLIENTS_WAIT        300
-#define SMTP_IPMAP_FILE         "smtp.ipmap.tab"
-#define SMTP_IPPROP_FILE        "smtp.ipprop.tab"
-#define SMTP_LOG_FILE           "smtp"
+#define SMTP_IPMAP_FILE         "etc/xmail/smtp.ipmap.tab"
+#define SMTP_IPPROP_FILE        "etc/xmail/smtp.ipprop.tab"
+#define SMTP_LOG_FILE           "var/log/xmail/smtp"
 #define SMTP_SERVER_NAME        "[" APP_NAME_VERSION_STR " ESMTP Server]"
 #define SMTP_PRE_DATA_FILTER    "pre-data"
 #define SMTP_POST_DATA_FILTER   "post-data"
@@ -62,9 +62,9 @@
 #define PLAIN_AUTH_PARAM_SIZE   1024
 #define LOGIN_AUTH_USERNAME     "Username:"
 #define LOGIN_AUTH_PASSWORD     "Password:"
-#define SVR_SMTP_AUTH_FILE      "smtpauth.tab"
+#define SVR_SMTP_AUTH_FILE      "etc/xmail/smtpauth.tab"
 #define SVR_SMTPAUTH_LINE_MAX   512
-#define SVR_SMTP_EXTAUTH_FILE   "smtpextauth.tab"
+#define SVR_SMTP_EXTAUTH_FILE   "etc/xmail/smtpextauth.tab"
 #define SVR_SMTP_EXTAUTH_LINE_MAX   1024
 #define SVR_SMTP_EXTAUTH_TIMEOUT    60
 #define SVR_SMTP_EXTAUTH_PRIORITY   SYS_PRIORITY_NORMAL
diff -Nuar ./SMTPUtils.cpp ../xmail-1.22.patched/SMTPUtils.cpp
--- ./SMTPUtils.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/SMTPUtils.cpp	2005-10-29 22:47:57.000000000 +0200
@@ -46,14 +46,14 @@
 
 #define STD_SMTP_TIMEOUT        STD_SERVER_TIMEOUT
 #define SMTPGW_LINE_MAX         1024
-#define SMTPGW_TABLE_FILE       "smtpgw.tab"
+#define SMTPGW_TABLE_FILE       "etc/xmail/smtpgw.tab"
 #define SMTPFWD_LINE_MAX        1024
-#define SMTPFWD_TABLE_FILE      "smtpfwd.tab"
+#define SMTPFWD_TABLE_FILE      "etc/xmail/smtpfwd.tab"
 #define SMTPRELAY_LINE_MAX      512
-#define SMTP_RELAY_FILE         "smtprelay.tab"
+#define SMTP_RELAY_FILE         "etc/xmail/smtprelay.tab"
 #define MAX_MX_RECORDS          32
-#define SMTP_SPAMMERS_FILE      "spammers.tab"
-#define SMTP_SPAM_ADDRESS_FILE  "spam-address.tab"
+#define SMTP_SPAMMERS_FILE      "etc/xmail/spammers.tab"
+#define SMTP_SPAM_ADDRESS_FILE  "etc/xmail/spam-address.tab"
 #define SPAMMERS_LINE_MAX       512
 #define SPAM_ADDRESS_LINE_MAX   512
 #define SMTPAUTH_LINE_MAX       512
diff -Nuar ./SvrUtils.cpp ../xmail-1.22.patched/SvrUtils.cpp
--- ./SvrUtils.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/SvrUtils.cpp	2005-10-29 22:48:12.000000000 +0200
@@ -39,9 +39,9 @@
 #include "MailSvr.h"
 #include "MiscUtils.h"
 
-#define SVR_PROFILE_FILE            "server.tab"
-#define MESSAGEID_FILE              "message.id"
-#define SMTP_SPOOL_DIR              "spool"
+#define SVR_PROFILE_FILE            "etc/xmail/server.tab"
+#define MESSAGEID_FILE              "etc/xmail/message.id"
+#define SMTP_SPOOL_DIR              "var/spool"
 #define MAX_MSG_FILENAME_LENGTH     80
 #define SVR_PROFILE_LINE_MAX        2048
 #define SYS_RES_CHECK_INTERVAL      8
diff -Nuar ./TACACS.cpp ../xmail-1.22.patched/TACACS.cpp
--- ./TACACS.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/TACACS.cpp	2005-10-29 22:48:31.000000000 +0200
@@ -50,7 +50,7 @@
 #define TACACS_DPORT            110
 #define TACACS_LPORT            0
 
-#define TACS_CONFIG_FILE        "tacacsservers.tab"
+#define TACS_CONFIG_FILE        "etc/xmail/tacacsservers.tab"
 #define TACS_ALIAS_LINE_MAX     1024
 
 #define	TACACS_SVC_NAME         "tacacs"
diff -Nuar ./UsrUtils.cpp ../xmail-1.22.patched/UsrUtils.cpp
--- ./UsrUtils.cpp	2005-10-12 23:35:22.000000000 +0200
+++ ../xmail-1.22.patched/UsrUtils.cpp	2005-10-29 22:49:28.000000000 +0200
@@ -45,18 +45,18 @@
 #include "AliasDomain.h"
 #include "UsrAuth.h"
 
-#define SVR_TABLE_FILE              "mailusers.tab"
-#define SVR_ALIAS_FILE              "aliases.tab"
+#define SVR_TABLE_FILE              "etc/xmail/mailusers.tab"
+#define SVR_ALIAS_FILE              "etc/xmail/aliases.tab"
 #define WILD_ALIASES_HASH           0
 #define USR_TABLE_LINE_MAX          2048
 #define USR_ALIAS_LINE_MAX          512
-#define USER_PROFILE_FILE           "user.tab"
-#define DEFAULT_USER_PROFILE_FILE   "userdef.tab"
-#define MAILBOX_DIRECTORY           "mailbox"
-
-#define POP3_LOCKS_DIR              "pop3locks"
-#define MLUSERS_TABLE_FILE          "mlusers.tab"
-#define MAILPROCESS_FILE            "mailproc.tab"
+#define USER_PROFILE_FILE           "etc/xmail/user.tab"
+#define DEFAULT_USER_PROFILE_FILE   "etc/xmail/userdef.tab"
+#define MAILBOX_DIRECTORY           "var/spool/xmail/mailbox"
+
+#define POP3_LOCKS_DIR              "var/run/xmail/pop3locks"
+#define MLUSERS_TABLE_FILE          "etc/xmail/mlusers.tab"
+#define MAILPROCESS_FILE            "etc/xmail/mailproc.tab"
 
 enum UsrFileds {
 	usrDomain = 0,
