--- ffmpeg-svn/libavcodec/avcodec.h.orig        2008-02-18 17:25:06.000000000 +0000
+++ ffmpeg-svn/libavcodec/avcodec.h     2008-02-18 17:27:07.000000000 +0000
@@ -30,6 +30,14 @@
 #include "avutil.h"
 #include <sys/types.h> /* size_t */
 
+#if !defined INT64_C
+#if defined __WORDSIZE && __WORDSIZE == 64
+#define INT64_C(c) c ## L
+#else
+#define INT64_C(c) c ## LL
+#endif
+#endif
+
 #define LIBAVCODEC_VERSION_INT  ((51<<16)+(50<<8)+1)
 #define LIBAVCODEC_VERSION      51.50.1
 #define LIBAVCODEC_BUILD        LIBAVCODEC_VERSION_INT

