Index: ffmpeg-0.cvs20070307/libavcodec/avcodec.h
===================================================================
--- ffmpeg-0.cvs20070307.orig/libavcodec/avcodec.h	2007-03-29 22:22:06.000000000 +0200
+++ ffmpeg-0.cvs20070307/libavcodec/avcodec.h	2007-03-29 22:23:03.000000000 +0200
@@ -34,6 +34,14 @@
 #include "avutil.h"
 #include <sys/types.h> /* size_t */
 
+#if !defined INT64_C
+#if defined __WORDSIZE && __WORDSIZE == 64
+#define INT64_C(c) c ## L
+#else
+#define INT64_C(c) c ## LL
+#endif
+#endif
+
 #define AV_STRINGIFY(s)         AV_TOSTRING(s)
 #define AV_TOSTRING(s) #s
 
