From 3e53a28e7810f40773ec8f2dbc239474a356cbf8 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 29 May 2021 13:51:48 +0200
Subject: [PATCH] Suppress compiler warning

---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index cbd820d..0438f6c 100644
--- a/meson.build
+++ b/meson.build
@@ -309,7 +309,6 @@ elif cc.get_id() == 'gcc' or cc.get_id() == 'clang'
     '-Werror=sequence-point',
     '-Werror=return-type',
     '-Werror=trigraphs',
-    '-Werror=array-bounds',
     '-Werror=write-strings',
     '-Werror=address',
     '-Werror=int-to-pointer-cast',
-- 
2.31.1

