From 63e3627213e6d7a352210f76c0004fb88d5e19e0 Mon Sep 17 00:00:00 2001
From: Nicolas Chauvet <kwizart@gmail.com>
Date: Mon, 18 Jun 2012 23:35:33 +0200
Subject: [PATCH] Fix build with gcc-4.5

---
 IlmBase/Imath/ImathMatrix.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/IlmBase/Imath/ImathMatrix.h b/IlmBase/Imath/ImathMatrix.h
index 5887412..bbf8cd1 100644
--- a/IlmBase/Imath/ImathMatrix.h
+++ b/IlmBase/Imath/ImathMatrix.h
@@ -49,6 +49,7 @@
 #include "ImathVec.h"
 #include "ImathShear.h"
 
+#include <cstring>
 #include <iostream>
 #include <iomanip>
 #include <string.h>
-- 
1.7.10

