--- Makefile	2004-02-27 22:00:36.000000000 +0100
+++ Makefile.cp	2004-02-27 22:06:16.000000000 +0100
@@ -54,11 +54,6 @@
 
 install:
 	set -e; for d in $(DIRS); do $(MAKE) -C $$d install; done
-	if [ ! -d $(TARGET_ROOT_ON_HOST)$(RC_DIR)/init.d ]; then \
-		mkdir -p $(TARGET_ROOT_ON_HOST)$(RC_DIR)/init.d; \
-	fi; \
-	cp rc.wlan $(TARGET_ROOT_ON_HOST)$(RC_DIR)/init.d/wlan; \
-	chmod 755 $(TARGET_ROOT_ON_HOST)$(RC_DIR)/init.d/wlan ;\
 	if [ -d $(TARGET_ROOT_ON_HOST)/etc/hotplug ]; then \
 		cp wlan.agent $(TARGET_ROOT_ON_HOST)/etc/hotplug; \
 		rm -f $(TARGET_ROOT_ON_HOST)/etc/hotplug/usb/prism2_usb; \
