diff -uNr sqlite-3.3.10/Makefile.in sqlite-3.3.10.cp/Makefile.in
--- sqlite-3.3.10/Makefile.in	2007-01-08 14:39:27.000000000 +0100
+++ sqlite-3.3.10.cp/Makefile.in	2007-01-16 11:11:58.000000000 +0100
@@ -391,7 +391,7 @@
 
 parse.c:	$(TOP)/src/parse.y lemon$(BEXE) $(TOP)/addopcodes.awk
 	cp $(TOP)/src/parse.y .
-	./lemon$(BEXE) $(OPTS) parse.y
+	./lemon$(BEXE) parse.y
 	mv parse.h parse.h.temp
 	awk -f $(TOP)/addopcodes.awk parse.h.temp >parse.h
 
@@ -424,7 +424,7 @@
 	$(LTCOMPILE) -c $(TOP)/src/tokenize.c
 
 keywordhash.h:	$(TOP)/tool/mkkeywordhash.c
-	$(BCC) -o mkkeywordhash$(BEXE) $(OPTS) $(TOP)/tool/mkkeywordhash.c
+	$(BCC) -o mkkeywordhash$(BEXE) $(TOP)/tool/mkkeywordhash.c
 	./mkkeywordhash$(BEXE) >keywordhash.h
 
 trigger.lo:	$(TOP)/src/trigger.c $(HDR)
