--- Makefile.in.orig	2009-09-10 23:18:17.000000000 +0400
+++ Makefile.in	2009-09-12 16:50:04.107891742 +0400
@@ -684,8 +684,9 @@
 status.lo:	$(TOP)/src/status.c $(HDR)
 	$(LTCOMPILE) $(TEMP_STORE) -c $(TOP)/src/status.c
 
-sqlite3.h:	$(TOP)/src/sqlite.h.in $(TOP)/manifest.uuid $(TOP)/VERSION
-	tclsh $(TOP)/tool/mksqlite3h.tcl $(TOP) >sqlite3.h
+sqlite3.h:	$(TOP)/src/sqlite.h.in 
+	sed -e s/--VERS--/$(RELEASE)/ $(TOP)/src/sqlite.h.in | \
+	sed -e s/--VERSION-NUMBER--/$(VERSION_NUMBER)/ >sqlite3.h
 
 table.lo:	$(TOP)/src/table.c $(HDR)
 	$(LTCOMPILE) $(TEMP_STORE) -c $(TOP)/src/table.c
