--- sqlite-autoconf-3070700/sqlite3.c.orig	2011-06-28 23:38:46.279329010 +0200
+++ sqlite-autoconf-3070700/sqlite3.c	2011-06-28 23:38:59.258185385 +0200
@@ -59049,6 +59049,7 @@
   memset(zCsr, 0, zEnd-zCsr);
   zCsr += (zCsr - (u8*)0)&7;
   assert( EIGHT_BYTE_ALIGNMENT(zCsr) );
+  p->expired = 0;
 
   /* Memory for registers, parameters, cursor, etc, is allocated in two
   ** passes.  On the first pass, we try to reuse unused space at the 
