--- Makefile.in	2007/12/13 18:20:46	1.191
+++ Makefile.in	2007/12/17 16:20:07	1.192
@@ -317,17 +317,10 @@
                 -rpath $(libdir)/sqlite \
 		-version-info "8:6:8"
 
-testcli$(TEXE):	$(TOP)/src/shell.c libsqlite3.la sqlite3.h
+sqlite3$(TEXE):	$(TOP)/src/shell.c libsqlite3.la sqlite3.h
 	$(LTLINK) $(READLINE_FLAGS) $(LIBPTHREAD) \
 		-o $@ $(TOP)/src/shell.c libsqlite3.la \
 		$(LIBREADLINE) $(TLIBS)
-
-sqlite3$(EXE):	$(TOP)/src/shell.c sqlite3.c sqlite3.h
-	$(LTLINK) $(READLINE_FLAGS) -o $@                          \
-		-DSQLITE_MAX_SQL_LENGTH=1000000000                  \
-		-USQLITE_THREADSAFE -DSQLITE_THREADSAFE=0           \
-		$(TOP)/src/shell.c sqlite3.c                        \
-		$(LIBREADLINE) $(TLIBS) $(THREADLIB)
 
 # This target creates a directory named "tsrc" and fills it with
 # copies of all of the C source code and header files needed to
--- main.mk	2007/12/13 18:20:46	1.164
+++ main.mk	2007/12/17 16:20:07	1.165
@@ -277,17 +277,10 @@
 	$(AR) libsqlite3.a $(LIBOBJ)
 	$(RANLIB) libsqlite3.a
 
-testcli$(EXE):	$(TOP)/src/shell.c libsqlite3.a sqlite3.h
-	$(TCCX) $(READLINE_FLAGS) -o testcli$(EXE)                  \
+sqlite3$(EXE):	$(TOP)/src/shell.c libsqlite3.a sqlite3.h
+	$(TCCX) $(READLINE_FLAGS) -o sqlite3$(EXE)                  \
 		$(TOP)/src/shell.c                                  \
 		libsqlite3.a $(LIBREADLINE) $(TLIBS) $(THREADLIB)
-
-sqlite3$(EXE):	$(TOP)/src/shell.c sqlite3.c sqlite3.h
-	$(TCCX) $(READLINE_FLAGS) -o sqlite3$(EXE)                  \
-		-DSQLITE_MAX_SQL_LENGTH=1000000000                  \
-		-USQLITE_THREADSAFE -DSQLITE_THREADSAFE=0           \
-		$(TOP)/src/shell.c sqlite3.c                        \
-		$(LIBREADLINE) $(TLIBS) $(THREADLIB)
 
 objects: $(LIBOBJ_ORIG)
 
