diff -NrU5 lavaps-2.7.orig/src/const_str.hh lavaps-2.7/src/const_str.hh
--- lavaps-2.7.orig/src/const_str.hh	2008-06-29 14:39:09.000000000 +0200
+++ lavaps-2.7/src/const_str.hh	2008-06-29 14:39:29.000000000 +0200
@@ -22,10 +22,11 @@
 
 
 #ifndef lavaps_const_str_h
 #define lavaps_const_str_h
 
+#include <cstdlib>
 #include <string.h>
 
 class const_str {
 protected:
 	const char *rep_;
diff -NrU5 lavaps-2.7.orig/src/lava_signal.cc lavaps-2.7/src/lava_signal.cc
--- lavaps-2.7.orig/src/lava_signal.cc	2008-06-29 14:38:04.000000000 +0200
+++ lavaps-2.7/src/lava_signal.cc	2008-06-29 14:40:13.000000000 +0200
@@ -28,10 +28,11 @@
 #include "main.hh"
 
 #include <sys/time.h>
 #include <sys/resource.h>
 #include <sys/types.h>
+#include <cstring>
 #include <signal.h>
 
 int
 lava_signal(int pid, int sig)
 {
