--- phonon-backend-gstreamer-4.4.4/gstreamer/message.h.orig	2012-01-15 15:11:10.270718918 +0100
+++ phonon-backend-gstreamer-4.4.4/gstreamer/message.h	2012-01-15 15:11:21.532577144 +0100
@@ -20,7 +20,7 @@
 
 #include <QtCore/QMetaType>
 
-#include <glib-2.0/glib/gmacros.h> // For G_*_DECLS macros primarily
+#include <glib.h> // For G_*_DECLS macros primarily
 #include <gst/gstmessage.h>
 
 QT_BEGIN_NAMESPACE
