From 0d0f0c22a6ce79ac25a30b22a96c67d47414e6a4 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 17:33:19 +0100
Subject: [PATCH] relax versions of dependencies

---
 async.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/async.cabal b/async.cabal
index 7f700c5..819fbf8 100644
--- a/async.cabal
+++ b/async.cabal
@@ -50,14 +50,14 @@ library
     if impl(ghc>=7.1)
         other-extensions: Trustworthy
     exposed-modules:     Control.Concurrent.Async
-    build-depends:       base >= 4.3 && < 4.12, hashable >= 1.1.1.0 && < 1.3, stm >= 2.2 && < 2.5
+    build-depends:       base >= 4.3, hashable >= 1.1.1.0 && < 1.3, stm >= 2.2
 
 test-suite test-async
     default-language: Haskell2010
     type:       exitcode-stdio-1.0
     hs-source-dirs: test
     main-is:    test-async.hs
-    build-depends: base >= 4.3 && < 4.12,
+    build-depends: base >= 4.3,
                    async,
                    stm,
                    test-framework,
-- 
2.19.1

