From 7474054626e4e3731b37945ed91d5b464aa8697f Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 17:49:25 +0100
Subject: [PATCH 08/35] Cleanup: test for 0 explicitly

---
 sample.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sample.cpp b/sample.cpp
index 916286d..fd40374 100644
--- a/sample.cpp
+++ b/sample.cpp
@@ -62,8 +62,9 @@ void printDB()
     cout << "[name=" << v[j]->name << "\t";
     cout << "age=" << v[j]->age << "\t";
     cout << "bio={";
-    for(size_t i=0;i<v[j]->bio.size();i++)
+    for(size_t i=0;i<v[j]->bio.size();i++) {
       i && cout << ", ", cout << v[j]->bio[i];
+    }
     cout << "}]\n";
   }
 }
-- 
2.24.0

