From 3c22a6c14818d690b698dc2f72238e663694edd5 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 17:53:06 +0100
Subject: [PATCH 10/35] Cleanup: fix include order

---
 sample.cpp | 4 ++--
 tests.cpp  | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/sample.cpp b/sample.cpp
index fd40374..593b1a3 100644
--- a/sample.cpp
+++ b/sample.cpp
@@ -1,9 +1,9 @@
 #include "hiberlite.h"
 
+#include <algorithm>
+#include <iostream>
 #include <string>
 #include <vector>
-#include <iostream>
-#include <algorithm>
 using namespace std;
 
 class Person{
diff --git a/tests.cpp b/tests.cpp
index c42cd8b..973ea2e 100644
--- a/tests.cpp
+++ b/tests.cpp
@@ -2,11 +2,11 @@
 #include "hiberlite.h"
 using namespace hiberlite;
 
-#include <iostream>
 #include <cmath>
-#include <vector>
+#include <iostream>
 #include <map>
 #include <stdlib.h>
+#include <vector>
 using namespace std;
 
 
-- 
2.24.0

