From cb9c76cc694ce4d3db1b22a2e2d7de34b580dc00 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 17:53:30 +0100
Subject: [PATCH 11/35] Cleanup: use 0U instead of casting

---
 include/Visitor_tmpl_impl.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/Visitor_tmpl_impl.hpp b/include/Visitor_tmpl_impl.hpp
index 925d98a..ab702e5 100644
--- a/include/Visitor_tmpl_impl.hpp
+++ b/include/Visitor_tmpl_impl.hpp
@@ -59,7 +59,7 @@ void AVisitor<C>::startWalk(Y& obj, bean_key key)
 template<class C> template<class Y>
 void AVisitor<C>::walk(Y& obj)
 {
-	hibernate(*this, obj, static_cast<const unsigned int>(0));
+	hibernate(*this, obj, 0U);
 }
 
 template<class C> template<class X>
-- 
2.24.0

