From 785cc4f877842fdb11e8449643e5a4e8455ab6ae Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 18:01:23 +0100
Subject: [PATCH 17/35] Cleanup: add missing override

---
 include/nvp.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/nvp.h b/include/nvp.h
index e37d43a..2046faa 100644
--- a/include/nvp.h
+++ b/include/nvp.h
@@ -41,7 +41,7 @@ class db_atom : public abstract_atom{
 		template<class Stmt,class Arg>
 		void loadValue(Stmt& res, Arg& arg);
 
-		inline virtual void bindValue(sqlite3_stmt* stmt, int col);
+		inline void bindValue(sqlite3_stmt* stmt, int col) override;
 };
 
 } //namespace hiberlite
-- 
2.24.0

