From 463428e78d50190d876166d59feb207d38a3a775 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 18:01:46 +0100
Subject: [PATCH 18/35] Cleanup: improve formatting

---
 src/Database.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Database.cpp b/src/Database.cpp
index 112e5b7..18e8d69 100644
--- a/src/Database.cpp
+++ b/src/Database.cpp
@@ -169,4 +169,4 @@ std::vector<sqlid_t> Database::dbSelectChildIds(std::string table, sqlid_t paren
 }
 
 
-} //namespace hiberlite;
+}  // namespace hiberlite
-- 
2.24.0

