From 7dd0a446b245be2b6f90f46889b7a1fc20b59f31 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 18:02:14 +0100
Subject: [PATCH 19/35] Cleanup: use =default instead of {}

---
 src/SQLiteStmt.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/SQLiteStmt.cpp b/src/SQLiteStmt.cpp
index e95e85f..6aeeddd 100644
--- a/src/SQLiteStmt.cpp
+++ b/src/SQLiteStmt.cpp
@@ -15,8 +15,7 @@ SQLiteSelect::SQLiteSelect(const shared_connection& _con, const std::string& que
 }
 
 SQLiteSelect::~SQLiteSelect()
-{
-}
+= default;
 
 bool SQLiteSelect::step()
 {
-- 
2.24.0

