From 8a84cc78170452656b9a9086ca589efc86d8996d Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 18:04:54 +0100
Subject: [PATCH 21/35] Cleanup: improve whitespace

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f68fb92..4f45972 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@ include(sqlite)
 
 find_package(Threads REQUIRED)
 
-set(SOURCES  
+set(SOURCES
   src/BeanLoader.cpp
   src/BeanUpdater.cpp
   src/ChildKiller.cpp
-- 
2.24.0

