From 6416ad9c3f18a321fd3e4a48afcf360d83e75f53 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 18:39:13 +0100
Subject: [PATCH 24/35] Cleanup: improve whitespace

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7109b24..d8f37ea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,4 +90,4 @@ add_executable(tests
 target_link_libraries(tests
   ${TARGET_NAME}
 )
-  
+
-- 
2.24.0

