From 594d59d95431735c7687f7e0e08d486281e1b206 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 18:39:54 +0100
Subject: [PATCH 26/35] Cleanup: initialize member in class instead of ctor

---
 include/shared_res.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/shared_res.h b/include/shared_res.h
index b5d2b83..e9488cd 100644
--- a/include/shared_res.h
+++ b/include/shared_res.h
@@ -10,9 +10,9 @@ protected:
 	friend class shared_res<C>;
 
 	C* res;
-	int refCount;
+	int refCount{0};
 
-	explicit inline shared_cnt_obj_pair(C* r) : res(r), refCount(0) {}
+	explicit inline shared_cnt_obj_pair(C* r) : res(r) {}
 
 public:
 
-- 
2.24.0

