From ee842bd06ee4ec35cf48243a01ffe03785987ed0 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 21:18:02 +0100
Subject: [PATCH 33/35] Integrate test cases via ctest

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 981c8cf..2b3f81d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.9)
 
 project(hiberlite VERSION 0.1)
 
+enable_testing()
+
 #find_package(Qt5 CONFIG REQUIRED Core Gui Widgets)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
@@ -113,3 +115,4 @@ target_link_libraries(tests
   ${TARGET_NAME}_shared
 )
 
+add_test(NAME tests COMMAND ${CMAKE_CURRENT_BINARY_DIR}/tests)
-- 
2.24.0

