From c5a562591f78bdb22338651ff6191043485ed058 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 1 Dec 2019 22:03:13 +0100
Subject: [PATCH 34/35] library fixup

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b3f81d..b0e0288 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,6 +83,12 @@ target_include_directories(${TARGET_NAME}_shared
   ${SQLITE_INCLUDES}
 )
 
+target_include_directories(${TARGET_NAME}_static
+  PUBLIC
+  include
+  ${SQLITE_INCLUDES}
+)
+
 target_link_libraries(${TARGET_NAME}_shared
   Threads::Threads
   dl
-- 
2.24.0

