Index: plugins/setupwindow/scimimenginesettings.cpp
===================================================================
RCS file: /cvsroot/scim/skim/plugins/setupwindow/scimimenginesettings.cpp,v
retrieving revision 1.27
diff -u -r1.27 scimimenginesettings.cpp
--- plugins/setupwindow/scimimenginesettings.cpp	29 Apr 2006 20:32:09 -0000	1.27
+++ plugins/setupwindow/scimimenginesettings.cpp	6 Jun 2006 14:25:44 -0000
@@ -166,7 +166,7 @@
     QStringList & uuids = d->uuids;
     QStringList names;
     QStringList langs;
-    std::vector <scim::String> icons;
+    QStringList icons;
     
     std::map <QString, std::vector <size_t> > groups;
     
@@ -207,7 +207,7 @@
 
           item->setPixmap(PIX_COLUMN,
                           KGlobal::iconLoader()->loadIcon(
-                                  QString::fromUtf8(icons [id].c_str ()), KIcon::User, ui->fontMetrics().height()));
+                                  icons [id], KIcon::User, ui->fontMetrics().height()));
           if(!disabledFactories.contains(uuids[id])) {
             item->setOn(true);
             info.enabled = true;
@@ -464,7 +464,7 @@
 void ScimIMEngineSettings::get_factory_list (QStringList &uuids,
                           QStringList &names,
                           QStringList &langs,
-                          std::vector <scim::String> &icons)
+                          QStringList &icons)
 {
     std::vector<scim::String>    module_list;
     scim::IMEngineFactoryPointer factory;
Index: plugins/setupwindow/scimimenginesettings.h
===================================================================
RCS file: /cvsroot/scim/skim/plugins/setupwindow/scimimenginesettings.h,v
retrieving revision 1.11
diff -u -r1.11 scimimenginesettings.h
--- plugins/setupwindow/scimimenginesettings.h	30 Jan 2006 11:56:42 -0000	1.11
+++ plugins/setupwindow/scimimenginesettings.h	6 Jun 2006 14:25:44 -0000
@@ -46,7 +46,7 @@
     void get_factory_list (QStringList &uuids,
                           QStringList &names,
                           QStringList &langs,
-                          std::vector <scim::String> &icons);
+                          QStringList &icons);
     void setIMFilters(QCheckListItem * item, std::vector <scim::String> & filter_list);
     void checkModification();
 
