From fea35625329d3184c5523ecf11597499a0c0fe9b Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Thu, 23 Jan 2025 00:46:35 +0100
Subject: [PATCH 3/3] Add CMake installation rules

Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 78d2cd8dce03..5ab8b92d379d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,3 +77,6 @@ if(PROJECT_IS_TOP_LEVEL)
 	# test target  
 	add_custom_target(test COMMAND test-internals COMMAND test-file COMMAND test-inline COMMAND test-c++inline COMMAND test-c++file  COMMAND test-shlib COMMAND test-statlib DEPENDS testcases.txt)
 endif()
+
+install(TARGETS zxcvbn-shared zxcvbn-static DESTINATION lib)
+install(FILES zxcvbn.h DESTINATION include)
-- 
2.47.1

