--- w3c-libwww-5.4.0/configure	2002-06-12 19:31:31.000000000 +1000
+++ w3c-libwww/configure	2003-10-03 14:06:48.000000000 +1000
@@ -7348,19 +7348,19 @@
   *)
     echo "$ac_t""yes" 1>&6
     if test "x$withval" = "xyes"; then
-      if test -f '/usr/local/lib/mysql/libmysqlclient.a'; then
-          echo $ac_n "checking for "/usr/local/lib/mysql/libmysqlclient.a"""... $ac_c" 1>&6
-echo "configure:7354: checking for "/usr/local/lib/mysql/libmysqlclient.a"" >&5
-  LP=`dirname "/usr/local/lib/mysql/libmysqlclient.a"`
-  LN=`basename "/usr/local/lib/mysql/libmysqlclient.a" | sed 's/lib\(.*\)\.a/\1/g'`
+      if test -f '/usr/lib/libmysqlclient.a'; then
+          echo $ac_n "checking for "/usr/lib/libmysqlclient.a"""... $ac_c" 1>&6
+echo "configure:7354: checking for "/usr/lib/libmysqlclient.a"" >&5
+  LP=`dirname "/usr/lib/libmysqlclient.a"`
+  LN=`basename "/usr/lib/libmysqlclient.a" | sed 's/lib\(.*\)\.a/\1/g'`
   echo "$ac_t""-L$LP -l$LN" 1>&6
   LIBS="$LIBS -L$LP -l$LN"
 
       elif test -f '/usr/lib/mysql/libmysqlclient.a'; then
-          echo $ac_n "checking for "/usr/lib/mysql/libmysqlclient.a"""... $ac_c" 1>&6
-echo "configure:7362: checking for "/usr/lib/mysql/libmysqlclient.a"" >&5
-  LP=`dirname "/usr/lib/mysql/libmysqlclient.a"`
-  LN=`basename "/usr/lib/mysql/libmysqlclient.a" | sed 's/lib\(.*\)\.a/\1/g'`
+          echo $ac_n "checking for "/usr/lib/libmysqlclient.a"""... $ac_c" 1>&6
+echo "configure:7362: checking for "/usr/lib/libmysqlclient.a"" >&5
+  LP=`dirname "/usr/lib/libmysqlclient.a"`
+  LN=`basename "/usr/lib/libmysqlclient.a" | sed 's/lib\(.*\)\.a/\1/g'`
   echo "$ac_t""-L$LP -l$LN" 1>&6
   LIBS="$LIBS -L$LP -l$LN"
 
