--- defaults/Makefile.in.orig	2009-09-22 16:54:11.000000000 +0400
+++ defaults/Makefile.in	2009-09-24 00:34:28.749574347 +0400
@@ -260,11 +260,11 @@
 	-I$(top_srcdir) \
 	-I$(top_builddir) \
 	-DSYSGCONFDIR=\"$(sysgconfdir)\" \
-	$(DEFAULTS_CFLAGS)
+	$(DEPENDENT_CFLAGS)
 
 gconf_defaults_mechanism_LDADD = \
 	$(top_builddir)/gconf/libgconf-2.la \
-	$(DEFAULTS_LIBS)
+	$(DEPENDENT_LIBS)
 
 BUILT_SOURCES = gconf-defaults-glue.h
 dbus_servicesdir = $(datadir)/dbus-1/system-services
