--- dmidecode-2.10/Makefile.orig	2010-09-15 22:05:41.000000000 +0200
+++ dmidecode-2.10/Makefile	2010-09-15 22:05:45.000000000 +0200
@@ -39,7 +39,7 @@
 # BSD make doesn't understand the $(shell) syntax above, it wants the !=
 # syntax below. GNU make ignores the line below so in the end both BSD
 # make and GNU make are happy.
-PROGRAMS != echo dmidecode ; test `uname -m 2>/dev/null` != ia64 && echo biosdecode ownership vpddecode
+#PROGRAMS != echo dmidecode ; test `uname -m 2>/dev/null` != ia64 && echo biosdecode ownership vpddecode
 
 all : $(PROGRAMS)
 
