diff -urp math3d-0.3.0/math3d/m2d.h math3d-0.3.0.gcc3/math3d/m2d.h
--- math3d-0.3.0/math3d/m2d.h	2000-10-09 14:17:54.000000000 +0200
+++ math3d-0.3.0.gcc3/math3d/m2d.h	2002-10-25 23:53:12.000000000 +0200
@@ -88,7 +88,7 @@ namespace Math3d {
       double squared() const;
       double length() const;
 
-      friend M3d;
+      friend class M3d;
   };
 
   inline double&
diff -urp math3d-0.3.0/math3d/m3d.h math3d-0.3.0.gcc3/math3d/m3d.h
--- math3d-0.3.0/math3d/m3d.h	2000-10-09 14:17:54.000000000 +0200
+++ math3d-0.3.0.gcc3/math3d/m3d.h	2002-10-25 23:52:56.000000000 +0200
@@ -100,10 +100,10 @@ namespace Math3d {
       double squared() const;
       double length() const;
 
-      friend M2d;
-      friend M4d;
-      friend MQuat;
-      friend M4x4;
+      friend class M2d;
+      friend class M4d;
+      friend class MQuat;
+      friend class M4x4;
   };
 
   inline double&
diff -urp math3d-0.3.0/math3d/m4d.h math3d-0.3.0.gcc3/math3d/m4d.h
--- math3d-0.3.0/math3d/m4d.h	2000-10-09 14:17:54.000000000 +0200
+++ math3d-0.3.0.gcc3/math3d/m4d.h	2002-10-25 23:53:46.000000000 +0200
@@ -99,8 +99,8 @@ namespace Math3d {
       double squared() const;
       double length() const;
 
-      friend M3d;
-      friend M4x4;
+      friend class M3d;
+      friend class M4x4;
   };
   
   inline double&
diff -urp math3d-0.3.0/math3d/m4x4.h math3d-0.3.0.gcc3/math3d/m4x4.h
--- math3d-0.3.0/math3d/m4x4.h	2000-10-09 19:55:51.000000000 +0200
+++ math3d-0.3.0.gcc3/math3d/m4x4.h	2002-10-25 23:54:35.000000000 +0200
@@ -118,7 +118,7 @@ namespace Math3d {
       bool operator!=(const M4x4& A) const;
       bool cmp(const M4x4& A, double epsilon=EPSILON) const;
 
-      friend MQuat;
+      friend class MQuat;
   };
 
   inline const double*
diff -urp math3d-0.3.0/math3d/mquat.h math3d-0.3.0.gcc3/math3d/mquat.h
--- math3d-0.3.0/math3d/mquat.h	2000-10-09 14:17:54.000000000 +0200
+++ math3d-0.3.0.gcc3/math3d/mquat.h	2002-10-25 23:54:14.000000000 +0200
@@ -99,7 +99,7 @@ namespace Math3d {
       double squared() const;
       double length() const;
 
-      friend M4x4;
+      friend class M4x4;
 };
 
   inline double&
