--- configure.ac.orig	2007-07-25 01:04:19.000000000 +0400
+++ configure.ac	2011-01-14 03:24:10.930926931 +0300
@@ -60,7 +60,7 @@
 then
 
 	# Check for gnuTLS.
-	AM_PATH_LIBGNUTLS(1.0.0, , [ AC_MSG_ERROR([[gnutls not found; install gnutls, gnutls-openssl and libtasn1 packages for your system or run configure with --disable-gnutls]]) ])
+	PKG_CHECK_MODULES(gnutls, gnutls >= 1.0.0, , [ AC_MSG_ERROR([[gnutls not found; install gnutls, gnutls-openssl and libtasn1 packages for your system or run configure with --disable-gnutls]]) ])
 
 	# This is a bloody hack for fedora core
 	CFLAGS="$CFLAGS $LIBGNUTLS_CFLAGS"
