--- glademm-2.6.0/src/writers/bonobo_dock.cc.orig	2010-04-17 18:03:45.605704595 +0200
+++ glademm-2.6.0/src/writers/bonobo_dock.cc	2010-04-17 18:04:15.178695540 +0200
@@ -19,6 +19,7 @@
  */
 
 #include "container.hh"
+#include <cstring>
 
 class Bonobo_Dock : public Gtk_Container
 {public:
--- glademm-2.6.0/src/writers/bonobo_dockitem.cc.orig	2010-04-17 18:04:40.993696230 +0200
+++ glademm-2.6.0/src/writers/bonobo_dockitem.cc	2010-04-17 18:04:52.224157567 +0200
@@ -19,6 +19,7 @@
  */
 
 #include "bin.hh"
+#include <cstring>
 
 class Bonobo_DockItem : public Gtk_Bin
 {public:
--- glademm-2.6.0/src/writers/clist.cc.orig	2010-04-17 18:05:27.062695862 +0200
+++ glademm-2.6.0/src/writers/clist.cc	2010-04-17 18:05:41.749695340 +0200
@@ -20,6 +20,7 @@
 
 #include "clist.hh"
 #include "../strsep.h"
+#include <cstring>
 
 static Gtk_CList Gtk_CList(false);
 
--- glademm-2.6.0/src/writers/combo.cc.orig	2010-04-17 18:06:05.262086301 +0200
+++ glademm-2.6.0/src/writers/combo.cc	2010-04-17 18:06:16.569695784 +0200
@@ -20,6 +20,7 @@
 
 #include "combo.hh"
 #include "../strsep.h"
+#include <cstring>
 
 static Gtk_Combo Gtk_Combo(false);
 
--- glademm-2.6.0/src/writers/ctree.cc.orig	2003-01-06 08:27:14.000000000 +0100
+++ glademm-2.6.0/src/writers/ctree.cc	2010-04-17 18:06:54.279943195 +0200
@@ -20,6 +20,7 @@
 // copied from clist -- might work
 
 #include "container.hh"
+#include <cstring>
 
 class Gtk_CTree : public Gtk_Container
 {public:
--- glademm-2.6.0/src/writers/gnome_about.cc.orig	2010-04-17 18:07:50.565696875 +0200
+++ glademm-2.6.0/src/writers/gnome_about.cc	2010-04-17 18:08:02.687419261 +0200
@@ -20,6 +20,7 @@
 
 #include "gnome_dialog.hh"
 #include "../strsep.h"
+#include <cstring>
 
 class Gnome_About : public Gnome_Dialog
 {public:
--- glademm-2.6.0/src/writers/gnome_app.hh.orig	2010-04-17 18:09:09.877764845 +0200
+++ glademm-2.6.0/src/writers/gnome_app.hh	2010-04-17 18:09:38.534195424 +0200
@@ -28,7 +28,7 @@
 	virtual bool NeedExplicitCtor(const Widget &w) const;
 	virtual void ConstructionArgs(const Widget &w, CxxFile &f) const;
 	virtual void Configure(const Widget &w, CxxFile &f,const std::string &instance) const;
-	virtual Subwidget Gnome_App::IsSubwidget(const Widget &w,const Widget &ch) const;
+	virtual Subwidget IsSubwidget(const Widget &w,const Widget &ch) const;
 	virtual const std::string InternalInstance(const Widget &parent,const Widget &w2) const;
 	virtual void AddChildren(const Widget &w,CxxFile &f,const std::string &instance,const WriterBase &writer_for_subw, const Widget &widget_for_subw) const;
 };
--- glademm-2.6.0/src/writers/optionmenu.cc.orig	2010-04-17 18:10:15.389695809 +0200
+++ glademm-2.6.0/src/writers/optionmenu.cc	2010-04-17 18:10:23.446675794 +0200
@@ -20,6 +20,7 @@
 
 #include "button.hh"
 #include "../strsep.h"
+#include <cstring>
 
 class Gtk_OptionMenu : public Gtk_Button
 {	static const char * const MenuTags;
--- glademm-2.6.0/src/writers/xpmname.cc.orig	2010-04-17 18:11:09.910695451 +0200
+++ glademm-2.6.0/src/writers/xpmname.cc	2010-04-17 18:11:18.385695785 +0200
@@ -1,6 +1,7 @@
 #include <cstdio>
 #include <string>
 #include <Configuration.hh>
+#include <cstring>
 
 // parts taken from
 /* xpmtoppm.c - read an X11 pixmap file and produce a portable pixmap
--- glademm-2.6.0/src/Tag.cc.orig	2010-04-17 18:11:57.921695792 +0200
+++ glademm-2.6.0/src/Tag.cc	2010-04-17 18:12:06.153698327 +0200
@@ -22,6 +22,7 @@
 #include <utility> // pair
 #include <locale.h>
 #include <iostream>
+#include <cstring>
 
 const Tag *Tag::find(const std::string &tp) const
 {  const_iterator i=find(begin(),tp);
--- glademm-2.6.0/src/TagStream.hh.orig	2010-04-17 18:12:37.619195903 +0200
+++ glademm-2.6.0/src/TagStream.hh	2010-04-17 18:12:52.726183340 +0200
@@ -25,6 +25,7 @@
 #include "Tag.hh"
 #include <fstream>
 #include <strstream>
+#include <cstring>
 
 class TagStream : public Tag
 {	// ---- for reading -----
--- glademm-2.6.0/src/TagStream.cc.orig	2010-04-17 18:13:34.830195720 +0200
+++ glademm-2.6.0/src/TagStream.cc	2010-04-17 18:18:15.777695538 +0200
@@ -20,7 +20,8 @@
 #include "TagStream.hh"
 #include <cstring>
 #include <unistd.h>
-#include <algo.h>
+#include <algorithm>
+#include <iostream>
 
 //#define REMEMBER_EMPTY_SPACE
 
@@ -73,10 +74,10 @@
 {  std::string ret;
    std::string::const_iterator i(cont.begin());
    while (i!=cont.end())
-   {  std::string::const_iterator verbatim(::find(i,cont.end(),'&'));
+   {  std::string::const_iterator verbatim(std::find(i,cont.end(),'&'));
       ret+=std::string(i,verbatim);
       if (verbatim!=cont.end())
-      {  std::string::const_iterator endtag(::find(verbatim,cont.end(),';'));
+      {  std::string::const_iterator endtag(std::find(verbatim,cont.end(),';'));
          if (endtag!=cont.end()) ++endtag;
          std::string tag(verbatim,endtag);
          if (tag[1]=='#' && tag[2]=='x')
