--- src/common/xmpp/tls_nb.py.orig	Wed Apr 24 20:00:14 2013 +0200
+++ src/common/xmpp/tls_nb.py	Wed May 15 14:46:31 2013 +0200
@@ -406,6 +406,8 @@
                 self._load_cert_file(os.path.join('/etc/ssl/certs', f), store,
                         logg=False)
 
+        tcpsock._sslContext.set_cipher_list('RC4-MD5')
+
         tcpsock._sslObj = OpenSSL.SSL.Connection(tcpsock._sslContext,
                 tcpsock._sock)
         tcpsock._sslObj.set_connect_state() # set to client mode
