--- makefile.in	2004-04-17 14:03:00.000000000 +0200
+++ makefile.in.cp	2004-04-18 15:18:16.000000000 +0200
@@ -22,57 +22,6 @@
 #
 ########################################################################
 
-package:	../Ted/Ted.static	\
-		TedBindist.tar		\
-		$(BINDIST).lsm		\
-		installTed.sh		\
-		README
-	:
-	@CHECK_ID_CMD@
-	:
-	rm -rf scratch
-	:
-	mkdir scratch
-	mkdir scratch/bin
-	:
-	$(CHOWN_CMD) scratch
-	$(CHOWN_CMD) scratch/bin
-	:
-	chmod 755 scratch
-	chmod 755 scratch/bin
-	:
-	( cd scratch && umask 0 && tar xvf ../TedBindist.tar )
-	cp scratch/Ted/TedDocument-en_US.rtf TedDocument-en_US.rtf
-	:
-	cp ../Ted/Ted.static scratch/bin/Ted
-	strip scratch/bin/Ted
-	$(CHOWN_CMD) scratch/bin/Ted
-	chmod 755 scratch/bin/Ted
-	:
-	( cd scratch && tar cvf - @TAR_CHOWN_ARGS@ * | gzip -9 > ../$(BINDIST).tar.gz )
-	:
-	$(CHOWN_CMD) README
-	$(CHOWN_CMD) TedDocument-en_US.rtf
-	$(CHOWN_CMD) $(BINDIST).tar.gz
-	$(CHOWN_CMD) $(BINDIST).lsm
-	$(CHOWN_CMD) installTed.sh
-	:
-	test x_$$RPM_BUILD_ROOT != x_ || chmod 644 README
-	test x_$$RPM_BUILD_ROOT != x_ || chmod 644 TedDocument-en_US.rtf
-	test x_$$RPM_BUILD_ROOT != x_ || chmod 644 $(BINDIST).tar.gz
-	test x_$$RPM_BUILD_ROOT != x_ || chmod 644 $(BINDIST).lsm
-	test x_$$RPM_BUILD_ROOT != x_ || chmod 755 installTed.sh
-	:
-	@echo Checking existence of distribution files
-	test -s README
-	test -s TedDocument-en_US.rtf
-	test -s $(BINDIST).tar.gz
-	test -s $(BINDIST).lsm
-	test -x installTed.sh
-	:
-	rm -rf scratch
-	@date >> package
-	@echo Distribution is complete
 
 ########################################################################
 #
@@ -88,17 +37,7 @@
 	:
 	@CHECK_ID_CMD@
 	:
-	rm -rf scratch
-	:
-	mkdir scratch
-	mkdir scratch/bin
-	:
-	$(CHOWN_CMD) scratch
-	$(CHOWN_CMD) scratch/bin
-	:
-	chmod 755 scratch
-	chmod 755 scratch/bin
-	:
+	mkdir -p scratch
 	( cd scratch && umask 0 && tar xvf ../TedBindist.tar )
 	cp scratch/Ted/TedDocument-en_US.rtf TedDocument-en_US.rtf
 	:
@@ -106,28 +45,6 @@
 	$(CHOWN_CMD) scratch/bin/Ted
 	chmod 755 scratch/bin/Ted
 	:
-	( cd scratch && tar cvf - @TAR_CHOWN_ARGS@ * | gzip -9 > ../$(BINDIST).tar.gz )
-	:
-	$(CHOWN_CMD) README
-	$(CHOWN_CMD) TedDocument-en_US.rtf
-	$(CHOWN_CMD) $(BINDIST).tar.gz
-	$(CHOWN_CMD) $(BINDIST).lsm
-	$(CHOWN_CMD) installTed.sh
-	:
-	test x_$$RPM_BUILD_ROOT != x_ || chmod 644 README
-	test x_$$RPM_BUILD_ROOT != x_ || chmod 644 TedDocument-en_US.rtf
-	test x_$$RPM_BUILD_ROOT != x_ || chmod 644 $(BINDIST).tar.gz
-	test x_$$RPM_BUILD_ROOT != x_ || chmod 644 $(BINDIST).lsm
-	test x_$$RPM_BUILD_ROOT != x_ || chmod 755 installTed.sh
-	:
-	@echo Checking existence of distribution files
-	test -s README
-	test -s TedDocument-en_US.rtf
-	test -s $(BINDIST).tar.gz
-	test -s $(BINDIST).lsm
-	test -x installTed.sh
-	:
-	rm -rf scratch
 	@date > package.shared
 	@echo Distribution is complete
 
