diff --git a/src/glsl/meson.build b/src/glsl/meson.build
index 19b4bc6..65f33f6 100644
--- a/src/glsl/meson.build
+++ b/src/glsl/meson.build
@@ -18,12 +18,6 @@ else
 
   glslang_deps = [
     cxx.find_library('glslang',             required: glslang_req),
-    cxx.find_library('MachineIndependent',  required: false),
-    cxx.find_library('OSDependent',         required: glslang_req),
-    cxx.find_library('HLSL',                required: glslang_req),
-    cxx.find_library('OGLCompiler',         required: glslang_req),
-    cxx.find_library('GenericCodeGen',      required: false),
-    cxx.find_library('SPVRemapper',         required: glslang_req),
     cxx.find_library('SPIRV',               required: glslang_req),
     cxx.find_library('SPIRV-Tools-opt',     required: false),
     cxx.find_library('SPIRV-Tools',         required: false),
