From 5a021be5fd4180dadb39ea5297c8acf1fc11b1d2 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 19:41:24 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 generic-deriving.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/generic-deriving.cabal b/generic-deriving.cabal
index ef56629..74cb833 100644
--- a/generic-deriving.cabal
+++ b/generic-deriving.cabal
@@ -75,9 +75,9 @@ library
     build-depends:      base >= 4.3 && < 4.9
     other-modules:      Generics.Deriving.TH.Pre4_9
 
-  build-depends:        containers       >= 0.1   && < 0.6
+  build-depends:        containers       >= 0.1
                       , ghc-prim                     < 1
-                      , template-haskell >= 2.4   && < 2.14
+                      , template-haskell >= 2.4
                       , th-abstraction   >= 0.2.7 && < 0.3
 
   default-language:     Haskell2010
-- 
2.19.1

