--- evolution-sharp-0.11.1/configure.in.evo28	2006-09-12 00:45:44.000000000 -0400
+++ evolution-sharp-0.11.1/configure.in	2006-09-12 00:45:44.000000000 -0400
@@ -15,6 +15,7 @@
 PKG_CHECK_MODULES(EDS12, evolution-data-server-1.2 >= 1.2, edsver="1.2", foo=bar)
 PKG_CHECK_MODULES(EDS14, evolution-data-server-1.2 >= 1.4, edsver="1.4", foo=bar)
 PKG_CHECK_MODULES(EDS16, evolution-data-server-1.2 >= 1.6, edsver="1.6", foo=bar)
+PKG_CHECK_MODULES(EDS18, evolution-data-server-1.2 >= 1.8, edsver="1.8", foo=bar)
 
 if test "x$edsver" = "x"; then
 	AC_MSG_ERROR([You need an evolution-data-server 1.2.x, 1.4.x, or 1.6.x development environment])
@@ -41,6 +42,13 @@
 	ECAL_SOVERSION=3
 fi
 
+if test "x$edsver" = "x1.8"; then
+	EDS_VERSION=-1.2
+	EDS_SOVERSION=7
+	EBOOK_SOVERSION=9
+	ECAL_SOVERSION=7
+fi
+
 AC_SUBST(EDS_VERSION)
 AC_SUBST(EDS_SOVERSION)
 AC_SUBST(EBOOK_SOVERSION)
@@ -49,6 +57,7 @@
 AM_CONDITIONAL(USE_EDS12, test "x$edsver" = "x1.2")
 AM_CONDITIONAL(USE_EDS14, test "x$edsver" = "x1.4")
 AM_CONDITIONAL(USE_EDS16, test "x$edsver" = "x1.6")
+AM_CONDITIONAL(USE_EDS18, test "x$edsver" = "x1.8")
 
 PKG_CHECK_MODULES(MONO, mono >= 1.0.0)
 AC_PATH_PROG(CSC, mcs, no)
@@ -122,6 +131,7 @@
 PKG_CHECK_MODULES(ESHELL22, "evolution-shell-2.2", eshellver="2.2", foo=bar)
 PKG_CHECK_MODULES(ESHELL24, "evolution-shell-2.4", eshellver="2.4", foo=bar)
 PKG_CHECK_MODULES(ESHELL26, "evolution-shell-2.6", eshellver="2.6", foo=bar)
+PKG_CHECK_MODULES(ESHELL28, "evolution-shell-2.8", eshellver="2.8", foo=bar)
 
 if test "x$eshellver" = "x"; then
 	AC_MSG_ERROR([You need to install evolution-shell])
--- evolution-sharp-0.11.1/evolution/Makefile.am.evo28	2006-09-12 00:46:17.000000000 -0400
+++ evolution-sharp-0.11.1/evolution/Makefile.am	2006-09-12 00:46:33.000000000 -0400
@@ -12,6 +12,10 @@
 RAW_API = evolution-api-1.6.raw
 endif
 
+if USE_EDS18
+RAW_API = evolution-api-1.6.raw
+endif
+
 API=evolution-api.xml
 METADATA = Evolution.metadata
 ASSEMBLY_NAME = evolution-sharp
