diff -up webkitgtk-2.4.9/Source/WebCore/platform/gtk/GtkClickCounter.cpp.orig webkitgtk-2.4.9/Source/WebCore/platform/gtk/GtkClickCounter.cpp
--- webkitgtk-2.4.9/Source/WebCore/platform/gtk/GtkClickCounter.cpp.orig	2016-04-15 20:59:30.679145001 +0200
+++ webkitgtk-2.4.9/Source/WebCore/platform/gtk/GtkClickCounter.cpp	2016-04-15 20:59:41.664149533 +0200
@@ -85,8 +85,8 @@ int GtkClickCounter::clickCountForGdkBut
     guint32 eventTime = getEventTime(event);
 
     if ((event->type == GDK_2BUTTON_PRESS || event->type == GDK_3BUTTON_PRESS)
-        || ((abs(buttonEvent->x - m_previousClickPoint.x()) < doubleClickDistance)
-            && (abs(buttonEvent->y - m_previousClickPoint.y()) < doubleClickDistance)
+        || ((std::abs(buttonEvent->x - m_previousClickPoint.x()) < doubleClickDistance)
+            && (std::abs(buttonEvent->y - m_previousClickPoint.y()) < doubleClickDistance)
             && (eventTime - m_previousClickTime < static_cast<guint>(doubleClickTime))
             && (buttonEvent->button == m_previousClickButton)))
         m_currentClickCount++;
diff -up webkitgtk-2.4.9/Source/WebCore/platform/gtk/LocalizedStringsGtk.cpp.orig webkitgtk-2.4.9/Source/WebCore/platform/gtk/LocalizedStringsGtk.cpp
--- webkitgtk-2.4.9/Source/WebCore/platform/gtk/LocalizedStringsGtk.cpp.orig	2016-04-16 10:27:25.371996415 +0200
+++ webkitgtk-2.4.9/Source/WebCore/platform/gtk/LocalizedStringsGtk.cpp	2016-04-16 10:34:52.624007845 +0200
@@ -659,7 +659,7 @@ String localizedMediaTimeDescription(flo
     if (!std::isfinite(time))
         return String::fromUTF8(_("indefinite time"));
 
-    int seconds = static_cast<int>(abs(time));
+    int seconds = abs(static_cast<int>(time));
     int days = seconds / (60 * 60 * 24);
     int hours = seconds / (60 * 60);
     int minutes = (seconds / 60) % 60;
