diff -Nuar webkitgtk-2.4.8.orig/Source/WebCore/platform/text/TextCodecUTF8.h webkitgtk-2.4.8/Source/WebCore/platform/text/TextCodecUTF8.h
--- webkitgtk-2.4.8.orig/Source/WebCore/platform/text/TextCodecUTF8.h	2015-04-25 15:26:50.930259771 -0400
+++ webkitgtk-2.4.8/Source/WebCore/platform/text/TextCodecUTF8.h	2015-04-25 15:29:27.546480274 -0400
@@ -27,6 +27,7 @@
 #define TextCodecUTF8_h
 
 #include "TextCodec.h"
+#include <unicode/utf8.h>
 
 namespace WebCore {
 
