diff -ur webkitgtk-2.4.8.orig/Source/autotools/CheckSystemAndBasicDependencies.m4 webkitgtk-2.4.8/Source/autotools/CheckSystemAndBasicDependencies.m4
--- webkitgtk-2.4.8.orig/Source/autotools/CheckSystemAndBasicDependencies.m4	2015-01-07 10:45:43.000000000 +0100
+++ webkitgtk-2.4.8/Source/autotools/CheckSystemAndBasicDependencies.m4	2015-04-25 00:54:32.627749635 +0200
@@ -87,7 +87,7 @@
 c_compiler="unknown"
 AC_LANG_PUSH([C])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
-#if !(defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 7)
+#if !(defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER) && (__GNUC__ >= 5 || __GNUC__ == 4 && __GNUC_MINOR__ >= 7))
 #error Not a supported GCC compiler
 #endif
 ])], [c_compiler="gcc"], [])
@@ -106,7 +106,7 @@
 cxx_compiler="unknown"
 AC_LANG_PUSH([C++])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
-#if !(defined(__GNUG__) && defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 7)
+#if !(defined(__GNUG__) && defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER) && (__GNUC__ >= 5 || __GNUC__ == 4 && __GNUC_MINOR__ >= 7))
 #error Not a supported G++ compiler
 #endif
 ])], [cxx_compiler="g++"], [])
