--- src/Makefile.in.orig	2009-08-05 09:05:12.000000000 -0500
+++ src/Makefile.in	2009-08-05 09:05:33.000000000 -0500
@@ -270,7 +270,7 @@
 LIBICONV = @LIBICONV@
 LIBINTL = @LIBINTL@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@ $(EB_LIBS) -lproxy $(GNOME_LIBDIR) $(GNOME_APPLETS_LIBS) -L/usr/local/lib/pth/ $(WIN32_LIBS)
+LIBS = @LIBS@ $(EB_LIBS) ../libproxy/libproxy.a $(GNOME_LIBDIR) $(GNOME_APPLETS_LIBS) -L/usr/local/lib/pth/ $(WIN32_LIBS)
 LIBTOOL = @LIBTOOL@
 LIPO = @LIPO@
 LN_S = @LN_S@
