--- po/Makefile.in.in.orig	2004-01-22 11:03:07.986163072 +0000
+++ po/Makefile.in.in	2004-01-22 11:04:42.492795880 +0000
@@ -27,7 +27,7 @@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
+mkinstalldirs = mkdir -p -- 
 
 GMSGFMT = @GMSGFMT@
 MSGFMT = @MSGFMT@
