--- rawstudio-1.2/src/dcraw.cc	2009-04-09 23:21:45.000000000 +0200
+++ rawstudio-1.2.fixed/src/dcraw.cc	2010-04-15 11:38:21.188270444 +0200
@@ -8510,7 +8510,7 @@
   argv[argc] = "";
   for (arg=1; (((opm = argv[arg][0]) - 2) | 2) == '+'; ) {
     opt = argv[arg++][1];
-    if ((cp = strchr (sp="nbrkStqmHACg", opt)))
+    if ((cp = const_cast<char *>(strchr (sp="nbrkStqmHACg", opt))))
       for (i=0; i < "114111111422"[cp-sp]-'0'; i++)
 	if (!isdigit(argv[arg+i][0])) {
 	  dcraw_message (DCRAW_ERROR,_("Non-numeric argument to \"-%c\"\n"), opt);
