--- mail-notification-4.1/m4/evolution-plugin.m4.evo2.12	2006-11-02 18:18:23.000000000 +0100
+++ mail-notification-4.1/m4/evolution-plugin.m4	2007-06-26 16:06:22.000000000 +0200
@@ -13,12 +13,17 @@ AC_DEFUN([AM_PATH_EVOLUTION_PLUGIN],
 
 evolution_branch=ifelse([$1],, [2.4], [$1])
 evolution_plugin_min_version=ifelse([$2],,, [>= $2])
+evolution_plugin_pkgconf=evolution-plugin-$evolution_branch
 
-PKG_CHECK_MODULES(EVOLUTION_PLUGIN, [evolution-plugin-$evolution_branch $evolution_plugin_min_version], [found=yes], [found=no])
+PKG_CHECK_MODULES(EVOLUTION_PLUGIN, [$evolution_plugin_pkgconf $evolution_plugin_min_version], [found=yes], [found=no])
+
+if test $found = no; then
+PKG_CHECK_MODULES(EVOLUTION_PLUGIN, [evolution-plugin $evolution_plugin_min_version], [found=yes;evolution_plugin_pkgconf=evolution-plugin], [found=no])
+fi
 
 if test $found = yes; then
 	AC_MSG_CHECKING([for the Evolution plugin directory])
-	evolution_plugindir=`$PKG_CONFIG --variable=plugindir evolution-plugin-$evolution_branch 2>/dev/null`
+	evolution_plugindir=`$PKG_CONFIG --variable=plugindir $evolution_plugin_pkgconf 2>/dev/null`
 	if test -n "$evolution_plugindir"; then
 		AC_MSG_RESULT([$evolution_plugindir])
 	else
