From 3fbbe5e2bb862c844608769b3671185afbf055a2 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Fri, 1 Jan 2021 11:55:05 +0100
Subject: [PATCH 1/9] cmake - fix copy files cmake - delete obsolete code

---
 src/CMakeLists.txt | 25 +++++++++++--------------
 1 file changed, 11 insertions(+), 14 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b890c4ea8b..fa7d018d98 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -939,7 +939,6 @@ add_custom_target(copy_files
     COMMAND ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_SOURCE_DIR}/src/yvalve/perf.h ${output_dir}/include/perf.h
 )
 add_dependencies_cc     (copy_files databases)
-add_dependencies_unix_cc(copy_files makeHeader)
 project_group(copy_files "Custom build steps")
 
 # headers
@@ -1026,27 +1025,25 @@ if (UNIX)
 
     if (NOT CMAKE_CROSSCOMPILING)
 
-    add_executable          (makeHeader misc/makeHeader.cpp)
-    set_output_directory    (makeHeader . CURRENT_DIR)
     set(files
-        ${CMAKE_SOURCE_DIR}/src/include/types_pub.h
-        ${CMAKE_SOURCE_DIR}/src/include/consts_pub.h
-        ${CMAKE_SOURCE_DIR}/src/dsql/sqlda_pub.h
-        ${CMAKE_SOURCE_DIR}/src/common/dsc_pub.h
-        ${CMAKE_SOURCE_DIR}/src/jrd/ibase.h
-        ${CMAKE_SOURCE_DIR}/src/jrd/inf_pub.h
-        ${CMAKE_SOURCE_DIR}/src/jrd/blr.h
+        ${CMAKE_SOURCE_DIR}/src/include/ibase.h
         ${CMAKE_SOURCE_DIR}/src/include/gen/iberror.h
     )
     foreach(F ${files})
         get_filename_component(name ${F} NAME)
         add_custom_command(TARGET copy_files POST_BUILD
-            COMMAND ${CMAKE_COMMAND} -E copy_if_different ${F} ${name})
+            COMMAND ${CMAKE_COMMAND} -E copy ${F} ${output_dir}/include/${name})
     endforeach()
-    add_custom_command(TARGET copy_files POST_BUILD
-        # include
-        COMMAND makeHeader < ibase.h > ${output_dir}/include/ibase.h
+
+
+    set(imp_files
+      blr.h consts_pub.h dsc_pub.h inf_pub.h sqlda_pub.h types_pub.h
     )
+    foreach(F ${imp_files})
+        add_custom_command(TARGET copy_files POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E copy 
+                   ${CMAKE_SOURCE_DIR}/src/include/firebird/impl/${F} ${output_dir}/include/firebird/impl/${F})
+    endforeach()
 
     endif() # if (NOT CMAKE_CROSSCOMPILING)
 endif()
-- 
2.32.0

