From dd053574821f3749148046eb2c0e14a8c5ff7430 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Tue, 14 Nov 2023 14:41:26 +0100
Subject: [PATCH 2/2] use system re2

---
 src/common/SimilarToRegex.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/common/SimilarToRegex.h b/src/common/SimilarToRegex.h
index 89b8121a1a..3ebcf36641 100644
--- a/src/common/SimilarToRegex.h
+++ b/src/common/SimilarToRegex.h
@@ -22,7 +22,8 @@
 #define FB_COMMON_SIMILAR_TO_REGEX_H
 
 #include "firebird.h"
-#include "re2/re2.h"
+#include <re2/re2.h>
+#include <re2/stringpiece.h>
 #include "../common/classes/auto.h"
 #include "../common/classes/array.h"
 #include "../common/classes/fb_string.h"
-- 
2.42.1

