From c9181049bd0d6a3a9246efc5cb9d44f513ba502c Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sat, 2 Jan 2021 12:33:02 +0100
Subject: [PATCH 5/9] fix FB_TZDATADIR usage

---
 src/include/gen/autoconfig.h.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/include/gen/autoconfig.h.in b/src/include/gen/autoconfig.h.in
index e0df757602..d5c49781e4 100644
--- a/src/include/gen/autoconfig.h.in
+++ b/src/include/gen/autoconfig.h.in
@@ -162,6 +162,10 @@
 #define FB_SECDBDIR "@FB_SECDBDIR@"
 
 
+/* Time Zone data */
+//#define FB_TZDATADIR "@FB_TZDATADIR@"
+#define FB_TZDATADIR "@tzdata@"
+
 /******************************************************************************
  *
  * Headers
-- 
2.32.0

