From e7bed8383e6ada4ed939279f811cec7233166424 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Fri, 9 Jul 2021 20:39:43 +0200
Subject: [PATCH 7/9] re2 now default to utf8

---
 src/common/SimilarToRegex.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/common/SimilarToRegex.cpp b/src/common/SimilarToRegex.cpp
index b54fce5cd2..723cad6e5a 100644
--- a/src/common/SimilarToRegex.cpp
+++ b/src/common/SimilarToRegex.cpp
@@ -110,7 +110,6 @@ namespace
 			options.set_log_errors(false);
 			options.set_dot_nl(true);
 			options.set_case_sensitive(!(flags & COMP_FLAG_CASE_INSENSITIVE));
-			options.set_utf8(!(flags & COMP_FLAG_LATIN));
 
 			re2::StringPiece sp((const char*) re2PatternStr.c_str(), re2PatternStr.length());
 			regexp = FB_NEW_POOL(pool) RE2(sp, options);
@@ -761,7 +760,6 @@ namespace
 			options.set_log_errors(false);
 			options.set_dot_nl(true);
 			options.set_case_sensitive(!(flags & COMP_FLAG_CASE_INSENSITIVE));
-			options.set_utf8(!(flags & COMP_FLAG_LATIN));
 
 			re2::StringPiece sp((const char*) finalRe2Pattern.c_str(), finalRe2Pattern.length());
 			regexp = FB_NEW_POOL(pool) RE2(sp, options);
-- 
2.32.0

