From 6305d7234624616371ba84251e96cc98810fe6c3 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Fri, 9 Jul 2021 21:50:31 +0200
Subject: [PATCH 9/9] cmake-3.16 fix

---
 src/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7cce834a0b..1e3199d6d6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -270,7 +270,7 @@ add_custom_target(parse
     SOURCES ${parse_src} y_tab.h y_tab.c
 )
 project_group(parse "Boot/Custom build steps")
-set_source_files_properties(dsql/parse.cpp include/gen/parse.h PROPERTIES GENERATED TRUE)
+set_source_files_properties(dsql/parse.cpp ${CMAKE_CURRENT_BINARY_DIR}/include/gen/parse.h PROPERTIES GENERATED TRUE)
 
 
 ########################################
@@ -374,7 +374,7 @@ endif()
 
 add_library                 (common ${common_src} ${common_os_src} ${common_include})
 target_link_libraries       (common ${LIB_mpr} tommath tomcrypt decNumber re2)
-add_dependencies_cc         (common UpdateCloopInterfaces)
+add_dependencies_cc         (common UpdateCloopInterfaces parse)
 if (UNIX)
 target_link_libraries       (common dl)
 endif()
-- 
2.32.0

