diff -u -r sabayon-orig/configure sabayon-2.19.2/configure
--- sabayon-orig/configure	2007-09-22 13:37:34.000000000 -0500
+++ sabayon-2.19.2/configure	2007-09-22 13:40:48.000000000 -0500
@@ -24739,6 +24739,12 @@
       install_xinitrc_script="yes"
       XINITRC_SYSCONFDIR="X11/xinit/xinitrc.d"
       ;;
+    smgl)
+      XSESSION="/etc/gdm/Xsession"
+      SESSION_NAME="gnome"
+      install_xinitrc_script="no"
+      XINITRC_SYSCONFDIR=""
+      ;;
     *)
       echo "Your distribution (${with_distro}) is not yet supported!  (patches welcome)"
       exit 1
diff -u -r sabayon-orig/configure.ac sabayon-2.19.2/configure.ac
--- sabayon-orig/configure.ac	2007-09-22 13:37:34.000000000 -0500
+++ sabayon-2.19.2/configure.ac	2007-09-22 13:41:32.000000000 -0500
@@ -162,6 +162,12 @@
       install_xinitrc_script="yes"
       XINITRC_SYSCONFDIR="X11/xinit/xinitrc.d"
       ;;
+    smgl)
+      XSESSION="/etc/gdm/Xsession"
+      SESSION_NAME="gnome"
+      install_xinitrc_script="no"
+      XINITRC_SYSCONFDIR=""
+      ;;
     *)
       echo "Your distribution (${with_distro}) is not yet supported!  (patches welcome)"
       exit 1
