From 83b108bc2d40c9e1374d7a3df6d7a896a6c88e3b Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Wed, 22 Mar 2017 16:33:37 +0100
Subject: [PATCH 2/2] do not downgrade C++

---
 CMakeLists.txt | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 65d0cb1c..012bb7d8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -608,22 +608,6 @@ if (BUILD_VC)
             set(ADDITIONAL_VC_FLAGS "-Wabi -fabi-version=0 -ffp-contract=fast -fPIC")
         endif()
 
-        #Handle Vc master
-        if(Vc_COMPILER_IS_GCC OR Vc_COMPILER_IS_CLANG)
-            AddCompilerFlag("-std=c++11" _ok)
-            if(NOT _ok)
-                AddCompilerFlag("-std=c++0x" _ok)
-            endif()
-        endif()
-
-        macro(ko_compile_for_all_implementations_no_scalar _objs _src)
-        if(PACKAGERS_BUILD)
-            vc_compile_for_all_implementations(${_objs} ${_src} FLAGS ${ADDITIONAL_VC_FLAGS} ONLY SSE2 SSSE3 SSE4_1 AVX AVX2+FMA+BMI2)
-        else()
-            set(${_objs} ${_src})
-        endif()
-        endmacro()
-
         macro(ko_compile_for_all_implementations _objs _src)
         if(PACKAGERS_BUILD)
             vc_compile_for_all_implementations(${_objs} ${_src} FLAGS ${ADDITIONAL_VC_FLAGS} ONLY Scalar SSE2 SSSE3 SSE4_1 AVX AVX2+FMA+BMI2)
-- 
2.16.1

