From 06056dbf542ed5f87f57ad1cda4cdec915279c36 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Tue, 30 Jan 2018 16:24:13 +0100
Subject: [PATCH 2/2] kross optional

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a302e498..fed3055a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -220,7 +220,6 @@ find_package(KF5 ${REQUIRED_KF5_VERSION} REQUIRED
     KCMUtils
     KDELibs4Support
     KIO
-    Kross
     Notifications
     NotifyConfig
     Parts
@@ -234,6 +233,7 @@ find_package(KF5 ${REQUIRED_KF5_VERSION} REQUIRED
 
 find_package(KF5Activities)
 find_package(KF5KHtml)
+find_package(KF5Kross)
 
 set_package_properties(KF5Activities PROPERTIES
     TYPE OPTIONAL
@@ -242,6 +242,10 @@ set_package_properties(KF5KHtml PROPERTIES
     PURPOSE "Required for HTML2ODS import filter"
     TYPE OPTIONAL
     )
+set_package_properties(KF5Kross PROPERTIES
+    PURPOSE "scripting support"
+    TYPE OPTIONAL
+    )
 
 if(KF5Activities_FOUND)
     set(HAVE_KACTIVITIES TRUE)
-- 
2.26.2

