From f399f12f26aa0c1983b7ab333843e94cbdd6bb1f Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Tue, 2 May 2017 10:02:41 +0200
Subject: [PATCH 3/3] find openjpeg-2.1

---
 cmake/modules/FindOpenJPEG.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cmake/modules/FindOpenJPEG.cmake b/cmake/modules/FindOpenJPEG.cmake
index ae859572..eedf7811 100644
--- a/cmake/modules/FindOpenJPEG.cmake
+++ b/cmake/modules/FindOpenJPEG.cmake
@@ -14,14 +14,14 @@ PATHS
     /usr/local/include/openjpeg
     /usr/local/include
     /usr/include/openjpeg
-    /usr/include/openjpeg-1.5
+    /usr/include/openjpeg-2.1
     /usr/include
 PATH_SUFFIXES
-    openjpeg-1.5
+    openjpeg-2.1
     openjpeg
 )
 
-set(OPENJPEG_NAMES ${OPENJPEG_NAMES} openjpeg)
+set(OPENJPEG_NAMES ${OPENJPEG_NAMES} openjp2)
 find_library(OPENJPEG_LIBRARY
   NAMES ${OPENJPEG_NAMES}
   PATHS
-- 
2.12.2

