From 245cb04154ecb7b1dda4723679034fa007f15fae Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Tue, 30 Jan 2018 16:24:13 +0100
Subject: [PATCH 3/3] kross optional

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 012bb7d8..76954886 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -213,7 +213,6 @@ find_package(KF5 ${REQUIRED_KF5_VERSION} REQUIRED
         KCMUtils
         KDELibs4Support
         KIO
-        Kross
         Notifications
         NotifyConfig
         Parts
@@ -227,6 +226,7 @@ find_package(KF5 ${REQUIRED_KF5_VERSION} REQUIRED
 
 find_package(KF5Activities)
 find_package(KF5KHtml)
+find_package(KF5Kross)
 
 set_package_properties(KF5Activities PROPERTIES
     TYPE OPTIONAL
@@ -235,6 +235,10 @@ set_package_properties(KF5KHtml PROPERTIES
     PURPOSE "Required for HTML2ODS import filter"
     TYPE OPTIONAL
 )
+set_package_properties(KF5Kross PROPERTIES
+    PURPOSE "scripting support"
+    TYPE OPTIONAL
+)
 
 if(KF5Activities_FOUND)
     set(HAVE_KACTIVITIES TRUE)
-- 
2.16.0

