#
# From http://bugs.sourcemage.org/show_bug.cgi?id=14387
#
diff -Naur xscreensaver-5.03.old/driver/demo-Gtk.c xscreensaver-5.03/driver/demo-Gtk.c
--- xscreensaver-5.03.old/driver/demo-Gtk.c	2008-03-22 10:27:57.000000000 -0600
+++ xscreensaver-5.03/driver/demo-Gtk.c	2008-03-22 10:28:57.000000000 -0600
@@ -1127,7 +1127,7 @@
   if (!was) gtk_widget_set_sensitive (parent, True);
 #ifdef HAVE_GTK2
   model = gtk_tree_view_get_model (GTK_TREE_VIEW (list));
-  STFU g_assert (model);
+  g_assert (model);
   if (gtk_tree_model_iter_nth_child (model, &iter, NULL, list_elt))
     {
       selection = gtk_tree_view_get_selection (GTK_TREE_VIEW (list));

