From 757dc94abf21fed5ad807659ffad913df5d2193a Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 20:21:11 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 th-expand-syns.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/th-expand-syns.cabal b/th-expand-syns.cabal
index 91c8049..8e9fcda 100644
--- a/th-expand-syns.cabal
+++ b/th-expand-syns.cabal
@@ -26,7 +26,7 @@ source-repository head
  location: git://github.com/DanielSchuessler/th-expand-syns.git
 
 Library
-    build-depends:       base >= 4 && < 5, template-haskell < 2.14, syb, containers
+    build-depends:       base >= 4 && < 5, template-haskell, syb, containers
     ghc-options:
     exposed-modules:     Language.Haskell.TH.ExpandSyns
     other-modules:       Language.Haskell.TH.ExpandSyns.SemigroupCompat
-- 
2.19.1

