diff -Naur Tk-804.028.orig/pTk/mTk/generic/tkBind.c Tk-804.028.new/pTk/mTk/generic/tkBind.c
--- Tk-804.028.orig/pTk/mTk/generic/tkBind.c	2007-09-16 13:08:48.000000000 -0700
+++ Tk-804.028.new/pTk/mTk/generic/tkBind.c	2009-01-13 16:35:38.492005068 -0800
@@ -629,6 +629,9 @@
    /* ColormapNotify */		COLORMAP,
    /* ClientMessage */		0,
    /* MappingNotify */		0,
+#ifdef GenericEvent
+   /* GenericEvent */       0,
+#endif
    /* VirtualEvent */		VIRTUAL,
    /* Activate */		ACTIVATE,
    /* Deactivate */		ACTIVATE,
