From 396a1df9c64d94d32561faa4de49429e2f2ce033 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 21 Jan 2017 12:33:15 +0100
Subject: [PATCH] install docs to the correct location, do not install test
 cases

---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0594d93..c13e325 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,6 +64,3 @@ CONFIGURE_FILE(${CMAKE_BINARY_DIR}/include/2.3.6/med_config.h.cmake ${CMAKE_BINA
 ADD_SUBDIRECTORY(include)
 ADD_SUBDIRECTORY(src)
 ADD_SUBDIRECTORY(tools)
-ADD_SUBDIRECTORY(tests)
-
-INSTALL(DIRECTORY doc DESTINATION ${CMAKE_INSTALL_PREFIX})
-- 
2.11.0

