diff -up cone-0.74/cone/nntpcommand.C.gcc43 cone-0.74/cone/nntpcommand.C
--- cone-0.74/cone/nntpcommand.C.gcc43	2008-03-28 23:58:51.000000000 -0400
+++ cone-0.74/cone/nntpcommand.C	2008-03-28 23:59:10.000000000 -0400
@@ -9,9 +9,10 @@
 #include "nntpcommand.H"
 #include "libmail/addmessage.H"
 #include "gettext.H"
-#include <errno.h>
-#include <stdio.h>
+#include <cerrno>
+#include <cstdio>
 #include <cstdlib>
+#include <cstring>
 
 #include <sys/types.h>
 #if HAVE_UNISTD_H
diff -up cone-0.74/cone/tags.C.gcc43 cone-0.74/cone/tags.C
--- cone-0.74/cone/tags.C.gcc43	2008-03-28 23:59:32.000000000 -0400
+++ cone-0.74/cone/tags.C	2008-03-28 23:59:44.000000000 -0400
@@ -9,6 +9,7 @@
 #include "tags.H"
 #include "gettext.H"
 #include <sstream>
+#include <cstring>
 
 using namespace std;
 
diff -up cone-0.74/cone/spellcheckerAspell.C.gcc43 cone-0.74/cone/spellcheckerAspell.C
--- cone-0.74/cone/spellcheckerAspell.C.gcc43	2008-03-29 00:00:31.000000000 -0400
+++ cone-0.74/cone/spellcheckerAspell.C	2008-03-29 00:00:49.000000000 -0400
@@ -9,7 +9,8 @@
 #include "spellcheckerAspell.H"
 #include "libmail/mail.H"
 
-#include <errno.h>
+#include <cerrno>
+#include <cstring>
 
 #if HAVE_ASPELL_CONFIG_HH
 #include "aspell/suggest.hh"
diff -up cone-0.74/cone/gettext.C.gcc43 cone-0.74/cone/gettext.C
--- cone-0.74/cone/gettext.C.gcc43	2008-03-28 23:57:20.000000000 -0400
+++ cone-0.74/cone/gettext.C	2008-03-28 23:57:49.000000000 -0400
@@ -6,14 +6,15 @@
 */
 
 #include "gettext.H"
-#include <ctype.h>
-#include <stdlib.h>
+#include <cctype>
+#include <cstdlib>
 #include <unistd.h>
 #include <sstream>
 #include <algorithm>
 #include "unicode/unicode.h"
 #include "libmail/mail.H"
-#include <errno.h>
+#include <cerrno>
+#include <cstring>
 
 #if HAVE_LANGINFO_H
 #include <langinfo.h>
diff -up cone-0.74/cone/spellcheckerPspell.C.gcc43 cone-0.74/cone/spellcheckerPspell.C
--- cone-0.74/cone/spellcheckerPspell.C.gcc43	2008-03-29 00:00:00.000000000 -0400
+++ cone-0.74/cone/spellcheckerPspell.C	2008-03-29 00:00:15.000000000 -0400
@@ -9,7 +9,8 @@
 #include "spellcheckerPspell.H"
 #include "libmail/mail.H"
 
-#include <errno.h>
+#include <cerrno>
+#include <cstring>
 
 using namespace std;
 
diff -up cone-0.74/libmail/attachments.C.gcc43 cone-0.74/libmail/attachments.C
--- cone-0.74/libmail/attachments.C.gcc43	2008-03-28 23:46:09.000000000 -0400
+++ cone-0.74/libmail/attachments.C	2008-03-29 00:02:57.000000000 -0400
@@ -11,10 +11,11 @@
 #include "rfc2045/rfc2045charset.h"
 #include <sys/time.h>
 #include "attachments.H"
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstring>
+#include <cstdlib>
+#include <cstdio>
 #include <unistd.h>
-#include <errno.h>
+#include <cerrno>
 #include <sstream>
 #include <iomanip>
 
diff -up cone-0.74/libmail/fd.C.gcc43 cone-0.74/libmail/fd.C
--- cone-0.74/libmail/fd.C.gcc43	2008-03-28 23:47:28.000000000 -0400
+++ cone-0.74/libmail/fd.C	2008-03-28 23:47:57.000000000 -0400
@@ -21,6 +21,7 @@
 
 #include	<iostream>
 #include	<sstream>
+#include	<cstdlib>
 
 using namespace std;
 
diff -up cone-0.74/libmail/mimetypes.C.gcc43 cone-0.74/libmail/mimetypes.C
--- cone-0.74/libmail/mimetypes.C.gcc43	2008-03-28 23:50:35.000000000 -0400
+++ cone-0.74/libmail/mimetypes.C	2008-03-28 23:51:28.000000000 -0400
@@ -8,7 +8,8 @@
 #include "mimetypes.H"
 #include "namespace.H"
 #include <fstream>
-#include <ctype.h>
+#include <cctype>
+#include <cstring>
 
 using namespace std;
 
diff -up cone-0.74/libmail/addmessageimport.C.gcc43 cone-0.74/libmail/addmessageimport.C
--- cone-0.74/libmail/addmessageimport.C.gcc43	2008-03-28 23:43:31.000000000 -0400
+++ cone-0.74/libmail/addmessageimport.C	2008-03-29 00:03:19.000000000 -0400
@@ -7,7 +7,8 @@
 #include "libmail_config.h"
 #include "addmessage.H"
 #include "attachments.H"
-#include <errno.h>
+#include <cerrno>
+#include <cstring>
 
 using namespace std;
 
diff -up cone-0.74/libmail/addressbookadd.C.gcc43 cone-0.74/libmail/addressbookadd.C
--- cone-0.74/libmail/addressbookadd.C.gcc43	2008-03-28 23:45:33.000000000 -0400
+++ cone-0.74/libmail/addressbookadd.C	2008-03-29 00:04:00.000000000 -0400
@@ -13,9 +13,10 @@
 #include "rfcaddr.H"
 #include "misc.H"
 #include "unicode/unicode.h"
-#include <ctype.h>
+#include <cctype>
 #include <vector>
-#include <errno.h>
+#include <cerrno>
+#include <cstring>
 
 using namespace std;
 
diff -up cone-0.74/libmail/rfc2047decode.C.gcc43 cone-0.74/libmail/rfc2047decode.C
--- cone-0.74/libmail/rfc2047decode.C.gcc43	2008-03-28 23:52:50.000000000 -0400
+++ cone-0.74/libmail/rfc2047decode.C	2008-03-28 23:53:00.000000000 -0400
@@ -9,6 +9,7 @@
 #include "rfc2047decode.H"
 #include "rfc822/rfc2047.h"
 #include "mail.H"
+#include <cstring>
 
 using namespace std;
 
diff -up cone-0.74/libmail/addmessageremoveattachments.C.gcc43 cone-0.74/libmail/addmessageremoveattachments.C
--- cone-0.74/libmail/addmessageremoveattachments.C.gcc43	2008-03-28 23:44:28.000000000 -0400
+++ cone-0.74/libmail/addmessageremoveattachments.C	2008-03-29 00:04:43.000000000 -0400
@@ -8,7 +8,8 @@
 #include "addmessage.H"
 #include "attachments.H"
 #include "headers.H"
-#include <errno.h>
+#include <cerrno>
+#include <cstring>
 #include <vector>
 
 using namespace std;
diff -up cone-0.74/libmail/sortfolders.C.gcc43 cone-0.74/libmail/sortfolders.C
--- cone-0.74/libmail/sortfolders.C.gcc43	2008-03-28 23:55:00.000000000 -0400
+++ cone-0.74/libmail/sortfolders.C	2008-03-28 23:55:11.000000000 -0400
@@ -6,6 +6,7 @@
 */
 #include "libmail_config.h"
 #include "mail.H"
+#include <cstring>
 
 mail::folder::sort::sort(bool foldersFirstArg) // TRUE: sort folders first
 	: foldersFirst(foldersFirstArg)
diff -up cone-0.74/libmail/addressbook.C.gcc43 cone-0.74/libmail/addressbook.C
--- cone-0.74/libmail/addressbook.C.gcc43	2008-03-28 23:45:00.000000000 -0400
+++ cone-0.74/libmail/addressbook.C	2008-03-29 00:05:01.000000000 -0400
@@ -13,7 +13,8 @@
 #include "addressbooksearch.H"
 #include "rfc2047decode.H"
 #include "misc.H"
-#include <errno.h>
+#include <cerrno>
+#include <cstring>
 
 using namespace std;
 
diff -up cone-0.74/libmail/fdtls.C.gcc43 cone-0.74/libmail/fdtls.C
--- cone-0.74/libmail/fdtls.C.gcc43	2008-03-28 23:48:21.000000000 -0400
+++ cone-0.74/libmail/fdtls.C	2008-03-28 23:48:32.000000000 -0400
@@ -6,6 +6,7 @@
 */
 #include "libmail_config.h"
 #include "fdtls.H"
+#include <cstdlib>
 
 #if HAVE_LIBCOURIERTLS
 
diff -up cone-0.74/libmail/addmessage.C.gcc43 cone-0.74/libmail/addmessage.C
--- cone-0.74/libmail/addmessage.C.gcc43	2004-06-22 20:55:25.000000000 -0400
+++ cone-0.74/libmail/addmessage.C	2008-03-29 00:05:15.000000000 -0400
@@ -7,7 +7,8 @@
 #include "libmail_config.h"
 #include "addmessage.H"
 #include "attachments.H"
-#include <errno.h>
+#include <cerrno>
+#include <cstring>
 
 // Default method implementation.
 
