From 466bd9484a235531700b664fba6fe81724727a32 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sun, 31 Jul 2022 19:04:29 +0200
Subject: [PATCH 2/4] Add missing function prototypes

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 include/vncauth.h     | 2 ++
 vncviewer/vncviewer.h | 1 +
 2 files changed, 3 insertions(+)

diff --git a/include/vncauth.h b/include/vncauth.h
index 86dc455ca722..ea6c023c9f3f 100644
--- a/include/vncauth.h
+++ b/include/vncauth.h
@@ -28,3 +28,5 @@ extern int vncEncryptAndStorePasswd(char *passwd, char *fname);
 extern char *vncDecryptPasswdFromFile(char *fname);
 extern void vncRandomBytes(unsigned char *bytes);
 extern void vncEncryptBytes(unsigned char *bytes, char *passwd);
+extern int vncEncryptAndStorePasswd2(char *, char *, char *);
+extern int vncDecryptPasswdFromFile2(char *, char *, char *);
diff --git a/vncviewer/vncviewer.h b/vncviewer/vncviewer.h
index 755a4807627c..9d92aa0b48cf 100644
--- a/vncviewer/vncviewer.h
+++ b/vncviewer/vncviewer.h
@@ -140,6 +140,7 @@ extern void SetVisualAndCmap();
 
 extern Bool HandleCursorShape(int xhot, int yhot, int width, int height,
                               CARD32 enc);
+extern Bool HandleCursorPos(int x, int y);
 extern void SoftCursorLockArea(int x, int y, int w, int h);
 extern void SoftCursorUnlockScreen(void);
 extern void SoftCursorMove(int x, int y);
-- 
2.37.1

