--- configure.orig	2004-01-12 12:40:34.000000000 +0100
+++ configure	2004-02-27 12:41:37.000000000 +0100
@@ -20352,6 +20352,9 @@
 
 echo "$as_me:$LINENO: checking for Tcl configuration" >&5
 echo $ECHO_N "checking for Tcl configuration... $ECHO_C" >&6
+if test x"$TCLPACKAGE" = "xno"; then
+        echo "disabled"
+else
 # First check to see if --with-tclconfig was specified.
 if test x"${with_tclconfig}" != x ; then
    if test -f "${with_tclconfig}/tclConfig.sh" ; then
@@ -20482,7 +20485,7 @@
 *-*-cygwin* | *-*-mingw*) TCLDYNAMICLINKING="$TCLLIB";;
 *)TCLDYNAMICLINKING="";;
 esac
-
+fi
 
 
 
