From 3083c004a32b6ce6e76eb7d4d20326defa2a7e08 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 17 Jan 2014 15:45:14 +0100
Subject: [PATCH] fix link error

---
 Makefile     | 2 +-
 src/Makefile | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index cb1c7a0..390320c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 .PHONY: all doc src include
-all: doc src include lib
+all: src include lib
 clean: cleandoc cleansrc cleaninclude cleanlib
 install: installinclude installlib
 
diff --git a/src/Makefile b/src/Makefile
index 8df1a95..7c63b11 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -17,7 +17,7 @@ libnoise.so.0: libnoise.so.0.3
 
 # Real build targets
 libnoise.so.0.3: $(OBJECTS)
-	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -shared -Wl,-soname=libnoise.so.0 -o $@ $(OBJECTS:.o=.lo)
+	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -Xcompiler -shared -Wl,-soname=libnoise.so.0 -o $@ $(OBJECTS:.o=.lo)
 
 libnoise.a: $(OBJECTS)
 	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -o $@ $(OBJECTS)
-- 
1.8.5.2

