diff -Nurp nn-6.6.5.orig/config.h-dist nn-6.6.5/config.h-dist
--- nn-6.6.5.orig/config.h-dist	2004-03-09 18:53:11.000000000 +0100
+++ nn-6.6.5/config.h-dist	2004-03-09 18:53:40.000000000 +0100
@@ -21,7 +21,7 @@
  * following to change the default headers
  */
 
-/* #define GNKSA	*/
+#define GNKSA
 
 
 /************************* DATABASE CONFIGURATION ***************************
@@ -100,7 +100,7 @@
  *	LIB_DIRECTORY defined below.
  */
 
-#define NNTP_SERVER	"/usr/local/lib/nntp_server"
+#define NNTP_SERVER	"/etc/nntp_server"
 
 /*
  *	Define NNTP_PATH_HOSTNAME to force a specific hostname into the
@@ -138,7 +138,7 @@
  *	conf/s-template.h as a starting point for writing you own.
  */
 
-#include "s-sunos5.h"
+#include "s-linux.h"
 
 
 /********************** MACHINE DEPENDENT DEFINITIONS **********************
@@ -149,7 +149,7 @@
  *	conf/m-template.h as a starting point for writing you own.
  */
 
-#include "m-sparc.h"
+#include "m-i80386.h"
 
 
 /**************************** LOCALIZATION ****************************
@@ -182,8 +182,8 @@
  *			   (= LIB_DIRECTORY/Log if undefined).
  */
 
-#define BIN_DIRECTORY	"/usr/local/bin"
-#define LIB_DIRECTORY	"/usr/local/lib/nn"
+#define BIN_DIRECTORY	"/usr/bin"
+#define LIB_DIRECTORY	"/usr/lib/nn"
 
 
 /*************************** NEWS TRANSPORT **************************
@@ -307,13 +307,13 @@
  *	DAEMON_MAN	- nnmaster
  */
 
-#define USER_MAN_DIR	"/usr/local/man/man1"
+#define USER_MAN_DIR	"/usr/share/man/man1"
 #define USER_MAN_SECTION	"1"
 
-#define SYS_MAN_DIR	"/usr/local/man/man1"
+#define SYS_MAN_DIR	"/usr/share/man/man1"
 #define SYS_MAN_SECTION		"1m"
 
-#define DAEMON_MAN_DIR	"/usr/local/man/man8"
+#define DAEMON_MAN_DIR	"/usr/share/man/man8"
 #define DAEMON_MAN_SECTION	"8"
 
 
