From e73ca9e3189667897678a900425055f111f580ec Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 18 Feb 2017 12:28:35 +0100
Subject: [PATCH] remove test for specific version of wxwidgets

---
 configure.ac | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6aacb20..83476ea 100644
--- a/configure.ac
+++ b/configure.ac
@@ -177,11 +177,6 @@ if test "$buildmain" = "yes"; then
         equivalent variable and wxWidgets version is $MIN_WX_VERSION or above.
     ])
   fi
-  if test "${WX_VERSION_MAJOR}.${WX_VERSION_MINOR}" = "3.1"; then
-    AC_MSG_ERROR([You must use wxWidgets 3.0.x, development versions of wxWidgets are not supported.])
-  elif test "${WX_VERSION_MAJOR}.${WX_VERSION_MINOR}" != "3.0"; then
-    AC_MSG_ERROR([You must use wxWidgets 3.0.x, wxWidgets 3.2 or higher is not yet supported.])
-  fi
 
   # --universal=no doesn't work correctly, it still accepts universal builds. Reject it manually here.
   if echo "$WX_CPPFLAGS" | grep __WXUNIVERSAL__ > /dev/null; then
-- 
2.11.1

