--- libexplain-1.4.D001.orig/test/04/t0461a.sh
+++ libexplain-1.4.D001/test/04/t0461a.sh
@@ -32,7 +32,7 @@ test $? -eq 0 || fail
 
 fmt -w 800 test.out.4 > test.out.3
 test $? -eq 0 || no_result
-sed -e 's|(42 ".*",|(42,|' test.out.3 > test.out.2
+sed -e 's|(pid = 42 ".*",|(pid = 42,|' test.out.3 > test.out.2
 test $? -eq 0 || no_result
 fmt test.out.2 > test.out
 test $? -eq 0 || no_result
--- libexplain-1.4.D001.orig/test/04/t0463a.sh
+++ libexplain-1.4.D001/test/04/t0463a.sh
@@ -32,7 +32,7 @@ test $? -eq 0 || fail
 
 fmt -w 800 test.out.4 > test.out.3
 test $? -eq 0 || no_result
-sed -e 's|(42 ".*",|(42,|' test.out.3 > test.out.2
+sed -e 's|(pid = 42 ".*",|(pid = 42,|' test.out.3 > test.out.2
 test $? -eq 0 || no_result
 fmt test.out.2 > test.out
 test $? -eq 0 || no_result
--- libexplain-1.4.D001.orig/test/06/t0652a.sh
+++ libexplain-1.4.D001/test/06/t0652a.sh
@@ -34,9 +34,9 @@ test $? -eq 0 || no_result
 # There is an alternative result, depending on the testing enviroment,
 # because stdin may actually be closed.
 fmt > test.ok.2 << 'fubar'
-fputs(s = "yuck", fp = stdin) failed, Bad file descriptor (EBADF)
-because fildes does not refer to an open file; this is more likely to
-be a software error (a bug) than it is to be a user error
+fputs(s = "yuck", fp = stdin) failed, Bad file descriptor (EBADF) because
+the fildes argument does not refer to an object that is open for writing
+(O_RDONLY | O_LARGEFILE)
 fubar
 test $? -eq 0 || no_result
 
