From 6509d952827826d73490df6f467411660706b032 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Wed, 27 Jan 2021 08:45:08 +0100
Subject: [PATCH] find fb3/4

---
 cmake/FindInterbase.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/FindInterbase.cmake b/cmake/FindInterbase.cmake
index 22f866d826..b705b88045 100644
--- a/cmake/FindInterbase.cmake
+++ b/cmake/FindInterbase.cmake
@@ -19,12 +19,12 @@
 #     The Interbase client library
 
 find_path(Interbase_INCLUDE_DIR
-          NAMES ibase.h
+          NAMES ibase.h PATHS /opt/firebird3/include  /opt/firebird4/include
           HINTS ${Interbase_INCLUDEDIR}
 )
 
 find_library(Interbase_LIBRARY
-             NAMES firebase_ms fbclient gds
+             NAMES firebase_ms fbclient gds PATHS /opt/firebird3/lib /opt/firebird4/lib
              HINTS ${Interbase_LIBDIR}
 )
 
-- 
2.30.0

