diff --git a/src/corelib/global/qsimd_p.h b/src/corelib/global/qsimd_p.h
index 1feedbbb238b61f8285381f7941de384863de46d..fac49d142a968e0daf73339a442d140369dc0658 100644
--- a/src/corelib/global/qsimd_p.h
+++ b/src/corelib/global/qsimd_p.h
@@ -211,9 +211,9 @@ asm(
 //
 // macOS's fat binaries support the "x86_64h" sub-architecture and the GNU libc
 // ELF loader also supports a "haswell/" subdir (e.g., /usr/lib/haswell).
-#  define ARCH_HASWELL_MACROS       (__AVX2__ + __BMI__ + __BMI2__ + __F16C__ + __FMA__ + __LZCNT__)
+#  define ARCH_HASWELL_MACROS       (__AVX2__ + __BMI__ + __BMI2__ + __FMA__ + __LZCNT__)
 #  if ARCH_HASWELL_MACROS != 0
-#    if ARCH_HASWELL_MACROS != 6
+#    if ARCH_HASWELL_MACROS != 5
 #      error "Please enable all x86-64-v3 extensions; you probably want to use -march=haswell or -march=x86-64-v3 instead of -mavx2"
 #    endif
 static_assert(ARCH_HASWELL_MACROS, "Undeclared identifiers indicate which features are missing.");
