diff --git a/src/streamprocessor/streamprocessor.cpp b/src/streamprocessor/streamprocessor.cpp
index aa27a66e422661149ce1068456134c3820182aa3..b30c96316e8b08ecfe07f9889e36da1dde1cdc8a 100644
--- a/src/streamprocessor/streamprocessor.cpp
+++ b/src/streamprocessor/streamprocessor.cpp
@@ -447,8 +447,8 @@ StreamProcessor::processAudio()
 						timeFrameDuration = frameResampled->nb_samples * 1000 / frameResampled->sample_rate;
 					} else {
 						frameSize = frame->nb_samples * av_get_bytes_per_sample(static_cast<AVSampleFormat>(frame->format));
-						if(frame->pkt_duration)
-							timeFrameDuration = frame->pkt_duration * 1000 * m_avStream->time_base.num / m_avStream->time_base.den;
+						if(frame->duration)
+							timeFrameDuration = frame->duration * 1000 * m_avStream->time_base.num / m_avStream->time_base.den;
 					}
 					timeFrameEnd = timeFrameStart + timeFrameDuration;
 
