From 718b6e51a3fad73e8dd4f6d8f442b51ef3a77432 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 1 Jan 2020 17:32:51 +0100
Subject: [PATCH] use our CFLAGS

---
 Make.rules | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Make.rules b/Make.rules
index ce5b033..7c87064 100644
--- a/Make.rules
+++ b/Make.rules
@@ -46,9 +46,7 @@ $(call set, INSTALL, cp)
 $(call set, INDENT, true)
 
 gcc = $(findstring gcc,$(CC))
-$(call set, CFLAGS, $(if $(gcc), -O2 -Wall, -O))
 $(call set, CPPFLAGS, )
-$(call set, LDFLAGS, -s)
 $(call set, LIBS, )
 
 
-- 
2.24.1

