--- apvlv-0.1.2-Source/src/ApvlvFile.cpp.orig	2011-08-20 09:49:41.549112054 +0200
+++ apvlv-0.1.2-Source/src/ApvlvFile.cpp	2011-08-20 09:51:19.901500636 +0200
@@ -304,7 +304,7 @@
 {
   PopplerRectangle rect = { x1, y1, x2, y2 };
   PopplerPage *page = poppler_document_get_page (mDoc, pn);
-  *out = poppler_page_get_text (page, POPPLER_SELECTION_WORD, &rect);
+  *out = poppler_page_get_selected_text (page, POPPLER_SELECTION_WORD, &rect);
   if (*out != NULL)
     {
       return true;
