--- src/plugins/quota/quota.c.old	2009-09-22 18:06:37.000000000 -0700
+++ src/plugins/quota/quota.c	2009-09-22 18:08:49.000000000 -0700
@@ -444,8 +444,10 @@
 	bool found;
 
 	if (!root->set->force_default_rule) {
-		if (root->backend.v.init_limits(root) < 0)
-			return -1;
+		if (root->backend.v.init_limits != NULL) {
+			if (root->backend.v.init_limits(root) < 0)
+				return -1;
+		}
 	}
 
 	bytes_limit = root->bytes_limit;
