--- spice-0.7.1.orig/client/x11/red_window.cpp	2010-12-16 10:23:56.000000000 -0500
+++ spice-0.7.1/client/x11/red_window.cpp	2011-01-12 23:14:47.362144898 -0500
@@ -2075,13 +2075,13 @@
 #ifdef USE_OGL
 RedGlContext RedWindow::create_context_gl()
 {
-    RedGlContext *context = NULL;
     if (XPlatform::get_fbconfig()[_screen]) {
         XLockDisplay(x_display);
-        context = glXCreateContext(x_display, XPlatform::get_vinfo()[_screen], NULL, GL_TRUE);
+        RedGlContext context = glXCreateContext(x_display, XPlatform::get_vinfo()[_screen], NULL, GL_TRUE);
         XUnlockDisplay(x_display);
+        return context;
     }
-    return context;
+    return NULL;
 }
 
 RedPbuffer RedWindow::create_pbuff(int width, int height)
