diff -Nur psiconv-0.9.6/program/psiconv/gen_html4.c psiconv-0.9.6-fixed/program/psiconv/gen_html4.c
--- psiconv-0.9.6/program/psiconv/gen_html4.c	2004-02-04 13:17:38.000000000 +0100
+++ psiconv-0.9.6-fixed/program/psiconv/gen_html4.c	2004-05-29 19:01:40.000000000 +0200
@@ -317,7 +317,7 @@
 }
                                                                                 
 
-static struct fileformat_s ffs[] =
+static struct fileformat_s psiconv_ffs[] =
   {
     {
       "HTML4",
@@ -337,6 +337,6 @@
 void init_html4(void)
 {
   int i;
-  for (i = 0; ffs[i].name; i++)
-    psiconv_list_add(fileformat_list,ffs+i);
+  for (i = 0; psiconv_ffs[i].name; i++)
+    psiconv_list_add(fileformat_list,psiconv_ffs+i);
 }
diff -Nur psiconv-0.9.6/program/psiconv/gen_xhtml.c psiconv-0.9.6-fixed/program/psiconv/gen_xhtml.c
--- psiconv-0.9.6/program/psiconv/gen_xhtml.c	2004-02-04 12:26:54.000000000 +0100
+++ psiconv-0.9.6-fixed/program/psiconv/gen_xhtml.c	2004-05-29 19:01:46.000000000 +0200
@@ -719,7 +719,7 @@
 }
                                                                                 
 
-static struct fileformat_s ffs[] =
+static struct fileformat_s psiconv_ffs[] =
   {
     {
       "XHTML",
@@ -739,6 +739,6 @@
 void init_xhtml(void)
 {
   int i;
-  for (i = 0; ffs[i].name; i++)
-    psiconv_list_add(fileformat_list,ffs+i);
+  for (i = 0; psiconv_ffs[i].name; i++)
+    psiconv_list_add(fileformat_list,psiconv_ffs+i);
 }
