diff -uNr mm3d-1.2.0-final.old/src/selectfacetool.cc mm3d-1.2.0-final/src/selectfacetool.cc
--- mm3d-1.2.0-final.old/src/selectfacetool.cc	2006-06-01 11:42:07.000000000 +0200
+++ mm3d-1.2.0-final/src/selectfacetool.cc	2006-06-01 11:56:24.000000000 +0200
@@ -131,7 +131,7 @@
          return result;
       }
 
-      bool NormalTest::shouldSelect( void * element )
+      bool shouldSelect( void * element )
       {
          Model::Triangle * tri = static_cast<Model::Triangle *>( element );
          if (tri)
