diff -Naur at-spi-1.30.1.orig/Makefile.in at-spi-1.30.1/Makefile.in
--- at-spi-1.30.1.orig/Makefile.in	2010-10-06 16:18:38.192000020 -0700
+++ at-spi-1.30.1/Makefile.in	2010-10-06 16:18:58.011000022 -0700
@@ -901,11 +901,11 @@
 @INTLTOOL_SCHEMAS_RULE@
 
 @GCONF_SCHEMAS_INSTALL_TRUE@install-data-local:
-@GCONF_SCHEMAS_INSTALL_TRUE@                if test -z "$(DESTDIR)" ; then \
-@GCONF_SCHEMAS_INSTALL_TRUE@                        for p in $(schemas_DATA) ; do \
-@GCONF_SCHEMAS_INSTALL_TRUE@                                GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(top_builddir)/data/$$p ; \
-@GCONF_SCHEMAS_INSTALL_TRUE@                        done \
-@GCONF_SCHEMAS_INSTALL_TRUE@                fi
+@GCONF_SCHEMAS_INSTALL_TRUE@	if test -z "$(DESTDIR)" ; then \
+@GCONF_SCHEMAS_INSTALL_TRUE@		for p in $(schemas_DATA) ; do \
+@GCONF_SCHEMAS_INSTALL_TRUE@			GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(top_builddir)/$$p ; \
+@GCONF_SCHEMAS_INSTALL_TRUE@		done \
+@GCONF_SCHEMAS_INSTALL_TRUE@	fi
 @GCONF_SCHEMAS_INSTALL_FALSE@install-data-local:
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
