From e0f5bcb5008ffb9dfcc40eb8d3a31a59db6897a3 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 29 Jan 2018 20:38:05 +0100
Subject: [PATCH] fix dependencies

---
 zlib-enum.cabal | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/zlib-enum.cabal b/zlib-enum.cabal
index 5bb4614..f04ae6e 100644
--- a/zlib-enum.cabal
+++ b/zlib-enum.cabal
@@ -24,7 +24,7 @@ library
          Buildable: False
     build-depends: base                  >= 4   && < 5
                  , bytestring            >= 0.9 && < 0.11
-                 , transformers          >= 0.2 && < 0.5
+                 , transformers          >= 0.2 && < 0.6
                  , enumerator            == 0.4.*
                  , zlib-bindings         == 0.1.*
     exposed-modules: Codec.Zlib.Enum
@@ -38,12 +38,12 @@ executable zlib-enum-test
         build-depends:
               base                       >= 4.0      &&   < 5.0
             , bytestring                 >= 0.9      &&   < 0.11
-            , transformers               >= 0.2      &&   < 0.5
+            , transformers               >= 0.2      &&   < 0.6
             , enumerator                 >= 0.4      &&   < 0.5
             , zlib-bindings              >= 0.1
-            , test-framework             >= 0.6      &&   < 0.7
-            , test-framework-quickcheck2 >= 0.2.12.1 &&   < 0.3.0
-            , QuickCheck                 >= 2.4      &&   < 2.5
+            , test-framework             >= 0.6      &&   < 0.8
+            , test-framework-quickcheck2 >= 0.2.12.1 &&   < 0.4.0
+            , QuickCheck                 >= 2.4      &&   < 2.12
     else
         Buildable: False
     ghc-options:     -Wall
-- 
2.15.1

