--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ common/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,5 @@
+SUBDIRS = \
+	rdr \
+	network \
+	Xregion \
+	rfb
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ common/rfb/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,136 @@
+lib_LTLIBRARIES = librfb.la
+
+librfb_la_SOURCES = \
+	Blacklist.cxx \
+	Blacklist.h \
+	CConnection.cxx \
+	CConnection.h \
+	CMsgHandler.cxx \
+	CMsgHandler.h \
+	CMsgReader.cxx \
+	CMsgReader.h \
+	CMsgReaderV3.cxx \
+	CMsgReaderV3.h \
+	CMsgWriter.cxx \
+	CMsgWriter.h \
+	CMsgWriterV3.cxx \
+	CMsgWriterV3.h \
+	ColourCube.h \
+	ColourMap.h \
+	ComparingUpdateTracker.cxx \
+	ComparingUpdateTracker.h \
+	Configuration.cxx \
+	Configuration.h \
+	ConnParams.cxx \
+	ConnParams.h \
+	CSecurity.h \
+	CSecurityNone.h \
+	CSecurityVncAuth.cxx \
+	CSecurityVncAuth.h \
+	Cursor.cxx \
+	Cursor.h \
+	Decoder.cxx \
+	Decoder.h \
+	d3des.c \
+	d3des.h \
+	Encoder.cxx \
+	Encoder.h \
+	encodings.cxx \
+	encodings.h \
+	Exception.h \
+	hextileConstants.h \
+	hextileDecode.h \
+	HextileDecoder.cxx \
+	HextileDecoder.h \
+	hextileEncode.h \
+	HextileEncoder.cxx \
+	HextileEncoder.h \
+	Hostname.h \
+	HTTPServer.cxx \
+	HTTPServer.h \
+	ImageGetter.h \
+	InputHandler.h \
+	KeyRemapper.cxx \
+	KeyRemapper.h \
+	keysymdef.h \
+	Logger.cxx \
+	Logger_file.cxx \
+	Logger_file.h \
+	Logger.h \
+	Logger_stdio.cxx \
+	Logger_stdio.h \
+	LogWriter.cxx \
+	LogWriter.h \
+	msgTypes.h \
+	Password.cxx \
+	Password.h \
+	PixelBuffer.cxx \
+	PixelBuffer.h \
+	PixelFormat.cxx \
+	PixelFormat.h \
+	Pixel.h \
+	RawDecoder.cxx \
+	RawDecoder.h \
+	RawEncoder.cxx \
+	RawEncoder.h \
+	Rect.h \
+	Region.cxx \
+	Region.h \
+	rreDecode.h \
+	RREDecoder.cxx \
+	RREDecoder.h \
+	rreEncode.h \
+	RREEncoder.cxx \
+	RREEncoder.h \
+	SConnection.cxx \
+	SConnection.h \
+	SDesktop.h \
+	secTypes.cxx \
+	secTypes.h \
+	ServerCore.cxx \
+	ServerCore.h \
+	SMsgHandler.cxx \
+	SMsgHandler.h \
+	SMsgReader.cxx \
+	SMsgReader.h \
+	SMsgReaderV3.cxx \
+	SMsgReaderV3.h \
+	SMsgWriter.cxx \
+	SMsgWriter.h \
+	SMsgWriterV3.cxx \
+	SMsgWriterV3.h \
+	SSecurityFactoryStandard.cxx \
+	SSecurityFactoryStandard.h \
+	SSecurity.h \
+	SSecurityNone.h \
+	SSecurityVncAuth.cxx \
+	SSecurityVncAuth.h \
+	Threading.h \
+	Timer.cxx \
+	Timer.h \
+	TransImageGetter.cxx \
+	TransImageGetter.h \
+	transInitTempl.h \
+	transTempl.h \
+	TrueColourMap.h \
+	UpdateTracker.cxx \
+	UpdateTracker.h \
+	UserPasswdGetter.h \
+	util.cxx \
+	util.h \
+	VNCSConnectionST.cxx \
+	VNCSConnectionST.h \
+	VNCServer.h \
+	VNCServerST.cxx \
+	VNCServerST.h \
+	zrleDecode.h \
+	ZRLEDecoder.cxx \
+	ZRLEDecoder.h \
+	zrleEncode.h \
+	ZRLEEncoder.cxx \
+	ZRLEEncoder.h
+
+librfb_la_LIBADD = \
+	$(top_srcdir)/network/libnetwork.la \
+	$(top_srcdir)/rdr/librdr.la \
+	$(top_srcdir)/Xregion/libXregion.la
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ common/rdr/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,30 @@
+noinst_LTLIBRARIES = librdr.la
+
+librdr_la_SOURCES = \
+	Exception.cxx \
+	FdInStream.h \
+	FixedMemOutStream.h \
+	HexOutStream.cxx \
+	InStream.h \
+	OutStream.h \
+	SubstitutingInStream.h \
+	ZlibInStream.h \
+	Exception.h \
+	FdOutStream.cxx \
+	HexInStream.cxx \
+	HexOutStream.h \
+	MemInStream.h \
+	RandomStream.cxx \
+	types.h \
+	ZlibOutStream.cxx \
+	FdInStream.cxx \
+	FdOutStream.h \
+	HexInStream.h \
+	InStream.cxx \
+	MemOutStream.h \
+	RandomStream.h \
+	ZlibInStream.cxx \
+	ZlibOutStream.h
+
+librdr_la_LIBADD = \
+	-lz
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ common/configure.ac	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,42 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ(2.60)
+AC_INIT([vnc-libs], 4.1.2, [bugzilla.redhat.com])
+AC_CONFIG_HEADER([config.h])
+AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_LIBTOOL
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_HEADER_STDC
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h limits.h netdb.h netinet/in.h stdlib.h string.h sys/socket.h sys/time.h sys/timeb.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_C_CONST
+AC_C_INLINE
+AC_TYPE_SIZE_T
+AC_HEADER_TIME
+AC_STRUCT_TM
+
+# Checks for library functions.
+AC_FUNC_ERROR_AT_LINE
+AC_FUNC_MEMCMP
+AC_FUNC_SELECT_ARGTYPES
+AC_TYPE_SIGNAL
+AC_FUNC_STRFTIME
+AC_FUNC_VPRINTF
+AC_CHECK_FUNCS([ftime gethostbyname getpass gettimeofday inet_ntoa memchr memmove memset select socket strcasecmp strchr strerror strncasecmp])
+
+AC_CONFIG_FILES([Makefile
+                 Xregion/Makefile
+                 network/Makefile
+                 rdr/Makefile
+                 rfb/Makefile])
+AC_OUTPUT
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ common/Xregion/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,6 @@
+noinst_LTLIBRARIES = libXregion.la
+
+libXregion_la_SOURCES = \
+	Region.c \
+	region.h \
+	Xregion.h
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ common/network/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,6 @@
+noinst_LTLIBRARIES = libnetwork.la
+
+libnetwork_la_SOURCES = \
+	Socket.h \
+	TcpSocket.cxx \
+	TcpSocket.h
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ unix/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,6 @@
+SUBDIRS = \
+	tx \
+	vncpasswd \
+	vncviewer \
+	vncconfig \
+	x0vncserver
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ unix/tx/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,25 @@
+noinst_LTLIBRARIES = libtx.la
+
+libtx_la_SOURCES = \
+	TXButton.h \
+	TXDialog.h \
+	TXEntry.h \
+	TXCheckbox.h \
+	TXImage.cxx \
+	TXImage.h \
+	TXLabel.h \
+	TXMenu.cxx \
+	TXMenu.h \
+	TXMsgBox.h \
+	TXScrollbar.cxx \
+	TXScrollbar.h \
+	TXViewport.cxx \
+	TXViewport.h \
+	TXWindow.cxx \
+	TXWindow.h
+
+AM_CPPFLAGS = \
+	-I$(top_srcdir)/../common/
+
+libtx_la_LIBADD = \
+	-lX11
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ unix/vncpasswd/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,10 @@
+bin_PROGRAMS = vncpasswd
+
+vncpasswd_SOURCES = \
+	vncpasswd.cxx
+
+AM_CPPFLAGS = \
+	-I$(top_srcdir)/../common
+
+vncpasswd_LDADD = \
+	$(top_srcdir)/../common/rfb/librfb.la
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ unix/vncconfig/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,17 @@
+bin_PROGRAMS = vncconfig
+
+vncconfig_SOURCES = \
+	buildtime.c \
+	QueryConnectDialog.cxx \
+	QueryConnectDialog.h \
+	vncconfig.cxx \
+	vncExt.c \
+	vncExt.h
+
+AM_CPPFLAGS = \
+	-I$(top_srcdir)/../common \
+	-I$(top_srcdir)/tx
+
+vncconfig_LDADD = \
+	$(top_srcdir)/../common/rfb/librfb.la \
+	$(top_srcdir)/tx/libtx.la
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ unix/configure.ac	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,52 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ(2.60)
+AC_INIT([vnc], 4.1.2, [bugzilla.redhat.com])
+AC_CONFIG_SRCDIR([x0vncserver/buildtime.c])
+AC_CONFIG_HEADER([config.h])
+AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AC_PROG_LIBTOOL
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_PATH_X
+AC_HEADER_STDC
+AC_HEADER_SYS_WAIT
+AC_CHECK_HEADERS([fcntl.h stdlib.h string.h strings.h sys/time.h termios.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_C_CONST
+AC_C_INLINE
+AC_C_RESTRICT
+AC_TYPE_SIZE_T
+AC_HEADER_TIME
+
+# Checks for library functions.
+AC_FUNC_ERROR_AT_LINE
+AC_FUNC_FORK
+AC_FUNC_LSTAT
+AC_FUNC_LSTAT_FOLLOWS_SLASHED_SYMLINK
+AC_FUNC_MALLOC
+AC_FUNC_MEMCMP
+AC_FUNC_REALLOC
+AC_FUNC_SELECT_ARGTYPES
+AC_TYPE_SIGNAL
+AC_FUNC_STAT
+AC_FUNC_VPRINTF
+AC_FUNC_WAIT3
+AC_CHECK_FUNCS([memset mkdir select setenv strchr strdup strerror strspn strtol])
+
+AC_CONFIG_FILES([Makefile
+                 tx/Makefile
+                 vncconfig/Makefile
+                 vncpasswd/Makefile
+                 vncviewer/Makefile
+                 x0vncserver/Makefile])
+AC_OUTPUT
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ unix/vncviewer/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,24 @@
+bin_PROGRAMS = vncviewer
+
+vncviewer_SOURCES = \
+	AboutDialog.h \
+	buildtime.c \
+	CConn.cxx \
+	CConn.h \
+	DesktopWindow.cxx \
+	DesktopWindow.h \
+	InfoDialog.h \
+	OptionsDialog.h \
+	parameters.h \
+	PasswdDialog.h \
+	ServerDialog.h \
+	vncviewer.cxx
+
+AM_CPPFLAGS = \
+	-I$(top_srcdir)/../common \
+	-I$(top_srcdir)/tx
+
+vncviewer_LDADD = \
+	$(top_srcdir)/../common/rfb/librfb.la \
+	$(top_srcdir)/tx/libtx.la \
+	-lXext
--- /dev/null	2007-02-16 09:39:42.585819347 +0100
+++ unix/x0vncserver/Makefile.am	2007-02-16 09:51:11.000000000 +0100
@@ -0,0 +1,20 @@
+bin_PROGRAMS = x0vncserver
+
+x0vncserver_SOURCES = \
+	buildtime.c \
+	Image.cxx \
+	Image.h \
+	x0vncserver.cxx \
+	$(top_srcdir)/vncconfig/QueryConnectDialog.h \
+	$(top_srcdir)/vncconfig/QueryConnectDialog.cxx
+
+AM_CPPFLAGS = \
+	-I$(top_srcdir)/../common \
+	-I$(top_srcdir)/vncconfig \
+	-I$(top_srcdir)/tx
+
+x0vncserver_LDADD = \
+	$(top_srcdir)/../common/rfb/librfb.la \
+	$(top_srcdir)/tx/libtx.la \
+	-lXext \
+	-lXtst
