--- unix/xorg-x11-server-source/hw/vnc/Makefile.am.autotools	2007-02-16 09:52:43.000000000 +0100
+++ unix/xorg-x11-server-source/hw/vnc/Makefile.am	2007-02-16 09:53:24.000000000 +0100
@@ -42,10 +42,7 @@
 		../xfree86/common/libcommon.la \
 		-lX11 \
 		libvnccommon.la \
-		../../../../common/rfb/librfb.a \
-		../../../../common/rdr/librdr.a \
-		../../../../common/network/libnetwork.a \
-		../../../../common/Xregion/libXregion.a
+		../../../../common/rfb/librfb.la
 
 Xvnc_CFLAGS =  -DHAVE_DIX_CONFIG_H \
 		-DNO_HW_ONLY_EXTS \
@@ -68,10 +65,7 @@
 libvnc_la_SOURCES = xf86vncModule.cc
 libvnc_la_LIBADD = \
 		libvnccommon.la \
-		../../../../common/rfb/librfb.a \
-		../../../../common/rdr/librdr.a \
-		../../../../common/network/libnetwork.a \
-		../../../../common/Xregion/libXregion.a
+		../../../../common/rfb/librfb.la
 libvnc_la_CXXFLAGS = \
 		-I../../../../common \
 		-I$(top_srcdir)/hw/xfree86/common \