diff -Nur vnc-4.0-unixsrc/rfb/HTTPServer.cxx vnc-4.0-unixsrc-fixed/rfb/HTTPServer.cxx
--- vnc-4.0-unixsrc/rfb/HTTPServer.cxx	2004-04-30 18:25:02.000000000 +0200
+++ vnc-4.0-unixsrc-fixed/rfb/HTTPServer.cxx	2004-06-30 16:19:40.671261545 +0200
@@ -16,6 +16,7 @@
  * USA.
  */
 
+#include <vector>
 #include <rfb/HTTPServer.h>
 #include <rfb/LogWriter.h>
 #include <rfb/util.h>
diff -Nur vnc-4.0-unixsrc/rfb/VNCSConnectionST.cxx vnc-4.0-unixsrc-fixed/rfb/VNCSConnectionST.cxx
--- vnc-4.0-unixsrc/rfb/VNCSConnectionST.cxx	2004-06-04 20:53:18.000000000 +0200
+++ vnc-4.0-unixsrc-fixed/rfb/VNCSConnectionST.cxx	2004-06-30 16:19:56.885194991 +0200
@@ -16,6 +16,7 @@
  * USA.
  */
 
+#include <vector>
 #include <rfb/VNCSConnectionST.h>
 #include <rfb/LogWriter.h>
 #include <rfb/secTypes.h>
diff -Nur vnc-4.0-unixsrc/rfb/VNCServerST.cxx vnc-4.0-unixsrc-fixed/rfb/VNCServerST.cxx
--- vnc-4.0-unixsrc/rfb/VNCServerST.cxx	2004-06-04 20:53:18.000000000 +0200
+++ vnc-4.0-unixsrc-fixed/rfb/VNCServerST.cxx	2004-06-30 16:20:10.135506166 +0200
@@ -53,6 +53,7 @@
 // closingSockets for this purpose.
 
 
+#include <vector>
 #include <rfb/ServerCore.h>
 #include <rfb/VNCServerST.h>
 #include <rfb/VNCSConnectionST.h>
diff -Nur vnc-4.0-unixsrc/x0vncserver/x0vncserver.cxx vnc-4.0-unixsrc-fixed/x0vncserver/x0vncserver.cxx
--- vnc-4.0-unixsrc/x0vncserver/x0vncserver.cxx	2004-06-11 18:22:21.000000000 +0200
+++ vnc-4.0-unixsrc-fixed/x0vncserver/x0vncserver.cxx	2004-06-30 16:20:28.982104065 +0200
@@ -15,6 +15,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
  * USA.
  */
+#include <vector>
 #include <strings.h>
 #include <sys/time.h>
 #include <sys/types.h>
