--- common/network/TcpSocket.h	2009-08-04 11:54:33.309180758 -0700
+++ common/network/TcpSocket.h	2009-08-04 11:54:44.254190679 -0700
@@ -29,7 +29,7 @@
 #define __NETWORK_TCP_SOCKET_H__
 
 #include <network/Socket.h>
-
+#include <cstdlib>
 #include <list>
 
 namespace network {
--- unix/tx/TXImage.h	2009-08-04 11:56:13.726180121 -0700
+++ unix/tx/TXImage.h	2009-08-04 11:56:28.816180860 -0700
@@ -32,7 +32,7 @@
 
 #ifndef __TXIMAGE_H__
 #define __TXIMAGE_H__
-
+#include <cstdlib>
 #include <X11/Xlib.h>
 #include <rfb/PixelBuffer.h>
 #include <rfb/ColourMap.h>
--- unix/x0vncserver/Image.h	2009-08-04 11:57:38.605253378 -0700
+++ unix/x0vncserver/Image.h	2009-08-04 11:57:50.430195205 -0700
@@ -21,7 +21,7 @@
 
 #ifndef __IMAGE_H__
 #define __IMAGE_H__
-
+#include <cstdlib>
 #include <X11/Xlib.h>
 #include <X11/extensions/XShm.h>
 
