From d1b8cb81f53ea37d10d0b553330be4187c1bd39c Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sun, 20 Mar 2016 16:01:57 +0100
Subject: [PATCH] do not include ncursesw/ncurses.h

---
 src/duc/cmd-ui.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/duc/cmd-ui.c b/src/duc/cmd-ui.c
index fa34ad9..c729a93 100644
--- a/src/duc/cmd-ui.c
+++ b/src/duc/cmd-ui.c
@@ -27,7 +27,7 @@
 #endif
 #endif
 #ifdef HAVE_LIBNCURSESW
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif
 
 enum {
-- 
2.7.4

