--- jack-keyboard.c.old	2008-03-04 17:56:51.000000000 +0200
+++ jack-keyboard.c	2008-03-04 17:57:56.000000000 +0200
@@ -394,7 +394,7 @@
 			gtk_list_store_append(connected_to_store, &iter);
 			gtk_list_store_set(connected_to_store, &iter, 0, available[i], -1);
 
-			if (connected != NULL && !strcmp(available[i], connected[0]))
+			if (connected != NULL && connected[0] != NULL && !strcmp(available[i], connected[0]))
 				gtk_combo_box_set_active_iter(GTK_COMBO_BOX(connected_to_combo), &iter);
 		}
 	}
