diff -uNr clucene-core-2.3.3.4.orig/src/core/CLucene/document/DateTools.cpp clucene-core-2.3.3.4/src/core/CLucene/document/DateTools.cpp
--- clucene-core-2.3.3.4.orig/src/core/CLucene/document/DateTools.cpp   2011-03-17 01:21:07.000000000 +0100
+++ clucene-core-2.3.3.4/src/core/CLucene/document/DateTools.cpp        2022-12-13 20:07:53.493774189 +0100
@@ -8,6 +8,7 @@
 
 #include "DateTools.h"
 #include "CLucene/util/Misc.h"
+#include <ctime>
 
 CL_NS_USE(util)
 CL_NS_DEF(document)
