From aeb2a2cb7903fb02bbeaf75e345978deb7fee918 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Wed, 22 Sep 2021 00:18:12 +0200
Subject: [PATCH 1/2] upmendex: ICU fix

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 texk/upmendex/mendex.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/texk/upmendex/mendex.h b/texk/upmendex/mendex.h
index d14e5c66753e..65b12055717c 100644
--- a/texk/upmendex/mendex.h
+++ b/texk/upmendex/mendex.h
@@ -6,6 +6,7 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include <unicode/utf16.h>
 #include <unicode/utypes.h>
 #include <unicode/ustring.h>
 #include <unicode/uchar.h>
-- 
2.33.0

