diff --git a/texk/dvisvgm/dvisvgm-src/src//Makefile.am b/texk/dvisvgm/dvisvgm-src/src/Makefile.am
--- a/texk/dvisvgm/dvisvgm-src/src/Makefile.am
+++ b/texk/dvisvgm/dvisvgm-src/src/Makefile.am
@@ -28,7 +28,7 @@ dvisvgm_LDADD += \
 	$(FREETYPE2_LIBS) \
 	$(FONTFORGE_LIBS) \
 	$(ZLIB_LIBS) \
-	$(LIBGS_LIBS)
+	$(LIBGS_LIBS) -ldl
 
 dvisvgm_DEPENDENCIES = $(noinst_LIBRARIES)
 dvisvgm_DEPENDENCIES += $(KPATHSEA_DEPEND) $(ZLIB_DEPEND) $(FREETYPE2_DEPEND)
diff --git a/texk/dvisvgm/dvisvgm-src/src//Makefile.in b/texk/dvisvgm/dvisvgm-src/src/Makefile.in
--- a/texk/dvisvgm/dvisvgm-src/src/Makefile.in
+++ b/texk/dvisvgm/dvisvgm-src/src/Makefile.in
@@ -482,7 +482,7 @@
 dvisvgm_LDADD = $(noinst_LIBRARIES) ../libs/clipper/libclipper.a \
 	../libs/xxHash/libxxhash.a $(am__append_1) $(KPATHSEA_LIBS) \
 	$(POTRACE_LIBS) $(FREETYPE2_LIBS) $(FONTFORGE_LIBS) \
-	$(ZLIB_LIBS) $(LIBGS_LIBS)
+	$(ZLIB_LIBS) $(LIBGS_LIBS) -ldl
 dvisvgm_DEPENDENCIES = $(noinst_LIBRARIES) $(KPATHSEA_DEPEND) \
 	$(ZLIB_DEPEND) $(FREETYPE2_DEPEND)
 libdvisvgm_a_SOURCES = AGLTable.hpp BasicDVIReader.cpp \
