diff --git a/texk/web2c/synctexdir/am/synctex.am b/texk/web2c/synctexdir/am/synctex.am
--- a/texk/web2c/synctexdir/am/synctex.am
+++ b/texk/web2c/synctexdir/am/synctex.am
@@ -55,7 +55,8 @@
 if SYNCTEX
 syncinclude_HEADERS = \
 	synctexdir/synctex_parser.h \
-	synctexdir/synctex_parser_utils.h
+	synctexdir/synctex_parser_utils.h \
+	synctexdir/synctex_version.h
 
 pkgconfig_DATA = synctexdir/synctex.pc
 endif SYNCTEX
