diff --git a/texk/web2c/Makefile.in b/texk/web2c/Makefile.in
--- a/texk/web2c/Makefile.in
+++ b/texk/web2c/Makefile.in
@@ -1762,7 +1762,7 @@
 MANS = $(dist_man_MANS) $(nodist_man_MANS)
 DATA = $(pkgconfig_DATA)
 am__syncinclude_HEADERS_DIST = synctexdir/synctex_parser.h \
-	synctexdir/synctex_parser_utils.h
+	synctexdir/synctex_parser_utils.h synctexdir/synctex_version.h
 HEADERS = $(syncinclude_HEADERS)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
@@ -4021,7 +4021,8 @@
 pkgconfigdir = ${libdir}/pkgconfig
 @SYNCTEX_TRUE@syncinclude_HEADERS = \
 @SYNCTEX_TRUE@	synctexdir/synctex_parser.h \
-@SYNCTEX_TRUE@	synctexdir/synctex_parser_utils.h
+@SYNCTEX_TRUE@	synctexdir/synctex_parser_utils.h \
+@SYNCTEX_TRUE@	synctexdir/synctex_version.h
 
 @SYNCTEX_TRUE@pkgconfig_DATA = synctexdir/synctex.pc
 @TEX_SYNCTEX_TRUE@tex_ch_synctex = \
diff --git a/texk/web2c/synctexdir/am/synctex.am b/texk/web2c/synctexdir/am/synctex.am
--- a/texk/web2c/synctexdir/am/synctex.am
+++ b/texk/web2c/synctexdir/am/synctex.am
@@ -55,7 +55,8 @@
 if SYNCTEX
 syncinclude_HEADERS = \
 	synctexdir/synctex_parser.h \
-	synctexdir/synctex_parser_utils.h
+	synctexdir/synctex_parser_utils.h \
+	synctexdir/synctex_version.h
 
 pkgconfig_DATA = synctexdir/synctex.pc
 endif SYNCTEX
