diff -uprN libuta-0.4.4.old/uta/blitters.cc libuta-0.4.4/uta/blitters.cc
--- libuta-0.4.4.old/uta/blitters.cc	2003-09-29 15:01:32.000000000 -0700
+++ libuta-0.4.4/uta/blitters.cc	2003-09-29 15:09:07.000000000 -0700
@@ -19,6 +19,7 @@
 */
 
 #include "blitters.h"
+#include <assert.h>
 #include <painter.h>
 
 namespace uta {
diff -uprN libuta-0.4.4.old/uta/painter.cc libuta-0.4.4/uta/painter.cc
--- libuta-0.4.4.old/uta/painter.cc	2003-09-29 15:01:32.000000000 -0700
+++ libuta-0.4.4/uta/painter.cc	2003-09-29 15:08:46.000000000 -0700
@@ -20,6 +20,7 @@
 */
 // written by Karsten Laux, June 1999  
 
+#include <assert.h>
 #include "painter.h"
 #include "sge_blib.h"
 
