diff -Naupr KWStyle/kwsCheckBlackList.cxx KWStyle.fixed/kwsCheckBlackList.cxx
--- KWStyle/kwsCheckBlackList.cxx	2006-04-05 19:36:48.000000000 +0200
+++ KWStyle.fixed/kwsCheckBlackList.cxx	2012-03-21 15:19:21.529638741 +0100
@@ -12,6 +12,7 @@
 
 =========================================================================*/
 #include "kwsParser.h"
+#include <cstring>
 
 namespace kws {
 
diff -Naupr KWStyle/kwsCheckHeader.cxx KWStyle.fixed/kwsCheckHeader.cxx
--- KWStyle/kwsCheckHeader.cxx	2007-02-26 11:30:42.000000000 +0100
+++ KWStyle.fixed/kwsCheckHeader.cxx	2012-03-21 15:18:37.392886548 +0100
@@ -12,6 +12,7 @@
 
 =========================================================================*/
 #include "kwsParser.h"
+#include <cstring>
 
 namespace kws {
 
diff -Naupr KWStyle/kwsCheckOperator.cxx KWStyle.fixed/kwsCheckOperator.cxx
--- KWStyle/kwsCheckOperator.cxx	2006-10-26 18:58:28.000000000 +0200
+++ KWStyle.fixed/kwsCheckOperator.cxx	2012-03-21 15:18:59.113256725 +0100
@@ -12,6 +12,7 @@
 
 =========================================================================*/
 #include "kwsParser.h"
+#include <cstring>
 
 namespace kws {
 
diff -Naupr KWStyle/kwsGenerator.cxx KWStyle.fixed/kwsGenerator.cxx
--- KWStyle/kwsGenerator.cxx	2007-05-11 12:49:08.000000000 +0200
+++ KWStyle.fixed/kwsGenerator.cxx	2012-03-21 15:18:20.016590395 +0100
@@ -13,6 +13,7 @@
 =========================================================================*/
 #include "kwsGenerator.h"
 #include "kwsXMLReader.h"
+#include <cstring>
 
 namespace kws {
 
diff -Naupr KWStyle/kwsParser.cxx KWStyle.fixed/kwsParser.cxx
--- KWStyle/kwsParser.cxx	2007-05-14 09:50:00.000000000 +0200
+++ KWStyle.fixed/kwsParser.cxx	2012-03-21 15:17:47.672039095 +0100
@@ -14,6 +14,7 @@
 #include "kwsParser.h"
 
 #include <kwssys/ios/sstream>
+#include <cstring>
 
 namespace kws {
 
diff -Naupr KWStyle/kwsStyle.cxx KWStyle.fixed/kwsStyle.cxx
--- KWStyle/kwsStyle.cxx	2007-05-11 12:05:46.000000000 +0200
+++ KWStyle.fixed/kwsStyle.cxx	2012-03-21 15:19:37.945918495 +0100
@@ -23,6 +23,7 @@
 #include <sstream>
 #include "kwsXMLReader.h"
 #include <vector>
+#include <cstring>
 
 #if defined(WIN32) && !defined(__CYGWIN__)
   #include <direct.h> // mkdir needs it
diff -Naupr KWStyle/kwsTestFile.h KWStyle.fixed/kwsTestFile.h
--- KWStyle/kwsTestFile.h	2006-04-05 19:38:20.000000000 +0200
+++ KWStyle.fixed/kwsTestFile.h	2012-03-21 15:20:07.010413761 +0100
@@ -16,6 +16,8 @@
 
 #include "itkXMLDefs.h"
 
+#include <cstdlib>
+
 namespace kws
 {
 
diff -Naupr KWStyle/kwsXMLReader.cxx KWStyle.fixed/kwsXMLReader.cxx
--- KWStyle/kwsXMLReader.cxx	2007-01-07 17:31:16.000000000 +0100
+++ KWStyle.fixed/kwsXMLReader.cxx	2012-03-21 15:18:07.216372224 +0100
@@ -12,6 +12,7 @@
 
 =========================================================================*/
 #include "kwsXMLReader.h"
+#include <cstring>
 
 namespace kws{
 
diff -Naupr KWStyle/metaCommand.cxx KWStyle.fixed/metaCommand.cxx
--- KWStyle/metaCommand.cxx	2006-07-27 15:14:40.000000000 +0200
+++ KWStyle.fixed/metaCommand.cxx	2012-03-21 15:17:27.183689859 +0100
@@ -18,6 +18,7 @@
 
 #include <stdio.h>
 #include <string>
+#include <cstring>
 
 #if (METAIO_USE_NAMESPACE)
 namespace METAIO_NAMESPACE {
diff -Naupr KWStyle/Testing/kwsTestMain.h KWStyle.fixed/Testing/kwsTestMain.h
--- KWStyle/Testing/kwsTestMain.h	2007-05-11 12:58:38.000000000 +0200
+++ KWStyle.fixed/Testing/kwsTestMain.h	2012-03-21 15:20:51.559172819 +0100
@@ -26,6 +26,8 @@
 // specified on the command line.
 #include <map>
 #include <string>
+#include <cstring>
+#include <cstdlib>
 #include <iostream>
 #include <fstream>
 #include "kwssys/SystemTools.hxx"
