--- beast-0.7.1.orig/bse/Makefile.in
+++ beast-0.7.1/bse/Makefile.in
@@ -422,7 +422,7 @@
 # need -I$(srcdir) for "bseserver.h" in .genprc.c
 # need -I. (builddir) for "bsecore.genidl.hh" in bsecore.cc
 INCLUDES = -I$(top_srcdir) -I$(top_builddir) -I$(srcdir) -I. \
-	$(BSE_CFLAGS) -DG_DISABLE_DEPRECATED -DG_DISABLE_CONST_RETURNS
+	$(BSE_CFLAGS) -DG_DISABLE_CONST_RETURNS
 EXTRA_HEADERS =
 MAINTAINERCLEANFILES = $(GENERATED) $(am__append_2)
 #	-Wl,-Bsymbolic
only in patch2:
unchanged:
