--- shell/Makefile.in~	2005-04-23 02:22:56.000000000 +0200
+++ shell/Makefile.in	2005-05-02 19:49:16.000000000 +0200
@@ -41,7 +41,7 @@
 @ENABLE_DJVU_TRUE@	$(NULL)
 
 @ENABLE_DVI_TRUE@am__append_2 = \
-@ENABLE_DVI_TRUE@	$(top_builddir)/dvi/libgtkdvi.la	\
+@ENABLE_DVI_TRUE@	$(top_builddir)/dvi/libgtkdvi.la $(top_builddir)/dvi/mdvi-lib/libmdvi.la	\
 @ENABLE_DVI_TRUE@	$(NULL)
 
 @WITH_TYPE1_FONTS_TRUE@am__append_3 = -lt1lib
--- dvi/Makefile.in~	2005-05-07 12:17:41.000000000 +0200
+++ dvi/Makefile.in	2005-05-07 14:21:54.000000000 +0200
@@ -36,7 +36,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 host_triplet = @host@
-@WITH_TYPE1_FONTS_TRUE@am__append_1 = -lt1lib
+@WITH_TYPE1_FONTS_TRUE@am__append_1 = -lt1 -lm
 ACLOCAL = @ACLOCAL@
 ACLOCAL_AMFLAGS = @ACLOCAL_AMFLAGS@
 AMDEP_FALSE = @AMDEP_FALSE@
--- configure~	2005-05-07 14:07:33.000000000 +0200
+++ configure	2005-05-07 14:05:06.000000000 +0200
@@ -26198,7 +26198,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lt1lib  $LIBS"
+LIBS="-lt1 -lm  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
--- dvi/mdvi-lib/t1.c~	2005-04-07 17:28:04.000000000 +0200
+++ dvi/mdvi-lib/t1.c	2005-05-07 14:36:06.000000000 +0200
@@ -174,7 +174,7 @@
 	DEBUG((DBG_TYPE1, "(t1) resetting device resolution (current: (%d,%d))\n",
 		t1lib_xdpi, t1lib_ydpi));
 
-	nfonts = T1_Get_no_fonts();
+	nfonts = T1_GetNoFonts();
 	for(i = 0; i < nfonts; i++)
 		T1_DeleteAllSizes(i);
 	/* reset device resolutions */
