--- src/udiskslinuxmanager.h.orig	2011-10-10 22:13:43.000000000 +0400
+++ src/udiskslinuxmanager.h	2012-11-24 17:52:35.021912864 +0400
@@ -23,6 +23,13 @@
 
 #include "udisksdaemontypes.h"
 
+/* /dev/loop-control interface */
+#ifndef LOOP_CTL_ADD
+# define LOOP_CTL_ADD      0x4C80
+# define LOOP_CTL_REMOVE   0x4C81
+# define LOOP_CTL_GET_FREE 0x4C82
+#endif
+
 G_BEGIN_DECLS
 
 #define UDISKS_TYPE_LINUX_MANAGER  (udisks_linux_manager_get_type ())
