diff -pruN wxwidgets2.6-2.6.3.2.1.5ubuntu9/src/gtk/button.cpp wxwidgets2.6-2.6.3.2.1.5ubuntu10/src/gtk/button.cpp
--- wxwidgets2.6-2.6.3.2.1.5ubuntu9/src/gtk/button.cpp	2006-05-01 07:38:40.000000000 +0300
+++ wxwidgets2.6-2.6.3.2.1.5ubuntu10/src/gtk/button.cpp	2007-06-25 23:40:33.000000000 +0300
@@ -89,7 +89,7 @@ gtk_button_style_set_callback( GtkWidget
             right_border += default_border->right;
             top_border += default_border->top;
             bottom_border += default_border->bottom;
-            g_free( default_border );
+            gtk_border_free( default_border );
         }
 #else
         left_border = 6;
diff -pruN wxwidgets2.6-2.6.3.2.1.5ubuntu9/src/gtk/window.cpp wxwidgets2.6-2.6.3.2.1.5ubuntu10/src/gtk/window.cpp
--- wxwidgets2.6-2.6.3.2.1.5ubuntu9/src/gtk/window.cpp	2006-05-01 07:38:40.000000000 +0300
+++ wxwidgets2.6-2.6.3.2.1.5ubuntu10/src/gtk/window.cpp	2007-06-25 23:47:35.000000000 +0300
@@ -3276,7 +3276,7 @@ void wxWindowGTK::DoSetSize( int x, int 
                 right_border += default_border->right;
                 top_border += default_border->top;
                 bottom_border += default_border->bottom;
-                g_free( default_border );
+                gtk_border_free( default_border );
             }
 #else
             left_border = 6;
