diff -Naupr wxGTK-2.8.10.orig/src/gtk/checklst.cpp wxGTK-2.8.10/src/gtk/checklst.cpp
--- wxGTK-2.8.10.orig/src/gtk/checklst.cpp	2009-09-09 13:00:26.444051291 +0200
+++ wxGTK-2.8.10/src/gtk/checklst.cpp	2009-09-09 13:09:57.719509230 +0200
@@ -17,7 +17,9 @@
 #include "wx/gtk/private.h"
 #include "wx/gtk/treeentry_gtk.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 
 //-----------------------------------------------------------------------------
diff -Naupr wxGTK-2.8.10.orig/src/gtk/colour.cpp wxGTK-2.8.10/src/gtk/colour.cpp
--- wxGTK-2.8.10.orig/src/gtk/colour.cpp	2009-09-09 13:00:26.296233293 +0200
+++ wxGTK-2.8.10/src/gtk/colour.cpp	2009-09-09 13:10:05.992841830 +0200
@@ -14,7 +14,9 @@
 
 #include "wx/gtk/private.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 
 //-----------------------------------------------------------------------------
 // wxColour
diff -Naupr wxGTK-2.8.10.orig/src/gtk/dcmemory.cpp wxGTK-2.8.10/src/gtk/dcmemory.cpp
--- wxGTK-2.8.10.orig/src/gtk/dcmemory.cpp	2009-09-09 13:00:26.299566494 +0200
+++ wxGTK-2.8.10/src/gtk/dcmemory.cpp	2009-09-09 13:10:14.689507541 +0200
@@ -12,7 +12,9 @@
 
 #include "wx/dcmemory.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 
 //-----------------------------------------------------------------------------
diff -Naupr wxGTK-2.8.10.orig/src/gtk/dcscreen.cpp wxGTK-2.8.10/src/gtk/dcscreen.cpp
--- wxGTK-2.8.10.orig/src/gtk/dcscreen.cpp	2009-09-09 13:00:26.479942581 +0200
+++ wxGTK-2.8.10/src/gtk/dcscreen.cpp	2009-09-09 13:10:26.484662954 +0200
@@ -16,7 +16,9 @@
     #include "wx/window.h"
 #endif
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gdk/gdkx.h>
 #include <gtk/gtk.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk/dialog.cpp wxGTK-2.8.10/src/gtk/dialog.cpp
--- wxGTK-2.8.10.orig/src/gtk/dialog.cpp	2009-09-09 13:00:26.410098837 +0200
+++ wxGTK-2.8.10/src/gtk/dialog.cpp	2009-09-09 13:10:38.459921951 +0200
@@ -20,7 +20,9 @@
 
 #include "wx/evtloop.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk/gsockgtk.cpp wxGTK-2.8.10/src/gtk/gsockgtk.cpp
--- wxGTK-2.8.10.orig/src/gtk/gsockgtk.cpp	2009-09-09 13:00:26.296233293 +0200
+++ wxGTK-2.8.10/src/gtk/gsockgtk.cpp	2009-09-09 13:10:47.456063068 +0200
@@ -15,8 +15,10 @@
 #include <stdlib.h>
 #include <stdio.h>
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
 #include <glib.h>
+#undef GSocket
 
 #include "wx/gsocket.h"
 #include "wx/unix/gsockunx.h"
diff -Naupr wxGTK-2.8.10.orig/src/gtk/listbox.cpp wxGTK-2.8.10/src/gtk/listbox.cpp
--- wxGTK-2.8.10.orig/src/gtk/listbox.cpp	2009-09-09 13:00:26.463316585 +0200
+++ wxGTK-2.8.10/src/gtk/listbox.cpp	2009-09-09 13:10:58.601494018 +0200
@@ -32,7 +32,9 @@
     #include "wx/tooltip.h"
 #endif
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk/mdi.cpp wxGTK-2.8.10/src/gtk/mdi.cpp
--- wxGTK-2.8.10.orig/src/gtk/mdi.cpp	2009-09-09 13:00:26.296233293 +0200
+++ wxGTK-2.8.10/src/gtk/mdi.cpp	2009-09-09 13:11:10.309501948 +0200
@@ -23,8 +23,10 @@
 #include "wx/notebook.h"
 #include "wx/gtk/private.h"
 
+#define GSocket GlibGSocket
 #include <glib.h>
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 #include "wx/gtk/win_gtk.h"
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk/pen.cpp wxGTK-2.8.10/src/gtk/pen.cpp
--- wxGTK-2.8.10.orig/src/gtk/pen.cpp	2009-09-09 13:00:26.487482771 +0200
+++ wxGTK-2.8.10/src/gtk/pen.cpp	2009-09-09 13:11:22.560018111 +0200
@@ -16,7 +16,9 @@
     #include "wx/colour.h"
 #endif
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 
 //-----------------------------------------------------------------------------
 // wxPen
diff -Naupr wxGTK-2.8.10.orig/src/gtk/popupwin.cpp wxGTK-2.8.10/src/gtk/popupwin.cpp
--- wxGTK-2.8.10.orig/src/gtk/popupwin.cpp	2009-09-09 13:00:26.452899826 +0200
+++ wxGTK-2.8.10/src/gtk/popupwin.cpp	2009-09-09 13:11:31.290815154 +0200
@@ -20,7 +20,9 @@
     #include "wx/cursor.h"
 #endif // WX_PRECOMP
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/bitmap.cpp wxGTK-2.8.10/src/gtk1/bitmap.cpp
--- wxGTK-2.8.10.orig/src/gtk1/bitmap.cpp	2009-09-09 13:00:23.832900225 +0200
+++ wxGTK-2.8.10/src/gtk1/bitmap.cpp	2009-09-09 13:06:11.866198603 +0200
@@ -23,7 +23,9 @@
 
 #include "wx/filefn.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 #include <gdk/gdkx.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/brush.cpp wxGTK-2.8.10/src/gtk1/brush.cpp
--- wxGTK-2.8.10.orig/src/gtk1/brush.cpp	2009-09-09 13:00:23.370005417 +0200
+++ wxGTK-2.8.10/src/gtk1/brush.cpp	2009-09-09 13:06:24.132864099 +0200
@@ -16,7 +16,9 @@
     #include "wx/colour.h"
 #endif
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 
 //-----------------------------------------------------------------------------
 // wxBrush
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/checklst.cpp wxGTK-2.8.10/src/gtk1/checklst.cpp
--- wxGTK-2.8.10.orig/src/gtk1/checklst.cpp	2009-09-09 13:00:24.749566738 +0200
+++ wxGTK-2.8.10/src/gtk1/checklst.cpp	2009-09-09 13:06:36.083053141 +0200
@@ -15,7 +15,9 @@
 #include "wx/checklst.h"
 #include "wx/gtk1/private.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 
 //-----------------------------------------------------------------------------
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/clipbrd.cpp wxGTK-2.8.10/src/gtk1/clipbrd.cpp
--- wxGTK-2.8.10.orig/src/gtk1/clipbrd.cpp	2009-09-09 13:00:23.339566955 +0200
+++ wxGTK-2.8.10/src/gtk1/clipbrd.cpp	2009-09-09 13:06:49.169938590 +0200
@@ -20,8 +20,10 @@
     #include "wx/dataobj.h"
 #endif
 
+#define GSocket GlibGSocket
 #include <glib.h>
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 
 //-----------------------------------------------------------------------------
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/colour.cpp wxGTK-2.8.10/src/gtk1/colour.cpp
--- wxGTK-2.8.10.orig/src/gtk1/colour.cpp	2009-09-09 13:00:23.689566774 +0200
+++ wxGTK-2.8.10/src/gtk1/colour.cpp	2009-09-09 13:06:59.551548662 +0200
@@ -18,7 +18,9 @@
 
 #include "wx/gtk1/private.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gdk/gdkx.h>
 #include <gdk/gdkprivate.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/cursor.cpp wxGTK-2.8.10/src/gtk1/cursor.cpp
--- wxGTK-2.8.10.orig/src/gtk1/cursor.cpp	2009-09-09 13:00:23.379566682 +0200
+++ wxGTK-2.8.10/src/gtk1/cursor.cpp	2009-09-09 13:07:08.097026441 +0200
@@ -17,7 +17,9 @@
     #include "wx/utils.h"
 #endif // WX_PRECOMP
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 
 //-----------------------------------------------------------------------------
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/dataobj.cpp wxGTK-2.8.10/src/gtk1/dataobj.cpp
--- wxGTK-2.8.10.orig/src/gtk1/dataobj.cpp	2009-09-09 13:00:25.807886309 +0200
+++ wxGTK-2.8.10/src/gtk1/dataobj.cpp	2009-09-09 13:07:17.808666606 +0200
@@ -23,7 +23,9 @@
 #include "wx/mstream.h"
 #include "wx/uri.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 
 //-------------------------------------------------------------------------
 // global data
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/dcclient.cpp wxGTK-2.8.10/src/gtk1/dcclient.cpp
--- wxGTK-2.8.10.orig/src/gtk1/dcclient.cpp	2009-09-09 13:00:24.700932560 +0200
+++ wxGTK-2.8.10/src/gtk1/dcclient.cpp	2009-09-09 13:07:29.259524380 +0200
@@ -28,7 +28,9 @@
 
 #include "wx/gtk1/win_gtk.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gdk/gdkx.h>
 #include <gdk/gdkprivate.h>
 #include <gtk/gtk.h>
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/dc.cpp wxGTK-2.8.10/src/gtk1/dc.cpp
--- wxGTK-2.8.10.orig/src/gtk1/dc.cpp	2009-09-09 13:00:24.672899840 +0200
+++ wxGTK-2.8.10/src/gtk1/dc.cpp	2009-09-09 13:07:39.036190063 +0200
@@ -12,7 +12,9 @@
 
 #include "wx/dc.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 
 //-----------------------------------------------------------------------------
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/dcmemory.cpp wxGTK-2.8.10/src/gtk1/dcmemory.cpp
--- wxGTK-2.8.10.orig/src/gtk1/dcmemory.cpp	2009-09-09 13:00:24.236764536 +0200
+++ wxGTK-2.8.10/src/gtk1/dcmemory.cpp	2009-09-09 13:07:50.597700851 +0200
@@ -12,7 +12,9 @@
 
 #include "wx/dcmemory.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 
 //-----------------------------------------------------------------------------
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/dcscreen.cpp wxGTK-2.8.10/src/gtk1/dcscreen.cpp
--- wxGTK-2.8.10.orig/src/gtk1/dcscreen.cpp	2009-09-09 13:00:25.832899863 +0200
+++ wxGTK-2.8.10/src/gtk1/dcscreen.cpp	2009-09-09 13:08:07.866187647 +0200
@@ -16,7 +16,9 @@
     #include "wx/window.h"
 #endif
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gdk/gdkx.h>
 #include <gtk/gtk.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/dialog.cpp wxGTK-2.8.10/src/gtk1/dialog.cpp
--- wxGTK-2.8.10.orig/src/gtk1/dialog.cpp	2009-09-09 13:00:24.271183641 +0200
+++ wxGTK-2.8.10/src/gtk1/dialog.cpp	2009-09-09 13:08:18.880334529 +0200
@@ -20,7 +20,9 @@
 
 #include "wx/evtloop.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/frame.cpp wxGTK-2.8.10/src/gtk1/frame.cpp
--- wxGTK-2.8.10.orig/src/gtk1/frame.cpp	2009-09-09 13:00:23.336233390 +0200
+++ wxGTK-2.8.10/src/gtk1/frame.cpp	2009-09-09 13:12:46.792888514 +0200
@@ -30,7 +30,9 @@
     #include "wx/statusbr.h"
 #endif // WX_PRECOMP
 
+#define GSocket GlibGSocket
 #include <glib.h>
+#undef GSocket
 #include "wx/gtk1/private.h"
 
 #include <gdk/gdkkeysyms.h>
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/gsockgtk.cpp wxGTK-2.8.10/src/gtk1/gsockgtk.cpp
--- wxGTK-2.8.10.orig/src/gtk1/gsockgtk.cpp	2009-09-09 13:00:23.689566774 +0200
+++ wxGTK-2.8.10/src/gtk1/gsockgtk.cpp	2009-09-09 13:08:36.662851137 +0200
@@ -15,8 +15,10 @@
 #include <stdlib.h>
 #include <stdio.h>
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
 #include <glib.h>
+#undef GSocket
 
 #include "wx/gsocket.h"
 #include "wx/unix/gsockunx.h"
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/listbox.cpp wxGTK-2.8.10/src/gtk1/listbox.cpp
--- wxGTK-2.8.10.orig/src/gtk1/listbox.cpp	2009-09-09 13:00:25.782900228 +0200
+++ wxGTK-2.8.10/src/gtk1/listbox.cpp	2009-09-09 13:08:47.052849741 +0200
@@ -29,7 +29,9 @@
     #include "wx/tooltip.h"
 #endif
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/mdi.cpp wxGTK-2.8.10/src/gtk1/mdi.cpp
--- wxGTK-2.8.10.orig/src/gtk1/mdi.cpp	2009-09-09 13:00:23.800739946 +0200
+++ wxGTK-2.8.10/src/gtk1/mdi.cpp	2009-09-09 13:08:56.206812100 +0200
@@ -23,8 +23,10 @@
 #include "wx/notebook.h"
 #include "wx/gtk1/private.h"
 
+#define GSocket GlibGSocket
 #include <glib.h>
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 #include "wx/gtk1/win_gtk.h"
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/minifram.cpp wxGTK-2.8.10/src/gtk1/minifram.cpp
--- wxGTK-2.8.10.orig/src/gtk1/minifram.cpp	2009-09-09 13:00:24.819844793 +0200
+++ wxGTK-2.8.10/src/gtk1/minifram.cpp	2009-09-09 13:09:07.377676869 +0200
@@ -22,7 +22,9 @@
 #include "wx/gtk1/win_gtk.h"
 #include "wx/gtk1/private.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gdk/gdkprivate.h>
 #include <gdk/gdkx.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/pen.cpp wxGTK-2.8.10/src/gtk1/pen.cpp
--- wxGTK-2.8.10.orig/src/gtk1/pen.cpp	2009-09-09 13:00:25.866325248 +0200
+++ wxGTK-2.8.10/src/gtk1/pen.cpp	2009-09-09 13:09:15.342846791 +0200
@@ -16,7 +16,9 @@
     #include "wx/colour.h"
 #endif
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 
 //-----------------------------------------------------------------------------
 // wxPen
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/popupwin.cpp wxGTK-2.8.10/src/gtk1/popupwin.cpp
--- wxGTK-2.8.10.orig/src/gtk1/popupwin.cpp	2009-09-09 13:00:24.956567244 +0200
+++ wxGTK-2.8.10/src/gtk1/popupwin.cpp	2009-09-09 13:09:24.590053283 +0200
@@ -20,7 +20,9 @@
     #include "wx/cursor.h"
 #endif // WX_PRECOMP
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/settings.cpp wxGTK-2.8.10/src/gtk1/settings.cpp
--- wxGTK-2.8.10.orig/src/gtk1/settings.cpp	2009-09-09 13:00:23.756233537 +0200
+++ wxGTK-2.8.10/src/gtk1/settings.cpp	2009-09-09 13:09:33.916908007 +0200
@@ -20,7 +20,9 @@
 
 #include "wx/fontutil.h"
 
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gdk/gdkx.h>
 #include <gdk/gdkprivate.h>
 #include <gtk/gtk.h>
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/tbargtk.cpp wxGTK-2.8.10/src/gtk1/tbargtk.cpp
--- wxGTK-2.8.10.orig/src/gtk1/tbargtk.cpp	2009-09-09 13:00:25.832899863 +0200
+++ wxGTK-2.8.10/src/gtk1/tbargtk.cpp	2009-09-09 13:13:02.026157608 +0200
@@ -27,7 +27,9 @@
     #include "wx/frame.h"
 #endif
 
+#define GSocket GlibGSocket
 #include <glib.h>
+#undef GSocket
 #include "wx/gtk1/private.h"
 
 // ----------------------------------------------------------------------------
diff -Naupr wxGTK-2.8.10.orig/src/gtk1/toplevel.cpp wxGTK-2.8.10/src/gtk1/toplevel.cpp
--- wxGTK-2.8.10.orig/src/gtk1/toplevel.cpp	2009-09-09 13:00:25.807886309 +0200
+++ wxGTK-2.8.10/src/gtk1/toplevel.cpp	2009-09-09 13:13:10.386249880 +0200
@@ -37,8 +37,10 @@
 #include "wx/gtk1/private.h"
 #include "wx/evtloop.h"
 
+#define GSocket GlibGSocket
 #include <glib.h>
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 #include <gdk/gdkx.h>
diff -Naupr wxGTK-2.8.10.orig/src/unix/fontutil.cpp wxGTK-2.8.10/src/unix/fontutil.cpp
--- wxGTK-2.8.10.orig/src/unix/fontutil.cpp	2009-09-09 13:00:27.852899602 +0200
+++ wxGTK-2.8.10/src/unix/fontutil.cpp	2009-09-09 13:11:46.569498482 +0200
@@ -390,7 +390,9 @@ bool wxGetNativeFontEncoding(wxFontEncod
     // known not to work at least with Sun CC 6.01
     #include <time.h>
 
+    #define GSocket GlibGSocket
     #include <gdk/gdk.h>
+    #undef GSocket
 #endif
 
 
diff -Naupr wxGTK-2.8.10.orig/src/unix/taskbarx11.cpp wxGTK-2.8.10/src/unix/taskbarx11.cpp
--- wxGTK-2.8.10.orig/src/unix/taskbarx11.cpp	2009-09-09 13:00:27.871904636 +0200
+++ wxGTK-2.8.10/src/unix/taskbarx11.cpp	2009-09-09 13:11:58.949669771 +0200
@@ -76,7 +76,9 @@
 // ----------------------------------------------------------------------------
 
 #if defined(__WXGTK__)
+    #define GSocket GlibGSocket
     #include <gdk/gdk.h>
+    #undef GSocket
     #include <gdk/gdkx.h>
     #include <gtk/gtk.h>
     #define GetDisplay()        GDK_DISPLAY()
diff -Naupr wxGTK-2.8.10.orig/src/unix/utilsx11.cpp wxGTK-2.8.10/src/unix/utilsx11.cpp
--- wxGTK-2.8.10.orig/src/unix/utilsx11.cpp	2009-09-09 13:00:28.375125570 +0200
+++ wxGTK-2.8.10/src/unix/utilsx11.cpp	2009-09-09 13:12:10.642829304 +0200
@@ -36,7 +36,9 @@
 #endif
 
 #ifdef __WXGTK__
+#define GSocket GlibGSocket
 #include <gdk/gdk.h>
+#undef GSocket
 #include <gdk/gdkx.h>
 #endif
 
