--- qwik-smtpd-0.3.orig/qwik-smtpd.c	2002-06-27 22:35:24.000000000 -0700
+++ qwik-smtpd.c	2004-10-30 19:56:14.579287608 -0700
@@ -208,7 +208,7 @@
               for(x = 0; x < max_recipients; x++)
               {
                 if(clientRcptTo[x] == NULL) break;
-                fprintf(fpout,clientRcptTo[x]);
+                fprintf(fpout,"%s", clientRcptTo[x]);
                 (void) fflush(fpout);
                 fprintf(fpout,"\n");
                 (void) fflush(fpout);
@@ -431,9 +431,9 @@
           }
           else
           {
-            fprintf(fpout,Received);
+            fprintf(fpout,"%s", Received);
             (void) fflush(fpout);
-            fprintf(fpout,messageID);
+            fprintf(fpout,"%s", messageID);
             (void) fflush(fpout);
             out(354, "type away!");
             alarm(data_timeout);
