--- src/dbiwrapped-0.0.1/Query.cpp.orig	2010-09-07 01:55:57.000000000 +0400
+++ src/dbiwrapped-0.0.1/Query.cpp	2013-01-21 15:58:14.000000000 +0400
@@ -50,6 +50,8 @@
 #include <dbi/dbi.h>
 #endif
 
+#include <boost/shared_ptr.hpp>
+
 #include "Database.h"
 #include "Query.h"
 #include "../dbierror.h"
--- src/dbiwrapped-0.0.1/StderrLog.cpp.orig	2010-09-07 01:55:57.000000000 +0400
+++ src/dbiwrapped-0.0.1/StderrLog.cpp	2013-01-21 16:04:30.000000000 +0400
@@ -42,6 +42,8 @@
 #include <time.h>
 #endif
 
+#include <boost/shared_ptr.hpp>
+
 #include "Database.h"
 #include "Query.h"
 #include "IError.h"
--- src/dbiwrapped-0.0.1/SysLog.cpp.orig	2010-09-07 01:55:57.000000000 +0400
+++ src/dbiwrapped-0.0.1/SysLog.cpp	2013-01-21 16:22:06.393442186 +0400
@@ -35,6 +35,8 @@
 #include <syslog.h>
 #include <cstring>
 
+#include <boost/shared_ptr.hpp>
+
 #include "Database.h"
 #include "Query.h"
 #include "IError.h"
--- src/simplescheduler.h.orig	2010-09-07 01:55:58.000000000 +0400
+++ src/simplescheduler.h	2013-01-21 16:31:11.000000000 +0400
@@ -21,6 +21,7 @@
 #include <map>
 #include <deque>
 #include <boost/thread.hpp>
+#include <boost/thread/condition.hpp>
 #include <boost/smart_ptr.hpp>
 #include <boost/detail/atomic_count.hpp>
 // #include "osdependant.h"
--- src/core.cpp.orig	2010-09-07 01:55:58.000000000 +0400
+++ src/core.cpp	2013-01-21 16:35:54.000000000 +0400
@@ -16,6 +16,9 @@
 */
 #include <iostream>
 #include <vector>
+
+#include <boost/shared_ptr.hpp>
+
 #include "defensive.h"
 #include "triplet.h"
 #include "core.h"
--- src/greylist.cpp.orig	2010-09-07 01:55:58.000000000 +0400
+++ src/greylist.cpp	2013-01-21 16:39:12.000000000 +0400
@@ -1,5 +1,8 @@
 /// $Id: greylist.cpp 21 2009-09-05 23:07:59Z mimo $
 #include <iostream>
+
+#include <boost/shared_ptr.hpp>
+
 // #include "greylist.h"
 #include "dbiwrapped.h"
 
