--- src/dbiwrapped-0.0.1/StderrLog.cpp.orig	2010-09-07 01:55:57.000000000 +0400
+++ src/dbiwrapped-0.0.1/StderrLog.cpp	2013-01-21 15:50:38.000000000 +0400
@@ -40,8 +40,6 @@
 #include <config-win.h>
 #include <mysql.h>
 #include <time.h>
-#else
-#include <mysql/mysql.h>
 #endif
 
 #include "Database.h"
--- src/dbiwrapped-0.0.1/SysLog.cpp.orig	2010-09-07 01:55:57.000000000 +0400
+++ src/dbiwrapped-0.0.1/SysLog.cpp	2013-01-21 16:11:51.000000000 +0400
@@ -31,7 +31,6 @@
 */
 #ifndef WIN32
 
-#include <mysql/mysql.h>
 #include <syslog.h>
 #include <cstring>
 
