From 68274f94f687fce197e0b59125774e9523b8cc7f Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Fri, 27 Dec 2024 16:50:59 +0100
Subject: [PATCH 2/5] qv4l2: Use <fcntl.h> instead of <sys/fcntl.h>

Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 utils/qv4l2/raw2sliced.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utils/qv4l2/raw2sliced.cpp b/utils/qv4l2/raw2sliced.cpp
index 0c4e6b698358..f3c4f9a899fc 100644
--- a/utils/qv4l2/raw2sliced.cpp
+++ b/utils/qv4l2/raw2sliced.cpp
@@ -5,7 +5,7 @@
 #include <string.h>
 #include <assert.h>
 #include <unistd.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 
 #include "raw2sliced.h"
 
-- 
2.47.1

