--- solvespace-2.1.orig/CMakeLists.txt	2016-06-13 15:16:15.125813398 -0400
+++ solvespace-2.1/CMakeLists.txt	2016-06-13 15:19:32.800205236 -0400
@@ -17,7 +17,6 @@

 # project

-include(GetGitCommitHash)

 project(solvespace)
  set(solvespace_VERSION_MAJOR 2)
@@ -158,7 +158,8 @@
     find_package(Freetype REQUIRED)
     find_library(APPKIT_LIBRARY AppKit REQUIRED)
 else() # Linux and compatible systems
-    find_package(SpaceWare)
+    #find_package(SpaceWare)
+    set(SPACEWARE_FOUND FALSE)
 
     # Use freedesktop's pkg-config to locate everything.
     find_package(PkgConfig REQUIRED)
