#
# From http://w1.fi/bugz/show_bug.cgi?id=251
#
diff -Naur wpa_supplicant-0.5.8-orig/wpa_gui/networkconfig.ui.h wpa_supplicant-0.5.8/wpa_gui/networkconfig.ui.h
--- wpa_supplicant-0.5.8-orig/wpa_gui/networkconfig.ui.h	2006-12-09 18:38:48.000000000 -0600
+++ wpa_supplicant-0.5.8/wpa_gui/networkconfig.ui.h	2007-12-31 18:36:06.000000000 -0600
@@ -10,6 +10,7 @@
 ** destructor.
 *****************************************************************************/
 
+#include <stdlib.h>
 
 enum {
     AUTH_NONE = 0,
diff -Naur wpa_supplicant-0.5.8-orig/wpa_gui/userdatarequest.ui.h wpa_supplicant-0.5.8/wpa_gui/userdatarequest.ui.h
--- wpa_supplicant-0.5.8-orig/wpa_gui/userdatarequest.ui.h	2005-06-05 10:19:57.000000000 -0600
+++ wpa_supplicant-0.5.8/wpa_gui/userdatarequest.ui.h	2007-12-31 18:34:57.000000000 -0600
@@ -10,6 +10,8 @@
 ** destructor.
 *****************************************************************************/
 
+#include <stdlib.h>
+
 int UserDataRequest::setParams(WpaGui *_wpagui, const char *reqMsg)
 {
     char *tmp, *pos, *pos2;
diff -Naur wpa_supplicant-0.5.8-orig/wpa_gui/wpagui.ui.h wpa_supplicant-0.5.8/wpa_gui/wpagui.ui.h
--- wpa_supplicant-0.5.8-orig/wpa_gui/wpagui.ui.h	2007-03-24 20:09:57.000000000 -0600
+++ wpa_supplicant-0.5.8/wpa_gui/wpagui.ui.h	2007-12-31 18:34:23.000000000 -0600
@@ -16,6 +16,7 @@
 #include <unistd.h>
 #endif
 
+#include <stdlib.h>
 
 void WpaGui::init()
 {

