--- db-5.3.28/dist/config.sub.orig	2019-05-14 01:47:55.953219891 +0200
+++ db-5.3.28/dist/config.sub	2019-05-14 01:48:35.713528717 +0200
@@ -125,6 +125,7 @@
 maybe_os=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\2/'`
 case $maybe_os in
   nto-qnx* | linux-gnu* | linux-android* | linux-dietlibc | linux-newlib* | \
+  linux-musl* | \
   linux-uclibc* | uclinux-uclibc* | uclinux-gnu* | kfreebsd*-gnu* | \
   knetbsd*-gnu* | netbsd*-gnu* | \
   kopensolaris*-gnu* | \
