diff -ur module-init-tools-3.1-pre2/Makefile.in module-init-tools-3.1-pre2-copy/Makefile.in
--- module-init-tools-3.1-pre2/Makefile.in	2004-06-14 07:54:21.000000000 +0200
+++ module-init-tools-3.1-pre2-copy/Makefile.in	2004-06-19 19:16:37.614111176 +0200
@@ -783,15 +783,6 @@
 		ln -sf `ls -l /sbin/$$f | sed 's/.* -> //'`.old /sbin/$$f;  \
 	    fi;								    \
 	    mv /sbin/$$f /sbin/$$f.old;					    \
-	    if [ -f /usr/share/man/man8/$$f.8.gz ]; then		    \
-		mv /usr/share/man/man8/$$f.8.gz				    \
-			/usr/share/man/man8/$$f.old.8.gz;		    \
-	    elif [ -f /usr/share/man/man8/$$f.8.bz2 ]; then		    \
-	        mv /usr/share/man/man8/$$f.8.bz2			    \
-			/usr/share/man/man8/$$f.old.8.bz2;		    \
-	    else							    \
-	        mv /usr/share/man/man8/$$f.8 /usr/share/man/man8/$$f.old.8; \
-	    fi;								    \
 	done
 	for f in kallsyms ksyms; do					    \
 	    if [ -L /sbin/$$f ]; then					    \
