--- Makefile.in~	2007-10-08 00:03:25.000000000 +0200
+++ Makefile.in	2007-11-02 16:45:49.000000000 +0100
@@ -751,7 +751,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
@@ -917,15 +917,6 @@
 		ln -sf `ls -l /sbin/$$f | sed 's/.* -> //'`.old /sbin/$$f;  \
 	    fi;								    \
 	    mv /sbin/$$f /sbin/$$f.old;					    \
-	    if [ -f /usr/share/man/man8/$$f.8.gz ]; then		    \
-		mv /usr/share/man/man8/$$f.8.gz				    \
-			/usr/share/man/man8/$$f.old.8.gz;		    \
-	    elif [ -f /usr/share/man/man8/$$f.8.bz2 ]; then		    \
-	        mv /usr/share/man/man8/$$f.8.bz2			    \
-			/usr/share/man/man8/$$f.old.8.bz2;		    \
-	    else							    \
-	        mv /usr/share/man/man8/$$f.8 /usr/share/man/man8/$$f.old.8; \
-	    fi;								    \
 	done
 	for f in kallsyms ksyms; do					    \
 	    if [ -L /sbin/$$f ]; then					    \
