From e66d4dd0846333217a64415883e4e3f4755f53ba Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 21:00:13 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 atomic-primops.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/atomic-primops.cabal b/atomic-primops.cabal
index 0aa961a..03e5fe9 100644
--- a/atomic-primops.cabal
+++ b/atomic-primops.cabal
@@ -49,7 +49,7 @@ Library
 
   -- casMutVar# had a bug in GHC 7.2, thus we require GHC 7.4 or greater
   -- (base 4.5 or greater). We also need the "Any" kind.
-  build-depends:     base >= 4.6.0.0 && < 4.12, ghc-prim, primitive
+  build-depends:     base >= 4.6.0.0, ghc-prim, primitive
 
   -- TODO: Try to push support back to 7.0, but make it default to an implementation
   -- other than Unboxed.
-- 
2.19.1

