--- e2fsck/e2fsck.h	2005-04-14 18:06:09.000000000 +0000
+++ e2fsck/e2fsck.h.new	2005-08-03 16:01:56.000000000 +0000
@@ -6,6 +6,7 @@
  * 
  */
 
+#include <memory.h>
 #include <stdio.h>
 #include <string.h>
 #ifdef HAVE_UNISTD_H
--- lib/ext2fs/ext2fs.h 2005-06-27 15:47:15.000000000 +0000
+++ lib/ext2fs/ext2fs.h.new     2005-08-03 16:24:17.000000000 +0000
@@ -58,6 +58,7 @@
 #include <sys/types.h>
 #endif

+#include <memory.h>
 #include <stdio.h>
 #include <stdlib.h>


