--- octave-3.4.0/configure	2011-02-08 18:03:31.000000000 +0800
+++ configure	2011-05-20 14:50:40.024000681 +0800
@@ -46257,11 +46257,14 @@
   return 0;
 }
 _ACEOF
+ac_link_hold="$ac_link"
+ac_link='$F77 -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 if ac_fn_c_try_link "$LINENO"; then :
   eval "$as_ac_Lib=yes"
 else
   eval "$as_ac_Lib=no"
 fi
+ac_link="$ac_link_hold"
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
