--- source/term.c~	2003-04-11 03:09:07.000000000 +0200
+++ source/term.c	2006-01-20 17:14:15.000000000 +0100
@@ -92,7 +92,6 @@
 #endif
 
 extern  char    *getenv();
-extern	char	*tparm();
 
 /*
  * The old code assumed termcap. termcap is almost always present, but on
--- source/ctcp.c~	2003-05-27 09:00:22.000000000 +0200
+++ source/ctcp.c	2006-01-20 17:10:36.000000000 +0100
@@ -176,7 +176,7 @@
 
 /* CDE do ops and unban logging */
 
-static char	*ctcp_type[] =
+char	*ctcp_type[] =
 {
 	"PRIVMSG",
 	"NOTICE"
--- source/timer.c~	2003-04-11 03:09:07.000000000 +0200
+++ source/timer.c	2006-01-20 17:14:56.000000000 +0100
@@ -148,7 +148,7 @@
  * This is put here on purpose -- we dont want any of the above functions
  * to have any knowledge of this struct.
  */
-static TimerList *PendingTimers;
+TimerList *PendingTimers;
 static char *schedule_timer (TimerList *ntimer);
 
 static char *current_exec_timer = empty_string;
