Index: gst-libs/gst/audio/gstbaseaudiosink.c
===================================================================
RCS file: /cvs/gstreamer/gst-plugins-base/gst-libs/gst/audio/gstbaseaudiosink.c,v
retrieving revision 1.48
diff -u -r1.48 gstbaseaudiosink.c
--- gst-libs/gst/audio/gstbaseaudiosink.c	2 Feb 2006 18:18:31 -0000	1.48
+++ gst-libs/gst/audio/gstbaseaudiosink.c	12 Feb 2006 14:38:56 -0000
@@ -528,7 +528,7 @@
     GST_DEBUG_OBJECT (sink, "clipping start to %" GST_TIME_FORMAT " %"
         G_GUINT64_FORMAT " samples", GST_TIME_ARGS (ctime), diff);
     samples -= diff;
-    data += samples * bps;
+    data += diff * bps;
     time = ctime;
   }
   diff = stop - cstop;
