diff -Naur gst-plugins-base-0.10.30.orig/docs/libs/Makefile.in gst-plugins-base-0.10.30/docs/libs/Makefile.in
--- gst-plugins-base-0.10.30.orig/docs/libs/Makefile.in	2010-10-06 16:38:30.085000005 -0700
+++ gst-plugins-base-0.10.30/docs/libs/Makefile.in	2010-10-06 16:38:54.269999903 -0700
@@ -955,7 +955,7 @@
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
diff -Naur gst-plugins-base-0.10.30.orig/docs/plugins/Makefile.in gst-plugins-base-0.10.30/docs/plugins/Makefile.in
--- gst-plugins-base-0.10.30.orig/docs/plugins/Makefile.in	2010-10-06 16:38:30.081000005 -0700
+++ gst-plugins-base-0.10.30/docs/plugins/Makefile.in	2010-10-06 16:40:59.705999982 -0700
@@ -1101,7 +1101,7 @@
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
