--- GNUmakefile.old	2005-03-17 14:45:32.000000000 +0100
+++ GNUmakefile	2005-04-26 12:57:09.000000000 +0200
@@ -70,7 +70,8 @@
 $(builddir)/tfm/%.tfm:
 	@rm -f $(builddir)/output-dir
 	@-mkdir $(builddir)/tfm 2>/dev/null
-	-mktextfm $(notdir $(basename $@)) > $(builddir)/output-dir
+	@-mkdir $(builddir)/output 2> /dev/null
+	-mktextfm --destdir $(builddir)/output $(notdir $(basename $@)) > $(builddir)/output-dir
 	cp `cat $(builddir)/output-dir` $(builddir)/tfm
 
 $(builddir)/tfm.make: $(builddir)/tfm/ecbx10.tfm
