From fff2e19c6afb85edfd74284aafdd4365014e848a Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sun, 20 Mar 2016 12:23:42 +0100
Subject: [PATCH] use channel_layout.h instead of audioconvert.h

---
 decoder_plugins/ffmpeg/ffmpeg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/decoder_plugins/ffmpeg/ffmpeg.c b/decoder_plugins/ffmpeg/ffmpeg.c
index 206b2b1..66538db 100644
--- a/decoder_plugins/ffmpeg/ffmpeg.c
+++ b/decoder_plugins/ffmpeg/ffmpeg.c
@@ -67,7 +67,7 @@ GCC_DIAG_OFF(deprecated-declarations)
 GCC_DIAG_ON(deprecated-declarations)
 #include <libavutil/mathematics.h>
 #ifdef HAVE_AV_GET_CHANNEL_LAYOUT_NB_CHANNELS
-#include <libavutil/audioconvert.h>
+#include <libavutil/channel_layout.h>
 #endif
 #else
 #include <ffmpeg/avformat.h>
-- 
2.7.4

