diff -Naur moc-2.4.3.orig/configure moc-2.4.3/configure
--- moc-2.4.3.orig/configure	2008-03-12 20:01:40.000000000 -0600
+++ moc-2.4.3/configure	2008-03-12 20:42:11.000000000 -0600
@@ -29013,6 +29013,7 @@
 fi
 
 
+	libavformat_CFLAGS='-I/usr/include/ffmpeg'
 	if test "x$FFMPEG_CONFIG" = "xyes"
 	then
 		libavformat_CFLAGAS=`ffmpeg-config --cflags`
diff -Naur moc-2.4.3.orig/decoder_plugins/ffmpeg/ffmpeg.c moc-2.4.3/decoder_plugins/ffmpeg/ffmpeg.c
--- moc-2.4.3.orig/decoder_plugins/ffmpeg/ffmpeg.c	2008-03-12 20:01:40.000000000 -0600
+++ moc-2.4.3/decoder_plugins/ffmpeg/ffmpeg.c	2008-03-12 20:10:43.000000000 -0600
@@ -22,7 +22,7 @@
 #ifdef HAVE_INTTYPES_H
 # include <inttypes.h>
 #endif
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 
 /* FFmpeg also likes common names, without that, our common.h and log.h would
  * not be included. */
