Bug: https://bugzilla.gnome.org/show_bug.cgi?id=584126

Index: gnome-python-extras-2.25.3/gdl/gdl.defs
===================================================================
--- gnome-python-extras-2.25.3.orig/gdl/gdl.defs	2009-10-06 15:14:54.324941311 +0200
+++ gnome-python-extras-2.25.3/gdl/gdl.defs	2009-10-06 15:15:05.043832032 +0200
@@ -550,18 +550,6 @@
   (return-type "gboolean")
 )
 
-(define-method get_ui
-  (of-object "GdlDockLayout")
-  (c-name "gdl_dock_layout_get_ui")
-  (return-type "GtkWidget*")
-)
-
-(define-method get_items_ui
-  (of-object "GdlDockLayout")
-  (c-name "gdl_dock_layout_get_items_ui")
-  (return-type "GtkWidget*")
-)
-
 (define-method get_layouts_ui
   (of-object "GdlDockLayout")
   (c-name "gdl_dock_layout_get_layouts_ui")
