--- Makefile	2006-05-23 09:34:52.000000000 +0200
+++ Makefile.cp	2006-05-23 09:58:20.000000000 +0200
@@ -99,45 +99,6 @@
 	echo Installing cups-config script...
 	$(INSTALL_DIR) -m 755 $(BINDIR)
 	$(INSTALL_SCRIPT) cups-config $(BINDIR)/cups-config
-	if test "x$(INITDIR)" != x; then \
-		echo Installing init scripts...; \
-		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(INITDIR)/init.d; \
-		$(INSTALL_SCRIPT) init/cups.sh $(BUILDROOT)$(INITDIR)/init.d/cups; \
-		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(INITDIR)/rc0.d; \
-		$(INSTALL_SCRIPT) init/cups.sh  $(BUILDROOT)$(INITDIR)/rc0.d/K00cups; \
-		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(INITDIR)/rc2.d; \
-		$(INSTALL_SCRIPT) init/cups.sh $(BUILDROOT)$(INITDIR)/rc2.d/S99cups; \
-		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(INITDIR)/rc3.d; \
-		$(INSTALL_SCRIPT) init/cups.sh $(BUILDROOT)$(INITDIR)/rc3.d/S99cups; \
-		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(INITDIR)/rc5.d; \
-		$(INSTALL_SCRIPT) init/cups.sh $(BUILDROOT)$(INITDIR)/rc5.d/S99cups; \
-	fi
-	if test "x$(INITDIR)" = x -a "x$(INITDDIR)" != x; then \
-		$(INSTALL_DIR) $(BUILDROOT)$(INITDDIR); \
-		if test "$(INITDDIR)" = "/System/Library/StartupItems/PrintingServices"; then \
-			echo Installing StartupItems files...; \
-			$(INSTALL_SCRIPT) init/PrintingServices $(BUILDROOT)$(INITDDIR)/PrintingServices; \
-			$(INSTALL_DATA) init/StartupParameters.plist $(BUILDROOT)$(INITDDIR)/StartupParameters.plist; \
-			$(INSTALL_DIR) -m 755 $(BUILDROOT)$(INITDDIR)/Resources/English.lproj; \
-			$(INSTALL_DATA) init/Localizable.strings $(BUILDROOT)$(INITDDIR)/Resources/English.lproj/Localizable.strings; \
-		elif test "$(INITDDIR)" = "/System/Library/LaunchDaemons"; then \
-			echo Installing LaunchDaemons configuration file...; \
-			$(INSTALL_DATA) init/org.cups.cupsd.plist $(BUILDROOT)$(DEFAULT_LAUNCHD_CONF); \
-		else \
-			echo Installing RC script...; \
-			$(INSTALL_SCRIPT) init/cups.sh $(BUILDROOT)$(INITDDIR)/cups; \
-		fi \
-	fi
-	if test "x$(DBUSDIR)" != x; then \
-		echo Installing cups.conf in $(DBUSDIR)...;\
-		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(DBUSDIR); \
-		$(INSTALL_DATA) packaging/cups-dbus.conf $(BUILDROOT)$(DBUSDIR)/cups.conf; \
-	fi
-	if test "x$(XINETD)" != x; then \
-		echo Installing xinetd configuration file for cups-lpd...; \
-		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(XINETD); \
-		$(INSTALL_DATA) init/cups-lpd $(BUILDROOT)$(XINETD)/cups-lpd; \
-	fi
 	if test -d /usr/share/applications; then \
 		echo Installing desktop icons...; \
 		$(INSTALL_DIR) -m 755 $(BUILDROOT)/usr/share/applications; \
