diff -Naur cups-1.3.9.orig/backend/serial.c cups-1.3.9/backend/serial.c
--- cups-1.3.9.orig/backend/serial.c	2009-02-06 22:33:09.226962859 -0800
+++ cups-1.3.9/backend/serial.c	2009-02-06 22:33:35.356190020 -0800
@@ -74,6 +74,7 @@
 #endif /* __APPLE__ */
 
 #if defined(__linux) && defined(TIOCGSERIAL)
+#  include <linux/types.h>
 #  include <linux/serial.h>
 #  include <linux/ioctl.h>
 #endif /* __linux && TIOCGSERIAL */
