--- config-scripts/cups-defaults.m4	(revision 5801)
+++ config-scripts/cups-defaults.m4	(working copy)
@@ -25,7 +25,9 @@
 
 dnl Default langugages...
 AC_ARG_WITH(languages, [  --with-languages        set installed languages, default="de es ja pl sv" ],
-	LANGUAGES="$withval",
+	if test "x$withval" != xno; then
+		LANGUAGES="$withval"
+	fi,
 	LANGUAGES="de es ja pl sv")
 AC_SUBST(LANGUAGES)
 
--- configure.in	(revision 5801)
+++ configure.in	(working copy)
@@ -54,9 +54,17 @@
 	INSTALL_LANGUAGES="install-languages"
 	UNINSTALL_LANGUAGES="uninstall-languages"
 	for lang in $LANGUAGES; do
-		LANGFILES="$LANGFILES doc/$lang/index.html"
-		LANGFILES="$LANGFILES templates/$lang/edit-config.tmpl"
-		LANGFILES="$LANGFILES templates/$lang/header.tmpl"
+		if test -f doc/$lang/index.html.in; then
+			LANGFILES="$LANGFILES doc/$lang/index.html"
+		fi
+
+		if test -f templates/$lang/edit-config.tmpl.in; then
+			LANGFILES="$LANGFILES templates/$lang/edit-config.tmpl"
+		fi
+
+		if test -f templates/$lang/header.tmpl.in; then
+			LANGFILES="$LANGFILES templates/$lang/header.tmpl"
+		fi
 	done
 fi
 
