diff -uNr partimage-ng.orig/src/CMakeLists.txt partimage-ng/src/CMakeLists.txt
 # check C library; need macro def HAVE_STDARG_H for libntfs
diff -uNr partimage-ng.orig/src/ext2partition.h partimage-ng/src/ext2partition.h
--- partimage-ng.orig/src/ext2partition.h	2008-09-26 08:55:12.000000000 +0200
+++ partimage-ng/src/ext2partition.h	2008-10-26 09:50:55.000000000 +0100
@@ -19,6 +19,17 @@
 #define PARTIMAGENG_EXT2PARTITION
 
 #include <ext2fs/ext2fs.h>
+struct ext2fs_struct_generic_bitmap {
+        errcode_t       magic;
+        ext2_filsys     fs;
+        __u32           start, end;
+        __u32           real_end;
+        char    *       description;
+        char    *       bitmap;
+        errcode_t       base_error_code;
+        __u32           reserved[7];
+};
+
 #include <sys/types.h>
 
 #include "partition.h"
