diff -uNr partimage-ng/src/ntfspartition_c.c partimage-ng.cp/src/ntfspartition_c.c
--- partimage-ng/src/ntfspartition_c.c	2009-01-23 14:44:55.000000000 +0100
+++ partimage-ng.cp/src/ntfspartition_c.c	2009-11-21 17:58:03.000000000 +0100
@@ -22,6 +22,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <limits.h>
 #include <errno.h>
 #include <string.h>
 #ifdef PING_USE_NTFS3G
@@ -320,13 +321,13 @@
         rl[1].lcn = -1;
         rl[1].length = 0;
 
-        old_mp_size = ntfs_get_size_for_mapping_pairs((ntfs_volume *)ntfs_vol, rl_bad, 0);
+        old_mp_size = ntfs_get_size_for_mapping_pairs((ntfs_volume *)ntfs_vol, rl_bad, 0, INT_MAX);
         if ( old_mp_size <= 0 ) {
                 fprintf(debugs(), "Cannot calculate the size of the old mapping pairs\n");
                 return 1;
         }
 
-        new_mp_size = ntfs_get_size_for_mapping_pairs((ntfs_volume *)ntfs_vol, rl, 0);
+        new_mp_size = ntfs_get_size_for_mapping_pairs((ntfs_volume *)ntfs_vol, rl, 0, INT_MAX);
         if ( new_mp_size <= 0 ) {
                 fprintf(debugs(), "Cannot calculate the size of the new mapping pairs\n");
                 return 1;
