--- SConstruct.org	2005-04-12 15:16:02.823848040 +0200
+++ SConstruct	2005-04-12 15:16:18.672438688 +0200
@@ -370,7 +370,7 @@
     pluginEnvironment['SHLIBSUFFIX'] = '.dylib'
 
 csoundProgramEnvironment = commonEnvironment.Copy()
-csoundProgramEnvironment.Append(LIBS = ['csound', 'sndfile'])
+csoundProgramEnvironment.Append(LIBS = ['csound', 'sndfile', 'dl', 'pthread'])
 csoundProgramEnvironment.ParseConfig('fltk-config --cflags --cxxflags --ldflags')
 
 vstEnvironment = commonEnvironment.Copy()
