diff -urN ardour-2.8.16/libs/ardour/ardour/lv2_plugin.h ardour-2.8.16-patched/libs/ardour/ardour/lv2_plugin.h
--- ardour-2.8.16/libs/ardour/ardour/lv2_plugin.h	2012-04-20 05:51:59.000000000 +1100
+++ ardour-2.8.16-patched/libs/ardour/ardour/lv2_plugin.h	2013-04-18 00:10:31.519543043 +1100
@@ -154,7 +154,7 @@
 	/** Find the LV2 input port with the given designation.
 	 * If found, bufptrs[port_index] will be set to bufptr.
 	 */
-	LilvPort* designated_input (const char* uri, void** bufptrs[], void** bufptr);
+	const LilvPort* designated_input (const char* uri, void** bufptrs[], void** bufptr);
 };
 
 
diff -urN ardour-2.8.16/libs/ardour/lv2_plugin.cc ardour-2.8.16-patched/libs/ardour/lv2_plugin.cc
--- ardour-2.8.16/libs/ardour/lv2_plugin.cc	2012-04-20 05:51:59.000000000 +1100
+++ ardour-2.8.16-patched/libs/ardour/lv2_plugin.cc	2013-04-18 00:07:18.358710010 +1100
@@ -644,10 +644,10 @@
 	deactivate ();
 }
 
-LilvPort*
+const LilvPort*
 LV2Plugin::designated_input (const char* uri, void** bufptrs[], void** bufptr)
 {
-	LilvPort* port = NULL;
+	const LilvPort* port = NULL;
 #ifdef HAVE_NEW_LILV
 	LilvNode* designation = lilv_new_uri(_world.world, uri);
 	port = lilv_plugin_get_port_by_designation(
