--- ardour-2.4.1/libs/sigc++2/sigc++/signal.h	2008-04-09 23:17:17.000000000 +0200
+++ ardour-2.4.1/libs/sigc++2/sigc++/signal.h-new	2008-05-17 04:38:03.000000000 +0200
@@ -17,6 +17,12 @@
   //Compilers, such as SUN Forte C++, that do not allow this also often
   //do not allow a typedef to have the same name as a class in the typedef's definition.
   #define SIGC_TYPEDEF_REDEFINE_ALLOWED 1
+
+  #ifdef __GNUC__
+    #if (__GNUC__ > 4) || (__GNUC__ == 4 && __GNUC_MINOR__ >= 3)
+      #undef SIGC_TYPEDEF_REDEFINE_ALLOWED
+    #endif
+  #endif
 #endif
 
 namespace sigc {
--- ardour-2.4.1/libs/pbd/base_ui.cc	2008-05-17 04:32:22.000000000 +0200
+++ ardour-2.4.1/libs/pbd/base_ui.cc-new	2008-05-17 04:32:38.000000000 +0200
@@ -21,6 +21,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <cstring>
 
 #include <pbd/base_ui.h>
 #include <pbd/error.h>
--- ardour-2.4.1/libs/pbd/mountpoint.cc	2008-05-17 04:43:57.000000000 +0200
+++ ardour-2.4.1/libs/pbd/mountpoint.cc-new	2008-05-17 04:43:28.000000000 +0200
@@ -21,6 +21,7 @@
 #include <cstdio>
 #include <string>
 #include <limits.h>
+#include <cstring>
 
 #include <pbd/mountpoint.h>
 
--- ardour-2.4.1/libs/pbd/pathscanner.cc	2008-04-09 23:17:27.000000000 +0200
+++ ardour-2.4.1/libs/pbd/pathscanner.cc-new	2008-05-17 04:45:32.000000000 +0200
@@ -22,6 +22,7 @@
 #include <cstdio>
 #include <vector>
 #include <dirent.h>
+#include <cstring>
 
 #include <pbd/error.h>
 #include <pbd/pathscanner.h>
--- ardour-2.4.1/libs/pbd/pool.cc	2008-04-09 23:17:27.000000000 +0200
+++ ardour-2.4.1/libs/pbd/pool.cc-new	2008-05-17 04:47:44.000000000 +0200
@@ -20,6 +20,8 @@
 
 #include <iostream>
 #include <vector>
+#include <cstring>
+#include <cstdlib>
 
 #include <pbd/pool.h>
 #include <pbd/error.h>
--- ardour-2.4.1/libs/glibmm2/glibmm/propertyproxy_base.cc	2008-04-09 23:17:08.000000000 +0200
+++ ardour-2.4.1/libs/glibmm2/glibmm/propertyproxy_base.cc-new	2008-05-17 05:00:28.000000000 +0200
@@ -24,6 +24,7 @@
 #include <glibmm/signalproxy_connectionnode.h>
 #include <glibmm/object.h>
 #include <glibmm/private/object_p.h>
+#include <cstring>
 
 namespace Glib
 {
--- ardour-2.4.1/libs/midi++2/fd_midiport.cc	2008-04-09 23:17:09.000000000 +0200
+++ ardour-2.4.1/libs/midi++2/fd_midiport.cc-new	2008-05-17 05:03:51.000000000 +0200
@@ -20,6 +20,7 @@
 
 #include <fcntl.h>
 #include <cerrno>
+#include <cstring>
 
 #include <pbd/error.h>
 #include <pbd/pathscanner.h>
--- ardour-2.4.1/libs/midi++2/midi.cc	2008-05-17 05:06:26.000000000 +0200
+++ ardour-2.4.1/libs/midi++2/midi.cc-new	2008-05-17 05:06:32.000000000 +0200
@@ -21,6 +21,7 @@
 #include <string>
 #include <cstdlib>
 #include <midi++/types.h>
+#include <cstring>
 
 const char *MIDI::controller_names[] = {
 	"bank (0)",
--- ardour-2.4.1/libs/midi++2/midichannel.cc	2008-04-09 23:17:09.000000000 +0200
+++ ardour-2.4.1/libs/midi++2/midichannel.cc-new	2008-05-17 05:08:05.000000000 +0200
@@ -21,6 +21,7 @@
 #include <midi++/types.h>
 #include <midi++/port.h>
 #include <midi++/channel.h>
+#include <cstring>
 
 using namespace sigc;
 using namespace MIDI;
--- ardour-2.4.1/libs/midi++2/midiport.cc	2008-04-09 23:17:09.000000000 +0200
+++ ardour-2.4.1/libs/midi++2/midiport.cc-new	2008-05-17 05:09:07.000000000 +0200
@@ -20,6 +20,7 @@
 #include <iostream>
 #include <cstdio>
 #include <fcntl.h>
+#include <cstring>
 
 #include <pbd/xml++.h>
 #include <pbd/failed_constructor.h>
--- ardour-2.4.1/libs/midi++2/midiparser.cc	2008-04-09 23:17:09.000000000 +0200
+++ ardour-2.4.1/libs/midi++2/midiparser.cc-new	2008-05-17 05:10:24.000000000 +0200
@@ -26,6 +26,7 @@
 #include <string>
 #include <iostream>
 #include <iterator>
+#include <cstring>
 
 #include <midi++/types.h>
 #include <midi++/parser.h>
--- ardour-2.4.1/libs/midi++2/mtc.cc	2008-04-09 23:17:09.000000000 +0200
+++ ardour-2.4.1/libs/midi++2/mtc.cc-new	2008-05-17 05:11:28.000000000 +0200
@@ -22,6 +22,7 @@
 #include <unistd.h>
 #include <string>
 #include <iostream>
+#include <cstring>
 
 #include <midi++/types.h>
 #include <midi++/parser.h>
--- ardour-2.4.1/libs/ardour/audioanalyser.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/audioanalyser.cc-new	2008-05-17 05:12:45.000000000 +0200
@@ -2,6 +2,7 @@
 #include <glibmm/miscutils.h>
 #include <glibmm/fileutils.h>
 #include <glib/gstdio.h> // for g_remove()
+#include <cstring>
 
 #include <pbd/error.h>
 
--- ardour-2.4.1/libs/ardour/audio_diskstream.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/audio_diskstream.cc-new	2008-05-17 05:13:15.000000000 +0200
@@ -29,6 +29,7 @@
 #include <ctime>
 #include <sys/stat.h>
 #include <sys/mman.h>
+#include <cstring>
 
 #include <pbd/error.h>
 #include <pbd/basename.h>
--- ardour-2.4.1/libs/ardour/audioengine.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/audioengine.cc-new	2008-05-17 05:16:15.000000000 +0200
@@ -21,6 +21,7 @@
 #include <cerrno>
 #include <vector>
 #include <sstream>
+#include <cstring>
 
 #include <glibmm/timer.h>
 #include <pbd/pthread_utils.h>
--- ardour-2.4.1/libs/ardour/ardour/buffer.h	2008-04-09 23:17:13.000000000 +0200
+++ ardour-2.4.1/libs/ardour/ardour/buffer.h-new	2008-05-17 05:15:12.000000000 +0200
@@ -24,6 +24,7 @@
 #include <cassert>
 #include <ardour/types.h>
 #include <ardour/data_type.h>
+#include <cstring>
 
 namespace ARDOUR {
 
--- ardour-2.4.1/libs/ardour/ardour/port.h	2008-04-09 23:17:12.000000000 +0200
+++ ardour-2.4.1/libs/ardour/ardour/port.h-new	2008-05-17 05:15:49.000000000 +0200
@@ -24,6 +24,7 @@
 #include <pbd/failed_constructor.h>
 #include <ardour/ardour.h>
 #include <jack/jack.h>
+#include <cstring>
 
 namespace ARDOUR {
 
--- ardour-2.4.1/libs/ardour/audiofilesource.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/audiofilesource.cc-new	2008-05-17 05:18:07.000000000 +0200
@@ -18,6 +18,7 @@
 */
 
 #include <vector>
+#include <cstring>
 
 #include <sys/time.h>
 #include <sys/stat.h>
--- ardour-2.4.1/libs/ardour/audiofilter.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/audiofilter.cc-new	2008-05-17 05:18:25.000000000 +0200
@@ -19,6 +19,7 @@
 
 #include <time.h>
 #include <cerrno>
+#include <cstring>
 
 #include <pbd/basename.h>
 #include <ardour/sndfilesource.h>
--- ardour-2.4.1/libs/ardour/audio_playlist.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/audio_playlist.cc-new	2008-05-17 05:20:08.000000000 +0200
@@ -20,6 +20,7 @@
 #include <algorithm>
 
 #include <cstdlib>
+#include <cstring>
 
 #include <sigc++/bind.h>
 
--- ardour-2.4.1/libs/ardour/audioregion.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/audioregion.cc-new	2008-05-17 05:20:25.000000000 +0200
@@ -24,6 +24,8 @@
 
 #include <set>
 
+#include <cstring>
+
 #include <sigc++/bind.h>
 #include <sigc++/class_slot.h>
 
--- ardour-2.4.1/libs/ardour/ardour/route.h	2008-04-09 23:17:12.000000000 +0200
+++ ardour-2.4.1/libs/ardour/ardour/route.h-new	2008-05-17 05:22:02.000000000 +0200
@@ -25,6 +25,7 @@
 #include <set>
 #include <map>
 #include <string>
+#include <cstring>
 
 #include <boost/shared_ptr.hpp>
 
--- ardour-2.4.1/libs/ardour/audiosource.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/audiosource.cc-new	2008-05-17 05:22:13.000000000 +0200
@@ -30,6 +30,7 @@
 #include <fstream>
 #include <algorithm>
 #include <vector>
+#include <cstring>
 
 #include <glibmm/fileutils.h>
 #include <glibmm/miscutils.h>
--- ardour-2.4.1/libs/ardour/crossfade.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/crossfade.cc-new	2008-05-17 05:27:25.000000000 +0200
@@ -28,6 +28,7 @@
 #include <ardour/playlist.h>
 #include <ardour/utils.h>
 #include <ardour/session.h>
+#include <cstring>
 
 #include "i18n.h"
 #include <locale.h>
--- ardour-2.4.1/libs/ardour/panner.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/panner.cc-new	2008-05-17 05:29:52.000000000 +0200
@@ -26,6 +26,7 @@
 #include <locale.h>
 #include <unistd.h>
 #include <float.h>
+#include <cstring>
 
 #include <glibmm.h>
 
--- ardour-2.4.1/libs/ardour/recent_sessions.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/recent_sessions.cc-new	2008-05-17 05:31:23.000000000 +0200
@@ -26,6 +26,7 @@
 #include <ardour/recent_sessions.h>
 #include <ardour/utils.h>
 #include "i18n.h"
+#include <cstring>
 
 using namespace std;
 using namespace ARDOUR;
--- ardour-2.4.1/libs/ardour/redirect.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/redirect.cc-new	2008-05-17 05:31:33.000000000 +0200
@@ -23,6 +23,7 @@
 #include <cerrno>
 #include <unistd.h>
 #include <sstream>
+#include <cstring>
 
 #include <sigc++/bind.h>
 
--- ardour-2.4.1/libs/ardour/session_click.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/session_click.cc-new	2008-05-17 05:33:53.000000000 +0200
@@ -19,6 +19,7 @@
 
 #include <list>
 #include <cerrno>
+#include <cstring>
 
 #include <ardour/ardour.h>
 #include <ardour/session.h>
--- ardour-2.4.1/libs/ardour/ardour/silentfilesource.h	2008-04-09 23:17:12.000000000 +0200
+++ ardour-2.4.1/libs/ardour/ardour/silentfilesource.h-new	2008-05-17 05:36:42.000000000 +0200
@@ -21,6 +21,7 @@
 #define __ardour_silentfilesource_h__
 
 #include <ardour/audiofilesource.h>
+#include <cstring>
 
 namespace ARDOUR {
 
--- ardour-2.4.1/libs/ardour/sndfilesource.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/sndfilesource.cc-new	2008-05-17 05:37:56.000000000 +0200
@@ -19,6 +19,7 @@
 
 #include <cerrno>
 #include <climits>
+#include <cstring>
 
 #include <pwd.h>
 #include <sys/utsname.h>
--- ardour-2.4.1/libs/ardour/utils.cc	2008-04-09 23:17:16.000000000 +0200
+++ ardour-2.4.1/libs/ardour/utils.cc-new	2008-05-17 05:39:39.000000000 +0200
@@ -21,6 +21,7 @@
 #include <cmath>
 #include <cctype>
 #include <string>
+#include <cstring>
 #include <cerrno>
 #include <iostream>
 #include <sys/types.h>
--- ardour-2.4.1/libs/vamp-sdk/vamp-sdk/Plugin.h	2008-04-09 23:17:05.000000000 +0200
+++ ardour-2.4.1/libs/vamp-sdk/vamp-sdk/Plugin.h-new	2008-05-17 05:42:02.000000000 +0200
@@ -43,6 +43,8 @@
 #include <string>
 #include <vector>
 #include <map>
+#include <cstring>
+#include <cstdlib>
 
 namespace Vamp {
 
--- ardour-2.4.1/libs/rubberband/src/Window.h	2008-04-09 23:17:28.000000000 +0200
+++ ardour-2.4.1/libs/rubberband/src/Window.h-new	2008-05-17 05:47:46.000000000 +0200
@@ -18,6 +18,7 @@
 #include <cmath>
 #include <iostream>
 #include <map>
+#include <cstdlib>
 
 namespace RubberBand {
 
--- ardour-2.4.1/libs/rubberband/src/RingBuffer.h	2008-04-09 23:17:28.000000000 +0200
+++ ardour-2.4.1/libs/rubberband/src/RingBuffer.h-new	2008-05-17 05:49:21.000000000 +0200
@@ -37,6 +37,8 @@
 #include <iostream>
 #endif
 
+#include <cstring>
+
 namespace RubberBand {
 
 /**
--- ardour-2.4.1/libs/rubberband/src/Thread.cpp	2008-04-09 23:17:28.000000000 +0200
+++ ardour-2.4.1/libs/rubberband/src/Thread.cpp-new	2008-05-17 05:50:37.000000000 +0200
@@ -15,6 +15,7 @@
 #include "Thread.h"
 
 #include <iostream>
+#include <cstdlib>
 
 #include <sys/time.h>
 #include <time.h>
--- ardour-2.4.1/libs/rubberband/src/main.cpp	2008-04-09 23:17:28.000000000 +0200
+++ ardour-2.4.1/libs/rubberband/src/main.cpp-new	2008-05-17 05:51:47.000000000 +0200
@@ -17,6 +17,8 @@
 #include <iostream>
 #include <sndfile.h>
 #include <cmath>
+#include <cstdlib>
+#include <cstring>
 #include <sys/time.h>
 #include <time.h>
 #include "sysutils.h"
--- ardour-2.4.1/libs/rubberband/src/StretchCalculator.cpp	2008-04-09 23:17:28.000000000 +0200
+++ ardour-2.4.1/libs/rubberband/src/StretchCalculator.cpp-new	2008-05-17 05:55:09.000000000 +0200
@@ -19,6 +19,7 @@
 #include <deque>
 #include <set>
 #include <cassert>
+#include <algorithm>
 
 namespace RubberBand
 {
--- ardour-2.4.1/libs/rubberband/src/FFT.cpp	2008-04-09 23:17:28.000000000 +0200
+++ ardour-2.4.1/libs/rubberband/src/FFT.cpp-new	2008-05-17 05:56:14.000000000 +0200
@@ -23,6 +23,7 @@
 #include <map>
 #include <cstdio>
 #include <vector>
+#include <cstdlib>
 
 namespace RubberBand {
 
--- ardour-2.4.1/gtk2_ardour/actions.cc	2008-04-09 23:17:02.000000000 +0200
+++ ardour-2.4.1/gtk2_ardour/actions.cc-new	2008-05-17 06:00:15.000000000 +0200
@@ -20,6 +20,7 @@
 #include <vector>
 #include <string>
 #include <list>
+#include <cstring>
 
 #include <gtk/gtkaccelmap.h>
 #include <gtk/gtkuimanager.h>
--- ardour-2.4.1/gtk2_ardour/ardour_ui_options.cc	2008-04-09 23:17:02.000000000 +0200
+++ ardour-2.4.1/gtk2_ardour/ardour_ui_options.cc-new	2008-05-17 06:03:44.000000000 +0200
@@ -19,6 +19,7 @@
 
 #include <pbd/convert.h>
 #include <pbd/stacktrace.h>
+#include <cstring>
 
 #include <gtkmm2ext/utils.h>
 
--- ardour-2.4.1/gtk2_ardour/editing.h	2008-04-09 23:17:02.000000000 +0200
+++ ardour-2.4.1/gtk2_ardour/editing.h-new	2008-05-17 06:10:46.000000000 +0200
@@ -21,6 +21,7 @@
 #define __gtk_ardour_editing_h__
 
 #include <string>
+#include <cstring>
 #include <map>
 #include <ardour/types.h>
 
--- ardour-2.4.1/gtk2_ardour/opts.cc	2008-04-09 23:17:02.000000000 +0200
+++ ardour-2.4.1/gtk2_ardour/opts.cc-new	2008-05-17 11:19:00.000000000 +0200
@@ -20,6 +20,7 @@
 #include <getopt.h>
 #include <iostream>
 #include <cstdlib>
+#include <cstring>
 
 #include <ardour/session.h>
 
--- ardour-2.4.1/gtk2_ardour/fft_result.cc	2008-04-09 23:17:02.000000000 +0200
+++ ardour-2.4.1/gtk2_ardour/fft_result.cc-new	2008-05-17 11:24:50.000000000 +0200
@@ -23,6 +23,7 @@
 #include <cstdlib>
 #include <string>
 #include <cmath>
+#include <cstring>
 
 #include <iostream>
 
--- ardour-2.4.1/libs/gtkmm2/gtk/gtkmm/scale.cc	2008-05-17 11:36:04.000000000 +0200
+++ ardour-2.4.1/libs/gtkmm2/gtk/gtkmm/scale.cc-new	2008-05-17 11:36:07.000000000 +0200
@@ -32,6 +32,8 @@
 #include <gtk/gtkvscale.h>
 #include <math.h>
 
+#include <cstring>
+
 namespace Gtk
 {
 
--- ardour-2.4.1/libs/gtkmm2/gtk/gtkmm/targetentry.cc	2008-04-09 23:17:25.000000000 +0200
+++ ardour-2.4.1/libs/gtkmm2/gtk/gtkmm/targetentry.cc-new	2008-05-17 11:46:19.000000000 +0200
@@ -21,6 +21,7 @@
  */
 
 #include <gtkmm/targetentry.h>
+#include <cstring>
 
 
 namespace Gtk
--- ardour-2.4.1/libs/gtkmm2ext/gtk_ui.cc	2008-04-09 23:17:05.000000000 +0200
+++ ardour-2.4.1/libs/gtkmm2ext/gtk_ui.cc-new	2008-05-17 11:50:35.000000000 +0200
@@ -25,6 +25,7 @@
 #include <cerrno>
 #include <climits>
 #include <cctype>
+#include <cstring>
 
 #include <gtkmm.h>
 #include <pbd/error.h>
--- ardour-2.4.1/libs/surfaces/tranzport/screen.cc	2008-04-09 23:17:32.000000000 +0200
+++ ardour-2.4.1/libs/surfaces/tranzport/screen.cc-new	2008-05-17 11:56:10.000000000 +0200
@@ -19,6 +19,7 @@
  *   */
 
 #include <tranzport_control_protocol.h>
+#include <cstring>
 
 void
 TranzportControlProtocol::screen_clear ()
--- ardour-2.4.1/libs/surfaces/mackie/mackie_control_protocol_poll.cc	2008-04-09 23:17:34.000000000 +0200
+++ ardour-2.4.1/libs/surfaces/mackie/mackie_control_protocol_poll.cc-new	2008-05-17 12:03:43.000000000 +0200
@@ -19,6 +19,7 @@
 #include <iostream>
 #include <string>
 #include <vector>
+#include <cstring>
 
 using namespace std;
 using namespace Mackie;
