--- Makefile.ori	2009-03-11 12:59:26.000000000 +0100
+++ Makefile	2009-03-11 13:00:02.000000000 +0100
@@ -16,13 +16,20 @@
 MANDIR     = $(ROOT)$(PREFIX)/man
 endif
 
-ELOGDIR    = $(ROOT)$(PREFIX)/elog
+ELOGDIR    = $(ROOT)$(PREFIX)/share/elog
 DESTDIR    = $(ROOT)$(PREFIX)/bin
 SDESTDIR   = $(ROOT)$(PREFIX)/sbin
 RCDIR      = $(ROOT)/etc/rc.d/init.d
 
 # flag for SSL support
+ifdef USE_SSL
 USE_SSL    = 1
+endif
+
+# support for crypt()
+ifdef USE_CRYPT
+USE_CRYPT  = 1
+endif
 
 #############################################################
 
@@ -76,6 +83,11 @@
 LIBS += -lssl
 endif 
 
+ifdef USE_CRYPT
+CFLAGS += -DHAVE_CRYPT
+LIBS += -lcrypt
+endif 
+
 WHOAMI = $(shell whoami)
 ifeq ($(WHOAMI),root)
 BINFLAGS = -o ${BINOWNER} -g ${BINGROUP}
