From f3f6fecea7b93934d107cea0710e95c7afc55a27 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Fri, 25 Mar 2016 10:13:17 +0100
Subject: [PATCH] fix compatibility with lua >= 5.2

---
 term/lua/gnuplot-tikz.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/term/lua/gnuplot-tikz.lua b/term/lua/gnuplot-tikz.lua
index a2cb684..967345c 100644
--- a/term/lua/gnuplot-tikz.lua
+++ b/term/lua/gnuplot-tikz.lua
@@ -1754,7 +1754,7 @@ term.options = function(opt_str, initial, t_count)
         o_next = string.sub(opt_str, s_start+1, s_end-1)
         if next_char == '"' then
           -- Wow! this is to resolve all string escapes, kind of "unescape string"
-          o_next = assert(loadstring("return(\""..o_next.."\")"))()
+          o_next = assert(load("return(\""..o_next.."\")"))()
         end
         o_type = "string"
       else
-- 
2.7.4

