From 240fdd5e802faee23ad1c716f18bf27c3e4f554c Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 31 Dec 2018 10:46:21 +0100
Subject: [PATCH] fix Makefiles

---
 apps/Makefile   | 5 +++--
 source/Makefile | 6 ++----
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/apps/Makefile b/apps/Makefile
index 9d48cb2..ca26511 100644
--- a/apps/Makefile
+++ b/apps/Makefile
@@ -24,8 +24,9 @@ PREFIX ?= /usr/local
 BINDIR ?= $(PREFIX)/bin
 
 VERSION = "0.3.2"
-CPPFLAGS +=  -MMD -MP -DVERSION=\"$(VERSION)\"
-CXXFLAGS += -O2 -Wall
+CPPFLAGS +=  -MMD -MP -DVERSION=\"$(VERSION)\" -I../source
+CXXFLAGS += -Wall
+LDFLAGS += -L../source
 
 
 all:	alsa_loopback alsa_delay
diff --git a/source/Makefile b/source/Makefile
index a99c5a3..0e315de 100644
--- a/source/Makefile
+++ b/source/Makefile
@@ -20,18 +20,16 @@
 
 # Modify as required.
 #
-SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 PREFIX ?= /usr/local
 INCDIR ?= $(PREFIX)/include
-LIBDIR ?= $(PREFIX)/lib$(SUFFIX)
+LIBDIR ?= $(PREFIX)/lib
 
 MAJVERS = 0
 MINVERS = 3.2
 VERSION = $(MAJVERS).$(MINVERS)
 
 CPPFLAGS += -DVERSION=\"$(VERSION)\" -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
-CXXFLAGS += -O2 -Wall -fPIC
-CXXFLAGS += -march=native
+CXXFLAGS += -Wall -fPIC -DPIC
 LDLFAGS +=
 LDLIBS +=
 
-- 
2.20.1

