diff -uNr ODBC_V2-0-BETA/OdbcJdbc/Builds/Gcc.lin/makefile.linux ODBC_V2-0-BETA.cp/OdbcJdbc/Builds/Gcc.lin/makefile.linux
--- ODBC_V2-0-BETA/OdbcJdbc/Builds/Gcc.lin/makefile.linux	2006-07-03 15:10:00.000000000 +0200
+++ ODBC_V2-0-BETA.cp/OdbcJdbc/Builds/Gcc.lin/makefile.linux	2006-07-03 22:10:28.000000000 +0200
@@ -55,8 +55,8 @@
 INCLUDEDIR      := -I/opt/interbase/include -I/usr/include/odbc $(INCLUDEDIR)
 EXTLIBDIR       := -L/opt/interbase/lib -L/usr/lib/unixODBC $(EXTLIBDIR)
 else
-INCLUDEDIR      = -I/opt/firebird/include -I/usr/include/odbc
-EXTLIBDIR       := -L/opt/firebird/lib -L/usr/lib/unixODBC $(EXTLIBDIR)
+INCLUDEDIR      = -I/usr/firebird/include -I/usr/include/odbc
+EXTLIBDIR       := -L/usr/firebird/lib -L/usr/lib/unixODBC $(EXTLIBDIR)
 endif
 #
 
@@ -75,8 +75,9 @@
 LIST_ODBCJDBCSETUPSRC	= $(addprefix $(ODBCJDBCDIR)/, $(ODBCJDBCSETUPSRC_LINUX))
 LIST_ODBCJDBCSETUPOBJ	= $(addprefix $(BUILDDIR)/, $(ODBCJDBCSETUPSRC_LINUX:.cpp=.o))
 #
-COMPFLAGS	= -g -w -D_REENTRANT -D_PTHREADS -DEXTERNAL -D$(ODBCMANAGER) $(INCLUDEDIR) -I$(FBINCDIR)
+COMPFLAGS	= -DPIC -fPIC $(CFLAGS) -w -D_REENTRANT -D_PTHREADS -DEXTERNAL -D$(ODBCMANAGER) $(INCLUDEDIR) -I$(FBINCDIR)
 #
+
 #LINKFLAGS	= -rdynamic -export-dynamic -shared 
 LINKFLAGS	= -shared
 EXTLIBS		= $(EXTLIBDIR) -lcrypt -ldl
diff -uNr ODBC_V2-0-BETA/OdbcJdbc/Builds/makefile.environ ODBC_V2-0-BETA.cp/OdbcJdbc/Builds/makefile.environ
--- ODBC_V2-0-BETA/OdbcJdbc/Builds/makefile.environ	2003-12-13 19:27:26.000000000 +0100
+++ ODBC_V2-0-BETA.cp/OdbcJdbc/Builds/makefile.environ	2006-07-03 21:56:27.000000000 +0200
@@ -3,8 +3,8 @@
 #FBINCDIR	= /mnt/third_party/interbase/sun5-2003-3-10/include
 #FBLIBDIR	= /mnt/third_party/interbase/sun5-2003-3-10/lib
 #
-FBINCDIR	= d:/Firebird/include
-FBLIBDIR	= d:/Firebird/lib
+#FBINCDIR	= /usr/firebird/include
+#FBLIBDIR	= /usr/firebird/lib
 #
 ISCDBCDIR 		= ../../IscDbc
 ODBCJDBCDIR 		= ../..
