--- dvdread/ifo_types.h~	2005-09-18 20:02:08.000000000 +0200
+++ dvdread/ifo_types.h	2006-01-23 16:09:36.000000000 +0100
@@ -21,18 +21,9 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#include <inttypes.h>
 #include <dvdread/dvd_reader.h>
 
-#if defined(__BEOS__)
-#if !defined(_INTTYPES_H_) && !defined(_INTTYPES_H) && !defined(_STDINT_H_) && !defined(_STDINT_H)
-#error "Must include <inttypes.h> or <stdint.h> before any libdvdread header."
-#endif
-#else
-#if !defined(UINT8_MAX) || !defined(UINT16_MAX) || !defined(INT32_MAX)
-#error "Must include <inttypes.h> or <stdint.h> before any libdvdread header."
-#endif
-#endif
-
 #undef ATTRIBUTE_PACKED
 #undef PRAGMA_PACK_BEGIN 
 #undef PRAGMA_PACK_END
