From 18f1f90be290394f0755caed738cad9cfa811b11 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Tue, 19 Sep 2017 22:55:58 +0200
Subject: [PATCH] allow base <= 4.10

---
 stm.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stm.cabal b/stm.cabal
index 9378d29..41fa2c1 100644
--- a/stm.cabal
+++ b/stm.cabal
@@ -34,7 +34,7 @@ library
         other-extensions: Safe
 
     build-depends:
-        base  >= 4.3 && < 4.10,
+        base  >= 4.3 && < 4.11,
         array >= 0.3 && < 0.6
 
     exposed-modules:
-- 
2.14.1

