--- rlfe.c.orig	2004-11-05 01:32:14.000000000 +0300
+++ rlfe.c	2015-07-21 11:14:12.857710397 +0300
@@ -588,7 +588,7 @@
 	}
       if (FD_ISSET (in_from_tty_fd, &in_set))
 	{
-	  extern int readline_echoing_p;
+	  extern int _rl_echoing_p;
 	  struct termios term_master;
 	  int do_canon = 1;
 	  int do_icrnl = 1;
@@ -605,7 +605,7 @@
 	    {
 	      do_canon = (term_master.c_lflag & ICANON) != 0;
 	      do_icrnl = (term_master.c_lflag & ICRNL) != 0;
-	      readline_echoing_p = (term_master.c_lflag & ECHO) != 0;
+	      _rl_echoing_p = (term_master.c_lflag & ECHO) != 0;
 	      DPRINT1 ("echo,canon,crnl:%03d\n",
 		       100 * readline_echoing_p
 		       + 10 * do_canon
