--- src/castfs.c.orig	2010-03-30 09:07:57.000000000 +0400
+++ src/castfs.c	2012-08-05 16:05:34.667167115 +0400
@@ -724,6 +724,13 @@
 	char *new_argv[argc], *logptr, *dbgptr, deflog[]="/tmp/castfs.log";
 	int new_argc = 0;
 
+	/* Currently, need: `castfs mnt-dir -o stage=stage-dir` */
+	if (argc < 4) {
+fprintf(stderr, "castfs: argc wrong\n");
+		usage();
+		exit(-1);
+	}
+
 	/* get environmental vars and open fd's for log file */
 	umask(0);
 
@@ -760,13 +767,6 @@
 		dbglvl = dbgdef;	// Toggle the needed bits.
 	}
 
-	/* Currently, need: `castfs mnt-dir -o stage=stage-dir` */
-	if (argc < 4) {
-fprintf(stderr, "castfs: argc wrong\n");
-		usage();
-		exit(-1);
-	}
-
 	/* dump some nice header information in the log file */
 	cast_log(CAST_DBG_MAIN, "main\n");
 	cast_log(CAST_DBG_MAIN, "logfile %s fd %d\n", getenv("CASTFS_LOGFILE"),
