From 61cdd06fced74438c36c26b863635a2d71202aa9 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sun, 19 Feb 2023 16:16:20 +0100
Subject: [PATCH 1/3] Fix missing includes

Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 src/errors.c | 1 +
 src/rpl.c    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/errors.c b/src/errors.c
index cf521f652c3b..80757f1e4471 100644
--- a/src/errors.c
+++ b/src/errors.c
@@ -26,6 +26,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 
 #include "global.h"
 #include "resources.h"
diff --git a/src/rpl.c b/src/rpl.c
index 13b0bacf0fcc..e33306bb787a 100644
--- a/src/rpl.c
+++ b/src/rpl.c
@@ -39,6 +39,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <string.h>
 
 #include "hp48.h"
 #include "hp48_emu.h"
-- 
2.39.1

