From c42528a5dde510d0ad1c2a9e934242d5468f5ec0 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sun, 19 Feb 2023 16:17:09 +0100
Subject: [PATCH 3/3] Make decode_8_thru_f static

Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 src/emulate.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/emulate.c b/src/emulate.c
index 0dd2eec1e92e..91c287b6ec7a 100644
--- a/src/emulate.c
+++ b/src/emulate.c
@@ -693,6 +693,7 @@ decode_group_1()
   }
 }
 
+static
 inline int
 #ifdef __FunctionProto__
 decode_8_thru_f(int op1)
-- 
2.39.1

