--- a/src/buffer.c.orig	2006-11-02 13:46:59.000000000 +0100
+++ b/src/buffer.c	2006-11-02 13:50:05.000000000 +0100
@@ -196,7 +196,8 @@
   ct_none,
   ct_compress,
   ct_gzip,
-  ct_bzip2
+  ct_bzip2,
+  ct_lzma
 };
 
 struct zip_magic
@@ -213,6 +214,7 @@
   { ct_compress, 2, "\037\235", "compress", "-Z" },
   { ct_gzip,     2, "\037\213", "gzip", "-z"  },
   { ct_bzip2,    3, "BZh",      "bzip2", "-j" },
+  { ct_lzma,     3, "]\000\000", "lzma", "-Y" },
 };
 
 #define NMAGIC (sizeof(magic)/sizeof(magic[0]))
--- a/src/tar.c.orig	2006-11-02 13:47:07.000000000 +0100
+++ b/src/tar.c	2006-11-02 13:51:53.000000000 +0100
@@ -568,6 +568,8 @@
    N_("control pax keywords"), GRID+8 },
   {"label", 'V', N_("TEXT"), 0,
    N_("create archive with volume name TEXT; at list/extract time, use TEXT as a globbing pattern for volume name"), GRID+8 },
+  {"lzma", 'Y', 0, 0,
+   N_("filter the archive through lzma"), GRID+8 },
   {"bzip2", 'j', 0, 0,
    N_("filter the archive through bzip2"), GRID+8 },
   {"gzip", 'z', 0, 0,
@@ -1423,6 +1425,10 @@
 	}
       break;
 
+    case 'Y':
+      set_use_compress_program_option ("lzma");
+      break;
+
     case 'z':
       set_use_compress_program_option ("gzip");
       break;
