--- configure.in.orig	2005-09-11 03:14:04.000000000 -0400
+++ configure.in	2018-06-10 10:39:30.671774766 -0400
@@ -116,7 +116,7 @@
 AC_ARG_ENABLE(
   libsn, [  --enable-libsn        include code for LIBSN [default=no]])
 
-if test -n "$enable_libsn"; then
+if test "x$enable_libsn" = "xyes"; then
                     PKG_CHECK_MODULES(LIBSN, libstartup-notification-1.0 >= 0.5)
       		    AC_MSG_RESULT([yes])
                     STARTUP="-DHAVE_STARTUP_NOTIFICATION -DSN_API_NOT_YET_FROZEN=1",
