From 776e88981949ee54d20485bc78d709506e329a21 Mon Sep 17 00:00:00 2001
From: Jürgen Buchmüller <pullmoll@t-online.de>
Date: Wed, 9 Sep 2020 00:59:07 +0200
Subject: [PATCH] libmspub: fix build w/ gcc10

Origin: Void Linux
---
 src/lib/MSPUBMetaData.h |    1 +
 1 file changed, 1 insertion(+)

--- a/src/lib/MSPUBMetaData.h	2018-02-28 12:54:49.000000000 +0100
+++ b/src/lib/MSPUBMetaData.h	2020-09-09 00:57:50.806188126 +0200
@@ -10,6 +10,7 @@
 #ifndef INCLUDED_MSPUBMETADATA_H
 #define INCLUDED_MSPUBMETADATA_H
 
+#include <cstdint>
 #include <map>
 #include <utility>
 #include <vector>
