--- userspace/sysdig/CMakeLists.txt.orig	2016-10-25 18:43:37.000000000 -0400
+++ userspace/sysdig/CMakeLists.txt	2016-10-26 16:50:36.791215074 -0400
@@ -39,7 +39,8 @@
 if(NOT WIN32)
 
 	target_link_libraries(sysdig
-		sinsp)
+		sinsp
+		"${OPENSSL_LIBRARIES}")
 
 	if(USE_BUNDLED_NCURSES)
 		add_dependencies(csysdig ncurses)
@@ -47,6 +48,7 @@
 
 	target_link_libraries(csysdig
 		sinsp
+		"${OPENSSL_LIBRARIES}"
 		"${CURSES_LIBRARIES}")
 
 	add_subdirectory(man)
