From 0212af79b7ccfe49e212aca526f9360d8f876540 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 31 Dec 2018 11:08:41 +0100
Subject: [PATCH] fix Makefile

---
 source/Makefile | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/source/Makefile b/source/Makefile
index 177115f..8973b3a 100644
--- a/source/Makefile
+++ b/source/Makefile
@@ -23,17 +23,16 @@
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 PREFIX ?= /usr/local
 INCDIR ?= $(PREFIX)/include
-LIBDIR ?= $(PREFIX)/lib$(SUFFIX)
+LIBDIR ?= $(PREFIX)/lib
 
 MAJVERS = 3
 MINVERS = 9.2
 VERSION = $(MAJVERS).$(MINVERS)
 
 
-CPPFLAGS += $(shell pkgconf --cflags freetype2) 
-CPPFLAGS += -DVERSION=\"$(VERSION)\" -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS $(PCONFCFL)
-CXXFLAGS += -Wall -O2 -fPIC 
-CXXFLAGS += -march=native
+CPPFLAGS += $(shell pkg-config --cflags freetype2) -I.
+CPPFLAGS += -DVERSION=\"$(VERSION)\" -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS $(PCONFCFL) -DPIC
+CXXFLAGS += -Wall -fPIC
 LDLIBS +=
 
 
-- 
2.20.1

