From e5fad99c3a6c76f6efcbe1eae964be137be6dae6 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Mon, 16 Oct 2023 14:15:14 +0200
Subject: [PATCH] unity fix

---
 src/keditbookmarks_p.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/keditbookmarks_p.h b/src/keditbookmarks_p.h
index 3f8cc78..ef6342e 100644
--- a/src/keditbookmarks_p.h
+++ b/src/keditbookmarks_p.h
@@ -4,7 +4,7 @@
 
     SPDX-License-Identifier: LGPL-2.0-only
 */
-
+#pragma once
 #include <QString>
 
 #include "kbookmarks_export.h"
-- 
2.42.0

