--- pad.c.ori	2000-07-23 18:14:10.000000000 +0200
+++ pad.c	2003-12-09 19:01:40.000000000 +0100
@@ -277,24 +277,25 @@
 /* print out the help message */
 void printhelp()
 {
-  fprintf(stderr, "Usage: pad [options] [input files]
-
-options:
-
-  -r [num]    - Pad will XOR in [num] pads generated from random data. if this
-                is ommited, pad will use either 1 or 0 random pads, depending on
-                if there is only one input file or more than one, respectively.
-
-  -o [output] - The result of all the XOR operations will be stored in this
-                file. If [output] is ommitted, pad will name the file according
-                to its MD5-sum, along with the rest of the random pads.
-
-  -s [size]   - The output data will be [size] bytes. If this is less than the
-                smallest input file, the output will be clipped to match, if it
-                is larger, random data will be appended to the end.
-
-  -h          - Show usage.
-
+  fprintf(stderr, "\
+Usage: pad [options] [input files]                                                \n\
+                                                                                  \n\
+options:                                                                          \n\
+                                                                                  \n\
+  -r [num]    - Pad will XOR in [num] pads generated from random data. if this    \n\
+                is ommited, pad will use either 1 or 0 random pads, depending on  \n\
+                if there is only one input file or more than one, respectively.   \n\
+                                                                                  \n\
+  -o [output] - The result of all the XOR operations will be stored in this       \n\
+                file. If [output] is ommitted, pad will name the file according   \n\
+                to its MD5-sum, along with the rest of the random pads.           \n\
+                                                                                  \n\
+  -s [size]   - The output data will be [size] bytes. If this is less than the    \n\
+                smallest input file, the output will be clipped to match, if it   \n\
+                is larger, random data will be appended to the end.               \n\
+                                                                                  \n\
+  -h          - Show usage.                                                       \n\
+                                                                                  \n\
 ");
 }
 
