From 929978f8a29dc54c36cd5d862fbe02385b6fc2a3 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 17 Jan 2018 19:55:14 +0100
Subject: [PATCH] fix dependency

---
 ed25519.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ed25519.cabal b/ed25519.cabal
index 5e5f715..c16618f 100644
--- a/ed25519.cabal
+++ b/ed25519.cabal
@@ -60,7 +60,7 @@ flag no-donna
 
 library
   build-depends:
-    ghc-prim    >= 0.1 && < 0.5,
+    ghc-prim    >= 0.1 && < 0.6,
     base        >= 4   && < 5,
     bytestring  >= 0.9 && < 0.11
 
-- 
2.15.1

