From d148c444e4ff9387da8ada8adf6c3af34e1c4729 Mon Sep 17 00:00:00 2001
From: Jürgen Buchmüller <pullmoll@t-online.de>
Date: Sat, 27 Jun 2020 12:21:58 +0200
Subject: [PATCH] libfreehand: fix build w/ newer icu

Origin: Void Linux

--- a/src/lib/libfreehand_utils.cpp
+++ b/src/lib/libfreehand_utils.cpp
@@ -10,6 +10,8 @@
 #include <cstdarg>
 #include <cstdio>
 
+#define UPRV_BLOCK_MACRO_BEGIN
+#define UPRV_BLOCK_MACRO_END
 #include <unicode/utf8.h>
 #include <unicode/utf16.h>
 #include "libfreehand_utils.h"
