From 84732ad38793ae5082f96f5acb1b784fd9028358 Mon Sep 17 00:00:00 2001
From: Jürgen Buchmüller <pullmoll@t-online.de>
Date: Thu, 9 Aug 2018 11:08:16 +0200
Subject: [PATCH] libfreehand: fix gcc-8.2.0 build

Origin: Void Linux

--- a/src/lib/FHCollector.cpp
+++ b/src/lib/FHCollector.cpp
@@ -1907,7 +1907,7 @@
   textObjectProps.insert("svg:width", width);
   for (int i=0; i<4; ++i) // osnola: let assume that there is no padding
   {
-    char const *(padding[])= {"fo:padding-left","fo:padding-right","fo:padding-top","fo:padding-bottom"};
+    char const *padding[] = {"fo:padding-left","fo:padding-right","fo:padding-top","fo:padding-bottom"};
     textObjectProps.insert(padding[i],0,librevenge::RVNG_POINT);
   }
   if (!FH_ALMOST_ZERO(rotation))
