--- CMakeLists.txt.orig	2008-10-01 15:21:46.000000000 -0500
+++ CMakeLists.txt	2008-10-01 15:19:26.000000000 -0500
@@ -113,10 +113,10 @@
 
 # Automagically create version header file.
 # does not work with mingw for me (jpc)
-if(UNIX)
-    include(CreateSVNVersionHeader)
-    create_svn_version_header()
-endif(UNIX)
+#if(UNIX)
+#    include(CreateSVNVersionHeader)
+#    create_svn_version_header()
+#endif(UNIX)
 
 if(EXISTS ${CMAKE_SOURCE_DIR}/include/config.h)
     # This file may exist (created by an alternate process to the svn test above),
