From 8786a9d34ed175bd18da182f53d939c446e743f9 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Thu, 30 Apr 2015 22:43:40 +0200
Subject: [PATCH 1/2] fix compile error

---
 src/gprbuild-compilation-slave.adb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gprbuild-compilation-slave.adb b/src/gprbuild-compilation-slave.adb
index 73b4b93..33383c9 100644
--- a/src/gprbuild-compilation-slave.adb
+++ b/src/gprbuild-compilation-slave.adb
@@ -817,7 +817,7 @@ package body Gprbuild.Compilation.Slave is
                S : Slave := Slave_S.Element (C);
             begin
                Proc (S);
-               Pool (C) := S;
+               Pool.Replace_Element(Position => C, New_Item => S);
             end;
          end loop;
       end Iterate;
-- 
2.3.5

