From ec9f15aeebd4f29f856e7a23153977f26d7818d0 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Thu, 30 Apr 2015 22:43:59 +0200
Subject: [PATCH 2/2] always assume that we are not on  a VMS

---
 gnat/hostparm.ads | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/gnat/hostparm.ads b/gnat/hostparm.ads
index 16b4b39..814ae07 100644
--- a/gnat/hostparm.ads
+++ b/gnat/hostparm.ads
@@ -41,10 +41,7 @@ package Hostparm is
    -- HOST Parameters --
    ---------------------
 
-   Gnat_VMSp : Integer;
-   pragma Import (C, Gnat_VMSp, "__gnat_vmsp");
-
-   OpenVMS : Boolean := Gnat_VMSp /= 0;
+   OpenVMS : Boolean := False;
    --  Set True for OpenVMS host. See also OpenVMS target boolean in
    --  system-vms.ads and system-vms_64.ads and OpenVMS_On_Target boolean in
    --  Targparm. This is not a constant, because it can be modified by -gnatdm.
-- 
2.3.5

