From dc3c01a438029d13e5ed98db0af78a86a9db4e7e Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 12 Nov 2018 22:08:43 +0100
Subject: [PATCH] fix pycairo.h include

---
 gi/pygi-foreign-cairo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gi/pygi-foreign-cairo.c b/gi/pygi-foreign-cairo.c
index 718f9a0..a0153a1 100644
--- a/gi/pygi-foreign-cairo.c
+++ b/gi/pygi-foreign-cairo.c
@@ -25,7 +25,7 @@
 #include <cairo.h>
 
 #if PY_VERSION_HEX < 0x03000000
-#include <pycairo.h>
+#include <pycairo/pycairo.h>
 static Pycairo_CAPI_t *Pycairo_CAPI;
 #else
 #include <py3cairo.h>
-- 
2.19.1

