diff -uNr gnomebaker-0.6.0.old/src/execfunctions.c gnomebaker-0.6.0/src/execfunctions.c
--- gnomebaker-0.6.0.old/src/execfunctions.c	2006-09-17 10:06:08.000000000 +0200
+++ gnomebaker-0.6.0/src/execfunctions.c	2006-09-17 10:06:44.000000000 +0200
@@ -271,8 +271,6 @@
 	if(preferences_get_bool(GB_CDRECORD_FORCE))
 	   exec_cmd_add_arg(cmd, "-force");
 
-	exec_cmd_add_arg(cmd, "gracetime=5");
-
     const gint speed = preferences_get_int(GB_CDWRITE_SPEED);
     if(speed > 0)
 	   exec_cmd_add_arg(cmd, "speed=%d", speed);
@@ -391,7 +389,6 @@
 	if(preferences_get_bool(GB_EJECT))
 		exec_cmd_add_arg(cmd, "-eject");
 
-	exec_cmd_add_arg(cmd, "gracetime=5");
 	exec_cmd_add_arg(cmd, "blank=%s", preferences_get_bool(GB_FAST_BLANK) ? "fast" : "all");
 }
 
