diff -urNp gnomebaker-0.5.1.orig/src/execfunctions.c gnomebaker-0.5.1/src/execfunctions.c
--- gnomebaker-0.5.1.orig/src/execfunctions.c	2006-05-23 13:37:04.000000000 +0200
+++ gnomebaker-0.5.1/src/execfunctions.c	2006-05-23 13:47:42.000000000 +0200
@@ -236,7 +236,6 @@ cdrecord_add_common_args(ExecCmd* cdBurn
 	if(preferences_get_bool(GB_CDRECORD_FORCE))
 	   exec_cmd_add_arg(cdBurn, "-force");
 	
-	exec_cmd_add_arg(cdBurn, "gracetime=5");
 	exec_cmd_add_arg(cdBurn, "speed=%d", preferences_get_int(GB_CDWRITE_SPEED));
 	exec_cmd_add_arg(cdBurn, "-v");	
 
@@ -385,7 +384,6 @@ cdrecord_add_blank_args(ExecCmd* cdBurn)
 	if(preferences_get_bool(GB_EJECT))
 		exec_cmd_add_arg(cdBurn, "-eject");
 	
-	exec_cmd_add_arg(cdBurn, "gracetime=5");
 	exec_cmd_add_arg(cdBurn, "blank=%s", preferences_get_bool(GB_FAST_BLANK) ? "fast" : "all");
 }
 
