--- configure.ac.orig	2007-08-27 00:51:31.000000000 +0200
+++ configure.ac	2007-08-27 00:52:14.000000000 +0200
@@ -197,7 +197,7 @@
 
     AC_CACHE_CHECK(whether gcc supports x86 inline asm,
                    ac_cv_c_inline_asm,
-                   [AC_TRY_LINK(,[{int x; asm volatile("movl %%eax, %%ebx\n"::
+                   [AC_TRY_LINK(,[{int x; asm volatile("movl %%eax, %%ebx\n\t pushal\n\t popal"::
                           "m" (x),"m" (x),"m" (x),"m" (x),"m" (x),"m" (x),
                           "m" (x),"m" (x),"m" (x),"m" (x),"m" (x),"m" (x));}],
                    ac_cv_c_inline_asm=yes,
@@ -840,11 +840,6 @@
 else
     gnustepdir=$appspath
 
-    if test "x$GNUSTEP_LOCAL_ROOT" != "x" ; then
-	gnustepdir=`echo "$GNUSTEP_LOCAL_ROOT" | sed -e "s|^${prefix}|prefix|"`
-	gnustepdir=`echo $gnustepdir | sed -e 's|^prefix|${prefix}|'`
-    fi
-
     wprefs_base_dir=$gnustepdir/Applications
     wprefs_datadir=$wprefs_base_dir/WPrefs.app
     wprefs_bindir=$wprefs_base_dir/WPrefs.app
