diff -Naupr QuteScoop.orig/ControllerDetails.cpp QuteScoop.linux/ControllerDetails.cpp
--- QuteScoop.orig/ControllerDetails.cpp	2007-12-30 21:43:01.000000000 +0100
+++ QuteScoop.linux/ControllerDetails.cpp	2008-01-12 23:48:39.740908442 +0100
@@ -99,6 +99,8 @@ void ControllerDetails::on_btnJoinChanne
 	
 #ifdef Q_WS_WIN
 	QString program = "start";
+#elif defined Q_WS_X11
+	QString program = "teamspeak";
 #else
 	QString program = "open";
 #endif
