diff -Naur LVM2.2.02.171.orig/daemons/dmeventd/Makefile.in LVM2.2.02.171/daemons/dmeventd/Makefile.in
--- LVM2.2.02.171.orig/daemons/dmeventd/Makefile.in	2017-08-28 23:54:14.525826374 -0700
+++ LVM2.2.02.171/daemons/dmeventd/Makefile.in	2017-08-28 23:57:59.031726462 -0700
@@ -67,7 +67,7 @@
 
 dmeventd.static: $(LIB_STATIC) dmeventd.o $(interfacebuilddir)/libdevmapper.a
 	$(CC) $(CFLAGS) $(LDFLAGS) $(ELDFLAGS) -static -L. -L$(interfacebuilddir) -o $@ \
-	dmeventd.o $(DL_LIBS) $(LVMLIBS) $(LIBS) $(STATIC_LIBS)
+	dmeventd.o $(DL_LIBS) $(LVMLIBS) $(LIBS) $(STATIC_LIBS) -lm
 
 ifeq ("@PKGCONFIG@", "yes")
   INSTALL_LIB_TARGETS += install_pkgconfig
diff -Naur LVM2.2.02.171.orig/tools/Makefile.in LVM2.2.02.171/tools/Makefile.in
--- LVM2.2.02.171.orig/tools/Makefile.in	2017-08-28 23:54:14.576827474 -0700
+++ LVM2.2.02.171/tools/Makefile.in	2017-08-28 23:54:36.258293501 -0700
@@ -134,7 +134,7 @@
 
 dmsetup.static: dmsetup.o $(interfacebuilddir)/libdevmapper.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -static -L$(interfacebuilddir) \
-	      -o $@ dmsetup.o -ldevmapper $(M_LIBS) $(PTHREAD_LIBS) $(STATIC_LIBS) $(LIBS)
+	      -o $@ dmsetup.o -ldevmapper $(M_LIBS) $(PTHREAD_LIBS) $(STATIC_LIBS) $(LIBS) -pthread
 
 all: device-mapper
 
@@ -161,7 +161,7 @@
 
 lvm.static: $(OBJECTS) lvm-static.o $(top_builddir)/lib/liblvm-internal.a  $(interfacebuilddir)/libdevmapper.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -static -L$(interfacebuilddir) -o $@ \
-	      $(OBJECTS) lvm-static.o $(LVMLIBS) $(STATIC_LIBS) $(LIBS)
+	      $(OBJECTS) lvm-static.o $(LVMLIBS) $(STATIC_LIBS) $(LIBS) -lm -luuid -pthread
 
 liblvm2cmd.a: $(top_builddir)/lib/liblvm-internal.a $(OBJECTS) lvmcmdlib.o lvm2cmd.o
 	cat $(top_builddir)/lib/liblvm-internal.a > $@
