diff -aur libgig-3.2.1-orig/src/DLS.cpp libgig-3.2.1/src/DLS.cpp
--- libgig-3.2.1-orig/src/DLS.cpp	2008-06-24 13:35:32.000000000 +0200
+++ libgig-3.2.1/src/DLS.cpp	2008-06-24 13:37:23.000000000 +0200
@@ -24,6 +24,7 @@
 #include "DLS.h"
 
 #include <time.h>
+#include <algorithm>
 
 #ifdef __APPLE__
 #include <CoreFoundation/CFUUID.h>
diff -aur libgig-3.2.1-orig/src/gig.cpp libgig-3.2.1/src/gig.cpp
--- libgig-3.2.1-orig/src/gig.cpp	2008-06-24 13:35:32.000000000 +0200
+++ libgig-3.2.1/src/gig.cpp	2008-06-24 13:36:57.000000000 +0200
@@ -27,6 +27,7 @@
 
 #include <math.h>
 #include <iostream>
+#include <algorithm>
 
 /// Initial size of the sample buffer which is used for decompression of
 /// compressed sample wave streams - this value should always be bigger than
diff -aur libgig-3.2.1-orig/src/RIFF.cpp libgig-3.2.1/src/RIFF.cpp
--- libgig-3.2.1-orig/src/RIFF.cpp	2008-06-24 13:35:32.000000000 +0200
+++ libgig-3.2.1/src/RIFF.cpp	2008-06-24 13:36:40.000000000 +0200
@@ -22,6 +22,7 @@
  ***************************************************************************/
 
 #include <string.h>
+#include <algorithm>
 
 #include "RIFF.h"
 
