From 89fb169f3d2c9bf742eba9087ce6e8ead3548652 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 7 Apr 2014 18:35:00 +0200
Subject: [PATCH] use system ucl

---
 src/Makefile | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 5375f81..83c0583 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -52,11 +52,7 @@ objext ?= .o
 upx_SOURCES := $(sort $(wildcard $(srcdir)/*.cpp))
 upx_OBJECTS := $(notdir $(upx_SOURCES:.cpp=$(objext)))
 
-# we need UCL and zlib - you can set envvar UPX_UCLDIR
-ifneq ($(wildcard $(UPX_UCLDIR)/include/ucl/ucl.h),)
-INCLUDES += -I$(UPX_UCLDIR)/include
-LIBS += $(addprefix -L,$(dir $(wildcard $(UPX_UCLDIR)/libucl$(libext) $(UPX_UCLDIR)/src/.libs/libucl$(libext))))
-endif
+INCLUDE += -I$(INSTALL_ROOT)/usr/include/ucl
 LIBS += -lucl -lz
 # you should set envvar UPX_LZMADIR to point to your unpacked LZMA SDK
 include $(top_srcdir)/src/stub/src/c/Makevars.lzma
-- 
1.9.0

