diff -uNr eigen2/CMakeLists.txt eigen2.cp/CMakeLists.txt
--- eigen2/CMakeLists.txt	2009-10-24 00:37:05.000000000 +0200
+++ eigen2.cp/CMakeLists.txt	2009-10-24 09:38:43.000000000 +0200
@@ -80,7 +80,7 @@
 include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
 
 if(EIGEN_BUILD_PKGCONFIG)
-    configure_file(eigen2.pc.in eigen2.pc)
+    configure_file(eigen2.pc.in ${CMAKE_INSTALL_PREFIX}/eigen2.pc)
     install(FILES ${CMAKE_INSTALL_PREFIX}/eigen2.pc
         DESTINATION lib/pkgconfig
         )
