diff -Naur playvtx-0.9.1/configure.in playvtx-0.9.1-fixed/configure.in
--- playvtx-0.9.1/configure.in	2005-11-17 13:04:47.000000000 +0100
+++ playvtx-0.9.1-fixed/configure.in	2008-02-07 20:47:40.000000000 +0100
@@ -7,7 +7,7 @@
 AC_PROG_CC
 
 # Checks for libraries.
-AC_CHECK_LIB(ayemu, ayemu_start,,
+AC_CHECK_LIB(ayemu, ayemu_init,,
 AC_MSG_ERROR(cannot find ayemu library. Check your ayemu library installation))
 AC_OUTPUT
 
