From 5aabcb01e7b6cf814f9076cb6f591859c97d0b2f Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Thu, 29 Jun 2017 09:01:18 +0200
Subject: [PATCH 3/3] more functioanl.h fixes

---
 Source/WebCore/dom/SlotAssignment.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/WebCore/dom/SlotAssignment.h b/Source/WebCore/dom/SlotAssignment.h
index 0fcd4dfa..9edd068f 100644
--- a/Source/WebCore/dom/SlotAssignment.h
+++ b/Source/WebCore/dom/SlotAssignment.h
@@ -33,6 +33,7 @@
 #include <wtf/Vector.h>
 #include <wtf/text/AtomicString.h>
 #include <wtf/text/AtomicStringHash.h>
+#include <functional>
 
 namespace WebCore {
 
-- 
2.13.2

