--- state_ui.h.orig	2006-11-11 18:32:15.000000000 +0400
+++ state_ui.h	2006-11-11 18:33:18.000000000 +0400
@@ -16,22 +16,6 @@
 #include <gtklog.h>
 #include <gtkutils.h>
 
-/* this got stripped from gtklog, so it is in here for now: */
-struct _GaimGtkLogViewer {
-	GList *logs;                 /**< The list of logs viewed in this viewer   */
-
-	GtkWidget        *window;    /**< The viewer's window                      */
-	GtkTreeStore     *treestore; /**< The treestore containing said logs       */
-	GtkWidget        *treeview;  /**< The treeview representing said treestore */
-	GtkWidget        *imhtml;    /**< The imhtml to display said logs          */
-	GtkWidget        *entry;     /**< The search entry, in which search terms
-	                              *   are entered                              */
-	GaimLogReadFlags flags;      /**< The most recently used log flags         */
-	char             *search;    /**< The string currently being searched for  */
-	GtkWidget        *label;     /**< The label at the top of the log viewer   */
-};
-typedef struct _GaimGtkLogViewer GaimGtkLogViewer;
-
 void GE_state_ui_init();
 void GE_state_ui_delete();
 
