--- src/Makefile.inc.old	2006-01-01 19:28:20.000000000 -0800
+++ src/Makefile.inc	2006-01-01 19:27:43.000000000 -0800
@@ -2,7 +2,7 @@
 
 # Where to install stuff
 
-MOSMLHOME=${HOME}/mosml
+MOSMLHOME=${INSTALL_ROOT}/usr
 
 # Various utility programs
 INSTALL_PROGRAM=cp
@@ -114,10 +114,10 @@
 # DOCDIR contains documentation
 
 BINDIR=${MOSMLHOME}/bin
-LIBDIR=${MOSMLHOME}/lib
-INCDIR=${MOSMLHOME}/include
-DOCDIR=${MOSMLHOME}/doc
-TOOLDIR=${MOSMLHOME}/tools
+LIBDIR=${MOSMLHOME}/lib/moscow_ml
+INCDIR=${MOSMLHOME}/include/moscow_ml
+DOCDIR=${MOSMLHOME}/share/doc
+TOOLDIR=${MOSMLHOME}/libexec/moscow_ml
 # LIBDIR=${MOSMLHOME}/lib/moscow_ml
 # TOOLDIR=${MOSMLHOME}/libexec/moscow_ml
 
