--- rss-glx_0.8.1/other_src/Makefile.in	2006-02-25 21:14:56.000000000 +0000
+++ rss-glx_0.8.1.new/other_src/Makefile.in	2008-03-27 16:00:53.000000000 +0000
@@ -232,6 +232,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/src
+AM_CFLAGS = $(Wand_CFLAGS)
 biof_SOURCES = driver.c biof.c
 busyspheres_SOURCES = driver.c busyspheres.c
 matrixview_SOURCES = driver.c matrixview.c matrixview_textures.c
