--- gcl-2.6.7/configure~	2005-01-15 19:17:17.000000000 +0000
+++ gcl-2.6.7/configure	2006-01-13 18:13:29.000000000 +0000
@@ -1697,11 +1697,9 @@
 done
 test -n "$MAKEINFO" || MAKEINFO=""false""
 
-
-
-if test -f /usr/lib/NextStep/software_version; then
-    system=NEXTSTEP-`${AWK} '/3/,/3/' /usr/lib/NextStep/software_version`
-else
+# if test -f /usr/lib/NextStep/software_version; then
+#    system=NEXTSTEP-`${AWK} '/3/,/3/' /usr/lib/NextStep/software_version`
+# else
     system=`uname -s`-`uname -r`
     if test "$?" -ne 0 ; then
 	echo "$ac_t""unknown (can't find uname command)" 1>&6
@@ -1710,15 +1708,15 @@
 	# Special check for weird MP-RAS system (uname returns weird
 	# results, and the version is kept in special file).
     
-	if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-	    system=MP-RAS-`${AWK} '{print $3}' /etc/.relid'`
-	fi
-	if test "`uname -s`" = "AIX" ; then
-	    system=AIX-`uname -v`.`uname -r`
-	fi
+	# if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
+	#    system=MP-RAS-`${AWK} '{print $3}' /etc/.relid'`
+	# fi
+	# if test "`uname -s`" = "AIX" ; then
+	#    system=AIX-`uname -v`.`uname -r`
+	# fi
 	echo "$ac_t""$system" 1>&6
     fi
-fi
+# fi
 
 # sysconf
 
