Created by Peng Chang (Charles) <chp@sourcemage.org>
diff -Nurp gcl-orig/xgcl-2/makefile gcl/xgcl-2/makefile
--- gcl-orig/xgcl-2/makefile	2009-06-11 13:44:40.000000000 +0800
+++ gcl/xgcl-2/makefile	2009-06-11 13:47:50.000000000 +0800
@@ -8,15 +8,15 @@ objects: $(LISP)
 #saved_%: $(LISP)
 #	echo '(load "sysdef.lisp")(xlib::compile-xgcl)(xlib::save-xgcl "$@")' | $(LISP)
 
-docs: dwdoc/dwdoccontents.html dwdoc.pdf
+docs: dwdoc/dwdoccontents.html
 
 dwdoc/dwdoccontents.html: $(LISP)
 	mkdir -p $(@D) && \
 	cd $(@D) && \
 	echo '(load "../sysdef.lisp")(in-package :xlib)(defmacro while (test &rest forms) `(loop (unless ,test (return)) ,@forms))(load "../gcl_tohtml.lsp")(load "../gcl_index.lsp")(tohtml "../dwdoc.tex" "dwdoc")(with-open-file (s "dwdoccontents.html" :direction :output) (let ((*standard-output* s)) (xlib::makecont "../dwdoc.tex" 1 "dwdoc")))(with-open-file (s "dwdocindex.html" :direction :output) (let ((*standard-output* s)) (xlib::printindex indexdata "dwdoc")))' | ../$<
 
-dwdoc.pdf: dwdoc.tex
-	pdflatex $<
+#dwdoc.pdf: dwdoc.tex
+#	pdflatex $<
 
 #dwdoc/index.html: dwdoc.tex
 #	latex2html $<
