From 5993fd4c4fa70b31525924322f031dd1039033d1 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Tue, 9 Apr 2013 18:49:58 +0200
Subject: [PATCH] respect CXXFLAGS

---
 configure.in | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index e9ab81d..e102c2a 100644
--- a/configure.in
+++ b/configure.in
@@ -7,8 +7,7 @@ AC_PREREQ(2.13)
 
 dnl Set our default compilation flags.
 
-CXXFLAGS="-O3 -fno-rtti -fno-exceptions"
-CFLAGS="-O3"
+CXXFLAGS="${CXXFLAGS} -fno-rtti -fno-exceptions"
 
 dnl Reset default linking directives.
 
-- 
1.8.2

