--- nxcomp-3.3.0-3/Loop.cpp.orig	2011-03-18 12:52:20.774151616 +0100
+++ nxcomp-3.3.0-3/Loop.cpp	2011-03-18 12:53:03.219512776 +0100
@@ -10979,7 +10979,7 @@
                                       packMethod == PACK_LOSSLESS ||
                                           packMethod == PACK_ADAPTIVE)
   {
-    char *dash = rindex(opt, '-');
+    const char *dash = rindex(opt, '-');
 
     if (dash != NULL && strlen(dash) == 2 &&
             *(dash + 1) >= '0' && *(dash + 1) <= '9')
