--- providers/bdb/gda-bdb-recordset.c.orig	2005-03-18 17:40:40.000000000 -0600
+++ providers/bdb/gda-bdb-recordset.c	2005-03-18 17:43:04.000000000 -0600
@@ -225,8 +225,12 @@
 	cnc_priv_data = g_object_get_data (G_OBJECT (cnc),
 					   OBJECT_DATA_BDB_HANDLE);
 
+
 	/* get the number of records in the database */
 	ret = dbp->stat (dbp,
+#if BDB_VERSION > 40300
+                        NULL,
+#endif
 			 &statp,
 #if BDB_VERSION < 40000
 			 NULL,
