--- src/text/fmt/xp/fp_TextRun.cpp.orig	2009/12/19 00:44:28
+++ src/text/fmt/xp/fp_TextRun.cpp	2010/04/03 13:54:44
@@ -3396,7 +3396,7 @@
 		
 	m_pRenderInfo->m_pText = text;
 	m_pRenderInfo->m_iOffset = iDocumentPosition - iRunOffset;
-	m_pRenderInfo->m_iLength = getLength();
+	m_pRenderInfo->m_iLength = iCount;
 	if(!getGraphics()->needsSpecialCaretPositioning(*m_pRenderInfo))
 	{
 		DELETEP(text);
