diff -Naur gradm2.old/Makefile gradm2/Makefile
--- gradm2.old/Makefile	2005-05-14 12:10:34.000000000 -0700
+++ gradm2/Makefile	2005-11-02 20:31:41.845625528 -0800
@@ -133,11 +133,6 @@
 	@echo "Installing gradm manpage..."
 	@mkdir -p $(DESTDIR)$(MANDIR)/man8
 	@$(INSTALL) -m 0644 gradm.8 $(DESTDIR)$(MANDIR)/man8/$(GRADM_BIN).8
-	@if [ -x /sbin/$(GRADM_BIN) ] ; then \
-		if [ -z $(DESTDIR) ] && [ ! -f $(GRSEC_DIR)/pw ] ; then \
-			/sbin/$(GRADM_BIN) -P ; \
-		fi \
-	fi
 	@true
 
 clean:
