From d3956fa4f7d2c82663a990f7b731c2a28054a29f Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 3 Feb 2017 10:04:11 +0100
Subject: [PATCH] do not hard-set prefix in Makefile

---
 Makefile | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 010fec8..8efe63f 100644
--- a/Makefile
+++ b/Makefile
@@ -15,11 +15,9 @@ VERSION := $(shell bin/get_version.sh --version)
 RELEASE := $(shell bin/get_version.sh --release)
 FULL    := $(shell bin/get_version.sh --full)
 
-PREFIX  := /usr/local
-
 CFG_DIR := $(PREFIX)/etc
-BIN_DIR := $(PREFIX)/bin
-MAN_DIR := $(PREFIX)/share/man
+BIN_DIR := $(PREFIX)/usr/bin
+MAN_DIR := $(PREFIX)/usr/share/man
 TMP_DIR := $(shell mktemp -d)
 FILES   := $(wildcard bin/*) $(wildcard man/*) README Makefile \
 	   $(wildcard rpm/*) lcovrc
-- 
2.11.1

