Index: dia-0.97.1/objects/custom/shape_typeinfo.c
===================================================================
--- dia-0.97.1.orig/objects/custom/shape_typeinfo.c
+++ dia-0.97.1/objects/custom/shape_typeinfo.c
@@ -27,7 +27,7 @@
 #include "custom_util.h"
 #include <string.h>
 #include <stdarg.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <glib/gstdio.h>
 #include <libxml/parser.h>
 
Index: dia-0.97.1/plug-ins/pgf/render_pgf.c
===================================================================
--- dia-0.97.1.orig/plug-ins/pgf/render_pgf.c
+++ dia-0.97.1/plug-ins/pgf/render_pgf.c
@@ -61,7 +61,7 @@ TODO:
 #endif
 #include <errno.h>
 
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <glib/gstdio.h>
 
 #include "intl.h"
Index: dia-0.97.1/tests/test-boundingbox.c
===================================================================
--- dia-0.97.1.orig/tests/test-boundingbox.c
+++ dia-0.97.1/tests/test-boundingbox.c
@@ -28,7 +28,7 @@
 #include <glib.h>
 #include <glib-object.h>
 
-#if GLIB_CHECK_VERSION(2,16,0)
+#if GLIB_CHECK_VERSION(2,16,0) && !GLIB_CHECK_VERSION(2,31,0)
 #include <glib/gtestutils.h>
 #endif
 #include "dialib.h"
Index: dia-0.97.1/tests/test-objects.c
===================================================================
--- dia-0.97.1.orig/tests/test-objects.c
+++ dia-0.97.1/tests/test-objects.c
@@ -28,7 +28,7 @@
 #include <glib.h>
 #include <glib-object.h>
 
-#if GLIB_CHECK_VERSION(2,16,0)
+#if GLIB_CHECK_VERSION(2,16,0) && !GLIB_CHECK_VERSION(2,31,0)
 #include <glib/gtestutils.h>
 #endif
 

