From 84bd8e3fb04383ee9272cc88130e6773dfe486a6 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Wed, 23 Nov 2016 11:54:41 -0200
Subject: [PATCH 1/2] Add knob to suppress building the toxav test

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7c08fd82554d..88ba43b4473c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -463,7 +463,9 @@ endif()
 #
 ################################################################################
 
-if(NOT WIN32 AND BUILD_TOXAV AND SNDFILE_FOUND AND PORTAUDIO_FOUND AND OPENCV_FOUND)
+option(BUILD_AV_TEST "Build toxav test" ON)
+if(NOT WIN32 AND BUILD_AV_TEST
+   AND BUILD_TOXAV AND SNDFILE_FOUND AND PORTAUDIO_FOUND AND OPENCV_FOUND)
   add_c_executable(av_test testing/av_test.c)
   target_link_modules(av_test
     toxav
-- 
2.10.2

