From b13629a72586c650462c918c0802db06e1ed9352 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Wed, 23 Nov 2016 11:55:10 -0200
Subject: [PATCH 2/2] Add knob to suppress building the nTox client

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 88ba43b4473c..da7e26391753 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -479,7 +479,8 @@ if(NOT WIN32 AND BUILD_AV_TEST
   endif()
 endif()
 
-if(NOT WIN32)
+option(BUILD_NTOX "Build nTox client" ON)
+if(NOT WIN32 AND BUILD_NTOX)
   add_c_executable(nTox testing/nTox.c)
   target_link_modules(nTox toxcore ${NCURSES_LIBRARIES})
 endif()
-- 
2.10.2

