From c6e424d812ad3ce0c402e88efcbf009eb8a29ca5 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Wed, 23 Nov 2016 22:20:12 -0200
Subject: [PATCH 3/3] Add nTox to the install target

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 74c25d7b5d96..1130be3126f5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -484,6 +484,7 @@ option(BUILD_NTOX "Build nTox client" ON)
 if(NOT WIN32 AND BUILD_NTOX)
   add_c_executable(nTox testing/nTox.c)
   target_link_modules(nTox toxcore ${NCURSES_LIBRARIES})
+  install(TARGETS nTox RUNTIME DESTINATION "bin")
 endif()

 add_c_executable(DHT_test testing/DHT_test.c)
--
2.10.2
