Description: Fix build with glibmm 2.4
Origin: patch imported from OpenSUSE, http://download.opensuse.org/factory/repo/src-oss/suse/src/libdbus-c++-0.9.0-12.2.src.rpm
Author: dimstar@opensuse.org
Bug-Debian: https://bugs.debian.org/793323
Last-Update: 2015-07-22

--- a/examples/glib/dbus-browser.h
+++ b/examples/glib/dbus-browser.h
@@ -1,9 +1,9 @@
 #ifndef __DEMO_DBUS_BROWSER_H
 #define __DEMO_DBUS_BROWSER_H
 
+#include <gtkmm.h>
 #include <dbus-c++/dbus.h>
 #include <dbus-c++/glib-integration.h>
-#include <gtkmm.h>
 
 #include "dbus-glue.h"
 
