diff -ur dbus-c++-20130217/configure.ac dbus-c++-mod/configure.ac
--- dbus-c++-20130217/configure.ac	2013-02-17 18:10:27.000000000 -0600
+++ dbus-c++-mod/configure.ac	2013-02-17 18:13:25.363189012 -0600
@@ -4,7 +4,7 @@
 AC_INIT([libdbus-c++], 0.9.0, [andreas.volz@tux-style.com])
 
 AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADER([config.h])
 AC_CONFIG_MACRO_DIR([m4])
 
 AC_CANONICAL_HOST
