From b9fe4d93203e0000072f5d02536e3e15f2b41b74 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sun, 21 Jun 2015 12:16:09 +0200
Subject: [PATCH] fix Makefile

---
 MANIFEST.gnat_util | 2 +-
 Makefile           | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/MANIFEST.gnat_util b/MANIFEST.gnat_util
index 08406e2..e8eba73 100644
--- a/MANIFEST.gnat_util
+++ b/MANIFEST.gnat_util
@@ -1,4 +1,4 @@
-Makefile.gnat_util
+Makefile
 MANIFEST.gnat_util
 README.gnat_util
 ali-util.adb
diff --git a/Makefile b/Makefile
index cf515e3..91554bf 100644
--- a/Makefile
+++ b/Makefile
@@ -1,12 +1,12 @@
 # This makefile is used to build and install Gnat_Util
 # For information on building and installing Gnat_Util see README.gnat_util.
 
-ENABLE_SHARED=no
+ENABLE_SHARED=yes
 BUILD_TYPE=Production
 
-prefix=/usr/local
+prefix=$(INSTALL_ROOT)/usr
 
-BUILDER=gprbuild -j0
+BUILDER=gprbuild
 INSTALLER=gprinstall
 
 OS_TYPE=unix
-- 
2.4.1

