--- back/Makefile	2004-12-14 18:55:05.000000000 +0100
+++ Makefile	2005-02-06 14:45:47.265954864 +0100
@@ -87,12 +87,12 @@
 $(error environment missing: MK_QTDIR=$(MK_QTDIR))
 endif
 
-ifeq ($(shell test -e $(MK_QTDIR)/include/qglobal.h$(colon) echo $$?),1)
+ifeq ($(shell test -e $(MK_QTDIR)/include/qt/qglobal.h$(colon) echo $$?),1)
 $(error environment incomplete: MK_QTDIR=$(MK_QTDIR))
 endif
 
 ifeq ($(QT_VER),0)
-QT_VER=$(shell grep QT_VERSION_STR $(MK_QTDIR)/include/qglobal.h | cut -d'"' -f2 | cut -d'.' -f1)
+QT_VER=$(shell grep QT_VERSION_STR $(MK_QTDIR)/include/qt/qglobal.h | cut -d'"' -f2 | cut -d'.' -f1)
 endif
 
 ifeq ($(QT_VER),0)
@@ -136,7 +136,7 @@
 DEFINES = -Dlinux -DFGLRX_USE_XEXTENSIONS -DFGLRX_OGL_INFO -DQT_NO_STL
 
 # for *.c
-CINCDIR = -I/usr/X11R6/include -I. -I./include_dummy
+CINCDIR = -I/usr/X11R6/include -I. -I./include_dummy -I/usr/include/qt
 CDEFINES = $(DEFINES)
 ifeq ($(DEBUG),0)
 CDEBFLAGS = -g
@@ -145,7 +145,7 @@
 LFLAGS = $(CDEBFLAGS) -L/usr/X11R6/$(LIB_DIR) -lXaw -lXmu -lXt -lXext -lX11
 
 # for *.cpp
-CCINCDIR = -I$(MK_QTDIR)/include -I. -I./include_dummy
+CCINCDIR = -I$(MK_QTDIR)/include -I. -I./include_dummy -I/usr/include/qt
 CCDEFINES = $(DEFINES)
 ifneq ($(DEBUG),0)
 CCDEBFLAGS = -ggdb
@@ -405,16 +405,6 @@
 	cd /usr/X11R6/bin/. ; rm -f $(BASENAME) ; ln -s $(TARGET0) $(BASENAME)
 	cp LICENSE.GPL /usr/X11R6/bin/.
 	cp LICENSE.QPL /usr/X11R6/bin/.
-	if  [ $KDEDIR ]; then\
-	  cp fireglcontrol.kdelnk $(KDEDIR)/share/applnk/. ;\
-	  cp ati.xpm $(KDEDIR)/share/icons/. ;\
-	fi
-	if test -d /usr/share/gnome/apps; then \
-	  cp fireglcontrol.desktop /usr/share/gnome/apps/. ;\
-	  if test -d /usr/share/pixmaps; then \
-	    cp ati.xpm /usr/share/pixmaps/. ;\
-	  fi ;\
-	fi
 
 source_package::
 	@if [ -f $(PACKET_NAME).tar ]; then      \
