diff -Naur minbif-1.0.1.orig/plugins/gayattitude/ga_message.c minbif-1.0.1/plugins/gayattitude/ga_message.c
--- minbif-1.0.1.orig/plugins/gayattitude/ga_message.c	2010-01-16 19:52:51.000000000 +0100
+++ minbif-1.0.1/plugins/gayattitude/ga_message.c	2010-03-18 18:23:06.439911513 +0100
@@ -208,7 +208,7 @@
 			message_content = g_parsing_quick_xpath_node_content(xpathCtx, "./td[4]/a", NULL, message_node);
 
 			/* check if ID is valid */
-			guint64 message_id;
+			guint64 message_id = 0;
 			if (message_idstr)
 			{
 				if (g_str_has_prefix(message_idstr, "msg"))
