diff -Nur libXrandr-1.2.0/include/X11/extensions/Xrandr.h libXrandr-1.2.0-fix/include/X11/extensions/Xrandr.h
--- libXrandr-1.2.0/include/X11/extensions/Xrandr.h	2006-12-13 04:24:25.000000000 -0500
+++ libXrandr-1.2.0-fix/include/X11/extensions/Xrandr.h	2007-02-20 15:24:11.000000000 -0500
@@ -296,7 +296,7 @@
 int
 XRRGetOutputProperty (Display *dpy, RROutput output,
 		      Atom property, long offset, long length,
-		      Bool delete, Bool pending, Atom req_type, 
+		      Bool del, Bool pending, Atom req_type, 
 		      Atom *actual_type, int *actual_format,
 		      unsigned long *nitems, unsigned long *bytes_after,
 		      unsigned char **prop);
diff -Nur libXrandr-1.2.0/src/XrrProperty.c libXrandr-1.2.0-fix/src/XrrProperty.c
--- libXrandr-1.2.0/src/XrrProperty.c	2007-01-31 17:23:05.000000000 -0500
+++ libXrandr-1.2.0-fix/src/XrrProperty.c	2007-02-20 15:23:41.000000000 -0500
@@ -250,7 +250,7 @@
 int
 XRRGetOutputProperty (Display *dpy, RROutput output,
 		      Atom property, long offset, long length,
-		      Bool delete, Bool pending, Atom req_type, 
+		      Bool del, Bool pending, Atom req_type, 
 		      Atom *actual_type, int *actual_format,
 		      unsigned long *nitems, unsigned long *bytes_after,
 		      unsigned char **prop)
@@ -273,7 +273,7 @@
     req->type = req_type;
     req->longOffset = offset;
     req->longLength = length;
-    req->delete = delete;
+    req->delete = del;
     req->pending = pending;
 
     if (!_XReply (dpy, (xReply *) &rep, 0, xFalse))

