From fdd6e05f278d33f1eb447a43ec817ab220ed6ab1 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Tue, 4 Aug 2015 22:37:18 +0200
Subject: [PATCH 1/2] fix incompatibility with glibc >= 2.20

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 566ef09..f493eef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c11")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
 
-add_definitions("-D_BSD_SOURCE -D_XOPEN_SOURCE=600")
+add_definitions("-D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600")
 
 set(CMAKE_SHARED_MODULE_PREFIX "")
 
-- 
2.4.1

