From d8108d1a9eda47c1a17ec2209571b04696614d00 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Tue, 4 Aug 2015 22:38:02 +0200
Subject: [PATCH 2/2] fix inlining error

---
 cmake/compilerwarnings.cmake | 2 --
 1 file changed, 2 deletions(-)

diff --git a/cmake/compilerwarnings.cmake b/cmake/compilerwarnings.cmake
index 7f6f935..fd20957 100644
--- a/cmake/compilerwarnings.cmake
+++ b/cmake/compilerwarnings.cmake
@@ -12,7 +12,6 @@ set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wlogical-op" )
 set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wmissing-include-dirs" )
 set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wno-write-strings" )
 set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wpointer-arith" )
-#set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wswitch-default" )
 set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wswitch-enum" )
 set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wundef" )
 set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wunused-macros" )
@@ -21,7 +20,6 @@ set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wshadow" )
 set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wpacked" )
 set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wredundant-decls" )
 set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Wunreachable-code" )
-set( WARNINGS_COMMONFLAGS "${WARNINGS_COMMONFLAGS} -Winline" )
 #
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${WARNINGS_COMMONFLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wstrict-prototypes")
-- 
2.4.1

