diff -Naur qtscriptgenerator-src-0.1.0.orig/qtbindings/qtbindings.pro qtscriptgenerator-src-0.1.0/qtbindings/qtbindings.pro
--- qtscriptgenerator-src-0.1.0.orig/qtbindings/qtbindings.pro	2009-02-20 12:42:24.000000000 +0100
+++ qtscriptgenerator-src-0.1.0/qtbindings/qtbindings.pro	2009-06-05 16:46:29.000000000 +0200
@@ -6,7 +6,6 @@
 	  qtscript_sql \
 	  qtscript_svg \
 	  qtscript_xml \
-	  qtscript_phonon \
 	  qtscript_webkit \
 	  qtscript_xmlpatterns \
 	  qtscript_uitools \
