diff --git a/docs/tools/shooter.c b/docs/tools/shooter.c
--- a/docs/tools/shooter.c
+++ b/docs/tools/shooter.c
@@ -170,7 +170,7 @@
       y_orig = 0;
     }
 
-  if (x_orig + width > gdk_screen_get_width (gdk_screen_get_dfeault ()))
+  if (x_orig + width > gdk_screen_get_width (gdk_screen_get_default ()))
     width = gdk_screen_width () - x_orig;
 
   if (y_orig + height > gdk_screen_get_height (gdk_screen_get_default ()))
