--- configure.orig	2004-03-15 16:23:45.721623080 -0600
+++ configure	2004-03-15 16:35:08.518822128 -0600
@@ -9018,7 +9018,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpng -lm $LIBS"
+LIBS="-lpng -lz -lm $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9061,7 +9061,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_png_main" >&5
 echo "${ECHO_T}$ac_cv_lib_png_main" >&6
 if test $ac_cv_lib_png_main = yes; then
-  have_png=yes; PNG_LIBS="$PNG_LIBS -lpng"
+  have_png=yes; PNG_LIBS="$PNG_LIBS -lpng -lz -lm"
 else
   have_png=no
 fi
@@ -9087,7 +9087,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <png.h>
+#include <libpng12/png.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -9122,7 +9122,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <png.h>
+#include <libpng12/png.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
@@ -9213,7 +9213,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <png.h>
+#include <libpng12/png.h>
 		#ifdef PNG_SETJMP_SUPPORTED
 			yes
 		#endif
