*** zoo-2.10/prterror.c	Fri Jul  5 18:00:00 1991
--- prterror.c	Fri Jan  9 15:40:54 2004
***************
*** 17,37 ****
  #endif
  #include "various.h"
  #include "zooio.h"
! #include "zoofns.h"
  
  #ifdef NEEDCTYP
  # include <ctype.h>	/* for isdigit() */
  #endif
  
- #ifdef STDARG
  # include <stdarg.h>
- #else
- # ifdef VARARGS
- #  include <varargs.h>
- # else
- #  include "MUST DEFINE STDARG OR VARARGS"
- # endif
- #endif
  
  #ifdef NEED_VPRINTF
  static int zvfprintf();
--- 17,29 ----
  #endif
  #include "various.h"
  #include "zooio.h"
! /*#include "zoofns.h"*/
  
  #ifdef NEEDCTYP
  # include <ctype.h>	/* for isdigit() */
  #endif
  
  # include <stdarg.h>
  
  #ifdef NEED_VPRINTF
  static int zvfprintf();
***************
*** 115,142 ****
  char could_not_open[] = "Could not open %s.\n";
  #endif
  
- #ifdef STDARG
  void prterror(int level, char *format, ...)
- #else
- /*VARARGS*/
- void prterror(va_alist)
- va_dcl
- #endif
  {
- 	va_list args;
     char string[120];       /* local format string */
- #ifdef VARARGS
- 	int level;
- 	char *format;
- #endif
  
- #ifdef STDARG
- 	va_start(args, format);
- #else
- 	va_start(args);
- 	level = va_arg(args, int);
- 	format = va_arg(args, char *);
- #endif
  
     *string = '\0';         /* get a null string to begin with */
  
--- 107,116 ----
***************
*** 163,173 ****
     strcat (string, format);      /* just append supplied format string */
  
  	/* and print the whole thing */
- #ifdef NEED_VPRINTF
- 	(void) zvfprintf(stdout, string, args);
- #else
-    (void) vprintf(string, args);
- #endif
  	fflush (stdout);
  
     if (level == 'f')       /* and abort on fatal error 'f' but not 'F' */
--- 137,142 ----
