--- LinNeighborhood-0.6.5_org/src/smbmount.c	2002-06-09 00:58:43.000000000 -0700
+++ LinNeighborhood-0.6.5/src/smbmount.c	2003-12-07 15:11:06.000000000 -0800
@@ -390,10 +390,14 @@
          escape_str(smbuser,&strlist),escape_str(smbpasswd,&strlist));
         break;
        case SMBMOUNT_206:
-        sprintf(commandstr,"%s %s %s -o username=%s%%%s,fmask=%d,dmask=%d,uid=%d,gid=%d,ip=%s,debug=0%s",
+//        sprintf(commandstr,"%s %s %s -o username=%s%%%s,fmask=%d,dmask=%d,uid=%d,gid=%d,ip=%s,debug=0%s",
+//         escape_str(pref_get_smbmount_exe(),&strlist),escape_str(servicename,&strlist),escape_str(mount_point,&strlist),
+//         escape_str(smbuser,&strlist),escape_str(smbpasswd,&strlist),
+//         filemode,dirmode,uid,gid,ipaddr,opt_group);
+        sprintf(commandstr,"%s %s %s -o username=%s%%%s,fmask=%d,dmask=%d,uid=%d,gid=%d,debug=0%s",
          escape_str(pref_get_smbmount_exe(),&strlist),escape_str(servicename,&strlist),escape_str(mount_point,&strlist),
          escape_str(smbuser,&strlist),escape_str(smbpasswd,&strlist),
-         filemode,dirmode,uid,gid,ipaddr,opt_group);
+         filemode,dirmode,uid,gid,opt_group);
          if ( (port >= 0) && (port < 65536) )
          {
           sprintf(commandstr, "%s,port=%d", commandstr, port);
@@ -485,8 +489,10 @@
       {
         argv[++argnr]="-o";
         argv[++argnr]=commandstr;
-        sprintf(commandstr,"username=%s%%%s,fmask=%d,dmask=%d,uid=%d,gid=%d,ip=%s,debug=0",
-           smbuser,smbpasswd,filemode,dirmode,uid,gid,ipaddr);
+//        sprintf(commandstr,"username=%s%%%s,fmask=%d,dmask=%d,uid=%d,gid=%d,ip=%s,debug=0",
+//           smbuser,smbpasswd,filemode,dirmode,uid,gid,ipaddr);
+        sprintf(commandstr,"username=%s%%%s,fmask=%d,dmask=%d,uid=%d,gid=%d,debug=0",
+           smbuser,smbpasswd,filemode,dirmode,uid,gid);
         if (*group_name)
         {
           strcat(commandstr,",workgroup=");
