*** pflogsumm.pl	2011-04-25 12:10:15.189036001 +0400
--- pflogsumm.pl.metalog	2011-04-25 12:10:04.633036001 +0400
***************
*** 539,544 ****
--- 539,549 ----
  while(<>) {
      next if(defined($dateStr) && ! /^$dateStr/o);
      s/: \[ID \d+ [^\]]+\] /: /o;	# lose "[ID nnnnnn some.thing]" stuff
+ 	#------------------------------------------------------------------------
+     # metalog hack by David B. Ferguson <david@linuxnet.ca> 08/19/2008
+     # see http://linuxnet.ca/postfix/pflogsumm_metalog.html for latest
+     s/\[(postfix|$syslogName)\/(.*)\] /metalog $1\/$2\[54321\]: /;
+     #------------------------------------------------------------------------
      my $logRmdr;
      next unless((($msgMonStr, $msgDay, $msgHr, $msgMin, $msgSec, $logRmdr) =
  	/^(...) +(\d+) (..):(..):(..) \S+ (.+)$/o) == 6);
