--- Makefile.orig	2007-04-04 16:13:16.000000000 -0700
+++ Makefile.am	2007-04-04 16:14:13.000000000 -0700
@@ -12,11 +12,13 @@
 diffimg_SOURCES = diffimg.c
 
 if WITH_LIBGD
+if WITH_MYLIBGD
 diffimg_LDADD = $(top_builddir)/lib/gd/libgvgd_C.la \
 	@FC_LIBS@ @FT_LIBS@ @JPEG_LIBS@ @PNG_LIBS@ @Z_LIBS@ @MATH_LIBS@
 else
 diffimg_LDADD = @GD_LIBS@
 endif
+endif
 
 GRAPH = "digraph G { hello -> world }"
 
