From: Ismael Luceno <ismael@sourcemage.org>
Subject: Replace non-standard canonicalize_file_name by realpath

--- a/src/frontend/standalone/main.c
+++ b/src/frontend/standalone/main.c
@@ -227,7 +227,7 @@ static void check_target (const char *target, bool noexec)
   else
     log_text (DI_LOG_LEVEL_ERROR, "Target check failed: %s", strerror (errno));

-  target_root = canonicalize_file_name(target);
+  target_root = realpath(target, NULL);
   if (!target_root)
     log_text (DI_LOG_LEVEL_ERROR, "Target check failed: %s", strerror (errno));
 }
