From 7a6405f6387c9005e4cfaa6c008ac3784e2ed77d Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sun, 16 Oct 2016 22:55:27 +0200
Subject: [PATCH] fix compile error with gcc 6

---
 libutil++/op_bfd.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libutil++/op_bfd.cpp b/libutil++/op_bfd.cpp
index 389c920..ba57552 100644
--- a/libutil++/op_bfd.cpp
+++ b/libutil++/op_bfd.cpp
@@ -456,8 +456,8 @@ get_symbol_contents(symbol_index_t sym_index, unsigned char * contents) const
 	op_bfd_symbol const & bfd_sym = syms[sym_index];
 	size_t size = bfd_sym.size();
 
-	if (!bfd_get_section_contents(ibfd.abfd, bfd_sym.symbol()->section, 
-				 contents, 
+	if (!bfd_get_section_contents(ibfd.abfd, bfd_sym.symbol()->section,
+				 contents,
 				 static_cast<file_ptr>(bfd_sym.value()), size)) {
 		return false;
 	}
@@ -535,7 +535,7 @@ void op_bfd::get_symbol_range(symbol_index_t sym_idx,
 {
 	op_bfd_symbol const & sym = syms[sym_idx];
 
-	bool const verbose = cverb << (vbfd & vlevel1);
+	bool const verbose = static_cast<bool>(cverb << (vbfd & vlevel1));
 
 	if (anon_obj)
 		start = sym.vma();
-- 
2.10.1

