--- oprofile-0.9.3.orig/libutil++/bfd_spu_support.cpp	2008-06-06 23:09:59.514434431 +0200
+++ oprofile-0.9.3/libutil++/bfd_spu_support.cpp	2008-06-06 23:30:55.628196519 +0200
@@ -20,6 +20,7 @@
 #include <sstream>
 #include <string>
 #include <sys/types.h>
+#include <cstring>
 
 struct spu_elf {
 	FILE * stream;
--- oprofile-0.9.3.orig/libutil++/bfd_support.cpp	2008-06-06 23:09:59.488601126 +0200
+++ oprofile-0.9.3/libutil++/bfd_support.cpp	2008-06-06 23:11:28.071092331 +0200
@@ -19,6 +19,8 @@
 #include <fstream>
 #include <sstream>
 #include <string>
+#include <cstdlib>
+#include <cstring>
 
 using namespace std;
 
--- oprofile-0.9.3.orig/libutil++/child_reader.cpp	2008-06-06 23:09:59.521100724 +0200
+++ oprofile-0.9.3/libutil++/child_reader.cpp	2008-06-06 23:29:04.014867476 +0200
@@ -15,6 +15,9 @@
 #include <cerrno>
 #include <sstream>
 #include <iostream>
+#include <cstring>
+#include <cstdlib>
+#include <climits>
 
 #include "op_libiberty.h"
 #include "child_reader.h"
--- oprofile-0.9.3.orig/pp/common_option.cpp	2008-06-06 23:09:59.441100877 +0200
+++ oprofile-0.9.3/pp/common_option.cpp	2008-06-06 23:48:48.007560937 +0200
@@ -12,6 +12,7 @@
 #include <iostream>
 #include <sstream>
 #include <iterator>
+#include <cstdlib>
 
 #include "op_config.h"
 #include "locate_images.h"
--- oprofile-0.9.3.orig/libutil++/cverb.cpp	2008-06-06 23:09:59.521100724 +0200
+++ oprofile-0.9.3/libutil++/cverb.cpp	2008-06-06 23:20:06.474363598 +0200
@@ -13,6 +13,7 @@
 #include <iostream>
 #include <map>
 #include <string>
+#include <cstring>
 
 #include "cverb.h"
 
--- oprofile-0.9.3.orig/libregex/demangle_symbol.cpp	2008-06-06 23:09:59.568173962 +0200
+++ oprofile-0.9.3/libregex/demangle_symbol.cpp	2008-06-06 23:38:01.551649174 +0200
@@ -8,6 +8,8 @@
  * @author John Levon
  */
 
+#include <cstdlib>
+
 #include "config.h"
 
 #include "demangle_symbol.h"
--- oprofile-0.9.3.orig/libutil++/file_manip.cpp	2008-06-06 23:09:59.504434369 +0200
+++ oprofile-0.9.3/libutil++/file_manip.cpp	2008-06-06 23:18:05.478523157 +0200
@@ -21,6 +21,8 @@
 #include <iostream>
 #include <fstream>
 #include <vector>
+#include <cstdlib>
+#include <climits>
 
 #include "op_file.h"
 
--- oprofile-0.9.3.orig/pp/opannotate_options.cpp	2008-06-06 23:09:59.404460347 +0200
+++ oprofile-0.9.3/pp/opannotate_options.cpp	2008-06-06 23:50:07.304288393 +0200
@@ -13,6 +13,7 @@
 #include <list>
 #include <iterator>
 #include <iostream>
+#include <cstdlib>
 
 #include "profile_spec.h"
 #include "arrange_profiles.h"
--- oprofile-0.9.3.orig/pp/oparchive.cpp	2008-06-06 23:09:59.441100877 +0200
+++ oprofile-0.9.3/pp/oparchive.cpp	2008-06-06 23:51:02.148223620 +0200
@@ -15,6 +15,7 @@
 
 #include <errno.h>
 #include <string.h>
+#include <cstdlib>
 
 #include <sys/types.h>
 #include <sys/stat.h>
--- oprofile-0.9.3.orig/pp/opgprof_options.cpp	2008-06-06 23:09:59.441100877 +0200
+++ oprofile-0.9.3/pp/opgprof_options.cpp	2008-06-06 23:51:59.058101869 +0200
@@ -13,6 +13,7 @@
 #include <list>
 #include <iterator>
 #include <iostream>
+#include <cstdlib>
 
 #include "opgprof_options.h"
 #include "popt_options.h"
--- oprofile-0.9.3.orig/libpp/op_header.cpp	2008-06-06 23:09:59.478178968 +0200
+++ oprofile-0.9.3/libpp/op_header.cpp	2008-06-06 23:40:26.434644777 +0200
@@ -14,6 +14,7 @@
 #include <iomanip>
 #include <set>
 #include <sstream>
+#include <cstring>
 
 #include "op_exception.h"
 #include "odb.h"
--- oprofile-0.9.3.orig/libabi/opimport.cpp	2008-06-06 23:09:59.564851229 +0200
+++ oprofile-0.9.3/libabi/opimport.cpp	2008-06-06 23:34:56.891631173 +0200
@@ -17,6 +17,8 @@
 #include <iostream>
 #include <vector>
 #include <cassert>
+#include <cstring>
+#include <cstdlib>
 
 #include <sys/types.h>
 #include <sys/stat.h>
--- oprofile-0.9.3.orig/gui/oprof_start_util.cpp	2008-06-06 23:09:59.474434501 +0200
+++ oprofile-0.9.3/gui/oprof_start_util.cpp	2008-06-06 23:55:37.104181832 +0200
@@ -19,6 +19,7 @@
 #include <sstream>
 #include <iostream>
 #include <fstream>
+#include <cstdlib>
 
 #include <qfiledialog.h>
 #include <qmessagebox.h>
--- oprofile-0.9.3.orig/libutil++/op_spu_bfd.cpp	2008-06-06 23:09:59.521100724 +0200
+++ oprofile-0.9.3/libutil++/op_spu_bfd.cpp	2008-06-06 23:32:21.870988592 +0200
@@ -14,6 +14,8 @@
 #include <sys/stat.h>
 
 #include <iostream>
+#include <cstdlib>
+#include <cstring>
 
 #include "op_bfd.h"
 #include "op_libiberty.h"
--- oprofile-0.9.3.orig/libpp/profile.cpp	2008-06-06 23:09:59.478178968 +0200
+++ oprofile-0.9.3/libpp/profile.cpp	2008-06-06 23:42:54.344226333 +0200
@@ -17,6 +17,7 @@
 #include <sstream>
 
 #include <cerrno>
+#include <cstring>
 
 #include "op_exception.h"
 #include "op_header.h"
--- oprofile-0.9.3.orig/libpp/sample_container.cpp	2008-06-06 23:09:59.481507620 +0200
+++ oprofile-0.9.3/libpp/sample_container.cpp	2008-06-06 23:44:28.834233072 +0200
@@ -13,6 +13,7 @@
 #include <numeric>
 #include <algorithm>
 #include <vector>
+#include <climits>
 
 #include "sample_container.h"
 
