From 462ee2b6bd77163404c5afc28e268b183a604756 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sat, 24 Jan 2015 12:48:36 +0100
Subject: [PATCH 2/8] use system libtommath

---
 CMakeLists.txt     | 5 -----
 src/CMakeLists.txt | 4 ++--
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86fed29d4e..1e25ff673b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -322,7 +322,6 @@ endif() # if (NOT CMAKE_CROSSCOMPILING)
 
 crosscompile_prebuild_steps()
 
-include_directories("extern/libtommath")
 include_directories("extern/icu/include")
 include_directories("extern/zlib")
 
@@ -335,10 +334,6 @@ include_directories("${CMAKE_CURRENT_BINARY_DIR}/src/include/gen")
 # LIBRARY libtommath
 ########################################
 
-file(GLOB libtommath_src "extern/libtommath/*.c" "extern/libtommath/*.h")
-
-add_library             (libtommath ${libtommath_src})
-project_group           (libtommath Extern)
 
 ########################################
 # EXECUTABLE btyacc
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 395965a69d..71ae606704 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -412,7 +412,7 @@ add_library                 (boot_yvalve SHARED $<TARGET_OBJECTS:yvalve_common>
 target_link_libraries       (boot_yvalve
     PRIVATE remote
     PRIVATE common
-    PUBLIC libtommath
+    PUBLIC tommath
     PUBLIC ${LIB_Ws2_32}
     PUBLIC ${LIB_readline}
     PUBLIC ${LIB_dl}
@@ -435,7 +435,7 @@ add_library                 (yvalve SHARED $<TARGET_OBJECTS:yvalve_common> ${yva
 target_link_libraries       (yvalve
     PRIVATE remote
     PRIVATE common
-    PUBLIC libtommath
+    PUBLIC tommath
     PUBLIC ${LIB_Ws2_32}
     PUBLIC ${LIB_readline}
     PUBLIC ${LIB_dl}
-- 
2.25.2

