From abc15abbeb2c083f5741c7ba4b0b761f3facecfe Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sat, 11 Feb 2017 14:56:02 +0100
Subject: [PATCH 3/6] suppress some includes

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e607732..47b0fa9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -174,7 +174,7 @@ configure_file(${AUTOCONFIG_SRC} ${AUTOCONFIG} @ONLY)
 #
 ################################################################################
 
-add_definitions(-DDEV_BUILD)
+#add_definitions(-DDEV_BUILD)
 
 if (WIN32)
     set(OS_DIR win32)
@@ -321,8 +321,8 @@ endif() # if (NOT CMAKE_CROSSCOMPILING)
 
 crosscompile_prebuild_steps()
 
-include_directories("extern/icu/include")
-include_directories("extern/zlib")
+#include_directories("extern/icu/include")
+#include_directories("extern/zlib")
 
 include_directories("src/include")
 include_directories("src/include/gen")
-- 
2.16.0

