From b2c773371ed786dd7ce8b3ed3ba2d30252b10305 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sat, 11 Feb 2017 14:56:02 +0100
Subject: [PATCH 3/9] suppress some includes

---
 CMakeLists.txt | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e25ff673b..0cdc0d10ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -174,8 +174,6 @@ configure_file(${AUTOCONFIG_SRC} ${AUTOCONFIG} @ONLY)
 #
 ################################################################################
 
-add_definitions(-DDEV_BUILD)
-
 if (WIN32)
     set(OS_DIR win32)
     set(VERSION_RC ${CMAKE_SOURCE_DIR}/src/jrd/version.rc)
@@ -322,12 +320,9 @@ endif() # if (NOT CMAKE_CROSSCOMPILING)
 
 crosscompile_prebuild_steps()
 
-include_directories("extern/icu/include")
-include_directories("extern/zlib")
-
 include_directories("src/include")
-include_directories("src/include/gen")
-include_directories("${CMAKE_CURRENT_BINARY_DIR}/src/include")
+include_directories("src/include/gen") 
+include_directories("${CMAKE_CURRENT_BINARY_DIR}/src/include") 
 include_directories("${CMAKE_CURRENT_BINARY_DIR}/src/include/gen")
 
 ########################################
-- 
2.28.0

