From 72ab3f3c38544459ee1f76300a83850f979e3676 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Mon, 13 Jun 2016 20:44:30 +0200
Subject: [PATCH 3/4] pcmet.epp no longer exists

---
 src/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 583c4c8..1344d1b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -54,7 +54,6 @@ set(epp_boot_gds_files
     jrd/grant.epp
     jrd/ini.epp
     jrd/met.epp
-    jrd/pcmet.epp
     jrd/scl.epp
     jrd/Function.epp
 )
@@ -484,7 +483,6 @@ set(engine12_generated_src
     jrd/grant.epp
     jrd/ini.epp
     jrd/met.epp
-    jrd/pcmet.epp
     jrd/scl.epp
     utilities/gstat/dba.epp
 )
-- 
2.12.1

