From 0192ecf2d8d09517036bcf024c95477039d70e65 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Tue, 3 Dec 2019 10:29:55 +0100
Subject: [PATCH 5/8] fix build with cmake-3.16

---
 src/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 144f6003d2..3c8641b743 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -266,7 +266,7 @@ add_custom_target(parse
     SOURCES ${parse_src}
 )
 project_group(parse "Boot/Custom build steps")
-set_source_files_properties(dsql/parse.cpp include/gen/parse.h PROPERTIES GENERATED TRUE)
+set_source_files_properties(dsql/parse.cpp  ${CMAKE_CURRENT_BINARY_DIR}/include/gen/parse.h PROPERTIES GENERATED TRUE)
 
 
 ########################################
@@ -357,7 +357,7 @@ add_library                 (burp ${burp_src} ${burp_generated_src_master})
 
 
 ########################################
-# LIBRARY common
+# LIBRARY Wcommon
 ########################################
 
 file(GLOB common_src "common/*.cpp" "common/classes/*.cpp" "common/config/*.cpp" "common/os/${OS_DIR}/*.cpp" "common/sha2/*.cpp")
@@ -371,7 +371,7 @@ endif()
 add_library                 (common ${common_src} ${common_os_src} ${common_include})
 set_source_files_properties(common/CRC32C.cpp PROPERTIES COMPILE_FLAGS -msse4)
 target_link_libraries       (common ${LIB_mpr})
-add_dependencies_cc         (common UpdateCloopInterfaces)
+add_dependencies_cc         (common UpdateCloopInterfaces parse)
 
 
 ################################################################################
-- 
2.25.2

