From 3333315b97704d0935ea2c513dd2a683fb473485 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sat, 1 Apr 2017 13:11:35 +0200
Subject: [PATCH 5/5] cmake fix for non sse4.2 machine

---
 CMakeLists.txt     | 2 +-
 src/CMakeLists.txt | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3be6969..73020c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.6)
 
 # In-source builds are not possible and so disabled.
 if (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1344d1b..6bc51de 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -369,6 +369,7 @@ if (APPLE)
 endif()
 
 add_library                 (common ${common_src} ${common_os_src} ${common_include})
+set_source_files_properties(common/CRC32C.cpp PROPERTIES COMPILE_FLAGS -msse4)
 target_link_libraries       (common ${LIB_mpr})
 add_dependencies_cc         (common UpdateCloopInterfaces)
 
-- 
2.12.2

