From c45f2a703178b5a49057e9404d7d21851545cfac Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sun, 7 Jan 2018 12:05:27 +0100
Subject: [PATCH 5/6] fb3client

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6bc51de..01fa3e4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -446,7 +446,7 @@ target_link_libraries       (yvalve
 add_dependencies_cc         (yvalve databases)
 set_exported_symbols        (yvalve firebird)
 set_output_directory_unix   (yvalve lib)
-set_target_properties       (yvalve PROPERTIES OUTPUT_NAME fbclient)
+set_target_properties       (yvalve PROPERTIES OUTPUT_NAME fb3client)
 
 
 ########################################
-- 
2.16.0

