From 90315362b464855c42602e0ee6a5c5def57214e0 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Thu, 18 Jan 2018 21:24:49 +0100
Subject: [PATCH 5/6] make pthread usage explicit

---
 src/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 144f6003d2..c5a0181cf4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -437,6 +437,7 @@ target_link_libraries       (yvalve
     PRIVATE remote
     PRIVATE common
     PUBLIC tommath
+    PUBLIC pthread
     PUBLIC ${LIB_Ws2_32}
     PUBLIC ${LIB_readline}
     PUBLIC ${LIB_dl}
-- 
2.24.0

