From 96a619f0cd33ebee530f0574a8f255fc2a00d503 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Thu, 18 Jan 2018 21:24:49 +0100
Subject: [PATCH 6/6] make pthread usage explicit

---
 src/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 01fa3e4..fa91d62 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -437,6 +437,7 @@ target_link_libraries       (yvalve
     PRIVATE remote
     PRIVATE common
     PUBLIC tommath
+    PUBLIC pthread
     PUBLIC ${LIB_Ws2_32}
     PUBLIC ${LIB_readline}
     PUBLIC ${LIB_dl}
-- 
2.16.0

