From 3d8984e5a90098bca674d3620fbd73664310d246 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Tue, 11 Feb 2020 16:06:58 +0100
Subject: [PATCH 7/9] cmake fix for glibc-2.31

---
 builds/cmake/Configure.cmake | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/builds/cmake/Configure.cmake b/builds/cmake/Configure.cmake
index 81a057583f..02e8e501a1 100644
--- a/builds/cmake/Configure.cmake
+++ b/builds/cmake/Configure.cmake
@@ -287,10 +287,11 @@ set(CMAKE_EXTRA_INCLUDE_FILES Windows.h)
 check_type_size("char[MAX_PATH]" MAXPATHLEN)
 set(CMAKE_EXTRA_INCLUDE_FILES)
 
-set(TIMEZONE_TYPE "struct timezone")
-if (APPLE OR MINGW)
+#set(TIMEZONE_TYPE "struct timezone")
+#if (APPLE OR MINGW)
+ # fix for glibc-2.31
     set(TIMEZONE_TYPE "void")
-endif()
+#endif()
 check_prototype_definition(
     gettimeofday
     "int gettimeofday(struct timeval *tv, ${TIMEZONE_TYPE} *tz)"
-- 
2.28.0

