From 2698ac9ad9f669dbf18cd64c04a08a8018d34477 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sun, 22 Mar 2020 12:03:18 +0100
Subject: [PATCH 8/8] tweak libedit usage

---
 src/CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d289e37ef9..5aa5514e07 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -416,7 +416,6 @@ target_link_libraries       (boot_yvalve
     PRIVATE common
     PUBLIC tommath
     PUBLIC ${LIB_Ws2_32}
-    PUBLIC ${LIB_readline}
     PUBLIC ${LIB_dl}
     PUBLIC ${LIB_iconv}
     PUBLIC ${LIB_CoreFoundation}
@@ -439,7 +438,6 @@ target_link_libraries       (yvalve
     PRIVATE common
     PUBLIC tommath
     PUBLIC ${LIB_Ws2_32}
-    PUBLIC ${LIB_readline}
     PUBLIC ${LIB_dl}
     PUBLIC ${LIB_iconv}
     PUBLIC ${LIB_CoreFoundation}
@@ -846,7 +844,7 @@ add_epp_suffix(isql_generated_src master)
 if (NOT CMAKE_CROSSCOMPILING)
 
 add_executable              (boot_isql ${isql_src} ${isql_generated_src_boot} ${VERSION_RC})
-target_link_libraries       (boot_isql common boot_yvalve)
+target_link_libraries       (boot_isql common boot_yvalve ${LIB_readline} )
 set_output_directory        (boot_isql . CURRENT_DIR)
 project_group               (boot_isql Boot)
 
@@ -897,7 +895,7 @@ endif()
 ########################################
 
 add_executable              (isql ${isql_src} ${isql_generated_src_master} ${VERSION_RC})
-target_link_libraries       (isql common yvalve)
+target_link_libraries       (isql common yvalve ${LIB_readline} )
 
 
 ########################################
-- 
2.25.2

