From 54b256aad5b6f7f16407e68f0d9c621b557bc0a4 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Fri, 7 Aug 2020 12:25:06 +0200
Subject: [PATCH 9/9] btyacc - fix mktemp usage

---
 extern/btyacc/main.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/extern/btyacc/main.c b/extern/btyacc/main.c
index c87c030b52..8d1b22d2b3 100644
--- a/extern/btyacc/main.c
+++ b/extern/btyacc/main.c
@@ -341,15 +341,15 @@ void create_file_names()
 	text_file_name[len + 5] = 't';
 	union_file_name[len + 5] = 'u';
 
-	if(mktemp(action_file_name) == NULL) {
+	if(mkstemp(action_file_name) == -1) {
 		fprintf(stderr, "btyacc: Cannot create temporary file\n");
 		exit(1);
 	}
-	if(mktemp(text_file_name) == NULL) {
+	if(mkstemp(text_file_name) == -1) {
 		fprintf(stderr, "btyacc: Cannot create temporary file\n");
 		exit(1);
 	}
-	if(mktemp(union_file_name) == NULL) {
+	if(mkstemp(union_file_name) == -1) {
 		fprintf(stderr, "btyacc: Cannot create temporary file\n");
 		exit(1);
 	}
-- 
2.28.0

