--- libcloud/compute/drivers/ec2.py.orig	2016-12-27 03:27:54.000000000 -0500
+++ libcloud/compute/drivers/ec2.py	2017-04-14 16:04:28.695062413 -0400
@@ -793,7 +793,7 @@
         'endpoint': 'ec2.us-east-2.amazonaws.com',
         'api_name': 'ec2_us_east_ohio',
         'country': 'USA',
-        'signature_version': '2',
+        'signature_version': '4',
         'instance_types': [
             't1.micro',
             'm1.small',
