--- timidity/jack_a.c.orig	2015-04-22 20:48:41.612316567 +0200
+++ timidity/jack_a.c	2015-04-22 20:50:19.767089464 +0200
@@ -508,6 +508,8 @@
 				pthread_cond_wait(&ctx->cond, &ctx->lock);
 		}
 		/* fallthrough */
+	/* Folks found that: https://bugs.archlinux.org/task/40906 */
+	case PM_REQ_PLAY_START: /* Please make -iA -Oj work! */
 	case PM_REQ_DISCARD:
 		ctx->running = 0;
 		ringbuf_clear(&ctx->rbuf);
