--- configure.in.in.orig	2008-08-22 02:35:26.000000000 +0500
+++ configure.in.in	2008-09-01 17:08:25.000000000 +0500
@@ -23,6 +23,7 @@
 AC_PROG_LIBTOOL()
 AC_PROG_INTLTOOL([0.31], [no-xml])
 
+AM_PROG_CC_C_O
 AC_SYS_LARGEFILE
 
 dnl check for i18n support
@@ -74,7 +75,7 @@
   CFLAGS="$save_CFLAGS"
 fi
 AC_SUBST([HAL_STORAGE_DEPENDENCY])
-
+AM_CONDITIONAL([HAVE_HAL], [test x"$HAL_FOUND" = x"yes"])
 
 dnl check for debugging support
 BM_DEBUG_SUPPORT()
