diff -uprN src.old/inputclient.h src/inputclient.h
--- src.old/inputclient.h	2003-09-20 21:41:59.219052200 -0700
+++ src/inputclient.h	2003-09-20 21:42:09.611472312 -0700
@@ -19,6 +19,7 @@
 #ifndef INPUTCLIENT_H
 #define INPUTCLIENT_H
 
+#include <assert.h>
 #include <string>
 #include "console.h"
 
diff -uprN src.old/mouse.cpp src/mouse.cpp
--- src.old/mouse.cpp	2003-09-20 21:41:59.220052048 -0700
+++ src/mouse.cpp	2003-09-20 21:42:18.347144288 -0700
@@ -19,6 +19,7 @@
 #include "debug.h"
 #include "mouse.h"
 #include <sys/signal.h>
+#include <assert.h>
 
 #if defined(__FreeBSD__)
 Mouse* Mouse::pMouse = NULL;
