diff -Naur /tmp/zhcon-0.2.1/src/zhcon.cpp src/zhcon.cpp
--- /tmp/zhcon-0.2.1/src/zhcon.cpp	2002-06-10 10:29:36.000000000 +0800
+++ src/zhcon.cpp	2002-11-10 21:38:06.000000000 +0800
@@ -365,7 +365,8 @@
     }
     
     if (!(strncmp("/dev/tty", TtyName, 8) == 0 ||
-        strncmp("/dev/vc/", TtyName, 8) == 0)) {
+        strncmp("/dev/vc/", TtyName, 8) == 0||
+        strncmp("/devices/vc/", TtyName, 12) == 0)) {
         printf("%s is not real tty or vc. Please exit current tty and try again.\r\n", TtyName);
         exit(1);
     }
