diff -uNr akonadi-1.1.0/cmake/modules/FindBoost.cmake 
akonadi-1.1.0.cp/cmake/modules/FindBoost.cmake
--- akonadi-1.1.0/cmake/modules/FindBoost.cmake 2009-01-03 17:11:36.000000000 +0100
+++ akonadi-1.1.0.cp/cmake/modules/FindBoost.cmake      2009-01-05 12:22:57.000000000 +0100
@@ -93,7 +93,7 @@

 OPTION(Boost_USE_MULTITHREADED "Use the multithreaded versions of the boost libraries" ON)

-SET( _boost_TEST_VERSIONS ${Boost_ADDITIONAL_VERSIONS} "1.36.1" "1.36.0" "1.35.1" "1.35.0" 
"1.35" "1.34.1" "1.34.0" "1.34" "1.33.1" "1.33.0" "1.33" )
+SET( _boost_TEST_VERSIONS ${Boost_ADDITIONAL_VERSIONS} "1.37.0" "1.36.1" "1.36.0" "1.35.1" 
"1.35.0" "1.35" "1.34.1" "1.34.0" "1.34" "1.33.1" "1.33.0" "1.33" )


 ############################################

