diff -u bsdiff-4.3/Makefile bsdiff-mod/Makefile
--- bsdiff-4.3/Makefile	2005-08-16 17:14:10.000000000 -0500
+++ bsdiff-mod/Makefile	2012-04-08 11:11:29.552182360 -0500
@@ -1,8 +1,8 @@
 CFLAGS		+=	-O3 -lbz2
 
-PREFIX		?=	/usr/local
-INSTALL_PROGRAM	?=	${INSTALL} -c -s -m 555
-INSTALL_MAN	?=	${INSTALL} -c -m 444
+PREFIX		?=	/usr
+INSTALL_PROGRAM	?=	install -c -s -m 555
+INSTALL_MAN	?=	install -c -m 444
 
 all:		bsdiff bspatch
 bsdiff:		bsdiff.c
@@ -10,6 +10,4 @@
 
 install:
 	${INSTALL_PROGRAM} bsdiff bspatch ${PREFIX}/bin
-.ifndef WITHOUT_MAN
-	${INSTALL_MAN} bsdiff.1 bspatch.1 ${PREFIX}/man/man1
-.endif
+	${INSTALL_MAN} bsdiff.1 bspatch.1 ${PREFIX}/share/man/man1
