diff -Naur libtunepimp-0.5.3.orig/lib/fileio.cpp libtunepimp-0.5.3/lib/fileio.cpp
--- libtunepimp-0.5.3.orig/lib/fileio.cpp	2010-04-20 22:08:13.355562828 -0700
+++ libtunepimp-0.5.3/lib/fileio.cpp	2010-04-20 22:09:05.204812864 -0700
@@ -124,10 +124,10 @@
 
 void tmktempname(const char *path, char *newPath, int newPathLen)
 {
-    char *ptr, *temp;
+    char *temp;
 
     temp = (char *)malloc(strlen(path) + 32);
-    ptr = strrchr(path, dirSepChar);
+    const char *ptr = strrchr(path, dirSepChar);
     if (ptr)
     {
         int len = (int)(ptr - path);
