diff -urN libtunepimp-0.3.0.old/lib/threads/posix/thread.cpp libtunepimp-0.3.0/lib/threads/posix/thread.cpp
--- libtunepimp-0.3.0.old/lib/threads/posix/thread.cpp	2004-01-11 09:27:52.000000000 +0000
+++ libtunepimp-0.3.0/lib/threads/posix/thread.cpp	2004-10-16 00:40:38.874226012 +0000
@@ -28,6 +28,7 @@
 #include <errno.h>
 #include <string.h>
 #include <sched.h>                                                                                              
+#include <signal.h>
 #include "thread.h"
 
 #define DB printf("%s:%d\n", __FILE__, __LINE__);
