From 954a484c3305de380b38180b5dce326b8a2dddea Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Mon, 16 Oct 2023 14:14:38 +0200
Subject: [PATCH] unity fix

---
 src/abstractrunner_p.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/abstractrunner_p.h b/src/abstractrunner_p.h
index 75385f3..5512313 100644
--- a/src/abstractrunner_p.h
+++ b/src/abstractrunner_p.h
@@ -2,6 +2,7 @@
     SPDX-FileCopyrightText: 2020-2023 Alexander Lohnau <alexander.lohnau@gmx.de>
     SPDX-License-Identifier: LGPL-2.0-or-later
 */
+#pragma once
 #include "abstractrunner.h"
 #include "runnersyntax.h"
 #include <QReadWriteLock>
-- 
2.42.0

