diff -Naur CEGUI-0.6.0.orig/Samples/TreeDemo/src/Sample_TreeDemo.cpp CEGUI-0.6.0/Samples/TreeDemo/src/Sample_TreeDemo.cpp
--- CEGUI-0.6.0.orig/Samples/TreeDemo/src/Sample_TreeDemo.cpp	2008-04-26 17:13:50.668243450 -0700
+++ CEGUI-0.6.0/Samples/TreeDemo/src/Sample_TreeDemo.cpp	2008-04-26 17:14:38.688219245 -0700
@@ -13,6 +13,7 @@
 #endif
 
 #include <time.h>
+#include <cstdlib>
 
 
 const unsigned int TreeDemoSample::TreeID    = 1;
