--- Makefile~	2006-03-30 18:56:17.000000000 +0200
+++ Makefile	2006-03-30 20:53:15.000000000 +0200
@@ -492,11 +492,6 @@
 	done
 endif
 	@$(INSTALL) -d $(DATADIR)
-	@$(INSTALL) -d $(DATADIR)/font
-	@if test ! -f $(DATADIR)/font/font.desc ; then \
-	echo "*** Download font at http://www.mplayerhq.hu/dload.html" ; \
-	echo "*** for OSD/Subtitles support and extract to $(DATADIR)/font/" ; \
-	fi
 ifeq ($(GUI),yes)
 	@$(INSTALL) -d $(DATADIR)/skins
 	@echo "*** Download skin(s) at http://www.mplayerhq.hu/dload.html"
