Index: libmpdemux/muxer_lavf.c
===================================================================
--- libmpdemux/muxer_lavf.c	(revision 25134)
+++ libmpdemux/muxer_lavf.c	(working copy)
@@ -317,7 +317,7 @@
 		av_freep(&(priv->oc->streams[i]));
 	}
 
-	url_fclose(&(priv->oc->pb));
+	url_fclose((priv->oc->pb));
 
 	av_free(priv->oc);
 }
@@ -400,7 +400,7 @@
 		goto fail;
         }
 	
-	((URLContext*)(priv->oc->pb.opaque))->priv_data= muxer;
+	((URLContext*)(priv->oc->pb->opaque))->priv_data= muxer;
 	
 	muxer->priv = (void *) priv;
 	muxer->cont_new_stream = &lavf_new_stream;
