--- Makefile~	2006-06-11 20:35:47.000000000 +0200
+++ Makefile	2006-06-12 09:31:16.000000000 +0200
@@ -471,11 +471,6 @@
 	done
 endif
 	@$(INSTALL) -d $(DATADIR)
-	@$(INSTALL) -d $(DATADIR)/font
-	@if test ! -f $(DATADIR)/font/font.desc ; then \
-	echo "*** Download font at http://www.mplayerhq.hu/dload.html" ; \
-	echo "*** for OSD/Subtitles support and extract to $(DATADIR)/font/" ; \
-	fi
 ifeq ($(GUI),yes)
 	@$(INSTALL) -d $(DATADIR)/skins
 	@echo "*** Download skin(s) at http://www.mplayerhq.hu/dload.html"
