--- Makefile~	2003-10-05 00:00:01.000000000 +0200
+++ Makefile	2003-11-30 18:03:02.000000000 +0100
@@ -284,11 +284,6 @@
 	done
 endif
 	@if test ! -d $(DATADIR) ; then mkdir -p $(DATADIR) ; fi
-	@if test ! -d $(DATADIR)/font ; then mkdir -p $(DATADIR)/font ; fi
-	@if test ! -f $(DATADIR)/font/font.desc ; then \
-	echo "*** Download font at http://www.mplayerhq.hu/homepage/dload.html" ; \
-	echo "*** for OSD/Subtitles support and extract to $(DATADIR)/font/" ; \
-	fi
 ifeq ($(GUI),yes)
 	@if test ! -d $(DATADIR)/Skin ; then mkdir -p $(DATADIR)/Skin ; fi
 	@echo "*** Download skin(s) at http://www.mplayerhq.hu/homepage/dload.html"
