http://www.museek-plus.org/ticket/191
--- python-bindings/museek/messages.py.orig	2009-02-20 13:42:45.000000000 +0300
+++ python-bindings/museek/messages.py	2011-02-06 17:12:33.179281238 +0300
@@ -774,7 +774,7 @@
 class DownloadFile(BaseMessage):
 	code = 0x0503
 	
-	def __init__(self, user = None, path = None, size = None):
+	def __init__(self, user = None, path = None, size = 0):
 		self.user = user
 		self.path = path
 		self.size = size
@@ -837,7 +837,7 @@
 class DownloadFileTo(BaseMessage):
 	code = 0x0507
 	
-	def __init__(self, user = None, path = None, dpath = None, size = None):
+	def __init__(self, user = None, path = None, dpath = None, size = 0):
 		self.user = user
 		self.path = path
 		self.dpath = dpath
