--- logitech_applet-0.4test1/logitech_applet.c.bak	2005-05-03 08:04:38.000000000 +0800
+++ logitech_applet-0.4test1/logitech_applet.c	2005-05-03 08:44:35.000000000 +0800
@@ -398,7 +398,7 @@
 				    break;
 				case 1: printf("Enabled\n");
 				    break;
-				default: printf("Unknown (Unexpected result)\n");
+				default: printf("Unknown (Unexpected result: %i)\n", cruise);
 				    break;
 				}
 			} 
