commit 8ef983574c4ce929879db3c759933a7ec23d63ab
Author: Treeve Jelbert <treeve@sourcemage.org>
Date:   Sun Aug 23 14:11:58 2020 +0200

    include stdio.h before readline.h

diff --git a/config/configure b/config/configure
index bbf5fb3..34ee442 100755
--- a/config/configure
+++ b/config/configure
@@ -192,12 +192,14 @@ if pkg-config readline; then
     # It looks like it's not uncommon for pkg-config to provide a -I
     # that doesn't support the documentation's specified #include
     # <readline/readline.h>.  See what's really going on.
-    if bup_try_c_code "#include <readline/readline.h> $readline_test_code" \
+    if bup_try_c_code "#include <stdio.h>
+                       #include <readline/readline.h> $readline_test_code" \
                       "$bup_readline_cflags"
     then
         bup_have_readline=1
         bup_readline_includes_in_subdir=1
-    elif bup_try_c_code "#include <readline.h> $readline_test_code" \
+    elif bup_try_c_code "#include <stdio.h>
+                         #include <readline.h> $readline_test_code" \
                         "$bup_readline_cflags"
     then
         bup_have_readline=1
@@ -210,11 +212,13 @@ if pkg-config readline; then
     fi
 fi
 if ! test "$bup_have_readline"; then
-    if bup_try_c_code "#include <readline/readline.h> $readline_test_code"; then
+    if bup_try_c_code "#include <stdio.h>
+                       #include <readline/readline.h> $readline_test_code"; then
         bup_readline_ldflags=-lreadline
         bup_have_readline=1
         bup_readline_includes_in_subdir=1
-    elif bup_try_c_code "#include <readline.h> $readline_test_code"; then
+    elif bup_try_c_code "#include <stdio.h>
+                         #include <readline.h> $readline_test_code"; then
         bup_readline_ldflags=-lreadline
         bup_have_readline=1
     fi
