diff -Naur iana-etc-1.04.orig/port-numbers.dist iana-etc-1.04/port-numbers.dist
--- iana-etc-1.04.orig/port-numbers.dist	2005-04-14 10:49:30.626611320 -0700
+++ iana-etc-1.04/port-numbers.dist	2005-04-14 10:50:35.421760960 -0700
@@ -12191,6 +12191,12 @@
 #               10009-10079 Unassigned
 amanda          10080/tcp  Amanda 
 amanda          10080/udp  Amanda 
+kamanda         10081/tcp  Amanda Kerberos authentication
+kamanda         10081/udp  Amanda Kerberos authentication
+amandaidx       10082/tcp  Amanda
+amandaidx       10082/udp  Amanda
+amidxtape       10083/tcp  Amanda
+amidxtape       10083/udp  Amanda
 #                          John Jackson <jrj@gandalf.cc.purdue.edu>
 #                          <amanda-core@amanda.org>
 #               10081-10099 Unassigned
