diff -Naur iana-etc-2.30.orig/port-numbers.dist iana-etc-2.30/port-numbers.dist
--- iana-etc-2.30.orig/port-numbers.dist	2008-03-05 18:32:33.000000000 +0100
+++ iana-etc-2.30/port-numbers.dist	2008-03-06 00:50:42.000000000 +0100
@@ -14312,6 +14312,12 @@
 #               10052-10079 Unassigned
 amanda          10080/tcp  Amanda 
 amanda          10080/udp  Amanda 
+kamanda         10081/tcp  Amanda Kerberos authentication
+kamanda         10081/udp  Amanda Kerberos authentication
+amandaidx       10082/tcp  Amanda
+amandaidx       10082/udp  Amanda
+amidxtape       10083/tcp  Amanda
+amidxtape       10083/udp  Amanda
 #                          John Jackson <jrj&gandalf.cc.purdue.edu>
 #                          <amanda-core&amanda.org>
 famdc		10081/tcp  FAM Archive Server
