diff -Naur iana-etc-1.04.orig/port-numbers.dist iana-etc-1.04/port-numbers.dist
--- iana-etc-1.04.orig/port-numbers.dist	2005-05-25 08:44:52.835050024 -0700
+++ iana-etc-1.04/port-numbers.dist	2005-05-25 08:46:01.556602760 -0700
@@ -2027,8 +2027,10 @@
 omginitialrefs  900/tcp    OMG Initial Refs
 omginitialrefs  900/udp    OMG Initial Refs
 #			   Christian Callsen <Christian.Callsen@eng.sun.com>
-smpnameres      901/tcp    SMPNAMERES
-smpnameres      901/udp    SMPNAMERES
+#			   Eric Sandall <eric@sandall.us> for Source Mage GNU/Linux
+#			   Changed smbnameres to swat
+swat            901/tcp    SWAT
+swat            901/udp    SWAT
 #                          Leif Ekblad <leif@rdos.net>
 ideafarm-chat   902/tcp    IDEAFARM-CHAT
 ideafarm-chat   902/udp    IDEAFARM-CHAT
