diff -Naur iana-etc-2.30.orig/port-numbers.dist iana-etc-2.30/port-numbers.dist
--- iana-etc-2.30.orig/port-numbers.dist	2008-03-05 18:32:33.000000000 +0100
+++ iana-etc-2.30/port-numbers.dist	2008-03-06 00:57:31.000000000 +0100
@@ -2100,6 +2100,10 @@
 #			   Christian Callsen <Christian.Callsen&eng.sun.com>
 smpnameres      901/tcp    SMPNAMERES
 smpnameres      901/udp    SMPNAMERES
+#
+swat            901/tcp    SWAT
+swat            901/udp    SWAT
+#
 #                          Leif Ekblad <leif&rdos.net>
 ideafarm-door   902/tcp    self documenting Telnet Door
 ideafarm-door   902/udp    self documenting Door: send 0x00 for info
