diff -Nur shadow-4.0.15/man/cs/Makefile.am shadow-4.0.15-patched/man/cs/Makefile.am
--- shadow-4.0.15/man/cs/Makefile.am	2006-03-07 21:28:54.000000000 +0100
+++ shadow-4.0.15-patched/man/cs/Makefile.am	2006-04-11 10:14:00.000000000 +0200
@@ -14,7 +14,6 @@
 	nologin.8 \
 	passwd.5 \
 	shadow.5 \
-	su.1 \
 	vipw.8
 
 EXTRA_DIST = $(man_MANS) \
diff -Nur shadow-4.0.15/man/cs/Makefile.in shadow-4.0.15-patched/man/cs/Makefile.in
--- shadow-4.0.15/man/cs/Makefile.in	2006-03-20 12:27:15.000000000 +0100
+++ shadow-4.0.15-patched/man/cs/Makefile.in	2006-04-11 10:13:45.000000000 +0200
@@ -196,7 +196,6 @@
 	nologin.8 \
 	passwd.5 \
 	shadow.5 \
-	su.1 \
 	vipw.8
 
 EXTRA_DIST = $(man_MANS) \
diff -Nur shadow-4.0.15/man/es/Makefile.am shadow-4.0.15-patched/man/es/Makefile.am
--- shadow-4.0.15/man/es/Makefile.am	2005-12-13 12:34:59.000000000 +0100
+++ shadow-4.0.15-patched/man/es/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -6,7 +6,6 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 
diff -Nur shadow-4.0.15/man/es/Makefile.in shadow-4.0.15-patched/man/es/Makefile.in
--- shadow-4.0.15/man/es/Makefile.in	2006-03-20 12:27:16.000000000 +0100
+++ shadow-4.0.15-patched/man/es/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -188,7 +188,6 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 
diff -Nur shadow-4.0.15/man/fi/Makefile.am shadow-4.0.15-patched/man/fi/Makefile.am
--- shadow-4.0.15/man/fi/Makefile.am	2005-11-12 18:01:00.000000000 +0100
+++ shadow-4.0.15-patched/man/fi/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -4,7 +4,6 @@
 man_MANS = \
 	chfn.1 \
 	chsh.1 \
-	passwd.1 \
-	su.1
+	passwd.1
 
 EXTRA_DIST = $(man_MANS)
diff -Nur shadow-4.0.15/man/fi/Makefile.in shadow-4.0.15-patched/man/fi/Makefile.in
--- shadow-4.0.15/man/fi/Makefile.in	2006-03-20 12:27:16.000000000 +0100
+++ shadow-4.0.15-patched/man/fi/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -183,8 +183,7 @@
 man_MANS = \
 	chfn.1 \
 	chsh.1 \
-	passwd.1 \
-	su.1
+	passwd.1
 
 EXTRA_DIST = $(man_MANS)
 all: all-am
diff -Nur shadow-4.0.15/man/fr/Makefile.am shadow-4.0.15-patched/man/fr/Makefile.am
--- shadow-4.0.15/man/fr/Makefile.am	2005-12-07 01:01:15.000000000 +0100
+++ shadow-4.0.15-patched/man/fr/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -36,8 +36,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -Nur shadow-4.0.15/man/fr/Makefile.in shadow-4.0.15-patched/man/fr/Makefile.in
--- shadow-4.0.15/man/fr/Makefile.in	2006-03-20 12:27:16.000000000 +0100
+++ shadow-4.0.15-patched/man/fr/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -219,8 +219,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -Nur shadow-4.0.15/man/hu/Makefile.am shadow-4.0.15-patched/man/hu/Makefile.am
--- shadow-4.0.15/man/hu/Makefile.am	2005-12-13 12:34:59.000000000 +0100
+++ shadow-4.0.15-patched/man/hu/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -10,8 +10,7 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	sg.1 \
-	su.1
+	sg.1
 
 EXTRA_DIST = $(man_MANS)
  
\ No newline at end of file
diff -Nur shadow-4.0.15/man/hu/Makefile.in shadow-4.0.15-patched/man/hu/Makefile.in
--- shadow-4.0.15/man/hu/Makefile.in	2006-03-20 12:27:16.000000000 +0100
+++ shadow-4.0.15-patched/man/hu/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -192,8 +192,7 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	sg.1 \
-	su.1
+	sg.1
 
 EXTRA_DIST = $(man_MANS)
 all: all-am
diff -Nur shadow-4.0.15/man/it/Makefile.am shadow-4.0.15-patched/man/it/Makefile.am
--- shadow-4.0.15/man/it/Makefile.am	2005-11-12 17:35:45.000000000 +0100
+++ shadow-4.0.15-patched/man/it/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -32,7 +32,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -Nur shadow-4.0.15/man/it/Makefile.in shadow-4.0.15-patched/man/it/Makefile.in
--- shadow-4.0.15/man/it/Makefile.in	2006-03-20 12:27:17.000000000 +0100
+++ shadow-4.0.15-patched/man/it/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -215,7 +215,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -Nur shadow-4.0.15/man/ja/Makefile.am shadow-4.0.15-patched/man/ja/Makefile.am
--- shadow-4.0.15/man/ja/Makefile.am	2005-07-10 19:59:40.000000000 +0200
+++ shadow-4.0.15-patched/man/ja/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -33,8 +33,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -44,5 +42,4 @@
 EXTRA_DIST = $(man_MANS) \
 	id.1 \
 	pw_auth.3 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
diff -Nur shadow-4.0.15/man/ja/Makefile.in shadow-4.0.15-patched/man/ja/Makefile.in
--- shadow-4.0.15/man/ja/Makefile.in	2006-03-20 12:27:17.000000000 +0100
+++ shadow-4.0.15-patched/man/ja/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -215,8 +215,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -226,8 +224,7 @@
 EXTRA_DIST = $(man_MANS) \
 	id.1 \
 	pw_auth.3 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
 
 all: all-am
 
diff -Nur shadow-4.0.15/man/ko/Makefile.am shadow-4.0.15-patched/man/ko/Makefile.am
--- shadow-4.0.15/man/ko/Makefile.am	2005-12-13 12:34:59.000000000 +0100
+++ shadow-4.0.15-patched/man/ko/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -7,7 +7,6 @@
 	groups.1 \
 	login.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 # newgrp.1 must be updated
diff -Nur shadow-4.0.15/man/ko/Makefile.in shadow-4.0.15-patched/man/ko/Makefile.in
--- shadow-4.0.15/man/ko/Makefile.in	2006-03-20 12:27:17.000000000 +0100
+++ shadow-4.0.15-patched/man/ko/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -189,7 +189,6 @@
 	groups.1 \
 	login.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 
diff -Nur shadow-4.0.15/man/Makefile.am shadow-4.0.15-patched/man/Makefile.am
--- shadow-4.0.15/man/Makefile.am	2006-03-12 19:53:48.000000000 +0100
+++ shadow-4.0.15-patched/man/Makefile.am	2006-04-11 10:12:23.000000000 +0200
@@ -40,8 +40,6 @@
 	shadow.3.xml \
 	shadow.5.xml \
 	sg.1.xml \
-	su.1.xml \
-	suauth.5.xml \
 	useradd.8.xml \
 	userdel.8.xml \
 	usermod.8.xml \
@@ -83,8 +81,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -96,10 +92,7 @@
 	id.1 \
 	id.1.xml \
 	pw_auth.3 \
-	pw_auth.3.xml \
-	sulogin.8 \
-	sulogin.8.xml \
-	sulogin.8.xml
+	pw_auth.3.xml
 
 all:
 
diff -Nur shadow-4.0.15/man/Makefile.in shadow-4.0.15-patched/man/Makefile.in
--- shadow-4.0.15/man/Makefile.in	2006-03-20 12:27:15.000000000 +0100
+++ shadow-4.0.15-patched/man/Makefile.in	2006-04-11 10:12:23.000000000 +0200
@@ -230,8 +230,6 @@
 	shadow.3.xml \
 	shadow.5.xml \
 	sg.1.xml \
-	su.1.xml \
-	suauth.5.xml \
 	useradd.8.xml \
 	userdel.8.xml \
 	usermod.8.xml \
@@ -273,8 +271,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -286,10 +282,7 @@
 	id.1 \
 	id.1.xml \
 	pw_auth.3 \
-	pw_auth.3.xml \
-	sulogin.8 \
-	sulogin.8.xml \
-	sulogin.8.xml
+	pw_auth.3.xml
 
 @ENABLE_REGENERATE_MAN_TRUE@CLEANFILES = $(man_MANS)
 all: all-recursive
diff -Nur shadow-4.0.15/man/pl/Makefile.am shadow-4.0.15-patched/man/pl/Makefile.am
--- shadow-4.0.15/man/pl/Makefile.am	2005-07-10 19:59:43.000000000 +0200
+++ shadow-4.0.15-patched/man/pl/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -33,8 +33,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -44,5 +42,4 @@
 EXTRA_DIST = $(man_MANS) \
 	id.1 \
 	pw_auth.3 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
diff -Nur shadow-4.0.15/man/pl/Makefile.in shadow-4.0.15-patched/man/pl/Makefile.in
--- shadow-4.0.15/man/pl/Makefile.in	2006-03-20 12:27:17.000000000 +0100
+++ shadow-4.0.15-patched/man/pl/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -215,8 +215,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -226,8 +224,7 @@
 EXTRA_DIST = $(man_MANS) \
 	id.1 \
 	pw_auth.3 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
 
 all: all-am
 
diff -Nur shadow-4.0.15/man/ru/Makefile.am shadow-4.0.15-patched/man/ru/Makefile.am
--- shadow-4.0.15/man/ru/Makefile.am	2006-03-11 16:39:54.000000000 +0100
+++ shadow-4.0.15-patched/man/ru/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -39,9 +39,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
-	sulogin.8 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -Nur shadow-4.0.15/man/ru/Makefile.in shadow-4.0.15-patched/man/ru/Makefile.in
--- shadow-4.0.15/man/ru/Makefile.in	2006-03-20 12:27:18.000000000 +0100
+++ shadow-4.0.15-patched/man/ru/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -223,9 +223,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
-	sulogin.8 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -Nur shadow-4.0.15/man/tr/Makefile.am shadow-4.0.15-patched/man/tr/Makefile.am
--- shadow-4.0.15/man/tr/Makefile.am	2005-12-07 21:20:18.000000000 +0100
+++ shadow-4.0.15-patched/man/tr/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -10,7 +10,6 @@
 	passwd.1 \
 	passwd.5 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -Nur shadow-4.0.15/man/tr/Makefile.in shadow-4.0.15-patched/man/tr/Makefile.in
--- shadow-4.0.15/man/tr/Makefile.in	2006-03-20 12:27:18.000000000 +0100
+++ shadow-4.0.15-patched/man/tr/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -193,7 +193,6 @@
 	passwd.1 \
 	passwd.5 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -Nur shadow-4.0.15/man/zh_CN/Makefile.am shadow-4.0.15-patched/man/zh_CN/Makefile.am
--- shadow-4.0.15/man/zh_CN/Makefile.am	2005-12-13 12:34:59.000000000 +0100
+++ shadow-4.0.15-patched/man/zh_CN/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -10,7 +10,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -Nur shadow-4.0.15/man/zh_CN/Makefile.in shadow-4.0.15-patched/man/zh_CN/Makefile.in
--- shadow-4.0.15/man/zh_CN/Makefile.in	2006-03-20 12:27:18.000000000 +0100
+++ shadow-4.0.15-patched/man/zh_CN/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -194,7 +194,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -Nur shadow-4.0.15/man/zh_TW/Makefile.am shadow-4.0.15-patched/man/zh_TW/Makefile.am
--- shadow-4.0.15/man/zh_TW/Makefile.am	2005-12-13 12:34:59.000000000 +0100
+++ shadow-4.0.15-patched/man/zh_TW/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -10,7 +10,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -Nur shadow-4.0.15/man/zh_TW/Makefile.in shadow-4.0.15-patched/man/zh_TW/Makefile.in
--- shadow-4.0.15/man/zh_TW/Makefile.in	2006-03-20 12:27:19.000000000 +0100
+++ shadow-4.0.15-patched/man/zh_TW/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -194,7 +194,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -Nur shadow-4.0.15/src/Makefile.am shadow-4.0.15-patched/src/Makefile.am
--- shadow-4.0.15/src/Makefile.am	2006-03-11 16:49:39.000000000 +0100
+++ shadow-4.0.15-patched/src/Makefile.am	2006-04-11 10:12:24.000000000 +0200
@@ -43,9 +43,9 @@
 	vipw
 
 # id and groups are from gnu, sulogin from sysvinit
-noinst_PROGRAMS = id sulogin
+noinst_PROGRAMS = id sulogin su
 
-suidbins       = su
+suidbins       =
 suidubins      = chage chfn chsh expiry gpasswd newgrp passwd
 
 LDADD 	       = $(top_builddir)/libmisc/libmisc.a \
diff -Nur shadow-4.0.15/src/Makefile.in shadow-4.0.15-patched/src/Makefile.in
--- shadow-4.0.15/src/Makefile.in	2006-03-20 12:27:19.000000000 +0100
+++ shadow-4.0.15-patched/src/Makefile.in	2006-04-11 10:12:24.000000000 +0200
@@ -36,7 +36,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-bin_PROGRAMS = groups$(EXEEXT) login$(EXEEXT) su$(EXEEXT)
+bin_PROGRAMS = groups$(EXEEXT) login$(EXEEXT)
 ubin_PROGRAMS = faillog$(EXEEXT) lastlog$(EXEEXT) chage$(EXEEXT) \
 	chfn$(EXEEXT) chsh$(EXEEXT) expiry$(EXEEXT) gpasswd$(EXEEXT) \
 	newgrp$(EXEEXT) passwd$(EXEEXT)
@@ -358,7 +358,7 @@
 	-I${top_srcdir}/lib \
 	-I$(top_srcdir)/libmisc
 
-suidbins = su
+suidbins =
 suidubins = chage chfn chsh expiry gpasswd newgrp passwd
 LDADD = $(top_builddir)/libmisc/libmisc.a \
 		 $(top_builddir)/lib/libshadow.la
