diff -ur shadow-4.0.17/man/cs/Makefile.am shadow-4.0.17.cp/man/cs/Makefile.am
--- shadow-4.0.17/man/cs/Makefile.am	2006-06-04 18:14:28.000000000 +0200
+++ shadow-4.0.17.cp/man/cs/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -16,7 +16,6 @@
 	nologin.8 \
 	passwd.5 \
 	shadow.5 \
-	su.1 \
 	vipw.8
 
 EXTRA_DIST = $(man_MANS) \
diff -ur shadow-4.0.17/man/cs/Makefile.in shadow-4.0.17.cp/man/cs/Makefile.in
--- shadow-4.0.17/man/cs/Makefile.in	2006-06-25 14:31:38.000000000 +0200
+++ shadow-4.0.17.cp/man/cs/Makefile.in	2006-07-12 09:59:09.000000000 +0200
@@ -200,7 +200,6 @@
 	nologin.8 \
 	passwd.5 \
 	shadow.5 \
-	su.1 \
 	vipw.8
 
 EXTRA_DIST = $(man_MANS) \
diff -ur shadow-4.0.17/man/es/Makefile.am shadow-4.0.17.cp/man/es/Makefile.am
--- shadow-4.0.17/man/es/Makefile.am	2005-12-13 12:34:59.000000000 +0100
+++ shadow-4.0.17.cp/man/es/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -6,7 +6,6 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 
diff -ur shadow-4.0.17/man/es/Makefile.in shadow-4.0.17.cp/man/es/Makefile.in
--- shadow-4.0.17/man/es/Makefile.in	2006-06-25 14:31:38.000000000 +0200
+++ shadow-4.0.17.cp/man/es/Makefile.in	2006-07-12 09:59:09.000000000 +0200
@@ -190,7 +190,6 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 
diff -ur shadow-4.0.17/man/fi/Makefile.am shadow-4.0.17.cp/man/fi/Makefile.am
--- shadow-4.0.17/man/fi/Makefile.am	2005-11-12 18:01:00.000000000 +0100
+++ shadow-4.0.17.cp/man/fi/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -4,7 +4,6 @@
 man_MANS = \
 	chfn.1 \
 	chsh.1 \
-	passwd.1 \
-	su.1
+	passwd.1
 
 EXTRA_DIST = $(man_MANS)
diff -ur shadow-4.0.17/man/fi/Makefile.in shadow-4.0.17.cp/man/fi/Makefile.in
--- shadow-4.0.17/man/fi/Makefile.in	2006-06-25 14:31:39.000000000 +0200
+++ shadow-4.0.17.cp/man/fi/Makefile.in	2006-07-12 09:59:09.000000000 +0200
@@ -185,8 +185,7 @@
 man_MANS = \
 	chfn.1 \
 	chsh.1 \
-	passwd.1 \
-	su.1
+	passwd.1
 
 EXTRA_DIST = $(man_MANS)
 all: all-am
diff -ur shadow-4.0.17/man/fr/Makefile.am shadow-4.0.17.cp/man/fr/Makefile.am
--- shadow-4.0.17/man/fr/Makefile.am	2006-05-20 04:25:42.000000000 +0200
+++ shadow-4.0.17.cp/man/fr/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -43,8 +43,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -ur shadow-4.0.17/man/fr/Makefile.in shadow-4.0.17.cp/man/fr/Makefile.in
--- shadow-4.0.17/man/fr/Makefile.in	2006-06-25 14:31:39.000000000 +0200
+++ shadow-4.0.17.cp/man/fr/Makefile.in	2006-07-12 09:59:09.000000000 +0200
@@ -225,8 +225,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -ur shadow-4.0.17/man/hu/Makefile.am shadow-4.0.17.cp/man/hu/Makefile.am
--- shadow-4.0.17/man/hu/Makefile.am	2005-12-13 12:34:59.000000000 +0100
+++ shadow-4.0.17.cp/man/hu/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -10,8 +10,7 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	sg.1 \
-	su.1
+	sg.1
 
 EXTRA_DIST = $(man_MANS)
  
\ No newline at end of file
diff -ur shadow-4.0.17/man/hu/Makefile.in shadow-4.0.17.cp/man/hu/Makefile.in
--- shadow-4.0.17/man/hu/Makefile.in	2006-06-25 14:31:39.000000000 +0200
+++ shadow-4.0.17.cp/man/hu/Makefile.in	2006-07-12 09:59:09.000000000 +0200
@@ -194,8 +194,7 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	sg.1 \
-	su.1
+	sg.1
 
 EXTRA_DIST = $(man_MANS)
 all: all-am
diff -ur shadow-4.0.17/man/it/Makefile.am shadow-4.0.17.cp/man/it/Makefile.am
--- shadow-4.0.17/man/it/Makefile.am	2006-05-25 15:48:53.000000000 +0200
+++ shadow-4.0.17.cp/man/it/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -38,7 +38,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -ur shadow-4.0.17/man/it/Makefile.in shadow-4.0.17.cp/man/it/Makefile.in
--- shadow-4.0.17/man/it/Makefile.in	2006-06-25 14:31:40.000000000 +0200
+++ shadow-4.0.17.cp/man/it/Makefile.in	2006-07-12 09:59:09.000000000 +0200
@@ -220,7 +220,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -ur shadow-4.0.17/man/ja/Makefile.am shadow-4.0.17.cp/man/ja/Makefile.am
--- shadow-4.0.17/man/ja/Makefile.am	2006-05-13 16:16:59.000000000 +0200
+++ shadow-4.0.17.cp/man/ja/Makefile.am	2006-07-12 10:01:46.000000000 +0200
@@ -40,8 +40,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -52,5 +50,4 @@
 	$(man_MANS) \
 	$(man_nonpam) \
 	id.1 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
diff -ur shadow-4.0.17/man/ja/Makefile.in shadow-4.0.17.cp/man/ja/Makefile.in
--- shadow-4.0.17/man/ja/Makefile.in	2006-06-25 14:31:40.000000000 +0200
+++ shadow-4.0.17.cp/man/ja/Makefile.in	2006-07-12 10:03:14.000000000 +0200
@@ -221,8 +221,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -233,9 +231,7 @@
 	$(man_MANS) \
 	$(man_nonpam) \
 	id.1 \
-	shadow.3 \
-	sulogin.8
-
+	shadow.3
 all: all-am
 
 .SUFFIXES:
diff -ur shadow-4.0.17/man/ko/Makefile.am shadow-4.0.17.cp/man/ko/Makefile.am
--- shadow-4.0.17/man/ko/Makefile.am	2005-12-13 12:34:59.000000000 +0100
+++ shadow-4.0.17.cp/man/ko/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -7,7 +7,6 @@
 	groups.1 \
 	login.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 # newgrp.1 must be updated
diff -ur shadow-4.0.17/man/ko/Makefile.in shadow-4.0.17.cp/man/ko/Makefile.in
--- shadow-4.0.17/man/ko/Makefile.in	2006-06-25 14:31:40.000000000 +0200
+++ shadow-4.0.17.cp/man/ko/Makefile.in	2006-07-12 09:59:09.000000000 +0200
@@ -191,7 +191,6 @@
 	groups.1 \
 	login.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 
diff -ur shadow-4.0.17/man/Makefile.am shadow-4.0.17.cp/man/Makefile.am
--- shadow-4.0.17/man/Makefile.am	2006-07-07 19:08:36.000000000 +0200
+++ shadow-4.0.17.cp/man/Makefile.am	2006-07-12 10:07:41.000000000 +0200
@@ -43,8 +43,6 @@
 	shadow.3.xml \
 	shadow.5.xml \
 	sg.1.xml \
-	su.1.xml \
-	suauth.5.xml \
 	useradd.8.xml \
 	userdel.8.xml \
 	usermod.8.xml \
@@ -94,8 +92,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -107,9 +103,7 @@
 	$(man_XMANS) \
 	$(man_nonpam) \
 	id.1 \
-	id.1.xml \
-	sulogin.8 \
-	sulogin.8.xml
+	id.1.xml 
 
 if ENABLE_REGENERATE_MAN
 
diff -ur shadow-4.0.17/man/Makefile.in shadow-4.0.17.cp/man/Makefile.in
--- shadow-4.0.17/man/Makefile.in	2006-07-10 16:00:56.000000000 +0200
+++ shadow-4.0.17.cp/man/Makefile.in	2006-07-12 10:08:05.000000000 +0200
@@ -234,8 +234,6 @@
 	shadow.3.xml \
 	shadow.5.xml \
 	sg.1.xml \
-	su.1.xml \
-	suauth.5.xml \
 	useradd.8.xml \
 	userdel.8.xml \
 	usermod.8.xml \
@@ -282,8 +280,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -295,9 +291,7 @@
 	$(man_XMANS) \
 	$(man_nonpam) \
 	id.1 \
-	id.1.xml \
-	sulogin.8 \
-	sulogin.8.xml
+	id.1.xml
 
 @ENABLE_REGENERATE_MAN_TRUE@CLEANFILES = $(man_MANS) $(ALL_TRANSLATED_XMLS)
 @ENABLE_REGENERATE_MAN_TRUE@POFILES = $(foreach lang, $(LINGUAS), $(lang)/$(lang).po)
diff -ur shadow-4.0.17/man/pl/Makefile.am shadow-4.0.17.cp/man/pl/Makefile.am
--- shadow-4.0.17/man/pl/Makefile.am	2006-05-13 16:15:46.000000000 +0200
+++ shadow-4.0.17.cp/man/pl/Makefile.am	2006-07-12 10:09:50.000000000 +0200
@@ -40,8 +40,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -52,5 +50,4 @@
 	$(man_MANS) \
 	$(man_nonpam) \
 	id.1 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
diff -ur shadow-4.0.17/man/pl/Makefile.in shadow-4.0.17.cp/man/pl/Makefile.in
--- shadow-4.0.17/man/pl/Makefile.in	2006-06-25 14:31:41.000000000 +0200
+++ shadow-4.0.17.cp/man/pl/Makefile.in	2006-07-12 10:10:09.000000000 +0200
@@ -221,8 +221,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -233,8 +231,7 @@
 	$(man_MANS) \
 	$(man_nonpam) \
 	id.1 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
 
 all: all-am
 
diff -ur shadow-4.0.17/man/ru/Makefile.am shadow-4.0.17.cp/man/ru/Makefile.am
--- shadow-4.0.17/man/ru/Makefile.am	2006-05-25 15:49:28.000000000 +0200
+++ shadow-4.0.17.cp/man/ru/Makefile.am	2006-07-12 10:11:28.000000000 +0200
@@ -56,5 +56,4 @@
 EXTRA_DIST = \
 	$(man_MANS) \
 	$(man_nonpam) \
-	id.1 \
-	sulogin.8
+	id.1
diff -ur shadow-4.0.17/man/ru/Makefile.in shadow-4.0.17.cp/man/ru/Makefile.in
--- shadow-4.0.17/man/ru/Makefile.in	2006-06-25 14:31:41.000000000 +0200
+++ shadow-4.0.17.cp/man/ru/Makefile.in	2006-07-12 10:11:46.000000000 +0200
@@ -239,8 +239,7 @@
 EXTRA_DIST = \
 	$(man_MANS) \
 	$(man_nonpam) \
-	id.1 \
-	sulogin.8
+	id.1
 
 all: all-am
 
diff -ur shadow-4.0.17/man/tr/Makefile.am shadow-4.0.17.cp/man/tr/Makefile.am
--- shadow-4.0.17/man/tr/Makefile.am	2005-12-07 21:20:18.000000000 +0100
+++ shadow-4.0.17.cp/man/tr/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -10,7 +10,6 @@
 	passwd.1 \
 	passwd.5 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -ur shadow-4.0.17/man/tr/Makefile.in shadow-4.0.17.cp/man/tr/Makefile.in
--- shadow-4.0.17/man/tr/Makefile.in	2006-06-25 14:31:41.000000000 +0200
+++ shadow-4.0.17.cp/man/tr/Makefile.in	2006-07-12 09:59:09.000000000 +0200
@@ -195,7 +195,6 @@
 	passwd.1 \
 	passwd.5 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -ur shadow-4.0.17/man/zh_CN/Makefile.am shadow-4.0.17.cp/man/zh_CN/Makefile.am
--- shadow-4.0.17/man/zh_CN/Makefile.am	2005-12-13 12:34:59.000000000 +0100
+++ shadow-4.0.17.cp/man/zh_CN/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -10,7 +10,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -ur shadow-4.0.17/man/zh_CN/Makefile.in shadow-4.0.17.cp/man/zh_CN/Makefile.in
--- shadow-4.0.17/man/zh_CN/Makefile.in	2006-06-25 14:31:42.000000000 +0200
+++ shadow-4.0.17.cp/man/zh_CN/Makefile.in	2006-07-12 09:59:09.000000000 +0200
@@ -196,7 +196,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -ur shadow-4.0.17/man/zh_TW/Makefile.am shadow-4.0.17.cp/man/zh_TW/Makefile.am
--- shadow-4.0.17/man/zh_TW/Makefile.am	2005-12-13 12:34:59.000000000 +0100
+++ shadow-4.0.17.cp/man/zh_TW/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -10,7 +10,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -ur shadow-4.0.17/man/zh_TW/Makefile.in shadow-4.0.17.cp/man/zh_TW/Makefile.in
--- shadow-4.0.17/man/zh_TW/Makefile.in	2006-06-25 14:31:42.000000000 +0200
+++ shadow-4.0.17.cp/man/zh_TW/Makefile.in	2006-07-12 09:59:09.000000000 +0200
@@ -196,7 +196,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -ur shadow-4.0.17/src/Makefile.am shadow-4.0.17.cp/src/Makefile.am
--- shadow-4.0.17/src/Makefile.am	2006-06-24 13:34:44.000000000 +0200
+++ shadow-4.0.17.cp/src/Makefile.am	2006-07-12 09:59:09.000000000 +0200
@@ -43,9 +43,9 @@
 	vipw
 
 # id and groups are from gnu, sulogin from sysvinit
-noinst_PROGRAMS = id sulogin
+noinst_PROGRAMS = id sulogin su
 
-suidbins       = su
+suidbins       =
 suidubins      = chage chfn chsh expiry gpasswd newgrp passwd
 
 LDADD 	       = $(top_builddir)/libmisc/libmisc.a \
diff -ur shadow-4.0.17/src/Makefile.in shadow-4.0.17.cp/src/Makefile.in
--- shadow-4.0.17/src/Makefile.in	2006-06-25 14:31:43.000000000 +0200
+++ shadow-4.0.17.cp/src/Makefile.in	2006-07-12 10:13:57.000000000 +0200
@@ -36,7 +36,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-bin_PROGRAMS = groups$(EXEEXT) login$(EXEEXT) su$(EXEEXT)
+bin_PROGRAMS = groups$(EXEEXT) login$(EXEEXT)
 sbin_PROGRAMS = nologin$(EXEEXT)
 ubin_PROGRAMS = faillog$(EXEEXT) lastlog$(EXEEXT) chage$(EXEEXT) \
 	chfn$(EXEEXT) chsh$(EXEEXT) expiry$(EXEEXT) gpasswd$(EXEEXT) \
@@ -361,7 +361,7 @@
 	-I${top_srcdir}/lib \
 	-I$(top_srcdir)/libmisc
 
-suidbins = su
+suidbins =
 suidubins = chage chfn chsh expiry gpasswd newgrp passwd
 LDADD = $(top_builddir)/libmisc/libmisc.a \
 		 $(top_builddir)/lib/libshadow.la
