diff -urN shadow-4.0.14/man/Makefile.am shadow/man/Makefile.am
--- shadow-4.0.14/man/Makefile.am	2005-12-07 15:09:49.000000000 -0500
+++ shadow/man/Makefile.am	2006-02-03 00:16:20.000000000 -0500
@@ -34,8 +34,6 @@
 	shadow.3.xml \
 	shadow.5.xml \
 	sg.1.xml \
-	su.1.xml \
-	suauth.5.xml \
 	useradd.8.xml \
 	userdel.8.xml \
 	usermod.8.xml \
@@ -76,8 +74,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -89,10 +85,7 @@
 	id.1 \
 	id.1.xml \
 	pw_auth.3 \
-	pw_auth.3.xml \
-	sulogin.8 \
-	sulogin.8.xml \
-	sulogin.8.xml
+	pw_auth.3.xml
 
 all:
 
diff -urN shadow-4.0.14/man/Makefile.in shadow/man/Makefile.in
--- shadow-4.0.14/man/Makefile.in	2006-01-03 11:12:26.000000000 -0500
+++ shadow/man/Makefile.in	2006-02-03 00:16:39.000000000 -0500
@@ -226,8 +226,6 @@
 	shadow.3.xml \
 	shadow.5.xml \
 	sg.1.xml \
-	su.1.xml \
-	suauth.5.xml \
 	useradd.8.xml \
 	userdel.8.xml \
 	usermod.8.xml \
@@ -268,8 +266,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -281,10 +277,7 @@
 	id.1 \
 	id.1.xml \
 	pw_auth.3 \
-	pw_auth.3.xml \
-	sulogin.8 \
-	sulogin.8.xml \
-	sulogin.8.xml
+	pw_auth.3.xml
 
 @ENABLE_REGENERATE_MAN_TRUE@CLEANFILES = $(man_MANS)
 all: all-recursive
diff -urN shadow-4.0.14/man/cs/Makefile.am shadow/man/cs/Makefile.am
--- shadow-4.0.14/man/cs/Makefile.am	2005-04-01 12:51:51.000000000 -0500
+++ shadow/man/cs/Makefile.am	2006-02-03 00:09:45.000000000 -0500
@@ -6,7 +6,6 @@
 	id.1 \
 	lastlog.8 \
 	passwd.5 \
-	shadow.5 \
-	su.1
+	shadow.5
 
 EXTRA_DIST = $(man_MANS)
diff -urN shadow-4.0.14/man/cs/Makefile.in shadow/man/cs/Makefile.in
--- shadow-4.0.14/man/cs/Makefile.in	2006-01-03 11:12:27.000000000 -0500
+++ shadow/man/cs/Makefile.in	2006-02-03 00:10:05.000000000 -0500
@@ -186,8 +186,7 @@
 	id.1 \
 	lastlog.8 \
 	passwd.5 \
-	shadow.5 \
-	su.1
+	shadow.5
 
 EXTRA_DIST = $(man_MANS)
 all: all-am
diff -urN shadow-4.0.14/man/es/Makefile.am shadow/man/es/Makefile.am
--- shadow-4.0.14/man/es/Makefile.am	2005-12-11 15:34:25.000000000 -0500
+++ shadow/man/es/Makefile.am	2006-02-03 00:04:24.000000000 -0500
@@ -6,7 +6,6 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 
diff -urN shadow-4.0.14/man/es/Makefile.in shadow/man/es/Makefile.in
--- shadow-4.0.14/man/es/Makefile.in	2006-01-03 11:12:28.000000000 -0500
+++ shadow/man/es/Makefile.in	2006-02-03 00:04:52.000000000 -0500
@@ -186,7 +186,6 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 
diff -urN shadow-4.0.14/man/fi/Makefile.am shadow/man/fi/Makefile.am
--- shadow-4.0.14/man/fi/Makefile.am	2005-11-12 13:00:03.000000000 -0500
+++ shadow/man/fi/Makefile.am	2006-02-03 00:10:48.000000000 -0500
@@ -4,7 +4,6 @@
 man_MANS = \
 	chfn.1 \
 	chsh.1 \
-	passwd.1 \
-	su.1
+	passwd.1
 
 EXTRA_DIST = $(man_MANS)
diff -urN shadow-4.0.14/man/fi/Makefile.in shadow/man/fi/Makefile.in
--- shadow-4.0.14/man/fi/Makefile.in	2006-01-03 11:12:28.000000000 -0500
+++ shadow/man/fi/Makefile.in	2006-02-03 00:11:01.000000000 -0500
@@ -181,8 +181,7 @@
 man_MANS = \
 	chfn.1 \
 	chsh.1 \
-	passwd.1 \
-	su.1
+	passwd.1
 
 EXTRA_DIST = $(man_MANS)
 all: all-am
diff -urN shadow-4.0.14/man/fr/Makefile.am shadow/man/fr/Makefile.am
--- shadow-4.0.14/man/fr/Makefile.am	2005-12-06 20:00:38.000000000 -0500
+++ shadow/man/fr/Makefile.am	2006-02-03 00:11:14.000000000 -0500
@@ -36,8 +36,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -urN shadow-4.0.14/man/fr/Makefile.in shadow/man/fr/Makefile.in
--- shadow-4.0.14/man/fr/Makefile.in	2006-01-03 11:12:28.000000000 -0500
+++ shadow/man/fr/Makefile.in	2006-02-03 00:11:25.000000000 -0500
@@ -217,8 +217,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -urN shadow-4.0.14/man/hu/Makefile.am shadow/man/hu/Makefile.am
--- shadow-4.0.14/man/hu/Makefile.am	2005-12-11 15:30:06.000000000 -0500
+++ shadow/man/hu/Makefile.am	2006-02-03 00:05:11.000000000 -0500
@@ -10,8 +10,7 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	sg.1 \
-	su.1
+	sg.1
 
 EXTRA_DIST = $(man_MANS)
  
\ No newline at end of file
diff -urN shadow-4.0.14/man/hu/Makefile.in shadow/man/hu/Makefile.in
--- shadow-4.0.14/man/hu/Makefile.in	2006-01-03 11:12:28.000000000 -0500
+++ shadow/man/hu/Makefile.in	2006-02-03 00:14:00.000000000 -0500
@@ -190,8 +190,7 @@
 	newgrp.1 \
 	passwd.1 \
 	passwd.5 \
-	sg.1 \
-	su.1
+	sg.1
 
 EXTRA_DIST = $(man_MANS)
 all: all-am
diff -urN shadow-4.0.14/man/it/Makefile.am shadow/man/it/Makefile.am
--- shadow-4.0.14/man/it/Makefile.am	2005-11-12 12:26:33.000000000 -0500
+++ shadow/man/it/Makefile.am	2006-02-03 00:14:11.000000000 -0500
@@ -32,7 +32,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -urN shadow-4.0.14/man/it/Makefile.in shadow/man/it/Makefile.in
--- shadow-4.0.14/man/it/Makefile.in	2006-01-03 11:12:29.000000000 -0500
+++ shadow/man/it/Makefile.in	2006-02-03 00:14:22.000000000 -0500
@@ -213,7 +213,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -urN shadow-4.0.14/man/ja/Makefile.am shadow/man/ja/Makefile.am
--- shadow-4.0.14/man/ja/Makefile.am	2005-12-13 06:29:44.000000000 -0500
+++ shadow/man/ja/Makefile.am	2006-02-03 00:20:50.000000000 -0500
@@ -33,8 +33,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -44,5 +42,4 @@
 EXTRA_DIST = $(man_MANS) \
 	id.1 \
 	pw_auth.3 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
diff -urN shadow-4.0.14/man/ja/Makefile.in shadow/man/ja/Makefile.in
--- shadow-4.0.14/man/ja/Makefile.in	2006-01-03 11:12:29.000000000 -0500
+++ shadow/man/ja/Makefile.in	2006-02-03 00:21:02.000000000 -0500
@@ -213,8 +213,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -224,8 +222,7 @@
 EXTRA_DIST = $(man_MANS) \
 	id.1 \
 	pw_auth.3 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
 
 all: all-am
 
diff -urN shadow-4.0.14/man/ko/Makefile.am shadow/man/ko/Makefile.am
--- shadow-4.0.14/man/ko/Makefile.am	2005-12-11 15:34:16.000000000 -0500
+++ shadow/man/ko/Makefile.am	2006-02-03 00:14:34.000000000 -0500
@@ -7,7 +7,6 @@
 	groups.1 \
 	login.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 # newgrp.1 must be updated
diff -urN shadow-4.0.14/man/ko/Makefile.in shadow/man/ko/Makefile.in
--- shadow-4.0.14/man/ko/Makefile.in	2006-01-03 11:12:30.000000000 -0500
+++ shadow/man/ko/Makefile.in	2006-02-03 00:14:43.000000000 -0500
@@ -187,7 +187,6 @@
 	groups.1 \
 	login.1 \
 	passwd.5 \
-	su.1 \
 	vigr.8 \
 	vipw.8
 
diff -urN shadow-4.0.14/man/pl/Makefile.am shadow/man/pl/Makefile.am
--- shadow-4.0.14/man/pl/Makefile.am	2005-07-10 14:00:33.000000000 -0400
+++ shadow/man/pl/Makefile.am	2006-02-03 00:21:11.000000000 -0500
@@ -33,8 +33,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -44,5 +42,4 @@
 EXTRA_DIST = $(man_MANS) \
 	id.1 \
 	pw_auth.3 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
diff -urN shadow-4.0.14/man/pl/Makefile.in shadow/man/pl/Makefile.in
--- shadow-4.0.14/man/pl/Makefile.in	2006-01-03 11:12:30.000000000 -0500
+++ shadow/man/pl/Makefile.in	2006-02-03 00:21:20.000000000 -0500
@@ -213,8 +213,6 @@
 	pwunconv.8 \
 	sg.1 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
@@ -224,8 +222,7 @@
 EXTRA_DIST = $(man_MANS) \
 	id.1 \
 	pw_auth.3 \
-	shadow.3 \
-	sulogin.8
+	shadow.3
 
 all: all-am
 
diff -urN shadow-4.0.14/man/ru/Makefile.am shadow/man/ru/Makefile.am
--- shadow-4.0.14/man/ru/Makefile.am	2005-12-02 18:23:35.000000000 -0500
+++ shadow/man/ru/Makefile.am	2006-02-03 00:15:01.000000000 -0500
@@ -38,9 +38,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
-	sulogin.8 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -urN shadow-4.0.14/man/ru/Makefile.in shadow/man/ru/Makefile.in
--- shadow-4.0.14/man/ru/Makefile.in	2006-01-03 11:12:31.000000000 -0500
+++ shadow/man/ru/Makefile.in	2006-02-03 00:15:12.000000000 -0500
@@ -220,9 +220,6 @@
 	sg.1 \
 	shadow.3 \
 	shadow.5 \
-	su.1 \
-	suauth.5 \
-	sulogin.8 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8 \
diff -urN shadow-4.0.14/man/tr/Makefile.am shadow/man/tr/Makefile.am
--- shadow-4.0.14/man/tr/Makefile.am	2005-12-07 15:09:05.000000000 -0500
+++ shadow/man/tr/Makefile.am	2006-02-03 00:15:17.000000000 -0500
@@ -10,7 +10,6 @@
 	passwd.1 \
 	passwd.5 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -urN shadow-4.0.14/man/tr/Makefile.in shadow/man/tr/Makefile.in
--- shadow-4.0.14/man/tr/Makefile.in	2006-01-03 11:12:31.000000000 -0500
+++ shadow/man/tr/Makefile.in	2006-02-03 00:15:27.000000000 -0500
@@ -191,7 +191,6 @@
 	passwd.1 \
 	passwd.5 \
 	shadow.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -urN shadow-4.0.14/man/zh_CN/Makefile.am shadow/man/zh_CN/Makefile.am
--- shadow-4.0.14/man/zh_CN/Makefile.am	2005-12-11 15:36:10.000000000 -0500
+++ shadow/man/zh_CN/Makefile.am	2006-02-03 00:15:32.000000000 -0500
@@ -10,7 +10,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -urN shadow-4.0.14/man/zh_CN/Makefile.in shadow/man/zh_CN/Makefile.in
--- shadow-4.0.14/man/zh_CN/Makefile.in	2006-01-03 11:12:32.000000000 -0500
+++ shadow/man/zh_CN/Makefile.in	2006-02-03 00:15:40.000000000 -0500
@@ -192,7 +192,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -urN shadow-4.0.14/man/zh_TW/Makefile.am shadow/man/zh_TW/Makefile.am
--- shadow-4.0.14/man/zh_TW/Makefile.am	2005-12-13 06:32:34.000000000 -0500
+++ shadow/man/zh_TW/Makefile.am	2006-02-03 00:15:45.000000000 -0500
@@ -10,7 +10,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -urN shadow-4.0.14/man/zh_TW/Makefile.in shadow/man/zh_TW/Makefile.in
--- shadow-4.0.14/man/zh_TW/Makefile.in	2006-01-03 11:12:32.000000000 -0500
+++ shadow/man/zh_TW/Makefile.in	2006-02-03 00:15:53.000000000 -0500
@@ -192,7 +192,6 @@
 	groupdel.8 \
 	groupmod.8 \
 	passwd.5 \
-	su.1 \
 	useradd.8 \
 	userdel.8 \
 	usermod.8
diff -urN shadow-4.0.14/src/Makefile.am shadow/src/Makefile.am
--- shadow-4.0.14/src/Makefile.am	2006-01-03 11:10:26.000000000 -0500
+++ shadow/src/Makefile.am	2006-02-03 00:02:57.000000000 -0500
@@ -42,9 +42,9 @@
 	vipw
 
 # id and groups are from gnu, sulogin from sysvinit
-noinst_PROGRAMS = id sulogin
+noinst_PROGRAMS = id sulogin su
 
-suidbins       = su
+suidbins       =
 suidubins      = chage chfn chsh expiry gpasswd newgrp passwd
 
 LDADD 	       = $(top_builddir)/libmisc/libmisc.a \
diff -urN shadow-4.0.14/src/Makefile.in shadow/src/Makefile.in
--- shadow-4.0.14/src/Makefile.in	2006-01-03 11:12:33.000000000 -0500
+++ shadow/src/Makefile.in	2006-02-03 00:02:57.000000000 -0500
@@ -36,7 +36,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-bin_PROGRAMS = groups$(EXEEXT) login$(EXEEXT) su$(EXEEXT)
+bin_PROGRAMS = groups$(EXEEXT) login$(EXEEXT)
 ubin_PROGRAMS = faillog$(EXEEXT) lastlog$(EXEEXT) chage$(EXEEXT) \
 	chfn$(EXEEXT) chsh$(EXEEXT) expiry$(EXEEXT) gpasswd$(EXEEXT) \
 	newgrp$(EXEEXT) passwd$(EXEEXT)
@@ -359,7 +359,7 @@
 	-I${top_srcdir}/lib \
 	-I$(top_srcdir)/libmisc
 
-suidbins = su
+suidbins =
 suidubins = chage chfn chsh expiry gpasswd newgrp passwd
 LDADD = $(top_builddir)/libmisc/libmisc.a \
 		 $(top_builddir)/lib/libshadow.la
