--- configure~	2007-04-29 01:16:11.000000000 +0200
+++ configure	2007-05-08 21:55:20.000000000 +0200
@@ -29816,23 +29816,23 @@
   else
      pkg_config_min_version=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $pkg_config_min_version; then
-        { echo "$as_me:$LINENO: checking for libgoffice-0.3 >= 0.3.0" >&5
-echo $ECHO_N "checking for libgoffice-0.3 >= 0.3.0... $ECHO_C" >&6; }
+        { echo "$as_me:$LINENO: checking for libgoffice-0.4 >= 0.3.0" >&5
+echo $ECHO_N "checking for libgoffice-0.4 >= 0.3.0... $ECHO_C" >&6; }
 
-        if $PKG_CONFIG --silence-errors --exists "libgoffice-0.3 >= 0.3.0" ; then
+        if $PKG_CONFIG --silence-errors --exists "libgoffice-0.4 >= 0.3.0" ; then
             { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
             succeeded=yes
 
             { echo "$as_me:$LINENO: checking GOFFICE_CFLAGS" >&5
 echo $ECHO_N "checking GOFFICE_CFLAGS... $ECHO_C" >&6; }
-            GOFFICE_CFLAGS=`$PKG_CONFIG --cflags "libgoffice-0.3 >= 0.3.0"`
+            GOFFICE_CFLAGS=`$PKG_CONFIG --cflags "libgoffice-0.4 >= 0.3.0"`
             { echo "$as_me:$LINENO: result: $GOFFICE_CFLAGS" >&5
 echo "${ECHO_T}$GOFFICE_CFLAGS" >&6; }
 
             { echo "$as_me:$LINENO: checking GOFFICE_LIBS" >&5
 echo $ECHO_N "checking GOFFICE_LIBS... $ECHO_C" >&6; }
-            GOFFICE_LIBS=`$PKG_CONFIG --libs "libgoffice-0.3 >= 0.3.0"`
+            GOFFICE_LIBS=`$PKG_CONFIG --libs "libgoffice-0.4 >= 0.3.0"`
             { echo "$as_me:$LINENO: result: $GOFFICE_LIBS" >&5
 echo "${ECHO_T}$GOFFICE_LIBS" >&6; }
         else
@@ -29842,7 +29842,7 @@
             GOFFICE_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            GOFFICE_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgoffice-0.3 >= 0.3.0" 2>&1`
+            GOFFICE_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgoffice-0.4 >= 0.3.0" 2>&1`
 
         fi
 
