--- acctproc.c.orig
+++ acctproc.c
@@ -399,7 +399,7 @@
 			goto out;
 		}
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2060000fL)
 		ec_sig_r = ec_sig->r;
 		ec_sig_s = ec_sig->s;
 #else
--- http.c.orig	2020-05-24 09:32:56.000000000 -0700
+++ http.c	2020-05-29 13:48:43.502662599 -0700
@@ -170,7 +170,7 @@
 		goto err;
 	}
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2060000fL)
 	const long blacklist_versions =   SSL_OP_NO_SSLv2 \
 	                                | SSL_OP_NO_SSLv3 \
 	                                | SSL_OP_NO_TLSv1 \
