From f85024e93a8b3292433b27a074fefafea0f979d4 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sun, 26 May 2013 13:35:01 +0200
Subject: [PATCH] use correct library suffix in install target

---
 project.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/project.mak b/project.mak
index 6f43306..865b7ea 100644
--- a/project.mak
+++ b/project.mak
@@ -109,7 +109,7 @@ endif
 
 ifeq ($(INSTMETH),findlib)
 install:
-	$(OCAMLFIND) install -destdir "$(INSTALLDIR)" zarith META $(TOINSTALL) dllzarith.dll
+	$(OCAMLFIND) install -destdir "$(INSTALLDIR)" zarith META $(TOINSTALL) dllzarith.$(DLLSUFFIX)
 
 uninstall:
 	$(OCAMLFIND) remove -destdir "$(INSTALLDIR)" zarith
-- 
1.8.2.3

