From 971d2ec8978152234d2a3fd99522ae06dc7b4782 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 23 Sep 2017 00:38:05 +0200
Subject: [PATCH] allow template-haskell version 2.12

---
 tagged.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tagged.cabal b/tagged.cabal
index 6bb6676..8eb3849 100644
--- a/tagged.cabal
+++ b/tagged.cabal
@@ -57,7 +57,7 @@ library
 
   if impl(ghc>=7.6)
     exposed-modules: Data.Proxy.TH
-    build-depends: template-haskell >= 2.8 && < 2.12
+    build-depends: template-haskell >= 2.8 && < 2.13
 
   if flag(deepseq)
     build-depends: deepseq >= 1.1 && < 1.5
-- 
2.14.1

