--- src/keymap.c.orig	2004-05-25 20:30:22.074925968 -0500
+++ src/keymap.c	2004-05-25 20:30:50.930539248 -0500
@@ -1149,7 +1149,7 @@
 			tmp = g_strdup(termcap_special);
 			cap = tgetstr(tmp, &cap);
 		}
-		if ((cap == NULL) && (strstr(terminal, "xterm") == 0)) {
+		if ((cap == NULL) && (strstr(terminal, "xterm") == terminal)) {
 			/* try, try again */
 			if (tgetent(ncurses_buffer, "xterm-xfree86") == 1) {
 				cap = ncurses_area;
