diff --color -urp audit-3.0.9.orig/bindings/swig/src/auditswig.i audit-3.0.9/bindings/swig/src/auditswig.i
--- audit-3.0.9.orig/bindings/swig/src/auditswig.i	2022-08-29 16:22:06.000000000 -0500
+++ audit-3.0.9/bindings/swig/src/auditswig.i	2022-10-29 13:26:54.888006805 -0500
@@ -39,7 +39,7 @@ signed
 #define __attribute(X) /*nothing*/
 typedef unsigned __u32;
 typedef unsigned uid_t;
-%include "/usr/include/linux/audit.h"
+%include "../lib/libaudit.h"
 #define __extension__ /*nothing*/
 %include <stdint.i>
 %include "../lib/libaudit.h"
diff --color -urp audit-3.0.9.orig/lib/audit.h audit-3.0.9/lib/audit.h
--- audit-3.0.9.orig/lib/audit.h	2022-10-29 13:24:59.853245373 -0500
+++ audit-3.0.9/lib/audit.h	2022-10-29 13:27:27.720241256 -0500
@@ -514,7 +514,7 @@ struct audit_rule_data {
 	__u32		values[AUDIT_MAX_FIELDS];
 	__u32		fieldflags[AUDIT_MAX_FIELDS];
 	__u32		buflen;	/* total length of string fields */
-	char		buf[];	/* string fields buffer */
+	char		buf[0];	/* string fields buffer */
 };
 
 #endif /* _LINUX_AUDIT_H_ */
diff --color -urp audit-3.0.9.orig/lib/libaudit.h audit-3.0.9/lib/libaudit.h
--- audit-3.0.9.orig/lib/libaudit.h	2022-08-29 16:22:06.000000000 -0500
+++ audit-3.0.9/lib/libaudit.h	2022-10-29 13:28:06.161512389 -0500
@@ -27,7 +27,7 @@
 #include <stdint.h>
 #include <sys/socket.h>
 #include <linux/netlink.h>
-#include <linux/audit.h>
+#include "audit.h"
 #include <stdarg.h>
 #include <syslog.h>
 #ifndef __attr_access
