--- _mysql.c.orig	2014-11-26 14:41:58.217304055 +0300
+++ _mysql.c	2014-11-26 14:44:08.990008082 +0300
@@ -26,6 +26,11 @@
 PERFORMANCE OF THIS SOFTWARE.
 */
 
+#if defined(MS_WINDOWS)
+#include <config-win.h>
+#else
+#include "my_config.h"
+#endif
 #include "Python.h"
 #if PY_MAJOR_VERSION >= 3
 #define IS_PY3K
@@ -38,11 +43,6 @@
 #endif
 #include "pymemcompat.h"
 #include "structmember.h"
-#if defined(MS_WINDOWS)
-#include <config-win.h>
-#else
-#include "my_config.h"
-#endif
 #include "mysql.h"
 #include "mysqld_error.h"
 #include "errmsg.h"
