From e92b260635e4fe542c0c606f2a2484334e8240ca Mon Sep 17 00:00:00 2001
From: Johannes Huber <johu@gmx.de>
Date: Wed, 13 Jul 2011 18:32:16 +0200
Subject: [PATCH] fix build for disabled qt4 option

---
 lib/CMakeLists.txt |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index ba07497..5c6ae79 100644
--- a/strigiclient/lib/CMakeLists.txt
+++ b/strigiclient/lib/CMakeLists.txt
@@ -1,4 +1,7 @@
 if(NOT WIN32)
     add_subdirectory(htmlgui)
 endif(NOT WIN32)
-add_subdirectory(searchclient)
+
+if(QT4_FOUND)
+ add_subdirectory(searchclient)
+endif(QT4_FOUND)
-- 
1.7.3.4


