diff -uNr strigi-0.5.1.cp/src/streams/mailinputstream.cpp strigi-0.5.1/src/streams/mailinputstream.cpp
--- strigi-0.5.1.cp/src/streams/mailinputstream.cpp	2007-05-02 20:33:11.000000000 +0200
+++ strigi-0.5.1/src/streams/mailinputstream.cpp	2007-05-04 15:20:54.000000000 +0200
@@ -29,6 +29,7 @@
 #include "base64inputstream.h"
 #include <cstring>
 #include <sstream>
+#include <iconv.h>
 
 using namespace std;
 using namespace Strigi;
