--- Makefile.orig	2013-10-23 12:10:46.000000000 +0400
+++ Makefile	2014-06-25 22:45:47.565785309 +0400
@@ -76,8 +76,10 @@
 	mkdir -p $(MANDIR)
 	install -m 644 pptp.8 $(MANDIR)
 	install -m 644 pptpsetup.8 $(MANDIR)
-	mkdir -p $(PPPDIR)
-	install -m 644 options.pptp $(PPPDIR)
+	if [ ! -f $(PPPDIR)/options.pptp ]; then \
+		mkdir -p $(PPPDIR) ; \
+		install -m 644 options.pptp $(PPPDIR) ; \
+	fi
 
 uninstall:
 	$(RM) $(BINDIR)/pptp $(MANDIR)/pptp.8
