From 938b2f938b8ca4308b596bc8b62fa71c089f3921 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 10 Jun 2021 21:44:24 +0200
Subject: [PATCH] Add missing include

---
 glib/gstrfuncs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/glib/gstrfuncs.c b/glib/gstrfuncs.c
index b6ff60f..716a562 100644
--- a/glib/gstrfuncs.c
+++ b/glib/gstrfuncs.c
@@ -28,6 +28,7 @@
 
 #include "config.h"
 
+#include <limits.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
-- 
2.32.0

