--- lib/ansible/modules/cloud/docker/docker_container.py.orig
+++ lib/ansible/modules/cloud/docker/docker_container.py
@@ -1310,7 +1310,6 @@
 
         # Map parameters to container inspect results
         config_mapping = dict(
-            auto_remove=host_config.get('AutoRemove'),
             expected_cmd=config.get('Cmd'),
             domainname=config.get('Domainname'),
             hostname=config.get('Hostname'),
@@ -1360,6 +1359,9 @@
             working_dir=host_config.get('WorkingDir')
         )
 
+        if HAS_DOCKER_PY_2 or HAS_DOCKER_PY_3:
+            config_mapping.update(auto_remove=host_config.get('AutoRemove'))
+
         differences = []
         for key, value in config_mapping.items():
             self.log('check differences %s %s vs %s' % (key, getattr(self.parameters, key), str(value)))
