--- configure.orig	2015-04-24 07:48:32.000000000 +0300
+++ configure	2015-04-24 07:51:07.626025323 +0300
@@ -2932,7 +2932,7 @@
     "libdirac_decoder/dirac_parser.h libdirac_encoder/dirac_encoder.h"  \
     "dirac_decoder_init dirac_encoder_init"
 enabled libfaac    && require2 libfaac "stdint.h faac.h" faacEncGetVersion -lfaac
-enabled libfreetype && require_pkg_config freetype2 "ft2build.h freetype/freetype.h" FT_Init_FreeType
+enabled libfreetype && require_pkg_config freetype2 "ft2build.h freetype.h" FT_Init_FreeType
 enabled libgsm     && require  libgsm gsm/gsm.h gsm_create -lgsm
 enabled libmp3lame && require  "libmp3lame >= 3.98.3" lame/lame.h lame_set_VBR_quality -lmp3lame
 enabled libnut     && require  libnut libnut.h nut_demuxer_init -lnut
--- libavfilter/vf_drawtext.c.orig	2013-10-06 20:18:09.000000000 +0400
+++ libavfilter/vf_drawtext.c	2015-04-24 08:00:29.846711104 +0300
@@ -41,7 +41,6 @@
 #undef time
 
 #include <ft2build.h>
-#include <freetype/config/ftheader.h>
 #include FT_FREETYPE_H
 #include FT_GLYPH_H
 
