From ee21e590076fe7b2a26713b274c17e278d12e09b Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sun, 2 Aug 2015 18:05:54 +0200
Subject: [PATCH] depend on base < 4.9

---
 split.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/split.cabal b/split.cabal
index 695306c..11c86df 100644
--- a/split.cabal
+++ b/split.cabal
@@ -51,7 +51,7 @@ Source-repository head
 
 Library
   ghc-options:       -Wall
-  build-depends:     base <4.8
+  build-depends:     base <4.9
   exposed-modules:   Data.List.Split, Data.List.Split.Internals
   default-language:  Haskell2010
   Hs-source-dirs:    src
-- 
2.4.1

