From 2532e07567ed6160f7303548354c1d6a7a14165d Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 20:08:16 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 split.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/split.cabal b/split.cabal
index a60b284..bf3a475 100644
--- a/split.cabal
+++ b/split.cabal
@@ -51,7 +51,7 @@ Source-repository head
 
 Library
   ghc-options:       -Wall
-  build-depends:     base <4.12
+  build-depends:     base
   exposed-modules:   Data.List.Split, Data.List.Split.Internals
   default-language:  Haskell2010
   Hs-source-dirs:    src
-- 
2.19.1

