--- Makefile.orig	2005-09-11 22:02:50.000000000 -0500
+++ Makefile	2005-09-11 22:07:31.000000000 -0500
@@ -1,19 +1,19 @@
 ##################################################################
 # Directories
 # Install dirs
-INSTALL_PREFIX = /usr/local
+INSTALL_PREFIX = /usr
 
 # Full default path is /usr/local/bin
 APG_BIN_DIR = /bin
 
 # Full default path is /usr/local/man/man1
-APG_MAN_DIR = /man/man1
+APG_MAN_DIR = /share/man/man1
 
 # Full default path is /usr/local/sbin
 APGD_BIN_DIR = /sbin
 
 # Full default path is /usr/local/man/man8
-APGD_MAN_DIR = /man/man8
+APGD_MAN_DIR = /share/man/man8
 
 # You should not edit 2 lines below
 APGBFM_CLIBS = -lm
@@ -44,10 +44,10 @@
 # If you want to use cracklib for password quality check then you
 # must uncomment the folowing 4 lines (you must not do this for WIN32)
 #
-#CRACKLIB_DICTPATH = "/usr/local/lib/pw_dict"
-#STANDALONE_OPTIONS += -DAPG_USE_CRACKLIB '-DCRACKLIB_DICTPATH=${CRACKLIB_DICTPATH}' -I/usr/local/include -L/usr/local/lib
-#CLISERV_OPTIONS += -DAPG_USE_CRACKLIB '-DCRACKLIB_DICTPATH=${CRACKLIB_DICTPATH}' -I/usr/local/include -L/usr/local/lib
-#APG_CLIBS += -lcrack
+CRACKLIB_DICTPATH = "/usr/lib/cracklib/pw_dict"
+STANDALONE_OPTIONS += -DAPG_USE_CRACKLIB '-DCRACKLIB_DICTPATH=${CRACKLIB_DICTPATH}' -I/usr/local/include -L/usr/local/lib
+CLISERV_OPTIONS += -DAPG_USE_CRACKLIB '-DCRACKLIB_DICTPATH=${CRACKLIB_DICTPATH}' -I/usr/local/include -L/usr/local/lib
+APG_CLIBS += -lcrack
 
 ##################################################################
 # Support for ANSI X9.17/SHA1 PRNG
@@ -81,7 +81,7 @@
 # Linux
 #
 # Uncoment line below for LINUX
-#APG_CS_CLIBS += -lnsl
+APG_CS_CLIBS += -lnsl
 
 ####################################################################
 # Solaris
