From 2d610a715e14bc2960a636cc2f35b32edabb0bb6 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 15 Oct 2017 00:36:45 +0200
Subject: [PATCH] gcc 7

---
 Makefile               | 2 +-
 platform/linux/linux.c | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0fb5b0d..f824484 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-override CFLAGS+=-Wall -Werror -D_GNU_SOURCE -g
+override CFLAGS+=-Wall -Werror -D_GNU_SOURCE -Wno-error=format-truncation
 OBJS=reptyr.o reallocarray.o attach.o
 UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S),Linux)
diff --git a/platform/linux/linux.c b/platform/linux/linux.c
index 45a6ff1..7d7aafb 100644
--- a/platform/linux/linux.c
+++ b/platform/linux/linux.c
@@ -27,6 +27,7 @@
 #include "../../reptyr.h"
 #include "../../ptrace.h"
 #include <stdint.h>
+#include <sys/sysmacros.h>
 
 int parse_proc_stat(int statfd, struct proc_stat *out) {
     char buf[1024];
-- 
2.14.2

