--- frontend/gtkmm/src/GtkUtil.hh~	2005-09-23 21:22:33.000000000 +0200
+++ frontend/gtkmm/src/GtkUtil.hh	2006-03-05 11:06:19.000000000 +0100
@@ -56,11 +56,11 @@
   create_label(std::string text, bool bold);
 
   static Gtk::Widget *
-  GtkUtil::create_label_with_tooltip(std::string text,
+  create_label_with_tooltip(std::string text,
                                      std::string tooltip);
 
   static EventImage *
-  GtkUtil::create_image_with_tooltip(std::string file,
+  create_image_with_tooltip(std::string file,
                                      std::string tooltip);
 
   static Gtk::Widget *
@@ -84,7 +84,7 @@
   static void set_wmclass(Gtk::Window &window, std::string class_postfix);
   static void center_window(Gtk::Window &window, HeadInfo &head);
 
-  static bool GtkUtil::has_button_images();
+  static bool has_button_images();
 };
 
 #endif // GTKMMGUI_HH
