diff -Naur /lib/arch/CArchDaemonUnix.cpp lib/arch/CArchDaemonUnix.cpp
--- lib/arch/CArchDaemonUnix.cpp	2005-04-23 23:02:12.000000000 -0400
+++ lib/arch/CArchDaemonUnix.cpp	2010-03-13 05:30:37.949473285 -0500
@@ -19,6 +19,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <cstdlib>
 
 //
 // CArchDaemonUnix
diff -Naur /lib/arch/CMultibyte.cpp lib/arch/CMultibyte.cpp
--- lib/arch/CMultibyte.cpp	2005-11-29 23:33:24.000000000 -0500
+++ lib/arch/CMultibyte.cpp	2010-03-13 05:51:56.399633465 -0500
@@ -19,6 +19,7 @@
 #include "CArch.h"
 #include <limits.h>
 #include <string.h>
+#include <cstdlib>
 #if HAVE_LOCALE_H
 #	include <locale.h>
 #endif
diff -Naur /lib/base/CStringUtil.cpp lib/base/CStringUtil.cpp
--- lib/base/CStringUtil.cpp	2005-12-14 12:25:20.000000000 -0500
+++ lib/base/CStringUtil.cpp	2010-03-13 05:37:56.222451090 -0500
@@ -20,6 +20,8 @@
 #include <cstdio>
 #include <cstdlib>
 #include <algorithm>
+#include <climits>
+#include <cstring>
 
 //
 // CStringUtil
diff -Naur /lib/net/CTCPSocket.cpp lib/net/CTCPSocket.cpp
--- lib/net/CTCPSocket.cpp	2006-04-01 20:47:03.000000000 -0500
+++ lib/net/CTCPSocket.cpp	2010-03-13 05:54:29.955407790 -0500
@@ -24,6 +24,7 @@
 #include "CArch.h"
 #include "XArch.h"
 #include <string.h>
+#include <cstdlib>
 
 //
 // CTCPSocket
diff -Naur /lib/synergy/CPacketStreamFilter.cpp lib/synergy/CPacketStreamFilter.cpp
--- lib/synergy/CPacketStreamFilter.cpp	2005-04-23 23:02:16.000000000 -0400
+++ lib/synergy/CPacketStreamFilter.cpp	2010-03-13 05:57:52.839459079 -0500
@@ -16,6 +16,7 @@
 #include "IEventQueue.h"
 #include "CLock.h"
 #include "TMethodEventJob.h"
+#include <cstring>
 
 //
 // CPacketStreamFilter
diff -Naur /lib/synergy/IKeyState.cpp lib/synergy/IKeyState.cpp
--- lib/synergy/IKeyState.cpp	2006-03-22 00:40:27.000000000 -0500
+++ lib/synergy/IKeyState.cpp	2010-03-13 06:02:51.155507908 -0500
@@ -14,6 +14,7 @@
 
 #include "IKeyState.h"
 #include <string.h>
+#include <cstdlib>
 
 //
 // IKeyState
diff -Naur /lib/synergy/IPrimaryScreen.cpp lib/synergy/IPrimaryScreen.cpp
--- lib/synergy/IPrimaryScreen.cpp	2006-03-22 00:40:27.000000000 -0500
+++ lib/synergy/IPrimaryScreen.cpp	2010-03-13 06:03:26.740507973 -0500
@@ -13,6 +13,7 @@
  */
 
 #include "IPrimaryScreen.h"
+#include <cstdlib>
 
 //
 // IPrimaryScreen
diff -Naur /lib/synergy/OptionTypes.h lib/synergy/OptionTypes.h
--- lib/synergy/OptionTypes.h	2005-08-03 22:08:25.000000000 -0400
+++ lib/synergy/OptionTypes.h	2010-03-13 06:07:26.538642237 -0500
@@ -17,6 +17,8 @@
 
 #include "BasicTypes.h"
 #include "stdvector.h"
+#include <cstdlib>
+#include <cstring>
 
 //! Option ID
 /*!
