From 8d80c2b7902445399f011b3f430095c74da0e39f Mon Sep 17 00:00:00 2001
From: Frank Morgner <frankmorgner@gmail.com>
Date: Fri, 16 Dec 2022 11:56:28 +0100
Subject: [PATCH 2/4] fixed compatibility with LibreSSL 3.7.0

Origin: Upstream
Upstream-Status: Backport [https://github.com/OpenSC/OpenSC/pull/2665/]
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 src/libopensc/sc-ossl-compat.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libopensc/sc-ossl-compat.h b/src/libopensc/sc-ossl-compat.h
index 8c0f96701c..4425da93f3 100644
--- a/src/libopensc/sc-ossl-compat.h
+++ b/src/libopensc/sc-ossl-compat.h
@@ -53,9 +53,11 @@ extern "C" {
 #define EVP_sha3_256()                          (NULL)
 #define EVP_sha3_384()                          (NULL)
 #define EVP_sha3_512()                          (NULL)
+#if LIBRESSL_VERSION_NUMBER < 0x3070000fL
 #define EVP_PKEY_new_raw_public_key(t, e, p, l) (NULL)
 #define EVP_PKEY_get_raw_public_key(p, pu, l)   (0)
 #endif
+#endif
 
 /* OpenSSL 1.1.1 has FIPS_mode function */
 #if OPENSSL_VERSION_NUMBER >= 0x30000000L
