--- libunicapgtk/unicapgtk_property_dialog.c.orig	2008-08-06 18:36:20.000000000 +0200
+++ libunicapgtk/unicapgtk_property_dialog.c	2008-08-27 22:11:53.000000000 +0200
@@ -111,8 +111,7 @@
 						      NULL, NULL,
 						      0, 99999,
 						      100,
-						      G_PARAM_READWRITE |
-                                                      G_PARAM_CONSTRUCT));
+						      G_PARAM_READWRITE ));
 
 }
 
