diff -Naur xulrunner-1.9.2.8.orig/mozilla-1.9.2/modules/libpr0n/encoders/png/nsPNGEncoder.cpp xulrunner-1.9.2.8/mozilla-1.9.2/modules/libpr0n/encoders/png/nsPNGEncoder.cpp
--- xulrunner-1.9.2.8.orig/mozilla-1.9.2/modules/libpr0n/encoders/png/nsPNGEncoder.cpp	2010-07-22 23:55:10.000000000 +0200
+++ xulrunner-1.9.2.8/mozilla-1.9.2/modules/libpr0n/encoders/png/nsPNGEncoder.cpp	2010-08-11 09:05:38.000000000 +0200
@@ -135,7 +135,7 @@
 
   // initialize
   mPNG = png_create_write_struct(PNG_LIBPNG_VER_STRING,
-                                 png_voidp_NULL,
+                                 nsnull,
                                  ErrorCallback,
                                  ErrorCallback);
   if (! mPNG)
