--- OpenThreads/Make/makerules.orig	2011-03-07 09:08:55.464739676 +0100
+++ OpenThreads/Make/makerules	2011-03-07 09:09:22.167351881 +0100
@@ -71,11 +71,11 @@
 
 $(EXEC:=.dbg) :
 	@[ -d $(DEBUGDIR) ] || mkdir $(DEBUGDIR)
-        @cd $(DEBUGDIR); \
+	@cd $(DEBUGDIR); \
               $(MAKE) -f ../GNUmakefile "DOF=$(DBGF)"  TOPDIR=../${TOPDIR} \
                         THISDIR=../$(THISDIR)\
                                   MAKEDEPEND=Makedepend $(EXEC)
-        @$(MAKE) LNSRC=$(DEBUGDIR)/$(EXEC) LNDEST=$(BININST)/$(OS)$(ARCH)/$(EXEC) __link
+	@$(MAKE) LNSRC=$(DEBUGDIR)/$(EXEC) LNDEST=$(BININST)/$(OS)$(ARCH)/$(EXEC) __link
 
 
 $(LIB:=.dbg) :
