--- xmenu.c.orig	2021-04-11 07:03:29.000000000 -0700
+++ xmenu.c	2022-07-29 19:26:12.652748944 -0700
@@ -1390,7 +1390,7 @@
 			case XK_1: case XK_2: case XK_3: case XK_4: case XK_5: case XK_6: case XK_7: case XK_8: case XK_9:
 				item = itemcycle(currmenu, ITEMFIRST);
 				lastitem = itemcycle(currmenu, ITEMLAST);
-				for (int i = ksym - XK_1; i > 0 && item != lastitem; i--) {
+				for (i = ksym - XK_1; i > 0 && item != lastitem; i--) {
 					currmenu->selected = item;
 					item = itemcycle(currmenu, ITEMNEXT);
 				}
