diff -Naur libofa-0.9.3.orig/examples/example.cpp libofa-0.9.3/examples/example.cpp
--- libofa-0.9.3.orig/examples/example.cpp	2008-03-22 16:33:04.564088179 -0700
+++ libofa-0.9.3/examples/example.cpp	2008-03-22 16:37:08.077087589 -0700
@@ -7,6 +7,7 @@
 
 -------------------------------------------------------------------*/
 
+#include <cstring>
 #include "protocol.h"
 
 AudioData* loadWaveFile(char *file);
diff -Naur libofa-0.9.3.orig/examples/protocol.cpp libofa-0.9.3/examples/protocol.cpp
--- libofa-0.9.3.orig/examples/protocol.cpp	2008-03-22 16:33:04.564088179 -0700
+++ libofa-0.9.3/examples/protocol.cpp	2008-03-22 16:38:42.205088364 -0700
@@ -9,6 +9,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string>
+#include <cstring>
 #include <map>
 #include <expat.h>
 #include <curl/curl.h>
diff -Naur libofa-0.9.3.orig/lib/signal_op.cpp libofa-0.9.3/lib/signal_op.cpp
--- libofa-0.9.3.orig/lib/signal_op.cpp	2008-03-22 16:33:04.565088992 -0700
+++ libofa-0.9.3/lib/signal_op.cpp	2008-03-22 16:36:53.076089354 -0700
@@ -13,6 +13,7 @@
 
 
 #include <math.h>
+#include <cstdlib>
 #include "signal_op.h"
 #include "AFLIB/aflibConverter.h"
 #include "error_op.h"
