diff -uNr libofa-0.9.4/examples/example.cpp libofa-0.9.4.cp/examples/example.cpp
--- libofa-0.9.4/examples/example.cpp	2008-07-21 23:27:55.000000000 +0200
+++ libofa-0.9.4.cp/examples/example.cpp	2009-07-04 18:07:02.000000000 +0200
@@ -9,6 +9,8 @@
 
 #include <ctype.h>
 #include "protocol.h"
+#include <cstring>
+#include <stdio.h>
 
 AudioData* loadWaveFile(char *file);
 AudioData* loadDataUsingLAME(char *file);
diff -uNr libofa-0.9.4/examples/protocol.cpp libofa-0.9.4.cp/examples/protocol.cpp
--- libofa-0.9.4/examples/protocol.cpp	2008-07-21 23:27:55.000000000 +0200
+++ libofa-0.9.4.cp/examples/protocol.cpp	2009-07-04 18:02:07.000000000 +0200
@@ -9,6 +9,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string>
+#include <cstring>
 #include <map>
 #include <expat.h>
 #include <curl/curl.h>
diff -uNr libofa-0.9.4/lib/signal_op.cpp libofa-0.9.4.cp/lib/signal_op.cpp
--- libofa-0.9.4/lib/signal_op.cpp	2008-07-21 23:27:52.000000000 +0200
+++ libofa-0.9.4.cp/lib/signal_op.cpp	2009-07-04 18:02:07.000000000 +0200
@@ -14,6 +14,7 @@
 
 #include <stdlib.h>
 #include <math.h>
+#include <cstdlib>
 #include "signal_op.h"
 #include "AFLIB/aflibConverter.h"
 #include "error_op.h"
