--- lib/JAMA/tnt_math_utils.h	2007-08-07 23:11:47.000000000 +0200
+++ tnt_math_utils.h	2007-08-07 23:03:47.000000000 +0200
@@ -19,6 +19,16 @@
 
 namespace TNT
 {
+
+/**
+	@returns the absolute value of a real (no-complex) scalar.
+*/
+template <class Real>
+Real abs(const Real &a)
+{
+	return  (a > 0 ? a : -a);
+}
+
 /**
 	@returns hypotenuse of real (non-complex) scalars a and b by 
 	avoiding underflow/overflow
@@ -55,16 +65,7 @@
 	return  a > b ? a : b;
 }
 */
-
-/**
-	@returns the absolute value of a real (no-complex) scalar.
-*/
-template <class Real>
-Real abs(const Real &a)
-{
-	return  (a > 0 ? a : -a);
 }
 
-}
 #endif
 /* MATH_UTILS_H */
