From 3af71fc918c6223d7e2d052b767842e6348c2995 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 19 Jun 2021 18:51:15 +0200
Subject: [PATCH] Relax versions requirement on astroid

---
 setup.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.cfg b/setup.cfg
index 4939385..4257f65 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -41,7 +41,7 @@ project_urls =
 [options]
 packages = find:
 install_requires = 
-	astroid==2.5.6
+	astroid>=2.5.6
 	isort>=4.2.5,<6
 	mccabe>=0.6,<0.7
 	toml>=0.7.1
-- 
2.32.0

