diff -Nuar setup.orig/Linux/oss/build/install.sh setup/Linux/oss/build/install.sh
--- setup.orig/Linux/oss/build/install.sh	2008-11-20 07:57:50.000000000 +0100
+++ setup/Linux/oss/build/install.sh	2008-11-20 07:58:50.000000000 +0100
@@ -336,7 +336,7 @@
 
 # Recompile libflashsupport.so if possible. Otherwise use the precompiled
 # version.
-(cd $OSSLIBDIR/lib;gcc -m32 -shared -fPIC -O2 -Wall -Werror flashsupport.c -o $OSSLIBDIR/lib/libflashsupport.so) > /dev/null 2>&1
+(cd $OSSLIBDIR/lib;gcc -m32 -shared -fPIC -O2 -Wall flashsupport.c -o $OSSLIBDIR/lib/libflashsupport.so) > /dev/null 2>&1
 
 if test ! -f $OSSLIBDIR/etc/userdefs
 then
diff -Nuar setup.orig/srcconf_linux.inc setup/srcconf_linux.inc
--- setup.orig/srcconf_linux.inc	2008-11-20 07:57:50.000000000 +0100
+++ setup/srcconf_linux.inc	2008-11-20 07:58:11.000000000 +0100
@@ -26,7 +26,7 @@
   if (system(cmd)==0)
      {
 	if (do_warning_checks)
-  	   strcpy(conf->OSflags, "-fno-common -fno-stack-protector -Wall -Werror");
+  	   strcpy(conf->OSflags, "-fno-common -fno-stack-protector -Wall");
 else
   	   strcpy(conf->OSflags, "-fno-common -fno-stack-protector");
 	unlink("srcconf.o");
@@ -34,7 +34,7 @@
   else
      {
 	if (do_warning_checks)
-  	   strcpy(conf->OSflags, "-fno-common -Wall -Werror");
+  	   strcpy(conf->OSflags, "-fno-common -Wall");
 	else
   	   strcpy(conf->OSflags, "-fno-common");
      }
