From c3dbf0db4fc575a67e2a4922a03f83d876b6cb08 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 21 Jan 2018 17:23:09 +0100
Subject: [PATCH] fix dependency

---
 tasty-rerun.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tasty-rerun.cabal b/tasty-rerun.cabal
index b950615..ae4640b 100644
--- a/tasty-rerun.cabal
+++ b/tasty-rerun.cabal
@@ -80,7 +80,7 @@ library
     split >= 0.1 && < 0.3,
     stm >= 2.4.2,
     tagged >= 0.7 && <0.9,
-    tasty >=0.10 && <1.0,
+    tasty >=0.10 && <1.1,
     transformers >= 0.3.0.0
   hs-source-dirs:      src
   default-language:    Haskell2010
-- 
2.15.1

