From 9219762779134112c636f30c180aaf91837bb4e2 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 3 Dec 2017 17:55:34 +0100
Subject: [PATCH] fix CFLAGS

---
 Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 535138b..37e33b4 100644
--- a/Makefile
+++ b/Makefile
@@ -3,14 +3,13 @@ BINDIR?=	$(PREFIX)/bin
 MANDIR?=	$(PREFIX)/share/man/man8
 
 CC?=		cc
-CFLAGS+=	-Wall -O2 -g -std=c99
+CFLAGS+=	-Wall -std=c99
 
 all: vmtouch vmtouch.8
 
 .PHONY: all install clean uninstall
 
 vmtouch: vmtouch.c
-	${CC} ${CFLAGS} -o vmtouch vmtouch.c
 
 vmtouch.8: vmtouch.pod
 	pod2man --section 8 --center "System Manager's Manual" --release " " vmtouch.pod > vmtouch.8
-- 
2.15.1

