diff -Naur libgui-0.0.4.orig/gui/commonlistbox.h libgui-0.0.4/gui/commonlistbox.h
--- libgui-0.0.4.orig/gui/commonlistbox.h	2005-07-28 08:32:22.660747250 -0700
+++ libgui-0.0.4/gui/commonlistbox.h	2005-07-28 08:37:35.088272750 -0700
@@ -46,10 +46,10 @@
     CSimpleList<CIdString *> m_Strings;
     bool m_bInternalSelect;
  protected:
-    virtual void SelectionChanged() = NULL;
-    virtual Widget InternalList() = NULL;
+    virtual void SelectionChanged() = 0;
+    virtual Widget InternalList() = 0;
     virtual void InternalCreate(unsigned long lStyle);
-    virtual void InternalListCreate(unsigned long lStyle)=NULL;
+    virtual void InternalListCreate(unsigned long lStyle)=0;
 		
     inline Widget List() { return m_List; }
 		
diff -Naur libgui-0.0.4.orig/gui/dialog.h libgui-0.0.4/gui/dialog.h
--- libgui-0.0.4.orig/gui/dialog.h	2005-07-28 08:32:22.664747500 -0700
+++ libgui-0.0.4/gui/dialog.h	2005-07-28 08:37:35.088272750 -0700
@@ -81,7 +81,7 @@
  public:
     enum {IDOK=-1, IDCANCEL=-2, IDYES=-3, IDNO=-4};
     virtual Widget Container() const;
-    virtual Widget Frame()=NULL;
+    virtual Widget Frame()=0;
     inline Widget Separator() { return m_wdgSeparator; }
     void SetCaption(const char *strCaption);
     const char *Caption();
diff -Naur libgui-0.0.4.orig/gui/graphicalobjectcontext.h libgui-0.0.4/gui/graphicalobjectcontext.h
--- libgui-0.0.4.orig/gui/graphicalobjectcontext.h	2005-07-28 08:32:22.660747250 -0700
+++ libgui-0.0.4/gui/graphicalobjectcontext.h	2005-07-28 08:37:35.092273000 -0700
@@ -42,7 +42,7 @@
  public:
     CGraphicalObjectContext(void *pObject);
 
-    virtual GOC Type()=NULL;
+    virtual GOC Type()=0;
 
     inline void *Object() { return m_pObject; }
  private:
diff -Naur libgui-0.0.4.orig/gui/guielement.h libgui-0.0.4/gui/guielement.h
--- libgui-0.0.4.orig/gui/guielement.h	2005-07-28 08:32:22.664747500 -0700
+++ libgui-0.0.4/gui/guielement.h	2005-07-28 08:43:13.217404500 -0700
@@ -58,7 +58,7 @@
     CGuiElement(unsigned long lStyle = 0L, int nId=0);
 		
     // STH sandalle SF virtual void InternalCreate(unsigned long lStyle) = 0;
-    virtual void InternalCreate(unsigned long lStyle) = NULL;
+    virtual void InternalCreate(unsigned long lStyle) = 0;
     virtual void InternalDestroy();
     virtual void PostCreate();
 
diff -Naur libgui-0.0.4.orig/gui/labelledfield.h libgui-0.0.4/gui/labelledfield.h
--- libgui-0.0.4.orig/gui/labelledfield.h	2005-07-28 08:32:22.664747500 -0700
+++ libgui-0.0.4/gui/labelledfield.h	2005-07-28 08:37:35.092273000 -0700
@@ -42,9 +42,9 @@
     void Create(unsigned long lStyle = 0L, Arg *args=NULL, Cardinal argcount=0);
  protected:
     virtual void InternalCreate(unsigned long lStyle=0L);
-    virtual Widget Field()=NULL;
+    virtual Widget Field()=0;
     inline Widget LabelParent() {return m_wdgForm;}
-    virtual void CreateField()=NULL;
+    virtual void CreateField()=0;
  private:
     Widget m_wdgForm;
     Widget m_wdgParent;
diff -Naur libgui-0.0.4.orig/gui/mainframe.h libgui-0.0.4/gui/mainframe.h
--- libgui-0.0.4.orig/gui/mainframe.h	2005-07-28 08:32:22.664747500 -0700
+++ libgui-0.0.4/gui/mainframe.h	2005-07-28 08:37:35.092273000 -0700
@@ -72,9 +72,9 @@
     virtual void InternalCreateMenu(Widget menu_bar);
     virtual void InternalDestroy();
 
-    virtual const char *Version() const=NULL;
-    virtual const char *Title() const=NULL;
-    virtual const char *PropertiesName() const=NULL;
+    virtual const char *Version() const=0;
+    virtual const char *Title() const=0;
+    virtual const char *PropertiesName() const=0;
     virtual void CreateToolBar(CToolBar *pToolBar);
 
     inline CToolBar *ToolBar() const { return m_pToolBar; }

diff -Naur libgui-0.0.4.orig/gui/textcontrol.h libgui-0.0.4/gui/textcontrol.h
--- libgui-0.0.4.orig/gui/textcontrol.h	2005-07-28 08:32:22.660747250 -0700
+++ libgui-0.0.4/gui/textcontrol.h	2005-07-28 08:37:35.096273250 -0700
@@ -40,8 +40,8 @@
     CStrings m_strText;
  protected:
     inline void SetTextControl(Widget wdgTextField) { m_wdgTextField = wdgTextField; }
-    virtual void InternalSetText(char *strText)=NULL;
-    virtual char *InternalText()=NULL;
+    virtual void InternalSetText(char *strText)=0;
+    virtual char *InternalText()=0;
     virtual void InternalDestroy();
  public:
     virtual Widget Container() const;
