--- configure.ac.orig	2013-02-10 20:14:25.000000000 +0400
+++ configure.ac	2013-02-11 02:11:54.065894982 +0400
@@ -517,7 +517,7 @@
 		AC_MSG_CHECKING([for fribidi/fribidi.h])
 		AC_TRY_COMPILE(
 #include <fribidi/fribidi.h>
-      , fribidi_version_info,
+      , FRIBIDI_VERSION,
 			AC_MSG_RESULT([yes])
 			AC_DEFINE(HAVE_FRIBIDI, [1], [Define to 1 if you have FRIBIDI])
 			LIBS="-lfribidi $LIBS",
