--- configure.in.orig	2011-02-19 15:09:14.000000000 +0300
+++ configure.in	2011-02-20 01:07:55.763381637 +0300
@@ -613,7 +613,7 @@
 		AC_MSG_CHECKING([for fribidi/fribidi.h])
 		AC_TRY_COMPILE(
 #include <fribidi/fribidi.h>
-      , fribidi_version_info,
+      , FRIBIDI_VERSION,
 			AC_MSG_RESULT([yes])
 			AC_DEFINE(HAVE_FRIBIDI, [1], [Define to 1 if you have FRIBIDI])
 			LIBS="$LIBS -lfribidi",
