--- speaker-test/speaker-test.c~	2006-04-06 10:24:12.000000000 -0700
+++ speaker-test/speaker-test.c	2006-04-20 18:23:04.704636250 -0700
@@ -44,9 +44,6 @@
 #include <getopt.h>
 #include <inttypes.h>
 #include <ctype.h>
-#ifdef ENABLE_NLS
-#include <locale.h>
-#endif
 #include <byteswap.h>
 
 #define ALSA_PCM_NEW_HW_PARAMS_API
@@ -56,6 +53,9 @@
 #include <math.h>
 #include "pink.h"
 #include "aconfig.h"
+#ifdef ENABLE_NLS
+#include <locale.h>
+#endif
 #include "gettext.h"
 #include "version.h"
 
