diff -uNr privoxy-3.0.5-BETA/GNUmakefile.in privoxy-3.0.5-BETA.cp/GNUmakefile.in
--- privoxy-3.0.5-BETA/GNUmakefile.in	2006-09-13 03:25:16.000000000 +0200
+++ privoxy-3.0.5-BETA.cp/GNUmakefile.in	2006-09-30 11:51:51.000000000 +0200
@@ -1240,10 +1240,6 @@
 			$(MKDIR) /etc/logrotate.d/ ;\
 			$(ECHO) "Installing logrotate script to /etc/logrotate.d/" ;\
 			$(INSTALL) -m 0644 privoxy.logrotate /etc/logrotate.d/privoxy ;\
-		elif [ -d /etc/init.d ] && [ -w /etc/init.d ] ; then \
-			$(ECHO) "Installing generic init script to /etc/init.d/privoxy" ;\
-			$(ECHO) "Please check that the PATHs are correct, and edit if needed." ;\
-			$(INSTALL) $(INSTALL_P) privoxy-generic.init /etc/init.d/privoxy ;\
 		fi ;\
 	else \
 		$(ECHO) "No init script installed, install it manually if needed" ;\
