--- sot/source/sdstor/stgole.cxx	22 Jul 2002 12:28:43 -0000	1.4
+++ sot/source/sdstor/stgole.cxx	12 Apr 2005 10:14:48 -0000
@@ -157,7 +157,7 @@ BOOL StgCompObjStream::Load()
 		INT32 nLen1 = 0;
 		*this >> nLen1;
 		sal_Char* p = new sal_Char[ (USHORT) nLen1 ];
-		if( Read( p, nLen1 ) == (ULONG) nLen1 )
+		if( Read( p, (USHORT) nLen1 ) == (ULONG) nLen1 )
 		{
 			aUserName = String( p, gsl_getSystemTextEncoding() );
 /*			// Now we can read the CB format
