diff -purN db-4.2.50/dist/configure db/dist/configure
--- db-4.2.50/dist/configure	2003-11-20 22:05:57.000000000 -0500
+++ db/dist/configure	2003-11-26 15:38:43.000000000 -0500
@@ -20729,7 +20729,6 @@ public static void main (String args[]) 
         System.exit (0);
 } }
 EOF
-if test x$ac_cv_prog_uudecode_base64 != xyes; then
         if { ac_try='$JAVAC $JAVACFLAGS $JAVA_TEST'
   { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
   (eval $ac_try) 2>&5
@@ -20744,7 +20743,6 @@ if test x$ac_cv_prog_uudecode_base64 != 
 echo "$as_me: error: The Java compiler $JAVAC failed (see config.log, check the CLASSPATH?)" >&2;}
    { (exit 1); exit 1; }; }
         fi
-fi
 if { ac_try='$JAVA $JAVAFLAGS $TEST'
   { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
   (eval $ac_try) 2>&5
