From c95a758eaca3baf466da25e91ebea89b609fe227 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 25 Aug 2017 20:39:33 +0200
Subject: [PATCH] std::isnan isnan

---
 xbmc/dialogs/GUIDialogExtendedProgressBar.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/dialogs/GUIDialogExtendedProgressBar.cpp b/xbmc/dialogs/GUIDialogExtendedProgressBar.cpp
index 42c2238..043a25f 100644
--- a/xbmc/dialogs/GUIDialogExtendedProgressBar.cpp
+++ b/xbmc/dialogs/GUIDialogExtendedProgressBar.cpp
@@ -52,7 +52,7 @@ void CGUIDialogProgressBarHandle::SetTitle(const std::string &strTitle)
 void CGUIDialogProgressBarHandle::SetProgress(int currentItem, int itemCount)
 {
   float fPercentage = (currentItem*100.0f)/itemCount;
-  if (!std::isnan(fPercentage))
+  if (!isnan(fPercentage))
     m_fPercentage = std::min(100.0f, fPercentage);
 }
 
-- 
2.14.1

