diff -Nuar xbmc-17.4-Krypton.orig/xbmc/utils/StringUtils.cpp xbmc-17.4-Krypton/xbmc/utils/StringUtils.cpp
--- xbmc-17.4-Krypton.orig/xbmc/utils/StringUtils.cpp	2017-08-24 12:59:16.508345154 -0400
+++ xbmc-17.4-Krypton/xbmc/utils/StringUtils.cpp	2017-08-24 13:12:39.486505409 -0400
@@ -1191,8 +1191,7 @@
 
 std::string StringUtils::CreateUUID()
 {
-  static GuidGenerator guidGenerator;
-  auto guid = guidGenerator.newGuid();
+  auto guid = xg::newGuid();
 
   std::stringstream strGuid; strGuid << guid;
   return strGuid.str();
