From ca928785a3ebd24a22cf7173bda0f7d30b4c637c Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Tue, 7 Aug 2018 10:36:59 +0200
Subject: [PATCH] fix linking error

---
 scrub/unicrash.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scrub/unicrash.c b/scrub/unicrash.c
index f60e07f..7153a68 100644
--- a/scrub/unicrash.c
+++ b/scrub/unicrash.c
@@ -27,6 +27,7 @@
 #include <unicode/ustring.h>
 #include <unicode/unorm2.h>
 #include <unicode/uspoof.h>
+#include <unicode/utf16.h>
 #include "path.h"
 #include "xfs_scrub.h"
 #include "common.h"
-- 
2.18.0

