From c65e8419274347c2e155df918bf725f4db5bb447 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 16 Nov 2019 10:15:39 +0100
Subject: [PATCH] fix linking error

---
 scrub/unicrash.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scrub/unicrash.c b/scrub/unicrash.c
index d5d2cf2..e02797a 100644
--- a/scrub/unicrash.c
+++ b/scrub/unicrash.c
@@ -13,6 +13,7 @@
 #include <unicode/ustring.h>
 #include <unicode/unorm2.h>
 #include <unicode/uspoof.h>
+#include <unicode/utf16.h>
 #include "libfrog/paths.h"
 #include "xfs_scrub.h"
 #include "common.h"
-- 
2.24.0

