--- a/src/dar_suite/my_getopt_long.h
+++ b/src/dar_suite/my_getopt_long.h
@@ -27,6 +27,10 @@
 #ifndef MY_GETOPT_LONG_H
 #define MY_GETOPT_LONG_H
 
+#if !defined(__GLIBC__) && defined(__linux__)
+extern int optreset; /* not defined by musl */
+#endif
+
 // getopt may be declated in <unistd.h> on systems like FreeBSD.
 // if you want to use libgnugetopt you need to include <getopt.h>
 // on this system. Thus a conflict appear because the getopt is
