diff -Naur a/widget/gtk/mozgtk/mozgtk.c b/widget/gtk/mozgtk/mozgtk.c
--- a/widget/gtk/mozgtk/mozgtk.c	2022-06-18 19:17:31.464160229 -0000
+++ b/widget/gtk/mozgtk/mozgtk.c	2022-06-18 19:24:34.966749520 -0000
@@ -8,6 +8,8 @@
 
 #include <gdk/gdk.h>
 
+MOZ_EXPORT unsigned int gtk_get_debug_flags(void) { return 0; }
+
 // Dummy call to gtk3 library to prevent the linker from removing
 // the gtk3 dependency with --as-needed.
 // see toolkit/library/moz.build for details.
diff -Naur a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	2022-06-18 19:17:31.464160229 -0000
+++ b/widget/gtk/nsWindow.cpp	2022-06-18 19:26:47.689851900 -0000
@@ -5312,6 +5312,9 @@
     SetCompositorHint(GTK_WIDGET_COMPOSIDED_ENABLED);
   }
 #endif
+// Dummy call to a function in mozgtk to prevent the linker from removing
+// the dependency with --as-needed.
+gtk_get_debug_flags();
 #ifdef MOZ_WAYLAND
   if (GdkIsWaylandDisplay()) {
     mSurfaceProvider.Initialize(this);
