diff -Nuar xneur-0.4.0.orig/etc/Makefile.am xneur-0.4.0/etc/Makefile.am
--- xneur-0.4.0.orig/etc/Makefile.am	2007-01-24 13:07:02.000000000 +0500
+++ xneur-0.4.0/etc/Makefile.am	2007-01-24 13:11:21.000000000 +0500
@@ -8,13 +8,3 @@
 etc_DATA = xneurrc xneurrc-ua
 
 EXTRA_DIST = $(etc_DATA)
-
-install-data-hook:
-	rm ~/.xneur.backup -Rf
-	if test -d ~/.xneur; then	\
-		mv ~/.xneur ~/.xneur.backup -f;	\
-		chmod -R a+r+w ~/.xneur.backup;	\
-		rm ~/.xneur -rf;		\
-	fi; 
-	cp $(DESTDIR)$(sysconfdir)/xneur ~/.xneur -rf;
-	chmod -R a+r+w ~/.xneur ;
diff -Nuar xneur-0.4.0.orig/etc/Makefile.in xneur-0.4.0/etc/Makefile.in
--- xneur-0.4.0.orig/etc/Makefile.in	2007-01-24 13:07:02.000000000 +0500
+++ xneur-0.4.0/etc/Makefile.in	2007-01-24 13:16:45.000000000 +0500
@@ -461,7 +461,6 @@
 
 install-data-am: install-etcDATA
 	@$(NORMAL_INSTALL)
-	$(MAKE) $(AM_MAKEFLAGS) install-data-hook
 
 install-exec-am:
 
@@ -496,7 +495,7 @@
 	ctags-recursive distclean distclean-generic distclean-libtool \
 	distclean-recursive distclean-tags distdir dvi dvi-am html \
 	html-am info info-am install install-am install-data \
-	install-data-am install-data-hook install-etcDATA install-exec \
+	install-data-am install-etcDATA install-exec \
 	install-exec-am install-info install-info-am install-man \
 	install-strip installcheck installcheck-am installdirs \
 	installdirs-am maintainer-clean maintainer-clean-generic \
@@ -506,15 +505,6 @@
 	uninstall-info-am
 
 
-install-data-hook:
-	rm ~/.xneur.backup -Rf
-	if test -d ~/.xneur; then	\
-		mv ~/.xneur ~/.xneur.backup -f;	\
-		chmod -R a+r+w ~/.xneur.backup;	\
-		rm ~/.xneur -rf;		\
-	fi; 
-	cp $(DESTDIR)$(sysconfdir)/xneur ~/.xneur -rf;
-	chmod -R a+r+w ~/.xneur ;
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
