*** fte/install	2001-08-28 05:06:34.000000000 +0200
--- fte/install	2002-04-06 16:58:58.000000000 +0200
***************
*** 34,40 ****
  fi
  
  # Ensure everything is made.
! make all
  if [ $? -ne 0 ]
  then
  	echo "Not everything could be made - not going to install it."
--- 34,40 ----
  fi
  
  # Ensure everything is made.
! make -e all
  if [ $? -ne 0 ]
  then
  	echo "Not everything could be made - not going to install it."
