--- camlidl-1.05/config/Makefile.unix.orig	2012-03-31 12:11:39.335097246 +0200
+++ camlidl-1.05/config/Makefile.unix	2012-03-31 12:12:36.311100004 +0200
@@ -31,7 +31,7 @@
 #RANLIB=:
 
 # Location of the Objective Caml library in your installation
-OCAMLLIB=/usr/local/lib/ocaml
+OCAMLLIB=/usr/lib/ocaml
 
 # Where to install the binaries
 BINDIR=/usr/local/bin
@@ -43,9 +43,6 @@
 OCAMLLEX=ocamllex
 OCAMLDEP=ocamldep
 
-# Extra flags to pass to the C compiler
-CFLAGS=-Wall -g
-
 # Suffixes for executables and libraries (do not change)
 EXE=
 LIBEXT=a
