diff -uprN rdate-1.5.old/rdate.c rdate-1.5/rdate.c
--- rdate-1.5.old/rdate.c	2003-09-20 21:17:52.290018848 -0700
+++ rdate-1.5/rdate.c	2003-09-20 21:18:36.249336016 -0700
@@ -215,19 +215,7 @@ int main(int argc, char *argv[])
     case 'm': maxchange = natural(optarg, "maxchange"); break;
     case 'c': mincount = natural(optarg, "mincount"); break;
     case 'h':
-      printf("\
-%s: print or set time based on remote time\n\
-Usage: %s [options] host..., where options are:\n\
- --help, -h -- print this help and exit\n\
- --udp, -u -- use UDP instead of TCP\n\
- --timeout, -t sec -- timeout for one host (0)\n\
- --print, -p -- print remote time (default)\n\
- --average, --avg, -a -- print the average remote time\n\
-  --delta, -d -- print delta between local time
- --set, -s -- set system time based on remote time\n\
-  --maxchange, -m sec -- max local time change allowed (0)\n\
-  --mincount, -c count -- min answers to set time (1)\n\
-", progname, progname);
+      printf("\ %s: print or set time based on remote time\n\ Usage: %s [options] host..., where options are:\n\ --help, -h -- print this help and exit\n\ --udp, -u -- use UDP instead of TCP\n\ --timeout, -t sec -- timeout for one host (0)\n\ --print, -p -- print remote time (default)\n\ --average, --avg, -a -- print the average remote time\n\ --delta, -d -- print delta between local time --set, -s -- set system time based on remote time\n\ --maxchange, -m sec -- max local time change allowed (0)\n\ --mincount, -c count -- min answers to set time (1)\n\ ", progname, progname);
       return 0;
     default:
       usage();
