--- include/utils.h~	2006-02-28 18:17:21.000000000 +0100
+++ include/utils.h	2006-02-28 18:17:35.000000000 +0100
@@ -4,6 +4,10 @@
 #include "avm_default.h"
 #include "formats.h"
 
+#ifndef __cplusplus
+#define __THROW
+#endif
+
 /**
  *  Utility functions.
  *    plain C functions are here
--- lib/aviread/ReadFile.cpp~	2006-02-28 18:06:55.000000000 +0100
+++ lib/aviread/ReadFile.cpp	2006-02-28 18:07:58.000000000 +0100
@@ -125,7 +125,7 @@
 };
 
 
-avm::IReadFile* avm::CreateReadFile(const char* name, unsigned int flags)
+avm::IReadFile* CreateReadFile(const char* name, unsigned int flags)
 {
     ReadFile* r = new ReadFile(name, flags);
     return r;
