diff -u -U 5 -r --new-file -x *~ -X /tmp/diffexclude tse3-0.2.7/src/tse3/Midi.h tse3-0.2.7.doj/src/tse3/Midi.h
--- tse3-0.2.7/src/tse3/Midi.h	Sun Jul 28 17:17:07 2002
+++ tse3-0.2.7.doj/src/tse3/Midi.h	Thu Jul  8 11:34:35 2004
@@ -817,8 +817,8 @@
                 }
             private:
                 const MidiEvent &e1;
         };
     };
-};
+}
 
 #endif
diff -u -U 5 -r --new-file -x *~ -X /tmp/diffexclude tse3-0.2.7/src/tse3/Mutex.h tse3-0.2.7.doj/src/tse3/Mutex.h
--- tse3-0.2.7/src/tse3/Mutex.h	Sun Jul 28 17:17:08 2002
+++ tse3-0.2.7.doj/src/tse3/Mutex.h	Thu Jul  8 12:14:04 2004
@@ -313,9 +313,9 @@
 #ifndef TSE3_WITHOUT_MUTEX
                     Mutex::mutex()->unlock();
 #endif
                 }
         };
-    };
+    }
 }
 
 #endif
diff -u -U 5 -r --new-file -x *~ -X /tmp/diffexclude tse3-0.2.7/src/tse3/Notifier.h tse3-0.2.7.doj/src/tse3/Notifier.h
--- tse3-0.2.7/src/tse3/Notifier.h	Tue Aug 20 16:59:15 2002
+++ tse3-0.2.7.doj/src/tse3/Notifier.h	Thu Jul  8 11:35:12 2004
@@ -288,11 +288,11 @@
                 const p2_type       &p2;
                 const p3_type       &p3;
                 const p4_type       &p4;                                 // (*)
         };
 
-    };
+    }
 
     /**************************************************************************
      * Public Notifier framework API
      *************************************************************************/
 
@@ -365,11 +365,11 @@
             /**
              * The type of @ref Listener that this Notifier class works with.
              */
             typedef Listener<interface_type> listener_type;
 
-            friend class listener_type;
+            friend class Listener<interface_type>;
 
         protected:
 
             /**
              * Creates a Notifier with no listeners.
@@ -572,10 +572,12 @@
             /**
              * The type of @ref Notifier class this listener works with.
              */
             typedef Notifier<interface_type> notifier_type;
 
+            friend class Notifier<interface_type>;
+
             /**
              * Attaches this Listener to @p notifier.
              *
              * You can only attach to a @ref Notifer object once - any
              * subsequent attempt to attach to it results in nothing happening;
@@ -599,12 +601,10 @@
              */
             void detachFrom(notifier_type *notifier)
             {
                 if (notifiers.erase(notifier)) notifier->detach(this);
             }
-
-            friend class notifier_type;
 
         protected:
 
             /**
              * Creates a new Listener which is not attached to any
diff -u -U 5 -r --new-file -x *~ -X /tmp/diffexclude tse3-0.2.7/src/tse3/app/Application.h tse3-0.2.7.doj/src/tse3/app/Application.h
--- tse3-0.2.7/src/tse3/app/Application.h	Sun Jul 28 17:17:09 2002
+++ tse3-0.2.7.doj/src/tse3/app/Application.h	Thu Jul  8 12:11:22 2004
@@ -33,16 +33,16 @@
     class PresetColours;
 
     namespace Cmd
     {
         class CommandHistory;
-    };
+    }
 
     namespace Ins
     {
         class Destination;
-    };
+    }
 
     /**
      * The App namespace contains classes that use the @ref TSE3
      * API, and provide a level of application support - including facilities
      * such as choices file saving.
diff -u -U 5 -r --new-file -x *~ -X /tmp/diffexclude tse3-0.2.7/src/tse3/app/Record.h tse3-0.2.7.doj/src/tse3/app/Record.h
--- tse3-0.2.7/src/tse3/app/Record.h	Sun Jul 28 17:17:09 2002
+++ tse3-0.2.7.doj/src/tse3/app/Record.h	Thu Jul  8 12:11:22 2004
@@ -34,11 +34,11 @@
     class MidiScheduler;
 
     namespace Cmd
     {
         class CommandHistory;
-    };
+    }
 
     namespace App
     {
         /**
          * This class provides help using the @ref Transport object's record
diff -u -U 5 -r --new-file -x *~ -X /tmp/diffexclude tse3-0.2.7/src/tse3/plt/Arts.cpp tse3-0.2.7.doj/src/tse3/plt/Arts.cpp
--- tse3-0.2.7/src/tse3/plt/Arts.cpp	Tue Oct 22 15:16:45 2002
+++ tse3-0.2.7.doj/src/tse3/plt/Arts.cpp	Thu Jul  8 11:39:49 2004
@@ -72,11 +72,11 @@
             }
     };
 
     ArtsCommon *artsCommon     = 0;
     int         artsCommonInit = 0;
-};
+}
 
 
 ArtsMidiScheduler::ArtsMidiScheduler()
 {
     if(!artsCommonInit) { assert(!artsCommon); artsCommon = new ArtsCommon(); }
diff -u -U 5 -r --new-file -x *~ -X /tmp/diffexclude tse3-0.2.7/src/tse3/util/MulDiv.h tse3-0.2.7.doj/src/tse3/util/MulDiv.h
--- tse3-0.2.7/src/tse3/util/MulDiv.h	Sun Jul 28 17:17:13 2002
+++ tse3-0.2.7.doj/src/tse3/util/MulDiv.h	Thu Jul  8 11:35:41 2004
@@ -32,9 +32,9 @@
          * @param  num Numerator
          * @param  div Divisor
          * @return val * num / div (which coped with integer overflows)
          */
         extern int muldiv(int val, int num, int div);
-    };
+    }
 }
 
 #endif
