diff -ruN gnome-alsamixer-0.9.6/src/gam-app.c gnome-alsamixer-0.9.6.new/src/gam-app.c
--- gnome-alsamixer-0.9.6/src/gam-app.c	2003-09-09 13:41:23.000000000 +0200
+++ gnome-alsamixer-0.9.6.new/src/gam-app.c	2005-08-31 15:34:42.000000000 +0200
@@ -25,7 +25,7 @@
 #include <gtk/gtklabel.h>
 #include <gtk/gtknotebook.h>
 #include <gtk/gtkmenuitem.h>
-#include <libgnome/gnome-i18n.h>
+#include <glib/gi18n.h>
 #include <libgnome/gnome-config.h>
 #include <libgnomeui/gnome-about.h>
 #include <libgnomeui/gnome-app-helper.h>
diff -ruN gnome-alsamixer-0.9.6/src/gam-prefs-dlg.c gnome-alsamixer-0.9.6.new/src/gam-prefs-dlg.c
--- gnome-alsamixer-0.9.6/src/gam-prefs-dlg.c	2003-09-09 13:37:23.000000000 +0200
+++ gnome-alsamixer-0.9.6.new/src/gam-prefs-dlg.c	2005-08-31 15:35:45.000000000 +0200
@@ -22,7 +22,7 @@
 #include <config.h>
 #endif
 
-#include <libgnome/gnome-i18n.h>
+#include <glib/gi18n.h>
 #include <libgnome/gnome-config.h>
 
 #include "gam-prefs-dlg.h"
diff -ruN gnome-alsamixer-0.9.6/src/gam-props-dlg.c gnome-alsamixer-0.9.6.new/src/gam-props-dlg.c
--- gnome-alsamixer-0.9.6/src/gam-props-dlg.c	2003-09-09 13:31:18.000000000 +0200
+++ gnome-alsamixer-0.9.6.new/src/gam-props-dlg.c	2005-08-31 15:35:52.000000000 +0200
@@ -22,7 +22,7 @@
 #include <config.h>
 #endif
 
-#include <libgnome/gnome-i18n.h>
+#include <glib/gi18n.h>
 
 #include "gam-props-dlg.h"
 #include "gam-slider.h"
