--- ./src/libavoid/connector.h~	2005-10-25 08:04:17.000000000 +0200
+++ ./src/libavoid/connector.h	2006-05-10 09:53:13.000000000 +0200
@@ -59,7 +59,7 @@
         bool isInitialised(void);
         void unInitialise(void);
         void setCallback(void (*cb)(void *), void *ptr);
-        void ConnRef::handleInvalid(void);
+        void handleInvalid(void);
         int generatePath(Point p0, Point p1);
         void makePathInvalid(void);
         
--- ./src/livarot/Path.h~	2005-11-07 07:04:16.000000000 +0100
+++ ./src/livarot/Path.h	2006-05-10 10:01:05.000000000 +0200
@@ -198,7 +198,7 @@
   cut_position*    CurvilignToPosition(int nbCv,double* cvAbs,int &nbCut);
   cut_position    PointToCurvilignPosition(NR::Point const &pos) const;
   //Should this take a cut_position as a param?
-  double           Path::PositionToLength(int piece, double t);
+  double           PositionToLength(int piece, double t);
   
   // caution: not tested on quadratic b-splines, most certainly buggy
   void             ConvertPositionsToMoveTo(int nbPos,cut_position* poss);
--- ./src/dialogs/tiledialog.h~	2005-05-25 16:07:31.000000000 +0200
+++ ./src/dialogs/tiledialog.h	2006-05-10 10:25:00.000000000 +0200
@@ -58,7 +58,7 @@
     /**
      * Respond to selection change
      */
-    void TileDialog::updateSelection();
+    void updateSelection();
 
 
     /**
--- ./src/extension/internal/ps.h~	2005-09-20 07:51:13.000000000 +0200
+++ ./src/extension/internal/ps.h	2006-05-10 11:03:23.000000000 +0200
@@ -36,10 +36,10 @@
 
 	void print_bpath (SVGOStringStream &os, const NArtBpath *bp);
 
-	void PrintPS::print_fill_style (SVGOStringStream &os, const SPStyle *style, const NRRect *pbox);
-	void PrintPS::print_stroke_style (SVGOStringStream &os, const SPStyle *style);
+	void print_fill_style (SVGOStringStream &os, const SPStyle *style, const NRRect *pbox);
+	void print_stroke_style (SVGOStringStream &os, const SPStyle *style);
 
-	const char* PrintPS::PSFontName (const SPStyle *style);
+	const char* PSFontName (const SPStyle *style);
 
 	unsigned int print_image (FILE *ofp, guchar *px, unsigned int width, unsigned int height, unsigned int rs,
 				       const NRMatrix *transform);
--- ./src/ui/dialog/aboutbox.h~	2005-09-05 22:18:41.000000000 +0200
+++ ./src/ui/dialog/aboutbox.h	2006-05-10 09:59:04.000000000 +0200
@@ -28,7 +28,7 @@
 class AboutBoxChild: public Gtk::Dialog
 {
 public:
-    AboutBoxChild::AboutBoxChild(Gtk::Window& parent, gchar * title)
+    AboutBoxChild(Gtk::Window& parent, gchar * title)
         : Gtk::Dialog(title,parent) {};
 protected:
     virtual void         on_response(int response_id);
--- ./src/ui/dialog/align-and-distribute.cpp~	2005-09-20 07:51:16.000000000 +0200
+++ ./src/ui/dialog/align-and-distribute.cpp	2006-05-10 11:19:50.000000000 +0200
@@ -954,7 +954,7 @@
         break;
 
     } // end of switch statement
-    return NULL;
+    //return NULL;
 }
 
 AlignAndDistribute::AlignTarget AlignAndDistribute::getAlignTarget()const {
