diff -ur gst-plugins-bad-0.10.22/ext/mpeg2enc/gstmpeg2encoder.cc gst-plugins-bad-mod/ext/mpeg2enc/gstmpeg2encoder.cc
--- gst-plugins-bad-0.10.22/ext/mpeg2enc/gstmpeg2encoder.cc	2010-04-29 03:59:29.000000000 -0500
+++ gst-plugins-bad-mod/ext/mpeg2enc/gstmpeg2encoder.cc	2012-01-13 16:46:44.177259807 -0600
@@ -27,9 +27,8 @@
 #include <mpegconsts.h>
 #include <quantize.hh>
 #if GST_MJPEGTOOLS_API >= 10900
-#include <ontheflyratectl.hh>
-#include <pass1ratectl.hh>
-#include <pass2ratectl.hh>
+#include <ontheflyratectlpass1.hh>
+#include <ontheflyratectlpass2.hh>
 #else
 #include <ratectl.hh>
 #endif
