--- ext/neon/gstneonhttpsrc.c    29 Apr 2006 15:36:15 -0000      1.13
+++ ext/neon/gstneonhttpsrc.c    17 Jun 2006 16:04:48 -0000
@@ -20,6 +20,10 @@
 #include <string.h>
 #include <unistd.h>

+#ifndef NE_FREE
+#define NEON_026_OR_LATER  1
+#endif
+
 #define HTTP_DEFAULT_HOST        "localhost"
 #define HTTP_DEFAULT_PORT        80
 #define HTTPS_DEFAULT_PORT       443
@@ -627,7 +631,11 @@ set_proxy (const char *uri, ne_uri * par
     goto clear;
   }

+#ifdef NEON_026_OR_LATER
+  if (!parsed->path || parsed->userinfo) {
+#else
   if (!parsed->path || parsed->authinfo) {
+#endif
     goto clear;
   }
