diff -uNr beecrypt-4.1.2/c++/beeyond/BeeCertificate.h beecrypt-4.1.2.cp/c++/beeyond/BeeCertificate.h
--- beecrypt-4.1.2/c++/beeyond/BeeCertificate.h	2004-11-03 10:37:27.000000000 +0100
+++ beecrypt-4.1.2.cp/c++/beeyond/BeeCertificate.h	2006-07-04 14:16:10.000000000 +0200
@@ -61,7 +61,7 @@
 		 */
 		class BEECRYPTCXXAPI BeeCertificate : public beecrypt::security::cert::Certificate, public beecrypt::lang::Cloneable
 		{
-			friend class BeeCertificateFactory;
+			friend class provider::BeeCertificateFactory;
 
 		public:
 			static const Date FOREVER;
diff -uNr beecrypt-4.1.2/c++/security/Security.h beecrypt-4.1.2.cp/c++/security/Security.h
--- beecrypt-4.1.2/c++/security/Security.h	2004-11-02 08:06:58.000000000 +0100
+++ beecrypt-4.1.2.cp/c++/security/Security.h	2006-07-04 14:14:46.000000000 +0200
@@ -76,7 +76,6 @@
 			typedef vector<const Provider*> provider_vector;
 			typedef provider_vector::iterator provider_vector_iterator;
 
-		private:
 			struct spi
 			{
 				Object* cspi;
@@ -90,6 +89,7 @@
 			static spi* getSpi(const String& algo, const String& type, const String& provider) throw (NoSuchAlgorithmException, NoSuchProviderException);
 			static spi* getSpi(const String& algo, const String& type, const Provider&) throw (NoSuchAlgorithmException);
 			static spi* getFirstSpi(const String& type);
+		private:
 
 			static const String& getKeyStoreDefault();
 
