--- db-5.3.28/dist/configure.orig	2025-04-28 01:00:42.322977454 +0200
+++ db-5.3.28/dist/configure	2025-04-28 01:04:29.314975385 +0200
@@ -20609,7 +20609,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_cond_t cond;
 	pthread_mutex_t mutex;
 	pthread_condattr_t condattr;
@@ -20669,7 +20669,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_cond_t cond;
 	pthread_condattr_t condattr;
 	exit(pthread_condattr_init(&condattr) ||
@@ -20719,7 +20719,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_rwlock_t rwlock;
 	pthread_rwlockattr_t rwlockattr;
 	exit(pthread_rwlockattr_init(&rwlockattr) ||
@@ -20783,7 +20783,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_cond_t cond;
 	pthread_mutex_t mutex;
 	pthread_condattr_t condattr;
@@ -20843,7 +20843,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_cond_t cond;
 	pthread_condattr_t condattr;
 	exit(pthread_condattr_init(&condattr) ||
@@ -20893,7 +20893,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_rwlock_t rwlock;
 	pthread_rwlockattr_t rwlockattr;
 	exit(pthread_rwlockattr_init(&rwlockattr) ||
@@ -20958,7 +20958,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_cond_t cond;
 	pthread_mutex_t mutex;
 	pthread_condattr_t condattr;
@@ -21016,7 +21016,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_cond_t cond;
 	pthread_condattr_t condattr;
 	exit(pthread_condattr_init(&condattr) ||
@@ -21066,7 +21066,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_rwlock_t rwlock;
 	pthread_rwlockattr_t rwlockattr;
 	exit(pthread_rwlockattr_init(&rwlockattr) ||
@@ -21128,7 +21128,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_cond_t cond;
 	pthread_mutex_t mutex;
 	pthread_condattr_t condattr;
@@ -21186,7 +21186,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_cond_t cond;
 	pthread_condattr_t condattr;
 	exit(pthread_condattr_init(&condattr) ||
@@ -21236,7 +21236,7 @@
 
 #include <stdlib.h>
 #include <pthread.h>
-main() {
+int main() {
 	pthread_rwlock_t rwlock;
 	pthread_rwlockattr_t rwlockattr;
 	exit(pthread_rwlockattr_init(&rwlockattr) ||
@@ -23454,7 +23454,7 @@
 /* end confdefs.h.  */
 
 #include <sys/time.h>
-main() {
+int main() {
 	struct timespec t;
 	return (clock_gettime(CLOCK_MONOTONIC, &t) != 0);
 }
