diff -Naur db-4.3.27.orig/dist/configure db-4.3.27/dist/configure
--- db-4.3.27.orig/dist/configure	2005-02-24 22:46:57.845051456 -0800
+++ db-4.3.27/dist/configure	2005-02-24 22:47:09.964209064 -0800
@@ -21468,7 +21468,6 @@
         System.exit (0);
 } }
 EOF
-if test x$ac_cv_prog_uudecode_base64 != xyes; then
         if { ac_try='$JAVAC $JAVACFLAGS $JAVA_TEST'
   { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
   (eval $ac_try) 2>&5
@@ -21483,7 +21482,6 @@
 echo "$as_me: error: The Java compiler $JAVAC failed (see config.log, check the CLASSPATH?)" >&2;}
    { (exit 1); exit 1; }; }
         fi
-fi
 if { ac_try='$JAVA $JAVAFLAGS $TEST'
   { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
   (eval $ac_try) 2>&5
