--- hk_kdecsvexportdialog.cpp	2005-09-29 09:25:20.000000000 +0000
+++ hk_kdecsvexportdialog.cpp.new	2005-09-28 17:07:28.000000000 +0000
@@ -72,7 +72,7 @@
     datetimeformatfield->setText(QString::fromUtf8(l2u(defaultdatetimeformat()).c_str()));
     for (unsigned int i=0;i<sizeof(charsets)/sizeof(hk_string);++i)
     {
-    charsetfield->insertItem(charsets[i]);
+    charsetfield->insertItem(QString::fromUtf8(l2u(charsets[i]).c_str()));
     }
 
     for (unsigned int i=0;i<sizeof(locales)/sizeof(hk_string);++i)
--- hk_kdecsvimportdialog.cpp	2005-09-29 09:25:07.000000000 +0000
+++ hk_kdecsvimportdialog.cpp.new	2005-09-28 17:07:28.000000000 +0000
@@ -77,7 +77,7 @@
 
     for (unsigned int i=0;i<sizeof(locales)/sizeof(hk_string);++i)
     {
-    localefield->insertItem(locales[i]);
+    localefield->insertItem(QString::fromUtf8(l2u(locales[i]).c_str()));
     }
 
     charsetfield->setCurrentText(QString::fromUtf8(l2u(filecharset()).c_str()));
