--- source/blender/python/intern/bpy_interface.c
+++ source/blender/python/intern/bpy_interface.c
@@ -203,11 +203,6 @@
	/* allow to use our own included python */
	PyC_SetHomePath(BLI_get_folder(BLENDER_SYSTEM_PYTHON, NULL));
 
-	/* Python 3.2 now looks for '2.58/python/include/python3.2d/pyconfig.h' to parse
-	 * from the 'sysconfig' module which is used by 'site', so for now disable site.
-	 * alternatively we could copy the file. */
-	Py_NoSiteFlag= 1;
-
 	Py_Initialize();
 	
 	// PySys_SetArgv(argc, argv); // broken in py3, not a huge deal
--- source/gameengine/Ketsji/KX_PythonInit.cpp
+++ source/gameengine/Ketsji/KX_PythonInit.cpp
@@ -1825,7 +1825,6 @@
 	STR_String pname = progname;
 	Py_SetProgramName(pname.Ptr());
 #endif
-	Py_NoSiteFlag=1;
 	Py_FrozenFlag=1;
 
 	/* must run before python initializes */
@@ -1898,7 +1897,6 @@
 	STR_String pname = progname;
 	Py_SetProgramName(pname.Ptr());
 #endif
-	Py_NoSiteFlag=1;
 	Py_FrozenFlag=1;
 
 	initPyTypes();
