--- extern/SConscript
+++ extern/SConscript
@@ -2,8 +2,6 @@
 
 Import('env')
 
-SConscript(['glew/SConscript'])
-
 if env['WITH_BF_GAMEENGINE']:
	 SConscript(['recastnavigation/SConscript'])

 if env['WITH_BF_ELTOPO']:
	 SConscript(['eltopo/SConscript'])

--- source/blender/editors/include/BIF_gl.h
+++ source/blender/editors/include/BIF_gl.h
@@ -35,7 +35,7 @@
 #ifndef BIF_GL_H
 #define BIF_GL_H
 
-#include "GL/glew.h"
+#include <GL/glew.h>
 
	/*
	 * these should be phased out. cpack should be replaced in
--- extern/CMakeLists.txt
+++ extern/CMakeLists.txt
@@ -39,9 +39,6 @@
	 add_subdirectory(binreloc)
 endif()

-if(WITH_BUILTIN_GLEW)
-	add_subdirectory(glew)
-endif()

 if(WITH_GAMEENGINE)
    add_subdirectory(recastnavigation)
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -150,7 +150,7 @@

 if(UNIX AND NOT APPLE)
 	option(WITH_X11_XINPUT "Enable X11 Xinput (tablet support)"	ON)
-	option(WITH_BUILTIN_GLEW "Use GLEW OpenGL wrapper library bundled with blender" ON)
+	option(WITH_BUILTIN_GLEW "Use GLEW OpenGL wrapper library bundled with blender" OFF)
 else()
 	# not an option for other OS's
 	set(WITH_BUILTIN_GLEW ON)
