diff -Npur release/plugins.orig/bmake release/plugins/bmake
--- release/plugins.orig/bmake	2011-11-11 15:00:26.339924887 +0100
+++ release/plugins/bmake	2011-11-11 15:01:40.637923565 +0100
@@ -57,10 +57,10 @@
	LDFLAGS="-Bshareable";
 elif ( test $UNAME = "Linux" ) then
	CC="gcc";
-	CFLAGS="-fPIC -funsigned-char -O2";
+#	CFLAGS="-fPIC -funsigned-char -O2";

	LD="ld";
-	LDFLAGS="-Bshareable";
+#	LDFLAGS="-Bshareable";
 elif ( test $UNAME = "SunOS" ) then
	CC="cc";
	CFLAGS="-O";
@@ -109,22 +109,20 @@ elif ( test -f "../plugin.h" ) then
 	INCLUDES=-I..;
 elif ( test -f "../include/plugin.h" ) then
 	INCLUDES=-I../include
+elif ( test -f "../../../source/blender/blenpluginapi/plugin.h" ) then
+	INCLUDES=-I../../../source/blender/blenpluginapi
 else 
 	echo "Couldn't find plugin.h";
 	exit;
 fi
 
 LIBM=`fgrep "#include <math.h>" $CFILE`
-LIBC=`fgrep "#include <std" $CFILE`
 
-LIBS=
+LIBS="-lc"
 
 if ( test -n "$LIBM" ) then
 	LIBS="$LIBS -lm"; 
 fi
-if ( test -n "$LIBC" ) then 
-	LIBS="$LIBS -lc"; 
-fi
 
 echo "$CC $CFLAGS -c $CFILE $INCLUDES"
 $CC $CFLAGS -c $CFILE $INCLUDES
diff -Npur release/plugins.orig/sequence/color-correction-hsv.c release/plugins/sequence/color-correction-hsv.c
--- release/plugins.orig/sequence/color-correction-hsv.c	2011-11-11 15:00:26.339924887 +0100
+++ release/plugins/sequence/color-correction-hsv.c	2011-11-11 15:03:47.233921332 +0100
@@ -15,7 +15,7 @@
  *
  */
 
-#include "math.h"
+#include <math.h>
 #include "plugin.h"
 #include <stdio.h>
 
diff -Npur release/plugins.orig/sequence/color-correction-yuv.c release/plugins/sequence/color-correction-yuv.c
--- release/plugins.orig/sequence/color-correction-yuv.c	2011-11-11 15:00:26.339924887 +0100
+++ release/plugins/sequence/color-correction-yuv.c	2011-11-11 15:02:10.446923031 +0100
@@ -15,7 +15,7 @@
  *
  */
 
-#include "math.h"
+#include <math.h>
 #include "plugin.h"
 #include <stdio.h>
 
diff -Npur release/plugins.orig/sequence/dnr.c release/plugins/sequence/dnr.c
--- release/plugins.orig/sequence/dnr.c	2011-11-11 15:00:26.339924887 +0100
+++ release/plugins/sequence/dnr.c	2011-11-11 15:03:30.049921648 +0100
@@ -15,7 +15,7 @@
  *
  */
 
-#include "math.h"
+#include <math.h>
 #include "plugin.h"
 #include <stdio.h>
 
diff -Npur release/plugins.orig/sequence/gamma.c release/plugins/sequence/gamma.c
--- release/plugins.orig/sequence/gamma.c	2011-11-11 15:00:26.339924887 +0100
+++ release/plugins/sequence/gamma.c	2011-11-11 15:01:58.982923268 +0100
@@ -15,7 +15,7 @@
  *
  */
 
-#include "math.h"
+#include <math.h>
 #include "plugin.h"
 #include "util.h"
 #include <stdio.h>
diff -Npur release/plugins.orig/texture/clouds2.c release/plugins/texture/clouds2.c
--- release/plugins.orig/texture/clouds2.c	2011-11-11 15:00:26.339924887 +0100
+++ release/plugins/texture/clouds2.c	2011-11-11 15:02:23.104922789 +0100
@@ -27,7 +27,7 @@
  * ***** END GPL LICENSE BLOCK *****
  */
  
-#include "math.h"
+#include <math.h>
 #include "plugin.h"
 
 /* ******************** GLOBAL VARIABLES ***************** */
diff -Npur release/plugins.orig/texture/tiles.c release/plugins/texture/tiles.c
--- release/plugins.orig/texture/tiles.c	2011-11-11 15:00:26.339924887 +0100
+++ release/plugins/texture/tiles.c	2011-11-11 15:04:05.749921038 +0100
@@ -27,7 +27,7 @@
  * ***** END GPL LICENSE BLOCK *****
  */
 
-#include "math.h"
+#include <math.h>
 #include "plugin.h"
 
 /* ******************** GLOBAL VARIABLES ***************** */
