diff -Npur blender-2.60a.orig/extern/libredcode/SConscript blender-2.60a/extern/libredcode/SConscript
--- blender-2.60a.orig/extern/libredcode/SConscript	2011-11-09 23:05:16.241465491 +0100
+++ blender-2.60a/extern/libredcode/SConscript	2011-11-09 23:06:24.188464236 +0100
@@ -7,7 +7,8 @@
 Import('env')

 sources = env.Glob('*.c')
-incs = '. ../libopenjpeg'
+#incs = '. ../libopenjpeg'
+incs = '. /usr/include'

 env.BlenderLib ( libname='extern_redcode', 
                  sources=sources, includes=Split(incs),
diff -Npur blender-2.60a.orig/extern/SConscript blender-2.60a/extern/SConscript
--- blender-2.60a.orig/extern/SConscript	2011-11-09 23:05:16.241465491 +0100
+++ blender-2.60a/extern/SConscript	2011-11-09 23:06:24.188464236 +0100
@@ -11,8 +11,5 @@
 if env['WITH_BF_BULLET']:
     SConscript(['bullet2/src/SConscript'])

-if env['WITH_BF_OPENJPEG'] and env['BF_OPENJPEG_LIB'] == '':
-    SConscript(['libopenjpeg/SConscript'])
-
 if env['WITH_BF_REDCODE'] and env['BF_REDCODE_LIB'] == '':
     SConscript(['libredcode/SConscript'])
diff -Npur blender-2.60a.orig/build_files/scons/tools/Blender.py blender-2.60a/build_files/scons/tools/Blender.py
--- blender-2.60a.orig/build_files/scons/tools/Blender.py	2011-11-09 23:05:16.241465491 +0100
+++ blender-2.60a/build_files/scons/tools/Blender.py	2011-11-09 23:06:24.188464236 +0100
@@ -246,6 +246,8 @@
         syslibs += Split(lenv['BF_FFMPEG_LIB'])
         if lenv['WITH_BF_OGG']:
             syslibs += Split(lenv['BF_OGG_LIB'])
+    if lenv['WITH_BF_OPENJPEG']:
+        syslibs += Split(lenv['BF_OPENJPEG_LIB'])
     if lenv['WITH_BF_JACK']:
             syslibs += Split(lenv['BF_JACK_LIB'])
     if lenv['WITH_BF_SNDFILE'] and not lenv['WITH_BF_STATICSNDFILE']:
