--- blender-2.58a/doc/doxygen/Doxyfile   2011-07-11 23:27:33.975642545 -0300
+++ blender-2.58a/doc/doxygen/Doxyfile   2011-07-11 23:28:54.913642574 -0300
@@ -42,7 +42,7 @@
 # exceed 55 pixels and the maximum width should not exceed 200 pixels.
 # Doxygen will copy the logo to the output directory.
 
-PROJECT_LOGO           = ../../release/freedesktop/icons/48x48/blender.png
+PROJECT_LOGO           = ../../release/freedesktop/icons/48x48/apps/blender.png
 
 # The OUTPUT_DIRECTORY tag is used to specify the (relative or absolute)
 # base path where the generated documentation will be put. 
