diff -ur blender-2.61/CMakeLists.txt blender-mod/CMakeLists.txt
--- blender-2.61/CMakeLists.txt	2011-12-13 13:57:52.000000000 -0600
+++ blender-mod/CMakeLists.txt	2011-12-15 16:43:02.083765807 -0600
@@ -156,7 +156,7 @@
 
 if(UNIX AND NOT APPLE)
 	option(WITH_X11_XINPUT "Enable X11 Xinput (tablet support and unicode input)"	ON)
-	option(WITH_BUILTIN_GLEW "Use GLEW OpenGL wrapper library bundled with blender" ON)
+	option(WITH_BUILTIN_GLEW "Use GLEW OpenGL wrapper library bundled with blender" OFF)     
 	option(WITH_XDG_USER_DIRS    "Build with XDG Base Directory Specification (only config and documents for now)" OFF)
 	mark_as_advanced(WITH_XDG_USER_DIRS)
 else()
diff -ur blender-2.61/extern/CMakeLists.txt blender-mod/extern/CMakeLists.txt
--- blender-2.61/extern/CMakeLists.txt	2011-12-13 13:57:20.000000000 -0600
+++ blender-mod/extern/CMakeLists.txt	2011-12-15 16:40:53.775760453 -0600
@@ -40,10 +40,6 @@
 	add_subdirectory(binreloc)
 endif()
 
-if(WITH_BUILTIN_GLEW)
-	add_subdirectory(glew)
-endif()
-
 if(WITH_GAMEENGINE)
    add_subdirectory(recastnavigation)
 endif()
diff -ur blender-2.61/extern/SConscript blender-mod/extern/SConscript
--- blender-2.61/extern/SConscript	2011-12-13 13:57:20.000000000 -0600
+++ blender-mod/extern/SConscript	2011-12-15 16:39:52.497757875 -0600
@@ -2,7 +2,6 @@
 
 Import('env')
 
-SConscript(['glew/SConscript'])
 SConscript(['colamd/SConscript'])
 
 if env['WITH_BF_GAMEENGINE']:
diff -ur blender-2.61/source/blender/editors/include/BIF_gl.h blender-mod/source/blender/editors/include/BIF_gl.h
--- blender-2.61/source/blender/editors/include/BIF_gl.h	2011-12-13 13:52:45.000000000 -0600
+++ blender-mod/source/blender/editors/include/BIF_gl.h	2011-12-15 16:41:59.989763237 -0600
@@ -33,7 +33,7 @@
 #ifndef BIF_GL_H
 #define BIF_GL_H
 
-#include "GL/glew.h"
+#include <GL/glew.h>
 
 	/*
 	 * these should be phased out. cpack should be replaced in
