diff -Naur blender-2.45.orig/intern/ghost/intern/GHOST_WindowX11.cpp blender-2.45/intern/ghost/intern/GHOST_WindowX11.cpp
--- blender-2.45.orig/intern/ghost/intern/GHOST_WindowX11.cpp	2008-04-19 17:10:04.005647128 -0700
+++ blender-2.45/intern/ghost/intern/GHOST_WindowX11.cpp	2008-04-19 17:22:52.032923670 -0700
@@ -42,6 +42,8 @@
 #include <strings.h>
 #endif
 
+#include <cstdio>
+
 
 // For obscure full screen mode stuuf
 // lifted verbatim from blut.
diff -Naur blender-2.45.orig/intern/string/STR_String.h blender-2.45/intern/string/STR_String.h
--- blender-2.45.orig/intern/string/STR_String.h	2008-04-19 17:10:03.994667450 -0700
+++ blender-2.45/intern/string/STR_String.h	2008-04-19 17:13:23.254673104 -0700
@@ -47,6 +47,9 @@
 
 #include <vector>
 #include <limits.h>
+#include <cstring>
+#include <cstdlib>
+
 using namespace std;
 
 
diff -Naur blender-2.45.orig/source/blender/yafray/intern/export_File.cpp blender-2.45/source/blender/yafray/intern/export_File.cpp
--- blender-2.45.orig/source/blender/yafray/intern/export_File.cpp	2008-04-19 17:10:03.940667287 -0700
+++ blender-2.45/source/blender/yafray/intern/export_File.cpp	2008-04-19 17:35:45.875893589 -0700
@@ -1,6 +1,7 @@
 #include "export_File.h"
 
 #include <math.h>
+#include <cstring>
 
 using namespace std;
 
diff -Naur blender-2.45.orig/source/blender/yafray/intern/export_Plugin.cpp blender-2.45/source/blender/yafray/intern/export_Plugin.cpp
--- blender-2.45.orig/source/blender/yafray/intern/export_Plugin.cpp	2008-04-19 17:10:03.940667287 -0700
+++ blender-2.45/source/blender/yafray/intern/export_Plugin.cpp	2008-04-19 17:33:48.316644087 -0700
@@ -1,6 +1,8 @@
 #include "export_Plugin.h"
 
 #include <math.h>
+#include <cstring>
+
 using namespace std;
 
 
diff -Naur blender-2.45.orig/source/blender/yafray/intern/yafray_Render.cpp blender-2.45/source/blender/yafray/intern/yafray_Render.cpp
--- blender-2.45.orig/source/blender/yafray/intern/yafray_Render.cpp	2008-04-19 17:10:03.939669059 -0700
+++ blender-2.45/source/blender/yafray/intern/yafray_Render.cpp	2008-04-19 17:37:16.293897581 -0700
@@ -7,6 +7,7 @@
 #include "yafray_Render.h"
 
 #include <math.h>
+#include <cstring>
 
 using namespace std;
 
diff -Naur blender-2.45.orig/source/gameengine/GamePlayer/common/GPC_KeyboardDevice.cpp blender-2.45/source/gameengine/GamePlayer/common/GPC_KeyboardDevice.cpp
--- blender-2.45.orig/source/gameengine/GamePlayer/common/GPC_KeyboardDevice.cpp	2008-04-19 17:10:03.980669133 -0700
+++ blender-2.45/source/gameengine/GamePlayer/common/GPC_KeyboardDevice.cpp	2008-04-19 17:42:29.390648003 -0700
@@ -32,6 +32,8 @@
 
 #include "GPC_KeyboardDevice.h"
 
+#include <cstdlib>
+
 /** 
  * NextFrame toggles currentTable with previousTable,
  * and copies relevant event information from previous to current table
