From eefa135633a91894819a97017f39563fe11e9eda Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sun, 10 Feb 2019 12:18:58 +0100
Subject: [PATCH] install libGL to lib/.mesa

---
 src/glx/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/glx/meson.build b/src/glx/meson.build
index a61f959..e863796 100644
--- a/src/glx/meson.build
+++ b/src/glx/meson.build
@@ -165,6 +165,7 @@ libgl = shared_library(
   ],
   version : gl_lib_version,
   install : true,
+  install_dir : 'lib/.mesa'
 )
 
 if with_tests
-- 
2.20.1

