From b05dff426cd49ec63646252c80cfdb25868b742e Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Tue, 7 Nov 2017 10:16:06 +0100
Subject: [PATCH 3/3] install libEGL.so* into lib/.mesa

---
 src/egl/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/egl/meson.build b/src/egl/meson.build
index 89a84fd890..dd9ee7936d 100644
--- a/src/egl/meson.build
+++ b/src/egl/meson.build
@@ -172,6 +172,7 @@ libegl = shared_library(
   dependencies : [deps_for_egl, dep_dl, dep_libdrm, dep_clock, dep_thread],
   install : true,
   version : egl_lib_version,
+  install_dir : 'lib/.mesa',
 )
 
 # If using glvnd the pkg-config header should not point to EGL_mesa, it should
-- 
2.18.0

