From f275e6addfa71b085e1585ca20e33bc74f3e19ae Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Tue, 7 Nov 2017 10:16:06 +0100
Subject: [PATCH 4/4] install libEGL.so* into lib/.mesa

---
 src/egl/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/egl/meson.build b/src/egl/meson.build
index 6537e4bdee..61173d6ba0 100644
--- a/src/egl/meson.build
+++ b/src/egl/meson.build
@@ -175,6 +175,7 @@ libegl = shared_library(
   dependencies : [deps_for_egl, dep_dl, dep_libdrm, dep_clock, dep_thread],
   install : true,
   version : egl_lib_version,
+  install_dir : 'lib/.mesa',
 )
 
 # If using glvnd the pkg-config header should not point to EGL_mesa, it should
-- 
2.17.0

