--- ltdl.c	2004-03-02 09:51:31.000000000 +0100
+++ ltdl.c.cp	2004-03-02 10:38:37.000000000 +0100
@@ -931,6 +931,8 @@
 #  define LT_GLOBAL		0
 #endif /* !LT_GLOBAL */
 
+#define LT_GLOBAL              RTLD_GLOBAL
+
 /* We may have to define LT_LAZY_OR_NOW in the command line if we
    find out it does not work in some platform. */
 #ifndef LT_LAZY_OR_NOW
@@ -980,15 +982,7 @@
 #ifdef QNX_LIBLTDL
   lt_module   module   = dlopen (filename, LT_GLOBAL | LT_LAZY_OR_NOW);
 #else
-#ifdef WITHOUT_RTLD_GROUP
-  lt_module   module   = dlopen (filename, LT_LAZY_OR_NOW | LT_MEMBER );
-#else
-#ifdef LT_GROUP
-  lt_module   module   = dlopen (filename, LT_GROUP | LT_LAZY_OR_NOW | LT_MEMBER );
-#else
-  lt_module   module   = dlopen (filename, LT_LAZY_OR_NOW | LT_MEMBER );
-#endif
-#endif
+  lt_module   module   = dlopen (filename, LT_GLOBAL | LT_LAZY_OR_NOW | LT_MEMBER );
 #endif
 
   if (!module)
