diff -urNp mp-5.1.0-rc1.orig/makefile.in mp-5.1.0-rc1/makefile.in
--- mp-5.1.0-rc1.orig/makefile.in	2008-09-18 21:27:51.000000000 +0200
+++ mp-5.1.0-rc1/makefile.in	2008-09-18 21:28:46.000000000 +0200
@@ -129,12 +129,12 @@ install: $(INSTALL_MSG) installdoc insta
 	[ -f $(APPNAME).1 ] && install -m 644 $(APPNAME).1 $(PREFIX)/share/man/man1/$(APPNAME).1 || true
 
 installdoc:
-	./mkinstalldirs $(PREFIX)/share/doc/$(APPNAME)
+	./mkinstalldirs $(PREFIX)/share/doc/mp
 	( cd $(MPDM); $(MAKE) installdoc )
 	( cd $(MPSL); $(MAKE) installdoc )
-	install -m 644 doc/* $(PREFIX)/share/doc/$(APPNAME)
-	install -m 644 $(ADD_DOCS) $(PREFIX)/share/doc/$(APPNAME)
-	[ -f doc/mp_index.html ] && install -m 644 doc/mp_index.html $(PREFIX)/share/doc/$(APPNAME)/index.html || true
+	install -m 644 doc/* $(PREFIX)/share/doc/mp
+	install -m 644 $(ADD_DOCS) $(PREFIX)/share/doc/mp
+	[ -f doc/mp_index.html ] && install -m 644 doc/mp_index.html $(PREFIX)/share/doc/mp/index.html || true
 
 uninstall: uninstall-mo
 	rm -f $(PREFIX)/bin/$(APPNAME)
