From 4d537395e0da1262937ede041e1793a9a18e66e4 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 26 Aug 2022 21:53:50 +0200
Subject: [PATCH] Fix rst2man file name searched for

---
 docs/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docs/meson.build b/docs/meson.build
index c4ba5e4..98d0baf 100644
--- a/docs/meson.build
+++ b/docs/meson.build
@@ -65,7 +65,7 @@ if build_docs
   )
 endif
 
-rst2man = find_program('rst2man', required: false)
+rst2man = find_program('rst2man.py', required: false)
 if get_option('man') and not rst2man.found()
   error('No rst2man found, but man pages were explicitly enabled')
 endif
-- 
2.37.1

