From 700ef65ffe33d5f078c5f9a8e4a92908b3615d33 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 18 Jul 2018 21:36:49 +0200
Subject: [PATCH] make gegl 0.4 acceptable

---
 configure | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/configure b/configure
index ee0a503..d3d5abd 100755
--- a/configure
+++ b/configure
@@ -16589,12 +16589,12 @@ if test -n "$GEGL_CFLAGS"; then
     pkg_cv_GEGL_CFLAGS="$GEGL_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gegl-0.2 >= 0.2.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "gegl-0.2 >= 0.2.0") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gegl-0.4 >= 0.2.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "gegl-0.4 >= 0.2.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_GEGL_CFLAGS=`$PKG_CONFIG --cflags "gegl-0.2 >= 0.2.0" 2>/dev/null`
+  pkg_cv_GEGL_CFLAGS=`$PKG_CONFIG --cflags "gegl-0.4 >= 0.2.0" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -16606,12 +16606,12 @@ if test -n "$GEGL_LIBS"; then
     pkg_cv_GEGL_LIBS="$GEGL_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gegl-0.2 >= 0.2.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "gegl-0.2 >= 0.2.0") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gegl-0.4 >= 0.2.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "gegl-0.4 >= 0.2.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_GEGL_LIBS=`$PKG_CONFIG --libs "gegl-0.2 >= 0.2.0" 2>/dev/null`
+  pkg_cv_GEGL_LIBS=`$PKG_CONFIG --libs "gegl-0.4 >= 0.2.0" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -16632,14 +16632,14 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        GEGL_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "gegl-0.2 >= 0.2.0" 2>&1`
+	        GEGL_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "gegl-0.4 >= 0.2.0" 2>&1`
         else
-	        GEGL_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "gegl-0.2 >= 0.2.0" 2>&1`
+	        GEGL_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "gegl-0.4 >= 0.2.0" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$GEGL_PKG_ERRORS" >&5
 
-	as_fn_error $? "Package requirements (gegl-0.2 >= 0.2.0) were not met:
+	as_fn_error $? "Package requirements (gegl-0.4 >= 0.2.0) were not met:
 
 $GEGL_PKG_ERRORS
 
-- 
2.18.0

