--- plug-ins/common/png.c.orig	2006-06-15 19:45:07.000000000 +0200
+++ plug-ins/common/png.c	2006-06-15 19:45:51.000000000 +0200
@@ -1012,7 +1012,7 @@
    * Done with the file...
    */
 
-  png_read_destroy (pp, info, NULL);
+  png_destroy_read_struct(&pp, &info, NULL);
 
   g_free (pixel);
   g_free (pixels);
@@ -1441,7 +1441,7 @@
     };
 
   png_write_end (pp, info);
-  png_write_destroy (pp);
+  png_destroy_write_struct(&pp, &info);
 
   g_free (pixel);
   g_free (pixels);
