diff -U10 -Naupr fvwm-2.5.21-16/libs/FlocaleCharset.c fvwm-2.5.21-17/libs/FlocaleCharset.c
--- fvwm-2.5.21-16/libs/FlocaleCharset.c	2007-03-03 13:25:06.000000000 +0100
+++ fvwm-2.5.21-17/libs/FlocaleCharset.c	2007-03-03 13:26:26.000000000 +0100
@@ -515,21 +515,21 @@ void FlocaleInit_X_Charset(Display *dpy,
 #endif
 			}
 		}
 	}
 	if (om)
 	{
 		XCloseOM(om);
 	}
 
 	if (FLCXOMCharsetList_num > 0 && FLCXOMCharsetList[0])
-		FLCXOMCharset = FLCXOMCharsetList[0];
+		FLCXOMCharset = FLCXOMCharsetList[FLCXOMCharsetList_num - 1];
 #endif
 }
 
 /* ---------------------------- interface functions ------------------------ */
 
 void FlocaleCharsetInit(Display *dpy, const char *module)
 {
 	static Bool initialized = False;
 	char *charset;
 
