diff -U10 -Naupr fvwm-2.5.21-18/fvwm/borders.c fvwm-2.5.21-19/fvwm/borders.c
--- fvwm-2.5.21-18/fvwm/borders.c	2007-03-03 13:26:38.000000000 +0100
+++ fvwm-2.5.21-19/fvwm/borders.c	2007-03-03 13:27:13.000000000 +0100
@@ -1388,21 +1388,21 @@ static void border_fill_pixmap_backgroun
 	if (alpha && alpha != bg->pixmap.alpha)
 	{
 		XFreePixmap(dpy, alpha);
 	}
 	return;
 }
 
 static void border_fill_fluxbox_handle(
 	Pixmap dest_pix, rectangle *dest_g, common_decorations_type *cd, Bool corner)
 {
-	corner = False;
+	//corner = False;
 	/* Main Color */
 	XFillRectangle(
 		dpy, dest_pix, corner ? cd->shadow_gc : cd->relief_gc, dest_g->x + 1, dest_g->y + 1,
 		dest_g->width - dest_g->x - 3, dest_g->height - dest_g->y - 2);
 	/* Right Shadow */
 	XFillRectangle(
 		dpy, dest_pix, cd->shadow_gc, dest_g->x + dest_g->width - 2, dest_g->y + 1,
 		1, dest_g->height - dest_g->y - 2);
 	/* Bottom Shadow */
 	XFillRectangle(
