--- drip-0.9.0rc3.old/plug-ins/magickfilter.cpp	2003-04-26 15:52:40.000000000 +0200
+++ drip-0.9.0rc3/plug-ins/magickfilter.cpp	2004-08-01 18:55:04.000000000 +0200
@@ -1052,7 +1052,7 @@
         #endif
     }
     if (value2 != -1) {
-        result = ContrastImage(ImageBuffer,(unsigned int)value2); //const unsigned int sharpen);
+        result = ContrastImage(ImageBuffer,(MagickBooleanType)value2); //const unsigned int sharpen);
         #ifdef DEBUG
         g_log(DRIP_LD,G_LOG_LEVEL_DEBUG,"Plugin: magickfilter Contrast");
         #endif
@@ -1099,7 +1099,7 @@
         #endif
     }
     if (value16 != -1) {
-        result = NegateImage(ImageBuffer,(unsigned int)value16); //const unsigned int grayscale);
+        result = NegateImage(ImageBuffer,(MagickBooleanType)value16); //const unsigned int grayscale);
         #ifdef DEBUG
         g_log(DRIP_LD,G_LOG_LEVEL_DEBUG,"Plugin: magickfilter Negate");
         #endif
@@ -1220,7 +1220,7 @@
         #endif
     }
     if (value33!=0 || value34!=0 || value35!=0) {
-        ImageBuffer2 = ShadeImage(ImageBuffer,(unsigned int)value33,value34,value35,&exception); //const unsigned int gray,double azimuth,double elevation
+        ImageBuffer2 = ShadeImage(ImageBuffer,(MagickBooleanType)value33,value34,value35,&exception); //const unsigned int gray,double azimuth,double elevation
         CatchException(&exception);
         DestroyImage(ImageBuffer);
         ImageBuffer = ImageBuffer2;
