--- git_version.sh.orig	2016-03-28 03:33:04.000000000 +0300
+++ git_version.sh	2016-03-29 14:27:28.981641741 +0300
@@ -88,7 +88,7 @@
                 if [ "x$outfile" = "x-" ]; then
                     : # keep default ifndef_symbol
                 else
-                    newdir=$(mktemp -t -d git_version_XXXXXX)
+                    newdir=$(mktemp -d git_version_XXXXXX)
                     outfilenew="$newdir/$1.new"
                     ifndef_symbol=`basename "$outfile" | $SED 's|\.|_|g; s|[^A-Za-z0-9_]||g' | tr a-z A-Z`
                 fi
@@ -369,7 +369,6 @@
         echo "$self: Output is new file, creating $outfile" >&2
         mv -f "$outfilenew" "$outfile"
     fi
-    rmdir "$newdir"
 fi
 
 # THE END.
