--- configure.in.orig	2003-09-27 09:33:24.000000000 -0400
+++ configure.in	2018-01-26 10:04:39.000000000 -0500
@@ -45,15 +45,17 @@
 BINS="packit"
 
 AC_ARG_ENABLE(debug, 
-    [ --enable-debug          Enable debugging],
-    [ if test "$GCC" = yes; then
-          DEBUG="yes"
-          CFLAGS="${CFLAGS} -DDEBUG -ggdb"
-      else
-          DEBUG="yes"
-          CFLAGS="${CFLAGS} -DDEBUG"
-      fi
-    ],)
+    [ --enable-debug          Enable debugging])
+
+AS_IF([test "x$enable_debug" != "xno"], [
+    if test "$GCC" = yes; then
+        DEBUG="yes"
+        CFLAGS="${CFLAGS} -DDEBUG -ggdb"
+    else
+        DEBUG="yes"
+        CFLAGS="${CFLAGS} -DDEBUG"
+    fi
+])
 
 AC_ARG_WITH(capture,
     [ --without-capture       Remove packet capture support],
