diff --git a/ud_socket.c b/ud_socket.c
index bb74241..a130fc8 100644
--- a/ud_socket.c
+++ b/ud_socket.c
@@ -58,7 +58,7 @@ ud_accept(int listenfd, struct ucred *cr
 	while (1) {
 		int newsock = 0;
 		struct sockaddr_un cliaddr;
-		int len = sizeof(struct sockaddr_un);
+		socklen_t len = sizeof(struct sockaddr_un);
 
 		newsock = accept(listenfd, (struct sockaddr *)&cliaddr, &len);
 		if (newsock < 0) {
