From c24501d2c139ff8e4fceb51e961e24c7848be0ec Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Fri, 18 Mar 2022 18:04:53 +0100
Subject: [PATCH 2/3] Use $(@D) instead of $(dir $@)

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 Makefile | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index eb6e99d6124c..2d6fb959b6b2 100644
--- a/Makefile
+++ b/Makefile
@@ -75,27 +75,27 @@ one.c: $(SRCS)
 jsdump.c: astnames.h opnames.h
 
 $(OUT)/%.o: %.c $(HDRS)
-	@ mkdir -p $(dir $@)
+	@ mkdir -p $(@D)
 	$(CC) $(CFLAGS) $(CPPFLAGS) -o $@ -c $<
 
 $(OUT)/libmujs.o: one.c $(HDRS)
-	@ mkdir -p $(dir $@)
+	@ mkdir -p $(@D)
 	$(CC) $(CFLAGS) $(CPPFLAGS) -o $@ -c $<
 
 $(OUT)/libmujs.a: $(OUT)/libmujs.o
-	@ mkdir -p $(dir $@)
+	@ mkdir -p $(@D)
 	$(AR) cr $@ $^
 
 $(OUT)/libmujs.$(SO_EXT): one.c $(HDRS)
-	@ mkdir -p $(dir $@)
+	@ mkdir -p $(@D)
 	$(CC) $(CFLAGS) $(CPPFLAGS) -fPIC -shared $(LDFLAGS) -o $@ $< -lm
 
 $(OUT)/mujs: $(OUT)/libmujs.o $(OUT)/main.o
-	@ mkdir -p $(dir $@)
+	@ mkdir -p $(@D)
 	$(CC) $(LDFLAGS) -o $@ $^ $(LIBREADLINE) -lm
 
 $(OUT)/mujs-pp: $(OUT)/libmujs.o $(OUT)/pp.o
-	@ mkdir -p $(dir $@)
+	@ mkdir -p $(@D)
 	$(CC) $(LDFLAGS) -o $@ $^ -lm
 
 .PHONY: $(OUT)/mujs.pc
-- 
2.35.1

