diff -ru nted-1.10.18.orig/dynarray.h nted-1.10.18/dynarray.h
--- nted-1.10.18.orig/dynarray.h	2011-01-27 14:09:44.000000000 +0100
+++ nted-1.10.18/dynarray.h	2014-12-07 13:08:15.461615576 +0100
@@ -26,6 +26,8 @@
 #ifndef DYN_ARRAY_H
 
 #define DYN_ARRAY_H
+
+#include <string.h>
 #include <gtk/gtk.h>
 
 #define CHECK_SIZES
diff -ru nted-1.10.18.orig/mainwindow.cpp nted-1.10.18/mainwindow.cpp
--- nted-1.10.18.orig/mainwindow.cpp	2011-01-27 14:09:44.000000000 +0100
+++ nted-1.10.18/mainwindow.cpp	2014-12-07 13:06:35.482865482 +0100
@@ -24,7 +24,7 @@
 #include <stdio.h>
 #include <time.h>
 #include <ft2build.h>
-#include <freetype/freetype.h>
+#include <freetype2/freetype.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 #include <cairo-ps.h>
