--- src/programpage.cpp	2003-10-24 07:47:00.000000000 -0700
+++ src.programpage.cpp	2003-10-24 07:48:21.000000000 -0700
@@ -49,43 +49,7 @@
  *	The contents and order must match
  *	the PROGGRP_ and *SEQ_ enums.
  */
-const char *program_sequence =
-"<?xml version='1.0'?><programs>
-	<group desc='%1'>
-		<program name='cdrdao'/>
-		<program name='cdrecord'/>
-	</group>
-
-	<group desc='%2'>
-		<program name='cdrecord'/>
-		<program name='cdrdao'/>
-	</group>
-
-	<group desc='%3'>
-		<program name='cdrdao'/>
-		<program name='readcd -> cdrecord'/>
-	</group>
-
-	<group desc='%4'>
-		<program name='bladeenc'/>
-		<program name='LAME'/>
-	</group>
-
-	<group desc='%5'>
-		<program name='cdrecord'/>
-		<program name='cdrdao'/>
-	</group>
-
-	<group desc='%6'>
-		<program name='cdda2wav'/>
-		<program name='cdparanoia'/>
-	</group>
-
-	<group desc='%7'>
-		<program name='cdrecord'/>
-		<program name='cdrdao'/>
-	</group>
-</programs>";
+const char *program_sequence = "<?xml version='1.0'?><programs> <group desc='%1'> <program name='cdrdao'/> <program name='cdrecord'/> </group> <group desc='%2'> <program name='cdrecord'/> <program name='cdrdao'/> </group> <group desc='%3'> <program name='cdrdao'/> <program name='readcd -> cdrecord'/> </group> <group desc='%4'> <program name='bladeenc'/> <program name='LAME'/> </group> <group desc='%5'> <program name='cdrecord'/> <program name='cdrdao'/> </group> <group desc='%6'> <program name='cdda2wav'/> <program name='cdparanoia'/> </group> <group desc='%7'> <program name='cdrecord'/> <program name='cdrdao'/> </group> </programs>";
 
 ArsonProgramPage::ArsonProgramPage (ArsonConfig &config, ArsonConfigDlg *pd, QWidget *parent)
 	: ArsonProgramPageBase(parent), ArsonConfigPage(config),
