diff -ur udev-146/rules/rules.d/50-udev-default.rules udev-146.cp/rules/rules.d/50-udev-default.rules
--- udev-146/rules/rules.d/50-udev-default.rules	2009-08-08 15:42:05.000000000 +0200
+++ udev-146.cp/rules/rules.d/50-udev-default.rules	2009-10-11 15:06:54.000000000 +0200
@@ -60,6 +60,10 @@
 SUBSYSTEM=="usb",		KERNEL=="lp*", NAME="usb/%k", SYMLINK+="usb%k", GROUP="lp"
 KERNEL=="lp[0-9]*",		GROUP="lp"
 KERNEL=="irlpt[0-9]*",		GROUP="lp"
+# hplip and cups 1.4+ use raw USB devices, so permissions should be similar to
+# the ones from the old usblp kernel module
+SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ENV{ID_USB_INTERFACES}=="", IMPORT{program}="usb_id --export %p"
+SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ENV{ID_USB_INTERFACES}==":0701*:", GROUP="lp", MODE="660"
 
 # block
 SUBSYSTEM=="block", GROUP="disk"
