Index: libgda-3.1.5/libgda/gda-command.h
===================================================================
--- libgda-3.1.5.orig/libgda/gda-command.h
+++ libgda-3.1.5/libgda/gda-command.h
@@ -25,8 +25,8 @@
 #define __GDA_COMMAND_H__
 
 #include <glib-object.h>
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
Index: libgda-3.1.5/libgda/gda-column.h
===================================================================
--- libgda-3.1.5.orig/libgda/gda-column.h
+++ libgda-3.1.5/libgda/gda-column.h
@@ -28,7 +28,7 @@
 
 #include <glib-object.h>
 #include <libgda/gda-value.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-decl.h>
 
 G_BEGIN_DECLS
Index: libgda-3.1.5/libgda/gda-value.h
===================================================================
--- libgda-3.1.5.orig/libgda/gda-value.h
+++ libgda-3.1.5/libgda/gda-value.h
@@ -27,8 +27,7 @@
 #define __GDA_VALUE_H__
 
 #include <time.h>
-#include <glib/glist.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <libxml/tree.h>
 #include <libgda/gda-decl.h>
Index: libgda-3.1.5/libgda/gda-config.h
===================================================================
--- libgda-3.1.5.orig/libgda/gda-config.h
+++ libgda-3.1.5/libgda/gda-config.h
@@ -23,7 +23,7 @@
 #ifndef __GDA_CONFIG_H__
 #define __GDA_CONFIG_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-data-model.h>
 #include <libgda/gda-parameter.h>
 
Index: libgda-3.1.5/libgda/gda-log.h
===================================================================
--- libgda-3.1.5.orig/libgda/gda-log.h
+++ libgda-3.1.5/libgda/gda-log.h
@@ -20,8 +20,7 @@
 #if !defined(__gda_log_h__)
 #  define __gda_log_h__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <time.h>
 
 G_BEGIN_DECLS
Index: libgda-3.1.5/libgda/gda-quark-list.h
===================================================================
--- libgda-3.1.5.orig/libgda/gda-quark-list.h
+++ libgda-3.1.5/libgda/gda-quark-list.h
@@ -24,8 +24,7 @@
 #ifndef __GDA_QUARK_LIST_H__
 #define __GDA_QUARK_LIST_H__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <glib-object.h>
 
 G_BEGIN_DECLS
Index: libgda-3.1.5/libgda/gda-row.h
===================================================================
--- libgda-3.1.5.orig/libgda/gda-row.h
+++ libgda-3.1.5/libgda/gda-row.h
@@ -27,7 +27,7 @@
 
 #include <glib-object.h>
 #include <libgda/gda-column.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-decl.h>
 
 G_BEGIN_DECLS
Index: libgda-3.1.5/libgda/gda-util.h
===================================================================
--- libgda-3.1.5.orig/libgda/gda-util.h
+++ libgda-3.1.5/libgda/gda-util.h
@@ -24,8 +24,7 @@
 #ifndef __GDA_UTIL_H__
 #define __GDA_UTIL_H__
 
-#include <glib/ghash.h>
-#include <glib/glist.h>
+#include <glib.h>
 #include "gda-parameter.h"
 #include "gda-row.h"
 #include "gda-connection.h"
Index: libgda-3.1.5/libgda/gda-column.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-column.c
+++ libgda-3.1.5/libgda/gda-column.c
@@ -23,8 +23,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <libgda/gda-column.h>
 #include <string.h>
 #include "gda-marshal.h"
Index: libgda-3.1.5/libgda/gda-command.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-command.c
+++ libgda-3.1.5/libgda/gda-command.c
@@ -21,9 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <libgda/gda-command.h>
 
 GType
Index: libgda-3.1.5/libgda/gda-config.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-config.c
+++ libgda-3.1.5/libgda/gda-config.c
@@ -38,7 +38,6 @@
 #include <sys/stat.h>
 #ifdef HAVE_FAM
 #include <fam.h>
-#include <glib/giochannel.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
Index: libgda-3.1.5/libgda/gda-data-model-array.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-data-model-array.c
+++ libgda-3.1.5/libgda/gda-data-model-array.c
@@ -22,7 +22,7 @@
  */
 
 #include <string.h>
-#include <glib/garray.h>
+#include <glib.h>
 #include <libgda/gda-data-model-array.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-row.h>
Index: libgda-3.1.5/libgda/gda-data-model-filter-sql.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-data-model-filter-sql.c
+++ libgda-3.1.5/libgda/gda-data-model-filter-sql.c
@@ -21,7 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gdataset.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <libsql/sql_parser.h>
 #include <glib/gi18n-lib.h>
Index: libgda-3.1.5/libgda/gda-data-model-hash.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-data-model-hash.c
+++ libgda-3.1.5/libgda/gda-data-model-hash.c
@@ -23,7 +23,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/ghash.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-data-model-hash.h>
 #include <libgda/gda-data-model-extra.h>
Index: libgda-3.1.5/libgda/gda-data-model.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-data-model.c
+++ libgda-3.1.5/libgda/gda-data-model.c
@@ -24,7 +24,7 @@
  */
 
 #include <glib/gi18n-lib.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <libgda/gda-data-model.h>
 #include <libgda/gda-data-model-private.h>
 #include <libgda/gda-data-model-extra.h>
Index: libgda-3.1.5/libgda/gda-init.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-init.c
+++ libgda-3.1.5/libgda/gda-init.c
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <gmodule.h>
 #include <libgda/libgda.h>
 #include <glib/gi18n-lib.h>
Index: libgda-3.1.5/libgda/gda-log.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-log.c
+++ libgda-3.1.5/libgda/gda-log.c
@@ -24,10 +24,6 @@
 #include <syslog.h>
 #endif
 #include <time.h>
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gutils.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-log.h>
 
Index: libgda-3.1.5/libgda/gda-quark-list.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-quark-list.c
+++ libgda-3.1.5/libgda/gda-quark-list.c
@@ -22,10 +22,7 @@
  */
 
 #include <libgda/gda-quark-list.h>
-#include <glib/ghash.h>
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 struct _GdaQuarkList {
 	GHashTable *hash_table;
Index: libgda-3.1.5/libgda/gda-server-operation.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-server-operation.c
+++ libgda-3.1.5/libgda/gda-server-operation.c
@@ -21,7 +21,7 @@
  */
 
 #include <stdlib.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <libgda/gda-marshal.h>
 #include <libgda/gda-dict.h>
 #include <libgda/gda-dict-type.h>
Index: libgda-3.1.5/libgda/gda-server-provider.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-server-provider.c
+++ libgda-3.1.5/libgda/gda-server-provider.c
@@ -22,7 +22,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <libgda/gda-server-provider.h>
 #include <libgda/gda-server-provider-extra.h>
 #include <libgda/gda-server-provider-private.h>
Index: libgda-3.1.5/libgda/gda-util.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-util.c
+++ libgda-3.1.5/libgda/gda-util.c
@@ -27,10 +27,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <string.h>
-#include <glib/gfileutils.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <libsql/sql_parser.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-log.h>
Index: libgda-3.1.5/libgda/gda-value.c
===================================================================
--- libgda-3.1.5.orig/libgda/gda-value.c
+++ libgda-3.1.5/libgda/gda-value.c
@@ -28,10 +28,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <time.h>
-#include <glib/gdate.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gstring.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-value.h>
 #include <libgda/gda-blob-op.h>
Index: libgda-3.1.5/libgda/sqlite/gda-sqlite.h
===================================================================
--- libgda-3.1.5.orig/libgda/sqlite/gda-sqlite.h
+++ libgda-3.1.5/libgda/sqlite/gda-sqlite.h
@@ -25,8 +25,7 @@
 #ifndef __GDA_SQLITE_H__
 #define __GDA_SQLITE_H__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <libgda/gda-value.h>
 #include <libgda/gda-connection.h>
 
Index: libgda-3.1.5/providers/bdb/gda-bdb.h
===================================================================
--- libgda-3.1.5.orig/providers/bdb/gda-bdb.h
+++ libgda-3.1.5/providers/bdb/gda-bdb.h
@@ -25,7 +25,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <db.h>
Index: libgda-3.1.5/providers/firebird/gda-firebird-provider.c
===================================================================
--- libgda-3.1.5.orig/providers/firebird/gda-firebird-provider.c
+++ libgda-3.1.5/providers/firebird/gda-firebird-provider.c
@@ -27,7 +27,7 @@
 #include <libgda/gda-server-provider-extra.h>
 #include <libgda/gda-command.h>
 #include <libgda/gda-parameter-list.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <string.h>
 #include "gda-firebird-provider.h"
 #include "gda-firebird-recordset.h"
Index: libgda-3.1.5/providers/firebird/gda-firebird-recordset.c
===================================================================
--- libgda-3.1.5.orig/providers/firebird/gda-firebird-recordset.c
+++ libgda-3.1.5/providers/firebird/gda-firebird-recordset.c
@@ -28,7 +28,7 @@
 #include <libgda/gda-quark-list.h>
 #include <libgda/gda-parameter-list.h>
 #include <glib/gi18n-lib.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <string.h>
 #include <math.h>
                                                                                                                              
Index: libgda-3.1.5/providers/freetds/gda-freetds-message.h
===================================================================
--- libgda-3.1.5.orig/providers/freetds/gda-freetds-message.h
+++ libgda-3.1.5/providers/freetds/gda-freetds-message.h
@@ -24,7 +24,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <tds.h>
Index: libgda-3.1.5/providers/freetds/gda-freetds-types.h
===================================================================
--- libgda-3.1.5.orig/providers/freetds/gda-freetds-types.h
+++ libgda-3.1.5/providers/freetds/gda-freetds-types.h
@@ -24,7 +24,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <tds.h>
Index: libgda-3.1.5/providers/freetds/gda-freetds.h
===================================================================
--- libgda-3.1.5.orig/providers/freetds/gda-freetds.h
+++ libgda-3.1.5/providers/freetds/gda-freetds.h
@@ -24,7 +24,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <tds.h>
Index: libgda-3.1.5/providers/postgres/gda-postgres.h
===================================================================
--- libgda-3.1.5.orig/providers/postgres/gda-postgres.h
+++ libgda-3.1.5/providers/postgres/gda-postgres.h
@@ -24,7 +24,7 @@
 #ifndef __GDA_POSTGRES_H__
 #define __GDA_POSTGRES_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include "gda-postgres-provider.h"
Index: libgda-3.1.5/providers/ibmdb2/gda-ibmdb2-types.h
===================================================================
--- libgda-3.1.5.orig/providers/ibmdb2/gda-ibmdb2-types.h
+++ libgda-3.1.5/providers/ibmdb2/gda-ibmdb2-types.h
@@ -23,7 +23,7 @@
 #ifndef __GDA_IBMDB2_TYPES_H__
 #define __GDA_IBMDB2_TYPES_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <sqlcli1.h>
Index: libgda-3.1.5/providers/ibmdb2/gda-ibmdb2.h
===================================================================
--- libgda-3.1.5.orig/providers/ibmdb2/gda-ibmdb2.h
+++ libgda-3.1.5/providers/ibmdb2/gda-ibmdb2.h
@@ -23,12 +23,11 @@
 #if !defined(__gda_ibmdb2_h__)
 #  define __gda_ibmdb2_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <libgda/gda-value.h>
-#include <glib/gi18n-lib.h>
 #include <sqlcli1.h>
 #include "gda-ibmdb2-provider.h"
 
Index: libgda-3.1.5/providers/ldap/gda-ldap.h
===================================================================
--- libgda-3.1.5.orig/providers/ldap/gda-ldap.h
+++ libgda-3.1.5/providers/ldap/gda-ldap.h
@@ -26,7 +26,7 @@
 #if !defined(__gda_ldap_h__)
 #  define __gda_ldap_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
Index: libgda-3.1.5/providers/mdb/gda-mdb-provider.c
===================================================================
--- libgda-3.1.5.orig/providers/mdb/gda-mdb-provider.c
+++ libgda-3.1.5/providers/mdb/gda-mdb-provider.c
@@ -22,7 +22,7 @@
  */
 
 #include <stdlib.h>
-#include <glib/gbacktrace.h>
+#include <glib.h>
 #include <virtual/gda-vconnection-data-model.h>
 #include <libgda/gda-data-model-array.h>
 #include <libgda/gda-data-model-private.h>
Index: libgda-3.1.5/providers/mdb/gda-mdb.h
===================================================================
--- libgda-3.1.5.orig/providers/mdb/gda-mdb.h
+++ libgda-3.1.5/providers/mdb/gda-mdb.h
@@ -24,7 +24,7 @@
 #ifndef __GDA_MDB_H__
 #define __GDA_MDB_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
Index: libgda-3.1.5/providers/msql/gda-msql.h
===================================================================
--- libgda-3.1.5.orig/providers/msql/gda-msql.h
+++ libgda-3.1.5/providers/msql/gda-msql.h
@@ -22,7 +22,7 @@
 #ifndef __gda_msql_h__
 #define __gda_msql_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
Index: libgda-3.1.5/providers/mysql/gda-mysql.h
===================================================================
--- libgda-3.1.5.orig/providers/mysql/gda-mysql.h
+++ libgda-3.1.5/providers/mysql/gda-mysql.h
@@ -25,7 +25,7 @@
 #ifndef __GDA_MYSQL_H__
 #define __GDA_MYSQL_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
Index: libgda-3.1.5/providers/odbc/gda-odbc.h
===================================================================
--- libgda-3.1.5.orig/providers/odbc/gda-odbc.h
+++ libgda-3.1.5/providers/odbc/gda-odbc.h
@@ -25,7 +25,7 @@
 #if !defined(__gda_odbc_h__)
 #  define __gda_odbc_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
Index: libgda-3.1.5/providers/oracle/gda-oracle.h
===================================================================
--- libgda-3.1.5.orig/providers/oracle/gda-oracle.h
+++ libgda-3.1.5/providers/oracle/gda-oracle.h
@@ -31,7 +31,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
Index: libgda-3.1.5/providers/sybase/gda-sybase.h
===================================================================
--- libgda-3.1.5.orig/providers/sybase/gda-sybase.h
+++ libgda-3.1.5/providers/sybase/gda-sybase.h
@@ -32,7 +32,7 @@
 #  include <config.h>
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <ctpublic.h>
