diff -ru xjadeo-0.4.1.orig/configure.ac xjadeo-0.4.1/configure.ac
--- xjadeo-0.4.1.orig/configure.ac	2007-06-16 18:53:10.000000000 +0200
+++ xjadeo-0.4.1/configure.ac	2008-02-14 09:49:08.000000000 +0100
@@ -289,6 +289,13 @@
 	])
 fi
 
+dnl Newer FFMPEG (svn) does not support img_convert, use swscaler instead.
+AC_CHECK_HEADER(ffmpeg/swscale.h, [HAVE_SWSCALE=yes])
+if test "x$HAVE_SWSCALE" = "xyes"; then
+	AC_DEFINE(HAVE_SWSCALE, 1, [Have ffmpeg swscaler.])
+	FFMPEG_LIBS="$FFMPEG_LIBS -lswscale"
+fi
+
 AC_SUBST(FFMPEG_CFLAGS)
 AC_SUBST(FFMPEG_LIBS)
 
diff -ru xjadeo-0.4.1.orig/src/xjadeo/xjadeo.c xjadeo-0.4.1/src/xjadeo/xjadeo.c
--- xjadeo-0.4.1.orig/src/xjadeo/xjadeo.c	2007-06-16 18:57:11.000000000 +0200
+++ xjadeo-0.4.1/src/xjadeo/xjadeo.c	2008-02-14 09:51:28.000000000 +0100
@@ -35,6 +35,11 @@
 
 #include <ffmpeg/avcodec.h>
 #include <ffmpeg/avformat.h>
+#ifdef HAVE_SWSCALE
+#include <ffmpeg/swscale.h>
+static int sws_flags = SWS_FAST_BILINEAR;
+struct SwsContext *scale_context = NULL;
+#endif
 
 #include <time.h>
 #include <getopt.h>
@@ -657,9 +662,23 @@
 			/* Did we get a video frame? */
 			if(frameFinished) {
 				/* Convert the image from its native format to FMT */
+#ifdef HAVE_SWSCALE
+			/* Inspired by rev 6649 of ffmpeg/trunk/vhook/watermark.c */
+			scale_context = sws_getCachedContext(scale_context,
+				pCodecCtx->width, pCodecCtx->height, pCodecCtx->pix_fmt,
+				pCodecCtx->width, pCodecCtx->height, render_fmt,
+				sws_flags, NULL, NULL, NULL);
+			if(scale_context == NULL)
+			{
+				fprintf(stderr, "Failed to setup scaler! That's fatal for now (FIXME).\n");
+				exit(100);
+			}
+			sws_scale(scale_context, pFrame->data, pFrame->linesize, 0, pCodecCtx->height, pFrameFMT->data, pFrameFMT->linesize); 
+#else
 				img_convert((AVPicture *)pFrameFMT, render_fmt, 
 					(AVPicture*)pFrame, pCodecCtx->pix_fmt, pCodecCtx->width, 
 					pCodecCtx->height);
+#endif
 
 				render_buffer(buffer); // in pFrameFMT
 				av_free_packet(&packet); /* XXX */
@@ -710,6 +729,9 @@
 
 	//Close the video file
 	av_close_input_file(pFormatCtx);
+#ifdef HAVE_SWSCALE
+	sws_freeContext(scale_context);
+#endif
 	return (0);
 }
 
