diff -Nuar Mesa-7.3.orig/src/mesa/Makefile Mesa-7.3/src/mesa/Makefile
--- Mesa-7.3.orig/src/mesa/Makefile	2009-02-08 22:21:58.000000000 +0000
+++ Mesa-7.3/src/mesa/Makefile	2009-02-08 22:26:32.000000000 +0000
@@ -114,14 +114,30 @@
 
 install-headers:
 	$(INSTALL) -d $(DESTDIR)$(INSTALL_INC_DIR)/GL
+	$(INSTALL) -d $(DESTDIR)$(INSTALL_INC_DIR)/GL/mesa
+	$(INSTALL) -m 644  $(TOP)/include/GL/gl.h \
+		$(DESTDIR)$(INSTALL_INC_DIR)/GL/mesa/
+	$(INSTALL) -m 644  $(TOP)/include/GL/glx.h \
+		$(DESTDIR)$(INSTALL_INC_DIR)/GL/mesa/
+	$(INSTALL) -m 644  $(TOP)/include/GL/glxext.h \
+		$(DESTDIR)$(INSTALL_INC_DIR)/GL/mesa/
+	$(INSTALL) -m 644  $(TOP)/include/GL/glext.h \
+		$(DESTDIR)$(INSTALL_INC_DIR)/GL/mesa/
+	mkdir -p $(TOP)/include/GL/backups
+	mv $(TOP)/include/GL/gl.h $(TOP)/include/GL/backups/
+	mv $(TOP)/include/GL/glx.h $(TOP)/include/GL/backups/
+	mv $(TOP)/include/GL/glxext.h $(TOP)/include/GL/backups/
+	mv $(TOP)/include/GL/glext.h $(TOP)/include/GL/backups/
 	$(INSTALL) -m 644 $(TOP)/include/GL/*.h \
 		$(DESTDIR)$(INSTALL_INC_DIR)/GL
+	mv $(TOP)/include/GL/backups/* $(TOP)/include/GL/
 
 install-libgl: default gl.pc install-headers
 	$(INSTALL) -d $(DESTDIR)$(INSTALL_LIB_DIR)
+	$(INSTALL) -d $(DESTDIR)$(INSTALL_LIB_DIR)/mesa
 	$(INSTALL) -d $(DESTDIR)$(INSTALL_LIB_DIR)/pkgconfig
 	$(INSTALL) $(TOP)/$(LIB_DIR)/$(GL_LIB_GLOB) \
-		$(DESTDIR)$(INSTALL_LIB_DIR)
+		$(DESTDIR)$(INSTALL_LIB_DIR)/mesa/
 	$(INSTALL) -m 644 gl.pc $(DESTDIR)$(INSTALL_LIB_DIR)/pkgconfig
 
 install-osmesa: default osmesa.pc

