diff -Naur ./mesalib-git.orig/src/mesa/Makefile mesalib-git/src/mesa/Makefile
--- ./mesalib-git.orig/src/mesa/Makefile	2009-07-24 12:06:06.000000000 +0200
+++ mesalib-git/src/mesa/Makefile	2009-07-25 21:02:07.827668651 +0200
@@ -123,14 +123,30 @@
 
 install-headers:
 	$(INSTALL) -d $(DESTDIR)$(INSTALL_INC_DIR)/GL
+	$(INSTALL) -d $(DESTDIR)$(INSTALL_INC_DIR)/GL/.mesa
+	$(INSTALL) -m 644  $(TOP)/include/GL/gl.h \
+		$(DESTDIR)$(INSTALL_INC_DIR)/GL/.mesa/
+	$(INSTALL) -m 644  $(TOP)/include/GL/glx.h \
+		$(DESTDIR)$(INSTALL_INC_DIR)/GL/.mesa/
+	$(INSTALL) -m 644  $(TOP)/include/GL/glxext.h \
+		$(DESTDIR)$(INSTALL_INC_DIR)/GL/.mesa/
+	$(INSTALL) -m 644  $(TOP)/include/GL/glext.h \
+		$(DESTDIR)$(INSTALL_INC_DIR)/GL/.mesa/
+	mkdir -p $(TOP)/include/GL/backups
+	mv $(TOP)/include/GL/gl.h $(TOP)/include/GL/backups/
+	mv $(TOP)/include/GL/glx.h $(TOP)/include/GL/backups/
+	mv $(TOP)/include/GL/glxext.h $(TOP)/include/GL/backups/
+	mv $(TOP)/include/GL/glext.h $(TOP)/include/GL/backups/
 	$(INSTALL) -m 644 $(TOP)/include/GL/*.h \
 		$(DESTDIR)$(INSTALL_INC_DIR)/GL
+	mv $(TOP)/include/GL/backups/* $(TOP)/include/GL/
 
 install-libgl: default gl.pc install-headers
 	$(INSTALL) -d $(DESTDIR)$(INSTALL_LIB_DIR)
+	$(INSTALL) -d $(DESTDIR)$(INSTALL_LIB_DIR)/.mesa
 	$(INSTALL) -d $(DESTDIR)$(INSTALL_LIB_DIR)/pkgconfig
 	$(MINSTALL) $(TOP)/$(LIB_DIR)/$(GL_LIB_GLOB) \
-		$(DESTDIR)$(INSTALL_LIB_DIR)
+		$(DESTDIR)$(INSTALL_LIB_DIR)/.mesa
 	$(INSTALL) -m 644 gl.pc $(DESTDIR)$(INSTALL_LIB_DIR)/pkgconfig
 
 install-osmesa: default osmesa.pc
