diff --git a/src/egl/Makefile.am b/src/egl/Makefile.am
index 5c2ba30..43e688b 100644
--- a/src/egl/Makefile.am
+++ b/src/egl/Makefile.am
@@ -32,7 +32,8 @@ AM_CFLAGS = \
 	$(EGL_CFLAGS) \
 	-D_EGL_NATIVE_PLATFORM=$(EGL_NATIVE_PLATFORM)
 
-lib_LTLIBRARIES = libEGL.la
+egllibdir = ${libdir}/.mesa
+egllib_LTLIBRARIES = libEGL.la
 
 libEGL_la_SOURCES = \
 	$(LIBEGL_C_FILES)
diff --git a/src/glx/Makefile.am b/src/glx/Makefile.am
index 6e50e09..0d6479d 100644
--- a/src/glx/Makefile.am
+++ b/src/glx/Makefile.am
@@ -52,7 +52,8 @@ AM_CFLAGS = \
 	$(GLPROTO_CFLAGS) \
 	$(X11_INCLUDES)
 
-lib_LTLIBRARIES = lib@GL_LIB@.la
+gldir = ${libdir}/.mesa
+gl_LTLIBRARIES = lib@GL_LIB@.la
 
 noinst_LTLIBRARIES = libglx.la
 
diff --git a/src/mapi/Makefile.am b/src/mapi/Makefile.am
index 307e05d..42c7e59 100644
--- a/src/mapi/Makefile.am
+++ b/src/mapi/Makefile.am
@@ -159,7 +159,8 @@ GLES_include_HEADERS = \
 	$(top_srcdir)/include/GLES/glext.h \
 	$(top_srcdir)/include/GLES/glplatform.h
 
-lib_LTLIBRARIES += es1api/libGLESv1_CM.la
+glesdir = ${libdir}/.mesa
+gles_LTLIBRARIES = es1api/libGLESv1_CM.la
 
 es1api_libGLESv1_CM_la_SOURCES = entry.c es1api/glapi_mapi_tmp.h
 es1api_libGLESv1_CM_la_CFLAGS = \
@@ -205,7 +206,8 @@ GLES3_include_HEADERS = \
 	$(top_srcdir)/include/GLES3/gl3ext.h \
 	$(top_srcdir)/include/GLES3/gl3platform.h
 
-lib_LTLIBRARIES += es2api/libGLESv2.la
+gles2dir = ${libdir}/.mesa
+gles2_LTLIBRARIES = es2api/libGLESv2.la
 
 es2api_libGLESv2_la_SOURCES = entry.c es2api/glapi_mapi_tmp.h
 es2api_libGLESv2_la_CFLAGS = \
diff --git a/src/mesa/Makefile.am b/src/mesa/Makefile.am
index 69506f2..7894ad6 100644
--- a/src/mesa/Makefile.am
+++ b/src/mesa/Makefile.am
@@ -33,7 +33,7 @@ if HAVE_OSMESA
 SUBDIRS += drivers/osmesa
 endif
 
-gldir = $(includedir)/GL
+gldir = $(includedir)/GL/.mesa
 gl_HEADERS = $(top_srcdir)/include/GL/*.h
 
 .PHONY: main/git_sha1.h.tmp
-- 
2.5.0

