--- a/growisofs.c	2007-10-24 07:40:22.000000000 -0700
+++ b/growisofs.c	2007-10-24 07:39:33.000000000 -0700
@@ -560,6 +560,7 @@
 #if defined(__linux)
 
 #include <linux/types.h>
+#include <limits.h>
 #include <linux/cdrom.h>
 #include <sys/ioctl.h>
 #include <sys/prctl.h>
--- a/transport.hxx	2007-10-24 07:40:57.000000000 -0700
+++ b/transport.hxx	2007-10-24 07:39:54.000000000 -0700
@@ -132,6 +132,7 @@
 #if defined(__linux)
 
 #include <sys/ioctl.h>
+#include <limits.h>
 #include <linux/cdrom.h>
 #include <mntent.h>
 #include <sys/wait.h>
