From c8828d072aec847801d4672cbe1c1f8105801117 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sat, 10 May 2014 08:46:26 +0200
Subject: [PATCH] fix scan-view's module search path

---
 tools/scan-view/scan-view | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/scan-view/scan-view b/tools/scan-view/scan-view
index fb27da6..5b66ff9 100755
--- a/tools/scan-view/scan-view
+++ b/tools/scan-view/scan-view
@@ -10,6 +10,8 @@ import time
 import urllib
 import webbrowser
 
+sys.path.append("/usr/share/llvm/scan-view")
+
 # How long to wait for server to start.
 kSleepTimeout = .05
 kMaxSleeps = int(60 / kSleepTimeout)
-- 
1.9.1

