From 72d31e2acc1dad39d4c778e21da3d632e61ae66f Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sun, 17 Sep 2017 23:02:48 +0200
Subject: [PATCH] fix compile error

---
 lib/Support/NativeFormatting.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/Support/NativeFormatting.cpp b/lib/Support/NativeFormatting.cpp
index b951a88..3ed982c 100644
--- a/lib/Support/NativeFormatting.cpp
+++ b/lib/Support/NativeFormatting.cpp
@@ -163,10 +163,10 @@ void llvm::write_double(raw_ostream &S, double N, FloatStyle Style,
                         Optional<size_t> Precision) {
   size_t Prec = Precision.getValueOr(getDefaultPrecision(Style));
 
-  if (std::isnan(N)) {
+  if (isnan(N)) {
     S << "nan";
     return;
-  } else if (std::isinf(N)) {
+  } else if (isinf(N)) {
     S << "INF";
     return;
   }
-- 
2.14.1

