--- dspam_logrotate-current     2006-04-17 13:25:36.000000000 -0500
+++ dspam_logrotate-new 2006-04-17 13:26:14.000000000 -0500
@@ -35,11 +35,11 @@
     my($cutoffTimestamp);
     my($printStats);
     ($filename, $cutoffTimestamp, $printStats) = @_;
-
-#preserve timestamps and ownership of the files
-my ( $mode, $uid, $gid ) = ( stat($filename) )[ 2, 4, 5 ];
-$mode = $mode & 07777;
-
+
+       #preserve timestamps and ownership of the files
+       my ( $mode, $uid, $gid ) = ( stat($filename) )[ 2, 4, 5 ];
+       $mode = $mode & 07777;
+
     # Generate names for the temporary files.
     my($tempInputFile) = $filename . ".in";
     my($tempOutputFile) = $filename . ".out";
@@ -83,7 +83,12 @@

     # Rename our temporary output file to the original log file name.
     close OUTFILE;
-    rename $tempOutputFile, $filename;
+
+       #apply previous permissions and ownerships to temp file
+       chmod $mode, $tempOutputFile;
+       chown $uid, $gid, $tempOutputFile;
