--- src/daemon.c.orig
+++ src/daemon.c
@@ -1,4 +1,4 @@
-/* $Id: daemon.c,v 1.22 2011/06/28 00:13:48 sbajic Exp $ */
+/* $Id: daemon.c,v 1.23 2012/02/09 22:05:07 sbajic Exp $ */
 
 /*
  DSPAM
@@ -992,33 +992,37 @@ char *daemon_expect(THREAD_CTX *TTX, const char *command) {
   char buf[128];
   char *cmd;
 
-  cmd = daemon_getline(TTX, 300); 
+  cmd = daemon_getline(TTX, 300);
   if (cmd == NULL)
     return NULL;
 
   while(strncasecmp(cmd, command, strlen(command))) {
     if (!strncasecmp(cmd, "QUIT", 4)) {
       free(cmd);
-      daemon_reply(TTX, LMTP_QUIT, "2.0.0", "OK"); 
+      daemon_reply(TTX, LMTP_QUIT, "2.0.0", "OK");
       return NULL;
-    }
-    if (!strncasecmp(cmd, "RSET", 4)) { 
+    } else if (!strncasecmp(cmd, "RSET", 4)) {
       snprintf(buf, sizeof(buf), "%d OK", LMTP_OK);
       if (send_socket(TTX, buf)<=0)
         return NULL;
       free(cmd);
-      return "RSET";
+      if (!strncasecmp(command, "LHLO", 4)) {
+        cmd = daemon_getline(TTX, 300);
+        if (cmd == NULL)
+          return NULL;
+      } else {
+        return "RSET";
+      }
+    } else {
+      snprintf(buf, sizeof(buf), "%d 5.0.0 Need %s here.", LMTP_BAD_CMD, command);
+      if (send_socket(TTX, buf)<=0)
+        return NULL;
+      free(cmd);
+      cmd = daemon_getline(TTX, 300);
+      if (cmd == NULL)
+        return NULL;
     }
-
-    snprintf(buf, sizeof(buf), "%d 5.0.0 Need %s here.", LMTP_BAD_CMD, command);
-
-    if (send_socket(TTX, buf)<=0)
-      return NULL;
-    free(cmd);
-    cmd = daemon_getline(TTX, 300);
-    if (cmd == NULL)
-      return NULL;
-  } 
+  }
   return cmd;
 }
 
-- 
