--- VEX/priv/host-x86/hdefs.c~	2005-09-17 16:06:10.000000000 +0200
+++ VEX/priv/host-x86/hdefs.c	2005-09-17 17:54:31.000000000 +0200
@@ -2315,7 +2315,6 @@
       if (0) vex_printf("EMIT FENCE\n");
       switch (i->Xin.MFence.subarch) {
          case VexSubArchX86_sse0:
-            vassert(0); /* awaiting test case */
             /* lock addl $0,0(%esp) */
             *p++ = 0xF0; *p++ = 0x83; *p++ = 0x44; 
             *p++ = 0x24; *p++ = 0x00; *p++ = 0x00;
