diff -ru valgrind-3.11.0.orig/configure.ac valgrind-3.11.0/configure.ac
--- valgrind-3.11.0.orig/configure.ac	2015-09-22 22:26:31.000000000 +0200
+++ valgrind-3.11.0/configure.ac	2016-02-07 09:44:39.568960224 +0100
@@ -3715,7 +3715,7 @@
 safe_CXXFLAGS=$CXXFLAGS
 CXXFLAGS="$mflag_primary"
 safe_LIBS="$LIBS"
-LIBS="-lboost_thread-mt -lboost_system-mt $LIBS"
+LIBS="-lboost_thread -lboost_system $LIBS"
 
 AC_LINK_IFELSE([AC_LANG_SOURCE([
 #include <boost/thread.hpp>
@@ -3730,7 +3730,7 @@
 [
 ac_have_boost_1_35=yes
 AC_SUBST([BOOST_CFLAGS], [])
-AC_SUBST([BOOST_LIBS], ["-lboost_thread-mt -lboost_system-mt"])
+AC_SUBST([BOOST_LIBS], ["-lboost_thread -lboost_system"])
 AC_MSG_RESULT([yes])
 ], [
 ac_have_boost_1_35=no
