--- odkda-cpp/compile.sh	2005-09-26 21:59:37.000000000 +0000
+++ odkda-cpp/compile.sh.cp	2005-09-26 22:01:30.000000000 +0000
@@ -16,7 +16,6 @@
 mkdir -p bin
 echo "Compiling ODKDA-CPP"
 
-export CFLAGS="-g -DDEBUG -DCPP"
 
 g++ provider.cpp -fPIC -I../include -I./include $CFLAGS -c -o bin/provider.o
 g++ value.cpp -fPIC -I../include -I./include $CFLAGS -c -o bin/value.o
@@ -24,4 +23,4 @@
 g++ row.cpp -fPIC -I../include -I./include $CFLAGS -c -o bin/row.o
 
 echo "    * Creating Library"
-g++ -shared -Wl,-soname,libodkda-cpp.so -o bin/libodkda-cpp.so bin/*.o -lodkda $CFLAGS
+g++ -shared -Wl,-soname,libodkda-cpp.so -o bin/libodkda-cpp.so bin/*.o 
