--- SConstruct.orig	2017-09-25 11:49:27.703917934 +0200
+++ SConstruct	2017-09-25 11:49:38.126918570 +0200
@@ -182,8 +182,8 @@
 # HANDLING OF OPTION VARIABLES
 
 unknown = opts.UnknownVariables()
-if unknown:
-  print 'Warning: Used unknown variables:', ', '.join(unknown.keys())
+#if unknown:
+#  print 'Warning: Used unknown variables:', ', '.join(unknown.keys())
 
 apr = str(env['APR'])
 apu = str(env['APU'])
