Only in panelmm-0.1/panel--: applet.cc
diff -ru panelmm-0.1/panel--/applet.gen_h /usr/src/panelmm-0.1/panel--/applet.gen_h
--- panelmm-0.1/panel--/applet.gen_h	1999-12-23 12:15:24.000000000 +0100
+++ /usr/src/panelmm-0.1/panel--/applet.gen_h	2002-06-24 20:38:19.000000000 +0200
@@ -23,6 +23,8 @@
 #include <gtk--/plug.h>
 #include <applet-widget.h>
 
+using std::string;
+
 CLASS_START(Gnome);
 class Applet : public Gtk::Plug
 {
Only in panelmm-0.1/panel--: applet.h
Only in panelmm-0.1/panel--: applet-main.cc
diff -ru panelmm-0.1/panel--/applet-main.gen_h /usr/src/panelmm-0.1/panel--/applet-main.gen_h
--- panelmm-0.1/panel--/applet-main.gen_h	1999-12-23 12:15:24.000000000 +0100
+++ /usr/src/panelmm-0.1/panel--/applet-main.gen_h	2002-06-24 20:39:44.000000000 +0200
@@ -24,6 +24,8 @@
 #include <gtk--/main.h>
 #include <applet-widget.h>
 
+using std::string;
+
 CLASS_START(Gnome);
 class Applet_Main : public Gtk::Main
 {
@@ -81,9 +83,9 @@
 Applet_Main::Applet_Main(const string &app_id,
 			       const string &app_version,
 			       int argc, char **argv,
-			       struct poptOption *options = 0,
-			       unsigned int flags = 0,
-			       poptContext *return_ctx = 0)
+			       struct poptOption *options,
+			       unsigned int flags,
+			       poptContext *return_ctx)
 {
   if (instance_) { 
     g_warning("Gnome_Applet_Main::Gnome_Applet_Main called twice!\n"); 
Only in panelmm-0.1/panel--: applet-main.h
Only in /usr/src/panelmm-0.1/: panelmmConf.sh
Only in /usr/src/panelmm-0.1/scripts: Makefile
