From d5458c0a4afd04251cd97e707ddfbb5ff3f2aa2f Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 19:23:49 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 microlens-th.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/microlens-th.cabal b/microlens-th.cabal
index d0e1f5b..0ce00c7 100644
--- a/microlens-th.cabal
+++ b/microlens-th.cabal
@@ -32,10 +32,10 @@ library
   -- other-extensions:
   build-depends:       base >=4.5 && <5
                      , microlens >=0.4.0 && <0.5
-                     , containers >=0.4.0 && <0.7
+                     , containers >=0.4.0
                      , transformers
                      -- lens has >=2.4, but GHC 7.4 shipped with 2.7
-                     , template-haskell >=2.7 && <2.15
+                     , template-haskell >=2.7
                      , th-abstraction >=0.2.1 && <0.3
 
   if flag(inlining)
-- 
2.19.1

