From 8c260555811bca4ffdca3903539ba62ae71506b0 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2012 09:14:28 +0100
Subject: [PATCH] remove example causing install error

---
 makefile.in | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/makefile.in b/makefile.in
index d8ba904..67069b4 100644
--- a/makefile.in
+++ b/makefile.in
@@ -1481,11 +1481,6 @@ install_bin:	all oforms install_gui
 
 # install recursive
 install:	install-main examples
-#ifdef QT
-	$(INSTALL) -m 755 examples/xcmsevents/qcmsevents "$(DESTDIR)$(bindir)"
-	mkdir -p "$(DESTDIR)$(desktopdir)"
-	$(INSTALL) -m 644 "$(SRCDIR)/examples/xcmsevents/qcmsevents-applet.desktop" "$(DESTDIR)$(desktopdir)"
-#endif
 	$(INSTALL) -m 755 examples/oyranos_profile_graph/oyranos-profile-graph "$(DESTDIR)$(bindir)"
 	echo ... $@ done
 
-- 
1.8.0

