diff -Naur oyranos-0.1.7.orig/makefile.in oyranos-0.1.7/makefile.in
--- oyranos-0.1.7.orig/makefile.in	2008-04-26 23:07:47.000000000 -0700
+++ oyranos-0.1.7/makefile.in	2008-04-26 23:13:15.000000000 -0700
@@ -31,7 +31,7 @@
 ELEKTRA_STATICTEST_LIBS = `test -f $(ELEKTRA_VERSION)/src/libelektra/libelektra.a && echo $(ELEKTRA_VERSION)/src/libelektra/libelektra.a || (test -f $(libdir)/libelektra.a && (echo $(libdir)/libelektra.a ; test -f $(libdir)/libelektra-filesys.a && echo $(libdir)/libelektra-filesys.a || (test -f $(libdir)/elektra/libelektra-filesys.a && echo $(libdir)/elektra/libelektra-filesys.a)) || echo $(ELEKTRA_LIBS))`
 
 LDLIBS = $(LDFLAGS) -L$(libdir) -L. \
-	$(ELEKTRA_STATICTEST_LIBS) -lc $(I18N_LIB)
+	$(ELEKTRA_STATICTEST_LIBS) -lc -lxml2 -ldb_cxx $(I18N_LIB)
 
 
 CPP_HEADERS = \
