diff -Naur oyranos-0.1.7.orig/configure oyranos-0.1.7/configure
--- oyranos-0.1.7.orig/configure	2008-04-26 22:30:50.719216830 -0700
+++ oyranos-0.1.7/configure	2008-04-26 22:37:22.139469402 -0700
@@ -409,11 +409,7 @@
   fi
 fi
 
-if [ "$internalelektra" != "no" ]; then
-  ELEKTRA_VERSION="elektra-0.6.4"
-  export ELEKTRA_VERSION
-fi
-export internalelektra
+export internalelektra="no"
 
 
 # set some compiler and linker options
