--- install.pl~	2007-06-04 02:13:34.000000000 +0000
+++ install.pl	2007-08-21 11:57:27.000000000 +0000
@@ -224,15 +224,7 @@
 
 if ($client_install) {
 
-    ### we are installing as a normal user instead of root, so see
-    ### if it is ok to install within the user's home directory
     $homedir = '';
-    if ($config_homedir) {
-        $homedir = $config_homedir;
-    } else {
-        $homedir = $ENV{'HOME'} or die '[*] Could not get home ',
-            "directory, set the $config_homedir var.";
-    }
 
     print
 "    The fwknop client will be installed at $homedir/bin/fwknop, and a few\n",
