--- install.pl~	2007-06-04 02:13:34.000000000 +0000
+++ install.pl	2007-08-21 13:27:36.000000000 +0000
@@ -640,16 +640,12 @@
             }
         }
 
-        if (-d $INIT_DIR) {
-            &enable_fwknop_at_boot($distro) unless $data_method =~ /client/i;
-        }
-
         if ($data_method =~ /client/i) {
             print "\n[+] fwknop has been installed!\n\n";
         } else {
             print
 "\n[+] fwknop has been installed!  To start in server mode, run\n\n",
-"    \"/etc/init.d/fwknop start\"\n\n",
+"    \"telinit run fwknopd start\"\n\n",
 "    Note: You will need to edit $config{'FWKNOP_CONF_DIR'}/access.conf for fwknop to\n",
 "    function properly in server mode.  More information can be found in\n",
 "    the fwknop manpage.\n\n";
