diff --unified --recursive --text --new-file gerbera-2.3.0.orig/src/web/config_load.cc gerbera-2.3.0/src/web/config_load.cc
--- gerbera-2.3.0.orig/src/web/config_load.cc	2024-12-29 10:46:14.272390575 +0100
+++ gerbera-2.3.0/src/web/config_load.cc	2024-12-29 10:46:28.379089688 +0100
@@ -658,7 +658,7 @@

         item = values.append_child(CONFIG_LOAD_ITEM);
         createItem(item, cs->getItemPath(indexList, { ConfigVal::A_TRANSCODING_MIMETYPE_FILTER, ConfigVal::A_TRANSCODING_PROFILES_PROFLE_NOTRANSCODING }), cs->option, ConfigVal::A_TRANSCODING_PROFILES_PROFLE_NOTRANSCODING, cs);
-        setValue(item, fmt::join(filter->getNoTranscodingMimeTypes(), ","));
+        setValue(item, fmt::format("{}", fmt::join(filter->getNoTranscodingMimeTypes(), ",")));

         item = values.append_child(CONFIG_LOAD_ITEM);
         createItem(item, cs->getItemPath(indexList, { ConfigVal::A_TRANSCODING_MIMETYPE_FILTER, ConfigVal::A_TRANSCODING_PROFILES_PROFLE_CLIENTFLAGS }), cs->option, ConfigVal::A_TRANSCODING_PROFILES_PROFLE_CLIENTFLAGS, cs);
@@ -789,7 +789,7 @@
             // scan mode (timed|inotify)
             item = values.append_child(CONFIG_LOAD_ITEM);
             createItem(item, ascs->getItemPath(indexList, { ConfigVal::A_AUTOSCAN_DIRECTORY_MODE }), ascs->option, ConfigVal::A_AUTOSCAN_DIRECTORY_MODE);
-            setValue(item, AutoscanDirectory::mapScanmode(adir->getScanMode()));
+            setValue(item, std::string(AutoscanDirectory::mapScanmode(adir->getScanMode())));

             // interval for timed
             item = values.append_child(CONFIG_LOAD_ITEM);
