--- src/winusb.orig	2012-01-19 16:28:30.000000000 +0400
+++ src/winusb	2012-01-21 02:05:58.093387384 +0400
@@ -330,7 +330,7 @@
 
 	# Create partiton
 	size=$(env LANG='' parted -s "$device" print | grep ^Disk | cut -f2 -d':' | sed 's/\s//g')
-	parted --align cylinder -s "$device" mkpart primary 1MB "$size" # We start at 1MB for grub (it needs a post-mbr gap for its code)
+	parted -s "$device" mkpart primary 1MB "$size" # We start at 1MB for grub (it needs a post-mbr gap for its code)
 	partition="${targetMedia}1" # first partition
 	partprobe "$device" # Reload partition table
 
@@ -360,7 +360,7 @@
 fi
 
 mkdir -p "$partitionMountPath"
-mount "$partition" "$partitionMountPath"
+mount.ntfs-3g "$partition" "$partitionMountPath"
 
 freeSpace=$(df --block-size 1 "$partitionMountPath" | grep "$partition" | awk '{print $4}')
 neededSpace=$(du -s "$isoMountPath" --bytes | awk '{print $1}')
