diff -Nur grubconf-0.5.1/Makefile.in grubconf-0.5.1-fixed/Makefile.in
--- grubconf-0.5.1/Makefile.in	2004-03-20 17:13:45.000000000 +0100
+++ grubconf-0.5.1-fixed/Makefile.in	2004-07-14 21:48:41.864866728 +0200
@@ -429,8 +429,7 @@
 install-exec-am:
 install-exec: install-exec-recursive
 
-install-data-am: install-man install-gnomemenuDATA \
-		install-grubconfdocDATA
+install-data-am: install-man install-gnomemenuDATA
 install-data: install-data-recursive
 
 install-am: all-am
diff -Nur grubconf-0.5.1/po/Makefile.in.in grubconf-0.5.1-fixed/po/Makefile.in.in
--- grubconf-0.5.1/po/Makefile.in.in	2003-01-13 20:42:38.000000000 +0100
+++ grubconf-0.5.1-fixed/po/Makefile.in.in	2004-07-14 21:49:00.280519553 +0200
@@ -29,7 +29,7 @@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
+mkinstalldirs = $(MKINSTALLDIRS)
 
 CC = @CC@
 GMSGFMT = @GMSGFMT@
