*** configure.old	2003-12-15 15:44:40.000000000 +0100
--- configure	2003-12-15 15:45:11.000000000 +0100
***************
*** 1142,1147 ****
  
  
! CFLAGS="$CFLAGS -I$QTDIR/include"
! CPPFLAGS="$CPPFLAGS -I$QTDIR/include"
  LIBS="-L$QTDIR/lib -L/usr/X11R6/lib"
  
--- 1142,1147 ----
  
  
! CFLAGS="$CFLAGS -I$QTDIR/include/qt"
! CPPFLAGS="$CPPFLAGS -I$QTDIR/include/qt"
  LIBS="-L$QTDIR/lib -L/usr/X11R6/lib"
  
