diff --git a/CMakeLists.txt b/CMakeLists.txt
index 12491d5..466857c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,6 +55,7 @@ find_package(KF${KF_MAJOR_VERSION} ${KF_MIN_VERSION} REQUIRED COMPONENTS
     Parts
     TextEditor
     WidgetsAddons
+    I18n
 )
 
 find_package(LibKompareDiff2 5.4 CONFIG)
diff --git a/src/dialogpages/CMakeLists.txt b/src/dialogpages/CMakeLists.txt
index c50bc93..ada709e 100644
--- a/src/dialogpages/CMakeLists.txt
+++ b/src/dialogpages/CMakeLists.txt
@@ -20,6 +20,7 @@ target_link_libraries(komparedialogpages
         KF${KF_MAJOR_VERSION}::Completion
         KF${KF_MAJOR_VERSION}::KIOWidgets
         KF${KF_MAJOR_VERSION}::WidgetsAddons
+	KF${KF_MAJOR_VERSION}::I18n
 )
 
 set_target_properties(komparedialogpages PROPERTIES
diff --git a/src/komparenavtreepart/CMakeLists.txt b/src/komparenavtreepart/CMakeLists.txt
index 32fa67d..7a5ed73 100644
--- a/src/komparenavtreepart/CMakeLists.txt
+++ b/src/komparenavtreepart/CMakeLists.txt
@@ -20,6 +20,7 @@ target_link_libraries(komparenavtreepart
     Qt::Widgets
     KF${KF_MAJOR_VERSION}::Parts
     KF${KF_MAJOR_VERSION}::IconThemes
+    KF${KF_MAJOR_VERSION}::I18n
     KompareDiff2
 )
 
