--- configure.orig	2004-08-01 01:43:57.889284992 -0500
+++ configure	2004-08-01 01:45:17.876125144 -0500
@@ -48,8 +48,7 @@
 		echo "LARGE_FILE=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE"
 
 		[ "$cpu" != "" ] &&
-		echo "TARGETARCH=-march=$cpu" &&
-		echo "TARGETCPU=-mcpu=$cpu"
+		echo "TARGETARCH=-march=$cpu"
 
 		[ "`uname`" = "Linux" ] &&
 		echo "LIBDL=-ldl"
@@ -72,7 +71,7 @@
 
 function build_pkgconfig
 {
-	for i in framework valerie miracle
+	for i in framework valerie
 	do
 		echo "prefix=$prefix" >mlt-$i.pc
 		echo "exec_prefix=$prefix" >>mlt-$i.pc
@@ -90,7 +89,7 @@
 
 # Define build directory for scripts called
 export build_dir=`dirname $0`
-export prefix=/usr/local
+export prefix=/usr
 export help=0
 export version=0.1.1
 export debug=true
@@ -115,7 +114,7 @@
 [ $help = 1 ] && show_help || build_config
 
 # Iterate through each of the components
-for i in framework modules inigo valerie miracle humperdink
+for i in framework modules inigo valerie
 do
 	if [ -x src/$i/configure ]
 	then
--- src/modules/kino/Makefile~	2005-04-15 16:28:26.000000000 +0200
+++ src/modules/kino/Makefile	2005-04-27 18:11:27.000000000 +0200
@@ -6,7 +6,7 @@
 OBJS=factory.o producer_kino.o
 CPPOBJS=kino_wrapper.o avi.o error.o filehandler.o riff.o
 CFLAGS+=-I../../
-LDFLAGS+=-lmlt -lstdc++
+LDFLAGS+=-L../../framework -lmlt -lstdc++
 CXXFLAGS+=$(CFLAGS) -Wno-deprecated
 
 ifdef HAVE_LIBQUICKTIME
