diff --git a/src/framework/mlt_property.h b/src/framework/mlt_property.h
index 404d513f..043f530b 100644
--- a/src/framework/mlt_property.h
+++ b/src/framework/mlt_property.h
@@ -3,7 +3,7 @@
  * \brief Property class declaration
  * \see mlt_property_s
  *
- * Copyright (C) 2003-2014 Meltytech, LLC
+ * Copyright (C) 2003-2017 Meltytech, LLC
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -31,7 +31,11 @@
 #endif
 
 #if defined(__GLIBC__) || defined(__APPLE__) || (__FreeBSD_version >= 900506)
-#include <xlocale.h>
+#  if __GLIBC_MINOR__ >= 26 && !defined(__APPLE__)
+#    include <locale.h>
+#  else
+#    include <xlocale.h>
+#  endif
 #else
 typedef char* locale_t;
 #endif
