--- src/daemon/daemon.c.orig	2007-03-05 15:44:17.000000000 +0400
+++ src/daemon/daemon.c	2007-03-05 15:47:10.000000000 +0400
@@ -1108,7 +1108,7 @@
 	dbus_conn = dbus_g_connection_get_connection(connection);
 
 	dbus_g_object_type_install_info(NOTIFY_TYPE_DAEMON,
-									&dbus_glib__object_info);
+									&dbus_glib_notification_daemon_object_info);
 
 	bus_proxy = dbus_g_proxy_new_for_name(connection,
 										  "org.freedesktop.DBus",
