From a8efd16d36a0f2ccf727d6a697ea7ca949d633fc Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Fri, 7 Aug 2020 12:25:06 +0200
Subject: [PATCH 1/3] btyacc - fix mktemp usage

---
 extern/btyacc/main.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/extern/btyacc/main.c b/extern/btyacc/main.c
index 6ac4cbd850..240520cdfd 100644
--- a/extern/btyacc/main.c
+++ b/extern/btyacc/main.c
@@ -341,15 +341,15 @@ void create_file_names()
 	text_file_name[len + 5] = 't';
 	union_file_name[len + 5] = 'u';
 
-	if(mktemp(action_file_name) == NULL) {
+	if(mkstemp(action_file_name) == -1) {
 		fprintf(stderr, "btyacc: Cannot create temporary file\n");
 		exit(1);
 	}
-	if(mktemp(text_file_name) == NULL) {
+	if(mkstemp(text_file_name) == -1) {
 		fprintf(stderr, "btyacc: Cannot create temporary file\n");
 		exit(1);
 	}
-	if(mktemp(union_file_name) == NULL) {
+	if(mkstemp(union_file_name) == -1) {
 		fprintf(stderr, "btyacc: Cannot create temporary file\n");
 		exit(1);
 	}
-- 
2.39.1

