--- include/AL/alc-orig.h	2008-04-13 15:30:33.000000000 +0500
+++ include/AL/alc.h	2008-04-13 15:35:26.000000000 +0500
@@ -67,7 +67,7 @@
 
 ALCAPI ALCenum ALCAPIENTRY alcGetError( ALCdevice *dev );
 
-ALCAPI ALCcontext * ALCAPIENTRY alcGetCurrentContext( ALCvoid );
+ALCAPI ALCcontext * ALCAPIENTRY alcGetCurrentContext( void );
 
 ALCAPI ALCdevice * ALCAPIENTRY alcOpenDevice( const ALubyte *tokstr );
 ALCAPI void ALCAPIENTRY alcCloseDevice( ALCdevice *dev );
@@ -93,7 +93,7 @@
       ALCAPI void            ALCAPIENTRY (*alcSuspendContext)( ALCcontext *alcHandle );
       ALCAPI void            ALCAPIENTRY (*alcDestroyContext)( ALCcontext* context );
       ALCAPI ALCenum         ALCAPIENTRY (*alcGetError)( ALCdevice *dev );
-      ALCAPI ALCcontext *    ALCAPIENTRY (*alcGetCurrentContext)( ALCvoid );
+      ALCAPI ALCcontext *    ALCAPIENTRY (*alcGetCurrentContext)( void );
       ALCAPI ALCdevice *     ALCAPIENTRY (*alcOpenDevice)( const ALCubyte *tokstr );
       ALCAPI void            ALCAPIENTRY (*alcCloseDevice)( ALCdevice *dev );
       ALCAPI ALCboolean      ALCAPIENTRY (*alcIsExtensionPresent)( ALCdevice *device, const ALCubyte *extName );
