From f7f50e5b77017e0f67bb4898f2917351906b86eb Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sat, 27 Aug 2016 11:18:48 +0200
Subject: [PATCH] fix compile error

---
 include/Inventor/SbBasic.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/Inventor/SbBasic.h b/include/Inventor/SbBasic.h
index d038bb1..80b8886 100644
--- a/include/Inventor/SbBasic.h
+++ b/include/Inventor/SbBasic.h
@@ -25,6 +25,7 @@
 \**************************************************************************/
 
 #include <Inventor/C/basic.h>
+#include <Inventor/C/errors/debugerror.h>
 
 /* ********************************************************************** */
 /* Trap people trying to use Inventor headers while compiling C source code.
-- 
2.9.3

