From d811f2e3d119c0b9cde3aff6604097b968969a41 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sun, 2 Jun 2013 13:40:58 +0200
Subject: [PATCH] add missing includes

---
 src/lib/random.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/lib/random.cpp b/src/lib/random.cpp
index 1007172..5b06658 100644
--- a/src/lib/random.cpp
+++ b/src/lib/random.cpp
@@ -34,6 +34,9 @@
 #include <QDataStream>
 #include <QTime>
 
+#include <sys/types.h>
+#include <unistd.h>
+
 void initStdRand();
 bool getNativeEntropy(quint8* buffer, int length);
 
-- 
1.8.3

