--- utils/gtf/gtfcalc.c.orig	2009-08-08 15:28:03.000000000 +0800
+++ utils/gtf/gtfcalc.c	2009-08-08 15:28:51.000000000 +0800
@@ -64,10 +64,12 @@ static GTF_constants GC = {
 
 /*-------------------------- Implementation -------------------------------*/
 
+#ifndef __USE_ISOC99
 static double round(double v)
 {
 	return floor(v + 0.5);
 }
+#endif
 
 static void GetInternalConstants(GTF_constants *c)
 /****************************************************************************
