diff -uNr svgalib-1.9.25/kernel/svgalib_helper/main.c svgalib-1.9.25.cp/kernel/svgalib_helper/main.c
--- svgalib-1.9.25/kernel/svgalib_helper/main.c	2006-01-12 19:17:53.000000000 +0100
+++ svgalib-1.9.25.cp/kernel/svgalib_helper/main.c	2006-09-28 21:43:53.000000000 +0200
@@ -17,7 +17,6 @@
 #include <linux/ioport.h>
 #include <linux/interrupt.h>
 #include <linux/pci.h>
-#include <linux/devfs_fs_kernel.h>
 #include <linux/mm.h>
 #include <linux/thread_info.h>
 #include <linux/smp.h>
