--- massdns.cpp	2000-06-13 09:05:57.000000000 +0200
+++ massdns.cpp	2002-05-15 18:18:01.000000000 +0200
@@ -10,6 +10,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <time.h>
 #include <unistd.h>
 
 class Request;
--- Makefile    2003-03-12 23:53:52.000000000 +0000
+++ Makefile    2003-03-12 23:54:00.000000000 +0000
@@ -1,6 +1,7 @@

 LDFLAGS= -s
 CFLAGS= -g0 -O99 -fomit-frame-pointer -I. -DNDEBUG
+CC=g++

 #LDFLAGS= -g
 #CFLAGS= -g3 -O0 -Wall -I. -D_DEBUG
