diff -purN SDL-1.2.8/include/SDL_endian.h SDL/include/SDL_endian.h
--- SDL-1.2.8/include/SDL_endian.h	2004-12-13 02:54:30.000000000 -0500
+++ SDL/include/SDL_endian.h	2005-04-22 09:55:25.000000000 -0400
@@ -68,7 +68,7 @@ static __inline__ Uint16 SDL_Swap16(Uint
 #elif defined(__GNUC__) && defined(__x86_64__)
 static __inline__ Uint16 SDL_Swap16(Uint16 x)
 {
-	__asm__("xchgb %b0,%h0" : "=q" (x) :  "0" (x));
+	__asm__("xchgb %b0,%h0" : "=Q" (x) :  "0" (x));
 	return x;
 }
 #elif defined(__GNUC__) && (defined(__powerpc__) || defined(__ppc__))
