--- bin/view-x-face~	2003-03-16 14:58:39.000000000 +0100
+++ bin/view-x-face	2003-03-16 15:00:18.000000000 +0100
@@ -50,10 +50,11 @@
 # Image viewer (should be able to read from stdin!): 
 if [ -n "$DISPLAY" ]
 then
-    VIEWER="xli stdin"		# Use xli as X11 image viewer
-    # VIEWER="xv -quit -"	# Use xv as X11 image viewer
+    # VIEWER="xli stdin"		# Use xli as X11 image viewer
+    VIEWER="xv -quit -"	# Use xv as X11 image viewer
 else
-    VIEWER=image2ascii		# Use image2ascii as ASCII image viewer
+    # VIEWER=image2ascii		# Use image2ascii as ASCII image viewer
+    VIEWER=pbmtoascii
 fi
 
 cat "$@" \
