--- src/Makefile.orig
+++ src/Makefile
@@ -9,8 +9,8 @@
 # This gets definitions for CC, CFLAGS, BINDIR etc.
 include ../config.mk
 
-CFLAGS+=`pkg-config --cflags gtk+-2.0` `pkg-config --cflags gdk-pixbuf-2.0`
-LDFLAGS+=`pkg-config --libs gtk+-2.0` `pkg-config --libs gdk-pixbuf-2.0`
+CFLAGS+=`pkg-config --cflags gtk+-2.0` `pkg-config --cflags gdk-pixbuf-2.0` `pkg-config --cflags x11`
+LDLIBS+=-lm `pkg-config --libs gtk+-2.0` `pkg-config --libs gdk-pixbuf-2.0 ` `pkg-config --libs x11`
 
 all: xzgv
 
@@ -23,7 +23,7 @@ OBJS=	main.o \
 	backend.o
 
 xzgv: $(OBJS)
-	$(CC) $(LDFLAGS) -o xzgv $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o xzgv $(OBJS) $(LDLIBS)
 
 installdirs:
 	/bin/sh ../mkinstalldirs $(BINDIR)
