diff --git a/openssl-sys-0.9.109/build/cfgs.rs b/openssl-sys-0.9.109/build/cfgs.rs
--- a/openssl-sys-0.9.109/build/cfgs.rs
+++ b/openssl-sys-0.9.109/build/cfgs.rs
@@ -71,6 +71,9 @@ pub fn get(openssl_version: Option<u64>, libressl_version: Option<u64>) -> Vec<&
         if libressl_version >= 0x4_01_00_00_0 {
             cfgs.push("libressl410");
         }
+        if libressl_version >= 0x4_02_00_00_0 {
+            cfgs.push("libressl420");
+        }
     } else {
         let openssl_version = openssl_version.unwrap();
 
diff --git a/openssl-sys-0.9.109/build/main.rs b/openssl-sys-0.9.109/build/main.rs
--- a/openssl-sys-0.9.109/build/main.rs
+++ b/openssl-sys-0.9.109/build/main.rs
@@ -451,6 +451,8 @@ See rust-openssl documentation for more information:
             (4, 0, _) => ('4', '0', 'x'),
             (4, 1, 0) => ('4', '1', '0'),
             (4, 1, _) => ('4', '1', 'x'),
+            (4, 2, 0) => ('4', '2', '0'),
+            (4, 2, _) => ('4', '2', 'x'),
             _ => version_error(),
         };
 
