diff --git a/openssl-sys-0.9.111/build/cfgs.rs b/openssl-sys-0.9.111/build/cfgs.rs
--- a/openssl-sys-0.9.111/build/cfgs.rs
+++ b/openssl-sys-0.9.111/build/cfgs.rs
@@ -47,6 +47,9 @@ pub fn get(openssl_version: Option<u64>, libressl_version: Option<u64>) -> Vec<&
         if libressl_version >= 0x4_02_00_00_0 {
             cfgs.push("libressl420");
         }
+        if libressl_version >= 0x4_03_00_00_0 {
+            cfgs.push("libressl430");
+        }
     } else {
         let openssl_version = openssl_version.unwrap();
         cfgs.push("ossl101");
diff --git a/openssl-sys-0.9.111/build/main.rs b/openssl-sys-0.9.111/build/main.rs
--- a/openssl-sys-0.9.111/build/main.rs
+++ b/openssl-sys-0.9.111/build/main.rs
@@ -434,6 +434,7 @@ See rust-openssl documentation for more information:
             (4, 1, 0) => ('4', '1', '0'),
             (4, 1, _) => ('4', '1', 'x'),
             (4, 2, _) => ('4', '2', 'x'),
+            (4, 3, _) => ('4', '3', 'x'),
             _ => version_error(),
         };
 
