From 68f770b51de3acc19f11fd8767f9e87b6e5e61ee Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 22:11:13 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 adjunctions.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/adjunctions.cabal b/adjunctions.cabal
index 53c7d4a..0fae9f1 100644
--- a/adjunctions.cabal
+++ b/adjunctions.cabal
@@ -49,7 +49,7 @@ library
     array               >= 0.3.0.2 && < 0.7,
     base                >= 4       && < 5,
     comonad             >= 4       && < 6,
-    containers          >= 0.3     && < 0.6,
+    containers          >= 0.3,
     contravariant       >= 1       && < 2,
     distributive        >= 0.5.1   && < 1,
     free                >= 4       && < 6,
-- 
2.19.1

