--- configure.old	2004-12-06 21:47:45.083366696 +0000
+++ configure	2004-12-06 21:50:28.560514392 +0000
@@ -6655,64 +6655,62 @@
    fi
 
         
-   echo $ac_n "checking for kdb2html""... $ac_c" 1>&6
-echo "configure:6660: checking for kdb2html" >&5
-   if test -n "$KDB2HTML"; then
-        kde_cv_path="$KDB2HTML";
-   else
-        kde_cache=`echo kdb2html | sed 'y%./+-%__p_%'`
-
-        if eval "test \"`echo '$''{'kde_cv_path_$kde_cache'+set}'`\" = set"; then
-  echo $ac_n "(cached) $ac_c" 1>&6
-else
-  
-        kde_cv_path="NONE"
-	dirs="$kde_default_bindirs"
-	kde_save_IFS=$IFS
-	IFS=':'
-	for dir in $PATH; do
-	  dirs="$dirs $dir"
-        done
-	IFS=$kde_save_IFS
-
-        for dir in $dirs; do
-	  if test -x "$dir/kdb2html"; then
-	    if test -n ""
-	    then
-              evalstr="$dir/kdb2html  2>&1 "
-	      if eval $evalstr; then
-                kde_cv_path="$dir/kdb2html"
-                break
-	      fi
-            else
-		kde_cv_path="$dir/kdb2html"
-                break
-	    fi
-          fi
-        done
-
-        eval "kde_cv_path_$kde_cache=$kde_cv_path"
-
-        
-fi
-
-
-      eval "kde_cv_path=\"`echo '$kde_cv_path_'$kde_cache`\""
-
-   fi
-
-   if test -z "$kde_cv_path" || test "$kde_cv_path" = NONE; then
-      echo "$ac_t""not found" 1>&6
-      
-    { echo "configure: error: The important program kdb2html was not found!
-Please check whether you installed KDE correctly.
-" 1>&2; exit 1; }
-
-   else
-      echo "$ac_t""$kde_cv_path" 1>&6
-      KDB2HTML=$kde_cv_path
-
-   fi
+#   echo $ac_n "checking for kdb2html""... $ac_c" 1>&6
+#echo "configure:6660: checking for kdb2html" >&5
+#   if test -n "$KDB2HTML"; then
+#        kde_cv_path="$KDB2HTML";
+#   else
+#        kde_cache=`echo kdb2html | sed 'y%./+-%__p_%'`
+#
+#        if eval "test \"`echo '$''{'kde_cv_path_$kde_cache'+set}'`\" = set"; then
+#  echo $ac_n "(cached) $ac_c" 1>&6
+#else
+#  
+#        kde_cv_path="NONE"
+#	dirs="$kde_default_bindirs"
+#	kde_save_IFS=$IFS
+#	IFS=':'
+#	for dir in $PATH; do
+#	  dirs="$dirs $dir"
+#        done
+#	IFS=$kde_save_IFS
+#        for dir in $dirs; do
+#	  if test -x "$dir/kdb2html"; then
+#	    if test -n ""
+#	    then
+#              evalstr="$dir/kdb2html  2>&1 "
+#	      if eval $evalstr; then
+#                kde_cv_path="$dir/kdb2html"
+#                break
+#	      fi
+#            else
+#		kde_cv_path="$dir/kdb2html"
+#                break
+#	    fi
+#          fi
+#        done
+#
+#        eval "kde_cv_path_$kde_cache=$kde_cv_path"
+#
+#        
+#fi
+#
+#      eval "kde_cv_path=\"`echo '$kde_cv_path_'$kde_cache`\""
+#
+#   fi
+#
+#   if test -z "$kde_cv_path" || test "$kde_cv_path" = NONE; then
+#      echo "$ac_t""not found" 1>&6
+#      
+#    { echo "configure: error: The important program kdb2html was not found!
+#Please check whether you installed KDE correctly.
+#" 1>&2; exit 1; }
+#
+#   else
+#      echo "$ac_t""$kde_cv_path" 1>&6
+#      KDB2HTML=$kde_cv_path
+#
+#   fi
 
         
    echo $ac_n "checking for artsc-config""... $ac_c" 1>&6
@@ -6904,7 +6902,7 @@
    
    LIB_SMB='-lsmb'
    
-   LIB_KFILE='-lkfile'
+   LIB_KFILE='-lkfilereplacepart'
    
    LIB_KAB='-lkab'
    
@@ -6923,7 +6921,7 @@
    
    LIB_KFM='-lkfm $(LIB_KDECORE)'
    
-   LIB_KFILE='-lkfile $(LIB_KFM) $(LIB_KDEUI)'
+   LIB_KFILE='-lkfilereplacepart $(LIB_KFM) $(LIB_KDEUI)'
    
    LIB_KAB='-lkab $(LIB_KIMGIO) $(LIB_KDECORE)'
    
