--- Makefile.am.old	2004-12-06 23:44:31.145283392 +0000
+++ Makefile.am	2004-12-06 23:42:27.899019672 +0000
@@ -1,7 +1,7 @@
 ####### kdevelop will overwrite this part!!! (begin)##########
 bin_PROGRAMS = dbdesign
 dbdesign_SOURCES = metaviewconf.ui viewconf.cpp seqedit.cpp metaseqedit.ui sequence.cpp svrdlg.cpp metasvrdialog.ui plugin.cpp fldcolors.cpp crelationitem.cpp creledit.cpp cmetareledit.ui cobjtreeitem.cpp cobjecttree.cpp calphavalidator.cpp cfldedit.cpp cmetafldedit.ui ctable.cpp ctableitem.cpp cmetatableed.ui ctableed.cpp cview.cpp dbdesign.cpp main.cpp 
-dbdesign_LDADD   = $(LIBDL) -lkfile -lkdeui -lkdecore $(LIB_QT) $(LIBSOCKET)
+dbdesign_LDADD   = $(LIBDL) -lkfilereplacepart -lkdeui -lkdecore $(LIB_QT) $(LIBSOCKET)
 
 SUBDIRS = plugins 
 
