--- Makefile.in.old	2004-12-06 23:44:31.145283392 +0000
+++ Makefile.in	2004-12-06 23:43:04.157507544 +0000
@@ -205,7 +205,7 @@
 #>- dbdesign_SOURCES = metaviewconf.ui viewconf.cpp seqedit.cpp metaseqedit.ui sequence.cpp svrdlg.cpp metasvrdialog.ui plugin.cpp fldcolors.cpp crelationitem.cpp creledit.cpp cmetareledit.ui cobjtreeitem.cpp cobjecttree.cpp calphavalidator.cpp cfldedit.cpp cmetafldedit.ui ctable.cpp ctableitem.cpp cmetatableed.ui ctableed.cpp cview.cpp dbdesign.cpp main.cpp 
 #>+ 1
 dbdesign_SOURCES= viewconf.cpp seqedit.cpp  sequence.cpp svrdlg.cpp  plugin.cpp fldcolors.cpp crelationitem.cpp creledit.cpp  cobjtreeitem.cpp cobjecttree.cpp calphavalidator.cpp cfldedit.cpp  ctable.cpp ctableitem.cpp  ctableed.cpp cview.cpp dbdesign.cpp main.cpp  metaviewconf.cpp metaseqedit.cpp metasvrdialog.cpp cmetareledit.cpp cmetafldedit.cpp cmetatableed.cpp
-dbdesign_LDADD = $(LIBDL) -lkfile -lkdeui -lkdecore $(LIB_QT) $(LIBSOCKET)
+dbdesign_LDADD = $(LIBDL) -L/usr/lib/kde3/ -lkfilereplacepart -lkdeui -lkdecore $(LIB_QT) $(LIBSOCKET)
 
 SUBDIRS = plugins 
 
@@ -741,6 +741,7 @@
 
 metaseqedit.h: $(srcdir)/metaseqedit.ui
 	$(UIC) -o metaseqedit.h $(srcdir)/metaseqedit.ui
+	patch -p0 <./metaseqedit.h.diff
 
 metaseqedit.moc: metaseqedit.h
 	$(MOC) metaseqedit.h -o metaseqedit.moc
