diff -ur acpica-unix-20110623/compiler/Makefile acpica-unix-mod/compiler/Makefile
--- acpica-unix-20110623/compiler/Makefile	2011-06-23 11:34:09.000000000 -0500
+++ acpica-unix-mod/compiler/Makefile	2011-08-10 19:25:14.815970671 -0500
@@ -43,7 +43,7 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
+      \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
@@ -243,16 +243,16 @@
 # by the utilities above and they are not necessarily ANSI C, etc.
 #
 aslcompilerlex.o :   aslcompilerlex.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall   -Wstrict-aliasing=0 -o$@ $?
 
 aslcompilerparse.o : aslcompilerparse.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall   -Wstrict-aliasing=0 -o$@ $?
 
 dtparserlex.o :      dtparserlex.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall   -Wstrict-aliasing=0 -o$@ $?
 
 dtparserparse.o :    dtparserparse.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall   -Wstrict-aliasing=0 -o$@ $?
 
 
 #
diff -ur acpica-unix-20110623/generate/unix/iasl/Makefile acpica-unix-mod/generate/unix/iasl/Makefile
--- acpica-unix-20110623/generate/unix/iasl/Makefile	2011-06-23 11:34:12.000000000 -0500
+++ acpica-unix-mod/generate/unix/iasl/Makefile	2011-08-10 19:25:54.095970658 -0500
@@ -200,16 +200,16 @@
 # by the utilities above and they are not necessarily ANSI C, etc.
 #
 aslcompilerlex.o :   aslcompilerlex.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall   -o$@ $?
 
 aslcompilerparse.o : aslcompilerparse.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall   -o$@ $?
 
 dtparserlex.o :      dtparserlex.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall   -o$@ $?
 
 dtparserparse.o :    dtparserparse.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall   -o$@ $?
 
 #
 # Compiler source
diff -ur acpica-unix-20110623/generate/unix/Makefile.config acpica-unix-mod/generate/unix/Makefile.config
--- acpica-unix-20110623/generate/unix/Makefile.config	2011-06-23 11:34:12.000000000 -0500
+++ acpica-unix-mod/generate/unix/Makefile.config	2011-08-10 19:29:36.743970583 -0500
@@ -58,7 +58,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
diff -ur acpica-unix-20110623/tools/acpibin/Makefile acpica-unix-mod/tools/acpibin/Makefile
--- acpica-unix-20110623/tools/acpibin/Makefile	2011-06-23 11:34:18.000000000 -0500
+++ acpica-unix-mod/tools/acpibin/Makefile	2011-08-10 19:26:40.389970642 -0500
@@ -41,7 +41,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
diff -ur acpica-unix-20110623/tools/acpiexec/Makefile acpica-unix-mod/tools/acpiexec/Makefile
--- acpica-unix-20110623/tools/acpiexec/Makefile	2011-06-23 11:34:18.000000000 -0500
+++ acpica-unix-mod/tools/acpiexec/Makefile	2011-08-10 19:28:08.378970613 -0500
@@ -37,6 +37,7 @@
     -D$(HOST) \
     -D_GNU_SOURCE \
     -DACPI_EXEC_APP \
+
     -I$(ACPICA_SRC)/include
 
 CWARNINGFLAGS = \
@@ -44,7 +45,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
diff -ur acpica-unix-20110623/tools/acpihelp/Makefile acpica-unix-mod/tools/acpihelp/Makefile
--- acpica-unix-20110623/tools/acpihelp/Makefile	2011-06-23 11:34:18.000000000 -0500
+++ acpica-unix-mod/tools/acpihelp/Makefile	2011-08-10 19:27:36.955970623 -0500
@@ -42,7 +42,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
diff -ur acpica-unix-20110623/tools/acpinames/Makefile acpica-unix-mod/tools/acpinames/Makefile
--- acpica-unix-20110623/tools/acpinames/Makefile	2011-06-23 11:34:18.000000000 -0500
+++ acpica-unix-mod/tools/acpinames/Makefile	2011-08-10 19:29:15.432970590 -0500
@@ -44,7 +44,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
diff -ur acpica-unix-20110623/tools/acpisrc/Makefile acpica-unix-mod/tools/acpisrc/Makefile
--- acpica-unix-20110623/tools/acpisrc/Makefile	2011-06-23 11:34:19.000000000 -0500
+++ acpica-unix-mod/tools/acpisrc/Makefile	2011-08-10 19:28:56.064970597 -0500
@@ -42,7 +42,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
diff -ur acpica-unix-20110623/tools/acpixtract/Makefile acpica-unix-mod/tools/acpixtract/Makefile
--- acpica-unix-20110623/tools/acpixtract/Makefile	2011-06-23 11:34:19.000000000 -0500
+++ acpica-unix-mod/tools/acpixtract/Makefile	2011-08-10 19:28:30.250970606 -0500
@@ -40,7 +40,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
