From ec91184183021519af972488c70338a97f0ea875 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sun, 6 Mar 2022 14:45:33 +0100
Subject: [PATCH] fix find gio-unix

---
 cmake/find-modules/FindGLIB.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/find-modules/FindGLIB.cmake b/cmake/find-modules/FindGLIB.cmake
index 0564b8d..c6a06f7 100644
--- a/cmake/find-modules/FindGLIB.cmake
+++ b/cmake/find-modules/FindGLIB.cmake
@@ -101,7 +101,7 @@ foreach (_component ${GLIB_FIND_COMPONENTS})
     elseif (${_component} STREQUAL "gio-unix")
         pkg_check_modules(GIO_UNIX gio-unix-2.0)
         find_path(GLIB_GIO_UNIX_INCLUDE_DIR
-                  NAMES gio/gunixconnection.h
+                  NAMES gio/gdesktopappinfo.h
                   HINTS ${GIO_UNIX_INCLUDEDIR}
                   PATH_SUFFIXES gio-unix-2.0)
 
-- 
2.35.1

