diff -urpN xprobe2-0.1.old/src/config_set.cc xprobe2-0.1/src/config_set.cc
--- xprobe2-0.1.old/src/config_set.cc	2003-09-27 23:27:28.000000000 -0700
+++ xprobe2-0.1/src/config_set.cc	2003-09-27 23:28:51.000000000 -0700
@@ -159,8 +159,7 @@ int Config_Section::read_sec(void) {
 
             case XP_CONFIG_TK_SECEND:
                 if (get_state() != XP_CONFIG_TK_SECBEGIN) {
-                    ui->error("[x] Multiple close sections on line:
-                    %i\n", cf->get_linenum());
+                    ui->error("[x] Multiple close sections on line: %i\n", cf->get_linenum());
                     return FAIL;
                 }
                 set_state(0);
diff -urpN xprobe2-0.1.old/src/os_matrix.cc xprobe2-0.1/src/os_matrix.cc
--- xprobe2-0.1.old/src/os_matrix.cc	2003-09-27 23:27:28.000000000 -0700
+++ xprobe2-0.1/src/os_matrix.cc	2003-09-27 23:29:11.000000000 -0700
@@ -85,8 +85,7 @@ void OS_Name::list_oses(void) {
     xprobe_mdebug(XPROBE_DEBUG_OSMATRIX,"Following systems are recognizable\n");
     for (osid_i = osid_name.begin();
          osid_i != osid_name.end(); osid_i++) {
-        xprobe_debug(XPROBE_DEBUG_OSMATRIX,"Id: %i\tOS:
-        %s\n",(*osid_i).first, (*osid_i).second.c_str());
+        xprobe_debug(XPROBE_DEBUG_OSMATRIX,"Id: %i\tOS: %s\n",(*osid_i).first, (*osid_i).second.c_str());
     }
 }
 
