From 5a4831c45cedc28c5dcba7c7f74c514271bfc3e5 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 29 Jan 2018 21:30:04 +0100
Subject: [PATCH] fix dependency on unix-compat

---
 snap-core.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/snap-core.cabal b/snap-core.cabal
index 6de9d37..d60ccdc 100644
--- a/snap-core.cabal
+++ b/snap-core.cabal
@@ -150,7 +150,7 @@ Library
     time                      >= 1.0     && < 1.9,
     transformers              >= 0.3     && < 0.6,
     transformers-base         >= 0.4     && < 0.5,
-    unix-compat               >= 0.3     && < 0.5,
+    unix-compat               >= 0.3     && < 0.6,
     unordered-containers      >= 0.1.4.3 && < 0.3,
     vector                    >= 0.6     && < 0.13
 
-- 
2.15.1

