diff -Naur xdtv-2.4.0.old/configure.in xdtv-2.4.0.new/configure.in
--- xdtv-2.4.0.old/configure.in	2007-02-19 22:04:14.000000000 +0100
+++ xdtv-2.4.0.new/configure.in	2007-03-09 23:15:49.000000000 +0100
@@ -223,8 +223,12 @@
    case ${host_cpu} in
       i386 | i486 | i586 | i686 | pentium | athlon) have_mmx=yes; host_arch=i386 ;;
       x86_64 | amd64) have_mmx=yes; host_arch=x86_64 ;;
-      alpha) have_mmx=no; host_arch=alpha ;;
+      alpha*) have_mmx=no; host_arch=alpha ;;
       sparc) have_mmx=no; host_arch=sparc ;;
+      sun4u | sparc64) have_mmx=no; host_arch=sparc64 ;;
+      ppc | powerpc) have_mmx=no; host_arch=powerpc ;;
+      ppc64) have_mmx=no; host_arch=powerpc64 ;;
+      ia64*) have_mmx=no; host_arch=ia64 ;;
       *) have_mmx=no;  host_arch=UNKNOWN ;;
    esac

@@ -2442,13 +2446,17 @@
        if test x"$host_arch" == xx86_64; then
          AC_DEFINE(ARCH_X86_64,1,FFMPEG is inside XdTV now!)
          AC_DEFINE(ARCH_X86,1,FFMPEG is inside XdTV now!)
+         AC_DEFINE(HAVE_FAST_64BIT,1,FFMPEG is inside XdTV now!)
          echo "TARGET_ARCH_X86_64=yes" >> config.mak.in
-        echo "TARGET_ARCH_X86=yes" >> config.mak.in
+         echo "TARGET_ARCH_X86=yes" >> config.mak.in
+         echo "HAVE_FAST_64BIT=yes" >> config.mak.in
        fi

        if test x"$host_arch" == xalpha; then
          AC_DEFINE(ARCH_ALPHA,1,FFMPEG is inside XdTV now!)
+         AC_DEFINE(HAVE_FAST_64BIT,1,FFMPEG is inside XdTV now!)
          echo "TARGET_ARCH_ALPHA=yes" >> config.mak.in
+         echo "HAVE_FAST_64BIT=yes" >> config.mak.in
        fi

        if test x"$host_arch" == xsparc; then
@@ -2456,6 +2464,31 @@
          echo "TARGET_ARCH_SPARC=yes" >> config.mak.in
        fi

+       if test x"$host_arch" == xsparc64; then
+         AC_DEFINE(ARCH_SPARC,1,FFMPEG is inside XdTV now!)
+         AC_DEFINE(HAVE_FAST_64BIT,1,FFMPEG is inside XdTV now!)
+         echo "TARGET_ARCH_SPARC=yes" >> config.mak.in
+         echo "HAVE_FAST_64BIT=yes" >> config.mak.in
+       fi
+
+       if test x"$host_arch" == xpowerpc; then
+         AC_DEFINE(ARCH_POWERPC,1,FFMPEG is inside XdTV now!)
+         echo "TARGET_ARCH_POWERPC=yes" >> config.mak.in
+       fi
+
+       if test x"$host_arch" == xpowerpc64; then
+         AC_DEFINE(ARCH_POWERPC,1,FFMPEG is inside XdTV now!)
+         AC_DEFINE(HAVE_FAST_64BIT,1,FFMPEG is inside XdTV now!)
+         echo "TARGET_ARCH_POWERPC=yes" >> config.mak.in
+         echo "HAVE_FAST_64BIT=yes" >> config.mak.in
+       fi
+
+       if test x"$host_arch" == xia64; then
+         AC_DEFINE(ARCH_IA64,1,FFMPEG is inside XdTV now!)
+         AC_DEFINE(HAVE_FAST_64BIT,1,FFMPEG is inside XdTV now!)
+         echo "TARGET_ARCH_IA64=yes" >> config.mak.in
+         echo "HAVE_FAST_64BIT=yes" >> config.mak.in
+       fi
     fi

     AC_SUBST(FFMPEGLINK)
diff -Naur xdtv-2.4.0.old/configure.in.Marillat xdtv-2.4.0.new/configure.in.Marillat
--- xdtv-2.4.0.old/configure.in.Marillat	2007-02-19 22:04:14.000000000 +0100
+++ xdtv-2.4.0.new/configure.in.Marillat	2007-03-09 23:16:00.000000000 +0100
@@ -223,8 +223,12 @@
    case ${host_cpu} in
       i386 | i486 | i586 | i686 | pentium | athlon) have_mmx=yes; host_arch=i386 ;;
       x86_64 | amd64) have_mmx=yes; host_arch=x86_64 ;;
-      alpha) have_mmx=no; host_arch=alpha ;;
+      alpha*) have_mmx=no; host_arch=alpha ;;
       sparc) have_mmx=no; host_arch=sparc ;;
+      sun4u | sparc64) have_mmx=no; host_arch=sparc64 ;;
+      ppc | powerpc) have_mmx=no; host_arch=powerpc ;;
+      ppc64) have_mmx=no; host_arch=powerpc64 ;;
+      ia64*) have_mmx=no; host_arch=ia64 ;;
       *) have_mmx=no;  host_arch=UNKNOWN ;;
    esac

@@ -2442,13 +2446,17 @@
        if test x"$host_arch" == xx86_64; then
          AC_DEFINE(ARCH_X86_64,1,FFMPEG is inside XdTV now!)
          AC_DEFINE(ARCH_X86,1,FFMPEG is inside XdTV now!)
+         AC_DEFINE(HAVE_FAST_64BIT,1,FFMPEG is inside XdTV now!)
          echo "TARGET_ARCH_X86_64=yes" >> config.mak.in
-        echo "TARGET_ARCH_X86=yes" >> config.mak.in
+         echo "TARGET_ARCH_X86=yes" >> config.mak.in
+         echo "HAVE_FAST_64BIT=yes" >> config.mak.in
        fi

        if test x"$host_arch" == xalpha; then
          AC_DEFINE(ARCH_ALPHA,1,FFMPEG is inside XdTV now!)
+         AC_DEFINE(HAVE_FAST_64BIT,1,FFMPEG is inside XdTV now!)
          echo "TARGET_ARCH_ALPHA=yes" >> config.mak.in
+         echo "HAVE_FAST_64BIT=yes" >> config.mak.in
        fi

        if test x"$host_arch" == xsparc; then
@@ -2456,6 +2464,32 @@
          echo "TARGET_ARCH_SPARC=yes" >> config.mak.in
        fi

+       if test x"$host_arch" == xsparc64; then
+         AC_DEFINE(ARCH_SPARC,1,FFMPEG is inside XdTV now!)
+         AC_DEFINE(HAVE_FAST_64BIT,1,FFMPEG is inside XdTV now!)
+         echo "TARGET_ARCH_SPARC=yes" >> config.mak.in
+         echo "HAVE_FAST_64BIT=yes" >> config.mak.in
+       fi
+
+       if test x"$host_arch" == xpowerpc; then
+         AC_DEFINE(ARCH_POWERPC,1,FFMPEG is inside XdTV now!)
+         echo "TARGET_ARCH_POWERPC=yes" >> config.mak.in
+       fi
+
+       if test x"$host_arch" == xpowerpc64; then
+         AC_DEFINE(ARCH_POWERPC,1,FFMPEG is inside XdTV now!)
+         AC_DEFINE(HAVE_FAST_64BIT,1,FFMPEG is inside XdTV now!)
+         echo "TARGET_ARCH_POWERPC=yes" >> config.mak.in
+         echo "HAVE_FAST_64BIT=yes" >> config.mak.in
+       fi
+
+       if test x"$host_arch" == xia64; then
+         AC_DEFINE(ARCH_IA64,1,FFMPEG is inside XdTV now!)
+         AC_DEFINE(HAVE_FAST_64BIT,1,FFMPEG is inside XdTV now!)
+         echo "TARGET_ARCH_IA64=yes" >> config.mak.in
+         echo "HAVE_FAST_64BIT=yes" >> config.mak.in
+       fi
+
     fi

     AC_SUBST(FFMPEGLINK)
diff -Naur xdtv-2.4.0.old/libavcodec/Makefile xdtv-2.4.0.new/libavcodec/Makefile
--- xdtv-2.4.0.old/libavcodec/Makefile	2007-02-21 23:39:57.000000000 +0100
+++ xdtv-2.4.0.new/libavcodec/Makefile	2007-03-10 09:59:21.000000000 +0100
@@ -11,7 +11,7 @@

 CFLAGS=-DHAVE_AV_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE              \
        -D_ISOC9X_SOURCE -I$(BUILD_ROOT) -I$(SRC_PATH)                             \
-       -I$(SRC_PATH)/libavutil $(OPTFLAGS) $(AMR_CFLAGS) 			  \
+       -I$(SRC_PATH)/libavutil $(OPTFLAGS) $(AMR_CFLAGS)                          \
        -fno-strict-aliasing

 OBJS= bitstream.o utils.o allcodecs.o             \
@@ -136,17 +136,32 @@
         i386/vp3dsp_mmx.o i386/vp3dsp_sse2.o          \
         i386/fft_3dn.o i386/fft_3dn2.o                \
         i386/snowdsp_mmx.o
-endif

 ifeq ($(CONFIG_GPL),yes)
 OBJS += i386/idct_mmx.o
 endif
+endif

 ifeq ($(TARGET_ARCH_SPARC),yes)
 OBJS+=sparc/dsputil_vis.o
 sparc/dsputil_vis.o: CFLAGS += -mcpu=ultrasparc -mtune=ultrasparc
 endif

+ifeq ($(TARGET_ARCH_ALPHA),yes)
+OBJS+= alpha/dsputil_alpha.o          \
+       alpha/mpegvideo_alpha.o        \
+       alpha/simple_idct_alpha.o      \
+       alpha/motion_est_alpha.o
+
+ASM_OBJS+= alpha/dsputil_alpha_asm.o  \
+           alpha/motion_est_mvi_asm.o
+endif
+
+ifeq ($(TARGET_ARCH_POWERPC),yes)
+OBJS+= ppc/dsputil_ppc.o              \
+       ppc/mpegvideo_ppc.o
+endif
+
 EXTRALIBS := -L$(BUILD_ROOT)/libavutil -lavutil$(BUILDSUF) $(EXTRALIBS)

 NAME=avcodec
diff -Naur xdtv-2.4.0.old/patch/Makefile.codec xdtv-2.4.0.new/patch/Makefile.codec
--- xdtv-2.4.0.old/patch/Makefile.codec	2007-02-21 23:14:57.000000000 +0100
+++ xdtv-2.4.0.new/patch/Makefile.codec	2007-03-10 09:58:50.000000000 +0100
@@ -11,7 +11,7 @@

 CFLAGS=-DHAVE_AV_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE              \
        -D_ISOC9X_SOURCE -I$(BUILD_ROOT) -I$(SRC_PATH)                             \
-       -I$(SRC_PATH)/libavutil $(OPTFLAGS) $(AMR_CFLAGS) 			  \
+       -I$(SRC_PATH)/libavutil $(OPTFLAGS) $(AMR_CFLAGS)                          \
        -fno-strict-aliasing

 OBJS= bitstream.o utils.o allcodecs.o             \
@@ -136,17 +136,32 @@
         i386/vp3dsp_mmx.o i386/vp3dsp_sse2.o          \
         i386/fft_3dn.o i386/fft_3dn2.o                \
         i386/snowdsp_mmx.o
-endif

 ifeq ($(CONFIG_GPL),yes)
 OBJS += i386/idct_mmx.o
 endif
+endif

 ifeq ($(TARGET_ARCH_SPARC),yes)
 OBJS+=sparc/dsputil_vis.o
 sparc/dsputil_vis.o: CFLAGS += -mcpu=ultrasparc -mtune=ultrasparc
 endif

+ifeq ($(TARGET_ARCH_ALPHA),yes)
+OBJS+= alpha/dsputil_alpha.o          \
+       alpha/mpegvideo_alpha.o        \
+       alpha/simple_idct_alpha.o      \
+       alpha/motion_est_alpha.o
+
+ASM_OBJS+= alpha/dsputil_alpha_asm.o  \
+           alpha/motion_est_mvi_asm.o
+endif
+
+ifeq ($(TARGET_ARCH_POWERPC),yes)
+OBJS+= ppc/dsputil_ppc.o              \
+       ppc/mpegvideo_ppc.o
+endif
+
 EXTRALIBS := -L$(BUILD_ROOT)/libavutil -lavutil$(BUILDSUF) $(EXTRALIBS)

 NAME=avcodec
