diff -Naur ncmpc-0.11.1.orig/src/list_window.c ncmpc-0.11.1/src/list_window.c
--- ncmpc-0.11.1.orig/src/list_window.c	2005-01-23 18:37:42.000000000 +0100
+++ ncmpc-0.11.1/src/list_window.c	2007-07-27 10:17:00.000000000 +0200
@@ -196,7 +196,7 @@
       if( label )
 	{
 	  int selected = lw->start+i == lw->selected;
-	  size_t len = strlen(label);
+	  size_t len, y;
 
 	  if( highlight )
 	    colors_use(lw->w, COLOR_LIST_BOLD);
@@ -207,6 +207,7 @@
 	    wattron(lw->w, A_REVERSE);
 	  
 	  waddnstr(lw->w, label, lw->cols);
+	  getyx(lw->w, y, len);
 	  if( fill && len<lw->cols )
 	    mvwhline(lw->w, i, len, ' ', lw->cols-len);
 
