From cbcd12b69fcb66891bfdd4d24ecd925f815c0fa5 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Fri, 22 Mar 2013 18:33:45 +0100
Subject: [PATCH] fix buffer overflow

---
 hw/kdrive/src/kinput.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/kdrive/src/kinput.c b/hw/kdrive/src/kinput.c
index b1068bb..aa5436f 100644
--- a/hw/kdrive/src/kinput.c
+++ b/hw/kdrive/src/kinput.c
@@ -221,7 +221,7 @@ KdUnregisterFd(void *closure, int fd, Bool do_close)
             if (do_close)
                 close(kdInputFds[i].fd);
             kdNumInputFds--;
-            for (j = i; j < kdNumInputFds; j++)
+            for (j = i; j < kdNumInputFds - 1; j++)
                 kdInputFds[j] = kdInputFds[j + 1];
             break;
         }
-- 
1.8.1.3

