diff -Naur xorg-server-1.6.4.orig/hw/xfree86/dixmods/Makefile.in xorg-server-1.6.4/hw/xfree86/dixmods/Makefile.in
--- xorg-server-1.6.4.orig/hw/xfree86/dixmods/Makefile.in	2009-10-08 18:44:03.073813053 -0700
+++ xorg-server-1.6.4/hw/xfree86/dixmods/Makefile.in	2009-10-08 18:44:15.656814046 -0700
@@ -473,8 +473,7 @@
 
 extsmoduledir = $(moduledir)/extensions
 extsmodule_LTLIBRARIES = $(RECORDMOD) \
-			 $(DBEMOD) \
-                         $(GLXMODS)
+			 $(DBEMOD)
 
 AM_CFLAGS = @XORG_CFLAGS@ @DIX_CFLAGS@
 INCLUDES = @XORG_INCS@ \
@@ -547,6 +546,14 @@
 $(am__aclocal_m4_deps):
 install-extsmoduleLTLIBRARIES: $(extsmodule_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
+	$(mkdir_p) "$(DESTDIR)$(extsmoduledir)/mesa"
+	@lst='$(GLXMODS)'; for r in $$lst; do \
+	if test -f $$r; then \
+		h=$(am__strip_dir) \
+		echo " $(LIBTOOL) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=install $(extsmoduleLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$r' '$(DESTDIR)$(extsmoduledir)/mesa/$$h'"; \
+		$(LIBTOOL) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=install $(extsmoduleLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$r" "$(DESTDIR)$(extsmoduledir)/mesa/$$h"; \
+	else :; fi; \
+	done
 	test -z "$(extsmoduledir)" || $(MKDIR_P) "$(DESTDIR)$(extsmoduledir)"
 	@list='$(extsmodule_LTLIBRARIES)'; test -n "$(extsmoduledir)" || list=; \
 	list2=; for p in $$list; do \
@@ -909,7 +916,7 @@
 	done
 check-am: all-am
 check: check-recursive
-all-am: Makefile $(LTLIBRARIES)
+all-am: Makefile $(LTLIBRARIES) $(GLXMODS)
 installdirs: installdirs-recursive
 installdirs-am:
 	for dir in "$(DESTDIR)$(extsmoduledir)" "$(DESTDIR)$(moduledir)"; do \
diff -Naur xorg-server-1.6.4.orig/hw/xfree86/dri/Makefile.in xorg-server-1.6.4/hw/xfree86/dri/Makefile.in
--- xorg-server-1.6.4.orig/hw/xfree86/dri/Makefile.in	2009-10-08 18:44:03.156814108 -0700
+++ xorg-server-1.6.4/hw/xfree86/dri/Makefile.in	2009-10-08 18:44:15.656814046 -0700
@@ -395,7 +395,7 @@
                    @GL_CFLAGS@
 
 libdri_la_LDFLAGS = -module -avoid-version @LIBDRM_LIBS@
-libdri_ladir = $(moduledir)/extensions
+libdri_ladir = $(moduledir)/extensions/mesa
 libdri_la_SOURCES = \
 	dri.c \
 	dri.h \
