diff -Nuar hw.orig/xfree86/dixmods/Makefile.in hw/xfree86/dixmods/Makefile.in
--- hw.orig/xfree86/dixmods/Makefile.in	2009-04-15 17:22:15.000000000 +0200
+++ hw/xfree86/dixmods/Makefile.in	2009-04-15 17:24:00.000000000 +0200
@@ -427,8 +427,7 @@
 
 extsmoduledir = $(moduledir)/extensions
 extsmodule_LTLIBRARIES = $(RECORDMOD) \
-			 $(DBEMOD) \
-                         $(GLXMODS)
+			 $(DBEMOD)
 
 AM_CFLAGS = @XORG_CFLAGS@ @DIX_CFLAGS@
 INCLUDES = @XORG_INCS@ \
@@ -500,6 +499,14 @@
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 install-extsmoduleLTLIBRARIES: $(extsmodule_LTLIBRARIES)
 	@$(NORMAL_INSTALL)
+	$(mkdir_p) "$(DESTDIR)$(extsmoduledir)/.mesa"
+	@lst='$(GLXMODS)'; for r in $$lst; do \
+	if test -f $$r; then \
+		h=$(am__strip_dir) \
+		echo " $(LIBTOOL) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=install $(INSTALL) $(INSTALL_STRIP_FLAG) '$$r' '$(DESTDIR)$(extsmoduledir)/.mesa/'"; \
+		$(LIBTOOL) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=install $(INSTALL) $(INSTALL_STRIP_FLAG) $$r "$(DESTDIR)$(extsmoduledir)/.mesa/"; \
+	else :; fi; \
+	done
 	test -z "$(extsmoduledir)" || $(MKDIR_P) "$(DESTDIR)$(extsmoduledir)"
 	@list='$(extsmodule_LTLIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
@@ -834,7 +841,7 @@
 	done
 check-am: all-am
 check: check-recursive
-all-am: Makefile $(LTLIBRARIES)
+all-am: Makefile $(LTLIBRARIES) $(GLXMODS)
 installdirs: installdirs-recursive
 installdirs-am:
 	for dir in "$(DESTDIR)$(extsmoduledir)" "$(DESTDIR)$(moduledir)"; do \
diff -Nuar hw.orig/xfree86/dri/Makefile.in hw/xfree86/dri/Makefile.in
--- hw.orig/xfree86/dri/Makefile.in	2009-04-15 17:22:15.000000000 +0200
+++ hw/xfree86/dri/Makefile.in	2009-04-15 17:22:48.000000000 +0200
@@ -377,7 +377,7 @@
                    @GL_CFLAGS@
 
 libdri_la_LDFLAGS = -module -avoid-version @LIBDRM_LIBS@
-libdri_ladir = $(moduledir)/extensions
+libdri_ladir = $(moduledir)/extensions/.mesa
 libdri_la_SOURCES = \
 	dri.c \
 	dri.h \
