--- hw/xfree86/common/xf86Init.c.orig	2006-03-17 23:31:45.000000000 +0200
+++ hw/xfree86/common/xf86Init.c	2006-03-17 23:32:03.000000000 +0200
@@ -1377,7 +1377,7 @@
     }
   
   /* First the options that are only allowed for root */
-  if (getuid() == 0 || geteuid != 0)
+  if (getuid() == 0 || geteuid() != 0)
   {
     if (!strcmp(argv[i], "-modulepath"))
     {
@@ -1677,7 +1677,7 @@
   }
   if (!strcmp(argv[i], "-configure"))
   {
-    if (getuid() != 0 && geteuid == 0) {
+    if (getuid() != 0 && geteuid() == 0) {
 	ErrorF("The '-configure' option can only be used by root.\n");
 	exit(1);
     }
