diff -purN gtkhtml-1.1.10/components/html-editor/popup.c gtkhtml/components/html-editor/popup.c
--- gtkhtml-1.1.10/components/html-editor/popup.c	2003-01-23 10:44:21.000000000 -0500
+++ gtkhtml/components/html-editor/popup.c	2004-06-25 16:34:20.120428672 -0400
@@ -496,6 +496,7 @@ prepare_properties_and_menu (GtkHTMLCont
 			ADD_PROP (PARAGRAPH);
 			break;
 		default:
+			;
 		}
 		if (obj->parent && obj->parent->parent && HTML_IS_TABLE_CELL (obj->parent->parent)) {
 			if (cd->format_html) {
