--- jack/Makefile.am.orig	2011-06-29 05:49:41.000000000 +0400
+++ jack/Makefile.am		2014-02-02 01:05:42.174435645 +0400
@@ -5,6 +5,7 @@
 libjackinclude_HEADERS = 	\
 	intclient.h		\
 	jack.h			\
+	jslist.h		\
 	ringbuffer.h		\
 	statistics.h            \
 	session.h               \
@@ -27,7 +28,6 @@
 	hardware.h 		\
 	internal.h 		\
 	intsimd.h 		\
-	jslist.h		\
 	memops.h		\
 	messagebuffer.h		\
 	pool.h			\
--- jack/Makefile.in.orig	2011-09-28 17:10:12.000000000 +0400
+++ jack/Makefile.in		2014-02-02 01:05:55.330784017 +0400
@@ -233,6 +233,7 @@
 libjackinclude_HEADERS = \
 	intclient.h		\
 	jack.h			\
+	jslist.h		\
 	ringbuffer.h		\
 	statistics.h            \
 	session.h               \
@@ -255,7 +256,6 @@
 	hardware.h 		\
 	internal.h 		\
 	intsimd.h 		\
-	jslist.h		\
 	memops.h		\
 	messagebuffer.h		\
 	pool.h			\
