--- Makefile.in	2006-10-19 13:22:27.000000000 +0000
+++ Makefile.in.noldap	2007-01-13 22:33:13.000000000 +0000
@@ -442,10 +442,6 @@
 
 install-data-local:
 	@$(NORMAL_INSTALL)
-	@if test ! -f $(DESTDIR)$(sysconfdir)/ldap.conf; then \
-		$(mkinstalldirs) $(DESTDIR)$(sysconfdir); \
-		$(INSTALL_DATA) -o root -g root $(srcdir)/ldap.conf $(DESTDIR)$(sysconfdir)/ldap.conf; \
-	fi
 
 uninstall-local:
 	@$(NORMAL_UNINSTALL)
