From 068a506c372dea3debc02ed01b43c71805c4c5bc Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 31 Dec 2018 11:30:27 +0100
Subject: [PATCH] fix Makefile

---
 source/Makefile | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/source/Makefile b/source/Makefile
index f9e75bd..503b192 100644
--- a/source/Makefile
+++ b/source/Makefile
@@ -17,15 +17,13 @@
 #
 # -------------------------------------------------------------------------
 
-SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 PREFIX ?= /usr/local
 BINDIR ?= $(PREFIX)/bin
-LIBDIR ?= $(PREFIX)/lib$(SUFFIX)
+LIBDIR ?= $(PREFIX)/lib
 
 VERSION = 0.9.7
 CPPFLAGS += -MMD -MP -DVERSION=\"$(VERSION)\" -DLIBDIR=\"$(LIBDIR)\"
-CXXFLAGS += -O2 -Wall
-CXXFLAGS += -march=native
+CXXFLAGS += -Wall
 
 
 all:	aeolus aeolus_x11.so aeolus_txt.so
@@ -45,7 +43,7 @@ $(AEOLUS_O):
 XIFACE_O =	styles.o mainwin.o midiwin.o audiowin.o instrwin.o editwin.o \
 	midimatrix.o multislider.o functionwin.o xiface.o addsynth.o
 aeolus_x11.so:	CPPFLAGS += -D_REENTRANT
-aeolus_x11.so:	CPPFLAGS += $(shell pkgconf --cflags freetype2)
+aeolus_x11.so:	CPPFLAGS += $(shell pkg-config --cflags freetype2)
 aeolus_x11.so:	CXXFLAGS += -shared -fPIC
 aeolus_x11.so:	LDFLAGS += -shared
 aeolus_x11.so:	LDLIBS += -lclthreads -lclxclient -lpthread -lXft -lX11
-- 
2.20.1

