--- Makefile.in.orig	2008-03-31 14:26:33.000000000 -0500
+++ Makefile.in	2008-03-31 14:32:28.000000000 -0500
@@ -50,13 +50,13 @@
 
 DESTDIR =
 
-EJABBERDDIR = $(DESTDIR)@prefix@/var/lib/ejabberd
+EJABBERDDIR = $(DESTDIR)@localstatedir@/lib/ejabberd
 BEAMDIR = $(EJABBERDDIR)/ebin
 PRIVDIR = $(EJABBERDDIR)/priv
 SODIR = $(PRIVDIR)/lib
 MSGSDIR = $(PRIVDIR)/msgs
-LOGDIR = $(DESTDIR)@prefix@/var/log/ejabberd
-ETCDIR = $(DESTDIR)@prefix@/etc/ejabberd
+LOGDIR = $(DESTDIR)@localstatedir@/log/ejabberd
+ETCDIR = $(DESTDIR)@sysconfdir@/ejabberd
 SBINDIR = $(DESTDIR)@prefix@/sbin
 
 ifeq ($(shell uname),Darwin)
