From 24f92892d4b022231b0cb5bd95b946dd5ac8b628 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 19 Jan 2018 18:52:28 +0100
Subject: [PATCH] fix search for ncurses

---
 cmake/ConkyPlatformChecks.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/ConkyPlatformChecks.cmake b/cmake/ConkyPlatformChecks.cmake
index 6a68852..ff1a4cc 100644
--- a/cmake/ConkyPlatformChecks.cmake
+++ b/cmake/ConkyPlatformChecks.cmake
@@ -122,7 +122,7 @@ if(BUILD_HTTP)
 endif(BUILD_HTTP)
 
 if(BUILD_NCURSES)
-	pkg_check_modules(NCURSES ncurses)
+	pkg_check_modules(NCURSES ncursesw)
 	if(NOT NCURSES_FOUND)
 		message(FATAL_ERROR "Unable to find ncurses library")
 	endif(NOT NCURSES_FOUND)
-- 
2.15.1

