--- src/ccurl_thread.c.orig	2010-10-06 01:29:36.000000000 +0400
+++ src/ccurl_thread.c	2011-09-14 23:09:00.762899542 +0400
@@ -33,7 +33,6 @@
 #endif /* DEBUG */
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 /*
--- src/eve.c.orig	2010-10-06 01:29:36.000000000 +0400
+++ src/eve.c	2011-09-14 23:09:15.386715713 +0400
@@ -37,7 +37,6 @@
 #include <libxml/xmlwriter.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include <time.h>
