From 510014d537d95de656af781590aab30e7d8763c5 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Wed, 1 Nov 2017 12:20:42 +0100
Subject: [PATCH 2/3] install config files

---
 include/meson.build | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/include/meson.build b/include/meson.build
index 00ec057..cf53c82 100644
--- a/include/meson.build
+++ b/include/meson.build
@@ -207,7 +207,9 @@ conf_data.set_quoted('BUILDERADDR', get_option('builder_addr'))
 conf_data.set_quoted('BUILDERSTRING', get_option('builder_string'))
 
 configure_file(output : 'dix-config.h',
-               configuration : conf_data)
+               configuration : conf_data,
+               install_dir: xorgsdkdir
+)
 
 
 
@@ -218,7 +220,9 @@ version_data.set_quoted('VENDOR_NAME_SHORT', get_option('vendor_name_short'))
 version_data.set_quoted('VENDOR_WEB', get_option('vendor_web'))
 version_data.set_quoted('VENDOR_MAN_VERSION', 'Version @0@.@1@.@2@'.format(major, minor, patch))
 configure_file(output : 'version-config.h',
-               configuration : version_data)
+               configuration : version_data,
+               install_dir: xorgsdkdir
+)
 
 xkb_data = configuration_data()
 
@@ -232,7 +236,9 @@ xkb_data.set_quoted('XKB_DFLT_OPTIONS', get_option('xkb_default_options'))
 xkb_data.set_quoted('XKM_OUTPUT_DIR', xkb_output_dir)
 
 configure_file(output : 'xkb-config.h',
-               configuration : xkb_data)
+               configuration : xkb_data,
+               install_dir: xorgsdkdir
+)
 
 xorg_data = configuration_data()
 
@@ -277,7 +283,9 @@ endif
 
 configure_file(output : 'xorg-config.h',
                input : 'xorg-config.h.meson.in',
-               configuration : xorg_data)
+               configuration : xorg_data,
+               install_dir: xorgsdkdir
+)
 
 xwin_data = configuration_data()
 xwin_data.set_quoted('DEFAULT_LOGDIR', log_dir)
@@ -291,7 +299,9 @@ xwin_data.set10('CYGMULTIWINDOW_DEBUG', enable_debugging)
 
 configure_file(output : 'xwin-config.h',
                input : 'xwin-config.h.meson.in',
-               configuration : xwin_data)
+               configuration : xwin_data,
+               install_dir: xorgsdkdir
+)
 
 if build_xorg
     install_data(
-- 
2.16.1

