From 20d214cddfc9b5ed4a64e8d3a5610256813b0f00 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Fri, 9 Feb 2018 14:04:21 +0100
Subject: [PATCH 3/3] automake fixes for smgl

---
 hw/xfree86/dixmods/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/xfree86/dixmods/Makefile.am b/hw/xfree86/dixmods/Makefile.am
index 856659f..e156043 100644
--- a/hw/xfree86/dixmods/Makefile.am
+++ b/hw/xfree86/dixmods/Makefile.am
@@ -8,7 +8,7 @@ module_LTLIBRARIES = libfb.la \
                      libwfb.la \
                      libshadow.la
 
-extsmoduledir = $(moduledir)/extensions
+extsmoduledir = $(moduledir)/extensions/.mesa
 extsmodule_LTLIBRARIES = $(GLXMODS)
 
 AM_CFLAGS = @XORG_CFLAGS@ @DIX_CFLAGS@
-- 
2.16.1

