diff -urN evms-2.5.1.ori/plugins/s390/format.c evms-2.5.1/plugins/s390/format.c
--- evms-2.5.1.ori/plugins/s390/format.c	2004-12-12 15:40:21.000000000 +0100
+++ evms-2.5.1/plugins/s390/format.c	2005-03-15 21:46:53.322211332 +0100
@@ -37,7 +37,7 @@
 #include "helpers.h"
 #include "format.h"
 
-#define BLKGETSIZE64 _IOR(0x12,114,sizeof(u_int64_t))
+#define BLKGETSIZE64 _IOR(0x12,114,size_t)
 #define BLKSSZGET    _IO(0x12,104)
 #define HDIO_GETGEO  0x0301
 
