--- SDL_ttf.c.orig	2003-02-21 18:38:15.000000000 +0100
+++ SDL_ttf.c	2004-06-12 14:58:35.928654641 +0200
@@ -39,10 +39,11 @@
 #define FREEA(p) free(p)
 #endif
 
-#include <freetype/freetype.h>
-#include <freetype/ftoutln.h>
-#include <freetype/ttnameid.h>
-#include <freetype/internal/ftobjs.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_OUTLINE_H
+#include FT_TRUETYPE_IDS_H
+#include FT_INTERNAL_OBJECTS_H
 
 #include "SDL.h"
 #include "SDL_endian.h"
