diff -purN Gtk-Perl-0.7009/GdkPixbuf/Makefile.PL Gtk-Perl/GdkPixbuf/Makefile.PL
--- Gtk-Perl-0.7009/GdkPixbuf/Makefile.PL	2003-06-17 05:47:56.000000000 -0400
+++ Gtk-Perl/GdkPixbuf/Makefile.PL	2004-06-25 20:57:26.688505536 -0400
@@ -38,6 +38,8 @@ foreach ($gpb->get_headers) {
         push @a, '-i', $_;
 }
 
+chop($::inc = `gdk-pixbuf-config --cflags`);
+chop($::libs = `gdk-pixbuf-config --libs`);
 $gpb->add_xs(gendefs::gendefs(@a));
 
 $depend = { 'build/GdkPixbufDefs.c' => 'build/GdkPixbufDefs.h build/PerlGdkPixbufInt.h' };
diff -purN Gtk-Perl-0.7009/Gnome/Makefile.PL Gtk-Perl/Gnome/Makefile.PL
--- Gtk-Perl-0.7009/Gnome/Makefile.PL	2003-06-17 05:47:56.000000000 -0400
+++ Gtk-Perl/Gnome/Makefile.PL	2004-06-25 21:03:39.317857264 -0400
@@ -66,6 +66,8 @@ foreach ($gnome->get_headers) {
 	push @a, '-i', $_;
 }
 
+chop($::inc = `gnome-config --cflags gnomeui zvt`);
+chop($::libs = `gnome-config --libs gnomeui zvt`);
 $gnome->add_xs(gendefs::gendefs('-L', @a));
 
 $depend = { 'build/GnomeDefs.c' => 'build/GnomeDefs.h build/PerlGnomeInt.h' };
diff -purN Gtk-Perl-0.7009/GnomePrint/Makefile.PL Gtk-Perl/GnomePrint/Makefile.PL
--- Gtk-Perl-0.7009/GnomePrint/Makefile.PL	2003-06-17 05:47:56.000000000 -0400
+++ Gtk-Perl/GnomePrint/Makefile.PL	2004-06-25 21:07:35.659927808 -0400
@@ -56,6 +56,8 @@ foreach ($gp->get_headers) {
 	push @a, '-i', $_;
 }
 
+chop($::inc = `gnome-config --cflags print`);
+chop($::libs = `gnome-config --libs print`);
 $gp->add_xs(gendefs::gendefs('-L', @a));
 
 $depend = { 'build/GnomePrintDefs.c' => 'build/GnomePrintDefs.h build/PerlGnomePrintInt.h' };
diff -purN Gtk-Perl-0.7009/GtkXmHTML/Makefile.PL Gtk-Perl/GtkXmHTML/Makefile.PL
--- Gtk-Perl-0.7009/GtkXmHTML/Makefile.PL	2003-06-17 05:47:56.000000000 -0400
+++ Gtk-Perl/GtkXmHTML/Makefile.PL	2004-06-25 20:58:36.817844256 -0400
@@ -44,6 +44,8 @@ foreach ($xm->get_headers) {
 	push @a, '-i', $_;
 }
 
+chop($::inc = `gnome-config --cflags gnomeui`);
+chop($::libs = `gnome-config --libs gtkxmhtml`);
 $xm->add_xs(gendefs::gendefs(@a));
 
 $depend = { 'build/GtkXmHTMLDefs.c' => 'build/GtkXmHTMLDefs.h build/PerlGtkXmHTMLInt.h' };
