From 9f70bd72df9a77db54d5ed205cfee168008d27e7 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 1 Jan 2020 13:48:40 +0100
Subject: [PATCH] fix possible buffer overflow

---
 xsel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xsel.c b/xsel.c
index 79cc68b..1016727 100644
--- a/xsel.c
+++ b/xsel.c
@@ -79,7 +79,7 @@ static Boolean do_follow = False;
 static Boolean no_daemon = False;
 
 /* logfile: name of file to log error messages to when detached */
-static char logfile[MAXFNAME];
+static char logfile[MAXFNAME + 1];
 
 /* fstat() on stdin and stdout */
 static struct stat in_statbuf, out_statbuf;
-- 
2.24.1

