From ce3183eb28cd77cb656eb4b77c005f96ecd35f15 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sun, 14 Apr 2024 20:47:10 +0200
Subject: [PATCH] fix x11locker.cpp

---
 x11locker.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/x11locker.cpp b/x11locker.cpp
index 577ee9c..724130d 100644
--- a/x11locker.cpp
+++ b/x11locker.cpp
@@ -51,8 +51,8 @@ void X11Locker::initialize()
     updateGeo();
 
     // virtual root property
-    gXA_VROOT = XInternAtom(X11Info::display(), "__SWM_VROOT", False);
-    gXA_SCREENSAVER_VERSION = XInternAtom(X11Info::display(), "_SCREENSAVER_VERSION", False);
+    gXA_VROOT = XInternAtom(X11Info::display(), "__SWM_VROOT", false);
+    gXA_SCREENSAVER_VERSION = XInternAtom(X11Info::display(), "_SCREENSAVER_VERSION", false);
 
     // read the initial information about all toplevel windows
     Window r, p;
@@ -102,7 +102,7 @@ void X11Locker::showLockWindow()
     qCDebug(KSCREENLOCKER) << "Lock window Id: " << m_background->winId();
 
     m_background->setPosition(0, 0);
-    XSync(X11Info::display(), False);
+    XSync(X11Info::display(), false);
 
     setVRoot(m_background->winId(), m_background->winId());
 }
@@ -123,7 +123,7 @@ void X11Locker::hideLockWindow()
         XChangeProperty(X11Info::display(), gVRoot, gXA_VROOT, XA_WINDOW, 32, PropModeReplace, (unsigned char *)vroot_data, 1);
         gVRoot = 0;
     }
-    XSync(X11Info::display(), False);
+    XSync(X11Info::display(), false);
     m_allowedWindows.clear();
 }
 
@@ -161,7 +161,7 @@ void X11Locker::saveVRoot()
                                     gXA_VROOT,
                                     0,
                                     1,
-                                    False,
+                                    false,
                                     XA_WINDOW,
                                     &actual_type,
                                     &actual_format,
@@ -244,7 +244,7 @@ void X11Locker::fakeFocusIn(WId window)
     ev.xfocus.window = window;
     ev.xfocus.mode = NotifyNormal;
     ev.xfocus.detail = NotifyAncestor;
-    XSendEvent(X11Info::display(), window, False, NoEventMask, &ev);
+    XSendEvent(X11Info::display(), window, false, NoEventMask, &ev);
     XFlush(X11Info::display());
 
     m_focusedLockWindow = window;
-- 
2.44.0

