--- gtk/src/rbgtk.h.orig	2006-12-29 16:17:29.000000000 +0300
+++ gtk/src/rbgtk.h	2008-04-23 15:14:41.623007786 +0400
@@ -115,7 +115,7 @@
  */
 typedef void (*rbgtkiter_set_value_func)(void *model, GtkTreeIter *iter,
                                          gint column, GValue *value);
-extern void rbgtk_register_treeiter_set_value_func(VALUE klass, rbgtkiter_set_value_func);
+extern void rbgtk_register_treeiter_set_value_func(GType klass, rbgtkiter_set_value_func);
 
 /*
  * Gtk::SelectionData
