From 5a52053068bec0216357c15c17a44059ba22d0ef Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Tue, 12 Apr 2022 16:54:32 +0200
Subject: [PATCH] Remove superflous include of non-standard <error.h>

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 tools/elfdeps.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tools/elfdeps.c b/tools/elfdeps.c
index d205935bbb49..9cb7b68ff319 100644
--- a/tools/elfdeps.c
+++ b/tools/elfdeps.c
@@ -5,7 +5,6 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <fcntl.h>
-#include <error.h>
 #include <errno.h>
 #include <popt.h>
 #include <gelf.h>
-- 
2.35.1

