#
# From https://bugs.kde.org/show_bug.cgi?id=193242
#
--- kdebase-3.5.10/kcontrol/kfontinst/kfontinst/Fontmap.cpp	2009-05-20 13:06:00.477670083 +0300
+++ kdebase-3.5.10/kcontrol/kfontinst/kfontinst/Fontmap.cpp	2009-05-20 15:09:44.061136748 +0300
@@ -44,7 +44,7 @@
 
 using namespace std;
 
-static char * findSpace(char *str)
+static const char * findSpace(const char *str)
 {
     while(str && *str!=' ' && *str!='\t')
         str++;
@@ -65,7 +65,7 @@
     char a[constMaxLen+1],
          b[constFileMaxLen+1];
 
-    char *slash1=strchr(line, '/'),
+    const char *slash1=strchr(line, '/'),
          *space1=slash1 ? findSpace(slash1) : NULL, //strchr(slash1, ' ') : NULL,
          *ob=slash1 ? strchr(slash1, '(') : NULL,
          *cb=ob ? strchr(ob, ')') : NULL,
