diff -Naur kdebase-3.5.10.orig/kcontrol/crypto/crypto.cpp kdebase-3.5.10/kcontrol/crypto/crypto.cpp
--- kdebase-3.5.10.orig/kcontrol/crypto/crypto.cpp	2006-10-01 19:31:49.000000000 +0200
+++ kdebase-3.5.10/kcontrol/crypto/crypto.cpp	2010-12-30 11:43:14.000000000 +0100
@@ -2321,7 +2321,7 @@
 unsigned int i;
 SSL_CTX *ctx;
 SSL *ssl;
-SSL_METHOD *meth;
+const SSL_METHOD *meth;
 
   SSLv2Box->clear();
   SSLv3Box->clear();
@@ -2337,7 +2337,7 @@
   CipherItem *item;
   for (i=0; ; i++) {
     int j, k;
-    SSL_CIPHER *sc;
+    const SSL_CIPHER *sc;
     sc = (meth->get_cipher)(i);
     if (!sc)
       break;
@@ -2365,7 +2365,7 @@
 
   for (i=0; ; i++) {
     int j, k;
-    SSL_CIPHER *sc;
+    const SSL_CIPHER *sc;
     sc = (meth->get_cipher)(i);
     if (!sc)
       break;
