--- a/CMakeLists.txt	2018-09-08 10:59:08.815543877 +0200
+++ b/CMakeLists.txt	2018-09-08 11:07:32.700184611 +0200
@@ -77,7 +77,7 @@
     set_package_properties(Phonon4Qt5 PROPERTIES
         DESCRIPTION "Qt-based audio library"
         # This is REQUIRED since you cannot tell CMake "either one of those two optional ones are required"
-        TYPE REQUIRED
+        TYPE OPTIONAL
         PURPOSE "Needed to build audio notification support when Canberra isn't available")
     if (Phonon4Qt5_FOUND)
         add_definitions(-DHAVE_PHONON4QT5)
