--- Makefile.org	2003-01-17 15:55:20.000000000 +0100
+++ Makefile	2003-01-17 16:05:10.000000000 +0100
@@ -1,8 +1,8 @@
 TGT=noip2
 CC=gcc
 
-PREFIX=/usr/local
-CONFDIR=${PREFIX}/etc
+PREFIX=/usr
+CONFDIR=/etc
 BINDIR=${PREFIX}/bin
 
 # these defines are for linux and BSD
@@ -14,7 +14,7 @@
 # ARCH=sun
 
 ${TGT}: Makefile ${TGT}.c 
-	${CC} -Wall -g -O2 -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS}
+	${CC} ${CFLAGS} -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS} ${LDFLAGS}
 
 install: ${TGT} 
 	cp ${TGT} ${BINDIR}/${TGT}
