--- addon/doxywizard/Makefile.in.orig	2008-12-29 08:15:15.000000000 -0600
+++ addon/doxywizard/Makefile.in	2008-12-29 08:15:52.000000000 -0600
@@ -16,10 +16,10 @@
 	$(MAKE) -f Makefile.doxywizard 
 
 Makefile.doxywizard: doxywizard.pro
-	$(QTDIR)/bin/qmake doxywizard.pro -o Makefile.doxywizard
+	$(QTDIR)/bin/qt4/qmake doxywizard.pro -o Makefile.doxywizard
 
 qmake: 
-	$(QTDIR)/bin/qmake doxywizard.pro -o Makefile.doxywizard
+	$(QTDIR)/bin/qt4/qmake doxywizard.pro -o Makefile.doxywizard
 
 clean: Makefile.doxywizard 
 	$(MAKE) -f Makefile.doxywizard clean 
