From d5e1eb0edca1125231b94c644ea1ef8ed19d8a31 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 14 Dec 2016 22:07:05 +0100
Subject: [PATCH] initialize locale, fixes #2043

---
 pelican/tools/pelican_quickstart.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pelican/tools/pelican_quickstart.py b/pelican/tools/pelican_quickstart.py
index 6b4eb5a5..a02f91e6 100755
--- a/pelican/tools/pelican_quickstart.py
+++ b/pelican/tools/pelican_quickstart.py
@@ -21,6 +21,8 @@ import six
 
 from pelican import __version__
 
+if (sys.version_info.major == 2):
+    locale.setlocale(locale.LC_ALL, '')
 
 _TEMPLATES_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)),
                               "templates")
-- 
2.11.0

