--- utils.c~	2005-11-11 15:55:39.000000000 +0100
+++ utils.c	2005-11-11 15:55:42.000000000 +0100
@@ -331,7 +331,7 @@
 // with printf strings and args.                                            //
 //////////////////////////////////////////////////////////////////////////////
 
-extern int debug_logging_mode;
+int debug_logging_mode = FALSE;
 
 void error_line (char *error, ...)
 {
--- wavpack.c~	2005-11-02 04:44:52.000000000 +0100
+++ wavpack.c	2005-11-11 15:46:35.000000000 +0100
@@ -91,7 +91,7 @@
 // this global is used to indicate the special "debug" mode where extra debug messages
 // are displayed and all messages are logged to the file \wavpack.log
 
-int debug_logging_mode;
+extern int debug_logging_mode;
 
 static int overwrite_all, num_files, file_index;
 
--- wvunpack.c~	2005-11-02 04:45:16.000000000 +0100
+++ wvunpack.c	2005-11-11 15:46:42.000000000 +0100
@@ -79,7 +79,7 @@
 // this global is used to indicate the special "debug" mode where extra debug messages
 // are displayed and all messages are logged to the file \wavpack.log
 
-int debug_logging_mode;
+extern int debug_logging_mode;
 
 static char overwrite_all = 0, delete_source = 0, raw_decode = 0, summary = 0,
     ignore_wvc = 0, quiet_mode = 0, calc_md5 = 0, copy_time = 0;
