diff -Nur spamprobe-1.4d/src/database/DatabaseConfig.cc spamprobe-1.4d-fixed/src/database/DatabaseConfig.cc
--- spamprobe-1.4d/src/database/DatabaseConfig.cc	2006-11-17 08:24:48.000000000 +0100
+++ spamprobe-1.4d-fixed/src/database/DatabaseConfig.cc	2009-08-10 18:54:20.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include <stdexcept>
 #include "File.h"
 #include "WordData.h"
diff -Nur spamprobe-1.4d/src/hdl/HdlTokenizer.cc spamprobe-1.4d-fixed/src/hdl/HdlTokenizer.cc
--- spamprobe-1.4d/src/hdl/HdlTokenizer.cc	2006-11-17 08:24:48.000000000 +0100
+++ spamprobe-1.4d-fixed/src/hdl/HdlTokenizer.cc	2009-08-10 18:56:20.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include "AbstractCharReader.h"
 #include "HdlError.h"
 #include "HdlToken.h"
diff -Nur spamprobe-1.4d/src/includes/MultiLineSubString.h spamprobe-1.4d-fixed/src/includes/MultiLineSubString.h
--- spamprobe-1.4d/src/includes/MultiLineSubString.h	2006-11-17 08:24:48.000000000 +0100
+++ spamprobe-1.4d-fixed/src/includes/MultiLineSubString.h	2009-08-10 18:54:35.000000000 +0200
@@ -31,6 +31,7 @@
 #ifndef _MultiLineSubString_h
 #define _MultiLineSubString_h
 
+#include <climits>
 #include "AbstractMultiLineString.h"
 
 class MultiLineSubString : public AbstractMultiLineString
diff -Nur spamprobe-1.4d/src/includes/util.h spamprobe-1.4d-fixed/src/includes/util.h
--- spamprobe-1.4d/src/includes/util.h	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/includes/util.h	2009-08-10 18:53:24.000000000 +0200
@@ -42,6 +42,7 @@
 #include <cassert>
 #include <iostream>
 #include <cstdio>
+#include <cstring>
 #include "Ptr.h"
 #include "Ref.h"
 
diff -Nur spamprobe-1.4d/src/parser/AutoTrainMailMessageReader.cc spamprobe-1.4d-fixed/src/parser/AutoTrainMailMessageReader.cc
--- spamprobe-1.4d/src/parser/AutoTrainMailMessageReader.cc	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/parser/AutoTrainMailMessageReader.cc	2009-08-10 18:53:48.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include "MailMessage.h"
 #include "AutoTrainMailMessageReader.h"
 
diff -Nur spamprobe-1.4d/src/parser/HtmlTokenizer.cc spamprobe-1.4d-fixed/src/parser/HtmlTokenizer.cc
--- spamprobe-1.4d/src/parser/HtmlTokenizer.cc	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/parser/HtmlTokenizer.cc	2009-08-10 18:54:02.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include "AbstractTokenReceiver.h"
 #include "StringReader.h"
 #include "RegularExpression.h"
diff -Nur spamprobe-1.4d/src/parser/MailMessageReader.cc spamprobe-1.4d-fixed/src/parser/MailMessageReader.cc
--- spamprobe-1.4d/src/parser/MailMessageReader.cc	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/parser/MailMessageReader.cc	2009-08-10 18:55:09.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include "RegularExpression.h"
 #include "MailMessage.h"
 #include "MailMessageList.h"
diff -Nur spamprobe-1.4d/src/parser/MbxMailMessageReader.cc spamprobe-1.4d-fixed/src/parser/MbxMailMessageReader.cc
--- spamprobe-1.4d/src/parser/MbxMailMessageReader.cc	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/parser/MbxMailMessageReader.cc	2009-08-10 18:55:34.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include <stdexcept>
 #include "MailMessage.h"
 #include "MailMessageList.h"
diff -Nur spamprobe-1.4d/src/spamprobe/Command_cleanup.cc spamprobe-1.4d-fixed/src/spamprobe/Command_cleanup.cc
--- spamprobe-1.4d/src/spamprobe/Command_cleanup.cc	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/spamprobe/Command_cleanup.cc	2009-08-10 18:56:56.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include "CleanupManager.h"
 #include "SpamFilter.h"
 #include "FrequencyDB.h"
diff -Nur spamprobe-1.4d/src/spamprobe/Command_edit_term.cc spamprobe-1.4d-fixed/src/spamprobe/Command_edit_term.cc
--- spamprobe-1.4d/src/spamprobe/Command_edit_term.cc	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/spamprobe/Command_edit_term.cc	2009-08-10 18:57:23.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include "SpamFilter.h"
 #include "FrequencyDB.h"
 #include "CommandConfig.h"
diff -Nur spamprobe-1.4d/src/spamprobe/Command_exec.cc spamprobe-1.4d-fixed/src/spamprobe/Command_exec.cc
--- spamprobe-1.4d/src/spamprobe/Command_exec.cc	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/spamprobe/Command_exec.cc	2009-08-10 18:57:55.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include "SpamFilter.h"
 #include "CommandConfig.h"
 #include "ConfigManager.h"
diff -Nur spamprobe-1.4d/src/spamprobe/Command_import.cc spamprobe-1.4d-fixed/src/spamprobe/Command_import.cc
--- spamprobe-1.4d/src/spamprobe/Command_import.cc	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/spamprobe/Command_import.cc	2009-08-10 18:58:21.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include <fstream>
 #include "LineReader.h"
 #include "IstreamCharReader.h"
diff -Nur spamprobe-1.4d/src/spamprobe/Command_purge.cc spamprobe-1.4d-fixed/src/spamprobe/Command_purge.cc
--- spamprobe-1.4d/src/spamprobe/Command_purge.cc	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/spamprobe/Command_purge.cc	2009-08-10 18:58:48.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include "CleanupManager.h"
 #include "SpamFilter.h"
 #include "FrequencyDB.h"
diff -Nur spamprobe-1.4d/src/spamprobe/spamprobe.cc spamprobe-1.4d-fixed/src/spamprobe/spamprobe.cc
--- spamprobe-1.4d/src/spamprobe/spamprobe.cc	2006-11-17 08:24:49.000000000 +0100
+++ spamprobe-1.4d-fixed/src/spamprobe/spamprobe.cc	2009-08-10 18:59:12.000000000 +0200
@@ -28,6 +28,7 @@
 //    http://www.cooldevtools.com/qpl.html
 //
 
+#include <cstdlib>
 #include <unistd.h>
 #include <locale.h>
 #include <signal.h>
