--- src/tools.cc.old	2009-05-21 11:27:57.000000000 +0200
+++ src/tools.cc	2009-05-21 11:28:03.000000000 +0200
@@ -44,9 +44,9 @@
 #if HAVE_SYS_CAPABILITY_H
 #undef _POSIX_SOURCE
 /* Ugly glue to get around linux header madness colliding with glibc */
-#define _LINUX_TYPES_H
+// #define _LINUX_TYPES_H
 #define _LINUX_FS_H
-typedef uint32_t __u32;
+// typedef uint32_t __u32;
 #include <sys/capability.h>
 #endif
 #endif

