--- configure.orig	2005-01-11 07:48:11.072871907 +0100
+++ configure	2005-01-11 07:48:17.755020230 +0100
@@ -32461,7 +32461,7 @@
 
 	if test "$dynamic_python" = "yes"
 	then
-		PYMAKE="lib-1.5.2 lib-2.0.0 lib-2.1.1 lib-2.2.0 lib-2.3.0"
+		PYMAKE="lib-1.5.2 lib-2.0.0 lib-2.1.1 lib-2.2.0 lib-2.3.0 lib-2.4.0"
 		PYSTUB="stub"
                 PYRTMAKE="lib-1.5.2_rt lib-2.0.0_rt lib-2.1.1_rt lib-2.2.0_rt lib-2.2.0_rt"
 		PYRTSTUB="stub_rt"
@@ -32489,6 +32489,9 @@
 			     USESYSPY=1
 			     ;;
 
+      2.4) SYSPYVER="2.4"
+           USESYSPY=1
+           ;;
 			*)   { { echo "$as_me:$LINENO: error: Unrecognised python version: $pyver" >&5
 echo "$as_me: error: Unrecognised python version: $pyver" >&2;}
    { (exit 1); exit 1; }; }
