--- mirage.py.orig	2010-07-23 16:57:44.000000000 +0400
+++ mirage.py	2010-07-27 01:33:16.957210477 +0400
@@ -2339,7 +2339,7 @@
 			dialog.destroy()
 			while gtk.events_pending():
 				gtk.main_iteration()
-			self.screenshot_delay = int(delay.get_text())
+			self.screenshot_delay = delay.get_value_as_int()
 			gobject.timeout_add(int(self.screenshot_delay*1000), self._screenshot_grab, area1.get_active())
 		else:
 			dialog.destroy()
