diff -urpN libgii-0.8.6.old/input/linux_evdev/eventparse.c libgii-0.8.6/input/linux_evdev/eventparse.c
--- libgii-0.8.6.old/input/linux_evdev/eventparse.c	2004-09-18 18:54:38.812165144 -0700
+++ libgii-0.8.6/input/linux_evdev/eventparse.c	2004-09-18 18:55:14.134795288 -0700
@@ -232,9 +232,6 @@ dispatch_event(struct gii_input *inp, st
 #ifdef EV_FF_STATUS
 	case EV_FF_STATUS:
 #endif
-#ifdef EV_RST
-	case EV_RST:
-#endif
 	default:
 		return 0;
 	}
