--- avahi-daemon/dbus-protocol.c (revision 1299)
+++ avahi-daemon/dbus-protocol.c (revision 1333)
@@ -1068,8 +1068,8 @@
             server->bus,
             AVAHI_DBUS_NAME,
-#if (DBUS_VERSION_MAJOR == 0) && (DBUS_VERSION_MINOR >= 60)
+#if (DBUS_VERSION_MAJOR == 0) && (DBUS_VERSION_MINOR < 60)
+            DBUS_NAME_FLAG_PROHIBIT_REPLACEMENT,
+#else
             DBUS_NAME_FLAG_DO_NOT_QUEUE,
-#else
-            DBUS_NAME_FLAG_PROHIBIT_REPLACEMENT,
 #endif
             &error) != DBUS_REQUEST_NAME_REPLY_PRIMARY_OWNER) {
@@ -1107,9 +1107,4 @@
 
     if (server->bus) {
-#ifdef HAVE_DBUS_CONNECTION_CLOSE
-        dbus_connection_close(server->bus);
-#else
-        dbus_connection_disconnect(server->bus);
-#endif
         dbus_connection_unref(server->bus);
         server->bus = NULL;
@@ -1128,9 +1123,4 @@
 
     if (server->bus) {
-#ifdef HAVE_DBUS_CONNECTION_CLOSE
-        dbus_connection_close(server->bus);
-#else
-        dbus_connection_disconnect(server->bus);
-#endif
         dbus_connection_unref(server->bus);
         server->bus = NULL;
@@ -1167,9 +1157,4 @@
 fail:
     if (server->bus) {
-#ifdef HAVE_DBUS_CONNECTION_CLOSE
-        dbus_connection_close(server->bus);
-#else
-        dbus_connection_disconnect(server->bus);
-#endif
         dbus_connection_unref(server->bus);
     }

