--- avahi-0.6.31/configure	2012-02-14 17:36:33.221047221 -0500
+++ avahi-0.6.31.o/configure	2012-06-13 00:55:43.876468604 -0400
@@ -20670,7 +20670,7 @@
   ;;
   *)
   as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $QT3_PREFIX
+for as_dir in /usr/bin 
 do
   IFS=$as_save_IFS
   test -z "$as_dir" && as_dir=.
@@ -20840,7 +20840,7 @@
   ;;
   *)
   as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $QT4_PREFIX
+for as_dir in /usr/bin/qt4 
 do
   IFS=$as_save_IFS
   test -z "$as_dir" && as_dir=.
