diff -uNr avahi-0.6.12/configure.ac avahi-0.6.12.cp/configure.ac
--- avahi-0.6.12/configure.ac	2006-07-22 14:39:40.000000000 +0200
+++ avahi-0.6.12.cp/configure.ac	2006-07-24 15:13:36.000000000 +0200
@@ -315,7 +315,7 @@
 	    QT3_PREFIX="`$PKG_CONFIG --variable=prefix qt-mt`/bin"
         MOC_QT3="no"
 	    AC_CHECK_FILE( "$QT3_PREFIX/moc-qt3", [ MOC_QT3=$QT3_PREFIX/moc-qt3 ], [
-        	    AC_CHECK_FILE("$QT3_PREFIX/moc", [ MOC_QT3=$QT3_PREFIX/moc ], [ 
+        	    AC_CHECK_FILE("/usr/bin/moc", [ MOC_QT3=/usr/bin/moc ], [ 
                 AC_MSG_ERROR([Couldn't find QT3 moc])])])
         AC_SUBST(MOC_QT3)
 fi
