diff -Naur powersave-0.12.20/config_files/dbus_powersave.conf powersave-0.12.20.new/config_files/dbus_powersave.conf
--- powersave-0.12.20/config_files/dbus_powersave.conf	2006-06-12 21:12:18.000000000 +0300
+++ powersave-0.12.20.new/config_files/dbus_powersave.conf	2006-07-10 19:39:49.000000000 +0300
@@ -54,7 +54,7 @@
   </policy>
 
   <!--    Allow desktop users to connect -->
-  <policy at_console="true">
+  <policy group="powerdev">
     <allow send_interface="com.novell.powersave"/>
     <allow send_interface="com.novell.powersave.manager"/>
     <allow send_interface="com.novell.powersave.request"/>
diff -Naur powersave-0.12.20/docs/powersave.8.in powersave-0.12.20.new/docs/powersave.8.in
--- powersave-0.12.20/docs/powersave.8.in	2006-06-12 21:12:33.000000000 +0300
+++ powersave-0.12.20.new/docs/powersave.8.in	2006-07-10 19:39:57.000000000 +0300
@@ -28,8 +28,10 @@
 
 .SH OPTIONS THAT NEED A RUNNING POWERSAVE DAEMON
 
-Be aware that the powersave system group exists and that 
-users allowed to use these functions are part of the powersave group.
+Be aware that the powerdev system group exists and that 
+users allowed to use these functions are part of the powerdev group. After
+adding users to the powerdev group you should reload the system message bus 
+with "/etc/init.d/dbus reload" to pick up the new group memberships.
 
 .TP
 .I  --suspend-to-disk -U
