diff -ru oscpack.old/examples/OscDump.cpp oscpack/examples/OscDump.cpp
--- oscpack.old/examples/OscDump.cpp	2006-11-20 14:49:40.000000000 +0000
+++ oscpack/examples/OscDump.cpp	2009-04-30 15:10:05.000000000 +0100
@@ -36,6 +36,8 @@
 
 
 #include <iostream>
+#include <cstdlib>
+#include <cstring>
 
 #include "osc/OscReceivedElements.h"
 #include "osc/OscPrintReceivedElements.h"
diff -ru oscpack.old/osc/OscPrintReceivedElements.cpp oscpack/osc/OscPrintReceivedElements.cpp
--- oscpack.old/osc/OscPrintReceivedElements.cpp	2006-11-20 14:49:38.000000000 +0000
+++ oscpack/osc/OscPrintReceivedElements.cpp	2009-04-30 15:07:56.000000000 +0100
@@ -32,6 +32,8 @@
 #include <iostream>
 #include <iomanip>
 #include <ctime>
+#include <cstdlib>
+#include <cstring>
 
 
 namespace osc{
diff -ru oscpack.old/tests/OscReceiveTest.cpp oscpack/tests/OscReceiveTest.cpp
--- oscpack.old/tests/OscReceiveTest.cpp	2006-11-20 14:49:38.000000000 +0000
+++ oscpack/tests/OscReceiveTest.cpp	2009-04-30 15:09:15.000000000 +0100
@@ -31,6 +31,7 @@
 
 #include <string.h>
 #include <iostream>
+#include <cstdlib>
 
 #include "osc/OscReceivedElements.h"
 
diff -ru oscpack.old/tests/OscSendTests.cpp oscpack/tests/OscSendTests.cpp
--- oscpack.old/tests/OscSendTests.cpp	2006-11-20 14:49:38.000000000 +0000
+++ oscpack/tests/OscSendTests.cpp	2009-04-30 15:08:32.000000000 +0100
@@ -31,6 +31,7 @@
 
 #include <iostream>
 #include <string.h>
+#include <cstdlib>
 
 #include "osc/OscOutboundPacketStream.h"
 
diff -ru oscpack.old/tests/OscUnitTests.cpp oscpack/tests/OscUnitTests.cpp
--- oscpack.old/tests/OscUnitTests.cpp	2006-11-20 14:49:38.000000000 +0000
+++ oscpack/tests/OscUnitTests.cpp	2009-04-30 14:56:24.000000000 +0100
@@ -29,6 +29,7 @@
 */
 #include "OscUnitTests.h"
 
+#include <cstring>
 #include <iostream>
 #include <iomanip>
 
