diff -Naur desktopcouch-0.6.4.orig/desktopcouch/records/server.py desktopcouch-0.6.4/desktopcouch/records/server.py
--- desktopcouch-0.6.4.orig/desktopcouch/records/server.py	2010-05-07 02:01:04.308859356 -0700
+++ desktopcouch-0.6.4/desktopcouch/records/server.py	2010-05-07 02:02:06.355867612 -0700
@@ -22,7 +22,7 @@
 """The Desktop Couch Records API."""
  
 from couchdb import Server
-from couchdb.client import Resource
+from couchdb.http import Resource
 import desktopcouch
 from desktopcouch.records import server_base
 import urlparse
diff -Naur desktopcouch-0.6.4.orig/desktopcouch/records/server_base.py desktopcouch-0.6.4/desktopcouch/records/server_base.py
--- desktopcouch-0.6.4.orig/desktopcouch/records/server_base.py	2010-05-07 02:01:04.308859356 -0700
+++ desktopcouch-0.6.4/desktopcouch/records/server_base.py	2010-05-07 02:01:53.851940948 -0700
@@ -39,7 +39,7 @@
 from oauth import oauth
 
 from couchdb import Server
-from couchdb.client import ResourceNotFound, ResourceConflict, uri as couchdburi
+from couchdb.http import ResourceNotFound, ResourceConflict, uri as couchdburi
 from couchdb.design import ViewDefinition
 from record import Record
 
