diff -Naur desktopcouch-0.6.4.orig/desktopcouch/records/server_base.py desktopcouch-0.6.4/desktopcouch/records/server_base.py
--- desktopcouch-0.6.4.orig/desktopcouch/records/server_base.py	2010-05-07 02:01:04.308859356 -0700
+++ desktopcouch-0.6.4/desktopcouch/records/server_base.py	2010-05-07 02:50:55.242862656 -0700
@@ -36,7 +36,7 @@
     import json
 # pylint: enable-msg=F0401
 
-from oauth import oauth
+import oauth2 as oauth
 
 from couchdb import Server
 from couchdb.client import ResourceNotFound, ResourceConflict, uri as couchdburi
diff -Naur desktopcouch-0.6.4.orig/desktopcouch/replication_services/ubuntuone.py desktopcouch-0.6.4/desktopcouch/replication_services/ubuntuone.py
--- desktopcouch-0.6.4.orig/desktopcouch/replication_services/ubuntuone.py	2010-05-07 02:01:04.307859496 -0700
+++ desktopcouch-0.6.4/desktopcouch/replication_services/ubuntuone.py	2010-05-07 02:50:46.389139860 -0700
@@ -1,4 +1,4 @@
-from oauth import oauth
+import oauth2 as oauth
 import logging
 import httplib2
 import gnomekeyring
