diff -Naur cppsocket.orig/config.sub cppsocket/config.sub
--- cppsocket.orig/config.sub	2008-04-26 12:35:47.000000000 -0700
+++ cppsocket/config.sub	2008-04-26 12:36:01.000000000 -0700
@@ -621,6 +621,9 @@
 	pentiumii-* | pentium2-*)
 		basic_machine=i786-`echo $basic_machine | sed 's/^[^-]*-//'`
 		;;
+	x86_64-*)
+		basic_machine=x86_64-`echo $basic_machine | sed 's/^[^-]*-//'`
+		;;
 	pn)
 		basic_machine=pn-gould
 		;;
