diff -Naur cppsocket.orig/cppsocket/exception.h cppsocket/cppsocket/exception.h
--- cppsocket.orig/cppsocket/exception.h	2008-04-26 12:26:07.000000000 -0700
+++ cppsocket/cppsocket/exception.h	2008-04-26 12:26:37.000000000 -0700
@@ -26,6 +26,7 @@
 #include <errno.h>
 #include <netdb.h>
 #include <string>
+#include <cstring>
 #include <exception>
 
 namespace CPPSocket
diff -Naur cppsocket.orig/cppsocket/stringbuffer.cpp cppsocket/cppsocket/stringbuffer.cpp
--- cppsocket.orig/cppsocket/stringbuffer.cpp	2008-04-26 12:26:07.000000000 -0700
+++ cppsocket/cppsocket/stringbuffer.cpp	2008-04-26 12:28:41.000000000 -0700
@@ -22,6 +22,8 @@
 
 #include <cppsocket/stringbuffer.h>
 
+#include <cstring>
+
 namespace CPPSocket
 {
   size_t StringBuffer::size() const throw()
