--- pl232src/src/autorange.c	2005-08-16 21:14:42.000000000 +0200
+++ patches/autorange.c	2006-02-17 18:17:04.000000000 +0100
@@ -400,7 +400,8 @@
 		DT_tomin( smax, &sec ); /* sec not used */
 		DT_gethms( &hour, &minute, &sec );
 		if( hoursblock ) hour = ((hour / hoursblock)+1) * hoursblock; 		/* scg 1/28/05 */
-		if( minute != 0 && sec != 0.0 ) hour++; 				/* scg 1/28/05 */
+		/* if( minute != 0 && sec != 0.0 ) hour++;  */				/* scg 1/28/05 */
+		if( minute != 0 || sec != 0.0 ) hour++; 				/* bug, scg 12/13/05 */
 		/* if( minute == 0 && sec == 0.0 ) */
 		 DT_maketime( hour, 0, 0.0, maxval ); 
 		/* else DT_maketime( hour+1, 0, 0.0, maxval ); */
--- pl232src/src/functions.c	2005-02-15 18:20:12.000000000 +0100
+++ patches/functions.c	2006-02-17 18:11:48.000000000 +0100
@@ -148,6 +148,7 @@
 	if( hash == 165 ) {     /* $def( varname ) - return 1 if variable has been defined, 0 if not */
 		if( TDH_getvalue( result, arg[0], TDH_dat, TDH_recid ) == 0 ) sprintf( result, "1" );
 		else sprintf( result, "0" );
+		*typ = NUMBER;
 		return( 0 );
 		}
 
@@ -474,6 +475,15 @@
 		return( 0 );
 		}
 
+	if( hash == 963 ) { /* $cleanname( s ) - remove everything but chars, digits, and underscores.. */
+		strcpy( result, "" );
+		for( i = 0, j = 0; arg[0][i] != '\0'; i++ ) {
+			if( isalnum( (int) arg[0][i] ) || arg[0][i] == '_' ) result[j++] = arg[0][i];
+			}
+		result[j] = '\0';
+		return( 0 );
+		}
+
 	if( hash == 992 ) goto EXT_DATE; /* $datevalid() */
 
 	if( hash == 997 ) goto EXT_SQL; /* $sqlerror() */
@@ -646,7 +656,7 @@
 
 	if( hash == 1708 ) { /* $htmldisplay( var1, .. varN ) - convert embedded HTML problem chars (<,>,&) 
 			      * to their html esc sequences for all variables given */
-		char *strchr();
+		/* char *strchr(); */
 		for( i = 0; i < nargs; i++ ) {
         		stat = TDH_getvalue( tok, arg[i], TDH_dat, TDH_recid );
 			if( strchr( tok, '&' ) != NULL ) GL_substitute( "&", "&amp;", tok );
--- pl232src/src/proc_scatterplot.c	2005-08-12 17:33:18.000000000 +0200
+++ patches/proc_scatterplot.c	2005-09-16 18:45:36.000000000 +0200
@@ -459,6 +459,7 @@
 			         /* sizefield scales up the AREA of symbol, not the diameter */
 		if( dorect ) {
 			char color[ COLORLEN ];
+			strcpy( color, "" ); /* added scg 9/1/05 - heatmap bug */
 			if( symfield >=0 || dupsleg ) sscanf( symbol, "%s", color ); /* strip off any trailing space */
 			Ecblock( x-rectw, y-recth, x+rectw, y+recth, color, rectoutline );
 			symbol[0] = '\0';
