--- auth/sql.c.orig	2005-03-08 14:24:11.000000000 -0800
+++ auth/sql.c	2005-05-13 19:12:38.000000000 -0700
@@ -74,7 +74,7 @@
   
   for (p = (const unsigned char *) ustr; *p; p++)
     {
-      if (strchr ("'\"", *p))
+      if (strchr ("'\"\\", *p))
 	len++;
     }
 
@@ -84,7 +84,7 @@
 
   for (p = (const unsigned char *) ustr, q = str; *p; p++)
     {
-      if (strchr ("'\"", *p))
+      if (strchr ("'\"\\", *p))
 	*q++ = '\\';
       *q++ = *p;
     }
