--- ./mailbox/getline.h.old	2005-01-01 16:39:25.619823831 +0100
+++ ./mailbox/getline.h	2005-01-01 16:28:47.798840941 +0100
@@ -18,6 +18,7 @@
 #ifndef _GETLINE_H_
 # define _GETLINE_H_ 1
 
+# include <sys/types.h>
 # include <stdio.h>
 
 # ifndef PARAMS
@@ -28,8 +29,8 @@
 #  endif
 # endif
 
-extern int getline PARAMS ((char **_lineptr, size_t *_n, FILE *_stream));
+extern ssize_t getline PARAMS ((char **_lineptr, size_t *_n, FILE *_stream));
 
-extern int getdelim PARAMS ((char **_lineptr, size_t *_n, int _delimiter, FILE *_stream));
+extern ssize_t getdelim PARAMS ((char **_lineptr, size_t *_n, int _delimiter, FILE *_stream));
 
 #endif /* ! _GETLINE_H_ */
