diff -ruN dermixd-1.1.0-beta2.orig/frontend/dermixd-control dermixd-1.1.0-beta2/frontend/dermixd-control
--- dermixd-1.1.0-beta2.orig/frontend/dermixd-control	2006-09-04 02:42:01.000000000 +0200
+++ dermixd-1.1.0-beta2/frontend/dermixd-control	2006-09-04 02:43:40.000000000 +0200
@@ -10,6 +10,7 @@
 use FindBin qw($Bin);
 use lib $Bin.'/../lib/perl';
 use lib $Bin.'/../lib/thorma/perl'; #we may operate withing ThOrMA
+use lib $Bin.'/../lib/dermixd'; #for standalone DerMixD install
 
 use Socket;
 use Param;
@@ -18,7 +19,8 @@
 my $param = &Param::Parse
 (
 	{ 'init', "Simple script for sending mixplayd a command... \n Usage: $0 [options] command+arg\nThe script tries to be smart at start/seek/pause commands when no channel is given..." }
-	,'remote','localhost','R','the (remote) server where mixplayd is running on'
+	,'remote','localhost','R','the (remote) server where dermixd is running on'
+	,'port','8888','p','the port to use'
 	,'verbose',0,'v','be more verbose about the server communication'
 	,'silent',0,'s','do not even give a note on success'
 	,'stay',0,'S','stay connected after commands have been issued (makes sense for scripting)'
@@ -33,7 +35,7 @@
 #Thus, returned undef on error... that was somehow not consistent with my different perl versions (5.6 vs 5.8)
 #so, now checking manually if there is a connection online
 
-my $pal = new DerMixD::Control({'remote',$param->{remote},'debug',$param->{verbose},'echo',$param->{echo}});
+my $pal = new DerMixD::Control({'port',$param->{port},'remote',$param->{remote},'debug',$param->{verbose},'echo',$param->{echo}});
 unless($pal->{online}){die "Not online ($!)\n";}
 
 local(*DAT);
diff -ruN dermixd-1.1.0-beta2.orig/frontend/simple_player dermixd-1.1.0-beta2/frontend/simple_player
--- dermixd-1.1.0-beta2.orig/frontend/simple_player	2006-09-04 02:42:01.000000000 +0200
+++ dermixd-1.1.0-beta2/frontend/simple_player	2006-09-04 02:43:40.000000000 +0200
@@ -11,6 +11,7 @@
 use lib $Bin.'/../lib/perl';
 use lib $Bin.'/../lib/thorma/perl'; #we may operate withing ThOrMA
 use lib $Bin; #for stand-alone package
+use lib $Bin.'/../lib/dermixd'; #for standalone DerMixD install
 
 #use Param;
 use DerMixD::Control;
