--- configure	2006-05-30 15:04:45.000000000 +0200
+++ configure	2006-05-30 15:05:35.000000000 +0200
@@ -30,22 +30,12 @@
 done
 
 echo -n "Checking for CPU type...  "
-if test -r "/proc/cpuinfo" ; then
-VENDO="`cat /proc/cpuinfo | grep 'vendor_id' | cut -d : -f2 | sed 's/ //g'`"
-if test "$VENDO" == "GenuineIntel" -o "$VENDO" == "AuthenticAMD" -o "$VENDO" == "CyrixInstead" \
- -o "$VENDO" == "UMC UMC UMC " -o "$VENDO" == "CentaurHauls" -o "$VENDO" == "NexGenDriven" \
- -o "$VENDO" == "RiseRiseRise" -o "$VENDO" == "GenuineTMx86" -o "$VENDO" == "TransmetaCPU"; then
-	FAMILY="`cat /proc/cpuinfo | grep 'cpu family' | cut -d : -f 2 | sed 's/ //g'`86"
-	echo "$FAMILY or clone."
+if [[ MYARCH == ia32 ]]; then
+  echo "x86 or clone."
 else
 	echo "Sorry. This library works only on Intel x86 and clone CPUs."
 	exit 1
 fi
-else
-	echo "error."
-	echo "Sorry, can't determine current your CPU type. Assuming CPU type to 386"
-fi
-
 
 echo -n "Checking for nasm...  "
 if test -z "`which nasm`"; then
