diff -uNr pyside-tools-0.1.2/pylupdate/CMakeLists.txt pyside-tools-0.1.2.cp/pylupdate/CMakeLists.txt
--- pyside-tools-0.1.2/pylupdate/CMakeLists.txt	2009-10-05 16:33:52.000000000 +0200
+++ pyside-tools-0.1.2.cp/pylupdate/CMakeLists.txt	2009-10-06 12:10:47.000000000 +0200
@@ -19,6 +19,7 @@
                     ${CMAKE_CURRENT_SOURCE_DIR}
                     ${QT_INCLUDE_DIR}
                     ${QT_QT_INCLUDE_DIR}
+                    ${QT_QTGUI_INCLUDE_DIR}
                     ${QT_QTXML_INCLUDE_DIR}
                     ${QT_QTCORE_INCLUDE_DIR})
 
