--- setup.py.orig	2009-08-12 00:22:38.000000000 +0400
+++ setup.py	2009-08-22 17:41:04.416503647 +0400
@@ -17,7 +17,7 @@
 setup(name="pyskein",
       version="0.3.1",
       description="Efficiently compute Skein hashes",
-      author="Hagen Fürstenau",
+      author="Hagen Furstenau",
       author_email="hfuerstenau@gmx.net",
       license="GPL",
       url=HOMEPAGE,
--- scripts/skeinsum.orig	2009-08-12 00:22:38.000000000 +0400
+++ scripts/skeinsum	2009-08-22 17:38:26.354503659 +0400
@@ -1,7 +1,7 @@
 #!/usr/local/bin/python3
 
 #   skeinsum.py
-#   Copyright 2008, 2009 Hagen Fürstenau <hfuerstenau@gmx.net>
+#   Copyright 2008, 2009 Hagen Furstenau <hfuerstenau@gmx.net>
 #
 #   Demonstrates Skein hashing with PySkein.
 #
--- scripts/threefish.orig	2009-08-12 00:22:38.000000000 +0400
+++ scripts/threefish	2009-08-22 17:38:20.602503287 +0400
@@ -1,7 +1,7 @@
 #!/usr/local/bin/python3
 
 #   threefish.py
-#   Copyright 2008, 2009 Hagen Fürstenau <hfuerstenau@gmx.net>
+#   Copyright 2008, 2009 Hagen Furstenau <hfuerstenau@gmx.net>
 #
 #   Demonstrates Threefish encryption with PySkein.
 #
