From 7e2e8428e5998a5ea54ed43f81e13a1f2fa92fd1 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 6 Aug 2018 22:31:39 +0200
Subject: [PATCH] fix linking with glibc 2.28

---
 src/fuser.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/fuser.c b/src/fuser.c
index b485f65..997bbcc 100644
--- a/src/fuser.c
+++ b/src/fuser.c
@@ -49,6 +49,8 @@
 #include <getopt.h>
 #include <setjmp.h>
 #include <limits.h>
+#include <sys/sysmacros.h>
+
 /* MAXSYMLINKS is a BSDism.  If it doesn't exist, fall back to SYMLINK_MAX,
    which is the POSIX name. */
 #ifndef MAXSYMLINKS
-- 
2.18.0

