--- ./src/Satellite.cpp.orig	2010-05-10 16:48:58.477891720 +0200
+++ ./src/Satellite.cpp	2010-05-10 16:50:14.099065314 +0200
@@ -1,6 +1,7 @@
 #include <clocale>
 #include <cstring>
 #include <iostream>
+#include <cstdio>
 using namespace std;
 
 #include <sys/time.h>
--- ./src/Separation.cpp.orig	2010-05-10 16:51:19.090375809 +0200
+++ ./src/Separation.cpp	2010-05-10 16:51:42.029875358 +0200
@@ -1,5 +1,6 @@
 #include <cmath>
 #include <iostream>
+#include <cstdio>
 
 #include "Separation.h"
 #include "View.h"
