diff -Naupr xplanet-1.2.0/src/buildPlanetMap.cpp xplanet-1.2.0.fixed/src/buildPlanetMap.cpp
--- xplanet-1.2.0/src/buildPlanetMap.cpp	2004-04-20 23:47:21.000000000 +0200
+++ xplanet-1.2.0.fixed/src/buildPlanetMap.cpp	2008-05-18 19:09:26.674392256 +0200
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include <cmath>
 #include <cstdio>
 #include <map>
diff -Naupr xplanet-1.2.0/src/libannotate/Icon.cpp xplanet-1.2.0.fixed/src/libannotate/Icon.cpp
--- xplanet-1.2.0/src/libannotate/Icon.cpp	2004-02-16 19:36:42.000000000 +0100
+++ xplanet-1.2.0.fixed/src/libannotate/Icon.cpp	2008-05-18 19:03:53.104274715 +0200
@@ -8,6 +8,7 @@ using namespace std;
 
 #include "libdisplay/libdisplay.h"
 #include "libimage/Image.h"
+#include <cstring>
 
 Icon::Icon(const int x, const int y, const std::string &filename,
            const unsigned char *transparent)
diff -Naupr xplanet-1.2.0/src/libplanet/Planet.cpp xplanet-1.2.0.fixed/src/libplanet/Planet.cpp
--- xplanet-1.2.0/src/libplanet/Planet.cpp	2005-06-14 20:04:05.000000000 +0200
+++ xplanet-1.2.0.fixed/src/libplanet/Planet.cpp	2008-05-18 19:05:12.344297804 +0200
@@ -1,3 +1,4 @@
+#include <cstring>
 #include <cmath>
 #include <cstdio>
 #include <cctype>
diff -Naupr xplanet-1.2.0/src/libprojection/getProjection.cpp xplanet-1.2.0.fixed/src/libprojection/getProjection.cpp
--- xplanet-1.2.0/src/libprojection/getProjection.cpp	2005-06-13 17:13:11.000000000 +0200
+++ xplanet-1.2.0.fixed/src/libprojection/getProjection.cpp	2008-05-18 19:07:42.291085791 +0200
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include <cctype>
 #include <cstring>
 #include <sstream>
diff -Naupr xplanet-1.2.0/src/Map.cpp xplanet-1.2.0.fixed/src/Map.cpp
--- xplanet-1.2.0/src/Map.cpp	2005-06-15 01:40:46.000000000 +0200
+++ xplanet-1.2.0.fixed/src/Map.cpp	2008-05-18 19:08:50.101035754 +0200
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include <map>
 #include <sstream>
 #include <string>
diff -Naupr xplanet-1.2.0/src/parseColor.cpp xplanet-1.2.0.fixed/src/parseColor.cpp
--- xplanet-1.2.0/src/parseColor.cpp	2004-11-23 18:49:48.000000000 +0100
+++ xplanet-1.2.0.fixed/src/parseColor.cpp	2008-05-18 19:10:34.168627069 +0200
@@ -1,3 +1,5 @@
+#include <cstdlib>
+#include <cstring>
 #include <cstdio>
 #include <fstream>
 #include <map>
diff -Naupr xplanet-1.2.0/src/PlanetProperties.h xplanet-1.2.0.fixed/src/PlanetProperties.h
--- xplanet-1.2.0/src/PlanetProperties.h	2005-04-02 02:43:06.000000000 +0200
+++ xplanet-1.2.0.fixed/src/PlanetProperties.h	2008-05-18 19:03:14.124299203 +0200
@@ -3,6 +3,7 @@
 
 #include <string>
 #include <vector>
+#include <cstring>
 
 #include "body.h"
 
diff -Naupr xplanet-1.2.0/src/xplanet.cpp xplanet-1.2.0.fixed/src/xplanet.cpp
--- xplanet-1.2.0/src/xplanet.cpp	2004-09-09 17:46:05.000000000 +0200
+++ xplanet-1.2.0.fixed/src/xplanet.cpp	2008-05-18 19:11:54.607880344 +0200
@@ -1,3 +1,4 @@
+#include <cstdlib>
 #include <clocale>
 #include <cstdio>
 #include <iostream>
diff -Naupr xplanet-1.2.0/src/xpUtil.cpp xplanet-1.2.0.fixed/src/xpUtil.cpp
--- xplanet-1.2.0/src/xpUtil.cpp	2005-04-02 02:22:51.000000000 +0200
+++ xplanet-1.2.0.fixed/src/xpUtil.cpp	2008-05-18 19:11:31.280905627 +0200
@@ -1,3 +1,5 @@
+#include <cstdlib>
+#include <cstring>
 #include <cerrno>
 #include <clocale>
 #include <cmath>
