--- render.h	2007-09-01 00:42:38.000000000 +0300
+++ render.h.new	2007-09-22 02:02:10.000000000 +0300
@@ -26,6 +26,27 @@
 #ifndef _RENDER_H_
 #define _RENDER_H_
 
+#ifndef _XTYPEDEF_XID
+#  define _XTYPEDEF_XID
+#  ifndef _XSERVER64
+typedef unsigned long XID;
+#  else
+#    if defined (_LP64) || \
+        defined(__alpha) || defined(__alpha__) || \
+        defined(__ia64__) || defined(ia64) || \
+        defined(__sparc64__) || \
+        defined(__s390x__) || \
+        (defined(__hppa__) && defined(__LP64__)) || \
+        defined(__amd64__) || defined(amd64) || \
+        defined(__powerpc64__) || \
+        (defined(sgi) && (_MIPS_SZLONG == 64))
+typedef unsigned int XID;
+#    else
+typedef unsigned long XID;
+#    endif
+#  endif
+#endif
+
 typedef XID		Glyph;
 typedef XID		GlyphSet;
 typedef XID		Picture;
