From a9f40400fd833e5ff2e7048614e48c6081eb53c9 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 10 Jun 2017 02:06:52 +0200
Subject: [PATCH] disable building manpages

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 98215a0..4044b66 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -87,7 +87,7 @@ PRE_UNINSTALL = :
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@HAVE_UNIX_TRUE@am__append_1 = man help
+@HAVE_UNIX_TRUE@am__append_1 = help
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/as-linguas.m4 \
-- 
2.13.1

