From a4d2aa2467ba081b77312e39fbf24df453003bf0 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 24 Oct 2020 16:13:39 +0200
Subject: [PATCH] fix make detection

---
 configure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 63ce1b9..4c7f597 100755
--- a/configure
+++ b/configure
@@ -18,7 +18,7 @@ rm -f config.mk builds/unix/unix-def.mk builds/unix/unix-cc.mk
 # respect GNUMAKE environment variable for backward compatibility
 if test "x$GNUMAKE" = x; then
   if test "x$MAKE" = x; then
-    if test "x`make -v 2>/dev/null | egrep 'GNU|makepp'`" = x; then
+    if test "x`make -v 2>/dev/null | grep -E 'GNU|makepp'`" = x; then
       MAKE=gmake
     else
       MAKE=make
@@ -30,7 +30,7 @@ fi
 
 echo $MAKE
 
-if test "x`$MAKE -v 2>/dev/null | egrep 'GNU|makepp'`" = x; then
+if test "x`$MAKE -v 2>/dev/null | grep -E 'GNU|makepp'`" = x; then
   echo "GNU make (>= 3.81) or makepp (>= 2.0) is required to build FreeType2." >&2
   echo "Please try" >&2
   echo >&2
-- 
2.29.0

