diff -Naur freetype-2.1.9.orig/include/freetype/freetype.h freetype-2.1.9/include/freetype/freetype.h
--- freetype-2.1.9.orig/include/freetype/freetype.h	2005-01-02 21:02:48.023840904 -0800
+++ freetype-2.1.9/include/freetype/freetype.h	2005-01-02 21:03:40.682835520 -0800
@@ -17,11 +17,13 @@
 
 
 #ifndef FT_FREETYPE_H
-#error "`ft2build.h' hasn't been included yet!"
-#error "Please always use macros to include FreeType header files."
-#error "Example:"
-#error "  #include <ft2build.h>"
-#error "  #include FT_FREETYPE_H"
+#include <ft2build.h>
+#warning "`ft2build.h' hasn't been included yet!"
+#warning "Please always use macros to include FreeType header files."
+#warning "Example:"
+#warning "  #include <ft2build.h>"
+#warning "  #include FT_FREETYPE_H"
+#warning "Source Mage has included this for you, but please fix it"
 #endif
 
 
