From f035a20a78737a680613f13667d006bf30921b54 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 5 Jun 2017 16:24:03 +0200
Subject: [PATCH] fix compile error with gcc 7

---
 dcraw.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dcraw.cc b/dcraw.cc
index 88c8068..cc865ef 100644
--- a/dcraw.cc
+++ b/dcraw.cc
@@ -9242,7 +9242,7 @@ canon_a5:
     if (make[0] == 'O') {
       i = find_green (12, 32, 1188864, 3576832);
       c = find_green (12, 32, 2383920, 2387016);
-      if (abs(i) < abs(c)) {
+      if (fabs(i) < fabs(c)) {
 	SWAP(i,c);
 	load_flags = 24;
       }
-- 
2.12.0

