--- lib/TagLib/TagLib/src/TagLib/IFD/Entries/Rational.cs.orig	2010-11-20 19:50:45.520002431 -0600
+++ lib/TagLib/TagLib/src/TagLib/IFD/Entries/Rational.cs	2010-11-20 19:51:47.440002796 -0600
@@ -62,8 +62,10 @@
 		/// </param>
 		public Rational (uint numerator, uint denominator)
 		{
-			Numerator = numerator;
-			Denominator = denominator;
+			if (denominator == 0)
+				throw new ArgumentException ("denominator");
+			this.numerator = numerator;
+			this.denominator = denominator;
 		}
 
 #endregion
--- lib/TagLib/TagLib/src/TagLib/IFD/Entries/SRational.cs.orig	2010-11-20 19:52:11.384002643 -0600
+++ lib/TagLib/TagLib/src/TagLib/IFD/Entries/SRational.cs	2010-11-20 19:52:43.948002767 -0600
@@ -62,8 +62,10 @@
 		/// </param>
 		public SRational (int numerator, int denominator)
 		{
-			Numerator = numerator;
-			Denominator = denominator;
+			if (denominator == 0)
+				throw new ArgumentException ("denominator");
+			this.numerator = numerator;
+			this.denominator = denominator;
 		}
 
 #endregion
