diff -urpN iproute2-2.6.8.orig/doc/Makefile iproute2-2.6.8/doc/Makefile
--- iproute2-2.6.8.orig/doc/Makefile	2004-08-20 07:15:45.300846296 -0700
+++ iproute2-2.6.8/doc/Makefile	2004-08-20 07:20:21.027929408 -0700
@@ -12,6 +12,7 @@ LPR=lpr -Zsduplex
 SHELL=bash
 PAGESIZE=a4
 PAGESPERPAGE=2
+DOCDIR=/usr/share/doc/iproute2
 
 HTMLFILES=$(subst .sgml,.html,$(shell echo *.sgml))
 DVIFILES=$(subst .ps,.dvi,$(PSFILES))
