diff -Naur iproute2-2.6.11.orig/ip/iptunnel.c iproute2-2.6.11/ip/iptunnel.c
--- iproute2-2.6.11.orig/ip/iptunnel.c	2005-04-07 10:55:29.690001184 -0700
+++ iproute2-2.6.11/ip/iptunnel.c	2005-04-07 10:55:38.098722864 -0700
@@ -35,6 +35,9 @@
 #endif
 
 #include <linux/if_tunnel.h>
+#include <endian.h>
+#include <asm/byteorder.h>
+#include <linux/byteorder/swab.h>
 
 #include "rt_names.h"
 #include "utils.h"
