diff -Naur iproute2-2.6.22-070710.orig/ip/iptunnel.c iproute2-2.6.22-070710/ip/iptunnel.c
--- iproute2-2.6.22-070710.orig/ip/iptunnel.c	2007-07-11 08:22:31.526850990 -0700
+++ iproute2-2.6.22-070710/ip/iptunnel.c	2007-07-11 08:22:44.024944328 -0700
@@ -28,6 +28,9 @@
 #include <linux/if_arp.h>
 #include <linux/ip.h>
 #include <linux/if_tunnel.h>
+#include <endian.h>
+#include <asm/byteorder.h>
+#include <linux/byteorder/swab.h>
 
 #include "rt_names.h"
 #include "utils.h"
