diff -Nuar iproute2-2.6.22-070710.orig/misc/Makefile iproute2-2.6.22-070710/misc/Makefile
--- iproute2-2.6.22-070710.orig/misc/Makefile	2007-07-11 06:34:14.000000000 +0500
+++ iproute2-2.6.22-070710/misc/Makefile	2007-07-25 23:55:37.000000000 +0500
@@ -1,7 +1,7 @@
 SSOBJ=ss.o ssfilter.o
 LNSTATOBJ=lnstat.o lnstat_util.o
 
-TARGETS=ss nstat ifstat rtacct arpd lnstat
+TARGETS=ss nstat ifstat rtacct lnstat
 
 include ../Config
 
@@ -18,9 +18,6 @@
 rtacct: rtacct.c
 	$(CC) $(CFLAGS) $(LDFLAGS) -o rtacct rtacct.c $(LIBNETLINK) -lm
 
-arpd: arpd.c
-	$(CC) $(CFLAGS) -I$(DBM_INCLUDE) $(LDFLAGS) -o arpd arpd.c $(LIBNETLINK) -ldb -lpthread
-
 ssfilter.c: ssfilter.y
 	bison ssfilter.y -o ssfilter.c
