From: Ismael Luceno <ismael@sourcemage.org>
Subject: Override some feature detection to build correctly

--
diff -ur aplus-fsf-4.22.orig/src/dap/Makefile.in aplus-fsf-4.22/src/dap/Makefile.in
--- aplus-fsf-4.22.orig/src/dap/Makefile.in	2008-04-08 17:15:28.000000000 -0300
+++ aplus-fsf-4.22/src/dap/Makefile.in	2015-03-04 22:09:07.973528149 -0300
@@ -366,7 +366,7 @@

 DEFS = @DEFS@
 DEFAULT_INCLUDES =  -I. -I$(srcdir)
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS = @CPPFLAGS@ -DHAVE_SVR4
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
 depcomp = $(SHELL) $(top_srcdir)/config/depcomp
diff -ur aplus-fsf-4.22.orig/src/MSTypes/Makefile.in aplus-fsf-4.22/src/MSTypes/Makefile.in
--- aplus-fsf-4.22.orig/src/MSTypes/Makefile.in	2008-04-08 17:15:26.000000000 -0300
+++ aplus-fsf-4.22/src/MSTypes/Makefile.in	2015-03-06 11:56:28.061053083 -0300
@@ -423,7 +423,7 @@

 DEFS = @DEFS@
 DEFAULT_INCLUDES =  -I. -I$(srcdir)
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS = @CPPFLAGS@ -DMS_OFFSETOF_IN_STDDEF
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
 depcomp = $(SHELL) $(top_srcdir)/config/depcomp
