From: Ismael Luceno <ismael@sourcemage.org>
Subject: Add missing includes

--
--- aplus-fsf-4.22/src/a/development.h.orig
+++ aplus-fsf-4.22/src/a/development.h
@@ -30,6 +30,8 @@
 #    endif
 #  endif

+#include <string.h>
+
 #  define bcopy(s1,s2,len) 	memmove(s2,s1,len)
 #  define bcmp(s1,s2,len)  	(memcmp(s2,s1,len)==0?0:1)
 #  define bzero(sp, len)   	memset(sp, 0, len)
--- aplus-fsf-4.22/src/dap/connopen.c.orig	2015-03-06 14:10:46.110262033 -0300
+++ aplus-fsf-4.22/src/dap/connopen.c	2015-03-06 14:11:00.748262413 -0300
@@ -10,7 +10,6 @@
 /* header file inclusions */
 #include <sys/types.h>
 #ifdef HAVE_SVR4
-#include  <sys/filio.h>
 #endif
 #include <sys/uio.h>
 #include <dap/notsunos4.h>
--- aplus-fsf-4.22/src/dap/lstnopen.c.orig	2015-03-06 14:36:26.060301965 -0300
+++ aplus-fsf-4.22/src/dap/lstnopen.c	2015-03-06 14:36:33.945302169 -0300
@@ -10,7 +10,6 @@
 /* header file inclusions */
 #include <sys/types.h>
 #if defined(HAVE_SVR4)
-#include <sys/filio.h>
 #endif
 #include <sys/uio.h>
 #include <dap/notsunos4.h>
