Fix bug in behavoir of tail modifier.

Upstream-Status: Backport [https://mailman.astron.com/pipermail/tcsh/2020-November/000092.html]

diff --git a/sh.lex.c b/sh.lex.c
index fdd24bc..9df132d 100644
--- a/sh.lex.c
+++ b/sh.lex.c
@@ -1046,7 +1046,7 @@ domod(Char *cp, Char type)
     case 't':
 	wp = Strrchr(cp, '/');
 	if (wp == NULL)
-	    return Strsave(type == 't' ? cp : STRNULL);
+	    return type == 't' ? Strsave(cp) : NULL;
 	if (type == 't')
 	    xp = Strsave(wp + 1);
 	else
