--- config.mk.orig	2016-10-31 12:24:54.000000000 -0400
+++ config.mk	2017-10-08 08:44:52.918989680 -0400
@@ -10,11 +10,6 @@
 INCS = -I. -I/usr/include
 LIBS = -L/usr/lib
 
-# libnotify, comment if you don't want it
-DEFS = -DNOTIFY
-INCS+= `pkg-config --cflags libnotify`
-LIBS+= `pkg-config --libs libnotify`
-
 # flags
 CPPFLAGS = -DVERSION=\"${VERSION}\" -D_POSIX_C_SOURCE=199309
 CFLAGS += -g -std=c99 -pedantic -Wall -Os ${INCS} ${DEFS} ${CPPFLAGS}
