--- tools/ioemu-qemu-xen/vl.c.orig	2012-04-24 21:35:40.000000000 +0400
+++ tools/ioemu-qemu-xen/vl.c	2012-09-19 14:44:19.190522811 +0400
@@ -4910,7 +4910,7 @@
     kernel_cmdline = "";
     cyls = heads = secs = 0;
     translation = BIOS_ATA_TRANSLATION_AUTO;
-    monitor_device = "vc:80Cx24C";
+    monitor_device = "null";
 
     serial_devices[0] = "vc:80Cx24C";
     for(i = 1; i < MAX_SERIAL_PORTS; i++)
