--- Makefile.am.orig	2005-04-24 10:27:46.000000000 +0200
+++ Makefile.am	2005-04-24 10:28:01.000000000 +0200
@@ -32,19 +32,6 @@
 	  chmod +x $(DESTDIR)$(scriptdir)/$$prog; \
 	done
 
-## Make versioned links.  We only run the transform on the root name;
-## then we make a versioned link with the transformed base name.  This
-## seemed like the most reasonable approach.
-install-exec-hook:
-	@$(POST_INSTALL)
-	@for p in $(bin_SCRIPTS); do \
-	  f="`echo $$p|sed '$(transform)'`"; \
-	  fv="$$f-$(APIVERSION)"; \
-	  rm -f $(DESTDIR)$(bindir)/$$fv; \
-	  echo " $(LN) $(DESTDIR)$(bindir)/$$f $(DESTDIR)$(bindir)/$$fv"; \
-	  $(LN) $(DESTDIR)$(bindir)/$$f $(DESTDIR)$(bindir)/$$fv; \
-	done
-
 uninstall-hook:
 	@for p in $(bin_SCRIPTS); do \
 	  f="`echo $$p|sed '$(transform)'`"; \
