From b51b3786f358fdffbabbb2b5ad9aa94d39090a56 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 2 Nov 2018 14:21:12 +0100
Subject: [PATCH] fix compatibility with basement 0.0.8

---
 foundation.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/foundation.cabal b/foundation.cabal
index 1011bd0..55caaa3 100644
--- a/foundation.cabal
+++ b/foundation.cabal
@@ -197,7 +197,7 @@ library
                       BangPatterns
                       DeriveDataTypeable
   build-depends:     base >= 4.7 && < 5
-                   , basement == 0.0.7
+                   , basement
                    , ghc-prim
   -- FIXME add suport for armel mipsel
   --  CPP-options: -DARCH_IS_LITTLE_ENDIAN
-- 
2.19.1

