diff --git a/lib/tevent/tevent.mk b/lib/tevent/tevent.mk
index ac5710f..53aeec3 100644
--- a/lib/tevent/tevent.mk
+++ b/lib/tevent/tevent.mk
@@ -21,7 +21,7 @@ installdirs::
 	mkdir -p $(DESTDIR)$(libdir)/pkgconfig
 
 installheaders:: installdirs
-	cp $(srcdir)/tevent.h $(DESTDIR)$(includedir)
+	cp $(srcdir)/tevent.h $(srcdir)/tevent_util.h $(DESTDIR)$(includedir)
 
 installlibs:: installdirs
 	cp tevent.pc $(DESTDIR)$(libdir)/pkgconfig
