From abb33a3e46d4038cb386d7e123440a377772eeac Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Tue, 30 Aug 2016 11:19:07 +0200
Subject: [PATCH] use LDFLAGS when linking

---
 Makefile.linux | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Makefile.linux b/Makefile.linux
index 020336b..c383bf1 100755
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -1,7 +1,5 @@
-CXXFLAGS=-Wall -O2
-
 mp3val: main.o mpegparse.o out.o report.o crossapi.o crc.o
-	$(CXX) $(CXXFLAGS) $^ -o $@
+	$(CXX) $(LDFLAGS) -o $@ $^ $(LOADLIBES) $(LDLIBS)
 
 clean:
 	rm -f mp3val
-- 
2.9.3

