diff -uNr bisonc++-1.4.0/icmake/install bisonc++-1.4.0.cp/icmake/install
--- bisonc++-1.4.0/icmake/install	2006-08-25 20:58:23.000000000 +0200
+++ bisonc++-1.4.0.cp/icmake/install	2006-09-22 11:49:49.000000000 +0200
@@ -30,8 +30,8 @@
     run("cp -r documentation/man/calculator " + where + DOC + "/examples");
     run("cp -r documentation/regression " + where + DOC);
 
-    printf("  installing manual\n");
-    run("cp -r tmp/manual " + where + DOC);
+//    printf("  installing manual\n");
+//    run("cp -r tmp/manual " + where + DOC);
 
     printf("  Installation completed\n");
 
