diff -uNr aspell-0.60.4/modules/filter/nroff.cpp aspell-0.60.4.cp/modules/filter/nroff.cpp
--- aspell-0.60.4/modules/filter/nroff.cpp	2005-02-20 22:49:18.000000000 +0100
+++ aspell-0.60.4.cp/modules/filter/nroff.cpp	2006-04-10 18:56:04.000000000 +0200
@@ -73,7 +73,7 @@
       return false;
     }
     
-    bool NroffFilter::process_char (FilterChar::Chr c);
+    bool process_char (FilterChar::Chr c);
       
   public:
 
