--- ./src/gtk/misc-gtk.c~	2007-04-27 01:47:33.000000000 +0200
+++ ./src/gtk/misc-gtk.c	2008-12-21 19:31:37.000000000 +0100
@@ -757,7 +757,8 @@
     tempstr = insert_commas (fle->size, NULL, 0);
 
   gtk_clist_set_text (GTK_CLIST (wdata->listbox), clist_num, 2, tempstr);
-  g_free (tempstr);
+  if (tempstr)
+      g_free (tempstr);
 
   if (fle->user)
     gtk_clist_set_text (GTK_CLIST (wdata->listbox), clist_num, 3, fle->user);
