diff -ur piave-0.2.4-orig/plugins/iostream/sndfile/sndfile.cc piave-0.2.4/plugins/iostream/sndfile/sndfile.cc
--- piave-0.2.4-orig/plugins/iostream/sndfile/sndfile.cc	2003-11-28 06:31:38.000000000 +0100
+++ piave-0.2.4/plugins/iostream/sndfile/sndfile.cc	2004-05-11 17:20:39.000000000 +0200
@@ -44,6 +44,7 @@
 using std::endl;
 #include <string>
 using std::string;
+#include <cmath>
 
 // sndfile API
 #include <sndfile.h>
diff -ur piave-0.2.4-orig/plugins/iostream/vorbisfile/vorbisfile.cc piave-0.2.4/plugins/iostream/vorbisfile/vorbisfile.cc
--- piave-0.2.4-orig/plugins/iostream/vorbisfile/vorbisfile.cc	2003-09-15 12:52:25.000000000 +0200
+++ piave-0.2.4/plugins/iostream/vorbisfile/vorbisfile.cc	2004-05-11 17:18:56.000000000 +0200
@@ -44,6 +44,7 @@
 using std::endl;
 #include <string>
 using std::string;
+#include <cmath>
 
 // this header
 #include "vorbisfile.hh"
