--- ext/syck/rubyext.c~	2005-12-20 05:13:26.000000000 +0100
+++ ext/syck/rubyext.c	2006-03-13 19:37:59.000000000 +0100
@@ -1157,6 +1157,9 @@
             }
             else if ( !NIL_P( target_class ) )
             {
+                if (subclass == rb_cBignum)
+                    obj = rb_str2inum(val, 10);
+                else
                 obj = rb_obj_alloc( subclass );
                 if ( rb_respond_to( obj, s_yaml_initialize ) )
                 {
