--- CMakeLists.txt.orig	2013-10-09 16:21:02.000000000 +0400
+++ CMakeLists.txt	2013-10-09 16:28:30.210499682 +0400
@@ -221,7 +221,9 @@
 ####### install
 # slim
 install(TARGETS slim RUNTIME DESTINATION bin)
-install(TARGETS slimlock RUNTIME DESTINATION bin)
+if(BUILD_SLIMLOCK)
+    install(TARGETS slimlock RUNTIME DESTINATION bin)
+endif(BUILD_SLIMLOCK)
 
 if (BUILD_SHARED_LIBS)
 	set_target_properties(libslim PROPERTIES
