From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: psykose <alice@ayaya.dev>
Date: Wed, 19 Apr 2023 18:45:01 +0000
Subject: [PATCH] Fix werror=format-security

Upstream-Status: Unknown
Origin: Alpine Linux
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 main.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/main.c b/main.c
index 664acac..d568fb1 100644
--- a/main.c
+++ b/main.c
@@ -588,10 +588,10 @@ static void callback(long inpos, int function){
 	    buffer[aheadposition+19]='>';
 	}
    
-	fprintf(stderr,buffer);
+	fprintf(stderr, "%s", buffer);
        
 	if (logfile != NULL && function==-1) {
-	  fprintf(logfile,buffer+1);
+	  fprintf(logfile, "%s", buffer+1);
 	  fprintf(logfile,"\n\n");
 	  fflush(logfile);
 	}
-- 
Fixed up by sm-checkpatch 0.1

