diff --git a/Exif/Info.cpp b/Exif/Info.cpp
index be4c88f3..b6aba88a 100644
--- a/Exif/Info.cpp
+++ b/Exif/Info.cpp
@@ -28,7 +28,6 @@
 #include <QFile>
 
 #include <exiv2/image.hpp>
-#include <exiv2/exif.hpp>
 
 using namespace Exif;
 
diff --git a/Exif/Info.h b/Exif/Info.h
index 7e6a28c9..c0b27819 100644
--- a/Exif/Info.h
+++ b/Exif/Info.h
@@ -21,6 +21,7 @@
 #include <qstringlist.h>
 #include "Utilities/StringSet.h"
 #include <exiv2/exif.hpp>
+#include <exiv2/exiv2.hpp>
 #include <exiv2/iptc.hpp>
 
 namespace DB { class FileName; }
-- 
2.20.1
