From c520f8fc524ecc58844fe3a365bc5a96582c08bb Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Mon, 4 Jan 2016 16:48:33 +0100
Subject: [PATCH] fix compile error

---
 src/uxa/intel_driver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/uxa/intel_driver.c b/src/uxa/intel_driver.c
index 2793da5..f936b18 100644
--- a/src/uxa/intel_driver.c
+++ b/src/uxa/intel_driver.c
@@ -642,7 +642,7 @@ redisplay_dirty(ScreenPtr screen, PixmapDirtyUpdatePtr dirty)
 		return;
 
 	PixmapRegionInit(&pixregion, dirty->slave_dst->master_pixmap);
-	PixmapSyncDirtyHelper(dirty, &pixregion);
+	PixmapSyncDirtyHelper(dirty);
 	RegionUninit(&pixregion);
 
         intel_flush(intel);
-- 
2.6.3

