From 4f23903d085bf05ca395ec3934b22be65bb5ad61 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 9 Feb 2020 18:41:24 +0100
Subject: [PATCH] use guile-config1

---
 aclocal.m4 | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/aclocal.m4 b/aclocal.m4
index 0a96f0c..2d933e1 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -679,8 +679,8 @@ AC_DEFUN([STEPMAKE_GUILE_FLAGS], [
 # If missing, add entry to missing-list ($1, one of 'OPTIONAL', 'REQUIRED')
 AC_DEFUN(STEPMAKE_GUILE_DEVEL, [
     ## First, let's just see if we can find Guile at all.
-    test -n "$target_alias" && target_guile_config=$target_alias-guile-config
-    test -n "$host_alias" && host_guile_config=$host_alias-guile-config
+    test -n "$target_alias" && target_guile_config=$target_alias-guile-config1
+    test -n "$host_alias" && host_guile_config=$host_alias-guile-config1
     AC_MSG_CHECKING([for guile-config])
     guile_config="guile-config"
     found="no"
@@ -693,7 +693,7 @@ AC_DEFUN(STEPMAKE_GUILE_DEVEL, [
              guile1-config   guile-1-config   guile-config-1   guile-config1 \
              guile1.9-config guile-1.9-config guile-config-1.9 guile-config1.9 \
              guile1.8-config guile-1.8-config guile-config-1.8 guile-config1.8 \
-             guile18-config; do
+             guile18-config guild-config1; do
         exe=`STEPMAKE_GET_EXECUTABLE($r)`
         if ! $exe --version > /dev/null 2>&1 ; then
             continue
-- 
2.25.0

