From 904260bb4d1bfe9a8815b6b992b2c2eeb7713e4c Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 24 May 2017 09:50:00 +0200
Subject: [PATCH] use guile-config1

---
 aclocal.m4 | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/aclocal.m4 b/aclocal.m4
index b15111d..2854d52 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -653,8 +653,8 @@ AC_DEFUN([STEPMAKE_GUILE_FLAGS], [
 # If missing, add entry to missing-list ($1, one of 'OPTIONAL', 'REQUIRED')
 AC_DEFUN(STEPMAKE_GUILE_DEVEL, [
     ## First, let's just see if we can find Guile at all.
-    test -n "$target_alias" && target_guile_config=$target_alias-guile-config
-    test -n "$host_alias" && host_guile_config=$host_alias-guile-config
+    test -n "$target_alias" && target_guile_config=$target_alias-guile-config1
+    test -n "$host_alias" && host_guile_config=$host_alias-guile-config1
     AC_MSG_CHECKING([for guile-config])
     guile_config="guile-config"
     found="no"
@@ -666,7 +666,7 @@ AC_DEFUN(STEPMAKE_GUILE_DEVEL, [
       guile2.0-config guile-2.0-config guile-config-2.0 \
       guile1-config   guile-1-config   guile-config-1   \
       guile1.9-config guile-1.9-config guile-config-1.9 \
-      guile1.8-config guile-1.8-config guile-config-1.8; \
+      guile1.8-config guile-1-config guile-1.9-config guile-1.8-config guile-config1; \
       do
 	exe=`STEPMAKE_GET_EXECUTABLE($r)`
 	if ! $exe --version > /dev/null 2>&1 ; then
-- 
2.12.0

