diff -Nur ./lilypond-2.2.6.old/mf/GNUmakefile ./lilypond-2.2.6/mf/GNUmakefile
--- ./lilypond-2.2.6.old/mf/GNUmakefile	2005-02-10 21:02:42.367140178 -0600
+++ ./lilypond-2.2.6/mf/GNUmakefile	2005-02-10 21:03:51.116065291 -0600
@@ -197,5 +197,5 @@
 	$(foreach i, $@, touch $i && ) true
 
 $(outdir)/%.pfa: $(outdir)/%.bla
-	$(MFTRACE) -I $(outdir)/ --pfa --simplify --keep-trying $(notdir $(basename $@))  && mv $(notdir $@) $(outdir)/
+	$(MFTRACE) -I $(outdir)/ --simplify --keep-trying $(notdir $(basename $@))  && mv $(notdir $@) $(outdir)/
 
diff -Nur ./lilypond-2.2.6.old/stepmake/stepmake/metafont-rules.make ./lilypond-2.2.6/stepmake/stepmake/metafont-rules.make
--- ./lilypond-2.2.6.old/stepmake/stepmake/metafont-rules.make	2005-02-10 21:02:42.495115824 -0600
+++ ./lilypond-2.2.6/stepmake/stepmake/metafont-rules.make	2005-02-10 21:04:47.085433033 -0600
@@ -30,11 +30,11 @@
 
 
 $(outdir)/%.pfa: %.mf
-	$(MFTRACE) $(MFTRACE_FLAGS) -I $(outdir)/ --pfa --simplify $(basename $(@F))
+	$(MFTRACE) $(MFTRACE_FLAGS) -I $(outdir)/ --simplify $(basename $(@F))
 	mv $(basename $(@F)).pfa $(outdir)
 
 $(outdir)/%.pfb: %.mf
-	$(MFTRACE)  $(MFTRACE_FLAGS) -I $(outdir)/ --pfa --pfb --simplify  $(basename $(@F))
+	$(MFTRACE)  $(MFTRACE_FLAGS) -I $(outdir)/ -f PFA,PFB --simplify  $(basename $(@F))
 	-mv $(basename $(@F)).pfa $(outdir)
 	mv $(basename $(@F)).pfb $(outdir)
 
