--- mongodb-2.4.6.orig/src/mongo/db/auth/privilege_set.cpp
+++ mongodb-2.4.6/src/mongo/db/auth/privilege_set.cpp
@@ -79,7 +79,7 @@ namespace mongo {
         resourceSearchList[1] = nsToDatabaseSubstring(desiredPrivilege.getResource());
 
         ActionSet unmetRequirements = desiredPrivilege.getActions();
-        for (int i = 0; i < boost::size(resourceSearchList); ++i) {
+        for (unsigned int i = 0; i < boost::size(resourceSearchList); ++i) {
             ResourcePrivilegeCacheEntry* entry = _lookupEntry(resourceSearchList[i]);
             if (NULL == entry)
                 continue;
--- mongodb-2.4.6.orig/src/mongo/db/cmdline_test.cpp
+++ mongodb-2.4.6/src/mongo/db/cmdline_test.cpp
@@ -71,7 +71,7 @@ namespace {
             "KEEP",
             "--servicePasswordFake=KEEP"
         };
-        const int argc = boost::size(argv);
+        const unsigned int argc = boost::size(argv);
         testCensoringArgv(argv, argv, argc);
     }
 
@@ -88,7 +88,7 @@ namespace {
             "--servicePassword",
             "get out of dodge"
         };
-        const int argc = boost::size(argv);
+        const unsigned int argc = boost::size(argv);
 
         const char* const expected[] = {
             "first",
@@ -120,7 +120,7 @@ namespace {
             "-servicePassword",
             "get out of dodge"
         };
-        const int argc = boost::size(argv);
+        const unsigned int argc = boost::size(argv);
 
         const char* const expected[] = {
             "first",
@@ -152,7 +152,7 @@ namespace {
             "KEEP",
             "--servicePasswordFake=KEEP"
         };
-        const int argc = boost::size(argv);
+        const unsigned int argc = boost::size(argv);
         testCensoringVector(argv, argv, argc);
     }
 
@@ -169,7 +169,7 @@ namespace {
             "--servicePassword",
             "get out of dodge"
         };
-        const int argc = boost::size(argv);
+        const unsigned int argc = boost::size(argv);
 
         const char* const expected[] = {
             "first",
@@ -201,7 +201,7 @@ namespace {
             "-servicePassword",
             "get out of dodge"
         };
-        const int argc = boost::size(argv);
+        const unsigned int argc = boost::size(argv);
 
         const char* const expected[] = {
             "first",
