--- SConstruct.orig	2014-08-22 01:56:32.000000000 +0400
+++ SConstruct	2014-09-27 15:28:11.496080339 +0400
@@ -704,7 +704,6 @@
     # -Winvalid-pch Warn if a precompiled header (see Precompiled Headers) is found in the search path but can't be used.
     env.Append( CCFLAGS=["-fPIC",
                          "-fno-strict-aliasing",
-                         "-ggdb",
                          "-pthread",
                          "-Wall",
                          "-Wsign-compare",
@@ -719,9 +718,10 @@
             env.Append( CCFLAGS=["-fno-builtin-memcmp"] ) # glibc's memcmp is faster than gcc's
 
     env.Append( CPPDEFINES=["_FILE_OFFSET_BITS=64"] )
-    env.Append( CXXFLAGS=["-Wnon-virtual-dtor", "-Woverloaded-virtual"] )
+    env.Append( CXXFLAGS=os.environ['CXXFLAGS'] )
+    env.Append( LINKFLAGS=os.environ['LDFLAGS'] )
     env.Append( LINKFLAGS=["-fPIC", "-pthread",  "-rdynamic"] )
-    env.Append( LIBS=[] )
+    env.Append( LIBS=['pcre', 'pcrecpp'] )
 
     #make scons colorgcc friendly
     for key in ('HOME', 'TERM'):
--- src/mongo/SConscript.orig	2014-08-22 01:56:32.000000000 +0400
+++ src/mongo/SConscript	2014-09-27 18:24:40.903897125 +0400
@@ -456,7 +456,7 @@
 env.Library('stacktrace', 'util/stacktrace.cpp')
 
 env.Program('lame_stacktrace_test', 'util/lame_stacktrace_test.cpp',
-            LIBDEPS=['stacktrace',
+            LIBDEPS=['stacktrace', 'foundation', 'mongocommon', 'alltools',
                      '$BUILD_DIR/mongo/base/base'])
 
 serverOnlyFiles += mmapFiles
