From 33cff189ff0f0d2d2f735211d72f1929db74485e Mon Sep 17 00:00:00 2001
From: Koen Holtman <koen.holtman@ieee.org>
Date: Tue, 25 Sep 2012 20:27:10 +0200
Subject: [PATCH 3/5] fix uint/size_t

---
 Makefile | 4 ++--
 afio.c   | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index cdb71e4..e9db102 100644
--- a/Makefile
+++ b/Makefile
@@ -68,8 +68,8 @@ g	= -DHAVEFNMATCH
 LARGEFILEFLAGS=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 
 # even more warnings flags..
-MW=
-#MW=-Wtraditional -Wcast-qual -Wcast-align -Wconversion -pedantic -Wlong-long -Wimplicit -Wuninitialized -W -Wshadow -Wsign-compare -Wstrict-prototypes -Wmissing-declarations
+#MW=-Wformat -Werror=format-security -Wall
+MW=-Wtraditional -Wcast-qual -Wcast-align -Wconversion -pedantic -Wlong-long -Wimplicit -Wuninitialized -W -Wshadow -Wsign-compare -Wstrict-prototypes -Wmissing-declarations
 
 CFLAGS1 = -Wall -Wstrict-prototypes -s -O2 -fomit-frame-pointer $(LARGEFILEFLAGS) $(MW)
 
diff --git a/afio.c b/afio.c
index 3cadb82..c9092ba 100644
--- a/afio.c
+++ b/afio.c
@@ -1558,7 +1558,7 @@ indata (fd, size, name)
   reg ssize_t sparse;
   reg int corrupt;
   auto char *buf;
-  auto size_t avail;
+  auto uint avail;
 
   corrupt = sparse = 0;
   oops = NULL;
-- 
2.3.4

