From 822904b09b36d70a40720b71c6a88ee17826726d Mon Sep 17 00:00:00 2001
From: Koen Holtman <koen.holtman@ieee.org>
Date: Tue, 25 Sep 2012 20:28:32 +0200
Subject: [PATCH 4/5] makefile uncomment fix

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index e9db102..cde1a7b 100644
--- a/Makefile
+++ b/Makefile
@@ -68,8 +68,8 @@ g	= -DHAVEFNMATCH
 LARGEFILEFLAGS=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 
 # even more warnings flags..
-#MW=-Wformat -Werror=format-security -Wall
-MW=-Wtraditional -Wcast-qual -Wcast-align -Wconversion -pedantic -Wlong-long -Wimplicit -Wuninitialized -W -Wshadow -Wsign-compare -Wstrict-prototypes -Wmissing-declarations
+MW=-Wformat -Werror=format-security -Wall
+#MW=-Wtraditional -Wcast-qual -Wcast-align -Wconversion -pedantic -Wlong-long -Wimplicit -Wuninitialized -W -Wshadow -Wsign-compare -Wstrict-prototypes -Wmissing-declarations
 
 CFLAGS1 = -Wall -Wstrict-prototypes -s -O2 -fomit-frame-pointer $(LARGEFILEFLAGS) $(MW)
 
-- 
2.3.4

