From ea98d67d572be666f37ad63c8d8a31f3bb2d0c0a Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Mon, 16 Oct 2023 14:14:12 +0200
Subject: [PATCH] unity fix

---
 src/parsers/plan2/FlexLexer.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/parsers/plan2/FlexLexer.h b/src/parsers/plan2/FlexLexer.h
index b251316..bc44e3a 100644
--- a/src/parsers/plan2/FlexLexer.h
+++ b/src/parsers/plan2/FlexLexer.h
@@ -24,6 +24,7 @@
 //  #include <FlexLexer.h>
 //  ...
 
+#pragma once
 #ifndef __FLEX_LEXER_H
 // Never included before - need to define base class.
 #define __FLEX_LEXER_H
-- 
2.42.0

