From 66c1f8c270f748286643f4368a0e7ad57a2344c0 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Tue, 21 Mar 2017 12:03:16 +0100
Subject: [PATCH] tree-wide: include <sys/sysmacros.h> directly

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/lxccontainer.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/lxc/lxccontainer.c b/src/lxc/lxccontainer.c
index 0dbbf2c..4786908 100644
--- a/src/lxc/lxccontainer.c
+++ b/src/lxc/lxccontainer.c
@@ -31,6 +31,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <arpa/inet.h>
+#include <sys/sysmacros.h>
 #include <sys/mman.h>
 #include <sys/mount.h>
 #include <sys/syscall.h>
@@ -64,9 +65,6 @@
 #ifdef MAJOR_IN_MKDEV
 #    include <sys/mkdev.h>
 #endif
-#ifdef MAJOR_IN_SYSMACROS
-#    include <sys/sysmacros.h>
-#endif
 
 #if HAVE_IFADDRS_H
 #include <ifaddrs.h>
