diff -uNr OdbcJdbc.orig/OdbcInstGetProp.cpp OdbcJdbc/OdbcInstGetProp.cpp
--- OdbcJdbc.orig/OdbcInstGetProp.cpp	2004-11-01 12:49:16.798557512 +0100
+++ OdbcJdbc/OdbcInstGetProp.cpp	2004-11-01 12:51:10.005347464 +0100
@@ -1,6 +1,14 @@
 
 #ifndef _WIN32
 #include <odbcinstext.h>
+#include <stdlib.h>
+#include <memory.h>
+#include <stdio.h>
+#include <string.h>
+#ifndef TRUE
+#define FALSE 0;
+#define TRUE 1;
+#endif
 
 #include "SetupAttributes.h"
 
