diff -Naur amarok-2.0.2.orig/src/collection/ipodcollection/handler/IpodHandler.cpp amarok-2.0.2/src/collection/ipodcollection/handler/IpodHandler.cpp
--- amarok-2.0.2.orig/src/collection/ipodcollection/handler/IpodHandler.cpp	2009-03-01 18:21:38.000000000 +0100
+++ amarok-2.0.2/src/collection/ipodcollection/handler/IpodHandler.cpp	2009-04-22 21:18:51.000000000 +0200
@@ -1339,12 +1339,17 @@
     QFileInfo tempImageFileInfo( tempImageFile ); // get info for path
     QString tempImagePath = tempImageFileInfo.absoluteFilePath(); // path
 
+#ifdef ITDB_THUMB_COVER_SMALL
     Itdb_Thumb *thumb = NULL;
+#endif 
     GdkPixbuf *gpixbuf = NULL;
     QString thumbPath;
 
     // pull image out of ipod
 
+#ifdef ITDB_THUMB_COVER_SMALL
+    // we've got libgpod 0.6.0 here
+
     if( ipodtrack->has_artwork == 0x01 )
     {
         // try small first
@@ -1406,6 +1411,14 @@
             }
         }
     }
+#else
+    // libgpod 0.7.0
+
+    if( itdb_track_has_thumbnails ( ipodtrack ) )
+    {
+        gpixbuf = (GdkPixbuf*) itdb_track_get_thumbnail (ipodtrack, -1, -1 );
+    }
+#endif
 
     if(gpixbuf != NULL)
     {
