From f062f976164eb84e0005f99fa4b2a6d1c4e5f0c8 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Tue, 17 Mar 2026 06:57:04 +0100
Subject: [PATCH 3/8] Makefile: Respect CFLAGS

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 Makefile        | 2 +-
 ripOLE/Makefile | 2 +-
 tnef/Makefile   | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 21685601176b..301aec845c16 100644
--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ VERSION=0.1.15
 #	use our recommended settings.
 #CFLAGS ?= -Wall -g -O2 -Werror $(CPU_OPTS)
 #CFLAGS=-Wall -g -O2 -Wundef -Wshadow -Wsign-compare -I.
-CFLAGS=-Wall  -I. -O2 
+CFLAGS ?= -Wall -Wextra -O2
 
 # OLE decoding is still considered to be 'beta' mode - so it 
 #	disabled in the stable release of ripMIME for now
diff --git a/ripOLE/Makefile b/ripOLE/Makefile
index 6276e04fd54a..ab1a912b8127 100644
--- a/ripOLE/Makefile
+++ b/ripOLE/Makefile
@@ -1,6 +1,6 @@
 
 OBJS= ole.o olestream-unwrap.o bytedecoders.o logger.o pldstr.o bt-int.o
-CFLAGS=-Wall -g -O2 -I. 
+CFLAGS ?= -Wall -Wextra -O2
 
 
 .c.o:
diff --git a/tnef/Makefile b/tnef/Makefile
index a527c5516f24..871b1af137f2 100644
--- a/tnef/Makefile
+++ b/tnef/Makefile
@@ -5,7 +5,7 @@
 # default is big endian.  You can do -D__TNEF_BYTE_ORDER 1234 for little
 # endian
 PLATFORM=-D___TNEF_BYTE_ORDER=4321 
-CFLAGS=-Wall -g -O2 -I.
+CFLAGS ?= -Wall -Wextra -O2
 
 default: config.h tnef.h tnef.c
 	$(CC) $(CFLAGS) $(PLATFORM) -c tnef.c
