--- configure~	2005-06-03 20:18:41.000000000 +0200
+++ configure	2005-12-09 16:41:45.000000000 +0100
@@ -220,18 +220,6 @@
 
 echo "Checking your current setup for incompatibilities..."
 
-echo -n "Checking for /proc... "
-
-if test -n "`/bin/mount | grep \"type proc\"`"; then
-	echo "yes"
-else
-	echo "no"
-	echo "*** ERROR:"
-	echo "*** proc-fs is needed for testing"
-	echo "*** We recommend you enable proc-fs in your kernel"
-	exit 1
-fi
-
 echo -n "Checking for kernel gcc... "
 
 kgcc=`cat /proc/version | sed 's/\(.*\)gcc[ -][vV]ersion \([0-9.]*\).*/\2/'`
