--- configure~	2004-02-27 20:47:15.000000000 +0100
+++ configure	2004-02-28 11:21:45.000000000 +0100
@@ -195,18 +195,6 @@
 
 echo "Checking your current setup for incompatibilities..."
 
-echo -n "Checking for /proc... "
-
-if test -n "`/bin/mount|grep \"type proc\"`"; then
-	echo "yes"
-else
-	echo "no"
-	echo "*** ERROR:"
-	echo "*** proc-fs is needed for testing"
-	echo "*** We recommend you enable proc-fs in your kernel"
-	exit 1
-fi
-
 echo -n "Checking for kernel gcc... "
 
 kgcc=`cat /proc/version | sed 's/\(.*\)gcc version \([0-9.]*\).*/\2/'`
