--- a/content/canvas/src/nsCanvasRenderingContext2D.cpp	2007-09-16 02:28:55.602763128 +0000
+++ b/content/canvas/src/nsCanvasRenderingContext2D.cpp	2007-09-16 02:28:27.527046255 +0000
@@ -98,6 +98,12 @@
 #include "nsTArray.h"
 
 #include "cairo.h"
+typedef enum _cairo_font_type {
+	CAIRO_FONT_TYPE_TOY,
+	CAIRO_FONT_TYPE_FT,
+	CAIRO_FONT_TYPE_WIN32,
+	CAIRO_FONT_TYPE_ATSUI
+} cairo_font_type_t;
 #include "imgIEncoder.h"
 #ifdef MOZILLA_1_8_BRANCH
 #define imgIEncoder imgIEncoder_MOZILLA_1_8_BRANCH
