--- Makefile.in~	2005-03-21 21:43:07.337253376 +0100
+++ Makefile.in	2005-03-21 21:44:52.123323472 +0100
@@ -206,7 +206,7 @@
 target_os = @target_os@
 target_vendor = @target_vendor@
 AUTOMAKE_OPTIONS = foreign 1.6 dist-zip dist-bzip2
-SUBDIRS = lib include doc examples debian
+SUBDIRS = lib include doc debian
 EXTRA_DIST = COPYING autogen.sh win32 libtheora.spec libtheora.spec.in \
 	theora-uninstalled.pc.in
 
