diff --git a/util/cairo-trace/meson.build b/util/cairo-trace/meson.build
index c4a504759..d90a994ea 100644
--- a/util/cairo-trace/meson.build
+++ b/util/cairo-trace/meson.build
@@ -12,7 +12,8 @@ libcairotrace = library('cairo-trace', cairo_trace_sources,
   include_directories: [incbase, incsrc],
   dependencies: deps,
   link_args: extra_link_args,
-  c_args: ['-DSHARED_LIB_EXT="@0@"'.format(shared_lib_ext),] + pthread_c_args,
+  c_args: ['-DSHARED_LIB_EXT="@0@"'.format(shared_lib_ext),
+           '-DPACKAGE=cairo -DPACKAGE_VERSION="@0@"'.format(meson.project_version()),] + pthread_c_args,
   install: true,
   install_dir: join_paths(get_option('prefix'), get_option('libdir'), 'cairo'),
 )
