diff -Naur xorg-cf-files-1.0.3.orig/linux.cf xorg-cf-files-1.0.3/linux.cf
--- xorg-cf-files-1.0.3.orig/linux.cf	2009-10-27 14:13:33.480912412 -0700
+++ xorg-cf-files-1.0.3/linux.cf	2009-10-27 14:14:01.877916656 -0700
@@ -1052,6 +1052,16 @@
 #define TtClientLibs $(TTLIB) $(XTOOLLIB) $(XLIB)
 #define TtClientDepLibs $(DEPTTLIB) $(DEPXTOOLLIB) $(DEPXLIB)
 
+#ifndef ExtraXawClientLibs
+# define ExtraXawClientLibs /**/
+#endif
+#ifndef ExtraXawClientDepLibs
+# define ExtraXawClientDepLibs /**/
+#endif
+#ifndef ExtraXawReqs
+# define ExtraXawReqs /**/
+#endif
+
 #if HaveLib64
 # ifndef LibDirName
 #  define LibDirName		lib64
