--- kaddressbook/Makefile.orig	2005-01-12 07:42:36.645030994 +0000
+++ kaddressbook/Makefile	2005-01-12 07:42:44.204814266 +0000
@@ -13,7 +13,7 @@ CXX      = g++
 LEX      = flex
 YACC     = yacc
 CFLAGS   = -pipe -Wall -W -O -march=athlon-4 -mmmx -mfpmath=sse -msse -m3dnow -pipe -DPIC -fPIC -O3  -DKAB_EMBEDDED -DKAB_NOSPLITTER -DDESKTOP_VERSION -DQT_NO_DEBUG
-CXXFLAGS = -pipe -Wall -W -O -march=athlon-4 -mmmx -mfpmath=sse -msse -m3dnow -pipe -DPIC -fPIC -O3  -DKAB_EMBEDDED -DKAB_NOSPLITTER -DDESKTOP_VERSION -DQT_NO_DEBUG
+CXXFLAGS = -pipe -Wall -W -O -march=athlon-4 -mmmx -mfpmath=sse -msse -m3dnow -pipe -DPIC -fPIC -O3  -DKAB_EMBEDDED -DKAB_NOSPLITTER -DDESKTOP_VERSION -DQT_NO_DEBUG -fpermissive
 LEXFLAGS = 
 YACCFLAGS= -d
 INCPATH  = -I/usr/share/qt/mkspecs/default -I. -I. -Idetails -Ifeatures -Ixxport -I../libkdepim -I../microkde -I../microkde/kdecore -I../microkde/kutils -I../microkde/kio/kfile -I../microkde/kio/kio -I../microkde/kdeui -I../microkde/kresources -I../kabc -I.. -Iinterfaces -I$(QTDIR)/include/qt -Imoc/unix/
