--- kmicromail/qpe/qdialog.h.orig	2005-01-12 07:37:50.925016925 +0000
+++ kmicromail/qpe/qdialog.h	2005-01-12 07:38:16.639878182 +0000
@@ -8,7 +8,7 @@
 #warning if you get a compiling error 
 #warning please adjust your path here!
 
-#include "/usr/local/qt/include/qdialog.h"
+#include "/usr/include/qt/qdialog.h"
 class QDialog_hacked : public QDialog
 {
     //Q__OBJECT
@@ -23,7 +23,7 @@ class QDialog_hacked : public QDialog
 #endif
 
 #else
-#warning "/usr/local/qt/include/qdialog.h" for KDialog
-#include "/usr/local/qt/include/qdialog.h"
+#warning "/usr/include/qt/qdialog.h" for KDialog
+#include "/usr/include/qt/qdialog.h"
 
 #endif
