--- htdig-3.2.0b6/htsearch/Collection.h.old    2006-09-10 01:11:13.000000000 +0200
+++ htdig-3.2.0b6/htsearch/Collection.h        2006-09-10 01:11:32.000000000+0200
@@ -36,9 +36,9 @@
                const char *docExcerpt);
     ~Collection();
 
-    void Collection::Open();
+    void Open();
 
-    void Collection::Close(); 
+    void Close(); 
 
     char *getWordFile() { return wordFile.get(); }
