diff -uNr kdegraphics-4.2.85/libs/libksane/libksane/ksane_widget_private.cpp kdegraphics-4.2.85.cp/libs/libksane/libksane/ksane_widget_private.cpp
--- kdegraphics-4.2.85/libs/libksane/libksane/ksane_widget_private.cpp	2009-03-10 13:24:04.000000000 +0100
+++ kdegraphics-4.2.85.cp/libs/libksane/libksane/ksane_widget_private.cpp	2009-05-08 10:55:04.000000000 +0200
@@ -738,6 +738,7 @@
         switch (status)
         {
             #ifndef SANE_CAP_ALWAYS_SETTABLE
+             #ifdef SANE_STATUS_WARMING_UP
             // should better be done by detecting SANE's version in configure and providing a HAS_SANE_1_1
             // FIXME remove these ifdefs and require sane 1.1.x as soon as possible
             case SANE_STATUS_WARMING_UP:
@@ -751,6 +752,7 @@
                     scanDone();
                 }
                 return;
+             #endif
             #endif
             case SANE_STATUS_NO_DOCS:
                 // if it is the first doc in batch mode this is an error
@@ -760,7 +762,9 @@
             case SANE_STATUS_DEVICE_BUSY:
             case SANE_STATUS_ACCESS_DENIED:
             #ifndef SANE_CAP_ALWAYS_SETTABLE
+             #ifdef SANE_STATUS_HW_LOCKED
             case SANE_STATUS_HW_LOCKED:
+             #endif
             #endif
                 KMessageBox::sorry(0, i18n(sane_strstatus(status)));
                 scanCancel();
