From 61997ad3bb87b6e4750101cbee5b2e944db86998 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 4 Oct 2020 20:52:19 +0200
Subject: [PATCH] Fix compilation error

---
 src/include/LUT.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/include/LUT.h b/src/include/LUT.h
index 5097fd3..874ff5a 100644
--- a/src/include/LUT.h
+++ b/src/include/LUT.h
@@ -63,6 +63,7 @@
 #include <cstdint>
 #ifndef NDEBUG
 #include <cassert>
+#include <cstdio>
 #endif
 #include "opthelper.h"
 #include "rt_math.h"
-- 
2.28.0

