diff -ruN xpp-1.5.orig/xpp.h xpp-1.5/xpp.h
--- xpp-1.5.orig/xpp.h	2006-12-31 22:07:11.000000000 +0100
+++ xpp-1.5/xpp.h	2006-12-31 22:08:45.000000000 +0100
@@ -177,7 +177,7 @@
    */
 
   int /* O - index of the record of the given option in the array */
-  printFiles::getSpecialOptionIndex(int *num_special_opts_in_arr,
+  getSpecialOptionIndex(int *num_special_opts_in_arr,
 				      /* number of options in the array */
 				      special_opt_t **special_opts_arr,
                                       /* array of numerical/string 
