--- bsd-finger-0.17/finger/lprint.c.old	2005-07-28 01:50:51.000000000 -0700
+++ bsd-finger-0.17/finger/lprint.c	2005-07-28 01:50:17.000000000 -0700
@@ -62,6 +62,7 @@
 #define	_PATH_PLAN	".plan"
 #define	_PATH_PROJECT	".project"
 #define _PATH_PGPKEY	".pgpkey"
+#define _PATH_SIGNATURE	".signature"
 
 void 
 lflag_print(void)
@@ -72,6 +73,7 @@
 		if (!pplan) {
 			show_text(pn->dir, _PATH_PGPKEY, "PGP key:\n");
 			show_text(pn->dir, _PATH_PROJECT, "Project:\n");
+			show_text(pn->dir, _PATH_SIGNATURE, "Signature:\n");
			if (!show_text(pn->dir, _PATH_PLAN, "Plan:\n")) {
				xprintf("No Plan.\n");
 			}
@@ -336,7 +338,7 @@
 
 	snprintf(tbuf, TBUFLEN, "%s/%s", directory, file_name);
 
-	if (lstat(tbuf, &sbuf1) || !S_ISREG(sbuf1.st_mode)) return 0;
+	if (stat(tbuf, &sbuf1) || !S_ISREG(sbuf1.st_mode)) return 0;
 	fd = open(tbuf, O_RDONLY);
 	if (fd<0) return 0;
 	if (fstat(fd, &sbuf2)) { close(fd); return 0; }
@@ -350,7 +352,7 @@
 
 	xprintf("%s", header);
 	while ((ch = getc(fp)) != EOF) {
-		xputc(ch);
+		putc(ch,stdout);
 		lastc = ch;
 	}
 	if (lastc != '\n') xputc('\n');
