From 761b60d6a8cce0f67601bf1ac119a707fc2e164e Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 20:18:13 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 text-short.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/text-short.cabal b/text-short.cabal
index 0b704c5..ee81a02 100644
--- a/text-short.cabal
+++ b/text-short.cabal
@@ -33,7 +33,7 @@ library
 
   other-modules:       Data.Text.Short.Internal
 
-  build-depends:       base        >= 4.7    && < 4.12
+  build-depends:       base        >= 4.7
                      , bytestring  >= 0.10.4 && < 0.11
                      , hashable    >= 1.2.6  && < 1.3
                      , deepseq     >= 1.3    && < 1.5
-- 
2.19.1

