--- scripts/exim_install.orig	2011-03-22 11:00:51.000000000 +0300
+++ scripts/exim_install	2011-04-21 19:30:29.361679691 +0400
@@ -316,8 +316,8 @@
         # If "exim" does not already exist just create a symbolic link.
 
         else
-          echo \(cd ${BIN_DIRECTORY}\; ${LN} -s ${version} exim\)
-          (${real} cd ${BIN_DIRECTORY}; ${real} ${LN} -s ${version} exim)
+          echo \(cd ${BIN_DIRECTORY}\; ${LN} -sf ${version} exim\)
+          (${real} cd ${BIN_DIRECTORY}; ${real} ${LN} -sf ${version} exim)
           if [ $? -ne 0 ]; then
             echo $com ""
             echo $com "*** Exim installation ${ver}failed ***"
@@ -378,21 +378,10 @@
 
 echo $com ""
 
-if [ `expr -- "${CONFIGURE_FILE}" : ".*:"` -ne 0 ] ; then
-  echo $com Runtime configuration is specified as the following list:
-  echo $com ' ' ${CONFIGURE_FILE}
-  echo $com Therefore, skipping automatic installation.
-
-elif [ ! -f ${CONFIGURE_FILE} ]; then
   echo $com Installing default configuration in ${CONFIGURE_FILE}
   echo $com because there is no existing configuration file.
-  if [ "${SYSTEM_ALIASES_FILE}" = "" ] ; then
-    SYSTEM_ALIASES_FILE=/etc/aliases
-    echo $com This configuration has system aliases in ${SYSTEM_ALIASES_FILE}.
-  fi
-
-  echo ${MKDIR} -p `${DIRNAME} ${CONFIGURE_FILE}`
-  ${real} ${MKDIR} -p `${DIRNAME} ${CONFIGURE_FILE}`
+  SYSTEM_ALIASES_FILE=aliases.dist
+	CONFIGURE_FILE=exim.conf.dist
 
   echo sed -e '\\'
   echo "  \"/SYSTEM_ALIASES_FILE/ s'SYSTEM_ALIASES_FILE'${ACTUAL_SYSTEM_ALIASES_FILE}'\"" '\\'
@@ -425,10 +414,6 @@
     ${real} ${CP} ../src/aliases.default ${SYSTEM_ALIASES_FILE}
   fi
 
-else
-  echo $com Configuration file ${CONFIGURE_FILE} already exists
-fi
-
 # Install info files if the directory is defined and the Texinfo
 # source documentation is present.
 
