diff -Naur exim-4.50.orig/scripts/exim_install exim-4.50/scripts/exim_install
--- exim-4.50.orig/scripts/exim_install	2005-06-02 22:59:51.867468752 -0700
+++ exim-4.50/scripts/exim_install	2005-06-03 09:01:46.733420976 -0700
@@ -353,21 +353,10 @@
 
 echo $com ""
 
-if [ `expr "${CONFIGURE_FILE}" : ".*:"` -ne 0 ] ; then
-  echo $com Runtime configuration is specified as the following list:
-  echo $com ' ' ${CONFIGURE_FILE}
-  echo $com Therefore, skipping automatic installation.
-
-elif [ ! -f ${CONFIGURE_FILE} ]; then
   echo $com Installing default configuration in ${CONFIGURE_FILE}
   echo $com because there is no existing configuration file.
-  if [ "${SYSTEM_ALIASES_FILE}" = "" ] ; then
-    SYSTEM_ALIASES_FILE=/etc/aliases
-    echo $com This configuration has system aliases in ${SYSTEM_ALIASES_FILE}.
-  fi
-
-  echo ${MKDIR} -p `${DIRNAME} ${CONFIGURE_FILE}`
-  ${real} ${MKDIR} -p `${DIRNAME} ${CONFIGURE_FILE}`
+    SYSTEM_ALIASES_FILE=aliases.dist
+    CONFIGURE_FILE=exim.conf.dist
 
   echo sed -e '\\'
   echo "  \"/SYSTEM_ALIASES_FILE/ s'SYSTEM_ALIASES_FILE'${ACTUAL_SYSTEM_ALIASES_FILE}'\"" '\\'
@@ -400,10 +389,6 @@
     ${real} ${CP} ../src/aliases.default ${SYSTEM_ALIASES_FILE}
   fi
 
-else
-  echo $com Configuration file ${CONFIGURE_FILE} already exists
-fi
-
 # Install info files if the directory is defined and the Texinfo
 # source documentation is present.
 
