--- src/IDocumentLink.h.orig	2006-10-17 18:23:19.000000000 +0400
+++ src/IDocumentLink.h	2008-04-28 13:56:56.762082838 +0400
@@ -33,7 +33,7 @@
     class IDocumentLink
     {
         public:
-            IDocumentLink (gdouble x1, gdouble y1, gdouble x2, gdouble y1);
+            IDocumentLink (gdouble x1, gdouble y1, gdouble x2, gdouble y2);
             virtual ~IDocumentLink (void);
 
             virtual void activate (IDocument *document) = 0;
