diff -Naur vacation-1.2.7.0-beta4/Makefile vacation-fixed/Makefile
--- vacation-1.2.7.0-beta4/Makefile	2007-04-16 13:01:29.000000000 +0200
+++ vacation-fixed/Makefile	2007-07-25 15:06:37.000000000 +0200
@@ -10,6 +10,7 @@
 #
 # Default CFLAGS for all builds, architecture flags get appended below.
 CFLAGS		= $(RPM_OPT_FLAGS) -Wall -DMAIN
+
 ifeq "$(ARCH)" "x86_64"
 # Uncomment below for backwards compatibility of gdbm files.
 #  CFLAGS       = $(CFLAGS) -m32
@@ -33,9 +34,9 @@
  
 ifeq "$(OS)" "FreeBSD"
 # PREFIX for FreeBSD
-PREFIX		= /usr/local
-IFLAGS		= -I/usr/local/include
-LIBS		= -L/usr/local/lib -lgdbm
+PREFIX		= /usr
+IFLAGS		= -I/usr/include
+LIBS		= -L/usr/lib -lgdbm
 else
 # PREFIX for Linux
 PREFIX		= /usr
@@ -45,7 +46,7 @@
 BINDIR		= $(PREFIX)/bin
 VACATION        = $(BINDIR)/vacation
 VACLOOK         = $(BINDIR)/vaclook
-MANDIR		= $(PREFIX)/man/man
+MANDIR		= $(PREFIX)/share/man/man
 MANEXT1		= 1
 
 VERSION 	= 1
diff -Naur vacation-1.2.7.0-beta4/vacation.c vacation-fixed/vacation.c
--- vacation-1.2.7.0-beta4/vacation.c	2007-01-21 03:16:22.000000000 +0100
+++ vacation-fixed/vacation.c	2007-07-25 14:22:08.000000000 +0200
@@ -140,9 +140,6 @@
     case 'i':			/* init the database */
       iflag = 1;
       break;
-    case 'i':			/* init the database */
-      iflag = 1;
-      break;
     case 't':
       if (isdigit(*optarg)) {
 	interval = atol(optarg) * SECSPERDAY;  /* unit is `days' */
