--- plugins/elements/gstfilesrc.c	2007/02/13 13:40:05	1.150
+++ plugins/elements/gstfilesrc.c	2007/04/27 07:34:10	1.152
@@ -1074,14 +1074,21 @@
      * "location" with uri + 16 because it provides unescaping */
     location = gst_uri_get_location (tmp);
     g_free (tmp);
+  } else if (strcmp (uri, "file://") == 0) {
+    /* Special case for "file://" as this is used by some applications
+     *  to test with gst_element_make_from_uri if there's an element
+     *  that supports the URI protocol. */
+    gst_file_src_set_location (src, NULL);
+    return TRUE;
   } else {
     location = gst_uri_get_location (uri);
-    if (!location)
-      return FALSE;
-    if (!g_path_is_absolute (location)) {
-      g_free (location);
-      return FALSE;
-    }
+  }
+
+  if (!location)
+    return FALSE;
+  if (!g_path_is_absolute (location)) {
+    g_free (location);
+    return FALSE;
   }
 
   ret = gst_file_src_set_location (src, location);

--- plugins/elements/gstfilesink.c	2007/02/13 13:40:05	1.84
+++ plugins/elements/gstfilesink.c	2007/04/27 07:34:10	1.86
@@ -554,14 +554,21 @@
      * "location" with uri + 16 because it provides unescaping */
     location = gst_uri_get_location (tmp);
     g_free (tmp);
+  } else if (strcmp (uri, "file://") == 0) {
+    /* Special case for "file://" as this is used by some applications
+     *  to test with gst_element_make_from_uri if there's an element
+     *  that supports the URI protocol. */
+    gst_file_sink_set_location (sink, NULL);
+    return TRUE;
   } else {
     location = gst_uri_get_location (uri);
-    if (!location)
-      return FALSE;
-    if (!g_path_is_absolute (location)) {
-      g_free (location);
-      return FALSE;
-    }
+  }
+
+  if (!location)
+    return FALSE;
+  if (!g_path_is_absolute (location)) {
+    g_free (location);
+    return FALSE;
   }
 
   ret = gst_file_sink_set_location (sink, location);
