--- configure.orig	2008-12-13 13:28:13.000000000 -0600
+++ configure	2008-12-13 13:29:26.000000000 -0600
@@ -25173,23 +25173,6 @@
    { (exit 1); exit 1; }; }
   fi
 
-          bison_min_version=1.875
-  bison_version=`$BISON_PATH --version | head -n 1 |  sed 's/^[^0-9]*//' | sed 's/[^0-9]*$//' | cut -d' ' -f1`
-  { echo "$as_me:$LINENO: checking bison version $bison_version >= $bison_min_version" >&5
-echo $ECHO_N "checking bison version $bison_version >= $bison_min_version... $ECHO_C" >&6; }
-
-  if perl -w <<EOF
-    exit ($bison_version < $bison_min_version) ? 0 : 1;
-EOF
-  then
-    { echo "$as_me:$LINENO: result: yes" >&5
-echo "${ECHO_T}yes" >&6; }
-  else
-    { { echo "$as_me:$LINENO: error: no" >&5
-echo "$as_me: error: no" >&2;}
-   { (exit 1); exit 1; }; }
-  fi
-
 
     # Extract the first word of "flex", so it can be a program name with args.
 set dummy flex; ac_word=$2
