--- common/m4/gst-parser.m4.orig	2008-09-10 11:16:47.000000000 +0200
+++ common/m4/gst-parser.m4	2008-12-14 19:14:43.000000000 +0100
@@ -15,10 +15,7 @@
   bison_version=`$BISON_PATH --version | head -n 1 |  sed 's/^[[^0-9]]*//' | sed 's/[[^0-9]]*$//' | cut -d' ' -f1`
   AC_MSG_CHECKING([bison version $bison_version >= $bison_min_version])
 
-  if perl -w <<EOF
-    exit ($bison_version < $bison_min_version) ? 0 : 1;
-EOF 
-  then 
+  if perl -we "exit ((v$bison_version ge v$bison_min_version) ? 0 : 1)"; then 
     AC_MSG_RESULT([yes])
   else 
     AC_MSG_ERROR([no])
