diff -Naur gstreamer-0.10.30.orig/docs/gst/Makefile.in gstreamer-0.10.30/docs/gst/Makefile.in
--- gstreamer-0.10.30.orig/docs/gst/Makefile.in	2010-10-06 16:31:14.559000006 -0700
+++ gstreamer-0.10.30/docs/gst/Makefile.in	2010-10-06 16:32:12.818002205 -0700
@@ -936,7 +936,7 @@
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
diff -Naur gstreamer-0.10.30.orig/docs/libs/Makefile.in gstreamer-0.10.30/docs/libs/Makefile.in
--- gstreamer-0.10.30.orig/docs/libs/Makefile.in	2010-10-06 16:31:14.569000006 -0700
+++ gstreamer-0.10.30/docs/libs/Makefile.in	2010-10-06 16:34:52.867999323 -0700
@@ -931,7 +931,7 @@
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
diff -Naur gstreamer-0.10.30.orig/docs/plugins/Makefile.in gstreamer-0.10.30/docs/plugins/Makefile.in
--- gstreamer-0.10.30.orig/docs/plugins/Makefile.in	2010-10-06 16:31:14.566000006 -0700
+++ gstreamer-0.10.30/docs/plugins/Makefile.in	2010-10-06 16:36:13.622000065 -0700
@@ -1043,7 +1043,7 @@
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
