diff -urN icedtea6-1.10.4/acinclude.m4 icedtea6-mod/acinclude.m4
--- icedtea6-1.10.4/acinclude.m4	2011-07-11 16:43:43.361068652 -0500
+++ icedtea6-mod/acinclude.m4	2011-10-27 19:22:33.939728683 -0500
@@ -630,101 +630,6 @@
   AC_SUBST(ALT_JAR_CMD)
 ])
 
-AC_DEFUN([IT_FIND_XALAN2_JAR],
-[
-  AC_MSG_CHECKING([for a xalan2 jar])
-  AC_ARG_WITH([xalan2-jar],
-              [AS_HELP_STRING(--with-xalan2-jar,specify location of the xalan2 jar)],
-  [
-    if test -f "${withval}" ; then
-      XALAN2_JAR="${withval}"
-    fi
-  ],
-  [
-    XALAN2_JAR=
-  ])
-  if test -z "${XALAN2_JAR}"; then
-    if test -e "/usr/share/java/xalan-j2.jar"; then
-      XALAN2_JAR=/usr/share/java/xalan-j2.jar
-    elif test -e "/usr/share/java/xalan2.jar"; then
-      XALAN2_JAR=/usr/share/java/xalan2.jar
-    elif test -e "/usr/share/xalan/lib/xalan.jar"; then
-      XALAN2_JAR=/usr/share/xalan/lib/xalan.jar
-    else
-      AC_MSG_RESULT(no)
-    fi
-  fi
-  if test -z "${XALAN2_JAR}"; then
-    AC_MSG_ERROR("A xalan2 jar was not found.")
-  fi
-  AC_MSG_RESULT(${XALAN2_JAR})
-  AC_SUBST(XALAN2_JAR)
-])
-
-AC_DEFUN([IT_FIND_XALAN2_SERIALIZER_JAR],
-[
-  AC_MSG_CHECKING([for a xalan2 serializer jar])
-  AC_ARG_WITH([xalan2-serializer-jar],
-              [AS_HELP_STRING(--with-xalan2-serializer-jar,specify location of the xalan2-serializer jar)],
-  [
-    if test -f "${withval}" ; then
-      XALAN2_SERIALIZER_JAR="${withval}"
-    fi
-  ],
-  [
-    XALAN2_SERIALIZER_JAR=
-  ])
-  if test -z "${XALAN2_SERIALIZER_JAR}"; then
-    if test -e "/usr/share/java/xalan-j2-serializer.jar"; then
-      XALAN2_SERIALIZER_JAR=/usr/share/java/xalan-j2-serializer.jar
-    elif test -e "/usr/share/xalan-serializer/lib/serializer.jar"; then
-      XALAN2_SERIALIZER_JAR=/usr/share/xalan-serializer/lib/serializer.jar
-    elif test -e "/usr/share/java/serializer.jar"; then
-      XALAN2_SERIALIZER_JAR=/usr/share/java/serializer.jar
-    else
-      AC_MSG_RESULT(no)
-    fi
-  fi
-  if test -z "${XALAN2_SERIALIZER_JAR}"; then
-    AC_MSG_ERROR("A xalan2-serializer jar was not found.")
-  fi
-  AC_MSG_RESULT(${XALAN2_SERIALIZER_JAR})
-  AC_SUBST(XALAN2_SERIALIZER_JAR)
-])
-
-AC_DEFUN([IT_FIND_XERCES2_JAR],
-[
-  AC_MSG_CHECKING([for a xerces2 jar])
-  AC_ARG_WITH([xerces2-jar],
-              [AS_HELP_STRING(--with-xerces2-jar,specify location of the xerces2 jar)],
-  [
-    if test -f "${withval}" ; then
-      XERCES2_JAR="${withval}"
-    fi
-  ],
-  [
-    XERCES2_JAR=
-  ])
-  if test -z "${XERCES2_JAR}"; then
-    if test -e "/usr/share/java/xerces-j2.jar"; then
-      XERCES2_JAR=/usr/share/java/xerces-j2.jar
-    elif test -e "/usr/share/java/xerces2.jar"; then
-      XERCES2_JAR=/usr/share/java/xerces2.jar
-    elif test -e "/usr/share/xerces-2/lib/xercesImpl.jar"; then
-      XERCES2_JAR=/usr/share/xerces-2/lib/xercesImpl.jar
-    elif test -e "/usr/share/java/xercesImpl.jar"; then
-      XERCES2_JAR=/usr/share/java/xercesImpl.jar
-    else
-      AC_MSG_RESULT(no)
-    fi
-  fi
-  if test -z "${XERCES2_JAR}"; then
-    AC_MSG_ERROR("A xerces2 jar was not found.")
-  fi
-  AC_MSG_RESULT(${XERCES2_JAR})
-  AC_SUBST(XERCES2_JAR)
-])
-
 AC_DEFUN([IT_FIND_RHINO_JAR],
 [
   AC_MSG_CHECKING([whether to include Javascript support via Rhino])
diff -urN icedtea6-1.10.4/configure.ac icedtea6-mod/configure.ac
--- icedtea6-1.10.4/configure.ac	2011-10-13 12:39:16.561204799 -0500
+++ icedtea6-mod/configure.ac	2011-10-27 19:24:20.455728608 -0500
@@ -156,9 +156,7 @@
 IT_FIND_NATIVE2ASCII
 if test "x$enable_bootstrap" = "xyes"; then
   IT_FIND_ECJ_JAR
-  IT_FIND_XALAN2_JAR
-  IT_FIND_XALAN2_SERIALIZER_JAR
-  IT_FIND_XERCES2_JAR
+  IT_FIND_TOOL([XSLTPROC], [xsltproc])
 fi
 AC_CONFIG_FILES([javac], [chmod +x javac])
 AC_CONFIG_FILES([javap], [chmod +x javap])
diff -urN icedtea6-1.10.4/Makefile.am icedtea6-mod/Makefile.am
--- icedtea6-1.10.4/Makefile.am	2011-10-13 11:27:35.063986484 -0500
+++ icedtea6-mod/Makefile.am	2011-10-27 19:20:27.399728763 -0500
@@ -40,7 +40,6 @@
 ICEDTEA_CLS_DIR = $(BUILD_OUTPUT_DIR)/classes
 ICEDTEA_CLS_DIR_ECJ = $(ECJ_BUILD_OUTPUT_DIR)/classes
 BOOT_DIR = $(abs_top_builddir)/bootstrap/jdk1.6.0
-ENDORSED_DIR = $(BOOT_DIR)/lib/endorsed
 ECJ_BOOT_DIR = $(abs_top_builddir)/bootstrap/ecj
 ICEDTEA_BOOT_DIR = $(abs_top_builddir)/bootstrap/icedtea
 JAMVM_IMPORT_PATH = $(abs_top_builddir)/jamvm/install/hotspot
@@ -436,7 +435,9 @@
 	patches/ecj/jaxp-dependency.patch \
 	patches/ecj/bootver.patch \
 	patches/ecj/getannotation-cast.patch \
-	patches/ecj/override.patch
+	patches/ecj/override.patch \
+	patches/ecj/xsltproc.patch
+ 
 
 if DTDTYPE_QNAME
 ICEDTEA_ECJ_PATCHES += \
@@ -491,8 +492,6 @@
 	ALT_BOOTDIR="$(BOOT_DIR)" \
 	BUILD_ARCH_DIR="$(BUILD_ARCH_DIR)" \
 	ICEDTEA_CLS_DIR="$(ICEDTEA_CLS_DIR)" \
-	ICEDTEA_ENDORSED_DIR="$(ENDORSED_DIR)" \
-	ENDORSED="-Djava.endorsed.dirs=$(ENDORSED_DIR)" \
 	CLASSPATH="" \
 	LD_LIBRARY_PATH="" \
 	ZERO_BUILD="$(ICEDTEA_ZERO_BUILD)" \
@@ -580,7 +579,8 @@
 	JAR_KNOWS_J_OPTIONS="$(JAR_KNOWS_J_OPTIONS)" \
 	JAR_ACCEPTS_STDIN_LIST="$(JAR_ACCEPTS_STDIN_LIST)" \
 	DISABLE_NIMBUS="true" \
-	NO_DOCS="true"
+	NO_DOCS="true" \
+	XSLT="$(XSLTPROC)"
 
 # Source files
 # FIXME (distclean): Add generated file list
@@ -1271,11 +1271,6 @@
 	ln -sf $(NATIVE2ASCII) $(ECJ_BOOT_DIR)/bin/native2ascii
 	ln -sf $(abs_top_builddir)/javac $(ECJ_BOOT_DIR)/bin/javac
 	ln -sf $(abs_top_builddir)/javap $(ECJ_BOOT_DIR)/bin/javap
-	mkdir -p $(ECJ_BOOT_DIR)/lib/endorsed && \
-	ln -sf $(XALAN2_JAR) $(ECJ_BOOT_DIR)/lib/endorsed/xalan-j2.jar && \
-	ln -sf $(XALAN2_SERIALIZER_JAR) \
-	  $(ECJ_BOOT_DIR)/lib/endorsed/xalan-j2-serializer.jar && \
-	ln -sf $(XERCES2_JAR) $(ECJ_BOOT_DIR)/lib/endorsed/xerces-j2.jar
 	mkdir -p $(ECJ_BOOT_DIR)/jre/lib && \
 	cp $(SYSTEM_JDK_DIR)/jre/lib/rt.jar $(ECJ_BOOT_DIR)/jre/lib/rt.jar && \
 	ln -sf $(SYSTEM_JDK_DIR)/jre/lib/$(JRE_ARCH_DIR) \
diff -urN icedtea6-1.10.4/patches/ecj/xsltproc.patch icedtea6-mod/patches/ecj/xsltproc.patch
--- icedtea6-1.10.4/patches/ecj/xsltproc.patch	1969-12-31 18:00:00.000000000 -0600
+++ icedtea6-mod/patches/ecj/xsltproc.patch	2011-10-27 19:28:04.711728486 -0500
@@ -0,0 +1,54 @@
+diff -r 591c7dc0b2ee make/linux/makefiles/jvmti.make
+--- openjdk-ecj.orig/hotspot/make/linux/makefiles/jvmti.make  Thu May 19 13:27:44 2011 +0200
++++ openjdk-ecj/hotspot/make/linux/makefiles/jvmti.make Wed May 25 20:01:25 2011 +0100
+@@ -59,8 +59,6 @@
+ 
+ JvmtiGeneratedFiles = $(JvmtiGeneratedNames:%=$(JvmtiOutDir)/%)
+ 
+-XSLT = $(QUIETLY) $(REMOTE) $(RUN.JAVA) $(ENDORSED) -classpath $(JvmtiOutDir) jvmtiGen
+-
+ .PHONY: all jvmtidocs clean cleanall
+ 
+ # #########################################################################
+@@ -77,34 +75,34 @@
+ 
+ $(JvmtiOutDir)/jvmtiEnter.cpp: $(both) $(JvmtiSrcDir)/jvmtiEnter.xsl
+ 	@echo Generating $@
+-	$(XSLT) -IN $(JvmtiSrcDir)/jvmti.xml -XSL $(JvmtiSrcDir)/jvmtiEnter.xsl -OUT $(JvmtiOutDir)/jvmtiEnter.cpp -PARAM interface jvmti
++	$(XSLT) -o $(JvmtiOutDir)/jvmtiEnter.cpp --stringparam interface jvmti $(JvmtiSrcDir)/jvmtiEnter.xsl $(JvmtiSrcDir)/jvmti.xml
+ 
+ $(JvmtiOutDir)/bytecodeInterpreterWithChecks.cpp: $(JvmtiGenClass) $(InterpreterSrcDir)/bytecodeInterpreter.cpp $(InterpreterSrcDir)/bytecodeInterpreterWithChecks.xml $(InterpreterSrcDir)/bytecodeInterpreterWithChecks.xsl
+ 	@echo Generating $@
+-	$(XSLT) -IN $(InterpreterSrcDir)/bytecodeInterpreterWithChecks.xml -XSL $(InterpreterSrcDir)/bytecodeInterpreterWithChecks.xsl -OUT $(JvmtiOutDir)/bytecodeInterpreterWithChecks.cpp 
++	$(XSLT) -o $(JvmtiOutDir)/bytecodeInterpreterWithChecks.cpp $(InterpreterSrcDir)/bytecodeInterpreterWithChecks.xsl $(InterpreterSrcDir)/bytecodeInterpreterWithChecks.xml
+ 
+ $(JvmtiOutDir)/jvmtiEnterTrace.cpp: $(both) $(JvmtiSrcDir)/jvmtiEnter.xsl
+ 	@echo Generating $@
+-	$(XSLT) -IN $(JvmtiSrcDir)/jvmti.xml -XSL $(JvmtiSrcDir)/jvmtiEnter.xsl -OUT $(JvmtiOutDir)/jvmtiEnterTrace.cpp -PARAM interface jvmti -PARAM trace Trace
++	$(XSLT) -o $(JvmtiOutDir)/jvmtiEnterTrace.cpp --stringparam interface jvmti --stringparam trace Trace $(JvmtiSrcDir)/jvmtiEnter.xsl $(JvmtiSrcDir)/jvmti.xml
+ 
+ $(JvmtiOutDir)/jvmtiEnvRecommended.cpp: $(both) $(JvmtiSrcDir)/jvmtiEnv.xsl $(JvmtiSrcDir)/jvmtiEnv.cpp $(JvmtiEnvFillClass)
+ 	@echo Generating $@
+-	$(XSLT) -IN $(JvmtiSrcDir)/jvmti.xml -XSL $(JvmtiSrcDir)/jvmtiEnv.xsl -OUT $(JvmtiOutDir)/jvmtiEnvStub.cpp
++	$(XSLT) -o $(JvmtiOutDir)/jvmtiEnvStub.cpp $(JvmtiSrcDir)/jvmtiEnv.xsl $(JvmtiSrcDir)/jvmti.xml
+ 	$(QUIETLY) $(REMOTE) $(RUN.JAVA) -classpath $(JvmtiOutDir) jvmtiEnvFill $(JvmtiSrcDir)/jvmtiEnv.cpp $(JvmtiOutDir)/jvmtiEnvStub.cpp $(JvmtiOutDir)/jvmtiEnvRecommended.cpp
+ 
+ $(JvmtiOutDir)/jvmtiEnv.hpp: $(both) $(JvmtiSrcDir)/jvmtiHpp.xsl
+ 	@echo Generating $@
+-	$(XSLT) -IN $(JvmtiSrcDir)/jvmti.xml -XSL $(JvmtiSrcDir)/jvmtiHpp.xsl -OUT $(JvmtiOutDir)/jvmtiEnv.hpp
++	$(XSLT) -o $(JvmtiOutDir)/jvmtiEnv.hpp $(JvmtiSrcDir)/jvmtiHpp.xsl $(JvmtiSrcDir)/jvmti.xml
+ 
+ $(JvmtiOutDir)/jvmti.h: $(both) $(JvmtiSrcDir)/jvmtiH.xsl
+ 	@echo Generating $@
+-	$(XSLT) -IN $(JvmtiSrcDir)/jvmti.xml -XSL $(JvmtiSrcDir)/jvmtiH.xsl -OUT $(JvmtiOutDir)/jvmti.h
++	$(XSLT) -o $(JvmtiOutDir)/jvmti.h $(JvmtiSrcDir)/jvmtiH.xsl $(JvmtiSrcDir)/jvmti.xml
+ 
+ jvmtidocs:  $(JvmtiOutDir)/jvmti.html 
+ 
+ $(JvmtiOutDir)/jvmti.html: $(both) $(JvmtiSrcDir)/jvmti.xsl
+ 	@echo Generating $@
+-	$(XSLT) -IN $(JvmtiSrcDir)/jvmti.xml -XSL $(JvmtiSrcDir)/jvmti.xsl -OUT $(JvmtiOutDir)/jvmti.html
++	$(XSLT) -o $(JvmtiOutDir)/jvmti.html $(JvmtiSrcDir)/jvmti.xsl $(JvmtiSrcDir)/jvmti.xml
+ 
+ # #########################################################################
+ 
