--- src/man.conf.old	2005-04-29 10:57:05.000000000 +0200
+++ src/man.conf	2005-04-29 11:10:45.000000000 +0200
@@ -40,7 +40,7 @@
 #
 # Every automatically generated MANPATH includes these fields
 #
-MANPATH	/usr/man
+#MANPATH	/usr/man
 MANPATH	/usr/share/man
 MANPATH	/usr/local/man
 MANPATH	/usr/local/share/man
@@ -49,8 +49,8 @@
 # Uncomment if you want to include one of these by default
 #
 # MANPATH	/opt/*/man
-# MANPATH	/usr/lib/*/man
-# MANPATH	/usr/share/*/man
+MANPATH	/usr/lib/*/man
+MANPATH	/usr/share/*/man
 # MANPATH	/usr/kerberos/man
 #
 # Set up PATH to MANPATH mapping
@@ -75,7 +75,7 @@
 # NOAUTOPATH keeps man from automatically adding directories that look like
 # manual page directories to the path.
 #
-#NOAUTOPATH
+NOAUTOPATH
 #
 # NOCACHE keeps man from creating cache pages ("cat pages")
 # (generally one enables/disable cat page creation by creating/deleting
