diff -uNr arc-5.21/marc.c arc-5.21/marc.c
--- arc-5.21/marc.c	2003-10-31 03:22:36.000000000 +0100
+++ arc-5.21/marc.c	2005-05-21 10:40:22.000000000 +0200
@@ -48,7 +48,6 @@
 int nargs;			       /* number of arguments */
 char *arg[];			       /* pointers to arguments */
 {
-    char *makefnam();		       /* filename fixup routine */
     char *envfind();
 #if	!_MTS
     char *arctemp2, *mktemp();		/* temp file stuff */
