--- a/Include/object.h	2020-10-05 17:07:58.000000000 +0200
+++ b/Include/object.h	2020-10-06 12:08:17.444833331 +0200
@@ -127,7 +127,7 @@
 static inline int _Py_IS_TYPE(const PyObject *ob, const PyTypeObject *type) {
     return ob->ob_type == type;
 }
-#define Py_IS_TYPE(ob, type) _Py_IS_TYPE(_PyObject_CAST_CONST(ob), type)
+#define Py_IS_TYPE(ob, type) (_Py_IS_TYPE(_PyObject_CAST_CONST(ob), type))
 
 static inline void _Py_SET_REFCNT(PyObject *ob, Py_ssize_t refcnt) {
     ob->ob_refcnt = refcnt;
@@ -630,7 +630,7 @@
     return ((flags & feature) != 0);
 }
 
-#define PyType_FastSubclass(type, flag) PyType_HasFeature(type, flag)
+#define PyType_FastSubclass(type, flag) (PyType_HasFeature(type, flag))
 
 static inline int _PyType_Check(PyObject *op) {
     return PyType_FastSubclass(Py_TYPE(op), Py_TPFLAGS_TYPE_SUBCLASS);
