--- portsentry-1.1/portsentry.conf	Wed Jul 11 18:57:49 2001
+++ portsentry-1.1/portsentry.conf.blah	Fri Jul 20 20:51:39 2001
@@ -80,11 +80,11 @@
 ######################
 #
 # Hosts to ignore
-IGNORE_FILE="/usr/local/psionic/portsentry/portsentry.ignore"
+IGNORE_FILE="/var/portsentry/portsentry.ignore"
 # Hosts that have been denied (running history)
-HISTORY_FILE="/usr/local/psionic/portsentry/portsentry.history"
+HISTORY_FILE="/var/portsentry/portsentry.history"
 # Hosts that have been denied this session only (temporary until next restart)
-BLOCKED_FILE="/usr/local/psionic/portsentry/portsentry.blocked"
+BLOCKED_FILE="/var/portsentry/portsentry.blocked"
 
 ##############################
 # Misc. Configuration Options#
@@ -198,6 +198,9 @@
 #
 # ipchain support for Linux
 #KILL_ROUTE="/sbin/ipchains -I input -s $TARGET$ -j DENY -l"
+#
+# New iptables support for Linux kernel 2.4.x
+KILL_ROUTE="/usr/sbin/iptables -A INPUT -s $TARGET$ -j DROP"
 #
 # ipchain support for Linux (no logging of denied packets)
 #KILL_ROUTE="/sbin/ipchains -I input -s $TARGET$ -j DENY"
