--- ATLAS.o/makes/Make.lib	2011-05-14 13:33:24.000000000 -0400
+++ ATLAS/makes/Make.lib	2012-06-22 14:40:42.068787951 -0400
@@ -33,11 +33,11 @@
 
 ptshared :
 	- rm -f libatlas.so liblapack.so
-	$(MAKE) libatlas.so liblapack.so libptf77blas.so libf77blas.so \
+	$(MAKE) libatlas.so liblapack.so \
                 libptcblas.so libcblas.so liblapack.so
 shared :
 	- rm -f libatlas.so liblapack.so
-	$(MAKE) libatlas.so liblapack.so libf77blas.so libcblas.so liblapack.so
+	$(MAKE) libatlas.so liblapack.so libcblas.so liblapack.so
 cptshared :
 	- rm -f libatlas.so libclapack.so
 	$(MAKE) libatlas.so libclapack.so libptcblas.so libcblas.so
