--- archinfo_x86.c	2011-05-15 12:43:52.101001247 +0800
+++ archinfo_x86.c	2011-05-15 12:44:16.568001255 +0800
@@ -309,11 +309,9 @@ enum MACHTYPE Chip2Mach(enum CHIP chip,
       case 0x1A:
       case 0x1E:
       case 0x1F:
-         iret = IntCorei1;
-         break;
       case 0x25:
       case 0x2A:
-         iret = IntCorei2;
+         iret = IntCorei1;
 	 break;
       default:
          iret = MACHOther;
