From f1f966df75958b46ef7569c8bceb54e900c95dc9 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 24 Sep 2017 19:17:36 +0200
Subject: [PATCH] fix dependencies

---
 MonadCatchIO-transformers.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/MonadCatchIO-transformers.cabal b/MonadCatchIO-transformers.cabal
index 7bad57b..f80f8a9 100644
--- a/MonadCatchIO-transformers.cabal
+++ b/MonadCatchIO-transformers.cabal
@@ -23,9 +23,9 @@ Source-repository head
 
 Library
   Build-Depends:
-    base < 4.9,
+    base < 4.11,
     extensible-exceptions == 0.1.*,
-    transformers >= 0.2 && < 0.5,
+    transformers >= 0.2 && < 0.6,
     monads-tf == 0.1.*
   Exposed-Modules:
     Control.Monad.CatchIO
-- 
2.14.1

