diff -uNr kdepim-4.3.67svn1020033/kaddressbook/mainwidget.cpp kdepim-4.3.67svn1020033.cp/kaddressbook/mainwidget.cpp
--- kdepim-4.3.67svn1020033/kaddressbook/mainwidget.cpp	2009-09-05 08:58:08.000000000 +0200
+++ kdepim-4.3.67svn1020033.cp/kaddressbook/mainwidget.cpp	2009-09-18 21:38:48.000000000 +0200
@@ -55,7 +55,7 @@
 #include <ktoolbar.h>
 #include <kxmlguiwindow.h>
 
-#include "akonadi_next/entitytreeview.h"
+#include <akonadi/entitytreeview.h>
 #include "contactfiltermodel.h"
 #include "contactstreemodel.h"
 #include "contactswitcher.h"
diff -uNr kdepim-4.3.67svn1020033/kaddressbook/printing/printingwizard.cpp kdepim-4.3.67svn1020033.cp/kaddressbook/printing/printingwizard.cpp
--- kdepim-4.3.67svn1020033/kaddressbook/printing/printingwizard.cpp	2009-09-05 08:58:08.000000000 +0200
+++ kdepim-4.3.67svn1020033.cp/kaddressbook/printing/printingwizard.cpp	2009-09-18 21:53:01.000000000 +0200
@@ -41,7 +41,7 @@
 
 #include "globalcontactmodel.h"
 #include "contactstreemodel.h"
-#include <akonadi_next/entitytreeview.h>
+#include <akonadi/entitytreeview.h>
 
 // including the styles
 #include "detailledstyle.h"
diff -uNr kdepim-4.3.67svn1020033/kaddressbook/xxportmanager.cpp kdepim-4.3.67svn1020033.cp/kaddressbook/xxportmanager.cpp
--- kdepim-4.3.67svn1020033/kaddressbook/xxportmanager.cpp	2009-08-28 18:02:28.000000000 +0200
+++ kdepim-4.3.67svn1020033.cp/kaddressbook/xxportmanager.cpp	2009-09-18 21:51:41.000000000 +0200
@@ -35,7 +35,7 @@
 #include <QtGui/QItemSelectionModel>
 #include <QtGui/QWidget>
 
-#include "akonadi_next/entitytreeview.h"
+#include <akonadi/entitytreeview.h>
 
 
 XXPortManager::XXPortManager( QWidget *parent )
diff -uNr kdepim-4.3.67svn1020033/libkdepim/CMakeLists.txt kdepim-4.3.67svn1020033.cp/libkdepim/CMakeLists.txt
--- kdepim-4.3.67svn1020033/libkdepim/CMakeLists.txt	2009-09-04 13:31:21.000000000 +0200
+++ kdepim-4.3.67svn1020033.cp/libkdepim/CMakeLists.txt	2009-09-18 12:47:40.000000000 +0200
@@ -97,7 +97,7 @@
   ${KDE4_KMIME_LIBRARY}
   ${KDEPIMLIBS_KPIMTEXTEDIT_LIBS}
   ${KDEPIMLIBS_AKONADI_LIBS}
-  akonadi-contact
+  ${KDEPIMLIBS_AKONADI_CONTACT_LIBS}
 )
 
 target_link_libraries(kdepim LINK_INTERFACE_LIBRARIES
