diff -uNr kdepim-4.3.67svn1020033/kaddressbook/mainwidget.cpp kdepim-4.3.67svn1020033.cp/kaddressbook/mainwidget.cpp
--- kdepim-4.3.67svn1020033/kaddressbook/mainwidget.cpp	2009-09-05 08:58:08.000000000 +0200
+++ kdepim-4.3.67svn1020033.cp/kaddressbook/mainwidget.cpp	2009-09-18 21:38:48.000000000 +0200
@@ -55,7 +55,7 @@
 #include <ktoolbar.h>
 #include <kxmlguiwindow.h>
 
-#include "akonadi_next/entitytreeview.h"
+#include <akonadi/entitytreeview.h>
 #include "contactfiltermodel.h"
 #include "contactstreemodel.h"
 #include "contactswitcher.h"
