From 2fca6f12aebcae5e7645ed3ac2e741cd96072c95 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 21 Aug 2017 20:35:25 +0200
Subject: [PATCH] compability fix for ncurses 6

---
 config.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.sh b/config.sh
index a9f0081..b97a83f 100755
--- a/config.sh
+++ b/config.sh
@@ -110,7 +110,7 @@ check_pkg "stfl" || fail "stfl"
 if [ `uname -s` = "Darwin" ]; then
 	check_custom "ncurses5.4" "ncurses5.4-config" || fail "ncurses5.4"
 elif [ `uname -s` != "OpenBSD" ]; then
-	check_custom "ncursesw5" "ncursesw5-config" ||  fail "ncursesw"
+	check_custom "ncursesw6" "ncursesw6-config" ||  fail "ncursesw"
 fi
 check_ssl_implementation
 all_aboard_the_fail_boat
-- 
2.14.1

