From 72283d261f36ae09187d6ed3cec51ea08206e318 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 31 Jan 2018 11:24:18 +0100
Subject: [PATCH] fix dependency on time

---
 ekg.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ekg.cabal b/ekg.cabal
index 48f68dc..8297f5e 100644
--- a/ekg.cabal
+++ b/ekg.cabal
@@ -48,7 +48,7 @@ library
     snap-core < 1.1,
     snap-server < 1.1,
     text < 1.3,
-    time < 1.7,
+    time < 1.9,
     transformers < 0.6,
     unordered-containers < 0.3
 
-- 
2.15.1

