diff -Naur schroot-1.1.2.orig/sbuild/sbuild-dirstream.h schroot-1.1.2/sbuild/sbuild-dirstream.h
--- schroot-1.1.2.orig/sbuild/sbuild-dirstream.h	2008-04-12 16:36:55.945090331 -0700
+++ schroot-1.1.2/sbuild/sbuild-dirstream.h	2008-04-12 16:47:17.218315729 -0700
@@ -25,6 +25,7 @@
 #include <iostream>
 #include <deque>
 #include <string>
+#include <cstring>
 
 #include <sys/types.h>
 #include <dirent.h>
diff -Naur schroot-1.1.2.orig/sbuild/sbuild-lock.cc schroot-1.1.2/sbuild/sbuild-lock.cc
--- schroot-1.1.2.orig/sbuild/sbuild-lock.cc	2008-04-12 16:36:55.945090331 -0700
+++ schroot-1.1.2/sbuild/sbuild-lock.cc	2008-04-12 16:49:43.798073471 -0700
@@ -23,6 +23,7 @@
 
 #include <cerrno>
 #include <cstdlib>
+#include <cstring>
 
 #include <unistd.h>
 
