diff -Naur schroot-1.1.2.orig/bin/schroot/Makefile.in schroot-1.1.2/bin/schroot/Makefile.in
--- schroot-1.1.2.orig/bin/schroot/Makefile.in	2008-04-12 20:43:09.532220489 -0700
+++ schroot-1.1.2/bin/schroot/Makefile.in	2008-04-12 20:47:19.966471871 -0700
@@ -556,15 +556,6 @@
 	  echo " rm -f '$(DESTDIR)$(man5dir)/$$inst'"; \
 	  rm -f "$(DESTDIR)$(man5dir)/$$inst"; \
 	done
-install-pkgsysconfDATA: $(pkgsysconf_DATA)
-	@$(NORMAL_INSTALL)
-	test -z "$(pkgsysconfdir)" || $(mkdir_p) "$(DESTDIR)$(pkgsysconfdir)"
-	@list='$(pkgsysconf_DATA)'; for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  f=$(am__strip_dir) \
-	  echo " $(pkgsysconfDATA_INSTALL) '$$d$$p' '$(DESTDIR)$(pkgsysconfdir)/$$f'"; \
-	  $(pkgsysconfDATA_INSTALL) "$$d$$p" "$(DESTDIR)$(pkgsysconfdir)/$$f"; \
-	done
 
 uninstall-pkgsysconfDATA:
 	@$(NORMAL_UNINSTALL)
