diff -Naupr bins-1.1.29.orig/install.sh bins-1.1.29/install.sh
--- bins-1.1.29.orig/install.sh	2010-08-16 12:02:34.000000000 +0200
+++ bins-1.1.29/install.sh	2010-08-16 12:03:59.000000000 +0200
@@ -36,7 +36,7 @@ SHARE="$PREFIX/share"
 DATA="$SHARE/bins"
 GLADE="$DATA"
 LOCALE="$SHARE/locale"
-MAN="$PREFIX/man/man1"
+MAN="$SHARE/man/man1"
 
 cd `dirname $0`
 
@@ -59,19 +59,16 @@ modified something for personalization.
 Press Enter to go on or Crtl-C to stop now...
 !EoF
 
-read dummy
-
 install_binaries()
 {
   mkdir -p $BIN 2>/dev/null || true
   if [ -w $BIN ] ; then
     echo "Installing program files in $BIN"
     # bins_cleanupgallery anti_bins
-    cp bins bins_edit bins-edit-gui $BIN
+    cp bins bins_edit $BIN
     chmod a+rx $BIN/bins
     chmod a+rx $BIN/bins_edit
     chmod a+rx $BIN/bins_cleanupgallery
-    chmod a+rx $BIN/bins-edit-gui
     chmod a+rx $BIN/anti_bins
 #    perl -pi -e "s{/usr/local/share}{$SHARE}g" \
 #     $BIN/bins $BIN/bins_edit $BIN/bins-edit-gui
@@ -169,13 +166,6 @@ if [ "$?" -eq "1" ] ; then
 fi
 set -o errexit
 
-# additionnal files
-mkdir -p $GLADE 2>/dev/null || true
-if [ -w $GLADE ] ; then
-  echo "Installing additionnal files in $GLADE"
-  cp bins-edit-gui.glade $GLADE
-fi
-
 # man pages
 mkdir -p $MAN 2>/dev/null || true
 if [ -w $MAN ] ; then
@@ -203,9 +193,6 @@ if [ -w $LOCALE ] ; then
   do
     mkdir -p $LOCALE/$L/LC_MESSAGES 2>/dev/null || true
     cp intl/$L.mo $LOCALE/$L/LC_MESSAGES/bins.mo
-    if [ -f intl/gui-$L.mo ]; then
-      cp intl/gui-$L.mo $LOCALE/$L/LC_MESSAGES/bins-edit-gui.mo
-    fi
   done
 else
   cat <<!EoF  
@@ -223,7 +210,7 @@ fi
 set +o errexit
 # set directory paths in source code
 echo -n "setting directory paths in programs :"
-for FILE in "$BIN/bins" "$BIN/bins_edit" "$BIN/bins-edit-gui"
+for FILE in "$BIN/bins" "$BIN/bins_edit"
 do
   echo -n " $FILE"
   ed "$FILE" <<EoF 2>/dev/null
