--- programs/pluto/state.h.orig
+++ programs/pluto/state.h
@@ -213,10 +213,8 @@ struct state {
 	so_serial_t st_clonedfrom;              /* serial number of parent */
 	int st_usage;
 
-#ifdef XAUTH_HAVE_PAM
 	pthread_mutex_t xauth_mutex;            /* per state xauth_mutex */
 	pthread_t xauth_tid;                    /* per state XAUTH_RO thread id */
-#endif
 
 	bool st_ikev2;                          /* is this an IKEv2 state? */
 	bool st_rekeytov2;                      /* true if this IKEv1 is about
--- programs/pluto/xauth.c.orig
+++ programs/pluto/xauth.c
	2014-05-05 21:25:14.112883644 +0400
@@ -99,9 +99,7 @@
 	chunk_t name;
 	chunk_t password;
 	chunk_t connname;
-#ifdef XAUTH_HAVE_PAM
 	st_jbuf_t *ptr;
-#endif
 };
 
 #ifdef XAUTH_HAVE_PAM
