From 0d37e5a916c59004f76b8dbf08420250eeef06df Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 31 Dec 2018 11:19:33 +0100
Subject: [PATCH] fix Makefiles

---
 apps/Makefile   | 6 +++---
 source/Makefile | 7 +++----
 2 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/apps/Makefile b/apps/Makefile
index 053b9a6..261a3fd 100644
--- a/apps/Makefile
+++ b/apps/Makefile
@@ -23,9 +23,9 @@ BINDIR ?= $(PREFIX)/bin
 MANDIR ?= /usr/share/man/man1
 
 VERSION = 1.6.0
-CPPFLAGS += -MMD -MP -DVERSION=\"$(VERSION)\"
-CXXFLAGS += -O2  -ffast-math -Wall
-CXXFLAGS += -march=native
+CPPFLAGS += -MMD -MP -DVERSION=\"$(VERSION)\" -I../source
+CXXFLAGS += -Wall
+LDFLAGS += -L../source
 
 
 all:	zresample zretune zresample.1.gz zretune.1.gz
diff --git a/source/Makefile b/source/Makefile
index cd54ff5..9fad8ae 100644
--- a/source/Makefile
+++ b/source/Makefile
@@ -23,7 +23,7 @@
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 PREFIX ?= /usr/local
 INCDIR ?= $(PREFIX)/include
-LIBDIR ?= $(PREFIX)/lib$(SUFFIX)
+LIBDIR ?= $(PREFIX)/lib
 
 MAJVERS = 1
 MINVERS = 6.2
@@ -31,9 +31,8 @@ VERSION = $(MAJVERS).$(MINVERS)
 DISTDIR = zita-resampler-$(VERSION)
 
 
-CPPFLAGS += -I. -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
-CXXFLAGS += -Wall -fPIC -O2 -ffast-math
-CXXFLAGS += -march=native
+CPPFLAGS += -I. -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -DPIC
+CXXFLAGS += -Wall -fPIC
 LDFLAGS += 
 LDLIBS +=
 
-- 
2.20.1

