diff  -u  config/Makefile.in  config/Makefile.in
--- config/Makefile.in	Mon Dec 31 20:03:19 2001
+++ config/Makefile.in	Sun Jan  6 15:17:15 2002
@@ -129,5 +129,4 @@
 ORBIT_LIBS = @ORBIT_LIBS@
 PACKAGE = @PACKAGE@
-PAM_PREFIX = @PAM_PREFIX@
 PIXBUF_CFLAGS = @PIXBUF_CFLAGS@
 PIXBUF_LIBS = @PIXBUF_LIBS@
@@ -365,20 +364,4 @@
 	fi
 
-	system=`uname`; \
-	if test -f /usr/include/security/pam_appl.h; then \
-	  if test '!' -d $(DESTDIR)$(PAM_PREFIX)/pam.d; then \
-		$(mkinstalldirs) $(DESTDIR)$(PAM_PREFIX)/pam.d; \
-		chmod 755 $(DESTDIR)$(PAM_PREFIX)/pam.d; \
-	   fi; \
-	   if test $$system = Linux && test '!' -f $(DESTDIR)$(PAM_PREFIX)/pam.d/gdm; then \
-		$(INSTALL_DATA) gdm $(DESTDIR)$(PAM_PREFIX)/pam.d/gdm; \
-	   fi; \
-	   if test $$system = Linux && test '!' -f $(DESTDIR)$(PAM_PREFIX)/pam.d/gdm-autologin; then \
-		$(INSTALL_DATA) gdm-autologin $(DESTDIR)$(PAM_PREFIX)/pam.d/gdm-autologin; \
-	   fi; \
-	   if test $$system = SunOS; then \
-		echo "Please add PAM authentication for gdm and gdm-autologin in $(PAM_PREFIX)/pam.conf!"; \
-	   fi; \
-	fi
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
