From dcca82715c4d3697cfec8763097c67ac4b8edd1c Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 17 Oct 2016 23:06:09 +0200
Subject: [PATCH] fix compile error with gcc 6

---
 src/engine.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/engine.c b/src/engine.c
index 07c3597..2dc5f2b 100644
--- a/src/engine.c
+++ b/src/engine.c
@@ -211,7 +211,7 @@ evaluateTriggers (Display* d)
   if (lockerPid)
   {
 #if !defined (UTEKV) && !defined (SYSV) && !defined (SVR4)
-    union wait  status;      /* childs process status */
+    int  status;      /* childs process status */
 #else /* !UTEKV && !SYSV && !SVR4 */
     int         status = 0;  /* childs process status */
 #endif /* !UTEKV && !SYSV && !SVR4 */
-- 
2.10.1

