Index: dvbcut-0.5.4/src/playaudio.cpp
===================================================================
--- dvbcut-0.5.4.orig/src/playaudio.cpp
+++ dvbcut-0.5.4/src/playaudio.cpp
@@ -38,7 +38,7 @@ void playaudio(const void *data, uint32_
   const uint8_t *d=(const uint8_t*)data;
 
   while (len>0) {
-    int16_t samples[MIN_BUFFER_SAMPLES >? avcc->frame_size];
+    int16_t samples[MIN_BUFFER_SAMPLES > avcc->frame_size ? MIN_BUFFER_SAMPLES : avcc->frame_size ];
     int frame_size;
 
     int bytesDecoded=avcodec_decode_audio(avcc,samples,&frame_size,(uint8_t*)d,len);
