diff -Naur dvbcut-newffmpeg/SConstruct dvbcut-SConstruct/SConstruct
--- dvbcut-newffmpeg/SConstruct	2007-03-30 22:55:19.000000000 +0200
+++ dvbcut-SConstruct/SConstruct	2007-03-30 23:01:22.000000000 +0200
@@ -46,6 +46,8 @@
 else:
   env.Append(CCFLAGS=['-O3','-Wall'])
 
+env.Append(CCFLAGS=['-I/usr/include/qt'])
+
 env.Replace(CXXFILESUFFIX=".cpp")
 
 env.Append(CPPDEFINES=[("_FILE_OFFSET_BITS", "64"), "_LARGEFILE_SOURCE"])
