Index: dvbcut-0.5.4/src/avframe.h
===================================================================
--- dvbcut-0.5.4.orig/src/avframe.h
+++ dvbcut-0.5.4/src/avframe.h
@@ -19,7 +19,9 @@
 #ifndef _DVBCUT_AVFRAME_H
 #define _DVBCUT_AVFRAME_H
 
+extern "C" {
 #include <ffmpeg/avcodec.h>
+}
 
 class QImage;
 
Index: dvbcut-0.5.4/src/lavfmuxer.cpp
===================================================================
--- dvbcut-0.5.4.orig/src/lavfmuxer.cpp
+++ dvbcut-0.5.4/src/lavfmuxer.cpp
@@ -16,7 +16,9 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+extern "C" {
 #include <ffmpeg/avformat.h>
+}
 #include <string.h>
 #include <utility>
 #include <list>
Index: dvbcut-0.5.4/src/lavfmuxer.h
===================================================================
--- dvbcut-0.5.4.orig/src/lavfmuxer.h
+++ dvbcut-0.5.4/src/lavfmuxer.h
@@ -19,7 +19,9 @@
 #ifndef _DVBCUT_LAVFMUXER_H
 #define _DVBCUT_LAVFMUXER_H
 
+extern "C" {
 #include <ffmpeg/avformat.h>
+}
 #include "mpgfile.h"
 #include "muxer.h"
 
Index: dvbcut-0.5.4/src/main.cpp
===================================================================
--- dvbcut-0.5.4.orig/src/main.cpp
+++ dvbcut-0.5.4/src/main.cpp
@@ -27,7 +27,9 @@
 #endif // HAVE_LIB_AO
 
 #include <qapplication.h>
+extern "C" {
 #include <ffmpeg/avformat.h>
+}
 #include <qimage.h>
 #include <qsettings.h>
 #include "dvbcut.h"
Index: dvbcut-0.5.4/src/mpgfile.h
===================================================================
--- dvbcut-0.5.4.orig/src/mpgfile.h
+++ dvbcut-0.5.4/src/mpgfile.h
@@ -19,7 +19,10 @@
 #ifndef _DVBCUT_MPGFILE_H
 #define _DVBCUT_MPGFILE_H
 
+extern "C" {
 #include <ffmpeg/avcodec.h>
+}
+
 #include <string>
 #include <vector>
 #include <list>
Index: dvbcut-0.5.4/src/playaudio.cpp
===================================================================
--- dvbcut-0.5.4.orig/src/playaudio.cpp
+++ dvbcut-0.5.4/src/playaudio.cpp
@@ -19,7 +19,10 @@
 #ifdef HAVE_LIB_AO
 #include <stdio.h>
 #include <ao/ao.h>
+extern "C" {
 #include <ffmpeg/avcodec.h>
+}
+
 #include "playaudio.h"
 
 #define MIN_BUFFER_SAMPLES (1536*6)
Index: dvbcut-0.5.4/src/psfile.cpp
===================================================================
--- dvbcut-0.5.4.orig/src/psfile.cpp
+++ dvbcut-0.5.4/src/psfile.cpp
@@ -19,7 +19,10 @@
 #include "psfile.h"
 #include "streamhandle.h"
 #include "stream.h"
+
+extern "C" {
 #include <ffmpeg/avcodec.h>
+}
 
 psfile::psfile(const std::string &filename, inbuffer &b, int initial_offset)
     : mpgfile(filename, b, initial_offset)
Index: dvbcut-0.5.4/src/streamdata.h
===================================================================
--- dvbcut-0.5.4.orig/src/streamdata.h
+++ dvbcut-0.5.4/src/streamdata.h
@@ -21,7 +21,9 @@
 
 #include <string>
 #include <list>
+extern "C" {
 #include <ffmpeg/avformat.h>
+}
 
 #include "port.h"
 #include "tsfile.h"
Index: dvbcut-0.5.4/src/tsfile.cpp
===================================================================
--- dvbcut-0.5.4.orig/src/tsfile.cpp
+++ dvbcut-0.5.4/src/tsfile.cpp
@@ -22,7 +22,9 @@
 #include <list>
 #include <utility>
 
+extern "C" {
 #include <ffmpeg/avcodec.h>
+}
 
 tsfile::tsfile(const std::string &filename, inbuffer &b, int initial_offset)
     : mpgfile(filename, b, initial_offset)
