Index: dvbcut-0.5.3/SConstruct
===================================================================
--- dvbcut-0.5.3.orig/SConstruct
+++ dvbcut-0.5.3/SConstruct
@@ -101,6 +101,7 @@ if (ffmpegpath!='/usr'): 
   env.Append(LIBPATH=os.path.join(str(ffmpegpath),'lib'))
 env.Append(LIBS=['avformat','avcodec','avutil'])
   
+env.Append(CPPDEFINES="__STDC_CONSTANT_MACROS")
 ###### WORK
 
 env.bin_targets=[]
Index: dvbcut-0.5.3/src/psfile.cpp
===================================================================
--- dvbcut-0.5.3.orig/src/psfile.cpp
+++ dvbcut-0.5.3/src/psfile.cpp
@@ -100,7 +100,7 @@ psfile::psfile(const std::string &filena
       streamnumber[sid]=audiostream(audiostreams);
       stream *S=&s[audiostream(audiostreams++)];
       S->id=sid;
-      S->dec=&ac3_decoder;
+      S->dec=&liba52_decoder;
       S->type=streamtype::ac3audio;
       if (audiostreams>=MAXAUDIOSTREAMS)
         break;
Index: dvbcut-0.5.3/src/tsfile.cpp
===================================================================
--- dvbcut-0.5.3.orig/src/tsfile.cpp
+++ dvbcut-0.5.3/src/tsfile.cpp
@@ -73,7 +73,7 @@ tsfile::tsfile(const std::string &filena
     stream *S=&s[audiostream(audiostreams++)];
     S->id=it->second;
     if (it->first==0xbd) {
-      S->dec=&ac3_decoder;
+      S->dec=&liba52_decoder;
       S->enc=&ac3_encoder;
       S->type=streamtype::ac3audio;
       } else {
