diff -Naupr dvbcut-0.6.0.orig/src/buffer.cpp dvbcut-0.6.0/src/buffer.cpp
--- dvbcut-0.6.0.orig/src/buffer.cpp	2010-04-14 13:03:04.202830292 +0200
+++ dvbcut-0.6.0/src/buffer.cpp	2010-04-14 13:03:26.907330296 +0200
@@ -26,7 +26,7 @@
 #include <unistd.h>
 #include <cstring>
 #include <cstdlib>
-// #include <cstdio>
+#include <stdio.h>
 #include <stdint.h>
 #include <cassert>
 
diff -Naupr dvbcut-0.6.0.orig/src/mpegmuxer.cpp dvbcut-0.6.0/src/mpegmuxer.cpp
--- dvbcut-0.6.0.orig/src/mpegmuxer.cpp	2010-04-14 13:03:04.202830292 +0200
+++ dvbcut-0.6.0/src/mpegmuxer.cpp	2010-04-14 13:03:34.703070639 +0200
@@ -26,6 +26,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <cassert>
+#include <stdio.h>
 #include "mpegmuxer.h"
 
 #ifndef O_BINARY
