--- dvbcut-0.6.0/src/main.cpp.orig	2011-02-16 21:39:05.229170751 +0100
+++ dvbcut-0.6.0/src/main.cpp	2011-02-16 21:40:07.274329059 +0100
@@ -171,7 +171,7 @@
       return 1;
     }
 
-    index::index idx(*mpg);
+    class index idx(*mpg);
     int pics = idx.generate();
     if (pics <= 0) {
       fprintf(stderr, "%s: %s: %s\n", argv0, mpgfilename.c_str(),
--- dvbcut-0.6.0/src/mpgfile.h.orig	2011-02-16 21:42:36.353918944 +0100
+++ dvbcut-0.6.0/src/mpgfile.h	2011-02-16 21:40:36.253804316 +0100
@@ -49,7 +49,7 @@
   stream s[MAXAVSTREAMS];
   int videostreams, audiostreams;
   int initialoffset;
-  index::index idx;
+  class index idx;
   int pictures;
 
   mpgfile(inbuffer &b, int initial_offset);
