From 58873a49460869f1ece1eca099f67ec4afbe3e9b Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 19 May 2021 21:16:05 +0200
Subject: [PATCH] Build a shared library

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1fc7991..6142f1c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,7 +51,7 @@ file(GLOB_RECURSE TVDEMOSRC "${CMAKE_CURRENT_SOURCE_DIR}/examples/tvdemo/*.cpp")
 file(GLOB_RECURSE TVDIRSRC "${CMAKE_CURRENT_SOURCE_DIR}/examples/tvdir/*.cpp")
 file(GLOB_RECURSE TVHELPSRC "${CMAKE_CURRENT_SOURCE_DIR}/examples/tvhelp/*.cpp")
 
-add_library(tvision STATIC ${TVSOURCE})
+add_library(tvision SHARED ${TVSOURCE})
 add_executable(hello ${HELLOSRC})
 add_executable(tvedit ${TVEDITSRC})
 add_executable(tvdemo ${TVDEMOSRC})
-- 
2.31.1

