--- libtransmission/natpmp.c.orig	2012-02-15 06:00:19.000000000 +0400
+++ libtransmission/natpmp.c	2012-02-21 01:34:04.292023453 +0400
@@ -120,7 +120,7 @@
 
     if( is_enabled && ( nat->state == TR_NATPMP_DISCOVER ) )
     {
-        int val = initnatpmp( &nat->natpmp );
+        int val = initnatpmp( &nat->natpmp, 0, 0 );
         logVal( "initnatpmp", val );
         val = sendpublicaddressrequest( &nat->natpmp );
         logVal( "sendpublicaddressrequest", val );
