--- third-party/libutp/utp.h.orig	2012-09-26 21:02:52.000000000 +0400
+++ third-party/libutp/utp.h	2012-09-30 12:20:52.264401960 +0400
@@ -111,10 +111,12 @@
 // Process an ICMP received UDP packet.
 bool UTP_HandleICMP(const byte* buffer, size_t len, const struct sockaddr *to, socklen_t tolen);
 
+#ifdef WITH_UTP
 // Write bytes to the uTP socket.
 // Returns true if the socket is still writable.
 bool UTP_Write(struct UTPSocket *socket, size_t count);
 
+#endif
 // Notify the uTP socket of buffer drain
 void UTP_RBDrained(struct UTPSocket *socket);
 
