diff -uNr sfs-cvs.old/adb/indexNode.h sfs-cvs/adb/indexNode.h
--- sfs-cvs.old/adb/indexNode.h	2006-05-29 12:40:31.000000000 +0200
+++ sfs-cvs/adb/indexNode.h	2006-05-29 12:52:41.000000000 +0200
@@ -71,7 +71,7 @@
   
  public:
   indexNode(nodeStorage *buf, btree* tree, int size);
-  indexNode::~indexNode();
+  ~indexNode();
   
   void insert(record *item, int policy, callback<void, int>::ref cb);
   void remove(void *key, bSize_t len);
diff -uNr sfs-cvs.old/adb/nodeBuf.h sfs-cvs/adb/nodeBuf.h
--- sfs-cvs.old/adb/nodeBuf.h	2006-05-29 12:40:31.000000000 +0200
+++ sfs-cvs/adb/nodeBuf.h	2006-05-29 12:42:10.000000000 +0200
@@ -87,7 +87,7 @@
   
   int idToOffset(nodeID_t id);
   nodeID_t blockToID(blockID_t b);
-  blockID_t nodeBuf::IDToBlock(nodeID_t n);
+  blockID_t IDToBlock(nodeID_t n);
  
   void verifySpace(bSize_t nsize);
   void initNode(void* buf, char, blockID_t, nodeID_t, bSize_t);
diff -uNr sfs-cvs.old/adb/record.h sfs-cvs/adb/record.h
--- sfs-cvs.old/adb/record.h	2006-05-29 12:40:31.000000000 +0200
+++ sfs-cvs/adb/record.h	2006-05-29 12:41:47.000000000 +0200
@@ -35,7 +35,7 @@
 
  public:
   record(void *key, int keyLen, void *value, int valueLen);
-  record::record(record *rec);
+  record(record *rec);
   ~record();
   void dealloc();
   void *getKey(bSize_t *len);
