--- configure.orig	2009-06-11 11:29:58.000000000 +0200
+++ configure	2009-06-11 11:28:50.000000000 +0200
@@ -5714,8 +5714,11 @@
 
 fi
 
-if test x"$modules" = xauto
+if test x"$modules" = xdisabled
 then
+	echo "Modules disabled, not checking for libltdl."
+else
+	have_ltdl=no
 
 { $as_echo "$as_me:$LINENO: checking for ANSI C header files" >&5
 $as_echo_n "checking for ANSI C header files... " >&6; }
@@ -6168,8 +6171,7 @@
 fi
 if test "x$ac_cv_header_ltdl_h" = x""yes; then
 
-				modules=enabled
-				LIBS="-lltdl $LIBS"
+				have_ltdl=yes
 
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_LTDL 1
@@ -6178,7 +6180,6 @@
 
 else
 
-				modules=disabled
 				{ $as_echo "$as_me:$LINENO: WARNING: Found ltdl library but NOT the header file on your system" >&5
 $as_echo "$as_me: WARNING: Found ltdl library but NOT the header file on your system" >&2;}
 
@@ -6188,13 +6189,23 @@
 
 else
 
-			modules=disabled
 			{ $as_echo "$as_me:$LINENO: WARNING: ltdl library not found" >&5
 $as_echo "$as_me: WARNING: ltdl library not found" >&2;}
 
 
 fi
 
+	if test x"$modules" = xenabled -a x"$have_ltdl" = xno; then
+		{ { $as_echo "$as_me:$LINENO: error: Modules enabled but no libltdl found! This will not work..." >&5
+$as_echo "$as_me: error: Modules enabled but no libltdl found! This will not work..." >&2;}
+   { (exit 1); exit 1; }; }
+	fi
+	if test x"$modules" = xauto -a x"$have_ltdl" = xyes; then
+		modules=enabled
+		echo "We found libltdl: Modules enabled."
+	else
+		modules=disabled
+	fi
 fi
 
 if test x"$modules" = xdisabled
@@ -6998,13 +7009,13 @@
 else
   lt_cv_nm_interface="BSD nm"
   echo "int some_variable = 0;" > conftest.$ac_ext
-  (eval echo "\"\$as_me:7001: $ac_compile\"" >&5)
+  (eval echo "\"\$as_me:7012: $ac_compile\"" >&5)
   (eval "$ac_compile" 2>conftest.err)
   cat conftest.err >&5
-  (eval echo "\"\$as_me:7004: $NM \\\"conftest.$ac_objext\\\"\"" >&5)
+  (eval echo "\"\$as_me:7015: $NM \\\"conftest.$ac_objext\\\"\"" >&5)
   (eval "$NM \"conftest.$ac_objext\"" 2>conftest.err > conftest.out)
   cat conftest.err >&5
-  (eval echo "\"\$as_me:7007: output\"" >&5)
+  (eval echo "\"\$as_me:7018: output\"" >&5)
   cat conftest.out >&5
   if $GREP 'External.*some_variable' conftest.out > /dev/null; then
     lt_cv_nm_interface="MS dumpbin"
@@ -8206,7 +8217,7 @@
   ;;
 *-*-irix6*)
   # Find out which ABI we are using.
-  echo '#line 8209 "configure"' > conftest.$ac_ext
+  echo '#line 8220 "configure"' > conftest.$ac_ext
   if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
   (eval $ac_compile) 2>&5
   ac_status=$?
@@ -9509,11 +9520,11 @@
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9512: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9523: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:9516: \$? = $ac_status" >&5
+   echo "$as_me:9527: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -9848,11 +9859,11 @@
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9851: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9862: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:9855: \$? = $ac_status" >&5
+   echo "$as_me:9866: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -9953,11 +9964,11 @@
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9956: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9967: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:9960: \$? = $ac_status" >&5
+   echo "$as_me:9971: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -10008,11 +10019,11 @@
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:10011: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:10022: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:10015: \$? = $ac_status" >&5
+   echo "$as_me:10026: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -12808,7 +12819,7 @@
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 12811 "configure"
+#line 12822 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -12904,7 +12915,7 @@
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 12907 "configure"
+#line 12918 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -23336,6 +23347,11 @@
 
 
 
+# Do that here to prevent the above tests usign -lltdl...
+if test "x$modules" = xenabled; then
+  LIBS="-lltdl $LIBS"
+fi
+
 ac_config_files="$ac_config_files Makefile libmpg123.pc doc/Makefile src/Makefile src/output/Makefile src/libmpg123/Makefile src/libmpg123/mpg123.h src/libmpg123/libmpg123.sym"
 
 
