--- mozilla/configure.orig	2011-11-21 14:29:14.000000000 +0400
+++ mozilla/configure	2011-12-10 16:04:46.337188291 +0400
@@ -14788,6 +14788,7 @@
 MOZ_FEEDS=1
 MOZ_INSTALLER=1
 MOZ_JSDEBUGGER=1
+MOZ_MAIL_NEWS=
 MOZ_AUTH_EXTENSION=1
 MOZ_OGG=1
 MOZ_RAW=
@@ -17714,8 +17715,17 @@
 fi
 
 
-
-
+# Check whether --enable-mailnews or --disable-mailnews was given.
+if test "${enable_mailnews+set}" = set; then
+  enableval="$enable_mailnews"
+  if test "$enableval" = "no"; then
+    MOZ_MAIL_NEWS=
+  elif test "$enableval" = "yes"; then
+    MOZ_MAIL_NEWS=1 
+  else
+    { echo "configure: error: Option, mailnews, does not take an argument ($enableval)." 1>&2; exit 1; }
+  fi
+fi
 
 
 # Check whether --enable-crypto or --disable-crypto was given.
@@ -27643,6 +27653,7 @@
     (''' RCC ''', r''' $RCC ''')
     (''' MOZ_X11 ''', r''' $MOZ_X11 ''')
     (''' MOZ_UI_LOCALE ''', r''' $MOZ_UI_LOCALE ''')
+    (''' MOZ_MAIL_NEWS ''', r''' $MOZ_MAIL_NEWS ''')
     (''' MOZ_OFFICIAL_BRANDING ''', r''' $MOZ_OFFICIAL_BRANDING ''')
     (''' MOZ_BRANDING_DIRECTORY ''', r''' $MOZ_BRANDING_DIRECTORY ''')
     (''' MOZ_DISTRIBUTION_ID ''', r''' $MOZ_DISTRIBUTION_ID ''')
