--- mozilla/configure.orig	2014-10-23 16:46:01.833461054 +0400
+++ mozilla/configure	2014-10-23 16:50:26.601734897 +0400
@@ -16258,6 +16258,7 @@
 MOZ_OFFICIAL_BRANDING=
 MOZ_FEEDS=1
 MOZ_WEBAPP_RUNTIME=
+MOZ_MAIL_NEWS=
 MOZ_AUTH_EXTENSION=1
 MOZ_RAW=
 MOZ_VORBIS=
@@ -18758,8 +18759,17 @@
 fi
 
 
-
-
+# Check whether --enable-mailnews or --disable-mailnews was given.
+if test "${enable_mailnews+set}" = set; then
+  enableval="$enable_mailnews"
+  if test "$enableval" = "no"; then
+    MOZ_MAIL_NEWS=
+  elif test "$enableval" = "yes"; then
+    MOZ_MAIL_NEWS=1 
+  else
+    { echo "configure: error: Option, mailnews, does not take an argument ($enableval)." 1>&2; exit 1; }
+  fi
+fi
 
 
 
@@ -29173,6 +29183,7 @@
     (''' RCC ''', r''' $RCC ''')
     (''' MOZ_X11 ''', r''' $MOZ_X11 ''')
     (''' MOZ_UI_LOCALE ''', r''' $MOZ_UI_LOCALE ''')
+    (''' MOZ_MAIL_NEWS ''', r''' $MOZ_MAIL_NEWS ''')
     (''' MOZ_OFFICIAL_BRANDING ''', r''' $MOZ_OFFICIAL_BRANDING ''')
     (''' MOZ_BRANDING_DIRECTORY ''', r''' $MOZ_BRANDING_DIRECTORY ''')
     (''' MOZ_DISTRIBUTION_ID ''', r''' $MOZ_DISTRIBUTION_ID ''')
