--- trunk/koffice/filters/kword/msword-odf/graphicshandler.cpp	2009/02/28 21:15:35	933412
+++ trunk/koffice/filters/kword/msword-odf/graphicshandler.cpp	2009/03/06 10:17:07	935799
@@ -66,7 +66,7 @@
         wvWare::U8* buf = new wvWare::U8[IMG_BUF_SIZE];
         size_t n = reader.read( buf, qMin( len, IMG_BUF_SIZE ) );
         Q_LONG n1 = m_store->write( (const char*)buf, n );
-        kDebug(30513) << "n=" << n << ", n1=" << n1 << "; buf contains " << (int) buf;
+        kDebug(30513) << "n=" << n << ", n1=" << n1 << "; buf contains " << (void*) buf;
         len -= n;
         delete [] buf;
         //error checking
@@ -109,7 +109,7 @@
         }
         //size_t n = reader.read( buf, qMin( len, IMG_BUF_SIZE ) );
         Q_LONG n1 = m_store->write( (const char*)buf, n );
-        kDebug(30513) << "n=" << n << ", n1=" << n1 << "; buf contains " << (int) buf;
+        kDebug(30513) << "n=" << n << ", n1=" << n1 << "; buf contains " << (void*) buf;
         len -= n;
         delete [] buf;
         //error checking
