From 024b2472fd71e7fed13b77142b2ca965d6ed65c6 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Fri, 10 Mar 2017 15:15:23 +0100
Subject: [PATCH] bypass broken cmake

---
 qt5/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/qt5/CMakeLists.txt b/qt5/CMakeLists.txt
index 6bd7a6be..cb7c5574 100644
--- a/qt5/CMakeLists.txt
+++ b/qt5/CMakeLists.txt
@@ -1,5 +1,5 @@
 set(CMAKE_AUTOMOC ON)
 
 add_subdirectory(src)
-add_subdirectory(tests)
-add_subdirectory(demos)
+#add_subdirectory(tests)
+#add_subdirectory(demos)
-- 
2.14.1

