From 263758e9844e24a37f90801d8f3a7c2838ee6a47 Mon Sep 17 00:00:00 2001
From: Pavel Vinogradov <public@sourcemage.org>
Date: Tue, 26 Sep 2017 21:32:38 -0400
Subject: [PATCH] Set CMP0002 to OLD in cmake/modules/PopplerMacros.cmake

---
 cmake/modules/PopplerMacros.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/modules/PopplerMacros.cmake b/cmake/modules/PopplerMacros.cmake
index ba4bceb3..ca1a1d8b 100644
--- a/cmake/modules/PopplerMacros.cmake
+++ b/cmake/modules/PopplerMacros.cmake
@@ -15,6 +15,7 @@ macro(POPPLER_ADD_TEST exe build_flag)
   if(NOT build_test AND NOT MSVC_IDE)
     get_directory_property(_buildtestsAdded BUILDTESTS_ADDED)
     if(NOT _buildtestsAdded)
+      cmake_policy(SET CMP0002 OLD)
       add_custom_target(buildtests)
       set_directory_properties(PROPERTIES BUILDTESTS_ADDED TRUE)
     endif(NOT _buildtestsAdded)
-- 
2.14.2

