diff --git a/configure b/configure
index 5b0bf69..a0f07e9 100755
--- a/configure
+++ b/configure
@@ -568,10 +568,10 @@ MFLAGS=
 MAKEFLAGS=
 
 # Identity of this package.
-PACKAGE_NAME='transmission'
-PACKAGE_TARNAME='transmission'
+PACKAGE_NAME='transmission-i2p'
+PACKAGE_TARNAME='transmission-i2p'
 PACKAGE_VERSION='2.82'
-PACKAGE_STRING='transmission 2.82'
+PACKAGE_STRING='transmission-i2p 2.82'
 PACKAGE_BUGREPORT='http://trac.transmissionbt.com/newticket'
 PACKAGE_URL=''
 
diff --git a/libtransmission/platform.c b/libtransmission/platform.c
index 59c8d5e..4a4cd6e 100644
--- a/libtransmission/platform.c
+++ b/libtransmission/platform.c
@@ -547,7 +547,7 @@ tr_getWebClientDir (const tr_session * session UNUSED)
           /* walk through the candidates & look for a match */
           for (l=candidates; l; l=l->next)
             {
-              char * path = tr_buildPath (l->data, "transmission", "web", NULL);
+              char * path = tr_buildPath (l->data, "transmission-i2p", "web", NULL);
               const int found = isWebClientDir (path);
               if (found)
                 {
