diff --git a/libtransmission/platform-quota.c b/libtransmission/platform-quota.c
index 1b8a33b..efdccb0 100644
--- a/libtransmission/platform-quota.c
+++ b/libtransmission/platform-quota.c
@@ -241,7 +241,7 @@ getquota (const char * device)
       spaceused = (int64_t) dq.dqb_curbytes;
 #elif defined(__UCLIBC__)
       spaceused = (int64_t) btodb(dq.dqb_curblocks);
-#elif defined(__sun) || (_LINUX_QUOTA_VERSION < 2)
+#elif defined(__sun) || (defined(_LINUX_QUOTA_VERSION) && _LINUX_QUOTA_VERSION < 2)
       spaceused = (int64_t) dq.dqb_curblocks >> 1;
 #else
       spaceused = btodb(dq.dqb_curspace);
