diff -Nau pks-0.9.6a/kd_since.c pks-0.9.6b/kd_since.c
--- pks-0.9.6a/kd_since.c      2003-04-08 12:14:27.000000000 +0000
+++ pks-0.9.6b/kd_since.c      2003-04-08 12:17:53.000000000 +0000
@@ -123,10 +123,10 @@
    ows.err = err;
    ows.append = kd_keys_elem_marshall;
 
-   for (i=0; i<entries.len; i+=12)
+   for (i=0; i<=entries.len; i+=12)
       if (i && memcmp((void *) (entries.buf+i-12),
 		      (void *) (entries.buf+i), 12) &&
-	  (!kd_output_wde((void *) (entries.buf+i), (void *) &ows)))
+	  (!kd_output_wde((void *) (entries.buf+i-12), (void *) &ows)))
 	 return(0);
 
    xbuffer_free(&entries);
diff -Nau pks-0.9.6a/globals.h pks-0.9.6b/globals.h
--- pks-0.9.6a/globals.h      2003-04-02 06:09:33.000000000 +0000
+++ pks-0.9.6b/globals.h      2003-04-08 13:01:08.000000000 +0000
@@ -13,7 +13,7 @@
    "42.17.2+magicfeature", or "42.17.3+joe".  Don't remove any
    existing modifiers. */
 
-#define PKS_VERSION "0.9.6"
+#define PKS_VERSION "0.9.6+smgl"
 
 #ifdef DEBUG
 extern int debug;
