--- libexif-gtk/Makefile.in.orig	2007-11-07 19:22:57.000000000 +0000
+++ libexif-gtk/Makefile.in		2007-11-07 19:24:00.000000000 +0000
@@ -251,8 +251,7 @@
 	-I$(top_srcdir)/intl				\
 	-I$(top_srcdir)/gtk-extensions			\
 	$(LIBEXIF_GTK_CFLAGS)				\
-	-DG_LOG_DOMAIN=\"libexif\"			\
-	-DGTK_DISABLE_DEPRECATED
+	-DG_LOG_DOMAIN=\"libexif\"			
 
 lib_LTLIBRARIES = libexif-gtk.la
 libexif_gtkincludedir = $(includedir)/libexif-gtk
