--- src/session.h.orig	2007-11-22 21:04:32.000000000 +0400
+++ src/session.h	2007-11-22 20:50:29.000000000 +0400
@@ -2,6 +2,8 @@
 #ifndef _GNM_SESSION_H_
 # define _GNM_SESSION_H_
 
+#include <glib.h>
+
 G_BEGIN_DECLS
 
 void gnm_session_init (char const *argv0);

--- src/session.c.orig	2007-11-22 21:04:28.000000000 +0400
+++ src/session.c	2007-11-22 20:50:03.000000000 +0400
@@ -23,7 +23,6 @@
 #ifdef GNM_WITH_GNOME
 #include "gnumeric.h"
 #include <gtk/gtk.h>
-#include <glib.h>
 #include "session.h"
 #include "application.h"
 #include <workbook.h>

