--- kdenetwork-3.5.10.orig/kopete/protocols/oscar/liboscar/rtf.cc	2005-09-10 10:20:23.000000000 +0200
+++ kdenetwork-3.5.10/kopete/protocols/oscar/liboscar/rtf.cc	2010-12-28 14:42:14.000000000 +0100
@@ -2131,7 +2131,7 @@
 
         FontDef& def = p->fonts[m_nFont-1];
 
-        char *pp = strchr(str, ';');
+        const char *pp = strchr(str, ';');
         unsigned size;
         if (pp != NULL)
            size = (pp - str);
--- kdenetwork-3.5.10.orig/kopete/protocols/groupwise/libgroupwise/rtf.cc	2005-09-10 10:20:20.000000000 +0200
+++ kdenetwork-3.5.10/kopete/protocols/groupwise/libgroupwise/rtf.cc	2010-12-28 14:37:54.000000000 +0100
@@ -2235,7 +2235,7 @@
 
         FontDef& def = p->fonts[m_nFont-1];
 
-        char *pp = strchr(str, ';');
+        const char *pp = strchr(str, ';');
         unsigned size;
         if (pp != NULL)
            size = (pp - str);
