From 3111b1b11330feb8b81d163bce67c5579a289796 Mon Sep 17 00:00:00 2001
From: Johnathan Jenkins <john@nixheads.co.uk>
Date: Fri, 4 Nov 2016 09:53:36 -0700
Subject: [PATCH] Fix compiling against libressl

---
 lib/mega.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/mega.c b/lib/mega.c
index 6ff99df..b3ddcc1 100644
--- a/lib/mega.c
+++ b/lib/mega.c
@@ -803,7 +803,7 @@ static gboolean rsa_key_gen(rsa_key* k)
     return FALSE;
   }
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if OPENSSL_VERSION_NUMBER >= 0x101000000L && !defined(LIBRESSL_VERSION_NUMBER)
   RSA_get0_key(key, &k->m, &k->e, &k->d);
   RSA_get0_factors(key, &k->q, &k->p);
   RSA_get0_crt_params(key, NULL, NULL, &k->u);
