--- orig.PL	2007-03-15 12:35:00.000000000 -0500
+++ Makefile.PL	2007-03-15 12:35:11.000000000 -0500
@@ -6,7 +6,7 @@
 requires('DBI');
 requires('Want');
 requires('Encode');
-requires('Class::ReturnValue', 0.40);
+#requires('Class::ReturnValue', 0.40);
 requires('Cache::Simple::TimedExpiry' => '0.21');
 requires('Clone');
 build_requires('Test::More' => 0.52);
@@ -17,7 +17,7 @@
 		'capitalization' => '0.03',
 	],
 	'Schema generation' => [
-	  -default => 1,
+	  -default => 0,
 	  'DBIx::DBSchema' => '',
 	  'Class::Accessor' => '',
 	],
