--- ../gtoaster/Makefile.in	2002-09-01 09:41:33.000000000 +0200
+++ Makefile.in	2003-12-01 20:53:29.000000000 +0100
@@ -179,13 +179,6 @@
 all: all-redirect
 .SUFFIXES:
 .SUFFIXES: .S .c .idl .o .s
-$(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) 
-	cd $(top_srcdir) && $(AUTOMAKE) --gnu Makefile
-
-Makefile: $(srcdir)/Makefile.in  $(top_builddir)/config.status $(BUILT_SOURCES)
-	cd $(top_builddir) \
-	  && CONFIG_FILES=$@ CONFIG_HEADERS= $(SHELL) ./config.status
-
 $(ACLOCAL_M4):  configure.in  m4/codeset.m4 m4/gettext.m4 m4/glibc21.m4 \
 		m4/iconv.m4 m4/intdiv0.m4 m4/inttypes-pri.m4 \
 		m4/inttypes.m4 m4/inttypes_h.m4 m4/isc-posix.m4 \
