--- configure.old	2005-05-09 00:17:06.000000000 +0200
+++ configure	2005-07-19 09:44:06.000000000 +0200
@@ -21111,7 +21111,7 @@
 
 
 #CFLAGS=`echo "$CFLAGS" | sed -e 's/-O[0-9*]//g'`
-CFLAGS="-g"
+CFLAGS="-g -mmmx"
 
                                                                                                                                                                                               ac_config_files="$ac_config_files Makefile doc/Makefile liboil/Makefile liboil/colorspace/Makefile liboil/conv/Makefile liboil/copy/Makefile liboil/dct/Makefile liboil/md5/Makefile liboil/jpeg/Makefile liboil/simdpack/Makefile liboil/utf8/Makefile testsuite/Makefile examples/Makefile examples/huffman/Makefile examples/jpeg/Makefile examples/md5/Makefile examples/uberopt/Makefile examples/work/Makefile liboil.pc"
 
