--- gps-5.0.1-src/shared.gpr.orig	2010-09-09 10:25:16.000000000 +0200
+++ gps-5.0.1-src/shared.gpr	2011-10-12 14:57:52.638925722 +0200
@@ -35,7 +35,7 @@
                "-gnatwaCJe", "-gnat05");
 
             for Default_Switches ("C") use
-              ("-g", "-O1", "-fdump-xref") & GnatColl_Shared.Gtk_Include;
+              ("-g", "-O1") & GnatColl_Shared.Gtk_Include;
 
          when "Production" =>
             for Default_Switches ("Ada") use
