--- gps-4.4.1-src/src_editor/src/src_editor_buffer.adb	2009-11-30 10:07:46.000000000 +0100
+++ gps-4.4.1-src.fixed/src_editor/src/src_editor_buffer.adb	2011-02-17 19:02:32.603974890 +0100
@@ -3119,7 +3119,7 @@
       Force_Write : Boolean := False;
       --  Whether the file mode has been forced to writable
       U_Buffer    : Unbounded_String;
-      S           : Ada.Strings.Unbounded.Aux.Big_String_Access;
+      S           : Ada.Strings.Unbounded.String_Access;
       Length      : Natural;
       Has_Errors  : Boolean := False;
 
