From bf31e0c5f34e25a6ab9f642c6b843f8cb4872306 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Wed, 14 Dec 2022 01:14:38 +0100
Subject: [PATCH] Work-around -Wpedantic flag added by meson/muon

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 meson.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/meson.build b/meson.build
index c942fd311b64..876a48acbe9b 100644
--- a/meson.build
+++ b/meson.build
@@ -61,6 +61,9 @@ possible_cc_flags = [
         # work-around for gcc 7.1 turning this on on its own.
         '-Wno-error=nonnull',
 
+        # work-around -Wpedantic added by meson/muon
+        '-Wno-pedantic',
+
         # Disable -Wmaybe-uninitialized, since it's noisy on gcc 8 with
         # optimizations enabled, producing essentially false positives.
         '-Wno-maybe-uninitialized',
-- 
2.38.1

