--- a/clang-ctags	2016-10-05 10:18:56.325760311 +0200
+++ b/clang-ctags	2016-10-05 20:52:03.275742387 +0200
@@ -44,6 +44,8 @@
     index = clang.cindex.Index.create()
     try:
         start = time.time()
+        compiler_command_line.append("-isystem")
+        compiler_command_line.append("/usr/lib/clang/3.9.0/include")
         tu = index.parse(None, compiler_command_line,
                          options=TranslationUnit.PARSE_SKIP_FUNCTION_BODIES)
         debug("  clang parse took %.2fs" % (time.time() - start))
