From 2e1e85aad08753f10c35f70d4400fe7a80f8242b Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Thu, 29 Dec 2022 11:46:26 +0100
Subject: [PATCH] fix qt5/6 build

---
 src/Main.cpp | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/src/Main.cpp b/src/Main.cpp
index 9eb650b..ef60184 100644
--- a/src/Main.cpp
+++ b/src/Main.cpp
@@ -27,13 +27,9 @@
 #include <unistd.h>
 #include <signal.h>
 
-#if QT_VERSION < QT_VERSION_CHECK( 6, 0, 0 )
-#include <DFL/DF5/Application.hpp>
-#include <DFL/DF5/Utils.hpp>
-#else
-#include <DFL/DF6/Application.hpp>
-#include <DFL/DF6/Utils.hpp>
-#endif
+// from dfl
+#include <Application.hpp>
+#include <Utils.hpp>
 
 #include <wayqt/WayQtUtils.hpp>
 #include <wayqt/Registry.hpp>
-- 
2.42.0

