diff -Naur libcanberra-0.23.orig/Makefile.in libcanberra-0.23/Makefile.in
--- libcanberra-0.23.orig/Makefile.in	2010-10-06 16:12:23.227000071 -0700
+++ libcanberra-0.23/Makefile.in	2010-10-06 16:16:31.841999901 -0700
@@ -936,11 +936,11 @@
 
 
 @GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@install-data-local:
-@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@                if test -z "$(DESTDIR)" ; then \
-@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@                        for p in $(schemas_DATA) ; do \
-@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@                                GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(top_builddir)/data/$$p ; \
-@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@                        done \
-@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@                fi
+@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@	if test -z "$(DESTDIR)" ; then \
+@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@		for p in $(schemas_DATA) ; do \
+@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@			GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(top_builddir)/$$p ; \
+@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@		done \
+@GCONF_SCHEMAS_INSTALL_TRUE@@HAVE_GTK_TRUE@	fi
 @GCONF_SCHEMAS_INSTALL_FALSE@@HAVE_GTK_TRUE@install-data-local:
 @HAVE_GTK_FALSE@install-data-local:
 
