--- configure~	2005-09-04 14:21:34.000000000 +0200
+++ configure	2005-11-14 19:20:16.000000000 +0100
@@ -35066,15 +35066,6 @@
 
 
 
-if test "x$HAVE_XFIXES" = "xyes"; then
-  HAVE_XFIXES_TRUE=
-  HAVE_XFIXES_FALSE='#'
-else
-  HAVE_XFIXES_TRUE='#'
-  HAVE_XFIXES_FALSE=
-fi
-
-
 
   succeeded=no
 
@@ -35180,15 +35171,6 @@
 
 
 
-if test "x$HAVE_XDAMAGE" = "xyes"; then
-  HAVE_XDAMAGE_TRUE=
-  HAVE_XDAMAGE_FALSE='#'
-else
-  HAVE_XDAMAGE_TRUE='#'
-  HAVE_XDAMAGE_FALSE=
-fi
-
-
   fi
 
 
@@ -35203,6 +35185,25 @@
     USE_X=no
   fi
 fi
+
+if test "x$HAVE_XFIXES" = "xyes"; then
+  HAVE_XFIXES_TRUE=
+  HAVE_XFIXES_FALSE='#'
+else
+  HAVE_XFIXES_TRUE='#'
+  HAVE_XFIXES_FALSE=
+fi
+
+if test "x$HAVE_XDAMAGE" = "xyes"; then
+  HAVE_XDAMAGE_TRUE=
+  HAVE_XDAMAGE_FALSE='#'
+else
+  HAVE_XDAMAGE_TRUE='#'
+  HAVE_XDAMAGE_FALSE=
+fi
+
+
+
 if test x$USE_X = xyes; then
   :
   if test "xximagesink" != "x"; then
