From 24716ce95bf5b10d685611de23489045cf2ca5cc Mon Sep 17 00:00:00 2001
From: Moritz Bunkus <mo@bunkus.online>
Date: Sat, 13 May 2023 22:25:34 +0200
Subject: [PATCH] fix compilation with fmt 10.0.0

Upstream-Status: Backport
Origin: Upstream
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 src/common/codec.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/common/codec.h b/src/common/codec.h
index f8bc1b4e3..f4a92c322 100644
--- a/src/common/codec.h
+++ b/src/common/codec.h
@@ -231,3 +231,7 @@ operator <<(std::ostream &out,
 
   return out;
 }
+
+#if FMT_VERSION >= 90000
+template <> struct fmt::formatter<codec_c> : ostream_formatter {};
+#endif  // FMT_VERSION >= 90000
-- 
GitLab

