From c79298d8813e010cfce30b31c66008c1a7c99392 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 24 May 2014 19:24:23 +0200
Subject: [PATCH] fix dependencies in Makefile.in

---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 7615e79..ed52d12 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -480,7 +480,7 @@ am__objects_18 = constraintList.$(OBJEXT) constraintResolve.$(OBJEXT) \
 	constraint.$(OBJEXT) loopHeuristics.$(OBJEXT)
 am__objects_19 =
 am__objects_20 = $(am__objects_19)
-am__objects_21 = $(am__objects_7) $(am__objects_16) $(am__objects_17) \
+am__objects_21 = $(am_objects_12) $(am__objects_7) $(am__objects_16) $(am__objects_17) \
 	$(am__objects_18) $(am__objects_20) $(am__objects_19)
 am_splint_OBJECTS = $(am__objects_21)
 splint_OBJECTS = $(am_splint_OBJECTS)
-- 
1.9.3

