--- Makefile	2006-06-09 09:27:50.000000000 -0700
+++ Makefile	2006-06-09 09:28:40.000000000 -0700
@@ -160,7 +160,7 @@
 MEINPROC = @MEINPROC@
 MICO_INCLUDES = @MICO_INCLUDES@
 MICO_LDFLAGS = @MICO_LDFLAGS@
-MOC = /usr/lib/qt3/bin/moc
+MOC = /usr/bin/moc
 MSGFMT = /usr//bin/msgfmt
 NOOPT_CFLAGS = 
 NOOPT_CXXFLAGS =  -fno-exceptions -fno-check-new
@@ -175,14 +175,14 @@
 PYTHONMODDIR = @PYTHONMODDIR@
 QTE_NORTTI = 
 QTOPIA_INCLUDES = @QTOPIA_INCLUDES@
-QT_INCLUDES = -I/usr/lib/qt3/include
-QT_LDFLAGS = -L/usr/lib/qt3/lib
+QT_INCLUDES = -I/usr/include/qt
+QT_LDFLAGS = -L/usr/lib
 RANLIB = ranlib
 REPO = -frepo
 SETUIDFLAGS = @SETUIDFLAGS@
 STRIP = strip
 TOPSUBDIRS =  f4lm
-UIC = /usr/lib/qt3/bin/uic -L $(kde_widgetdir) -nounload
+UIC = /usr/bin/uic -L $(kde_widgetdir) -nounload
 UIC_TR = tr2i18n
 USER_INCLUDES = 
 USER_LDFLAGS = 
@@ -198,8 +198,8 @@
 X_INCLUDES = -I/usr/X11R6/include
 X_LDFLAGS = -L/usr/X11R6/lib
 X_PRE_LIBS = 
-all_includes = -I/usr/lib/qt3/include -I/usr/X11R6/include
-all_libraries = -L/usr/X11R6/lib -L/usr/lib/qt3/lib
+all_includes = -I/usr/include/qt -I/usr/X11R6/include
+all_libraries = -L/usr/X11R6/lib -L/usr/lib
 idldir = @idldir@
 jni_includes = @jni_includes@
 kde_appsdir = @kde_appsdir@
@@ -224,8 +224,8 @@
 kde_wallpaperdir = @kde_wallpaperdir@
 kde_widgetdir = @kde_widgetdir@
 micodir = @micodir@
-qt_includes = /usr/lib/qt3/include
-qt_libraries = /usr/lib/qt3/lib
+qt_includes = /usr/include/qt
+qt_libraries = /usr/lib
 x_includes = /usr/X11R6/include
 x_libraries = /usr/X11R6/lib
 
