--- a/suff.c
+++ b/suff.c
@@ -67,5 +67,8 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
+
+#define _BSD_SOURCE
+#include <strings.h> /* for ffs */

 #ifndef MAKE_NATIVE
--- a/targ.c
+++ b/targ.c
@@ -67,5 +67,8 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
+
+#define _BSD_SOURCE
+#include <strings.h> /* for ffs */

 #ifndef MAKE_NATIVE
--- a/unit-tests/dotwait.exp
+++ b/unit-tests/dotwait.exp
@@ -22,10 +22,10 @@
 shared.2.1
 shared.2.99
 shared.2.99
+cycle.1.99
 make: Graph cycles through `cycle.2.99'
 make: Graph cycles through `cycle.2.98'
 make: Graph cycles through `cycle.2.97'
-cycle.1.99
 cycle.1.99
 exit status 0
