From 56aa92974c8f88935b5d64b75ec59b3ac92c3653 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sun, 7 Mar 2021 16:21:42 +0100
Subject: [PATCH 2/2] fix glibc-2.33 compatability

---
 installwatch.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/installwatch.c b/installwatch.c
index 610a824..cbea001 100644
--- a/installwatch.c
+++ b/installwatch.c
@@ -48,6 +48,15 @@
 #include <dlfcn.h>
 #include <dirent.h>
 
+/* glibc-2.33 removed these defs */
+#ifndef _STAT_VER
+#define _STAT_VER 0
+#endif
+#ifndef _MKNOD_VER
+#define _MKNOD_VER 0
+#endif
+
+
 /* There's no d_off on GNU/kFreeBSD */
 #if defined(__FreeBSD_kernel__)
 #define D_OFF(X) (-1)
-- 
2.30.1

