diff --git a/vendor/openssl-sys/build/main.rs b/vendor/openssl-sys/build/main.rs
--- a/vendor/openssl-sys/build/main.rs
+++ b/vendor/openssl-sys/build/main.rs
@@ -285,6 +285,8 @@ See rust-openssl documentation for more information:
             (3, 7, 0) => ('3', '7', '0'),
             (3, 7, 1) => ('3', '7', '1'),
             (3, 7, _) => ('3', '7', 'x'),
+            (3, 8, 0) => ('3', '8', '0'),
+            (3, 8, _) => ('3', '8', 'x'),
             _ => version_error(),
         };
 
diff --git a/vendor/openssl-sys/build/cfgs.rs b/vendor/openssl-sys/build/cfgs.rs
index 960515f..f901e5a 100644
--- a/vendor/openssl-sys/build/cfgs.rs
+++ b/vendor/openssl-sys/build/cfgs.rs
@@ -52,6 +52,9 @@ pub fn get(openssl_version: Option<u64>, libressl_version: Option<u64>) -> Vec<&
         if libressl_version >= 0x3_07_00_00_0 {
             cfgs.push("libressl370");
         }
+        if libressl_version >= 0x3_08_00_00_0 {
+            cfgs.push("libressl380");
+        }
     } else {
         let openssl_version = openssl_version.unwrap();
 
diff --git a/vendor/openssl-sys-0.9.85/build/main.rs b/vendor/openssl-sys-0.9.85/build/main.rs
--- a/vendor/openssl-sys-0.9.85/build/main.rs
+++ b/vendor/openssl-sys-0.9.85/build/main.rs
@@ -285,6 +285,8 @@ See rust-openssl documentation for more information:
             (3, 7, 0) => ('3', '7', '0'),
             (3, 7, 1) => ('3', '7', '1'),
             (3, 7, _) => ('3', '7', 'x'),
+            (3, 8, 0) => ('3', '8', '0'),
+            (3, 8, _) => ('3', '8', 'x'),
             _ => version_error(),
         };
 
diff --git a/vendor/openssl-sys-0.9.85/build/cfgs.rs b/vendor/openssl-sys-0.9.85/build/cfgs.rs
index 960515f..f901e5a 100644
--- a/vendor/openssl-sys-0.9.85/build/cfgs.rs
+++ b/vendor/openssl-sys-0.9.85/build/cfgs.rs
@@ -52,6 +52,9 @@ pub fn get(openssl_version: Option<u64>, libressl_version: Option<u64>) -> Vec<&
         if libressl_version >= 0x3_07_00_00_0 {
             cfgs.push("libressl370");
         }
+        if libressl_version >= 0x3_08_00_00_0 {
+            cfgs.push("libressl380");
+        }
     } else {
         let openssl_version = openssl_version.unwrap();
 
