diff --git a/vendor/openssl-sys-0.9.106/build/main.rs b/vendor/openssl-sys-0.9.106/build/main.rs
--- a/vendor/openssl-sys-0.9.106/build/main.rs
+++ b/vendor/openssl-sys-0.9.106/build/main.rs
@@ -383,6 +383,8 @@ See rust-openssl documentation for more information:
             (3, 9, _) => ('3', '9', 'x'),
             (4, 0, 0) => ('4', '0', '0'),
             (4, 0, _) => ('4', '0', 'x'),
+            (4, 1, 0) => ('4', '1', '0'),
+            (4, 1, _) => ('4', '1', 'x'),
             _ => version_error(),
         };
 
