diff --git a/vendor/openssl-sys-0.9.109/build/main.rs b/vendor/openssl-sys-0.9.109/build/main.rs
index 4007a26..65c8ce9 100644
--- a/vendor/openssl-sys-0.9.109/build/main.rs
+++ b/vendor/openssl-sys-0.9.109/build/main.rs
@@ -451,6 +451,8 @@ See rust-openssl documentation for more information:
             (4, 0, _) => ('4', '0', 'x'),
             (4, 1, 0) => ('4', '1', '0'),
             (4, 1, _) => ('4', '1', 'x'),
+            (4, 2, 0) => ('4', '2', '0'),
+            (4, 2, _) => ('4', '2', 'x'),
             _ => version_error(),
         };
 
