From 7ba0212765c29cd61df1e10c0f8a132a6d28d9c1 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Tue, 29 Dec 2020 18:11:00 +0100
Subject: [PATCH 01/10] Allow to build against libressl 3.2.x and 3.3.x

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 vendor/openssl-sys/build/main.rs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/vendor/openssl-sys/build/main.rs b/vendor/openssl-sys/build/main.rs
index 49f26d2b7f37..6bacc16237ab 100644
--- a/vendor/openssl-sys/build/main.rs
+++ b/vendor/openssl-sys/build/main.rs
@@ -221,6 +221,8 @@ See rust-openssl README for more information:
             (3, 1, 0) => ('3', '1', '0'),
             (3, 1, _) => ('3', '1', 'x'),
             (3, 2, 0) => ('3', '2', '0'),
+            (3, 2, _) => ('3', '2', 'x'),
+            (3, 3, _) => ('3', '3', 'x'),
             _ => version_error(),
         };
 
-- 
2.30.0

