diff --git a/vendor/openssl-sys/build/main.rs b/vendor/openssl-sys/build/main.rs
--- a/vendor/openssl-sys/build/main.rs
+++ b/vendor/openssl-sys/build/main.rs
@@ -285,6 +285,8 @@ See rust-openssl documentation for more information:
             (3, 7, 0) => ('3', '7', '0'),
             (3, 7, 1) => ('3', '7', '1'),
             (3, 7, _) => ('3', '7', 'x'),
+            (3, 8, 0) => ('3', '8', '0'),
+            (3, 8, _) => ('3', '8', 'x'),
             _ => version_error(),
         };
 
