diff --git a/src/bootstrap/src/core/build_steps/dist.rs b/src/bootstrap/src/core/build_steps/dist.rs
index cfcb144..529be48 100644
--- a/src/bootstrap/src/core/build_steps/dist.rs
+++ b/src/bootstrap/src/core/build_steps/dist.rs
@@ -1065,7 +1108,6 @@ fn copy_src_dirs(
 
         static LLVM_PROJECTS: &[&str] = &[
             "llvm-project/clang",
-            "llvm-project/libunwind",
             "llvm-project/lld",
             "llvm-project/lldb",
             "llvm-project/llvm",
@@ -1191,7 +1238,7 @@ impl Step for Src {
         copy_src_dirs(
             builder,
             &builder.src,
-            &["library", "src/llvm-project/libunwind"],
+            &["library"],
             &[
                 // not needed and contains symlinks which rustup currently
                 // chokes on when unpacking.
