diff --git a/vendor/openssl-sys-0.9.112/build/main.rs b/vendor/openssl-sys-0.9.112/build/main.rs
index 3f9eae0..f159337 100644
--- a/vendor/openssl-sys-0.9.112/build/main.rs
+++ b/vendor/openssl-sys-0.9.112/build/main.rs
@@ -425,6 +425,7 @@ See rust-openssl documentation for more information:
             (4, 1, 0) => ('4', '1', '0'),
             (4, 1, _) => ('4', '1', 'x'),
             (4, 2, _) => ('4', '2', 'x'),
+            (4, 3, _) => ('4', '3', 'x'),
             _ => version_error(),
         };
 
