From 584ab4467bb72f89bbc24445c29722c59bb27984 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Fri, 8 Sep 2017 00:04:29 -0500
Subject: [PATCH 02/10] Fix LLVM build

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 src/bootstrap/lib.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/bootstrap/lib.rs b/src/bootstrap/lib.rs
index ff8548a0f0bb..210084cf5ea6 100644
--- a/src/bootstrap/lib.rs
+++ b/src/bootstrap/lib.rs
@@ -834,7 +834,8 @@ impl Build {
             .args()
             .iter()
             .map(|s| s.to_string_lossy().into_owned())
-            .filter(|s| !s.starts_with("-O") && !s.starts_with("/O"))
+            .filter(|s| !s.starts_with("-O") && !s.starts_with("/O")
+                    && !s.starts_with("-static"))
             .collect::<Vec<String>>();
 
         // If we're compiling on macOS then we add a few unconditional flags
-- 
2.30.0

