From c9787296f8198b50c89a5e655bbd81576c3e1d06 Mon Sep 17 00:00:00 2001
From: Tim Neumann <mail@timnn.me>
Date: Fri, 21 Jul 2017 13:13:37 +0200
Subject: [PATCH] rustllvm: define LLVM_VERSION_LT

---
 src/rustllvm/rustllvm.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/rustllvm/rustllvm.h b/src/rustllvm/rustllvm.h
index 0baf5528e9..20816af2f1 100644
--- a/src/rustllvm/rustllvm.h
+++ b/src/rustllvm/rustllvm.h
@@ -55,6 +55,8 @@
   (LLVM_VERSION_MAJOR < (major) ||                                             \
    LLVM_VERSION_MAJOR == (major) && LLVM_VERSION_MINOR <= (minor))
 
+#define LLVM_VERSION_LT(major, minor) (!LLVM_VERSION_GE((major), (minor)))
+
 #if LLVM_VERSION_GE(3, 7)
 #include "llvm/IR/LegacyPassManager.h"
 #else
-- 
2.14.2

