diff --git a/src/bootstrap/builder.rs b/src/bootstrap/builder.rs
--- a/src/bootstrap/builder.rs
+++ b/src/bootstrap/builder.rs
@@ -145,7 +145,7 @@
             only_hosts: S::ONLY_HOSTS,
             should_run: S::should_run,
             make_run: S::make_run,
-            name: unsafe { ::std::intrinsics::type_name::<S>() },
+            name: { ::std::intrinsics::type_name::<S>() },
         }
     }
 
