diff -Naur mozilla.orig/xpcom/base/nsStackFrameUnix.cpp mozilla/xpcom/base/nsStackFrameUnix.cpp
--- mozilla.orig/xpcom/base/nsStackFrameUnix.cpp	2007-04-17 12:44:25.090270741 -0700
+++ mozilla/xpcom/base/nsStackFrameUnix.cpp	2007-04-17 13:09:43.169678209 -0700
@@ -77,9 +77,20 @@
 #endif // MOZ_DEMANGLE_SYMBOLS
 }
 
+#if defined(linux) // Linux
+#if (__GLIBC__ > 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 3)) // use glibc backtrace()
+#include <execinfo.h>
+void DumpStackToFile(FILE* aStream)
+{
+  void *array[20];
+  size_t size;
 
-#if defined(linux) && defined(__GLIBC__) && (defined(__i386) || defined(PPC)) // i386 or PPC Linux stackwalking code
+  fflush(aStream);
+  size = backtrace (array, 20);
+  backtrace_symbols_fd (array, size, fileno(aStream));
+}
 
+#elif defined(__GLIBC__) && (defined(__i386) || defined(PPC)) // old style i386 or PPC Linux stackwalking code
 #include <setjmp.h>
 //
 
@@ -135,6 +146,13 @@
   }
 }
 
+#else // not implemented
+void DumpStackToFile(FILE* aStream)
+{
+  fprintf(aStream, "Info: Stacktrace not implemented for this Linux platform\n");
+}
+#endif // Linux
+
 #elif defined(__sun) && (defined(__sparc) || defined(sparc) || defined(__i386) || defined(i386))
 
 /*
diff -Naur mozilla.orig/xpcom/base/nsTraceRefcntImpl.cpp mozilla/xpcom/base/nsTraceRefcntImpl.cpp
--- mozilla.orig/xpcom/base/nsTraceRefcntImpl.cpp	2007-04-17 12:44:25.089270894 -0700
+++ mozilla/xpcom/base/nsTraceRefcntImpl.cpp	2007-04-17 13:09:43.174677447 -0700
@@ -857,8 +857,8 @@
 }
 
 // WIN32 x86 stack walking code
-// i386 or PPC Linux stackwalking code or Solaris
-#elif (defined(linux) && defined(__GLIBC__) && (defined(__i386) || defined(PPC))) || (defined(__sun) && (defined(__sparc) || defined(sparc) || defined(__i386) || defined(i386)))
+// Linux stackwalking code or Solaris
+#elif (defined(linux) && defined(__GLIBC__)) || (defined(__sun) && (defined(__sparc) || defined(sparc) || defined(__i386) || defined(i386)))
 #include "nsStackFrameUnix.h"
 void
 nsTraceRefcntImpl::WalkTheStack(FILE* aStream)
