diff -Naur mozilla.orig/gfx/src/gtk/Makefile.in mozilla/gfx/src/gtk/Makefile.in
--- mozilla.orig/gfx/src/gtk/Makefile.in	2007-04-17 12:44:26.861000605 -0700
+++ mozilla/gfx/src/gtk/Makefile.in	2007-04-17 13:03:22.151804875 -0700
@@ -197,9 +197,9 @@
 endif
 
 ifdef MOZ_ENABLE_XPRINT
+EXTRA_DSO_LDOPTS += $(DIST)/lib/$(LIB_PREFIX)xprintutil.$(LIB_SUFFIX)
 DEFINES         += -DUSE_XPRINT -DUSE_MOZILLA_TYPES
 EXTRA_DSO_LDOPTS += \
-		-lxprintutil \
 		$(MOZ_XPRINT_LDFLAGS)
 endif
 
