diff -u  netkit-ntalk-0.17/talkd/announce.c  netkit-ntalk-0.17/talkd/announce.c
--- netkit-ntalk-0.17/talkd/announce.c	Tue Jan 15 11:03:40 2002
+++ netkit-ntalk-0.17/talkd/announce.cw	Tue Jan 15 11:03:40 2002
@@ -50,4 +50,5 @@
 #include <string.h>
 #include <paths.h>
+#include <time.h>
 #include "prot_talkd.h"
 #include "proto.h"
diff -u  netkit-ntalk-0.17/talkd/table.c  netkit-ntalk-0.17/talkd/table.c
--- netkit-ntalk-0.17/talkd/table.c	Tue Jan 15 11:03:40 2002
+++ netkit-ntalk-0.17/talkd/table.c	Tue Jan 15 11:03:40 2002
@@ -55,4 +55,5 @@
 #include <string.h>
 #include <netinet/in.h>
+#include <time.h>
 
 #include "prot_talkd.h"
