diff -ruN at-3.1.8.orig/parsetime.y at-3.1.8.orig.fixed/parsetime.y
--- at-3.1.8.orig/parsetime.y	2002-12-10 22:04:25.000000000 +0100
+++ at-3.1.8.orig.fixed/parsetime.y	2002-12-10 22:03:17.000000000 +0100
@@ -62,7 +62,7 @@
 		;
 
 time_or_not     : time
-		|
+		|;
 
 time            : hr24clock_hr_min
                 | hr24clock_hr_min timezone_name
@@ -111,7 +111,7 @@
                 ;
 
 inc_or_dec	: increment
-		| decrement
+		| decrement;
 
 increment       : '+' inc_number inc_period
 		    {
@@ -286,7 +286,7 @@
 			    exectm.tm_mon = mnum -1;
 			    free($1);
 			}
-		    }
+		    };
 day_number	: INT
                      {
 			exectm.tm_mday = -1;
