--- configure.in.orig	2007-10-12 20:39:27.000000000 +0200
+++ configure.in	2007-10-12 20:40:49.000000000 +0200
@@ -2657,8 +2657,7 @@
                        ])
         if test "$ac_cv_have_visibility_builtin_bug" = "no" -a \
                 "$ac_cv_have_visibility_class_bug" = "no"; then
-          VISIBILITY_FLAGS='-I$(DIST)/include/system_wrappers -include $(topsrcdir)/config/gcc_hidden.h'
-          WRAP_SYSTEM_INCLUDES=1
+          VISIBILITY_FLAGS='-fvisibility=hidden'
         else
           VISIBILITY_FLAGS='-fvisibility=hidden'
         fi # have visibility pragma bug
