diff -Naur ntop-3.3.9.orig/Makefile.am ntop-3.3.9/Makefile.am
--- ntop-3.3.9.orig/Makefile.am	2009-12-30 12:37:37.007745774 -0800
+++ ntop-3.3.9/Makefile.am	2009-12-30 12:38:40.538426478 -0800
@@ -270,8 +270,6 @@
 
 install: install-recursive
 
-	cd @GEO_DIR@; make install
-
 	@mkdir -p $(DESTDIR)/$(CFG_DBFILE_DIR)
 
 	@echo ""
diff -Naur ntop-3.3.9.orig/configure.in ntop-3.3.9/configure.in
--- ntop-3.3.9.orig/configure.in	2009-12-30 12:37:36.997426119 -0800
+++ ntop-3.3.9/configure.in	2009-12-30 12:39:44.841488896 -0800
@@ -1916,10 +1916,8 @@
 fi
 
 
-GEO_DIR=`find $PWD -type d -name "GeoIP-*"`
-GEO_IP="$GEO_DIR/libGeoIP/"
-CFLAGS="$CFLAGS -I$GEO_IP"
-LDFLAGS="$LDFLAGS -L$GEO_IP.libs/ -lGeoIP"
+CFLAGS="$CFLAGS"
+LDFLAGS="$LDFLAGS -lGeoIP"
 
 dnl> NTOPCONFIGDEBUG_SETTINGS([precet])
 
@@ -2084,7 +2082,6 @@
 AC_SUBST(SO_VERSION_PATCH)
 AC_SUBST(RRD_LIB)
 AC_SUBST(RRD_INC)
-AC_SUBST(GEO_DIR)
 
 AC_CONFIG_FILES([Makefile])
 
