--- gal/a11y/e-table/gal-a11y-e-table-item.c	2004/05/21 14:10:56	1.1
+++ gal/a11y/e-table/gal-a11y-e-table-item.c	2004/05/21 14:11:48
@@ -48,6 +48,7 @@ struct _GalA11yETableItemPrivate {
 static gboolean gal_a11y_e_table_item_ref_selection (GalA11yETableItem *a11y,
 						     ESelectionModel *selection);
 static gboolean gal_a11y_e_table_item_unref_selection (GalA11yETableItem *a11y);
+static gpointer * eti_reinit_data (AtkTable *table, ETableItem *item);
 
 #if 0
 static void
