diff -Naur hellanzb-0.13/Hellanzb/Core.py hellanzb-0.13/Hellanzb/Core.py
--- hellanzb-0.13/Hellanzb/Core.py	2007-03-27 00:20:43.000000000 -0400
+++ hellanzb-0.13/Hellanzb/Core.py	2007-10-25 10:04:38.831982064 -0400
@@ -39,6 +39,7 @@
     try:
         confDirs.append(os.path.join(os.getcwd(), 'etc'))
         confDirs.append(os.getcwd())
+        confDirs.append('/etc')
     except OSError, ose:
         if ose.errno != 2:
             raise
diff -Naur ./hellanzb-0.13/setup.py ./hellanzb-0.13/setup.py
--- hellanzb-0.13/setup.py	2007-03-27 00:20:43.000000000 -0400
+++ hellanzb-0.13/setup.py	2007-10-25 10:02:28.970175892 -0400
@@ -38,8 +38,7 @@
         packages = [ 'Hellanzb', 'Hellanzb.NZBLeecher', 'Hellanzb.HellaXMLRPC',
                      'Hellanzb.external', 'Hellanzb.external.elementtree' ],
         scripts = [ 'hellanzb.py' ],
-        data_files = [ ( 'etc', [ 'etc/hellanzb.conf.sample' ] ),
-                       ( 'share/doc/hellanzb', [ 'CHANGELOG', 'CREDITS', 'README', 'LICENSE' ] ) ],
+        data_files = [ ( 'share/doc/hellanzb', [ 'CHANGELOG', 'CREDITS', 'README', 'LICENSE' ] ) ],
         )
     py2app_options = dict(
         app = [ 'hellanzb.py' ],
