From cf831a5bcaeeda65cb0f162b998fd3055274886c Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 18 Dec 2016 11:18:03 +0100
Subject: [PATCH] do not start xconsole

---
 config/Xsetup_0 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/Xsetup_0 b/config/Xsetup_0
index 29d7aa4..13f4793 100644
--- a/config/Xsetup_0
+++ b/config/Xsetup_0
@@ -1,2 +1,2 @@
 #!/bin/sh
-xconsole -geometry 480x130-0-0 -daemon -notify -verbose -fn fixed -exitOnFail
+
-- 
2.11.0

