From db7616d6daa62a2299b09200bd3d0cba97d8db9d Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 23 May 2022 17:28:41 +0200
Subject: [PATCH] Use dbus-launcher to start xsession

---
 config/Xsession.in | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/config/Xsession.in b/config/Xsession.in
index c1d7d11..3868a91 100644
--- a/config/Xsession.in
+++ b/config/Xsession.in
@@ -57,16 +57,17 @@ fi
 
 startup=$HOME/.xsession
 resources=$HOME/.Xresources
+dbus_launch=/usr/bin/dbus-launch
 
 if [ -s "$startup" ]; then
 	if [ -x "$startup" ]; then
-		exec "$startup"
+		exec "$dbus_launch" --exit-with-session "$startup"
 	else
-		exec /bin/sh "$startup"
+		exec "$dbus_launch" --exit-with-session /bin/sh "$startup"
 	fi
 else
 	if [ -r "$resources" ]; then
 		BINDIR/xrdb -load "$resources"
 	fi
-	exec BINDIR/xsm
+	exec "$dbus_launch" --exit-with-session BINDIR/xsm
 fi
-- 
2.36.1

