From f8e1ba94509391955b0bffffb11f2ec3bed55bf1 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Fri, 15 Mar 2013 20:34:51 +0100
Subject: [PATCH] use X's default authentication mechanism instead of xdm's
 specific one

---
 config/xdm-config.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/config/xdm-config.cpp b/config/xdm-config.cpp
index a685129..5203cce 100644
--- a/config/xdm-config.cpp
+++ b/config/xdm-config.cpp
@@ -37,3 +37,4 @@ DisplayManager*loginmoveInterval:	10
 ! SECURITY: do not listen for XDMCP or Chooser requests
 ! Comment out this line if you want to manage X terminals with xdm
 DisplayManager.requestPort:	0
+DisplayManager*authName:  MIT-MAGIC-COOKIE-1
-- 
1.8.1.3

