diff -uNr libxml2-2.9.10.orig/encoding.c libxml2-2.9.10/encoding.c
--- libxml2-2.9.10.orig/encoding.c	2019-10-22 20:46:01.000000000 +0200
+++ libxml2-2.9.10/encoding.c	2020-11-04 13:20:06.944305628 +0100
@@ -44,6 +44,11 @@
 #endif
 #include <libxml/globals.h>
 #include <libxml/xmlerror.h>
+#ifndef FALSE
+# define FALSE (1 == 0)
+# define TRUE (! FALSE)
+#endif
+
 
 #include "buf.h"
 #include "enc.h"
