From 7ada7d12db384f096c3abe74f4fc25dbe89893b5 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sun, 9 Dec 2012 15:36:08 +0100
Subject: [PATCH] fix install target, build shared and static library

---
 src/Makefile.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 752d627..86b8a7a 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -8,12 +8,14 @@ DESTDIR=
 
 OBJS = libircclient.o
 
-all:	@TARGET@
+all:	lib
 
 static:	libircclient.a
 shared: libircclient.so
 shared_mingw: libircclient.dll
 
+lib: shared static
+
 install: lib
 	-mkdir -p $(DESTDIR)@prefix@/include
 	-mkdir -p $(DESTDIR)@prefix@/lib
-- 
1.8.0

