From a186f8dfc22d24b03d103df1f303cb22fb38d149 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sat, 5 May 2012 11:44:24 +0200
Subject: [PATCH] fix build with gcc 4.7

---
 src/qtsingleapplication/qtlocalpeer.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/qtsingleapplication/qtlocalpeer.cpp b/src/qtsingleapplication/qtlocalpeer.cpp
index 4c4382d..b666197 100644
--- a/src/qtsingleapplication/qtlocalpeer.cpp
+++ b/src/qtsingleapplication/qtlocalpeer.cpp
@@ -101,7 +101,7 @@ QtLocalPeer::QtLocalPeer(QObject* parent, const QString &appId)
         socketName += QLatin1Char('-') + QString::number(sessionId, 16);
     }
 #else
-    socketName += QLatin1Char('-') + QString::number(::getuid(), 16);
+    socketName += QLatin1Char('-') + QString::number(QtLP_Private::getuid(), 16);
 #endif
 
     server = new QLocalServer(this);
-- 
1.7.10.1

