--- Makefile	2004-10-31 16:52:41.243237328 +0000
+++ Makefile.new	2004-10-31 16:55:13.808043952 +0000
@@ -141,8 +141,8 @@
 am__quote = 
 install_sh = /usr/src/gnomeicu-0.99.5/install-sh
 
-SUBDIRS = src gnomeicu-applet ui po icons emoticons sounds doc
-#SUBDIRS = src ui po icons emoticons sounds doc
+#SUBDIRS = src gnomeicu-applet ui po icons emoticons sounds doc
+SUBDIRS = src ui po icons emoticons sounds doc
 
 desktopdir = $(datadir)/applications
 desktop_in_files = GnomeICU.desktop.in
