diff -uNr krusader-2.0.0/krusader/VFS/kiojobwrapper.cpp krusader-2.0.0.cp/krusader/VFS/kiojobwrapper.cpp
--- krusader-2.0.0/krusader/VFS/kiojobwrapper.cpp	2009-04-11 21:40:15.000000000 +0200
+++ krusader-2.0.0.cp/krusader/VFS/kiojobwrapper.cpp	2009-10-13 14:08:24.000000000 +0200
@@ -38,6 +38,8 @@
 #include <kio/job.h>
 #include <qapplication.h>
 #include <iostream>
+#include <cstdio>
+
 #include <klocale.h>
 #include "virtualcopyjob.h"
 #include <QTextDocument>
