From 9e3aee69c0ab82f264567aa1d1bb46c3240a932e Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@sourcemage.org>
Date: Tue, 26 May 2020 20:28:18 +0200
Subject: [PATCH] corelib: Fix memory_resource feature check

diff --git a/src/corelib/tools/qduplicatetracker_p.h b/src/corelib/tools/qduplicatetracker_p.h
index 99068c01a335..7355da9ceb83 100644
--- a/src/corelib/tools/qduplicatetracker_p.h
+++ b/src/corelib/tools/qduplicatetracker_p.h
@@ -52,7 +52,7 @@
 
 #include <qglobal.h>
 
-#if QT_HAS_INCLUDE(<memory_resource>) && __cplusplus > 201402L
+#if defined __GLIBC__ && QT_HAS_INCLUDE(<memory_resource>) && __cplusplus > 201402L
 #  include <unordered_set>
 #  include <memory_resource>
 #else
