From 0bdcb647f8fe187abde0e4b140745f150876ddbb Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Tue, 28 Jul 2020 20:42:37 +0200
Subject: [PATCH] Disable O_TMPFILE; conflicts with castfs

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 src/corelib/io/qtemporaryfile_p.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/corelib/io/qtemporaryfile_p.h b/src/corelib/io/qtemporaryfile_p.h
index 6bcff936b4a9..10971f94f909 100644
--- a/src/corelib/io/qtemporaryfile_p.h
+++ b/src/corelib/io/qtemporaryfile_p.h
@@ -64,7 +64,6 @@
 // some early libc support had the wrong values for O_TMPFILE
 // (see https://bugzilla.gnome.org/show_bug.cgi?id=769453#c18)
 #    if (O_TMPFILE & O_DIRECTORY) == O_DIRECTORY
-#      define LINUX_UNNAMED_TMPFILE
 #    endif
 #  endif
 #endif
-- 
2.28.0

