diff -Nuar powertweak-0.99.5.orig/src/backends/CPU/x86/mtrr.c powertweak-0.99.5/src/backends/CPU/x86/mtrr.c
--- powertweak-0.99.5.orig/src/backends/CPU/x86/mtrr.c	2006-06-28 21:13:03.247832176 -0700
+++ powertweak-0.99.5/src/backends/CPU/x86/mtrr.c	2006-06-28 21:13:14.875064568 -0700
@@ -22,6 +22,8 @@
 
 #define ERRSTRING strerror (errno)
 
+extern char** mtrr_strings;
+
 void ShowMTRRInfo (struct cpu_identity *id)
 {
 	int fd;
