diff --git a/meson.build b/meson.build
index 970f49a52..14736e12e 100644
--- a/meson.build
+++ b/meson.build
@@ -720,7 +720,9 @@ endif
 bfd_dep = cc.find_library('bfd', has_headers: ['bfd.h'], required: get_option('symbol-lookup'))
 if bfd_dep.found() and \
    cc.has_function('bfd_openr', dependencies: [bfd_dep]) and \
-   cc.links(files('meson-cc-tests/bfd-section-flags.c'), name: 'bfd_section_flags', dependencies: bfd_dep)
+   cc.links(files('meson-cc-tests/bfd-section-flags.c'),
+     args: ['-DPACKAGE=cairo -DPACKAGE_VERSION="@0@"'.format(meson.project_version()),],
+     name: 'bfd_section_flags', dependencies: bfd_dep)
   conf.set('HAVE_BFD', 1)
   deps += [bfd_dep]
 elif get_option('symbol-lookup').enabled()
diff --git a/util/cairo-trace/meson.build b/util/cairo-trace/meson.build
index 8d4ac131b..cf05352c5 100644
--- a/util/cairo-trace/meson.build
+++ b/util/cairo-trace/meson.build
@@ -11,7 +11,8 @@ shared_lib_ext = libcairo.full_path().split('.')[-1]
 libcairotrace = library('cairo-trace', cairo_trace_sources,
   include_directories: [incbase, incsrc],
   dependencies: deps,
-  c_args: ['-DSHARED_LIB_EXT="@0@"'.format(shared_lib_ext),] + pthread_c_args,
+  c_args: ['-DSHARED_LIB_EXT="@0@"'.format(shared_lib_ext),
+    '-DPACKAGE=cairo -DPACKAGE_VERSION="@0@"'.format(meson.project_version()),] + pthread_c_args,
   link_args: extra_link_args,
   gnu_symbol_visibility: 'hidden',
   install: true,
