--- cairo-1.18.2/meson-cc-tests/bfd-section-flags.c.orig	2025-02-28 12:33:31.306663221 +0100
+++ cairo-1.18.2/meson-cc-tests/bfd-section-flags.c	2025-02-28 12:43:50.864663147 +0100
@@ -1,3 +1,5 @@
+/* The bfd.h requires some trace of an autoconf config.h file. */
+#define PACKAGE "cairo"
 #include <bfd.h>
 
 int
--- cairo-1.18.2/util/cairo-trace/lookup-symbol.c.orig	2025-02-28 12:56:08.216663059 +0100
+++ cairo-1.18.2/util/cairo-trace/lookup-symbol.c	2025-02-28 12:57:02.155663052 +0100
@@ -60,6 +60,10 @@
 #include <pthread.h>
 
 #if HAVE_BFD
+/* bfd.h requiring some autoconf-like traces of config.h */
+#ifndef PACKAGE
+#define PACKAGE "cairo"
+#endif
 #include <bfd.h>
 
 struct symtab {
