From 6e58c2c34383ac24ab4ff2f0aa6c70f18091a359 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sat, 10 Sep 2016 13:03:44 +0200
Subject: [PATCH] fix path to header

---
 menu-cache-gen/menu-tags.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/menu-cache-gen/menu-tags.h b/menu-cache-gen/menu-tags.h
index f3fd7d3..e9ee422 100644
--- a/menu-cache-gen/menu-tags.h
+++ b/menu-cache-gen/menu-tags.h
@@ -19,7 +19,7 @@
  *      Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-#include <libfm/fm-extra.h>
+#include <libfm-1.0/fm-extra.h>
 #include <menu-cache.h>
 
 FmXmlFileTag menuTag_Menu;
-- 
2.10.0

