--- src/configuration.h.old	2006-07-24 11:12:39.675026456 +0200
+++ src/configuration.h	2006-07-24 11:13:15.182628480 +0200
@@ -322,8 +322,8 @@ public:
     ~Configuration();
 
 private:
-    QValueList<int> Configuration::listString2Int( QStringList qs);
-    QStringList Configuration::listInt2String( QValueList<int> qi);
+    QValueList<int> listString2Int( QStringList qs);
+    QStringList listInt2String( QValueList<int> qi);
     int startuptip, readid3v2, writeid3v2, autoconn, autohide, avoidhdd, avoidsmb, avoidnfs;
 #ifdef HAVE_MEXTRAS
   int checkartist, checkcover, checklyrics, checksyncedlyrics, checkalbum, checktracks, checkbiography, checkyears;
--- src/progressdialogscroll.h.old	2006-07-24 11:12:46.931923240 +0200
+++ src/progressdialogscroll.h	2006-07-24 11:13:27.178804784 +0200
@@ -32,7 +32,7 @@ public:
         setMinimumHeight( 200 );
         setMinimumWidth( 500 );
     }
-    void ProgressDialogScroll::resizeEvent( QResizeEvent *ev ) {
+    void resizeEvent( QResizeEvent *ev ) {
         QProgressDialog::resizeEvent( ev );
         QRect rct = static_cast<QWidget*>(this->child("label","QLabel"))->geometry();
         rct.setTop(2);
--- src/database_mysql.h.old	2006-07-24 11:14:14.165661696 +0200
+++ src/database_mysql.h	2006-07-24 11:14:30.470183032 +0200
@@ -116,7 +116,7 @@ public:
     void alterYearsActiveArtistsMextras(QString& years_active, int artists_id);
     void alterBiographyArtistsMextras(int biography_id, int artists_id);
     void alterLyricsTracksMextras(int lyrics_id, int tracks_id);
-    void DataBase_MySQL::alterSyncedLyricsTracksMextras(int synced_lyrics_id, int tracks_id);
+    void alterSyncedLyricsTracksMextras(int synced_lyrics_id, int tracks_id);
     void alterCoverTracksMextras(int, int, const QString, const QString);
     void alterAlbumReviewTracksMextras(int album_review_id, int album_id);
     void alterAlbumTracksTracksMextras(int album_tracks_id, int album_id);
