diff -urN BlockCacheSet.cpp BlockCacheSet.cpp
--- BlockCacheSet.cpp	2006-11-05 13:22:45.963189321 +0000
+++ BlockCacheSet.cpp	2006-11-05 13:23:25.000000000 +0000
@@ -15,7 +15,9 @@
 /*                                                                        */
 /**************************************************************************/
 
-#include <iostream.h>
+using namespace std;
+
+#include <iostream>
 #include "BlockCacheSet.h"
 
 
diff -urN Loop.cpp Loop.cpp
--- Loop.cpp	2006-11-05 13:22:45.854205948 +0000
+++ Loop.cpp	2006-11-05 13:23:25.000000000 +0000
@@ -15,6 +15,8 @@
 /*                                                                        */
 /**************************************************************************/
 
+using namespace std;
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -29,7 +31,7 @@
 #include <dirent.h>
 #include <pwd.h>
 
-#include <fstream.h>
+#include <fstream>
 #include <fcntl.h>
 
 #include <sys/types.h>
diff -urN Message.cpp Message.cpp
--- Message.cpp	2006-11-05 13:22:45.963189321 +0000
+++ Message.cpp	2006-11-05 13:23:25.000000000 +0000
@@ -15,8 +15,10 @@
 /*                                                                        */
 /**************************************************************************/
 
+using namespace std;
+
 #include <stdio.h>
-#include <fstream.h>
+#include <fstream>
 #include <unistd.h>
 #include <string.h>
 
diff -urN Misc.cpp Misc.cpp
--- Misc.cpp	2006-11-05 13:22:45.701229285 +0000
+++ Misc.cpp	2006-11-05 13:23:25.000000000 +0000
@@ -15,7 +15,9 @@
 /*                                                                        */
 /**************************************************************************/
 
-#include <iostream.h>
+using namespace std;
+
+#include <iostream>
 #include <stdio.h>
 #include <ctype.h>
 #include <stdlib.h>
diff -urN Misc.h Misc.h
--- Misc.h	2006-11-05 13:22:45.643238132 +0000
+++ Misc.h	2006-11-05 13:23:25.000000000 +0000
@@ -18,8 +18,10 @@
 #ifndef Misc_H
 #define Misc_H
 
-#include <iostream.h>
-#include <fstream.h>
+using namespace std;
+
+#include <iostream>
+#include <fstream>
 
 #include <errno.h>
 #include <string.h>
diff -urN Proxy.cpp Proxy.cpp
--- Proxy.cpp	2006-11-05 13:22:45.856205642 +0000
+++ Proxy.cpp	2006-11-05 13:23:25.000000000 +0000
@@ -15,10 +15,12 @@
 /*                                                                        */
 /**************************************************************************/
 
+using namespace std;
+
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
-#include <fstream.h>
+#include <fstream>
 #include <sys/types.h>
 #include <sys/stat.h>
 
diff -urN Split.cpp Split.cpp
--- Split.cpp	2006-11-05 13:22:45.797214643 +0000
+++ Split.cpp	2006-11-05 13:23:25.000000000 +0000
@@ -15,7 +15,9 @@
 /*                                                                        */
 /**************************************************************************/
 
-#include <fstream.h>
+using namespace std;
+
+#include <fstream>
 #include <unistd.h>
 #include <string.h>
 #include <sys/stat.h>
