--- configure.in	2006-11-04 19:37:56.000000000 +0000
+++ configure.in	2006-11-04 19:38:23.000000000 +0000
@@ -52,35 +52,6 @@
 AC_PROG_CC
 AC_LANG_CPLUSPLUS
 
-dnl Check whether option -Wno-deprecated
-dnl is needed by GCC compiler.
-
-AC_MSG_CHECKING([whether compiler needs -Wno-deprecated])
-gcc_version=`${CC} --version | grep 'gcc (GCC) [[3-4]].' | head -n 1`
-case "${gcc_version}" in
-     gcc*)
-        AC_MSG_RESULT([yes])
-        CXXFLAGS="$CXXFLAGS -Wno-deprecated"
-        ;;
-
-     *)
-        AC_MSG_RESULT([no])
-        ;;
-esac
-
-AC_MSG_CHECKING([whether compiler accepts -Wmissing-declarations and -Wnested-externs])
-gcc_version=`${CC} --version | grep 'gcc (GCC) [[3-4]].' | head -n 1`
-case "${gcc_version}" in
-     gcc*)
-        AC_MSG_RESULT([no])
-        ;;
-
-     *)
-        AC_MSG_RESULT([yes])
-        CXXFLAGS="$CXXFLAGS -Wmissing-declarations -Wnested-externs"
-        ;;
-esac
-
 dnl Check for BSD compatible install.
 
 AC_PROG_INSTALL
