--- Loop.cpp.orig	2009-07-06 12:38:10.000000000 +0200
+++ Loop.cpp	2009-07-06 12:38:14.000000000 +0200
@@ -10979,7 +10979,7 @@
                                       packMethod == PACK_LOSSLESS ||
                                           packMethod == PACK_ADAPTIVE)
   {
-    char *dash = rindex(opt, '-');
+    const char *dash = rindex(opt, '-');
 
     if (dash != NULL && strlen(dash) == 2 &&
             *(dash + 1) >= '0' && *(dash + 1) <= '9')
