--- wscript.orig	2008-08-14 16:39:37.000000000 +0200
+++ wscript	2008-08-14 16:39:52.000000000 +0200
@@ -263,8 +263,8 @@
         conf.check_message("uncommitted changes", "", bool(changed))
         conf.env["VERSION"] = BASEVERSION + " (git commit: %s%s)" % (nam, dirty)
 
-    conf.env["CCFLAGS"] = Utils.to_list(conf.env["CCFLAGS"]) + ['-g', '-O0']
-    conf.env["CXXFLAGS"] = Utils.to_list(conf.env["CXXFLAGS"]) + ['-g', '-O0']
+    conf.env["CCFLAGS"] = Utils.to_list(conf.env["CCFLAGS"])
+    conf.env["CXXFLAGS"] = Utils.to_list(conf.env["CXXFLAGS"])
     conf.env['XMMS_PKGCONF_FILES'] = []
     conf.env['XMMS_OUTPUT_PLUGINS'] = [(-1, "NONE")]
 
