diff -Naur nekobee-0.1.5~/configure.in nekobee-0.1.5/configure.in
--- nekobee-0.1.5~/configure.in	2006-07-21 22:50:59.000000000 +0000
+++ nekobee-0.1.5/configure.in	2007-03-01 20:15:15.000000000 +0000
@@ -28,10 +28,6 @@
 if test "x${try_gtk2}" = 'xyes'; then
     AM_PATH_GTK_2_0(2.0.0, with_gtk='yes (version 2.0)', with_gtk=no)
 fi
-dnl  No GTK+ 2.x found, look for GTK+ 1.2 instead
-if test "x${with_gtk}" = 'xno'; then
-    AM_PATH_GTK(1.2.0, with_gtk='yes (version 1.2)', AC_MSG_WARN([GUI will not be built because no version of GTK+ was found]))
-fi
 AM_CONDITIONAL(BUILD_GUI, test "x${with_gtk}" != 'xno')
 echo "GTK support: $with_gtk"
 
