--- include/uniconfkey.h~	2005-11-24 20:46:46.000000000 +0100
+++ include/uniconfkey.h	2006-03-15 10:14:37.000000000 +0100
@@ -113,7 +113,7 @@
     bool iswild() const;
 
     /** Returns true if the key has a trailing slash. */
-    bool UniConfKey::hastrailingslash() const;
+    bool hastrailingslash() const;
 
     /**
      * Returns the number of segments in this path.
--- include/wvhttppool.h~	2005-11-24 20:46:46.000000000 +0100
+++ include/wvhttppool.h	2006-03-15 10:26:08.000000000 +0100
@@ -57,7 +57,7 @@
     WvString linkname;
     WvUrl url;
 
-    WvUrlLink::WvUrlLink(WvStringParm _linkname, WvStringParm _url)
+    WvUrlLink(WvStringParm _linkname, WvStringParm _url)
 	: linkname(_linkname), url(_url)
     {}
 };
--- include/wvstring.h~	2005-11-24 20:46:46.000000000 +0100
+++ include/wvstring.h	2006-03-15 10:12:01.000000000 +0100
@@ -393,7 +393,7 @@
     
 protected:
     void copy_constructor(const WvFastString &s);
-    inline void WvString::construct(const char *_str)
+    inline void construct(const char *_str)
         {
             link(&nullbuf, _str);
     
--- include/wvx509.h~	2006-01-09 22:42:53.000000000 +0100
+++ include/wvx509.h	2006-03-15 10:20:39.000000000 +0100
@@ -349,19 +349,19 @@
      * Set the Certificate to use X509v3, since that's all modern
      * PKI uses anyways :)
      */
-    void WvX509Mgr::set_version();
+    void set_version();
 
     /**
      * Get and set the keyUsage field.
      */
-    WvString WvX509Mgr::get_key_usage();
-    void WvX509Mgr::set_key_usage(WvStringParm values);
+    WvString get_key_usage();
+    void set_key_usage(WvStringParm values);
 
     /**
      * Get and set the extendedKeyUsage field.
      */
-    WvString WvX509Mgr::get_ext_key_usage();
-    void WvX509Mgr::set_ext_key_usage(WvStringParm values);
+    WvString get_ext_key_usage();
+    void set_ext_key_usage(WvStringParm values);
     
     /**
      * Return the Subject alt name if it exists, and WvString::null if
