diff -Naur libmsn-4.1.orig/msntest/msntest.cpp libmsn-4.1/msntest/msntest.cpp
--- libmsn-4.1.orig/msntest/msntest.cpp	2010-01-05 13:07:08.000000000 +0100
+++ libmsn-4.1/msntest/msntest.cpp	2010-12-29 10:37:11.000000000 +0100
@@ -259,7 +259,7 @@
                     if(mySocketsSsl[i].isSSL && !mySocketsSsl[i].isConnected)
                     {
                         BIO *bio_socket_new;
-                        SSL_METHOD *meth=NULL;
+                        const SSL_METHOD *meth=NULL;
                         meth=SSLv23_client_method();
                         SSLeay_add_ssl_algorithms();
                         mySocketsSsl[i].ctx = SSL_CTX_new(meth);
