diff --git a/src/shrpx_tls.cc b/src/shrpx_tls.cc
index 74ca9e1..cb3e68f 100644
--- a/src/shrpx_tls.cc
+++ b/src/shrpx_tls.cc
@@ -362,7 +362,7 @@ int tls_session_new_cb(SSL *ssl, SSL_SESSION *session) {

 namespace {
 SSL_SESSION *tls_session_get_cb(SSL *ssl,
-#if OPENSSL_1_1_API
+#if OPENSSL_1_1_API || LIBRESSL_2_8_API
                                 const unsigned char *id,
 #else  // !OPENSSL_1_1_API
                                 unsigned char *id,
diff --git a/src/ssl_compat.h b/src/ssl_compat.h
index 777b83c..c962d21 100644
--- a/src/ssl_compat.h
+++ b/src/ssl_compat.h
@@ -32,12 +32,14 @@
 #define LIBRESSL_IN_USE 1
 #define LIBRESSL_LEGACY_API (LIBRESSL_VERSION_NUMBER < 0x20700000L)
 #define LIBRESSL_2_7_API (LIBRESSL_VERSION_NUMBER >= 0x20700000L)
+#define LIBRESSL_2_8_API (LIBRESSL_VERSION_NUMBER >= 0x20800000L)
 #else // !defined(LIBRESSL_VERSION_NUMBER)
 #define OPENSSL_1_1_API (OPENSSL_VERSION_NUMBER >= 0x1010000fL)
 #define OPENSSL_1_1_1_API (OPENSSL_VERSION_NUMBER >= 0x10101000L)
 #define LIBRESSL_IN_USE 0
 #define LIBRESSL_LEGACY_API 0
 #define LIBRESSL_2_7_API 0
+#define LIBRESSL_2_8_API 0
 #endif // !defined(LIBRESSL_VERSION_NUMBER)

 #endif // OPENSSL_COMPAT_H
