diff -urN mod_python-3.2.7/configure mod_python/configure
--- mod_python-3.2.7/configure	2005-10-28 12:06:22.000000000 -0400
+++ mod_python/configure	2006-02-08 07:56:21.000000000 -0500
@@ -3035,7 +3035,7 @@
 # get the mod_python version
 
 MP_VERSION=`awk '/MPV_STRING/ {print $3}' src/include/mpversion.h`
-MP_VERSION=`echo $MP_VERSION | sed s/\\"//g`
+MP_VERSION=`echo $MP_VERSION | sed s/\"//g`
 
 # get --with-python-src. The python src is required to generate the documentation
 # It is not required to compile or install mod_python itself
