#
# Fixes http://www.mangler.org/trac/ticket/194
#
diff -Naur mangler-1.2.0.orig/Makefile.in mangler-1.2.0/Makefile.in
--- mangler-1.2.0.orig/Makefile.in	2010-07-03 00:35:17.079075663 -0700
+++ mangler-1.2.0/Makefile.in	2010-07-03 00:55:21.392083891 -0700
@@ -168,6 +168,8 @@
 PACKAGE_VERSION = @PACKAGE_VERSION@
 PATH_SEPARATOR = @PATH_SEPARATOR@
 PKG_CONFIG = @PKG_CONFIG@
+PKG_CONFIG_LIBDIR = @PKG_CONFIG_LIBDIR@
+PKG_CONFIG_PATH = @PKG_CONFIG_PATH@
 PTHREAD_CC = @PTHREAD_CC@
 PTHREAD_CFLAGS = @PTHREAD_CFLAGS@
 PTHREAD_LIBS = @PTHREAD_LIBS@
@@ -248,6 +250,8 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
+xi_CFLAGS = @xi_CFLAGS@
+xi_LIBS = @xi_LIBS@
 ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = libventrilo3 icons sounds src 
 all: config.h
diff -Naur mangler-1.2.0.orig/configure.ac mangler-1.2.0/configure.ac
--- mangler-1.2.0.orig/configure.ac	2010-07-03 00:35:17.079075663 -0700
+++ mangler-1.2.0/configure.ac	2010-07-03 00:55:21.392083891 -0700
@@ -138,6 +138,10 @@
 AC_SUBST(gtkmm_CFLAGS)
 AC_SUBST(gtkmm_LIBS)
 
+PKG_CHECK_MODULES(xi, xi)
+AC_SUBST(xi_CFLAGS)
+AC_SUBST(xi_LIBS)
+
 AC_ARG_WITH([pulseaudio],
 	[AS_HELP_STRING([--without-pulseaudio], [disable support for PulseAudio])],
 	[:],
diff -Naur mangler-1.2.0.orig/src/Makefile.am mangler-1.2.0/src/Makefile.am
--- mangler-1.2.0.orig/src/Makefile.am	2010-07-03 00:35:17.061326047 -0700
+++ mangler-1.2.0/src/Makefile.am	2010-07-03 00:55:24.801082580 -0700
@@ -36,7 +36,7 @@
 	manglerg15.cpp manglerg15.h \
 	inilib.cpp inilib.h
 
-mangler_LDADD = @top_srcdir@/libventrilo3/libventrilo3.la $(gtkmm_LIBS) $(gthread_LIBS) $(pulse_LIBS) $(alsa_LIBS) $(dbus_LIBS)
+mangler_LDADD = @top_srcdir@/libventrilo3/libventrilo3.la $(gtkmm_LIBS) $(gthread_LIBS) $(pulse_LIBS) $(alsa_LIBS) $(dbus_LIBS) $(xi_LIBS)
 
 INCLUDES = -I@top_srcdir@/libventrilo3/ -I@top_srcdir@/icons/ -I@top_srcdir@/sounds/ $(gtkmm_CFLAGS) $(gthread_CFLAGS) $(pulse_CFLAGS) $(alsa_CFLAGS) $(oss_CFLAGS) $(dbus_CFLAGS)
 
