From 94ab14414606b794b7ccf60595493cd8715e195b Mon Sep 17 00:00:00 2001
From: ProgAndy <code@progandy.de>
Date: Fri, 2 Apr 2021 14:33:18 +0200
Subject: [PATCH] Fix mime type XZ after "Add checksum for XZ"

---
 magic/Magdir/compress | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/magic/Magdir/compress b/magic/Magdir/compress
index 64d11a73..aef4bb49 100644
--- a/magic/Magdir/compress
+++ b/magic/Magdir/compress
@@ -265,14 +265,13 @@
 
 # http://tukaani.org/xz/xz-file-format.txt
 0	ustring		\xFD7zXZ\x00		XZ compressed data, checksum
+!:strength * 2
+!:mime	application/x-xz
 >7	byte&0xf	0x0			NONE
 >7	byte&0xf	0x1			CRC32
 >7	byte&0xf	0x4			CRC64
 >7	byte&0xf	0xa			SHA-256
 
-!:strength * 2
-!:mime	application/x-xz
-
 # https://github.com/ckolivas/lrzip/blob/master/doc/magic.header.txt
 0	string		LRZI			LRZIP compressed data
 >4	byte		x			- version %d
-- 
2.31.1

