From 948b95efd679a362d9f21033a26f0e4b37741da6 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 24 Mar 2014 23:36:32 +0100
Subject: [PATCH] disable tests

---
 wscript | 1 -
 1 file changed, 1 deletion(-)

diff --git a/wscript b/wscript
index 83ad7b8..13d941c 100644
--- a/wscript
+++ b/wscript
@@ -267,7 +267,6 @@ def build(bld):
         pass
     if bld.env['DEST_OS'] not in ['ios', 'iosimulator', 'android']:
         bld.recurse('examples')
-        bld.recurse('tests')
 
     bld( source = 'aubio.pc.in' )
 
-- 
1.9.0

