--- Makefile.am~	2003-07-29 22:27:41.000000000 +0200
+++ Makefile.am	2005-04-24 09:26:05.000000000 +0200
@@ -24,7 +24,7 @@
 ## tests.
 SUBDIRS = . m4 lib tests
 
-bin_SCRIPTS = automake aclocal
+bin_SCRIPTS = automake-@APIVERSION@ aclocal-@APIVERSION@
 info_TEXINFOS = automake.texi
 
 CLEANFILES = $(bin_SCRIPTS)
@@ -40,19 +40,6 @@
 EXTRA_DIST = ChangeLog.96 ChangeLog.98 ChangeLog.00 ChangeLog.01 \
   $(AUTOMAKESOURCES)
 
-## Make versioned links.  We only run the transform on the root name;
-## then we make a versioned link with the transformed base name.  This
-## seemed like the most reasonable approach.
-install-exec-hook:
-	@$(POST_INSTALL)
-	@for p in $(bin_SCRIPTS); do \
-	  f="`echo $$p|sed '$(transform)'`"; \
-	  fv="$$f-$(APIVERSION)"; \
-	  rm -f $(DESTDIR)$(bindir)/$$fv; \
-	  echo " $(LN) $(DESTDIR)$(bindir)/$$f $(DESTDIR)$(bindir)/$$fv"; \
-	  $(LN) $(DESTDIR)$(bindir)/$$f $(DESTDIR)$(bindir)/$$fv; \
-	done
-
 uninstall-hook:
 	@for p in $(bin_SCRIPTS); do \
 	  f="`echo $$p|sed '$(transform)'`"; \
@@ -71,13 +58,13 @@
 	    -e 's,[@]APIVERSION[@],$(APIVERSION),g' \
 	    -e 's,[@]SHELL[@],$(SHELL),g'
 
-automake: automake.in Makefile
-	$(do_subst) < $(srcdir)/automake.in > automake
-	chmod +x automake
-
-aclocal: aclocal.in Makefile
-	$(do_subst) < $(srcdir)/aclocal.in > aclocal
-	chmod +x aclocal
+automake-@APIVERSION@: automake.in Makefile
+	$(do_subst) < $(srcdir)/automake.in > automake-@APIVERSION@
+	chmod +x automake-@APIVERSION@
+
+aclocal-@APIVERSION@: aclocal.in Makefile
+	$(do_subst) < $(srcdir)/aclocal.in > aclocal-@APIVERSION@
+	chmod +x aclocal-@APIVERSION@
 
 ## The master location for INSTALL is lib/INSTALL.
 ## This is where `make fetch' will install new versions.
