diff -r -N -u gnote-0.7.2.orig/libtomboy/tomboyutil.c gnote-0.7.2/libtomboy/tomboyutil.c
--- gnote-0.7.2.orig/libtomboy/tomboyutil.c	2010-07-02 08:43:17.748133704 -0500
+++ gnote-0.7.2/libtomboy/tomboyutil.c	2010-07-02 08:45:23.418853833 -0500
@@ -153,9 +153,9 @@
 void
 tomboy_window_present_hardcore (GtkWindow *window)
 {
-	if (!GTK_WIDGET_REALIZED (window))
+	if (!gtk_widget_get_realized (window))
 		gtk_widget_realize (GTK_WIDGET (window));
-	else if (GTK_WIDGET_VISIBLE (window))
+	else if (gtk_widget_get_visible (window))
 		tomboy_window_move_to_current_workspace (window);
 
 	tomboy_window_override_user_time (window);
