--- a/liblash/client.c	2005-09-15 11:15:30.000000000 +0300
+++ b/liblash/client.c	2006-09-16 10:52:11.000000000 +0300
@@ -20,7 +20,7 @@
 
 #include <lash/types.h>
 #include <lash/client.h>
-#include <lash/xmalloc.h>
+#include <lash/list.h>
 #include <lash/internal.h>
 
 lash_client_t *
--- a/liblash/config.c	2005-09-15 11:15:30.000000000 +0300
+++ b/liblash/config.c	2006-09-16 10:52:11.000000000 +0300
@@ -22,7 +22,7 @@
 #include <stdio.h>
 
 #include <lash/config.h>
-#include <lash/xmalloc.h>
+#include <lash/list.h>
 #include <lash/internal.h>
 
 void
--- a/liblash/event.c	2005-09-15 11:15:30.000000000 +0300
+++ b/liblash/event.c	2006-09-16 10:52:11.000000000 +0300
@@ -22,7 +22,7 @@
 
 #include <lash/types.h>
 #include <lash/internal.h>
-#include <lash/xmalloc.h>
+#include <lash/list.h>
 #include <lash/event.h>
 
 lash_event_t *
--- a/liblash/socket.c	2005-10-04 06:26:41.000000000 +0300
+++ b/liblash/socket.c	2006-09-16 10:52:11.000000000 +0300
@@ -34,6 +34,10 @@
 #include <unistd.h>
 #include <ctype.h>
 
+
+#include <lash/list.h> /* Something weird with mutual includes between list.h and xmalloc.h, which leaves
+                          lash_malloc undefined. I am including list. h before xmalloc.h, which will pull in
+                          xmalloc.h. In fact xmalloc.h include can be removed */
 #include <lash/xmalloc.h>
 #include <lash/debug.h>
 
