--- configure.ac.orig	2015-08-17 09:40:48.119201650 +0200
+++ configure.ac	2015-08-17 09:42:59.912212267 +0200
@@ -140,6 +140,7 @@
                    #include "list.h"])
 fi
 
+if false; then
 AC_CHECK_HEADERS([glamor.h],[found_glamor_header=yes],[found_glamor_header=no],
 		 [#include "xorg-server.h"])
 AC_MSG_CHECKING([whether to include GLAMOR support])
@@ -150,6 +151,7 @@
 else
 	AC_MSG_RESULT([no])
 fi
+fi
 
 AC_CONFIG_FILES([
 	Makefile
