--- utils/msgfmt.py.old	2005-01-23 00:12:30.838573232 +0100
+++ utils/msgfmt.py	2005-01-23 00:08:20.280663824 +0100
@@ -82,7 +82,7 @@
         koffsets += [l1, o1+keystart]
         voffsets += [l2, o2+valuestart]
     offsets = koffsets + voffsets
-    output = struct.pack("iiiiiii",
+    output = struct.pack("Iiiiiii",
                          0x950412de,        # Magic
                          0,                 # Version
                          len(keys),         # # of entries
