From 7bdeb3377938821279843ecf70584a11fa475239 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Fri, 21 Apr 2017 20:18:20 +0200
Subject: [PATCH 2/2] icu-59 fix

---
 Source/JavaScriptCore/API/JSStringRef.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/JavaScriptCore/API/JSStringRef.h b/Source/JavaScriptCore/API/JSStringRef.h
index 6f45fcb..b9cb5e6 100644
--- a/Source/JavaScriptCore/API/JSStringRef.h
+++ b/Source/JavaScriptCore/API/JSStringRef.h
@@ -43,7 +43,7 @@ extern "C" {
 @typedef JSChar
 @abstract A Unicode character.
 */
-    typedef unsigned short JSChar;
+    typedef char16_t JSChar; // icu-59
 #else
     typedef wchar_t JSChar;
 #endif
-- 
2.12.2

