diff -Naur cdfs-2.6.23/daemon.c cdfs-2.6.23-fixed/daemon.c
--- cdfs-2.6.23/daemon.c	2006-10-24 21:44:49.000000000 +0200
+++ cdfs-2.6.23-fixed/daemon.c	2007-11-22 03:00:05.000000000 +0100
@@ -28,6 +28,7 @@
 
 
 #include "cdfs.h"
+#include <linux/sched.h>
 
 int kcdfsd_pid = 0;
 static int kcdfsd_running = 0;
diff -Naur cdfs-2.6.23/root.c cdfs-2.6.23-fixed/root.c
--- cdfs-2.6.23/root.c	2006-10-24 21:41:12.000000000 +0200
+++ cdfs-2.6.23-fixed/root.c	2007-11-22 03:00:05.000000000 +0100
@@ -34,6 +34,7 @@
 
 
 #include <linux/seq_file.h>
+#include <linux/sched.h>
 
 #ifndef OLD_KERNEL
 #include <linux/statfs.h>
