From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
Subject: [PATCH] fix segfault when wd empty
Date: Sat Dec 26 10:35:06 2015 -0700
From: Dan Kociela <dkociela@gmail.com>

Upstream-Status: Unknown
Origin: Void Linux
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 src/tabbar.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: src/tabbar.c
===================================================================
--- a/src/tabbar.c	(revision 337)
+++ b/src/tabbar.c	(working copy)
@@ -1327,7 +1327,7 @@
 		if( p.we_wordc > 1)
 		    rxvt_msg( DBG_ERROR, DBG_TABBAR,
 			    "Too many words when expanding %s\n", cwdOption );
-		else
+		else if( NOT_NULL( *p.we_wordv ) )
 		{
 		    filename = *p.we_wordv;
 
