diff --git a/CMakeLists.txt b/CMakeLists.txt
index e420043..9e7784e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,8 +13,8 @@ set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
 
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
                     Plasma DocTools Runner JsEmbed NotifyConfig Su NewStuff Wallet KCMUtils
-                    IdleTime Declarative TextWidgets KDELibs4Support Crash GlobalAccel
-                    NetworkManagerQt DBusAddons)
+                    IdleTime Declarative TextWidgets KDELibs4Support Crash GlobalAccel DBusAddons)
+find_package(KF5NetworkManagerQt)
 find_package(KF5XmlRpcClient REQUIRED)
 
 # WARNING PlasmaQuick provides unversioned CMake config
diff --git a/dataengines/CMakeLists.txt b/dataengines/CMakeLists.txt
index 014da10..72fb3fc 100644
--- a/dataengines/CMakeLists.txt
+++ b/dataengines/CMakeLists.txt
@@ -9,7 +9,11 @@ add_subdirectory(dict)
 add_subdirectory(executable)
 add_subdirectory(favicons)
 add_subdirectory(filebrowser)
-add_subdirectory(geolocation)
+
+if(KF5NetworkManagerQt_FOUND)
+  add_subdirectory(geolocation)
+endif()
+
 add_subdirectory(hotplug)
 add_subdirectory(keystate)
 add_subdirectory(mpris2)
