Index: configure.ac
===================================================================
RCS file: /cvsroot/yafc/yafc/configure.ac,v
retrieving revision 1.1
diff -u -r1.1 configure.ac
--- configure.ac	5 Dec 2002 22:10:21 -0000	1.1
+++ configure.ac	14 Mar 2003 05:30:17 -0000
@@ -64,7 +64,8 @@
                  arpa/inet.h \
                  sys/ioctl.h \
                  memory.h \
-                 setjmp.h
+                 setjmp.h \
+                 errno.h
 )
 
 AC_CHECK_HEADERS(dirent.h, [], [
