From c851f18f38abc6c04f70dc7f6fa825873ad93edb Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 1 Dec 2018 13:23:28 +0100
Subject: [PATCH] fix path to awk

---
 scripts/mk_bits_lut | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/mk_bits_lut b/scripts/mk_bits_lut
index 7321c2c..ba9c752 100755
--- a/scripts/mk_bits_lut
+++ b/scripts/mk_bits_lut
@@ -1,4 +1,4 @@
-#!/usr/bin/awk -f
+#!/bin/awk -f
 
 # Check for bits being set by right shifting until the
 # desired bit is the first bit, and then checking mod 2.
-- 
2.19.2

