From 4b38c274e174b0428b66f2c6c324050c385d1aa5 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Fri, 18 Sep 2015 20:52:20 +0200
Subject: [PATCH] remove invocation of git, remove exit 1 at end of
 Makefile.inc rule

---
 Makefile | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index 2584ba7..87e3e0f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,12 +1,7 @@
 NAME=hwdata
 VERSION=$(shell awk '/Version:/ { print $$2 }' hwdata.spec)
 RELEASE=$(shell rpm -q --define 'dist %{nil}' --specfile --qf "%{release}" hwdata.spec)
-ifeq ($(shell git rev-parse --abbrev-ref HEAD | sed -n 's/^\([^0-9-]\+\).*/\L\1/p'), rhel)
-    # add revision to tag name for rhel branches
-    TAGNAME := $(NAME)-$(VERSION)-$(RELEASE)
-else
-    TAGNAME := $(NAME)-$(VERSION)
-endif
+TAGNAME := $(NAME)-$(VERSION)
 SOURCEDIR := $(shell pwd)
 ARCHIVE := $(TAGNAME).tar.bz2
 
@@ -26,7 +21,6 @@ all:
 Makefile.inc: configure
 	./configure
 	@echo "$@ generated. Run the make again."
-	@exit 1
 
 install: Makefile.inc
 	mkdir -p -m 755 $(DESTDIR)$(datadir)/$(NAME)
-- 
2.5.2

