diff -Naur dpkg-1.14.16.6/scripts/Makefile.am dpkg-1.14.16.6-fixed/scripts/Makefile.am
--- dpkg-1.14.16.6/scripts/Makefile.am	2008-01-18 11:12:53.000000000 +0100
+++ dpkg-1.14.16.6-fixed/scripts/Makefile.am	2008-02-13 03:02:33.000000000 +0100
@@ -47,7 +47,6 @@
 	cleanup-info.pl \
 	dpkg-divert.pl \
 	dpkg-statoverride.pl \
-	install-info.pl \
 	update-alternatives.pl \
 	changelog/debian.pl \
 	t/000_pod.t \
@@ -125,11 +124,7 @@
 
 
 # Automake has its own install-info rule, gah
-all-local: install-info-stamp
-install-info-stamp: $(srcdir)/install-info.pl
-	$(do_perl_subst) <$< >install-info
-	chmod +x install-info
-	touch $@
+all-local:
 
 install-exec-local: install-info-stamp
 	$(mkdir_p) $(DESTDIR)$(sbindir)
diff -Naur dpkg-1.14.16.6/scripts/Makefile.in dpkg-1.14.16.6-fixed/scripts/Makefile.in
--- dpkg-1.14.16.6/scripts/Makefile.in	2008-01-26 20:38:33.000000000 +0100
+++ dpkg-1.14.16.6-fixed/scripts/Makefile.in	2008-02-13 03:02:33.000000000 +0100
@@ -251,7 +251,6 @@
 	cleanup-info.pl \
 	dpkg-divert.pl \
 	dpkg-statoverride.pl \
-	install-info.pl \
 	update-alternatives.pl \
 	changelog/debian.pl \
 	t/000_pod.t \
@@ -284,7 +283,6 @@
 
 CLEANFILES = \
 	$(bin_SCRIPTS) $(sbin_SCRIPTS) $(changelog_SCRIPTS) \
-	install-info install-info-stamp
 
 perllibdir = $(PERL_LIBDIR)
 nobase_dist_perllib_DATA = \
@@ -726,15 +724,7 @@
 	chmod +x $@
 
 # Automake has its own install-info rule, gah
-all-local: install-info-stamp
-install-info-stamp: $(srcdir)/install-info.pl
-	$(do_perl_subst) <$< >install-info
-	chmod +x install-info
-	touch $@
-
-install-exec-local: install-info-stamp
-	$(mkdir_p) $(DESTDIR)$(sbindir)
-	$(INSTALL_SCRIPT) install-info $(DESTDIR)$(sbindir)
+all-local:
 
 install-data-local:
 	$(mkdir_p) $(DESTDIR)$(sysconfdir)/alternatives
