--- a/src/combined/ffmpeg/ff_audio_decoder.c       Fri Sep 05 21:26:59 2008 +0200
+++ b/src/combined/ffmpeg/ff_audio_decoder.c       Sat Sep 13 12:48:10 2008 +0100
@@ -322,7 +322,7 @@ static void ff_audio_decode_data (audio_
 
     if (!this->output_open) {
       if (!this->audio_bits || !this->audio_sample_rate || !this->audio_channels) {
-        avcodec_decode_audio (this->context,
+        avcodec_decode_audio2 (this->context,
                               (int16_t *)this->decode_buffer,
                               &decode_buffer_size,
                               &this->buf[0],
--- a/src/combined/ffmpeg/ffmpeg_decoder.h       Fri Sep 05 21:26:59 2008 +0200
+++ b/src/combined/ffmpeg/ffmpeg_decoder.h       Sat Sep 13 12:48:10 2008 +0100
@@ -33,6 +33,10 @@
 #  include "../../libffmpeg/libavcodec/avcodec.h"
 #endif
 
+#if LIBAVCODEC_VERSION_MAJOR > 51
+#define bits_per_sample bits_per_coded_sample
+#endif
+
 typedef struct ff_codec_s {
   uint32_t          type;
   enum CodecID      id;
