diff -Naur xine-lib-1.1.15/src/xine-engine/buffer.h xine-lib-1.1.15.new/src/xine-engine/buffer.h
--- xine-lib-1.1.15/src/xine-engine/buffer.h	2008-07-16 23:01:56.000000000 +0200
+++ xine-lib-1.1.15.new/src/xine-engine/buffer.h	2008-09-03 19:52:41.000000000 +0200
@@ -676,7 +676,7 @@
 /* convert xine_waveformatex struct from little endian */
 void _x_waveformatex_le2me( xine_waveformatex *wavex ) XINE_PROTECTED;
 
-static inline _x_is_fourcc(void *ptr, void *tag) {
+static __inline int _x_is_fourcc(void *ptr, void *tag) {
   return memcmp(ptr, tag, 4) == 0;
 }
 
