Index: trunk/src/pentaxmn.cpp
===================================================================
--- trunk/src/pentaxmn.cpp	(revision 2958)
+++ trunk/src/pentaxmn.cpp	(revision 2959)
@@ -675,6 +675,7 @@
         { 0x0700, "smc PENTAX-DA 21mm F3.2 AL Limited" },
         { 0x074b, "Tamron SP AF 70-200mm F2.8 Di LD [IF] Macro (A001)" },
         { 0x07d6, "smc PENTAX-DA 35mm F2.4 AL" },
+        { 0x07d6, "smc PENTAX-DA 35mm F2.4 AL" },
         { 0x07d9, "smc PENTAX-DA 50-200mm F4-5.6 ED WR" },
         { 0x07da, "smc PENTAX-DA 18-55mm F3.5-5.6 AL WR" },
         { 0x07dc, "Tamron SP AF 10-24mm F3.5-4.5 Di II LD Aspherical [IF]" },
@@ -694,6 +695,8 @@
         { 0x07f2, "smc PENTAX-DA* 16-50mm F2.8 ED AL [IF] SDM (SDM not used)" },
         { 0x07f3, "smc PENTAX-DA 70mm F2.4 Limited" },
         { 0x07f4, "smc PENTAX-DA 21mm F3.2 AL Limited" },
+        { 0x0808, "SIGMA 18-250mm F3.5-6.3 DC OS HSM" },
+        { 0x08d7, "smc PENTAX-DA 18-135mm F3.5-5.6 ED AL [IF] DC WR" },
         { 0x08e2, "smc PENTAX-DA* 55mm F1.4 SDM" },
         { 0x08e3, "smc PENTAX DA* 60-250mm F4 [IF] SDM"},
         { 0x08e8, "smc PENTAX-DA 17-70mm F4 AL [IF] SDM" },
