Index: trunk/src/olympusmn.cpp
===================================================================
--- trunk/src/olympusmn.cpp	(revision 2981)
+++ trunk/src/olympusmn.cpp	(revision 2982)
@@ -1280,7 +1280,7 @@
             { { 0, 51,  0 }, N_("Olympus Zuiko Digital 25mm F2.8")                      },
             { { 0, 52,  0 }, N_("Olympus Zuiko Digital ED 9-18mm F4.0-5.6")             },
             { { 0, 53,  0 }, N_("Olympus Zuiko Digital 14-54mm F2.8-3.5 II")            },
-            { { 1,  1,  0 }, N_("Sigma 18-50mm F3.5-5.6")                               },
+            { { 1,  1,  0 }, N_("Sigma 18-50mm F3.5-5.6 DC")                            },
             { { 1,  1, 16 }, N_("Sigma 30mm F2.8 EX DN")                                },
             { { 1,  2,  0 }, N_("Sigma 55-200mm F4.0-5.6 DC")                           },
             { { 1,  2, 16 }, N_("Sigma 19mm F2.8 EX DN")                                },
@@ -1290,6 +1290,7 @@
             { { 1,  6,  0 }, N_("Sigma 50-500mm F4.0-6.3 EX DG APO HSM RF")             },
             { { 1,  7,  0 }, N_("Sigma 105mm F2.8 DG")                                  },
             { { 1,  8,  0 }, N_("Sigma 150mm F2.8 DG HSM")                              },
+            { { 1,  9,  0 }, N_("Sigma 18-50mm F2.8 EX DC Macro")                       },
             { { 1, 16,  0 }, N_("Sigma 24mm F1.8 EX DG Aspherical Macro")               },
             { { 1, 17,  0 }, N_("Sigma 135-400mm F4.5-5.6 DG ASP APO RF")               },
             { { 1, 18,  0 }, N_("Sigma 300-800mm F5.6 EX DG APO")                       },
