Index: trunk/src/canonmn.cpp
===================================================================
--- trunk/src/canonmn.cpp	(revision 2993)
+++ trunk/src/canonmn.cpp	(revision 2994)
@@ -729,7 +729,8 @@
         { 161, printCsLensByFocalLength },
         { 169, printCsLensByFocalLength },
         { 173, printCsLensByFocalLength }, // works partly
-        { 174, printCsLensByFocalLength }  // not tested
+        { 174, printCsLensByFocalLength }, // not tested
+        { 183, printCsLensByFocalLength }  // not tested
     };
 
     //! FlashActivity, tag 0x001c
