Index: trunk/src/nikonmn.cpp
===================================================================
--- trunk/src/nikonmn.cpp	(revision 2968)
+++ trunk/src/nikonmn.cpp	(revision 2969)
@@ -2146,6 +2146,7 @@
 {0x00,0x49,0x30,0x48,0x22,0x2B,0x00,0x02,0x00,0x00,0x02, "Tamron", "166D", "SP AF 20-40mm F/2.7-3.5"},
 {0x0E,0x4A,0x31,0x48,0x23,0x2D,0x0E,0x02,0x00,0x00,0x02, "Tamron", "166D", "SP AF 20-40mm F/2.7-3.5"},
 //M                                         "Tamron" "266D"   "SP AF 20-40mm F/2.7-3.5 Aspherical-IF";
+{0xFE,0x48,0x37,0x5C,0x24,0x24,0xDF,0x0E,0x00,0x00,0x02, "Tamron", "A007", "SP 24-70mm F/2.8 Di VC USD"},
 //M                                         "Tamron" "73D"    "AF 24-70mm F/3.3-5.6 Aspherical";
 {0x45,0x41,0x37,0x72,0x2C,0x3C,0x48,0x02,0x00,0x00,0x02, "Tamron", "190D", "SP AF 24-135mm F/3.5-5.6 AD Aspherical (IF) Macro"},
 //M                                         "Tamron" "159D"   "AF 28-70mm F/3.5-4.5";
