Index: trunk/src/easyaccess.cpp
===================================================================
--- trunk/src/easyaccess.cpp	(revision 2987)
+++ trunk/src/easyaccess.cpp	(revision 2988)
@@ -256,6 +256,7 @@
     ExifData::const_iterator lensName(const ExifData& ed)
     {
         static const char* keys[] = {
+            "Exif.Photo.LensModel",
             "Exif.CanonCs.LensType",
             "Exif.NikonLd1.LensIDNumber",
             "Exif.NikonLd2.LensIDNumber",
