--- a/src/exif.cpp	(revision 2743)
+++ b/src/exif.cpp	(revision 2744)
@@ -752,6 +752,9 @@
             { pttLen, "Exif.Pentax.PreviewLength"                     },
             { pttTag, "Exif.Pentax.PreviewOffset"                     },
             { pttTag, "Exif.Pentax.PreviewResolution"                 },
+            { pttLen, "Exif.PentaxDng.PreviewLength"                  },
+            { pttTag, "Exif.PentaxDng.PreviewOffset"                  },
+            { pttTag, "Exif.PentaxDng.PreviewResolution"              },
             { pttLen, "Exif.SamsungPreview.JPEGInterchangeFormatLength" },
             { pttIfd, "SamsungPreview"                                },
             { pttLen, "Exif.Thumbnail.StripByteCounts"                },
