Index: trunk/src/minoltamn.cpp
===================================================================
--- trunk/src/minoltamn.cpp	(revision 2988)
+++ trunk/src/minoltamn.cpp	(revision 2989)
@@ -1863,15 +1863,7 @@
                  "Tamron SP AF 90mm F2.5"},
         { 45751, "1.4x Teleconverter " },
         { 45851, "Tamron SP AF 300mm F2.8 LD IF" },
-        { 65535, "T-Mount | "
-                 "Arax MC 35mm F2.8 Tilt+Shift | "
-                 "Arax MC 80mm F2.8 Tilt+Shift | "
-                 "Zenitar MF 16mm F2.8 Fisheye M42 | "
-                 "Samyang 500mm Mirror F8.0 | "
-                 "Pentacon Auto 135mm F2.8 | "
-                 "Pentacon Auto 29mm F2.8 | "
-                 "Helios 44-2 58mm F2.0 | "
-                 "No Lens" }
+        { 65535, "Manual Lens or No Lens" }
     };
 
     std::ostream& printMinoltaSonyLensID(std::ostream& os, const Value& value, const ExifData* metadata)
