diff -Naur exiv2-0.16.orig/xmpsdk/src/XMPCore_Impl.hpp exiv2-0.16/xmpsdk/src/XMPCore_Impl.hpp
--- exiv2-0.16.orig/xmpsdk/src/XMPCore_Impl.hpp	2008-04-18 21:41:14.674009158 -0700
+++ exiv2-0.16/xmpsdk/src/XMPCore_Impl.hpp	2008-04-18 21:42:24.759030670 -0700
@@ -21,6 +21,7 @@
 
 #include <vector>
 #include <string>
+#include <cstring>
 #include <map>
 
 #include <cassert>
diff -Naur exiv2-0.16.orig/xmpsdk/src/XMPMeta.cpp exiv2-0.16/xmpsdk/src/XMPMeta.cpp
--- exiv2-0.16.orig/xmpsdk/src/XMPMeta.cpp	2008-04-18 21:41:14.674009158 -0700
+++ exiv2-0.16/xmpsdk/src/XMPMeta.cpp	2008-04-18 21:41:59.981260226 -0700
@@ -24,6 +24,8 @@
 	#include <iostream>
 #endif
 
+#include <cstring>
+
 using namespace std;
 
 #if XMP_WinBuild
diff -Naur exiv2-0.16.orig/xmpsdk/src/XMPMeta-GetSet.cpp exiv2-0.16/xmpsdk/src/XMPMeta-GetSet.cpp
--- exiv2-0.16.orig/xmpsdk/src/XMPMeta-GetSet.cpp	2008-04-18 21:41:14.675010110 -0700
+++ exiv2-0.16/xmpsdk/src/XMPMeta-GetSet.cpp	2008-04-18 21:42:47.998017709 -0700
@@ -25,6 +25,8 @@
 	#include <iostream>
 #endif
 
+#include <cstring>
+
 using namespace std;
 
 #if XMP_WinBuild
diff -Naur exiv2-0.16.orig/xmpsdk/src/XMPMeta-Parse.cpp exiv2-0.16/xmpsdk/src/XMPMeta-Parse.cpp
--- exiv2-0.16.orig/xmpsdk/src/XMPMeta-Parse.cpp	2008-04-18 21:41:14.674009158 -0700
+++ exiv2-0.16/xmpsdk/src/XMPMeta-Parse.cpp	2008-04-18 21:44:03.867034243 -0700
@@ -23,6 +23,8 @@
 	#include <iostream>
 #endif
 
+#include <cstring>
+
 using namespace std;
 
 #if XMP_WinBuild
