--- configure.ac.orig	2019-05-16 03:42:56.000000000 -0700
+++ configure.ac	2019-06-01 18:23:07.029227730 -0700
@@ -251,7 +251,7 @@
 )
 AC_ARG_WITH(libvncserver,
     [  --with-libvncserver     make rdp2vnc],
-    [ 
+    [AS_IF([test "x$withval" != xno],
     VNCINC=`$vncserverconfig --cflags`
     AC_SUBST(VNCINC)
     LDVNC=`$vncserverconfig --libs`
@@ -260,7 +260,7 @@
     AC_SUBST(VNCLINK)
     RDP2VNCTARGET="rdp2vnc"
     AC_SUBST(RDP2VNCTARGET)
-    ]
+    )]
 )
 
 #
