diff -urp connmon-0.13.0/src/connmon/Makefile connmon-0.13.0.fixed/src/connmon/Makefile
--- connmon-0.13.0/src/connmon/Makefile	2002-10-10 21:29:30.000000000 +0200
+++ connmon-0.13.0.fixed/src/connmon/Makefile	2003-01-19 18:28:40.000000000 +0100
@@ -21,7 +21,7 @@ clean:
 	rm -f *.o connmon config.c
 
 connmon: $(LDOBJS)
-	$(CC) $(CFLAGS) -o connmon $(LDFLAGS) $(LDOBJS)
+	$(CC) $(CFLAGS) -o connmon $(LDOBJS) $(LDFLAGS)
 
 %.o: %.c connmon.h curses-util.h
 	$(CC) $(CFLAGS) -c $<
diff -urp connmon-0.13.0/src/gconnmon/Makefile connmon-0.13.0.fixed/src/gconnmon/Makefile
--- connmon-0.13.0/src/gconnmon/Makefile	2002-10-10 21:29:30.000000000 +0200
+++ connmon-0.13.0.fixed/src/gconnmon/Makefile	2003-01-19 18:34:30.000000000 +0100
@@ -20,7 +20,7 @@ clean:
 	rm -f *.o gconnmon
 
 gconnmon: $(LDOBJS)
-	$(CC) -o $@ $(LIBS) $(LDOBJS)
+	$(CC) -o $@ $(LDOBJS) $(LIBS)
 
 %.o: %.c gconnmon.h
 	$(CC) $(CFLAGS) -c $<
