diff -Naur mozilla.orig/xpfe/bootstrap/Makefile.in mozilla/xpfe/bootstrap/Makefile.in
--- mozilla.orig/xpfe/bootstrap/Makefile.in	2005-01-05 10:13:48.000000000 -0800
+++ mozilla/xpfe/bootstrap/Makefile.in	2005-01-05 10:30:19.657789128 -0800
@@ -357,7 +357,7 @@
 
 CXXFLAGS	+= $(MOZ_TOOLKIT_REGISTRY_CFLAGS)
 
-LOCAL_INCLUDES	= -I$(srcdir) -I.
+LOCAL_INCLUDES	= -I$(srcdir) -I. -I$(DIST)/sdk/include
 
 ifneq (,$(filter-out OS2 WINNT,$(OS_ARCH)))
 $(MOZ_APP_NAME).1: mozilla.man.in Makefile.in Makefile $(DEPTH)/config/autoconf.mk
