From e5ff7c27ba4652aeeef461cad6ecf0de08c8a864 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Thu, 24 Jan 2013 19:02:13 +0100
Subject: [PATCH] fix gettext dependency

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6157d69..ef50bf3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -154,7 +154,7 @@ if(USE_NLS)
   find_package(Gettext)
   if(Gettext_FOUND)
     include_directories(${Gettext_INCLUDE_DIRS})
-    list(APPEND LIBS ${Gettext_LIBRARIES})
+    list(APPEND LIBS ${Gettext_LIBRARIES} -lintl)
     add_definitions("-DUSE_GETTEXT")
     message(STATUS "Internationalization: Enabled")
   else()
-- 
1.8.1

