From ab9d6a3eca3a5a17aa5016d53b050ae1da9ed7ff Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Fri, 22 Mar 2013 18:19:24 +0100
Subject: [PATCH] remove -Werror from CXXFLAGS

---
 src/CMakeLists.txt | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a43d2e6..30ac41d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -277,12 +277,6 @@ endif(NOT WIN32)
 
 if(NOT CUSTOM_CFLAGS)
 
-if(NOT APPLE AND NOT CMAKE_C_COMPILER_ID STREQUAL "Clang")
-  # apple build fails on deprecated warnings..
-  # and too many warnings reported by Clang for now
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
-endif(NOT APPLE AND NOT CMAKE_C_COMPILER_ID STREQUAL "Clang")
-
 if(NOT BINARY_PACKAGE_BUILD)
 	MESSAGE("-- Checking for -march=native support")
 	CHECK_C_COMPILER_FLAG("-march=native" MARCHNATIVE)
-- 
1.8.1.3

