From 7a26b2751a34f485f18a4106e489f9e09415574e Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sun, 7 Aug 2016 15:28:58 +0200
Subject: [PATCH] do not build darktable-viewer

---
 src/dtview/CMakeLists.txt | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/src/dtview/CMakeLists.txt b/src/dtview/CMakeLists.txt
index d45bb0c..c220712 100644
--- a/src/dtview/CMakeLists.txt
+++ b/src/dtview/CMakeLists.txt
@@ -12,15 +12,15 @@ include_directories(SYSTEM ${SDL_INCLUDE_DIR})
 include_directories(SYSTEM ${OPENGL_INCLUDE_DIR})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/..)
 include_directories(${CMAKE_CURRENT_BINARY_DIR}/..)
-add_executable(darktable-viewer main.c)
-
-set_target_properties(darktable-viewer PROPERTIES CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
-set_target_properties(darktable-viewer PROPERTIES CMAKE_INSTALL_RPATH_USE_LINK_PATH FALSE)
-if(APPLE)
-  set_target_properties(darktable-viewer PROPERTIES INSTALL_RPATH @loader_path/../${LIB_INSTALL}/darktable)  
-else(APPLE)
-  set_target_properties(darktable-viewer PROPERTIES INSTALL_RPATH $ORIGIN/../${LIB_INSTALL}/darktable)
-endif(APPLE)
-set_target_properties(darktable-viewer PROPERTIES LINKER_LANGUAGE C)
-target_link_libraries(darktable-viewer lib_darktable ${SDL_LIBRARY} ${OPENGL_LIBRARIES})
-install(TARGETS darktable-viewer DESTINATION bin)
+#add_executable(darktable-viewer main.c)
+#
+#set_target_properties(darktable-viewer PROPERTIES CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
+#set_target_properties(darktable-viewer PROPERTIES CMAKE_INSTALL_RPATH_USE_LINK_PATH FALSE)
+#if(APPLE)
+#  set_target_properties(darktable-viewer PROPERTIES INSTALL_RPATH @loader_path/../${LIB_INSTALL}/darktable)  
+#else(APPLE)
+#  set_target_properties(darktable-viewer PROPERTIES INSTALL_RPATH $ORIGIN/../${LIB_INSTALL}/darktable)
+#endif(APPLE)
+#set_target_properties(darktable-viewer PROPERTIES LINKER_LANGUAGE C)
+#target_link_libraries(darktable-viewer lib_darktable ${SDL_LIBRARY} ${OPENGL_LIBRARIES})
+#install(TARGETS darktable-viewer DESTINATION bin)
-- 
2.9.2

