From e8581a1d45d08ed034843e3e724293c6cd4916ee Mon Sep 17 00:00:00 2001
From: Pascal de Bruijn <pmjdebruijn@pcode.nl>
Date: Fri, 8 Feb 2013 20:23:22 +0100
Subject: [PATCH] wb_presets: Add Megaman liliput-plus (SAMSUNG NX100)

---
 src/iop/wb_presets.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/src/iop/wb_presets.c b/src/iop/wb_presets.c
index 701887b..bdb724b 100644
--- a/src/iop/wb_presets.c
+++ b/src/iop/wb_presets.c
@@ -4269,6 +4269,7 @@ const wb_data wb_preset[] = {
   { "SAMSUNG", "NX100", Flash, 5,		{ 2.249317, 1, 0.859903, 0 } },
   { "SAMSUNG", "NX100", Flash, 6,		{ 2.284664, 1, 0.836022, 0 } },
   { "SAMSUNG", "NX100", Flash, 7,		{ 2.320238, 1, 0.812302, 0 } },
+  { "SAMSUNG", "NX100", "Megaman liliput plus", 0, { 1.800000, 1, 1.350000, 0 } },
   { "SAMSUNG", "NX100", "5000K", 0,		{ 1.684339, 1, 1.094063, 0 } },
   { "SAMSUNG", "NX100", "5500K", 0,		{ 1.692022, 1, 0.843950, 0 } },
   { "SAMSUNG", "NX100", "6500K", 0,		{ 2.110371, 1, 0.879107, 0 } },
-- 
1.7.10.4

