From 3120ed5e8a744ea018ad57ea538d3b7aead46b7a Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Fri, 12 Apr 2024 17:47:34 +0200
Subject: [PATCH] Define _POSIX_C_SOURCE to expose popen/pclose

Origin: Source Mage
Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 front/sdl/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/front/sdl/main.c b/front/sdl/main.c
index 8f088aa09e55..0faacad1efb0 100644
--- a/front/sdl/main.c
+++ b/front/sdl/main.c
@@ -21,6 +21,7 @@
 //
 // 3. This notice may not be removed or altered from any source distribution.
 
+#define _POSIX_C_SOURCE 2
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdbool.h>
-- 
2.43.0

