From 8c479ed9ca7490fa5e5620aabf610ba3e0beb3fc Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 30 Jun 2019 13:17:34 +0200
Subject: [PATCH] fix build error with newer versions of mesa

---
 src/apps/ociodisplay/main.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/apps/ociodisplay/main.cpp b/src/apps/ociodisplay/main.cpp
index f8184ad..c29f03f 100644
--- a/src/apps/ociodisplay/main.cpp
+++ b/src/apps/ociodisplay/main.cpp
@@ -56,7 +56,6 @@ namespace OCIO = OCIO_NAMESPACE;
 #else
 #include <GL/glew.h>
 #include <GL/gl.h>
-#include <GL/glext.h>
 #include <GL/glut.h>
 #endif
 
-- 
2.22.0

