diff -urNp wifi-radar-20060511.orig/Makefile wifi-radar-20060511/Makefile
--- wifi-radar-20060511.orig/Makefile	2006-05-11 08:42:52.000000000 +0200
+++ wifi-radar-20060511/Makefile	2006-05-11 08:43:37.000000000 +0200
@@ -42,8 +42,8 @@ install:	all
 	test -d ${mandir}/man5 || install -d -m 755 ${mandir}/man5
 	install -m 644 wifi-radar.conf.5 	${mandir}/man5
 	
-	test -d ${initdir} || install -d -m 755 ${initdir}
-	install -m 755 init_script 			${initdir}/wifi-radar
+#	test -d ${initdir} || install -d -m 755 ${initdir}
+#	install -m 755 init_script 			${initdir}/wifi-radar
 	
 	test -d ${pixmapsdir} || install -d -m 755 ${pixmapsdir}
 	install -m 644 pixmaps/wifi-radar.svg	${pixmapsdir}
