--- wifi-radar-1.9.3/wifi-radar		2005-03-28 03:27:37.000000000 -0600
+++ wifi-radar-1.9.3/wifi-radar.new	2005-03-31 09:51:48.000000000 -0600
@@ -26,7 +26,7 @@
 # You may set this to true for cards that require the interface to be brought up first
 IFUP_REQUIRED	= False
 # Where the conf file is could be different for your distro.
-CONF_FILE		= "/etc/conf.d/wifi-radar.conf"
+CONF_FILE		= "/etc/sysconfig/wifi-radar.conf"
 IWLIST_COMMAND	= "iwlist"
 IWCONFIG_COMMAND= "iwconfig"
 IFCONFIG_COMMAND= "ifconfig"
