--- Fig.old	2003-01-06 10:34:21.000000000 -0800
+++ Fig	2003-01-06 10:36:06.000000000 -0800
@@ -18,9 +18,9 @@
 
 ! Browser - put your favorite browser here.  
 ! 		This is for viewing the xfig html reference.
-! For netscape, this command will open the help pages in a running netscape,
-!     or start a new netscape if one isn't already running
-Fig.browser:	netscape -remote 'openFile(%f)' || netscape %f
+! For mozilla, this command will open the help pages in a running mozilla,
+!     or start a new mozilla if one isn't already running
+Fig.browser:	mozilla -remote 'openFile(%f)' || mozilla %f
 
 ! pdfviewer - put your favorite pdf viewer here.  
 !		This is for viewing the xfig how-to guide and man pages
