diff -urN openMotif-2.2.2.orig/clients/mwm/WmResParse.c openMotif-2.2.2/clients/mwm/WmResParse.c
--- openMotif-2.2.2.orig/clients/mwm/WmResParse.c	2002-01-10 21:55:37.000000000 +0100
+++ openMotif-2.2.2/clients/mwm/WmResParse.c	2003-06-04 22:20:22.000000000 +0200
@@ -2411,7 +2411,7 @@
 	if (MwmBehavior)
 	{
 	    strcpy(cfileName, LIBDIR);
-	    strncat(cfileName, "/", MAXWMPATH-strlen(cfileName));
+	    strncat(cfileName, "/mwm/", MAXWMPATH-strlen(cfileName));
 	    strncat(cfileName, LANG, MAXWMPATH-strlen(cfileName));
 	    strncat(cfileName, SLASH_MWMRC, MAXWMPATH - strlen(cfileName));
 	}
@@ -2427,7 +2427,7 @@
 	* Try /$LANG/system.mwmrc within the install tree
 	*/
 	strcpy(cfileName, LIBDIR);
-	strncat(cfileName, "/", MAXWMPATH-strlen(cfileName));
+	strncat(cfileName, "/mwm/", MAXWMPATH-strlen(cfileName));
 	strncat(cfileName, LANG, MAXWMPATH-strlen(cfileName));
 	strncat(cfileName, SLASH_MWMRC, MAXWMPATH - strlen(cfileName));
 #endif /* WSM */
@@ -2449,7 +2449,7 @@
     if (MwmBehavior)
     {
 	strcpy(cfileName, LIBDIR);
-	strncat(cfileName, SLASH_MWMRC, MAXWMPATH - strlen(cfileName));
+	strncat(cfileName, "/mwm" SLASH_MWMRC, MAXWMPATH - strlen(cfileName));
 #ifdef PANELIST
 	fileP = fopen (cfileName, "r");
 #else /* PANELIST */
@@ -2473,7 +2473,7 @@
      * Try /system.mwmrc within the install tree
      */
     strcpy(cfileName, LIBDIR);
-    strncat(cfileName, SLASH_MWMRC, MAXWMPATH - strlen(cfileName));
+    strncat(cfileName, "/mwm" SLASH_MWMRC, MAXWMPATH - strlen(cfileName));
 
     if (LANG != NULL) 
     {
