--- openMotif-2.2.3/lib/Xm/XpmWrFFrI.c.CAN-2004-0914_sec8	2004-11-29 15:04:10.105373839 +0100
+++ openMotif-2.2.3/lib/Xm/XpmWrFFrI.c	2004-11-29 15:04:10.150367679 +0100
@@ -322,10 +322,7 @@
 #ifndef NO_ZPIPE
 	size_t len = strlen(filename);
 
-	if(len == 0                        ||
-	   filename[0] == '/'              ||
-	   strstr(filename, "../") != NULL ||
-	   filename[len-1] == '/')
+	if(len == 0)
 		return(XpmOpenFailed);
 
 	if (len > 2 && !strcmp(".Z", filename + (len - 2))) {
