--- qtiplot/scidavis.pro.orig	2007-07-12 17:42:26.000000000 +0200
+++ qtiplot/scidavis.pro	2007-07-12 17:45:40.000000000 +0200
@@ -26,11 +26,12 @@
 #############################################################################
 
 INCLUDEPATH       += ../3rdparty/muParser
-INCLUDEPATH       += ../3rdparty/qwtplot3d/include
-INCLUDEPATH       += ../3rdparty/qwt/src
 INCLUDEPATH       += ../3rdparty/liborigin
-INCLUDEPATH       += ../3rdparty/gsl/include
 INCLUDEPATH       += ../3rdparty/zlib123/include
+INCLUDEPATH       += /usr/include/qwtplot3d
+INCLUDEPATH       += /usr/include/qwt
+INCLUDEPATH       += /usr/include/gsl
+
 
 ##################### 3rd PARTY LIBRARIES SECTION ###########################
 #!!! Warning: You must modify these paths according to your computer settings
@@ -43,18 +44,14 @@
 unix:LIBS         += -L /usr/lib$${libsuff}
 
 # statically link against Qwt(3D) in 3rdparty
-unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
 # dynamically link against Qwt(3D) installed system-wide
 # WARNING: make sure they are compiled against Qt4
-#unix:LIBS         += -lqwtplot3d
-#unix:LIBS         += -lqwt
+unix:LIBS         += -lqwtplot3d
+unix:LIBS         += -lqwt
 
 # statically link against GSL in 3rdparty
-unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
 #dynamically link against GSL installed system-wide
-#unix:LIBS         += -lgsl -lgslcblas
+unix:LIBS         += -lgsl -lgslcblas
 
 ##################### Windows ###############################################
 
