--- scidavis-0.2.4/scidavis/scidavis.pro.orig	2011-10-01 21:41:35.508464199 +0200
+++ scidavis-0.2.4/scidavis/scidavis.pro	2011-10-01 21:42:02.618143390 +0200
@@ -114,6 +114,8 @@
 ##dynamically link against GSL and zlib installed system-wide
 #unix:LIBS         += -lgsl -lgslcblas -lz -lmuparser
 
+unix:INCLUDEPATH += /usr/include/qt4/QtAssistant
+
 #############################################################################
 ### Default settings for Windows
 #############################################################################
