--- scidavis-0.2.4/scidavis/scidavis.pro.orig	2011-10-01 18:33:46.959397655 +0200
+++ scidavis-0.2.4/scidavis/scidavis.pro	2011-10-01 18:35:01.406552063 +0200
@@ -87,11 +87,11 @@
 ### are compiled against Qt4), dynamically against everything else.
 #############################################################################
 
-unix:INCLUDEPATH  += ../3rdparty/qwtplot3d/include
-unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
+#unix:INCLUDEPATH  += ../3rdparty/qwtplot3d/include
+#unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
 
-unix:INCLUDEPATH  += ../3rdparty/qwt/src
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+#unix:INCLUDEPATH  += ../3rdparty/qwt/src
+#unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
 
 unix:LIBS         += -L/usr/lib$${libsuff}
 unix:LIBS         += -lgsl -lgslcblas -lz
@@ -108,9 +108,9 @@
 ## WARNING: make sure they are compiled against >= Qt4.2
 ## Mixing Qt 4.2 and Qt >= 4.3 compiled stuff may also 
 ## cause problems.
-#unix:INCLUDEPATH  += /usr/include/qwtplot3d
-#unix:LIBS         += -lqwtplot3d
-#unix:LIBS         += -lqwt
+unix:INCLUDEPATH  += /usr/include/qwtplot3d
+unix:LIBS         += -lqwtplot3d
+unix:LIBS         += -lqwt
 ##dynamically link against GSL and zlib installed system-wide
 #unix:LIBS         += -lgsl -lgslcblas -lz -lmuparser
 
