diff -Naur thunderbird.orig/chrome/app-chrome.manifest thunderbird/chrome/app-chrome.manifest
--- thunderbird.orig/chrome/app-chrome.manifest	1969-12-31 16:00:00.000000000 -0800
+++ thunderbird/chrome/app-chrome.manifest	2008-02-12 09:12:43.594567541 -0800
@@ -0,0 +1,73 @@
+
+locale	wallet	en-US	jar:en-US.jar!/locale/en-US/wallet/
+
+content	enigmime	jar:enigmime.jar!/content/enigmime/
+
+skin	enigmail	classic/1.0	jar:enigmail-skin-tbird.jar!/skin/classic/enigmail/
+
+overlay	chrome://messenger/content/messengercompose/messengercompose.xul	chrome://enigmail/content/enigmailCheckLanguage.xul
+overlay	chrome://messenger/content/messengercompose/messengercompose.xul	chrome://enigmail/content/enigmailMsgComposeOverlay.xul
+overlay	chrome://messenger/content/mailWindowOverlay.xul	chrome://enigmail/content/enigmailCheckLanguage.xul
+overlay	chrome://messenger/content/mailWindowOverlay.xul	chrome://enigmail/content/enigmailMessengerOverlay.xul
+overlay	chrome://messenger/content/msgHdrViewOverlay.xul	chrome://enigmail/content/enigmailMsgHdrViewOverlay.xul
+overlay	chrome://messenger/content/msgPrintEngine.xul	chrome://enigmail/content/enigmailMsgPrintOverlay.xul
+overlay	chrome://messenger/content/am-identity-edit.xul	chrome://enigmail/content/enigmailAmIdEditOverlay.xul
+overlay	chrome://messenger/content/am-identity-edit.xul	chrome://enigmail/content/enigmailEditIdentity.xul
+overlay	chrome://messenger/content/addressbook/addressbook.xul	chrome://enigmail/content/enigmailAbCardViewOverlay.xul
+overlay	chrome://messenger/content/addressbook/abContactsPanel.xul	chrome://enigmail/content/enigmailAbContactsPanel.xul
+overlay	chrome://global/content/customizeToolbar.xul	chrome://enigmail/content/enigmailCustToolOverlay.xul
+overlay	chrome://communicator/content/pref/preftree.xul	chrome://enigmail/content/enigmailPrefsOverlay.xul
+overlay	chrome://enigmail/content/am-enigprefs.xul	chrome://enigmail/content/enigmailEditIdentity.xul
+overlay	chrome://enigmail/content/am-enigprefs-edit.xul	chrome://enigmail/content/enigmailEditIdentity.xul
+overlay	chrome://browser/content/browser.xul	chrome://enigmail/content/enigmailNavigatorOverlay.xul
+content	enigmail	jar:enigmail.jar!/content/enigmail/
+
+locale	enigmail	en-US	jar:enigmail.jar!/locale/en-US/enigmail/
+
+locale	enigmail	en-US	jar:enigmail-en-US.jar!/locale/en-US/enigmail/
+
+skin	enigmail	modern/1.0	jar:enigmail-skin.jar!/skin/modern/enigmail/
+
+skin	enigmail	classic/1.0	jar:enigmail-skin.jar!/skin/classic/enigmail/
+
+locale	enigmail	ca-AD	jar:enigmail-locale.jar!/locale/ca-AD/enigmail/
+
+locale	enigmail	de-AT	jar:enigmail-locale.jar!/locale/de-AT/enigmail/
+
+locale	enigmail	de-DE	jar:enigmail-locale.jar!/locale/de-DE/enigmail/
+
+locale	enigmail	el	jar:enigmail-locale.jar!/locale/el/enigmail/
+
+locale	enigmail	es-ES	jar:enigmail-locale.jar!/locale/es-ES/enigmail/
+
+locale	enigmail	fi-FI	jar:enigmail-locale.jar!/locale/fi-FI/enigmail/
+
+locale	enigmail	fr-FR	jar:enigmail-locale.jar!/locale/fr-FR/enigmail/
+
+locale	enigmail	hu-HU	jar:enigmail-locale.jar!/locale/hu-HU/enigmail/
+
+locale	enigmail	it-IT	jar:enigmail-locale.jar!/locale/it-IT/enigmail/
+
+locale	enigmail	ja-JP	jar:enigmail-locale.jar!/locale/ja-JP/enigmail/
+
+locale	enigmail	ko-KR	jar:enigmail-locale.jar!/locale/ko-KR/enigmail/
+
+locale	enigmail	nb-NO	jar:enigmail-locale.jar!/locale/nb-NO/enigmail/
+
+locale	enigmail	pl-PL	jar:enigmail-locale.jar!/locale/pl-PL/enigmail/
+
+locale	enigmail	pt-BR	jar:enigmail-locale.jar!/locale/pt-BR/enigmail/
+
+locale	enigmail	pt-PT	jar:enigmail-locale.jar!/locale/pt-PT/enigmail/
+
+locale	enigmail	ru-RU	jar:enigmail-locale.jar!/locale/ru-RU/enigmail/
+
+locale	enigmail	sl-SI	jar:enigmail-locale.jar!/locale/sl-SI/enigmail/
+
+locale	enigmail	sv-SE	jar:enigmail-locale.jar!/locale/sv-SE/enigmail/
+
+locale	enigmail	tr-TR	jar:enigmail-locale.jar!/locale/tr-TR/enigmail/
+
+locale	enigmail	zh-CN	jar:enigmail-locale.jar!/locale/zh-CN/enigmail/
+
+locale	enigmail	zh-TW	jar:enigmail-locale.jar!/locale/zh-TW/enigmail/
