--- mozilla/configure.vis	2009-08-19 14:43:58.000000000 +0200
+++ mozilla/configure	2009-08-19 15:15:11.000000000 +0200
@@ -8015,7 +8015,7 @@ EOF
                       if ${CC-cc} -Werror -S conftest.c -o conftest.s >/dev/null 2>&1; then
                         if egrep '\.(hidden|extern_private).*foo_hidden' conftest.s >/dev/null; then
                           if ! egrep '\.(hidden|extern_private).*foo_default' conftest.s > /dev/null; then
-                            ac_cv_visibility_pragma=yes
+                            ac_cv_visibility_pragma=no
                           fi
                         fi
                       fi



