diff -Naur mozilla.orig/configure mozilla/configure
--- mozilla.orig/configure	2006-04-25 13:40:12.593290920 -0700
+++ mozilla/configure	2006-04-25 13:41:41.698744832 -0700
@@ -7650,7 +7650,7 @@
                       if ${CC-cc} -Werror -S conftest.c -o conftest.s >/dev/null 2>&1; then
                         if grep '\.hidden.*foo_hidden' conftest.s >/dev/null; then
                           if ! grep '\.hidden.*foo_default' conftest.s > /dev/null; then
-                            ac_cv_visibility_pragma=yes
+                            ac_cv_visibility_pragma=no
                           fi
                         fi
                       fi
