diff -Naur dillo2-cvs/CVS/Entries dillo2-cvs-fixed/CVS/Entries
--- dillo2-cvs/CVS/Entries	2008-08-26 20:23:07.000000000 +0200
+++ dillo2-cvs-fixed/CVS/Entries	2008-08-26 20:17:28.420559878 +0200
@@ -11,9 +11,4 @@
 /configure.in/1.18/Thu Jul  3 13:11:55 2008//THEAD
 /dillo2rc/1.1/Sat Aug 23 21:53:05 2008//THEAD
 /install-dpi-local/1.1.1.1/Sat Oct  6 22:36:34 2007//THEAD
-D/dlib////
-D/doc////
-D/dpi////
-D/dpid////
-D/dpip////
-D/src////
+D
diff -Naur dillo2-cvs/CVS/Entries.Log dillo2-cvs-fixed/CVS/Entries.Log
--- dillo2-cvs/CVS/Entries.Log	1970-01-01 01:00:00.000000000 +0100
+++ dillo2-cvs-fixed/CVS/Entries.Log	2008-08-26 20:17:28.420559878 +0200
@@ -0,0 +1,6 @@
+A D/dlib////
+A D/doc////
+A D/dpi////
+A D/dpid////
+A D/dpip////
+A D/src////
diff -Naur dillo2-cvs/src/CVS/Entries dillo2-cvs-fixed/src/CVS/Entries
--- dillo2-cvs/src/CVS/Entries	2008-08-26 20:23:07.000000000 +0200
+++ dillo2-cvs-fixed/src/CVS/Entries	2008-08-26 20:17:28.410559229 +0200
@@ -72,4 +72,4 @@
 /url.h/1.5/Sat May 10 19:52:09 2008//THEAD
 /web.cc/1.6/Fri Jun 27 18:21:53 2008//THEAD
 /web.hh/1.1.1.1/Sat Oct  6 22:36:34 2007//THEAD
-D/IO////
+D
diff -Naur dillo2-cvs/src/CVS/Entries.Log dillo2-cvs-fixed/src/CVS/Entries.Log
--- dillo2-cvs/src/CVS/Entries.Log	1970-01-01 01:00:00.000000000 +0100
+++ dillo2-cvs-fixed/src/CVS/Entries.Log	2008-08-26 20:17:28.410559229 +0200
@@ -0,0 +1 @@
+A D/IO////
diff -Naur dillo2-cvs/src/Makefile.am dillo2-cvs-fixed/src/Makefile.am
--- dillo2-cvs/src/Makefile.am	2008-08-23 23:53:05.000000000 +0200
+++ dillo2-cvs-fixed/src/Makefile.am	2008-08-26 20:19:47.598031864 +0200
@@ -1,6 +1,6 @@
 AM_CPPFLAGS=-DDILLORC_SYS='"$(sysconfdir)/dillo2rc"' @LIBJPEG_CPPFLAGS@
-AM_CFLAGS = -I../../dw-testbed @LIBPNG_CFLAGS@
-AM_CXXFLAGS = -I../../dw-testbed @LIBPNG_CFLAGS@ @LIBFLTK_CXXFLAGS@
+AM_CFLAGS = -I/usr/include/dw @LIBPNG_CFLAGS@
+AM_CXXFLAGS = -I/usr/include/dw @LIBPNG_CFLAGS@ @LIBFLTK_CXXFLAGS@
 
 SUBDIRS = IO
 
@@ -10,10 +10,10 @@
 	../dlib/libDlib.a \
 	../dpip/libDpip.a \
 	IO/libDiof.a \
-	../../dw-testbed/dw/libDw-widgets.a \
-	../../dw-testbed/dw/libDw-fltk.a \
-	../../dw-testbed/dw/libDw-core.a \
-	../../dw-testbed/lout/liblout.a \
+	/usr/lib/dw/libDw-widgets.a \
+	/usr/lib/dw/libDw-fltk.a \
+	/usr/lib/dw/libDw-core.a \
+	/usr/lib/lout/liblout.a \
 	@LIBJPEG_LIBS@ @LIBPNG_LIBS@ @LIBFLTK_LIBS@ @LIBZ_LIBS@ @LIBICONV_LIBS@
 
 dillo_fltk_SOURCES = \
diff -Naur dillo2-cvs/src/srch dillo2-cvs-fixed/src/srch
--- dillo2-cvs/src/srch	2007-10-07 00:36:34.000000000 +0200
+++ dillo2-cvs-fixed/src/srch	2008-08-26 20:21:17.973869139 +0200
@@ -28,6 +28,5 @@
 
 #egrep $FLAGS "$1" dw/*[ch]
 #egrep $FLAGS "$1" lout/*[ch]
-egrep $FLAGS "$1" ../../dw-testbed/dw/*[ch]
-egrep $FLAGS "$1" ../../dw-testbed/lout/*[ch]
-
+egrep $FLAGS "$1" /usr/include/dw/*[ch]
+egrep $FLAGS "$1" /usr/include/lout/*[ch]
