diff -urNp bless-0.5.1.orig/doc/user/Makefile.in bless-0.5.1/doc/user/Makefile.in
--- bless-0.5.1.orig/doc/user/Makefile.in	2007-06-05 20:28:00.000000000 +0200
+++ bless-0.5.1/doc/user/Makefile.in	2007-06-05 20:32:05.000000000 +0200
@@ -330,11 +330,7 @@ uninstall-am: uninstall-local
 
 
 $(OMF): $(OMF).in
-	if [ -x `which scrollkeeper-preinstall` ]; then \
-	    scrollkeeper-preinstall file:$(datadir)/doc/$(PACKAGE_NAME)-$(PACKAGE_VERSION)/user/bless-manual.xml $(OMF).in $(OMF); \
-	else \
 	    cp $(OMF).in $(OMF); \
-	fi
 
 doc:
 	xsltproc /usr/share/xml/docbook/stylesheet/nwalsh/html/chunk.xsl bless-manual.xml
