--- prelink/src/prelink.h.bak	2004-06-09 13:08:06.503039570 -0700
+++ prelink/src/prelink.h	2004-06-09 13:08:29.314428019 -0700
@@ -44,6 +44,23 @@
 #define SHT_GNU_LIBLIST		0x6ffffff7
 #endif
 
+#ifndef PT_GNU_STACK
+#define PT_GNU_STACK 0x6474e551 /* Indicates stack executability */
+#endif
+
+#define R_PPC_TLS		67
+#define R_PPC_DTPMOD32		68
+#define R_PPC_TPREL16		69
+#define R_PPC_TPREL16_LO	70
+#define R_PPC_TPREL16_HI	71
+#define R_PPC_TPREL16_HA	72
+#define R_PPC_TPREL32		73
+#define R_PPC_DTPREL16		74
+#define R_PPC_DTPREL16_LO	75
+#define R_PPC_DTPREL16_HI	76
+#define R_PPC_DTPREL16_HA	77
+#define R_PPC_DTPREL32		78
+
 struct prelink_entry;
 struct prelink_info;
 struct PLArch;
