diff --git a/configure.ac b/configure.ac
index 7dff9a01a8ef28c88bc76b81568262646ff80995..ef9f5ecdcf68fef4093978cf042ee3734df7d45e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -113,7 +113,6 @@ AC_CHECK_SIZEOF([void *])
 CXXFLAGS="$CXXFLAGS -DSIZEOF_VOID_P=$ac_cv_sizeof_void_p"
 
 BOOST_REQUIRE([1.36])
-BOOST_FILESYSTEM
 
 # =====
 # Debug
diff --git a/src/libixion/module.cpp b/src/libixion/module.cpp
index b6a4dfff3df7c24283c6b49e55fce7ac6dfb1c6b..7bdf32cbde2979328ec58b3eebf449996d3479d5 100644
--- a/src/libixion/module.cpp
+++ b/src/libixion/module.cpp
@@ -10,7 +10,6 @@
 #include "ixion/compute_engine.hpp"
 #include <sstream>
 #include <vector>
-#include <boost/filesystem.hpp>
 
 #ifdef _WIN32
 #include <windows.h>
@@ -18,8 +17,6 @@
 #include <dlfcn.h>
 #endif
 
-namespace fs = boost::filesystem;
-
 namespace ixion { namespace draft {
 
 namespace {
