--- Makefile.orig	2004-04-25 02:15:41.365193792 -0500
+++ Makefile	2004-04-25 02:15:49.838905592 -0500
@@ -2,7 +2,7 @@
 DISTNAME = replex-$(VERSION)
 TARNAME = $(DISTNAME).tar.gz	
 INCS   = -I..
-CFLAGS =  -g -Wall -O2 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
+#CFLAGS =  -g -Wall -O2 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
 LIBS   = -L. 
 MFLAG  = -M
 OBJS = element.o pes.o mpg_common.o ts.o ringbuffer.o avi.o multiplex.o
