--- element.c.orig	2004-08-31 19:20:24.021680147 +0200
+++ element.c	2004-08-31 19:18:46.625093873 +0200
@@ -625,6 +625,7 @@
 				break;
 			default:
 				//fprintf(stderr," no pulldown detected \n");
+        break;
 			}
 		}
 		break;
--- replex.c.orig	2004-08-31 19:20:14.684870174 +0200
+++ replex.c	2004-08-31 19:21:22.569217944 +0200
@@ -1845,7 +1845,7 @@
         }
 
         if (optind == argc-1) {
-                if ((rx.fd_in = open(argv[optind] ,O_RDONLY| O_LARGEFILE)) < 0) {
+                if ((rx.fd_in = open(argv[optind] ,O_RDONLY)) < 0) {
                         perror("Error opening input file ");
                         exit(1);
                 }
@@ -1864,7 +1864,7 @@
 	if (!rx.demux){
 		if (filename){
 			if ((rx.fd_out = open(filename,O_WRONLY|O_CREAT
-					      |O_TRUNC|O_LARGEFILE,
+					      |O_TRUNC,
 					      S_IRUSR|S_IWUSR|S_IRGRP|
 					      S_IWGRP|
 					      S_IROTH|S_IWOTH)) < 0){
@@ -1921,8 +1921,7 @@
 			
 		snprintf(fname,256,"%s.mv2",filename);
 		if ((rx.dmx_out[0] = open(fname,O_WRONLY|
-					  O_CREAT|O_TRUNC|
-					  O_LARGEFILE,
+					  O_CREAT|O_TRUNC,
 					  S_IRUSR|S_IWUSR|
 					  S_IRGRP|S_IWGRP|
 					  S_IROTH|S_IWOTH)) 
@@ -1938,8 +1937,7 @@
 				 ,i);
 			if ((rx.dmx_out[i+1] = 
 			     open(fname,O_WRONLY|
-				  O_CREAT|O_TRUNC|
-				  O_LARGEFILE,
+				  O_CREAT|O_TRUNC,
 				  S_IRUSR|S_IWUSR|
 				  S_IRGRP|S_IWGRP|
 				  S_IROTH|S_IWOTH)) 
@@ -1956,8 +1954,7 @@
 				 ,i);
 			if ((rx.dmx_out[i+1+rx.apidn] = 
 			     open(fname,O_WRONLY|
-				  O_CREAT|O_TRUNC|
-				  O_LARGEFILE,
+				  O_CREAT|O_TRUNC,
 				  S_IRUSR|S_IWUSR|
 				  S_IRGRP|S_IWGRP|
 				  S_IROTH|S_IWOTH)) 
