--- sonata/main.py.orig	2009-09-22 01:02:16.000000000 +0400
+++ sonata/main.py	2010-10-31 16:40:39.759577403 +0300
@@ -1993,9 +1993,9 @@
 		# When withdrawing an app, extra configure events (with wrong coords)
 		# are fired (at least on Openbox). This prevents a user from moving
 		# the window, withdrawing it, then unwithdrawing it and finding it in
-		# an older position
-		if not window.props.visible:
-			return
+		# an older position. Broken with recent PyGTK (2.22).
+		# if not window.props.visible:
+		# 	return
 		
 		width, height = window.get_size()
 		if self.config.expanded: self.config.w, self.config.h = width, height
