--- sonata/main.py.orig	2009-09-22 01:02:16.000000000 +0400
+++ sonata/main.py	2010-05-15 20:58:27.362298551 +0400
@@ -961,7 +961,7 @@
 		self.actionGroupProfiles.get_action(index).activate()
 		self.skip_on_profiles_click = False
 		# Connect:
-		self.mpd_connect(force=True)
+		self.mpd_connect(blocking=True, force=True)
 		self.iterate_now()
 
 	def on_disconnectkey_pressed(self, _event):
@@ -976,7 +976,7 @@
 	def update_status(self):
 		try:
 			if not self.conn:
-				self.mpd_connect()
+				self.mpd_connect(blocking=True)
 			if self.conn:
 				self.iterate_time = self.iterate_time_when_connected
 				self.status = mpdh.status(self.client)
