--- mime/Makefile.in.orig	2014-03-01 15:31:41.000000000 +0400
+++ mime/Makefile.in	2014-05-12 14:00:03.431211914 +0400
@@ -531,11 +531,6 @@
 #	$(mimeicon_DATA)
 
 install-data-local: install-mimexmlDATA
-	if [ -f $(DESTDIR)$(datadir)/mime/packages/freedesktop.org.xml ] ; then \
-	  if which update-mime-database>/dev/null 2>&1; then \
-	    update-mime-database $(DESTDIR)$(datadir)/mime; \
-	  fi \
-	fi
 
 uninstall-local:
 	if [ -f $(DESTDIR)$(datadir)/mime/packages/freedesktop.org.xml ] ; then \
