diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -25099,21 +25099,21 @@ printf "%s\n" "found" >&6; }
       PKG_CONFIG_LIBDIR="$OPENSSL_PCDIR"
       export PKG_CONFIG_LIBDIR
     fi
-         $PKGCONFIG --libs-only-l --libs-only-other openssl 2>/dev/null`
+         $PKGCONFIG --libs-only-l --libs-only-other libngtcp2_crypto_openssl 2>/dev/null`
 
       SSL_LDFLAGS=`
     if test -n "$OPENSSL_PCDIR"; then
       PKG_CONFIG_LIBDIR="$OPENSSL_PCDIR"
       export PKG_CONFIG_LIBDIR
     fi
-         $PKGCONFIG --libs-only-L openssl 2>/dev/null`
+         $PKGCONFIG --libs-only-L libngtcp2_crypto_openssl 2>/dev/null`
 
       SSL_CPPFLAGS=`
     if test -n "$OPENSSL_PCDIR"; then
       PKG_CONFIG_LIBDIR="$OPENSSL_PCDIR"
       export PKG_CONFIG_LIBDIR
     fi
-         $PKGCONFIG --cflags-only-I openssl 2>/dev/null`
+         $PKGCONFIG --cflags-only-I libngtcp2_crypto_openssl 2>/dev/null`
 
 
       { printf "%s\n" "$as_me:${as_lineno-$LINENO}: pkg-config: SSL_LIBS: \"$SSL_LIBS\"" >&5
@@ -25132,14 +25132,13 @@ printf "%s\n" "$as_me: pkg-config: SSL_CPPFLAGS: \"$SSL_CPPFLAGS\"" >&6;}
     CPPFLAGS="$CPPFLAGS $SSL_CPPFLAGS"
   LDFLAGS="$LDFLAGS $SSL_LDFLAGS"
 
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for HMAC_Update in -lcrypto" >&5
-printf %s "checking for HMAC_Update in -lcrypto... " >&6; }
+  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for HMAC_Update in -lngtcp2_crypto_openssl" >&5
+printf %s "checking for HMAC_Update in -lngtcp2_crypto_openssl... " >&6; }
 if test ${ac_cv_lib_crypto_HMAC_Update+y}
 then :
   printf %s "(cached) " >&6
 else $as_nop
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lcrypto  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -25171,7 +25170,6 @@ if test "x$ac_cv_lib_crypto_HMAC_Update" = xyes
 then :
 
      HAVECRYPTO="yes"
-     LIBS="-lcrypto $LIBS"
 
 else $as_nop
 
