diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -25099,21 +25099,21 @@ printf "%s\n" "found" >&6; }
       PKG_CONFIG_LIBDIR="$OPENSSL_PCDIR"
       export PKG_CONFIG_LIBDIR
     fi
-         $PKGCONFIG --libs-only-l --libs-only-other openssl 2>/dev/null`
+         $PKGCONFIG --libs-only-l --libs-only-other libngtcp2_crypto_openssl 2>/dev/null`
 
       SSL_LDFLAGS=`
     if test -n "$OPENSSL_PCDIR"; then
       PKG_CONFIG_LIBDIR="$OPENSSL_PCDIR"
       export PKG_CONFIG_LIBDIR
     fi
-         $PKGCONFIG --libs-only-L openssl 2>/dev/null`
+         $PKGCONFIG --libs-only-L libngtcp2_crypto_openssl 2>/dev/null`
 
       SSL_CPPFLAGS=`
     if test -n "$OPENSSL_PCDIR"; then
       PKG_CONFIG_LIBDIR="$OPENSSL_PCDIR"
       export PKG_CONFIG_LIBDIR
     fi
-         $PKGCONFIG --cflags-only-I openssl 2>/dev/null`
+         $PKGCONFIG --cflags-only-I libngtcp2_crypto_openssl 2>/dev/null`
 
 
       { printf "%s\n" "$as_me:${as_lineno-$LINENO}: pkg-config: SSL_LIBS: \"$SSL_LIBS\"" >&5
@@ -25132,14 +25132,13 @@ printf "%s\n" "$as_me: pkg-config: SSL_CPPFLAGS: \"$SSL_CPPFLAGS\"" >&6;}
     CPPFLAGS="$CPPFLAGS $SSL_CPPFLAGS"
   LDFLAGS="$LDFLAGS $SSL_LDFLAGS"
 
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for HMAC_Update in -lcrypto" >&5
-printf %s "checking for HMAC_Update in -lcrypto... " >&6; }
+  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for HMAC_Update in -lngtcp2_crypto_openssl" >&5
+printf %s "checking for HMAC_Update in -lngtcp2_crypto_openssl... " >&6; }
 if test ${ac_cv_lib_crypto_HMAC_Update+y}
 then :
   printf %s "(cached) " >&6
 else $as_nop
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lcrypto  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -25171,7 +25170,6 @@ if test "x$ac_cv_lib_crypto_HMAC_Update" = xyes
 then :
 
      HAVECRYPTO="yes"
-     LIBS="-lcrypto $LIBS"
 
 else $as_nop
 
@@ -25309,14 +25309,13 @@ fi
 
   if test X"$HAVECRYPTO" = X"yes"; then
 
-    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for SSL_connect in -lssl" >&5
-printf %s "checking for SSL_connect in -lssl... " >&6; }
+    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for SSL_connect in -lngtcp2_crypto_openssl" >&5
+printf %s "checking for SSL_connect in -lngtcp2_crypto_openssl... " >&6; }
 if test ${ac_cv_lib_ssl_SSL_connect+y}
 then :
   printf %s "(cached) " >&6
 else $as_nop
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lssl  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -25348,7 +25347,6 @@ if test "x$ac_cv_lib_ssl_SSL_connect" = xyes
 then :
   printf "%s\n" "#define HAVE_LIBSSL 1" >>confdefs.h
 
-  LIBS="-lssl $LIBS"
 
 fi
 
@@ -25358,14 +25356,13 @@ fi
 printf %s "checking for ssl with RSAglue/rsaref libs in use... " >&6; };
         OLIBS=$LIBS
         LIBS="-lRSAglue -lrsaref $LIBS"
-        { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for SSL_connect in -lssl" >&5
-printf %s "checking for SSL_connect in -lssl... " >&6; }
+        { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for SSL_connect in -lngtcp2_crypto_openssl" >&5
+printf %s "checking for SSL_connect in -lngtcp2_crypto_openssl... " >&6; }
 if test ${ac_cv_lib_ssl_SSL_connect+y}
 then :
   printf %s "(cached) " >&6
 else $as_nop
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lssl  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -25397,7 +25394,6 @@ if test "x$ac_cv_lib_ssl_SSL_connect" = xyes
 then :
   printf "%s\n" "#define HAVE_LIBSSL 1" >>confdefs.h
 
-  LIBS="-lssl $LIBS"
 
 fi
 
@@ -25435,7 +25431,7 @@ done
                                 ac_fn_c_check_header_compile "$LINENO" "x509.h" "ac_cv_header_x509_h" "$ac_includes_default"
 if test "x$ac_cv_header_x509_h" = xyes
 then :
-  printf "%s\n" "#define HAVE_X509_H 1" >>confdefs.h
+  printe "%s\n" "#define HAVE_X509_H 0" >>confdefs.h
 
 fi
 ac_fn_c_check_header_compile "$LINENO" "rsa.h" "ac_cv_header_rsa_h" "$ac_includes_default"
