--- src/Makefile.orig	2013-06-18 08:49:36.000000000 -0400
+++ src/Makefile	2016-06-23 17:21:05.637925725 -0400
@@ -1,4 +1,4 @@
-LIBRARIES = libb64.a
+LIBRARIES = libb64.a libb64.so
 
 # Build flags (uncomment one)
 #############################
@@ -27,6 +27,9 @@
 libb64.a: cencode.o cdecode.o
 	$(AR) $(ARFLAGS) $@ $^
 
+libb64.so: libb64.a
+	ld $(LDFLAGS) -shared --whole-archive -o $@ $^
+
 strip:
 	strip $(BINARIES) *.exe
 
