--- src/Makefile.in.orig
+++ src/Makefile.in
@@ -35,7 +35,7 @@
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-sbin_PROGRAMS = fping$(EXEEXT)
+sbin_PROGRAMS = fping$(EXEEXT) fping6$(EXEEXT)
 subdir = src
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -49,7 +49,9 @@
 am__installdirs = "$(DESTDIR)$(sbindir)"
 PROGRAMS = $(sbin_PROGRAMS)
 am_fping_OBJECTS = fping.$(OBJEXT)
+am_fping6_OBJECTS = fping6.$(OBJEXT)
 fping_OBJECTS = $(am_fping_OBJECTS)
+fping6_OBJECTS = $(am_fping6_OBJECTS)
 fping_LDADD = $(LDADD)
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -240,6 +242,10 @@
 	@rm -f fping$(EXEEXT)
 	$(LINK) $(fping_OBJECTS) $(fping_LDADD) $(LIBS)
 
+fping6$(EXEEXT): $(fping6_OBJECTS) $(fping_DEPENDENCIES) 
+	@rm -f fping6$(EXEEXT)
+	$(LINK) $(fping6_OBJECTS) $(fping_LDADD) $(LIBS)
+
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
 
@@ -255,6 +261,9 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(COMPILE) -c $<
 
+fping6.o:
+	$(COMPILE) -DIPV6 -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ fping.c
+
 .c.obj:
 @am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ `$(CYGPATH_W) '$<'`
 @am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
