--- Makefile.in.orig	2011-12-28 17:52:03.000000000 +0400
+++ Makefile.in	2012-01-08 05:07:08.116384872 +0400
@@ -35,7 +35,7 @@
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-sbin_PROGRAMS = fping$(EXEEXT)
+sbin_PROGRAMS = fping$(EXEEXT) fping6$(EXEEXT)
 subdir = .
 DIST_COMMON = README $(am__configure_deps) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(srcdir)/config.h.in \
@@ -55,7 +55,9 @@
 am__installdirs = "$(DESTDIR)$(sbindir)" "$(DESTDIR)$(man8dir)"
 PROGRAMS = $(sbin_PROGRAMS)
 am_fping_OBJECTS = fping.$(OBJEXT)
+am_fping6_OBJECTS = fping6.$(OBJEXT)
 fping_OBJECTS = $(am_fping_OBJECTS)
+fping6_OBJECTS = $(am_fping6_OBJECTS)
 fping_LDADD = $(LDADD)
 DEFAULT_INCLUDES = -I.@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -304,6 +306,10 @@
 	@rm -f fping$(EXEEXT)
 	$(LINK) $(fping_OBJECTS) $(fping_LDADD) $(LIBS)
 
+fping6$(EXEEXT): $(fping6_OBJECTS) $(fping_DEPENDENCIES) 
+	@rm -f fping6$(EXEEXT)
+	$(LINK) $(fping6_OBJECTS) $(fping_LDADD) $(LIBS)
+
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
 
@@ -319,6 +325,9 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(COMPILE) -c $<
 
+fping6.o:
+	$(COMPILE) -DIPV6 -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ fping.c
+
 .c.obj:
 @am__fastdepCC_TRUE@	$(COMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ `$(CYGPATH_W) '$<'`
 @am__fastdepCC_TRUE@	$(am__mv) $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
