From 2665d9d08968d8e23fc9c8dc3f1ca7f354fcc0c6 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sat, 25 Jan 2025 18:03:19 +0100
Subject: [PATCH] Fix version

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 librecad/src/src.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/librecad/src/src.pro b/librecad/src/src.pro
index 4d3fd44088e7..f67f105d77b1 100644
--- a/librecad/src/src.pro
+++ b/librecad/src/src.pro
@@ -10,7 +10,7 @@ DISABLE_POSTSCRIPT = false
 DEFINES += DWGSUPPORT
 DEFINES -= JWW_WRITE_SUPPORT
 
-LC_VERSION="2.2.1_rc4"
+LC_VERSION="2.2.1"
 VERSION=$${LC_VERSION}
 
 # Store intermedia stuff somewhere else
-- 
2.48.1

