diff -Naur mcdp-0.4a/Makefile mcdp-0.4a-fixed/Makefile
--- mcdp-0.4a/Makefile	2007-04-26 22:38:40.000000000 -0600
+++ mcdp-0.4a-fixed/Makefile	2007-04-26 22:39:46.000000000 -0600
@@ -6,7 +6,8 @@
 MANDIR	= /usr/share/man
 BINDIR	= /usr/bin
 
-ARCH	= $(shell uname -m | sed -e 's/i[4-9]86/i386/' -e 's/armv[3-6][lb]/arm/')
+#ARCH	= $(shell uname -m | sed -e 's/i[4-9]86/i386/' -e 's/armv[3-6][lb]/arm/')
+ARCH	= $(shell uname -m | sed -e 's/armv[3-6][lb]/arm/')
 DIET	= $(shell basename `which diet 2>/dev/null` 2>/dev/null)
 
 # compiler
@@ -16,6 +17,7 @@
 
 # extraflags
 #CFLAGS += -mpreferred-stack-boundary=2 -march=$(ARCH) -mcpu=$(ARCH)
+CFLAGS += -mpreferred-stack-boundary=2 -march=$(ARCH)
 #CFLAGS += -W -Wchar-subscripts -Wmissing-prototypes -Wmissing-declarations -Wswitch -Wredundant-decls -Wunused
 #CFLAGS += -Werror
 #CFLAGS += -funroll-loops
@@ -45,7 +47,7 @@
 	@echo "You may take a look at \"doc/profile.sh\" ... "
 
 uninstall:
-	rm -f mcdp $(BINDIR)
+	rm -f $(BINDIR)/mcdp
 	rm -f $(MANDIR)/man1/mcdp.1.gz
 
 .c.o:
