--- configure.in.origin	2007-08-15 18:33:33.000000000 +0500
+++ configure.in	2007-08-15 18:47:53.000000000 +0500
@@ -11,6 +11,7 @@
 AM_DISABLE_STATIC
 
 AC_PATH_PROG(GCONFTOOL, gconftool-2)
+AC_PATH_PROG(PKG_CONFIG, pkg-config)
 AC_PROG_INTLTOOL([0.22])
 
 AC_CONFIG_MACRO_DIR(m4)
@@ -188,11 +189,11 @@
 AM_CONDITIONAL(MAEMO_SUPPORT, test "x${enable_maemo_support}" = "xyes")
 
 if test "x${enable_gpe_support}" = "xyes" ; then
-  DEP_MODULES="gtk+-2.0 >= 2.4 glib-2.0 >= 2.2 libgpewidget >= 0.109"
+  DEP_MODULES="gtk+-2.0 >= 2.4 glib-2.0 >= 2.2 gthread-2.0 >= 2.1 libgpewidget >= 0.109"
 elif test "x${enable_maemo_support}" = "xyes" ; then
   DEP_MODULES="gtk+-2.0 >= 2.6 gconf-2.0 >= 2.6 hildon-libs >= 0.12"
 elif test "x${enable_gnome_support}" = "xno" ; then
-  DEP_MODULES="gtk+-2.0 >= 2.6"
+  DEP_MODULES="gtk+-2.0 >= 2.6 gthread-2.0 >= 2.1"
 else
   DEP_MODULES="libgnomeui-2.0 >= 2.2.0"
 fi
