diff -Nuar stardict-3.0.1.orig/src/gconf_file.cpp stardict-3.0.1/src/gconf_file.cpp
--- stardict-3.0.1.orig/src/gconf_file.cpp	2009-12-07 19:21:18.252449785 -0500
+++ stardict-3.0.1/src/gconf_file.cpp	2009-12-07 19:22:35.372274388 -0500
@@ -22,6 +22,8 @@
 #  include "config.h"
 #endif
 
+#include <memory>
+
 #include <glib/gi18n.h>
 
 #include "gconf_file.hpp"
diff -Nuar stardict-3.0.1.orig/src/inifile.cpp stardict-3.0.1/src/inifile.cpp
--- stardict-3.0.1.orig/src/inifile.cpp	2009-12-07 19:21:18.254447900 -0500
+++ stardict-3.0.1/src/inifile.cpp	2009-12-07 19:22:35.372274388 -0500
@@ -25,6 +25,7 @@
 #include <vector>
 #include <cerrno>
 #include <cstring>
+#include <cstdlib>
 #include <cstdio>
 #include <glib/gi18n.h>
 #include <glib/gstdio.h>
diff -Nuar stardict-3.0.1.orig/src/lib/common.cpp stardict-3.0.1/src/lib/common.cpp
--- stardict-3.0.1.orig/src/lib/common.cpp	2009-12-07 19:21:18.256449750 -0500
+++ stardict-3.0.1/src/lib/common.cpp	2009-12-07 19:22:35.371274971 -0500
@@ -23,6 +23,8 @@
 #ifdef HAVE_CONFIG_H
 #  include "config.h"
 #endif
+#include <cstdlib>
+#include <cstring>
 
 #include "common.hpp"
 
diff -Nuar stardict-3.0.1.orig/src/lib/data.cpp stardict-3.0.1/src/lib/data.cpp
--- stardict-3.0.1.orig/src/lib/data.cpp	2009-12-07 19:21:18.258453418 -0500
+++ stardict-3.0.1/src/lib/data.cpp	2009-12-07 19:22:35.370274420 -0500
@@ -26,6 +26,8 @@
 
 #include "data.hpp"
 #include "getuint32.h"
+#include <cstdlib>
+#include <cstring>
 
 
 DictBase::DictBase()
diff -Nuar stardict-3.0.1.orig/src/lib/http_client.cpp stardict-3.0.1/src/lib/http_client.cpp
--- stardict-3.0.1.orig/src/lib/http_client.cpp	2009-12-07 19:21:18.257443116 -0500
+++ stardict-3.0.1/src/lib/http_client.cpp	2009-12-07 19:22:35.371274971 -0500
@@ -1,6 +1,8 @@
 #ifdef HAVE_CONFIG_H
 #  include "config.h"
 #endif
+#include <cstdlib>
+#include <cstring>
 
 #include "http_client.h"
 #include "sockets.hpp"
diff -Nuar stardict-3.0.1.orig/src/lib/pluginmanager.cpp stardict-3.0.1/src/lib/pluginmanager.cpp
--- stardict-3.0.1.orig/src/lib/pluginmanager.cpp	2009-12-07 19:21:18.256449750 -0500
+++ stardict-3.0.1/src/lib/pluginmanager.cpp	2009-12-07 19:22:35.371274971 -0500
@@ -1,6 +1,7 @@
 #include "pluginmanager.h"
 #include "file.hpp"
-#include <string>
+#include <cstdlib>
+#include <cstring>
 
 StarDictPluginBaseObject::StarDictPluginBaseObject(const char *filename, GModule *module_, plugin_configure_func_t configure_func_):
 	plugin_filename(filename), module(module_), configure_func(configure_func_)
diff -Nuar stardict-3.0.1.orig/src/lib/stardict_client.cpp stardict-3.0.1/src/lib/stardict_client.cpp
--- stardict-3.0.1.orig/src/lib/stardict_client.cpp	2009-12-07 19:21:18.256449750 -0500
+++ stardict-3.0.1/src/lib/stardict_client.cpp	2009-12-07 19:22:35.370274420 -0500
@@ -28,6 +28,8 @@
 #include "sockets.hpp"
 #include "md5.h"
 #include "getuint32.h"
+#include <cstdlib>
+#include <cstring>
 
 #include "stardict_client.hpp"
 
diff -Nuar stardict-3.0.1.orig/src/lib/treedict.cpp stardict-3.0.1/src/lib/treedict.cpp
--- stardict-3.0.1.orig/src/lib/treedict.cpp	2009-12-07 19:21:18.258453418 -0500
+++ stardict-3.0.1/src/lib/treedict.cpp	2009-12-07 19:22:35.371274971 -0500
@@ -24,6 +24,8 @@
 #ifdef HAVE_CONFIG_H
 #  include "config.h"
 #endif
+#include <cstdlib>
+#include <cstring>
 
 #include "file.hpp"
 #include "getuint32.h"
diff -Nuar stardict-3.0.1.orig/src/pangoview.cpp stardict-3.0.1/src/pangoview.cpp
--- stardict-3.0.1.orig/src/pangoview.cpp	2009-12-07 19:21:18.262454280 -0500
+++ stardict-3.0.1/src/pangoview.cpp	2009-12-07 19:22:35.372274388 -0500
@@ -22,6 +22,8 @@
 #  include "config.h"
 #endif
 
+#include <cstring>
+
 #include "gtktextviewpango.h"
 #include "utils.h"
 #include "skin.h"//for SkinCursor definition
diff -Nuar stardict-3.0.1.orig/src/prefsdlg.cpp stardict-3.0.1/src/prefsdlg.cpp
--- stardict-3.0.1.orig/src/prefsdlg.cpp	2009-12-07 19:21:18.263449875 -0500
+++ stardict-3.0.1/src/prefsdlg.cpp	2009-12-07 19:22:35.373267936 -0500
@@ -21,6 +21,8 @@
 #  include "config.h"
 #endif
 
+#include <cstdlib>
+
 #include <glib/gi18n.h>
 #include <glib/gstdio.h>
 
diff -Nuar stardict-3.0.1.orig/src/readword.cpp stardict-3.0.1/src/readword.cpp
--- stardict-3.0.1.orig/src/readword.cpp	2009-12-07 19:21:18.251449003 -0500
+++ stardict-3.0.1/src/readword.cpp	2009-12-07 19:22:35.372274388 -0500
@@ -3,6 +3,7 @@
 #endif
 
 #include <cstring>
+#include <cstdlib>
 #include <string>
 
 #include <glib/gi18n.h>
diff -Nuar stardict-3.0.1.orig/src/sigc++/signal.h stardict-3.0.1/src/sigc++/signal.h
--- stardict-3.0.1.orig/src/sigc++/signal.h	2009-12-07 19:21:18.261443104 -0500
+++ stardict-3.0.1/src/sigc++/signal.h	2009-12-07 19:22:35.374274986 -0500
@@ -18,7 +18,7 @@
   //Compilers, such as older versions of SUN Forte C++, that do not allow this also often
   //do not allow a typedef to have the same name as a class in the typedef's definition.
   //For Sun Forte CC 5.7 (SUN Workshop 10), comment this out to fix the build.
-  #define SIGC_TYPEDEF_REDEFINE_ALLOWED 1
+//  #define SIGC_TYPEDEF_REDEFINE_ALLOWED 1
 #endif
 
 namespace sigc {
diff -Nuar stardict-3.0.1.orig/src/utils.cpp stardict-3.0.1/src/utils.cpp
--- stardict-3.0.1.orig/src/utils.cpp	2009-12-07 19:21:18.255450024 -0500
+++ stardict-3.0.1/src/utils.cpp	2009-12-07 19:22:35.372274388 -0500
@@ -22,6 +22,8 @@
 #  include "config.h"
 #endif
 
+#include <cstring>
+
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <cstdlib>
diff -Nuar stardict-3.0.1.orig/src/x11_iskeyspressed.hpp stardict-3.0.1/src/x11_iskeyspressed.hpp
--- stardict-3.0.1.orig/src/x11_iskeyspressed.hpp	2009-12-07 19:21:18.255450024 -0500
+++ stardict-3.0.1/src/x11_iskeyspressed.hpp	2009-12-07 19:22:35.372274388 -0500
@@ -1,6 +1,8 @@
 #ifndef _X11_ISKEYSPRESSED_HPP_
 #define _X11_ISKEYSPRESSED_HPP_
 
+#include <memory>
+
 #include <gdk/gdkx.h>
 #include <X11/keysym.h>
 #include <gtk/gtk.h>
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-dictdotcn-plugin/stardict_dictdotcn.cpp stardict-3.0.1/stardict-plugins/stardict-dictdotcn-plugin/stardict_dictdotcn.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-dictdotcn-plugin/stardict_dictdotcn.cpp	2009-12-07 19:21:18.268450532 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-dictdotcn-plugin/stardict_dictdotcn.cpp	2009-12-07 19:22:35.375267925 -0500
@@ -1,6 +1,6 @@
 #include "stardict_dictdotcn.h"
 #include <glib/gi18n.h>
-#include <string>
+#include <cstring>
 #include <list>
 
 #ifdef _WIN32
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-espeak-tts-plugin/stardict_espeak.cpp stardict-3.0.1/stardict-plugins/stardict-espeak-tts-plugin/stardict_espeak.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-espeak-tts-plugin/stardict_espeak.cpp	2009-12-07 19:21:18.270452511 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-espeak-tts-plugin/stardict_espeak.cpp	2009-12-07 19:22:35.375267925 -0500
@@ -1,4 +1,5 @@
 #include "stardict_espeak.h"
+#include <cstring>
 #include <espeak/speak_lib.h>
 #include <glib/gi18n.h>
 
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp stardict-3.0.1/stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp	2009-12-07 19:21:18.271450734 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp	2009-12-07 19:22:35.376274365 -0500
@@ -1,7 +1,8 @@
 #include "stardict_gucharmap.h"
 #include <glib/gi18n.h>
 #include <gucharmap/gucharmap.h>
-#include <string>
+#include <cstring>
+#include <cstdlib>
 
 static char *build_dictdata(char type, const char *definition)
 {
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-html-parsedata-plugin/stardict_html_parsedata.cpp stardict-3.0.1/stardict-plugins/stardict-html-parsedata-plugin/stardict_html_parsedata.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-html-parsedata-plugin/stardict_html_parsedata.cpp	2009-12-07 19:21:18.266453935 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-html-parsedata-plugin/stardict_html_parsedata.cpp	2009-12-07 19:22:35.374274986 -0500
@@ -1,4 +1,6 @@
 #include "stardict_html_parsedata.h"
+#include <cstring>
+#include <cstdlib>
 #include <glib/gi18n.h>
 
 #ifdef _WIN32
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-man-plugin/stardict_man.cpp stardict-3.0.1/stardict-plugins/stardict-man-plugin/stardict_man.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-man-plugin/stardict_man.cpp	2009-12-07 19:21:18.268450532 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-man-plugin/stardict_man.cpp	2009-12-07 19:22:35.375267925 -0500
@@ -1,6 +1,6 @@
 #include "stardict_man.h"
 #include <glib/gi18n.h>
-#include <string>
+#include <cstring>
 
 static const StarDictPluginSystemInfo *plugin_info = NULL;
 static bool need_prefix;
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-powerword-parsedata-plugin/stardict_powerword_parsedata.cpp stardict-3.0.1/stardict-plugins/stardict-powerword-parsedata-plugin/stardict_powerword_parsedata.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-powerword-parsedata-plugin/stardict_powerword_parsedata.cpp	2009-12-07 19:21:18.268450532 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-powerword-parsedata-plugin/stardict_powerword_parsedata.cpp	2009-12-07 19:22:35.376274365 -0500
@@ -1,4 +1,5 @@
 #include "stardict_powerword_parsedata.h"
+#include <cstring>
 #include <glib/gi18n.h>
 
 #ifdef _WIN32
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-qqwry-plugin/stardict_qqwry.cpp stardict-3.0.1/stardict-plugins/stardict-qqwry-plugin/stardict_qqwry.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-qqwry-plugin/stardict_qqwry.cpp	2009-12-07 19:21:18.268450532 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-qqwry-plugin/stardict_qqwry.cpp	2009-12-07 19:22:35.375267925 -0500
@@ -1,7 +1,8 @@
 #include "stardict_qqwry.h"
 #include <glib/gi18n.h>
 #include <glib/gstdio.h>
-#include <string>
+#include <cstring>
+#include <cstdlib>
 
 #ifdef _WIN32
 #include <windows.h>
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-spell-plugin/stardict_spell.cpp stardict-3.0.1/stardict-plugins/stardict-spell-plugin/stardict_spell.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-spell-plugin/stardict_spell.cpp	2009-12-07 19:21:18.271450734 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-spell-plugin/stardict_spell.cpp	2009-12-07 19:22:35.376274365 -0500
@@ -1,4 +1,5 @@
 #include "stardict_spell.h"
+#include <cstring>
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <enchant.h>
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-update-info-plugin/stardict_update_info.cpp stardict-3.0.1/stardict-plugins/stardict-update-info-plugin/stardict_update_info.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-update-info-plugin/stardict_update_info.cpp	2009-12-07 19:21:18.269443361 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-update-info-plugin/stardict_update_info.cpp	2009-12-07 19:22:35.376274365 -0500
@@ -1,5 +1,6 @@
 #include "stardict_update_info.h"
 #include <glib/gi18n.h>
+#include <cstdlib>
 
 #ifdef _WIN32
 #include <windows.h>
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-wiki-parsedata-plugin/stardict_wiki2xml.cpp stardict-3.0.1/stardict-plugins/stardict-wiki-parsedata-plugin/stardict_wiki2xml.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-wiki-parsedata-plugin/stardict_wiki2xml.cpp	2009-12-07 19:21:18.270452511 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-wiki-parsedata-plugin/stardict_wiki2xml.cpp	2009-12-07 19:22:35.374274986 -0500
@@ -1,5 +1,6 @@
 #include "stardict_wiki2xml.h"
 #include "WIKI2XML.h"
+#include <cstring>
 #include <glib.h>
 
 std::string wiki2xml(std::string &str)
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-wiki-parsedata-plugin/stardict_wiki_parsedata.cpp stardict-3.0.1/stardict-plugins/stardict-wiki-parsedata-plugin/stardict_wiki_parsedata.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-wiki-parsedata-plugin/stardict_wiki_parsedata.cpp	2009-12-07 19:21:18.270452511 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-wiki-parsedata-plugin/stardict_wiki_parsedata.cpp	2009-12-07 19:22:35.374274986 -0500
@@ -1,5 +1,6 @@
 #include "stardict_wiki_parsedata.h"
 #include "stardict_wiki2xml.h"
+#include <cstring>
 #include <glib/gi18n.h>
 
 #ifdef _WIN32
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-wordnet-plugin/court_widget.cpp stardict-3.0.1/stardict-plugins/stardict-wordnet-plugin/court_widget.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-wordnet-plugin/court_widget.cpp	2009-12-07 19:21:18.269443361 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-wordnet-plugin/court_widget.cpp	2009-12-07 19:22:35.375267925 -0500
@@ -1,4 +1,5 @@
 #include "court_widget.h"
+#include <cstring>
 #include <math.h>
 #include <list>
 
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet.cpp stardict-3.0.1/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet.cpp	2009-12-07 19:21:18.269443361 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet.cpp	2009-12-07 19:22:35.375267925 -0500
@@ -1,5 +1,6 @@
 #include "stardict_wordnet.h"
 #include "court_widget.h"
+#include <cstring>
 #include <glib/gi18n.h>
 
 #ifdef _WIN32
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet_parsedata.cpp stardict-3.0.1/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet_parsedata.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet_parsedata.cpp	2009-12-07 19:21:18.269443361 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-wordnet-plugin/stardict_wordnet_parsedata.cpp	2009-12-07 19:22:35.374274986 -0500
@@ -1,4 +1,5 @@
 #include "stardict_wordnet_parsedata.h"
+#include <cstring>
 #include <glib/gi18n.h>
 
 #ifdef _WIN32
diff -Nuar stardict-3.0.1.orig/stardict-plugins/stardict-xdxf-parsedata-plugin/stardict_xdxf_parsedata.cpp stardict-3.0.1/stardict-plugins/stardict-xdxf-parsedata-plugin/stardict_xdxf_parsedata.cpp
--- stardict-3.0.1.orig/stardict-plugins/stardict-xdxf-parsedata-plugin/stardict_xdxf_parsedata.cpp	2009-12-07 19:21:18.270452511 -0500
+++ stardict-3.0.1/stardict-plugins/stardict-xdxf-parsedata-plugin/stardict_xdxf_parsedata.cpp	2009-12-07 19:22:35.374274986 -0500
@@ -1,4 +1,5 @@
 #include "stardict_xdxf_parsedata.h"
+#include <cstring>
 #include <glib/gi18n.h>
 
 #ifdef _WIN32
diff -Nuar stardict-3.0.1.orig/tests/t_config_file.cpp stardict-3.0.1/tests/t_config_file.cpp
--- stardict-3.0.1.orig/tests/t_config_file.cpp	2009-12-07 19:21:18.271450734 -0500
+++ stardict-3.0.1/tests/t_config_file.cpp	2009-12-07 19:22:35.371274971 -0500
@@ -8,6 +8,7 @@
 #include <cstring>
 #include <iterator>
 #include <iostream>
+#include <memory>
 #include <gtk/gtk.h>
 
 #include "config_file.hpp"
diff -Nuar stardict-3.0.1.orig/tests/t_xml.cpp stardict-3.0.1/tests/t_xml.cpp
--- stardict-3.0.1.orig/tests/t_xml.cpp	2009-12-07 19:21:18.271450734 -0500
+++ stardict-3.0.1/tests/t_xml.cpp	2009-12-07 19:22:35.371274971 -0500
@@ -5,6 +5,7 @@
 #include <glib.h>
 #include <cstdlib>
 #include <string>
+#include <cstring>
 
 
 static void xml_decode(const char *str, std::string& decoded)
