--- Makefile.orig	2003-10-08 21:22:54.000000000 -0700
+++ Makefile	2003-10-09 05:31:19.000000000 -0700
@@ -127,28 +127,6 @@
 			echo "You may need to add $(sbindir) and $(bindir) to your PATH." ; \
 		fi ; \
 		echo ; \
-		for dir in /etc/init.d /etc/rc.d/init.d /sbin/init.d XXXX ; do \
-			if test $$dir = XXXX ; then \
-				if ! test -d /usr/local/etc/rc.d ; then \
-					echo "Warning: couldn't find SysV init script directory!" ; \
-					echo "Refer to setup-build-details.html for instructions on manually"; \
-					echo "setting ptal-init to start at bootup."; \
-				else \
-					echo "Setting /usr/local/etc/rc.d/ptal-init.sh symlink." ; \
-					$(LN_S) -f $(sbindir)/ptal-init /usr/local/etc/rc.d/ptal-init.sh ; \
-				fi ; \
-			elif test -d $$dir ; then \
-				echo "Setting $$dir/ptal-init symlink." ; \
-				$(LN_S) -f $(sbindir)/ptal-init $$dir/ptal-init ; \
-				if ! chkconfig --add ptal-init ; then \
-					echo "Warning: chkconfig not available to set up SysV init script symlinks!" ; \
-					echo "Refer to setup-build-details.html for instructions on manually"; \
-					echo "setting ptal-init to start at bootup."; \
-				fi ; \
-				break ; \
-			fi ; \
-		done ; \
-		echo ; \
 		if test -n "$(CUPS_BACKEND_DIR)" ; then \
 			echo "Setting $(CUPS_BACKEND_DIR)/ptal symlink." ; \
 			$(LN_S) -f $(sbindir)/ptal-cups $(CUPS_BACKEND_DIR)/ptal ; \
