diff -Naur xmms-vtx-0.8.1/configure.in xmms-vtx-0.8.1-fixed/configure.in
--- xmms-vtx-0.8.1/configure.in	2005-11-17 13:10:40.000000000 +0100
+++ xmms-vtx-0.8.1-fixed/configure.in	2008-02-07 21:06:09.000000000 +0100
@@ -18,7 +18,7 @@
 dnl check for libraries
 AC_CHECK_LIB(m,main)
 
-AC_CHECK_LIB(ayemu, ayemu_start,,
+AC_CHECK_LIB(ayemu, ayemu_init,,
 AC_MSG_ERROR(cannot find ayemu library. Check your ayemu library installation))
 AC_OUTPUT
 
