--- schroot-1.6.12.orig/man/CMakeLists.txt	2022-07-21 07:05:10.000000000 +0200
+++ schroot-1.6.12/man/CMakeLists.txt	2022-07-26 03:09:32.632463653 +0200
@@ -81,7 +81,7 @@
                              "${CMAKE_CURRENT_SOURCE_DIR}/po4a.cfg"
                      VERBATIM)
 else(PO4A_EXECUTABLE)
-  warn("po4a not available; not translating manual pages")
+  message(WARNING "po4a not available; not translating manual pages")
 endif(PO4A_EXECUTABLE)
 
 set(manpage_sources
