--- GNUmakefile.orig	2022-07-06 01:11:27.000000000 -0700
+++ GNUmakefile	2022-08-22 20:27:28.302874319 -0700
@@ -9,8 +9,8 @@
 GIT_TAG = $(shell git tag --points-at HEAD 2>/dev/null || cat VERSION)
 MAKEFLAGS += --no-builtin-rules
 
-WARNINGS = -Wall -Wextra -Wpedantic -Wformat=2 -Wstrict-aliasing=3 -Wstrict-overflow=5 -Wstack-usage=12500 \
-	-Wcast-align -Wpointer-arith -Wchar-subscripts -Warray-bounds=2 -Wno-unknown-warning-option
+WARNINGS = -Wall -Wextra -Wformat=2 -Wstrict-aliasing=3 -Wstrict-overflow=5 \
+	-Wcast-align -Wpointer-arith -Wchar-subscripts -Wno-unknown-warning-option
 VISIBILITY ?= -fvisibility=hidden
 
 override CFLAGS ?= -g -O2 $(WARNINGS) $(EXTRA_WARNINGS)
