--- src/mp3.c.orig	2008-11-02 21:54:31.000000000 +0100
+++ src/mp3.c	2009-02-25 11:48:53.000000000 +0100
@@ -504,7 +504,7 @@
 
 SOX_FORMAT_HANDLER(mp3)
 {
-  static char const * const names[] = {"mp3", "mp2", "audio/mpeg", NULL};
+  static char const * const names[] = {"mp3", "mp2", "audio/mpeg", "audio/mpeg; charset=binary", NULL};
   static unsigned const write_encodings[] = {
     SOX_ENCODING_GSM, 0, 0};
   static sox_format_handler_t const handler = {SOX_LIB_VERSION_CODE,
