From 9eb068527c1f7d43ff2a3e335509bc533d0c1d5d Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 4 Jun 2021 13:42:18 +0200
Subject: [PATCH] Fix linking against libbsd

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 5b41f7e..f99ba66 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 LAYOUT ?= linux
 SCDOC := scdoc
-LIBBSD_CFLAGS =
-LIBBSD_LIBS =
+LIBBSD_CFLAGS := $(shell pkg-config --cflags libbsd-overlay)
+LIBBSD_LIBS := $(shell pkg-config --libs libbsd-overlay)
 
 PACKAGE_NAME := ifupdown-ng
 PACKAGE_VERSION := 0.11.3
-- 
2.31.1

