From 76a56e3f8ee79c7a70acd96a3530d02ca9cf1f7a Mon Sep 17 00:00:00 2001
From: Leonardo Taccari <iamleot@gmail.com>
Date: Mon, 15 Jun 2020 19:06:39 +0200
Subject: [PATCH] Always wait childs to avoid zombies

---
 entr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/entr.c b/entr.c
index 1532a9c..db414ab 100644
--- a/entr.c
+++ b/entr.c
@@ -254,8 +254,8 @@ void
 proc_exit(int sig) {
 	int status;
 
+	xwaitpid(child_pid, &status, 0);
 	if ((oneshot_opt == 1) && (terminating == 0)) {
-		xwaitpid(child_pid, &status, 0);
 		if ((shell_opt == 1) && (restart_opt == 0)) {
 			fprintf(stdout, "%s returned exit code %d\n",
 			    basename(getenv("SHELL")), WEXITSTATUS(status));
