diff -urN simpleinit-msb-1.2/sysvtools/Makefile simpleinit-msb-1.2-new/sysvtools/Makefile
--- simpleinit-msb-1.2/sysvtools/Makefile	2001-09-26 13:01:08.000000000 +0200
+++ simpleinit-msb-1.2-new/sysvtools/Makefile	2005-04-15 00:43:31.281944078 +0200
@@ -8,6 +8,8 @@
 # Version:	@(#)Makefile  2.78  13-Nov-1999  miquels@cistron.nl
 #
 
+include ../make_include
+
 CC	= cc
 CFLAGS	= -Wall -O2 -D_GNU_SOURCE
 LDFLAGS	= -s
@@ -22,7 +24,7 @@
 MANDIR		= /usr/share/man
 
 # Additional libs for Gnu Libc
-ifneq ($(wildcard /usr/lib/libcrypt.a),)
+ifeq "$(NEED_LIBCRYPT)" "yes"
 LCRYPT		= -lcrypt
 endif
 
