diff -Naur simpleinit-msb-1.3.orig/sysvtools/bootlogd.c simpleinit-msb-1.3/sysvtools/bootlogd.c
--- simpleinit-msb-1.3.orig/sysvtools/bootlogd.c	2011-05-12 15:20:38.263708528 -0700
+++ simpleinit-msb-1.3/sysvtools/bootlogd.c	2011-05-12 15:22:44.089708477 -0700
@@ -228,6 +228,9 @@
 	}
 
 #ifdef TIOCGDEV
+	#ifndef  ENOIOCTLCMD
+		#define ENOIOCTLCMD	515
+	#endif
 	if (ioctl(0, TIOCGDEV, &kdev) == 0)
 		return findtty(res, rlen, (dev_t)kdev);
 	if (errno != ENOIOCTLCMD) return -1;
