--- login-utils/shutdown.c.orig	2014-06-25 15:09:23.000000000 +0400
+++ login-utils/shutdown.c	2014-06-25 15:11:12.534462171 +0400
@@ -210,10 +210,10 @@
 	prog = argv[0];
 	if((ptr = strrchr(argv[0], '/'))) prog = ++ptr;
 
-	/* All names (halt, reboot, fasthalt, fastboot, shutdown)
+	/* All names (halt, poweroff, reboot, fasthalt, fastboot, shutdown)
 	   refer to the same program with the same options,
 	   only the defaults differ. */
-	if(!strcmp("halt", prog)) {
+	if(!strcmp("halt", prog) || !strcmp("poweroff", prog)) {
 		opt_reboot = 0;
 		opt_quiet = 1;
 		opt_fast = 0;
