--- login-utils/simpleinit.c.orig	2006-11-19 15:42:20.000000000 +0100
+++ login-utils/simpleinit.c	2006-11-19 15:43:16.000000000 +0100
@@ -408,10 +407,6 @@
 	  my_reboot (LINUX_REBOOT_CMD_CAD_OFF);
 	} 
 
-	setlocale(LC_ALL, "");
-	bindtextdomain(PACKAGE, LOCALEDIR);
-	textdomain(PACKAGE);
-	
 	/*  Find script to run. Command-line overrides config file overrides
 	    built-in default  */
 	for (i = 0; i < NUMCMD; i++) inittab[i].pid = -1;
--- login-utils/shutdown.c.orig	2006-11-19 15:43:36.000000000 +0100
+++ login-utils/shutdown.c	2006-11-19 15:44:43.000000000 +0100
@@ -197,9 +197,6 @@
 	}
 	sigsetmask (0); /*  simpleinit(8) blocks all signals: undo for ALRM  */
 	for (i = 1; i < NSIG; i++) signal (i, SIG_DFL); /* unnecessary */
-        setlocale(LC_ALL, "");
-        bindtextdomain(PACKAGE, LOCALEDIR);
-        textdomain(PACKAGE);
 
 #ifndef DEBUGGING
 	if(setreuid (0, 0)) {
