--- sysvtools/Makefile.orig	2007-11-09 00:16:54.000000000 +0300
+++ sysvtools/Makefile	2008-07-30 23:00:07.328860627 +0400
@@ -8,6 +8,8 @@
 # Version:	@(#)Makefile  2.85-13  23-Mar-2004  miquels@cistron.nl
 #
 
+include ../make_include
+
 CC	= gcc
 CFLAGS	= -Wall -O2 -fomit-frame-pointer -D_GNU_SOURCE
 LDFLAGS	= -s
@@ -36,7 +38,7 @@
 MANDIR		= /usr/share/man
 
 # Additional libs for GNU libc.
-ifneq ($(wildcard /usr/lib/libcrypt.a),)
+ifeq "$(NEED_LIBCRYPT)" "yes"
 LCRYPT		= -lcrypt
 endif
 
