diff -ur xbase64-3.1.2/xbase64/xbase64.cpp /usr/src/xbase64-3.1.2/xbase64/xbase64.cpp
--- xbase64-3.1.2/xbase64/xbase64.cpp	2004-03-16 13:55:52.000000000 +0100
+++ /usr/src/xbase64-3.1.2/xbase64/xbase64.cpp	2004-06-08 10:37:45.836404280 +0200
@@ -51,6 +51,7 @@
 #include <xbase64/xbase64.h>
 #include <ctype.h>
 #include <string.h>
+#include <cerrno>
 
 //#include <xbase64/xbexcept.h>
 
diff -ur xbase64-3.1.2/xbase64/xblock.cpp /usr/src/xbase64-3.1.2/xbase64/xblock.cpp
--- xbase64-3.1.2/xbase64/xblock.cpp	2004-03-16 13:55:56.000000000 +0100
+++ /usr/src/xbase64-3.1.2/xbase64/xblock.cpp	2004-06-08 10:39:41.845768176 +0200
@@ -61,6 +61,7 @@
 #ifdef HAVE_DOS_H    // _sleep
 #include <dos.h>
 #endif
+#include <cerrno>
 
 //#include <stdio.h>
 //#include <stdlib.h>
diff -ur xbase64-3.1.2/xbase64/xbnode.cpp /usr/src/xbase64-3.1.2/xbase64/xbnode.cpp
--- xbase64-3.1.2/xbase64/xbnode.cpp	2004-06-01 21:21:48.000000000 +0200
+++ /usr/src/xbase64-3.1.2/xbase64/xbnode.cpp	2004-06-08 10:43:09.681172392 +0200
@@ -1,6 +1,6 @@
-#include "xbNode.h"
-
-void xbNodeLink::AddNode(xbNodeLink* node)
-{
-  nextNode_=node;
-}
+#include "xbnode.h"
+
+void xbNodeLink::AddNode(xbNodeLink* node)
+{
+  nextNode_=node;
+}
