From a39f0b296fc0ddcf31a21625ba6397040ea08521 Mon Sep 17 00:00:00 2001
From: Vlad Glagolev <vaygr@users.noreply.github.com>
Date: Thu, 27 Oct 2016 11:45:30 -0400
Subject: [PATCH] respect theme foreground color

---
 gtk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk.c b/gtk.c
index 41cf596..6a37a9a 100644
--- a/gtk.c
+++ b/gtk.c
@@ -495,7 +495,7 @@ static void update_tree_row(struct mtr_ctl *ctl, int row, GtkTreeIter *iter)
     COL_WORST, net_worst(row)/1000,
     COL_STDEV, (float)(net_stdev(row)/1000.0),
     
-    COL_COLOR, net_up(row) ? "black" : "red",
+    COL_COLOR, net_up(row) ? NULL : "red",
 
     -1);
 #ifdef HAVE_IPINFO
-- 
2.7.4

