--- scripts/Makefile.in.orig	2010-02-26 15:39:16.000000000 +0300
+++ scripts/Makefile.in	2011-01-15 01:59:40.080000128 +0300
@@ -57,8 +57,6 @@
 
 install-utils:
 ifeq ($(WITH_UTILS),yes)
-	install -d $(DESTDIR)$(INITDIR)
-	install -m 755 drbd $(DESTDIR)$(INITDIR)/
 	@ if [ ! -e $(DESTDIR)$(sysconfdir)/drbd.conf ]; then \
 		install -d $(DESTDIR)$(sysconfdir)/; \
 		install -m 644 drbd.conf $(DESTDIR)$(sysconfdir)/; \
