--- Makefile.orig	2017-02-02 10:43:26.000000000 -0500
+++ Makefile	2017-02-02 13:24:50.665780890 -0500
@@ -8,10 +8,9 @@
 SHA := $(shell cat COMMIT 2>/dev/null || git rev-parse --short=8 HEAD)
 DATE := $(shell date -u '+%FT%T%z')
 
-LDFLAGS += -X "main.Version=$(VERSION)"
-LDFLAGS += -X "main.BuildTime=$(DATE)"
-LDFLAGS += -X "main.Commit=$(SHA)"
-LDFLAGS += -extldflags '-static'
+LDFLAGS += -X main.Version $(VERSION)
+LDFLAGS += -X main.BuildTime $(DATE)
+LDFLAGS += -X main.Commit $(SHA)
 
 PREFIX ?= /usr
 BINDIR ?= $(PREFIX)/bin
