--- qtada-gpl-3.1.0-20100503-3734/aclocal.m4.orig	2010-09-12 11:59:25.000000000 +0200
+++ qtada-gpl-3.1.0-20100503-3734/aclocal.m4	2010-09-12 11:59:02.000000000 +0200
@@ -332,6 +332,7 @@
 dnl Autoconf wants to disallow AM_ names.  We explicitly allow
 dnl the ones we care about.
 m4_pattern_allow([^AM_[A-Z]+FLAGS$])dnl
+m4_pattern_allow(AS_MESSAGE_LOG_FD)dnl
 AC_REQUIRE([AM_SET_CURRENT_AUTOMAKE_VERSION])dnl
 AC_REQUIRE([AC_PROG_INSTALL])dnl
 if test "`cd $srcdir && pwd`" != "`pwd`"; then
--- qtada-gpl-3.1.0-20100503-3734/configure.ac.orig	2010-09-12 12:11:12.000000000 +0200
+++ qtada-gpl-3.1.0-20100503-3734/configure.ac	2010-09-12 12:11:56.000000000 +0200
@@ -22,6 +22,8 @@
 
 AM_CONDITIONAL(GNU_LD, [test "$with_gnu_ld" = yes])
 
+m4_pattern_allow(AS_MESSAGE_LOG_FD)
+
 # Формирование ключей компилятора и редактора связей для сборки с Qt в
 # различных операционных системах.
 
