--- Makefile.am.orig	2010-08-29 21:31:07.000000000 +0200
+++ Makefile.am	2010-08-29 21:31:30.000000000 +0200
@@ -28,8 +28,8 @@
 # We build source code distribution on the Linux, so we enable GtkAda
 # Integration Kit to allow AutoTools to do additional checks.
 
-SUBDIRS = . tools source examples documentation $(subdirs)
-DIST_SUBDIRS = . tools source examples documentation
+SUBDIRS = . tools source documentation $(subdirs)
+DIST_SUBDIRS = . tools source documentation
 # $(subdirs) may include optional directory of the in-place testsuite which is
 # never included in the source code distribution.
 
--- Makefile.am.orig	2010-08-29 22:37:46.000000000 +0200
+++ Makefile.am	2010-08-29 22:37:54.000000000 +0200
@@ -102,6 +102,5 @@
 dist-hook:
 	rm -f gnat/config.cgpr
 if MAINTAINER_MODE
-	rm -rf $(top_distdir)/tools/generator
 	rm -rf $(top_distdir)/testsuite
 endif
--- configure.ac.orig	2010-08-29 22:39:44.000000000 +0200
+++ configure.ac	2010-08-29 22:39:52.000000000 +0200
@@ -467,13 +467,6 @@
         AC_CONFIG_FILES([
 		documentation/qtada.qhcp
 		documentation/qtada_ugr/qtada_ugr.qhp
-		tools/generator/Makefile
-		tools/generator/testsuite/Makefile
-		tools/generator/testsuite/00-000-0000/Makefile
-		tools/generator/testsuite/00-000-0001/Makefile
-		tools/generator/testsuite/00-000-0002/Makefile
-		tools/generator/testsuite/00-000-0003/Makefile
-		tools/generator/testsuite/00-000-0004/Makefile
         ])
 fi
 
--- tools/Makefile.am.orig	2010-08-29 22:43:22.000000000 +0200
+++ tools/Makefile.am	2010-08-29 22:43:27.000000000 +0200
@@ -22,9 +22,5 @@
 ##----------------------------------------------------------------------------
 include $(top_srcdir)/Makefile.common
 
-if MAINTAINER_MODE
-    MAINTAINER_TOOLS_DIRS = generator
-endif
-
 SUBDIRS = . gen_sizes gen_pvd $(MAINTAINER_TOOLS_DIRS)
 DIST_SUBDIRS = . gen_sizes gen_pvd $(MAINTAINER_TOOLS_DIRS)
