--- source/sql/qt4-sql_records-sql_indices.ads.orig	2010-08-27 14:25:24.000000000 +0200
+++ source/sql/qt4-sql_records-sql_indices.ads	2010-08-27 14:25:41.000000000 +0200
@@ -87,13 +87,10 @@
 
    type Q_Sql_Index is new Qt4.Sql_Records.Q_Sql_Record with null record;
 
-   overriding
    procedure Adjust (Self : in out Q_Sql_Index);
 
-   overriding
    procedure Finalize (Self : in out Q_Sql_Index);
 
-   overriding
    procedure Initialize (Self : in out Q_Sql_Index);
 
 end Qt4.Sql_Records.Sql_Indices;
