--- src/filename_list.cxx	2007-06-01 08:13:08.000000000 -0500
+++ src/filename_list.cxx.gcc44	2009-06-01 15:38:51.425093964 -0500
@@ -63,7 +63,7 @@ int fltk::filename_list(const char *d, d
   // some Unix systems):
   int n = scandir(d, list, 0, sort);
 #elif defined(__linux) || defined (__FreeBSD__) || defined (__NetBSD__)
-  int n = scandir(d, list, 0, (int(*)(const void*,const void*))sort);
+  int n = scandir(d, list, 0, (int(*)(const dirent **,const dirent **))sort);
 #elif defined(__hpux) || defined(__CYGWIN__)
   // HP-UX, Cygwin define the comparison function like this:
   int n = scandir(d, list, 0, (int(*)(const dirent **, const dirent **))sort);
