--- drivers/ipoe/ipoe.c.orig	2014-05-08 14:41:12.000000000 +0400
+++ drivers/ipoe/ipoe.c	2014-08-12 12:40:34.776058280 +0400
@@ -395,7 +395,9 @@
 	if (!ses->peer_addr)
 		goto drop;
 	
+#ifdef CONFIG_NET_CLS_ACT
 	skb->tc_verd = SET_TC_NCLS(0);
+#endif
 	
 	noff = skb_network_offset(skb);
 
--- drivers/ipoe/CMakeLists.txt.orig	2014-05-08 14:41:12.000000000 +0400
+++ drivers/ipoe/CMakeLists.txt	2014-08-12 13:36:56.475218335 +0400
@@ -14,6 +14,4 @@
 	DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/driver/ipoe.ko
 )
 
-
-INSTALL(CODE "EXECUTE_PROCESS(COMMAND make -C ${KDIR} M=${CMAKE_CURRENT_BINARY_DIR}/drivers/ipoe modules_install)")
-
+INSTALL(CODE "EXECUTE_PROCESS(COMMAND make -C ${KDIR} M=${CMAKE_CURRENT_BINARY_DIR}/driver modules_install)")
