diff -Nur gnome-print-0.37.orig/installer/gf-pfb.c gnome-print-0.37-ftfix/installer/gf-pfb.c
--- gnome-print-0.37.orig/installer/gf-pfb.c	2002-01-19 01:23:00.000000000 +0100
+++ gnome-print-0.37-ftfix/installer/gf-pfb.c	2004-05-01 13:12:34.000000000 +0200
@@ -12,7 +12,8 @@
 #include <fcntl.h>
 #include <string.h>
 #include <ctype.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include "gf-pfb.h"
 
 #define noVERBOSE
diff -Nur gnome-print-0.37.orig/installer/gf-ttf.c gnome-print-0.37-ftfix/installer/gf-ttf.c
--- gnome-print-0.37.orig/installer/gf-ttf.c	2001-11-05 15:07:28.000000000 +0100
+++ gnome-print-0.37-ftfix/installer/gf-ttf.c	2004-05-01 13:13:08.000000000 +0200
@@ -32,7 +32,8 @@
 #include <fcntl.h>
 #include <string.h>
 #include <ctype.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 
 #include "gf-ttf.h"
 
diff -Nur gnome-print-0.37.orig/installer/gnome-font-install.c gnome-print-0.37-ftfix/installer/gnome-font-install.c
--- gnome-print-0.37.orig/installer/gnome-font-install.c	2002-01-22 22:44:55.000000000 +0100
+++ gnome-print-0.37-ftfix/installer/gnome-font-install.c	2004-05-01 13:11:58.000000000 +0200
@@ -26,7 +26,8 @@
 #include <parser.h>
 #include <xmlmemory.h>
 /* End of ugly thing */
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <libgnome/gnome-defs.h>
 #include <libgnome/gnome-util.h>
 #include <libgnome/gnome-i18n.h>
diff -Nur gnome-print-0.37.orig/libgnomeprint/gnome-font-private.h gnome-print-0.37-ftfix/libgnomeprint/gnome-font-private.h
--- gnome-print-0.37.orig/libgnomeprint/gnome-font-private.h	2001-12-05 23:07:58.000000000 +0100
+++ gnome-print-0.37-ftfix/libgnomeprint/gnome-font-private.h	2004-05-01 13:07:56.000000000 +0200
@@ -22,7 +22,8 @@
 
 BEGIN_GNOME_DECLS
 
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <libgnomeprint/gp-character-block.h>
 #include <libgnomeprint/gp-fontmap.h>
 #include <libgnomeprint/gnome-font.h>
diff -Nur gnome-print-0.37.orig/libgnomeprint/parseTT.h gnome-print-0.37-ftfix/libgnomeprint/parseTT.h
--- gnome-print-0.37.orig/libgnomeprint/parseTT.h	2001-11-05 20:27:09.000000000 +0100
+++ gnome-print-0.37-ftfix/libgnomeprint/parseTT.h	2004-05-01 13:09:28.000000000 +0200
@@ -22,7 +22,8 @@
 BEGIN_GNOME_DECLS
 
 #include <glib.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 
 /*
  * Convert loaded Freetype TTF face to Type1
