diff -uNr libdbi-drivers-0.8.1/drivers/firebird/dbd_firebird.c libdbi-drivers-0.8.1.cp/drivers/firebird/dbd_firebird.c
--- libdbi-drivers-0.8.1/drivers/firebird/dbd_firebird.c	2005-09-21 20:17:52.000000000 +0200
+++ libdbi-drivers-0.8.1.cp/drivers/firebird/dbd_firebird.c	2006-05-28 09:22:24.000000000 +0200
@@ -45,7 +45,6 @@
 
 
 #include <ibase.h>
-#include <gds.h>
 
 
 #include "dbd_firebird.h"
@@ -462,7 +461,7 @@
 {
         ibase_conn_t *iconn = conn->connection;
         
-	TEXT errbuf[MAXLEN];
+	ISC_SCHAR errbuf[MAXLEN];
 	long sqlcode;
 
 	if ( conn->connection == NULL) {
diff -uNr libdbi-drivers-0.8.1/drivers/firebird/utility.c libdbi-drivers-0.8.1.cp/drivers/firebird/utility.c
--- libdbi-drivers-0.8.1/drivers/firebird/utility.c	2005-09-21 00:28:11.000000000 +0200
+++ libdbi-drivers-0.8.1.cp/drivers/firebird/utility.c	2006-05-28 09:22:44.000000000 +0200
@@ -14,7 +14,6 @@
 #include <dbi/dbd.h>
 
 #include <ibase.h>
-#include <gds.h>
 
 #include "dbd_firebird.h"
 #include "utility.h"
