From 1f67aa148aa95bf709b140a7e5036622f4001de0 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 31 Jan 2018 11:18:08 +0100
Subject: [PATCH] fix dependency on unix-compat

---
 snap-server.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/snap-server.cabal b/snap-server.cabal
index 993c841..758f0c9 100644
--- a/snap-server.cabal
+++ b/snap-server.cabal
@@ -112,7 +112,7 @@ Library
     snap-core                           >= 1.0      && < 1.1,
     text                                >= 0.11     && < 1.3,
     time                                >= 1.0      && < 1.9,
-    unix-compat                         >= 0.2      && < 0.5,
+    unix-compat                         >= 0.2      && < 0.6,
     vector                              >= 0.7      && < 0.13
 
   other-extensions:
-- 
2.15.1

