From 6a8af96e65c6799411606b54df3bca0133bb4ef8 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 9 Feb 2020 16:35:48 +0100
Subject: [PATCH] Do not generate documentation

---
 Makefile.am | 2 +-
 Makefile.in | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 9f3dd14..7ed751b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -40,7 +40,7 @@ if WITH_SYSTEMDSYSTEMUNITDIR
     SYSTEMD_DIR         = systemd
 endif
 
-SUBDIRS                 = lib daemons tools plugins configs doc $(SYSTEMD_DIR)
+SUBDIRS                 = lib daemons tools plugins configs $(SYSTEMD_DIR)
 
 if INSTALL_ETC
 
diff --git a/Makefile.in b/Makefile.in
index 8f2f951..bf41b36 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -268,7 +268,7 @@ am__define_uniq_tagged_files = \
 ETAGS = etags
 CTAGS = ctags
 CSCOPE = cscope
-DIST_SUBDIRS = lib daemons tools plugins configs doc systemd
+DIST_SUBDIRS = lib daemons tools plugins configs systemd
 am__DIST_COMMON = $(srcdir)/Makefile.in $(srcdir)/VERSION.in \
 	$(srcdir)/config.h.in $(srcdir)/lirc-driver.pc.in \
 	$(srcdir)/lirc.pc.in $(srcdir)/pylint.mak AUTHORS COPYING \
@@ -509,7 +509,7 @@ AM_DISTCHECK_CONFIGURE_FLAGS = \
 
 BUILT_SOURCES = paths.h python-pkg/lirc/config.py python-pkg/VERSION
 @WITH_SYSTEMDSYSTEMUNITDIR_TRUE@SYSTEMD_DIR = systemd
-SUBDIRS = lib daemons tools plugins configs doc $(SYSTEMD_DIR)
+SUBDIRS = lib daemons tools plugins configs $(SYSTEMD_DIR)
 @INSTALL_ETC_TRUE@lirc_confdir = $(sysconfdir)/lirc
 @INSTALL_ETC_TRUE@dist_lirc_conf_DATA = lircd.conf \
 @INSTALL_ETC_TRUE@                          lircmd.conf \
-- 
2.25.0

