--- gfx/2d/DrawTargetCairo.cpp.orig	2015-06-06 15:30:46.000000000 +0300
+++ gfx/2d/DrawTargetCairo.cpp	2015-07-23 18:45:45.406135553 +0300
@@ -638,8 +638,10 @@
 DrawTargetCairo::SetPermitSubpixelAA(bool aPermitSubpixelAA)
 {
   DrawTarget::SetPermitSubpixelAA(aPermitSubpixelAA);
+#ifdef MOZ_TREE_CAIRO
   cairo_surface_set_subpixel_antialiasing(mSurface,
     aPermitSubpixelAA ? CAIRO_SUBPIXEL_ANTIALIASING_ENABLED : CAIRO_SUBPIXEL_ANTIALIASING_DISABLED);
+#endif
 }
 
 void
