--- Makefile.orig	2017-09-28 12:31:05.024452057 -0400
+++ Makefile	2017-09-28 12:32:06.987691205 -0400
@@ -137,11 +137,10 @@
 .PHONY: install uninstall
 
 install: all
-	install -d -m 755 $(DOCDIR)/examples $(PREFIX)/sbin $(SERVICEDIR)
+	install -d -m 755 $(DOCDIR)/examples $(PREFIX)/sbin
 	install -m 644 AUTHORS COPYING NEWS README.rst TODO $(DOCDIR)
 	install -m 644 examples/*.ferm $(DOCDIR)/examples
 	install -m 755 src/ferm src/import-ferm $(PREFIX)/sbin/
-	install -m 644 ferm.service $(SERVICEDIR)
 
 	install -d -m 755 $(DOCDIR) $(MANDIR)
 	install -m 644 doc/ferm.txt doc/ferm.html $(DOCDIR)
