--- Makefile.orig	2016-03-30 08:16:02.000000000 -0400
+++ Makefile	2016-10-03 20:01:28.914713926 -0400
@@ -137,11 +137,10 @@
 .PHONY: install uninstall
 
 install: all
-	install -d -m 755 $(DOCDIR)/examples $(PREFIX)/sbin $(SERVICEDIR)
+	install -d -m 755 $(DOCDIR)/examples $(PREFIX)/sbin
 	install -m 644 AUTHORS COPYING NEWS README TODO $(DOCDIR)
 	install -m 644 examples/*.ferm $(DOCDIR)/examples
 	install -m 755 src/ferm src/import-ferm $(PREFIX)/sbin/
-	install -m 644 ferm.service $(SERVICEDIR)
 
 	install -d -m 755 $(DOCDIR) $(MANDIR)
 	install -m 644 doc/ferm.txt doc/ferm.html $(DOCDIR)
