--- Xvnc/include/Xpoll.h~	2004-06-30 17:26:29.150359031 +0200
+++ Xvnc/include/Xpoll.h	2004-06-30 17:31:44.504165469 +0200
@@ -115,50 +115,50 @@
  * array.
  */
 #define XFD_ANYSET(p) \
-		((p)->fds_bits[0] || (p)->fds_bits[1] || \
-		(p)->fds_bits[2] || (p)->fds_bits[3] || \
-		(p)->fds_bits[4] || (p)->fds_bits[5] || \
-		(p)->fds_bits[6] || (p)->fds_bits[7])
+		((p)->__fds_bits[0] || (p)->__fds_bits[1] || \
+		(p)->__fds_bits[2] || (p)->__fds_bits[3] || \
+		(p)->__fds_bits[4] || (p)->__fds_bits[5] || \
+		(p)->__fds_bits[6] || (p)->__fds_bits[7])
 
 #define XFD_COPYSET(src,dst) \
-		(dst)->fds_bits[0] = (src)->fds_bits[0]; \
-		(dst)->fds_bits[1] = (src)->fds_bits[1]; \
-		(dst)->fds_bits[2] = (src)->fds_bits[2]; \
-		(dst)->fds_bits[3] = (src)->fds_bits[3]; \
-		(dst)->fds_bits[4] = (src)->fds_bits[4]; \
-		(dst)->fds_bits[5] = (src)->fds_bits[5]; \
-		(dst)->fds_bits[6] = (src)->fds_bits[6]; \
-		(dst)->fds_bits[7] = (src)->fds_bits[7];
+		(dst)->__fds_bits[0] = (src)->__fds_bits[0]; \
+		(dst)->__fds_bits[1] = (src)->__fds_bits[1]; \
+		(dst)->__fds_bits[2] = (src)->__fds_bits[2]; \
+		(dst)->__fds_bits[3] = (src)->__fds_bits[3]; \
+		(dst)->__fds_bits[4] = (src)->__fds_bits[4]; \
+		(dst)->__fds_bits[5] = (src)->__fds_bits[5]; \
+		(dst)->__fds_bits[6] = (src)->__fds_bits[6]; \
+		(dst)->__fds_bits[7] = (src)->__fds_bits[7];
 
 #define XFD_ANDSET(dst,b1,b2) \
-		(dst)->fds_bits[0] = ((b1)->fds_bits[0] & (b2)->fds_bits[0]); \
-		(dst)->fds_bits[1] = ((b1)->fds_bits[1] & (b2)->fds_bits[1]); \
-		(dst)->fds_bits[2] = ((b1)->fds_bits[2] & (b2)->fds_bits[2]); \
-		(dst)->fds_bits[3] = ((b1)->fds_bits[3] & (b2)->fds_bits[3]); \
-		(dst)->fds_bits[4] = ((b1)->fds_bits[4] & (b2)->fds_bits[4]); \
-		(dst)->fds_bits[5] = ((b1)->fds_bits[5] & (b2)->fds_bits[5]); \
-		(dst)->fds_bits[6] = ((b1)->fds_bits[6] & (b2)->fds_bits[6]); \
-		(dst)->fds_bits[7] = ((b1)->fds_bits[7] & (b2)->fds_bits[7]);
+		(dst)->__fds_bits[0] = ((b1)->__fds_bits[0] & (b2)->__fds_bits[0]); \
+		(dst)->__fds_bits[1] = ((b1)->__fds_bits[1] & (b2)->__fds_bits[1]); \
+		(dst)->__fds_bits[2] = ((b1)->__fds_bits[2] & (b2)->__fds_bits[2]); \
+		(dst)->__fds_bits[3] = ((b1)->__fds_bits[3] & (b2)->__fds_bits[3]); \
+		(dst)->__fds_bits[4] = ((b1)->__fds_bits[4] & (b2)->__fds_bits[4]); \
+		(dst)->__fds_bits[5] = ((b1)->__fds_bits[5] & (b2)->__fds_bits[5]); \
+		(dst)->__fds_bits[6] = ((b1)->__fds_bits[6] & (b2)->__fds_bits[6]); \
+		(dst)->__fds_bits[7] = ((b1)->__fds_bits[7] & (b2)->__fds_bits[7]);
 
 #define XFD_ORSET(dst,b1,b2) \
-		(dst)->fds_bits[0] = ((b1)->fds_bits[0] | (b2)->fds_bits[0]); \
-		(dst)->fds_bits[1] = ((b1)->fds_bits[1] | (b2)->fds_bits[1]); \
-		(dst)->fds_bits[2] = ((b1)->fds_bits[2] | (b2)->fds_bits[2]); \
-		(dst)->fds_bits[3] = ((b1)->fds_bits[3] | (b2)->fds_bits[3]); \
-		(dst)->fds_bits[4] = ((b1)->fds_bits[4] | (b2)->fds_bits[4]); \
-		(dst)->fds_bits[5] = ((b1)->fds_bits[5] | (b2)->fds_bits[5]); \
-		(dst)->fds_bits[6] = ((b1)->fds_bits[6] | (b2)->fds_bits[6]); \
-		(dst)->fds_bits[7] = ((b1)->fds_bits[7] | (b2)->fds_bits[7]);
+		(dst)->__fds_bits[0] = ((b1)->__fds_bits[0] | (b2)->__fds_bits[0]); \
+		(dst)->__fds_bits[1] = ((b1)->__fds_bits[1] | (b2)->__fds_bits[1]); \
+		(dst)->__fds_bits[2] = ((b1)->__fds_bits[2] | (b2)->__fds_bits[2]); \
+		(dst)->__fds_bits[3] = ((b1)->__fds_bits[3] | (b2)->__fds_bits[3]); \
+		(dst)->__fds_bits[4] = ((b1)->__fds_bits[4] | (b2)->__fds_bits[4]); \
+		(dst)->__fds_bits[5] = ((b1)->__fds_bits[5] | (b2)->__fds_bits[5]); \
+		(dst)->__fds_bits[6] = ((b1)->__fds_bits[6] | (b2)->__fds_bits[6]); \
+		(dst)->__fds_bits[7] = ((b1)->__fds_bits[7] | (b2)->__fds_bits[7]);
 
 #define XFD_UNSET(dst,b1) \
-		(dst)->fds_bits[0] &= ~((b1)->fds_bits[0]); \
-		(dst)->fds_bits[1] &= ~((b1)->fds_bits[1]); \
-		(dst)->fds_bits[2] &= ~((b1)->fds_bits[2]); \
-		(dst)->fds_bits[3] &= ~((b1)->fds_bits[3]); \
-		(dst)->fds_bits[4] &= ~((b1)->fds_bits[4]); \
-		(dst)->fds_bits[5] &= ~((b1)->fds_bits[5]); \
-		(dst)->fds_bits[6] &= ~((b1)->fds_bits[6]); \
-		(dst)->fds_bits[7] &= ~((b1)->fds_bits[7]);
+		(dst)->__fds_bits[0] &= ~((b1)->__fds_bits[0]); \
+		(dst)->__fds_bits[1] &= ~((b1)->__fds_bits[1]); \
+		(dst)->__fds_bits[2] &= ~((b1)->__fds_bits[2]); \
+		(dst)->__fds_bits[3] &= ~((b1)->__fds_bits[3]); \
+		(dst)->__fds_bits[4] &= ~((b1)->__fds_bits[4]); \
+		(dst)->__fds_bits[5] &= ~((b1)->__fds_bits[5]); \
+		(dst)->__fds_bits[6] &= ~((b1)->__fds_bits[6]); \
+		(dst)->__fds_bits[7] &= ~((b1)->__fds_bits[7]);
 
 #else /* USE_POLL */
 #include <sys/poll.h>
--- Xvnc/programs/Xserver/os/WaitFor.c~	2000-06-11 14:00:52.000000000 +0200
+++ Xvnc/programs/Xserver/os/WaitFor.c	2004-06-30 17:43:05.445375799 +0200
@@ -390,7 +390,7 @@
 	    XFD_ANDSET(&devicesReadable, &LastSelectMask, &EnabledDevices);
 	    XFD_ANDSET(&clientsReadable, &LastSelectMask, &AllClients); 
 #ifndef WIN32
-	    if (LastSelectMask.fds_bits[0] & WellKnownConnections.fds_bits[0]) 
+	    if (LastSelectMask.__fds_bits[0] & WellKnownConnections.__fds_bits[0]) 
 #else
 	    XFD_ANDSET(&tmp_set, &LastSelectMask, &WellKnownConnections);
 	    if (XFD_ANYSET(&tmp_set))
@@ -414,11 +414,11 @@
 	{
 	    int highest_priority;
 
-	    while (clientsReadable.fds_bits[i])
+	    while (clientsReadable.__fds_bits[i])
 	    {
 	        int client_priority, client_index;
 
-		curclient = ffs (clientsReadable.fds_bits[i]) - 1;
+		curclient = ffs (clientsReadable.__fds_bits[i]) - 1;
 		client_index = ConnectionTranslation[curclient + (i << 5)];
 #else
 	int highest_priority;
@@ -463,7 +463,7 @@
 		    pClientsReady[nready++] = client_index;
 		}
 #ifndef WIN32
-		clientsReadable.fds_bits[i] &= ~(((fd_mask)1) << curclient);
+		clientsReadable.__fds_bits[i] &= ~(((fd_mask)1) << curclient);
 	    }
 #else
 	    FD_CLR(curclient, &clientsReadable);
--- Xvnc/programs/Xserver/os/connection.c~	2004-06-30 17:55:38.654375206 +0200
+++ Xvnc/programs/Xserver/os/connection.c	2004-06-30 17:55:49.991930173 +0200
@@ -783,7 +783,7 @@
 #ifndef AMOEBA
     XFD_ANDSET (&tmask, (fd_set*)closure, &WellKnownConnections);
 #ifndef WIN32
-    readyconnections = tmask.fds_bits[0];
+    readyconnections = tmask.__fds_bits[0];
     if (!readyconnections)
 	return TRUE;
 #else
@@ -984,7 +984,7 @@
 #ifndef WIN32
     for (i=0; i<howmany(XFD_SETSIZE, NFDBITS); i++)
     {
-	mask = AllClients.fds_bits[i];
+	mask = AllClients.__fds_bits[i];
         while (mask)
     	{
 	    curoff = ffs (mask) - 1;
--- Xvnc/programs/Xserver/os/io.c~	2000-06-11 14:00:52.000000000 +0200
+++ Xvnc/programs/Xserver/os/io.c	2004-06-30 17:57:09.493797226 +0200
@@ -992,8 +992,8 @@
 #ifndef WIN32
     for (base = 0; base < howmany(XFD_SETSIZE, NFDBITS); base++)
     {
-	mask = OutputPending.fds_bits[ base ];
-	OutputPending.fds_bits[ base ] = 0;
+	mask = OutputPending.__fds_bits[ base ];
+	OutputPending.__fds_bits[ base ] = 0;
 	while (mask)
 	{
 	    index = ffs(mask) - 1;
