--- Xvnc/lib/Xau/AuRead.c~	2000-06-11 14:00:52.000000000 +0200
+++ Xvnc/lib/Xau/AuRead.c	2004-06-30 17:52:59.126707877 +0200
@@ -49,7 +49,7 @@
 FILE	*file;
 {
     unsigned short  len;
-    char	    *data, *malloc ();
+    char	    *data;
 
     if (read_short (&len, file) == 0)
 	return 0;
@@ -76,7 +76,6 @@
 {
     Xauth   local;
     Xauth   *ret;
-    char    *malloc ();
 
     if (read_short (&local.family, auth_file) == 0)
 	return 0;
--- Xvnc/lib/Xau/AuWrite.c~	2000-06-11 14:00:52.000000000 +0200
+++ Xvnc/lib/Xau/AuWrite.c	2004-06-30 17:53:17.939310108 +0200
@@ -61,7 +61,6 @@
 FILE	*auth_file;
 Xauth	*auth;
 {
-    char    *malloc ();
 
     if (write_short (auth->family, auth_file) == 0)
 	return 0;
