From e67d91c9ab7da671eb16c931fff5cdd6a6bca25c Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 31 Dec 2017 14:13:57 +0100
Subject: [PATCH] don't install ChangeLog

---
 Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 113719b..69b96ef 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,9 +10,9 @@ man_MANS = geeqie.1
 readmedir = @readmedir@
 
 if HAVE_MARKDOWN
-readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS README.html ChangeLog.html
+readme_DATA = README.md COPYING TODO README.lirc AUTHORS README.html
 else
-readme_DATA = README.md COPYING ChangeLog TODO README.lirc AUTHORS ChangeLog.html
+readme_DATA = README.md COPYING TODO README.lirc AUTHORS
 endif
 
 desktopdir = $(datadir)/applications
-- 
2.15.1

