--- systrace-1.6d/regress/regress.sh	2006-03-24 04:17:16.000000000 +0100
+++ systrace-1.6d-regress_patch/regress/regress.sh	2006-04-18 00:43:48.000000000 +0200
@@ -12,8 +12,9 @@ for POL in *.policy.$OS; do
 	fi
 
 	echo -n "$PROG:"
-	SYSTR_RES=`eval ../systrace -f $POL -a $PROG $ARGS 2>/dev/null`
-	# echo -e "\t(../systrace -f $POL -a $PROG $ARGS)"
+	SYSTR_POL=`eval ../systrace -t -d . -A $PROG $ARGS 2>/dev/null`
+	SYSTR_RES=`eval ../systrace -t -d . -a $PROG $ARGS 2>/dev/null`
+	echo -e "\t(../systrace -f $POL.learn -a $PROG $ARGS)"
 	NORM_RES=`$PROG $ARGS`
 
 	if [ -z "$SYSTR_RES" ] ; then
