From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@sourcemage.org>
Date: Sat,  1 Feb 2025 19:38:05 +0100
Subject: [PATCH] Fix build against musl libc

Upstream-Status: Pending
Origin: Source Mage
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 libobs/util/threading-posix.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libobs/util/threading-posix.c b/libobs/util/threading-posix.c
index 6d7c40ee74b6..8960e70cc95c 100644
--- a/libobs/util/threading-posix.c
+++ b/libobs/util/threading-posix.c
@@ -261,7 +261,7 @@ void os_set_thread_name(const char *name)
 	pthread_setname_np(name);
 #elif defined(__FreeBSD__)
 	pthread_set_name_np(pthread_self(), name);
-#elif defined(__GLIBC__) && !defined(__MINGW32__)
+#elif !defined(__MINGW32__)
 	if (strlen(name) <= 15) {
 		pthread_setname_np(pthread_self(), name);
 	} else {
