From f45ad40561266483293dc1a4f64d18ecaede4b91 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sat, 5 Oct 2019 15:26:22 +0200
Subject: [PATCH] fix compile errors

---
 CMakeLists.txt    | 2 +-
 src/test/tgkill.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 596f826..d22bcf8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,7 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${FLAGS_COMMON} -Wstrict-prototypes -std=gnu
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${FLAGS_COMMON} -D__STDC_LIMIT_MACROS -D__STDC_FORMAT_MACROS -std=c++14")
 set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -g3")
 
-set(RR_FLAGS_DEBUG "-Wall -Wextra -Werror -O0 -DDEBUG -UNDEBUG")
+set(RR_FLAGS_DEBUG "-Wall -Wextra -Wno-error -O0 -DDEBUG -UNDEBUG")
 set(RR_FLAGS_RELEASE "-Wall -Wextra -O2 -UDEBUG -DNDEBUG")
 set(RR_TEST_FLAGS "${RR_FLAGS_DEBUG}")
 
diff --git a/src/test/tgkill.c b/src/test/tgkill.c
index 1446b7c..c7de4dc 100644
--- a/src/test/tgkill.c
+++ b/src/test/tgkill.c
@@ -4,7 +4,7 @@
 
 static int num_signals_caught;
 
-static int tgkill(int tgid, int tid, int sig) {
+int tgkill(int tgid, int tid, int sig) {
   return syscall(SYS_tgkill, tgid, tid, sig);
 }
 
-- 
2.23.0

