diff -Naur libsndfile-1.0.17.orig/acinclude.m4 libsndfile-1.0.17/acinclude.m4
--- libsndfile-1.0.17.orig/acinclude.m4	2008-04-13 10:18:59.000000000 +0000
+++ libsndfile-1.0.17/acinclude.m4	2008-04-13 10:19:09.000000000 +0000
@@ -265,13 +265,7 @@
 else
 	ac_cv_c_big_endian=0
 	ac_cv_c_little_endian=0
-
-	AC_MSG_WARN([[*****************************************************************]])
-	AC_MSG_WARN([[*** Not able to determine endian-ness of target processor.       ]])
-	AC_MSG_WARN([[*** The constants CPU_IS_BIG_ENDIAN and CPU_IS_LITTLE_ENDIAN in  ]])
-	AC_MSG_WARN([[*** src/config.h may need to be hand editied.                    ]])
-	AC_MSG_WARN([[*****************************************************************]])
-	fi
+fi
 
 )# AC_C_FIND_ENDIAN
 
