diff -Naur libgksu-2.0.12.orig/Makefile.am libgksu-2.0.12/Makefile.am
--- libgksu-2.0.12.orig/Makefile.am	2011-03-10 16:15:00.190590002 -0800
+++ libgksu-2.0.12/Makefile.am	2011-03-10 16:18:49.910590000 -0800
@@ -17,7 +17,7 @@
 
 if GCONF_SCHEMAS_INSTALL
 install-data-local:
-        if test -z "$(DESTDIR)" ; then \
+	if test -z "$(DESTDIR)" ; then \
           for p in $(schemas_DATA) ; do \
             GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$$p ; \
           done \
diff -Naur libgksu-2.0.12.orig/Makefile.in libgksu-2.0.12/Makefile.in
--- libgksu-2.0.12.orig/Makefile.in	2011-03-10 16:15:00.190590002 -0800
+++ libgksu-2.0.12/Makefile.in	2011-03-10 16:19:05.715590000 -0800
@@ -729,7 +729,7 @@
 @INTLTOOL_SCHEMAS_RULE@
 
 @GCONF_SCHEMAS_INSTALL_TRUE@install-data-local:
-@GCONF_SCHEMAS_INSTALL_TRUE@        if test -z "$(DESTDIR)" ; then \
+@GCONF_SCHEMAS_INSTALL_TRUE@	if test -z "$(DESTDIR)" ; then \
 @GCONF_SCHEMAS_INSTALL_TRUE@          for p in $(schemas_DATA) ; do \
 @GCONF_SCHEMAS_INSTALL_TRUE@            GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$$p ; \
 @GCONF_SCHEMAS_INSTALL_TRUE@          done \
