--- etc.old/switch.conf	2006-08-14 16:24:03.000000000 +0200
+++ etc/switch.conf	2006-08-14 16:28:42.000000000 +0200
@@ -1,4 +1,4 @@
-ALLOWED="@adm"
+ALLOWED="@wheel"
 # La ligne suivante permet de définir le répertoire contenant les 
 # profils réseaux
 CONFIG_PREFIX="/etc/netswitch/profiles"
@@ -11,19 +11,19 @@
 SMTPDIR="/etc/nullmailer"
 SMTP_SCRIPT="/etc/init.d/svscan"
 
-IWBIN="/sbin/iwconfig"
-IWLIST="/sbin/iwlist"
+IWBIN="/usr/sbin/iwconfig"
+IWLIST="/usr/sbin/iwlist"
 
 WPABIN="/usr/sbin/wpa_supplicant"
 WPA_PASSPHRASE="/usr/sbin/wpa_passphrase"
 WPA_CONF="/etc/netswitch/wpa.conf"
 WPA_PRECONF="/etc/netswitch/wpa.tmpl"
 
-IPBIN="/sbin/ip"
+IPBIN="/usr/sbin/ip"
 IFBIN="/sbin/ifconfig"
 PPPBIN="/usr/sbin/pppd"
 ROUTEBIN="/sbin/route"
-INITDIR="/etc/xinit.d"
+INITDIR="/etc/init.d"
 
 DHCP_CLIENT="/sbin/dhcpcd"
 if [ ! -x "$DHCP_CLIENT" ]; then
--- Makefile.in.old	2006-08-14 16:54:52.000000000 +0200
+++ Makefile.in	2006-08-14 16:55:05.000000000 +0200
@@ -211,7 +211,7 @@
 pcdata_DATA = netswitch.pc
 pcdatadir = $(libdir)/pkgconfig
 man_MANS = netswitch.1
-EXTRA_DIST = netswitch.png netswitch-16x16.png netswitch.1 etc/switch etc/switch.conf init.d
+EXTRA_DIST = netswitch.png netswitch-16x16.png netswitch.1 etc/switch etc/switch.conf 
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
@@ -728,13 +728,11 @@
 	$(INSTALL_DATA) -D netswitch-16x16.png $(DESTDIR)$(datadir)/icons/hicolor/24x24/apps/netswitch.png
 	$(INSTALL_DATA) -D etc/switch.conf $(DESTDIR)$(sysconfdir)/netswitch/switch.conf
 	$(INSTALL_SCRIPT) -D etc/switch $(DESTDIR)$(bindir)/netswitch
-	$(INSTALL_SCRIPT) -D init.d $(DESTDIR)$(sysconfdir)/init.d/netswitch
 
 uninstall-local:
 	@$(NORMAL_UNINSTALL)
 	rm -f $(DESTDIR)$(datadir)/pixmaps/netswitch.png
 	rm -f $(DESTDIR)$(bindir)/netswitch
-	rm -f $(DESTDIR)$(sysconfdir)/init.d/netswitch
 	rm -rf $(DESTDIR)$(syscondif)/netswitch
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
