*** gui.c.old	2003-12-04 11:34:46.000000000 +0100
--- gui.c	2003-12-04 11:36:18.000000000 +0100
***************
*** 26,32 ****
  
  static const char about_text[] =
! "Crystality Plugin v0.92
  
! Bandwidth Extender, Harmonic Booster and 3D Echo
  You will need a good stereo and a good ear to notice quality
  improvement, otherwise this is not for you.  This plugin tries
--- 26,32 ----
  
  static const char about_text[] =
! "Crystality Plugin v0.92";
  
! /*Bandwidth Extender, Harmonic Booster and 3D Echo
  You will need a good stereo and a good ear to notice quality
  improvement, otherwise this is not for you.  This plugin tries
***************
*** 48,52 ****
  along with this program; if not, write to the Free Software
  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
! ";
  
  GtkWidget *conf_dialog = NULL, *about_dialog = NULL;
--- 48,52 ----
  along with this program; if not, write to the Free Software
  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
! ";*/
  
  GtkWidget *conf_dialog = NULL, *about_dialog = NULL;
