--- Makefile.PL	2005-02-10 13:27:15.000000000 +0100
+++ Makefile.PL.new	2005-02-21 18:38:29.277746320 +0100
@@ -5,19 +5,11 @@
 
 my @programs;
 
-for (split "\n", <<'QUERIES') {
-xml_pp|y|XML pretty printer
-xml_grep|y|XML grep - grep XML files using XML::Twig's subset of XPath
-xml_split|y|split big XML files
-xml_merge|y|merge back files created by xml_split
-xml_spellcheck|y|spellcheck XML files skipping tags
-QUERIES
-    my ($program, $default, $description) = split /\|/, $_;
-    if (prompt("Do you want to install '$program' ($description)?", 
-               $default) =~ /^y/) {
-	push(@programs, $program);
-    }
-}
+push(@programs,"xml_pp");
+push(@programs,"xml_grep");
+push(@programs,"xml_split");
+push(@programs,"xml_merge");
+push(@programs,"xml_spellcheck");
 
 WriteMakefile(
     NAME	     => 'XML::Twig',
