diff -purN ZopeInterface-3.0.1/Dependencies/zope.interface-ZopeInterface-3.0.1/zope.interface/_zope_interface_coptimizations.c ZopeInterface/Dependencies/zope.interface-ZopeInterface-3.0.1/zope.interface/_zope_interface_coptimizations.c
--- ZopeInterface-3.0.1/Dependencies/zope.interface-ZopeInterface-3.0.1/zope.interface/_zope_interface_coptimizations.c	2004-08-04 06:03:17.000000000 -0400
+++ ZopeInterface/Dependencies/zope.interface-ZopeInterface-3.0.1/zope.interface/_zope_interface_coptimizations.c	2005-04-24 21:31:59.000000000 -0400
@@ -70,7 +70,7 @@ import_declarations(void)
   return 0;
 }
 
-extern PyTypeObject SpecType;   /* Forward */
+static PyTypeObject SpecType;   /* Forward */
 
 static PyObject *
 implementedByFallback(PyObject *cls)
