diff -Naur krb5-1.6.3.orig/src/appl/bsd/compat_recv.c krb5-1.6.3/src/appl/bsd/compat_recv.c
--- krb5-1.6.3.orig/src/appl/bsd/compat_recv.c	2008-03-21 22:47:40.315090009 -0700
+++ krb5-1.6.3/src/appl/bsd/compat_recv.c	2008-03-21 23:11:55.522085601 -0700
@@ -41,7 +41,7 @@
 #ifdef KRB5_KRB4_COMPAT
 #include <kerberosIV/krb.h>
 #endif
-#include "com_err.h"
+#include "et/com_err.h"
 #include <errno.h>
 
 #include <stdio.h>
diff -Naur krb5-1.6.3.orig/src/appl/bsd/krcp.c krb5-1.6.3/src/appl/bsd/krcp.c
--- krb5-1.6.3.orig/src/appl/bsd/krcp.c	2008-03-21 22:47:40.315090009 -0700
+++ krb5-1.6.3/src/appl/bsd/krcp.c	2008-03-21 23:11:55.513078286 -0700
@@ -70,7 +70,7 @@
 #ifdef KERBEROS
 #include <krb5.h>
 #include <k5-util.h>
-#include <com_err.h>
+#include <et/com_err.h>
 
 #ifdef KRB5_KRB4_COMPAT
 #include <kerberosIV/krb.h>
diff -Naur krb5-1.6.3.orig/src/appl/bsd/krlogin.c krb5-1.6.3/src/appl/bsd/krlogin.c
--- krb5-1.6.3.orig/src/appl/bsd/krlogin.c	2008-03-21 22:47:40.314091152 -0700
+++ krb5-1.6.3/src/appl/bsd/krlogin.c	2008-03-21 23:11:55.500085875 -0700
@@ -156,7 +156,7 @@
 
 #ifdef KERBEROS
 #include <krb5.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #ifdef KRB5_KRB4_COMPAT
 #include <kerberosIV/krb.h>
 #endif
diff -Naur krb5-1.6.3.orig/src/appl/bsd/krlogind.c krb5-1.6.3/src/appl/bsd/krlogind.c
--- krb5-1.6.3.orig/src/appl/bsd/krlogind.c	2008-03-21 22:47:40.314091152 -0700
+++ krb5-1.6.3/src/appl/bsd/krlogind.c	2008-03-21 23:11:55.484085719 -0700
@@ -256,7 +256,7 @@
 Key_schedule v4_schedule;
 #endif
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "defines.h"
      
 #define SECURE_MESSAGE  "This rlogin session is encrypting all data transmissions.\r\n"
diff -Naur krb5-1.6.3.orig/src/appl/bsd/krsh.c krb5-1.6.3/src/appl/bsd/krsh.c
--- krb5-1.6.3.orig/src/appl/bsd/krsh.c	2008-03-21 22:47:40.314091152 -0700
+++ krb5-1.6.3/src/appl/bsd/krsh.c	2008-03-21 23:11:55.470085234 -0700
@@ -63,7 +63,7 @@
 
 #ifdef KERBEROS
 #include <krb5.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #ifdef KRB5_KRB4_COMPAT
 #include <kerberosIV/krb.h>
 #endif
diff -Naur krb5-1.6.3.orig/src/appl/bsd/krshd.c krb5-1.6.3/src/appl/bsd/krshd.c
--- krb5-1.6.3.orig/src/appl/bsd/krshd.c	2008-03-21 22:47:40.313091456 -0700
+++ krb5-1.6.3/src/appl/bsd/krshd.c	2008-03-21 23:11:55.442085100 -0700
@@ -161,7 +161,7 @@
 
 #ifdef KERBEROS
 #include <krb5.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include "loginpaths.h"
 #ifdef KRB5_KRB4_COMPAT
 #include <kerberosIV/krb.h>
diff -Naur krb5-1.6.3.orig/src/appl/bsd/login.c krb5-1.6.3/src/appl/bsd/login.c
--- krb5-1.6.3.orig/src/appl/bsd/login.c	2008-03-21 22:47:40.314091152 -0700
+++ krb5-1.6.3/src/appl/bsd/login.c	2008-03-21 23:11:55.460085485 -0700
@@ -155,7 +155,7 @@
 /* #include "krb5.h" */
 /* need k5-int.h to get ->profile from krb5_context */
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "osconf.h"
 #endif /* KRB5_GET_TICKETS */
 
diff -Naur krb5-1.6.3.orig/src/appl/bsd/v4rcp.c krb5-1.6.3/src/appl/bsd/v4rcp.c
--- krb5-1.6.3.orig/src/appl/bsd/v4rcp.c	2008-03-21 22:47:40.315090009 -0700
+++ krb5-1.6.3/src/appl/bsd/v4rcp.c	2008-03-21 23:11:55.535085553 -0700
@@ -37,7 +37,7 @@
  */
 #ifdef KERBEROS
 #include <krb5.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <k5-util.h>
 #endif
 
diff -Naur krb5-1.6.3.orig/src/appl/libpty/cleanup.c krb5-1.6.3/src/appl/libpty/cleanup.c
--- krb5-1.6.3.orig/src/appl/libpty/cleanup.c	2008-03-21 22:47:40.320089604 -0700
+++ krb5-1.6.3/src/appl/libpty/cleanup.c	2008-03-21 23:11:55.588086175 -0700
@@ -20,7 +20,7 @@
  * 
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 #ifdef HAVE_SYS_WAIT_H
diff -Naur krb5-1.6.3.orig/src/appl/libpty/getpty.c krb5-1.6.3/src/appl/libpty/getpty.c
--- krb5-1.6.3.orig/src/appl/libpty/getpty.c	2008-03-21 22:47:40.321090417 -0700
+++ krb5-1.6.3/src/appl/libpty/getpty.c	2008-03-21 23:11:55.605086585 -0700
@@ -20,7 +20,7 @@
  * 
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/libpty/init.c krb5-1.6.3/src/appl/libpty/init.c
--- krb5-1.6.3.orig/src/appl/libpty/init.c	2008-03-21 22:47:40.322090113 -0700
+++ krb5-1.6.3/src/appl/libpty/init.c	2008-03-21 23:11:55.633086160 -0700
@@ -22,7 +22,7 @@
  * 
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/libpty/init_slave.c krb5-1.6.3/src/appl/libpty/init_slave.c
--- krb5-1.6.3.orig/src/appl/libpty/init_slave.c	2008-03-21 22:47:40.320089604 -0700
+++ krb5-1.6.3/src/appl/libpty/init_slave.c	2008-03-21 23:11:55.542085657 -0700
@@ -19,7 +19,7 @@
  * 
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/libpty/logwtmp.c krb5-1.6.3/src/appl/libpty/logwtmp.c
--- krb5-1.6.3.orig/src/appl/libpty/logwtmp.c	2008-03-21 22:47:40.320089604 -0700
+++ krb5-1.6.3/src/appl/libpty/logwtmp.c	2008-03-21 23:11:55.570086628 -0700
@@ -19,7 +19,7 @@
  * 
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/libpty/open_ctty.c krb5-1.6.3/src/appl/libpty/open_ctty.c
--- krb5-1.6.3.orig/src/appl/libpty/open_ctty.c	2008-03-21 22:47:40.322090113 -0700
+++ krb5-1.6.3/src/appl/libpty/open_ctty.c	2008-03-21 23:11:55.626086057 -0700
@@ -19,7 +19,7 @@
  * 
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/libpty/open_slave.c krb5-1.6.3/src/appl/libpty/open_slave.c
--- krb5-1.6.3.orig/src/appl/libpty/open_slave.c	2008-03-21 22:47:40.320089604 -0700
+++ krb5-1.6.3/src/appl/libpty/open_slave.c	2008-03-21 23:11:55.549086039 -0700
@@ -21,7 +21,7 @@
  * 
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/libpty/pty_paranoia.c krb5-1.6.3/src/appl/libpty/pty_paranoia.c
--- krb5-1.6.3.orig/src/appl/libpty/pty_paranoia.c	2008-03-21 22:47:40.320089604 -0700
+++ krb5-1.6.3/src/appl/libpty/pty_paranoia.c	2008-03-21 23:11:55.598085923 -0700
@@ -184,7 +184,7 @@
  * non-root.
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 #include <sys/wait.h>
diff -Naur krb5-1.6.3.orig/src/appl/libpty/sane_hostname.c krb5-1.6.3/src/appl/libpty/sane_hostname.c
--- krb5-1.6.3.orig/src/appl/libpty/sane_hostname.c	2008-03-21 22:47:40.321090417 -0700
+++ krb5-1.6.3/src/appl/libpty/sane_hostname.c	2008-03-21 23:11:55.612085851 -0700
@@ -20,7 +20,7 @@
  * express or implied warranty.
  * 
  */
-#include "com_err.h"
+#include "et/com_err.h"
 #include "pty-int.h"
 #include <sys/socket.h>
 #include "libpty.h"
diff -Naur krb5-1.6.3.orig/src/appl/libpty/update_utmp.c krb5-1.6.3/src/appl/libpty/update_utmp.c
--- krb5-1.6.3.orig/src/appl/libpty/update_utmp.c	2008-03-21 22:47:40.320089604 -0700
+++ krb5-1.6.3/src/appl/libpty/update_utmp.c	2008-03-21 23:11:55.581085793 -0700
@@ -316,7 +316,7 @@
  * username both in utmp and wtmp.
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/libpty/update_wtmp.c krb5-1.6.3/src/appl/libpty/update_wtmp.c
--- krb5-1.6.3.orig/src/appl/libpty/update_wtmp.c	2008-03-21 22:47:40.320089604 -0700
+++ krb5-1.6.3/src/appl/libpty/update_wtmp.c	2008-03-21 23:11:55.563086245 -0700
@@ -19,7 +19,7 @@
  * 
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/libpty/vhangup.c krb5-1.6.3/src/appl/libpty/vhangup.c
--- krb5-1.6.3.orig/src/appl/libpty/vhangup.c	2008-03-21 22:47:40.322090113 -0700
+++ krb5-1.6.3/src/appl/libpty/vhangup.c	2008-03-21 23:11:55.619085674 -0700
@@ -20,7 +20,7 @@
  * 
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/libpty/void_assoc.c krb5-1.6.3/src/appl/libpty/void_assoc.c
--- krb5-1.6.3.orig/src/appl/libpty/void_assoc.c	2008-03-21 22:47:40.320089604 -0700
+++ krb5-1.6.3/src/appl/libpty/void_assoc.c	2008-03-21 23:11:55.555085609 -0700
@@ -20,7 +20,7 @@
  * 
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "libpty.h"
 #include "pty-int.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/sample/sclient/sclient.c krb5-1.6.3/src/appl/sample/sclient/sclient.c
--- krb5-1.6.3.orig/src/appl/sample/sclient/sclient.c	2008-03-21 22:47:40.322090113 -0700
+++ krb5-1.6.3/src/appl/sample/sclient/sclient.c	2008-03-21 23:11:55.649085478 -0700
@@ -30,7 +30,7 @@
  */
 
 #include "krb5.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include <stdio.h>
 #include <string.h>
diff -Naur krb5-1.6.3.orig/src/appl/sample/sserver/sserver.c krb5-1.6.3/src/appl/sample/sserver/sserver.c
--- krb5-1.6.3.orig/src/appl/sample/sserver/sserver.c	2008-03-21 22:47:40.322090113 -0700
+++ krb5-1.6.3/src/appl/sample/sserver/sserver.c	2008-03-21 23:11:55.641086517 -0700
@@ -38,7 +38,7 @@
  */
 
 #include "krb5.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include <stdio.h>
 #include <string.h>
diff -Naur krb5-1.6.3.orig/src/appl/simple/client/sim_client.c krb5-1.6.3/src/appl/simple/client/sim_client.c
--- krb5-1.6.3.orig/src/appl/simple/client/sim_client.c	2008-03-21 22:47:40.311090669 -0700
+++ krb5-1.6.3/src/appl/simple/client/sim_client.c	2008-03-21 23:11:55.416086313 -0700
@@ -38,7 +38,7 @@
 #include <errno.h>
 
 #include <krb5.h>
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include "simple.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/simple/server/sim_server.c krb5-1.6.3/src/appl/simple/server/sim_server.c
--- krb5-1.6.3.orig/src/appl/simple/server/sim_server.c	2008-03-21 22:47:40.311090669 -0700
+++ krb5-1.6.3/src/appl/simple/server/sim_server.c	2008-03-21 23:11:55.424086671 -0700
@@ -41,7 +41,7 @@
 #include <netinet/in.h>
 #include <netdb.h>
 
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include "simple.h"
 
diff -Naur krb5-1.6.3.orig/src/appl/telnet/libtelnet/kerberos5.c krb5-1.6.3/src/appl/telnet/libtelnet/kerberos5.c
--- krb5-1.6.3.orig/src/appl/telnet/libtelnet/kerberos5.c	2008-03-21 22:47:40.308089906 -0700
+++ krb5-1.6.3/src/appl/telnet/libtelnet/kerberos5.c	2008-03-21 23:11:55.380085822 -0700
@@ -67,7 +67,7 @@
 #include <stdio.h>
 #include "krb5.h"
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include <netdb.h>
 #include <ctype.h>
 #include <syslog.h>
diff -Naur krb5-1.6.3.orig/src/appl/telnet/telnetd/sys_term.c krb5-1.6.3/src/appl/telnet/telnetd/sys_term.c
--- krb5-1.6.3.orig/src/appl/telnet/telnetd/sys_term.c	2008-03-21 22:47:40.310090135 -0700
+++ krb5-1.6.3/src/appl/telnet/telnetd/sys_term.c	2008-03-21 23:11:55.393084937 -0700
@@ -35,7 +35,7 @@
 
 #include "telnetd.h"
 #include "pathnames.h"
-#include <com_err.h>
+#include <et/com_err.h>
 
 #ifndef LOGIN_PROGRAM
 #define LOGIN_PROGRAM _PATH_LOGIN
diff -Naur krb5-1.6.3.orig/src/appl/telnet/telnetd/telnetd.c krb5-1.6.3/src/appl/telnet/telnetd/telnetd.c
--- krb5-1.6.3.orig/src/appl/telnet/telnetd/telnetd.c	2008-03-21 22:47:40.310090135 -0700
+++ krb5-1.6.3/src/appl/telnet/telnetd/telnetd.c	2008-03-21 23:11:55.408084839 -0700
@@ -58,7 +58,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <libpty.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #if	defined(_SC_CRAY_SECURE_SYS)
 #include <sys/sysv.h>
 #include <sys/secdev.h>
diff -Naur krb5-1.6.3.orig/src/appl/user_user/client.c krb5-1.6.3/src/appl/user_user/client.c
--- krb5-1.6.3.orig/src/appl/user_user/client.c	2008-03-21 22:47:40.302090616 -0700
+++ krb5-1.6.3/src/appl/user_user/client.c	2008-03-21 23:11:55.368085566 -0700
@@ -37,7 +37,7 @@
 #include <errno.h>
 
 #include "krb5.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 int main (argc, argv)
 int argc;
diff -Naur krb5-1.6.3.orig/src/appl/user_user/server.c krb5-1.6.3/src/appl/user_user/server.c
--- krb5-1.6.3.orig/src/appl/user_user/server.c	2008-03-21 22:47:40.302090616 -0700
+++ krb5-1.6.3/src/appl/user_user/server.c	2008-03-21 23:11:55.360085767 -0700
@@ -40,7 +40,7 @@
 
 #include "port-sockets.h"
 #include "krb5.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 /* fd 0 is a tcp socket used to talk to the client */
 
diff -Naur krb5-1.6.3.orig/src/clients/kdestroy/kdestroy.c krb5-1.6.3/src/clients/kdestroy/kdestroy.c
--- krb5-1.6.3.orig/src/clients/kdestroy/kdestroy.c	2008-03-21 22:47:40.298088482 -0700
+++ krb5-1.6.3/src/clients/kdestroy/kdestroy.c	2008-03-21 23:11:55.274085975 -0700
@@ -29,7 +29,7 @@
 
 #include "autoconf.h"
 #include <krb5.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <string.h>
 #include <stdio.h>
 #ifdef HAVE_UNISTD_H
diff -Naur krb5-1.6.3.orig/src/clients/kinit/kinit.c krb5-1.6.3/src/clients/kinit/kinit.c
--- krb5-1.6.3.orig/src/clients/kinit/kinit.c	2008-03-21 22:47:40.296089091 -0700
+++ krb5-1.6.3/src/clients/kinit/kinit.c	2008-03-21 23:11:55.248086071 -0700
@@ -39,7 +39,7 @@
 #include <stdio.h>
 #include <time.h>
 #include <errno.h>
-#include <com_err.h>
+#include <et/com_err.h>
 
 #ifdef GETOPT_LONG
 #include <getopt.h>
diff -Naur krb5-1.6.3.orig/src/clients/klist/klist.c krb5-1.6.3/src/clients/klist/klist.c
--- krb5-1.6.3.orig/src/clients/klist/klist.c	2008-03-21 22:47:40.297089345 -0700
+++ krb5-1.6.3/src/clients/klist/klist.c	2008-03-21 23:11:55.267085872 -0700
@@ -32,7 +32,7 @@
 #ifdef KRB5_KRB4_COMPAT
 #include <kerberosIV/krb.h>
 #endif
-#include <com_err.h>
+#include <et/com_err.h>
 #include <stdlib.h>
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
diff -Naur krb5-1.6.3.orig/src/clients/ksu/ksu.h krb5-1.6.3/src/clients/ksu/ksu.h
--- krb5-1.6.3.orig/src/clients/ksu/ksu.h	2008-03-21 22:47:40.298088482 -0700
+++ krb5-1.6.3/src/clients/ksu/ksu.h	2008-03-21 23:11:55.282085774 -0700
@@ -28,14 +28,14 @@
 #include "k5-int.h"
 #include "k5-util.h"
 #include <stdio.h>
-#include "com_err.h"
+#include "et/com_err.h"
 #include <sys/types.h> 
 #include <sys/param.h>
 #include <pwd.h>
 #include <unistd.h>
 #include <string.h>
 #include <syslog.h>
-/* <stdarg.h> or <varargs.h> is already included by com_err.h.  */
+/* <stdarg.h> or <varargs.h> is already included by et/com_err.h.  */
 
 #define NO_TARGET_FILE '.'
 #define SOURCE_USER_LOGIN "."
diff -Naur krb5-1.6.3.orig/src/clients/kvno/kvno.c krb5-1.6.3/src/clients/kvno/kvno.c
--- krb5-1.6.3.orig/src/clients/kvno/kvno.c	2008-03-21 22:47:40.297089345 -0700
+++ krb5-1.6.3/src/clients/kvno/kvno.c	2008-03-21 23:11:55.256085032 -0700
@@ -57,7 +57,7 @@
                         char *ccachestr, char *etypestr, char *keytab_name,
 			char *sname);
 
-#include <com_err.h>
+#include <et/com_err.h>
 static void extended_com_err_fn (const char *, errcode_t, const char *,
 				 va_list);
 
diff -Naur krb5-1.6.3.orig/src/include/k5-int.h krb5-1.6.3/src/include/k5-int.h
--- krb5-1.6.3.orig/src/include/k5-int.h	2008-03-21 22:47:40.396088530 -0700
+++ krb5-1.6.3/src/include/k5-int.h	2008-03-21 23:11:55.728085166 -0700
@@ -1188,7 +1188,7 @@
 	(krb5_context, krb5_pa_enc_ts *);
 
 /* #include "krb5/wordsize.h" -- comes in through base-defs.h. */
-#include "com_err.h"
+#include "et/com_err.h"
 #include "k5-plugin.h"
 
 struct _krb5_context {
diff -Naur krb5-1.6.3.orig/src/kadmin/dbutil/dump.c krb5-1.6.3/src/kadmin/dbutil/dump.c
--- krb5-1.6.3.orig/src/kadmin/dbutil/dump.c	2008-03-21 22:47:40.289089826 -0700
+++ krb5-1.6.3/src/kadmin/dbutil/dump.c	2008-03-21 23:11:55.159085238 -0700
@@ -32,7 +32,7 @@
 #include <kadm5/admin.h>
 #include <kadm5/server_internal.h>
 #include <kdb.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include "kdb5_util.h"
 #if defined(HAVE_REGEX_H) && defined(HAVE_REGCOMP)
 #include <regex.h>
diff -Naur krb5-1.6.3.orig/src/kadmin/dbutil/dumpv4.c krb5-1.6.3/src/kadmin/dbutil/dumpv4.c
--- krb5-1.6.3.orig/src/kadmin/dbutil/dumpv4.c	2008-03-21 22:47:40.288090130 -0700
+++ krb5-1.6.3/src/kadmin/dbutil/dumpv4.c	2008-03-21 23:11:55.134085587 -0700
@@ -56,7 +56,7 @@
 #ifdef KRB5_KRB4_COMPAT
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include <des.h>
 #include <krb.h>
diff -Naur krb5-1.6.3.orig/src/kadmin/dbutil/kdb5_destroy.c krb5-1.6.3/src/kadmin/dbutil/kdb5_destroy.c
--- krb5-1.6.3.orig/src/kadmin/dbutil/kdb5_destroy.c	2008-03-21 22:47:40.289089826 -0700
+++ krb5-1.6.3/src/kadmin/dbutil/kdb5_destroy.c	2008-03-21 23:11:55.141086249 -0700
@@ -31,7 +31,7 @@
 
 #include "k5-int.h"
 #include <stdio.h>
-#include "com_err.h"
+#include "et/com_err.h"
 #include <kadm5/admin.h>
 #include <kdb.h>
 #include "kdb5_util.h"
diff -Naur krb5-1.6.3.orig/src/kadmin/dbutil/kdb5_stash.c krb5-1.6.3/src/kadmin/dbutil/kdb5_stash.c
--- krb5-1.6.3.orig/src/kadmin/dbutil/kdb5_stash.c	2008-03-21 22:47:40.288090130 -0700
+++ krb5-1.6.3/src/kadmin/dbutil/kdb5_stash.c	2008-03-21 23:11:55.124086398 -0700
@@ -54,7 +54,7 @@
  */
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include <kadm5/admin.h>
 #include <stdio.h>
 #include "kdb5_util.h"
diff -Naur krb5-1.6.3.orig/src/kadmin/dbutil/loadv4.c krb5-1.6.3/src/kadmin/dbutil/loadv4.c
--- krb5-1.6.3.orig/src/kadmin/dbutil/loadv4.c	2008-03-21 22:47:40.288090130 -0700
+++ krb5-1.6.3/src/kadmin/dbutil/loadv4.c	2008-03-21 23:11:55.116086040 -0700
@@ -57,7 +57,7 @@
 #ifdef KRB5_KRB4_COMPAT
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include <des.h>
 #include <krb.h>
diff -Naur krb5-1.6.3.orig/src/kadmin/ktutil/ktutil.c krb5-1.6.3/src/kadmin/ktutil/ktutil.c
--- krb5-1.6.3.orig/src/kadmin/ktutil/ktutil.c	2008-03-21 22:47:40.285088809 -0700
+++ krb5-1.6.3/src/kadmin/ktutil/ktutil.c	2008-03-21 23:11:55.104086342 -0700
@@ -28,7 +28,7 @@
 
 #include "k5-int.h"
 #include "ktutil.h"
-#include <com_err.h>
+#include <et/com_err.h>
 #include "adm_proto.h"
 #include <ss/ss.h>
 #include <stdio.h>
diff -Naur krb5-1.6.3.orig/src/kadmin/testing/util/tcl_kadm5.c krb5-1.6.3/src/kadmin/testing/util/tcl_kadm5.c
--- krb5-1.6.3.orig/src/kadmin/testing/util/tcl_kadm5.c	2008-03-21 22:47:40.284089952 -0700
+++ krb5-1.6.3/src/kadmin/testing/util/tcl_kadm5.c	2008-03-21 23:11:55.079085295 -0700
@@ -8,7 +8,7 @@
 #endif
 #define USE_KADM5_API_VERSION 2
 #include <kadm5/admin.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <errno.h>
 #include <stdlib.h>
 #include <adb_err.h>
diff -Naur krb5-1.6.3.orig/src/kadmin/testing/util/tcl_ovsec_kadm.c krb5-1.6.3/src/kadmin/testing/util/tcl_ovsec_kadm.c
--- krb5-1.6.3.orig/src/kadmin/testing/util/tcl_ovsec_kadm.c	2008-03-21 22:47:40.284089952 -0700
+++ krb5-1.6.3/src/kadmin/testing/util/tcl_ovsec_kadm.c	2008-03-21 23:11:55.096084868 -0700
@@ -8,7 +8,7 @@
 #endif
 #define USE_KADM5_API_VERSION 1
 #include <kadm5/admin.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <errno.h>
 #include <stdlib.h>
 #include "tcl_kadm5.h"
diff -Naur krb5-1.6.3.orig/src/kdc/do_as_req.c krb5-1.6.3/src/kdc/do_as_req.c
--- krb5-1.6.3.orig/src/kdc/do_as_req.c	2008-03-21 22:47:40.299089015 -0700
+++ krb5-1.6.3/src/kdc/do_as_req.c	2008-03-21 23:11:55.291085548 -0700
@@ -28,7 +28,7 @@
  */
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include <syslog.h>
 #ifdef HAVE_NETINET_IN_H
diff -Naur krb5-1.6.3.orig/src/kdc/do_tgs_req.c krb5-1.6.3/src/kdc/do_tgs_req.c
--- krb5-1.6.3.orig/src/kdc/do_tgs_req.c	2008-03-21 22:47:40.300088432 -0700
+++ krb5-1.6.3/src/kdc/do_tgs_req.c	2008-03-21 23:11:55.315085223 -0700
@@ -28,7 +28,7 @@
  */
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include <syslog.h>
 #ifdef HAVE_NETINET_IN_H
diff -Naur krb5-1.6.3.orig/src/kdc/fakeka.c krb5-1.6.3/src/kdc/fakeka.c
--- krb5-1.6.3.orig/src/kdc/fakeka.c	2008-03-21 22:47:40.300088432 -0700
+++ krb5-1.6.3/src/kdc/fakeka.c	2008-03-21 23:11:55.353085943 -0700
@@ -48,7 +48,7 @@
 
 #include <krb5.h>
 #include <kadm5/admin.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <kerberosIV/krb.h>
 #include <kerberosIV/des.h>
 
diff -Naur krb5-1.6.3.orig/src/kdc/kerberos_v4.c krb5-1.6.3/src/kdc/kerberos_v4.c
--- krb5-1.6.3.orig/src/kdc/kerberos_v4.c	2008-03-21 22:47:40.300088432 -0700
+++ krb5-1.6.3/src/kdc/kerberos_v4.c	2008-03-21 23:11:55.328084617 -0700
@@ -140,7 +140,7 @@
 /*
  * v5 include files:
  */
-#include "com_err.h"
+#include "et/com_err.h"
 #include "extern.h"		/* to pick up master_princ */
 
 static krb5_data *response;
diff -Naur krb5-1.6.3.orig/src/kdc/main.c krb5-1.6.3/src/kdc/main.c
--- krb5-1.6.3.orig/src/kdc/main.c	2008-03-21 22:47:40.300088432 -0700
+++ krb5-1.6.3/src/kdc/main.c	2008-03-21 23:11:55.339085178 -0700
@@ -33,7 +33,7 @@
 #include <netdb.h>
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "adm.h"
 #include "adm_proto.h"
 #include "kdc_util.h"
diff -Naur krb5-1.6.3.orig/src/kdc/network.c krb5-1.6.3/src/kdc/network.c
--- krb5-1.6.3.orig/src/kdc/network.c	2008-03-21 22:47:40.299089015 -0700
+++ krb5-1.6.3/src/kdc/network.c	2008-03-21 23:11:55.304085500 -0700
@@ -27,7 +27,7 @@
  */
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "kdc_util.h"
 #include "extern.h"
 #include "kdc5_err.h"
diff -Naur krb5-1.6.3.orig/src/krb524/k524init.c krb5-1.6.3/src/krb524/k524init.c
--- krb5-1.6.3.orig/src/krb524/k524init.c	2008-03-21 22:47:40.183088862 -0700
+++ krb5-1.6.3/src/krb524/k524init.c	2008-03-21 23:11:54.849086055 -0700
@@ -22,7 +22,7 @@
 
 #include "autoconf.h"
 #include <krb5.h>
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include <stdio.h>
 #ifdef HAVE_UNISTD_H
diff -Naur krb5-1.6.3.orig/src/krb524/krb524d.c krb5-1.6.3/src/krb524/krb524d.c
--- krb5-1.6.3.orig/src/krb524/krb524d.c	2008-03-21 22:47:40.183088862 -0700
+++ krb5-1.6.3/src/krb524/krb524d.c	2008-03-21 23:11:54.841085418 -0700
@@ -44,7 +44,7 @@
 #include <k5-int.h>
 #include <kadm5/admin.h>
 #include <adm_proto.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <stdarg.h>
 
 #include <assert.h>
diff -Naur krb5-1.6.3.orig/src/krb524/test.c krb5-1.6.3/src/krb524/test.c
--- krb5-1.6.3.orig/src/krb524/test.c	2008-03-21 22:47:40.183088862 -0700
+++ krb5-1.6.3/src/krb524/test.c	2008-03-21 23:11:54.857085853 -0700
@@ -32,7 +32,7 @@
 
 #include <des.h>
 #include <krb.h>
-#include "com_err.h"
+#include "et/com_err.h"
 
 #define KEYSIZE 8
 #define CRED_BUFSIZ 2048
diff -Naur krb5-1.6.3.orig/src/lib/crypto/des/destest.c krb5-1.6.3/src/lib/crypto/des/destest.c
--- krb5-1.6.3.orig/src/lib/crypto/des/destest.c	2008-03-21 22:47:40.262089667 -0700
+++ krb5-1.6.3/src/lib/crypto/des/destest.c	2008-03-21 23:11:55.059085799 -0700
@@ -55,7 +55,7 @@
  */
 
 #include "des_int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include <stdio.h>
 
diff -Naur krb5-1.6.3.orig/src/lib/crypto/des/t_verify.c krb5-1.6.3/src/lib/crypto/des/t_verify.c
--- krb5-1.6.3.orig/src/lib/crypto/des/t_verify.c	2008-03-21 22:47:40.261090530 -0700
+++ krb5-1.6.3/src/lib/crypto/des/t_verify.c	2008-03-21 23:11:55.051085721 -0700
@@ -60,7 +60,7 @@
 #include "k5-int.h"
 #include "des_int.h"
 #include <stdio.h>
-#include "com_err.h"
+#include "et/com_err.h"
 
 static void do_encrypt(unsigned char *, unsigned char *);
 static void do_decrypt(unsigned char *, unsigned char *);
diff -Naur krb5-1.6.3.orig/src/lib/gssapi/generic/disp_com_err_status.c krb5-1.6.3/src/lib/gssapi/generic/disp_com_err_status.c
--- krb5-1.6.3.orig/src/lib/gssapi/generic/disp_com_err_status.c	2008-03-21 22:47:40.217091079 -0700
+++ krb5-1.6.3/src/lib/gssapi/generic/disp_com_err_status.c	2008-03-21 23:11:55.036086377 -0700
@@ -26,7 +26,7 @@
 
 #include "gssapiP_generic.h"
 #include "gss_libinit.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 /* XXXX internationalization!! */
 
diff -Naur krb5-1.6.3.orig/src/lib/gssapi/krb5/disp_status.c krb5-1.6.3/src/lib/gssapi/krb5/disp_status.c
--- krb5-1.6.3.orig/src/lib/gssapi/krb5/disp_status.c	2008-03-21 22:47:40.227089151 -0700
+++ krb5-1.6.3/src/lib/gssapi/krb5/disp_status.c	2008-03-21 23:11:55.043085922 -0700
@@ -22,7 +22,7 @@
 
 #include "gssapiP_krb5.h"
 #include "gss_libinit.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 /* XXXX internationalization!! */
 
diff -Naur krb5-1.6.3.orig/src/lib/kadm5/adb.h krb5-1.6.3/src/lib/kadm5/adb.h
--- krb5-1.6.3.orig/src/lib/kadm5/adb.h	2008-03-21 22:47:40.210090417 -0700
+++ krb5-1.6.3/src/lib/kadm5/adb.h	2008-03-21 23:11:54.960085776 -0700
@@ -17,7 +17,7 @@
 #include <db.h>
 #include <kadm5/admin.h>
 #include <kadm5/adb_err.h>
-#include <com_err.h>
+#include <et/com_err.h>
 
 typedef	long		osa_adb_ret_t;
 
diff -Naur krb5-1.6.3.orig/src/lib/kadm5/admin.h krb5-1.6.3/src/lib/kadm5/admin.h
--- krb5-1.6.3.orig/src/lib/kadm5/admin.h	2008-03-21 22:47:40.210090417 -0700
+++ krb5-1.6.3/src/lib/kadm5/admin.h	2008-03-21 23:11:54.982085502 -0700
@@ -41,7 +41,7 @@
 #include	<gssrpc/rpc.h>
 #include	<krb5.h>
 #include	<kdb.h>
-#include	<com_err.h>
+#include	<et/com_err.h>
 #include	<kadm5/kadm_err.h>
 #include	<kadm5/chpass_util_strings.h>
 
diff -Naur krb5-1.6.3.orig/src/lib/kadm5/clnt/client_init.c krb5-1.6.3/src/lib/kadm5/clnt/client_init.c
--- krb5-1.6.3.orig/src/lib/kadm5/clnt/client_init.c	2008-03-21 22:47:40.210090417 -0700
+++ krb5-1.6.3/src/lib/kadm5/clnt/client_init.c	2008-03-21 23:11:54.992085809 -0700
@@ -35,7 +35,7 @@
 #include <memory.h>
 #endif
 #include <string.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
diff -Naur krb5-1.6.3.orig/src/lib/kadm5/logger.c krb5-1.6.3/src/lib/kadm5/logger.c
--- krb5-1.6.3.orig/src/lib/kadm5/logger.c	2008-03-21 22:47:40.210090417 -0700
+++ krb5-1.6.3/src/lib/kadm5/logger.c	2008-03-21 23:11:54.971085220 -0700
@@ -33,7 +33,7 @@
  */
 #include "k5-int.h"
 #include "adm_proto.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include <stdio.h>
 #include <ctype.h>
 #ifdef	HAVE_SYSLOG_H
diff -Naur krb5-1.6.3.orig/src/lib/kadm5/srv/server_init.c krb5-1.6.3/src/lib/kadm5/srv/server_init.c
--- krb5-1.6.3.orig/src/lib/kadm5/srv/server_init.c	2008-03-21 22:47:40.211089274 -0700
+++ krb5-1.6.3/src/lib/kadm5/srv/server_init.c	2008-03-21 23:11:55.001085582 -0700
@@ -12,7 +12,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include "k5-int.h"		/* needed for gssapiP_krb5.h */
 #include <kadm5/admin.h>
 #include <krb5.h>
diff -Naur krb5-1.6.3.orig/src/lib/kadm5/unit-test/destroy-test.c krb5-1.6.3/src/lib/kadm5/unit-test/destroy-test.c
--- krb5-1.6.3.orig/src/lib/kadm5/unit-test/destroy-test.c	2008-03-21 22:47:40.215090012 -0700
+++ krb5-1.6.3/src/lib/kadm5/unit-test/destroy-test.c	2008-03-21 23:11:55.028085182 -0700
@@ -1,5 +1,5 @@
 #include <kadm5/admin.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <stdio.h>
 #include <krb5.h>
 #include <sys/socket.h>
diff -Naur krb5-1.6.3.orig/src/lib/kadm5/unit-test/handle-test.c krb5-1.6.3/src/lib/kadm5/unit-test/handle-test.c
--- krb5-1.6.3.orig/src/lib/kadm5/unit-test/handle-test.c	2008-03-21 22:47:40.215090012 -0700
+++ krb5-1.6.3/src/lib/kadm5/unit-test/handle-test.c	2008-03-21 23:11:55.015085788 -0700
@@ -1,5 +1,5 @@
 #include <kadm5/admin.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <stdio.h>
 #include <krb5.h>
 #include <sys/socket.h>
diff -Naur krb5-1.6.3.orig/src/lib/kadm5/unit-test/init-test.c krb5-1.6.3/src/lib/kadm5/unit-test/init-test.c
--- krb5-1.6.3.orig/src/lib/kadm5/unit-test/init-test.c	2008-03-21 22:47:40.214089199 -0700
+++ krb5-1.6.3/src/lib/kadm5/unit-test/init-test.c	2008-03-21 23:11:55.008086523 -0700
@@ -1,6 +1,6 @@
 #undef USE_KADM5_API_VERSION
 #include <kadm5/admin.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <krb5.h>
diff -Naur krb5-1.6.3.orig/src/lib/kadm5/unit-test/randkey-test.c krb5-1.6.3/src/lib/kadm5/unit-test/randkey-test.c
--- krb5-1.6.3.orig/src/lib/kadm5/unit-test/randkey-test.c	2008-03-21 22:47:40.215090012 -0700
+++ krb5-1.6.3/src/lib/kadm5/unit-test/randkey-test.c	2008-03-21 23:11:55.021085358 -0700
@@ -1,5 +1,5 @@
 #include <kadm5/admin.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <stdio.h>
 #include <krb5.h>
 #include <sys/socket.h>
diff -Naur krb5-1.6.3.orig/src/lib/krb5/ccache/cc_mslsa.c krb5-1.6.3/src/lib/krb5/ccache/cc_mslsa.c
--- krb5-1.6.3.orig/src/lib/krb5/ccache/cc_mslsa.c	2008-03-21 22:47:40.188089574 -0700
+++ krb5-1.6.3/src/lib/krb5/ccache/cc_mslsa.c	2008-03-21 23:11:54.886085124 -0700
@@ -54,7 +54,7 @@
 #define _UNICODE
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "cc-int.h"
 
 #include <stdio.h>
diff -Naur krb5-1.6.3.orig/src/lib/krb5/ccache/t_cc.c krb5-1.6.3/src/lib/krb5/ccache/t_cc.c
--- krb5-1.6.3.orig/src/lib/krb5/ccache/t_cc.c	2008-03-21 22:47:40.188089574 -0700
+++ krb5-1.6.3/src/lib/krb5/ccache/t_cc.c	2008-03-21 23:11:54.894085760 -0700
@@ -34,7 +34,7 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-#include "com_err.h"
+#include "et/com_err.h"
 
 #define KRB5_OK 0
 
diff -Naur krb5-1.6.3.orig/src/lib/krb5/krb/gic_pwd.c krb5-1.6.3/src/lib/krb5/krb/gic_pwd.c
--- krb5-1.6.3.orig/src/lib/krb5/krb/gic_pwd.c	2008-03-21 22:47:40.192089752 -0700
+++ krb5-1.6.3/src/lib/krb5/krb/gic_pwd.c	2008-03-21 23:11:54.912085028 -0700
@@ -1,5 +1,5 @@
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 static krb5_error_code
 krb5_get_as_key_password(
diff -Naur krb5-1.6.3.orig/src/lib/krb5/krb/recvauth.c krb5-1.6.3/src/lib/krb5/krb/recvauth.c
--- krb5-1.6.3.orig/src/lib/krb5/krb/recvauth.c	2008-03-21 22:47:40.194090261 -0700
+++ krb5-1.6.3/src/lib/krb5/krb/recvauth.c	2008-03-21 23:11:54.920085106 -0700
@@ -29,7 +29,7 @@
 
 #include "k5-int.h"
 #include "auth_con.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
diff -Naur krb5-1.6.3.orig/src/lib/krb5/krb/sendauth.c krb5-1.6.3/src/lib/krb5/krb/sendauth.c
--- krb5-1.6.3.orig/src/lib/krb5/krb/sendauth.c	2008-03-21 22:47:40.191089498 -0700
+++ krb5-1.6.3/src/lib/krb5/krb/sendauth.c	2008-03-21 23:11:54.902086118 -0700
@@ -29,7 +29,7 @@
 
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "auth_con.h"
 #include <errno.h>
 #include <stdio.h>
diff -Naur krb5-1.6.3.orig/src/lib/krb5/krb/t_kerb.c krb5-1.6.3/src/lib/krb5/krb/t_kerb.c
--- krb5-1.6.3.orig/src/lib/krb5/krb/t_kerb.c	2008-03-21 22:47:40.195089956 -0700
+++ krb5-1.6.3/src/lib/krb5/krb/t_kerb.c	2008-03-21 23:11:54.945086153 -0700
@@ -18,7 +18,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 
-#include "com_err.h"
+#include "et/com_err.h"
 
 void test_string_to_timestamp (krb5_context, char *);
 void test_425_conv_principal (krb5_context, char *, char*, char *);
diff -Naur krb5-1.6.3.orig/src/lib/krb5/krb/t_ser.c krb5-1.6.3/src/lib/krb5/krb/t_ser.c
--- krb5-1.6.3.orig/src/lib/krb5/krb/t_ser.c	2008-03-21 22:47:40.194090261 -0700
+++ krb5-1.6.3/src/lib/krb5/krb/t_ser.c	2008-03-21 23:11:54.930085413 -0700
@@ -29,7 +29,7 @@
  * t_ser.c - Test serialization.
  */
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "auth_con.h"
 
 #include <ctype.h>
diff -Naur krb5-1.6.3.orig/src/lib/krb5/krb/t_walk_rtree.c krb5-1.6.3/src/lib/krb5/krb/t_walk_rtree.c
--- krb5-1.6.3.orig/src/lib/krb5/krb/t_walk_rtree.c	2008-03-21 22:47:40.195089956 -0700
+++ krb5-1.6.3/src/lib/krb5/krb/t_walk_rtree.c	2008-03-21 23:11:54.938086050 -0700
@@ -4,7 +4,7 @@
 
 #include "k5-int.h"
 #include <stdio.h>
-#include "com_err.h"
+#include "et/com_err.h"
 
 int
 main(int argc, char **argv)
diff -Naur krb5-1.6.3.orig/src/lib/krb5/krb5_libinit.c krb5-1.6.3/src/lib/krb5/krb5_libinit.c
--- krb5-1.6.3.orig/src/lib/krb5/krb5_libinit.c	2008-03-21 22:47:40.187088761 -0700
+++ krb5-1.6.3/src/lib/krb5/krb5_libinit.c	2008-03-21 23:11:54.864086236 -0700
@@ -1,7 +1,7 @@
 #include <assert.h>
 
 #include "autoconf.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "k5-int.h"
 #include "krb5_err.h"
 #include "kv5m_err.h"
diff -Naur krb5-1.6.3.orig/src/lib/krb5/os/t_locate_kdc.c krb5-1.6.3/src/lib/krb5/os/t_locate_kdc.c
--- krb5-1.6.3.orig/src/lib/krb5/os/t_locate_kdc.c	2008-03-21 22:47:40.199088738 -0700
+++ krb5-1.6.3/src/lib/krb5/os/t_locate_kdc.c	2008-03-21 23:11:54.952085977 -0700
@@ -2,7 +2,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include "port-sockets.h"
-#include <com_err.h>
+#include <et/com_err.h>
 
 #define TEST
 #include "dnsglue.c"
diff -Naur krb5-1.6.3.orig/src/plugins/kdb/db2/policy_db.h krb5-1.6.3/src/plugins/kdb/db2/policy_db.h
--- krb5-1.6.3.orig/src/plugins/kdb/db2/policy_db.h	2008-03-21 22:47:40.407089091 -0700
+++ krb5-1.6.3/src/plugins/kdb/db2/policy_db.h	2008-03-21 23:11:55.736086362 -0700
@@ -31,7 +31,7 @@
 #include <gssrpc/xdr.h>
 #include <db.h>
 #include "adb_err.h"
-#include <com_err.h>
+#include <et/com_err.h>
 
 typedef	long		osa_adb_ret_t;
 
diff -Naur krb5-1.6.3.orig/src/slave/kprop.c krb5-1.6.3/src/slave/kprop.c
--- krb5-1.6.3.orig/src/slave/kprop.c	2008-03-21 22:47:40.183088862 -0700
+++ krb5-1.6.3/src/slave/kprop.c	2008-03-21 23:11:54.819084854 -0700
@@ -43,7 +43,7 @@
 #include <fcntl.h>
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "kprop.h"
 
 #ifndef GETSOCKNAME_ARG3_TYPE
diff -Naur krb5-1.6.3.orig/src/slave/kpropd.c krb5-1.6.3/src/slave/kpropd.c
--- krb5-1.6.3.orig/src/slave/kpropd.c	2008-03-21 22:47:40.183088862 -0700
+++ krb5-1.6.3/src/slave/kpropd.c	2008-03-21 23:11:54.831085390 -0700
@@ -77,7 +77,7 @@
 #include <syslog.h>
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include <errno.h>
 
 #include "kprop.h"
diff -Naur krb5-1.6.3.orig/src/tests/asn.1/krb5_decode_test.c krb5-1.6.3/src/tests/asn.1/krb5_decode_test.c
--- krb5-1.6.3.orig/src/tests/asn.1/krb5_decode_test.c	2008-03-21 22:47:40.295089395 -0700
+++ krb5-1.6.3/src/tests/asn.1/krb5_decode_test.c	2008-03-21 23:11:55.224085558 -0700
@@ -1,6 +1,6 @@
 #include "k5-int.h"
 #include "ktest.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "utility.h"
 #include "ktest_equal.h"
 
diff -Naur krb5-1.6.3.orig/src/tests/asn.1/krb5_encode_test.c krb5-1.6.3/src/tests/asn.1/krb5_encode_test.c
--- krb5-1.6.3.orig/src/tests/asn.1/krb5_encode_test.c	2008-03-21 22:47:40.295089395 -0700
+++ krb5-1.6.3/src/tests/asn.1/krb5_encode_test.c	2008-03-21 23:11:55.235085839 -0700
@@ -1,6 +1,6 @@
 #include <stdio.h>
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "utility.h"
 
 #include "ktest.h"
diff -Naur krb5-1.6.3.orig/src/tests/create/kdb5_mkdums.c krb5-1.6.3/src/tests/create/kdb5_mkdums.c
--- krb5-1.6.3.orig/src/tests/create/kdb5_mkdums.c	2008-03-21 22:47:40.294089421 -0700
+++ krb5-1.6.3/src/tests/create/kdb5_mkdums.c	2008-03-21 23:11:55.208085401 -0700
@@ -29,7 +29,7 @@
 
 #include "k5-int.h"
 #include "kdb.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include <ss/ss.h>
 #include <stdio.h>
 
diff -Naur krb5-1.6.3.orig/src/tests/dejagnu/t_inetd.c krb5-1.6.3/src/tests/dejagnu/t_inetd.c
--- krb5-1.6.3.orig/src/tests/dejagnu/t_inetd.c	2008-03-21 22:47:40.291089496 -0700
+++ krb5-1.6.3/src/tests/dejagnu/t_inetd.c	2008-03-21 23:11:55.167085874 -0700
@@ -53,7 +53,7 @@
 #include <unistd.h>
 #endif
 
-#include "com_err.h"
+#include "et/com_err.h"
 
 
 char *progname;
diff -Naur krb5-1.6.3.orig/src/tests/hammer/kdc5_hammer.c krb5-1.6.3/src/tests/hammer/kdc5_hammer.c
--- krb5-1.6.3.orig/src/tests/hammer/kdc5_hammer.c	2008-03-21 22:47:40.293089166 -0700
+++ krb5-1.6.3/src/tests/hammer/kdc5_hammer.c	2008-03-21 23:11:55.190086134 -0700
@@ -31,7 +31,7 @@
 #include <sys/time.h>
 
 #include "k5-int.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 #define KRB5_DEFAULT_OPTIONS 0
 #define KRB5_DEFAULT_LIFE 60*60*8 /* 8 hours */
diff -Naur krb5-1.6.3.orig/src/tests/threads/prof1.c krb5-1.6.3/src/tests/threads/prof1.c
--- krb5-1.6.3.orig/src/tests/threads/prof1.c	2008-03-21 22:47:40.293089166 -0700
+++ krb5-1.6.3/src/tests/threads/prof1.c	2008-03-21 23:11:55.181086081 -0700
@@ -7,7 +7,7 @@
 #include <time.h>
 #include <sys/time.h>
 #include <utime.h>
-#include <com_err.h>
+#include <et/com_err.h>
 #include <profile.h>
 
 int nthreads = 10;
diff -Naur krb5-1.6.3.orig/src/tests/threads/t_rcache.c krb5-1.6.3/src/tests/threads/t_rcache.c
--- krb5-1.6.3.orig/src/tests/threads/t_rcache.c	2008-03-21 22:47:40.293089166 -0700
+++ krb5-1.6.3/src/tests/threads/t_rcache.c	2008-03-21 23:11:55.174086536 -0700
@@ -29,7 +29,7 @@
 
 
 #include <stdio.h>
-#include <com_err.h>
+#include <et/com_err.h>
 
 #include "k5-int.h"
 #include <krb5.h>
diff -Naur krb5-1.6.3.orig/src/tests/verify/kdb5_verify.c krb5-1.6.3/src/tests/verify/kdb5_verify.c
--- krb5-1.6.3.orig/src/tests/verify/kdb5_verify.c	2008-03-21 22:47:40.293089166 -0700
+++ krb5-1.6.3/src/tests/verify/kdb5_verify.c	2008-03-21 23:11:55.199085907 -0700
@@ -29,7 +29,7 @@
 
 #include "k5-int.h"
 #include "kdb.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include <ss/ss.h>
 #include <stdio.h>
 
diff -Naur krb5-1.6.3.orig/src/util/et/com_err.3 krb5-1.6.3/src/util/et/com_err.3
--- krb5-1.6.3.orig/src/util/et/com_err.3	2008-03-21 22:47:40.177087616 -0700
+++ krb5-1.6.3/src/util/et/com_err.3	2008-03-21 23:11:54.752085703 -0700
@@ -8,7 +8,7 @@
 com_err \- common error display routine
 .SH SYNOPSIS
 .nf
- #include "com_err.h"
+ #include "et/com_err.h"
 .PP
 void com_err (whoami, code, format, ...);
 	const char *whoami;
@@ -77,7 +77,7 @@
 used.
 
 The
-.B com_err.h
+.B et/com_err.h
 header file should be included in any source file that uses routines
 from the
 .I com_err
diff -Naur krb5-1.6.3.orig/src/util/et/com_err.c krb5-1.6.3/src/util/et/com_err.c
--- krb5-1.6.3.orig/src/util/et/com_err.c	2008-03-21 22:47:40.176089317 -0700
+++ krb5-1.6.3/src/util/et/com_err.c	2008-03-21 23:11:54.668086140 -0700
@@ -23,7 +23,7 @@
 #include <string.h>
 #include <stdlib.h>
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "error_table.h"
 
 #if defined(_WIN32)
diff -Naur krb5-1.6.3.orig/src/util/et/com_err.texinfo krb5-1.6.3/src/util/et/com_err.texinfo
--- krb5-1.6.3.orig/src/util/et/com_err.texinfo	2008-03-21 22:47:40.177087616 -0700
+++ krb5-1.6.3/src/util/et/com_err.texinfo	2008-03-21 23:11:54.731084834 -0700
@@ -244,7 +244,7 @@
 @section Run-time support routines
 
 Any source file which uses the routines supplied with or produced by the
-com_err package should include the header file @file{<com_err.h>}.  It
+com_err package should include the header file @file{<et/com_err.h>}.  It
 contains declarations and definitions which may be needed on some systems.
 (Some functions cannot be referenced properly without the return type
 declarations in this file.  Some functions may work properly on most
@@ -501,7 +501,7 @@
 @t{profiled} should be created to hold objects compiled for profiling.
 Running ``make all'' should then be sufficient to build the library and
 error-table compiler.  The files @samp{libcom_err.a},
-@samp{libcom_err_p.a}, @samp{com_err.h}, and @samp{compile_et} should be
+@samp{libcom_err_p.a}, @samp{et/com_err.h}, and @samp{compile_et} should be
 installed for use; @samp{com_err.3} and @samp{compile_et.1} can also be
 installed as manual pages.
 
diff -Naur krb5-1.6.3.orig/src/util/et/error_message.c krb5-1.6.3/src/util/et/error_message.c
--- krb5-1.6.3.orig/src/util/et/error_message.c	2008-03-21 22:47:40.177087616 -0700
+++ krb5-1.6.3/src/util/et/error_message.c	2008-03-21 23:11:54.720085949 -0700
@@ -25,7 +25,7 @@
 #include <stdlib.h>
 #endif
 #include <string.h>
-#include "com_err.h"
+#include "et/com_err.h"
 #include "error_table.h"
 #include "k5-platform.h"
 
diff -Naur krb5-1.6.3.orig/src/util/et/error_table.y krb5-1.6.3/src/util/et/error_table.y
--- krb5-1.6.3.orig/src/util/et/error_table.y	2008-03-21 22:47:40.176089317 -0700
+++ krb5-1.6.3/src/util/et/error_table.y	2008-03-21 23:11:54.712086430 -0700
@@ -70,7 +70,7 @@
 #include <ctype.h>
 #include <sys/types.h>
 #include <sys/time.h>
-#include "com_err.h"
+#include "et/com_err.h"
 #include "internal.h"
 #include "error_table.h"
 #include "mit-sipb-copyright.h"
diff -Naur krb5-1.6.3.orig/src/util/et/et_c.awk krb5-1.6.3/src/util/et/et_c.awk
--- krb5-1.6.3.orig/src/util/et/et_c.awk	2008-03-21 22:47:40.176089317 -0700
+++ krb5-1.6.3/src/util/et/et_c.awk	2008-03-21 23:11:54.683085763 -0700
@@ -191,7 +191,7 @@
 	print "    0" > outfile
 	print "};" > outfile
 	print "" > outfile
-	print "#include <com_err.h>" > outfile
+	print "#include <et/com_err.h>" > outfile
 	print "" > outfile
 	if (tab_base_high == 0) {
 	    print "const struct error_table et_" table_name "_error_table = { text, " \
diff -Naur krb5-1.6.3.orig/src/util/et/et_c.pl krb5-1.6.3/src/util/et/et_c.pl
--- krb5-1.6.3.orig/src/util/et/et_c.pl	2008-03-21 22:47:40.177087616 -0700
+++ krb5-1.6.3/src/util/et/et_c.pl	2008-03-21 23:11:54.760085222 -0700
@@ -242,7 +242,7 @@
 &Pick('>', $outfile) &&
     (print $fh '');
 &Pick('>', $outfile) &&
-    (print $fh '#include <com_err.h>');
+    (print $fh '#include <et/com_err.h>');
 &Pick('>', $outfile) &&
     (print $fh '');
 if ($tab_base_high == 0) {
diff -Naur krb5-1.6.3.orig/src/util/et/et_h.awk krb5-1.6.3/src/util/et/et_h.awk
--- krb5-1.6.3.orig/src/util/et/et_h.awk	2008-03-21 22:47:40.176089317 -0700
+++ krb5-1.6.3/src/util/et/et_h.awk	2008-03-21 23:11:54.675085685 -0700
@@ -111,7 +111,7 @@
 	print " * This file is automatically generated; please do not edit it." > outfile
 	print " */" > outfile
 	print "" > outfile
-	print "#include <com_err.h>" > outfile
+	print "#include <et/com_err.h>" > outfile
 	print "" > outfile
 	table_item_count = 0
 }
diff -Naur krb5-1.6.3.orig/src/util/et/et_h.pl krb5-1.6.3/src/util/et/et_h.pl
--- krb5-1.6.3.orig/src/util/et/et_h.pl	2008-03-21 22:47:40.176089317 -0700
+++ krb5-1.6.3/src/util/et/et_h.pl	2008-03-21 23:11:54.697085690 -0700
@@ -137,7 +137,7 @@
 	&Pick('>', $outfile) &&
 	    (print $fh '');
 	&Pick('>', $outfile) &&
-	    (print $fh '#include <com_err.h>');
+	    (print $fh '#include <et/com_err.h>');
 	&Pick('>', $outfile) &&
 	    (print $fh '');
 	$table_item_count = 0;
diff -Naur krb5-1.6.3.orig/src/util/et/et_name.c krb5-1.6.3/src/util/et/et_name.c
--- krb5-1.6.3.orig/src/util/et/et_name.c	2008-03-21 22:47:40.177087616 -0700
+++ krb5-1.6.3/src/util/et/et_name.c	2008-03-21 23:11:54.738085775 -0700
@@ -19,7 +19,7 @@
  * provided "as is" without express or implied warranty.
  */
 
-#include "com_err.h"
+#include "et/com_err.h"
 #include "error_table.h"
 
 static const char char_set[] =
diff -Naur krb5-1.6.3.orig/src/util/et/init_et.c krb5-1.6.3/src/util/et/init_et.c
--- krb5-1.6.3.orig/src/util/et/init_et.c	2008-03-21 22:47:40.176089317 -0700
+++ krb5-1.6.3/src/util/et/init_et.c	2008-03-21 23:11:54.705085768 -0700
@@ -21,7 +21,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include "com_err.h"
+#include "et/com_err.h"
 #include "error_table.h"
 
 #if 0
diff -Naur krb5-1.6.3.orig/src/util/et/t_com_err.c krb5-1.6.3/src/util/et/t_com_err.c
--- krb5-1.6.3.orig/src/util/et/t_com_err.c	2008-03-21 22:47:40.176089317 -0700
+++ krb5-1.6.3/src/util/et/t_com_err.c	2008-03-21 23:11:54.690085028 -0700
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include "com_err.h"
+#include "et/com_err.h"
 #include "et1.h"
 #include "et2.h"
 
diff -Naur krb5-1.6.3.orig/src/util/et/test_et.c krb5-1.6.3/src/util/et/test_et.c
--- krb5-1.6.3.orig/src/util/et/test_et.c	2008-03-21 22:47:40.177087616 -0700
+++ krb5-1.6.3/src/util/et/test_et.c	2008-03-21 23:11:54.745085879 -0700
@@ -1,6 +1,6 @@
 #include <stdio.h>
 #include <errno.h>
-#include "com_err.h"
+#include "et/com_err.h"
 #include "test1.h"
 #include "test2.h"
 #include "autoconf.h"
diff -Naur krb5-1.6.3.orig/src/util/profile/profile.swg krb5-1.6.3/src/util/profile/profile.swg
--- krb5-1.6.3.orig/src/util/profile/profile.swg	2008-03-21 22:47:40.178088987 -0700
+++ krb5-1.6.3/src/util/profile/profile.swg	2008-03-21 23:11:54.776085378 -0700
@@ -26,7 +26,7 @@
  * Input for wrapper generator program SWIG for profile routines.
  */
 #include <errno.h>
-#include "com_err.h"
+#include "et/com_err.h"
 #include "profile.h"
 
 #ifdef SWIGTCL
diff -Naur krb5-1.6.3.orig/src/util/profile/profile_tcl.c krb5-1.6.3/src/util/profile/profile_tcl.c
--- krb5-1.6.3.orig/src/util/profile/profile_tcl.c	2008-03-21 22:47:40.179088124 -0700
+++ krb5-1.6.3/src/util/profile/profile_tcl.c	2008-03-21 23:11:54.794085763 -0700
@@ -1096,7 +1096,7 @@
  * Input for wrapper generator program SWIG for profile routines.
  */
 #include <errno.h>
-#include "com_err.h"
+#include "et/com_err.h"
 #include "profile.h"
 
 #ifdef SWIGTCL
diff -Naur krb5-1.6.3.orig/src/util/profile/prof_int.h krb5-1.6.3/src/util/profile/prof_int.h
--- krb5-1.6.3.orig/src/util/profile/prof_int.h	2008-03-21 22:47:40.178088987 -0700
+++ krb5-1.6.3/src/util/profile/prof_int.h	2008-03-21 23:11:54.768085300 -0700
@@ -12,7 +12,7 @@
 
 #include "k5-thread.h"
 #include "k5-platform.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "profile.h"
 
 typedef long prf_magic_t;
diff -Naur krb5-1.6.3.orig/src/util/profile/test_profile.c krb5-1.6.3/src/util/profile/test_profile.c
--- krb5-1.6.3.orig/src/util/profile/test_profile.c	2008-03-21 22:47:40.179088124 -0700
+++ krb5-1.6.3/src/util/profile/test_profile.c	2008-03-21 23:11:54.802086399 -0700
@@ -11,7 +11,7 @@
 #endif
 
 #include "argv_parse.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 const char *program_name = "test_profile";
 
diff -Naur krb5-1.6.3.orig/src/util/ss/error.c krb5-1.6.3/src/util/ss/error.c
--- krb5-1.6.3.orig/src/util/ss/error.c	2008-03-21 22:47:40.180089496 -0700
+++ krb5-1.6.3/src/util/ss/error.c	2008-03-21 23:11:54.808085131 -0700
@@ -8,7 +8,7 @@
 #include <stdio.h>
 
 #include "copyright.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #include "ss_internal.h"
 
 char * ss_name(sci_idx)
diff -Naur krb5-1.6.3.orig/src/windows/cns/cns.h krb5-1.6.3/src/windows/cns/cns.h
--- krb5-1.6.3.orig/src/windows/cns/cns.h	2008-03-21 22:47:40.386090737 -0700
+++ krb5-1.6.3/src/windows/cns/cns.h	2008-03-21 23:11:55.679086399 -0700
@@ -25,7 +25,7 @@
 #include "winsock.h"
 #include "krb5.h"
 #include "krbini.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 #define DEFAULT_TKT_LIFE    120             /* In 5 minute units */
 #define ANAME_SZ	        40
diff -Naur krb5-1.6.3.orig/src/windows/cns/kpasswd.c krb5-1.6.3/src/windows/cns/kpasswd.c
--- krb5-1.6.3.orig/src/windows/cns/kpasswd.c	2008-03-21 22:47:40.386090737 -0700
+++ krb5-1.6.3/src/windows/cns/kpasswd.c	2008-03-21 23:11:55.685085410 -0700
@@ -9,7 +9,7 @@
 #include <string.h>
 
 #include "krb5.h"
-#include "com_err.h"
+#include "et/com_err.h"
 
 #include "cns.h"
 
diff -Naur krb5-1.6.3.orig/src/windows/cns/tktlist.c krb5-1.6.3/src/windows/cns/tktlist.c
--- krb5-1.6.3.orig/src/windows/cns/tktlist.c	2008-03-21 22:47:40.385090203 -0700
+++ krb5-1.6.3/src/windows/cns/tktlist.c	2008-03-21 23:11:55.671085204 -0700
@@ -30,7 +30,7 @@
 #ifdef KRB5
 #include "winsock.h"
 #include "krb5.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #endif
 
 #include "cns.h"
diff -Naur krb5-1.6.3.orig/src/windows/identity/plugins/krb4/errorfuncs.h krb5-1.6.3/src/windows/identity/plugins/krb4/errorfuncs.h
--- krb5-1.6.3.orig/src/windows/identity/plugins/krb4/errorfuncs.h	2008-03-21 22:47:40.381090583 -0700
+++ krb5-1.6.3/src/windows/identity/plugins/krb4/errorfuncs.h	2008-03-21 23:11:55.663086522 -0700
@@ -31,9 +31,9 @@
    and AFS plugins */
 
 #include <errno.h>
-#include <com_err.h>
+#include <et/com_err.h>
 /*
- * This is a hack needed because the real com_err.h does
+ * This is a hack needed because the real et/com_err.h does
  * not define err_func.  We need it in the case where
  * we pull in the real com_err instead of the krb4 
  * impostor.
diff -Naur krb5-1.6.3.orig/src/windows/identity/plugins/krb5/errorfuncs.h krb5-1.6.3/src/windows/identity/plugins/krb5/errorfuncs.h
--- krb5-1.6.3.orig/src/windows/identity/plugins/krb5/errorfuncs.h	2008-03-21 22:47:40.378089262 -0700
+++ krb5-1.6.3/src/windows/identity/plugins/krb5/errorfuncs.h	2008-03-21 23:11:55.656085023 -0700
@@ -31,9 +31,9 @@
    and AFS plugins */
 
 #include <errno.h>
-#include <com_err.h>
+#include <et/com_err.h>
 /*
- * This is a hack needed because the real com_err.h does
+ * This is a hack needed because the real et/com_err.h does
  * not define err_func.  We need it in the case where
  * we pull in the real com_err instead of the krb4 
  * impostor.
diff -Naur krb5-1.6.3.orig/src/windows/wintel/auth.c krb5-1.6.3/src/windows/wintel/auth.c
--- krb5-1.6.3.orig/src/windows/wintel/auth.c	2008-03-21 22:47:40.392090027 -0700
+++ krb5-1.6.3/src/windows/wintel/auth.c	2008-03-21 23:11:55.696085133 -0700
@@ -13,7 +13,7 @@
 #include <time.h>
 #include <string.h>
 #include "krb5.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #endif
 
 #include "telnet.h"
diff -Naur krb5-1.6.3.orig/src/windows/wintel/encrypt.c krb5-1.6.3/src/windows/wintel/encrypt.c
--- krb5-1.6.3.orig/src/windows/wintel/encrypt.c	2008-03-21 22:47:40.392090027 -0700
+++ krb5-1.6.3/src/windows/wintel/encrypt.c	2008-03-21 23:11:55.707085136 -0700
@@ -73,7 +73,7 @@
 #include <time.h>
 #include <string.h>
 #include "krb5.h"
-#include "com_err.h"
+#include "et/com_err.h"
 #endif
 
 #include "telnet.h"
