--- a/sysdeps/unix/sysv/linux/configure.in	2007-09-13 17:56:51.519950207 -0700
+++ b/sysdeps/unix/sysv/linux/configure.in	2007-09-13 17:40:35.186895935 -0700
@@ -24,6 +24,7 @@
   CPPFLAGS="$CPPFLAGS $SYSINCLUDES"
 fi
 define([libc_cv_linuxVER], [libc_cv_linux]patsubst(LIBC_LINUX_VERSION,[\.]))dnl
+as_test_x="test -x"
 AC_CACHE_CHECK(installed Linux kernel header files, libc_cv_linuxVER, [dnl
 AC_EGREP_CPP([eat flaming death], [#include <linux/version.h>
 #if !defined LINUX_VERSION_CODE || LINUX_VERSION_CODE < ]dnl
