--- configure.orig	2009-10-19 12:57:29.000000000 +0200
+++ configure	2009-10-19 13:01:02.000000000 +0200
@@ -4841,6 +4841,8 @@
     '') ac_prog_version="v. ?.??, bad"; ac_verc_fail=yes;;
     2.1[3-9]*)
        ac_prog_version="$ac_prog_version, ok"; ac_verc_fail=no;;
+    2.2[0-9]*)
+       ac_prog_version="$ac_prog_version, ok"; ac_verc_fail=no;;
     *) ac_prog_version="$ac_prog_version, bad"; ac_verc_fail=yes;;
 
   esac
@@ -4904,6 +4906,8 @@
     '') ac_prog_version="v. ?.??, bad"; ac_verc_fail=yes;;
     2.1[3-9]*)
        ac_prog_version="$ac_prog_version, ok"; ac_verc_fail=no;;
+    2.2[0-9]*)
+       ac_prog_version="$ac_prog_version, ok"; ac_verc_fail=no;;
     *) ac_prog_version="$ac_prog_version, bad"; ac_verc_fail=yes;;
 
   esac
