diff -urpN linux-libc-headers-2.6.8.1.orig/include/linux/config.h linux-libc-headers-2.6.8.1/include/linux/config.h
--- linux-libc-headers-2.6.8.1.orig/include/linux/config.h	2004-11-29 21:24:16.893466608 -0800
+++ linux-libc-headers-2.6.8.1/include/linux/config.h	2004-11-29 21:24:46.574954336 -0800
@@ -1,3 +1,3 @@
-#error "Compilation aborted. Please read the FAQ for linux-libc-headers package."
-#error "(can be found at http://ep09.pld-linux.org/~mmazur/linux-libc-headers/doc/)"
+#warning "Compilation would have been aborted. Please read the FAQ for linux-libc-headers package."
+#warning "(can be found at http://ep09.pld-linux.org/~mmazur/linux-libc-headers/doc/)"
 
