--- csu/Makefile.old	2005-06-05 17:15:06.000000000 -0700
+++ csu/Makefile	2005-06-05 17:15:54.000000000 -0700
@@ -241,7 +241,7 @@
 	 esac; \
 	 files="$(all-Banner-files)";				\
 	 if test -n "$$files"; then				\
-	   echo "\"Available extensions:\\n\"";			\
+	   printf '"Available extensions:\\n"\n';		\
 	   sed -e '/^#/d' -e 's/^[[:space:]]*/	/'		\
 	       -e 's/^\(.*\)$$/\"\1\\n\"/' $$files;		\
 	 fi) > $@T
