diff --git a/inet/protocols/timed.h b/inet/protocols/timed.h
index 4345bed..b5d4702 100644
--- a/inet/protocols/timed.h
+++ b/inet/protocols/timed.h
@@ -32,7 +32,8 @@
 #ifndef	_PROTOCOLS_TIMED_H
 #define	_PROTOCOLS_TIMED_H 1
 
-#include <rpc/types.h>
+#include <sys/types.h>
+#include <sys/time.h>
 
 /*
  * Time Synchronization Protocol
@@ -40,6 +41,7 @@
 
 #define	TSPVERSION	1
 #define ANYADDR 	NULL
+#define MAXHOSTNAMELEN	64
 
 struct tsp {
 	u_char	tsp_type;
diff --git a/nis/Makefile b/nis/Makefile
index f63ada6..e7e5f0c 100644
--- a/nis/Makefile
+++ b/nis/Makefile
@@ -23,9 +23,9 @@ subdir	:= nis
 
 aux			:= nis_hash
 
-headers			:= $(wildcard rpcsvc/*.[h])
 distribute		:= nss-nis.h nss-nisplus.h nis_intern.h Banner \
-			   nisplus-parser.h nis_xdr.h nss
+			   nisplus-parser.h nis_xdr.h nss \
+			   $(wildcard rpcsvc/*.[hx])
 
 # These are the databases available for the nis (and perhaps later nisplus)
 # service.  This must be a superset of the services in nss.
