diff -Naur glibc-2.3.5.orig/sysdeps/unix/sysv/linux/i386/sysdep.h glibc-2.3.5/sysdeps/unix/sysv/linux/i386/sysdep.h
--- glibc-2.3.5.orig/sysdeps/unix/sysv/linux/i386/sysdep.h	2005-04-11 11:06:13.530536216 -0700
+++ glibc-2.3.5/sysdeps/unix/sysv/linux/i386/sysdep.h	2005-04-11 11:08:00.636253672 -0700
@@ -454,7 +454,7 @@
 
 #define LOADARGS_0
 #ifdef __PIC__
-# if defined I386_USE_SYSENTER
+# if defined I386_USE_SYSENTER && defined SHARED
 #  define LOADARGS_1 \
     "bpushl .L__X'%k3, %k3\n\t"
 #  define LOADARGS_5 \
