diff -purN web2ldap-0.15.1/fcgi/web2ldap.py web2ldap/fcgi/web2ldap.py
--- web2ldap-0.15.1/fcgi/web2ldap.py	2004-07-07 08:06:11.000000000 +0000
+++ web2ldap/fcgi/web2ldap.py	2004-08-14 16:42:03.000000000 +0000
@@ -20,7 +20,7 @@ except ImportError:
 else:
   psyco.full()
 
-import sys,os,signal,time,fcgi,threading
+import sys,os,signal,time,threading
 
 exec_startdir = os.path.dirname(os.path.dirname(os.path.abspath(sys.argv[0])))
 sys.path.insert(0,os.sep.join([exec_startdir,'etc','web2ldap']))
@@ -31,7 +31,7 @@ if os.name == 'posix':
   # For finding web2ldapcnf.py in /etc/web2ldap on Unix systems
   sys.path.append('/etc/web2ldap')
 
-import mssignals,web2ldapcnf.misc,web2ldapcnf.fastcgi,web2ldapcnf.plugins
+import fcgi,mssignals,web2ldapcnf.misc,web2ldapcnf.fastcgi,web2ldapcnf.plugins
 
 open(web2ldapcnf.fastcgi.pid_file,'w').write(str(os.getpid()))
 
