--- gscanbus.c.orig	2003-07-06 13:31:10.000000000 +0100
+++ gscanbus.c	2003-07-06 13:35:19.000000000 +0100
@@ -109,14 +109,9 @@
 
 static GdkPixmap *pixmap = NULL;
 
-const char not_compatible[] = "\
-This libraw1394 does not work with your version of Linux. You need a different
-version that matches your kernel (see kernel help text for the raw1394 option to
-find out which is the correct version).\n";
-
-const char not_loaded[] = "\
-This probably means that you don't have raw1394 support in the kernel or that
-you haven't loaded the raw1394 module.\n";
+const char not_compatible[] = "\nThis libraw1394 does not work with your version of Linux. You need a different\nversion that matches your kernel (see kernel help text for the raw1394 option to\nfind out which is the correct version).\n";
+
+const char not_loaded[] = "\nThis probably means that you don't have raw1394 support in the kernel or that\nyou haven't loaded the raw1394 module.\n";
 
 /*---------------------------------------------------------------------------
  * Drawing routines
