--- daemon/UPnP.h.orig	2016-10-17 00:37:40.000000000 -0400
+++ daemon/UPnP.h	2016-10-19 20:56:18.628270468 -0400
@@ -60,7 +60,7 @@
 		std::mutex m_StartedMutex;
 		boost::asio::io_service m_Service;
 		boost::asio::deadline_timer m_Timer;
-        bool m_upnpUrlsInitialized=false;
+        bool m_upnpUrlsInitialized;
 		struct UPNPUrls m_upnpUrls;
 		struct IGDdatas m_upnpData;
 
@@ -65,7 +65,7 @@
 		struct IGDdatas m_upnpData;
 
 		// For miniupnpc
-		struct UPNPDev * m_Devlist = 0;
+		struct UPNPDev * m_Devlist;
 		char m_NetworkAddr[64];
 		char m_externalIPAddress[40];
 	};
--- daemon/UPnP.cpp.orig	2016-10-17 00:37:40.000000000 -0400
+++ daemon/UPnP.cpp	2016-10-19 21:02:23.135699101 -0400
@@ -22,7 +22,7 @@
 {
 namespace transport
 {
-	UPnP::UPnP () : m_IsRunning(false), m_Thread (nullptr), m_Timer (m_Service)
+	UPnP::UPnP () : m_IsRunning(false), m_Thread (nullptr), m_Timer (m_Service), m_Devlist(0), m_upnpUrlsInitialized(false)
 	{
 	}
 
