--- libgda-1.99.1/providers/firebird/gda-firebird-recordset.c	2006-10-12 10:10:58.000000000 +0200
+++ gda-firebird-recordset.c	2006-11-01 17:06:49.000000000 +0100
@@ -432,7 +432,11 @@
 	GdaBlob *blob = NULL;
 	struct tm *a_date_time;
 	gpointer tmp_big, field_data;
+#if FB_API_VER < 20
 	struct vary *var_text;
+#else
+	PARAMVARY *var_text;
+#endif
 	gdouble i64_n;
 	XSQLDA *sql_result;
 	
@@ -509,7 +513,11 @@
 			break;
 		case SQL_VARYING:
 		case SQL_VARYING+1:
+#if FB_API_VER < 20
 			var_text = (struct vary *) field_data;
+#else
+			var_text = (PARAMVARY *) field_data;
+#endif
 			var_text->vary_string[var_text->vary_length] = '\0';
 
 			g_value_set_string (gda_value, (const gchar *) var_text->vary_string);
