diff -u -r libgda-orig/Makefile.am libgda-1.9.103/Makefile.am
--- libgda-orig/Makefile.am	2006-10-12 18:27:45.000000000 -0500
+++ libgda-1.9.103/Makefile.am	2006-10-12 18:29:22.000000000 -0500
@@ -7,7 +7,6 @@
 	providers \
 	report \
 	tools \
-	gda-sharp \
 	testing \
 	data \
 	doc
diff -u -r libgda-orig/Makefile.in libgda-1.9.103/Makefile.in
--- libgda-orig/Makefile.in	2006-10-12 18:27:45.000000000 -0500
+++ libgda-1.9.103/Makefile.in	2006-10-12 18:30:25.000000000 -0500
@@ -347,7 +347,6 @@
 	providers \
 	report \
 	tools \
-	gda-sharp \
 	testing \
 	data \
 	doc
