--- lib/fribidi-types.h.orig
+++ lib/fribidi-types.h
@@ -40,8 +40,7 @@
 
 #if FRIBIDI_USE_GLIB+0
 # ifndef __FRIBIDI_DOC
-#  include <glib/gtypes.h>
-#  include <glib/gunicode.h>
+#  include <glib.h>
 # endif	/* !__FRIBIDI_DOC */
 # define FRIBIDI_INT8_LOCAL		gint8
 # define FRIBIDI_INT16_LOCAL		gint16
--- lib/common.h.orig	2011-11-25 16:25:05.609799032 -0500
+++ lib/common.h	2011-11-25 16:29:52.262106021 -0500
@@ -57,21 +57,15 @@
 #  define SIZEOF_VOID_P GLIB_SIZEOF_VOID_P
 # endif	/* !SIZEOF_VOID_P */
 # ifndef __FRIBIDI_DOC
-#  include <glib/gmem.h>
+#  include <glib.h>
 # endif	/* !__FRIBIDI_DOC */
 # ifndef fribidi_malloc
 #  define fribidi_malloc g_try_malloc
 #  define fribidi_free g_free
 # endif	/* !fribidi_malloc */
 # ifndef fribidi_assert
-#  ifndef __FRIBIDI_DOC
-#   include <glib/gmessages.h>
-#  endif /* !__FRIBIDI_DOC */
 #  define fribidi_assert g_assert
 # endif	/* !fribidi_assert */
-# ifndef __FRIBIDI_DOC
-#  include <glib/gmacros.h>
-# endif	/* !__FRIBIDI_DOC */
 # ifndef FRIBIDI_BEGIN_STMT
 #  define FRIBIDI_BEGIN_STMT G_STMT_START {
 #  define FRIBIDI_END_STMT } G_STMT_END
--- lib/fribidi-common.h.orig	2011-11-25 16:03:33.664620104 -0500
+++ lib/fribidi-common.h	2011-11-25 16:23:06.487671454 -0500
@@ -62,7 +62,7 @@
 
 #if FRIBIDI_USE_GLIB+0
 # ifndef __FRIBIDI_DOC
-#  include <glib/gmacros.h>
+#  include <glib.h>
 # endif	/* !__FRIBIDI_DOC */
 # define FRIBIDI_BEGIN_DECLS		G_BEGIN_DECLS
 # define FRIBIDI_END_DECLS		G_END_DECLS
--- lib/mem.h.orig	2011-11-25 16:34:53.162576464 -0500
+++ lib/mem.h	2011-11-25 16:36:38.266690592 -0500
@@ -42,7 +42,7 @@
 #if FRIBIDI_USE_GLIB+0
 
 #ifndef __FRIBIDI_DOC
-# include <glib/gmem.h>
+# include <glib.h>
 #endif /* !__FRIBIDI_DOC */
 
 #define FriBidiMemChunk GMemChunk
--- charset/fribidi-char-sets.c.orig	2011-11-25 16:40:17.758928935 -0500
+++ charset/fribidi-char-sets.c	2011-11-25 16:47:16.620383771 -0500
@@ -114,7 +114,7 @@
 };
 
 #if FRIBIDI_USE_GLIB+0
-# include <glib/gstrfuncs.h>
+# include <glib.h>
 # define fribidi_strcasecmp g_ascii_strcasecmp
 #else /* !FRIBIDI_USE_GLIB */
 static char
