--- gmp3info.c.old	2001-07-16 02:17:16.000000000 -0700
+++ gmp3info.c	2005-02-19 00:42:21.881300184 -0800
@@ -484,12 +484,7 @@
 }
 
 void about_mp3info(void) {
-	quick_popup(VERSION,"MP3Info
-			An ID3 Tag Editor\n\n
-			Original author: Ricardo Cerqueira <rmc@plug.pt>
-			Current maintainer: Cedric Tefft <cedric@earthling.net>
-			GTK Version: Ricardo Cerqueira <rmc@plug.pt>\n\n
-			Ricardo Cerqueira, Cedric Tefft   1999-2001\n\n");
+	quick_popup(VERSION,"MP3Info An ID3 Tag Editor\n\nOriginal author: Ricardo Cerqueira <rmc@plug.pt>Current maintainer: Cedric Tefft <cedric@earthling.net>GTK Version: Ricardo Cerqueira <rmc@plug.pt>\n\nRicardo Cerqueira, Cedric Tefft   1999-2001\n\n");
 }
 
 /* rmcc has left the building */
