diff -ru Miro-2.5.2/portable/libtorrent/include/libtorrent/storage.hpp Miro-2.5.2-patched/portable/libtorrent/include/libtorrent/storage.hpp
--- Miro-2.5.2/portable/libtorrent/include/libtorrent/storage.hpp	2009-07-31 11:56:04.000000000 +0900
+++ Miro-2.5.2-patched/portable/libtorrent/include/libtorrent/storage.hpp	2009-10-02 22:10:11.869173280 +0900
@@ -42,6 +42,7 @@
 
 #include <boost/limits.hpp>
 #include <boost/thread.hpp>
+#include <boost/function.hpp>
 #include <boost/shared_ptr.hpp>
 #include <boost/intrusive_ptr.hpp>
 #include <boost/filesystem/path.hpp>
