diff -Nur Miro-2.5.2-orig/platform/gtk-x11/run.sh Miro-2.5.2/platform/gtk-x11/run.sh
--- Miro-2.5.2-orig/platform/gtk-x11/run.sh	2009-07-31 04:56:01.000000000 +0200
+++ Miro-2.5.2/platform/gtk-x11/run.sh	2009-08-04 02:03:53.000000000 +0200
@@ -48,14 +48,6 @@
 MIROPYTHONPATH=$MIROPYTHONPATH:dist/$PREFIX/lib64/python$PYTHON_VERSION/site-packages/
 MIROPYTHONPATH=$MIROPYTHONPATH:$PYTHONPATH
 
-$PYTHON setup.py install --root=./dist --prefix=$PREFIX
+$PYTHON setup.py install --root=/ --prefix=$PREFIX
 
 echo ""
-
-if [ $? = 0 ]
-then
-    echo "Compile successful.  Running...."
-    PATH=dist/$PREFIX/bin:$PATH PYTHONPATH=$MIROPYTHONPATH dist/$PREFIX/bin/miro "$@"
-else
-    echo "Compile failed.  Review the console text and make necessary changes."
-fi
