--- setup.sh.old	2005-06-17 15:12:07.000000000 -0700
+++ setup.sh	2005-06-17 15:15:09.000000000 -0700
@@ -396,9 +396,7 @@
            --menu "$MAIN_MENU_TEXT" 13 74 5 \
              1 "Driver configuration ($SELECTED_DRIVER)" \
              2 "Software configuration" \
-             3 "Save configuration & run configure" \
-             4 "Save configuration & exit" \
-             5 "Exit WITHOUT doing anything" 2>$TEMP
+             3 "Save configuration" 2>$TEMP
 
     if test "$?" != "0"; then
         {
@@ -413,7 +411,6 @@
         elif test "$1" = "3"; then
             {
             SaveConfig
-            CONFIGURE="yes"
             }
         elif test "$1" = "4"; then SaveConfig
         elif test "$1" = "5"; then
