From 073159da041bae4b62ed2a1a23be52a65a033999 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 18 Mar 2017 11:12:46 +0100
Subject: [PATCH] fix Makefile

---
 Makefile | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index 26518ec..0bbaaa8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,4 @@
-CC = gcc
-CXX = g++
-
-CFLAGS = -W -Wall -Wextra -ansi -pedantic -lm -O2
-CXXFLAGS = -W -Wall -Wextra -ansi -pedantic -O2
+LDLIBS := -lm
 
 ZOPFLILIB_SRC = src/zopfli/blocksplitter.c src/zopfli/cache.c\
                 src/zopfli/deflate.c src/zopfli/gzip_container.c\
@@ -16,11 +12,11 @@ LODEPNG_SRC := src/zopflipng/lodepng/lodepng.cpp src/zopflipng/lodepng/lodepng_u
 ZOPFLIPNGLIB_SRC := src/zopflipng/zopflipng_lib.cc
 ZOPFLIPNGBIN_SRC := src/zopflipng/zopflipng_bin.cc
 
-.PHONY: zopfli zopflipng
+all: zopfli zopflipng libzopfli libzopflipng
 
 # Zopfli binary
 zopfli:
-	$(CC) $(ZOPFLILIB_SRC) $(ZOPFLIBIN_SRC) $(CFLAGS) -o zopfli
+	$(CC) $(ZOPFLILIB_SRC) $(ZOPFLIBIN_SRC) $(CFLAGS) -o zopfli $(LDLIBS)
 
 # Zopfli shared library
 libzopfli:
@@ -30,7 +26,7 @@ libzopfli:
 # ZopfliPNG binary
 zopflipng:
 	$(CC) $(ZOPFLILIB_SRC) $(CFLAGS) -c
-	$(CXX) $(ZOPFLILIB_OBJ) $(LODEPNG_SRC) $(ZOPFLIPNGLIB_SRC) $(ZOPFLIPNGBIN_SRC) $(CFLAGS) -o zopflipng
+	$(CXX) $(ZOPFLILIB_OBJ) $(LODEPNG_SRC) $(ZOPFLIPNGLIB_SRC) $(ZOPFLIPNGBIN_SRC) $(CFLAGS) -o zopflipng $(LDLIBS)
 
 # ZopfliPNG shared library
 libzopflipng:
-- 
2.12.0

