From 825b70ac11282df1065cd7d1a968bdcc0d524938 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Thu, 27 Jan 2022 14:20:59 +0100
Subject: [PATCH 2/2] meson - fix fbdev usage

---
 meson.build       | 2 +-
 meson_options.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index afadcb2c..61ab239f 100644
--- a/meson.build
+++ b/meson.build
@@ -121,7 +121,7 @@ config_h.set10('TEST_GL_RENDERER', get_option('test-gl-renderer'))
 
 backend_default = get_option('backend-default')
 if backend_default == 'auto'
-	foreach b : [ 'headless', 'fbdev', 'x11', 'wayland', 'drm' ]
+	foreach b : [ 'headless', 'x11', 'wayland', 'drm' ]
 		if get_option('backend-' + b)
 			backend_default = b
 		endif
diff --git a/meson_options.txt b/meson_options.txt
index 8a527d74..f8a715d8 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -53,7 +53,7 @@ option(
 option(
 	'backend-default',
 	type: 'combo',
-	choices: [ 'auto', 'drm', 'wayland', 'x11', 'fbdev', 'headless' ],
+	choices: [ 'auto', 'drm', 'wayland', 'x11',  'headless' ],
 	value: 'drm',
 	description: 'Default backend when no parent display server detected'
 )
-- 
2.35.1

