--- src/ui_callbacks.c.old	2006-09-09 12:21:09.000000000 +0300
+++ src/ui_callbacks.c	2006-09-09 12:58:40.000000000 +0300
@@ -238,6 +238,7 @@
   if (ui->shutdown)
     {
       gtk_main_quit ();
+      process_info_destroy(ui->procinfo);
       return;
     }
   
@@ -557,6 +558,7 @@
 
       case CTRLMSG_QUIT:
         gtk_main_quit ();
+        process_info_destroy(ui->procinfo);
         break;
         
       }
