--- lemon.c.orig	2009-04-29 13:05:36.000000000 +0200
+++ lemon.c	2009-04-29 13:07:11.000000000 +0200
@@ -3082,6 +3082,8 @@
     tpltname = buf;
   }else if( access(templatename,004)==0 ){
     tpltname = templatename;
+  }else if( access("INSTALL_ROOT/usr/share/lemon/lempar.c",004)==0 ){
+    tpltname = "INSTALL_ROOT/usr/share/lemon/lempar.c";
   }else{
     tpltname = pathsearch(lemp->argv0,templatename,0);
   }
