--- install.sh.orig	2009-04-02 19:03:31.000000000 +0200
+++ install.sh	2009-04-02 19:05:55.000000000 +0200
@@ -32,30 +32,30 @@
 mkdir -p `dirname $SCRIPT_DEST`
 cp -a hibernate.sh $SCRIPT_DEST
 
-echo "Installing configuration files to $CONFIG_DIR ..."
-mkdir -p $CONFIG_DIR
-# We assume that if hibernate.conf does not exist, no config files do.
-# Let a package management system figure this one out.
-if [ -f $CONFIG_FILE ] ; then
-    ext=.dist
-    echo "  **"
-    echo "  ** You already have a configuration file at $CONFIG_FILE"
-    echo "  ** The new version will be installed to ${CONFIG_FILE}${ext}"
-    echo "  **"
-    EXISTING_CONFIG=1
-else
-    ext=
-fi
-cp -a hibernate.conf ${CONFIG_FILE}${ext}
-cp -a ram.conf ${RAM_CONFIG_FILE}${ext}
-cp -a disk.conf ${DISK_CONFIG_FILE}${ext}
-cp -a tuxonice.conf ${S2_CONFIG_FILE}${ext}
-cp -a ususpend-ram.conf ${US_RAM_CONFIG_FILE}${ext}
-cp -a ususpend-disk.conf ${US_DISK_CONFIG_FILE}${ext}
-cp -a ususpend-both.conf ${US_BOTH_CONFIG_FILE}${ext}
-cp -a sysfs-ram.conf ${SYSFS_RAM_CONFIG_FILE}${ext}
-cp -a sysfs-disk.conf ${SYSFS_DISK_CONFIG_FILE}${ext}
-cp -a common.conf ${COMMON_CONFIG_FILE}${ext}
+#echo "Installing configuration files to $CONFIG_DIR ..."
+#mkdir -p $CONFIG_DIR
+## We assume that if hibernate.conf does not exist, no config files do.
+## Let a package management system figure this one out.
+#if [ -f $CONFIG_FILE ] ; then
+#    ext=.dist
+#    echo "  **"
+#    echo "  ** You already have a configuration file at $CONFIG_FILE"
+#    echo "  ** The new version will be installed to ${CONFIG_FILE}${ext}"
+#    echo "  **"
+#    EXISTING_CONFIG=1
+#else
+#    ext=
+#fi
+#cp -a hibernate.conf ${CONFIG_FILE}${ext}
+#cp -a ram.conf ${RAM_CONFIG_FILE}${ext}
+#cp -a disk.conf ${DISK_CONFIG_FILE}${ext}
+#cp -a tuxonice.conf ${S2_CONFIG_FILE}${ext}
+#cp -a ususpend-ram.conf ${US_RAM_CONFIG_FILE}${ext}
+#cp -a ususpend-disk.conf ${US_DISK_CONFIG_FILE}${ext}
+#cp -a ususpend-both.conf ${US_BOTH_CONFIG_FILE}${ext}
+#cp -a sysfs-ram.conf ${SYSFS_RAM_CONFIG_FILE}${ext}
+#cp -a sysfs-disk.conf ${SYSFS_DISK_CONFIG_FILE}${ext}
+#cp -a common.conf ${COMMON_CONFIG_FILE}${ext}
 
 if [ -n "$DISTRIBUTION" ] ; then
     sed -i -e "s/^# \\(Distribution\\) .*/\1 $DISTRIBUTION/" ${COMMON_CONFIG_FILE}${ext}
@@ -105,13 +105,13 @@
 
 echo "Installed."
 echo
-if [ -z "$EXISTING_CONFIG" ] ; then
-    echo "Edit $CONFIG_FILE to taste, and see `basename $SCRIPT_DEST` -h for help."
-else
-    echo "You may want to merge $CONFIG_FILE with"
-    echo "$CONFIG_FILE.dist"
-    echo "See `basename $SCRIPT_DEST` -h for help on any extra options."
-fi
+#if [ -z "$EXISTING_CONFIG" ] ; then
+#    echo "Edit $CONFIG_FILE to taste, and see `basename $SCRIPT_DEST` -h for help."
+#else
+#    echo "You may want to merge $CONFIG_FILE with"
+#    echo "$CONFIG_FILE.dist"
+#    echo "See `basename $SCRIPT_DEST` -h for help on any extra options."
+#fi
 
 
 
