--- common.conf.old	2006-06-28 10:30:42.000000000 +0200
+++ common.conf	2006-06-28 10:31:05.000000000 +0200
@@ -14,7 +14,8 @@
 # AlwaysForce yes
 # AlwaysKill yes
 # HibernateVT 15
-# Distribution debian (or fedora/gentoo/mandrake/redhat/slackware/suse)
+# Distribution debian (or fedora/gentoo/mandrake/redhat/slackware/suse/sourcemage)
+Distribution sourcemage
 # XDisplay :0
 
 ##############################################################################
