diff -Naur hibernate-script-1.10.orig/hibernate.sh hibernate-script-1.10/hibernate.sh
--- hibernate-script-1.10.orig/hibernate.sh	2005-07-20 21:13:14.547868232 -0700
+++ hibernate-script-1.10/hibernate.sh	2005-07-20 21:21:17.888389304 -0700
@@ -670,7 +670,7 @@
     AddConfigHelp "LogVerbosity N" "Same as Verbosity, but controls what is written to the logfile."
     AddConfigHelp "AlwaysForce <boolean>" "If set to yes, the script will always run as if --force had been passed."
     AddConfigHelp "AlwaysKill <boolean>" "If set to yes, the script will always run as if --kill had been passed."
-    AddConfigHelp "Distribution <debian|fedora|mandrake|redhat|gentoo|suse|slackware>" "If specified, tweaks some scriptlets to be more integrated with the given distribution."
+    AddConfigHelp "Distribution <debian|fedora|mandrake|redhat|gentoo|suse|slackware|sourcemage>" "If specified, tweaks some scriptlets to be more integrated with the given distribution."
     AddConfigHelp "Include <filename>" "Read configuration directives from the given file."
     AddConfigHelp "XDisplay <display location>" "Specifies where scriptlets that use the X server should find one. (Default: :0)"
 }
