--- hibernate.sh.old	2006-06-28 10:37:41.000000000 +0200
+++ hibernate.sh	2006-06-28 10:38:08.000000000 +0200
@@ -702,7 +702,7 @@
     AddConfigHelp "LogVerbosity N" "Same as Verbosity, but controls what is written to the logfile."
     AddConfigHelp "AlwaysForce <boolean>" "If set to yes, the script will always run as if --force had been passed."
     AddConfigHelp "AlwaysKill <boolean>" "If set to yes, the script will always run as if --kill had been passed."
-    AddConfigHelp "Distribution <debian|fedora|mandrake|redhat|gentoo|suse|slackware>" "If specified, tweaks some scriptlets to be more integrated with the given distribution."
+    AddConfigHelp "Distribution <debian|fedora|mandrake|redhat|gentoo|suse|slackware|sourcemage>" "If specified, tweaks some scriptlets to be more integrated with the given distribution."
     AddConfigHelp "Include <filename>" "Immediately include configuration directives from the given file."
     AddConfigHelp "XDisplay <display location>" "Specifies where scriptlets that use the X server should find one. (Default: :0)"
 }
