diff -Naur hibernate-script-1.10.orig/hibernate.conf hibernate-script-1.10/hibernate.conf
--- hibernate-script-1.10.orig/hibernate.conf	2005-07-20 21:13:14.545868536 -0700
+++ hibernate-script-1.10/hibernate.conf	2005-07-20 21:16:33.740586336 -0700
@@ -64,7 +64,8 @@
 # AlwaysForce yes
 # AlwaysKill yes
 # HibernateVT 15
-# Distribution debian (or fedora/gentoo/mandrake/redhat/slackware/suse)
+# Distribution (or fedora/gentoo/mandrake/redhat/slackware/suse/sourcemage)
+Distribution sourcemage
 # XDisplay :0
 
 ##############################################################################
diff -Naur hibernate-script-1.10.orig/ram.conf hibernate-script-1.10/ram.conf
--- hibernate-script-1.10.orig/ram.conf	2005-07-20 21:13:14.547868232 -0700
+++ hibernate-script-1.10/ram.conf	2005-07-20 21:16:04.034102408 -0700
@@ -17,6 +17,7 @@
 # AlwaysKill yes
 # HibernateVT 15
 # Distribution debian (or fedora/gentoo/mandrake/redhat/slackware/suse)
+Distribution  sourcemage
 # XDisplay :0
 
 ##############################################################################
