--- src/libguppiplot/guppi-layout-constraint.c.orig	2004-07-22 10:10:45.481338456 +0100
+++ src/libguppiplot/guppi-layout-constraint.c	2004-07-22 10:11:05.595280672 +0100
@@ -165,6 +165,7 @@
 
   default:
     /* Fall through, do nothing. */
+	;;
   }
 
   ti = term_info_new (type, factor, geom);
--- src/libguppiplot/guppi-text-block.c.orig	2004-07-22 10:09:43.993686000 +0100
+++ src/libguppiplot/guppi-text-block.c	2004-07-22 10:10:13.209244560 +0100
@@ -900,6 +900,7 @@
 
     default:
       /* Do nothing. */
+	;;
     }
 
     render_stack_evolve (stack, tt);
--- src/libguppiplot/guppi-root-group-item.c.orig	2004-07-22 10:11:45.379232592 +0100
+++ src/libguppiplot/guppi-root-group-item.c	2004-07-22 10:12:23.891377856 +0100
@@ -548,6 +548,7 @@
     return motion_notify_event (root, &(ev->motion));
 
   default:
+	;;
   }
 
   return FALSE;
@@ -777,6 +778,7 @@
   case ROOT_GROUP_RESIZE_NONE:
   default:
     /* insure we do nothing */
+	;;
   }
 
   if (please_resize_canvas)
--- src/libguppiplot/guppi-view-interval.c.orig	2004-07-22 10:07:20.323527184 +0100
+++ src/libguppiplot/guppi-view-interval.c	2004-07-22 10:09:05.293569312 +0100
@@ -227,6 +227,7 @@
 #endif
 
   default:
+	;;
   }
 
   return TRUE;
--- src/demo.c.orig	2004-07-22 10:14:39.986688224 +0100
+++ src/demo.c	2004-07-22 10:14:53.046702800 +0100
@@ -451,6 +451,7 @@
     compass_pos = GUPPI_NORTH;
     break;
   default:
+	;;
   }
 
   guppi_element_state_set (state,
--- libguppitank/guppi-object-barchart.c.orig	2004-07-22 10:16:50.599832008 +0100
+++ libguppitank/guppi-object-barchart.c	2004-07-22 10:17:37.104762184 +0100
@@ -701,6 +701,7 @@
     break;
 
   default:
+	;;
   }
 }
 
@@ -733,6 +734,7 @@
     break;
 
   default:
+	;;
   }
 }
 
