--- fbnews.py.orig	2005-05-31 19:02:24.000000000 +0400
+++ fbnews.py	2015-03-31 18:03:11.204129918 +0300
@@ -24,7 +24,7 @@
 __revision__ = "0.5"
 
 DEBUG = False
-ENCODING = 'iso8859_15'
+ENCODING = 'utf-8'
 
 import urllib, time, os, codecs, sys, getopt, StringIO
 import fbnconf
@@ -77,7 +77,7 @@
     	format += config.siteformat(caption) + ') {}\n'
     
     for key in lst[1].keys():
-        format += u'[exec] (' + key + ') {' \
+        format += u'[exec] (' + key.replace(')', '\\)') + ') {' \
                   + config.browser(lst[1][key])  + u'}\n'
 
     format += '[end]\n'
