--- Makefile	2004-06-26 20:16:04.000000000 +0200
+++ Makefile-fixed	2011-04-14 13:19:22.996849506 +0200
@@ -11,13 +11,13 @@
 
 DESTDIR =
 
-PREFIX = $(DESTDIR)/usr/local
-MANDIR = $(PREFIX)/man/man1
+PREFIX = $(DESTDIR)/usr
+MANDIR = $(PREFIX)/share/man/man1
 DOCDIR = $(PREFIX)/share/doc
 
 # Image library
-USE_GDKPIXBUF = 1
-#USE_GDKPIXBUF2 = 1
+#USE_GDKPIXBUF = 1
+USE_GDKPIXBUF2 = 1
 #USE_IMLIB = 1
 
 DEFS = $(OS)
@@ -30,8 +30,8 @@
 
 ifdef USE_GDKPIXBUF2
 IMGLIB_DEF = -DUSE_GDKPIXBUF2
-CFLAGS = -O3 -Wall `pkg-config --cflags gdk-pixbuf-xlib-2.0` -I/usr/X11R6/include
-LDFLAGS = `pkg-config --libs gdk-pixbuf-xlib-2.0` -rdynamic -lX11 -lXi -lXext -L/usr/X11R6/lib
+CFLAGS = -O3 -Wall `pkg-config --cflags gdk-pixbuf-xlib-2.0` -I/usr/include/X11
+LDFLAGS = `pkg-config --libs gdk-pixbuf-xlib-2.0` -rdynamic -lX11 -lXi -lXext -L/usr/lib
 endif
 
 ifdef USE_IMLIB
