diff --git a/alsactl/90-alsa-restore.rules.in b/alsactl/90-alsa-restore.rules.in
index dae2ed8c..5db6080b 100644
--- a/alsactl/90-alsa-restore.rules.in
+++ b/alsactl/90-alsa-restore.rules.in
@@ -14,7 +14,7 @@ DRIVERS=="snd_hda_intel", TEST=="device/pcmC$env{ALSA_CARD_NUMBER}D0p", RUN+="/b
 TEST=="device/device/acp3x-dmic-capture", GOTO="alsa_hda_analog"
 TEST=="device/device/acp6x-dmic-capture", GOTO="alsa_hda_analog"
 TEST=="device/device/acp63-dmic-capture", GOTO="alsa_hda_analog"
-TEST=="device/device/acp-pdm-dmic", GOTO="alsa_hda_analog"
+TEST=="device/device/acp-dmic-codec", GOTO="alsa_hda_analog"
 GOTO="alsa_restore_std"
 
 LABEL="alsa_hda_analog"
