--- src/parser/attack_scanner.l.orig	2020-06-23 23:01:26.000000000 +0000
+++ src/parser/attack_scanner.l	2023-10-14 19:59:29.032617779 +0000
@@ -225,7 +225,7 @@
 "authentication failure "[^0-9]+                                { return CUCIPOP_AUTHFAIL; }
 
  /* Exim */
-"authenticator failed for ".+" ["                               { BEGIN(exim_esmtp_autherr); return EXIM_ESMTP_AUTHFAIL_PREF; }
+[A-Za-z0-9_-]+" authenticator failed for ".+" ["                               { BEGIN(exim_esmtp_autherr); return EXIM_ESMTP_AUTHFAIL_PREF; }
 <exim_esmtp_autherr>"]"(:{NUMBER})?(" I=".+)?(" U=".+)?": 535 Incorrect authentication data"(" (set_id=".+")")?    { BEGIN(INITIAL); return EXIM_ESMTP_AUTHFAIL_SUFF; }
 
 "SMTP protocol error in \"AUTH LOGIN\" H="({HOSTNAME}" ")?"(".*") ["              { BEGIN(exim_esmtp_loginerr); return EXIM_ESMTP_LOGINFAIL_PREF; }
