diff -Naupr boxbackup-0.11rc2.orig/bin/bbackupctl/bbackupctl.cpp boxbackup-0.11rc2/bin/bbackupctl/bbackupctl.cpp
--- boxbackup-0.11rc2.orig/bin/bbackupctl/bbackupctl.cpp	2008-04-15 16:28:40.359658673 +0200
+++ boxbackup-0.11rc2/bin/bbackupctl/bbackupctl.cpp	2008-04-15 16:29:06.718426918 +0200
@@ -49,6 +49,7 @@
 #include "Box.h"
 
 #include <stdio.h>
+#include <stdlib.h>
 
 #ifdef HAVE_UNISTD_H
 	#include <unistd.h>
diff -Naupr boxbackup-0.11rc2.orig/bin/bbackupquery/bbackupquery.cpp boxbackup-0.11rc2/bin/bbackupquery/bbackupquery.cpp
--- boxbackup-0.11rc2.orig/bin/bbackupquery/bbackupquery.cpp	2008-04-15 16:28:40.359658673 +0200
+++ boxbackup-0.11rc2/bin/bbackupquery/bbackupquery.cpp	2008-04-15 16:29:06.735573369 +0200
@@ -54,6 +54,7 @@
 
 #include <errno.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 #ifdef HAVE_SYS_TYPES_H
 	#include <sys/types.h>
diff -Naupr boxbackup-0.11rc2.orig/bin/bbstoreaccounts/bbstoreaccounts.cpp boxbackup-0.11rc2/bin/bbstoreaccounts/bbstoreaccounts.cpp
--- boxbackup-0.11rc2.orig/bin/bbstoreaccounts/bbstoreaccounts.cpp	2008-04-15 16:28:40.315117284 +0200
+++ boxbackup-0.11rc2/bin/bbstoreaccounts/bbstoreaccounts.cpp	2008-04-15 16:29:06.735573369 +0200
@@ -52,6 +52,7 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <limits.h>
+#include <string.h>
 #include <vector>
 #include <algorithm>
 
diff -Naupr boxbackup-0.11rc2.orig/lib/backupclient/BackupClientFileAttributes.cpp boxbackup-0.11rc2/lib/backupclient/BackupClientFileAttributes.cpp
--- boxbackup-0.11rc2.orig/lib/backupclient/BackupClientFileAttributes.cpp	2008-04-15 16:28:40.141742232 +0200
+++ boxbackup-0.11rc2/lib/backupclient/BackupClientFileAttributes.cpp	2008-04-15 16:29:06.735573369 +0200
@@ -54,8 +54,8 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <string.h>
 #include <limits.h>
+#include <cstring>
 #include <algorithm>
 #include <new>
 #include <vector>
diff -Naupr boxbackup-0.11rc2.orig/lib/backupclient/BackupStoreFileDiff.cpp boxbackup-0.11rc2/lib/backupclient/BackupStoreFileDiff.cpp
--- boxbackup-0.11rc2.orig/lib/backupclient/BackupStoreFileDiff.cpp	2008-04-15 16:28:40.141742232 +0200
+++ boxbackup-0.11rc2/lib/backupclient/BackupStoreFileDiff.cpp	2008-04-15 16:29:06.735573369 +0200
@@ -51,6 +51,8 @@
 #include <new>
 #include <map>
 
+#include <string.h>
+
 #ifdef HAVE_TIME_H
 	#include <time.h>
 #elif HAVE_SYS_TIME_H
diff -Naupr boxbackup-0.11rc2.orig/lib/backupclient/BackupStoreFileEncodeStream.cpp boxbackup-0.11rc2/lib/backupclient/BackupStoreFileEncodeStream.cpp
--- boxbackup-0.11rc2.orig/lib/backupclient/BackupStoreFileEncodeStream.cpp	2008-04-15 16:28:40.141742232 +0200
+++ boxbackup-0.11rc2/lib/backupclient/BackupStoreFileEncodeStream.cpp	2008-04-15 16:29:06.735573369 +0200
@@ -63,6 +63,8 @@
 
 #include "MemLeakFindOn.h"
 
+#include <string.h>
+
 using namespace BackupStoreFileCryptVar;
 
 
diff -Naupr boxbackup-0.11rc2.orig/lib/backupclient/BackupStoreFile.h boxbackup-0.11rc2/lib/backupclient/BackupStoreFile.h
--- boxbackup-0.11rc2.orig/lib/backupclient/BackupStoreFile.h	2008-04-15 16:28:40.155148417 +0200
+++ boxbackup-0.11rc2/lib/backupclient/BackupStoreFile.h	2008-04-15 16:29:06.735573369 +0200
@@ -53,6 +53,7 @@
 #include "BackupClientFileAttributes.h"
 #include "BackupStoreFilename.h"
 
+#include <stdlib.h>
 #include <memory>
 
 typedef struct 
diff -Naupr boxbackup-0.11rc2.orig/lib/common/Configuration.cpp boxbackup-0.11rc2/lib/common/Configuration.cpp
--- boxbackup-0.11rc2.orig/lib/common/Configuration.cpp	2008-04-15 16:28:40.085155989 +0200
+++ boxbackup-0.11rc2/lib/common/Configuration.cpp	2008-04-15 16:29:06.735573369 +0200
@@ -50,6 +50,7 @@
 
 #include <stdlib.h>
 #include <limits.h>
+#include <string.h>
 
 #include "Configuration.h"
 #include "CommonException.h"
diff -Naupr boxbackup-0.11rc2.orig/lib/common/Logging.cpp boxbackup-0.11rc2/lib/common/Logging.cpp
--- boxbackup-0.11rc2.orig/lib/common/Logging.cpp	2008-04-15 16:28:40.085155989 +0200
+++ boxbackup-0.11rc2/lib/common/Logging.cpp	2008-04-15 16:30:31.228397630 +0200
@@ -58,6 +58,7 @@
 #include "Logging.h"
 
 #include <iomanip>
+#include <cstring>
 
 #include "BoxTime.h"
 
diff -Naupr boxbackup-0.11rc2.orig/lib/common/Timer.cpp boxbackup-0.11rc2/lib/common/Timer.cpp
--- boxbackup-0.11rc2.orig/lib/common/Timer.cpp	2008-04-15 16:28:40.085155989 +0200
+++ boxbackup-0.11rc2/lib/common/Timer.cpp	2008-04-15 16:32:52.535039650 +0200
@@ -50,6 +50,7 @@
 #include "Box.h"
 
 #include <signal.h>
+#include <cstring>
 
 #include "Timer.h"
 #include "Logging.h"
diff -Naupr boxbackup-0.11rc2.orig/lib/common/WaitForEvent.h boxbackup-0.11rc2/lib/common/WaitForEvent.h
--- boxbackup-0.11rc2.orig/lib/common/WaitForEvent.h	2008-04-15 16:28:40.065075319 +0200
+++ boxbackup-0.11rc2/lib/common/WaitForEvent.h	2008-04-15 16:29:06.735573369 +0200
@@ -59,6 +59,8 @@
 	#endif
 #endif
 
+#include <stdlib.h>
+
 #include "CommonException.h"
 
 #include "MemLeakFindOn.h"
diff -Naupr boxbackup-0.11rc2.orig/lib/raidfile/RaidFileRead.cpp boxbackup-0.11rc2/lib/raidfile/RaidFileRead.cpp
--- boxbackup-0.11rc2.orig/lib/raidfile/RaidFileRead.cpp	2008-04-15 16:28:40.141742232 +0200
+++ boxbackup-0.11rc2/lib/raidfile/RaidFileRead.cpp	2008-04-15 16:29:06.735573369 +0200
@@ -52,6 +52,7 @@
 #include <fcntl.h>
 #include <stdarg.h>
 #include <unistd.h>
+#include <stdlib.h>
 
 #include <sys/stat.h>
 #include <sys/types.h>
