--- src/callback.h.orig	2007-09-11 19:14:32.000000000 +0400
+++ src/callback.h	2015-04-13 15:24:54.000000000 +0300
@@ -27,7 +27,7 @@
 void on_file_print_preview(void);
 void on_file_print(void);
 void on_file_close(void);
-void on_file_quit(void);
+gboolean on_file_quit(void);
 void on_edit_undo(void);
 void on_edit_redo(void);
 void on_edit_cut(void);
--- src/callback.c.orig	2010-12-18 11:19:01.000000000 +0300
+++ src/callback.c	2015-04-13 15:25:34.000000000 +0300
@@ -169,12 +169,15 @@
 	}
 }
 
-void on_file_quit(void)
+gboolean on_file_quit(void)
 {
 	if (!check_text_modification()) {
 		save_config_file();
 		gtk_main_quit();
+		return FALSE;
 	}
+
+	return TRUE;
 }
 
 void on_edit_undo(void)
