From 034b91a065177cfc70b4cc9318bd25b7eddd73d7 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 1 May 2015 10:11:12 +0200
Subject: [PATCH] add missing include

---
 src/celmath/intersect.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/celmath/intersect.h b/src/celmath/intersect.h
index 38fb01f..a1aab6a 100644
--- a/src/celmath/intersect.h
+++ b/src/celmath/intersect.h
@@ -15,6 +15,7 @@
 #include "ray.h"
 #include "sphere.h"
 #include "ellipsoid.h"
+#include <celmath/mathlib.h>
 
 
 template<class T> bool testIntersection(const Ray3<T>& ray,
-- 
2.3.5

