Index: release-0.2/src/libStats/CMakeLists.txt
===================================================================
--- release-0.2/src/libStats/CMakeLists.txt	(révision 85)
+++ release-0.2/src/libStats/CMakeLists.txt	(copie de travail)
@@ -26,3 +26,5 @@
 SET(libraries libStats libalgebra libgmp.so libgmpxx.so)
 
 TARGET_LINK_LIBRARIES(libStatsTest ${libraries})
+
+INSTALL(TARGETS libStats LIBRARY DESTINATION /usr/lib)
\ No newline at end of file
Index: release-0.2/src/libNumberTheory/CMakeLists.txt
===================================================================
--- release-0.2/src/libNumberTheory/CMakeLists.txt	(révision 85)
+++ release-0.2/src/libNumberTheory/CMakeLists.txt	(copie de travail)
@@ -32,3 +32,5 @@
 SET(libraries libNumberTheory libalgebra libgmp.so libgmpxx.so)
 
 TARGET_LINK_LIBRARIES(libNumberTheoryTest ${libraries})
+
+INSTALL(TARGETS libNumberTheory LIBRARY DESTINATION /usr/lib)
\ No newline at end of file
Index: release-0.2/src/libalgebra/CMakeLists.txt
===================================================================
--- release-0.2/src/libalgebra/CMakeLists.txt	(révision 85)
+++ release-0.2/src/libalgebra/CMakeLists.txt	(copie de travail)
@@ -70,3 +70,4 @@
 #INSTALL_PROGRAMS(...)
 #INSTALL_TARGET(...)
 
+INSTALL(TARGETS libalgebra LIBRARY DESTINATION /usr/lib)
\ No newline at end of file
Index: release-0.2/src/libClassicalAnalysis/CMakeLists.txt
===================================================================
--- release-0.2/src/libClassicalAnalysis/CMakeLists.txt	(révision 85)
+++ release-0.2/src/libClassicalAnalysis/CMakeLists.txt	(copie de travail)
@@ -28,3 +28,5 @@
 SET(libraries libClassicalAnalysis libalgebra libgmp.so libgmpxx.so)
 
 TARGET_LINK_LIBRARIES(libClassicalAnalysisTest ${libraries})
+
+INSTALL(TARGETS libClassicalAnalysis LIBRARY DESTINATION /usr/lib)
\ No newline at end of file
Index: release-0.2/src/libAnalysis/CMakeLists.txt
===================================================================
--- release-0.2/src/libAnalysis/CMakeLists.txt	(révision 85)
+++ release-0.2/src/libAnalysis/CMakeLists.txt	(copie de travail)
@@ -36,3 +36,5 @@
 SET(libraries libAnalysis libNumberTheory libalgebra libgmp.so libgmpxx.so)
 
 TARGET_LINK_LIBRARIES(libAnalysisTest ${libraries})
+
+INSTALL(TARGETS libAnalysis LIBRARY DESTINATION /usr/lib)
\ No newline at end of file
Index: release-0.2/src/libClassicalCrypto/CMakeLists.txt
===================================================================
--- release-0.2/src/libClassicalCrypto/CMakeLists.txt	(révision 85)
+++ release-0.2/src/libClassicalCrypto/CMakeLists.txt	(copie de travail)
@@ -32,3 +32,4 @@
 
 TARGET_LINK_LIBRARIES(libClassicalCryptoTest ${libraries})
 
+INSTALL(TARGETS libClassicalCrypto LIBRARY DESTINATION /usr/lib)
\ No newline at end of file
Index: release-0.2/CMakeLists.txt
===================================================================
--- release-0.2/CMakeLists.txt	(révision 85)
+++ release-0.2/CMakeLists.txt	(copie de travail)
@@ -10,4 +10,4 @@
 INSTALL(DIRECTORY doc/translation DESTINATION /usr/share/cryptoolinux
 	PATTERN .svn EXCLUDE PATTERN *.ts EXCLUDE)
 
-INSTALL(DIRECTORY doc/help DESTINATION /usr/share/cryptoolinux PATTERN .svn EXCLUDE)
+#INSTALL(DIRECTORY doc/help DESTINATION /usr/share/cryptoolinux PATTERN .svn EXCLUDE)
