diff -Naur sendmail-8.13.4.orig/cf/cf/Makefile sendmail-8.13.4/cf/cf/Makefile
--- sendmail-8.13.4.orig/cf/cf/Makefile	2005-06-03 11:00:18.000000000 -0700
+++ sendmail-8.13.4/cf/cf/Makefile	2005-06-03 11:00:27.000000000 -0700
@@ -80,10 +80,8 @@
 install-cf:  install-sendmail-cf install-submit-cf
 
 install-sendmail-cf: $(CF).cf
-	$(INSTALL) -c -o $(CFOWN) -g $(CFGRP) -m $(CFMODE) $(CF).cf ${DESTDIR}$(MAILDIR)/sendmail.cf
 
 install-submit-cf: $(SUBMIT).cf
-	$(INSTALL) -c -o $(CFOWN) -g $(CFGRP) -m $(CFMODE) $(SUBMIT).cf ${DESTDIR}$(MAILDIR)/submit.cf
 
 depend:
 
