From 0a06c46c6e57ef507185a8b71f242069849fd4d5 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Fri, 31 Mar 2023 20:50:24 +0200
Subject: [PATCH 3/5] Replace <sys/signal.h> by the standard <signal.h>

Upstream-Status: Inappropriate [CLA involved]
Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 src/OSD/OSD_signal.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/OSD/OSD_signal.cxx b/src/OSD/OSD_signal.cxx
index e9cac5aea2dd..7b03d6097bd3 100644
--- a/src/OSD/OSD_signal.cxx
+++ b/src/OSD/OSD_signal.cxx
@@ -764,7 +764,7 @@ typedef void (* SIG_PFV) (int);
 #include <signal.h>
 
 #if !defined(__ANDROID__) && !defined(__QNX__) && !defined(__EMSCRIPTEN__)
-  #include <sys/signal.h>
+  #include <signal.h>
 #endif
 
 # define _OSD_FPX (FE_INVALID | FE_DIVBYZERO | FE_OVERFLOW)
-- 
2.40.0

