From 75ff9dfa85b96bf5a364671ac3a279d85b46cc32 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sat, 1 Apr 2023 20:27:07 +0200
Subject: [PATCH 5/5] Add missing include of <limits>

Upstream-Status: Inappropriate [CLA involved]
Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 src/ViewerTest/ViewerTest_CmdParser.cxx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ViewerTest/ViewerTest_CmdParser.cxx b/src/ViewerTest/ViewerTest_CmdParser.cxx
index 04cb7d4b1cf4..a8619da14425 100644
--- a/src/ViewerTest/ViewerTest_CmdParser.cxx
+++ b/src/ViewerTest/ViewerTest_CmdParser.cxx
@@ -21,6 +21,7 @@
 
 #include <algorithm>
 #include <iostream>
+#include <limits>
 
 namespace
 {
-- 
2.40.0

