--- util/check_lin_tif.c.orig	2016-02-07 00:38:43.473467770 +0100
+++ util/check_lin_tif.c	2016-02-07 00:40:11.723364495 +0100
@@ -84,9 +84,7 @@
                 "risking instability and probable data loss.  You are very brave!\n"
                 "Please do not report bugs to the NEdit developers unless you can\n"
                 "reproduce them with a known-good NEdit binary downloaded from:\n"
-                "\thttp://www.nedit.org\n"
-                "\nHIT ENTER TO CONTINUE\n");
-            fgets(buf, 2, stdin);
+                "\thttp://www.nedit.org\n");
             return EXIT_SUCCESS;
         }
 #else
@@ -122,9 +120,7 @@
                 "\n========================== WARNING ===========================\n"
                 "You have chosen to build NEdit with an untested version of Motif.\n"
                 "Please report your success or failure with this version to:\n"
-                "\tdevelop@nedit.org\n"
-                "\nHIT ENTER TO CONTINUE\n");
-            fgets(buf, 2, stdin);
+                "\tdevelop@nedit.org\n");
             return EXIT_SUCCESS;
         }
 #else
