diff -ruN nedit-5.5.orig/source/window.c nedit-5.5/source/window.c
--- nedit-5.5.orig/source/window.c	2004-09-16 00:50:58.000000000 +0200
+++ nedit-5.5/source/window.c	2006-09-21 18:31:45.000000000 +0200
@@ -144,6 +144,9 @@
    0x00, 0x00, 0x00, 0x00, 0x04, 0x01, 0x84, 0x01, 0xc4, 0x00, 0x64, 0x00,
    0xc4, 0x00, 0x84, 0x01, 0x04, 0x01, 0x00, 0x00, 0x00, 0x00};
 
+/* ThOr: svrname in WM_CLASS hack */
+static const char* def_wmclass = "textShell";
+
 extern void _XmDismissTearOff(Widget, XtPointer, XtPointer);
 
 static void hideTooltip(Widget tab);
@@ -359,9 +362,20 @@
     XtSetArg(al[ac], XmNgeometry, newGeometry[0]=='\0'?NULL:newGeometry); ac++;
     XtSetArg(al[ac], XmNinitialState,
             iconic ? IconicState : NormalState); ac++;
-
-    winShell = CreateWidget(TheAppShell, "textShell",
-                topLevelShellWidgetClass, al, ac);
+    {
+   	/* ThOr: I want different WM_CLASS for different servers */
+		char* wmclass = NULL;
+   	if (GetPrefServerName()[0] != '\0'){
+   	  wmclass = (char*) malloc(sizeof(char) * (strlen(def_wmclass)+strlen(GetPrefServerName())+2));
+		  wmclass[0] = 0;
+		  strcat(wmclass, def_wmclass);
+		  strcat(wmclass, ":");
+  		  strcat(wmclass, GetPrefServerName());
+   	}
+   	winShell = CreateWidget(TheAppShell, wmclass == NULL ? def_wmclass : wmclass,
+               	topLevelShellWidgetClass, al, ac);
+		if(wmclass != NULL) free(wmclass);
+	 }
     window->shell = winShell;
 
 #ifdef EDITRES
@@ -2429,7 +2443,8 @@
     	WidgetList tabs;
 	int tabCount;
 	
-	if (strcmp(XtName(children[n]), "textShell") ||
+	/* ncmp for possible textShell:svrname */
+	if (strncmp(XtName(children[n]), def_wmclass, 9) ||
 	  ((topWin = WidgetToWindow(children[n])) == NULL))
 	    continue;   /* skip non-editor windows */ 
 
@@ -3407,7 +3422,8 @@
 	
 	/* get list of tabs in all windows */
     	for (n=0; n<nItems; n++) {
-	    if (strcmp(XtName(children[n]), "textShell") ||
+	   /* ncmp for possible textShell:svrname */
+	   if (strncmp(XtName(children[n]), def_wmclass, 9) ||
 	      ((win = WidgetToWindow(children[n])) == NULL))
 	    	continue;   /* skip non-text-editor windows */ 
 	    
