--- plugins/mtpdevice/rb-mtp-source.c
+++ plugins/mtpdevice/rb-mtp-source.c
@@ -681,7 +681,7 @@
 		trackmeta->filetype = mimetype_to_filetype (mimetype);
 	}
 
-	if (LIBMTP_Send_Track_From_File (device, filename, trackmeta, NULL, NULL, 0) != 0) {
+	if (LIBMTP_Send_Track_From_File (device, filename, trackmeta, NULL, NULL) != 0) {
 		LIBMTP_destroy_track_t (trackmeta);
 		rb_debug ("Tracktransfer failed\n");
 		return NULL;
