--- tkrat-2.1.5/lib/ratDummy.c.orig	2008-03-22 20:44:31.000000000 +0000
+++ tkrat-2.1.5/lib/ratDummy.c	2008-03-22 21:10:39.000000000 +0000
@@ -11,6 +11,7 @@
  * COPYRIGHT, included with this distribution.
  */
 
+#define Tk_ManageGeometry_TCL_DECLARED
 #include <tk.h>
 #ifndef CONST84
 #   define CONST84
--- tkrat-2.1.5/util/rat_dummy.c.orig	2008-03-22 21:02:32.000000000 +0000
+++ tkrat-2.1.5/util/rat_dummy.c	2008-03-22 21:10:42.000000000 +0000
@@ -11,6 +11,7 @@
  * COPYRIGHT, included with this distribution.
  */
 
+#define Tk_ManageGeometry_TCL_DECLARED
 #include <tk.h>
 
 #ifndef CONST84
