From c332e2d5043252d9a9b523b8aed9490a43339433 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 13 Jan 2017 20:41:39 +0100
Subject: [PATCH] fix zlib.pc install path

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 1852192..fd061cf 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -52,7 +52,7 @@ sharedlibdir = ${libdir}
 includedir = ${prefix}/include
 mandir = ${prefix}/share/man
 man3dir = ${mandir}/man3
-pkgconfigdir = ${libdir}/pkgconfig
+pkgconfigdir = ${prefix}/lib/pkgconfig
 SRCDIR=
 ZINC=
 ZINCOUT=-I.
-- 
2.11.0

