--- Makefile.in	2004-06-19 23:39:08.637359368 -0700
+++ Makefile.in	2004-06-19 23:39:26.166694504 -0700
@@ -94,7 +94,7 @@
 	-@if [ ! -d $(includedir)  ]; then mkdir -p $(includedir); fi
 	-@if [ ! -d $(libdir)      ]; then mkdir -p $(libdir); fi
 	-@if [ ! -d $(man3dir)     ]; then mkdir -p $(man3dir); fi
-	cp zlib.h zconf.h $(includedir)
+	cp zlib.h zconf.h zutil.h $(includedir)
 	chmod 644 $(includedir)/zlib.h $(includedir)/zconf.h
 	cp $(LIBS) $(libdir)
 	cd $(libdir); chmod 755 $(LIBS)
