--- Makefile.in.orig	2010-03-14 05:10:27.000000000 +0300
+++ Makefile.in	2010-03-16 11:16:50.868545002 +0300
@@ -48,7 +48,7 @@
 includedir = ${prefix}/include
 mandir = ${prefix}/share/man
 man3dir = ${mandir}/man3
-pkgconfigdir = ${libdir}/pkgconfig
+pkgconfigdir = ${prefix}/lib/pkgconfig
 
 OBJC = adler32.o compress.o crc32.o deflate.o gzclose.o gzlib.o gzread.o \
 	gzwrite.o infback.o inffast.o inflate.o inftrees.o trees.o uncompr.o zutil.o
