--- Makefile.in.orig	2010-03-14 05:10:27.000000000 +0300
+++ Makefile.in	2010-03-16 15:01:38.585868726 +0300
@@ -48,7 +48,7 @@
 includedir = ${prefix}/include
 mandir = ${prefix}/share/man
 man3dir = ${mandir}/man3
-pkgconfigdir = ${libdir}/pkgconfig
+pkgconfigdir = ${prefix}/lib/pkgconfig
 
 OBJC = adler32.o compress.o crc32.o deflate.o gzclose.o gzlib.o gzread.o \
 	gzwrite.o infback.o inffast.o inflate.o inftrees.o trees.o uncompr.o zutil.o
@@ -182,11 +182,11 @@
 
 install: install-libs
 	-@if [ ! -d $(DESTDIR)$(includedir)   ]; then mkdir -p $(DESTDIR)$(includedir); fi
-	cp zlib.h zconf.h $(DESTDIR)$(includedir)
-	chmod 644 $(DESTDIR)$(includedir)/zlib.h $(DESTDIR)$(includedir)/zconf.h
+	cp zlib.h zconf.h zutil.h $(DESTDIR)$(includedir)
+	chmod 644 $(DESTDIR)$(includedir)/zlib.h $(DESTDIR)$(includedir)/zconf.h $(DESTDIR)$(includedir)/zutil.h
 
 uninstall:
-	cd $(DESTDIR)$(includedir); rm -f zlib.h zconf.h
+	cd $(DESTDIR)$(includedir); rm -f zlib.h zconf.h zutil.h
 	cd $(DESTDIR)$(libdir); rm -f libz.a; \
 	if test "$(SHAREDLIBV)" -a -f $(SHAREDLIBV); then \
 	  rm -f $(SHAREDLIBV) $(SHAREDLIB) $(SHAREDLIBM); \
