--- cyrus-sasl-2.1.21/plugins/ntlm.c.openssl-098	2004-12-24 17:06:14.000000000 +0100
+++ cyrus-sasl-2.1.21/plugins/ntlm.c	2005-11-08 23:49:45.000000000 +0100
@@ -77,6 +77,7 @@
 #include <openssl/hmac.h>
 #include <openssl/des.h>
 #include <openssl/opensslv.h>
+#include <openssl/md5.h>
 #if (OPENSSL_VERSION_NUMBER >= 0x0090700f) && \
      !defined(OPENSSL_ENABLE_OLD_DES_SUPPORT)
 # define des_cblock DES_cblock
--- cyrus-sasl-2.1.21/plugins/srp.c.openssl-098	2004-06-23 20:43:37.000000000 +0200
+++ cyrus-sasl-2.1.21/plugins/srp.c	2005-11-08 23:51:08.000000000 +0100
@@ -87,6 +87,7 @@
 /* for digest and cipher support */
 #include <openssl/evp.h>
 #include <openssl/hmac.h>
+#include <openssl/md5.h>
 
 #include <sasl.h>
 #define MD5_H  /* suppress internal MD5 */
--- cyrus-sasl-2.1.21/plugins/passdss.c.openssl-098	2004-11-27 17:17:43.000000000 +0100
+++ cyrus-sasl-2.1.21/plugins/passdss.c	2005-11-08 23:50:25.000000000 +0100
@@ -67,6 +67,7 @@
 /* for digest and cipher support */
 #include <openssl/evp.h>
 #include <openssl/hmac.h>
+#include <openssl/md5.h>
 
 #include <sasl.h>
 #define MD5_H  /* suppress internal MD5 */
