--- texk/web2c/fmtutil.in	2005-02-07 11:59:59.895582760 +0100
+++ texk/web2c/fmtutil.in	2005-02-07 12:01:42.773942872 +0100
@@ -88,10 +88,10 @@
 #! pdfmex-pl	pdfetex		mexconf.tex	nls=pdfetex-pl,il2-pl *pdfmex.ini
 
 # - Czech / Slovak:
-#! csplain	pdfetex		-		-translate-file=cp227.tcx *csplain.ini
-#! cslatex	pdfetex		-	-translate-file=cp227.tcx *cslatex.ini
-#! pdfcsplain	pdfetex		-		-translate-file=cp227.tcx *csplain.ini
-#! pdfcslatex	pdfetex		-    -translate-file=cp227.tcx *cslatex.ini
+csplain	pdfetex		-		-translate-file=cp227.tcx *csplain.ini
+cslatex	pdfetex		-	-translate-file=cp227.tcx *cslatex.ini
+pdfcsplain	pdfetex		-		-translate-file=cp227.tcx *csplain.ini
+pdfcslatex	pdfetex		-    -translate-file=cp227.tcx *cslatex.ini
 
 # Other formats:
 
