diff -u -r gir-repository.orig/gir/Makefile.am gir-repository.new/gir/Makefile.am
--- gir-repository.orig/gir/Makefile.am	2010-07-23 17:51:18.978774095 -0500
+++ gir-repository.new/gir/Makefile.am	2010-07-23 17:54:34.640562414 -0500
@@ -881,17 +881,17 @@
 endif BUILD_AVAHI
 
 if BUILD_BABL
-BABL_INCLUDEDIR=`pkg-config --variable=includedir babl`/babl-0.0
-Babl-0.0.gir: $(INTROSPECTION_SCANNER)
-	$(INTROSPECTION_SCANNER) -v --namespace Babl --nsversion=0.0 \
+BABL_INCLUDEDIR=`pkg-config --variable=includedir babl`/babl-0.1
+Babl-0.1.gir: $(INTROSPECTION_SCANNER)
+	$(INTROSPECTION_SCANNER) -v --namespace Babl --nsversion=0.1 \
             --add-include-path=$(srcdir) --add-include-path=. \
 	    --include=GObject-2.0 \
-            --library=babl-0.0 \
+            --library=babl-0.1 \
 	    --libtool="$(LIBTOOL)" \
             --output $@ \
             --pkg babl \
             $(BABL_INCLUDEDIR)/babl/*.h
-BUILT_GIRSOURCES += Babl-0.0.gir
+BUILT_GIRSOURCES += Babl-0.1.gir
 endif BUILD_BABL
 
 if BUILD_UNIQUE
diff -u -r gir-repository.orig/gir/Makefile.in gir-repository.new/gir/Makefile.in
--- gir-repository.orig/gir/Makefile.in	2010-07-23 17:51:18.987774509 -0500
+++ gir-repository.new/gir/Makefile.in	2010-07-23 17:55:38.366984541 -0500
@@ -86,7 +86,7 @@
 @BUILD_GSSDP_TRUE@am__append_33 = GSSDP-1.0.gir
 @BUILD_GUPNP_TRUE@am__append_34 = GUPnP-1.0.gir
 @BUILD_AVAHI_TRUE@am__append_35 = Avahi-0.6.gir
-@BUILD_BABL_TRUE@am__append_36 = Babl-0.0.gir
+@BUILD_BABL_TRUE@am__append_36 = Babl-0.1.gir
 @BUILD_UNIQUE_TRUE@am__append_37 = Unique-1.0.gir
 @BUILD_NAUTILUS_TRUE@am__append_38 = Nautilus-1.0.gir
 @BUILD_WNCK_TRUE@am__append_39 = Wnck-1.0.gir
@@ -549,7 +549,7 @@
 @BUILD_GSSDP_TRUE@GSSDP_INCLUDEDIR = `pkg-config --variable=includedir gssdp-1.0`/gssdp-1.0
 @BUILD_GUPNP_TRUE@GUPNP_INCLUDEDIR = `pkg-config --variable=includedir gupnp-1.0`/gupnp-1.0
 @BUILD_AVAHI_TRUE@AVAHI_INCLUDEDIR = `pkg-config --variable=includedir avahi-gobject`/avahi-gobject
-@BUILD_BABL_TRUE@BABL_INCLUDEDIR = `pkg-config --variable=includedir babl`/babl-0.0
+@BUILD_BABL_TRUE@BABL_INCLUDEDIR = `pkg-config --variable=includedir babl`/babl-0.1
 @BUILD_UNIQUE_TRUE@UNIQUE_INCLUDEDIR = `pkg-config --variable=includedir unique-1.0`/unique-1.0/unique
 @BUILD_NAUTILUS_TRUE@NAUTILUS_INCLUDEDIR = `pkg-config --variable=includedir libnautilus-extension`/nautilus/libnautilus-extension
 @BUILD_WNCK_TRUE@WNCK_INCLUDEDIR = `pkg-config --variable=includedir libwnck-1.0`/libwnck-1.0/libwnck
@@ -1529,11 +1529,11 @@
 @BUILD_AVAHI_TRUE@             --output $@ \
 @BUILD_AVAHI_TRUE@             --pkg avahi-gobject \
 @BUILD_AVAHI_TRUE@             $(AVAHI_INCLUDEDIR)/*.h
-@BUILD_BABL_TRUE@Babl-0.0.gir: $(INTROSPECTION_SCANNER)
-@BUILD_BABL_TRUE@	$(INTROSPECTION_SCANNER) -v --namespace Babl --nsversion=0.0 \
+@BUILD_BABL_TRUE@Babl-0.1.gir: $(INTROSPECTION_SCANNER)
+@BUILD_BABL_TRUE@	$(INTROSPECTION_SCANNER) -v --namespace Babl --nsversion=0.1 \
 @BUILD_BABL_TRUE@            --add-include-path=$(srcdir) --add-include-path=. \
 @BUILD_BABL_TRUE@	    --include=GObject-2.0 \
-@BUILD_BABL_TRUE@            --library=babl-0.0 \
+@BUILD_BABL_TRUE@            --library=babl-0.1 \
 @BUILD_BABL_TRUE@	    --libtool="$(LIBTOOL)" \
 @BUILD_BABL_TRUE@            --output $@ \
 @BUILD_BABL_TRUE@            --pkg babl \
