--- configure.ac.orig	2006-10-12 19:44:43.000000000 -0500
+++ configure.ac	2006-10-12 19:45:40.000000000 -0500
@@ -183,7 +183,7 @@
               enable_poppler_glib=$enableval,
               enable_poppler_glib="try")
 if test x$enable_poppler_glib = xyes; then
-  PKG_CHECK_MODULES(POPPLER_GLIB, gdk-2.0 >= 2.4.0 glib >= 2.6)
+  PKG_CHECK_MODULES(POPPLER_GLIB, gdk-2.0 >= 2.4.0 glib-2.0 >= 2.6)
 elif test x$enable_poppler_glib = xtry; then
   PKG_CHECK_MODULES(POPPLER_GLIB, gdk-2.0 >= 2.4.0,
                     [enable_poppler_glib="yes"],
