--- configure.in~	2006-04-28 18:21:20.000000000 +0200
+++ configure.in	2006-05-06 10:31:30.000000000 +0200
@@ -701,7 +701,7 @@
     fi
     if test "$with_python" != ""
     then
-        pythondir=$(PYTHON_SITE_PACKAGES)
+        pythondir=$PYTHON_SITE_PACKAGES
     else
         pythondir='$(libdir)/python$(PYTHON_VERSION)/site-packages'
     fi
--- configure~	2006-05-06 10:31:56.000000000 +0200
+++ configure	2006-05-06 10:32:03.000000000 +0200
@@ -26958,7 +26958,7 @@
     fi
     if test "$with_python" != ""
     then
-        pythondir=$(PYTHON_SITE_PACKAGES)
+        pythondir=$PYTHON_SITE_PACKAGES
     else
         pythondir='$(libdir)/python$(PYTHON_VERSION)/site-packages'
     fi
