--- libxml2-2.6.24-orig/configure	2006-05-04 23:14:58.000000000 -0500
+++ libxml2-2.6.24/configure	2006-05-04 23:16:48.000000000 -0500
@@ -26935,17 +26935,17 @@
 	   -d $with_python/lib/python$PYTHON_VERSION/site-packages
 	then
 	    PYTHON_INCLUDES=$with_python/include/python$PYTHON_VERSION
-	    PYTHON_SITE_PACKAGES='$(libdir)/lib/python$(PYTHON_VERSION)/site-packages'
+	    PYTHON_SITE_PACKAGES='$libdir/lib/python$PYTHON_VERSION/site-packages'
 	else
 	    if test -r $prefix/include/python$PYTHON_VERSION/Python.h
 	    then
 	        PYTHON_INCLUDES=$prefix/include/python$PYTHON_VERSION
-		PYTHON_SITE_PACKAGES='$(libdir)/python$(PYTHON_VERSION)/site-packages'
+		PYTHON_SITE_PACKAGES='$libdir/python$PYTHON_VERSION/site-packages'
 	    else
 		if test -r /usr/include/python$PYTHON_VERSION/Python.h
 		then
 		    PYTHON_INCLUDES=/usr/include/python$PYTHON_VERSION
-		    PYTHON_SITE_PACKAGES='$(libdir)/python$(PYTHON_VERSION)/site-packages'
+		    PYTHON_SITE_PACKAGES='$libdir/python$PYTHON_VERSION/site-packages'
 		else
 		    echo could not find python$PYTHON_VERSION/Python.h
 		fi
@@ -26958,9 +26958,9 @@
     fi
     if test "$with_python" != ""
     then
-        pythondir=$(PYTHON_SITE_PACKAGES)
+        pythondir=$PYTHON_SITE_PACKAGES
     else
-        pythondir='$(libdir)/python$(PYTHON_VERSION)/site-packages'
+        pythondir='$libdir/python$PYTHON_VERSION/site-packages'
     fi
 else
     PYTHON=
--- libxml2-2.6.24-orig/configure.in	2006-05-04 23:14:58.000000000 -0500
+++ libxml2-2.6.24/configure.in	2006-05-04 23:28:06.000000000 -0500
@@ -678,17 +678,17 @@
 	   -d $with_python/lib/python$PYTHON_VERSION/site-packages
 	then
 	    PYTHON_INCLUDES=$with_python/include/python$PYTHON_VERSION
-	    PYTHON_SITE_PACKAGES='$(libdir)/lib/python$(PYTHON_VERSION)/site-packages'
+	    PYTHON_SITE_PACKAGES='$libdir/lib/python$PYTHON_VERSION/site-packages'
 	else
 	    if test -r $prefix/include/python$PYTHON_VERSION/Python.h
 	    then
 	        PYTHON_INCLUDES=$prefix/include/python$PYTHON_VERSION
-		PYTHON_SITE_PACKAGES='$(libdir)/python$(PYTHON_VERSION)/site-packages'
+		PYTHON_SITE_PACKAGES='$libdir/python$PYTHON_VERSION/site-packages'
 	    else
 		if test -r /usr/include/python$PYTHON_VERSION/Python.h
 		then
 		    PYTHON_INCLUDES=/usr/include/python$PYTHON_VERSION
-		    PYTHON_SITE_PACKAGES='$(libdir)/python$(PYTHON_VERSION)/site-packages'
+		    PYTHON_SITE_PACKAGES='$libdir/python$PYTHON_VERSION/site-packages'
 		else
 		    echo could not find python$PYTHON_VERSION/Python.h
 		fi
@@ -701,9 +701,9 @@
     fi
     if test "$with_python" != ""
     then
-        pythondir=$(PYTHON_SITE_PACKAGES)
+        pythondir=$PYTHON_SITE_PACKAGES
     else
-        pythondir='$(libdir)/python$(PYTHON_VERSION)/site-packages'
+        pythondir='$libdir/python$PYTHON_VERSION/site-packages'
     fi
 else
     PYTHON=
