--- qtiplot/qtiplot.pro.orig	2007-04-21 23:23:49.000000000 +0200
+++ qtiplot/qtiplot.pro	2007-04-21 23:25:01.000000000 +0200
@@ -4,7 +4,7 @@
 
 # building without muParser doesn't work yet
 SCRIPTING_LANGS += muParser
-SCRIPTING_LANGS += Python
+#SCRIPTING_LANGS += Python
 
 # a console displaying output of scripts; particularly useful on Windows
 # where running QtiPlot from a terminal is inconvenient
@@ -27,13 +27,13 @@
 
 INCLUDEPATH       += ../3rdparty/muParser
 
-INCLUDEPATH       += ../3rdparty/qwtplot3d/include
-INCLUDEPATH       += ../3rdparty/qwt/src
-#INCLUDEPATH       += /usr/include/qwtplot3d
-#INCLUDEPATH       += /usr/include/qwt5
+#INCLUDEPATH       += ../3rdparty/qwtplot3d/include
+#INCLUDEPATH       += ../3rdparty/qwt/src
+INCLUDEPATH       += /usr/include/qwtplot3d
+INCLUDEPATH       += /usr/include/qwt
 
 INCLUDEPATH       += ../3rdparty/liborigin
-INCLUDEPATH       += ../3rdparty/gsl/include
+INCLUDEPATH       += /usr/include/gsl
 INCLUDEPATH       += ../3rdparty/zlib123/include
 
 ##################### 3rd PARTY LIBRARIES SECTION ###########################
@@ -47,18 +47,18 @@
 unix:LIBS         += -L /usr/lib$${libsuff}
 
 # statically link against Qwt(3D) in 3rdparty
-unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
-unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
+#unix:LIBS         += ../3rdparty/qwtplot3d/lib/libqwtplot3d.a
+#unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
 # dynamically link against Qwt(3D) installed system-wide
 # WARNING: make sure they are compiled against Qt4
-#unix:LIBS         += -lqwtplot3d
-#unix:LIBS         += -lqwt
+unix:LIBS         += -lqwtplot3d
+unix:LIBS         += -lqwt
 
 # statically link against GSL in 3rdparty
-unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
+#unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
 #dynamically link against GSL installed system-wide
-#unix:LIBS         += -lgsl -lgslcblas
+unix:LIBS         += -lgsl -lgslcblas
 
 
 ##################### Windows ###############################################
