From 6c3f44fe548becf54366628f87c1235005a22377 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 21 Jul 2018 16:32:16 +0200
Subject: [PATCH] do not use git

---
 CMakeLists.txt | 36 ++++++++++++++++++------------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a7a8bdd..b52098e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,24 +23,24 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 	set(SKIP_INSTALL_ALL ${OSX_PACKAGE})
 endif()
 
-include(FindGit OPTIONAL)
-if (GIT_FOUND)
-	execute_process(
-		COMMAND ${GIT_EXECUTABLE} rev-parse --show-toplevel
-		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-		OUTPUT_VARIABLE LIBAD9361_GIT_REPO
-		OUTPUT_STRIP_TRAILING_WHITESPACE
-	)
-
-if (${LIBAD9361_GIT_REPO} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
-		execute_process(
-			COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
-			WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-			OUTPUT_VARIABLE LIBAD9361_VERSION_GIT
-			OUTPUT_STRIP_TRAILING_WHITESPACE
-		)
-	endif()
-endif()
+#include(FindGit OPTIONAL)
+#if (GIT_FOUND)
+#	execute_process(
+#		COMMAND ${GIT_EXECUTABLE} rev-parse --show-toplevel
+#		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+#		OUTPUT_VARIABLE LIBAD9361_GIT_REPO
+#		OUTPUT_STRIP_TRAILING_WHITESPACE
+#	)
+#
+#if (${LIBAD9361_GIT_REPO} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
+#		execute_process(
+#			COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
+#			WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+#			OUTPUT_VARIABLE LIBAD9361_VERSION_GIT
+#			OUTPUT_STRIP_TRAILING_WHITESPACE
+#		)
+#	endif()
+#endif()
 
 if (NOT LIBAD9361_VERSION_GIT)
 	set(LIBAD9361_VERSION_GIT v${VERSION})
-- 
2.18.0

