From 5c6fbf1d8080ab56483e85bf8e2d314dcd124809 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Mon, 22 Oct 2018 11:54:38 +0200
Subject: [PATCH 2/2] fix OPENSSL_zalloc usage

---
 SWIG/_lib11_compat.i | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SWIG/_lib11_compat.i b/SWIG/_lib11_compat.i
index 1ec42dd..b781952 100644
--- a/SWIG/_lib11_compat.i
+++ b/SWIG/_lib11_compat.i
@@ -8,7 +8,7 @@
  */
 
 %{
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined (LIBRESSL_VERSION_NUMBER)
 
 #include <string.h>
 #include <openssl/engine.h>
-- 
2.23.0

