Also link to -ldl to prevent undefined references.

Origin: Void Linux

--- a/runtimes/libcxxabi/src/CMakeLists.txt
+++ b/runtimes/libcxxabi/src/CMakeLists.txt
@@ -73,6 +73,7 @@
   endif()
 
   add_library_flags_if(LIBCXXABI_HAS_C_LIB c)
+  add_library_flags_if(LIBCXXABI_HAS_C_LIB dl)
 endif()
 
 if (LIBCXXABI_USE_LLVM_UNWINDER)
--- a/runtimes/libcxx/CMakeLists.txt
+++ b/runtimes/libcxx/CMakeLists.txt
@@ -745,6 +745,10 @@
     target_link_libraries(${target} PRIVATE m)
   endif()
 
+  if (LIBCXX_HAS_C_LIB)
+    target_link_libraries(${target} PRIVATE dl)
+  endif()
+
   if (LIBCXX_HAS_RT_LIB)
     target_link_libraries(${target} PRIVATE rt)
   endif()
