diff -Naur gnome-keyring-2.30.1.orig/daemon/gnome-keyring-pkcs11.desktop.in gnome-keyring-2.30.1/daemon/gnome-keyring-pkcs11.desktop.in
--- gnome-keyring-2.30.1.orig/daemon/gnome-keyring-pkcs11.desktop.in	2010-05-07 00:27:39.792857884 -0700
+++ gnome-keyring-2.30.1/daemon/gnome-keyring-pkcs11.desktop.in	2010-05-07 00:32:08.503107570 -0700
@@ -3,7 +3,6 @@
 _Name=Certificate and Key Storage
 _Comment=GNOME Keyring: PKCS#11 Component
 Exec= gnome-keyring-daemon --start --components=pkcs11
-OnlyShowIn=GNOME;LXDE;
 AutostartCondition=GNOME /apps/gnome-keyring/daemon-components/pkcs11
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
diff -Naur gnome-keyring-2.30.1.orig/daemon/gnome-keyring-pkcs11.desktop.in.in gnome-keyring-2.30.1/daemon/gnome-keyring-pkcs11.desktop.in.in
--- gnome-keyring-2.30.1.orig/daemon/gnome-keyring-pkcs11.desktop.in.in	2010-05-07 00:27:39.791857675 -0700
+++ gnome-keyring-2.30.1/daemon/gnome-keyring-pkcs11.desktop.in.in	2010-05-07 00:32:10.361857745 -0700
@@ -3,7 +3,6 @@
 _Name=Certificate and Key Storage
 _Comment=GNOME Keyring: PKCS#11 Component
 Exec=@VALGRIND_RUN@ gnome-keyring-daemon --start --components=pkcs11
-OnlyShowIn=GNOME;LXDE;
 AutostartCondition=GNOME /apps/gnome-keyring/daemon-components/pkcs11
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
diff -Naur gnome-keyring-2.30.1.orig/daemon/gnome-keyring-secrets.desktop.in gnome-keyring-2.30.1/daemon/gnome-keyring-secrets.desktop.in
--- gnome-keyring-2.30.1.orig/daemon/gnome-keyring-secrets.desktop.in	2010-05-07 00:27:39.792857884 -0700
+++ gnome-keyring-2.30.1/daemon/gnome-keyring-secrets.desktop.in	2010-05-07 00:27:52.119857815 -0700
@@ -3,7 +3,6 @@
 _Name=Secret Storage Service
 _Comment=GNOME Keyring: Secret Service
 Exec= gnome-keyring-daemon --start --components=secrets
-OnlyShowIn=GNOME;LXDE;
 AutostartCondition=GNOME /apps/gnome-keyring/daemon-components/secrets
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
diff -Naur gnome-keyring-2.30.1.orig/daemon/gnome-keyring-secrets.desktop.in.in gnome-keyring-2.30.1/daemon/gnome-keyring-secrets.desktop.in.in
--- gnome-keyring-2.30.1.orig/daemon/gnome-keyring-secrets.desktop.in.in	2010-05-07 00:27:39.791857675 -0700
+++ gnome-keyring-2.30.1/daemon/gnome-keyring-secrets.desktop.in.in	2010-05-07 00:27:48.458857814 -0700
@@ -3,7 +3,6 @@
 _Name=Secret Storage Service
 _Comment=GNOME Keyring: Secret Service
 Exec=@VALGRIND_RUN@ gnome-keyring-daemon --start --components=secrets
-OnlyShowIn=GNOME;LXDE;
 AutostartCondition=GNOME /apps/gnome-keyring/daemon-components/secrets
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
diff -Naur gnome-keyring-2.30.1.orig/daemon/gnome-keyring-ssh.desktop.in gnome-keyring-2.30.1/daemon/gnome-keyring-ssh.desktop.in
--- gnome-keyring-2.30.1.orig/daemon/gnome-keyring-ssh.desktop.in	2010-05-07 00:27:39.792857884 -0700
+++ gnome-keyring-2.30.1/daemon/gnome-keyring-ssh.desktop.in	2010-05-07 00:32:14.603857675 -0700
@@ -3,7 +3,6 @@
 _Name=SSH Key Agent
 _Comment=GNOME Keyring: SSH Agent
 Exec= gnome-keyring-daemon --start --components=ssh
-OnlyShowIn=GNOME;LXDE;
 AutostartCondition=GNOME /apps/gnome-keyring/daemon-components/ssh
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
diff -Naur gnome-keyring-2.30.1.orig/daemon/gnome-keyring-ssh.desktop.in.in gnome-keyring-2.30.1/daemon/gnome-keyring-ssh.desktop.in.in
--- gnome-keyring-2.30.1.orig/daemon/gnome-keyring-ssh.desktop.in.in	2010-05-07 00:27:39.790857884 -0700
+++ gnome-keyring-2.30.1/daemon/gnome-keyring-ssh.desktop.in.in	2010-05-07 00:32:16.422857745 -0700
@@ -3,7 +3,6 @@
 _Name=SSH Key Agent
 _Comment=GNOME Keyring: SSH Agent
 Exec=@VALGRIND_RUN@ gnome-keyring-daemon --start --components=ssh
-OnlyShowIn=GNOME;LXDE;
 AutostartCondition=GNOME /apps/gnome-keyring/daemon-components/ssh
 X-GNOME-Autostart-Phase=Initialization
 X-GNOME-AutoRestart=false
