--- module/embed/gecko/gtkmozembed/gtkmozembed.h	2008/08/03 11:35:25	3508
+++ module/embed/gecko/gtkmozembed/gtkmozembed.h	2008/08/05 00:10:37	3509
@@ -30,10 +30,10 @@
 G_BEGIN_DECLS
 
 #define GTK_TYPE_MOZ_EMBED             (gtk_moz_embed_get_type())
-#define GTK_MOZ_EMBED(obj)             GTK_CHECK_CAST((obj), GTK_TYPE_MOZ_EMBED, GtkMozEmbed)
-#define GTK_MOZ_EMBED_CLASS(klass)     GTK_CHEK_CLASS_CAST((klass), GTK_TYPE_MOZ_EMBED, GtkMozEmbedClass)
-#define GTK_IS_MOZ_EMBED(obj)          GTK_CHECK_TYPE((obj), GTK_TYPE_MOZ_EMBED)
-#define GTK_IS_MOZ_EMBED_CLASS(klass)  GTK_CHECK_CLASS_TYPE((klass), GTK_TYPE_MOZ_EMBED)
+#define GTK_MOZ_EMBED(obj)             G_TYPE_CHECK_INSTANCE_CAST((obj), GTK_TYPE_MOZ_EMBED, GtkMozEmbed)
+#define GTK_MOZ_EMBED_CLASS(klass)     G_TYPE_CHEK_CLASS_CAST((klass), GTK_TYPE_MOZ_EMBED, GtkMozEmbedClass)
+#define GTK_IS_MOZ_EMBED(obj)          G_TYPE_CHECK_INSTANCE_TYPE((obj), GTK_TYPE_MOZ_EMBED)
+#define GTK_IS_MOZ_EMBED_CLASS(klass)  G_TYPE_CHECK_CLASS_TYPE((klass), GTK_TYPE_MOZ_EMBED)
 
 typedef struct _GtkMozEmbed      GtkMozEmbed;
 typedef struct _GtkMozEmbedClass GtkMozEmbedClass;
@@ -117,10 +117,10 @@
 #define GTK_TYPE_MOZ_EMBED_CHROME_FLAGS \
              (gtk_moz_embed_chrome_flags_get_type())
 
-GtkType      gtk_moz_embed_progress_flags_get_type (void) G_GNUC_CONST;
-GtkType      gtk_moz_embed_status_enums_get_type (void) G_GNUC_CONST;
-GtkType      gtk_moz_embed_reload_flags_get_type (void) G_GNUC_CONST;
-GtkType      gtk_moz_embed_chrome_flags_get_type (void) G_GNUC_CONST;
+GType      gtk_moz_embed_progress_flags_get_type (void) G_GNUC_CONST;
+GType      gtk_moz_embed_status_enums_get_type (void) G_GNUC_CONST;
+GType      gtk_moz_embed_reload_flags_get_type (void) G_GNUC_CONST;
+GType      gtk_moz_embed_chrome_flags_get_type (void) G_GNUC_CONST;
 
 /* These are straight out of nsIWebProgressListener.h */

--- src/kz-download-box.h	2008/08/03 11:35:25	3508
+++ src/kz-download-box.h	2008/08/05 00:10:37	3509
@@ -28,11 +28,11 @@
 G_BEGIN_DECLS
 
 #define KZ_TYPE_DOWNLOAD_BOX		(kz_download_box_get_type ())
-#define KZ_DOWNLOAD_BOX(obj)		(GTK_CHECK_CAST ((obj), KZ_TYPE_DOWNLOAD_BOX, KzDownloadBox))
-#define KZ_DOWNLOAD_BOX_CLASS(klass)	(GTK_CHECK_CLASS_CAST ((klass), KZ_TYPE_DOWNLOAD_BOX, KzDownloadBoxClass))
-#define KZ_IS_DOWNLOAD_BOX(obj)		(GTK_CHECK_TYPE ((obj), KZ_TYPE_DOWNLOAD_BOX))
-#define KZ_IS_DOWNLOAD_BOX_CLASS(klass)	(GTK_CHECK_CLASS_TYPE ((klass), KZ_TYPE_DOWNLOAD_BOX))
-#define KZ_DOWNLOAD_BOX_GET_CLASS(obj)	(GTK_CHECK_GET_CLASS ((obj), KZ_TYPE_DOWNLOAD_BOX, KzDownloadBoxClass))
+#define KZ_DOWNLOAD_BOX(obj)		(G_TYPE_CHECK_INSTANCE_CAST ((obj), KZ_TYPE_DOWNLOAD_BOX, KzDownloadBox))
+#define KZ_DOWNLOAD_BOX_CLASS(klass)	(G_TYPE_CHECK_CLASS_CAST ((klass), KZ_TYPE_DOWNLOAD_BOX, KzDownloadBoxClass))
+#define KZ_IS_DOWNLOAD_BOX(obj)		(G_TYPE_CHECK_INSTANCE_TYPE ((obj), KZ_TYPE_DOWNLOAD_BOX))
+#define KZ_IS_DOWNLOAD_BOX_CLASS(klass)	(G_TYPE_CHECK_CLASS_TYPE ((klass), KZ_TYPE_DOWNLOAD_BOX))
+#define KZ_DOWNLOAD_BOX_GET_CLASS(obj)	(G_TYPE_INSTANCE_GET_CLASS ((obj), KZ_TYPE_DOWNLOAD_BOX, KzDownloadBoxClass))
 
 typedef struct _KzDownloadBox      KzDownloadBox;
 typedef struct _KzDownloadBoxClass KzDownloadBoxClass;

--- src/kz-feed-info.h	2008/08/03 11:35:25	3508
+++ src/kz-feed-info.h	2008/08/05 00:10:37	3509
@@ -8,11 +8,11 @@
 G_BEGIN_DECLS
 
 #define KZ_TYPE_FEED_INFO		(kz_feed_info_get_type ())
-#define KZ_FEED_INFO(obj)		(GTK_CHECK_CAST ((obj), KZ_TYPE_FEED_INFO, KzFeedInfo))
-#define KZ_FEED_INFO_CLASS(klass)	(GTK_CHECK_CLASS_CAST ((klass), KZ_TYPE_FEED_INFO, KzFeedInfoClass))
-#define KZ_IS_FEED_INFO(obj)		(GTK_CHECK_TYPE ((obj), KZ_TYPE_FEED_INFO))
-#define KZ_IS_FEED_INFO_CLASS(klass)	(GTK_CHECK_CLASS_TYPE ((klass), KZ_TYPE_FEED_INFO))
-#define KZ_FEED_INFO_GET_CLASS(obj)	(GTK_CHECK_GET_CLASS ((obj), KZ_TYPE_FEED_INFO, KzFeedInfoClass))
+#define KZ_FEED_INFO(obj)		(G_TYPE_CHECK_INSTANCE_CAST ((obj), KZ_TYPE_FEED_INFO, KzFeedInfo))
+#define KZ_FEED_INFO_CLASS(klass)	(G_TYPE_CHECK_CLASS_CAST ((klass), KZ_TYPE_FEED_INFO, KzFeedInfoClass))
+#define KZ_IS_FEED_INFO(obj)		(G_TYPE_CHECK_INSTANCE_TYPE ((obj), KZ_TYPE_FEED_INFO))
+#define KZ_IS_FEED_INFO_CLASS(klass)	(G_TYPE_CHECK_CLASS_TYPE ((klass), KZ_TYPE_FEED_INFO))
+#define KZ_FEED_INFO_GET_CLASS(obj)	(G_TYPE_INSTANCE_GET_CLASS ((obj), KZ_TYPE_FEED_INFO, KzFeedInfoClass))
 
 typedef struct _KzFeedInfo      KzFeedInfo;
 typedef struct _KzFeedInfoClass KzFeedInfoClass;

--- src/kz-notebook.h	2008/08/03 11:35:25	3508
+++ src/kz-notebook.h	2008/08/05 00:10:37	3509
@@ -9,11 +9,11 @@
 G_BEGIN_DECLS
 
 #define KZ_TYPE_NOTEBOOK		(kz_notebook_get_type ())
-#define KZ_NOTEBOOK(obj)		(GTK_CHECK_CAST ((obj), KZ_TYPE_NOTEBOOK, KzNotebook))
-#define KZ_NOTEBOOK_CLASS(klass)	(GTK_CHECK_CLASS_CAST ((klass), KZ_TYPE_NOTEBOOK, KzNotebookClass))
-#define KZ_IS_NOTEBOOK(obj)		(GTK_CHECK_TYPE ((obj), KZ_TYPE_NOTEBOOK))
-#define KZ_IS_NOTEBOOK_CLASS(klass)	(GTK_CHECK_CLASS_TYPE ((klass), KZ_TYPE_NOTEBOOK))
-#define KZ_NOTEBOOK_GET_CLASS(obj)	(GTK_CHECK_GET_CLASS ((obj), KZ_TYPE_NOTEBOOK, KzNotebookClass))
+#define KZ_NOTEBOOK(obj)		(G_TYPE_CHECK_INSTANCE_CAST ((obj), KZ_TYPE_NOTEBOOK, KzNotebook))
+#define KZ_NOTEBOOK_CLASS(klass)	(G_TYPE_CHECK_CLASS_CAST ((klass), KZ_TYPE_NOTEBOOK, KzNotebookClass))
+#define KZ_IS_NOTEBOOK(obj)		(G_TYPE_CHECK_INSTANCE_TYPE ((obj), KZ_TYPE_NOTEBOOK))
+#define KZ_IS_NOTEBOOK_CLASS(klass)	(G_TYPE_CHECK_CLASS_TYPE ((klass), KZ_TYPE_NOTEBOOK))
+#define KZ_NOTEBOOK_GET_CLASS(obj)	(G_TYPE_INSTANCE_GET_CLASS ((obj), KZ_TYPE_NOTEBOOK, KzNotebookClass))
 
 typedef struct _KzNotebook      KzNotebook;
 typedef struct _KzNotebookClass KzNotebookClass;

--- src/kz-statusbar.h	2008/08/03 11:35:25	3508
+++ src/kz-statusbar.h	2008/08/05 00:10:37	3509
@@ -8,11 +8,11 @@
 G_BEGIN_DECLS
 
 #define KZ_TYPE_STATUSBAR		(kz_statusbar_get_type ())
-#define KZ_STATUSBAR(obj)		(GTK_CHECK_CAST ((obj), KZ_TYPE_STATUSBAR, KzStatusbar))
-#define KZ_STATUSBAR_CLASS(klass)	(GTK_CHECK_CLASS_CAST ((klass), KZ_TYPE_STATUSBAR, KzStatusbarClass))
-#define KZ_IS_STATUSBAR(obj)		(GTK_CHECK_TYPE ((obj), KZ_TYPE_STATUSBAR))
-#define KZ_IS_STATUSBAR_CLASS(klass)	(GTK_CHECK_CLASS_TYPE ((klass), KZ_TYPE_STATUSBAR))
-#define KZ_STATUSBAR_GET_CLASS(obj)	(GTK_CHECK_GET_CLASS ((obj), KZ_TYPE_STATUSBAR, KzStatusbarClass))
+#define KZ_STATUSBAR(obj)		(G_TYPE_CHECK_INSTANCE_CAST ((obj), KZ_TYPE_STATUSBAR, KzStatusbar))
+#define KZ_STATUSBAR_CLASS(klass)	(G_TYPE_CHECK_CLASS_CAST ((klass), KZ_TYPE_STATUSBAR, KzStatusbarClass))
+#define KZ_IS_STATUSBAR(obj)		(G_TYPE_CHECK_INSTANCE_TYPE ((obj), KZ_TYPE_STATUSBAR))
+#define KZ_IS_STATUSBAR_CLASS(klass)	(G_TYPE_CHECK_CLASS_TYPE ((klass), KZ_TYPE_STATUSBAR))
+#define KZ_STATUSBAR_GET_CLASS(obj)	(G_TYPE_INSTANCE_GET_CLASS ((obj), KZ_TYPE_STATUSBAR, KzStatusbarClass))
 
 typedef struct _KzStatusbar      KzStatusbar;
 typedef struct _KzStatusbarClass KzStatusbarClass;



