From 9d14633496ae504557c95d124b97a71147635f04 Mon Sep 17 00:00:00 2001
From: Michael Drake <tlsa@netsurf-browser.org>
Date: Fri, 24 May 2024 21:30:56 +0100
Subject: Squash -Wcalloc-transposed-args (gcc-14)

Origin: Upstream
Upstream-Status: Backport
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 src/svgtiny.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/svgtiny.c b/src/svgtiny.c
index ee0c59c..6859fbd 100644
--- a/src/svgtiny.c
+++ b/src/svgtiny.c
@@ -586,7 +586,7 @@ struct svgtiny_diagram *svgtiny_create(void)
 {
 	struct svgtiny_diagram *diagram;
 
-	diagram = calloc(sizeof(*diagram), 1);
+	diagram = calloc(1, sizeof(*diagram));
 	if (!diagram)
 		return 0;
 
-- 
cgit v1.2.3

