diff -Naur -u linux-5.6/arch/alpha/lib/Makefile linux-5.6.noWerror/arch/alpha/lib/Makefile
--- linux-5.6/arch/alpha/lib/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/alpha/lib/Makefile	2020-05-10 14:03:43.429793471 +0200
@@ -4,7 +4,7 @@
 #
 
 asflags-y := $(KBUILD_CFLAGS)
-ccflags-y := -Werror
+ccflags-y := 
 
 # Many of these routines have implementations tuned for ev6.
 # Choose them iff we're targeting ev6 specifically.
diff -Naur -u linux-5.6/arch/alpha/mm/Makefile linux-5.6.noWerror/arch/alpha/mm/Makefile
--- linux-5.6/arch/alpha/mm/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/alpha/mm/Makefile	2020-05-10 14:03:43.433126804 +0200
@@ -3,7 +3,7 @@
 # Makefile for the linux alpha-specific parts of the memory manager.
 #
 
-ccflags-y := -Werror
+ccflags-y := 
 
 obj-y	:= init.o fault.o
 
diff -Naur -u linux-5.6/arch/alpha/oprofile/Makefile linux-5.6.noWerror/arch/alpha/oprofile/Makefile
--- linux-5.6/arch/alpha/oprofile/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/alpha/oprofile/Makefile	2020-05-10 14:03:43.433126804 +0200
@@ -1,5 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0
-ccflags-y := -Werror -Wno-sign-compare
+ccflags-y := -Wno-sign-compare
 
 obj-$(CONFIG_OPROFILE) += oprofile.o
 
diff -Naur -u linux-5.6/arch/arm64/kernel/vdso32/Makefile linux-5.6.noWerror/arch/arm64/kernel/vdso32/Makefile
--- linux-5.6/arch/arm64/kernel/vdso32/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/arm64/kernel/vdso32/Makefile	2020-05-10 14:03:43.536460135 +0200
@@ -72,7 +72,6 @@
 # KBUILD_CFLAGS from top-level Makefile
 VDSO_CFLAGS += -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
                -fno-strict-aliasing -fno-common \
-               -Werror-implicit-function-declaration \
                -Wno-format-security \
                -std=gnu89
 VDSO_CFLAGS  += -O2
@@ -80,9 +79,6 @@
 VDSO_CFLAGS += $(call cc32-option,-Wdeclaration-after-statement,)
 VDSO_CFLAGS += $(call cc32-option,-Wno-pointer-sign)
 VDSO_CFLAGS += $(call cc32-option,-fno-strict-overflow)
-VDSO_CFLAGS += $(call cc32-option,-Werror=strict-prototypes)
-VDSO_CFLAGS += $(call cc32-option,-Werror=date-time)
-VDSO_CFLAGS += $(call cc32-option,-Werror=incompatible-pointer-types)
 
 # The 32-bit compiler does not provide 128-bit integers, which are used in
 # some headers that are indirectly included from the vDSO code.
diff -Naur -u linux-5.6/arch/mips/Kbuild linux-5.6.noWerror/arch/mips/Kbuild
--- linux-5.6/arch/mips/Kbuild	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/mips/Kbuild	2020-05-10 14:03:43.563126801 +0200
@@ -3,7 +3,7 @@
 # -Werror can be disabled for specific files using:
 # CFLAGS_<file.o> := -Wno-error
 ifeq ($(W),)
-subdir-ccflags-y := -Werror
+subdir-ccflags-y := 
 endif
 
 # platform specific definitions
diff -Naur -u linux-5.6/arch/powerpc/Kbuild linux-5.6.noWerror/arch/powerpc/Kbuild
--- linux-5.6/arch/powerpc/Kbuild	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/powerpc/Kbuild	2020-05-10 14:03:43.606460134 +0200
@@ -1,5 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0
-subdir-ccflags-$(CONFIG_PPC_WERROR) := -Werror
+subdir-ccflags-$(CONFIG_PPC_WERROR) := 
 
 obj-y += kernel/
 obj-y += mm/
diff -Naur -u linux-5.6/arch/sh/cchips/hd6446x/Makefile linux-5.6.noWerror/arch/sh/cchips/hd6446x/Makefile
--- linux-5.6/arch/sh/cchips/hd6446x/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/sh/cchips/hd6446x/Makefile	2020-05-10 14:03:43.653126799 +0200
@@ -1,4 +1,4 @@
 # SPDX-License-Identifier: GPL-2.0-only
 obj-$(CONFIG_HD64461)	+= hd64461.o
 
-ccflags-y := -Werror
+ccflags-y := 
diff -Naur -u linux-5.6/arch/sh/kernel/Makefile linux-5.6.noWerror/arch/sh/kernel/Makefile
--- linux-5.6/arch/sh/kernel/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/sh/kernel/Makefile	2020-05-10 14:03:43.656460133 +0200
@@ -48,4 +48,4 @@
 obj-$(CONFIG_DMA_NONCOHERENT)	+= dma-coherent.o
 obj-$(CONFIG_HAVE_HW_BREAKPOINT)		+= hw_breakpoint.o
 
-ccflags-y := -Werror
+ccflags-y := 
diff -Naur -u linux-5.6/arch/sh/lib/Makefile linux-5.6.noWerror/arch/sh/lib/Makefile
--- linux-5.6/arch/sh/lib/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/sh/lib/Makefile	2020-05-10 14:03:43.659793466 +0200
@@ -31,4 +31,4 @@
 lib-$(CONFIG_MCOUNT)		+= mcount.o
 lib-y				+= $(memcpy-y) $(memset-y) $(udivsi3-y)
 
-ccflags-y := -Werror
+ccflags-y := 
diff -Naur -u linux-5.6/arch/sh/mm/Makefile linux-5.6.noWerror/arch/sh/mm/Makefile
--- linux-5.6/arch/sh/mm/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/sh/mm/Makefile	2020-05-10 14:03:43.659793466 +0200
@@ -71,4 +71,4 @@
 	-ffixed-r60 -ffixed-r61 -ffixed-r62 \
 	-fomit-frame-pointer
 
-ccflags-y := -Werror
+ccflags-y := 
diff -Naur -u linux-5.6/arch/sparc/kernel/Makefile linux-5.6.noWerror/arch/sparc/kernel/Makefile
--- linux-5.6/arch/sparc/kernel/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/sparc/kernel/Makefile	2020-05-10 14:03:43.666460132 +0200
@@ -5,7 +5,7 @@
 #
 
 asflags-y := -ansi
-ccflags-y := -Werror
+ccflags-y := 
 
 extra-y     := head_$(BITS).o
 
diff -Naur -u linux-5.6/arch/sparc/lib/Makefile linux-5.6.noWerror/arch/sparc/lib/Makefile
--- linux-5.6/arch/sparc/lib/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/sparc/lib/Makefile	2020-05-10 14:03:43.669793466 +0200
@@ -3,7 +3,7 @@
 #
 
 asflags-y := -ansi -DST_DIV0=0x02
-ccflags-y := -Werror
+ccflags-y := 
 
 lib-$(CONFIG_SPARC32) += ashrdi3.o
 lib-$(CONFIG_SPARC32) += memcpy.o memset.o
diff -Naur -u linux-5.6/arch/sparc/mm/Makefile linux-5.6.noWerror/arch/sparc/mm/Makefile
--- linux-5.6/arch/sparc/mm/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/sparc/mm/Makefile	2020-05-10 14:03:43.669793466 +0200
@@ -3,7 +3,7 @@
 #
 
 asflags-y := -ansi
-ccflags-y := -Werror
+ccflags-y := 
 
 obj-$(CONFIG_SPARC64)   += ultra.o tlb.o tsb.o
 obj-y                   += fault_$(BITS).o
diff -Naur -u linux-5.6/arch/sparc/prom/Makefile linux-5.6.noWerror/arch/sparc/prom/Makefile
--- linux-5.6/arch/sparc/prom/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/sparc/prom/Makefile	2020-05-10 14:03:43.673126799 +0200
@@ -3,7 +3,7 @@
 # Linux.
 #
 asflags := -ansi
-ccflags := -Werror
+ccflags := 
 
 lib-y                 := bootstr_$(BITS).o
 lib-y                 += init_$(BITS).o
diff -Naur -u linux-5.6/arch/x86/kvm/Makefile linux-5.6.noWerror/arch/x86/kvm/Makefile
--- linux-5.6/arch/x86/kvm/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/arch/x86/kvm/Makefile	2020-05-10 14:03:43.693126799 +0200
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 
 ccflags-y += -Iarch/x86/kvm
-ccflags-$(CONFIG_KVM_WERROR) += -Werror
+ccflags-$(CONFIG_KVM_WERROR) += 
 
 KVM := ../../../virt/kvm
 
diff -Naur -u linux-5.6/drivers/gpu/drm/i915/Makefile linux-5.6.noWerror/drivers/gpu/drm/i915/Makefile
--- linux-5.6/drivers/gpu/drm/i915/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/drivers/gpu/drm/i915/Makefile	2020-05-10 14:03:43.893126795 +0200
@@ -22,7 +22,7 @@
 subdir-ccflags-y += $(call cc-disable-warning, sometimes-uninitialized)
 subdir-ccflags-y += $(call cc-disable-warning, initializer-overrides)
 subdir-ccflags-y += $(call cc-disable-warning, uninitialized)
-subdir-ccflags-$(CONFIG_DRM_I915_WERROR) += -Werror
+subdir-ccflags-$(CONFIG_DRM_I915_WERROR) += 
 
 # Fine grained warnings disable
 CFLAGS_i915_pci.o = $(call cc-disable-warning, override-init)
diff -Naur -u linux-5.6/drivers/gpu/drm/tilcdc/Makefile linux-5.6.noWerror/drivers/gpu/drm/tilcdc/Makefile
--- linux-5.6/drivers/gpu/drm/tilcdc/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/drivers/gpu/drm/tilcdc/Makefile	2020-05-10 14:03:43.946460127 +0200
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
 ifeq (, $(findstring -W,$(EXTRA_CFLAGS)))
-	ccflags-y += -Werror
+	ccflags-y += 
 endif
 
 tilcdc-y := \
diff -Naur -u linux-5.6/drivers/misc/cxl/Makefile linux-5.6.noWerror/drivers/misc/cxl/Makefile
--- linux-5.6/drivers/misc/cxl/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/drivers/misc/cxl/Makefile	2020-05-10 14:03:44.063126791 +0200
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
 ccflags-y			:= $(call cc-disable-warning, unused-const-variable)
-ccflags-$(CONFIG_PPC_WERROR)	+= -Werror
+ccflags-$(CONFIG_PPC_WERROR)	+= 
 
 cxl-y				+= main.o file.o irq.o fault.o native.o
 cxl-y				+= context.o sysfs.o pci.o trace.o
diff -Naur -u linux-5.6/drivers/misc/ocxl/Makefile linux-5.6.noWerror/drivers/misc/ocxl/Makefile
--- linux-5.6/drivers/misc/ocxl/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/drivers/misc/ocxl/Makefile	2020-05-10 14:03:44.069793458 +0200
@@ -1,5 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0+
-ccflags-$(CONFIG_PPC_WERROR)	+= -Werror
+ccflags-$(CONFIG_PPC_WERROR)	+= 
 
 ocxl-y				+= main.o pci.o config.o file.o pasid.o mmio.o
 ocxl-y				+= link.o context.o afu_irq.o sysfs.o trace.o
diff -Naur -u linux-5.6/drivers/scsi/aic7xxx/Makefile linux-5.6.noWerror/drivers/scsi/aic7xxx/Makefile
--- linux-5.6/drivers/scsi/aic7xxx/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/drivers/scsi/aic7xxx/Makefile	2020-05-10 14:03:44.219793454 +0200
@@ -35,7 +35,7 @@
 						   aic79xx_osm_pci.o
 
 ifdef WARNINGS_BECOME_ERRORS
-ccflags-y += -Werror
+ccflags-y += 
 endif
 
 # Files generated that shall be removed upon make clean
diff -Naur -u linux-5.6/drivers/scsi/lpfc/Makefile linux-5.6.noWerror/drivers/scsi/lpfc/Makefile
--- linux-5.6/drivers/scsi/lpfc/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/drivers/scsi/lpfc/Makefile	2020-05-10 14:03:44.229793454 +0200
@@ -25,7 +25,7 @@
 ccflags-$(GCOV) += -O0
 
 ifdef WARNINGS_BECOME_ERRORS
-ccflags-y += -Werror
+ccflags-y += 
 endif
 
 obj-$(CONFIG_SCSI_LPFC) := lpfc.o
diff -Naur -u linux-5.6/Makefile linux-5.6.noWerror/Makefile
--- linux-5.6/Makefile	2020-05-10 14:04:16.183126128 +0200
+++ linux-5.6.noWerror/Makefile	2020-05-10 14:03:44.326460119 +0200
@@ -456,9 +456,8 @@
 		$(USERINCLUDE)
 
 KBUILD_AFLAGS   := -D__ASSEMBLY__ -fno-PIE
-KBUILD_CFLAGS   := -Wall -Wundef -Werror=strict-prototypes -Wno-trigraphs \
+KBUILD_CFLAGS   := -Wall -Wundef -Wno-trigraphs \
 		   -fno-strict-aliasing -fno-common -fshort-wchar -fno-PIE \
-		   -Werror=implicit-function-declaration -Werror=implicit-int \
 		   -Wno-format-security \
 		   -std=gnu89
 KBUILD_CPPFLAGS := -D__KERNEL__
@@ -538,7 +537,7 @@
 ifeq ($(shell $(AS) --version 2>&1 | head -n 1 | grep clang),)
 CLANG_FLAGS	+= -no-integrated-as
 endif
-CLANG_FLAGS	+= -Werror=unknown-warning-option
+CLANG_FLAGS	+= 
 KBUILD_CFLAGS	+= $(CLANG_FLAGS)
 KBUILD_AFLAGS	+= $(CLANG_FLAGS)
 export CLANG_FLAGS
@@ -880,13 +879,10 @@
 KBUILD_CFLAGS   += $(call cc-option,-fconserve-stack)
 
 # Prohibit date/time macros, which would make the build non-deterministic
-KBUILD_CFLAGS   += $(call cc-option,-Werror=date-time)
 
 # enforce correct pointer usage
-KBUILD_CFLAGS   += $(call cc-option,-Werror=incompatible-pointer-types)
 
 # Require designated initializers for all marked structures
-KBUILD_CFLAGS   += $(call cc-option,-Werror=designated-init)
 
 # change __FILE__ to the relative path from the srctree
 KBUILD_CFLAGS	+= $(call cc-option,-fmacro-prefix-map=$(srctree)/=)
diff -Naur -u linux-5.6/scripts/Kbuild.include linux-5.6.noWerror/scripts/Kbuild.include
--- linux-5.6/scripts/Kbuild.include	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/scripts/Kbuild.include	2020-05-10 14:03:44.509793448 +0200
@@ -116,7 +116,7 @@
 # __cc-option
 # Usage: MY_CFLAGS += $(call __cc-option,$(CC),$(MY_CFLAGS),-march=winchip-c6,-march=i586)
 __cc-option = $(call try-run,\
-	$(1) -Werror $(2) $(3) -c -x c /dev/null -o "$$TMP",$(3),$(4))
+	$(1) $(2) $(3) -c -x c /dev/null -o "$$TMP",$(3),$(4))
 
 # Do not attempt to build with gcc plugins during cc-option tests.
 # (And this uses delayed resolution so the flags will be up to date.)
@@ -131,12 +131,12 @@
 # cc-option-yn
 # Usage: flag := $(call cc-option-yn,-march=winchip-c6)
 cc-option-yn = $(call try-run,\
-	$(CC) -Werror $(KBUILD_CPPFLAGS) $(CC_OPTION_CFLAGS) $(1) -c -x c /dev/null -o "$$TMP",y,n)
+	$(CC) $(KBUILD_CPPFLAGS) $(CC_OPTION_CFLAGS) $(1) -c -x c /dev/null -o "$$TMP",y,n)
 
 # cc-disable-warning
 # Usage: cflags-y += $(call cc-disable-warning,unused-but-set-variable)
 cc-disable-warning = $(call try-run,\
-	$(CC) -Werror $(KBUILD_CPPFLAGS) $(CC_OPTION_CFLAGS) -W$(strip $(1)) -c -x c /dev/null -o "$$TMP",-Wno-$(strip $(1)))
+	$(CC) $(KBUILD_CPPFLAGS) $(CC_OPTION_CFLAGS) -W$(strip $(1)) -c -x c /dev/null -o "$$TMP",-Wno-$(strip $(1)))
 
 # cc-ifversion
 # Usage:  EXTRA_CFLAGS += $(call cc-ifversion, -lt, 0402, -O1)
diff -Naur -u linux-5.6/scripts/Kconfig.include linux-5.6.noWerror/scripts/Kconfig.include
--- linux-5.6/scripts/Kconfig.include	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/scripts/Kconfig.include	2020-05-10 14:03:44.509793448 +0200
@@ -25,7 +25,7 @@
 
 # $(cc-option,<flag>)
 # Return y if the compiler supports <flag>, n otherwise
-cc-option = $(success,$(CC) -Werror $(CLANG_FLAGS) $(1) -S -x c /dev/null -o /dev/null)
+cc-option = $(success,$(CC) $(CLANG_FLAGS) $(1) -S -x c /dev/null -o /dev/null)
 
 # $(ld-option,<flag>)
 # Return y if the linker supports <flag>, n otherwise
@@ -48,6 +48,6 @@
 # machine bit flags
 #  $(m32-flag): -m32 if the compiler supports it, or an empty string otherwise.
 #  $(m64-flag): -m64 if the compiler supports it, or an empty string otherwise.
-cc-option-bit = $(if-success,$(CC) -Werror $(1) -E -x c /dev/null -o /dev/null,$(1))
+cc-option-bit = $(if-success,$(CC) $(1) -E -x c /dev/null -o /dev/null,$(1))
 m32-flag := $(cc-option-bit,-m32)
 m64-flag := $(cc-option-bit,-m64)
diff -Naur -u linux-5.6/tools/build/feature/Makefile linux-5.6.noWerror/tools/build/feature/Makefile
--- linux-5.6/tools/build/feature/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/build/feature/Makefile	2020-05-10 14:03:44.573126781 +0200
@@ -78,10 +78,10 @@
 
 all: $(FILES)
 
-__BUILD = $(CC) $(CFLAGS) -MD -Wall -Werror -o $@ $(patsubst %.bin,%.c,$(@F)) $(LDFLAGS)
+__BUILD = $(CC) $(CFLAGS) -MD -Wall -o $@ $(patsubst %.bin,%.c,$(@F)) $(LDFLAGS)
   BUILD = $(__BUILD) > $(@:.bin=.make.output) 2>&1
 
-__BUILDXX = $(CXX) $(CXXFLAGS) -MD -Wall -Werror -o $@ $(patsubst %.bin,%.cpp,$(@F)) $(LDFLAGS)
+__BUILDXX = $(CXX) $(CXXFLAGS) -MD -Wall -o $@ $(patsubst %.bin,%.cpp,$(@F)) $(LDFLAGS)
   BUILDXX = $(__BUILDXX) > $(@:.bin=.make.output) 2>&1
 
 ###############################
@@ -230,10 +230,10 @@
 	$(BUILD)
 
 $(OUTPUT)test-libbfd-liberty.bin:
-	$(CC) $(CFLAGS) -Wall -Werror -o $@ test-libbfd.c -DPACKAGE='"perf"' $(LDFLAGS) -lbfd -ldl -liberty
+	$(CC) $(CFLAGS) -Wall -o $@ test-libbfd.c -DPACKAGE='"perf"' $(LDFLAGS) -lbfd -ldl -liberty
 
 $(OUTPUT)test-libbfd-liberty-z.bin:
-	$(CC) $(CFLAGS) -Wall -Werror -o $@ test-libbfd.c -DPACKAGE='"perf"' $(LDFLAGS) -lbfd -ldl -liberty -lz
+	$(CC) $(CFLAGS) -Wall -o $@ test-libbfd.c -DPACKAGE='"perf"' $(LDFLAGS) -lbfd -ldl -liberty -lz
 
 $(OUTPUT)test-cplus-demangle.bin:
 	$(BUILD) -liberty
diff -Naur -u linux-5.6/tools/lib/api/Makefile linux-5.6.noWerror/tools/lib/api/Makefile
--- linux-5.6/tools/lib/api/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/lib/api/Makefile	2020-05-10 14:03:44.576460114 +0200
@@ -34,7 +34,7 @@
 
 # Treat warnings as errors unless directed not to
 ifneq ($(WERROR),0)
-  CFLAGS += -Werror
+  CFLAGS += 
 endif
 
 CFLAGS += -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
diff -Naur -u linux-5.6/tools/lib/bpf/Makefile linux-5.6.noWerror/tools/lib/bpf/Makefile
--- linux-5.6/tools/lib/bpf/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/lib/bpf/Makefile	2020-05-10 14:03:44.576460114 +0200
@@ -108,7 +108,7 @@
 
 # Append required CFLAGS
 override CFLAGS += $(EXTRA_WARNINGS)
-override CFLAGS += -Werror -Wall
+override CFLAGS += -Wall
 override CFLAGS += -fPIC
 override CFLAGS += $(INCLUDES)
 override CFLAGS += -fvisibility=hidden
diff -Naur -u linux-5.6/tools/lib/perf/Makefile linux-5.6.noWerror/tools/lib/perf/Makefile
--- linux-5.6/tools/lib/perf/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/lib/perf/Makefile	2020-05-10 14:03:44.579793447 +0200
@@ -69,7 +69,7 @@
 
 # Append required CFLAGS
 override CFLAGS += $(EXTRA_WARNINGS)
-override CFLAGS += -Werror -Wall
+override CFLAGS += -Wall
 override CFLAGS += -fPIC
 override CFLAGS += $(INCLUDES)
 override CFLAGS += -fvisibility=hidden
diff -Naur -u linux-5.6/tools/lib/subcmd/Makefile linux-5.6.noWerror/tools/lib/subcmd/Makefile
--- linux-5.6/tools/lib/subcmd/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/lib/subcmd/Makefile	2020-05-10 14:03:44.579793447 +0200
@@ -37,7 +37,7 @@
 
 # Treat warnings as errors unless directed not to
 ifneq ($(WERROR),0)
-  CFLAGS += -Werror
+  CFLAGS += 
 endif
 
 CFLAGS += -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE
diff -Naur -u linux-5.6/tools/objtool/Makefile linux-5.6.noWerror/tools/objtool/Makefile
--- linux-5.6/tools/objtool/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/objtool/Makefile	2020-05-10 14:03:44.583126780 +0200
@@ -31,7 +31,7 @@
 	    -I$(srctree)/tools/arch/$(HOSTARCH)/include/uapi \
 	    -I$(srctree)/tools/arch/$(SRCARCH)/include
 WARNINGS := $(EXTRA_WARNINGS) -Wno-switch-default -Wno-switch-enum -Wno-packed
-CFLAGS   := -Werror $(WARNINGS) $(KBUILD_HOSTCFLAGS) -g $(INCLUDES) $(LIBELF_FLAGS)
+CFLAGS   := $(WARNINGS) $(KBUILD_HOSTCFLAGS) -g $(INCLUDES) $(LIBELF_FLAGS)
 LDFLAGS  += $(LIBELF_LIBS) $(LIBSUBCMD) $(KBUILD_HOSTLDFLAGS)
 
 # Allow old libelf to be used:
diff -Naur -u linux-5.6/tools/perf/Makefile.config linux-5.6.noWerror/tools/perf/Makefile.config
--- linux-5.6/tools/perf/Makefile.config	2020-05-10 14:04:16.329792792 +0200
+++ linux-5.6.noWerror/tools/perf/Makefile.config	2020-05-10 14:03:44.609793446 +0200
@@ -188,8 +188,8 @@
 
 # Treat warnings as errors unless directed not to
 ifneq ($(WERROR),0)
-  CORE_CFLAGS += -Werror
-  CXXFLAGS += -Werror
+  CORE_CFLAGS += 
+  CXXFLAGS += 
 endif
 
 ifndef DEBUG
diff -Naur -u linux-5.6/tools/perf/Makefile.perf linux-5.6.noWerror/tools/perf/Makefile.perf
--- linux-5.6/tools/perf/Makefile.perf	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/perf/Makefile.perf	2020-05-10 14:03:44.583126780 +0200
@@ -729,12 +729,12 @@
 
 ifndef NO_PERF_READ_VDSO32
 $(OUTPUT)perf-read-vdso32: perf-read-vdso.c util/find-map.c
-	$(QUIET_CC)$(CC) -m32 $(filter -static,$(LDFLAGS)) -Wall -Werror -o $@ perf-read-vdso.c
+	$(QUIET_CC)$(CC) -m32 $(filter -static,$(LDFLAGS)) -Wall -o $@ perf-read-vdso.c
 endif
 
 ifndef NO_PERF_READ_VDSOX32
 $(OUTPUT)perf-read-vdsox32: perf-read-vdso.c util/find-map.c
-	$(QUIET_CC)$(CC) -mx32 $(filter -static,$(LDFLAGS)) -Wall -Werror -o $@ perf-read-vdso.c
+	$(QUIET_CC)$(CC) -mx32 $(filter -static,$(LDFLAGS)) -Wall -o $@ perf-read-vdso.c
 endif
 
 ifndef NO_JVMTI
diff -Naur -u linux-5.6/tools/testing/selftests/kvm/Makefile linux-5.6.noWerror/tools/testing/selftests/kvm/Makefile
--- linux-5.6/tools/testing/selftests/kvm/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/testing/selftests/kvm/Makefile	2020-05-10 14:03:44.626460112 +0200
@@ -54,11 +54,11 @@
 	-I$(<D) -Iinclude/$(UNAME_M) -I..
 
 no-pie-option := $(call try-run, echo 'int main() { return 0; }' | \
-        $(CC) -Werror -no-pie -x c - -o "$$TMP", -no-pie)
+        $(CC) -no-pie -x c - -o "$$TMP", -no-pie)
 
 # On s390, build the testcases KVM-enabled
 pgste-option = $(call try-run, echo 'int main() { return 0; }' | \
-	$(CC) -Werror -Wl$(comma)--s390-pgste -x c - -o "$$TMP",-Wl$(comma)--s390-pgste)
+	$(CC) -Wl$(comma)--s390-pgste -x c - -o "$$TMP",-Wl$(comma)--s390-pgste)
 
 
 LDFLAGS += -pthread $(no-pie-option) $(pgste-option)
diff -Naur -u linux-5.6/tools/testing/selftests/nsfs/Makefile linux-5.6.noWerror/tools/testing/selftests/nsfs/Makefile
--- linux-5.6/tools/testing/selftests/nsfs/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/testing/selftests/nsfs/Makefile	2020-05-10 14:03:44.633126779 +0200
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0-only
 TEST_GEN_PROGS := owner pidns
 
-CFLAGS := -Wall -Werror
+CFLAGS := -Wall
 
 include ../lib.mk
diff -Naur -u linux-5.6/tools/testing/selftests/powerpc/Makefile linux-5.6.noWerror/tools/testing/selftests/powerpc/Makefile
--- linux-5.6/tools/testing/selftests/powerpc/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/testing/selftests/powerpc/Makefile	2020-05-10 14:03:44.633126779 +0200
@@ -9,7 +9,7 @@
 
 GIT_VERSION = $(shell git describe --always --long --dirty || echo "unknown")
 
-CFLAGS := -std=gnu99 -O2 -Wall -Werror -DGIT_VERSION='"$(GIT_VERSION)"' -I$(CURDIR)/include $(CFLAGS)
+CFLAGS := -std=gnu99 -O2 -Wall -DGIT_VERSION='"$(GIT_VERSION)"' -I$(CURDIR)/include $(CFLAGS)
 
 export CFLAGS
 
diff -Naur -u linux-5.6/tools/testing/selftests/powerpc/pmu/ebb/Makefile linux-5.6.noWerror/tools/testing/selftests/powerpc/pmu/ebb/Makefile
--- linux-5.6/tools/testing/selftests/powerpc/pmu/ebb/Makefile	2020-05-10 14:04:16.333126125 +0200
+++ linux-5.6.noWerror/tools/testing/selftests/powerpc/pmu/ebb/Makefile	2020-05-10 14:03:44.636460112 +0200
@@ -10,7 +10,7 @@
 TMPOUT = $(OUTPUT)/
 # Toolchains may build PIE by default which breaks the assembly
 no-pie-option := $(call try-run, echo 'int main() { return 0; }' | \
-        $(CC) -Werror $(KBUILD_CPPFLAGS) $(CC_OPTION_CFLAGS) -no-pie -x c - -o "$$TMP", -no-pie)
+        $(CC) $(KBUILD_CPPFLAGS) $(CC_OPTION_CFLAGS) -no-pie -x c - -o "$$TMP", -no-pie)
 
 LDFLAGS += $(no-pie-option)
 
diff -Naur -u linux-5.6/tools/testing/selftests/timens/Makefile linux-5.6.noWerror/tools/testing/selftests/timens/Makefile
--- linux-5.6/tools/testing/selftests/timens/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/testing/selftests/timens/Makefile	2020-05-10 14:03:44.643126779 +0200
@@ -1,7 +1,7 @@
 TEST_GEN_PROGS := timens timerfd timer clock_nanosleep procfs exec
 TEST_GEN_PROGS_EXTENDED := gettime_perf
 
-CFLAGS := -Wall -Werror -pthread
+CFLAGS := -Wall -pthread
 LDLIBS := -lrt -ldl
 
 include ../lib.mk
diff -Naur -u linux-5.6/tools/testing/vsock/Makefile linux-5.6.noWerror/tools/testing/vsock/Makefile
--- linux-5.6/tools/testing/vsock/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/testing/vsock/Makefile	2020-05-10 14:03:44.646460112 +0200
@@ -4,7 +4,7 @@
 vsock_test: vsock_test.o timeout.o control.o util.o
 vsock_diag_test: vsock_diag_test.o timeout.o control.o util.o
 
-CFLAGS += -g -O2 -Werror -Wall -I. -I../../include -I../../../usr/include -Wno-pointer-sign -fno-strict-overflow -fno-strict-aliasing -fno-common -MMD -U_FORTIFY_SOURCE -D_GNU_SOURCE
+CFLAGS += -g -O2 -Wall -I. -I../../include -I../../../usr/include -Wno-pointer-sign -fno-strict-overflow -fno-strict-aliasing -fno-common -MMD -U_FORTIFY_SOURCE -D_GNU_SOURCE
 .PHONY: all test clean
 clean:
 	${RM} *.o *.d vsock_test vsock_diag_test
diff -Naur -u linux-5.6/tools/usb/usbip/configure.ac linux-5.6.noWerror/tools/usb/usbip/configure.ac
--- linux-5.6/tools/usb/usbip/configure.ac	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/usb/usbip/configure.ac	2020-05-10 14:03:44.646460112 +0200
@@ -18,7 +18,7 @@
 # Silent build for automake >= 1.11
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
-AC_SUBST([EXTRA_CFLAGS], ["-Wall -Werror -Wextra -std=gnu99"])
+AC_SUBST([EXTRA_CFLAGS], ["-Wall -Wextra -std=gnu99"])
 
 # Checks for programs.
 AC_PROG_CC
diff -Naur -u linux-5.6/tools/virtio/Makefile linux-5.6.noWerror/tools/virtio/Makefile
--- linux-5.6/tools/virtio/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/tools/virtio/Makefile	2020-05-10 14:03:44.649793446 +0200
@@ -4,7 +4,7 @@
 virtio_test: virtio_ring.o virtio_test.o
 vringh_test: vringh_test.o vringh.o virtio_ring.o
 
-CFLAGS += -g -O2 -Werror -Wall -I. -I../include/ -I ../../usr/include/ -Wno-pointer-sign -fno-strict-overflow -fno-strict-aliasing -fno-common -MMD -U_FORTIFY_SOURCE
+CFLAGS += -g -O2 -Wall -I. -I../include/ -I ../../usr/include/ -Wno-pointer-sign -fno-strict-overflow -fno-strict-aliasing -fno-common -MMD -U_FORTIFY_SOURCE
 vpath %.c ../../drivers/virtio ../../drivers/vhost
 mod:
 	${MAKE} -C `pwd`/../.. M=`pwd`/vhost_test V=${V}
diff -Naur -u linux-5.6/usr/include/Makefile linux-5.6.noWerror/usr/include/Makefile
--- linux-5.6/usr/include/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6.noWerror/usr/include/Makefile	2020-05-10 14:03:44.649793446 +0200
@@ -6,7 +6,7 @@
 #
 # -std=c90 (equivalent to -ansi) catches the violation of those.
 # We cannot go as far as adding -Wpedantic since it emits too many warnings.
-UAPI_CFLAGS := -std=c90 -Wall -Werror=implicit-function-declaration
+UAPI_CFLAGS := -std=c90 -Wall
 
 override c_flags = $(UAPI_CFLAGS) -Wp,-MD,$(depfile) -I$(objtree)/usr/include
 
