--- Makefile.orig	2008-03-16 20:31:25.000000000 +0000
+++ Makefile	2008-03-16 20:36:46.000000000 +0000
@@ -9,9 +9,10 @@
 
 # Set this to the location of the X installation you want to compile against
 XROOT = /usr/X11R6
+INSTALL_ROOT = /usr
 
 # Set this to the location of the global configuration files
-SYSCONFDIR = $(PREFIX)/etc/X11/windowlab
+SYSCONFDIR = $(PREFIX)/etc/windowlab
 
 # Information about the location of the menurc file
 ifndef MENURC
@@ -39,8 +40,8 @@
 CFLAGS = -g -O2 -Wall -W
 endif
 
-BINDIR = $(DESTDIR)$(PREFIX)$(XROOT)/bin
-MANDIR = $(DESTDIR)$(PREFIX)$(XROOT)/man/man1
+BINDIR = $(DESTDIR)$(PREFIX)$(INSTALL_ROOT)/bin
+MANDIR = $(DESTDIR)$(PREFIX)$(INSTALL_ROOT)/man/man1
 CFGDIR = $(DESTDIR)$(SYSCONFDIR)
 INCLUDES = -I$(XROOT)/include $(EXTRA_INC)
 LDPATH = -L$(XROOT)/lib
