--- Makefile	2009-11-05 22:04:45.000000000 +0100
+++ Makefile.fixed	2009-12-18 17:07:37.000000000 +0100
@@ -5,18 +5,18 @@
 EXTRA_LIBS += -lXext
 
 # Set this to the hardcoded location of all files if it's not /
-PREFIX = /usr/local
+PREFIX = /usr
 
 # Set this to the directory, below PREFIX, where man pages 
 # are expected. Below this directory, the target "install"
 # will put "windowlab.1x" in section "man1".
-MANBASE = /man
+MANBASE = /share/man
 
 # Set this to the location of the X installation you want to compile against
 XROOT = /usr/X11R6
 
 # Some flexibility for configuration location
-CONFPREFIX = $(PREFIX)
+CONFPREFIX = /
 CONFDIR = /etc/X11/windowlab
 
 # Set this to the location of the global configuration files

