diff -Naur qiv-2.2.4.orig/Makefile qiv-2.2.4/Makefile
--- qiv-2.2.4.orig/Makefile	2011-05-13 15:11:24.000000000 +0200
+++ qiv-2.2.4/Makefile	2013-06-24 19:03:32.000000000 +0200
@@ -47,7 +47,8 @@
 
 CC        = gcc
 #CFLAGS    = -O -g -Wall
-CFLAGS    = -O2 -Wall \
+ENV_CFLAGS := $(CFLAGS)
+CFLAGS    = $(ENV_CFLAGS) -Wall \
 	    -fcaller-saves -ffast-math -fno-strength-reduce \
 	    -fthread-jumps #-march=pentium #-DSTAT_MACROS_BROKEN
 #CFLAGS    = -O2 -Wall -fomit-frame-pointer -finline-functions \
@@ -99,7 +100,7 @@
 all: $(PROGRAM)
 
 $(PROGRAM): $(OBJS)
-	$(CC) $(CFLAGS) $(DEFINES) $(OBJS) $(LIBS) -o $(PROGRAM)
+	$(CC) $(CFLAGS) $(DEFINES) $(LDFLAGS) -o $(PROGRAM) $(OBJS) $(LIBS)
 
 $(OBJS): %.o: %.c $(HEADERS)
 	$(CC) -c $(CFLAGS) $(DEFINES) $(INCLUDES) $< -o $@
@@ -111,7 +112,7 @@
 debug: $(PROGRAM_G)
 
 $(PROGRAM_G): $(OBJS_G)
-	$(CC) -g $(CFLAGS) $(DEFINES_G) $(LIBS) $(OBJS_G) -o $(PROGRAM_G)
+	$(CC) -g $(CFLAGS) $(DEFINES_G) $(LFLAGS) -o $(PROGRAM_G) $(OBJS_G) $(LIBS)
 
 $(OBJS_G): %.g: %.c $(HEADERS)
 	$(CC) -c -g $(CFLAGS) $(DEFINES_G) $(INCLUDES) $< -o $@
