From dba0c127bacf334b6255d51ef21564ff99fb66a7 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 23 Jul 2018 18:48:55 +0200
Subject: [PATCH] use our CFLAGS, honour installation path

---
 Makefile.conf | 19 +++++++------------
 1 file changed, 7 insertions(+), 12 deletions(-)

diff --git a/Makefile.conf b/Makefile.conf
index a1cfb24..a7aea84 100644
--- a/Makefile.conf
+++ b/Makefile.conf
@@ -45,11 +45,11 @@ endif
 # Installation locations (edit if required)
 
 ifneq ($(HOST),MINGW)
-prefix=/usr/local
+prefix=/usr
 bindir=$(prefix)/bin
 incdir=$(prefix)/include
 libdir=$(prefix)/lib
-docdir=$(prefix)/doc/routino
+docdir=$(prefix)/share/doc/routino
 datadir=$(prefix)/share/routino
 else
 prefix="c:/Program Files/Routino"
@@ -69,23 +69,18 @@ LIBVERSION=$(SOVERSION).0.0
 
 
 # Compilation programs
-CC=gcc
-LD=gcc
+LD=$(CC)
 
 
 # Maths library
 LDFLAGS=-lm
 
 # Language dialect selection
-CFLAGS=-std=c99
+CFLAGS+=-std=c99
 
 # Warning options
 CFLAGS+=-Wall -Wmissing-prototypes -Wextra -Wno-unused-parameter -pedantic
 
-# Optimisation options
-CFLAGS+=-O3
-CFLAGS+=-ffast-math
-
 # Optimisation option (only works if compilation and execution use exactly the same CPU architecture).
 #CFLAGS+=-march=native
 
@@ -123,7 +118,7 @@ LDFLAGS_LDSO=-Wl,-R.
 
 # Required for multi-threaded support (comment these two lines out if not required)
 CFLAGS+=-pthread -DUSE_PTHREADS
-LDFLAGS+=-pthread -lpthread
+LDFLAGS+=-pthread
 
 
 ifneq ($(HOST),MINGW)
@@ -139,8 +134,8 @@ LDFLAGS+=-lz
 
 
 # Required for xz support (uncomment these two lines if required)
-#CFLAGS+=-DUSE_XZ
-#LDFLAGS+=-llzma
+CFLAGS+=-DUSE_XZ
+LDFLAGS+=-llzma
 
 
 # Required to use stdio with files > 2GiB on 32-bit system.
-- 
2.18.0

