From b39f4b6b732f3f13b031e34548d8908399be5cee Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sat, 6 Feb 2016 12:39:48 +0100
Subject: [PATCH] use our CFLAGS, honour installation path

---
 Makefile.conf | 28 ++++------------------------
 1 file changed, 4 insertions(+), 24 deletions(-)

diff --git a/Makefile.conf b/Makefile.conf
index cc112e0..7e38466 100644
--- a/Makefile.conf
+++ b/Makefile.conf
@@ -45,11 +45,10 @@ endif
 # Installation locations (edit if required)
 
 ifneq ($(HOST),MINGW)
-prefix=/usr/local
 bindir=$(prefix)/bin
 incdir=$(prefix)/include
 libdir=$(prefix)/lib
-docdir=$(prefix)/doc/routino
+docdir=$(prefix)/share/doc/routino
 datadir=$(prefix)/share/routino
 else
 prefix="c:/Program Files/Routino"
@@ -60,28 +59,9 @@ docdir=$(prefix)/doc
 datadir=$(prefix)/xml
 endif
 
+LD=$(CC)
 
-# Compilation programs
-CC=gcc
-LD=gcc
-
-
-# Language dialect selection
-CFLAGS=-std=c99
-
-# Warning options
-CFLAGS+=-Wall -Wmissing-prototypes -Wextra -Wno-unused-parameter -pedantic
-
-# Optimisation options
-CFLAGS+=-O3
-CFLAGS+=-ffast-math
-
-# Optimisation option (only works if compilation and execution use exactly the same CPU architecture).
-#CFLAGS+=-march=native
-
-# Debugging symbols
-#CFLAGS+=-g
-
+CFLAGS+=-std=c99
 
 # Maths library
 LDFLAGS=-lm
@@ -103,7 +83,7 @@ LDFLAGS_LDSO=-Wl,-R.
 
 # Required for multi-threaded support (comment these two lines out if not required)
 CFLAGS+=-pthread -DUSE_PTHREADS
-LDFLAGS+=-pthread -lpthread
+LDFLAGS+=-pthread
 
 
 ifneq ($(HOST),MINGW)
-- 
2.7.0

