--- calendar.c.orig	2008-10-26 15:58:01.768236917 +0100
+++ calendar.c	2008-10-26 15:59:48.214892700 +0100
@@ -48,7 +48,7 @@
 
 
 
-void kill (GtkWidget * widget)
+void kill_widget (GtkWidget * widget)
 {
     gtk_widget_destroy ((GtkWidget*)gtk_widget_get_toplevel(widget));
     gtk_main_quit ();
@@ -193,7 +193,7 @@
 	gtk_signal_connect(GTK_OBJECT (event_box), "button_press_event",
 			   GTK_SIGNAL_FUNC (move), NULL);
 	gtk_signal_connect(GTK_OBJECT (event_box2), "button_press_event",
-			   (GtkSignalFunc) kill, GTK_OBJECT(dayView));
+			   (GtkSignalFunc) kill_widget, GTK_OBJECT(dayView));
 
 	gtk_widget_realize(dayView);
 	//	gtk_window_set_position(GTK_WINDOW (dayView),GTK_WIN_POS_CENTER_ON_PARENT);
--- calendarfunc.c.orig	2008-10-26 15:56:12.068247584 +0100
+++ calendarfunc.c	2008-10-26 15:56:37.728245314 +0100
@@ -22,6 +22,7 @@
 
 #include "calendarfunc.h"
 #include <stdio.h>
+#include <math.h>
 
 
 
--- dockapp.c.orig	2008-10-26 15:56:59.058242478 +0100
+++ dockapp.c	2008-10-26 15:57:44.694925854 +0100
@@ -24,6 +24,7 @@
  */
 
 #include "dockapp.h"
+#include <stdlib.h>
 
 #define WINDOWED_SIZE_W 64
 #define WINDOWED_SIZE_H 64
