--- src/gigedit/mainwindow.cpp~	2008-05-24 18:02:31.000000000 +0300
+++ src/gigedit/mainwindow.cpp	2008-05-24 18:02:56.000000000 +0300
@@ -46,6 +46,7 @@
 #endif
 
 #include <stdio.h>
+#include <string.h>
 #include <sndfile.h>
 
 #include "mainwindow.h"
--- src/gigedit/paramedit.cpp~	2008-05-24 18:09:37.000000000 +0300
+++ src/gigedit/paramedit.cpp	2008-05-24 18:09:56.000000000 +0300
@@ -17,6 +17,8 @@
  * 02110-1301 USA.
  */
 
+#include <string.h>
+
 #include "paramedit.h"
 
 namespace {
