diff -p -u -5 -r1.119 wv.h
--- wv.h	18 Oct 2005 04:03:30 -0000	1.119
+++ wv.h	29 Aug 2006 14:10:03 -0000
@@ -2683,11 +2683,10 @@ that indicates their length.
 	/*public */
 	void *userData;
 
 	/*protected */
         GsfInput *ole_file;
-        GsfInput *input;
 	wvStream *mainfd;
 	wvStream *tablefd;
 	wvStream *data;
 	wvStream *summary;
 	FIB fib;
@@ -2740,11 +2739,12 @@ that indicates their length.
 
 	int fieldstate;
 	int fieldmiddle;
 	char *filename;
 	char *dir;
-    } wvParseStruct;
+         GsfInput *input;
+   } wvParseStruct;
 
     void wvSetPassword (const char *password, wvParseStruct * ps);
     void wvSetTableInfo (wvParseStruct * ps, TAP * ptap, int no);
     int wvDecrypt95 (wvParseStruct * ps);
     int wvDecrypt97 (wvParseStruct * ps);

