--- Makefile.orig	2005-02-07 15:56:48.000000000 +0100
+++ Makefile	2005-02-07 23:23:20.422596896 +0100
@@ -36,7 +36,7 @@
 # If unset Intel 386 with optimization for Pentium is selected.
 
 ifndef ARCH
-ARCH     = -march=i586 -m$(TUNE)=i686
+ARCH     = #-march=i586 -m$(TUNE)=i686
 endif
 
 ifdef MINGW
@@ -176,9 +176,9 @@
 	-fno-finite-math-only -fno-unsafe-math-optimizations
 endif
 
-OPTIM_SPEED = -O2 $(FLAGS) $(UNBREAK)
+OPTIM_SPEED = $(UNBREAK)
 
-OPTIM_SIZE = -Os $(FLAGS) $(UNBREAK)
+OPTIM_SIZE = $(UNBREAK)
 
 # Options to generate Assembly code for inspecting
 
