diff -Naur motif-2.3.4.orig/clients/mwm/Makefile.am motif-2.3.4/clients/mwm/Makefile.am
--- motif-2.3.4.orig/clients/mwm/Makefile.am	2012-10-22 16:50:39.000000000 +0200
+++ motif-2.3.4/clients/mwm/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -11,7 +11,7 @@
 bin_PROGRAMS = mwm
 mwm_LDADD = ./WmWsmLib/libWsm.a ../../lib/Xm/libXm.la 
 
-INCLUDES = -DLARGECURSORS -DR2_COMPAT -DUNMAP_ON_RESTART \
+AM_CPPFLAGS = -DLARGECURSORS -DR2_COMPAT -DUNMAP_ON_RESTART \
            -DCDE_INSTALLATION_TOP=\"@CDE_INSTALLATION_TOP@\" \
            -DCDE_CONFIGURATION_TOP=\"@CDE_CONFIGURATION_TOP@\" \
            -DBITMAPDIR=\"@INCDIR@/bitmaps\" \
diff -Naur motif-2.3.4.orig/clients/mwm/WmWsmLib/Makefile.am motif-2.3.4/clients/mwm/WmWsmLib/Makefile.am
--- motif-2.3.4.orig/clients/mwm/WmWsmLib/Makefile.am	2012-10-22 16:50:39.000000000 +0200
+++ motif-2.3.4/clients/mwm/WmWsmLib/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -4,7 +4,7 @@
 
 noinst_HEADERS = utm_send.h	wsm_funcs.h	wsm_proto.h
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 
 libWsm_a_SOURCES = $(SRCS)
 
diff -Naur motif-2.3.4.orig/clients/uil/Makefile.am motif-2.3.4/clients/uil/Makefile.am
--- motif-2.3.4.orig/clients/uil/Makefile.am	2012-10-22 16:50:39.000000000 +0200
+++ motif-2.3.4/clients/uil/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -15,7 +15,7 @@
 
 uil_LDADD = ../../lib/Mrm/libMrm.la ../../lib/Xm/libXm.la 
 
-INCLUDES = -DINCDIR=\"@INCDIR@\" \
+AM_CPPFLAGS = -DINCDIR=\"@INCDIR@\" \
            -DLIBDIR=\"@LIBDIR@\" \
            -I$(srcdir)/../../include \
            -I$(srcdir)/../../lib \
diff -Naur motif-2.3.4.orig/clients/xmbind/Makefile.am motif-2.3.4/clients/xmbind/Makefile.am
--- motif-2.3.4.orig/clients/xmbind/Makefile.am	2012-10-22 16:50:39.000000000 +0200
+++ motif-2.3.4/clients/xmbind/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -4,4 +4,4 @@
 
 LDADD = ../../lib/Xm/libXm.la
 
-INCLUDES = -I$(srcdir)/../../lib -I$(top_builddir)/lib ${X_CFLAGS}
+AM_CPPFLAGS = -I$(srcdir)/../../lib -I$(top_builddir)/lib ${X_CFLAGS}
diff -Naur motif-2.3.4.orig/config/util/Makefile.am motif-2.3.4/config/util/Makefile.am
--- motif-2.3.4.orig/config/util/Makefile.am	2012-10-22 16:50:39.000000000 +0200
+++ motif-2.3.4/config/util/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -7,7 +7,7 @@
 	     mkdirhier.man mkdirhier.sh mkshadow syminst.sh x11mf.sh \
 	     xmkmf.cpp xmkmf.man Makefile.ini
 
-INCLUDES = ${X_CFLAGS}
+AM_CPPFLAGS = ${X_CFLAGS}
 noinst_PROGRAMS = makestrs
 
 
diff -Naur motif-2.3.4.orig/configure.ac motif-2.3.4/configure.ac
--- motif-2.3.4.orig/configure.ac	2012-10-22 16:50:39.000000000 +0200
+++ motif-2.3.4/configure.ac	2013-06-23 17:28:17.000000000 +0200
@@ -1,8 +1,13 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT
+AC_PREREQ(2.59)
+AC_INIT([motif], [2.3.4])
 AC_CONFIG_SRCDIR([lib/Xm/Form.c])
-AC_PREREQ(2.52)
 AC_CONFIG_AUX_DIR(.)
+AC_CANONICAL_TARGET
+
+AM_INIT_AUTOMAKE([foreign no-define])
+dnl AM_MAINTAINER_MODE
+
 AC_CHECK_FILE(/usr/X/include/X11/X.h,
   AC_PREFIX_DEFAULT(/usr/X),
   AC_PREFIX_DEFAULT(/usr))
@@ -10,9 +15,6 @@
   AC_PREFIX_DEFAULT(/usr/X11R6),
   AC_PREFIX_DEFAULT(/usr))
 
-dnl AM_MAINTAINER_MODE
-AC_CANONICAL_TARGET
-
 dnl The version of the Motif libraries
 CURRENT=4
 REVISION=4
@@ -21,9 +23,6 @@
 AC_SUBST(REVISION)
 AC_SUBST(AGE)
 
-dnl This is really dumb but it seems to be bug
-AM_INIT_AUTOMAKE(motif,2.3.4,no-define)
-
 LIBTOOL_VERSION=$CURRENT:$REVISION:$AGE
 
 dnl
@@ -31,12 +30,11 @@
 dnl
 dnl AC_DISABLE_STATIC
 
-AM_CONFIG_HEADER(include/config.h lib/Xm/Xm.h)
+AC_CONFIG_HEADERS(include/config.h lib/Xm/Xm.h)
 
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_CPP
-AM_PROG_CC_STDC
 AM_PROG_LIBTOOL
 
 AC_EXEEXT
diff -Naur motif-2.3.4.orig/configure.ac.orig motif-2.3.4/configure.ac.orig
--- motif-2.3.4.orig/configure.ac.orig	1970-01-01 01:00:00.000000000 +0100
+++ motif-2.3.4/configure.ac.orig	2013-06-23 17:28:06.000000000 +0200
@@ -0,0 +1,400 @@
+dnl Process this file with autoconf to produce a configure script.
+AC_PREREQ(2.59)
+AC_INIT([motif], [2.3.4])
+AC_CONFIG_SRCDIR([lib/Xm/Form.c])
+AC_CONFIG_AUX_DIR(.)
+AC_CANONICAL_TARGET
+
+AM_INIT_AUTOMAKE([foreign no-define])
+dnl AM_MAINTAINER_MODE
+
+AC_CHECK_FILE(/usr/X/include/X11/X.h,
+  AC_PREFIX_DEFAULT(/usr/X),
+  AC_PREFIX_DEFAULT(/usr))
+AC_CHECK_FILE(/usr/X11R6/include/X11/X.h,
+  AC_PREFIX_DEFAULT(/usr/X11R6),
+  AC_PREFIX_DEFAULT(/usr))
+
+dnl The version of the Motif libraries
+CURRENT=4
+REVISION=4
+AGE=0
+AC_SUBST(CURRENT)
+AC_SUBST(REVISION)
+AC_SUBST(AGE)
+
+LIBTOOL_VERSION=$CURRENT:$REVISION:$AGE
+
+dnl
+dnl Override a default in libtool.
+dnl
+dnl AC_DISABLE_STATIC
+
+AM_CONFIG_HEADER(include/config.h lib/Xm/Xm.h)
+
+dnl Checks for programs.
+AC_PROG_CC
+AC_PROG_CPP
+AM_PROG_CC_STDC
+AM_PROG_LIBTOOL
+
+AC_EXEEXT
+
+dnl AC_PROG_YACC
+dnl Do this the old fashioned way. 'bison -y' doesn't cut it
+AC_CHECK_PROGS(YACC, byacc, yacc)
+
+dnl AC_PROG_SPLINT
+AC_CHECK_PROGS(SPLINT, splint)
+if test x$SPLINT = x
+then
+    AC_MSG_WARN([No splint is dound])
+fi
+
+AM_PROG_LEX
+AC_PROG_INSTALL
+AC_PROG_LN_S
+AC_PROG_MAKE_SET
+
+
+dnl Checks for libraries.
+
+AC_PATH_X
+AC_PATH_XTRA
+LT_LIB_XTHREADS
+
+dnl Checks for Xos_r.h
+
+AC_CHECK_HEADERS([X11/Xos_r.h])
+AC_CHECK_HEADERS([X11/Xpoll.h])
+
+dnl Compile in EditRes support if we can find Xmu
+
+save_LIBS="$LIBS"
+save_CFLAGS="$CFLAGS"
+LIBS="$X_LIBS $LIBS"
+CFLAGS="$X_CFLAGS $CFLAGS"
+AC_CHECK_HEADERS(X11/Xmu/Editres.h,
+AC_CHECK_LIB(Xmu, _XEditResCheckMessages,
+X_XMU=-lXmu
+AC_DEFINE(HAVE_LIBXMU, 1, This System has libXmu)
+, X_XMU="", -lXt -lX11)
+AC_CHECK_LIB(Xmu,XmuNCopyISOLatin1Lowered ,
+AC_DEFINE(HAVE_XMU_N_COPY_ISO,1,Have XmuNCopyISOLatin1Lowered)
+, , -lXt -lX11)
+,,[[#include <X11/Intrinsic.h>]]
+)
+
+
+AC_SUBST(X_XMU)
+LIBS="$save_LIBS"
+CFLAGS="$save_CFLAGS"
+AM_CONDITIONAL(OM_XMU, test x$X_XMU != x)
+
+dnl Checks for header files.
+AC_HEADER_DIRENT
+AC_HEADER_STDC
+AC_HEADER_SYS_WAIT
+AC_CHECK_HEADERS(fcntl.h limits.h malloc.h sys/malloc.h strings.h sys/file.h sys/time.h unistd.h wchar.h)
+AC_CHECK_HEADERS(wctype.h, AC_DEFINE(HAS_WIDECHAR_FUNCTIONS,1,System supports wchar))
+AC_CHECK_HEADER(langinfo.h,,AC_DEFINE(CSRG_BASED,1,System Has langinfo.h))
+AC_CHECK_HEADER(X11/Xos_r.h,,AC_DEFINE(NEED_XOS_R_H,1,System Missing Xos_r.h))
+AC_CHECK_HEADER(X11/Xpoll.h,,AC_DEFINE(NEED_XPOLL_H,1,system Missing X11/Xpoll.h))
+dnl Checks for typedefs, structures, and compiler characteristics.
+AC_C_CONST
+AC_TYPE_MODE_T
+AC_TYPE_OFF_T
+AC_TYPE_PID_T
+AC_TYPE_SIZE_T
+AC_HEADER_TIME
+AC_STRUCT_TM
+AC_TYPE_UID_T
+
+dnl Checks for library functions.
+AC_FUNC_ALLOCA
+AC_FUNC_MEMCMP
+AC_FUNC_SETPGRP
+AC_TYPE_SIGNAL
+AC_FUNC_STRCOLL
+AC_FUNC_STRFTIME
+AC_FUNC_FORK([])
+AC_FUNC_VPRINTF
+AM_FUNC_VOID_SPRINTF
+AC_CHECK_FUNCS(wcslen wcscpy wcsncpy wcschr wcscat wcsncat getcwd gettimeofday mkdir re_comp regcmp select strcspn strerror strstr strtod strtol uname)
+AC_CHECK_FUNCS(strdup,,AC_DEFINE(NEED_STRDUP, 1, no strdup))
+AC_CHECK_FUNCS(strcasecmp,,AC_DEFINE(NEED_STRCASECMP, 1, no strcasecmp))
+AC_CHECK_FUNCS(putenv,,AC_DEFINE(NO_PUTENV, 1, No PUTENV))
+AC_CHECK_FUNCS(regcomp,,AC_DEFINE(NO_REGCOMP, 1, no regcmp))
+AC_CHECK_FUNCS(memmove,,AC_DEFINE(NO_MEMMOVE, 1, no memmove))
+AC_CHECK_TYPE(XICProc,,AC_DEFINE(NO_XICPROC, 1, XICProc isn't defined), [#include <X11/Xlib.h>])
+
+AC_ARG_ENABLE(message-catalog, [  --enable-message-catalog
+                          Enable building of the message catalog (default=no)])
+if test "$enable_message_catalog" != "yes"
+then
+    AC_DEFINE(NO_MESSAGE_CATALOG, 1, No Defined Message Catalog)
+fi
+AM_CONDITIONAL(MessageCatalog, test x$enable_message_catalog = xyes)
+
+AC_ARG_ENABLE(themes, [  --enable-themes         Enable themes (default=no)])
+if test "$enable_themes" = "yes"
+then
+    if test -f $srcdir/lib/Xm/Theme.h
+    then
+	AC_DEFINE(USE_XMTHEMES, 1, "Use XmThemes")
+	XMTHEME_DIST = "Theme.h ThemeP.h DynLibP.h Theme.c DynLib.c"
+    else
+	AC_MSG_WARN([--enable-themes specified, but I don't have the code. Disabled])
+	enable_themes="no"
+    fi
+fi
+AC_SUBST(XMTHEME_DIST)
+AM_CONDITIONAL(OM_XMTHEMES, test "$enable_themes" = "yes")
+
+AC_ARG_ENABLE(debug-themes, [  --enable-debug-themes   Enable themes debugging (default=no)])
+if test "$enable_themes" = "yes"
+then
+    if test "$enable_debug_themes" = "yes"
+    then
+	AC_DEFINE(DEBUGTHEMES, 1, "Debug Themes")
+    fi
+fi
+
+if test x$GCC = xyes
+then
+    CFLAGS="$CFLAGS -Wall -g -fno-strict-aliasing -Wno-unused -Wno-comment"
+    if test ` $CC -dumpversion | sed -e 's/\(^.\).*/\1/'` = "4" ; then
+        CFLAGS="$CFLAGS -fno-tree-ter"
+    fi
+fi
+AC_DEFINE(NO_OL_COMPAT, 1, "No OL Compatability")
+
+CDE_INSTALLATION_TOP="\${prefix}/dt"
+AC_SUBST(CDE_INSTALLATION_TOP)
+
+CDE_CONFIGURATION_TOP="\${prefix}/etc/dt"
+AC_SUBST(CDE_CONFIGURATION_TOP)
+
+LIBDIR="${libdir}/X11"
+AC_SUBST(LIBDIR)
+
+MWMRCDIR="${libdir}/X11"
+AC_SUBST(MWMRCDIR)
+
+INCDIR="${includedir}/X11"
+AC_SUBST(INCDIR)
+
+XMBINDDIR_FALLBACK="${libdir}/X11/bindings"
+AC_SUBST(XMBINDDIR_FALLBACK)
+
+RM="rm -f"
+AC_SUBST(RM)
+
+AC_ARG_ENABLE(motif22-compatibility, [  --disable-motif22-compatibility
+                       Disable binary compatibility with OpenMotif 2.2])
+AC_ARG_ENABLE(utf8, [  --disable-utf8
+                       Disable UTF-8 support])
+
+case "$host_os" in
+    freebsd*) CFLAGS="$CFLAGS -DCSRG_BASED -DXNO_MTSAFE_API -DXNO_MTSAFE_PWDAPI";
+        LDFLAGS="$LDFLAGS -L/usr/local/lib";;
+    cygwin*) CFLAGS="$CFLAGS -DXNO_MTSAFE_DIRENTDAPI";;
+    solaris* | sunos*) X_LIBS="$X_LIBS -L/usr/X/lib" &&
+        if test x$enable_motif22_compatibility = x
+        then
+            enable_motif22_compatibility="no"
+        fi;
+        if test x$enable_utf8 = x
+        then
+            enable_utf8="no"
+        fi;
+        if test x$enable_xft = x
+        then
+            enable_xft="no"
+        fi;;
+esac
+
+if test x$enable_motif22_compatibility = x
+then
+    enable_motif22_compatibility="yes"
+fi
+
+if test "$enable_motif22_compatibility" = "yes"
+then
+  AC_DEFINE([OM22_COMPATIBILITY], [1], [Compatibility with OpenMotif 2.2])
+  OM22_COMPATIBILITY_FALSE=!
+  OM22_COMPATIBILITY_TRUE=
+else
+  OM22_COMPATIBILITY_FALSE=
+  OM22_COMPATIBILITY_TRUE=!
+fi
+AC_SUBST(OM22_COMPATIBILITY_FALSE)
+AC_SUBST(OM22_COMPATIBILITY_TRUE)
+
+if test x$enable_utf8 = x
+then
+    enable_utf8="yes"
+fi
+
+if test "$enable_utf8" = "yes"
+then
+  AC_DEFINE([UTF8_SUPPORTED], [1], [UTF8 is supported])
+  AC_SEARCH_LIBS([iconv_open], [iconv], ,
+    AC_SEARCH_LIBS([libiconv_open], [iconv]))
+fi
+
+AC_ARG_ENABLE(printing, [  --disable-printing
+                       Disable printing support])
+
+if test x$enable_printing = x
+then
+    enable_printing="yes"
+fi
+
+if test "$enable_printing" = "yes"
+then
+  AC_MSG_CHECKING([for libXp])
+  AC_CHECK_HEADERS(X11/extensions/Print.h,
+    AC_CHECK_LIB(Xp, XpCreateContext, ,enable_printing="no"),
+  enable_printing="no")
+fi
+
+if test "$enable_printing" = "yes"
+then
+  LIB_XP=-lXp
+  AC_DEFINE([PRINTING_SUPPORTED], [1], [Printing is supported])
+  PRINTING_SUPPORTED_FALSE=!
+  PRINTING_SUPPORTED_TRUE=
+else
+  LIB_XP=
+  PRINTING_SUPPORTED_FALSE=
+  PRINTING_SUPPORTED_TRUE=!
+fi
+AC_SUBST(PRINTING_SUPPORTED_FALSE)
+AC_SUBST(PRINTING_SUPPORTED_TRUE)
+
+AM_CONDITIONAL(PRINTING, test "$enable_printing" = "yes")
+AC_SUBST(LIB_XP)
+
+AC_FIND_XFT
+AC_IMAGE_SUPPORT
+
+
+
+# AM_CONDITIONAL(Motif22Compatibility, test x$enable_motif22_compatibility = xyes)
+
+AC_CONFIG_FILES([Makefile \
+bindings/Makefile \
+bitmaps/Makefile \
+clients/Makefile \
+clients/mwm/Makefile \
+clients/mwm/WmWsmLib/Makefile \
+clients/uil/Makefile \
+clients/xmbind/Makefile \
+config/Makefile \
+config/cf/Makefile \
+config/imake/Makefile \
+config/util/Makefile \
+config/makedepend/Makefile \
+include/Makefile \
+include/Dt/Makefile \
+lib/Makefile \
+lib/Xm/Makefile \
+lib/Mrm/Makefile \
+localized/Makefile \
+localized/util/Makefile \
+doc/Makefile \
+doc/man/Makefile \
+doc/man/man1/Makefile \
+doc/man/man3/Makefile \
+doc/man/man4/Makefile \
+doc/man/man5/Makefile \
+tools/Makefile \
+tools/wml/Makefile \
+demos/Makefile \
+demos/lib/Makefile \
+demos/lib/Xmd/Makefile \
+demos/lib/Wsm/Makefile \
+demos/lib/Exm/Makefile \
+demos/lib/Exm/wml/Makefile \
+demos/programs/Makefile \
+demos/programs/Exm/Makefile \
+demos/programs/Exm/app_in_c/Makefile \
+demos/programs/Exm/app_in_uil/Makefile \
+demos/programs/Exm/simple_app/Makefile \
+demos/programs/airport/Makefile \
+demos/programs/animate/Makefile \
+demos/programs/drag_and_drop/Makefile \
+demos/programs/draw/Makefile \
+demos/programs/earth/Makefile \
+demos/programs/filemanager/Makefile \
+demos/programs/fileview/Makefile \
+demos/programs/getsubres/Makefile \
+demos/programs/hellomotif/Makefile \
+demos/programs/hellomotifi18n/Makefile \
+demos/programs/hellomotifi18n/C/Makefile \
+demos/programs/hellomotifi18n/C/uid/Makefile \
+demos/programs/hellomotifi18n/english/Makefile \
+demos/programs/hellomotifi18n/english/uid/Makefile \
+demos/programs/hellomotifi18n/french/Makefile \
+demos/programs/hellomotifi18n/french/uid/Makefile \
+demos/programs/hellomotifi18n/hebrew/Makefile \
+demos/programs/hellomotifi18n/hebrew/uid/Makefile \
+demos/programs/hellomotifi18n/japan/Makefile \
+demos/programs/hellomotifi18n/japan/uid/Makefile \
+demos/programs/hellomotifi18n/japanese/Makefile \
+demos/programs/hellomotifi18n/japanese/uid/Makefile \
+demos/programs/hellomotifi18n/swedish/Makefile \
+demos/programs/hellomotifi18n/swedish/uid/Makefile \
+demos/programs/i18ninput/Makefile \
+demos/programs/panner/Makefile \
+demos/programs/periodic/Makefile \
+demos/programs/piano/Makefile \
+demos/programs/popups/Makefile \
+demos/programs/sampler2_0/Makefile \
+demos/programs/setdate/Makefile \
+demos/programs/todo/Makefile \
+demos/programs/workspace/Makefile \
+demos/programs/tooltips/Makefile \
+demos/programs/FontSel/Makefile \
+demos/programs/ButtonBox/Makefile \
+demos/programs/ColorSel/Makefile \
+demos/programs/Column/Makefile \
+demos/programs/Combo2/Makefile \
+demos/programs/Ext18List/Makefile \
+demos/programs/Ext18List/pixmaps/Makefile \
+demos/programs/IconB/Makefile \
+demos/programs/Outline/Makefile \
+demos/programs/Paned/Makefile \
+demos/programs/TabStack/Makefile \
+demos/programs/Tree/Makefile \
+demos/programs/pixmaps/Makefile \
+demos/unsupported/Makefile \
+demos/unsupported/Exm/Makefile \
+demos/unsupported/aicon/Makefile \
+demos/unsupported/dainput/Makefile \
+demos/unsupported/dogs/Makefile \
+demos/unsupported/hellomotif/Makefile \
+demos/unsupported/motifshell/Makefile \
+demos/unsupported/uilsymdump/Makefile \
+demos/unsupported/xmapdef/Makefile \
+demos/unsupported/xmfonts/Makefile \
+demos/unsupported/xmforc/Makefile \
+demos/unsupported/xmform/Makefile \
+demos/doc/Makefile \
+demos/doc/programGuide/Makefile \
+demos/doc/programGuide/ch05/Makefile \
+demos/doc/programGuide/ch05/Scale/Makefile \
+demos/doc/programGuide/ch06/Makefile \
+demos/doc/programGuide/ch06/spin_box/Makefile \
+demos/doc/programGuide/ch06/combo_box/Makefile \
+demos/doc/programGuide/ch08/Makefile \
+demos/doc/programGuide/ch08/Notebook/Makefile \
+demos/doc/programGuide/ch08/Container/Makefile \
+demos/doc/programGuide/ch16/Makefile \
+demos/doc/programGuide/ch17/Makefile \
+demos/doc/programGuide/ch17/simple_drop/Makefile \
+demos/doc/programGuide/ch17/simple_drag/Makefile \
+lib/Xm/xmstring.list \
+])
+AC_OUTPUT
diff -Naur motif-2.3.4.orig/demos/doc/programGuide/ch05/Scale/Makefile.am motif-2.3.4/demos/doc/programGuide/ch05/Scale/Makefile.am
--- motif-2.3.4.orig/demos/doc/programGuide/ch05/Scale/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/doc/programGuide/ch05/Scale/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -6,6 +6,6 @@
 
 tic_mark_SOURCES = tic_mark.c interface.c
 
-INCLUDES = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
 
 LDADD = -L$(top_builddir)/demos/lib/Xmd -lXmd $(top_builddir)/lib/Xm/libXm.la
diff -Naur motif-2.3.4.orig/demos/doc/programGuide/ch06/combo_box/Makefile.am motif-2.3.4/demos/doc/programGuide/ch06/combo_box/Makefile.am
--- motif-2.3.4.orig/demos/doc/programGuide/ch06/combo_box/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/doc/programGuide/ch06/combo_box/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -7,6 +7,6 @@
 combo_box_SOURCES = combo_box.c interface.c
 
 
-INCLUDES = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
 
 LDADD = -L$(top_builddir)/demos/lib/Xmd -lXmd $(top_builddir)/lib/Xm/libXm.la
diff -Naur motif-2.3.4.orig/demos/doc/programGuide/ch06/spin_box/Makefile.am motif-2.3.4/demos/doc/programGuide/ch06/spin_box/Makefile.am
--- motif-2.3.4.orig/demos/doc/programGuide/ch06/spin_box/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/doc/programGuide/ch06/spin_box/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -7,6 +7,6 @@
 spin_box_SOURCES = spin_box.c interface.c
 
 
-INCLUDES = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
 
 LDADD = -L$(top_builddir)/demos/lib/Xmd -lXmd $(top_builddir)/lib/Xm/libXm.la
diff -Naur motif-2.3.4.orig/demos/doc/programGuide/ch08/Container/Makefile.am motif-2.3.4/demos/doc/programGuide/ch08/Container/Makefile.am
--- motif-2.3.4.orig/demos/doc/programGuide/ch08/Container/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/doc/programGuide/ch08/Container/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -6,6 +6,6 @@
 
 container_SOURCES = container.c  interface.c
 
-INCLUDES = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
 
 LDADD = -L$(top_builddir)/demos/lib/Xmd -lXmd $(top_builddir)/lib/Xm/libXm.la
diff -Naur motif-2.3.4.orig/demos/doc/programGuide/ch08/Notebook/Makefile.am motif-2.3.4/demos/doc/programGuide/ch08/Notebook/Makefile.am
--- motif-2.3.4.orig/demos/doc/programGuide/ch08/Notebook/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/doc/programGuide/ch08/Notebook/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -6,6 +6,6 @@
 
 notebook_SOURCES = interface.c  notebook.c
 
-INCLUDES = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
 
 LDADD = -L$(top_builddir)/demos/lib/Xmd -lXmd $(top_builddir)/lib/Xm/libXm.la
diff -Naur motif-2.3.4.orig/demos/doc/programGuide/ch16/Makefile.am motif-2.3.4/demos/doc/programGuide/ch16/Makefile.am
--- motif-2.3.4.orig/demos/doc/programGuide/ch16/Makefile.am	2012-10-22 16:50:39.000000000 +0200
+++ motif-2.3.4/demos/doc/programGuide/ch16/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -7,6 +7,6 @@
 UTMXmText_SOURCES = UTMXmText.c
 
 
-INCLUDES = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
 
 LDADD = -L$(top_builddir)/demos/lib/Xmd -lXmd $(top_builddir)/lib/Xm/libXm.la
diff -Naur motif-2.3.4.orig/demos/doc/programGuide/ch17/simple_drag/Makefile.am motif-2.3.4/demos/doc/programGuide/ch17/simple_drag/Makefile.am
--- motif-2.3.4.orig/demos/doc/programGuide/ch17/simple_drag/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/doc/programGuide/ch17/simple_drag/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -7,6 +7,6 @@
 simple_drag_SOURCES = simple_drag.c 
 
 
-INCLUDES = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
 
 LDADD = -L$(top_builddir)/demos/lib/Xmd -lXmd $(top_builddir)/lib/Xm/libXm.la
diff -Naur motif-2.3.4.orig/demos/doc/programGuide/ch17/simple_drop/Makefile.am motif-2.3.4/demos/doc/programGuide/ch17/simple_drop/Makefile.am
--- motif-2.3.4.orig/demos/doc/programGuide/ch17/simple_drop/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/doc/programGuide/ch17/simple_drop/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -7,6 +7,6 @@
 simple_drop_SOURCES = simple_drop.c 
 
 
-INCLUDES = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
+AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib -I$(top_srcdir)/demos/lib $(X_CFLAGS)
 
 LDADD = -L$(top_builddir)/demos/lib/Xmd -lXmd $(top_builddir)/lib/Xm/libXm.la
diff -Naur motif-2.3.4.orig/demos/lib/Exm/Makefile.am motif-2.3.4/demos/lib/Exm/Makefile.am
--- motif-2.3.4.orig/demos/lib/Exm/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/lib/Exm/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -4,7 +4,7 @@
 
 noinst_LIBRARIES = libExm.a libExmMrm.a
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(srcdir)/.. -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(srcdir)/.. -I$(top_builddir)/lib ${X_CFLAGS} 
 
 CLEANFILES = Grid.E
    
diff -Naur motif-2.3.4.orig/demos/lib/Exm/wml/Makefile.am motif-2.3.4/demos/lib/Exm/wml/Makefile.am
--- motif-2.3.4.orig/demos/lib/Exm/wml/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/lib/Exm/wml/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,7 +9,7 @@
 
 WMLDIR = $(top_srcdir)/tools/wml
 
-INCLUDES = -I$(top_srcdir)/lib \
+AM_CPPFLAGS = -I$(top_srcdir)/lib \
         -I$(top_builddir)/lib \
 	-I$(top_srcdir)/include \
 	-I$(srcdir)/../.. \
@@ -30,7 +30,7 @@
 	./wmldbcreate -o $@
 
 
-# INCLUDES = -I. -I$(INCLUDESRC) -I$(MINCLUDESRC) -I$(MLIBSRC) -I../../. -I$(MTOP)/include
+# AM_CPPFLAGS = -I. -I$(INCLUDESRC) -I$(MINCLUDESRC) -I$(MLIBSRC) -I../../. -I$(MTOP)/include
 # DEPLIBS = $(DEPXMLIB)  $(DEPXTOOLLIB) $(DEPXLIB)
 # LOCAL_LIBRARIES = XmClientLibs
 # 
diff -Naur motif-2.3.4.orig/demos/lib/Wsm/Makefile.am motif-2.3.4/demos/lib/Wsm/Makefile.am
--- motif-2.3.4.orig/demos/lib/Wsm/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/lib/Wsm/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -6,4 +6,4 @@
 
 noinst_HEADERS = wsm_funcs.h wsm_proto.h utm_send.h
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/lib/Xmd/Makefile.am motif-2.3.4/demos/lib/Xmd/Makefile.am
--- motif-2.3.4.orig/demos/lib/Xmd/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/lib/Xmd/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -14,7 +14,7 @@
 
 LDADD = libXmd.a ../../../lib/Xm/libXm.la
 
-INCLUDES = -I$(top_builddir)/lib -I$(srcdir)/../../../lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(top_builddir)/lib -I$(srcdir)/../../../lib ${X_CFLAGS} 
 
 EXTRA_DIST = Imakefile
 
diff -Naur motif-2.3.4.orig/demos/programs/ButtonBox/Makefile.am motif-2.3.4/demos/programs/ButtonBox/Makefile.am
--- motif-2.3.4.orig/demos/programs/ButtonBox/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/ButtonBox/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -10,4 +10,4 @@
 TOPDIR = ../../..
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
 
-INCLUDES = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/ColorSel/Makefile.am motif-2.3.4/demos/programs/ColorSel/Makefile.am
--- motif-2.3.4.orig/demos/programs/ColorSel/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/ColorSel/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,4 +9,4 @@
 data_DATA = $(colordemo_SOURCES) $(EXTRA_DIST)
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
 TOPDIR = ../../..
-INCLUDES = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/Column/Makefile.am motif-2.3.4/demos/programs/Column/Makefile.am
--- motif-2.3.4.orig/demos/programs/Column/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/Column/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,4 +9,4 @@
 data_DATA = $(EXTRA_DIST) $(column_SOURCES)
 TOPDIR = ../../..
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
-INCLUDES = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/Combo2/Makefile.am motif-2.3.4/demos/programs/Combo2/Makefile.am
--- motif-2.3.4.orig/demos/programs/Combo2/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/Combo2/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,4 +9,4 @@
 data_DATA = $(EXTRA_DIST) $(combo_SOURCES)
 TOPDIR = ../../..
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
-INCLUDES = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/Exm/app_in_c/Makefile.am motif-2.3.4/demos/programs/Exm/app_in_c/Makefile.am
--- motif-2.3.4.orig/demos/programs/Exm/app_in_c/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/Exm/app_in_c/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,7 +9,7 @@
 
 EXTRA_DIST = exm_in_c.man Imakefile
 
-INCLUDES = -I$(srcdir)/../../../../lib \
+AM_CPPFLAGS = -I$(srcdir)/../../../../lib \
            -I$(srcdir)/../../../lib \
            -I$(top_builddir)/lib \
             ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/Exm/app_in_uil/Makefile.am motif-2.3.4/demos/programs/Exm/app_in_uil/Makefile.am
--- motif-2.3.4.orig/demos/programs/Exm/app_in_uil/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/Exm/app_in_uil/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -10,7 +10,7 @@
 LDADD = -L../../../lib/Exm -lExmMrm -lExm  $(top_builddir)/lib/Mrm/libMrm.la
 
 
-INCLUDES = -I$(top_srcdir)/lib -I$(top_srcdir)/lib -I../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(top_srcdir)/lib -I$(top_srcdir)/lib -I../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 
 SUFFIXES = .uil .uid
 
diff -Naur motif-2.3.4.orig/demos/programs/Exm/simple_app/Makefile.am motif-2.3.4/demos/programs/Exm/simple_app/Makefile.am
--- motif-2.3.4.orig/demos/programs/Exm/simple_app/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/Exm/simple_app/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -10,7 +10,7 @@
 
 EXTRA_DIST = simpleDemo.man README Imakefile
 
-INCLUDES = -I$(srcdir)/../../../../lib \
+AM_CPPFLAGS = -I$(srcdir)/../../../../lib \
            -I$(srcdir)/../../../lib \
            -I$(top_builddir)/lib \
            ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/Ext18List/Makefile.am motif-2.3.4/demos/programs/Ext18List/Makefile.am
--- motif-2.3.4.orig/demos/programs/Ext18List/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/Ext18List/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -12,4 +12,4 @@
 TOPDIR = ../../..
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
 
-INCLUDES = -I$(TOPDIR)/lib -I.. -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I.. -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/FontSel/Makefile.am motif-2.3.4/demos/programs/FontSel/Makefile.am
--- motif-2.3.4.orig/demos/programs/FontSel/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/FontSel/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,4 +9,4 @@
 data_DATA = $(EXTRA_DIST) $(fontsel_SOURCES)
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
 TOPDIR = ../../..
-INCLUDES = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/IconB/Makefile.am motif-2.3.4/demos/programs/IconB/Makefile.am
--- motif-2.3.4.orig/demos/programs/IconB/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/IconB/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -10,6 +10,6 @@
 TOPDIR = ../../..
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
 
-INCLUDES = -I$(TOPDIR)/lib -I$(srcdir)/.. \
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I$(srcdir)/.. \
            -I$(srcdir)/$(TOPDIR)/lib \
            ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/Outline/Makefile.am motif-2.3.4/demos/programs/Outline/Makefile.am
--- motif-2.3.4.orig/demos/programs/Outline/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/Outline/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,4 +9,4 @@
 data_DATA = $(outline_SOURCES) $(EXTRA_DIST)
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
 TOPDIR = ../../..
-INCLUDES = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/Paned/Makefile.am motif-2.3.4/demos/programs/Paned/Makefile.am
--- motif-2.3.4.orig/demos/programs/Paned/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/Paned/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -10,4 +10,4 @@
 TOPDIR = ../../..
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
 
-INCLUDES = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/TabStack/Makefile.am motif-2.3.4/demos/programs/TabStack/Makefile.am
--- motif-2.3.4.orig/demos/programs/TabStack/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/TabStack/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -10,4 +10,4 @@
 TOPDIR = ../../..
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
 
-INCLUDES = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/Tree/Makefile.am motif-2.3.4/demos/programs/Tree/Makefile.am
--- motif-2.3.4.orig/demos/programs/Tree/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/Tree/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -10,4 +10,4 @@
 TOPDIR = ../../..
 LDADD = $(TOPDIR)/lib/Xm/libXm.la
 
-INCLUDES = -I$(TOPDIR)/lib -I$(srcdir)/.. -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(TOPDIR)/lib -I$(srcdir)/.. -I$(srcdir)/$(TOPDIR)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/airport/Makefile.am motif-2.3.4/demos/programs/airport/Makefile.am
--- motif-2.3.4.orig/demos/programs/airport/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/airport/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,4 +9,4 @@
 EXTRA_DIST = Imakefile XmdAirport.ad README
 data_DATA = $(airport_SOURCES) $(EXTRA_DIST)
 LDADD = ../../../lib/Xm/libXm.la
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/animate/Makefile.am motif-2.3.4/demos/programs/animate/Makefile.am
--- motif-2.3.4.orig/demos/programs/animate/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/animate/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -19,7 +19,7 @@
 
 LDADD = ../../lib/Xmd/libXmd.a ../../../lib/Mrm/libMrm.la ../../../lib/Xm/libXm.la
 
-INCLUDES = -I$(srcdir)/../../../lib \
+AM_CPPFLAGS = -I$(srcdir)/../../../lib \
            -I$(srcdir)/../../lib \
            -I$(top_builddir)/lib \
            ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/drag_and_drop/Makefile.am motif-2.3.4/demos/programs/drag_and_drop/Makefile.am
--- motif-2.3.4.orig/demos/programs/drag_and_drop/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/drag_and_drop/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -12,4 +12,4 @@
 noinst_HEADERS = DNDDemo.h
 LDADD = ../../../lib/Xm/libXm.la
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/draw/Makefile.am motif-2.3.4/demos/programs/draw/Makefile.am
--- motif-2.3.4.orig/demos/programs/draw/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/draw/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,4 +8,4 @@
 EXTRA_DIST = Imakefile XmdDraw.ad README draw.help draw.man
 data_DATA = $(EXTRA_DIST) $(draw_SOURCES)
 LDADD = ../../lib/Xmd/libXmd.a ../../../lib/Xm/libXm.la
-INCLUDES = -I$(srcdir)/../../lib -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../lib -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/earth/Makefile.am motif-2.3.4/demos/programs/earth/Makefile.am
--- motif-2.3.4.orig/demos/programs/earth/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/earth/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,4 +8,4 @@
 EXTRA_DIST = terre.xbm earth.man Imakefile README
 data_DATA = $(EXTRA_DIST) $(earth_SOURCES)
 LDADD = ../../../lib/Xm/libXm.la
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/filemanager/Makefile.am motif-2.3.4/demos/programs/filemanager/Makefile.am
--- motif-2.3.4.orig/demos/programs/filemanager/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/filemanager/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -12,6 +12,6 @@
            s_none.xpm s_pix.xpm
 data_DATA = $(EXTRA_DIST) $(filemanager_SOURCES)
 LDADD = ../../lib/Xmd/libXmd.a ../../../lib/Xm/libXm.la
-INCLUDES = -I$(srcdir)/../../lib -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../lib -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 
 
diff -Naur motif-2.3.4.orig/demos/programs/fileview/Makefile.am motif-2.3.4/demos/programs/fileview/Makefile.am
--- motif-2.3.4.orig/demos/programs/fileview/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/fileview/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -13,7 +13,7 @@
 
 LDADD = ../../../lib/Xm/libXm.la ../../../lib/Mrm/libMrm.la
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 
 all: French.uid German.uid English.uid
 
diff -Naur motif-2.3.4.orig/demos/programs/getsubres/Makefile.am motif-2.3.4/demos/programs/getsubres/Makefile.am
--- motif-2.3.4.orig/demos/programs/getsubres/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/getsubres/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,4 +8,4 @@
 getsubres_SOURCES = getsubres.c
 data_DATA = $(EXTRA_DIST) $(getsubres_SOURCES)
 LDADD = ../../lib/Xmd/libXmd.a ../../../lib/Xm/libXm.la
-INCLUDES = -I$(srcdir)/../../lib -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../lib -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/hellomotif/Makefile.am motif-2.3.4/demos/programs/hellomotif/Makefile.am
--- motif-2.3.4.orig/demos/programs/hellomotif/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/hellomotif/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,7 +8,7 @@
 hellomotif_SOURCES = hellomotif.c
 data_DATA = $(EXTRA_DIST) $(hellomotif_SOURCES) hellomotif.uid
 LDADD = ../../../lib/Xm/libXm.la ../../../lib/Mrm/libMrm.la
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 
 all: hellomotif.uid
 
diff -Naur motif-2.3.4.orig/demos/programs/hellomotifi18n/Makefile.am motif-2.3.4/demos/programs/hellomotifi18n/Makefile.am
--- motif-2.3.4.orig/demos/programs/hellomotifi18n/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/hellomotifi18n/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -11,7 +11,7 @@
 data_DATA = $(EXTRA_DIST) $(helloint_SOURCES) helloint.uid
 helloint_DEPENDENCIES = helloint.uid C/uid/l_strings.uid english/uid/l_strings.uid french/uid/l_strings.uid hebrew/uid/l_strings.uid japan/uid/l_strings.uid japanese/uid/l_strings.uid swedish/uid/l_strings.uid
 
-INCLUDES = -I$(top_srcdir)/lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(top_srcdir)/lib -I$(top_builddir)/lib ${X_CFLAGS} 
 LDADD = $(top_builddir)/lib/Mrm/libMrm.la $(top_builddir)/lib/Xm/libXm.la
 
 SUFFIXES = .uil .uid
diff -Naur motif-2.3.4.orig/demos/programs/i18ninput/Makefile.am motif-2.3.4/demos/programs/i18ninput/Makefile.am
--- motif-2.3.4.orig/demos/programs/i18ninput/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/i18ninput/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -10,4 +10,4 @@
 data_DATA = $(EXTRA_DIST) $(i18ninput_SOURCES)
 LDADD = ../../lib/Xmd/libXmd.a ../../../lib/Xm/libXm.la
 
-INCLUDES = -I$(srcdir)/../../lib -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../lib -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/panner/Makefile.am motif-2.3.4/demos/programs/panner/Makefile.am
--- motif-2.3.4.orig/demos/programs/panner/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/panner/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,4 +8,4 @@
 EXTRA_DIST = README panner.man Imakefile
 data_DATA = $(EXTRA_DIST) $(panner_SOURCES)
 LDADD = ../../../lib/Xm/libXm.la
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/periodic/Makefile.am motif-2.3.4/demos/programs/periodic/Makefile.am
--- motif-2.3.4.orig/demos/programs/periodic/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/periodic/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,7 +9,7 @@
 data_DATA = $(EXTRA_DIST) $(periodic_SOURCES) periodic.uid
 LDADD = ../../../lib/Xm/libXm.la ../../../lib/Mrm/libMrm.la
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 
 all: periodic.uid
 
diff -Naur motif-2.3.4.orig/demos/programs/piano/Makefile.am motif-2.3.4/demos/programs/piano/Makefile.am
--- motif-2.3.4.orig/demos/programs/piano/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/piano/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -10,4 +10,4 @@
 data_DATA  = $(EXTRA_DIST) $(piano_SOURCES) 
 LDADD = ../../../lib/Xm/libXm.la -lm
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/popups/Makefile.am motif-2.3.4/demos/programs/popups/Makefile.am
--- motif-2.3.4.orig/demos/programs/popups/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/popups/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,4 +9,4 @@
 data_DATA = $(EXTRA_DIST) $(autopopups_SOURCES)
 LDADD = ../../../lib/Xm/libXm.la
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/sampler2_0/Makefile.am motif-2.3.4/demos/programs/sampler2_0/Makefile.am
--- motif-2.3.4.orig/demos/programs/sampler2_0/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/sampler2_0/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -12,4 +12,4 @@
 data_DATA = $(sampler2_0_SOURCES) $(EXTRA_DIST)
 LDADD = ../../lib/Xmd/libXmd.a ../../../lib/Xm/libXm.la
 
-INCLUDES = -I$(srcdir)/../../lib -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../lib -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/setdate/Makefile.am motif-2.3.4/demos/programs/setdate/Makefile.am
--- motif-2.3.4.orig/demos/programs/setdate/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/setdate/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,4 +9,4 @@
 data_DATA = $(EXTRA_DIST) $(setDate_SOURCES)
 LDADD = ../../../lib/Xm/libXm.la
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/todo/Makefile.am motif-2.3.4/demos/programs/todo/Makefile.am
--- motif-2.3.4.orig/demos/programs/todo/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/todo/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,5 +8,5 @@
 todo_SOURCES = todo.c io.c actions.c page.h
 data_DATA = $(EXTRA_DIST) $(todo_SOURCES)
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(srcdir)/../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(srcdir)/../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 LDADD = -L../../lib/Xmd -lXmd -L../../lib/Exm -lExm ../../../lib/Xm/libXm.la 
diff -Naur motif-2.3.4.orig/demos/programs/tooltips/Makefile.am motif-2.3.4/demos/programs/tooltips/Makefile.am
--- motif-2.3.4.orig/demos/programs/tooltips/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/tooltips/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -9,4 +9,4 @@
 data_DATA = $(tooltips_SOURCES) $(EXTRA_DIST)
 LDADD = ../../../lib/Xm/libXm.la
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/programs/workspace/Makefile.am motif-2.3.4/demos/programs/workspace/Makefile.am
--- motif-2.3.4.orig/demos/programs/workspace/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/programs/workspace/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -10,5 +10,5 @@
               command_ui.h wsm_cb.h wsm_create.h wsmData.h wsmDebug.h \
               wsm.h wsmSend.h wsmStruct.h wsm_ui.h xrmLib.h
 data_DATA = $(wsm_SOURCES) $(EXTRA_DIST)
-INCLUDES = -I$(srcdir)/../../../lib -I$(srcdir)/../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(srcdir)/../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 LDADD = ../../../lib/Xm/libXm.la -L../../lib/Wsm -lWsm
diff -Naur motif-2.3.4.orig/demos/unsupported/Exm/Makefile.am motif-2.3.4/demos/unsupported/Exm/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/Exm/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/Exm/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -7,4 +7,4 @@
 
 EXTRA_DIST = Imakefile
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(srcdir)/.. -I$(srcdir)/../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(srcdir)/.. -I$(srcdir)/../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/unsupported/aicon/Makefile.am motif-2.3.4/demos/unsupported/aicon/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/aicon/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/aicon/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -4,6 +4,6 @@
  
 LDADD = -L../../lib/Xmd -lXmd ../../../lib/Xm/libXm.la 
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(srcdir)/../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(srcdir)/../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 
 EXTRA_DIST = Imakefile
diff -Naur motif-2.3.4.orig/demos/unsupported/dainput/Makefile.am motif-2.3.4/demos/unsupported/dainput/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/dainput/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/dainput/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,6 +8,6 @@
 
 EXTRA_DIST = Imakefile
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 
 noinst_HEADERS = dainput.h
diff -Naur motif-2.3.4.orig/demos/unsupported/dogs/Makefile.am motif-2.3.4/demos/unsupported/dogs/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/dogs/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/dogs/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -18,7 +18,7 @@
 
 LDADD = $(top_builddir)/lib/Mrm/libMrm.la $(top_builddir)/lib/Xm/libXm.la
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
 
 SUFFIXES = .uil .uid
 
diff -Naur motif-2.3.4.orig/demos/unsupported/hellomotif/Makefile.am motif-2.3.4/demos/unsupported/hellomotif/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/hellomotif/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/hellomotif/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,4 +8,4 @@
 
 EXTRA_DIST = Imakefile
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/unsupported/motifshell/Makefile.am motif-2.3.4/demos/unsupported/motifshell/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/motifshell/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/motifshell/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,4 +8,4 @@
 
 EXTRA_DIST = Imakefile
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/unsupported/uilsymdump/Makefile.am motif-2.3.4/demos/unsupported/uilsymdump/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/uilsymdump/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/uilsymdump/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -2,7 +2,7 @@
 
 noinst_PROGRAMS = uilsymdump
 
-INCLUDES = -I$(srcdir)/../../../clients -I$(srcdir)/../../../lib -I../../../clients -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../clients -I$(srcdir)/../../../lib -I../../../clients -I$(top_builddir)/lib ${X_CFLAGS} 
 
 EXTRA_DIST = Imakefile
 
diff -Naur motif-2.3.4.orig/demos/unsupported/xmapdef/Makefile.am motif-2.3.4/demos/unsupported/xmapdef/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/xmapdef/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/xmapdef/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,4 +8,4 @@
 
 EXTRA_DIST = Imakefile
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/unsupported/xmfonts/Makefile.am motif-2.3.4/demos/unsupported/xmfonts/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/xmfonts/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/xmfonts/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,4 +8,4 @@
 
 EXTRA_DIST = Imakefile
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/unsupported/xmforc/Makefile.am motif-2.3.4/demos/unsupported/xmforc/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/xmforc/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/xmforc/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,4 +8,4 @@
 
 EXTRA_DIST = Imakefile
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS} 
diff -Naur motif-2.3.4.orig/demos/unsupported/xmform/Makefile.am motif-2.3.4/demos/unsupported/xmform/Makefile.am
--- motif-2.3.4.orig/demos/unsupported/xmform/Makefile.am	2012-10-22 16:50:38.000000000 +0200
+++ motif-2.3.4/demos/unsupported/xmform/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -8,4 +8,4 @@
 
 LDADD = ../../../lib/Xm/libXm.la
 
-INCLUDES = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS}
+AM_CPPFLAGS = -I$(srcdir)/../../../lib -I$(top_builddir)/lib ${X_CFLAGS}
diff -Naur motif-2.3.4.orig/lib/Mrm/Makefile.am motif-2.3.4/lib/Mrm/Makefile.am
--- motif-2.3.4.orig/lib/Mrm/Makefile.am	2012-10-22 16:50:34.000000000 +0200
+++ motif-2.3.4/lib/Mrm/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -2,7 +2,7 @@
 CLEANFILES = $(BUILT_SOURCES)
 EXTRA_DIST = Mrm.msg Imakefile
 
-INCLUDES = -I$(srcdir)/.. -I.. -I../Xm ${X_CFLAGS}
+AM_CPPFLAGS = -I$(srcdir)/.. -I.. -I../Xm ${X_CFLAGS}
 
 mrmdir = $(includedir)/Mrm
 
diff -Naur motif-2.3.4.orig/lib/Xm/Makefile.am motif-2.3.4/lib/Xm/Makefile.am
--- motif-2.3.4.orig/lib/Xm/Makefile.am	2012-10-22 16:50:34.000000000 +0200
+++ motif-2.3.4/lib/Xm/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -5,7 +5,7 @@
 EXTRA_DIST = xmstring.list Xm.msg Imakefile \
              XmStrDefs.ct
 
-INCLUDES = -I.. -I$(srcdir)/.. -DXMBINDDIR_FALLBACK=\"@XMBINDDIR_FALLBACK@\" \
+AM_CPPFLAGS = -I.. -I$(srcdir)/.. -DXMBINDDIR_FALLBACK=\"@XMBINDDIR_FALLBACK@\" \
            -DINCDIR=\"@INCDIR@\" \
            -DLIBDIR=\"@LIBDIR@\" \
            ${X_CFLAGS}
diff -Naur motif-2.3.4.orig/tools/wml/Makefile.am motif-2.3.4/tools/wml/Makefile.am
--- motif-2.3.4.orig/tools/wml/Makefile.am	2012-10-22 16:50:39.000000000 +0200
+++ motif-2.3.4/tools/wml/Makefile.am	2013-06-23 17:28:06.000000000 +0200
@@ -32,7 +32,7 @@
 
 wmldbcreate_LDADD = ../../lib/Xm/libXm.la
 
-INCLUDES = -I../../lib -I$(srcdir)/../../lib -I$(srcdir)/../../include ${X_CFLAGS} 
+AM_CPPFLAGS = -I../../lib -I$(srcdir)/../../lib -I$(srcdir)/../../include ${X_CFLAGS} 
 
 noinst_LIBRARIES = libwml.a
 libwml_a_SOURCES = wmlparse.y wml.c wmloutkey.c wmlouth.c wmloutmm.c wmloutp1.c wmlresolve.c wmlsynbld.c wmlutils.c
