From ba735cd0edc79a0880c43313e5b954d6e08a677f Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sun, 28 Feb 2016 16:17:31 +0100
Subject: [PATCH] do not build gr-trellis documentation

---
 gr-trellis/CMakeLists.txt | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/gr-trellis/CMakeLists.txt b/gr-trellis/CMakeLists.txt
index a3f0ad4..cb6ab71 100644
--- a/gr-trellis/CMakeLists.txt
+++ b/gr-trellis/CMakeLists.txt
@@ -56,11 +56,11 @@ if(ENABLE_GR_TRELLIS)
 include(GrPackage)
 CPACK_SET(CPACK_COMPONENT_GROUP_TRELLIS_DESCRIPTION "GNU Radio Trellis Blocks")
 
-CPACK_COMPONENT("trellis_docs"
-    GROUP        "Trellis"
-    DISPLAY_NAME "Documentation"
-    DESCRIPTION  "Doxygen HTML and XML"
-)
+#CPACK_COMPONENT("trellis_docs"
+#    GROUP        "Trellis"
+#    DISPLAY_NAME "Documentation"
+#    DESCRIPTION  "Doxygen HTML and XML"
+#)
 
 CPACK_COMPONENT("trellis_runtime"
     GROUP        "Trellis"
@@ -102,7 +102,7 @@ CPACK_COMPONENT("trellis_swig"
 ########################################################################
 add_subdirectory(include/gnuradio/trellis)
 add_subdirectory(lib)
-add_subdirectory(doc)
+#add_subdirectory(doc)
 if(ENABLE_PYTHON)
     add_subdirectory(swig)
     add_subdirectory(python/trellis)
-- 
2.7.0

