--- src/conf.c.orig	2012-01-18 12:31:45.000000000 +0400
+++ src/conf.c	2014-05-11 15:40:03.205332624 +0400
@@ -416,6 +416,7 @@
 	conf.localpartcmp = strcmp;
 	conf.max_defer_time = 86400 * 4;  /* 4 days */
 	conf.max_msg_size = 0; /* no limit on msg size */
+	conf.log_dir = LOG_DIR;
 	conf.spool_dir = SPOOL_DIR;
 	conf.mail_dir = "/var/mail";
 	conf.listen_addresses = g_list_append(NULL, parse_interface("localhost", 25));
--- src/masqmail.c.orig	2012-01-18 12:31:45.000000000 +0400
+++ src/masqmail.c	2014-05-11 15:48:18.197013226 +0400
@@ -640,14 +640,10 @@
 		}
 	}
 
-	conf.log_dir = LOG_DIR;
-	/* FIXME: fails if we run as user */
-	logopen();
 	if (!read_conf(conf_file)) {
 		logwrite(LOG_ALERT, "SHUTTING DOWN due to problems reading config\n");
 		exit(5);
 	}
-	logclose();
 
 	if (do_queue) {
 		conf.do_queue = TRUE;
