--- src/conf.c.orig	2015-11-18 13:43:02.334209797 +0300
+++ src/conf.c	2015-11-18 14:04:57.175247427 +0300
@@ -406,6 +406,7 @@
 	conf.max_defer_time = 86400 * 4;  /* 4 days */
 	conf.max_msg_size = 0; /* no limit on msg size */
 	conf.lock_dir = LOCK_DIR;
+	conf.log_dir = LOG_DIR;
 	conf.spool_dir = SPOOL_DIR;
 	conf.mail_dir = "/var/mail";
 
--- src/masqmail.c.orig	2015-11-18 13:43:02.372210826 +0300
+++ src/masqmail.c	2015-11-18 14:04:02.332759360 +0300
@@ -712,16 +712,12 @@
 		}
 	}
 
-	conf.log_dir = LOG_DIR;
 	conf.debug_level = debug_level;  /* for debuggin during read_conf() */
-	/* FIXME: fails if we run as user */
-	logopen();
 	if (!read_conf(conf_file)) {
 		logwrite(LOG_ALERT, "SHUTTING DOWN due to problems reading "
 				"config\n");
 		exit(5);
 	}
-	logclose();
 
 	if (do_queue) {
 		conf.do_queue = TRUE;
