diff -Naur laptop-mode-tools_1.47.old/install.sh laptop-mode-tools_1.47/install.sh
--- laptop-mode-tools_1.47.old/install.sh	2009-01-26 20:45:20.000000000 +0100
+++ laptop-mode-tools_1.47/install.sh	2009-03-12 17:52:42.000000000 +0100
@@ -212,9 +212,6 @@
 		echo "$0: Failed to install ACPI event file in $DESTDIR/etc/acpi/events. Installation failed."
 		exit 14
 	fi
-	if [ -z "$DESTDIR" ] ; then
-		killall -HUP acpid
-	fi
 	echo "Installed ACPI support."
 	ACPI_DONE=1
 fi
diff -Naur laptop-mode-tools_1.47.old/usr/sbin/laptop_mode laptop-mode-tools_1.47/usr/sbin/laptop_mode
--- laptop-mode-tools_1.47.old/usr/sbin/laptop_mode	2009-01-26 20:45:20.000000000 +0100
+++ laptop-mode-tools_1.47/usr/sbin/laptop_mode	2009-03-12 17:56:35.000000000 +0100
@@ -393,19 +393,8 @@
 
 #############################################################################
 
-KLEVEL="$(uname -r |
-			 {
-				 IFS='.-' read a b c
-				 echo $a.$b
-			 }
-)"
-KMINOR="$(uname -r |
-			{
-				IFS='.-' read a b c d
-				# Strip any stuff from the end -- only the initial digits are part of the KMINOR.
-				echo $c | sed -e 's/\([[:digit:]]*\).*/\1/'
-			}
-)"
+KLEVEL=`uname -r | sed 's/^\([0-9][0-9]*\.[0-9][0-9]*\)\..*/\1/'`
+KMINOR=`uname -r | sed 's/^[0-9][0-9]*\.[0-9][0-9]*\.\(.*\)/\1/'`
 
 case "$KLEVEL" in
 	"2.4" ) ;;
@@ -895,7 +884,7 @@
 if [ -f /var/run/laptop-mode-tools/state ] ; then
 	read WAS_ACTIVE WAS_ON_AC WAS_ACTIVATE_WITH_POSSIBLE_DATA_LOSS WAS_STATE < /var/run/laptop-mode-tools/state
 	if [ "$WAS_STATE" != "" ] ; then
-		if [ "$WAS_ACTIVE" -eq "$ACTIVATE" -a "$WAS_ON_AC" -eq "$ON_AC" -a "$WAS_ACTIVATE_WITH_POSSIBLE_DATA_LOSS" -eq "$ACTIVATE_WITH_POSSIBLE_DATA_LOSS" -a "$WAS_STATE" = "$STATE" -a "$FORCE" -eq 0 ] ; then
+		if [ "$WAS_ACTIVE" = "$ACTIVATE" -a "$WAS_ON_AC" = "$ON_AC" -a "$WAS_ACTIVATE_WITH_POSSIBLE_DATA_LOSS" = "$ACTIVATE_WITH_POSSIBLE_DATA_LOSS" -a "$WAS_STATE" = "$STATE" -a "$FORCE" = 0 ] ; then
 			echo -n "$STATE, "
 			if [ "$WAS_ACTIVE" -eq 1 ] ; then
 				echo -n "active [unchanged]"
@@ -934,7 +923,7 @@
 
 # Note that the /usr/local/lib path is deprecated.
 export FORCE STATE ON_AC ACTIVATE ACTIVATE_WITH_POSSIBLE_DATA_LOSS KLEVEL KMINOR WAS_ACTIVE LM_VERBOSE
-for SCRIPT in /usr/share/laptop-mode-tools/modules/* /usr/local/lib/laptop-mode-tools/modules/* /usr/local/share/laptop-mode-tools/modules/* /etc/laptop-mode/modules/*; do
+for SCRIPT in /usr/share/laptop-mode-tools/modules/* /etc/laptop-mode/modules/*; do
 	if [ -x "$SCRIPT" ] ; then
 		$LM_VERBOSE && echo "Invoking module $SCRIPT." >> $OUTPUT
 		"$SCRIPT" >> $OUTPUT
