--- src/gtkknob.c~	2007-05-03 22:57:55.000000000 +0300
+++ src/gtkknob.c	2007-05-03 22:59:02.000000000 +0300
@@ -238,7 +238,7 @@
     attributes_mask = GDK_WA_X | GDK_WA_Y | GDK_WA_VISUAL | GDK_WA_COLORMAP;
     widget->window = gdk_window_new(widget->parent->window, &attributes, attributes_mask);
 
-    widget->style = gtk_style_attach(widget->parent->style, widget->window);
+    widget->style = gtk_style_attach(widget->style, widget->window);
 
     gdk_window_set_user_data(widget->window, widget);
 
