*** xglobe-0.52/src/earthapp.moc.cpp.orig	2004-12-10 10:19:17.506405480 -0800
--- xglobe-0.52/src/earthapp.moc.cpp	2004-12-10 10:19:25.239229912 -0800
***************
*** 11,25 ****
  #include "earthapp.h"
  #include <qmetaobject.h>
  #include <qapplication.h>
  
  #include <private/qucomextra_p.h>
- #if !defined(Q_MOC_OUTPUT_REVISION) || (Q_MOC_OUTPUT_REVISION != 19)
- #error "This file was generated using the moc from 3.0.6-snapshot-20020712. It"
- #error "cannot be used with the include files from this version of Qt."
- #error "(The moc has changed too much.)"
- #endif
  
  const char *EarthApplication::className() const
  {
      return "EarthApplication";
  }
--- 11,20 ----
