From d5b190f47fa53c5440316170690fdaca96a76667 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sun, 23 Aug 2015 15:20:26 +0200
Subject: [PATCH] fix compile error due to missing include for scoped_ptr

---
 include/avhttp/multi_download.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/avhttp/multi_download.hpp b/include/avhttp/multi_download.hpp
index fd17ad5..72ac8fa 100644
--- a/include/avhttp/multi_download.hpp
+++ b/include/avhttp/multi_download.hpp
@@ -25,6 +25,7 @@
 #include <boost/shared_ptr.hpp>
 #include <boost/make_shared.hpp>
 #include <boost/thread.hpp>
+#include <boost/scoped_ptr.hpp>
 #include <boost/thread/condition.hpp>
 #include <boost/lexical_cast.hpp>
 #include <boost/date_time.hpp>
-- 
2.4.1

