From b3ea76eebbd5980450d1ef01d00e8668e10001a3 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sun, 28 Aug 2016 00:32:19 +0200
Subject: [PATCH] fix build of qtwebkit with gcc 6

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 10ad7ca..1f233fd 100755
--- a/configure
+++ b/configure
@@ -7734,7 +7734,7 @@ case "$XPLATFORM" in
     *-g++*)
 	# Check gcc's version
 	case "$(${QMAKE_CONF_COMPILER} -dumpversion)" in
-	    5*|4*|3.4*)
+	    6*|5*|4*|3.4*)
 		;;
             3.3*)
                 canBuildWebKit="no"
-- 
2.9.3

