--- src/network/ssl/qsslcertificate.cpp	2009-09-05 11:24:45.000000000 +0200
+++ src/network/ssl/qsslcertificate.cpp.cp	2009-09-05 11:54:28.000000000 +0200
@@ -377,7 +377,7 @@
             }
 
             const char *altNameStr = reinterpret_cast<const char *>(q_ASN1_STRING_data(genName->d.ia5));
-            const QString altName = QLatin1String(QByteArray(altNameStr, len));
+            const QString altName = QString::fromLatin1(altNameStr, len);
             if (genName->type == GEN_DNS)
                 result.insert(QSsl::DnsEntry, altName);
             else if (genName->type == GEN_EMAIL)
