--- src/sql/drivers/odbc/qsql_odbc.cpp.orig	2009-05-03 10:13:19.000000000 -0500
+++ src/sql/drivers/odbc/qsql_odbc.cpp	2009-05-03 10:13:58.000000000 -0500
@@ -69,13 +69,13 @@
 #endif
 
 // newer platform SDKs use SQLLEN instead of SQLINTEGER
-#if defined(SQLLEN) || defined(Q_OS_WIN64)
+#if defined(SQLLEN) || defined(Q_OS_WIN64) || defined(Q_OS_UNIX)
 # define QSQLLEN SQLLEN
 #else
 # define QSQLLEN SQLINTEGER
 #endif
 
-#if defined(SQLULEN) || defined(Q_OS_WIN64)
+#if defined(SQLULEN) || defined(Q_OS_WIN64) || defined(Q_OS_UNIX)
 # define QSQLULEN SQLULEN
 #else
 # define QSQLULEN SQLUINTEGER
