diff -uNr picoODBC-2.0.2/odbccfg/odbccfg.c picoODBC-2.0.2.cp/odbccfg/odbccfg.c
--- picoODBC-2.0.2/odbccfg/odbccfg.c	2004-05-10 12:16:40.000000000 +0200
+++ picoODBC-2.0.2.cp/odbccfg/odbccfg.c	2006-05-27 18:04:42.000000000 +0200
@@ -23,6 +23,8 @@
  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 #include <odbcinstext.h>
+#include <string.h>
+#include <memory.h>
 
 static const char *server[] =
 {
