--- udftools-1.0.0b3.orig/wrudf/wrudf.c	2002-11-25 23:18:51.000000000 -0800
+++ udftools-1.0.0b3/wrudf/wrudf.c	2006-04-18 10:34:59.177536143 -0700
@@ -245,7 +245,7 @@ initialise(char *devicename) 
 	    } else if( strncmp( spm->partIdent.ident, UDF_ID_VIRTUAL, strlen(UDF_ID_VIRTUAL)) == 0 )
 		virtualPartitionNum = i;
 	}
-	(char*)spm += spm->partitionMapLength;
+	spm = (char*)spm + spm->partitionMapLength;
     }
 
     if( medium == CDR ) {
