From 39ae5dc10de7ec7645db464a6d4e5b8955255250 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 20:14:29 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 stringsearch.cabal | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/stringsearch.cabal b/stringsearch.cabal
index f94e05d..004baf8 100644
--- a/stringsearch.cabal
+++ b/stringsearch.cabal
@@ -82,14 +82,14 @@ Library
 
   -- Packages needed in order to build this package.
   if flag(base4)
-    Build-depends:   base >= 4 && < 5, array >= 0.3 && < 0.6,
-                     bytestring >= 0.9 && < 1, containers >= 0.3 && < 0.6
+    Build-depends:   base >= 4, array >= 0.3,
+                     bytestring >= 0.9 && < 1, containers >= 0.3
   else
     if flag(base3)
-      Build-depends: base >= 3 && < 4, array >= 0.1 && < 0.4,
-                     bytestring >= 0.9 && < 1, containers >= 0.1 && < 0.4
+      Build-depends: base >= 3, array >= 0.1,
+                     bytestring >= 0.9 && < 1, containers >= 0.1
     else
-      Build-depends: base >= 2 && < 3
+      Build-depends: base >= 2
 
   Extensions:        BangPatterns
   if flag(base4)
-- 
2.19.1

