Index: camel-mime-utils.c
===================================================================
RCS file: /cvs/gnome/evolution/camel/camel-mime-utils.c,v
retrieving revision 1.146.6.6
diff -u -r1.146.6.6 camel-mime-utils.c
--- camel-mime-utils.c	12 Apr 2002 06:43:56 -0000	1.146.6.6
+++ camel-mime-utils.c	3 May 2002 03:05:46 -0000
@@ -2367,7 +2367,7 @@
 			} else {
 				/* Fix for stupidly-broken-mailers that like to put '.''s in names unquoted */
 				/* see bug #8147 */
-				if (*inptr && *inptr != '<') {
+				while (!pre && *inptr && *inptr != '<') {
 					w(g_warning("Working around stupid mailer bug #5: unescaped characters in names"));
 					name = g_string_append_c(name, *inptr++);
 					pre = header_decode_word(&inptr);
