--- Makefile~	2004-10-23 22:10:40.000000000 +0200
+++ Makefile	2005-02-05 14:24:34.432207398 +0100
@@ -110,6 +110,9 @@
 install: install-symbolic-link
 .PHONY: install-symbolic-link
 install-symbolic-link: subdir_install
+	-test ! -x $(common-objpfx)elf/ldconfig || LC_ALL=C LANGUAGE=C \
+	  $(common-objpfx)elf/ldconfig $(addprefix -r ,$(install_root)) \
+				       $(slibdir) $(libdir)
 	$(symbolic-link-prog) $(symbolic-link-list)
 	rm -f $(symbolic-link-list)
 
