diff -ru /usr/src/libgda-0.2.96/bindings/c++/gdaBatch.cpp libgda-0.2.96/bindings/c++/gdaBatch.cpp
--- /usr/src/libgda-0.2.96/bindings/c++/gdaBatch.cpp	2001-12-07 22:07:48.000000000 +0100
+++ libgda-0.2.96/bindings/c++/gdaBatch.cpp	2002-06-23 23:15:58.000000000 +0200
@@ -127,7 +127,7 @@
 }
 
 GdaBatch*
-Batch::getCStruct (bool refn = true) const
+Batch::getCStruct (bool refn) const
 {
 	if (refn)
 	  ref ();
diff -ru /usr/src/libgda-0.2.96/bindings/c++/gdaCommand.cpp libgda-0.2.96/bindings/c++/gdaCommand.cpp
--- /usr/src/libgda-0.2.96/bindings/c++/gdaCommand.cpp	2001-12-07 22:07:48.000000000 +0100
+++ libgda-0.2.96/bindings/c++/gdaCommand.cpp	2002-06-23 23:15:58.000000000 +0200
@@ -139,7 +139,7 @@
 */
 
 GdaCommand*
-Command::getCStruct (bool refn = true) const
+Command::getCStruct (bool refn) const
 {
 	if (refn)
 		ref ();
diff -ru /usr/src/libgda-0.2.96/bindings/c++/gdaCommand.h libgda-0.2.96/bindings/c++/gdaCommand.h
--- /usr/src/libgda-0.2.96/bindings/c++/gdaCommand.h	2001-12-07 22:07:48.000000000 +0100
+++ libgda-0.2.96/bindings/c++/gdaCommand.h	2002-06-23 23:15:58.000000000 +0200
@@ -21,6 +21,8 @@
 
 namespace gda {
 
+using std::vector;
+
 class Recordset;
 class Value;
 
diff -ru /usr/src/libgda-0.2.96/bindings/c++/gdaConnection.cpp libgda-0.2.96/bindings/c++/gdaConnection.cpp
--- /usr/src/libgda-0.2.96/bindings/c++/gdaConnection.cpp	2001-12-07 22:07:48.000000000 +0100
+++ libgda-0.2.96/bindings/c++/gdaConnection.cpp	2002-06-23 23:15:58.000000000 +0200
@@ -294,7 +294,7 @@
 	return gda_connection_get_version (_gda_connection);
 }
 
-GdaConnection* Connection::getCStruct (bool refn = true) const
+GdaConnection* Connection::getCStruct (bool refn) const
 {
 	if (refn) ref ();
 	return _gda_connection;
diff -ru /usr/src/libgda-0.2.96/bindings/c++/gdaConnection.h libgda-0.2.96/bindings/c++/gdaConnection.h
--- /usr/src/libgda-0.2.96/bindings/c++/gdaConnection.h	2001-12-07 22:07:48.000000000 +0100
+++ libgda-0.2.96/bindings/c++/gdaConnection.h	2002-06-23 23:15:58.000000000 +0200
@@ -22,6 +22,8 @@
 
 namespace gda {
 
+using std::string;
+
 class Recordset;
 class Error;
 class ErrorList;
diff -ru /usr/src/libgda-0.2.96/bindings/c++/gdaError.cpp libgda-0.2.96/bindings/c++/gdaError.cpp
--- /usr/src/libgda-0.2.96/bindings/c++/gdaError.cpp	2001-12-07 22:07:48.000000000 +0100
+++ libgda-0.2.96/bindings/c++/gdaError.cpp	2002-06-23 23:15:58.000000000 +0200
@@ -98,7 +98,7 @@
 }
 
 GdaError*
-Error::getCStruct (bool refn = true) const
+Error::getCStruct (bool refn) const
 {
 	if (refn) ref ();
 	return _gda_error;
diff -ru /usr/src/libgda-0.2.96/bindings/c++/gdaField.cpp libgda-0.2.96/bindings/c++/gdaField.cpp
--- /usr/src/libgda-0.2.96/bindings/c++/gdaField.cpp	2001-12-11 21:16:14.000000000 +0100
+++ libgda-0.2.96/bindings/c++/gdaField.cpp	2002-06-23 23:15:58.000000000 +0200
@@ -428,7 +428,7 @@
 }
 
 GdaField*
-Field::getCStruct (bool refn = true) const
+Field::getCStruct (bool refn) const
 {
 	if (refn) ref ();
 	return _gda_field;
diff -ru /usr/src/libgda-0.2.96/bindings/c++/gdaHelpers.cpp libgda-0.2.96/bindings/c++/gdaHelpers.cpp
--- /usr/src/libgda-0.2.96/bindings/c++/gdaHelpers.cpp	2001-11-18 18:21:55.000000000 +0100
+++ libgda-0.2.96/bindings/c++/gdaHelpers.cpp	2002-06-23 23:15:58.000000000 +0200
@@ -23,6 +23,8 @@
 
 //using namespace gda;
 
+using std::string;
+
 string
 gda::gda_return_string (gchar* pszString)
 {
diff -ru /usr/src/libgda-0.2.96/bindings/c++/gdaHelpers.h libgda-0.2.96/bindings/c++/gdaHelpers.h
--- /usr/src/libgda-0.2.96/bindings/c++/gdaHelpers.h	2001-11-18 18:21:55.000000000 +0100
+++ libgda-0.2.96/bindings/c++/gdaHelpers.h	2002-06-23 23:15:58.000000000 +0200
@@ -20,6 +20,6 @@
 //#define __gda_helpers_cpp_gdaHelpersH
 
 namespace gda {
-string gda_return_string (gchar* pszString);
+std::string gda_return_string (gchar* pszString);
 };
 //#endif // __gda_helpers_cpp_gdaHelpersH
diff -ru /usr/src/libgda-0.2.96/bindings/c++/gdaRecordset.cpp libgda-0.2.96/bindings/c++/gdaRecordset.cpp
--- /usr/src/libgda-0.2.96/bindings/c++/gdaRecordset.cpp	2001-12-07 22:07:48.000000000 +0100
+++ libgda-0.2.96/bindings/c++/gdaRecordset.cpp	2002-06-23 23:15:58.000000000 +0200
@@ -254,7 +254,7 @@
 }
 
 GdaRecordset*
-Recordset::getCStruct (bool refn = true) const
+Recordset::getCStruct (bool refn) const
 {
 	if (refn) ref ();
 	return _gda_recordset;
