From 262107bee42f272eb96b471f60d01ab34849116b Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 13 Jan 2013 13:34:50 +0100
Subject: [PATCH] add libm to EXTERNAL_LIBRARIES

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 82c4111..b7ce128 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,6 +44,9 @@ endif(DEFINED CMAKE_BUILD_TYPE)
 
 project(gl2ps C)
 
+find_library(M_LIB m)
+list(APPEND EXTERNAL_LIBRARIES ${M_LIB})
+
 option(ENABLE_ZLIB "Enable compression using ZLIB" ON)
 option(ENABLE_PNG "Enable PNG support" ON)
 
-- 
1.8.1

