From a52db48fbf45620de1b18a937f387528f6abd44b Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 22 May 2022 17:29:35 +0200
Subject: [PATCH] Source Xsession.d files

---
 config/Xsession.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/config/Xsession.in b/config/Xsession.in
index 42a8b7d..c1d7d11 100644
--- a/config/Xsession.in
+++ b/config/Xsession.in
@@ -49,6 +49,12 @@ esac
 
 XCOMM The startup script is not intended to have arguments.
 
+if [ -d /etc/X11/Xsession.d ]; then
+	for i in /etc/X11/Xsession.d/*; do
+		. "$i"
+	done
+fi
+
 startup=$HOME/.xsession
 resources=$HOME/.Xresources
 
-- 
2.36.0

