--- configure.sh.orig	2009-07-25 16:58:00.000000000 +0200
+++ configure.sh	2009-07-25 16:59:10.000000000 +0200
@@ -30,9 +30,9 @@
 
 # Check qmake version
 echo "Check Qt version..."
-QMAKE_VERSION="$($QMAKE_PATH -query QT_VERSION | awk -F "." '{print$1"."$2}')"
+QMAKE_VERSION="$($QMAKE_PATH -query QT_VERSION | awk -F "." '{print$1""$2}')"
 
-if [ $QMAKE_VERSION = "4.4" ]; then
+if [ "$QMAKE_VERSION" -ge "44" ]; then
 	echo -e "OK. Qt Version is $QMAKE_VERSION.\n"
 else
 	echo "Error! Your qt version $QMAKE_VERSION is too low install a newer one."
