--- conf.c.orig	2020-02-05 02:59:09.000000000 -0800
+++ conf.c	2021-07-09 15:36:24.482626925 -0700
@@ -226,7 +226,8 @@
 			if (fingerprint == NULL) {
 				errlogx(EX_CONFIG, "fingerprint allocation failed");
 			}
-			for (unsigned int i = 0; i < SHA256_DIGEST_LENGTH; i++) {
+			unsigned int i;
+			for (i = 0; i < SHA256_DIGEST_LENGTH; i++) {
 				if(sscanf(data + 2 * i, "%02hhx", &fingerprint[i]) != 1) {
 					errlogx(EX_CONFIG, "failed to read fingerprint");
 				}
