diff -ur mlview-0.9.0/src/mlview-editor.cc mlview-0.9.0-patched//src/mlview-editor.cc
--- mlview-0.9.0/src/mlview-editor.cc	2005-08-07 09:42:28.000000000 +0200
+++ mlview-0.9.0-patched//src/mlview-editor.cc	2010-04-06 19:50:37.000000000 +0200
@@ -157,7 +157,7 @@
 			contextual_menu (NULL)
 	{}
 
-	static void EditorPriv::schemas_window_destroy_cb
+	static void schemas_window_destroy_cb
 	(GtkWidget *a_widget, struct DocumentWindowData *a_win) ;
 
 	static void validation_window_destroy_cb
diff -ur mlview-0.9.0/src/mlview-plugin.h mlview-0.9.0-patched//src/mlview-plugin.h
--- mlview-0.9.0/src/mlview-plugin.h	2005-08-07 09:47:54.000000000 +0200
+++ mlview-0.9.0-patched//src/mlview-plugin.h	2010-04-06 19:49:49.000000000 +0200
@@ -46,12 +46,12 @@
 		
 		//forbid assignation/copy
 		Plugin& operator= (Plugin const&) ;
-		Plugin::Plugin (Plugin const&) ;
+		Plugin (Plugin const&) ;
 
 	public:
 		Plugin (const PluginDescriptor &a_descr) ;
 
-		Plugin::Plugin (const UString &a_url) ;
+		Plugin (const UString &a_url) ;
 		
 		virtual ~Plugin () ;
 		
diff -ur mlview-0.9.0/src/mlview-source-view.h mlview-0.9.0-patched//src/mlview-source-view.h
--- mlview-0.9.0/src/mlview-source-view.h	2005-08-07 09:42:28.000000000 +0200
+++ mlview-0.9.0-patched//src/mlview-source-view.h	2010-04-06 19:46:35.000000000 +0200
@@ -53,7 +53,7 @@
 
 	enum MlViewStatus set_default_options () ;
 
-	MlViewXMLDocument* SourceView::get_document () ;
+	MlViewXMLDocument* get_document () ;
 
 	GtkUIManager * get_ui_manager () ;
 
diff -ur mlview-0.9.0/src/mlview-tree-view.h mlview-0.9.0-patched//src/mlview-tree-view.h
--- mlview-0.9.0/src/mlview-tree-view.h	2005-07-22 21:14:28.000000000 +0200
+++ mlview-0.9.0-patched//src/mlview-tree-view.h	2010-04-06 19:46:17.000000000 +0200
@@ -178,7 +178,7 @@
 
 	enum MlViewStatus build_app_edit_menu () ;
 
-	GtkDialog * TreeView::get_expand_tree_dialog ()  ;
+	GtkDialog * get_expand_tree_dialog ()  ;
 
 	void clear_completion_popup_submenus () ;
 
diff -ur mlview-0.9.0/src/mlview-validator.cc mlview-0.9.0-patched//src/mlview-validator.cc
--- mlview-0.9.0/src/mlview-validator.cc	2005-07-31 21:16:16.000000000 +0200
+++ mlview-0.9.0-patched//src/mlview-validator.cc	2010-04-06 19:48:50.000000000 +0200
@@ -26,6 +26,7 @@
 #include "mlview-validator.h"
 #include "mlview-safe-ptr-utils.h"
 
+#include <backward/auto_ptr.h>
 #define MESSAGE_LINE_LENGTH 55
 namespace mlview
 {
