diff -Naur kbd-1.12.orig/data/Makefile kbd-1.12/data/Makefile
--- kbd-1.12.orig/data/Makefile	2005-04-28 15:16:46.446294848 -0700
+++ kbd-1.12/data/Makefile	2005-04-28 15:30:03.916061016 -0700
@@ -29,7 +29,11 @@
 # compress data files - do not touch the distribution but copy first
 $(KEYMAPDIR)_Z:
 	cp -r $(KEYMAPDIR) $(KEYMAPDIR)_Z
-	cd $(KEYMAPDIR)_Z && $(MYGZIP) */*.map */*/*.map
+	for i in $(KEYMAPSUBDIRS); do \
+		if [ "$$i"  !=  "include" ]; then\
+		cd  $(KEYMAPDIR)_Z/$$i  &&  $(MYGZIP) *.map  &&  cd  -;\
+		fi;\
+	done
 
 $(FONTDIR)_Z:
 	cp -r $(FONTDIR) $(FONTDIR)_Z
