diff  -u  Makefile  Makefile
--- Makefile	Wed Dec 12 10:04:08 2001
+++ Makefile	Wed Dec 12 10:15:19 2001
@@ -5,9 +5,9 @@
 
 # Paths
 
-INSTALL = /usr/ucb/install
-EBIN = /usr/local/sbin
-MAN = /usr/local/man/man
+INSTALL = install
+EBIN = /usr/sbin
+MAN = /usr/man/man
 
 # OS-Dependant settings
 
@@ -33,24 +33,24 @@
 # -DCIDR_KEY	# keys in CIDR format
 # -DTERM_KD	# keys and data nul-terminated
 
-DEFS = -DTI_RPC -DFCNTL_LOCK -DSYSINFO
+DEFS = -DSOCK_RPC -DFCNTL_LOCK -DGETHOST -DDASH_C
 
 # Compiler flags 
 CC = cc
 RANLIB = :
-CFLAGS = $(DEFS) -g -I/usr/local/src/db/db-3.1.17/build_unix
+CFLAGS = $(DEFS) -g
 #CFLAGS = $(DEFS) -g -I/usr/local/src/db/db-2.4.14/Unix
 #CFLAGS = $(DEFS) -g -I/usr/local/src/db/db.1.85/PORT/sunos.5.2/include
-LDLIBS = -L/usr/local/src/db/db-3.1.17/build_unix -lnsl -ldb
+LDLIBS = -ldb
 #LDLIBS = -L/usr/local/src/db/db-2.4.14/Unix -lnsl -ldb
 #LDLIBS = -L/usr/local/src/db/db.1.85/PORT/sunos.5.2 -lnsl -ldb
-TSTLIBS = -L. -ldrac -lnsl
-RPCGENFLAGS = 
+TSTLIBS = -L. -ldrac
+RPCGENFLAGS = -C -I
 #RPCGENFLAGS = -C
 
 # Man sections
 MANLIB = 3
-MANADM = 1m
+MANADM = 8
 
 ## Nothing to change after this point
 
