diff --git a/src/tools/pw-cat.c b/src/tools/pw-cat.c
--- a/src/tools/pw-cat.c
+++ b/src/tools/pw-cat.c
@@ -448,10 +448,10 @@ static int av_codec_params_to_audio_info(struct data *data, AVCodecParameters *c
 		info->info.dts.rate = data->rate;
 		info->info.dts.channels = data->channels;
 		switch (codec_params->profile) {
-		case FF_PROFILE_DTS_HD_MA:
+		case AV_PROFILE_DTS_HD_MA:
 			info->info.dts.ext_type = SPA_AUDIO_DTS_EXT_HD_MA;
 			break;
-		case FF_PROFILE_DTS_HD_HRA:
+		case AV_PROFILE_DTS_HD_HRA:
 			info->info.dts.ext_type = SPA_AUDIO_DTS_EXT_HD_HRA;
 			break;
 		default:
