--- src/lib/howl/Posix/posix_interface.c.orig	2007-10-03 10:50:56.000000000 +0200
+++ src/lib/howl/Posix/posix_interface.c	2007-10-03 10:51:02.000000000 +0200
@@ -41,6 +41,10 @@
 typedef __uint32_t 	u32;
 typedef __uint16_t 	u16;
 typedef __uint8_t		u8;
+typedef uint64_t __u64;
+typedef uint32_t __u32;
+typedef uint16_t __u16;
+typedef uint8_t __u8;
 #	include <linux/sockios.h>
 #	include <linux/ethtool.h> 
 #endif
