diff -Nur gSculpt-0.3/Makefile gSculpt-0.3-gcc3/Makefile
--- gSculpt-0.3/Makefile	2002-02-22 15:20:12.000000000 +0100
+++ gSculpt-0.3-gcc3/Makefile	2003-02-09 13:46:24.000000000 +0100
@@ -18,7 +18,7 @@
 
 # makemake variables
 
-DEPENDFLAGS  := -O3 -Wall -Werror -I./ -I./src/ -I./src/gSculptConfig -I./src/MultiThreading -I./src/Patch -I./src/App/ -I./src/Editor -I./src/Events -I./src/Faces -I./src/FileIO -I./src/LogGrid -I./src/Math -I./src/Misc -I./src/Model -I./src/Polyline -I./src/Mesh -I./src/Templates -I./src/Tools -I./src/Vertices -I./src/Edges -I./src/Widgets -I./src/WorkArea -I./src/EventHandlers -I./src/Xform -I./src/main -I./src/Tools/Polyline -I./src/Tools/Select -I./src/Tools/Xform -I./src/CompGeometry -I./src/Journal -I./src/ModelTools -I./src/XformTools -I./src/ObjEdit -I./src/CreateObj -I./src/TypeManager -I./src/ViewTools -I./src/SelectTools -I./src/LoftPolyTools -I./src/PolylineTools -I./src/ToolComponents `gtk-config --cflags` ${SEARCHDIRS}
+DEPENDFLAGS  := -O3 -Wall -I./ -I./src/ -I./src/gSculptConfig -I./src/MultiThreading -I./src/Patch -I./src/App/ -I./src/Editor -I./src/Events -I./src/Faces -I./src/FileIO -I./src/LogGrid -I./src/Math -I./src/Misc -I./src/Model -I./src/Polyline -I./src/Mesh -I./src/Templates -I./src/Tools -I./src/Vertices -I./src/Edges -I./src/Widgets -I./src/WorkArea -I./src/EventHandlers -I./src/Xform -I./src/main -I./src/Tools/Polyline -I./src/Tools/Select -I./src/Tools/Xform -I./src/CompGeometry -I./src/Journal -I./src/ModelTools -I./src/XformTools -I./src/ObjEdit -I./src/CreateObj -I./src/TypeManager -I./src/ViewTools -I./src/SelectTools -I./src/LoftPolyTools -I./src/PolylineTools -I./src/ToolComponents `gtk-config --cflags` ${SEARCHDIRS}
 TOUCHHEADERS := ${MYCODEDIR}/*.h
 
 # C preprocessor (C, C++, FORTRAN)
diff -Nur gSculpt-0.3/src/Edges/CEdgeContainer.h gSculpt-0.3-gcc3/src/Edges/CEdgeContainer.h
--- gSculpt-0.3/src/Edges/CEdgeContainer.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Edges/CEdgeContainer.h	2003-02-09 13:56:21.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Editor/StatusBar.h gSculpt-0.3-gcc3/src/Editor/StatusBar.h
--- gSculpt-0.3/src/Editor/StatusBar.h	2002-02-22 15:22:36.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Editor/StatusBar.h	2003-02-09 13:52:41.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Editor/Toolbox.h gSculpt-0.3-gcc3/src/Editor/Toolbox.h
--- gSculpt-0.3/src/Editor/Toolbox.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Editor/Toolbox.h	2003-02-09 13:58:00.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Events/Event.h gSculpt-0.3-gcc3/src/Events/Event.h
--- gSculpt-0.3/src/Events/Event.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Events/Event.h	2003-02-09 13:51:31.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Faces/CFace.h gSculpt-0.3-gcc3/src/Faces/CFace.h
--- gSculpt-0.3/src/Faces/CFace.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Faces/CFace.h	2003-02-09 13:56:48.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/FileIO/fileio.h gSculpt-0.3-gcc3/src/FileIO/fileio.h
--- gSculpt-0.3/src/FileIO/fileio.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/FileIO/fileio.h	2003-02-09 13:49:15.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Journal/JournalProcess.h gSculpt-0.3-gcc3/src/Journal/JournalProcess.h
--- gSculpt-0.3/src/Journal/JournalProcess.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Journal/JournalProcess.h	2003-02-09 13:53:52.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Journal/JournalTaskProgress.h gSculpt-0.3-gcc3/src/Journal/JournalTaskProgress.h
--- gSculpt-0.3/src/Journal/JournalTaskProgress.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Journal/JournalTaskProgress.h	2003-02-09 13:54:26.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Math/BBox.h gSculpt-0.3-gcc3/src/Math/BBox.h
--- gSculpt-0.3/src/Math/BBox.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Math/BBox.h	2003-02-09 13:50:02.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Model/CModel.cpp gSculpt-0.3-gcc3/src/Model/CModel.cpp
--- gSculpt-0.3/src/Model/CModel.cpp	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Model/CModel.cpp	2003-02-09 13:59:42.000000000 +0100
@@ -68,17 +68,17 @@
 }
 
 
-BBox CModel::getBBox(bool selectedOnly = true) const
+BBox CModel::getBBox(bool selectedOnly) const
 {
 	return BBox();
 }
 
-BBox CModel::getBBox(const Matrix4& m, bool selectedOnly = true) const
+BBox CModel::getBBox(const Matrix4& m, bool selectedOnly) const
 {
 	return BBox();
 }
 
-BBox CModel::getBBox(const Transformation *x, bool selectedOnly = true) const
+BBox CModel::getBBox(const Transformation *x, bool selectedOnly) const
 {
 	return BBox();
 }
diff -Nur gSculpt-0.3/src/ObjEdit/ObjectCollection.cpp gSculpt-0.3-gcc3/src/ObjEdit/ObjectCollection.cpp
--- gSculpt-0.3/src/ObjEdit/ObjectCollection.cpp	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/ObjEdit/ObjectCollection.cpp	2003-02-09 14:02:39.000000000 +0100
@@ -326,7 +326,7 @@
 UniqueId ObjectCollection::getUniqueId(const string& name) const
 {
 	IdTable::const_iterator iter = nameToId.find(name);
-	if ( iter != objects.end() )
+	if ( iter != nameToId.end() )
 	{
 		return (*iter).second;
 	}
diff -Nur gSculpt-0.3/src/Patch/Patch.h gSculpt-0.3-gcc3/src/Patch/Patch.h
--- gSculpt-0.3/src/Patch/Patch.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Patch/Patch.h	2003-02-09 13:47:02.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/SelectTools/ToolSelectBase.cpp gSculpt-0.3-gcc3/src/SelectTools/ToolSelectBase.cpp
--- gSculpt-0.3/src/SelectTools/ToolSelectBase.cpp	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/SelectTools/ToolSelectBase.cpp	2003-02-09 14:09:58.000000000 +0100
@@ -146,7 +146,7 @@
 ToolSelectBase::ToolSelectBase( CadView *v,
 										  ObjPainterManager *pman,
 										  const string& funcName, const string& optTitle,
-										  int tableWidth = 1 )
+										  int tableWidth)
 				: ToolSelectModel( v, pman, funcName, optTitle, tableWidth ),
 				  CadClickAndDragEventHandler( v->getCadArea() )
 {
diff -Nur gSculpt-0.3/src/Tools/ToolGroup.h gSculpt-0.3-gcc3/src/Tools/ToolGroup.h
--- gSculpt-0.3/src/Tools/ToolGroup.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Tools/ToolGroup.h	2003-02-09 13:57:23.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/TypeManager/TypeManager.h gSculpt-0.3-gcc3/src/TypeManager/TypeManager.h
--- gSculpt-0.3/src/TypeManager/TypeManager.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/TypeManager/TypeManager.h	2003-02-09 13:53:12.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Vertices/CVertexContainer.h gSculpt-0.3-gcc3/src/Vertices/CVertexContainer.h
--- gSculpt-0.3/src/Vertices/CVertexContainer.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Vertices/CVertexContainer.h	2003-02-09 13:56:07.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Widgets/ListManager.h gSculpt-0.3-gcc3/src/Widgets/ListManager.h
--- gSculpt-0.3/src/Widgets/ListManager.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Widgets/ListManager.h	2003-02-09 14:14:58.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Widgets/messageBox.h gSculpt-0.3-gcc3/src/Widgets/messageBox.h
--- gSculpt-0.3/src/Widgets/messageBox.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Widgets/messageBox.h	2003-02-09 14:03:19.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Widgets/NameDialog.cpp gSculpt-0.3-gcc3/src/Widgets/NameDialog.cpp
--- gSculpt-0.3/src/Widgets/NameDialog.cpp	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Widgets/NameDialog.cpp	2003-02-09 14:17:08.000000000 +0100
@@ -44,8 +44,8 @@
 }
 
 NameDialog::NameDialog(NameDialogListener *l,
-							  const string& title = "Name",
-							  const string& prompt = "name:")
+							  const string& title,
+							  const string& prompt)
 {
 	listener = l;
 	constructorCommon(title, prompt);
diff -Nur gSculpt-0.3/src/Widgets/NameDialog.h gSculpt-0.3-gcc3/src/Widgets/NameDialog.h
--- gSculpt-0.3/src/Widgets/NameDialog.h	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Widgets/NameDialog.h	2003-02-09 14:16:07.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/Widgets/VectorWidget.cpp gSculpt-0.3-gcc3/src/Widgets/VectorWidget.cpp
--- gSculpt-0.3/src/Widgets/VectorWidget.cpp	2002-02-22 15:30:42.000000000 +0100
+++ gSculpt-0.3-gcc3/src/Widgets/VectorWidget.cpp	2003-02-09 14:14:34.000000000 +0100
@@ -1,3 +1,4 @@
+using namespace std;
 /************************
  This code is released under the terms of the GNU GPL, v2 (or any later version).
  The GNU GPL can be found in the file 'COPYING'.  Copyright Geoffrey French 1999-2001.
diff -Nur gSculpt-0.3/src/XformTools/ToolTransformModel.cpp gSculpt-0.3-gcc3/src/XformTools/ToolTransformModel.cpp
--- gSculpt-0.3/src/XformTools/ToolTransformModel.cpp	2002-01-09 23:22:01.000000000 +0100
+++ gSculpt-0.3-gcc3/src/XformTools/ToolTransformModel.cpp	2003-02-09 14:07:22.000000000 +0100
@@ -15,7 +15,7 @@
 													ObjPainterManager *pman,
 													const string& funcName,
 													const string& optTitle,
-													int tableWidth = 1)
+													int tableWidth)
 					: ToolModifyModel(v, pman, funcName, optTitle, tableWidth)
 {
 }
diff -uNr gSculpt-0.3.old/src/Templates/array.h gSculpt-0.3/src/Templates/array.h
--- gSculpt-0.3.old/src/Templates/array.h	2006-05-29 11:21:06.000000000 +0200
+++ gSculpt-0.3/src/Templates/array.h	2006-05-29 11:21:32.000000000 +0200
@@ -99,20 +99,6 @@
 		calcEnd();
 	}
 
-	//construct from C array
-	array(T *inData, int aSize, int initialCap = aSize)
-	{
-		sz = aSize;
-		cap = initialCap;
-		dlt = __array_defaultDelta;
-		data = new T[cap];
-		for (int i = 0; i < sz; i++)
-		{
-			data[i] = inData[i];
-		}
-		calcEnd();
-	}
-	
 	//construct from C array (2)
 	array(T *inData, int aSize, int initialCap, int delta)
 	{
