diff -purN gnome-vfs-1.0.5/libgnomevfs/gnome-vfs-private-utils.c gnome-vfs/libgnomevfs/gnome-vfs-private-utils.c
--- gnome-vfs-1.0.5/libgnomevfs/gnome-vfs-private-utils.c	2001-07-16 14:21:07.000000000 -0400
+++ gnome-vfs/libgnomevfs/gnome-vfs-private-utils.c	2004-06-22 18:45:53.676653576 -0400
@@ -263,7 +263,7 @@ gnome_vfs_forkexec (const gchar *file_na
  * 
  * Return value: 
  **/
-GnomeVFSProcessResult
+GnomeVFSProcessRunResult
 gnome_vfs_process_run_cancellable (const gchar *file_name,
 				   const gchar * const argv[],
 				   GnomeVFSProcessOptions options,
diff -purN gnome-vfs-1.0.5/libgnomevfs/gnome-vfs-process.c gnome-vfs/libgnomevfs/gnome-vfs-process.c
--- gnome-vfs-1.0.5/libgnomevfs/gnome-vfs-process.c	2001-07-16 14:21:07.000000000 -0400
+++ gnome-vfs/libgnomevfs/gnome-vfs-process.c	2004-06-22 18:51:41.144830384 -0400
@@ -258,7 +258,7 @@ gnome_vfs_process_free (GnomeVFSProcess 
  * 
  * Return value: A numeric value reporting the result of the operation.
  **/
-GnomeVFSProcessRunResult
+GnomeVFSProcessResult
 gnome_vfs_process_signal (GnomeVFSProcess *process,
 			  guint signal_number)
 {
