--- gnome-vfs/modules/cdda-cddb.c.orig	2005-03-11 21:26:03.789158107 -0500
+++ gnome-vfs/modules/cdda-cddb.c	2005-03-11 21:26:47.324426192 -0500
@@ -440,7 +440,7 @@
     query->query_match=MATCH_INEXACT;
     query->query_matches=0;
 
-    while(!CDDBReadLine(socket,inbuffer,256)) {
+    while(query->query_matches < MAX_INEXACT_MATCHES && !CDDBReadLine(socket,inbuffer,256)) {
       query->query_list[query->query_matches].list_genre=
 	CDDBGenreValue(ChopWhite(strtok(inbuffer," ")));
       
