--- src/call_script.cpp	2006-02-06 09:34:44.000000000 +0100
+++ src/call_script.cpp.cp	2006-02-06 09:40:49.000000000 +0100
@@ -149,7 +149,7 @@
 		// NOTE: MEMMAN audits are not called as all pointers will be deleted
 		//       automatically when the child process dies
 		//	 Also, the child process has a copy of the MEMMAN object
-		char **argv = new (char *)[arg_list.size() + 1];
+		char **argv = new char *[arg_list.size() + 1];
 		
 		int idx = 0;
 		for (list<string>::iterator i = arg_list.begin(); 
@@ -175,7 +175,7 @@
 		int start_twinkle_env = environ_size; // Position of Twinkle variables
 		environ_size += 1;
 		
-		char **env = new (char*)[environ_size + 1];
+		char **env = new char*[environ_size + 1];
 		
 		// Copy current environment to child
 		for (int i = 0; environ[i] != NULL; i++) {
