--- installer.sh.old	2005-07-11 20:08:15.000000000 -0400
+++ installer.sh	2005-07-11 20:43:38.000000000 -0400
@@ -49,7 +49,7 @@
 esac
 
 # Default installation dir (you can change this with the --installdir option)
-INSTALLDIR="/usr/local/rkhunter"
+INSTALLDIR="/usr/rkhunter"
 INSTALLPREFIX="./files/"
 
 SHOWUSAGE=0
@@ -113,7 +113,7 @@
 overwrite:filehashsha1.pl:/scripts/filehashsha1.pl:SHA1%%Digest%%generator
 overwrite:showfiles.pl:/scripts/showfiles.pl:Directory%%viewer
 overwrite:backdoorports.dat:/db/backdoorports.dat:Database%%Backdoor%%ports
-overwrite:mirrors.dat:/db/mirrors.dat:Database%%Update%%mirrors
+overwrite:mirrors.dat:/db/mirrors.dat:Database%%Updathttp://lists.ibiblio.org/pipermail/sm-discuss/2005-February/009895.htmle%%mirrors
 overwrite:os.dat:/db/os.dat:Database%%Operating%%Systems
 overwrite:programs_bad.dat:/db/programs_bad.dat:Database%%Program%%versions
 overwrite:programs_good.dat:/db/programs_good.dat:Database%%Program%%versions
@@ -126,8 +126,8 @@
 
 # Prefix: INSTALLDIR
 INSTALLFILES2="
-nooverwrite:rkhunter.conf:/usr/local/etc/rkhunter.conf:RK%%Hunter%%configuration%%file
-overwrite:rkhunter:/usr/local/bin/rkhunter:RK%%Hunter%%binary
+nooverwrite:rkhunter.conf:/usr/etc/rkhunter.conf:RK%%Hunter%%configuration%%file
+overwrite:rkhunter:/usr/bin/rkhunter:RK%%Hunter%%binary
 "
 
 # Create directories (only if they do not exist)
@@ -138,12 +138,12 @@
 ${INSTALLDIR}/lib/rkhunter/db
 ${INSTALLDIR}/lib/rkhunter/docs
 ${INSTALLDIR}/lib/rkhunter/scripts
-${INSTALLDIR}/lib/rkhunter/tmp
-/usr/local/etc
-/usr/local/bin
+${INSTALLDIR}/lib/rkhunter/tmphttp://lists.ibiblio.org/pipermail/sm-discuss/2005-February/009895.html
+/usr/etc
+/usr/bin
 "
 
-CHECKDIR="/usr/local"
+CHECKDIR="/usr"
 
 
 # Functions
@@ -320,7 +320,7 @@
         t15="Instalando "
         t16="Falhou (impossivel encontrar "
         t17="Instalação Concluida."
-        t18="Leia a documentação em /usr/local/rkhunter/docs para maiores informações. Digite 'rkhunter' para executar o rootkit"
+        t18="Leia a documentação em /usr/rkhunter/docs para maiores informações. Digite 'rkhunter' para executar o rootkit"
         t19="Falha na Instalação"
         ;;
 
@@ -534,11 +534,11 @@
 done
 
 # Installation dir to configuration file
-INSTALLDIRCHECK=`cat /usr/local/etc/rkhunter.conf | grep "INSTALLDIR="`
+INSTALLDIRCHECK=`cat /usr/etc/rkhunter.conf | grep "INSTALLDIR="`
 if [ "${INSTALLDIRCHECK}" = "" ]
   then
-    echo "" >> /usr/local/etc/rkhunter.conf
-    echo "INSTALLDIR=${INSTALLDIR}" >> /usr/local/etc/rkhunter.conf
+    echo "" >> /usr/etc/rkhunter.conf
+    echo "INSTALLDIR=${INSTALLDIR}" >> /etc/rkhunter.conf
     echo "Configuration updated with installation path (${INSTALLDIR})"
   else
     echo "Configuration already updated."
@@ -548,7 +548,7 @@
 then
 	echo ""
 	echo $E "$t17"
-	echo "$t18 (/usr/local/bin/rkhunter)"
+	echo "$t18 (/usr/bin/rkhunter)"
 else
 	echo ""
 	echo $E "$t19"
