--- include/Makefile.in.orig	2006-01-24 10:26:18.000000000 -0600
+++ include/Makefile.in	2006-01-24 10:27:03.000000000 -0600
@@ -174,7 +174,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-nobase_include_HEADERS = mhash.h mutils/mhash.h mutils/mglobal.h mutils/mtypes.h mutils/mutils.h mutils/mhash_config.h
+nobase_include_HEADERS = mhash.h mutils/mhash.h mutils/mglobal.h mutils/mtypes.h mutils/mutils.h mutils/mhash_config.h mutils/mincludes.h
 all: all-am
 
 .SUFFIXES:
