--- xcin.orig/etc/xcinrc.LINUX	2003-02-08 17:36:55.000000000 +0800
+++ xcin/etc/xcinrc.LINUX	2003-02-08 21:18:22.000000000 +0800
@@ -111,7 +111,7 @@
     (DEFAULT_IM_MODULE	"gen_inp")
     (DEFAULT_IM_SINMD	"DEFAULT")
     (PHRASE		"default.phr")
-    (CINPUT		(cj chewing simplex phone bimspinyin bimsphone jyutping
+    (CINPUT		(cj chewing simplex phone bimspinyin jyutping
 			 array30 jyutping0 zh_hex))
     (FONTSET		"-sony-*-24-*-iso8859-1,-*-medium-r-*-24-*-big5-0")
     (OVERSPOT_FONTSET	"-sony-*-16-*-iso8859-1,-*-medium-r-*-16-*-big5-0")))
@@ -134,9 +134,10 @@
 	  (END_KEY		YES)
       	  (BEEP_DUPCHAR		YES)))
 
-(define bimsphone@big5
-	'((SETKEY		6)
-	  (MODULE		"bimsphone")))
+(define chewing@big5
+        '((SETKEY       6)
+         (KB_TYPE       "KB_DEFAULT")
+         (MODULE        "chewing")))
 
 (define bimspinyin@big5
 	'((SETKEY		7)
@@ -204,7 +205,7 @@
     (DEFAULT_IM_MODULE	"gen_inp")
     (DEFAULT_IM_SINMD	"DEFAULT")
     (PHRASE		"default.phr")
-    (CINPUT             (cj chewing simplex phone bimspinyin bimsphone jyutping
+    (CINPUT             (cj chewing simplex phone bimspinyin jyutping
 			 array30 jyutping0 zh_hex))
     (FONTSET		"-sony-*-24-*-iso8859-1,-*-medium-r-*-24-*-big5hkscs-0,-*-24-*-c-*-big5hkscs-0")
     (OVERSPOT_FONTSET	"-sony-*-16-*-iso8859-1,-*-medium-r-*-16-*-big5hkscs-0,-*-16-*-c-*-big5hkscs-0")))
@@ -227,9 +228,10 @@
 	  (END_KEY		YES)
 	  (BEEP_DUPCHAR		YES)))
 
-(define bimsphone@big5hkscs
-	'((SETKEY		6)
-	  (MODULE		"bimsphone")))
+(define chewing@big5hkscs
+        '((SETKEY       6)
+         (KB_TYPE       "KB_DEFAULT")
+         (MODULE        "chewing")))
 
 (define bimspinyin@big5hkscs
 	'((SETKEY		7)
@@ -273,8 +275,4 @@
 	  (INP_CNAME_gbk	"0xc4da+0xc2eb")
 	  (BEEP_WRONG		YES)))
 
-(define chewing@big5
-        '((SETKEY       7)
-         (KB_TYPE       "KB_DEFAULT")
-         (MODULE        "chewing")))
 
