--- xcin.orig/src/xim.c	2003-02-08 17:36:55.000000000 +0800
+++ xcin/src/xim.c	2003-02-08 17:37:20.000000000 +0800
@@ -1058,7 +1058,8 @@
     make_trigger_keys(&on_keys);
 
     if (xccore->irc->xim_name[0] == '\0') {
-        if (strcasecmp("zh_TW.Big5", locale->lc_ctype))
+        if (strcasecmp("zh_TW.Big5", locale->lc_ctype)&&
+		strcmp("zh_TW", locale->lc_ctype))
 	    snprintf(xim_name, 128, "%s-%s", DEFAULT_XIMNAME, locale->lc_ctype);
         else
 	    strncpy(xim_name, DEFAULT_XIMNAME, sizeof(xim_name));
