diff -Naur jpeg-mmx-orig/jquant_x86simd.c jpeg-mmx/jquant_x86simd.c
--- jpeg-mmx-orig/jquant_x86simd.c	2006-01-22 21:37:14.000000000 -0600
+++ jpeg-mmx/jquant_x86simd.c	2006-01-22 21:37:43.000000000 -0600
@@ -104,7 +104,7 @@
 
 	/* Initialise zero block flags */
 	/* Set up SSE rounding mode */
-	__asm__ ( "ldmxcsr %0\n" : : "X" (trunc_mxcsr) );
+	__asm__ ( "ldmxcsr %0\n" : : "m" (trunc_mxcsr) );
 
 	for (i=0; i < 64 ; i+=4)
 	{
