--- build/Makefile.asound~	2005-06-16 23:07:20.000000000 +0200
+++ build/Makefile.asound	2006-06-04 15:18:50.000000000 +0200
@@ -116,7 +116,7 @@
 $(shell mkdir -p $(DIRS) )
 $(shell echo  $(CFLAGS) > $(ALSA_DIR)/$(OBJDIR)/c_flags)
 $(shell echo  $(OBJS) > $(ALSA_DIR)/$(OBJDIR)/objs)
-$(shell if ! test -L $(ALSA_DIR)/include/alsa \; then cd $(ALSA_DIR)/include \; ln -sf . alsa \; fi )
+$(shell if ! test -L $(ALSA_DIR)/include/alsa ; then cd $(ALSA_DIR)/include ; ln -sf . alsa ; fi )
 
 
 
--- build/Makefile.libavc~	2005-04-21 08:57:00.000000000 +0200
+++ build/Makefile.libavc	2006-06-04 15:21:21.000000000 +0200
@@ -15,9 +15,9 @@
 
 CFLAGS += -DHAVE_CONFIG_H -I$(AVCDIR) -I$(RAWDIR)
 
-$(shell sh -c 'if ! test -d $(AVCDIR)/$(OBJDIR)/common\; then mkdir -p $(AVCDIR)/$(OBJDIR)/common\; fi' )
-$(shell sh -c 'if ! test -d $(AVCDIR)/$(OBJDIR)/libavc1394\; then mkdir -p $(AVCDIR)/$(OBJDIR)/libavc1394\; fi' )
-$(shell sh -c 'if ! test -d $(AVCDIR)/$(OBJDIR)/librom1394\; then mkdir -p $(AVCDIR)/$(OBJDIR)/librom1394\; fi' )
+$(shell sh -c 'if ! test -d $(AVCDIR)/$(OBJDIR)/common; then mkdir -p $(AVCDIR)/$(OBJDIR)/common; fi' )
+$(shell sh -c 'if ! test -d $(AVCDIR)/$(OBJDIR)/libavc1394; then mkdir -p $(AVCDIR)/$(OBJDIR)/libavc1394; fi' )
+$(shell sh -c 'if ! test -d $(AVCDIR)/$(OBJDIR)/librom1394; then mkdir -p $(AVCDIR)/$(OBJDIR)/librom1394; fi' )
 $(shell echo $(CFLAGS) > $(AVCDIR)/$(OBJDIR)/c_flags)
 
 $(OUTPUT): $(OBJS)
--- build/Makefile.sndfile~	2005-04-20 22:15:32.000000000 +0200
+++ build/Makefile.sndfile	2006-06-04 15:21:49.000000000 +0200
@@ -5,7 +5,7 @@
 DIRS := $(SNDFILE_DIR)/$(OBJDIR)/src/G72x\
 	$(SNDFILE_DIR)/$(OBJDIR)/src/GSM610
 
-$(shell if ! test -d $(SNDFILE_DIR)/$(OBJDIR) \; then mkdir -p $(DIRS) \; fi )
+$(shell if ! test -d $(SNDFILE_DIR)/$(OBJDIR) ; then mkdir -p $(DIRS) ; fi )
 
 
 OBJS := \
--- build/Makefile.libraw~	2005-04-20 22:15:32.000000000 +0200
+++ build/Makefile.libraw	2006-06-04 15:22:13.000000000 +0200
@@ -24,8 +24,8 @@
 DIRS += $(OBJDIR)/libraw1394/src
 
 
-$(shell sh -c 'if ! test -d $(DIR)/$(OBJDIR)/src/\; then mkdir -p $(DIR)/$(OBJDIR)/src/\; fi' )
-$(shell sh -c 'if ! test -d $(DIR)/libraw1394\; then mkdir -p $(DIR)/libraw1394\; fi' )
+$(shell sh -c 'if ! test -d $(DIR)/$(OBJDIR)/src/; then mkdir -p $(DIR)/$(OBJDIR)/src/; fi' )
+$(shell sh -c 'if ! test -d $(DIR)/libraw1394; then mkdir -p $(DIR)/libraw1394; fi' )
 $(shell echo $(CFLAGS) > $(DIR)/$(OBJDIR)/c_flags)
 
 $(OUTPUT): $(OBJS) $(HEADERS)
--- esound/Makefile~	2005-04-20 22:15:12.000000000 +0200
+++ esound/Makefile	2006-06-04 15:28:01.000000000 +0200
@@ -1,6 +1,6 @@
 include ../global_config
 
-$(shell sh -c 'if ! test -d $(OBJDIR)\; then mkdir $(OBJDIR)\; fi' )
+$(shell sh -c 'if ! test -d $(OBJDIR); then mkdir $(OBJDIR); fi' )
 
 OBJS = \
 	$(OBJDIR)/audio.o \
--- guicast/Makefile~	2005-07-08 10:29:23.000000000 +0200
+++ guicast/Makefile	2006-06-04 15:28:19.000000000 +0200
@@ -26,7 +26,7 @@
 endif
 
 
-$(shell sh -c 'if ! test -d $(OBJDIR)\; then mkdir $(OBJDIR)\; fi' )
+$(shell sh -c 'if ! test -d $(OBJDIR); then mkdir $(OBJDIR); fi' )
 
 OBJS = \
 	$(OBJDIR)/bcbar.o \
--- cinelerra/Makefile~	2005-09-07 23:25:21.000000000 +0200
+++ cinelerra/Makefile	2006-06-04 15:32:04.000000000 +0200
@@ -436,7 +436,7 @@
 
 
 
-$(shell echo "\#define BUILDDATE \"`date`\"" > builddate.h )
+$(shell echo "#define BUILDDATE \"`date`\"" > builddate.h )
 $(shell echo $(CFLAGS) > $(OBJDIR)/c_flags)
 $(shell echo $(LDFLAGS1) $(OBJS) $(DCRAW) $(FILEEXR) $(AVIOBJS) $(CXXREPOSITORY) $(STATICLIBS) $(LIBS) $(LDFLAGS2) > $(OBJDIR)/objs)
 
--- global_config~	2005-08-04 09:27:47.000000000 +0200
+++ global_config	2006-06-04 15:55:48.000000000 +0200
@@ -83,7 +83,7 @@
 CXXREPOSITORY = cxx_repository/*.o
 endif
 
-$(shell sh -c 'if ! test -d $(OBJDIR)\; then mkdir $(OBJDIR)\; fi' )
+$(shell sh -c 'if ! test -d $(OBJDIR); then mkdir $(OBJDIR); fi' )
 
 ########################## LINKER
 
--- build/Makefile.exr~	2005-04-20 22:15:32.000000000 +0200
+++ build/Makefile.exr	2006-06-04 15:56:04.000000000 +0200
@@ -99,7 +99,7 @@
 	-DHAVE_IOS_BASE=1 \
 	-DHAVE_STL_LIMITS=1
 
-$(shell sh -c 'if ! test -d $(EXRDIR)/$(OBJDIR)\; then mkdir -p $(DIRS)\; fi' )
+$(shell sh -c 'if ! test -d $(EXRDIR)/$(OBJDIR); then mkdir -p $(DIRS); fi' )
 $(shell echo $(CFLAGS) > $(EXRDIR)/$(OBJDIR)/c_flags)
 $(shell echo $(OBJS) > $(EXRDIR)/$(OBJDIR)/objs)
 
--- quicktime/Makefile~	2005-10-02 00:13:17.000000000 +0200
+++ quicktime/Makefile	2006-06-04 15:56:14.000000000 +0200
@@ -240,7 +240,7 @@
 LIBS := $(OUTPUT) $(MPEG3_DIR)/$(OBJDIR)/libmpeg3.a -lpng -lz -lpthread -ldl -lm
 
 
-$(shell if ! test -d $(OBJDIR) \; then mkdir -p $(DIRS) \; fi )
+$(shell if ! test -d $(OBJDIR) ; then mkdir -p $(DIRS) ; fi )
 $(shell echo  $(CFLAGS) > $(OBJDIR)/c_flags)
 $(shell echo  $(LAME_CFLAGS) > $(OBJDIR)/lame_flags)
 $(shell echo  $(OBJS) $(JPEG_LIB)/*.o $(DV_LIB)/*.o $(FAAD_LIB)/*.o $(FAAC_LIB)/*.o $(FFMPEG_LIB)/*.o $(X264_LIB)/*.o $(OGG_LIB)/*.o $(VORBIS_LIB)/*.o $(LAME_OBJS) > $(OBJDIR)/objs)
--- libmpeg3/Makefile~	2005-08-23 20:11:16.000000000 +0200
+++ libmpeg3/Makefile	2006-06-04 17:45:39.000000000 +0200
@@ -146,7 +146,7 @@
 
 LIBS = -lm -lpthread
 
-$(shell if ! test -d $(OBJDIR) \; then mkdir -p $(OBJDIR) \; fi )
+$(shell if ! test -d $(OBJDIR) ; then mkdir -p $(OBJDIR) ; fi )
 
 $(shell echo $(CFLAGS) > $(OBJDIR)/c_flags)
 $(shell echo $(A52CFLAGS) > $(OBJDIR)/a52_flags)
