--- include/my_base.h.orig	2009-01-30 00:42:12.000000000 +0300
+++ include/my_base.h	2009-02-20 02:06:46.452883556 +0300
@@ -41,6 +41,15 @@
 #endif	/* stdin */
 #include <my_list.h>
 
+/*
+  Clean workaround for inconsistent BIG_TABLES define issue
+  http://bugs.mysql.com/35346
+  Author: Antony Curtis, patch created by Arjen Lentz
+*/
+
+#undef BIG_TABLES
+#define BIG_TABLES 1
+
 /* The following is bits in the flag parameter to ha_open() */
 
 #define HA_OPEN_ABORT_IF_LOCKED		0	/* default */
