diff -Naur mysql-5.1.32.orig/include/my_base.h mysql-5.1.32/include/my_base.h
--- mysql-5.1.32.orig/include/my_base.h	2009-04-15 21:14:57.795163964 -0700
+++ mysql-5.1.32/include/my_base.h	2009-04-15 21:15:17.947918898 -0700
@@ -36,6 +36,15 @@
 #endif	/* stdin */
 #include <my_list.h>
 
+/*
+  Clean workaround for inconsistent BIG_TABLES define issue
+  http://bugs.mysql.com/35346
+  Author: Antony Curtis, patch created by Arjen Lentz
+*/
+
+#undef BIG_TABLES
+#define BIG_TABLES 1
+
 /* The following is bits in the flag parameter to ha_open() */
 
 #define HA_OPEN_ABORT_IF_LOCKED		0	/* default */
