From 5031bce345b8393a28526477d1a065436d669dca Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sat, 27 Jun 2015 23:02:33 +0200
Subject: [PATCH] fix curses include

---
 pinentry/pinentry-curses.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pinentry/pinentry-curses.c b/pinentry/pinentry-curses.c
index d25ab2e..40650ba 100644
--- a/pinentry/pinentry-curses.c
+++ b/pinentry/pinentry-curses.c
@@ -23,7 +23,7 @@
 #endif
 #include <assert.h>
 #ifdef HAVE_NCURSESW
-#include <ncursesw/curses.h>
+#include <curses.h>
 #else
 #include <curses.h>
 #endif
-- 
2.4.1

