diff --git a/src/song/ModifiedSinceSongFilter.cxx b/src/song/ModifiedSinceSongFilter.cxx
index 43879c7..fe5cc05 100644
--- a/src/song/ModifiedSinceSongFilter.cxx
+++ b/src/song/ModifiedSinceSongFilter.cxx
@@ -7,6 +7,7 @@
 #include "util/StringBuffer.hxx"
 
 #include <fmt/core.h>
+#include <fmt/format.h>
 
 using std::string_view_literals::operator""sv;
 
diff --git a/src/song/AddedSinceSongFilter.cxx b/src/song/AddedSinceSongFilter.cxx
index efc394f..89af2b7 100644
--- a/src/song/AddedSinceSongFilter.cxx
+++ b/src/song/AddedSinceSongFilter.cxx
@@ -7,6 +7,7 @@
 #include "util/StringBuffer.hxx"
 
 #include <fmt/core.h>
+#include <fmt/format.h>
 
 using std::string_view_literals::operator""sv;
 
diff --git a/src/song/AudioFormatSongFilter.cxx b/src/song/AudioFormatSongFilter.cxx
index 69bd2b1..6fa8f91 100644
--- a/src/song/AudioFormatSongFilter.cxx
+++ b/src/song/AudioFormatSongFilter.cxx
@@ -6,6 +6,7 @@
 #include "util/StringBuffer.hxx"
 
 #include <fmt/core.h>
+#include <fmt/format.h>
 
 using std::string_view_literals::operator""sv;
 
diff --git a/src/input/plugins/AlsaInputPlugin.cxx b/src/input/plugins/AlsaInputPlugin.cxx
index 6bf8034..71eaca3 100644
--- a/src/input/plugins/AlsaInputPlugin.cxx
+++ b/src/input/plugins/AlsaInputPlugin.cxx
@@ -28,6 +28,7 @@
 #include <alsa/asoundlib.h>
 
 #include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <cassert>
 
diff --git a/src/net/ToString.cxx b/src/net/ToString.cxx
index 41421d8..5acbed6 100644
--- a/src/net/ToString.cxx
+++ b/src/net/ToString.cxx
@@ -7,6 +7,7 @@
 #include "IPv4Address.hxx"
 
 #include <fmt/core.h>
+#include <fmt/format.h>
 
 #include <algorithm>
 #include <cassert>
