diff --git a/src/client/Response.hxx b/src/client/Response.hxx
--- a/src/client/Response.hxx
+++ b/src/client/Response.hxx
@@ -82,7 +82,10 @@ public:
 
 	template<typename S, typename... Args>
 	bool Fmt(const S &format_str, Args&&... args) noexcept {
-#if FMT_VERSION >= 70000
+#if FMT_VERSION >= 90000
+		return VFmt(fmt::detail::to_string_view(format_str),
+			    fmt::make_format_args<fmt::v9::buffer_context<fmt::v9::char_t<S>>>(args...));
+#elif FMT_VERSION >= 70000
 		return VFmt(fmt::to_string_view(format_str),
 			    fmt::make_args_checked<Args...>(format_str,
 							    args...));
@@ -109,7 +112,10 @@ public:
 	template<typename S, typename... Args>
 	void FmtError(enum ack code,
 		      const S &format_str, Args&&... args) noexcept {
-#if FMT_VERSION >= 70000
+#if FMT_VERSION >= 90000
+		return VFmtError(code, fmt::detail::to_string_view(format_str),
+			   fmt::make_format_args<fmt::v9::buffer_context<fmt::v9::char_t<S>>>(args...));
+#elif FMT_VERSION >= 70000
 		return VFmtError(code, fmt::to_string_view(format_str),
 				 fmt::make_args_checked<Args...>(format_str,
 								 args...));
diff --git a/src/Log.hxxb/src/Log.hxx
--- a/src/Log.hxx
+++ b/src/Log.hxx
@@ -45,7 +45,10 @@ void
 LogFmt(LogLevel level, const Domain &domain,
        const S &format_str, Args&&... args) noexcept
 {
-#if FMT_VERSION >= 70000
+#if FMT_VERSION >= 90000
+	return LogVFmt(level, domain, fmt::detail::to_string_view(format_str),
+		       fmt::make_format_args<fmt::v9::buffer_context<fmt::v9::char_t<S>>>(args...));
+#elif FMT_VERSION >= 70000
 	return LogVFmt(level, domain, fmt::to_string_view(format_str),
 		       fmt::make_args_checked<Args...>(format_str,
 						       args...));
