diff -Naur wbar-1.3.3.orig/IconLoader.cc wbar-1.3.3/IconLoader.cc
--- wbar-1.3.3.orig/IconLoader.cc	2008-04-24 12:06:30.222538804 -0700
+++ wbar-1.3.3/IconLoader.cc	2008-04-24 12:08:41.678298214 -0700
@@ -1,5 +1,6 @@
 extern "C" {
 #include <unistd.h>
+#include <cstdlib>
 #include <sys/stat.h>
 #include <sys/types.h>
 };
diff -Naur wbar-1.3.3.orig/SuperBar.cc wbar-1.3.3/SuperBar.cc
--- wbar-1.3.3.orig/SuperBar.cc	2008-04-24 12:06:30.223539071 -0700
+++ wbar-1.3.3/SuperBar.cc	2008-04-24 12:09:47.454291462 -0700
@@ -1,3 +1,5 @@
+#include <cstdlib>
+
 #include "SuperBar.h"
 #include "SuperIcon.h"
 
