--- cdrtools-2.01.01/cdrecord/cdrecord.c.orig	2007-07-19 18:50:02.000000000 +0200
+++ cdrtools-2.01.01/cdrecord/cdrecord.c	2007-07-19 18:51:17.000000000 +0200
@@ -4214,16 +4214,17 @@
 	 * with e.g. vendor unique commands, where cdrecord would miss
 	 * functionality or fail completely after starting to write.
 	 */
+  /* kernel >= 2.6.12 does not like that as user
 	seterrno(0);
 	scgp->silent++;
-	code = rezero_unit(scgp); /* Not supported by some drives */
+	code = rezero_unit(scgp);
 	scgp->silent--;
 	err = geterrno();
 	if (code < 0 && (err == EPERM || err == EACCES)) {
-		linuxcheck();	/* For version 1.349 of cdrecord.c */
+		linuxcheck();
 		scg_openerr("");
 	}
-
+  */
 
 	test_unit_ready(scgp);
 	scsi_start_stop_unit(scgp, 1, 0, immed);
