diff -Naur cdrtools-2.01.01.orig/libscg/scsi-linux-ata.c cdrtools-2.01.01/libscg/scsi-linux-ata.c
--- cdrtools-2.01.01.orig/libscg/scsi-linux-ata.c	2007-07-09 10:56:56.113108163 -0700
+++ cdrtools-2.01.01/libscg/scsi-linux-ata.c	2007-07-09 10:57:29.108074574 -0700
@@ -394,7 +394,8 @@
 						"try open(%s) return %i, errno %i, cancel\n",
 						device, f, errno);
 					}
-					return (-2);
+					/* return (-2); */
+					break;
 				} else if (errno == ENOENT || errno == ENODEV) {
 					break;
 				}
diff -Naur cdrtools-2.01.01.orig/libscg/scsi-linux-sg.c cdrtools-2.01.01/libscg/scsi-linux-sg.c
--- cdrtools-2.01.01.orig/libscg/scsi-linux-sg.c	2007-07-09 10:56:56.113108163 -0700
+++ cdrtools-2.01.01/libscg/scsi-linux-sg.c	2007-07-09 10:57:29.108074574 -0700
@@ -464,7 +464,7 @@
 				if (scgp->errstr)
 					js_snprintf(scgp->errstr, SCSI_ERRSTR_SIZE,
 							"Cannot open '%s'", devname);
-				return (0);
+				//return (0);
 			}
 		} else {
 			int	iparm;
