--- cdrtools-2.01.01/libscg/scsi-linux-ata.c.scanbus	2005-01-19 21:13:34.000000000 +0100
+++ cdrtools-2.01.01/libscg/scsi-linux-ata.c	2005-01-19 21:32:59.634364995 +0100
@@ -384,7 +384,8 @@
 						"try open(%s) return %i, errno %i, cancel\n",
 						device, f, errno);
 					}
-					return (-2);
+					/* return (-2); */
+					break;
 				} else if (errno == ENOENT || errno == ENODEV) {
 					break;
 				}
--- cdrtools-2.01.01/libscg/scsi-linux-sg.c.scanbus	2005-01-19 21:34:38.264068893 +0100
+++ cdrtools-2.01.01/libscg/scsi-linux-sg.c	2005-01-19 21:34:54.386932733 +0100
@@ -428,7 +428,7 @@
 				if (scgp->errstr)
 					js_snprintf(scgp->errstr, SCSI_ERRSTR_SIZE,
 							"Cannot open '%s'", devname);
-				return (0);
+				//return (0);
 			}
 		} else {
 			int	iparm;
