--- cdrtools-2.01.01/cdrecord/cdrecord.c.rezero	2005-08-03 12:50:28.000000000 +0200
+++ cdrtools-2.01.01/cdrecord/cdrecord.c	2005-08-03 12:50:53.000000000 +0200
@@ -3880,7 +3880,8 @@
 	scsi_start_stop_unit(scgp, 1, 0, immed);
 	wait_unit_ready(scgp, 120);
 	scgp->silent++;
-	rezero_unit(scgp);	/* Is this needed? Not supported by some drvives */
+	/* kernel >= 2.6.12 does not like that as user
+	rezero_unit(scgp); */	/* Is this needed? Not supported by some drvives */
 	scgp->silent--;
 	test_unit_ready(scgp);
 	scsi_start_stop_unit(scgp, 1, 0, immed);
