diff -uNr cdrtools-2.01.01.old/cdrecord/cdrecord.c cdrtools-2.01.01/cdrecord/cdrecord.c
--- cdrtools-2.01.01.old/cdrecord/cdrecord.c	2006-10-02 11:28:45.000000000 +0200
+++ cdrtools-2.01.01/cdrecord/cdrecord.c	2006-10-02 11:29:42.000000000 +0200
@@ -4048,7 +4048,8 @@
 	scsi_start_stop_unit(scgp, 1, 0, immed);
 	wait_unit_ready(scgp, 120);
 	scgp->silent++;
-	rezero_unit(scgp);	/* Is this needed? Not supported by some drives */
+	/* kernel >= 2.6.12 does not like that as user
+	rezero_unit(scgp); */   /* Is this needed? Not supported by some drvives */
 	scgp->silent--;
 	test_unit_ready(scgp);
 	scsi_start_stop_unit(scgp, 1, 0, immed);
