diff -Naur cdrtools-2.01.01.orig/cdrecord/cdrecord.c cdrtools-2.01.01/cdrecord/cdrecord.c
--- cdrtools-2.01.01.orig/cdrecord/cdrecord.c	2007-07-09 10:56:55.613184429 -0700
+++ cdrtools-2.01.01/cdrecord/cdrecord.c	2007-07-09 10:59:10.092668741 -0700
@@ -4187,7 +4187,8 @@
 	scsi_start_stop_unit(scgp, 1, 0, immed);
 	wait_unit_ready(scgp, 120);
 	scgp->silent++;
-	rezero_unit(scgp);	/* Is this needed? Not supported by some drives */
+	/* kernel >= 2.6.12 does not like that as user
+	rezero_unit(scgp); */   /* Is this needed? Not supported by some drvives */
 	scgp->silent--;
 	test_unit_ready(scgp);
 	scsi_start_stop_unit(scgp, 1, 0, immed);
