From 9429519747eaf2fb297751fb9d363672b5b023c6 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Mon, 3 Oct 2016 16:50:33 +0200
Subject: [PATCH] do not build manpage

---
 wscript | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/wscript b/wscript
index 41f3be7..eefb470 100644
--- a/wscript
+++ b/wscript
@@ -119,8 +119,7 @@ def build(bld):
                           public_headers=('' if private_library else 'talloc.h'),
                           pc_files='talloc.pc',
                           public_headers_install=not private_library,
-                          private_library=private_library,
-                          manpages='man/talloc.3')
+                          private_library=private_library)
 
     if not bld.CONFIG_SET('USING_SYSTEM_PYTALLOC_UTIL') and not bld.env.disable_python:
         for env in bld.gen_python_environments(['PKGCONFIGDIR']):
-- 
2.10.0

