diff -Naur rpcbind-0.2.4.orig/src/security.c rpcbind-0.2.4/src/security.c
--- rpcbind-0.2.4.orig/src/security.c	2017-08-16 00:16:28.077548192 +0000
+++ rpcbind-0.2.4/src/security.c	2017-08-16 00:16:47.661547024 +0000
@@ -26,7 +26,7 @@
 #include <rpcsvc/mount.h>
 #include <rpcsvc/rquota.h>
 #include <rpcsvc/nfs_prot.h>
-#include <rpcsvc/yp.h>
+#include <rpcsvc/yp_prot.h>
 #include <rpcsvc/ypclnt.h>
 #include <rpcsvc/yppasswd.h>
 #else
