Index: exscalibar-1.0.4/src/exinfo/exinfo.pro
===================================================================
--- exscalibar-1.0.4.orig/src/exinfo/exinfo.pro
+++ exscalibar-1.0.4/src/exinfo/exinfo.pro
@@ -8,8 +8,7 @@ include( ../../global.pro )
 INSTALLS += target
 target.path = $$PREFIX/bin
 TARGETDEPS += ../../src/geddei/libgeddei.so ../../src/qtextra/libqtextra.so
-LIBS +=	-lqtextra \
-	-lgeddei
+LIBS += -lgeddei -lqtextra
 QMAKE_LIBDIR = ../../src/qtextra \
                ../../src/geddei
 INCLUDEPATH += ../../src/geddei \
Index: exscalibar-1.0.4/src/geddei/processorforwarder.cpp
===================================================================
--- exscalibar-1.0.4.orig/src/geddei/processorforwarder.cpp
+++ exscalibar-1.0.4/src/geddei/processorforwarder.cpp
@@ -118,7 +118,7 @@ void ProcessorForwarder::newConnection(i
 			SubProcessor *sub = SubProcessorFactory::create(type);
 			if(MESSAGES) qDebug("Created SubProcessor at %p", sub);
 			// Return the subProc's pointer for decoupling later...
-			header << ((uint)sub) << endl;
+			header << ((ptrdiff_t)sub) << endl;
 			new RSCoupling(link, sub);
 			// return here to make sure that link isn't deleted.
 			return;
Index: exscalibar-1.0.4/src/tests/testall/testall.pro
===================================================================
--- exscalibar-1.0.4.orig/src/tests/testall/testall.pro
+++ exscalibar-1.0.4/src/tests/testall/testall.pro
@@ -9,8 +9,8 @@ TARGETDEPS += ../../../src/processors/to
               ../../../src/qtextra/libqtextra.so \
               ../../../src/geddei/libgeddei.so
 LIBS += -ltoolkit \
-        -lqtextra \
-        -lgeddei
+        -lgeddei \
+	-lqtextra
 INCLUDEPATH = ../../../src/processors/toolkit \
               ../../../src/geddei \
               ../../../src/qtextra 
Index: exscalibar-1.0.4/src/tests/testdemux/testdemux.pro
===================================================================
--- exscalibar-1.0.4.orig/src/tests/testdemux/testdemux.pro
+++ exscalibar-1.0.4/src/tests/testdemux/testdemux.pro
@@ -10,8 +10,8 @@ TARGETDEPS += ../../../src/processors/to
               ../../../src/qtextra/libqtextra.so \
               ../../../src/geddei/libgeddei.so
 LIBS += -ltoolkit \
-        -lqtextra \
-        -lgeddei
+        -lgeddei \
+	-lqtextra
 INCLUDEPATH = ../../../src/processors/toolkit \
               ../../../src/geddei \
               ../../../src/qtextra 
