--- user-list.vala	2011-09-28 07:32:51.000000000 +0200
+++ user-list.vala-fixed	2011-11-12 00:34:32.217649083 +0100
@@ -249,7 +249,7 @@
 
         /* Draw overlay */
         bc.set_source (overlay);
-        bc.rectangle (grid_size - 1, grid_size - 1, width - grid_size * 2 + 2, height - grid_size * 2 + 2);
+//        bc.rectangle (grid_size - 1, grid_size - 1, width - grid_size * 2 + 2, height - grid_size * 2 + 2);
         bc.fill ();
 
         /* Draw logo */
