From c4bbf67d20aabad507e8db5432ec71d34e818aba Mon Sep 17 00:00:00 2001
From: Lauri Nurmi <lanurmi@iki.fi>
Date: Thu, 17 Sep 2015 01:04:53 +0300
Subject: [PATCH] Fix a sometimes uninitialized variable.

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 src/con_x11.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/con_x11.cpp b/src/con_x11.cpp
index 42677bcc1379..dec75a797358 100644
--- a/src/con_x11.cpp
+++ b/src/con_x11.cpp
@@ -414,7 +414,7 @@ static int InitXFonts(void) {
 }
 
 static int SetupXWindow(int argc, char **argv) {
-    unsigned long mask;
+    unsigned long mask = 0;
 
 #ifdef WINHCLX
     HCLXlibInit(); /* HCL - Initialize the X DLL */
-- 
2.36.0

