From: Boyuan Yang <byang@debian.org>
Date: Wed, 14 Apr 2021 23:52:47 -0400
Subject: Let cmake respect CMAKE_CXX_FLAGS

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 067ef11..73600ea 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -86,7 +86,7 @@ IF( UNIX AND ${CMAKE_SYSTEM_NAME} MATCHES "Darwin" )
 ENDIF( UNIX AND ${CMAKE_SYSTEM_NAME} MATCHES "Darwin" )
 
 IF( CMAKE_COMPILER_IS_GNUCC )
-  SET( CMAKE_CXX_FLAGS "-pedantic -Wall -Wno-long-long -Wpointer-arith -Wconversion -Wwrite-strings -Winline -fno-rtti -fno-exceptions" )
+  SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Wall -Wno-long-long -Wpointer-arith -Wconversion -Wwrite-strings -Winline -fno-rtti -fno-exceptions" )
 ENDIF( CMAKE_COMPILER_IS_GNUCC )
 
 IF( CMAKE_SYSTEM_NAME MATCHES "HP/UX" )
