--- egenix-mx-base-2.0.5/mx/DateTime/mxDateTime/mxDateTime.c	2001-12-10 10:15:49.000000000 +0000
+++ egenix-mx-base-2.0.5.orig/mx/DateTime/mxDateTime/mxDateTime.c	2004-07-06 07:40:04.193376848 +0000
@@ -324,7 +324,7 @@
 mxDateTimeObject *mxDateTime_New(void)
 {
     mxDateTimeObject *datetime;
-
+/*
 #ifdef MXDATETIME_FREELIST
     if (mxDateTime_FreeList) {
 	datetime = mxDateTime_FreeList;
@@ -334,11 +334,12 @@
     }
     else
 #endif 
-	 {
+*/	 
+	{
 	datetime = PyObject_NEW(mxDateTimeObject,&mxDateTime_Type);
 	if (datetime == NULL)
 	    return NULL;
-    }
+  }
 
     /* Init vars */
     datetime->argument = 0;
