From ffe24fa1352d1a6f0fb3d5ea6f6634fb18551979 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Tue, 8 Apr 2025 14:01:43 +0200
Subject: [PATCH 2/2] fix QDataStream usage

---
 kmymoney/mymoney/mymoneycategory.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kmymoney/mymoney/mymoneycategory.h b/kmymoney/mymoney/mymoneycategory.h
index b7a6399..a5b70c7 100644
--- a/kmymoney/mymoney/mymoneycategory.h
+++ b/kmymoney/mymoney/mymoneycategory.h
@@ -11,6 +11,7 @@
 // QT Includes
 
 #include "qcontainerfwd.h"
+#include <QDataStream>
 #include <qglobal.h>
 
 class QString;
-- 
2.49.0

