diff -Naur wdm-1.28.orig/configs/Xclients.in wdm-1.28/configs/Xclients.in
--- wdm-1.28.orig/configs/Xclients.in	2006-07-05 20:01:07.000000000 +0200
+++ wdm-1.28/configs/Xclients.in	2006-07-05 20:11:53.000000000 +0200
@@ -66,6 +66,14 @@
 			exec $WINMGR_PATH >>$HOME/.xwm.msgs 2>&1
 		fi
 		;;
+	
+	start*)
+		START_PATH="/usr/bin/$WMSTYLE"
+		if [ -x "$START_PATH" ]; then
+			echo "Starting $WMSTYLE" > $HOME/.xwm.msgs
+			exec $START_PATH >> $HOME/.xwm.msgs 2>&1
+		fi
+	;;
 
 	wmaker*|WindowMaker*)
 		# startup WindowMaker
