diff -Naur wdm-1.28.orig/configs/TakeConsole wdm-1.28/configs/TakeConsole
--- wdm-1.28.orig/configs/TakeConsole	2006-07-05 20:01:07.000000000 +0200
+++ wdm-1.28/configs/TakeConsole	2006-07-06 00:40:53.000000000 +0200
@@ -5,3 +5,5 @@
 #
 chmod 622 /dev/console
 chown root /dev/console
+
+/usr/bin/update_wdm_wm_list
diff -Naur wdm-1.28.orig/configs/wdm-config.in wdm-1.28/configs/wdm-config.in
--- wdm-1.28.orig/configs/wdm-config.in	2006-07-05 20:01:07.000000000 +0200
+++ wdm-1.28/configs/wdm-config.in	2006-07-05 20:50:11.000000000 +0200
@@ -25,7 +25,7 @@
 ! display screen background--if not set or set to none, then the
 !                            background is not set by wdm.
 ! DisplayManager*wdmBg:         none
-DisplayManager*wdmBg:           solid:RoyalBlue
+DisplayManager*wdmBg:           solid:black
 ! DisplayManager*wdmBg:         dgradient:RoyalBlue,FireBrick,Black
 ! DisplayManager*wdmBg:         hgradient:RoyalBlue,FireBrick,Black
 ! DisplayManager*wdmBg:         vgradient:RoyalBlue,FireBrick,Black
@@ -53,6 +53,8 @@
 ! list of window managers which can be started
 ! or 'None' if you want only NoChange to appear
 ! e.g.: DisplayManager*wdmWm:   None
+! Beware: Do not edit the list of window managers by hand as it will be
+! replaced on every restart of wdm!
 DisplayManager*wdmWm:           @WMLIST@
 !
 ! commands for reboot and halt 
diff -Naur wdm-1.28.orig/configs/Xsetup_0.in wdm-1.28/configs/Xsetup_0.in
--- wdm-1.28.orig/configs/Xsetup_0.in	2006-07-05 20:01:07.000000000 +0200
+++ wdm-1.28/configs/Xsetup_0.in	2006-07-06 00:56:42.000000000 +0200
@@ -1,6 +1,7 @@
 #!@SYSTEM_SHELL@
 # @configure_input@
 # $XConsortium: Xsetup_0,v 1.3 93/09/28 14:30:31 gildea Exp $
+/usr/bin/update_wdm_wm_list
 @XCONSOLE@ -geometry 480x130-0-0 -daemon -notify -verbose -fn fixed -exitOnFail
 # /usr/X11R6/bin/xbanner
 # if xbanner is used, change Xsession to do freetemp
