diff -ru kdepim-3.5.10/kmail/editorwatcher.cpp kdepim-3.5.10.inotify/kmail/editorwatcher.cpp
--- kdepim-3.5.10/kmail/editorwatcher.cpp	2008-02-13 10:39:37.000000000 +0100
+++ kdepim-3.5.10.inotify/kmail/editorwatcher.cpp	2009-02-18 15:18:49.000000000 +0100
@@ -32,30 +32,14 @@
 #include <cassert>
 
 // inotify stuff taken from kdelibs/kio/kio/kdirwatch.cpp
+// fixed for linux header usage
 #ifdef HAVE_INOTIFY
 #include <sys/ioctl.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/syscall.h>
-#include <linux/types.h>
-// Linux kernel headers are documented to not compile
-#define _S390_BITOPS_H
-#include <linux/inotify.h>
-
-static inline int inotify_init (void)
-{
-  return syscall (__NR_inotify_init);
-}
-
-static inline int inotify_add_watch (int fd, const char *name, __u32 mask)
-{
-  return syscall (__NR_inotify_add_watch, fd, name, mask);
-}
-
-static inline int inotify_rm_watch (int fd, __u32 wd)
-{
-  return syscall (__NR_inotify_rm_watch, fd, wd);
-}
+#include <sys/types.h>
+#include <sys/inotify.h>
 #endif
 
 using namespace KMail;
