diff --git a/meson.build b/meson.build
index fed3dc055..e99e5ccac 100644
--- a/meson.build
+++ b/meson.build
@@ -4,7 +4,7 @@ project('harfbuzz', 'c', 'cpp',
   default_options: [
     'cpp_eh=none',          # Just to support msvc, we are passing -fno-exceptions also anyway
     # 'cpp_rtti=false',     # Do NOT enable, wraps inherit it and ICU needs RTTI
-    'cpp_std=c++11',
+    'cpp_std=c++17',
     'wrap_mode=nofallback', # Use --wrap-mode=default to revert, https://github.com/harfbuzz/harfbuzz/pull/2548
   ],
 )
