--- xpvm/src/Makefile.aimk	Thu Apr  9 23:11:22 1998
+++ xpvm/src/Makefile.aimk	Thu May  9 17:41:41 2002
@@ -25,8 +25,8 @@
 CFLOPTS		=	-O
 #CFLOPTS	=	-g
 
-CC			=	cc
-#CC			=	gcc
+#CC			=	cc
+CC			=	gcc
 
 CFLAGS		=	$(CFLOPTS) $(PVMVERSION) -DIMA_$(PVM_ARCH) $(ARCHCFLAGS)
 LFLAGS		=	$(ARCHCFLAGS)
@@ -63,12 +63,15 @@
 #TCLLIBDIR	=	-L$(TCLTKHOME)
 #TKLIBDIR	=	-L$(TCLTKHOME)
 
+# Tcl / Tk 8.4
+TCLLIB		=	-ltcl8.4
+TKLIB		=	-ltk8.4
 # Tcl / Tk 8.1
 #TCLLIB		=	-ltcl8.1
 #TKLIB		=	-ltk8.1
 # Tcl / Tk 8.0
-TCLLIB		=	-ltcl8.0
-TKLIB		=	-ltk8.0
+#TCLLIB		=	-ltcl8.0
+#TKLIB		=	-ltk8.0
 # Tcl 7.6 / Tk 4.2
 #TCLLIB		=	-ltcl7.6
 #TKLIB		=	-ltk4.2
@@ -80,14 +83,14 @@
 #TKLIB		=	-ltk
 
 # Generic X11 System
-XINCL		=
-XLIBDIR		=
+#XINCL		=
+#XLIBDIR	=
 # Default Locations...
 #XINCL		=	-I/usr/include/X11
 #XLIBDIR	=	-L/usr/lib
 # Linux Like Systems
-#XINCL		=	-I/usr/X11R6/include
-#XLIBDIR	=	-L/usr/X11R6/lib
+XINCL		=	-I/usr/X11R6/include
+XLIBDIR		=	-L/usr/X11R6/lib
 
 XLIB		=	-lX11 
 #XLIB		=	-lX11 -lXpm
