--- eel/eel-wrap-table.c~	2001-04-30 13:36:04.000000000 +0200
+++ eel/eel-wrap-table.c	2006-06-10 21:34:04.000000000 +0200
@@ -518,6 +518,7 @@
 					item_allocation.x += (max_child_dimensions.width - (int) item_allocation.width);
 					break;
 				default:
+          ;
 				}
 				
 				switch (wrap_table->details->y_justification) {
@@ -528,6 +529,7 @@
 					item_allocation.y += (max_child_dimensions.height - (int) item_allocation.height);
 					break;
 				default:
+          ;
 				}
 			}
 			
--- eel/eel-wrap-table.h~	2001-04-04 09:51:37.000000000 +0200
+++ eel/eel-wrap-table.h	2006-06-10 21:34:15.000000000 +0200
@@ -74,7 +74,7 @@
 							   int                 x,
 							   int                 y);
 void             eel_wrap_table_set_x_justification       (EelWrapTable       *wrap_table,
-							   GtkJustification    justification);
+							   EelJustification    justification);
 EelJustification eel_wrap_table_get_x_justification       (const EelWrapTable *wrap_table);
 void             eel_wrap_table_set_y_justification       (EelWrapTable       *wrap_table,
 							   EelJustification    justification);
--- test/test-eel-gtk-style.c~	2001-05-01 12:32:50.000000000 +0200
+++ test/test-eel-gtk-style.c	2006-06-10 21:59:15.000000000 +0200
@@ -64,6 +64,7 @@
 	case 5: return style->text[state]; break;
 	case 6: return style->base[state]; break;
 	default:
+          ;
 	}
 	g_assert_not_reached ();
 	return empty;
@@ -89,6 +90,7 @@
 	case 5: return style->text_gc[state]; break;
 	case 6: return style->base_gc[state]; break;
 	default:
+          ;
 	}
 	g_assert_not_reached ();
 	return NULL;
