diff -Naur swfdec-mozilla-0.4.3.orig/configure.ac swfdec-mozilla-0.4.3/configure.ac
--- swfdec-mozilla-0.4.3.orig/configure.ac	2007-04-02 14:15:48.286730883 -0700
+++ swfdec-mozilla-0.4.3/configure.ac	2007-04-02 14:49:09.712400849 -0700
@@ -112,23 +112,50 @@
   AC_MSG_ERROR([cannot find cairo, which is required for build])
 fi
 
-PKG_CHECK_MODULES(MOZILLA, mozilla-plugin, HAVE_MOZILLA=yes, HAVE_MOZILLA=no)
-MOZILLA_DIR=mozilla
-if test "$HAVE_MOZILLA" = no ; then
-  PKG_CHECK_MODULES(MOZILLA, firefox-plugin, HAVE_MOZILLA=yes, HAVE_MOZILLA=no)
-  MOZILLA_DIR=mozilla-firefox
-fi
-if test "$HAVE_MOZILLA" = no ; then
-  PKG_CHECK_MODULES(MOZILLA, iceape-plugin, HAVE_MOZILLA=yes, HAVE_MOZILLA=no)
-  MOZILLA_DIR=iceape
-fi
-AC_SUBST(MOZILLA_CFLAGS)
-AC_SUBST(MOZILLA_LIBS)
-AC_SUBST(MOZILLA_DIR)
-AM_CONDITIONAL(HAVE_MOZILLA, [test $HAVE_MOZILLA = yes])
-if test "x$HAVE_MOZILLA" = "xno"; then
-  AC_MSG_ERROR([Mozilla not found. Mozilla/Netscape plugin will not be built.])
+dnl ******************
+dnl Mozilla name check
+dnl ******************
+AC_MSG_CHECKING([which mozilla to use])
+MOZILLA=
+AC_ARG_WITH([mozilla],
+         AS_HELP_STRING([--with-mozilla@<:@=mozilla|firefox|seamonkey|xulrunner@:>@],
+                        [Which mozilla to use for plugin (default: autodetect)]),
+            [MOZILLA="$withval"])
+if test "x$MOZILLA" = "x"; then
+  if $PKG_CONFIG --exists mozilla-plugin; then
+    MOZILLA=mozilla
+  elif $PKG_CONFIG --exists firefox-plugin; then
+    MOZILLA=firefox
+  elif $PKG_CONFIG --exists seamonkey-plugin; then
+    MOZILLA=seamonkey
+  elif $PKG_CONFIG --exists xulrunner-plugin; then
+    MOZILLA=xulrunner
+  else
+    AC_MSG_WARN([no mozilla installation found. Mozilla plugin will not be built])
+    HAVE_MOZILLA=no
+  fi
+elif test "x$with_mozilla" != "xmozilla" -a "x$with_mozilla" != "xfirefox" -a "x$with_mozilla" != "xseamonkey" -a "x$with_mozilla" != "xxulrunner" -a "x$with_mozilla" != "xno"; then
+  AC_MSG_ERROR([unknown mozilla name ($MOZILLA)])
+ fi
+  
+AC_SUBST(MOZILLA)
+AC_MSG_RESULT([$MOZILLA])
+
+if test "x$with_mozilla" != "xno"; then
+  PKG_CHECK_MODULES(MOZILLA, $MOZILLA-plugin, HAVE_MOZILLA="yes", HAVE_MOZILLA="no")
+  AC_SUBST(MOZILLA_CFLAGS)
+  AC_SUBST(MOZILLA_LIBS)
+  if test "x$HAVE_MOZILLA" = "xyes"; then
+    MOZILLA_PLUGIN_DIR=`pkg-config --variable=libdir $MOZILLA-plugin`/plugins
+  else
+    MOZILLA_PLUGIN_DIR=
+  fi
+else
+  MOZILLA_PLUGIN_DIR=
+  HAVE_MOZILLA=no
 fi
+AC_SUBST(MOZILLA_PLUGIN_DIR)
+AM_CONDITIONAL(HAVE_MOZILLA,[test "$HAVE_MOZILLA" = "yes"])
 
 dnl define standard mozilla defines
 dnl this needs updates if we want to compile on Windows or OS X
diff -Naur swfdec-mozilla-0.4.3.orig/src/Makefile.am swfdec-mozilla-0.4.3/src/Makefile.am
--- swfdec-mozilla-0.4.3.orig/src/Makefile.am	2007-04-02 14:15:48.279731951 -0700
+++ swfdec-mozilla-0.4.3/src/Makefile.am	2007-04-02 14:33:03.215845958 -0700
@@ -8,7 +8,7 @@
 BUILT_SOURCES = swfdec_playback.c
 CLEANFILES = swfdec_playback.c
 
-plugindir = $(libdir)/$(MOZILLA_DIR)/plugins
+plugindir = @MOZILLA_PLUGIN_DIR@
 plugin_LTLIBRARIES = libswfdecmozilla.la
 
 libswfdecmozilla_la_CFLAGS = $(GLOBAL_CFLAGS) $(MOZILLA_CFLAGS) $(SWFDEC_CFLAGS) \
