--- configure.in.orig	2005-06-03 10:00:08.000000000 -0500
+++ configure.in	2005-06-03 10:06:50.000000000 -0500
@@ -92,7 +92,7 @@
 
 # TODO: This should really depend on glibmm instead of gtkmm, but we want to use Gtk::Main::init_gtkmm_internals().
 # We should be able to add something more appropriate to glibmm 2.4:
-PKG_CHECK_MODULES(LIBGNOMEPRINTUIMM, gtkmm-2.4 >= $GTKMM_VERSION_NEEDED libgnomeprintmm-2.6 >= $LIBGNOMEPRINTMM_VERSION_NEEDED libgnomeprintui-2.2 >= $LIBGNOMEPRINTUI_VERSION_NEEDED)
+PKG_CHECK_MODULES(LIBGNOMEPRINTUIMM, gtkmm-2.4 >= $GTKMM_VERSION_NEEDED gdkmm-2.4 >= $GTKMM_VERSION_NEEDED atkmm-1.6 >= 1.6.0 pangomm-1.4 >= 1.4.0 libgnomeprintmm-2.6 >= $LIBGNOMEPRINTMM_VERSION_NEEDED libgnomeprintui-2.2 >= $LIBGNOMEPRINTUI_VERSION_NEEDED)
 AC_SUBST(LIBGNOMEPRINTUIMM_CFLAGS)
 AC_SUBST(LIBGNOMEPRINTUIMM_LIBS)
 
