diff -Naur kdebase-workspace-4.3.4.orig/plasma/dataengines/geolocation/location_gps.cpp kdebase-workspace-4.3.4/plasma/dataengines/geolocation/location_gps.cpp
--- kdebase-workspace-4.3.4.orig/plasma/dataengines/geolocation/location_gps.cpp	2009-05-06 12:13:48.000000000 +0200
+++ kdebase-workspace-4.3.4/plasma/dataengines/geolocation/location_gps.cpp	2009-12-13 13:24:45.000000000 +0100
@@ -41,7 +41,7 @@
 
 void Gpsd::run()
 {
-    gps_query(m_gpsdata, "w+x\n");
+    gps_stream(m_gpsdata, WATCH_ENABLE, NULL);
 
     while (!m_abort) {
         Plasma::DataEngine::Data d;
