--- openssl.c.orig
+++ openssl.c
@@ -56,6 +56,7 @@
 static struct termios	otio;
 static sigjmp_buf	ssljmp;
 
+#include <stdint.h>
 #include <openssl/crypto.h>
 #include <openssl/ssl.h>
 #include <openssl/err.h>
@@ -137,11 +138,6 @@
 
 	if ((cp = value("ssl-rand-egd")) != NULL) {
 		cp = expand(cp);
-		if (RAND_egd(cp) == -1) {
-			fprintf(stderr, catgets(catd, CATSET, 245,
-				"entropy daemon at \"%s\" not available\n"),
-					cp);
-		} else
 			state = 1;
 	} else if ((cp = value("ssl-rand-file")) != NULL) {
 		cp = expand(cp);
