Index: /tags/v0.16.x/src/xpra/client/gtk_base/gtk_tray_menu_base.py
===================================================================
--- /tags/v0.16.x/src/xpra/client/gtk_base/gtk_tray_menu_base.py	(revision 12407)
+++ /tags/v0.16.x/src/xpra/client/gtk_base/gtk_tray_menu_base.py	(revision 12408)
@@ -547,6 +547,7 @@
             copts = self.client.get_clipboard_helper_classes()
             #ugly alert: the helper does not exist yet.. we just check the helper classnames:
-            for c in copts:
-                if c.find("translated_clipboard")>0:
+            if len(copts)>0:
+                chname = copts[0].lower()
+                if chname.find("translated_clipboard")>0:
                     return self.make_translatedclipboard_optionsmenuitem()
         except:
