--- wxWidgets/src/common/intl.cpp	2005/09/18 23:38:06	1.166
+++ wxWidgets/src/common/intl.cpp	2005/09/28 09:48:03	1.166.2.1
@@ -5,7 +5,7 @@
 // Modified by: Michael N. Filippov <michael@idisys.iae.nsk.su>
 //              (2003/09/30 - PluralForms support)
 // Created:     29/01/98
-// RCS-ID:      $Id: intl.cpp,v 1.166 2005/09/18 23:38:06 VZ Exp $
+// RCS-ID:      $Id: intl.cpp,v 1.166.2.1 2005/09/28 09:48:03 VZ Exp $
 // Copyright:   (c) 1998 Vadim Zeitlin <zeitlin@dptmaths.ens-cachan.fr>
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
@@ -1260,9 +1260,14 @@
             inputConv =
             csConv = new wxCSConv(m_charset);
     }
-    else // no conversion needed
+    else // no need to convert the encoding
     {
+        // we still need the conversion for Unicode build
+#if wxUSE_UNICODE
+        inputConv = wxConvCurrent;
+#else // !wxUSE_UNICODE
         inputConv = NULL;
+#endif
     }
 
     // conversion to apply to msgid strings before looking them up: we only
