--- src/common/sckfile.cpp.orig	2005-04-16 16:50:11.000000000 -0700
+++ src/common/sckfile.cpp	2005-04-16 16:50:14.000000000 -0700
@@ -9,7 +9,7 @@
 // Licence:     wxWindows license
 /////////////////////////////////////////////////////////////////////////////
 #ifdef __GNUG__
-#pragma implementation "sckfile.h"
+#pragma implementation "file.h"
 #endif
 
 // For compilers that support precompilation, includes "wx.h".
