--- Makefile.in	2003-09-21 12:32:39.000000000 +0100
+++ Makefile.in.new	2003-09-28 13:51:51.006341440 +0100
@@ -1,6 +1,6 @@
 # $Id: wxGTK-2.4.2-contrib_animate.patch,v 1.2 2007/07/24 22:52:52 dirtyepic dead $
 
-CONTRIB_SUBDIRS=canvas fl gizmos mmedia net ogl plot stc svg xrc animate #applet
+CONTRIB_SUBDIRS=canvas fl gizmos mmedia net ogl plot stc svg xrc #animate #applet
 
 all:
 	@for d in $(CONTRIB_SUBDIRS); do (cd $$d && $(MAKE)); done
