--- wxGTK-2.4.1/contrib/src/xrc/Makefile.in	2003-09-09 16:46:13.468009816 +0100
+++ wxGTK-2.4.1/contrib/src/xrc/Makefile.in	2003-09-09 16:46:50.065446160 +0100
@@ -17,7 +17,9 @@
 HEADER_SUBDIR=xrc
 
 EXPAT_DEFS=-I$(expat_dir)/xmlparse -I$(expat_dir)/xmltok
-EXPAT_OBJECTS=xmltok.o xmlrole.o xmlparse.o
+EXPAT_OBJECTS=$(expat_dir)/xmltok/xmltok.o \
+	   $(expat_dir)/xmltok/xmlrole.o \
+	   $(expat_dir)/xmlparse/xmlparse.o
 
 HEADERS=xh_all.h xh_bttn.h xh_chckb.h xh_chckl.h xh_choic.h xh_combo.h \
         xh_dlg.h xh_gauge.h xh_html.h xh_menu.h xh_notbk.h xh_panel.h \
