diff -Naur gtk-vnc-0.3.9.orig/plugin/npshell.c gtk-vnc-0.3.9/plugin/npshell.c
--- gtk-vnc-0.3.9.orig/plugin/npshell.c	2009-08-20 15:02:56.509101940 -0700
+++ gtk-vnc-0.3.9/plugin/npshell.c	2009-08-20 15:10:52.489077176 -0700
@@ -81,8 +81,7 @@
 #include <string.h>
 
 #include <npapi.h>
-#include <npupp.h>
-
+#include <npfunctions.h>
 #include "gtk-vnc-plugin.h"
 
 /***********************************************************************
diff -Naur gtk-vnc-0.3.9.orig/plugin/npunix.c gtk-vnc-0.3.9/plugin/npunix.c
--- gtk-vnc-0.3.9.orig/plugin/npunix.c	2009-08-20 15:02:56.509101940 -0700
+++ gtk-vnc-0.3.9/plugin/npunix.c	2009-08-20 15:10:59.748825742 -0700
@@ -55,7 +55,7 @@
 
 #include <stdio.h>
 #include <npapi.h>
-#include <npupp.h>
+#include <npfunctions.h>
 
 #include <glib.h>               /* just for G_GNUC_UNUSED */
 
