From 39a93a1686668f228335dc313e3fdd489b027ea6 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 23 Sep 2017 01:05:56 +0200
Subject: [PATCH] fix base dependency

---
 primitive.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/primitive.cabal b/primitive.cabal
index af15214..06745d4 100644
--- a/primitive.cabal
+++ b/primitive.cabal
@@ -45,7 +45,7 @@ Library
         Data.Primitive.Internal.Compat
         Data.Primitive.Internal.Operations
 
-  Build-Depends: base >= 4.5 && < 4.10
+  Build-Depends: base >= 4.5 && < 4.11
                , ghc-prim >= 0.2 && < 0.6
                , transformers >= 0.2 && < 0.6
 
-- 
2.14.1

