From 9cddb7d81d3d5754d77c58836eb85fb07dfd1f08 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 28 Jan 2018 16:11:43 +0100
Subject: [PATCH] fix dependency on hunit

---
 test-framework-hunit.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test-framework-hunit.cabal b/test-framework-hunit.cabal
index ea8e2da..99b5767 100644
--- a/test-framework-hunit.cabal
+++ b/test-framework-hunit.cabal
@@ -24,7 +24,7 @@ Flag Base3
 Library
         Exposed-Modules:        Test.Framework.Providers.HUnit
 
-        Build-Depends:          test-framework >= 0.2.0, HUnit >= 1.2 && < 1.4, extensible-exceptions >= 0.1.1 && < 0.2.0
+        Build-Depends:          test-framework >= 0.2.0, HUnit >= 1.2 && < 1.7, extensible-exceptions >= 0.1.1 && < 0.2.0
         if flag(base3)
                 Build-Depends:          base >= 3 && < 4
         else
-- 
2.15.1

