From 41e5323c895f613a62b9e5b2ab1b7e85ddd793ec Mon Sep 17 00:00:00 2001
From: Bastien Dejean <nihilhill@gmail.com>
Date: Tue, 12 Apr 2016 10:43:08 +0200
Subject: [PATCH] Use the proper type for the return value of getopt

---
 xlsw.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xlsw.c b/xlsw.c
index 174b82c..cb90133 100644
--- a/xlsw.c
+++ b/xlsw.c
@@ -13,7 +13,7 @@
 int main(int argc, char *argv[])
 {
     recursive = false;
-    char opt;
+    int opt;
 
     while ((opt = getopt(argc, argv, "hvr")) != -1) {
         switch (opt) {
