--- old/src/copy.c	2007/01/03 12:43:18	1.1
+++ coreutils-6.4/src/copy.c	2007/01/03 12:43:44
@@ -1431,7 +1431,7 @@
 	 where you'd replace `18' with the integer in parentheses that
 	 was output from the perl one-liner above.
 	 If necessary, of course, change `/tmp' to some other directory.  */
-      if (errno != EXDEV)
+      if (errno != EXDEV && errno != EPERM)
 	{
 	  /* There are many ways this can happen due to a race condition.
 	     When something happens between the initial XSTAT and the
