diff -Naur coreutils-5.2.0/src/copy.c coreutils/src/copy.c
--- coreutils-5.2.0/src/copy.c	2004-02-08 02:42:04.000000000 +1100
+++ coreutils/src/copy.c	2004-02-25 02:21:10.000000000 +1100
@@ -232,7 +232,7 @@
 
   /* Compare the source dev/ino from the open file to the incoming,
      saved ones obtained via a previous call to stat.  */
-  if (! SAME_INODE (*src_sb, src_open_sb))
+  if (! SAME_INODE_ONLY (*src_sb, src_open_sb))
     {
       error (0, 0,
 	     _("skipping file %s, as it was replaced while being copied"),
diff -Naur coreutils-5.2.0/src/system.h coreutils/src/system.h
--- coreutils-5.2.0/src/system.h	2004-02-05 20:23:23.000000000 +1100
+++ coreutils/src/system.h	2004-02-25 02:21:10.000000000 +1100
@@ -750,6 +750,9 @@
   ((Stat_buf_1).st_ino == (Stat_buf_2).st_ino \
    && (Stat_buf_1).st_dev == (Stat_buf_2).st_dev)
 
+#define SAME_INODE_ONLY(Stat_buf_1, Stat_buf_2) \
+ ((Stat_buf_1).st_ino == (Stat_buf_2).st_ino)
+
 #define DOT_OR_DOTDOT(Basename) \
   (Basename[0] == '.' && (Basename[1] == '\0' \
 			  || (Basename[1] == '.' && Basename[2] == '\0')))
