diff -uNr coreutils-5.93/man/Makefile.am coreutils-5.93.orig/man/Makefile.am
--- coreutils-5.93/man/Makefile.am	2005-10-13 14:12:51.000000000 +0000
+++ coreutils-5.93.orig/man/Makefile.am	2005-11-19 13:11:38.000000000 +0000
@@ -3,7 +3,7 @@
   basename.1 cat.1 chgrp.1 chmod.1 chown.1 chroot.1 cksum.1 comm.1 \
   cp.1 csplit.1 cut.1 date.1 dd.1 df.1 dir.1 dircolors.1 dirname.1 du.1 \
   echo.1 env.1 expand.1 expr.1 factor.1 false.1 fmt.1 fold.1 groups.1 \
-  head.1 hostid.1 hostname.1 id.1 install.1 join.1 kill.1 \
+  head.1 hostid.1 id.1 install.1 join.1 kill.1 \
   link.1 ln.1 logname.1 \
   ls.1 md5sum.1 mkdir.1 mkfifo.1 mknod.1 mv.1 nice.1 nl.1 nohup.1 od.1 \
   paste.1 pathchk.1 pinky.1 pr.1 printenv.1 printf.1 ptx.1 pwd.1 readlink.1 \
@@ -52,7 +52,6 @@
 groups.1:	$(common_dep)	$(srcdir)/groups.x	../src/groups.sh
 head.1:		$(common_dep)	$(srcdir)/head.x	../src/head.c
 hostid.1:	$(common_dep)	$(srcdir)/hostid.x	../src/hostid.c
-hostname.1:	$(common_dep)	$(srcdir)/hostname.x	../src/hostname.c
 id.1:		$(common_dep)	$(srcdir)/id.x		../src/id.c
 install.1:	$(common_dep)	$(srcdir)/install.x	../src/install.c
 join.1:		$(common_dep)	$(srcdir)/join.x	../src/join.c
diff -uNr coreutils-5.93/man/Makefile.in coreutils-5.93.orig/man/Makefile.in
--- coreutils-5.93/man/Makefile.in	2005-11-05 22:12:21.000000000 +0000
+++ coreutils-5.93.orig/man/Makefile.in	2005-11-19 13:12:32.000000000 +0000
@@ -68,7 +68,7 @@
 	$(top_srcdir)/m4/fts.m4 $(top_srcdir)/m4/getaddrinfo.m4 \
 	$(top_srcdir)/m4/getcwd-path-max.m4 $(top_srcdir)/m4/getcwd.m4 \
 	$(top_srcdir)/m4/getdate.m4 $(top_srcdir)/m4/getdelim.m4 \
-	$(top_srcdir)/m4/getgroups.m4 $(top_srcdir)/m4/gethostname.m4 \
+	$(top_srcdir)/m4/getgroups.m4 \
 	$(top_srcdir)/m4/gethrxtime.m4 $(top_srcdir)/m4/getline.m4 \
 	$(top_srcdir)/m4/getndelim2.m4 $(top_srcdir)/m4/getopt.m4 \
 	$(top_srcdir)/m4/getpagesize.m4 $(top_srcdir)/m4/getpass.m4 \
@@ -303,7 +303,7 @@
   basename.1 cat.1 chgrp.1 chmod.1 chown.1 chroot.1 cksum.1 comm.1 \
   cp.1 csplit.1 cut.1 date.1 dd.1 df.1 dir.1 dircolors.1 dirname.1 du.1 \
   echo.1 env.1 expand.1 expr.1 factor.1 false.1 fmt.1 fold.1 groups.1 \
-  head.1 hostid.1 hostname.1 id.1 install.1 join.1 kill.1 \
+  head.1 hostid.1 id.1 install.1 join.1 kill.1 \
   link.1 ln.1 logname.1 \
   ls.1 md5sum.1 mkdir.1 mkfifo.1 mknod.1 mv.1 nice.1 nl.1 nohup.1 od.1 \
   paste.1 pathchk.1 pinky.1 pr.1 printenv.1 printf.1 ptx.1 pwd.1 readlink.1 \
@@ -571,7 +571,6 @@
 groups.1:	$(common_dep)	$(srcdir)/groups.x	../src/groups.sh
 head.1:		$(common_dep)	$(srcdir)/head.x	../src/head.c
 hostid.1:	$(common_dep)	$(srcdir)/hostid.x	../src/hostid.c
-hostname.1:	$(common_dep)	$(srcdir)/hostname.x	../src/hostname.c
 id.1:		$(common_dep)	$(srcdir)/id.x		../src/id.c
 install.1:	$(common_dep)	$(srcdir)/install.x	../src/install.c
 join.1:		$(common_dep)	$(srcdir)/join.x	../src/join.c
diff -uNr coreutils-5.93/src/Makefile.am coreutils-5.93.orig/src/Makefile.am
--- coreutils-5.93/src/Makefile.am	2005-07-18 07:52:18.000000000 +0000
+++ coreutils-5.93.orig/src/Makefile.am	2005-11-19 13:13:06.000000000 +0000
@@ -26,7 +26,7 @@
   cat cksum comm csplit cut expand fmt fold head join md5sum \
   nl od paste pr ptx sha1sum sort split sum tac tail tr tsort unexpand uniq wc \
   basename date dirname echo env expr factor false \
-  hostname id kill logname pathchk printenv printf pwd seq sleep tee \
+  id kill logname pathchk printenv printf pwd seq sleep tee \
   test true tty whoami yes \
   $(OPTIONAL_BIN_PROGS) $(DF_PROG)
 
diff -uNr coreutils-5.93/src/Makefile.in coreutils-5.93.orig/src/Makefile.in
--- coreutils-5.93/src/Makefile.in	2005-11-05 22:21:57.000000000 +0000
+++ coreutils-5.93.orig/src/Makefile.in	2005-11-19 13:15:31.000000000 +0000
@@ -55,7 +55,7 @@
 	tail$(EXEEXT) tr$(EXEEXT) tsort$(EXEEXT) unexpand$(EXEEXT) \
 	uniq$(EXEEXT) wc$(EXEEXT) basename$(EXEEXT) date$(EXEEXT) \
 	dirname$(EXEEXT) echo$(EXEEXT) env$(EXEEXT) expr$(EXEEXT) \
-	factor$(EXEEXT) false$(EXEEXT) hostname$(EXEEXT) id$(EXEEXT) \
+	factor$(EXEEXT) false$(EXEEXT) id$(EXEEXT) \
 	kill$(EXEEXT) logname$(EXEEXT) pathchk$(EXEEXT) \
 	printenv$(EXEEXT) printf$(EXEEXT) pwd$(EXEEXT) seq$(EXEEXT) \
 	sleep$(EXEEXT) tee$(EXEEXT) test$(EXEEXT) true$(EXEEXT) \
@@ -95,7 +95,7 @@
 	$(top_srcdir)/m4/fts.m4 $(top_srcdir)/m4/getaddrinfo.m4 \
 	$(top_srcdir)/m4/getcwd-path-max.m4 $(top_srcdir)/m4/getcwd.m4 \
 	$(top_srcdir)/m4/getdate.m4 $(top_srcdir)/m4/getdelim.m4 \
-	$(top_srcdir)/m4/getgroups.m4 $(top_srcdir)/m4/gethostname.m4 \
+	$(top_srcdir)/m4/getgroups.m4 \
 	$(top_srcdir)/m4/gethrxtime.m4 $(top_srcdir)/m4/getline.m4 \
 	$(top_srcdir)/m4/getndelim2.m4 $(top_srcdir)/m4/getopt.m4 \
 	$(top_srcdir)/m4/getpagesize.m4 $(top_srcdir)/m4/getpass.m4 \
@@ -335,11 +335,6 @@
 hostid_LDADD = $(LDADD)
 hostid_DEPENDENCIES = ../lib/libcoreutils.a $(am__DEPENDENCIES_1) \
 	../lib/libcoreutils.a
-hostname_SOURCES = hostname.c
-hostname_OBJECTS = hostname.$(OBJEXT)
-hostname_LDADD = $(LDADD)
-hostname_DEPENDENCIES = ../lib/libcoreutils.a $(am__DEPENDENCIES_1) \
-	../lib/libcoreutils.a
 id_SOURCES = id.c
 id_OBJECTS = id.$(OBJEXT)
 id_LDADD = $(LDADD)
@@ -621,7 +616,7 @@
 	$(chown_SOURCES) chroot.c cksum.c comm.c $(cp_SOURCES) \
 	csplit.c cut.c date.c dd.c df.c $(dir_SOURCES) dircolors.c \
 	dirname.c du.c echo.c env.c expand.c expr.c factor.c false.c \
-	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c hostname.c \
+	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c  \
 	id.c join.c kill.c link.c ln.c logname.c $(ls_SOURCES) \
 	$(md5sum_SOURCES) mkdir.c mkfifo.c mknod.c $(mv_SOURCES) \
 	nice.c nl.c nohup.c od.c paste.c pathchk.c pinky.c pr.c \
@@ -635,7 +630,7 @@
 	$(chown_SOURCES) chroot.c cksum.c comm.c $(cp_SOURCES) \
 	csplit.c cut.c date.c dd.c df.c $(dir_SOURCES) dircolors.c \
 	dirname.c du.c echo.c env.c expand.c expr.c factor.c false.c \
-	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c hostname.c \
+	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c \
 	id.c join.c kill.c link.c ln.c logname.c $(ls_SOURCES) \
 	$(md5sum_SOURCES) mkdir.c mkfifo.c mknod.c $(mv_SOURCES) \
 	nice.c nl.c nohup.c od.c paste.c pathchk.c pinky.c pr.c \
@@ -1067,9 +1062,6 @@
 hostid$(EXEEXT): $(hostid_OBJECTS) $(hostid_DEPENDENCIES) 
 	@rm -f hostid$(EXEEXT)
 	$(LINK) $(hostid_LDFLAGS) $(hostid_OBJECTS) $(hostid_LDADD) $(LIBS)
-hostname$(EXEEXT): $(hostname_OBJECTS) $(hostname_DEPENDENCIES) 
-	@rm -f hostname$(EXEEXT)
-	$(LINK) $(hostname_LDFLAGS) $(hostname_OBJECTS) $(hostname_LDADD) $(LIBS)
 id$(EXEEXT): $(id_OBJECTS) $(id_DEPENDENCIES) 
 	@rm -f id$(EXEEXT)
 	$(LINK) $(id_LDFLAGS) $(id_OBJECTS) $(id_LDADD) $(LIBS)
@@ -1318,7 +1310,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/fold.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/head.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hostid.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hostname.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/id.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/install.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/join.Po@am__quote@
