diff -uNr coreutils-6.4.old/man/Makefile.am coreutils-6.4/man/Makefile.am
--- coreutils-6.4.old/man/Makefile.am	2006-11-07 15:31:03.000000000 +0100
+++ coreutils-6.4/man/Makefile.am	2006-11-07 15:34:00.000000000 +0100
@@ -22,7 +22,7 @@
   base64.1 basename.1 cat.1 chgrp.1 chmod.1 chown.1 chroot.1 cksum.1 comm.1 \
   cp.1 csplit.1 cut.1 date.1 dd.1 df.1 dir.1 dircolors.1 dirname.1 du.1 \
   echo.1 env.1 expand.1 expr.1 factor.1 false.1 fmt.1 fold.1 groups.1 \
-  head.1 hostid.1 hostname.1 id.1 install.1 join.1 kill.1 \
+  head.1 hostid.1 id.1 install.1 join.1 kill.1 \
   link.1 ln.1 logname.1 \
   ls.1 md5sum.1 mkdir.1 mkfifo.1 mknod.1 mv.1 nice.1 nl.1 nohup.1 od.1 \
   paste.1 pathchk.1 pinky.1 pr.1 printenv.1 printf.1 ptx.1 pwd.1 readlink.1 \
@@ -73,7 +73,6 @@
 groups.1:	$(common_dep)	$(srcdir)/groups.x	../src/groups.sh
 head.1:		$(common_dep)	$(srcdir)/head.x	../src/head.c
 hostid.1:	$(common_dep)	$(srcdir)/hostid.x	../src/hostid.c
-hostname.1:	$(common_dep)	$(srcdir)/hostname.x	../src/hostname.c
 id.1:		$(common_dep)	$(srcdir)/id.x		../src/id.c
 install.1:	$(common_dep)	$(srcdir)/install.x	../src/install.c
 join.1:		$(common_dep)	$(srcdir)/join.x	../src/join.c
diff -uNr coreutils-6.4.old/man/Makefile.in coreutils-6.4/man/Makefile.in
--- coreutils-6.4.old/man/Makefile.in	2006-11-07 15:31:03.000000000 +0100
+++ coreutils-6.4/man/Makefile.in	2006-11-07 15:35:10.000000000 +0100
@@ -380,7 +380,7 @@
   base64.1 basename.1 cat.1 chgrp.1 chmod.1 chown.1 chroot.1 cksum.1 comm.1 \
   cp.1 csplit.1 cut.1 date.1 dd.1 df.1 dir.1 dircolors.1 dirname.1 du.1 \
   echo.1 env.1 expand.1 expr.1 factor.1 false.1 fmt.1 fold.1 groups.1 \
-  head.1 hostid.1 hostname.1 id.1 install.1 join.1 kill.1 \
+  head.1 hostid.1 id.1 install.1 join.1 kill.1 \
   link.1 ln.1 logname.1 \
   ls.1 md5sum.1 mkdir.1 mkfifo.1 mknod.1 mv.1 nice.1 nl.1 nohup.1 od.1 \
   paste.1 pathchk.1 pinky.1 pr.1 printenv.1 printf.1 ptx.1 pwd.1 readlink.1 \
@@ -659,7 +659,6 @@
 groups.1:	$(common_dep)	$(srcdir)/groups.x	../src/groups.sh
 head.1:		$(common_dep)	$(srcdir)/head.x	../src/head.c
 hostid.1:	$(common_dep)	$(srcdir)/hostid.x	../src/hostid.c
-hostname.1:	$(common_dep)	$(srcdir)/hostname.x	../src/hostname.c
 id.1:		$(common_dep)	$(srcdir)/id.x		../src/id.c
 install.1:	$(common_dep)	$(srcdir)/install.x	../src/install.c
 join.1:		$(common_dep)	$(srcdir)/join.x	../src/join.c
diff -uNr coreutils-6.4.old/src/Makefile.am coreutils-6.4/src/Makefile.am
--- coreutils-6.4.old/src/Makefile.am	2006-11-07 15:31:03.000000000 +0100
+++ coreutils-6.4/src/Makefile.am	2006-11-07 15:35:35.000000000 +0100
@@ -27,7 +27,7 @@
   nl od paste pr ptx sha1sum sha224sum sha256sum sha384sum sha512sum \
   shuf sort split sum tac tail tr tsort unexpand uniq wc \
   basename date dirname echo env expr factor false \
-  hostname id kill logname pathchk printenv printf pwd seq sleep tee \
+  id kill logname pathchk printenv printf pwd seq sleep tee \
   test true tty whoami yes \
   base64 \
   $(OPTIONAL_BIN_PROGS) $(DF_PROG)
diff -uNr coreutils-6.4.old/src/Makefile.in coreutils-6.4/src/Makefile.in
--- coreutils-6.4.old/src/Makefile.in	2006-11-07 15:31:03.000000000 +0100
+++ coreutils-6.4/src/Makefile.in	2006-11-07 15:36:59.000000000 +0100
@@ -53,7 +53,7 @@
 	tsort$(EXEEXT) unexpand$(EXEEXT) uniq$(EXEEXT) wc$(EXEEXT) \
 	basename$(EXEEXT) date$(EXEEXT) dirname$(EXEEXT) echo$(EXEEXT) \
 	env$(EXEEXT) expr$(EXEEXT) factor$(EXEEXT) false$(EXEEXT) \
-	hostname$(EXEEXT) id$(EXEEXT) kill$(EXEEXT) logname$(EXEEXT) \
+	id$(EXEEXT) kill$(EXEEXT) logname$(EXEEXT) \
 	pathchk$(EXEEXT) printenv$(EXEEXT) printf$(EXEEXT) \
 	pwd$(EXEEXT) seq$(EXEEXT) sleep$(EXEEXT) tee$(EXEEXT) \
 	test$(EXEEXT) true$(EXEEXT) tty$(EXEEXT) whoami$(EXEEXT) \
@@ -349,11 +349,6 @@
 hostid_LDADD = $(LDADD)
 hostid_DEPENDENCIES = ../lib/libcoreutils.a $(am__DEPENDENCIES_1) \
 	../lib/libcoreutils.a
-hostname_SOURCES = hostname.c
-hostname_OBJECTS = hostname.$(OBJEXT)
-hostname_LDADD = $(LDADD)
-hostname_DEPENDENCIES = ../lib/libcoreutils.a $(am__DEPENDENCIES_1) \
-	../lib/libcoreutils.a
 id_SOURCES = id.c
 id_OBJECTS = id.$(OBJEXT)
 id_LDADD = $(LDADD)
@@ -663,7 +658,7 @@
 	chmod.c $(chown_SOURCES) chroot.c cksum.c comm.c $(cp_SOURCES) \
 	csplit.c cut.c date.c dd.c df.c $(dir_SOURCES) dircolors.c \
 	dirname.c du.c echo.c env.c expand.c expr.c factor.c false.c \
-	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c hostname.c \
+	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c \
 	id.c join.c kill.c link.c ln.c logname.c $(ls_SOURCES) \
 	$(md5sum_SOURCES) mkdir.c mkfifo.c mknod.c $(mv_SOURCES) \
 	nice.c nl.c nohup.c od.c paste.c pathchk.c pinky.c pr.c \
@@ -679,7 +674,7 @@
 	chmod.c $(chown_SOURCES) chroot.c cksum.c comm.c $(cp_SOURCES) \
 	csplit.c cut.c date.c dd.c df.c $(dir_SOURCES) dircolors.c \
 	dirname.c du.c echo.c env.c expand.c expr.c factor.c false.c \
-	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c hostname.c \
+	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c \
 	id.c join.c kill.c link.c ln.c logname.c $(ls_SOURCES) \
 	$(md5sum_SOURCES) mkdir.c mkfifo.c mknod.c $(mv_SOURCES) \
 	nice.c nl.c nohup.c od.c paste.c pathchk.c pinky.c pr.c \
@@ -1185,9 +1180,6 @@
 hostid$(EXEEXT): $(hostid_OBJECTS) $(hostid_DEPENDENCIES) 
 	@rm -f hostid$(EXEEXT)
 	$(LINK) $(hostid_OBJECTS) $(hostid_LDADD) $(LIBS)
-hostname$(EXEEXT): $(hostname_OBJECTS) $(hostname_DEPENDENCIES) 
-	@rm -f hostname$(EXEEXT)
-	$(LINK) $(hostname_OBJECTS) $(hostname_LDADD) $(LIBS)
 id$(EXEEXT): $(id_OBJECTS) $(id_DEPENDENCIES) 
 	@rm -f id$(EXEEXT)
 	$(LINK) $(id_OBJECTS) $(id_LDADD) $(LIBS)
@@ -1452,7 +1444,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/fold.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/head.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hostid.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hostname.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/id.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/install.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/join.Po@am__quote@
