diff -ur coreutils-6.4.old/man/Makefile.am coreutils-6.4/man/Makefile.am
--- coreutils-6.4.old/man/Makefile.am	2006-10-22 17:54:15.000000000 +0100
+++ coreutils-6.4/man/Makefile.am	2007-02-12 19:36:45.000000000 +0000
@@ -22,7 +22,7 @@
   base64.1 basename.1 cat.1 chgrp.1 chmod.1 chown.1 chroot.1 cksum.1 comm.1 \
   cp.1 csplit.1 cut.1 date.1 dd.1 df.1 dir.1 dircolors.1 dirname.1 du.1 \
   echo.1 env.1 expand.1 expr.1 factor.1 false.1 fmt.1 fold.1 groups.1 \
-  head.1 hostname.1 id.1 install.1 join.1 kill.1 \
+  head.1 id.1 install.1 join.1 \
   link.1 ln.1 logname.1 \
   ls.1 md5sum.1 mkdir.1 mkfifo.1 mknod.1 mv.1 nice.1 nl.1 nohup.1 od.1 \
   paste.1 pathchk.1 pinky.1 pr.1 printenv.1 printf.1 ptx.1 pwd.1 readlink.1 \
@@ -73,11 +73,9 @@
 groups.1:	$(common_dep)	$(srcdir)/groups.x	../src/groups.sh
 head.1:		$(common_dep)	$(srcdir)/head.x	../src/head.c
 hostid.1:	$(common_dep)	$(srcdir)/hostid.x	../src/hostid.c
-hostname.1:	$(common_dep)	$(srcdir)/hostname.x	../src/hostname.c
 id.1:		$(common_dep)	$(srcdir)/id.x		../src/id.c
 install.1:	$(common_dep)	$(srcdir)/install.x	../src/install.c
 join.1:		$(common_dep)	$(srcdir)/join.x	../src/join.c
-kill.1:		$(common_dep)	$(srcdir)/kill.x	../src/kill.c
 link.1:		$(common_dep)	$(srcdir)/link.x	../src/link.c
 ln.1:		$(common_dep)	$(srcdir)/ln.x		../src/ln.c
 logname.1:	$(common_dep)	$(srcdir)/logname.x	../src/logname.c
diff -ur coreutils-6.4.old/man/Makefile.in coreutils-6.4/man/Makefile.in
--- coreutils-6.4.old/man/Makefile.in	2006-10-22 21:36:24.000000000 +0100
+++ coreutils-6.4/man/Makefile.in	2007-02-12 19:37:37.000000000 +0000
@@ -380,7 +380,7 @@
   base64.1 basename.1 cat.1 chgrp.1 chmod.1 chown.1 chroot.1 cksum.1 comm.1 \
   cp.1 csplit.1 cut.1 date.1 dd.1 df.1 dir.1 dircolors.1 dirname.1 du.1 \
   echo.1 env.1 expand.1 expr.1 factor.1 false.1 fmt.1 fold.1 groups.1 \
-  head.1 hostname.1 id.1 install.1 join.1 kill.1 \
+  head.1 id.1 install.1 join.1 \
   link.1 ln.1 logname.1 \
   ls.1 md5sum.1 mkdir.1 mkfifo.1 mknod.1 mv.1 nice.1 nl.1 nohup.1 od.1 \
   paste.1 pathchk.1 pinky.1 pr.1 printenv.1 printf.1 ptx.1 pwd.1 readlink.1 \
@@ -659,11 +659,9 @@
 groups.1:	$(common_dep)	$(srcdir)/groups.x	../src/groups.sh
 head.1:		$(common_dep)	$(srcdir)/head.x	../src/head.c
 hostid.1:	$(common_dep)	$(srcdir)/hostid.x	../src/hostid.c
-hostname.1:	$(common_dep)	$(srcdir)/hostname.x	../src/hostname.c
 id.1:		$(common_dep)	$(srcdir)/id.x		../src/id.c
 install.1:	$(common_dep)	$(srcdir)/install.x	../src/install.c
 join.1:		$(common_dep)	$(srcdir)/join.x	../src/join.c
-kill.1:		$(common_dep)	$(srcdir)/kill.x	../src/kill.c
 link.1:		$(common_dep)	$(srcdir)/link.x	../src/link.c
 ln.1:		$(common_dep)	$(srcdir)/ln.x		../src/ln.c
 logname.1:	$(common_dep)	$(srcdir)/logname.x	../src/logname.c
diff -ur coreutils-6.4.old/src/Makefile.am coreutils-6.4/src/Makefile.am
--- coreutils-6.4.old/src/Makefile.am	2006-10-22 17:54:15.000000000 +0100
+++ coreutils-6.4/src/Makefile.am	2007-02-12 19:38:20.000000000 +0000
@@ -27,7 +27,7 @@
   nl od paste pr ptx sha1sum sha224sum sha256sum sha384sum sha512sum \
   shuf sort split sum tac tail tr tsort unexpand uniq wc \
   basename date dirname echo env expr factor false \
-  hostname id kill logname pathchk printenv printf pwd seq sleep tee \
+  id logname pathchk printenv printf pwd seq sleep tee \
   test true tty whoami yes \
   base64 \
   $(OPTIONAL_BIN_PROGS) $(DF_PROG)
diff -ur coreutils-6.4.old/src/Makefile.in coreutils-6.4/src/Makefile.in
--- coreutils-6.4.old/src/Makefile.in	2006-10-22 22:41:18.000000000 +0100
+++ coreutils-6.4/src/Makefile.in	2007-02-12 19:40:02.000000000 +0000
@@ -53,7 +53,7 @@
 	tsort$(EXEEXT) unexpand$(EXEEXT) uniq$(EXEEXT) wc$(EXEEXT) \
 	basename$(EXEEXT) date$(EXEEXT) dirname$(EXEEXT) echo$(EXEEXT) \
 	env$(EXEEXT) expr$(EXEEXT) factor$(EXEEXT) false$(EXEEXT) \
-	hostname$(EXEEXT) id$(EXEEXT) kill$(EXEEXT) logname$(EXEEXT) \
+	id$(EXEEXT) logname$(EXEEXT) \
 	pathchk$(EXEEXT) printenv$(EXEEXT) printf$(EXEEXT) \
 	pwd$(EXEEXT) seq$(EXEEXT) sleep$(EXEEXT) tee$(EXEEXT) \
 	test$(EXEEXT) true$(EXEEXT) tty$(EXEEXT) whoami$(EXEEXT) \
@@ -349,11 +349,6 @@
 hostid_LDADD = $(LDADD)
 hostid_DEPENDENCIES = ../lib/libcoreutils.a $(am__DEPENDENCIES_1) \
 	../lib/libcoreutils.a
-hostname_SOURCES = hostname.c
-hostname_OBJECTS = hostname.$(OBJEXT)
-hostname_LDADD = $(LDADD)
-hostname_DEPENDENCIES = ../lib/libcoreutils.a $(am__DEPENDENCIES_1) \
-	../lib/libcoreutils.a
 id_SOURCES = id.c
 id_OBJECTS = id.$(OBJEXT)
 id_LDADD = $(LDADD)
@@ -364,11 +359,6 @@
 join_LDADD = $(LDADD)
 join_DEPENDENCIES = ../lib/libcoreutils.a $(am__DEPENDENCIES_1) \
 	../lib/libcoreutils.a
-kill_SOURCES = kill.c
-kill_OBJECTS = kill.$(OBJEXT)
-kill_LDADD = $(LDADD)
-kill_DEPENDENCIES = ../lib/libcoreutils.a $(am__DEPENDENCIES_1) \
-	../lib/libcoreutils.a
 link_SOURCES = link.c
 link_OBJECTS = link.$(OBJEXT)
 link_LDADD = $(LDADD)
@@ -663,8 +653,8 @@
 	chmod.c $(chown_SOURCES) chroot.c cksum.c comm.c $(cp_SOURCES) \
 	csplit.c cut.c date.c dd.c df.c $(dir_SOURCES) dircolors.c \
 	dirname.c du.c echo.c env.c expand.c expr.c factor.c false.c \
-	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c hostname.c \
-	id.c join.c kill.c link.c ln.c logname.c $(ls_SOURCES) \
+	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c \
+	id.c join.c link.c ln.c logname.c $(ls_SOURCES) \
 	$(md5sum_SOURCES) mkdir.c mkfifo.c mknod.c $(mv_SOURCES) \
 	nice.c nl.c nohup.c od.c paste.c pathchk.c pinky.c pr.c \
 	printenv.c printf.c ptx.c pwd.c readlink.c $(rm_SOURCES) \
@@ -679,8 +669,8 @@
 	chmod.c $(chown_SOURCES) chroot.c cksum.c comm.c $(cp_SOURCES) \
 	csplit.c cut.c date.c dd.c df.c $(dir_SOURCES) dircolors.c \
 	dirname.c du.c echo.c env.c expand.c expr.c factor.c false.c \
-	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c hostname.c \
-	id.c join.c kill.c link.c ln.c logname.c $(ls_SOURCES) \
+	fmt.c fold.c $(ginstall_SOURCES) head.c hostid.c \
+	id.c join.c link.c ln.c logname.c $(ls_SOURCES) \
 	$(md5sum_SOURCES) mkdir.c mkfifo.c mknod.c $(mv_SOURCES) \
 	nice.c nl.c nohup.c od.c paste.c pathchk.c pinky.c pr.c \
 	printenv.c printf.c ptx.c pwd.c readlink.c $(rm_SOURCES) \
@@ -1185,18 +1175,12 @@
 hostid$(EXEEXT): $(hostid_OBJECTS) $(hostid_DEPENDENCIES) 
 	@rm -f hostid$(EXEEXT)
 	$(LINK) $(hostid_OBJECTS) $(hostid_LDADD) $(LIBS)
-hostname$(EXEEXT): $(hostname_OBJECTS) $(hostname_DEPENDENCIES) 
-	@rm -f hostname$(EXEEXT)
-	$(LINK) $(hostname_OBJECTS) $(hostname_LDADD) $(LIBS)
 id$(EXEEXT): $(id_OBJECTS) $(id_DEPENDENCIES) 
 	@rm -f id$(EXEEXT)
 	$(LINK) $(id_OBJECTS) $(id_LDADD) $(LIBS)
 join$(EXEEXT): $(join_OBJECTS) $(join_DEPENDENCIES) 
 	@rm -f join$(EXEEXT)
 	$(LINK) $(join_OBJECTS) $(join_LDADD) $(LIBS)
-kill$(EXEEXT): $(kill_OBJECTS) $(kill_DEPENDENCIES) 
-	@rm -f kill$(EXEEXT)
-	$(LINK) $(kill_OBJECTS) $(kill_LDADD) $(LIBS)
 link$(EXEEXT): $(link_OBJECTS) $(link_DEPENDENCIES) 
 	@rm -f link$(EXEEXT)
 	$(LINK) $(link_OBJECTS) $(link_LDADD) $(LIBS)
@@ -1452,11 +1436,9 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/fold.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/head.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hostid.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hostname.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/id.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/install.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/join.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/kill.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/lbracket.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/link.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ln.Po@am__quote@
