--- src/mixerctl.cc.old	2005-05-14 19:34:28.000000000 -0700
+++ src/mixerctl.cc	2005-05-14 19:33:50.000000000 -0700
@@ -50,7 +50,7 @@
 
     if((fd = ::open(devname.c_str(), O_RDWR)) != -1) {
 	if(ioctl(fd, SOUND_MIXER_READ_DEVMASK, &devmask) != -1) {
-	    for(ct = ctVolume; ct != channeltype_end; (int) ct += 1) {
+	    for(ct = ctVolume; ct != channeltype_end; ct = (mixerctl::channeltype) ((int) ct + 1)) {
 		if((1 << (int) ct) & devmask) {
 		    channels.insert(ct);
 		}
