https://bugs.gentoo.org/show_bug.cgi?id=271843

--- mplayerplug-in-3.55-orig/Source/plugin-list.cpp
+++ mplayerplug-in-3.55/Source/plugin-list.cpp
@@ -132,9 +132,9 @@ void insert_area(Node * parent, char *ta
 
 void find_area_tags(const char *smilbuffer, Node * parent)
 {
-    char *startarea;
-    char *endvideo;
-    char *start;
+    const char *startarea;
+    const char *endvideo;
+    const char *start;
 
     int tagtime = 0;
     char tagtarget[128];
--- mplayerplug-in-3.55-orig/Source/plugin-setup.h
+++ mplayerplug-in-3.55/Source/plugin-setup.h
@@ -174,7 +174,7 @@ char *getURLHostname(char *url);
 char *getURLFilename(const char *url);
 int isMms(char *url, int nomediacache);
 void mmsToHttp(char *dest, char *src);
-int sendCommand(nsPluginInstance * instance, char *command);
+int sendCommand(nsPluginInstance * instance, const char *command);
 int URLcmp(const char *url1, const char *url2);
 extern void remove_quotes(char *url);
 void killmplayer(nsPluginInstance * instance);
--- mplayerplug-in-3.55-orig/Source/plugin-support.cpp
+++ mplayerplug-in-3.55/Source/plugin-support.cpp
@@ -109,7 +109,7 @@ char *getURLFilename(const char *url)
 {
 
     char *filename;
-    char *tmp;
+    const char *tmp;
     int len;
 
     if (DEBUG > 1)
@@ -454,7 +454,7 @@ void remove_quotes(char *url)
 // in cleanup routines (like destroyCB and shut), when we know that
 // the player thread is not running, it is safe to call without locking
 
-int sendCommand(nsPluginInstance * instance, char *command)
+int sendCommand(nsPluginInstance * instance, const char *command)
 {
     int retval;
     char buffer[1024];
