--- configure.in	2008-06-24 09:12:43.000000000 -0400
+++ configure.in	2008-09-19 00:32:17.000000000 -0400
@@ -28,7 +28,7 @@
    AC_SUBST(PKG_CONFIG_PATH)
    AC_MSG_NOTICE("Using pkg-config path of ${PKG_CONFIG_PATH}")
 fi   
-   
+
 #tell us where gecko-sdk is, try to find if not provided
 
 AC_ARG_WITH([gecko-sdk],[  --with-gecko-sdk=<path> prefix of gecko-sdk installation],
@@ -53,48 +53,49 @@
 fi
 
 ],[
-	AC_MSG_NOTICE([Determining mozilla/firefox packages to build against])
-	PKG_CHECK_MODULES(MOZPLUG,[mozilla-plugin mozilla-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir mozilla-plugin`"
-MOZILLA_HOME=`$PKG_CONFIG --variable=libdir mozilla-plugin`
-GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir mozilla-xpcom`
-],[AC_MSG_WARN([mozilla-plugin not found])])
-
-	if test "$MOZPLUG_CFLAGS" = ""; then
-		PKG_CHECK_MODULES(MOZPLUG,[firefox-plugin firefox-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir firefox-plugin`"
-MOZILLA_HOME=`$PKG_CONFIG --variable=libdir firefox-plugin`
-GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir firefox-xpcom`
-],[AC_MSG_WARN([firefox-plugin not found])])
-	fi
-	if test "$MOZPLUG_CFLAGS" = ""; then
-		PKG_CHECK_MODULES(MOZPLUG,[seamonkey-plugin seamonkey-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir seamonkey-plugin`"
-MOZILLA_HOME=`$PKG_CONFIG --variable=libdir seamonkey-plugin`
-GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir seamonkey-xpcom`
-],[AC_MSG_WARN([seamonkey-plugin not found])])
-	fi
-	if test "$MOZPLUG_CFLAGS" = ""; then
-		PKG_CHECK_MODULES(MOZPLUG,[xulrunner-plugin xulrunner-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir xulrunner-plugin`"
-MOZILLA_HOME=`$PKG_CONFIG --variable=libdir xulrunner-plugin`
-GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir xulrunner-xpcom`
-],[AC_MSG_WARN([xulrunner-plugin not found])])
-	fi
-	if test "$MOZPLUG_CFLAGS" = ""; then
-		PKG_CHECK_MODULES(MOZPLUG,[libxul],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir libxul`"
-MOZILLA_HOME=`$PKG_CONFIG --variable=libdir libxul`
-GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir libxul`
-],[AC_MSG_WARN([libxul not found])])
-	fi
-	if test "$MOZPLUG_CFLAGS" = ""; then
-		PKG_CHECK_MODULES(MOZPLUG,[iceape-plugin iceape-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir iceape-plugin`"
-MOZILLA_HOME=`$PKG_CONFIG --variable=libdir iceape-plugin`
-GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir iceape-xpcom`
-],[AC_MSG_WARN([iceape-plugin not found])])
-	fi
-
-	if test "$MOZPLUG_CFLAGS" = ""; then
-		AC_MSG_ERROR([Unable to find mozilla or firefox development files])
-	fi
-  ]
-)
+
+PKG_CHECK_MODULES(MOZPLUG,[mozilla-plugin mozilla-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir mozilla-plugin`"
+	MOZILLA_HOME=`$PKG_CONFIG --variable=libdir mozilla-plugin`
+	GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir mozilla-xpcom`
+	],[AC_MSG_WARN([mozilla-plugin not found])])
+
+if test "$MOZPLUG_CFLAGS" = ""; then
+	PKG_CHECK_MODULES(MOZPLUG,[firefox-plugin firefox-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir firefox-plugin`"
+		MOZILLA_HOME=`$PKG_CONFIG --variable=libdir firefox-plugin`
+		GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir firefox-xpcom`
+		],[AC_MSG_WARN([firefox-plugin not found])])
+fi
+if test "$MOZPLUG_CFLAGS" = ""; then
+	PKG_CHECK_MODULES(MOZPLUG,[seamonkey-plugin seamonkey-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir seamonkey-plugin`"
+		MOZILLA_HOME=`$PKG_CONFIG --variable=libdir seamonkey-plugin`
+		GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir seamonkey-xpcom`
+		],[AC_MSG_WARN([seamonkey-plugin not found])])
+fi
+
+if test "$MOZPLUG_CFLAGS" = ""; then
+	PKG_CHECK_MODULES(MOZPLUG,[xulrunner-plugin xulrunner-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir xulrunner-plugin`"
+		MOZILLA_HOME=`$PKG_CONFIG --variable=libdir xulrunner-plugin`
+		GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir xulrunner-xpcom`
+		],[AC_MSG_WARN([xulrunner-plugin not found])])
+fi
+
+if test "$MOZPLUG_CFLAGS" = ""; then
+	PKG_CHECK_MODULES(MOZPLUG,[libxul],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir libxul`"
+		MOZILLA_HOME=`$PKG_CONFIG --variable=sdkdir libxul`
+		GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir libxul`/`pkg-config --variable=includetype libxul`
+		],[AC_MSG_WARN([xulrunner not found])])
+fi
+if test "$MOZPLUG_CFLAGS" = ""; then
+	PKG_CHECK_MODULES(MOZPLUG,[iceape-plugin iceape-xpcom],[AC_SUBST(MOZPLUG_CFLAGS) MOZPLUG_CFLAGS="${MOZPLUG_CFLAGS} -I`$PKG_CONFIG --variable=includedir iceape-plugin`"
+		MOZILLA_HOME=`$PKG_CONFIG --variable=libdir iceape-plugin`
+		GECKO_IDLDIR=`$PKG_CONFIG --variable=idldir iceape-xpcom`
+		],[AC_MSG_WARN([iceape-plugin not found])])
+fi
+])
+
+if test "$MOZPLUG_CFLAGS" = ""; then
+	AC_MSG_ERROR([Unable to find mozilla or firefox development files])
+fi
 
 AC_ARG_WITH([mozilla-home],[  --with-mozilla-home=<path> prefix of mozilla/firefox installation],
 [MOZ_HOME=$with_mozilla_home
