--- Makefile.in.orig
+++ Makefile.in
@@ -102,7 +102,7 @@ AM_RECURSIVE_TARGETS = $(RECURSIVE_TARGETS:-recursive=
 	distdir dist dist-all distcheck
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = libratbox libltdl src modules tools doc help bandb ssld \
+DIST_SUBDIRS = libratbox src modules tools doc help bandb ssld \
 	resolver
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
@@ -295,10 +295,10 @@ servicesdir = @moduledir@/autoload
 AM_CFLAGS = $(WARNFLAGS)
 ircd_SOURCES = main.c
 @MINGW_TRUE@EXTRA_FLAGS = -no-undefined -Wl,--enable-runtime-pseudo-reloc -export-symbols-regex '*'
-@STATIC_MODULES_FALSE@SUBDIRS = libratbox libltdl src modules tools doc help bandb ssld resolver
-@STATIC_MODULES_TRUE@SUBDIRS = libratbox libltdl modules src tools doc help bandb ssld resolver
-@STATIC_MODULES_FALSE@ircd_LDADD = libratbox/src/libratbox.la src/libcore.la $(LIBLTDL)
-@STATIC_MODULES_TRUE@ircd_LDADD = libratbox/src/libratbox.la modules/libmodules.la src/libcore.la modules/static_modules.o $(LIBLTDL) $(DLOPEN)
+@STATIC_MODULES_FALSE@SUBDIRS = libratbox src modules tools doc help bandb ssld resolver
+@STATIC_MODULES_TRUE@SUBDIRS = libratbox modules src tools doc help bandb ssld resolver
+@STATIC_MODULES_FALSE@ircd_LDADD = libratbox/src/libratbox.la src/libcore.la
+@STATIC_MODULES_TRUE@ircd_LDADD = libratbox/src/libratbox.la modules/libmodules.la src/libcore.la modules/static_modules.o $(DLOPEN)
 @STATIC_MODULES_FALSE@ircd_LDFLAGS = $(EXTRA_FLAGS) -dlopen self
 all: all-recursive
 
--- configure.ac.orig
+++ configure.ac
@@ -109,7 +109,7 @@ done
 
 LIBTOOL="$LIBTOOL --silent"
 
-AC_CONFIG_SUBDIRS(libltdl libratbox)
+AC_CONFIG_SUBDIRS(libratbox)
 
 case "$host_os" in
 *cygwin*)
--- Makefile.am.orig
+++ Makefile.am
@@ -24,14 +24,14 @@
 
 if !STATIC_MODULES
 
-SUBDIRS = libratbox libltdl src modules tools doc help bandb ssld resolver
+SUBDIRS = libratbox src modules tools doc help bandb ssld resolver
 
 ircd_LDADD = libratbox/src/libratbox.la src/libcore.la $(LIBLTDL)
 ircd_LDFLAGS = $(EXTRA_FLAGS) -dlopen self
 
 else
 
-SUBDIRS = libratbox libltdl modules src tools doc help bandb ssld resolver
+SUBDIRS = libratbox modules src tools doc help bandb ssld resolver
 ircd_LDADD = libratbox/src/libratbox.la modules/libmodules.la src/libcore.la modules/static_modules.o $(LIBLTDL) $(DLOPEN)
 
 
