--- libratbox/src/openssl.c.orig	2016-01-13 21:10:17.000000000 +0300
+++ libratbox/src/openssl.c	2016-07-12 17:09:15.750938732 +0300
@@ -329,7 +329,11 @@
 	}
 	tls_opts = SSL_CTX_get_options(ssl_server_ctx);
 	/* Disable SSLv2, make the client use our settings */
-	tls_opts |= SSL_OP_NO_SSLv2 | SSL_OP_NO_COMPRESSION | SSL_OP_CIPHER_SERVER_PREFERENCE | SSL_OP_NO_SSLv3;
+	tls_opts |= SSL_OP_NO_SSLv2 | SSL_OP_CIPHER_SERVER_PREFERENCE | SSL_OP_NO_SSLv3;
+
+#ifdef SSL_OP_NO_COMPRESSION
+	tls_opts |= SSL_OP_NO_COMPRESSION;
+#endif
 
 #ifdef SSL_OP_NO_TICKET
 	tls_opts |= SSL_OP_NO_TICKET;
