--- configure.in.orig	2012-01-02 04:47:37.000000000 +0400
+++ configure.in	2012-06-04 23:41:10.871001151 +0400
@@ -431,14 +431,14 @@
 
 	if test x$with_libnl != xno ; then
 		#
-		# Try libnl 2.x first.
+		# Try libnl 3.x first.
 		#
-		AC_CHECK_LIB(nl, nl_socket_alloc,
+		AC_CHECK_LIB(nl-3, nl_socket_alloc,
 		[
 			#
-			# Yes, we have libnl 2.x.
+			# Yes, we have libnl 3.x.
 			#
-			LIBS="-lnl-genl -lnl $LIBS"
+			LIBS="-lnl-genl-3 -lnl-3 $LIBS"
 			AC_DEFINE(HAVE_LIBNL,1,[if libnl exists])
 			AC_DEFINE(HAVE_LIBNL_2_x,1,[if libnl exists and is version 2.x])
 		],
