--- src/gnome-mount.c.orig	2008-04-17 03:41:32.000000000 +0400
+++ src/gnome-mount.c		2011-10-25 01:24:30.000000000 +0400
@@ -30,6 +30,7 @@
 #include <fcntl.h>
 #include <string.h>
 #include <stdlib.h>
+#include <locale.h>
 
 #include <dbus/dbus.h>
 #include <dbus/dbus-glib-lowlevel.h>
@@ -39,6 +40,8 @@
 #include <glib/gi18n.h>
 #include <gconf/gconf-client.h>
 #include <gdk/gdkx.h>
+#include <gtk/gtkmessagedialog.h>
+#include <gtk/gtkstock.h>
 #include "copy-paste/gnome-password-dialog.h"
 #include <gnome-keyring.h>
 #include <libnotify/notify.h>
