diff -uprN gnome-mount-0.8.orig/src/gnome-mount.c gnome-mount-0.8/src/gnome-mount.c
--- gnome-mount-0.8.orig/src/gnome-mount.c	2009-06-24 08:15:31.000000000 +0800
+++ gnome-mount-0.8/src/gnome-mount.c	2009-06-24 08:16:00.000000000 +0800
@@ -30,6 +30,7 @@
 #include <fcntl.h>
 #include <string.h>
 #include <stdlib.h>
+#include <locale.h>
 
 #include <dbus/dbus.h>
 #include <dbus/dbus-glib-lowlevel.h>
