diff --git a/configure.ac b/configure.ac
index e2433da..431cf51 100644
--- a/configure.ac
+++ b/configure.ac
@@ -75,14 +75,9 @@ fi
 AC_SUBST(MAGICK_CPPFLAGS)
 AC_SUBST(MAGICK_LIBS)
 
-AC_CHECK_PROGS(FRIBIDICONFIG, [fribidi-config])
-if test -n "$FRIBIDICONFIG"; then
-   FRIBIDI_CPPFLAGS="`$FRIBIDICONFIG --cflags`"
-   FRIBIDI_LIBS="`$FRIBIDICONFIG --libs`"
-   AC_DEFINE(HAVE_FRIBIDI, 1, [Whether FriBiDi is available])
-fi
+PKG_CHECK_MODULES([FRIBIDI], [fribidi], [AC_DEFINE(HAVE_FRIBIDI, 1, [Whether FriBiDi is available])])
 
-AC_SUBST(FRIBIDI_CPPFLAGS)
+AC_SUBST(FRIBIDI_CFLAGS)
 AC_SUBST(FRIBIDI_LIBS)
 
 AC_CHECK_PROGS(FREETYPECONFIG, [freetype-config])
diff --git a/src/Makefile.am b/src/Makefile.am
index 41c8520..33cd80f 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -7,7 +7,7 @@ endif
 
 nodist_bin_SCRIPTS = dvddirdel
 
-AM_CPPFLAGS   = -DSYSCONFDIR="\"$(sysconfdir)\"" @XML_CPPFLAGS@ @MAGICK_CPPFLAGS@ @FREETYPE_CPPFLAGS@ @FRIBIDI_CPPFLAGS@
+AM_CPPFLAGS   = -DSYSCONFDIR="\"$(sysconfdir)\"" @XML_CPPFLAGS@ @MAGICK_CPPFLAGS@ @FREETYPE_CPPFLAGS@ @FRIBIDI_CFLAGS@
 AM_CFLAGS = -Wall
 
 dvdauthor_SOURCES = dvdauthor.c dvdauthor.h da-internal.h \
