diff --git a/src/subreader.c b/src/subreader.c
index 01f23e3..6838ea9 100644
--- a/src/subreader.c
+++ b/src/subreader.c
@@ -1082,8 +1082,8 @@ subtitle* sub_fribidi (subtitle *sub, int sub_utf8)
   int char_set_num;
   fribidi_boolean log2vis;
   if(flip_hebrew) { // Please fix the indentation someday
-  fribidi_set_mirroring (FRIBIDI_TRUE);
-  fribidi_set_reorder_nsm (FRIBIDI_FALSE);
+  fribidi_set_mirroring (1);
+  fribidi_set_reorder_nsm (0);
 
   if( sub_utf8 == 0 ) {
     char_set_num = fribidi_parse_c
