--- klamd.cpp.orig	2005-09-26 17:46:30.000000000 +0000
+++ klamd.cpp	2005-09-26 18:49:07.000000000 +0000
@@ -1,6 +1,7 @@
 /*
  * Copyright (C) 2004 Robert Hogan <robert at roberthogan dot net>
  */
+#include <stdlib.h>
 
 #include "klamd.h"
 #include "klamav.h"
--- kuarantine.cpp.orig	2005-09-26 18:54:06.000000000 +0000
+++ kuarantine.cpp	2005-09-26 18:54:45.000000000 +0000
@@ -26,7 +26,7 @@
 #include <qlayout.h>
 #include <kmessagebox.h>
 #include <kstandarddirs.h>
-
+#include <stdlib.h>
 
 Kuarantine::Kuarantine(QWidget *parent, const char *name)
     : QWidget(parent, name)
