From 5770e518db21e9e1d74cd04f8d6dc26b7695656b Mon Sep 17 00:00:00 2001
From: nic <nic@boet.cc>
Date: Wed, 7 Oct 2020 15:14:42 -0500
Subject: [PATCH] implicit declaration of function crypt

---
 src/auth_passwd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/auth_passwd.c b/src/auth_passwd.c
index 7b2fe21..0af2ded 100644
--- a/src/auth_passwd.c
+++ b/src/auth_passwd.c
@@ -17,6 +17,7 @@
 #include "alock.h"
 
 #include <string.h>
+#include <crypt.h>
 #include <unistd.h>
 #include <errno.h>
 #include <grp.h>
