From b22a460def5f066227ec1486ffcf642bc1dd3315 Mon Sep 17 00:00:00 2001
From: Vlad Glagolev <scm@vaygr.net>
Date: Sat, 27 Aug 2022 23:06:05 +0000
Subject: [PATCH] Check crypt() value to avoid possible NULL ptr dereference

---
 src/auth_passwd.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/auth_passwd.c b/src/auth_passwd.c
index 0af2ded..7c4dd43 100644
--- a/src/auth_passwd.c
+++ b/src/auth_passwd.c
@@ -65,13 +65,14 @@ static int module_init(Display *display) {
 }
 
 static int module_authenticate(const char *pass) {
+    char *cpw;
 
     if (pass == NULL || pwd_entry == NULL)
         return -1;
 
     /* Simpler, and should work with crypt() algorithms using longer
      * salt strings (like the md5-based one on freebsd).  --marekm */
-    return strcmp(crypt(pass, pwd_entry->pw_passwd), pwd_entry->pw_passwd);
+    return !(cpw = crypt(pass, pwd_entry->pw_passwd)) || strcmp(pwd_entry->pw_passwd, (const char *)cpw) != 0;
 }
 
 
