From e4c3ed6d7be482c1885b189e9639efd91328d502 Mon Sep 17 00:00:00 2001
From: Vlad Glagolev <scm@vaygr.net>
Date: Fri, 26 Aug 2022 00:04:59 +0000
Subject: [PATCH] Fix pkill for older procps

---
 src/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main.c b/src/main.c
index ad2fdab..285becc 100644
--- a/src/main.c
+++ b/src/main.c
@@ -572,7 +572,7 @@ int main(int argc, char **argv) {
 
 #if WITH_DUNST
     /* pause notification daemon */
-    system("pkill -x -SIGUSR1 dunst");
+    system("pkill -SIGUSR1 -x dunst");
 #endif
 
     { /* try to initialize selected modules */
@@ -664,7 +664,7 @@ return_success:
 
 #if WITH_DUNST
     /* resume notification daemon */
-    system("pkill -x -SIGUSR2 dunst");
+    system("pkill -SIGUSR2 -x dunst");
 #endif
 
     return retval;
-- 
2.25.1

