--- tests/test.cpp.orig	2017-09-27 09:33:27.704970834 +0200
+++ tests/test.cpp	2017-09-27 09:35:50.973979582 +0200
@@ -480,7 +480,7 @@
 	jack_nframes_t delta_time = cur_time - last_time;
 
 	Log("calling process4 callback : jack_frame_time = %ld delta_time = %ld\n", cur_time, delta_time);
-	if (delta_time > 0  && (jack_nframes_t)abs(delta_time - cur_buffer_size) > tolerance) {
+	if (delta_time > 0  && (jack_nframes_t)abs((long long)(delta_time - cur_buffer_size)) > tolerance) {
 		printf("!!! ERROR !!! jack_frame_time seems to return incorrect values cur_buffer_size = %d, delta_time = %d tolerance %d\n", cur_buffer_size, delta_time, tolerance);
 	}
 
--- tests/iodelay.cpp.orig	2017-09-27 09:41:22.342999815 +0200
+++ tests/iodelay.cpp	2017-09-27 09:43:49.582008805 +0200
@@ -167,8 +167,8 @@
 static jack_port_t    *jack_capt;
 static jack_port_t    *jack_play;
 
-jack_latency_range_t   capture_latency = {-1, -1};
-jack_latency_range_t   playback_latency = {-1, -1};
+jack_latency_range_t   capture_latency = {(jack_nframes_t)-1, (jack_nframes_t)-1};
+jack_latency_range_t   playback_latency = {(jack_nframes_t)-1, (jack_nframes_t)-1};
 
 void
 latency_cb (jack_latency_callback_mode_t mode, void *arg)
@@ -266,4 +266,4 @@
     return 0;
 }
 
-// --------------------------------------------------------------------------------
\ No newline at end of file
+// --------------------------------------------------------------------------------
