--- config/Makefile.in.orig	2005-02-03 20:08:51.278437392 -0600
+++ config/Makefile.in	2005-02-03 20:09:53.304008072 -0600
@@ -586,23 +586,6 @@
 		chown root:gdm $(DESTDIR)$(authdir); \
 		chmod 1770 $(DESTDIR)$(authdir); \
 	fi
-
-	system=`uname`; \
-	if test -f /usr/include/security/pam_appl.h; then \
-	  if test '!' -d $(DESTDIR)$(PAM_PREFIX)/pam.d; then \
-		$(mkinstalldirs) $(DESTDIR)$(PAM_PREFIX)/pam.d; \
-		chmod 755 $(DESTDIR)$(PAM_PREFIX)/pam.d; \
-	   fi; \
-	   if test $$system = Linux && test '!' -f $(DESTDIR)$(PAM_PREFIX)/pam.d/gdm; then \
-		$(INSTALL_DATA) gdm $(DESTDIR)$(PAM_PREFIX)/pam.d/gdm; \
-	   fi; \
-	   if test $$system = Linux && test '!' -f $(DESTDIR)$(PAM_PREFIX)/pam.d/gdm-autologin; then \
-		$(INSTALL_DATA) gdm-autologin $(DESTDIR)$(PAM_PREFIX)/pam.d/gdm-autologin; \
-	   fi; \
-	   if test $$system = SunOS; then \
-		echo "Please add PAM authentication for gdm and gdm-autologin in $(PAM_PREFIX)/pam.conf!"; \
-	   fi; \
-	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
