diff -u -a -b -B -r gdm-2.13.0.10/config/Makefile.am gdm-mod/config/Makefile.am
--- gdm-2.13.0.10/config/Makefile.am	2006-02-13 15:09:09.000000000 -0600
+++ gdm-mod/config/Makefile.am	2006-03-05 10:58:18.000000000 -0600
@@ -12,7 +12,7 @@
 predir = $(sysconfdir)/gdm/PreSession
 postlogindir = $(sysconfdir)/gdm/PostLogin
 
-noinst_DATA = gdm.conf
+#noinst_DATA = gdm.conf
 
 DESKTOP_FILES = default.desktop CDE.desktop gnome.desktop @SSHDESKTOP@
 
diff -u -a -b -B -r gdm-2.13.0.10/config/Makefile.in gdm-mod/config/Makefile.in
--- gdm-2.13.0.10/config/Makefile.in	2006-03-01 18:54:03.000000000 -0600
+++ gdm-mod/config/Makefile.in	2006-03-05 10:57:43.000000000 -0600
@@ -309,7 +309,7 @@
 postdir = $(sysconfdir)/gdm/PostSession
 predir = $(sysconfdir)/gdm/PreSession
 postlogindir = $(sysconfdir)/gdm/PostLogin
-noinst_DATA = gdm.conf
+#noinst_DATA = gdm.conf
 DESKTOP_FILES = default.desktop CDE.desktop gnome.desktop @SSHDESKTOP@
 EXTRA_DIST = \
 	gdm.conf.in \
