--- config/Xsession.in	2005-07-19 09:53:41.000000000 +0200
+++ config/Xsession.dbus	2005-12-03 15:36:36.496457750 +0100
@@ -232,6 +232,14 @@
     echo "$0: ssh-agent not found!"
 fi
 
+# add dbus-launch if found
+dbuslaunch="`gdmwhich dbus-launch`"
+if [ -n "$dbuslaunch" ] && [ -x "$dbuslaunch" ]; then
+    command="$dbuslaunch --auto-syntax --exit-with-session $command"
+elif [ -z "$dbuslaunch" ] ; then
+    echo "$0: dbus-launch not found!"
+fi
+
 echo "$0: Setup done, will execute: $command"
 
 eval exec $command
