diff -ur calibre/src/calibre/gui2/update.py calibre-mod/src/calibre/gui2/update.py
--- calibre/src/calibre/gui2/update.py	2011-09-09 11:46:16.000000000 -0500
+++ calibre-mod/src/calibre/gui2/update.py	2011-09-17 17:52:39.505641255 -0500
@@ -20,14 +20,15 @@
 VSEP = '|'
 
 def get_newest_version():
-    br = browser()
-    req = mechanize.Request(URL)
-    req.add_header('CALIBRE_VERSION', __version__)
-    req.add_header('CALIBRE_OS',
-            'win' if iswindows else 'osx' if isosx else 'oth')
-    req.add_header('CALIBRE_INSTALL_UUID', prefs['installation_uuid'])
-    version = br.open(req).read().strip()
-    return version
+    #br = browser()
+    #req = mechanize.Request(URL)
+    #req.add_header('CALIBRE_VERSION', __version__)
+    #req.add_header('CALIBRE_OS',
+    #        'win' if iswindows else 'osx' if isosx else 'oth')
+    #req.add_header('CALIBRE_INSTALL_UUID', prefs['installation_uuid'])
+    #version = br.open(req).read().strip()
+    #return version
+    return '-0.0.0'
 
 class CheckForUpdates(QThread):
 
