diff -ur calibre.orig/src/calibre/gui2/update.py calibre/src/calibre/gui2/update.py
--- calibre.orig/src/calibre/gui2/update.py	2012-09-20 16:27:22.398340171 -0500
+++ calibre/src/calibre/gui2/update.py	2012-09-20 16:27:58.594339911 -0500
@@ -20,18 +20,6 @@
 VSEP = '|'
 
 def get_newest_version():
-    br = browser()
-    req = mechanize.Request(URL)
-    req.add_header('CALIBRE_VERSION', __version__)
-    req.add_header('CALIBRE_OS',
-            'win' if iswindows else 'osx' if isosx else 'oth')
-    req.add_header('CALIBRE_INSTALL_UUID', prefs['installation_uuid'])
-    version = br.open(req).read().strip()
-    try:
-        version = version.decode('utf-8')
-    except UnicodeDecodeError:
-        version = u''
-    return version
 
 class CheckForUpdates(QThread):
 
