--- setup.py.orig	2009-03-14 08:21:11.000000000 +0100
+++ setup.py	2009-03-14 08:21:50.000000000 +0100
@@ -177,8 +177,3 @@
                       },
          )
 
-    if 'develop' in ' '.join(sys.argv) and islinux:
-        subprocess.check_call('calibre_postinstall --do-not-reload-udev-hal', shell=True)
-    if 'install' in sys.argv and islinux:
-        subprocess.check_call('calibre_postinstall', shell=True)
-
