Author: Joachim Reichel <reichel@debian.org>
Description: fix typo w.r.t. flac support
--- a/src/normalize-mp3.in
+++ b/src/normalize-mp3.in
@@ -295,7 +295,7 @@
     $path = find_prog("flac");
     if ($path) {
 	$path .= " -s -d -o %w %m";
-	$FLAC = $path;
+	$FLACDECODE = $path;
     }
 }
 
@@ -304,7 +304,7 @@
     $path = find_prog("flac");
     if ($path) {
 	$path .= " -s -o %m %w";
-	$FLAC = $path;
+	$FLACENCODE = $path;
     }
 }
 
