--- normalize-0.7.7/src/normalize-mp3.in.orig	2025-05-31 16:55:10.567869537 +0200
+++ normalize-0.7.7/src/normalize-mp3.in	2025-05-31 16:56:48.309869525 +0200
@@ -241,11 +241,11 @@
 
 sub find_mp3decode {
     my ($path);
-    $path = find_prog("madplay");
-    if ($path) { $path .= " -q -o %w %m"; }
+    $path = find_prog("mpg123");
+    if ($path) { $path .= " -q -w %w %m"; }
     unless ($path) {
-	$path = find_prog("mpg123");
-	if ($path) { $path .= " -q -w %w %m"; }
+        $path = find_prog("madplay");
+        if ($path) { $path .= " -q -o %w %m"; }
     }
     if ($path) { $MP3DECODE = $path; }
 }
