Index: normalize-audio/nid3lib/write.c
===================================================================
--- normalize-audio.orig/nid3lib/write.c
+++ normalize-audio/nid3lib/write.c
@@ -20,8 +20,6 @@
  * id3_write(), and many helper functions.
  */
 
-#define _POSIX_C_SOURCE 2
-
 #include "config.h"
 
 #include <stdio.h>
@@ -59,14 +57,6 @@
 #if HAVE_INTTYPES_H
 # include <inttypes.h>
 #endif
-#if HAVE_FTRUNCATE
-/*
- * ftruncate() shouldn't be declared yet, since we turn on
- * _POSIX_C_SOURCE, but you never know, so we use an old style
- * declaration to lessen the chance of conflict.
- */
-int ftruncate();
-#endif
 
 #include "nid3P.h"
 
