diff -aur monodevelop-0.9-orig/Core/src/MonoDevelop.Projects/MonoDevelop.Projects/Combine.cs monodevelop-0.9/Core/src/MonoDevelop.Projects/MonoDevelop.Projects/Combine.cs
--- monodevelop-0.9-orig/Core/src/MonoDevelop.Projects/MonoDevelop.Projects/Combine.cs	2005-11-04 04:01:53.000000000 -0800
+++ monodevelop-0.9/Core/src/MonoDevelop.Projects/MonoDevelop.Projects/Combine.cs	2006-01-12 23:17:58.000000000 -0800
@@ -17,7 +17,7 @@
 using System.Threading;
 
 using Mono.Unix.Native;
-using FileMode = Mono.Unix.FilePermissions;
+using FileMode = Mono.Unix.Native.FilePermissions;
 
 using MonoDevelop.Core;
 using MonoDevelop.Core.Properties;
