diff --git a/mythtv/libs/libmyth/libmyth.pro b/mythtv/libs/libmyth/libmyth.pro
index 24ec735b18..1fd5cc01c0 100644
--- a/mythtv/libs/libmyth/libmyth.pro
+++ b/mythtv/libs/libmyth/libmyth.pro
@@ -1,10 +1,8 @@
 include ( ../../settings.pro )
 
 QT += network xml sql script
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
 android: QT += androidextras
-}
 
 TEMPLATE = lib
 TARGET = myth-$$LIBVERSION
@@ -177,11 +175,7 @@ unix:!cygwin {
     SOURCES += mediamonitor-unix.cpp
     HEADERS += mediamonitor-unix.h
     !android {
-    contains(QT_VERSION, ^5\\.[0-9]\\..*) {
         using_qtdbus: QT += dbus
-    } else {
-        using_qtdbus: CONFIG += qdbus
-    }
     }
 }
 
diff --git a/mythtv/libs/libmyth/test/test_audioconvert/test_audioconvert.pro b/mythtv/libs/libmyth/test/test_audioconvert/test_audioconvert.pro
index 9e73a8ca23..d0d77e8027 100644
--- a/mythtv/libs/libmyth/test/test_audioconvert/test_audioconvert.pro
+++ b/mythtv/libs/libmyth/test/test_audioconvert/test_audioconvert.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_audioconvert
diff --git a/mythtv/libs/libmyth/test/test_audioutils/test_audioutils.pro b/mythtv/libs/libmyth/test/test_audioutils/test_audioutils.pro
index 3d62f741cb..df659aeb19 100644
--- a/mythtv/libs/libmyth/test/test_audioutils/test_audioutils.pro
+++ b/mythtv/libs/libmyth/test/test_audioutils/test_audioutils.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_audioutils
diff --git a/mythtv/libs/libmyth/test/test_programinfo/test_programinfo.pro b/mythtv/libs/libmyth/test/test_programinfo/test_programinfo.pro
index bfd9d24287..e59584844d 100644
--- a/mythtv/libs/libmyth/test/test_programinfo/test_programinfo.pro
+++ b/mythtv/libs/libmyth/test/test_programinfo/test_programinfo.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_programinfo
diff --git a/mythtv/libs/libmythbase/libmythbase.pro b/mythtv/libs/libmythbase/libmythbase.pro
index aa7ac151db..444abea063 100644
--- a/mythtv/libs/libmythbase/libmythbase.pro
+++ b/mythtv/libs/libmythbase/libmythbase.pro
@@ -153,10 +153,7 @@ win32-msvc* {
 }
 
 QT += xml sql network
-
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 include ( ../libs-targetfix.pro )
 
diff --git a/mythtv/libs/libmythbase/test/test_mythsystem/test_mythsystem.pro b/mythtv/libs/libmythbase/test/test_mythsystem/test_mythsystem.pro
index fee4e38226..159dcb28f3 100644
--- a/mythtv/libs/libmythbase/test/test_mythsystem/test_mythsystem.pro
+++ b/mythtv/libs/libmythbase/test/test_mythsystem/test_mythsystem.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_mythsystem
diff --git a/mythtv/libs/libmythbase/test/test_mythsystemlegacy/test_mythsystemlegacy.pro b/mythtv/libs/libmythbase/test/test_mythsystemlegacy/test_mythsystemlegacy.pro
index b116e7a5ae..d97c6c8112 100644
--- a/mythtv/libs/libmythbase/test/test_mythsystemlegacy/test_mythsystemlegacy.pro
+++ b/mythtv/libs/libmythbase/test/test_mythsystemlegacy/test_mythsystemlegacy.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_mythsystemlegacy
diff --git a/mythtv/libs/libmythbase/test/test_mythtimer/test_mythtimer.pro b/mythtv/libs/libmythbase/test/test_mythtimer/test_mythtimer.pro
index d0e332b74d..91c6a3f773 100644
--- a/mythtv/libs/libmythbase/test/test_mythtimer/test_mythtimer.pro
+++ b/mythtv/libs/libmythbase/test/test_mythtimer/test_mythtimer.pro
@@ -1,11 +1,6 @@
 include ( ../../../../settings.pro )
 
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_mythtimer
diff --git a/mythtv/libs/libmythbase/test/test_template/test_template.pro b/mythtv/libs/libmythbase/test/test_template/test_template.pro
index 2cf0a3e615..c8229039b7 100644
--- a/mythtv/libs/libmythbase/test/test_template/test_template.pro
+++ b/mythtv/libs/libmythbase/test/test_template/test_template.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_template
diff --git a/mythtv/libs/libmythmetadata/libmythmetadata.pro b/mythtv/libs/libmythmetadata/libmythmetadata.pro
index d06405c930..695900c985 100644
--- a/mythtv/libs/libmythmetadata/libmythmetadata.pro
+++ b/mythtv/libs/libmythmetadata/libmythmetadata.pro
@@ -1,9 +1,7 @@
 include ( ../../settings.pro )
 
 QT += network xml sql
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = lib
 TARGET = mythmetadata-$$LIBVERSION
diff --git a/mythtv/libs/libmythmetadata/test/test_videometadata/test_videometadata.pro b/mythtv/libs/libmythmetadata/test/test_videometadata/test_videometadata.pro
index 218378d363..cc283b73c0 100644
--- a/mythtv/libs/libmythmetadata/test/test_videometadata/test_videometadata.pro
+++ b/mythtv/libs/libmythmetadata/test/test_videometadata/test_videometadata.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_videometadata
diff --git a/mythtv/libs/libmythprotoserver/libmythprotoserver.pro b/mythtv/libs/libmythprotoserver/libmythprotoserver.pro
index d9c5643ec8..a1d2446cc1 100644
--- a/mythtv/libs/libmythprotoserver/libmythprotoserver.pro
+++ b/mythtv/libs/libmythprotoserver/libmythprotoserver.pro
@@ -1,9 +1,7 @@
 include ( ../../settings.pro )
 
 QT += network sql
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = lib
 TARGET = mythprotoserver-$$LIBVERSION
diff --git a/mythtv/libs/libmythtv/libmythtv.pro b/mythtv/libs/libmythtv/libmythtv.pro
index 77dd7917a0..b88964a3ea 100644
--- a/mythtv/libs/libmythtv/libmythtv.pro
+++ b/mythtv/libs/libmythtv/libmythtv.pro
@@ -1,9 +1,7 @@
 include ( ../../settings.pro )
 
 QT += network xml sql
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = lib
 TARGET = mythtv-$$LIBVERSION
diff --git a/mythtv/libs/libmythtv/test/test_copyframes/test_copyframes.pro b/mythtv/libs/libmythtv/test/test_copyframes/test_copyframes.pro
index b365d57cd3..15a39d0986 100644
--- a/mythtv/libs/libmythtv/test/test_copyframes/test_copyframes.pro
+++ b/mythtv/libs/libmythtv/test/test_copyframes/test_copyframes.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_copyframes
diff --git a/mythtv/libs/libmythtv/test/test_eitfixups/test_eitfixups.pro b/mythtv/libs/libmythtv/test/test_eitfixups/test_eitfixups.pro
index 6e8881b4eb..a6be44cc62 100644
--- a/mythtv/libs/libmythtv/test/test_eitfixups/test_eitfixups.pro
+++ b/mythtv/libs/libmythtv/test/test_eitfixups/test_eitfixups.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_eitfixups
diff --git a/mythtv/libs/libmythtv/test/test_iptvrecorder/test_iptvrecorder.pro b/mythtv/libs/libmythtv/test/test_iptvrecorder/test_iptvrecorder.pro
index c051d80319..bbc9e89a46 100644
--- a/mythtv/libs/libmythtv/test/test_iptvrecorder/test_iptvrecorder.pro
+++ b/mythtv/libs/libmythtv/test/test_iptvrecorder/test_iptvrecorder.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_iptvrecorder
diff --git a/mythtv/libs/libmythtv/test/test_mpegtables/test_mpegtables.pro b/mythtv/libs/libmythtv/test/test_mpegtables/test_mpegtables.pro
index 62f3124bb5..b280b0eaa7 100644
--- a/mythtv/libs/libmythtv/test/test_mpegtables/test_mpegtables.pro
+++ b/mythtv/libs/libmythtv/test/test_mpegtables/test_mpegtables.pro
@@ -1,13 +1,7 @@
 include ( ../../../../settings.pro )
 
 QT += xml sql network
-
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-CONFIG += qtestlib
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += testlib
-}
 
 TEMPLATE = app
 TARGET = test_mpegtables
diff --git a/mythtv/libs/libmythui/libmythui.pro b/mythtv/libs/libmythui/libmythui.pro
index 6b534ad2c2..092098dd4a 100644
--- a/mythtv/libs/libmythui/libmythui.pro
+++ b/mythtv/libs/libmythui/libmythui.pro
@@ -1,10 +1,7 @@
 include ( ../../settings.pro )
 
 QT += xml sql network opengl
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-QT += webkit
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*):using_qtwebkit {
+using_qtwebkit {
 QT += widgets
 QT += webkitwidgets
 android: QT += androidextras
