diff --git a/mythplugins/mythbrowser/mythbrowser/mythbrowser.pro b/mythplugins/mythbrowser/mythbrowser/mythbrowser.pro
index 693e621b08..c7a39b5545 100644
--- a/mythplugins/mythbrowser/mythbrowser/mythbrowser.pro
+++ b/mythplugins/mythbrowser/mythbrowser/mythbrowser.pro
@@ -5,8 +5,7 @@ include (../../programs-libs.pro )
 LIBS += -lmythtv-$$LIBVERSION
 
 QT += network xml sql opengl
-contains(QT_VERSION, ^4\\.[0-9]\\..*) : QT += webkit
-contains(QT_VERSION, ^5\\.[0-9]\\..*) : QT += widgets webkitwidgets
+QT += widgets webkitwidgets
 
 TEMPLATE = lib
 CONFIG += thread opengl plugin warn_on
diff --git a/mythplugins/mythmusic/mythmusic/mythmusic.pro b/mythplugins/mythmusic/mythmusic/mythmusic.pro
index 0703da590f..88b35c3782 100644
--- a/mythplugins/mythmusic/mythmusic/mythmusic.pro
+++ b/mythplugins/mythmusic/mythmusic/mythmusic.pro
@@ -4,8 +4,7 @@ include ( ../../programs-libs.pro )
 include (config.pro)
 
 QT += xml sql opengl network
-contains(QT_VERSION, ^4\\.[0-9]\\..*) : QT += webkit
-contains(QT_VERSION, ^5\\.[0-9]\\..*) : QT += widgets webkitwidgets
+QT += widgets webkitwidgets
 
 !exists( config.pro ) {
    error(Missing config.pro: please run the configure script)
diff --git a/mythplugins/mythnetvision/mythnetvision/mythnetvision.pro b/mythplugins/mythnetvision/mythnetvision/mythnetvision.pro
index 2bf47ec5ee..a2f343ecef 100644
--- a/mythplugins/mythnetvision/mythnetvision/mythnetvision.pro
+++ b/mythplugins/mythnetvision/mythnetvision/mythnetvision.pro
@@ -3,9 +3,7 @@ include ( ../../settings.pro )
 include ( ../../programs-libs.pro )
 
 QT += network sql xml 
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = lib
 CONFIG += plugin thread warn_on debug
diff --git a/mythplugins/mythnews/mythnews/mythnews.pro b/mythplugins/mythnews/mythnews/mythnews.pro
index d2bff02684..b757dffe07 100644
--- a/mythplugins/mythnews/mythnews/mythnews.pro
+++ b/mythplugins/mythnews/mythnews/mythnews.pro
@@ -3,9 +3,7 @@ include ( ../../settings.pro )
 include ( ../../programs-libs.pro )
 
 QT += network sql xml
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = lib
 CONFIG += plugin thread warn_on debug
diff --git a/mythplugins/mythweather/mythweather/mythweather.pro b/mythplugins/mythweather/mythweather/mythweather.pro
index ab0654b92d..3fb32a5c7c 100644
--- a/mythplugins/mythweather/mythweather/mythweather.pro
+++ b/mythplugins/mythweather/mythweather/mythweather.pro
@@ -3,9 +3,7 @@ include ( ../../settings.pro )
 include ( ../../programs-libs.pro )
 
 QT += sql xml network
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = lib
 CONFIG += plugin thread 
diff --git a/mythplugins/mythzoneminder/mythzoneminder/mythzoneminder.pro b/mythplugins/mythzoneminder/mythzoneminder/mythzoneminder.pro
index e9b418c3d7..803be487ea 100644
--- a/mythplugins/mythzoneminder/mythzoneminder/mythzoneminder.pro
+++ b/mythplugins/mythzoneminder/mythzoneminder/mythzoneminder.pro
@@ -3,9 +3,7 @@ include ( ../../settings.pro )
 include ( ../../programs-libs.pro )
 
 QT += network
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = lib
 CONFIG += plugin thread
