diff --git a/mythtv/programs/mythavtest/mythavtest.pro b/mythtv/programs/mythavtest/mythavtest.pro
index b993647179..35ed274fd9 100644
--- a/mythtv/programs/mythavtest/mythavtest.pro
+++ b/mythtv/programs/mythavtest/mythavtest.pro
@@ -3,9 +3,7 @@ include ( ../../version.pro )
 include ( ../programs-libs.pro )
 
 QT += network xml sql
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythbackend/mythbackend.pro b/mythtv/programs/mythbackend/mythbackend.pro
index ab59bcbbe1..5e54d3bfeb 100644
--- a/mythtv/programs/mythbackend/mythbackend.pro
+++ b/mythtv/programs/mythbackend/mythbackend.pro
@@ -10,9 +10,7 @@ mingw | win32-msvc* {
    # need the following for QUrl::addQueryItem
    DEFINES += QT_DISABLE_DEPRECATED_BEFORE
 }
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythccextractor/mythccextractor.pro b/mythtv/programs/mythccextractor/mythccextractor.pro
index fbe8c2e50a..a0e70a9ef7 100644
--- a/mythtv/programs/mythccextractor/mythccextractor.pro
+++ b/mythtv/programs/mythccextractor/mythccextractor.pro
@@ -2,9 +2,7 @@ include (../../settings.pro)
 include ( ../programs-libs.pro )
 
 QT += sql network
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythcommflag/mythcommflag.pro b/mythtv/programs/mythcommflag/mythcommflag.pro
index 997966e295..a3fde4f4bd 100644
--- a/mythtv/programs/mythcommflag/mythcommflag.pro
+++ b/mythtv/programs/mythcommflag/mythcommflag.pro
@@ -2,9 +2,7 @@ include (../../settings.pro)
 include (../../version.pro)
 include ( ../programs-libs.pro )
 
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythfilerecorder/mythfilerecorder.pro b/mythtv/programs/mythfilerecorder/mythfilerecorder.pro
index 73300742f1..ba568e091a 100644
--- a/mythtv/programs/mythfilerecorder/mythfilerecorder.pro
+++ b/mythtv/programs/mythfilerecorder/mythfilerecorder.pro
@@ -3,9 +3,7 @@ include (../../version.pro)
 include ( ../programs-libs.pro )
 
 QT += network xml sql script
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythfilldatabase/mythfilldatabase.pro b/mythtv/programs/mythfilldatabase/mythfilldatabase.pro
index 52e6e0a256..0f4f944cf7 100644
--- a/mythtv/programs/mythfilldatabase/mythfilldatabase.pro
+++ b/mythtv/programs/mythfilldatabase/mythfilldatabase.pro
@@ -3,9 +3,7 @@ include ( ../../version.pro )
 include ( ../programs-libs.pro )
 
 QT += network xml sql
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythfrontend/mythfrontend.pro b/mythtv/programs/mythfrontend/mythfrontend.pro
index 744fe1f2bb..2e5f313243 100644
--- a/mythtv/programs/mythfrontend/mythfrontend.pro
+++ b/mythtv/programs/mythfrontend/mythfrontend.pro
@@ -7,11 +7,7 @@ mingw | win32-msvc* {
    # script debugger currently only enabled for WIN32 builds
    QT += scripttools
 }
-contains(QT_VERSION, ^4\\.[0-9]\\..*) {
-QT += webkit
-using_qtdbus: CONFIG += qdbus
-}
-contains(QT_VERSION, ^5\\.[0-9]\\..*):using_qtwebkit {
+using_qtwebkit {
 QT += widgets
 QT += webkitwidgets
 using_qtdbus: QT += dbus
diff --git a/mythtv/programs/mythjobqueue/mythjobqueue.pro b/mythtv/programs/mythjobqueue/mythjobqueue.pro
index 9405ab597f..389eec7843 100644
--- a/mythtv/programs/mythjobqueue/mythjobqueue.pro
+++ b/mythtv/programs/mythjobqueue/mythjobqueue.pro
@@ -3,9 +3,7 @@ include ( ../../version.pro)
 include ( ../programs-libs.pro)
 
 QT += sql network
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 CONFIG -= opengl
 
diff --git a/mythtv/programs/mythlcdserver/mythlcdserver.pro b/mythtv/programs/mythlcdserver/mythlcdserver.pro
index fd0a98bf91..766715ee61 100644
--- a/mythtv/programs/mythlcdserver/mythlcdserver.pro
+++ b/mythtv/programs/mythlcdserver/mythlcdserver.pro
@@ -3,9 +3,8 @@ include ( ../../version.pro )
 include ( ../programs-libs.pro )
 
 QT += network xml sql
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
+
 using_opengl:QT += opengl
 
 TEMPLATE = app
diff --git a/mythtv/programs/mythlogserver/mythlogserver.pro b/mythtv/programs/mythlogserver/mythlogserver.pro
index fbe8c2e50a..a0e70a9ef7 100644
--- a/mythtv/programs/mythlogserver/mythlogserver.pro
+++ b/mythtv/programs/mythlogserver/mythlogserver.pro
@@ -2,9 +2,7 @@ include (../../settings.pro)
 include ( ../programs-libs.pro )
 
 QT += sql network
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythmediaserver/mythmediaserver.pro b/mythtv/programs/mythmediaserver/mythmediaserver.pro
index 2f9a804939..e042941fba 100644
--- a/mythtv/programs/mythmediaserver/mythmediaserver.pro
+++ b/mythtv/programs/mythmediaserver/mythmediaserver.pro
@@ -3,9 +3,7 @@ include ( ../../version.pro )
 include ( ../programs-libs.pro )
 
 QT += network sql
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythpreviewgen/mythpreviewgen.pro b/mythtv/programs/mythpreviewgen/mythpreviewgen.pro
index d21797ede9..3d085385ba 100644
--- a/mythtv/programs/mythpreviewgen/mythpreviewgen.pro
+++ b/mythtv/programs/mythpreviewgen/mythpreviewgen.pro
@@ -3,9 +3,7 @@ include ( ../../version.pro )
 include ( ../programs-libs.pro )
 
 QT += network sql
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythscreenwizard/mythscreenwizard.pro b/mythtv/programs/mythscreenwizard/mythscreenwizard.pro
index 42c4669620..5a4d08d417 100644
--- a/mythtv/programs/mythscreenwizard/mythscreenwizard.pro
+++ b/mythtv/programs/mythscreenwizard/mythscreenwizard.pro
@@ -3,9 +3,7 @@ include ( ../../version.pro)
 include ( ../programs-libs.pro)
 
 QT += sql network
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythtranscode/mythtranscode.pro b/mythtv/programs/mythtranscode/mythtranscode.pro
index 4d4c3bcbcb..1f66fade2b 100644
--- a/mythtv/programs/mythtranscode/mythtranscode.pro
+++ b/mythtv/programs/mythtranscode/mythtranscode.pro
@@ -3,9 +3,7 @@ include ( ../../version.pro)
 include ( ../programs-libs.pro)
 
 QT += network xml sql
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythtv-setup/mythtv-setup.pro b/mythtv/programs/mythtv-setup/mythtv-setup.pro
index 24a9255fb7..9ef3f1c61a 100644
--- a/mythtv/programs/mythtv-setup/mythtv-setup.pro
+++ b/mythtv/programs/mythtv-setup/mythtv-setup.pro
@@ -3,9 +3,7 @@ include ( ../../version.pro )
 include ( ../programs-libs.pro )
 
 QT += network xml sql
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 INCLUDEPATH += ../../libs/libmythtv/channelscan
 DEPENDPATH += ../../libs/libmythtv/channelscan
diff --git a/mythtv/programs/mythutil/mythutil.pro b/mythtv/programs/mythutil/mythutil.pro
index ab8897f17c..0ad14ef856 100644
--- a/mythtv/programs/mythutil/mythutil.pro
+++ b/mythtv/programs/mythutil/mythutil.pro
@@ -3,9 +3,7 @@ include ( ../../version.pro )
 include ( ../programs-libs.pro )
 
 QT += network sql xml
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
diff --git a/mythtv/programs/mythwelcome/mythwelcome.pro b/mythtv/programs/mythwelcome/mythwelcome.pro
index edddbf18a6..fa83a39852 100644
--- a/mythtv/programs/mythwelcome/mythwelcome.pro
+++ b/mythtv/programs/mythwelcome/mythwelcome.pro
@@ -3,9 +3,7 @@ include (../../version.pro)
 include (../programs-libs.pro)
 
 QT += xml sql network
-contains(QT_VERSION, ^5\\.[0-9]\\..*) {
 QT += widgets
-}
 
 TEMPLATE = app
 CONFIG += thread
