From d1d97e7d0b3024816934fd0a64efe3d1c567fff6 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 21 Mar 2014 01:28:06 +0100
Subject: [PATCH] clean up Makefile

---
 Makefile | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 46c1331..f9c3110 100644
--- a/Makefile
+++ b/Makefile
@@ -1,10 +1,9 @@
 # go on and adjust here if you don't like those flags
-CFLAGS=-Os -fomit-frame-pointer -s -pipe
+CFLAGS ?= -Os -fomit-frame-pointer -s -pipe
 #CFLAGS=-Wall -Os -fomit-frame-pointer -s -pipe -DDEBUG
-CC=gcc
 # likewise, if you want to change the destination prefix
 DESTDIR=
-DESTPREFIX=/usr/local
+DESTPREFIX ?= /usr/local
 MANDIR=$(DESTPREFIX)/share/man/man8
 GZIP=gzip -9
 TARGET=schedtool
@@ -19,7 +18,7 @@ clean:
 distclean: clean unzipman
 	rm -f *~ *.s
 
-install: all install-doc zipman
+install: all zipman
 	install -d $(DESTDIR)$(DESTPREFIX)/bin
 	install -p -c $(TARGET) $(DESTDIR)$(DESTPREFIX)/bin
 	install -d $(DESTDIR)$(MANDIR)
-- 
1.9.0

