--- modules/codec/ffmpeg/mux.c.orig	2007-11-21 21:47:16.000000000 +0100
+++ modules/codec/ffmpeg/mux.c	2007-11-21 21:49:41.000000000 +0100
@@ -133,7 +133,11 @@
         return VLC_EGENERIC;
     }
 
+#if (LIBAVFORMAT_BUILD >= ((52<<16)+(0<<8)+0))
+    p_sys->oc->pb = &p_sys->io;
+#else
     p_sys->oc->pb = p_sys->io;
+#endif
     p_sys->oc->nb_streams = 0;
 
     p_sys->b_write_header = VLC_TRUE;
