--- mozilla/Makefile.in.orig	2005-05-27 15:23:14.000000000 +0200
+++ mozilla/Makefile.in	2005-05-27 15:23:52.000000000 +0200
@@ -289,10 +289,10 @@
 @BUILD_MOZILLA_TRUE@BUILT_SOURCES_mozilla = vlcintf.h
 
 @BUILD_MOZILLA_TRUE@plugin_DATA = $(LIBRARIES_mozilla) $(BUNDLE_mozilla)
-@BUILD_MOZILLA_TRUE@plugindir = $(libdir)/mozilla/plugins
+@BUILD_MOZILLA_TRUE@plugindir = `$(MOZILLA_CONFIG) --defines | sed 's/.*MOZ_DEFAULT_MOZILLA_FIVE_HOME="\([^"]*\)".*/\1/'`/plugins
 
 @BUILD_MOZILLA_TRUE@vlcintf_xpt_DATA = vlcintf.xpt
-@BUILD_MOZILLA_TRUE@vlcintf_xptdir = $(libdir)/mozilla/components
+@BUILD_MOZILLA_TRUE@vlcintf_xptdir = `$(MOZILLA_CONFIG) --defines | sed 's/.*MOZ_DEFAULT_MOZILLA_FIVE_HOME="\([^"]*\)".*/\1/'`/components
 
 @BUILD_MOZILLA_TRUE@@HAVE_WIN32_TRUE@DATA_npvlc_rc = $(noinst_npvlc_rc_DATA)
 @BUILD_MOZILLA_TRUE@@HAVE_WIN32_TRUE@noinst_npvlc_rc_DATA = npvlc_rc.$(OBJEXT)
