--- modules/codec/faad.c~	2007-02-15 19:50:20.000000000 +0100
+++ modules/codec/faad.c	2007-02-15 19:50:23.000000000 +0100
@@ -316,6 +316,7 @@
         p_dec->fmt_out.audio.i_channels = frame.channels;
 
         /* Adjust stream info when dealing with SBR/PS */
+#ifdef NEAACDECAPI
         if( (p_sys->b_sbr != frame.sbr || p_sys->b_ps != frame.ps) &&
             p_dec->p_parent->i_object_type == VLC_OBJECT_INPUT )
         {
@@ -336,6 +337,7 @@
           free( psz_cat );
           p_sys->b_sbr = frame.sbr; p_sys->b_ps = frame.ps;
         }
+#endif
 
         /* Convert frame.channel_position to our own channel values */
         for( i = 0; i < frame.channels; i++ )
