diff -ur vlc-0.8.6e/configure vlc-0.8.6e-ffmpegfix/configure
--- vlc-0.8.6e/configure	2008-02-24 20:07:09.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/configure	2008-04-01 20:00:19.000000000 +0200
@@ -44443,7 +44443,7 @@
   if test "${FFMPEG_CONFIG}" != "no"
   then
 
-for ac_header in ffmpeg/avcodec.h
+for ac_header in libavcodec/avcodec.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -44579,15 +44579,15 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file ffmpeg/avcodec.h." >&5
-echo "$as_me: error: Missing header file ffmpeg/avcodec.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libavcodec/avcodec.h." >&5
+echo "$as_me: error: Missing header file libavcodec/avcodec.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
 done
 
 
-for ac_header in ffmpeg/avformat.h
+for ac_header in libavformat/avformat.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -44727,7 +44727,7 @@
 done
 
 
-for ac_header in ffmpeg/avutil.h
+for ac_header in libavutil/avutil.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -44867,7 +44867,7 @@
 done
 
 
-for ac_header in postproc/postprocess.h
+for ac_header in libpostproc/postprocess.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -45003,8 +45003,8 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file postproc/postprocess.h." >&5
-echo "$as_me: error: Missing header file postproc/postprocess.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libpostproc/postprocess.h." >&5
+echo "$as_me: error: Missing header file libpostproc/postprocess.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
@@ -45099,7 +45099,7 @@
                   CPPFLAGS="${CPPFLAGS_save} ${CPPFLAGS_ffmpeg}"
       LDFLAGS="${LDFLAGS_save} ${LDFLAGS_ffmpeg}"
 
-for ac_header in ffmpeg/avcodec.h
+for ac_header in libavcodec/avcodec.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -45235,15 +45235,15 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file ffmpeg/avcodec.h." >&5
-echo "$as_me: error: Missing header file ffmpeg/avcodec.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libavcodec/avcodec.h." >&5
+echo "$as_me: error: Missing header file libavcodec/avcodec.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
 done
 
 
-for ac_header in ffmpeg/avformat.h
+for ac_header in libavformat/avformat.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -45383,7 +45383,7 @@
 done
 
 
-for ac_header in ffmpeg/avutil.h
+for ac_header in libavutil/avutil.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -45523,7 +45523,7 @@
 done
 
 
-for ac_header in postproc/postprocess.h
+for ac_header in libpostproc/postprocess.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -45659,8 +45659,8 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file postproc/postprocess.h." >&5
-echo "$as_me: error: Missing header file postproc/postprocess.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libpostproc/postprocess.h." >&5
+echo "$as_me: error: Missing header file libpostproc/postprocess.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
@@ -45981,7 +45981,7 @@
                   CPPFLAGS="${CPPFLAGS_save} ${CPPFLAGS_ffmpeg}"
       LDFLAGS="${LDFLAGS_save} ${LDFLAGS_ffmpeg}"
 
-for ac_header in ffmpeg/avcodec.h
+for ac_header in libavcodec/avcodec.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -46117,15 +46117,15 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file ffmpeg/avcodec.h." >&5
-echo "$as_me: error: Missing header file ffmpeg/avcodec.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libavcodec/avcodec.h." >&5
+echo "$as_me: error: Missing header file libavcodec/avcodec.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
 done
 
 
-for ac_header in ffmpeg/avformat.h
+for ac_header in libavformat/avformat.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -46265,7 +46265,7 @@
 done
 
 
-for ac_header in ffmpeg/avutil.h
+for ac_header in libavutil/avutil.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -46405,7 +46405,7 @@
 done
 
 
-for ac_header in postproc/postprocess.h
+for ac_header in libpostproc/postprocess.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -46541,8 +46541,8 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file postproc/postprocess.h." >&5
-echo "$as_me: error: Missing header file postproc/postprocess.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libpostproc/postprocess.h." >&5
+echo "$as_me: error: Missing header file libpostproc/postprocess.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
@@ -46864,7 +46864,7 @@
 echo "${ECHO_T}yes" >&6; }
 
 
-for ac_header in ffmpeg/avcodec.h
+for ac_header in libavcodec/avcodec.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -47000,15 +47000,15 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file ffmpeg/avcodec.h." >&5
-echo "$as_me: error: Missing header file ffmpeg/avcodec.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libavcodec/avcodec.h." >&5
+echo "$as_me: error: Missing header file libavcodec/avcodec.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
 done
 
 
-for ac_header in ffmpeg/avformat.h
+for ac_header in libavformat/avformat.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -47148,7 +47148,7 @@
 done
 
 
-for ac_header in ffmpeg/avutil.h
+for ac_header in libavutil/avutil.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -47288,7 +47288,7 @@
 done
 
 
-for ac_header in postproc/postprocess.h
+for ac_header in libpostproc/postprocess.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
@@ -47424,8 +47424,8 @@
 _ACEOF
 
 else
-  { { echo "$as_me:$LINENO: error: Missing header file postproc/postprocess.h." >&5
-echo "$as_me: error: Missing header file postproc/postprocess.h." >&2;}
+  { { echo "$as_me:$LINENO: error: Missing header file libpostproc/postprocess.h." >&5
+echo "$as_me: error: Missing header file libpostproc/postprocess.h." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
diff -ur vlc-0.8.6e/modules/codec/ffmpeg/audio.c vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/audio.c
--- vlc-0.8.6e/modules/codec/ffmpeg/audio.c	2007-11-21 23:23:29.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/audio.c	2008-04-01 20:13:27.000000000 +0200
@@ -30,9 +30,9 @@
 
 /* ffmpeg header */
 #ifdef HAVE_FFMPEG_AVCODEC_H
-#   include <ffmpeg/avcodec.h>
+#   include <ffmpeg/libavcodec/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #include "ffmpeg.h"
diff -ur vlc-0.8.6e/modules/codec/ffmpeg/chroma.c vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/chroma.c
--- vlc-0.8.6e/modules/codec/ffmpeg/chroma.c	2007-11-21 23:23:29.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/chroma.c	2008-04-01 20:13:27.000000000 +0200
@@ -29,9 +29,9 @@
 
 /* ffmpeg header */
 #ifdef HAVE_FFMPEG_AVCODEC_H
-#   include <ffmpeg/avcodec.h>
+#   include <ffmpeg/libavcodec/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #include "ffmpeg.h"
diff -ur vlc-0.8.6e/modules/codec/ffmpeg/demux.c vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/demux.c
--- vlc-0.8.6e/modules/codec/ffmpeg/demux.c	2007-11-21 23:23:29.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/demux.c	2008-04-01 20:15:17.000000000 +0200
@@ -35,7 +35,7 @@
 #ifdef HAVE_FFMPEG_AVFORMAT_H
 #   include <ffmpeg/avformat.h>
 #else
-#   include <avformat.h>
+#   include <libavformat/avformat.h>
 #endif
 
 #include "ffmpeg.h"
diff -ur vlc-0.8.6e/modules/codec/ffmpeg/encoder.c vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/encoder.c
--- vlc-0.8.6e/modules/codec/ffmpeg/encoder.c	2007-11-21 23:23:29.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/encoder.c	2008-04-01 20:13:27.000000000 +0200
@@ -37,9 +37,9 @@
 /* ffmpeg header */
 #define HAVE_MMX 1
 #ifdef HAVE_FFMPEG_AVCODEC_H
-#   include <ffmpeg/avcodec.h>
+#   include <ffmpeg/libavcodec/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #if LIBAVCODEC_BUILD < 4704
diff -ur vlc-0.8.6e/modules/codec/ffmpeg/ffmpeg.c vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/ffmpeg.c
--- vlc-0.8.6e/modules/codec/ffmpeg/ffmpeg.c	2007-11-21 23:23:29.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/ffmpeg.c	2008-04-01 20:13:27.000000000 +0200
@@ -31,9 +31,9 @@
 /* ffmpeg header */
 #define HAVE_MMX 1
 #ifdef HAVE_FFMPEG_AVCODEC_H
-#   include <ffmpeg/avcodec.h>
+#   include <ffmpeg/libavcodec/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #if LIBAVCODEC_BUILD < 4680
diff -ur vlc-0.8.6e/modules/codec/ffmpeg/mux.c vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/mux.c
--- vlc-0.8.6e/modules/codec/ffmpeg/mux.c	2007-12-29 16:25:53.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/mux.c	2008-04-01 20:15:45.000000000 +0200
@@ -34,7 +34,7 @@
 #ifdef HAVE_FFMPEG_AVFORMAT_H
 #   include <ffmpeg/avformat.h>
 #else
-#   include <avformat.h>
+#   include <libavformat/avformat.h>
 #endif
 
 #include "ffmpeg.h"
diff -ur vlc-0.8.6e/modules/codec/ffmpeg/postprocess.c vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/postprocess.c
--- vlc-0.8.6e/modules/codec/ffmpeg/postprocess.c	2007-11-21 23:23:29.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/postprocess.c	2008-04-01 20:13:27.000000000 +0200
@@ -28,9 +28,9 @@
 
 /* ffmpeg header */
 #ifdef HAVE_FFMPEG_AVCODEC_H
-#   include <ffmpeg/avcodec.h>
+#   include <ffmpeg/libavcodec/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #include "ffmpeg.h"
diff -ur vlc-0.8.6e/modules/codec/ffmpeg/video.c vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/video.c
--- vlc-0.8.6e/modules/codec/ffmpeg/video.c	2007-11-21 23:23:29.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/video.c	2008-04-01 20:13:27.000000000 +0200
@@ -31,9 +31,9 @@
 
 /* ffmpeg header */
 #ifdef HAVE_FFMPEG_AVCODEC_H
-#   include <ffmpeg/avcodec.h>
+#   include <ffmpeg/libavcodec/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #include "ffmpeg.h"
diff -ur vlc-0.8.6e/modules/codec/ffmpeg/video_filter.c vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/video_filter.c
--- vlc-0.8.6e/modules/codec/ffmpeg/video_filter.c	2008-02-24 20:01:53.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/modules/codec/ffmpeg/video_filter.c	2008-04-01 20:13:27.000000000 +0200
@@ -31,9 +31,9 @@
 
 /* ffmpeg header */
 #ifdef HAVE_FFMPEG_AVCODEC_H
-#   include <ffmpeg/avcodec.h>
+#   include <ffmpeg/libavcodec/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #include "ffmpeg.h"
diff -ur vlc-0.8.6e/modules/stream_out/switcher.c vlc-0.8.6e-ffmpegfix/modules/stream_out/switcher.c
--- vlc-0.8.6e/modules/stream_out/switcher.c	2007-11-21 23:23:22.000000000 +0100
+++ vlc-0.8.6e-ffmpegfix/modules/stream_out/switcher.c	2008-04-02 19:17:25.000000000 +0200
@@ -39,7 +39,7 @@
 #ifdef HAVE_FFMPEG_AVCODEC_H
 #   include <ffmpeg/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #ifdef HAVE_POSTPROC_POSTPROCESS_H
