--- modules/gui/wxwindows/wizard.cpp.orig	2005-08-07 13:47:17.000000000 +0200
+++ modules/gui/wxwindows/wizard.cpp	2005-08-07 14:06:41.000000000 +0200
@@ -1085,7 +1085,11 @@
         }
     }
     p_parent->SetStream( methods_array[i_method].psz_access ,
+#if wxUSE_UNICODE
+                         (char *)address_txtctrl->GetValue().mb_str().data() );
+#else
                          (char *)address_txtctrl->GetValue().mb_str() );
+#endif
 
     /* Set the action for the muxer page */
     ((wizEncapPage*)GetNext())->SetAction( p_parent->GetAction() );
