diff -Naur rekonq-0.9.1.ori/doc/nl/index.docbook rekonq-0.9.1/doc/nl/index.docbook
--- rekonq-0.9.1.ori/doc/nl/index.docbook	2012-04-01 09:11:53.000000000 +0200
+++ rekonq-0.9.1/doc/nl/index.docbook	2012-04-02 16:43:28.000000000 +0200
@@ -82,7 +82,7 @@
 ></surname
 > </personname>
     </author>
-&ged.vertaald;&Freek.de.Kruijf; 
+&Freek.de.Kruijf; 
     </authorgroup>
 <legalnotice
 >&FDLNotice;</legalnotice>
