diff -Naur gpm-1.20.1.orig/src/prog/gpm-root.y gpm-1.20.1/src/prog/gpm-root.y
--- gpm-1.20.1.orig/src/prog/gpm-root.y	2007-12-01 01:32:11.225537231 -0800
+++ gpm-1.20.1/src/prog/gpm-root.y	2007-12-01 01:33:21.339840854 -0800
@@ -525,7 +525,9 @@
 	            open("/dev/null",O_RDONLY); /* stdin  */
 	            open(consolename,O_WRONLY); /* stdout */
 	            dup(1);                     /* stderr */  
-	            for (i=3;i<OPEN_MAX; i++) close(i);
+		    int open_max = sysconf(_SC_OPEN_MAX);
+		    if (open_max == -1) open_max = 1024;
+	            for (i=3;i<open_max; i++) close(i);
 	            execl("/bin/sh","sh","-c",self->arg,(char *)NULL);
 	            exit(1); /* shouldn't happen */
 	         default: return 0;
