diff -Naur gpm-1.20.1.orig/src/lib/liblow.c gpm-1.20.1/src/lib/liblow.c
--- gpm-1.20.1.orig/src/lib/liblow.c	2005-07-30 13:19:07.969733048 -0700
+++ gpm-1.20.1/src/lib/liblow.c	2005-07-30 13:19:31.889096752 -0700
@@ -199,7 +199,8 @@
    Gpm_Stst *new = NULL;
    char* sock_name = 0;
 
-   option.consolename = NULL;
+   if (!checked_con)
+      option.consolename = NULL;
 
    gpm_report(GPM_PR_DEBUG,"VC: %d",flag);
 
