From c592030942bcb7a78ebde5f370d4892ca523253c Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Mon, 11 Jun 2018 14:53:06 +0200
Subject: [PATCH] initialize variable before use

---
 driconf_simpleui.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/driconf_simpleui.py b/driconf_simpleui.py
index 769aa67..4a9dad4 100644
--- a/driconf_simpleui.py
+++ b/driconf_simpleui.py
@@ -257,6 +257,9 @@ def lineWrap (string, chars=30):
             return head + tail
         else:
             i = chars
+
+            j = i
+
             while i >= chars/3:
                 if tail[i] == ' ':
                     j = i + 1
-- 
2.17.1

