--- configure.orig	2010-11-16 15:22:06.000000000 +0300
+++ configure	2010-11-16 15:22:21.694157446 +0300
@@ -22845,8 +22845,8 @@
                    -f "$mysql_zlib_dir/lib/libz.so" -o \
                    -f "$mysql_zlib_dir/lib/libz.sl" -o \
                    -f "$mysql_zlib_dir/lib/libz.dylib" \) \
-                -a -f "$mysql_zlib_dir/include/zlib.h"; then
-          ZLIB_INCLUDES="-I$mysql_zlib_dir/include"
+                -a -f "$mysql_zlib_dir/usr/include/zlib.h"; then
+          ZLIB_INCLUDES="-I$mysql_zlib_dir/usr/include"
           ZLIB_LIBS="-L$mysql_zlib_dir/lib -lz"
 
 save_CPPFLAGS="$CPPFLAGS"
