--- x.c.orig	2022-08-22 15:30:31.070291506 -0700
+++ x.c	2022-08-22 15:30:54.346325460 -0700
@@ -2168,7 +2168,6 @@
 {
 	xw.l = xw.t = 0;
 	xw.isfixed = False;
-	xsetcursor(cursorstyle);
 
 	ARGBEGIN {
 	case 'a':
@@ -2227,6 +2226,7 @@
 	setlocale(LC_CTYPE, "");
 	XSetLocaleModifiers("");
 	xrdb_load();
+	xsetcursor(cursorstyle);
 	signal(SIGUSR1, reload);
 	cols = MAX(cols, 1);
 	rows = MAX(rows, 1);
