--- bootstrap.orig	2010-04-02 13:02:21.394443953 +0200
+++ bootstrap	2010-04-02 13:03:15.957443368 +0200
@@ -42,15 +42,6 @@
 	exit 1;
 fi
 
-if test -n "`$AUTOMAKE --version 2>&1|head -n 1|egrep '1\.[0-6](\.[0-9]+)?$'`"; then
-	echo "Automake 1.7 or above is required. Aborting build...";
-	exit 1;
-fi
-
-if test -n "`$AUTOMAKE --version 2>&1|head -n 1|egrep '1\.7(\.[0-9]+)?$|1\.8(\.[0-9]+)?$'`"; then
-	echo "make dist only supported with automake 1.9 or above" >&2;
-fi
-
 if test -n "`$LIBTOOL --version 2>&1|head -n 1|cut -f 4 -d ' '|egrep '1\.[0-3](\.[0-9]+)?$'`"; then
 	echo "Libtool 1.4 or above is required. Aborting build...";
 	exit 1;
