--- src/setup/mime/Makefile.am.orig	2008-12-18 15:47:39.812769956 +0100
+++ src/setup/mime/Makefile.am	2008-12-18 15:52:38.879406921 +0100
@@ -27,9 +27,6 @@
 	  if which update-desktop-database>/dev/null 2>&1; then \
 		update-desktop-database; \
 	  fi; \
-	  if which gtk-update-icon-cache>/dev/null 2>&1; then \
-		gtk-update-icon-cache --force $(DESTDIR)$(datadir)/icons/gnome; \
-	  fi \
 	fi
 
 uninstall-hook:
@@ -40,7 +37,4 @@
 	  if which update-desktop-database>/dev/null 2>&1; then \
 		update-desktop-database; \
 	  fi; \
-	  if which gtk-update-icon-cache>/dev/null 2>&1; then \
-		gtk-update-icon-cache --force $(DESTDIR)$(datadir)/icons/gnome; \
-	  fi \
 	fi
--- src/setup/mime/Makefile.in.orig	2008-12-18 15:47:45.706102794 +0100
+++ src/setup/mime/Makefile.in	2008-12-18 15:53:03.912727516 +0100
@@ -490,9 +490,6 @@
 	  if which update-desktop-database>/dev/null 2>&1; then \
 		update-desktop-database; \
 	  fi; \
-	  if which gtk-update-icon-cache>/dev/null 2>&1; then \
-		gtk-update-icon-cache --force $(DESTDIR)$(datadir)/icons/gnome; \
-	  fi \
 	fi
 
 uninstall-hook:
@@ -503,9 +500,6 @@
 	  if which update-desktop-database>/dev/null 2>&1; then \
 		update-desktop-database; \
 	  fi; \
-	  if which gtk-update-icon-cache>/dev/null 2>&1; then \
-		gtk-update-icon-cache --force $(DESTDIR)$(datadir)/icons/gnome; \
-	  fi \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
