diff -Naur mesa-demos-8.0.1.orig/src/trivial/Makefile.in mesa-demos-8.0.1/src/trivial/Makefile.in
--- mesa-demos-8.0.1.orig/src/trivial/Makefile.in	2010-10-06 17:09:29.101000005 -0700
+++ mesa-demos-8.0.1/src/trivial/Makefile.in	2010-10-06 17:13:24.339001231 -0700
@@ -58,7 +58,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@HAVE_GLUT_TRUE@bin_PROGRAMS = clear$(EXEEXT) clear-fbo$(EXEEXT) \
+@HAVE_GLUT_TRUE@bin_PROGRAMS = clear-gl$(EXEEXT) clear-fbo$(EXEEXT) \
 @HAVE_GLUT_TRUE@	clear-fbo-tex$(EXEEXT) clear-random$(EXEEXT) \
 @HAVE_GLUT_TRUE@	clear-repeat$(EXEEXT) clear-scissor$(EXEEXT) \
 @HAVE_GLUT_TRUE@	clear-undefined$(EXEEXT) createwin$(EXEEXT) \
@@ -1001,7 +1001,7 @@
 	list=`for p in $$list; do echo "$$p"; done | sed 's/$(EXEEXT)$$//'`; \
 	echo " rm -f" $$list; \
 	rm -f $$list
-clear$(EXEEXT): $(clear_OBJECTS) $(clear_DEPENDENCIES) 
+clear-gl$(EXEEXT): $(clear_OBJECTS) $(clear_DEPENDENCIES) 
 	@rm -f clear$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(clear_OBJECTS) $(clear_LDADD) $(LIBS)
 clear-fbo$(EXEEXT): $(clear_fbo_OBJECTS) $(clear_fbo_DEPENDENCIES) 
