--- addrfilt.c~	2006-03-05 18:04:59.000000000 +0100
+++ addrfilt.c	2006-03-05 18:05:04.000000000 +0100
@@ -45,7 +45,7 @@
 
 struct _TableNode;
 
-typedef struct _TableNode ExtendedTable[TABLE_SIZE];
+typedef struct _TableNode *ExtendedTable;
 
 typedef enum {DENY, ALLOW, AS_PARENT} State;
 
@@ -124,7 +124,7 @@
 
   if (node->extended == NULL) {
 
-    node->extended = MallocNew(ExtendedTable);
+    node->extended = (ExtendedTable *)malloc(sizeof (ExtendedTable) * TABLE_SIZE);
 
     for (i=0; i<TABLE_SIZE; i++) {
       child_node = &((*(node->extended))[i]);
