=== modified file 'grub-core/kern/emu/getroot.c'
--- grub-core/kern/emu/getroot.c	2011-05-21 05:03:55 +0000
+++ grub-core/kern/emu/getroot.c	2011-05-30 23:56:40 +0000
@@ -220,9 +220,17 @@ grub_find_root_device_from_mountinfo (co
   /* Now scan visible mounts for the ones we're interested in.  */
   for (i = entry_len - 1; i >= 0; i--)
     {
+      struct stat st;
+
       if (!*entries[i].device)
 	continue;
 
+      if (stat (entries[i].device, &st) < 0)
+	/* The root filesystem device recorded in mountinfo does not
+	   exist, perhaps due to a udev bug.  Fall back to more primitive
+	   methods.  */
+	break;
+
       ret = strdup (entries[i].device);
       if (relroot)
 	*relroot = strdup (entries[i].enc_root);
