From 5c7b234752f9e995e13c2c08f9d3dcd0509020d6 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 2 Jan 2020 00:15:06 +0100
Subject: [PATCH] Install to correct lib directory

---
 source/Makefile | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/source/Makefile b/source/Makefile
index f841f3f..718561a 100644
--- a/source/Makefile
+++ b/source/Makefile
@@ -20,10 +20,9 @@
 
 # Modify as required.
 #
-SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 PREFIX ?= /usr/local
 INCDIR ?= $(PREFIX)/include
-LIBDIR ?= $(PREFIX)/lib$(SUFFIX)
+LIBDIR ?= $(PREFIX)/lib
 
 
 MAJVERS = 2
@@ -51,8 +50,8 @@ $(CLTHREADS_MIN): $(CLTHREADS_O)
 
 
 install:	$(CLTHREADS_MIN)
-	install -d $(DESTDIR)$(PREFIX)/$(INCDIR)
-	install -d $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	install -d $(DESTDIR)/$(INCDIR)
+	install -d $(DESTDIR)/$(LIBDIR)
 	install -m 644 $(CLTHREADS_H) $(DESTDIR)$(INCDIR)
 	install -m 755 $(CLTHREADS_MIN) $(DESTDIR)$(LIBDIR)
 	ldconfig
-- 
2.24.1

