diff -Naur amrecgui-0.2.3.orig/src/recoverview.C amrecgui-0.2.3/src/recoverview.C
--- amrecgui-0.2.3.orig/src/recoverview.C	2005-07-27 11:01:19.103577500 -0700
+++ amrecgui-0.2.3/src/recoverview.C	2005-07-27 11:03:02.642048250 -0700
@@ -480,9 +480,9 @@
     }
 }
 
-CStrings CRecoverView::Read() return strRead
+CStrings CRecoverView::Read()
 {
-    CStrings strLine;
+    CStrings strLine, strRead;
 
     int nChar;
 
@@ -525,7 +525,7 @@
     } while (nChar != EOF);
 }
 
-CStrings CRecoverView::Write(const char *strCommand, bool bRead) return strRead
+CStrings CRecoverView::Write(const char *strCommand, bool bRead)
 {
     CStrings strLine = strCommand;
 
@@ -534,7 +534,7 @@
     fflush(m_Command);
     if (bRead)
     {
-	strRead = Read();
+	CStrings strRead = Read();
     }
 }
 
@@ -830,9 +830,9 @@
     }
 }
 
-CStrings CRecoverView::File(CGraphicalTreeNode *pNode) return strFile
+CStrings CRecoverView::File(CGraphicalTreeNode *pNode)
 {
-    strFile = pNode->Node();
+    CStrings strFile = pNode->Node();
 
     if (strFile.Right(1) == "/")
     {
