From 01360a68ee9ad11b00cb7fa26f3b31abf06fccdc Mon Sep 17 00:00:00 2001
From: Bastien Dejean <nihilhill@gmail.com>
Date: Tue, 12 Apr 2016 10:47:55 +0200
Subject: [PATCH] Use the proper type for the return value of getopt

---
 sres.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sres.c b/sres.c
index ffd6cce..ee31cf4 100644
--- a/sres.c
+++ b/sres.c
@@ -13,7 +13,7 @@ int main (int argc, char *argv[]) {
 
     int return_what = RETURN_BOTH;
     char format[MAXLEN] = {0};
-    char opt;
+    int opt;
 
     while ((opt = getopt(argc, argv, "vhHW")) != -1) {
         switch (opt) {
