diff -Naur aewm-1.3.12/Makefile aewm-1.3.12-fixed/Makefile
--- aewm-1.3.12/Makefile	2007-11-15 23:46:18.000000000 +0100
+++ aewm-1.3.12-fixed/Makefile	2008-01-03 16:19:36.000000000 +0100
@@ -20,9 +20,9 @@
 CC = gcc
 CFLAGS = -g -O2 -Wall
 
-BINDIR = $(DESTDIR)$(XROOT)/bin
-MANDIR = $(DESTDIR)$(XROOT)/man/man1
-CFGDIR = $(DESTDIR)/etc/X11/aewm
+BINDIR = $(DESTDIR)/usr/bin
+MANDIR = $(DESTDIR)/usr/share/man/man1
+CFGDIR = $(DESTDIR)/etc/aewm
 
 PLAINOBJ = aesession.o parser.o
 CLIENTOBJ = common.o atom.o
@@ -78,6 +78,7 @@
 
 AEMAN = aewm.1x aeclients.1x
 AERC = aewmrc clientsrc
+AEDOC = doc
 
 install: all
 	mkdir -p $(BINDIR) $(MANDIR) $(CFGDIR)
