--- opts.mk	2009-10-14 17:53:17.000000000 +0200
+++ opts.mk.fixed	2009-10-14 18:03:44.000000000 +0200
@@ -2,16 +2,15 @@
 
 # Change compile options here if desired
 CC = gcc
-CFLAGS = -g -O2 -fno-strict-aliasing -Wall
 
 # If you are using X11R7, change this to /usr
-XROOT = /usr/X11R6
+XROOT = /usr
 
 # Where to install
-PREFIX = /usr/local
+PREFIX = /usr
 BINDIR = $(PREFIX)/bin
 MANDIR = $(PREFIX)/share/man/man1
-CFGDIR = $(PREFIX)/etc/aewm
+CFGDIR = /etc/aewm
 
 # Uncomment to enable Shape extension support
 #OPT_WMFLAGS += -DSHAPE

