--- etc/esehttpd.conf.in.orig	2003-10-27 13:25:20.000000000 +0100
+++ etc/esehttpd.conf.in	2003-10-27 13:32:48.000000000 +0100
@@ -1,15 +1,15 @@
 
-ServerRoot		@prefix@/lib/esehttpd
+ServerRoot		/var
 Port			80
-# User			nobody
-# Group			nobody
+User			nobody
+Group			nogroup
 # BindAddress		*
 # ServerName		localhost
 # SSLPort		443
-# SSLCertificateFile	conf/server.crt
-# SSLCertificateKeyFile	conf/server.key
-# SSLDHParamFile	conf/server.dh
-# SSLCACertificateFile	conf/ca.crt
+# SSLCertificateFile	/etc/esehttpd/server.crt
+# SSLCertificateKeyFile	/etc/esehttpd/server.key
+# SSLDHParamFile	/etc/esehttpd/server.dh
+# SSLCACertificateFile	/etc/esehttpd/ca.crt
 # SSLCipherSuite	EDH-RSA-DES-CBC3-SHA:EDH-DSS-DES-CBC3-SHA:DES-CBC3-SHA
 
 # MaxFiles		1024
@@ -26,9 +26,9 @@ Port			80
 
 # ErrorDocumentDir	www/errors/
 
-TypesConfig		conf/mime.types
+TypesConfig		/etc/esehttpd/mime.types
 DefaultType		text/plain
-CustomLog		log/access_log
+CustomLog		/var/log/esehttpd/access_log
 
 ChangeRoot	www/
 DocumentRoot	www/html/
@@ -53,10 +53,10 @@ ScriptAlias	/cgi-bin	www/cgi-bin/		cgi-s
 
 # <VirtualHost foo.bar.org>
 # SSLPort		443
-# SSLCertificateFile	@prefix@/lib/esehttpd/conf/foobar.crt
-# SSLCertificateKeyFile	@prefix@/lib/esehttpd/conf/foobar.key
-# SSLDHParamFile	@prefix@/lib/esehttpd/conf/foobar.dh
-# <Directory @prefix@/lib/esehttpd/www/>
+# SSLCertificateFile	/etc/esehttpd/foobar.crt
+# SSLCertificateKeyFile	/etc/esehttpd/foobar.key
+# SSLDHParamFile	/etc/esehttpd/foobar.dh
+# <Directory /var/www/>
 #     SSLVerifyClient require
 # </Directory>
 # </VirtualHost>
