diff -Nur cdbakeoven-2.0beta2/cdbakeoven/cdboshared/CdboConstants.h cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboshared/CdboConstants.h
--- cdbakeoven-2.0beta2/cdbakeoven/cdboshared/CdboConstants.h	2003-02-17 12:44:03.000000000 +0100
+++ cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboshared/CdboConstants.h	2004-07-09 14:24:40.000000000 +0200
@@ -73,7 +73,7 @@
 		MediaTypeMixed
 	};
 
-};
+}
 
 #endif
 
diff -Nur cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboplugins/cdbo_konqy_plugins.cpp cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboplugins/cdbo_konqy_plugins.cpp
--- cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboplugins/cdbo_konqy_plugins.cpp	2003-02-17 12:44:03.000000000 +0100
+++ cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboplugins/cdbo_konqy_plugins.cpp	2004-07-09 14:38:14.752941147 +0200
@@ -41,7 +41,7 @@
 
 #include <qdir.h>
 
-K_EXPORT_COMPONENT_FACTORY( cdbokonqyplugins, KGenericFactory<CdboKonqyPlugins> );
+K_EXPORT_COMPONENT_FACTORY( cdbokonqyplugins, KGenericFactory<CdboKonqyPlugins> )
 
 CdboKonqyPlugins::CdboKonqyPlugins(QObject* parent, const char* name, const QStringList& )
 	:KParts::Plugin(parent, name){
diff -Nur cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboplugins/cdboplugins.cpp cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboplugins/cdboplugins.cpp
--- cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboplugins/cdboplugins.cpp	2003-02-17 12:44:03.000000000 +0100
+++ cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboplugins/cdboplugins.cpp	2004-07-09 14:38:30.970874083 +0200
@@ -51,7 +51,7 @@
 #include "CdboComOutDialog.h"
 #include "CdboPluginManager.h"
 
-K_EXPORT_COMPONENT_FACTORY( cdboplugins, KGenericFactory<CdboPlugins> );
+K_EXPORT_COMPONENT_FACTORY( cdboplugins, KGenericFactory<CdboPlugins> )
 
 CdboPlugins::CdboPlugins(QObject* parent, const char* name, const QStringList& )
 	:KParts::Plugin(parent, name){
diff -Nur cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboviewparts/CdboAudioViewPart.cpp cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboviewparts/CdboAudioViewPart.cpp
--- cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboviewparts/CdboAudioViewPart.cpp	2003-02-17 12:44:03.000000000 +0100
+++ cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboviewparts/CdboAudioViewPart.cpp	2004-07-09 14:33:57.382744381 +0200
@@ -52,7 +52,7 @@
 #include "CdboVersion.h"
 
 
-K_EXPORT_COMPONENT_FACTORY( libcdboaudioview, CdboAudioViewFactory );
+K_EXPORT_COMPONENT_FACTORY( libcdboaudioview, CdboAudioViewFactory )
 
 //===================================
 CdboAudioViewPart::CdboAudioViewPart(QWidget *parentWidget, const char *widgetName,
diff -Nur cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboviewparts/CdboDataViewPart.cpp cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboviewparts/CdboDataViewPart.cpp
--- cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboviewparts/CdboDataViewPart.cpp	2003-02-17 12:44:03.000000000 +0100
+++ cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboviewparts/CdboDataViewPart.cpp	2004-07-09 14:31:24.569221298 +0200
@@ -57,7 +57,7 @@
 
 #include "CdboVersion.h"
 
-K_EXPORT_COMPONENT_FACTORY( libcdbodataview, CdboDataViewFactory );
+K_EXPORT_COMPONENT_FACTORY( libcdbodataview, CdboDataViewFactory )
 
 //===================================
 CdboDataViewPart::CdboDataViewPart(QWidget *parentWidget, const char *widgetName,
diff -Nur cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboviewparts/CdboNewcdViewPart.cpp cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboviewparts/CdboNewcdViewPart.cpp
--- cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboviewparts/CdboNewcdViewPart.cpp	2003-02-17 12:44:03.000000000 +0100
+++ cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboviewparts/CdboNewcdViewPart.cpp	2004-07-09 14:35:18.192444744 +0200
@@ -43,7 +43,7 @@
 
 #include "CdboVersion.h"
 
-K_EXPORT_COMPONENT_FACTORY( libcdbonewcdview, CdboNewcdViewFactory );
+K_EXPORT_COMPONENT_FACTORY( libcdbonewcdview, CdboNewcdViewFactory )
 
 //===================================
 CdboNewcdViewPart::CdboNewcdViewPart(QWidget *parentWidget, const char *widgetName,
diff -Nur cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboviewparts/CdboSrcVeiwPart.cpp cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboviewparts/CdboSrcVeiwPart.cpp
--- cdbakeoven-2.0beta2/cdbakeoven/cdboviews/cdboviewparts/CdboSrcVeiwPart.cpp	2003-02-17 12:44:03.000000000 +0100
+++ cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/cdboviewparts/CdboSrcVeiwPart.cpp	2004-07-09 14:32:37.099976858 +0200
@@ -41,7 +41,7 @@
 #include "CdboFileListView.h"
 #include "CdboVersion.h"
 
-K_EXPORT_COMPONENT_FACTORY( libcdbosrcview, CdboSrcVeiwFactory );
+K_EXPORT_COMPONENT_FACTORY( libcdbosrcview, CdboSrcVeiwFactory )
 
 //===================================
 CdboSrcVeiwPart::CdboSrcVeiwPart(QWidget *parentWidget, const char *widgetName,
diff -Nur cdbakeoven-2.0beta2/cdbakeoven/cdboviews/kdedmodule/cdrwatcher.cpp cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/kdedmodule/cdrwatcher.cpp
--- cdbakeoven-2.0beta2/cdbakeoven/cdboviews/kdedmodule/cdrwatcher.cpp	2003-02-17 12:44:03.000000000 +0100
+++ cdbakeoven-2.0beta2-fixes/cdbakeoven/cdboviews/kdedmodule/cdrwatcher.cpp	2004-07-09 14:26:15.000000000 +0200
@@ -244,7 +244,7 @@
 	KDEDModule *create_cdrwatcher(const QCString &obj){
 		return new CDRWatcherModule(obj);
 	}
-};
+}
 
 
 
diff -Nur cdbakeoven-2.0beta2/Makefile.in cdbakeoven-2.0beta2-fixes/Makefile.in
--- cdbakeoven-2.0beta2/Makefile.in	2003-02-17 12:47:40.000000000 +0100
+++ cdbakeoven-2.0beta2-fixes/Makefile.in	2004-07-09 14:24:40.088774536 +0200
@@ -278,23 +278,6 @@
 
 am__CONFIG_DISTCLEAN_FILES = config.status config.cache config.log \
  configure.lineno
-$(srcdir)/Makefile.in:  Makefile.am  $(top_srcdir)/configure.in $(ACLOCAL_M4)
-#>- 	cd $(top_srcdir) && \
-#>- 	  $(AUTOMAKE) --foreign  Makefile
-#>+ 3
-	cd $(top_srcdir) && \
-	  $(AUTOMAKE) --foreign  Makefile
-	cd $(top_srcdir) && perl admin/am_edit Makefile.in
-Makefile:  $(srcdir)/Makefile.in  $(top_builddir)/config.status
-	cd $(top_builddir) && $(SHELL) ./config.status $@ $(am__depfiles_maybe)
-
-$(top_builddir)/config.status: $(srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
-	$(SHELL) ./config.status --recheck
-$(srcdir)/configure:  $(srcdir)/configure.in $(ACLOCAL_M4) $(CONFIGURE_DEPENDENCIES)
-	cd $(srcdir) && $(AUTOCONF)
-
-$(ACLOCAL_M4):  configure.in acinclude.m4
-	cd $(srcdir) && $(ACLOCAL) $(ACLOCAL_AMFLAGS)
 
 config.h: stamp-h1
 	@if test ! -f $@; then \
@@ -306,10 +289,6 @@
 	@rm -f stamp-h1
 	cd $(top_builddir) && $(SHELL) ./config.status config.h
 
-$(srcdir)/config.h.in:  $(top_srcdir)/configure.in $(ACLOCAL_M4) 
-	cd $(top_srcdir) && $(AUTOHEADER)
-	touch $(srcdir)/config.h.in
-
 distclean-hdr:
 	-rm -f config.h stamp-h1
 
@@ -631,15 +610,6 @@
 	@echo "****************************************************"
 	@echo ""
 
-$(top_srcdir)/acinclude.m4: $(shell find . -name acinclude.m4.in) $(top_srcdir)/admin/acinclude.m4.in $(top_srcdir)/admin/libtool.m4.in
-	@cd $(top_srcdir) && cat $^ > acinclude.m4;
-
-$(top_srcdir)/configure.in: $(top_srcdir)/configure.files $(shell test -f $(top_srcdir)/configure.files && sed -e "s%^%\$(top_srcdir)/%" $(top_srcdir)/configure.files) $(top_srcdir)/subdirs
-	cd $(top_srcdir) && $(MAKE) -f admin/Makefile.common configure.in
-$(top_srcdir)/subdirs:
-	cd $(top_srcdir) && $(MAKE) -f admin/Makefile.common subdirs
-$(top_srcdir)/configure.files:
-	cd $(top_srcdir) && $(MAKE) -f admin/Makefile.common configure.files
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
@@ -647,13 +617,6 @@
 #>+ 2
 docs-am:
 
-#>+ 6
-force-reedit:
-		cd $(top_srcdir) && \
-	  $(AUTOMAKE) --foreign  Makefile
-	cd $(top_srcdir) && perl admin/am_edit Makefile.in
-
-
 #>+ 5
 bcheck: bcheck-recursive
 
