From 496ed6f1d4573f272579f12d909172de0e954868 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 2 Nov 2018 11:57:15 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 unliftio-core.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/unliftio-core.cabal b/unliftio-core.cabal
index 23923a5..de4e637 100644
--- a/unliftio-core.cabal
+++ b/unliftio-core.cabal
@@ -26,8 +26,8 @@ library
   hs-source-dirs:
       src
   build-depends:
-      base >=4.5 && <4.12
-    , transformers >=0.2 && <0.6
+      base >=4.5
+    , transformers >=0.2
   exposed-modules:
       Control.Monad.IO.Unlift
   other-modules:
-- 
2.19.1

