From 810806d4d5003dbc61f0efa38f363d0417c66ed3 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sat, 8 Jun 2013 23:07:57 +0200
Subject: [PATCH] fix failure during install

---
 src/Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index feb144e..5d12cad 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -243,9 +243,8 @@ install:
 	if [ -f diag1.img ]; then cp -f diag1.img $$DESTDIR$(BOOT_DIR); fi
 	if [ -f diag2.img ]; then cp -f diag2.img $$DESTDIR$(BOOT_DIR); fi
 
-	strip lilo
 	cp lilo $$DESTDIR$(SBIN_DIR)/lilo
-	[ -x lilo.static ] && strip lilo.static
+	strip $$DESTDIR$(SBIN_DIR)/lilo
 	cp ../mkrescue $$DESTDIR$(USRSBIN_DIR)/mkrescue
 	cp ../keytab-lilo.pl $$DESTDIR$(USRSBIN_DIR)/keytab-lilo
 
@@ -264,6 +263,8 @@ mkloader:	temp.c
 tidy:
 	rm -f core $(LIS) *.shs *.crc $(TMP)*
 	rm -rf sbin boot usr
+	if [ -x lilo ]; then strip lilo; fi
+	if [ -x lilo.static ]; then strip lilo.static; fi
 
 clean:
 	rm -f *.o *.s *.i *.img *.b tmp_make version
-- 
1.8.3

