diff -Nuar DeviceKit-003/autogen.sh DeviceKit-003.mod/autogen.sh
--- DeviceKit-003/autogen.sh	2009-03-02 21:01:50.000000000 +0100
+++ DeviceKit-003.mod/autogen.sh	2009-04-20 07:49:18.000000000 +0200
@@ -31,13 +31,6 @@
   }
 }
 
-(gtkdocize --flavour no-tmpl) < /dev/null > /dev/null 2>&1 || {
-	echo
-	echo "You must have gtk-doc installed to compile $PROJECT."
-	echo "Install the appropriate package for your distribution,"
-	echo "or get the source tarball at http://ftp.gnome.org/pub/GNOME/sources/gtk-doc/"
-	DIE=1
-}
 
 (automake --version) < /dev/null > /dev/null 2>&1 || {
   echo
@@ -92,7 +85,7 @@
       echo "Running autoconf ..."
       autoconf
 
-conf_flags="--enable-maintainer-mode --enable-gtk-doc"
+conf_flags="--enable-maintainer-mode"
 
 if test x$NOCONFIGURE = x; then
   echo "Running $srcdir/configure $conf_flags $@ ..."
diff -Nuar DeviceKit-003/doc/Makefile.am DeviceKit-003.mod/doc/Makefile.am
--- DeviceKit-003/doc/Makefile.am	2009-03-02 21:01:50.000000000 +0100
+++ DeviceKit-003.mod/doc/Makefile.am	2009-04-20 07:41:35.000000000 +0200
@@ -60,14 +60,8 @@
 
 # Extra options to supply to gtkdoc-fixref
 FIXXREF_OPTIONS=
-
-if ENABLE_GTK_DOC
-include $(top_srcdir)/gtk-doc.make
-else
-EXTRA_DIST =
-CLEANFILES =
-endif
-
+EXTRA_DIST=
+CLEANFILES=
 # Version information for marking the documentation
 EXTRA_DIST += version.xml.in
 
