From 2db4f5aa6d7a0bba5fcca46d998fff2a65ebf8af Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 17 Jan 2018 20:24:21 +0100
Subject: [PATCH] fix dependencies

---
 echo.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/echo.cabal b/echo.cabal
index 4a18d27..e8de149 100644
--- a/echo.cabal
+++ b/echo.cabal
@@ -39,7 +39,7 @@ library
                        System.IO.Echo.Internal
 
   build-depends:       base    >= 4.3     && < 5
-                     , process >= 1.0.1.1 && < 1.5
+                     , process >= 1.0.1.1 && < 1.7
   if os(windows)
     cpp-options:       "-DWINDOWS"
     build-depends:     mintty >= 0.1 && < 0.2
-- 
2.15.1

