From 5c7cc50ee395e5d1ac5d2d4337df9d2769ee9ec8 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 26 Jul 2020 11:59:03 +0200
Subject: [PATCH] Do not download during build

---
 Makefile.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.in b/Makefile.in
index 5ace8da..807c515 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -282,6 +282,7 @@ gpsbabel.html: FORCE # gpsbabel
 	  --stringparam html.valid.html "1" \
 	  --stringparam html.stylesheet \
 	  "https://www.gpsbabel.org/style3.css" \
+    --nonet \
 	  http://docbook.sourceforge.net/release/xsl/current/xhtml/docbook.xsl \
 	xmldoc/readme.xml
 
-- 
2.27.0

