From 51966b307deb21e0a35999aa2e2cc413fcdd9f9e Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Thu, 26 Feb 2015 20:23:53 +0100
Subject: [PATCH] do not download during build

---
 Makefile.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.in b/Makefile.in
index c767fe2..e3da604 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -271,6 +271,7 @@ gpsbabel.html: FORCE # gpsbabel
 	  --stringparam html.valid.html "1" \
 	  --stringparam html.stylesheet \
 	  "http://www.gpsbabel.org/style3.css" \
+	  --nonet \
 	  http://docbook.sourceforge.net/release/xsl/current/xhtml/docbook.xsl \
 	xmldoc/readme.xml
 
-- 
2.1.0

