--- doc/asn2deb.pod.orig	2013-05-21 03:00:25.000000000 +0400
+++ doc/asn2deb.pod	2013-06-09 12:28:54.264075709 +0400
@@ -1,3 +1,4 @@
+=encoding ISO8859-1
 
 =head1 NAME
 
--- doc/idl2deb.pod.orig	2013-05-21 03:00:25.000000000 +0400
+++ doc/idl2deb.pod	2013-06-09 12:29:10.085321169 +0400
@@ -1,3 +1,4 @@
+=encoding ISO8859-1
 
 =head1 NAME
 
--- doc/wireshark.pod.template.orig	2013-05-21 02:17:14.000000000 +0400
+++ doc/wireshark.pod.template	2013-06-09 12:28:31.164717330 +0400
@@ -1,3 +1,4 @@
+=encoding UTF-8
 
 =head1 NAME
 
