Only in libfreebob-1.0.11.gcc43: autom4te.cache
diff -ru libfreebob-1.0.11/src/libfreebobavc/avc_extended_plug_info.cpp libfreebob-1.0.11.gcc43/src/libfreebobavc/avc_extended_plug_info.cpp
--- libfreebob-1.0.11/src/libfreebobavc/avc_extended_plug_info.cpp	2007-10-15 20:15:08.000000000 +0200
+++ libfreebob-1.0.11.gcc43/src/libfreebobavc/avc_extended_plug_info.cpp	2008-07-09 03:15:50.000000000 +0200
@@ -23,6 +23,7 @@
 #include "ieee1394service.h"
 
 #include <cstdlib>
+#include <cstring>
 #include <netinet/in.h>
 #include <iostream>
 
diff -ru libfreebob-1.0.11/src/libfreebobavc/avc_generic.cpp libfreebob-1.0.11.gcc43/src/libfreebobavc/avc_generic.cpp
--- libfreebob-1.0.11/src/libfreebobavc/avc_generic.cpp	2007-10-15 20:15:08.000000000 +0200
+++ libfreebob-1.0.11.gcc43/src/libfreebobavc/avc_generic.cpp	2008-07-09 03:16:26.000000000 +0200
@@ -23,6 +23,7 @@
 #include "ieee1394service.h"
 
 #include <cstdlib>
+#include <cstring>
 #include <netinet/in.h>
 
 #define DEBUG_EXTRA_VERBOSE 2
Only in libfreebob-1.0.11.gcc43: stamp-h.in
diff -ru libfreebob-1.0.11/tests/test-extplugcmd.cpp libfreebob-1.0.11.gcc43/tests/test-extplugcmd.cpp
--- libfreebob-1.0.11/tests/test-extplugcmd.cpp	2007-10-15 20:15:07.000000000 +0200
+++ libfreebob-1.0.11.gcc43/tests/test-extplugcmd.cpp	2008-07-09 03:17:19.000000000 +0200
@@ -24,6 +24,7 @@
 #include "libfreebobavc/ieee1394service.h"
 
 #include <argp.h>
+#include <cstdlib>
 
 using namespace std;
 
diff -ru libfreebob-1.0.11/tests/test-volume.cpp libfreebob-1.0.11.gcc43/tests/test-volume.cpp
--- libfreebob-1.0.11/tests/test-volume.cpp	2007-10-15 20:15:07.000000000 +0200
+++ libfreebob-1.0.11.gcc43/tests/test-volume.cpp	2008-07-09 03:17:44.000000000 +0200
@@ -21,6 +21,7 @@
 #include "libfreebobavc/avc_function_block.h"
 #include "libfreebobavc/serialize.h"
 #include "libfreebobavc/ieee1394service.h"
+#include <cstdlib>
 
 const bool bVerbose = false;
 
