--- Makefile.old	2004-12-18 19:22:40.413965128 +0000
+++ Makefile	2004-12-18 19:22:15.669726824 +0000
@@ -16,7 +16,7 @@
 KERNELHEADER    = /usr/src/linux/include
 CFLAGS		= -g -Wall -O2 -I $(KERNELHEADER)
 
-DESTDIR		=
+DESTDIR		= 
 PREFIX		= /usr/local
 
 INSDIR		= $(PREFIX)/sbin
@@ -50,13 +50,13 @@
 	$(INS) -m 644 $(MANPAGE) $(DESTDIR)$(MANDIR)
 	gzip -9f $(DESTDIR)$(MANDIR)/$(MANPAGE)
 
-	$(INS) -m 755 $(RCFILE) $(DESTDIR)$(RCFILETO)/$(RCFILEFINAL)
+#	$(INS) -m 755 $(RCFILE) $(DESTDIR)$(RCFILETO)/$(RCFILEFINAL)
 
-ifndef DESTDIR
-		chkconfig --add $(RCFILEFINAL)
-else
-		echo "MAKE: Skipping chkconfig operation (rpm build?)"
-endif
+#ifndef DESTDIR
+#		chkconfig --add $(RCFILEFINAL)
+#else
+#		echo "MAKE: Skipping chkconfig operation (rpm build?)"
+#endif
 
 device:
 	mkdir -p $(DESTDIR)/dev/cpu
