diff -Naur libmtp-0.2.6.1/src/music-players.h libmtp-0.2.6.1-patched/src/music-players.h
--- libmtp-0.2.6.1/src/music-players.h	2008-02-29 23:48:49.000000000 +0100
+++ libmtp-0.2.6.1-patched/src/music-players.h	2008-05-23 01:06:04.000000000 +0200
@@ -489,6 +489,8 @@
    */
   // Reported by Øyvind Stegard <stegaro@users.sourceforge.net>
   { "SonyEricsson", 0x0fce, "K850i", 0x0075, DEVICE_FLAG_NONE },
+  // Reported by Arjan Bouter <abouter@sourcemage.org>
+  { "SonyEricsson", 0x0fce, "W890i", 0x00b3, DEVICE_FLAG_NONE },
   // Reported by Michael Eriksson
   { "SonyEricsson", 0x0fce, "W910", 0x0076, DEVICE_FLAG_NONE },
 
