From 78cfc92044f925a312ef8b70c324e7d78f535f2a Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 13:48:03 +0100
Subject: [PATCH] relax dependency on base version

---
 hashable.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hashable.cabal b/hashable.cabal
index 74f6512..05a30f8 100644
--- a/hashable.cabal
+++ b/hashable.cabal
@@ -48,7 +48,7 @@ Library
   Exposed-modules:   Data.Hashable
                      Data.Hashable.Lifted
   Other-modules:     Data.Hashable.Class
-  Build-depends:     base >= 4.4 && < 4.12,
+  Build-depends:     base >= 4.4,
                      bytestring >= 0.9 && < 0.11,
                      deepseq >= 1.3 && < 1.5
   if impl(ghc)
-- 
2.19.1

