--- mozilla/nsprpub/pr/include/prlog.h:3.14	Sun Apr 25 15:00:47 2004
+++ mozilla/nsprpub/pr/include/prlog.h	Mon May 28 14:48:26 2007
@@ -251,6 +251,14 @@
 
 #endif /* defined(DEBUG) || defined(FORCE_PR_ASSERT) */
 
+/*
+** Compile-time assert. "condition" must be a constant expression.
+** The macro can be used only in places where an "extern" declaration is
+** allowed.
+*/
+#define PR_STATIC_ASSERT(condition) \
+    extern void pr_static_assert(int arg[(condition) ? 1 : -1])
+
 PR_END_EXTERN_C
 
 #endif /* prlog_h___ */

