diff --git a/core/document.cpp b/core/document.cpp
index 3215a1abce6292a6cc25c5f8b645232c92d75ec5..0aa5b698019a2660f2d6baabd54cef1e82002b0e 100644
--- a/core/document.cpp
+++ b/core/document.cpp
@@ -4388,7 +4388,8 @@ void Document::processAction( const Action * action )
                 {
                     const QUrl realUrl = KIO::upUrl(d->m_url).resolved(url);
                     // KRun autodeletes
-                    new KRun( realUrl, d->m_widget );
+                    KRun *r = new KRun( realUrl, d->m_widget );
+                    r->setRunExecutables(false);
                 }
             }
             } break;
