From f6e76cf8890195d46331a6e41eda2fb3c83c0328 Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Sun, 7 Apr 2019 18:37:56 +0200
Subject: [PATCH] fix meson

---
 gusb/meson.build | 5 +++--
 meson.build      | 4 +++-
 2 files changed, 6 insertions(+), 3 deletions(-)
diff --git a/meson.build b/meson.build
index c6f16c1..d076d6e 100644
--- a/meson.build
+++ b/meson.build
@@ -1,7 +1,7 @@
 project('libgusb', 'c',
   version : '0.3.0',
   license : 'LGPL-2.1+',
-  meson_version : '>=0.37.0',
+  meson_version : '>=0.50.0',
   default_options : ['c_std=c99']
 )
 
@@ -108,4 +108,6 @@ root_incdir = include_directories('.')
 
 subdir('gusb')
 subdir('tools')
+if get_option('docs')
 subdir('docs')
+endif
-- 
2.21.0

