diff -Nur xmovie-1.9.12/build/Makefile.xmovie xmovie-1.9.12-shared/build/Makefile.xmovie
--- xmovie-1.9.12/build/Makefile.xmovie	2003-11-11 07:43:15.000000000 +0100
+++ xmovie-1.9.12-shared/build/Makefile.xmovie	2004-02-13 13:03:03.000000000 +0100
@@ -7,8 +7,6 @@
 
 
 DIRS := \
-	libmpeg3 \
-	quicktime \
 	guicast \
 	xmovie \
 
diff -Nur xmovie-1.9.12/xmovie/Makefile xmovie-1.9.12-shared/xmovie/Makefile
--- xmovie-1.9.12/xmovie/Makefile	2004-02-03 09:09:36.000000000 +0100
+++ xmovie-1.9.12-shared/xmovie/Makefile	2004-02-13 13:03:51.000000000 +0100
@@ -37,8 +37,6 @@
 	-I$(SNDFILE_DIR)/src
 
 LIBS += -L/usr/X11R6/lib \
-	../libmpeg3/$(OBJDIR)/libmpeg3.a \
-	../quicktime/$(OBJDIR)/libquicktime.a \
 	../guicast/$(OBJDIR)/libguicast.a \
 	$(SNDFILE_DIR)/$(OBJDIR)/libsndfile.a \
  	$(GLLIBS) \
@@ -51,7 +49,9 @@
 	-lpng \
 	-lz \
 	-ldl \
-	-lXxf86vm
+	-lXxf86vm \
+	-lquicktime \
+	-lmpeg3
 
 # Static libraries for distribution
 ifeq ($(USE_STATIC),y)
