From 62ea09194932bf48e24b149dd9343e54303f2073 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 6 Jan 2018 18:36:40 +0100
Subject: [PATCH] fix warning about undefined function

---
 src/misc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/misc.c b/src/misc.c
index ef27833..1433b7d 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -30,6 +30,9 @@
 /*  IMPLIED WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED */
 /*  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR */
 /*  PURPOSE. */
+
+#include <malloc.h>
+
 #include "flexdef.h"
 #include "tables.h"
 
-- 
2.15.1

