--- setup.py.orig	2009-05-07 19:25:54.000000000 +0400
+++ setup.py	2010-12-02 00:38:07.275726702 +0300
@@ -14,7 +14,7 @@
     license='BSD', 
     url='http://trac-hacks.org/wiki/LdapPlugin',
     keywords = "trac ldap permission group acl",
-    install_requires = [ 'Trac>=0.11', 'Trac<0.12'],
+    install_requires = [ 'Trac>=0.11', 'Trac<0.13'],
     packages = find_packages(exclude=['ez_setup', '*.tests*']),
     package_data = { },
     entry_points = {
