diff --git a/gtk/Makefile.am b/gtk/Makefile.am
index 842c2f9..63f88e9 100644
--- a/gtk/Makefile.am
+++ b/gtk/Makefile.am
@@ -481,6 +481,7 @@ gtk_private_h_sources =		\
 	gtkfilesystemmodel.h	\
 	gtkfontchooserprivate.h	\
 	gtkfontchooserutils.h	\
+	gtkfontchooserwidgetprivate.h \
 	gtkgestureprivate.h	\
 	gtkgesturedragprivate.h	\
 	gtkgesturelongpressprivate.h	\
diff --git a/gtk/gtkfontchooserwidgetprivate.h b/gtk/gtkfontchooserwidgetprivate.h
new file mode 100644
index 0000000..f51447e
--- /dev/null
+++ b/gtk/gtkfontchooserwidgetprivate.h
@@ -0,0 +1,30 @@
+/* GTK - The GIMP Toolkit
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __GTK_FONT_CHOOSER_WIDGET_PRIVATE_H__
+#define __GTK_FONT_CHOOSER_WIDGET_PRIVATE_H__
+
+#include "gtkfontchooserwidget.h"
+
+G_BEGIN_DECLS
+
+gboolean gtk_font_chooser_widget_handle_event (GtkWidget   *widget,
+                                               GdkEventKey *event);
+
+G_END_DECLS
+
+#endif /* __GTK_FONT_CHOOSER_WIDGET_PRIVATE_H__ */
