diff --git a/testsuite/gtk/Makefile.am b/testsuite/gtk/Makefile.am
index 99718b6..45eed14 100644
--- a/testsuite/gtk/Makefile.am
+++ b/testsuite/gtk/Makefile.am
@@ -157,9 +157,6 @@ keyhash_SOURCES	=		\
 gtkkeyhash.c: $(top_srcdir)/gtk/gtkkeyhash.c
 	$(AM_V_GEN) $(LN_S) $^ $@
 
-gtkresources.c: $(top_builddir)/gtk/gtkresources.c
-	$(AM_V_GEN) $(LN_S) $^ $@
-
 gtkprivate.c: $(top_srcdir)/gtk/gtkprivate.c
 	$(AM_V_GEN) $(LN_S) $^ $@
 
