--- projects/makefile/Makefile.orig	2011-12-25 23:15:38.397716914 +0100
+++ projects/makefile/Makefile	2011-12-25 23:15:52.212551519 +0100
@@ -1,10 +1,8 @@
-MAKE = make
-CC = gcc
 INSTALL_FILE = install -p -o root -g root -m 644
 INSTALL_DIR = install -p -o root -g root -d
 LN = ln -s
 RM = rm -fv
-CFLAGS += -c -O2 -Wall
+CFLAGS += -c
 LDFLAGS +=
 
 CFILES = image_DXT.c image_helper.c SOIL.c stb_image_aug.c
