--- gv-3.5.8.orig/doc/gv-1.html
+++ gv-3.5.8/doc/gv-1.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Synopsis </TITLE>
--- gv-3.5.8.orig/doc/gv-10.html
+++ gv-3.5.8/doc/gv-10.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Copyright </TITLE>
--- gv-3.5.8.orig/doc/gv-11.html
+++ gv-3.5.8/doc/gv-11.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Suggestions and bug reports </TITLE>
@@ -10,7 +11,7 @@
 <H2><A NAME="Suggestions_and_bug_reports"></A> <A NAME="s11">11. Suggestions and bug reports </A></H2>
 
 <P>Please send suggestions or bug reports to 
-<A HREF="mailto:plass@thep.physik.uni-mainz.de">plass@thep.physik.uni-mainz.de</A>.
+<A HREF="mailto:ckesselh@debian.org">ckesselh@debian.org</A>.
 </P>
 
 <HR>
--- gv-3.5.8.orig/doc/gv-12.html
+++ gv-3.5.8/doc/gv-12.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Acknowledgements </TITLE>
--- gv-3.5.8.orig/doc/gv-13.html
+++ gv-3.5.8/doc/gv-13.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: References </TITLE>
--- gv-3.5.8.orig/doc/gv-2.html
+++ gv-3.5.8/doc/gv-2.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Description </TITLE>
--- gv-3.5.8.orig/doc/gv-3.html
+++ gv-3.5.8/doc/gv-3.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Options </TITLE>
--- gv-3.5.8.orig/doc/gv-4.html
+++ gv-3.5.8/doc/gv-4.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Resources </TITLE>
@@ -328,13 +329,13 @@
 GV.gsArguments:
 GV.uncompressCommand:   gzip -d -c %s &gt; %s
 GV.printCommand:        lpr
-GV.scratchDir:          ~/
+GV.scratchDir:          /tmp/
 GV.defaultSaveDir:      ~/
 GV.fallbackPageMedia:   a4
 GV.useBackingPixmap:    True
 GV*dirs:                Home\n\ 
                         Tmp\n\ 
-                        /usr/doc\n\ 
+                        /usr/share/doc\n\ 
                         /usr/local/doc
 GV*filter:              no .*
 GV*filters:             None\n\ 
--- gv-3.5.8.orig/doc/gv-5.html
+++ gv-3.5.8/doc/gv-5.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Mouse and key bindings </TITLE>
--- gv-3.5.8.orig/doc/gv-6.html
+++ gv-3.5.8/doc/gv-6.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Comments about the user interface </TITLE>
--- gv-3.5.8.orig/doc/gv-8.html
+++ gv-3.5.8/doc/gv-8.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Availability </TITLE>
@@ -14,8 +15,6 @@
 <DL>
 <DT><B>World Wide Web:</B><DD><P>
 <A HREF="http://wwwthep.physik.uni-mainz.de/~plass/gv/">http://wwwthep.physik.uni-mainz.de/~plass/gv/</A></P>
-<DT><B>Anonymous ftp:</B><DD><P>
-<A HREF="ftp://thep.physik.uni-mainz.de/pub/gv/">ftp://thep.physik.uni-mainz.de/pub/gv/</A></P>
 </DL>
 
 </P>
--- gv-3.5.8.orig/doc/gv-9.html
+++ gv-3.5.8/doc/gv-9.html
@@ -1,3 +1,4 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997: Authors </TITLE>
--- gv-3.5.8.orig/doc/gv.html
+++ gv-3.5.8/doc/gv.html
@@ -1,20 +1,4 @@
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
 <HTML>
 <HEAD>
 <TITLE> gv 3.5, June 1997</TITLE>
--- gv-3.5.8.orig/doc/gv.man
+++ gv-3.5.8/doc/gv.man
@@ -238,7 +238,7 @@
 \fBa system specific resource file\fP.
 The name of this file is as described by the \*QXFILESEARCHPATH\*U
 environment variable, or, if this variable is not defined,
-implementation dependant. In most cases it is  
+implementation dependent. In most cases it is  
 \*Q/usr/X11/lib/X11/app-defaults/GV\*U.
 Administrators may want to modify this file in order
 to set up gv according to the local needs of their system.
@@ -538,7 +538,7 @@
 .nr bi 1
 .Pp
 Defines the paper-size to be used when automatic orientation
-detetction fails.  Valid values are 
+detection fails.  Valid values are 
 \*Qportrait\*U, \*Qlandscape\*U,\*Qseascape\*U 
 and \*Qupside-down\*U.
 .IP "medias"
@@ -552,7 +552,7 @@
 .nr bi 1
 .Pp
 Sets the paper size. 
-Allowd values are as given in the above list of paper-sizes.
+Allowed values are as given in the above list of paper-sizes.
 Specifying
 \*Qautomatic\*U causes gv to attempt
 to derive the correct paper-size from document structuring
@@ -561,7 +561,7 @@
 .nr bi 1
 .Pp
 Defines the paper-size to be used when automatic paper-size 
-detetction fails.  Valid values are as given in the  above
+detection fails.  Valid values are as given in the  above
 list of paper-sizes.
 .if \n(ll>1 .RE
 .nr ll -1
@@ -694,13 +694,13 @@
 GV.gsArguments:
 GV.uncompressCommand:   gzip -d -c %s > %s
 GV.printCommand:        lpr
-GV.scratchDir:          ~/
+GV.scratchDir:          /tmp/
 GV.defaultSaveDir:      ~/
 GV.fallbackPageMedia:   a4
 GV.useBackingPixmap:    True
 GV*dirs:                Home\\n\\
                         Tmp\\n\\
-                        /usr/doc\\n\\
+                        /usr/share/doc\\n\\
                         /usr/local/doc
 GV*filter:              no .*
 GV*filters:             None\\n\\
@@ -859,7 +859,7 @@
 Display the next page.
 .if \n(ll>1 .RE
 .nr ll -1
-.IP "\fBMouse bindings in the ">>" and "<<" button windows\fP"
+.IP "\fBMouse bindings in the "">>"" and ""<<"" button windows\fP"
 .nr bi 1
 .Pp
 The ">>" and "<<" buttons are used to move 
@@ -1172,10 +1172,6 @@
 .nr bi 1
 .Pp
 http://wwwthep.physik.uni-mainz.de/~plass/gv/
-.IP "Anonymous ftp:"
-.nr bi 1
-.Pp
-ftp://thep.physik.uni-mainz.de/pub/gv/
 .if \n(ll>1 .RE
 .nr ll -1
 
@@ -1207,7 +1203,7 @@
 .SH Suggestions and bug reports 
 
 .Pp
-Please send suggestions or bug reports to plass@thep.physik.uni-mainz.de.
+Please send suggestions or bug reports to ckesselh@debian.org.
 .Pp
 .Pp
 .Pp
--- gv-3.5.8.orig/doc/source/gv.doc
+++ gv-3.5.8/doc/source/gv.doc
@@ -757,7 +757,6 @@
 obtained from the following sources:
 <descrip>
 <tag/World Wide Web:/FTP_ANCHOR(http://wwwthep.physik.uni-mainz.de/&tilde;plass/gv/,http://wwwthep.physik.uni-mainz.de/&tilde;plass/gv/)
-<tag/Anonymous ftp:/FTP_ANCHOR(ftp://thep.physik.uni-mainz.de/pub/gv/,ftp://thep.physik.uni-mainz.de/pub/gv/)
 </descrip>
 <!--
 #####################################################
--- gv-3.5.8.orig/README
+++ gv-3.5.8/README
@@ -19,8 +19,6 @@
 
 The most recent version may be obtained from gv's homepage at
    http://wwwthep.physik.uni-mainz.de/~plass/gv/
-or via anonymous ftp from
-   ftp://thep.physik.uni-mainz.de/pub/gv/
 
 Please send suggestions or bug reports to plass@thep.physik.uni-mainz.de.
 
--- gv-3.5.8.orig/config.Unix
+++ gv-3.5.8/config.Unix
@@ -32,10 +32,10 @@
 XCOMM --------------------------------------------------------------------
 XCOMM  Destination directories.
 XCOMM --------------------------------------------------------------------
-      GV_BINDIR    = /usr/local/bin/
-      GV_DOCDIR    = /usr/local/doc/gv/
-      GV_LIBDIR    = /usr/local/lib/gv/
-      GV_MANDIR    = /usr/local/man/man1/
+      GV_BINDIR    = /usr/X11R6/bin/
+      GV_DOCDIR    = /usr/share/doc/gv/
+      GV_LIBDIR    = /usr/X11R6/lib/X11/gv/
+      GV_MANDIR    = /usr/X11R6/man/man1/
 
 XCOMM --------------------------------------------------------------------
 XCOMM --- The name of the Xaw3d library.
@@ -194,7 +194,7 @@
 XCOMM  Your temporary directory.
 XCOMM  Must be terminated with a trailing '/'.
 XCOMM --------------------------------------------------------------------
-      SCRATCH_DIR = ~/
+      SCRATCH_DIR = /tmp/
 
 XCOMM --------------------------------------------------------------------
 XCOMM  The default directory for saving.
--- gv-3.5.8.orig/source/Imakefile
+++ gv-3.5.8/source/Imakefile
@@ -291,7 +291,7 @@
 	@echo "GV.useBackingPixmap:	$(USE_BACKING_PIXMAP)"		>> $(GV_MAKE_RES_DAT)
 	@echo "GV*dirs:		Home\n\\"				>> $(GV_MAKE_RES_DAT)
 	@echo "			Tmp\n\\"				>> $(GV_MAKE_RES_DAT)
-	@echo "			/usr/doc\n\\"				>> $(GV_MAKE_RES_DAT)
+	@echo "			/usr/share/doc\n\\"			>> $(GV_MAKE_RES_DAT)
 	@echo "			/usr/local/doc"				>> $(GV_MAKE_RES_DAT)
 	@echo "GV*filter:		no .*"				>> $(GV_MAKE_RES_DAT)
 	@echo "GV*filters:		None\n\\"			>> $(GV_MAKE_RES_DAT)
@@ -311,8 +311,8 @@
 	@echo "!########## gv_intern_res.dat (generated by makefile)" >> $(GV_INTERN_RES_DAT)
 	@echo ""							>> $(GV_INTERN_RES_DAT)
 	@echo "GV.gsInterpreter:	gs" >> $(GV_INTERN_RES_DAT)
-	@echo "GV.gsCmdScanPDF:	gs -dNODISPLAY -dQUIET -sPDFname=%s -sDSCname=%s pdf2dsc.ps -c quit" >> $(GV_INTERN_RES_DAT)
-	@echo "GV.gsCmdConvPDF:	gs -dNODISPLAY -dQUIET $(PS_LEVEL) -dNOPAUSE -sPSFile=%s %s -c quit" >> $(GV_INTERN_RES_DAT)
+	@echo "GV.gsCmdScanPDF:	pdf2dsc %pdf %dsc" >> $(GV_INTERN_RES_DAT)
+	@echo "GV.gsCmdConvPDF:	pdf2ps $(PS_LEVEL) %pdf %ps" >> $(GV_INTERN_RES_DAT)
 	@echo "GV.gsX11Device:		-sDEVICE=x11" >> $(GV_INTERN_RES_DAT)
 	@echo "GV.gsX11AlphaDevice:	-dNOPLATFONTS -sDEVICE=x11alpha" >> $(GV_INTERN_RES_DAT)
 	@echo "GV.gsSafer:		True" >> $(GV_INTERN_RES_DAT)
--- gv-3.5.8.orig/source/Vlist.c
+++ gv-3.5.8/source/Vlist.c
@@ -623,6 +623,8 @@
     if (y < 0) entry = -1;
     else       entry = (vw->vlist.entries*y)/(int)vw->label.label_height;
   }
+  if (entry >= vw->vlist.entries)
+	  entry = vw->vlist.entries-1;
   IMESSAGE(entry)
   ENDMESSAGE(VlistEntryOfPosition)
   return(entry);
--- gv-3.5.8.orig/source/actions.c
+++ gv-3.5.8/source/actions.c
@@ -94,7 +94,7 @@
   }
   if (new_width && new_height) {
     if (width != new_width || height != new_height) {
-       misc_setPageMarker(0,2); /* bring selected in sight */
+       misc_setPageMarker(0,2,event,True); /* bring selected in sight */
        width = new_width;
        height=new_height;
     }
@@ -253,7 +253,7 @@
 	}
       }
       h = doc_putPageInRange(doc,h);
-      misc_setPageMarker(h,1);
+      misc_setPageMarker(h,1,event,False);
     }
 
     nx = x = (int)aaa->core.x;
@@ -428,7 +428,7 @@
     if (!scrolling && toc_text) {
       entry = VlistEntryOfPosition(newtoc,(int)event->xbutton.y);
       if (entry >=0 && entry < doc->numpages) {
-	misc_setPageMarker(entry,0);
+	misc_setPageMarker(entry,0,event,True);
         show_page(entry,NULL);
       }
     }
--- gv-3.5.8.orig/source/callbacks.c
+++ gv-3.5.8/source/callbacks.c
@@ -870,7 +870,7 @@
   char *s;
 
   BEGINMESSAGE(cb_page)
-  if (gv_psfile && client_data) {
+  if (gv_filename && client_data) {
     s = (char*)client_data;
     if (*s=='-' || *s=='+') {  
       k = 1;
@@ -894,7 +894,7 @@
 	np = np + ((k==1) ? cp : 0);
 	np=doc_putPageInRange(doc,np);
 	IIMESSAGE(np,current_page)
-	misc_setPageMarker(np,0);
+	misc_setPageMarker(np,0,NULL,True);
 	if (np != current_page) show_page(np,NULL);
       } else {
 	if (k==1 && np > 0) show_page(np,NULL);
--- gv-3.5.8.orig/source/config.h
+++ gv-3.5.8/source/config.h
@@ -40,7 +40,7 @@
 ----------------------------------------------------------
 */
 
-/* #define GV_RESTART_IF_BUSY */
+/*#define GV_RESTART_IF_BUSY*/
 
 /*
 ##########################################################
--- gv-3.5.8.orig/source/descrip.mms
+++ gv-3.5.8/source/descrip.mms
@@ -262,7 +262,7 @@
  @ WRITE_ file ""
  @ WRITE_ file "GV.gsInterpreter:	gs"
  @ WRITE_ file "GV.gsCmdScanPDF:	gs ""-dNODISPLAY"" ""-dQUIET"" ""-sPDFname""=%s ""-sDSCname""=%s pdf2dsc.ps -c quit"
- @ WRITE_ file "GV.gsCmdConvPDF:	gs ""-dNODISPLAY"" ""-dQUIET"" $(PS_LEVEL) ""-dNOPAUSE"" ""-sPSFile""=%s %s -c quit"
+ @ WRITE_ file "GV.gsCmdConvPDF:       gs ""-dNODISPLAY"" ""-dQUIET"" $(PS_LEVEL) ""-dNOPAUSE"" ""-sPSFile""=%s %s -c quit"
  @ WRITE_ file "GV.gsX11Device:	""-sDEVICE=x11"""
  @ WRITE_ file "GV.gsX11AlphaDevice:""-dNOPLATFONTS"" ""-sDEVICE=x11alpha"""
  @ WRITE_ file "GV.gsSafer:	True"
--- gv-3.5.8.orig/source/doc_misc.c
+++ gv-3.5.8/source/doc_misc.c
@@ -216,8 +216,8 @@
 	  (dm->width+1==dbb[URX] && dm->height+1==dbb[URY])) media=dm;
     for (j=0; gv_medias[j] && !media ; j++) {
        dm = gv_medias[j];
-       if ((dm->width==dbb[URX] && dm->height==dbb[URY]) ||
-	   (dm->width+1==dbb[URX] && dm->height+1==dbb[URY])) media=dm;
+       if (dm->used==1 && ((dm->width==dbb[URX] && dm->height==dbb[URY]) ||
+	   (dm->width+1==dbb[URX] && dm->height+1==dbb[URY]))) media=dm;
     }
     if (media) found=1;
   }
--- gv-3.5.8.orig/source/file.c
+++ gv-3.5.8/source/file.c
@@ -145,6 +145,25 @@
 }
 
 /*############################################################*/
+/* file_assureDirectory */
+/*############################################################*/
+
+void
+file_assureDirectory(to,from)
+   char *to;
+   char *from;
+{
+   int len;
+   BEGINMESSAGE(file_assureDirectory)
+   strcpy(to,from);
+#  ifndef VMS
+      len = strlen(to);
+      if (to[len-1] != '/') { to[len] = '/'; to[len+1] = '\0'; }
+#  endif
+   ENDMESSAGE(file_assureDirectory)
+}
+
+/*############################################################*/
 /* file_getTmpFilename */
 /* provide some temporary file name */
 /*############################################################*/
@@ -164,11 +183,13 @@
 
    BEGINMESSAGE(file_getTmpFilename)
 
-   if (!baseDirectory) baseDirectory = app_res.scratch_dir;
-   strcpy(tmpDirBuf,baseDirectory);
-   pos = file_locateFilename(tmpDirBuf);
-   if (pos) { ++pos; *pos='\0'; }
-   else strcpy(tmpDirBuf,app_res.scratch_dir);
+   pos = NULL;
+   if (baseDirectory) {
+     strcpy(tmpDirBuf,baseDirectory); 
+     pos = file_locateFilename(tmpDirBuf);
+   }
+   if (pos) *pos='\0';
+   else file_assureDirectory(tmpDirBuf,app_res.scratch_dir);
 
    if (!baseFilename) baseFilename= ".";
    strcpy(tmpNameBuf,baseFilename);
--- gv-3.5.8.orig/source/gv_misc_res.dat
+++ gv-3.5.8/source/gv_misc_res.dat
@@ -11,7 +11,7 @@
 
 !##### Version (do not modify this resource)
 
-GV.version:				gv 3.5.8
+GV.version:				gv 3.5.8 (debian)
 
 !##### Background, Foreground
 
@@ -525,9 +525,9 @@
  <LeaveWindow>:		GV_Toc(unhighlight)	\n\
  <Btn1Down>:		GV_Toc(scrollon)	\n\
  <Btn1Motion>:		GV_Toc(scroll)		\n\
- <Btn1Up>:		GV_Toc(select)		\
+ <Btn1Up>:		GV_Toc(highlight)	\
 			GV_Toc(scrolloff)	\
-			GV_Toc(highlight)	\n\
+			GV_Toc(select)		\n\
  <Btn2Down>:		GV_Toc(toggleon)	\n\
  <Btn2Motion>:		GV_Toc(toggleextend)	\
 			GV_Toc(highlight)	\n\
--- gv-3.5.8.orig/source/gv_user_res.dat
+++ gv-3.5.8/source/gv_user_res.dat
@@ -8,7 +8,7 @@
 GV.fallbackOrientation:	portrait
 GV.swapLandscape:	False
 GV.autoCenter:		True
-GV.antialias:		False
+GV.antialias:		True
 GV.respectDSC:		True
 GV.ignoreEOF:		True
 GV.confirmPrint:	True
--- gv-3.5.8.orig/source/main.c
+++ gv-3.5.8/source/main.c
@@ -985,9 +985,9 @@
   main_setInternResource(db,&gv_gs_x11_device,"gsX11Device");
   main_setInternResource(db,&gv_gs_x11_alpha_device,"gsX11AlphaDevice");
   main_setInternResource(db,&gv_gs_arguments,"gsArguments");
-  s = resource_getResource(db,gv_class,gv_class,"gsSafer",NULL);
+  s = resource_getResource(db,gv_class,gv_name,"gsSafer",NULL);
   if (!strcasecmp(s,"true"))  gv_gs_safer = 1; else gv_gs_safer = 0;
-  s = resource_getResource(db,gv_class,gv_class,"gsQuiet",NULL);
+  s = resource_getResource(db,gv_class,gv_name,"gsQuiet",NULL);
   if (!strcasecmp(s,"true"))  gv_gs_quiet = 1; else gv_gs_quiet = 0;
   ENDMESSAGE(main_setGhostscriptResources)
 }
--- gv-3.5.8.orig/source/main_resources.h
+++ gv-3.5.8/source/main_resources.h
@@ -285,10 +285,10 @@
  { "-spartan"		, ".style"			, XrmoptionNoArg	, "gv_spartan.dat"},
  { "-pixmap"		, ".useBackingPixmap"		, XrmoptionNoArg	, "True"	},
  { "-nopixmap"		, ".useBackingPixmap"		, XrmoptionNoArg	, "False"	},
- { "-quiet"		, ".quiet"    		        , XrmoptionNoArg	, "True"	},
- { "-noquiet"		, ".quiet"    		        , XrmoptionNoArg	, "False"	},
- { "-safer"		, ".safer"    		        , XrmoptionNoArg	, "True"	},
- { "-nosafer"		, ".safer"    		        , XrmoptionNoArg	, "False"	},
+ { "-quiet"		, ".gsQuiet"   		        , XrmoptionNoArg	, "True"	},
+ { "-noquiet"		, ".gsQuiet"   		        , XrmoptionNoArg	, "False"	},
+ { "-safer"		, ".gsSafer"   		        , XrmoptionNoArg	, "True"	},
+ { "-nosafer"		, ".gsSafer"  		        , XrmoptionNoArg	, "False"	},
  { "-arguments"		, ".arguments"			, XrmoptionSepArg	, NULL		},
  { "-h"		        , ".help"			, XrmoptionNoArg	, "?"		},
  { "-?"		        , ".help"			, XrmoptionNoArg	, "?"		},
--- gv-3.5.8.orig/source/misc.c
+++ gv-3.5.8/source/misc.c
@@ -301,9 +301,11 @@
 /*############################################################*/
 
 void
-misc_setPageMarker(entry,kind)
+misc_setPageMarker(entry,kind,event,check_toc)
   int entry;
   int kind; /* 0 = selected, 1 = highlighted , 2 = bring selected in sight*/
+  XEvent *event;
+  Boolean check_toc;
 {
   int x,y,yl,yu,ny=99999;
   Boolean b = False;
@@ -330,9 +332,12 @@
 	ny = (int)newtocClip->core.height - ((int)newtoc->core.y + yl + 14);
         if (y>ny) b = True;
       }
-      if (b) {
+      if (event && b) {
 	INFIMESSAGE(jumping to,ny)
 	ClipWidgetSetCoordinates(newtocClip,x,ny);
+	entry = VlistEntryOfPosition(newtoc,(int)event->xbutton.y+(y-ny));
+	if (entry != VlistHighlighted(newtoc) && check_toc)
+		VlistChangeHighlighted(newtoc,entry,XawVlistSet);
       }
     }
   }
@@ -689,7 +694,7 @@
        }
        if (toc_text) {
 	 INFMESSAGE(marking current_page as current)
-         misc_setPageMarker(current_page,0);
+         misc_setPageMarker(current_page,0,NULL,True);
        }
     }
 
@@ -1131,7 +1136,11 @@
 	}
 	s[i] = '\0';
 							n=0;
-	XtSetArg(args[n], XtNvlist, s);			n++;
+	if (	(!gv_filename_old)			||
+		(!olddoc)				||
+		(strcmp(gv_filename_old, gv_filename))	||
+		(doc->numpages != olddoc->numpages)	)
+		  XtSetArg(args[n], XtNvlist, s);	n++;
 	XtSetArg(args[n], XtNlabel, toc_text);		n++;
       } else {
 	s = NULL;
@@ -1154,7 +1163,7 @@
     misc_setSensitive(w_printAllPages     , show_printAllPages     , (gv_psfile   != NULL));
     misc_setSensitive(w_checkFile         , show_checkFile         , (gv_filename != NULL));
     misc_setSensitive(w_updateFile        , show_updateFile        , (gv_filename != NULL));
-    misc_setSensitive(w_showThisPage      , show_showThisPage      , (gv_filename != NULL));
+    misc_setSensitive(w_showThisPage      , show_showThisPage      , (gv_psfile   != NULL));
     misc_setSensitive(w_prevPage          , show_prevPage          , (toc_text    != NULL));
     misc_setSensitive(w_nextPage          , show_nextPage          , (gv_filename != NULL));
     misc_setSensitive(w_toggleCurrentPage , show_toggleCurrentPage , (toc_text    != NULL));
@@ -1168,7 +1177,7 @@
     XtSetSensitive(saveAllEntry,     (gv_psfile   != NULL));
     XtSetSensitive(saveMarkedEntry,  (toc_text    != NULL));
     XtSetSensitive(nextEntry,        (gv_filename != NULL));
-    XtSetSensitive(redisplayEntry,   (gv_filename != NULL));
+    XtSetSensitive(redisplayEntry,   (gv_psfile   != NULL));
     XtSetSensitive(prevEntry,        (toc_text    != NULL));
     XtSetSensitive(currentEntry,     (toc_text    != NULL));
     XtSetSensitive(oddEntry,         (toc_text    != NULL));
@@ -1718,3 +1727,51 @@
     ENDMESSAGE(catch_Xerror)
     return 0;
 }
+
+/*############################################################*/
+/* quote_filename */
+/* Quotes special characters in filenames */
+/* (taken from bash sources) */
+/*############################################################*/
+
+char *
+quote_filename (string) 
+     char *string;
+{
+    int c;
+    char *result, *r, *s;
+
+    BEGINMESSAGE(quote_filename)
+
+    result = (char*) GV_XtMalloc((2 * strlen (string) + 1) * sizeof(char));
+
+    for (r = result, s = string; s && (c = *s); s++)
+    {
+      switch (c)
+	{
+	case ' ': case '\t': case '\n':		/* IFS white space */
+	case '\'': case '"': case '\\':		/* quoting chars */
+	case '|': case '&': case ';':		/* shell metacharacters */
+	case '(': case ')': case '<': case '>':
+	case '!': case '{': case '}':		/* reserved words */
+	case '*': case '[': case '?': case ']':	/* globbing chars */
+	case '^':
+	case '$': case '`':			/* expansion chars */
+	  *r++ = '\\';
+	  *r++ = c;
+	  break;
+	case '#':				/* comment char */
+	  if (s == string)
+	    *r++ = '\\';
+	  /* FALLTHROUGH */
+	default:
+	  *r++ = c;
+	  break;
+	}
+    }
+    *r = '\0';
+
+    ENDMESSAGE(quote_filename)
+
+    return (result);
+}
--- gv-3.5.8.orig/source/misc.h
+++ gv-3.5.8/source/misc.h
@@ -57,7 +57,9 @@
 extern void			misc_setPageMarker (
 #if NeedFunctionPrototypes
     int, /* entry */
-    int	 /* kind */
+    int, /* kind */
+    XEvent*,
+    Boolean
 #endif
 );
 
@@ -115,6 +117,12 @@
 #endif
 );
  
+extern char *                   quote_filename (
+#if NeedFunctionPrototypes
+    char*
+#endif
+);
+
 #endif	/* _GV_MISC_H_ */
 
 
--- gv-3.5.8.orig/source/miscmenu.c
+++ gv-3.5.8/source/miscmenu.c
@@ -60,7 +60,7 @@
 
 static MiscMenuEntryStruct miscmenu_entries[] = {
   { "update",cb_checkFile,(XtPointer)CHECK_FILE_DATE,2 },
-  { "redisplay",cb_redisplay,NULL,2 },
+  { "redisplay",cb_redisplay,NULL,3 },
   { "toggle_current" , cb_setPageMark, (XtPointer)(SPM_CURRENT|SPM_TOGGLE),1 },
   { "toggle_even"    , cb_setPageMark, (XtPointer)(SPM_EVEN|SPM_TOGGLE),1 },
   { "toggle_odd"     , cb_setPageMark, (XtPointer)(SPM_ODD|SPM_TOGGLE),1 },
@@ -123,7 +123,7 @@
     if (nl) *nl='\0';
     name[0]='\0';
     if (*c == '#' || *c == '!') i=0;
-    else i=sscanf(c," %s ",name);
+    else i=sscanf(c," %99s ",name);
     if (i==1) {
       for (j=0; miscmenu_entries[j].name; j++)
 	if (!strcasecmp(miscmenu_entries[j].name,name)) { 
--- gv-3.5.8.orig/source/options_fs.c
+++ gv-3.5.8/source/options_fs.c
@@ -105,7 +105,6 @@
    Widget	w;
    XtPointer	client_data, call_data;
 {
-   BEGINMESSAGE(options_fs_cb_apply)
    Arg args[5];
    Cardinal n;
    static Boolean s_scratch_dir = False;
--- gv-3.5.8.orig/source/paths.h
+++ gv-3.5.8/source/paths.h
@@ -34,9 +34,9 @@
 #   define INC_XMU(aaa) <XMU_DIRECTORY/aaa>
 #   define INC_XAW(aaa) <XAW_DIRECTORY/aaa>
 #else
-#   define INC_X11(aaa) <X11/##aaa##>
-#   define INC_XMU(aaa) <X11/Xmu/##aaa##>
-#   define INC_XAW(aaa) <X11/Xaw3d/##aaa##>
+#   define INC_X11(aaa) <X11/aaa>
+#   define INC_XMU(aaa) <X11/Xmu/aaa>
+#   define INC_XAW(aaa) <X11/Xaw3d/aaa>
 #endif
 
 #endif /* _PATHS_H_ */
--- gv-3.5.8.orig/source/process.c
+++ gv-3.5.8/source/process.c
@@ -272,8 +272,7 @@
    pid = fork();
 
    if (pid == 0) { /* child */
-      int  argc=0;
-      char *argv[20];
+      char *argv[4];
       char *c;
 
       INFMESSAGE(child process)
@@ -286,15 +285,10 @@
        */
       system(c);
 #else
-      while (isspace(*c)) c++;
-      while (*c) {
-         argv[argc++] = c;
-         while (*c && !isspace(*c)) c++;
-         if (*c) *c++ = '\0';
-         while (isspace(*c)) c++;
-         SMESSAGE(argv[argc-1])
-      }
-      argv[argc] = NULL;
+      argv[0] = "sh";
+      argv[1] = "-c";
+      argv[2] = c;
+      argv[3] = NULL;
 
       INFMESSAGE(spawning conversion process)
 /*
--- gv-3.5.8.orig/source/ps.c
+++ gv-3.5.8/source/ps.c
@@ -93,10 +93,35 @@
 #define memset(a,b,c) bzero(a,c)
 #endif
 
+/* We use this helper function for providing proper */
+/* case and colon :-) insensitive DSC matching */
+static int dsc_strncmp(s1, s2, n)
+  char *s1;
+  char *s2;
+  size_t n;
+{
+ char *tmp;	
+
+ if (strncasecmp(s1, s2, n) == 0)
+	 return 0;
+ if (s2[n-1] == ':'){
+	 tmp = (char *) PS_malloc(n*sizeof(char));
+	 strncpy(tmp, s2, (n-1));
+	 tmp[n-1]=' ';
+	 if (strncasecmp(s1, tmp, n) == 0){
+		 PS_free(tmp);
+		 return 0;
+	 }
+	 PS_free(tmp);
+ }
+ 
+ return 1;
+}
+
 /* length calculates string length at compile time */
 /* can only be used with character constants */
 #define length(a)       (sizeof((a))-1)
-#define iscomment(a, b) (strncmp((a), (b), length((b))) == 0)
+#define iscomment(a, b) (dsc_strncmp((a), (b), length((b))) == 0)
 #define DSCcomment(a)   ((a)[0] == '%' && (a)[1] == '%')
 
 /* list of standard paper sizes from Adobe's PPD. */
@@ -417,6 +417,7 @@
       struct document *retval = NULL;
       FILE *tmpfile = (FILE*)NULL;
       char *filename_unc;
+      char *quoted_filename, *quoted_filename_unc;
       char cmd[512];
       char s[512];
       filename_unc=file_getTmpFilename(NULL,filename_raw);
@@ -430,7 +431,11 @@
 	  ENDMESSAGE(psscan)
 	  return(NULL);
 	}
-      sprintf(cmd,cmd_uncompress,filename,filename_unc);
+      quoted_filename = quote_filename(filename);
+      quoted_filename_unc = quote_filename(filename_unc);
+      sprintf(cmd,cmd_uncompress,quoted_filename,quoted_filename_unc);
+      GV_XtFree(quoted_filename);
+      GV_XtFree(quoted_filename_unc);
       INFMESSAGE(is compressed)
       INFSMESSAGE(uncompress command,cmd)
       if (ps_system(cmd) || file_fileIsNotUseful(filename_unc)) {
@@ -478,7 +508,7 @@
       doc = (struct document *) PS_malloc(sizeof(struct document));
       CHECK_MALLOCED(doc);
       memset(doc, 0, sizeof(struct document));
-      sscanf(line, "%*s %s", text);
+      sscanf(line, "%*s %256s", text);
       /*###jp###*/
       /*doc->epsf = iscomment(text, "EPSF-");*/
       doc->epsf = iscomment(text, "EPSF");
@@ -498,6 +498,9 @@
       struct document *retval = NULL;
       FILE *tmpfile = (FILE*)NULL;
       char *filename_dsc;
+      char *quoted_filename, *quoted_filename_dsc;
+      char *pdfpos;
+      char *dscpos;
       char cmd[512];
       char s[512];
       filename_dsc=file_getTmpFilename(NULL,filename_raw);
@@ -511,7 +514,29 @@
 	  ENDMESSAGE(psscan)
 	  return(NULL);
 	}
-      sprintf(cmd,cmd_scan_pdf,filename,filename_dsc);
+      quoted_filename = quote_filename(filename);
+      quoted_filename_dsc = quote_filename(filename_dsc);
+      if ((pdfpos = strstr(cmd_scan_pdf,"%pdf")) &&
+         (dscpos = strstr(cmd_scan_pdf,"%dsc"))) {
+       cmd[0] = '\0';
+       if (pdfpos < dscpos) {
+         strncat(cmd,cmd_scan_pdf,(pdfpos-cmd_scan_pdf));
+         strcat(cmd,quoted_filename);
+         strncat(cmd,pdfpos+4,(dscpos-pdfpos-4));
+         strcat(cmd,quoted_filename_dsc);
+         strcat(cmd,dscpos+4);
+       } else {
+         strncat(cmd,cmd_scan_pdf,(dscpos-cmd_scan_pdf));
+         strcat(cmd,quoted_filename_dsc);
+         strncat(cmd,dscpos+4,(pdfpos-dscpos-4));
+         strcat(cmd,quoted_filename);
+         strcat(cmd,pdfpos+4);
+       }
+      } else {
+       sprintf(cmd,cmd_scan_pdf,quoted_filename,quoted_filename_dsc);
+      }
+      GV_XtFree(quoted_filename);
+      GV_XtFree(quoted_filename_dsc);
       INFMESSAGE(is PDF)
       INFSMESSAGE(scan command,cmd)
 #ifdef VMS
@@ -546,7 +601,7 @@
 	} else if (doc->date == NULL && iscomment(line+2, "CreationDate:")) {
 	    doc->date = gettextline(line+length("%%CreationDate:"));
 	} else if (bb_set == NONE && iscomment(line+2, "BoundingBox:")) {
-	    sscanf(line+length("%%BoundingBox:"), "%s", text);
+	    sscanf(line+length("%%BoundingBox:"), "%256s", text);
 	    if (strcmp(text, "(atend)") == 0) {
 		bb_set = ATEND;
 	    } else {
@@ -578,7 +633,7 @@
 	    }
 	} else if (orientation_set == NONE &&
 		   iscomment(line+2, "Orientation:")) {
-	    sscanf(line+length("%%Orientation:"), "%s", text);
+	    sscanf(line+length("%%Orientation:"), "%256s", text);
 	    if (strcmp(text, "(atend)") == 0) {
 		orientation_set = ATEND;
 	    } else if (strcmp(text, "Portrait") == 0) {
@@ -589,7 +644,7 @@
 		orientation_set = 1;
 	    }
 	} else if (page_order_set == NONE && iscomment(line+2, "PageOrder:")) {
-	    sscanf(line+length("%%PageOrder:"), "%s", text);
+	    sscanf(line+length("%%PageOrder:"), "%256s", text);
 	    if (strcmp(text, "(atend)") == 0) {
 		page_order_set = ATEND;
 	    } else if (strcmp(text, "Ascend") == 0) {
@@ -603,7 +658,7 @@
 		page_order_set = 1;
 	    }
 	} else if (pages_set == NONE && iscomment(line+2, "Pages:")) {
-	    sscanf(line+length("%%Pages:"), "%s", text);
+	    sscanf(line+length("%%Pages:"), "%256s", text);
 	    if (strcmp(text, "(atend)") == 0) {
 		pages_set = ATEND;
 	    } else {
@@ -823,7 +878,7 @@
 		/* Do nothing */
 	    } else if (doc->default_page_orientation == NONE &&
 		iscomment(line+2, "PageOrientation:")) {
-		sscanf(line+length("%%PageOrientation:"), "%s", text);
+		sscanf(line+length("%%PageOrientation:"), "%256s", text);
 		if (strcmp(text, "Portrait") == 0) {
 		    doc->default_page_orientation = PORTRAIT;
 		} else if (strcmp(text, "Landscape") == 0) {
@@ -945,7 +1000,7 @@
 		/* Do nothing */
 	    } else if (doc->default_page_orientation == NONE &&
 		iscomment(line+2, "PageOrientation:")) {
-		sscanf(line+length("%%PageOrientation:"), "%s", text);
+		sscanf(line+length("%%PageOrientation:"), "%256s", text);
 		if (strcmp(text, "Portrait") == 0) {
 		    doc->default_page_orientation = PORTRAIT;
 		} else if (strcmp(text, "Landscape") == 0) {
@@ -1082,7 +1137,7 @@
 		/* Do nothing */
 	    } else if (doc->pages[doc->numpages].orientation == NONE &&
 		iscomment(line+2, "PageOrientation:")) {
-		sscanf(line+length("%%PageOrientation:"), "%s", text);
+		sscanf(line+length("%%PageOrientation:"), "%256s", text);
 		if (strcmp(text, "Portrait") == 0) {
 		    doc->pages[doc->numpages].orientation = PORTRAIT;
 		} else if (strcmp(text, "Landscape") == 0) {
@@ -1114,7 +1169,7 @@
 		PS_free(cp);
 	    } else if ((page_bb_set == NONE || page_bb_set == ATEND) &&
 		       iscomment(line+2, "PageBoundingBox:")) {
-		sscanf(line+length("%%PageBoundingBox:"), "%s", text);
+		sscanf(line+length("%%PageBoundingBox:"), "%256s", text);
 		if (strcmp(text, "(atend)") == 0) {
 		    page_bb_set = ATEND;
 		} else {
@@ -1228,14 +1283,14 @@
 	    }
 	} else if (orientation_set == ATEND &&
 		   iscomment(line+2, "Orientation:")) {
-	    sscanf(line+length("%%Orientation:"), "%s", text);
+	    sscanf(line+length("%%Orientation:"), "%256s", text);
 	    if (strcmp(text, "Portrait") == 0) {
 		doc->orientation = PORTRAIT;
 	    } else if (strcmp(text, "Landscape") == 0) {
 		doc->orientation = LANDSCAPE;
 	    }
 	} else if (page_order_set == ATEND && iscomment(line+2, "PageOrder:")) {
-	    sscanf(line+length("%%PageOrder:"), "%s", text);
+	    sscanf(line+length("%%PageOrder:"), "%256s", text);
 	    if (strcmp(text, "Ascend") == 0) {
 		doc->pageorder = ASCEND;
 	    } else if (strcmp(text, "Descend") == 0) {
@@ -1738,6 +1793,8 @@
            (DSCcomment(line) && iscomment(line+2,(comment)))
 #define IS_BEGIN(comment)				\
            (iscomment(line+7,(comment)))
+#define IS_END(comment)				\
+           (iscomment(line+5,(comment)))
 #define SKIP_WHILE(cond)				\
 	   while (readline(fd, &line, NULL, &nbytes) && (cond)) *line_lenP += nbytes;\
            skipped=1;
@@ -1773,7 +1830,12 @@
    else
 #endif
    if  (!IS_COMMENT("Begin"))     {} /* Do nothing */
-   else if IS_BEGIN("Document:")  SKIP_UNTIL_1("EndDocument")
+   else if IS_BEGIN("Document:")  {  /* Skip the EPS without handling its content */
+            while (line && !IS_END("Document")) {
+               line = ps_io_fgetchars(fd,-1);
+               if (line) *line_lenP += FD_LINE_LEN;
+            }
+   }
    else if IS_BEGIN("Feature:")   SKIP_UNTIL_1("EndFeature")
 #ifdef USE_ACROREAD_WORKAROUND
    else if IS_BEGIN("File")       SKIP_UNTIL_2("EndFile","EOF")
@@ -1789,7 +1851,7 @@
       INFMESSAGE(encountered "BeginData:")
       if (FD_LINE_LEN > 100) FD_BUF[100] = '\0';
       text[0] = '\0';
-      if (sscanf(line+length("%%BeginData:"), "%d %*s %s", &num, text) >= 1) {
+      if (sscanf(line+length("%%BeginData:"), "%d %*s %100s", &num, text) >= 1) {
          if (strcmp(text, "Lines") == 0) {
             INFIMESSAGE(number of lines to skip:,num)
             while (num) {
@@ -1888,7 +1950,7 @@
          INFMESSAGE(encountered "BeginData:")
          if (FD_LINE_LEN > 100) FD_BUF[100] = '\0';
          text[0] = '\0';
-         if (sscanf(line+length("%%BeginData:"), "%d %*s %s", &num, text) >= 1) {
+         if (sscanf(line+length("%%BeginData:"), "%d %*s %100s", &num, text) >= 1) {
             if (strcmp(text, "Lines") == 0) {
                INFIMESSAGE(number of lines:,num)
                while (num) {
@@ -1985,7 +2047,7 @@
           PS_free(comment);
           continue;
        }
-       sscanf(comment+length("%%Pages:"), "%s", text);
+       sscanf(comment+length("%%Pages:"), "%256s", text);
        if (strcmp(text, "(atend)") == 0) {
           fputs(comment, dest_file);
           pages_atend = True;
--- gv-3.5.8.orig/source/save.c
+++ gv-3.5.8/source/save.c
@@ -112,17 +112,19 @@
    String print_filename;
 {
    String error=NULL;
+   char *print_quoted_filename;
    char *c,*p;
    Cardinal m,n;
    String printfail=GV_ERROR_PRINT_FAIL;
 
    BEGINMESSAGE(print_file)
 
+   print_quoted_filename = quote_filename(print_filename);
    p = GV_XtNewString(print_command);
    n=0;
    c=p;
    while ((c=strstr(c,"%s"))) { c+=2; n++; }
-   m = (strlen(p)+(n>0?n:1)*strlen(print_filename)+5)*sizeof(char);
+   m = (strlen(p)+(n>0?n:1)*strlen(print_quoted_filename)+5)*sizeof(char);
    c = (char*) GV_XtMalloc(m);
    if (n>0) {
      char *e,*s;
@@ -133,13 +135,13 @@
        if (s) *s='\0';
        strcat(c,e);
        if (s) {
-	 strcat(c,print_filename);
+	 strcat(c,print_quoted_filename);
          e=s+2;
        } 
        else s=NULL;
      }
    } else {
-     sprintf(c, "%s %s",p,print_filename);
+     sprintf(c, "%s %s",p,print_quoted_filename);
    }
    INFSMESSAGE(printing:,c)
    if (SYSTEM_FAILED_ON(c)) {
@@ -149,6 +151,7 @@
    }
    GV_XtFree(c);
    GV_XtFree(p);
+   GV_XtFree(print_quoted_filename);
    ENDMESSAGE(print_file)
    return(error);
 }
@@ -198,6 +201,9 @@
    char proc_name[256];
    char *error=NULL;
    char *pos;
+   char *pdfpos;
+   char *pspos;
+   char *quoted_src_fn, *quoted_conv_fn;
 
    BEGINMESSAGE(save_forkPDFToPSConversion)
 
@@ -205,7 +211,30 @@
    strcpy(proc_name,pos);
    strcat(proc_name," conversion");
 
-   sprintf(command,gv_gs_cmd_conv_pdf,sd->conv_fn,sd->src_fn);
+   quoted_src_fn = quote_filename(sd->src_fn);
+   quoted_conv_fn = quote_filename(sd->conv_fn);
+   if ((pdfpos = strstr(gv_gs_cmd_conv_pdf,"%pdf")) &&
+       (pspos = strstr(gv_gs_cmd_conv_pdf,"%ps"))) {
+     command[0] = '\0';
+     if (pdfpos < pspos) {
+       strncat(command,gv_gs_cmd_conv_pdf,(pdfpos-gv_gs_cmd_conv_pdf));
+       strcat(command,quoted_src_fn);
+       strncat(command,pdfpos+4,(pspos-pdfpos-4));
+       strcat(command,quoted_conv_fn);
+       strcat(command,pspos+3);
+     } else {
+       strncat(command,gv_gs_cmd_conv_pdf,(pspos-gv_gs_cmd_conv_pdf));
+       strcat(command,quoted_conv_fn);
+       strncat(command,pspos+3,(pdfpos-pspos-3));
+       strcat(command,quoted_src_fn);
+       strcat(command,pdfpos+4);
+     }
+   } else {
+     sprintf(command,gv_gs_cmd_conv_pdf,quoted_conv_fn,quoted_src_fn);
+   }
+   GV_XtFree(quoted_src_fn);
+   GV_XtFree(quoted_conv_fn);
+
    INFSMESSAGE(starting conversion:,command)
    process_fork(proc_name,command,save_forkPDFToPSConversionDone,(XtPointer)sd);
    ENDMESSAGE(save_forkPDFToPSConversion)
--- gv-3.5.8.orig/source/versionp.h
+++ gv-3.5.8/source/versionp.h
@@ -35,8 +35,8 @@
 #define  VERSION_APPLIC_CONTEXT	app_con  			/* the Application Context */
 
 char *versionIdentification[] = {
-        "gv 3.5.8",
-        "June 1997",
+        "gv 3.5.8 (debian)",
+        "February 2002",
 	0
 };
 
@@ -84,7 +84,7 @@
       FAX: (608)262-9777        Madison, WI   53706\
 ",
 "\
- Please send error reports to:  plass@thep.physik.uni-mainz.de \n\
+ Please send error reports to:  ckesselh@debian.org \n\
 ",
 0 
 };
