diff -u swh-plugins-0.4.15+1.orig/alias_1407.c swh-plugins-0.4.15+1/alias_1407.c
--- swh-plugins-0.4.15+1.orig/alias_1407.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/alias_1407.c	2012-05-13 01:06:55.846742739 +0200
@@ -168,7 +168,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/alias_1407.so.c swh-plugins-0.4.15+1/alias_1407.so.c
--- swh-plugins-0.4.15+1.orig/alias_1407.so.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/alias_1407.so.c	2012-05-13 01:06:55.850742753 +0200
@@ -168,7 +168,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/allpass_1895.c swh-plugins-0.4.15+1/allpass_1895.c
--- swh-plugins-0.4.15+1.orig/allpass_1895.c	2012-05-13 01:05:58.094460405 +0200
+++ swh-plugins-0.4.15+1/allpass_1895.c	2012-05-13 01:06:55.866742832 +0200
@@ -1124,7 +1124,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/allpass_1895.so.c swh-plugins-0.4.15+1/allpass_1895.so.c
--- swh-plugins-0.4.15+1.orig/allpass_1895.so.c	2012-05-13 01:05:58.214460980 +0200
+++ swh-plugins-0.4.15+1/allpass_1895.so.c	2012-05-13 01:06:55.882742910 +0200
@@ -1124,7 +1124,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/amp_1181.c swh-plugins-0.4.15+1/amp_1181.c
--- swh-plugins-0.4.15+1.orig/amp_1181.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/amp_1181.c	2012-05-13 01:06:55.886742929 +0200
@@ -161,7 +161,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/amp_1181.so.c swh-plugins-0.4.15+1/amp_1181.so.c
--- swh-plugins-0.4.15+1.orig/amp_1181.so.c	2012-05-13 01:05:58.166460749 +0200
+++ swh-plugins-0.4.15+1/amp_1181.so.c	2012-05-13 01:06:55.886742929 +0200
@@ -161,7 +161,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/am_pitchshift_1433.c swh-plugins-0.4.15+1/am_pitchshift_1433.c
--- swh-plugins-0.4.15+1.orig/am_pitchshift_1433.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/am_pitchshift_1433.c	2012-05-13 01:06:55.894742967 +0200
@@ -366,7 +366,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/am_pitchshift_1433.so.c swh-plugins-0.4.15+1/am_pitchshift_1433.so.c
--- swh-plugins-0.4.15+1.orig/am_pitchshift_1433.so.c	2012-05-13 01:05:58.218461009 +0200
+++ swh-plugins-0.4.15+1/am_pitchshift_1433.so.c	2012-05-13 01:06:55.898742980 +0200
@@ -366,7 +366,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/analogue_osc_1416.c swh-plugins-0.4.15+1/analogue_osc_1416.c
--- swh-plugins-0.4.15+1.orig/analogue_osc_1416.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/analogue_osc_1416.c	2012-05-13 01:06:55.902743005 +0200
@@ -304,7 +304,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/analogue_osc_1416.so.c swh-plugins-0.4.15+1/analogue_osc_1416.so.c
--- swh-plugins-0.4.15+1.orig/analogue_osc_1416.so.c	2012-05-13 01:05:58.214460980 +0200
+++ swh-plugins-0.4.15+1/analogue_osc_1416.so.c	2012-05-13 01:06:55.906743040 +0200
@@ -304,7 +304,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/bandpass_a_iir_1893.c swh-plugins-0.4.15+1/bandpass_a_iir_1893.c
--- swh-plugins-0.4.15+1.orig/bandpass_a_iir_1893.c	2012-05-13 01:05:58.206460944 +0200
+++ swh-plugins-0.4.15+1/bandpass_a_iir_1893.c	2012-05-13 01:06:55.910743046 +0200
@@ -203,7 +203,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/bandpass_a_iir_1893.so.c swh-plugins-0.4.15+1/bandpass_a_iir_1893.so.c
--- swh-plugins-0.4.15+1.orig/bandpass_a_iir_1893.so.c	2012-05-13 01:05:58.230461069 +0200
+++ swh-plugins-0.4.15+1/bandpass_a_iir_1893.so.c	2012-05-13 01:06:55.914743062 +0200
@@ -203,7 +203,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/bandpass_iir_1892.c swh-plugins-0.4.15+1/bandpass_iir_1892.c
--- swh-plugins-0.4.15+1.orig/bandpass_iir_1892.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/bandpass_iir_1892.c	2012-05-13 01:06:55.918743082 +0200
@@ -258,7 +258,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/bandpass_iir_1892.so.c swh-plugins-0.4.15+1/bandpass_iir_1892.so.c
--- swh-plugins-0.4.15+1.orig/bandpass_iir_1892.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/bandpass_iir_1892.so.c	2012-05-13 01:06:55.922743110 +0200
@@ -258,7 +258,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/bode_shifter_1431.c swh-plugins-0.4.15+1/bode_shifter_1431.c
--- swh-plugins-0.4.15+1.orig/bode_shifter_1431.c	2012-05-13 01:05:58.210460960 +0200
+++ swh-plugins-0.4.15+1/bode_shifter_1431.c	2012-05-13 01:06:55.930743143 +0200
@@ -372,7 +372,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/bode_shifter_1431.so.c swh-plugins-0.4.15+1/bode_shifter_1431.so.c
--- swh-plugins-0.4.15+1.orig/bode_shifter_1431.so.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/bode_shifter_1431.so.c	2012-05-13 01:06:55.934743164 +0200
@@ -372,7 +372,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/bode_shifter_cv_1432.c swh-plugins-0.4.15+1/bode_shifter_cv_1432.c
--- swh-plugins-0.4.15+1.orig/bode_shifter_cv_1432.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/bode_shifter_cv_1432.c	2012-05-13 01:06:55.938743191 +0200
@@ -400,7 +400,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/bode_shifter_cv_1432.so.c swh-plugins-0.4.15+1/bode_shifter_cv_1432.so.c
--- swh-plugins-0.4.15+1.orig/bode_shifter_cv_1432.so.c	2012-05-13 01:05:58.202460932 +0200
+++ swh-plugins-0.4.15+1/bode_shifter_cv_1432.so.c	2012-05-13 01:06:55.946743219 +0200
@@ -400,7 +400,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/butterworth_1902.c swh-plugins-0.4.15+1/butterworth_1902.c
--- swh-plugins-0.4.15+1.orig/butterworth_1902.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/butterworth_1902.c	2012-05-13 01:06:55.954743268 +0200
@@ -523,7 +523,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/butterworth_1902.so.c swh-plugins-0.4.15+1/butterworth_1902.so.c
--- swh-plugins-0.4.15+1.orig/butterworth_1902.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/butterworth_1902.so.c	2012-05-13 01:06:55.962743299 +0200
@@ -523,7 +523,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/chebstortion_1430.c swh-plugins-0.4.15+1/chebstortion_1430.c
--- swh-plugins-0.4.15+1.orig/chebstortion_1430.c	2012-05-13 01:05:58.110460476 +0200
+++ swh-plugins-0.4.15+1/chebstortion_1430.c	2012-05-13 01:06:55.966743321 +0200
@@ -321,7 +321,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/chebstortion_1430.so.c swh-plugins-0.4.15+1/chebstortion_1430.so.c
--- swh-plugins-0.4.15+1.orig/chebstortion_1430.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/chebstortion_1430.so.c	2012-05-13 01:06:55.970743348 +0200
@@ -321,7 +321,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/comb_1190.c swh-plugins-0.4.15+1/comb_1190.c
--- swh-plugins-0.4.15+1.orig/comb_1190.c	2012-05-13 01:05:58.186460843 +0200
+++ swh-plugins-0.4.15+1/comb_1190.c	2012-05-13 01:06:55.974743361 +0200
@@ -258,7 +258,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/comb_1190.so.c swh-plugins-0.4.15+1/comb_1190.so.c
--- swh-plugins-0.4.15+1.orig/comb_1190.so.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/comb_1190.so.c	2012-05-13 01:06:55.978743377 +0200
@@ -258,7 +258,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/comb_1887.c swh-plugins-0.4.15+1/comb_1887.c
--- swh-plugins-0.4.15+1.orig/comb_1887.c	2012-05-13 01:05:58.242461117 +0200
+++ swh-plugins-0.4.15+1/comb_1887.c	2012-05-13 01:06:55.986743426 +0200
@@ -1117,7 +1117,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/comb_1887.so.c swh-plugins-0.4.15+1/comb_1887.so.c
--- swh-plugins-0.4.15+1.orig/comb_1887.so.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/comb_1887.so.c	2012-05-13 01:06:55.990743439 +0200
@@ -1117,7 +1117,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/comb_splitter_1411.c swh-plugins-0.4.15+1/comb_splitter_1411.c
--- swh-plugins-0.4.15+1.orig/comb_splitter_1411.c	2012-05-13 01:05:58.110460476 +0200
+++ swh-plugins-0.4.15+1/comb_splitter_1411.c	2012-05-13 01:06:55.990743439 +0200
@@ -262,7 +262,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/comb_splitter_1411.so.c swh-plugins-0.4.15+1/comb_splitter_1411.so.c
--- swh-plugins-0.4.15+1.orig/comb_splitter_1411.so.c	2012-05-13 01:05:58.094460405 +0200
+++ swh-plugins-0.4.15+1/comb_splitter_1411.so.c	2012-05-13 01:06:55.994743455 +0200
@@ -262,7 +262,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/const_1909.c swh-plugins-0.4.15+1/const_1909.c
--- swh-plugins-0.4.15+1.orig/const_1909.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/const_1909.c	2012-05-13 01:06:55.994743455 +0200
@@ -178,7 +178,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/crossover_dist_1404.c swh-plugins-0.4.15+1/crossover_dist_1404.c
--- swh-plugins-0.4.15+1.orig/crossover_dist_1404.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/crossover_dist_1404.c	2012-05-13 01:06:55.994743455 +0200
@@ -194,7 +194,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/crossover_dist_1404.so.c swh-plugins-0.4.15+1/crossover_dist_1404.so.c
--- swh-plugins-0.4.15+1.orig/crossover_dist_1404.so.c	2012-05-13 01:05:58.118460515 +0200
+++ swh-plugins-0.4.15+1/crossover_dist_1404.so.c	2012-05-13 01:06:55.994743455 +0200
@@ -191,7 +191,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/dc_remove_1207.c swh-plugins-0.4.15+1/dc_remove_1207.c
--- swh-plugins-0.4.15+1.orig/dc_remove_1207.c	2012-05-13 01:05:58.094460405 +0200
+++ swh-plugins-0.4.15+1/dc_remove_1207.c	2012-05-13 01:06:55.998743477 +0200
@@ -173,7 +173,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/dc_remove_1207.so.c swh-plugins-0.4.15+1/dc_remove_1207.so.c
--- swh-plugins-0.4.15+1.orig/dc_remove_1207.so.c	2012-05-13 01:05:58.110460476 +0200
+++ swh-plugins-0.4.15+1/dc_remove_1207.so.c	2012-05-13 01:06:55.998743477 +0200
@@ -173,7 +173,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
Gemeinsame Unterverzeichnisse: swh-plugins-0.4.15+1.orig/debian und swh-plugins-0.4.15+1/debian.
diff -u swh-plugins-0.4.15+1.orig/debug_1184.so.c swh-plugins-0.4.15+1/debug_1184.so.c
--- swh-plugins-0.4.15+1.orig/debug_1184.so.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/debug_1184.so.c	2012-05-13 01:06:55.998743477 +0200
@@ -207,7 +207,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/decay_1886.c swh-plugins-0.4.15+1/decay_1886.c
--- swh-plugins-0.4.15+1.orig/decay_1886.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/decay_1886.c	2012-05-13 01:06:55.998743477 +0200
@@ -259,7 +259,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/decay_1886.so.c swh-plugins-0.4.15+1/decay_1886.so.c
--- swh-plugins-0.4.15+1.orig/decay_1886.so.c	2012-05-13 01:05:58.118460515 +0200
+++ swh-plugins-0.4.15+1/decay_1886.so.c	2012-05-13 01:06:55.998743477 +0200
@@ -259,7 +259,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/decimator_1202.c swh-plugins-0.4.15+1/decimator_1202.c
--- swh-plugins-0.4.15+1.orig/decimator_1202.c	2012-05-13 01:05:58.110460476 +0200
+++ swh-plugins-0.4.15+1/decimator_1202.c	2012-05-13 01:06:56.002743504 +0200
@@ -247,7 +247,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/decimator_1202.so.c swh-plugins-0.4.15+1/decimator_1202.so.c
--- swh-plugins-0.4.15+1.orig/decimator_1202.so.c	2012-05-13 01:05:58.202460932 +0200
+++ swh-plugins-0.4.15+1/decimator_1202.so.c	2012-05-13 01:06:56.002743504 +0200
@@ -247,7 +247,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/declip_1195.c swh-plugins-0.4.15+1/declip_1195.c
--- swh-plugins-0.4.15+1.orig/declip_1195.c	2012-05-13 01:05:58.210460960 +0200
+++ swh-plugins-0.4.15+1/declip_1195.c	2012-05-13 01:06:56.002743504 +0200
@@ -167,7 +167,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/declip_1195.so.c swh-plugins-0.4.15+1/declip_1195.so.c
--- swh-plugins-0.4.15+1.orig/declip_1195.so.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/declip_1195.so.c	2012-05-13 01:06:56.002743504 +0200
@@ -167,7 +167,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/delay_1898.c swh-plugins-0.4.15+1/delay_1898.c
--- swh-plugins-0.4.15+1.orig/delay_1898.c	2012-05-13 01:05:58.214460980 +0200
+++ swh-plugins-0.4.15+1/delay_1898.c	2012-05-13 01:06:56.006743517 +0200
@@ -903,7 +903,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/delay_1898.so.c swh-plugins-0.4.15+1/delay_1898.so.c
--- swh-plugins-0.4.15+1.orig/delay_1898.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/delay_1898.so.c	2012-05-13 01:06:56.010743533 +0200
@@ -903,7 +903,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/delayorama_1402.c swh-plugins-0.4.15+1/delayorama_1402.c
--- swh-plugins-0.4.15+1.orig/delayorama_1402.c	2012-05-13 01:05:58.094460405 +0200
+++ swh-plugins-0.4.15+1/delayorama_1402.c	2012-05-13 01:06:56.014743555 +0200
@@ -675,7 +675,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/delayorama_1402.so.c swh-plugins-0.4.15+1/delayorama_1402.so.c
--- swh-plugins-0.4.15+1.orig/delayorama_1402.so.c	2012-05-13 01:05:58.110460476 +0200
+++ swh-plugins-0.4.15+1/delayorama_1402.so.c	2012-05-13 01:06:56.014743555 +0200
@@ -675,7 +675,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/diode_1185.c swh-plugins-0.4.15+1/diode_1185.c
--- swh-plugins-0.4.15+1.orig/diode_1185.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/diode_1185.c	2012-05-13 01:06:56.014743555 +0200
@@ -192,7 +192,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/diode_1185.so.c swh-plugins-0.4.15+1/diode_1185.so.c
--- swh-plugins-0.4.15+1.orig/diode_1185.so.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/diode_1185.so.c	2012-05-13 01:06:56.018743582 +0200
@@ -192,7 +192,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/divider_1186.c swh-plugins-0.4.15+1/divider_1186.c
--- swh-plugins-0.4.15+1.orig/divider_1186.c	2012-05-13 01:05:58.118460515 +0200
+++ swh-plugins-0.4.15+1/divider_1186.c	2012-05-13 01:06:56.018743582 +0200
@@ -256,7 +256,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/divider_1186.so.c swh-plugins-0.4.15+1/divider_1186.so.c
--- swh-plugins-0.4.15+1.orig/divider_1186.so.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/divider_1186.so.c	2012-05-13 01:06:56.018743582 +0200
@@ -256,7 +256,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/dj_eq_1901.c swh-plugins-0.4.15+1/dj_eq_1901.c
--- swh-plugins-0.4.15+1.orig/dj_eq_1901.c	2012-05-13 01:05:58.214460980 +0200
+++ swh-plugins-0.4.15+1/dj_eq_1901.c	2012-05-13 01:06:56.022743595 +0200
@@ -470,7 +470,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/dj_eq_1901.so.c swh-plugins-0.4.15+1/dj_eq_1901.so.c
--- swh-plugins-0.4.15+1.orig/dj_eq_1901.so.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/dj_eq_1901.so.c	2012-05-13 01:06:56.022743595 +0200
@@ -470,7 +470,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/dj_flanger_1438.c swh-plugins-0.4.15+1/dj_flanger_1438.c
--- swh-plugins-0.4.15+1.orig/dj_flanger_1438.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/dj_flanger_1438.c	2012-05-13 01:06:56.022743595 +0200
@@ -373,7 +373,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/dj_flanger_1438.so.c swh-plugins-0.4.15+1/dj_flanger_1438.so.c
--- swh-plugins-0.4.15+1.orig/dj_flanger_1438.so.c	2012-05-13 01:05:58.166460749 +0200
+++ swh-plugins-0.4.15+1/dj_flanger_1438.so.c	2012-05-13 01:06:56.026743611 +0200
@@ -373,7 +373,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/dyson_compress_1403.c swh-plugins-0.4.15+1/dyson_compress_1403.c
--- swh-plugins-0.4.15+1.orig/dyson_compress_1403.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/dyson_compress_1403.c	2012-05-13 01:06:56.026743611 +0200
@@ -783,7 +783,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/dyson_compress_1403.so.c swh-plugins-0.4.15+1/dyson_compress_1403.so.c
--- swh-plugins-0.4.15+1.orig/dyson_compress_1403.so.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/dyson_compress_1403.so.c	2012-05-13 01:06:56.030743633 +0200
@@ -783,7 +783,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/fad_delay_1192.c swh-plugins-0.4.15+1/fad_delay_1192.c
--- swh-plugins-0.4.15+1.orig/fad_delay_1192.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/fad_delay_1192.c	2012-05-13 01:06:56.030743633 +0200
@@ -312,7 +312,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/fad_delay_1192.so.c swh-plugins-0.4.15+1/fad_delay_1192.so.c
--- swh-plugins-0.4.15+1.orig/fad_delay_1192.so.c	2012-05-13 01:05:58.214460980 +0200
+++ swh-plugins-0.4.15+1/fad_delay_1192.so.c	2012-05-13 01:06:56.034743660 +0200
@@ -312,7 +312,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/fast_lookahead_limiter_1913.c swh-plugins-0.4.15+1/fast_lookahead_limiter_1913.c
--- swh-plugins-0.4.15+1.orig/fast_lookahead_limiter_1913.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/fast_lookahead_limiter_1913.c	2012-05-13 01:06:56.034743660 +0200
@@ -580,7 +580,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/flanger_1191.c swh-plugins-0.4.15+1/flanger_1191.c
--- swh-plugins-0.4.15+1.orig/flanger_1191.c	2012-05-13 01:05:58.210460960 +0200
+++ swh-plugins-0.4.15+1/flanger_1191.c	2012-05-13 01:06:56.034743660 +0200
@@ -442,7 +442,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/flanger_1191.so.c swh-plugins-0.4.15+1/flanger_1191.so.c
--- swh-plugins-0.4.15+1.orig/flanger_1191.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/flanger_1191.so.c	2012-05-13 01:06:56.038743673 +0200
@@ -428,7 +428,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/fm_osc_1415.c swh-plugins-0.4.15+1/fm_osc_1415.c
--- swh-plugins-0.4.15+1.orig/fm_osc_1415.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/fm_osc_1415.c	2012-05-13 01:06:56.038743673 +0200
@@ -186,7 +186,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/fm_osc_1415.so.c swh-plugins-0.4.15+1/fm_osc_1415.so.c
--- swh-plugins-0.4.15+1.orig/fm_osc_1415.so.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/fm_osc_1415.so.c	2012-05-13 01:06:56.038743673 +0200
@@ -186,7 +186,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/foldover_1213.c swh-plugins-0.4.15+1/foldover_1213.c
--- swh-plugins-0.4.15+1.orig/foldover_1213.c	2012-05-13 01:05:58.178460807 +0200
+++ swh-plugins-0.4.15+1/foldover_1213.c	2012-05-13 01:06:56.038743673 +0200
@@ -173,7 +173,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/foldover_1213.so.c swh-plugins-0.4.15+1/foldover_1213.so.c
--- swh-plugins-0.4.15+1.orig/foldover_1213.so.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/foldover_1213.so.c	2012-05-13 01:06:56.042743691 +0200
@@ -173,7 +173,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/foverdrive_1196.c swh-plugins-0.4.15+1/foverdrive_1196.c
--- swh-plugins-0.4.15+1.orig/foverdrive_1196.c	2012-05-13 01:05:58.218461009 +0200
+++ swh-plugins-0.4.15+1/foverdrive_1196.c	2012-05-13 01:06:56.042743691 +0200
@@ -162,7 +162,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/foverdrive_1196.so.c swh-plugins-0.4.15+1/foverdrive_1196.so.c
--- swh-plugins-0.4.15+1.orig/foverdrive_1196.so.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/foverdrive_1196.so.c	2012-05-13 01:06:56.042743691 +0200
@@ -162,7 +162,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/freq_tracker_1418.c swh-plugins-0.4.15+1/freq_tracker_1418.c
--- swh-plugins-0.4.15+1.orig/freq_tracker_1418.c	2012-05-13 01:05:58.242461117 +0200
+++ swh-plugins-0.4.15+1/freq_tracker_1418.c	2012-05-13 01:06:56.042743691 +0200
@@ -247,7 +247,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/freq_tracker_1418.so.c swh-plugins-0.4.15+1/freq_tracker_1418.so.c
--- swh-plugins-0.4.15+1.orig/freq_tracker_1418.so.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/freq_tracker_1418.so.c	2012-05-13 01:06:56.042743691 +0200
@@ -247,7 +247,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/gate_1410.c swh-plugins-0.4.15+1/gate_1410.c
--- swh-plugins-0.4.15+1.orig/gate_1410.c	2012-05-13 01:05:58.210460960 +0200
+++ swh-plugins-0.4.15+1/gate_1410.c	2012-05-13 01:06:56.046743711 +0200
@@ -440,7 +440,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/gate_1410.so.c swh-plugins-0.4.15+1/gate_1410.so.c
--- swh-plugins-0.4.15+1.orig/gate_1410.so.c	2012-05-13 01:05:58.110460476 +0200
+++ swh-plugins-0.4.15+1/gate_1410.so.c	2012-05-13 01:06:56.046743711 +0200
@@ -440,7 +440,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/gate_1921.so.c swh-plugins-0.4.15+1/gate_1921.so.c
--- swh-plugins-0.4.15+1.orig/gate_1921.so.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/gate_1921.so.c	2012-05-13 01:06:56.050743738 +0200
@@ -858,7 +858,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/giant_flange_1437.c swh-plugins-0.4.15+1/giant_flange_1437.c
--- swh-plugins-0.4.15+1.orig/giant_flange_1437.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/giant_flange_1437.c	2012-05-13 01:06:56.050743738 +0200
@@ -508,7 +508,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/giant_flange_1437.so.c swh-plugins-0.4.15+1/giant_flange_1437.so.c
--- swh-plugins-0.4.15+1.orig/giant_flange_1437.so.c	2012-05-13 01:05:58.186460843 +0200
+++ swh-plugins-0.4.15+1/giant_flange_1437.so.c	2012-05-13 01:06:56.054743750 +0200
@@ -508,7 +508,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/gong_1424.c swh-plugins-0.4.15+1/gong_1424.c
--- swh-plugins-0.4.15+1.orig/gong_1424.c	2012-05-13 01:05:58.110460476 +0200
+++ swh-plugins-0.4.15+1/gong_1424.c	2012-05-13 01:06:56.058743766 +0200
@@ -615,7 +615,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/gong_1424.so.c swh-plugins-0.4.15+1/gong_1424.so.c
--- swh-plugins-0.4.15+1.orig/gong_1424.so.c	2012-05-13 01:05:58.230461069 +0200
+++ swh-plugins-0.4.15+1/gong_1424.so.c	2012-05-13 01:06:56.058743766 +0200
@@ -615,7 +615,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/gong_beater_1439.c swh-plugins-0.4.15+1/gong_beater_1439.c
--- swh-plugins-0.4.15+1.orig/gong_beater_1439.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/gong_beater_1439.c	2012-05-13 01:06:56.062743787 +0200
@@ -308,7 +308,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/gong_beater_1439.so.c swh-plugins-0.4.15+1/gong_beater_1439.so.c
--- swh-plugins-0.4.15+1.orig/gong_beater_1439.so.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/gong_beater_1439.so.c	2012-05-13 01:06:56.062743787 +0200
@@ -308,7 +308,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
Gemeinsame Unterverzeichnisse: swh-plugins-0.4.15+1.orig/gsm und swh-plugins-0.4.15+1/gsm.
diff -u swh-plugins-0.4.15+1.orig/gsm_1215.c swh-plugins-0.4.15+1/gsm_1215.c
--- swh-plugins-0.4.15+1.orig/gsm_1215.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/gsm_1215.c	2012-05-13 01:06:56.062743787 +0200
@@ -394,7 +394,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/gsm_1215.so.c swh-plugins-0.4.15+1/gsm_1215.so.c
--- swh-plugins-0.4.15+1.orig/gsm_1215.so.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/gsm_1215.so.c	2012-05-13 01:06:56.066743815 +0200
@@ -394,7 +394,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
Gemeinsame Unterverzeichnisse: swh-plugins-0.4.15+1.orig/gverb und swh-plugins-0.4.15+1/gverb.
diff -u swh-plugins-0.4.15+1.orig/gverb_1216.c swh-plugins-0.4.15+1/gverb_1216.c
--- swh-plugins-0.4.15+1.orig/gverb_1216.c	2012-05-13 01:05:58.230461069 +0200
+++ swh-plugins-0.4.15+1/gverb_1216.c	2012-05-13 01:06:56.070743829 +0200
@@ -290,7 +290,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/gverb_1216.so.c swh-plugins-0.4.15+1/gverb_1216.so.c
--- swh-plugins-0.4.15+1.orig/gverb_1216.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/gverb_1216.so.c	2012-05-13 01:06:56.074743848 +0200
@@ -290,7 +290,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/hard_limiter_1413.c swh-plugins-0.4.15+1/hard_limiter_1413.c
--- swh-plugins-0.4.15+1.orig/hard_limiter_1413.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/hard_limiter_1413.c	2012-05-13 01:06:56.078743875 +0200
@@ -194,7 +194,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/hard_limiter_1413.so.c swh-plugins-0.4.15+1/hard_limiter_1413.so.c
--- swh-plugins-0.4.15+1.orig/hard_limiter_1413.so.c	2012-05-13 01:05:58.118460515 +0200
+++ swh-plugins-0.4.15+1/hard_limiter_1413.so.c	2012-05-13 01:06:56.082743887 +0200
@@ -194,7 +194,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/harmonic_gen_1220.c swh-plugins-0.4.15+1/harmonic_gen_1220.c
--- swh-plugins-0.4.15+1.orig/harmonic_gen_1220.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/harmonic_gen_1220.c	2012-05-13 01:06:56.086743903 +0200
@@ -374,7 +374,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/hermes_filter_1200.c swh-plugins-0.4.15+1/hermes_filter_1200.c
--- swh-plugins-0.4.15+1.orig/hermes_filter_1200.c	2012-05-13 01:05:58.186460843 +0200
+++ swh-plugins-0.4.15+1/hermes_filter_1200.c	2012-05-13 01:06:56.114744044 +0200
@@ -1410,7 +1410,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/hermes_filter_1200.so.c swh-plugins-0.4.15+1/hermes_filter_1200.so.c
--- swh-plugins-0.4.15+1.orig/hermes_filter_1200.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/hermes_filter_1200.so.c	2012-05-13 01:06:56.138744168 +0200
@@ -1410,7 +1410,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/highpass_iir_1890.c swh-plugins-0.4.15+1/highpass_iir_1890.c
--- swh-plugins-0.4.15+1.orig/highpass_iir_1890.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/highpass_iir_1890.c	2012-05-13 01:06:56.142744183 +0200
@@ -203,7 +203,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/highpass_iir_1890.so.c swh-plugins-0.4.15+1/highpass_iir_1890.so.c
--- swh-plugins-0.4.15+1.orig/highpass_iir_1890.so.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/highpass_iir_1890.so.c	2012-05-13 01:06:56.142744183 +0200
@@ -203,7 +203,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/hilbert_1440.c swh-plugins-0.4.15+1/hilbert_1440.c
--- swh-plugins-0.4.15+1.orig/hilbert_1440.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/hilbert_1440.c	2012-05-13 01:06:56.146744199 +0200
@@ -240,7 +240,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/hilbert_1440.so.c swh-plugins-0.4.15+1/hilbert_1440.so.c
--- swh-plugins-0.4.15+1.orig/hilbert_1440.so.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/hilbert_1440.so.c	2012-05-13 01:06:56.146744199 +0200
@@ -240,7 +240,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/imp_1199.c swh-plugins-0.4.15+1/imp_1199.c
--- swh-plugins-0.4.15+1.orig/imp_1199.c	2012-05-13 01:05:58.102460437 +0200
+++ swh-plugins-0.4.15+1/imp_1199.c	2012-05-13 01:06:56.146744199 +0200
@@ -527,7 +527,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/imp_1199.so.c swh-plugins-0.4.15+1/imp_1199.so.c
--- swh-plugins-0.4.15+1.orig/imp_1199.so.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/imp_1199.so.c	2012-05-13 01:06:56.150744221 +0200
@@ -525,7 +525,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/impulse_1885.c swh-plugins-0.4.15+1/impulse_1885.c
--- swh-plugins-0.4.15+1.orig/impulse_1885.c	2012-05-13 01:05:58.118460515 +0200
+++ swh-plugins-0.4.15+1/impulse_1885.c	2012-05-13 01:06:56.150744221 +0200
@@ -193,7 +193,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/impulse_1885.so.c swh-plugins-0.4.15+1/impulse_1885.so.c
--- swh-plugins-0.4.15+1.orig/impulse_1885.so.c	2012-05-13 01:05:58.186460843 +0200
+++ swh-plugins-0.4.15+1/impulse_1885.so.c	2012-05-13 01:06:56.150744221 +0200
@@ -193,7 +193,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
Gemeinsame Unterverzeichnisse: swh-plugins-0.4.15+1.orig/impulses und swh-plugins-0.4.15+1/impulses.
diff -u swh-plugins-0.4.15+1.orig/inv_1429.c swh-plugins-0.4.15+1/inv_1429.c
--- swh-plugins-0.4.15+1.orig/inv_1429.c	2012-05-13 01:05:58.218461009 +0200
+++ swh-plugins-0.4.15+1/inv_1429.c	2012-05-13 01:06:56.150744221 +0200
@@ -145,7 +145,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/inv_1429.so.c swh-plugins-0.4.15+1/inv_1429.so.c
--- swh-plugins-0.4.15+1.orig/inv_1429.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/inv_1429.so.c	2012-05-13 01:06:56.154744248 +0200
@@ -145,7 +145,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/karaoke_1409.c swh-plugins-0.4.15+1/karaoke_1409.c
--- swh-plugins-0.4.15+1.orig/karaoke_1409.c	2012-05-13 01:05:58.202460932 +0200
+++ swh-plugins-0.4.15+1/karaoke_1409.c	2012-05-13 01:06:56.154744248 +0200
@@ -188,7 +188,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/karaoke_1409.so.c swh-plugins-0.4.15+1/karaoke_1409.so.c
--- swh-plugins-0.4.15+1.orig/karaoke_1409.so.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/karaoke_1409.so.c	2012-05-13 01:06:56.154744248 +0200
@@ -188,7 +188,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/latency_1914.c swh-plugins-0.4.15+1/latency_1914.c
--- swh-plugins-0.4.15+1.orig/latency_1914.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/latency_1914.c	2012-05-13 01:06:56.154744248 +0200
@@ -180,7 +180,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/lcr_delay_1436.c swh-plugins-0.4.15+1/lcr_delay_1436.c
--- swh-plugins-0.4.15+1.orig/lcr_delay_1436.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/lcr_delay_1436.c	2012-05-13 01:06:56.158744260 +0200
@@ -547,7 +547,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/lcr_delay_1436.so.c swh-plugins-0.4.15+1/lcr_delay_1436.so.c
--- swh-plugins-0.4.15+1.orig/lcr_delay_1436.so.c	2012-05-13 01:05:58.242461117 +0200
+++ swh-plugins-0.4.15+1/lcr_delay_1436.so.c	2012-05-13 01:06:56.158744260 +0200
@@ -547,7 +547,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/lookahead_limiter_1435.so.c swh-plugins-0.4.15+1/lookahead_limiter_1435.so.c
--- swh-plugins-0.4.15+1.orig/lookahead_limiter_1435.so.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/lookahead_limiter_1435.so.c	2012-05-13 01:06:56.162744276 +0200
@@ -400,7 +400,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/lookahead_limiter_const_1906.so.c swh-plugins-0.4.15+1/lookahead_limiter_const_1906.so.c
--- swh-plugins-0.4.15+1.orig/lookahead_limiter_const_1906.so.c	2012-05-13 01:05:58.178460807 +0200
+++ swh-plugins-0.4.15+1/lookahead_limiter_const_1906.so.c	2012-05-13 01:06:56.162744276 +0200
@@ -456,7 +456,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/lowpass_iir_1891.c swh-plugins-0.4.15+1/lowpass_iir_1891.c
--- swh-plugins-0.4.15+1.orig/lowpass_iir_1891.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/lowpass_iir_1891.c	2012-05-13 01:06:56.162744276 +0200
@@ -205,7 +205,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/lowpass_iir_1891.so.c swh-plugins-0.4.15+1/lowpass_iir_1891.so.c
--- swh-plugins-0.4.15+1.orig/lowpass_iir_1891.so.c	2012-05-13 01:05:58.202460932 +0200
+++ swh-plugins-0.4.15+1/lowpass_iir_1891.so.c	2012-05-13 01:06:56.166744297 +0200
@@ -205,7 +205,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/ls_filter_1908.c swh-plugins-0.4.15+1/ls_filter_1908.c
--- swh-plugins-0.4.15+1.orig/ls_filter_1908.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/ls_filter_1908.c	2012-05-13 01:06:56.166744297 +0200
@@ -216,7 +216,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/ls_filter_1908.so.c swh-plugins-0.4.15+1/ls_filter_1908.so.c
--- swh-plugins-0.4.15+1.orig/ls_filter_1908.so.c	2012-05-13 01:05:58.210460960 +0200
+++ swh-plugins-0.4.15+1/ls_filter_1908.so.c	2012-05-13 01:06:56.166744297 +0200
@@ -216,7 +216,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
Gemeinsame Unterverzeichnisse: swh-plugins-0.4.15+1.orig/m4 und swh-plugins-0.4.15+1/m4.
diff -u swh-plugins-0.4.15+1.orig/makestub.pl swh-plugins-0.4.15+1/makestub.pl
--- swh-plugins-0.4.15+1.orig/makestub.pl	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/makestub.pl	2012-05-13 01:07:32.718922991 +0200
@@ -146,7 +146,6 @@
 
 \#ifdef ENABLE_NLS
 \#define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 \#else
 \#define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/matrix_ms_st_1421.c swh-plugins-0.4.15+1/matrix_ms_st_1421.c
--- swh-plugins-0.4.15+1.orig/matrix_ms_st_1421.c	2012-05-13 01:05:58.206460944 +0200
+++ swh-plugins-0.4.15+1/matrix_ms_st_1421.c	2012-05-13 01:06:56.166744297 +0200
@@ -180,7 +180,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/matrix_ms_st_1421.so.c swh-plugins-0.4.15+1/matrix_ms_st_1421.so.c
--- swh-plugins-0.4.15+1.orig/matrix_ms_st_1421.so.c	2012-05-13 01:05:58.242461117 +0200
+++ swh-plugins-0.4.15+1/matrix_ms_st_1421.so.c	2012-05-13 01:06:56.166744297 +0200
@@ -180,7 +180,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/matrix_spatialiser_1422.c swh-plugins-0.4.15+1/matrix_spatialiser_1422.c
--- swh-plugins-0.4.15+1.orig/matrix_spatialiser_1422.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/matrix_spatialiser_1422.c	2012-05-13 01:06:56.170744325 +0200
@@ -338,7 +338,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/matrix_spatialiser_1422.so.c swh-plugins-0.4.15+1/matrix_spatialiser_1422.so.c
--- swh-plugins-0.4.15+1.orig/matrix_spatialiser_1422.so.c	2012-05-13 01:05:58.166460749 +0200
+++ swh-plugins-0.4.15+1/matrix_spatialiser_1422.so.c	2012-05-13 01:06:56.170744325 +0200
@@ -338,7 +338,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/matrix_st_ms_1420.c swh-plugins-0.4.15+1/matrix_st_ms_1420.c
--- swh-plugins-0.4.15+1.orig/matrix_st_ms_1420.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/matrix_st_ms_1420.c	2012-05-13 01:06:56.170744325 +0200
@@ -169,7 +169,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/matrix_st_ms_1420.so.c swh-plugins-0.4.15+1/matrix_st_ms_1420.so.c
--- swh-plugins-0.4.15+1.orig/matrix_st_ms_1420.so.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/matrix_st_ms_1420.so.c	2012-05-13 01:06:56.170744325 +0200
@@ -169,7 +169,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/mbeq_1197.c swh-plugins-0.4.15+1/mbeq_1197.c
--- swh-plugins-0.4.15+1.orig/mbeq_1197.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/mbeq_1197.c	2012-05-13 01:06:56.174744338 +0200
@@ -664,7 +664,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/mbeq_1197.so.c swh-plugins-0.4.15+1/mbeq_1197.so.c
--- swh-plugins-0.4.15+1.orig/mbeq_1197.so.c	2012-05-13 01:05:58.230461069 +0200
+++ swh-plugins-0.4.15+1/mbeq_1197.so.c	2012-05-13 01:06:56.178744356 +0200
@@ -654,7 +654,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
Gemeinsame Unterverzeichnisse: swh-plugins-0.4.15+1.orig/metadata und swh-plugins-0.4.15+1/metadata.
diff -u swh-plugins-0.4.15+1.orig/mod_delay_1419.c swh-plugins-0.4.15+1/mod_delay_1419.c
--- swh-plugins-0.4.15+1.orig/mod_delay_1419.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/mod_delay_1419.c	2012-05-13 01:06:56.178744356 +0200
@@ -236,7 +236,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/mod_delay_1419.so.c swh-plugins-0.4.15+1/mod_delay_1419.so.c
--- swh-plugins-0.4.15+1.orig/mod_delay_1419.so.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/mod_delay_1419.so.c	2012-05-13 01:06:56.178744356 +0200
@@ -236,7 +236,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/multivoice_chorus_1201.c swh-plugins-0.4.15+1/multivoice_chorus_1201.c
--- swh-plugins-0.4.15+1.orig/multivoice_chorus_1201.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/multivoice_chorus_1201.c	2012-05-13 01:06:56.182744376 +0200
@@ -534,7 +534,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/multivoice_chorus_1201.so.c swh-plugins-0.4.15+1/multivoice_chorus_1201.so.c
--- swh-plugins-0.4.15+1.orig/multivoice_chorus_1201.so.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/multivoice_chorus_1201.so.c	2012-05-13 01:06:56.182744376 +0200
@@ -534,7 +534,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/notch_iir_1894.c swh-plugins-0.4.15+1/notch_iir_1894.c
--- swh-plugins-0.4.15+1.orig/notch_iir_1894.c	2012-05-13 01:05:58.110460476 +0200
+++ swh-plugins-0.4.15+1/notch_iir_1894.c	2012-05-13 01:06:56.182744376 +0200
@@ -258,7 +258,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/notch_iir_1894.so.c swh-plugins-0.4.15+1/notch_iir_1894.so.c
--- swh-plugins-0.4.15+1.orig/notch_iir_1894.so.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/notch_iir_1894.so.c	2012-05-13 01:06:56.186744403 +0200
@@ -258,7 +258,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/phasers_1217.c swh-plugins-0.4.15+1/phasers_1217.c
--- swh-plugins-0.4.15+1.orig/phasers_1217.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/phasers_1217.c	2012-05-13 01:06:56.190744415 +0200
@@ -1037,7 +1037,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/phasers_1217.so.c swh-plugins-0.4.15+1/phasers_1217.so.c
--- swh-plugins-0.4.15+1.orig/phasers_1217.so.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/phasers_1217.so.c	2012-05-13 01:06:56.194744431 +0200
@@ -1037,7 +1037,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/pitch_scale_1193.c swh-plugins-0.4.15+1/pitch_scale_1193.c
--- swh-plugins-0.4.15+1.orig/pitch_scale_1193.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/pitch_scale_1193.c	2012-05-13 01:06:56.194744431 +0200
@@ -248,7 +248,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/pitch_scale_1193.so.c swh-plugins-0.4.15+1/pitch_scale_1193.so.c
--- swh-plugins-0.4.15+1.orig/pitch_scale_1193.so.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/pitch_scale_1193.so.c	2012-05-13 01:06:56.194744431 +0200
@@ -248,7 +248,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/pitch_scale_1194.c swh-plugins-0.4.15+1/pitch_scale_1194.c
--- swh-plugins-0.4.15+1.orig/pitch_scale_1194.c	2012-05-13 01:05:58.202460932 +0200
+++ swh-plugins-0.4.15+1/pitch_scale_1194.c	2012-05-13 01:06:56.194744431 +0200
@@ -232,7 +232,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/pitch_scale_1194.so.c swh-plugins-0.4.15+1/pitch_scale_1194.so.c
--- swh-plugins-0.4.15+1.orig/pitch_scale_1194.so.c	2012-05-13 01:05:58.118460515 +0200
+++ swh-plugins-0.4.15+1/pitch_scale_1194.so.c	2012-05-13 01:06:56.198744452 +0200
@@ -232,7 +232,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/plate_1423.c swh-plugins-0.4.15+1/plate_1423.c
--- swh-plugins-0.4.15+1.orig/plate_1423.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/plate_1423.c	2012-05-13 01:06:56.198744452 +0200
@@ -302,7 +302,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/plate_1423.so.c swh-plugins-0.4.15+1/plate_1423.so.c
--- swh-plugins-0.4.15+1.orig/plate_1423.so.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/plate_1423.so.c	2012-05-13 01:06:56.198744452 +0200
@@ -302,7 +302,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
Gemeinsame Unterverzeichnisse: swh-plugins-0.4.15+1.orig/po und swh-plugins-0.4.15+1/po.
diff -u swh-plugins-0.4.15+1.orig/pointer_cast_1910.c swh-plugins-0.4.15+1/pointer_cast_1910.c
--- swh-plugins-0.4.15+1.orig/pointer_cast_1910.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/pointer_cast_1910.c	2012-05-13 01:06:56.198744452 +0200
@@ -228,7 +228,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/rate_shifter_1417.c swh-plugins-0.4.15+1/rate_shifter_1417.c
--- swh-plugins-0.4.15+1.orig/rate_shifter_1417.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/rate_shifter_1417.c	2012-05-13 01:06:56.202744480 +0200
@@ -242,7 +242,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/rate_shifter_1417.so.c swh-plugins-0.4.15+1/rate_shifter_1417.so.c
--- swh-plugins-0.4.15+1.orig/rate_shifter_1417.so.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/rate_shifter_1417.so.c	2012-05-13 01:06:56.202744480 +0200
@@ -242,7 +242,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/retro_flange_1208.c swh-plugins-0.4.15+1/retro_flange_1208.c
--- swh-plugins-0.4.15+1.orig/retro_flange_1208.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/retro_flange_1208.c	2012-05-13 01:06:56.202744480 +0200
@@ -499,7 +499,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/retro_flange_1208.so.c swh-plugins-0.4.15+1/retro_flange_1208.so.c
--- swh-plugins-0.4.15+1.orig/retro_flange_1208.so.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/retro_flange_1208.so.c	2012-05-13 01:06:56.222744573 +0200
@@ -499,7 +499,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/revdelay_1605.c swh-plugins-0.4.15+1/revdelay_1605.c
--- swh-plugins-0.4.15+1.orig/revdelay_1605.c	2012-05-13 01:05:58.218461009 +0200
+++ swh-plugins-0.4.15+1/revdelay_1605.c	2012-05-13 01:06:56.226744589 +0200
@@ -436,7 +436,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/revdelay_1605.so.c swh-plugins-0.4.15+1/revdelay_1605.so.c
--- swh-plugins-0.4.15+1.orig/revdelay_1605.so.c	2012-05-13 01:05:58.182460823 +0200
+++ swh-plugins-0.4.15+1/revdelay_1605.so.c	2012-05-13 01:06:56.234744638 +0200
@@ -432,7 +432,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/ringmod_1188.c swh-plugins-0.4.15+1/ringmod_1188.c
--- swh-plugins-0.4.15+1.orig/ringmod_1188.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/ringmod_1188.c	2012-05-13 01:06:56.242744667 +0200
@@ -436,7 +436,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/ringmod_1188.so.c swh-plugins-0.4.15+1/ringmod_1188.so.c
--- swh-plugins-0.4.15+1.orig/ringmod_1188.so.c	2012-05-13 01:05:58.206460944 +0200
+++ swh-plugins-0.4.15+1/ringmod_1188.so.c	2012-05-13 01:06:56.246744689 +0200
@@ -436,7 +436,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/satan_maximiser_1408.c swh-plugins-0.4.15+1/satan_maximiser_1408.c
--- swh-plugins-0.4.15+1.orig/satan_maximiser_1408.c	2012-05-13 01:05:58.202460932 +0200
+++ swh-plugins-0.4.15+1/satan_maximiser_1408.c	2012-05-13 01:06:56.250744716 +0200
@@ -262,7 +262,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/satan_maximiser_1408.so.c swh-plugins-0.4.15+1/satan_maximiser_1408.so.c
--- swh-plugins-0.4.15+1.orig/satan_maximiser_1408.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/satan_maximiser_1408.so.c	2012-05-13 01:06:56.254744729 +0200
@@ -262,7 +262,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sc1_1425.c swh-plugins-0.4.15+1/sc1_1425.c
--- swh-plugins-0.4.15+1.orig/sc1_1425.c	2012-05-13 01:05:58.242461117 +0200
+++ swh-plugins-0.4.15+1/sc1_1425.c	2012-05-13 01:06:56.262744773 +0200
@@ -351,7 +351,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sc1_1425.so.c swh-plugins-0.4.15+1/sc1_1425.so.c
--- swh-plugins-0.4.15+1.orig/sc1_1425.so.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/sc1_1425.so.c	2012-05-13 01:06:56.266744786 +0200
@@ -351,7 +351,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sc2_1426.c swh-plugins-0.4.15+1/sc2_1426.c
--- swh-plugins-0.4.15+1.orig/sc2_1426.c	2012-05-13 01:05:58.178460807 +0200
+++ swh-plugins-0.4.15+1/sc2_1426.c	2012-05-13 01:06:56.270744803 +0200
@@ -362,7 +362,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sc2_1426.so.c swh-plugins-0.4.15+1/sc2_1426.so.c
--- swh-plugins-0.4.15+1.orig/sc2_1426.so.c	2012-05-13 01:05:58.242461117 +0200
+++ swh-plugins-0.4.15+1/sc2_1426.so.c	2012-05-13 01:06:56.278744853 +0200
@@ -362,7 +362,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sc3_1427.c swh-plugins-0.4.15+1/sc3_1427.c
--- swh-plugins-0.4.15+1.orig/sc3_1427.c	2012-05-13 01:05:58.178460807 +0200
+++ swh-plugins-0.4.15+1/sc3_1427.c	2012-05-13 01:06:56.282744868 +0200
@@ -409,7 +409,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sc3_1427.so.c swh-plugins-0.4.15+1/sc3_1427.so.c
--- swh-plugins-0.4.15+1.orig/sc3_1427.so.c	2012-05-13 01:05:58.122460531 +0200
+++ swh-plugins-0.4.15+1/sc3_1427.so.c	2012-05-13 01:06:56.290744904 +0200
@@ -409,7 +409,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sc4_1434.so.c swh-plugins-0.4.15+1/sc4_1434.so.c
--- swh-plugins-0.4.15+1.orig/sc4_1434.so.c	2012-05-13 01:05:58.166460749 +0200
+++ swh-plugins-0.4.15+1/sc4_1434.so.c	2012-05-13 01:06:56.294744931 +0200
@@ -429,7 +429,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sc4_1882.c swh-plugins-0.4.15+1/sc4_1882.c
--- swh-plugins-0.4.15+1.orig/sc4_1882.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/sc4_1882.c	2012-05-13 01:06:56.302744960 +0200
@@ -452,7 +452,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sc4_1882.so.c swh-plugins-0.4.15+1/sc4_1882.so.c
--- swh-plugins-0.4.15+1.orig/sc4_1882.so.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/sc4_1882.so.c	2012-05-13 01:06:56.302744960 +0200
@@ -448,7 +448,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sc4m_1916.c swh-plugins-0.4.15+1/sc4m_1916.c
--- swh-plugins-0.4.15+1.orig/sc4m_1916.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/sc4m_1916.c	2012-05-13 01:06:56.302744960 +0200
@@ -416,7 +416,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/se4_1883.c swh-plugins-0.4.15+1/se4_1883.c
--- swh-plugins-0.4.15+1.orig/se4_1883.c	2012-05-13 01:05:58.182460823 +0200
+++ swh-plugins-0.4.15+1/se4_1883.c	2012-05-13 01:06:56.306744981 +0200
@@ -448,7 +448,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/se4_1883.so.c swh-plugins-0.4.15+1/se4_1883.so.c
--- swh-plugins-0.4.15+1.orig/se4_1883.so.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/se4_1883.so.c	2012-05-13 01:06:56.306744981 +0200
@@ -448,7 +448,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/shaper_1187.c swh-plugins-0.4.15+1/shaper_1187.c
--- swh-plugins-0.4.15+1.orig/shaper_1187.c	2012-05-13 01:05:58.202460932 +0200
+++ swh-plugins-0.4.15+1/shaper_1187.c	2012-05-13 01:06:56.310745009 +0200
@@ -182,7 +182,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/shaper_1187.so.c swh-plugins-0.4.15+1/shaper_1187.so.c
--- swh-plugins-0.4.15+1.orig/shaper_1187.so.c	2012-05-13 01:05:58.246461146 +0200
+++ swh-plugins-0.4.15+1/shaper_1187.so.c	2012-05-13 01:06:56.310745009 +0200
@@ -182,7 +182,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sifter_1210.c swh-plugins-0.4.15+1/sifter_1210.c
--- swh-plugins-0.4.15+1.orig/sifter_1210.c	2012-05-13 01:05:58.098460420 +0200
+++ swh-plugins-0.4.15+1/sifter_1210.c	2012-05-13 01:06:56.310745009 +0200
@@ -362,7 +362,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sifter_1210.so.c swh-plugins-0.4.15+1/sifter_1210.so.c
--- swh-plugins-0.4.15+1.orig/sifter_1210.so.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/sifter_1210.so.c	2012-05-13 01:06:56.310745009 +0200
@@ -362,7 +362,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sin_cos_1881.c swh-plugins-0.4.15+1/sin_cos_1881.c
--- swh-plugins-0.4.15+1.orig/sin_cos_1881.c	2012-05-13 01:05:58.210460960 +0200
+++ swh-plugins-0.4.15+1/sin_cos_1881.c	2012-05-13 01:06:56.314745023 +0200
@@ -214,7 +214,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sin_cos_1881.so.c swh-plugins-0.4.15+1/sin_cos_1881.so.c
--- swh-plugins-0.4.15+1.orig/sin_cos_1881.so.c	2012-05-13 01:05:58.210460960 +0200
+++ swh-plugins-0.4.15+1/sin_cos_1881.so.c	2012-05-13 01:06:56.314745023 +0200
@@ -214,7 +214,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/single_para_1203.c swh-plugins-0.4.15+1/single_para_1203.c
--- swh-plugins-0.4.15+1.orig/single_para_1203.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/single_para_1203.c	2012-05-13 01:06:56.314745023 +0200
@@ -214,7 +214,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/single_para_1203.so.c swh-plugins-0.4.15+1/single_para_1203.so.c
--- swh-plugins-0.4.15+1.orig/single_para_1203.so.c	2012-05-13 01:05:58.242461117 +0200
+++ swh-plugins-0.4.15+1/single_para_1203.so.c	2012-05-13 01:06:56.314745023 +0200
@@ -214,7 +214,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sinus_wavewrapper_1198.c swh-plugins-0.4.15+1/sinus_wavewrapper_1198.c
--- swh-plugins-0.4.15+1.orig/sinus_wavewrapper_1198.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/sinus_wavewrapper_1198.c	2012-05-13 01:06:56.314745023 +0200
@@ -166,7 +166,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/sinus_wavewrapper_1198.so.c swh-plugins-0.4.15+1/sinus_wavewrapper_1198.so.c
--- swh-plugins-0.4.15+1.orig/sinus_wavewrapper_1198.so.c	2012-05-13 01:05:58.206460944 +0200
+++ swh-plugins-0.4.15+1/sinus_wavewrapper_1198.so.c	2012-05-13 01:06:56.318745041 +0200
@@ -166,7 +166,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/smooth_decimate_1414.c swh-plugins-0.4.15+1/smooth_decimate_1414.c
--- swh-plugins-0.4.15+1.orig/smooth_decimate_1414.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/smooth_decimate_1414.c	2012-05-13 01:06:56.318745041 +0200
@@ -247,7 +247,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/smooth_decimate_1414.so.c swh-plugins-0.4.15+1/smooth_decimate_1414.so.c
--- swh-plugins-0.4.15+1.orig/smooth_decimate_1414.so.c	2012-05-13 01:05:58.182460823 +0200
+++ swh-plugins-0.4.15+1/smooth_decimate_1414.so.c	2012-05-13 01:06:56.318745041 +0200
@@ -247,7 +247,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/split_1406.c swh-plugins-0.4.15+1/split_1406.c
--- swh-plugins-0.4.15+1.orig/split_1406.c	2012-05-13 01:05:58.170460768 +0200
+++ swh-plugins-0.4.15+1/split_1406.c	2012-05-13 01:06:56.318745041 +0200
@@ -162,7 +162,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/split_1406.so.c swh-plugins-0.4.15+1/split_1406.so.c
--- swh-plugins-0.4.15+1.orig/split_1406.so.c	2012-05-13 01:05:58.178460807 +0200
+++ swh-plugins-0.4.15+1/split_1406.so.c	2012-05-13 01:06:56.322745062 +0200
@@ -162,7 +162,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/step_muxer_1212.c swh-plugins-0.4.15+1/step_muxer_1212.c
--- swh-plugins-0.4.15+1.orig/step_muxer_1212.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/step_muxer_1212.c	2012-05-13 01:06:56.322745062 +0200
@@ -404,7 +404,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/step_muxer_1212.so.c swh-plugins-0.4.15+1/step_muxer_1212.so.c
--- swh-plugins-0.4.15+1.orig/step_muxer_1212.so.c	2012-05-13 01:05:58.206460944 +0200
+++ swh-plugins-0.4.15+1/step_muxer_1212.so.c	2012-05-13 01:06:56.322745062 +0200
@@ -404,7 +404,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/surround_encoder_1401.c swh-plugins-0.4.15+1/surround_encoder_1401.c
--- swh-plugins-0.4.15+1.orig/surround_encoder_1401.c	2012-05-13 01:05:58.230461069 +0200
+++ swh-plugins-0.4.15+1/surround_encoder_1401.c	2012-05-13 01:06:56.326745088 +0200
@@ -324,7 +324,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/surround_encoder_1401.so.c swh-plugins-0.4.15+1/surround_encoder_1401.so.c
--- swh-plugins-0.4.15+1.orig/surround_encoder_1401.so.c	2012-05-13 01:05:58.186460843 +0200
+++ swh-plugins-0.4.15+1/surround_encoder_1401.so.c	2012-05-13 01:06:56.326745088 +0200
@@ -278,7 +278,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/svf_1214.c swh-plugins-0.4.15+1/svf_1214.c
--- swh-plugins-0.4.15+1.orig/svf_1214.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/svf_1214.c	2012-05-13 01:06:56.326745088 +0200
@@ -300,7 +300,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/svf_1214.so.c swh-plugins-0.4.15+1/svf_1214.so.c
--- swh-plugins-0.4.15+1.orig/svf_1214.so.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/svf_1214.so.c	2012-05-13 01:06:56.330745100 +0200
@@ -303,7 +303,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/tape_delay_1211.c swh-plugins-0.4.15+1/tape_delay_1211.c
--- swh-plugins-0.4.15+1.orig/tape_delay_1211.c	2012-05-13 01:05:58.110460476 +0200
+++ swh-plugins-0.4.15+1/tape_delay_1211.c	2012-05-13 01:06:56.330745100 +0200
@@ -483,7 +483,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/tape_delay_1211.so.c swh-plugins-0.4.15+1/tape_delay_1211.so.c
--- swh-plugins-0.4.15+1.orig/tape_delay_1211.so.c	2012-05-13 01:05:58.234461081 +0200
+++ swh-plugins-0.4.15+1/tape_delay_1211.so.c	2012-05-13 01:06:56.330745100 +0200
@@ -483,7 +483,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/transient_1206.c swh-plugins-0.4.15+1/transient_1206.c
--- swh-plugins-0.4.15+1.orig/transient_1206.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/transient_1206.c	2012-05-13 01:06:56.334745115 +0200
@@ -377,7 +377,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/transient_1206.so.c swh-plugins-0.4.15+1/transient_1206.so.c
--- swh-plugins-0.4.15+1.orig/transient_1206.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/transient_1206.so.c	2012-05-13 01:06:56.334745115 +0200
@@ -377,7 +377,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/triple_para_1204.c swh-plugins-0.4.15+1/triple_para_1204.c
--- swh-plugins-0.4.15+1.orig/triple_para_1204.c	2012-05-13 01:05:58.202460932 +0200
+++ swh-plugins-0.4.15+1/triple_para_1204.c	2012-05-13 01:06:56.338745134 +0200
@@ -375,7 +375,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/triple_para_1204.so.c swh-plugins-0.4.15+1/triple_para_1204.so.c
--- swh-plugins-0.4.15+1.orig/triple_para_1204.so.c	2012-05-13 01:05:58.230461069 +0200
+++ swh-plugins-0.4.15+1/triple_para_1204.so.c	2012-05-13 01:06:56.338745134 +0200
@@ -375,7 +375,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
Gemeinsame Unterverzeichnisse: swh-plugins-0.4.15+1.orig/util und swh-plugins-0.4.15+1/util.
diff -u swh-plugins-0.4.15+1.orig/valve_1209.c swh-plugins-0.4.15+1/valve_1209.c
--- swh-plugins-0.4.15+1.orig/valve_1209.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/valve_1209.c	2012-05-13 01:06:56.338745134 +0200
@@ -250,7 +250,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/valve_1209.so.c swh-plugins-0.4.15+1/valve_1209.so.c
--- swh-plugins-0.4.15+1.orig/valve_1209.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/valve_1209.so.c	2012-05-13 01:06:56.346745181 +0200
@@ -246,7 +246,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/valve_rect_1405.c swh-plugins-0.4.15+1/valve_rect_1405.c
--- swh-plugins-0.4.15+1.orig/valve_rect_1405.c	2012-05-13 01:05:58.178460807 +0200
+++ swh-plugins-0.4.15+1/valve_rect_1405.c	2012-05-13 01:06:56.350745198 +0200
@@ -319,7 +319,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/valve_rect_1405.so.c swh-plugins-0.4.15+1/valve_rect_1405.so.c
--- swh-plugins-0.4.15+1.orig/valve_rect_1405.so.c	2012-05-13 01:05:58.186460843 +0200
+++ swh-plugins-0.4.15+1/valve_rect_1405.so.c	2012-05-13 01:06:56.354745218 +0200
@@ -319,7 +319,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/vynil_1905.c swh-plugins-0.4.15+1/vynil_1905.c
--- swh-plugins-0.4.15+1.orig/vynil_1905.c	2012-05-13 01:05:58.230461069 +0200
+++ swh-plugins-0.4.15+1/vynil_1905.c	2012-05-13 01:06:56.362745256 +0200
@@ -586,7 +586,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/vynil_1905.so.c swh-plugins-0.4.15+1/vynil_1905.so.c
--- swh-plugins-0.4.15+1.orig/vynil_1905.so.c	2012-05-13 01:05:58.202460932 +0200
+++ swh-plugins-0.4.15+1/vynil_1905.so.c	2012-05-13 01:06:56.370745299 +0200
@@ -584,7 +584,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/wave_terrain_1412.c swh-plugins-0.4.15+1/wave_terrain_1412.c
--- swh-plugins-0.4.15+1.orig/wave_terrain_1412.c	2012-05-13 01:05:58.218461009 +0200
+++ swh-plugins-0.4.15+1/wave_terrain_1412.c	2012-05-13 01:06:56.374745316 +0200
@@ -162,7 +162,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/wave_terrain_1412.so.c swh-plugins-0.4.15+1/wave_terrain_1412.so.c
--- swh-plugins-0.4.15+1.orig/wave_terrain_1412.so.c	2012-05-13 01:05:58.238461097 +0200
+++ swh-plugins-0.4.15+1/wave_terrain_1412.so.c	2012-05-13 01:06:56.378745334 +0200
@@ -162,7 +162,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/xfade_1915.c swh-plugins-0.4.15+1/xfade_1915.c
--- swh-plugins-0.4.15+1.orig/xfade_1915.c	2012-05-13 01:05:58.214460980 +0200
+++ swh-plugins-0.4.15+1/xfade_1915.c	2012-05-13 01:06:56.382745355 +0200
@@ -392,7 +392,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/zm1_1428.c swh-plugins-0.4.15+1/zm1_1428.c
--- swh-plugins-0.4.15+1.orig/zm1_1428.c	2012-05-13 01:05:58.178460807 +0200
+++ swh-plugins-0.4.15+1/zm1_1428.c	2012-05-13 01:06:56.386745381 +0200
@@ -170,7 +170,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
diff -u swh-plugins-0.4.15+1.orig/zm1_1428.so.c swh-plugins-0.4.15+1/zm1_1428.so.c
--- swh-plugins-0.4.15+1.orig/zm1_1428.so.c	2012-05-13 01:05:58.174460795 +0200
+++ swh-plugins-0.4.15+1/zm1_1428.so.c	2012-05-13 01:06:56.390745393 +0200
@@ -170,7 +170,6 @@
 
 #ifdef ENABLE_NLS
 #define D_(s) dgettext(PACKAGE, s)
-	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, PACKAGE_LOCALE_DIR);
 #else
 #define D_(s) (s)
