From 076c27e7cb97fafde9a0ce110b2a7c835d49a490 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 2 Nov 2018 12:14:48 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 io-streams-haproxy.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/io-streams-haproxy.cabal b/io-streams-haproxy.cabal
index f95cf45..59776d1 100644
--- a/io-streams-haproxy.cabal
+++ b/io-streams-haproxy.cabal
@@ -35,7 +35,7 @@ library
   other-modules:     System.IO.Streams.Network.Internal.Address
   c-sources:         cbits/byteorder.c
 
-  build-depends:     base              >= 4.5 && < 4.11,
+  build-depends:     base              >= 4.5,
                      attoparsec        >= 0.7 && < 0.14,
                      bytestring        >= 0.9 && < 0.11,
                      io-streams        >= 1.3 && < 1.6,
-- 
2.19.1

