--- src/fe-gtk/xtext.h	2008-08-18 00:23:20.000000000 -0700
+++ src/fe-gtk/xtext.h	2008-08-18 00:23:27.000000000 -0700
@@ -270,6 +270,6 @@
 xtext_buffer *gtk_xtext_buffer_new (GtkXText *xtext);
 void gtk_xtext_buffer_free (xtext_buffer *buf);
 void gtk_xtext_buffer_show (GtkXText *xtext, xtext_buffer *buf, int render);
-GtkType gtk_xtext_get_type (void);
+GType gtk_xtext_get_type (void);
 
 #endif


