diff -Nur xchat-2.6.8/src/fe-text/fe-text.c xchat-2.6.8-hacked/src/fe-text/fe-text.c
--- src/fe-text/fe-text.c	2006-07-16 13:34:21.000000000 +0900
+++ src/fe-text/fe-text.c	2006-10-17 14:55:11.000000000 +0900
@@ -584,7 +584,7 @@
 {
 }
 void
-fe_set_tab_color (struct session *sess, int col, int flash)
+fe_set_tab_color (struct session *sess, int col)
 {
 }
 void
@@ -759,7 +759,7 @@
 {
 }
 void
-fe_lastlog (session * sess, session * lastlog_sess, char *sstr)
+fe_lastlog (session * sess, session * lastlog_sess, char *sstr, gboolean regexp)
 {
 }
 void
@@ -845,3 +845,12 @@
 fe_server_event (server *serv, int type, int arg)
 {
 }
+void
+fe_flash_window (struct session *sess)
+{
+}
+void fe_get_file (const char *title, char *initial,
+				 void (*callback) (void *userdata, char *file), void *userdata,
+				 int flags)
+{
+}
