diff -u -a -b -B -r libggi-2.2.1-orig/programs/check/speed.c libggi-2.2.1/programs/check/speed.c
--- libggi-2.2.1-orig/programs/check/speed.c	2006-04-23 20:40:34.000000000 -0500
+++ libggi-2.2.1/programs/check/speed.c	2006-04-23 20:45:25.000000000 -0500
@@ -31,6 +31,9 @@
 #ifndef CLK_TCK
 #include <time.h>
 #endif
+#ifndef CLK_TCK
+# define  CLK_TCK       sysconf(_SC_CLK_TCK)
+#endif
 
 static struct {
 	ggi_visual_t vis;
