diff -ru cairo-1.12.16.orig/build/configure.ac.warnings cairo-1.12.16/build/configure.ac.warnings
--- cairo-1.12.16.orig/build/configure.ac.warnings	2013-08-26 17:07:21.000000000 +0200
+++ cairo-1.12.16/build/configure.ac.warnings	2014-08-08 08:13:14.566163013 +0200
@@ -39,7 +39,7 @@
 dnl -flto working really needs a test link, not just a compile
 
 safe_MAYBE_WARN="$MAYBE_WARN"
-MAYBE_WARN="$MAYBE_WARN -flto"
+MAYBE_WARN="$MAYBE_WARN -flto -ffat-lto-objects -fuse-linker-plugin"
 AC_TRY_LINK([],[
 	int main(int argc, char **argv) { return 0; }
 ],[],[
Only in cairo-1.12.16/build: configure.ac.warnings.orig
diff -ru cairo-1.12.16.orig/configure cairo-1.12.16/configure
--- cairo-1.12.16.orig/configure	2013-08-26 19:00:41.000000000 +0200
+++ cairo-1.12.16/configure	2014-08-08 08:15:19.576158790 +0200
@@ -18147,7 +18147,7 @@
 
 
 safe_MAYBE_WARN="$MAYBE_WARN"
-MAYBE_WARN="$MAYBE_WARN -flto"
+MAYBE_WARN="$MAYBE_WARN -flto -ffat-lto-objects -fuse-linker-plugin"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
Only in cairo-1.12.16: configure.orig
