From 6078970c70f2cead517115f5310d794fd52e393f Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Tue, 8 Jun 2021 20:42:40 +0200
Subject: [PATCH] Remove dependency on git

---
 CMakeLists.txt | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 600d7ed..1a8db82 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,17 +1,6 @@
 cmake_minimum_required(VERSION 3.0)
 
-find_package(Git QUIET)
-if (Git_FOUND)
-  message("git found")
-  execute_process(
-    COMMAND ${GIT_EXECUTABLE} rev-list --count HEAD
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    OUTPUT_VARIABLE git_version
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-  )
-else()
-  set(git_version 0)
-endif()
+set(git_version 0)
 
 project(ftxui
   LANGUAGES CXX
-- 
2.32.0

