diff -Naur httpd-2.0.53.orig/server/listen.c httpd-2.0.53/server/listen.c
--- httpd-2.0.53.orig/server/listen.c	2005-02-07 13:07:46.808512368 -0800
+++ httpd-2.0.53/server/listen.c	2005-02-07 14:55:45.926536712 -0800
@@ -75,19 +75,6 @@
         return stat;
     }
 
-#if APR_HAVE_IPV6
-    if (server->bind_addr->family == APR_INET6) {
-        stat = apr_socket_opt_set(s, APR_IPV6_V6ONLY, v6only_setting);
-        if (stat != APR_SUCCESS && stat != APR_ENOTIMPL) {
-            ap_log_perror(APLOG_MARK, APLOG_CRIT, stat, p,
-                          "make_sock: for address %pI, apr_socket_opt_set: "
-                          "(IPV6_V6ONLY)",
-                          server->bind_addr);
-            apr_socket_close(s);
-            return stat;
-        }
-    }
-#endif
 
     /*
      * To send data over high bandwidth-delay connections at full
