diff -Naur httpd-2.0.53.orig/server/mpm/prefork/prefork.c httpd-2.0.53/server/mpm/prefork/prefork.c
--- httpd-2.0.53.orig/server/mpm/prefork/prefork.c	2005-02-18 12:58:45.347325288 -0800
+++ httpd-2.0.53/server/mpm/prefork/prefork.c	2005-02-18 12:59:16.689560544 -0800
@@ -382,7 +382,7 @@
 	ap_log_error(APLOG_MARK, APLOG_WARNING, errno, ap_server_conf, "sigaction(SIGXCPU)");
 #endif
 #ifdef SIGXFSZ
-    sa.sa_handler = SIG_DFL;
+    sa.sa_handler = SIG_IGN;
     if (sigaction(SIGXFSZ, &sa, NULL) < 0)
 	ap_log_error(APLOG_MARK, APLOG_WARNING, errno, ap_server_conf, "sigaction(SIGXFSZ)");
 #endif
