diff -Naur httpd-2.0.53.orig/modules/ssl/Makefile.in httpd-2.0.53/modules/ssl/Makefile.in
--- httpd-2.0.53.orig/modules/ssl/Makefile.in	2005-02-07 13:07:46.828509328 -0800
+++ httpd-2.0.53/modules/ssl/Makefile.in	2005-02-07 14:58:19.320217328 -0800
@@ -24,10 +24,6 @@
 #   (we really don't expect end users to use these targets!)
 #
 
-ssl_expr_scan.c: $(top_srcdir)/modules/ssl/ssl_expr_scan.l ssl_expr_parse.h
-	flex -Pssl_expr_yy -s -B $(top_srcdir)/modules/ssl/ssl_expr_scan.l
-	sed -e '/$$Header:/d' <lex.ssl_expr_yy.c >ssl_expr_scan.c && rm -f lex.ssl_expr_yy.c
-
 ssl_expr_parse.c ssl_expr_parse.h: $(top_srcdir)/modules/ssl/ssl_expr_parse.y
 	yacc -d $(top_srcdir)/modules/ssl/ssl_expr_parse.y
 	sed -e 's;yy;ssl_expr_yy;g' \
