--- httpd-2.0.54.orig/modules/ssl/ssl_toolkit_compat.h	2005-02-04 12:21:18.000000000 -0800
+++ httpd-2.0.54/modules/ssl/ssl_toolkit_compat.h	2005-07-06 08:50:38.468750000 -0700
@@ -24,6 +24,12 @@
 
 #ifdef OPENSSL_VERSION_NUMBER
 
+#if (OPENSSL_VERSION_NUMBER >= 0x0090800fL)
+#ifndef PEM_F_DEF_CALLBACK
+#define PEM_F_DEF_CALLBACK PEM_F_PEM_DEF_CALLBACK 
+#endif
+#endif
+
 /*
  * rsa sslc uses incomplete types for most structures
  * so we macroize for OpenSSL those which cannot be dereferenced
