diff -Naur httpd-2.0.53.orig/buildconf httpd-2.0.53/buildconf
--- httpd-2.0.53.orig/buildconf	2005-02-07 13:07:47.183455368 -0800
+++ httpd-2.0.53/buildconf	2005-02-07 13:09:11.810590088 -0800
@@ -66,7 +66,7 @@
     should_exit=1
 fi
 
-if [ ! -d "$apu_src_dir" -o ! -f "$apu_src_dir/Makefile.in" ]; then
+if [ ! -d "$apu_src_dir" -o ! -f "$apu_src_dir/build/find_apu.m4" ]; then
     echo ""
     echo "You don't have a copy of the apr-util source in $apu_src_dir. "
     echo "Please get one the source using the following instructions, "
