diff -urN httpd-2.0.58/buildconf httpd/buildconf
--- httpd-2.0.58/buildconf	2006-04-24 13:12:21.000000000 -0400
+++ httpd/buildconf	2006-05-15 22:28:32.000000000 -0400
@@ -67,7 +67,7 @@
     should_exit=1
 fi
 
-if [ ! -f "$apu_src_dir/Makefile.in" ]; then
+if [ ! -f "$apu_src_dir/build/apu-conf.m4" ]; then
     echo ""
     echo "You don't have a copy of the apr-util source in $apu_src_dir. "
     echo "Please get one the source using the following instructions, "
