From 3a1af8afcafbc83b8f087833ae845829aa67765b Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Mon, 18 Apr 2022 19:04:44 +0200
Subject: [PATCH 2/4] Fix cleanup of shared secret

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 vpnc.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/vpnc.c b/vpnc.c
index 39eb40e86fa5..474d94029c89 100644
--- a/vpnc.c
+++ b/vpnc.c
@@ -1853,9 +1853,11 @@ static void do_phase1_am(const char *key_id, const char *shared_key, struct sa_b
 			static const unsigned char c012[3] = { 0, 1, 2 };
 			unsigned char *skeyid_e;
 			unsigned char *dh_shared_secret;
+			size_t dh_shared_secret_len;
 
 			/* Determine the shared secret.  */
-			dh_shared_secret = xallocc(dh_getlen(dh_grp));
+			dh_shared_secret_len = dh_getlen(dh_grp);
+			dh_shared_secret = xallocc(dh_shared_secret_len);
 			dh_create_shared(dh_grp, dh_shared_secret, ke->u.ke.data);
 			hex_dump("dh_shared_secret", dh_shared_secret, dh_getlen(dh_grp), NULL);
 
@@ -1899,7 +1901,7 @@ static void do_phase1_am(const char *key_id, const char *shared_key, struct sa_b
 			gcry_md_close(hm);
 			hex_dump("skeyid_e", skeyid_e, s->ike.md_len, NULL);
 
-			memset(dh_shared_secret, 0, sizeof(dh_shared_secret));
+			memset(dh_shared_secret, 0, dh_shared_secret_len);
 			free(dh_shared_secret);
 
 			/* Determine the IKE encryption key.  */
-- 
2.35.3

