From f7cb5f93d85063590841d60bb10cf7b306d0c246 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Mon, 18 Apr 2022 19:07:40 +0200
Subject: [PATCH 4/5] Fix debug message format string

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 isakmp-pkt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/isakmp-pkt.c b/isakmp-pkt.c
index d241ad8c3a2a..0d5cb714fe3d 100644
--- a/isakmp-pkt.c
+++ b/isakmp-pkt.c
@@ -820,7 +820,7 @@ struct isakmp_packet *parse_isakmp_packet(const uint8_t * data, size_t data_len,
 	}
 
 	DEBUG(3, printf("BEGIN_PARSE\n"));
-	DEBUG(3, printf("Recieved Packet Len: %d\n", data_len));
+	DEBUG(3, printf("Recieved Packet Len: %zu\n", data_len));
 	fetchn(r->i_cookie, ISAKMP_COOKIE_LENGTH);
 	hex_dump("i_cookie", r->i_cookie, ISAKMP_COOKIE_LENGTH, NULL);
 	fetchn(r->r_cookie, ISAKMP_COOKIE_LENGTH);
-- 
2.35.3

