diff -r -U2 nc.ori/Makefile nc/Makefile
--- nc.ori/Makefile	Wed Mar 20 04:16:06 1996
+++ nc/Makefile	Sat Feb  9 14:22:52 2002
@@ -10,4 +10,5 @@
 # debugging
 # DFLAGS = -DTEST -DDEBUG
+DFLAGS = -DTELNET -DGAPING_SECURITY_HOLE
 CFLAGS = -O
 XFLAGS = 	# xtra cflags, set by systype targets
@@ -68,5 +69,5 @@
 
 linux:
-	make -e $(ALL) $(MFLAGS) XFLAGS='-DLINUX' STATIC=-static
+	make -e $(ALL) $(MFLAGS) XFLAGS='-DLINUX' STATIC=
 
 # irix 5.2, dunno 'bout earlier versions.  If STATIC='-non_shared' doesn't
diff -r -U2 nc.ori/netcat.c nc/netcat.c
--- nc.ori/netcat.c	Thu Mar 21 01:38:04 1996
+++ nc/netcat.c	Sat Feb  9 12:04:18 2002
@@ -28,4 +28,5 @@
 
 #include "generic.h"		/* same as with L5, skey, etc */
+#include <resolv.h>
 
 /* conditional includes -- a very messy section which you may have to dink
