diff -urN iptraf-2.7.0.orig/src/othptab.c iptraf-2.7.0/src/othptab.c
--- iptraf-2.7.0.orig/src/othptab.c	2001-12-28 03:23:59.000000000 +0100
+++ iptraf-2.7.0/src/othptab.c	2005-10-22 18:39:15.000000000 +0200
@@ -17,13 +17,10 @@
 ***/
 
 #include <asm/types.h>
-#include <linux/if_ether.h>
-#include <linux/if_tr.h>
-#include <linux/if_fddi.h>
 #include <winops.h>
+#include "tcptable.h"
 #include "arphdr.h"
 #include "options.h"
-#include "tcptable.h"
 #include "othptab.h"
 #include "deskman.h"
 #include "attrs.h"
diff -urN iptraf-2.7.0.orig/src/packet.c iptraf-2.7.0/src/packet.c
--- iptraf-2.7.0.orig/src/packet.c	2002-05-08 11:44:08.000000000 +0200
+++ iptraf-2.7.0/src/packet.c	2005-10-22 18:39:15.000000000 +0200
@@ -32,6 +32,9 @@
 #include <sys/time.h>
 #include <net/if_arp.h>
 #include <net/if.h>
+#ifdef _NET_IF_H
+#define  _LINUX_IF_H
+#endif
 #include <sys/ioctl.h>
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
diff -urN iptraf-2.7.0.orig/src/tcptable.h iptraf-2.7.0/src/tcptable.h
--- iptraf-2.7.0.orig/src/tcptable.h	2001-12-28 03:16:39.000000000 +0100
+++ iptraf-2.7.0/src/tcptable.h	2005-10-22 18:39:15.000000000 +0200
@@ -20,11 +20,15 @@
 #include <arpa/inet.h>
 #include <net/if_arp.h>
 #include <asm/types.h>
+#include <net/if.h>
+#ifdef _NET_IF_H
+#define  _LINUX_IF_H
+#endif
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
 #include <linux/if_fddi.h>
 #include <linux/if_tr.h>
-#include <net/if.h>
+//#include <net/if.h>
 #include <netinet/ip.h>
 #include <netinet/udp.h>
 #include "servname.h"
