diff -Naur iptraf-3.0.0_orig/src/install.sh iptraf-3.0.0/src/install.sh
--- iptraf-3.0.0_orig/src/install.sh	2002-12-26 04:07:18.000000000 +0100
+++ iptraf-3.0.0/src/install.sh	2006-08-21 21:07:42.000000000 +0200
@@ -85,7 +85,7 @@
 echo "this version. You can view this file now (will require the 'less'"
 echo "program in /usr/bin.  Press Q to quit when done)."
 echo
-echo -n "Would you like to view the RELEASE-NOTES file now (Y/N)? "; read YESNO
+echo -n "Would you like to view the RELEASE-NOTES file now (Y/N)? "; YESNO == y
 
 if [ "$YESNO" = "y" -o "$YESNO" = "Y" ]; then
     less ../RELEASE-NOTES
diff -Naur iptraf-3.0.0_orig/src/Makefile iptraf-3.0.0/src/Makefile
--- iptraf-3.0.0_orig/src/Makefile	2005-09-13 11:11:17.000000000 +0200
+++ iptraf-3.0.0/src/Makefile	2006-08-21 21:07:42.000000000 +0200
@@ -68,12 +68,12 @@
 # installation target directory.  The iptraf and rvnamed programs get
 # stored here.  iptraf also exec's rvnamed from this directory.
 
-TARGET		= /usr/local/bin
+TARGET		= /usr/bin
 
 # The IPTraf working directory; if you change this.  Starting with this
 # version, this definition overrides dirs.h.
 
-WORKDIR		= /var/local/iptraf
+WORKDIR		= /var/iptraf
 
 # The IPTraf log file directory.  IPTraf log files are placed here.
 # This definition overrides dirs.h
diff -Naur iptraf-3.0.0_orig/src/othptab.c iptraf-3.0.0/src/othptab.c
--- iptraf-3.0.0_orig/src/othptab.c	2005-09-13 08:42:54.000000000 +0200
+++ iptraf-3.0.0/src/othptab.c	2006-08-21 21:14:49.000000000 +0200
@@ -18,8 +18,8 @@
 
 #include <asm/types.h>
 #include <linux/if_ether.h>
-#include <linux/if_tr.h>
-#include <linux/if_fddi.h>
+/*#include <linux/if_tr.h>*/
+/*#include <linux/if_fddi.h>*/
 #include <winops.h>
 #include "arphdr.h"
 #include "options.h"
diff -Naur iptraf-3.0.0_orig/src/tcptable.h iptraf-3.0.0/src/tcptable.h
--- iptraf-3.0.0_orig/src/tcptable.h	2005-09-13 08:42:54.000000000 +0200
+++ iptraf-3.0.0/src/tcptable.h	2006-08-21 21:13:25.000000000 +0200
@@ -20,11 +20,11 @@
 #include <arpa/inet.h>
 #include <net/if_arp.h>
 #include <asm/types.h>
+#include <net/if.h>
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
 #include <linux/if_fddi.h>
 #include <linux/if_tr.h>
-#include <net/if.h>
 #include <netinet/ip.h>
 #include <netinet/udp.h>
 #include "servname.h"
