From 1a05cd9d1d5dd743416aa3554408a118a832aa51 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 26 Oct 2012 22:22:51 +0200
Subject: [PATCH 1/2] add missing header

---
 src/celeste/Utilities.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/celeste/Utilities.cpp b/src/celeste/Utilities.cpp
index 56bb2f7..6e6c2e6 100755
--- a/src/celeste/Utilities.cpp
+++ b/src/celeste/Utilities.cpp
@@ -27,6 +27,7 @@ Copyright (c) 2002-3 Adriaan Tijsseling
 
 #include <math.h>
 #include <stdlib.h>
+#include <unistd.h>
 #include "Utilities.h"
 
 using namespace std; 
@@ -285,4 +286,4 @@ int SafeAbs( int val )
 	else return val;
 }
 
-}; // namespace
\ No newline at end of file
+}; // namespace
-- 
1.8.0

