diff -ur hugin-0.8.0/src/hugin_base/vigra_ext/ImageTransforms.h hugin-0.8.0-patched/src/hugin_base/vigra_ext/ImageTransforms.h
--- hugin-0.8.0/src/hugin_base/vigra_ext/ImageTransforms.h	2009-07-08 07:08:48.000000000 +0900
+++ hugin-0.8.0-patched/src/hugin_base/vigra_ext/ImageTransforms.h	2009-09-17 08:22:46.837407980 +0900
@@ -42,6 +42,7 @@
 #include "MultiThreadOperations.h"
 #include <boost/thread/thread.hpp>
 #include <boost/bind.hpp>
+#include <boost/function.hpp>
 
 namespace vigra_ext
 {
