diff --color -urp libdbusmenu-16.04.0.orig/libdbusmenu-glib/Makefile.am libdbusmenu-16.04.0/libdbusmenu-glib/Makefile.am
--- libdbusmenu-16.04.0.orig/libdbusmenu-glib/Makefile.am	2016-02-27 12:58:43.000000000 -0600
+++ libdbusmenu-16.04.0/libdbusmenu-glib/Makefile.am	2022-11-26 09:09:29.156611178 -0600
@@ -66,7 +66,7 @@ libdbusmenu_glib_la_LDFLAGS = \
 libdbusmenu_glib_la_CFLAGS = \
 	$(DBUSMENUGLIB_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
-	-Wall -Werror -Wno-error=deprecated-declarations \
+	-Wall -Wno-error=deprecated-declarations \
 	-DG_LOG_DOMAIN="\"LIBDBUSMENU-GLIB\""
 
 libdbusmenu_glib_la_LIBADD = \
diff --color -urp libdbusmenu-16.04.0.orig/libdbusmenu-glib/Makefile.in libdbusmenu-16.04.0/libdbusmenu-glib/Makefile.in
--- libdbusmenu-16.04.0.orig/libdbusmenu-glib/Makefile.in	2016-02-27 13:00:13.000000000 -0600
+++ libdbusmenu-16.04.0/libdbusmenu-glib/Makefile.in	2022-11-26 09:09:40.889687102 -0600
@@ -498,7 +498,7 @@ libdbusmenu_glib_la_LDFLAGS = \
 libdbusmenu_glib_la_CFLAGS = \
 	$(DBUSMENUGLIB_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
-	-Wall -Werror -Wno-error=deprecated-declarations \
+	-Wall -Wno-error=deprecated-declarations \
 	-DG_LOG_DOMAIN="\"LIBDBUSMENU-GLIB\""
 
 libdbusmenu_glib_la_LIBADD = \
diff --color -urp libdbusmenu-16.04.0.orig/libdbusmenu-gtk/Makefile.am libdbusmenu-16.04.0/libdbusmenu-gtk/Makefile.am
--- libdbusmenu-16.04.0.orig/libdbusmenu-gtk/Makefile.am	2016-02-27 12:58:43.000000000 -0600
+++ libdbusmenu-16.04.0/libdbusmenu-gtk/Makefile.am	2022-11-26 09:08:09.300088399 -0600
@@ -66,7 +66,7 @@ libdbusmenu_gtk_la_CFLAGS = \
 	$(DBUSMENUGTK_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-I$(top_srcdir) \
-	-Wall -Werror -Wno-error=deprecated-declarations \
+	-Wall -Wno-error=deprecated-declarations \
 	-DG_LOG_DOMAIN="\"LIBDBUSMENU-GTK\""
 
 libdbusmenu_gtk_la_LIBADD = \
diff --color -urp libdbusmenu-16.04.0.orig/libdbusmenu-gtk/Makefile.in libdbusmenu-16.04.0/libdbusmenu-gtk/Makefile.in
--- libdbusmenu-16.04.0.orig/libdbusmenu-gtk/Makefile.in	2016-02-27 13:00:14.000000000 -0600
+++ libdbusmenu-16.04.0/libdbusmenu-gtk/Makefile.in	2022-11-26 09:08:21.151166683 -0600
@@ -507,7 +507,7 @@ libdbusmenu_gtk_la_CFLAGS = \
 	$(DBUSMENUGTK_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-I$(top_srcdir) \
-	-Wall -Werror -Wno-error=deprecated-declarations \
+	-Wall -Wno-error=deprecated-declarations \
 	-DG_LOG_DOMAIN="\"LIBDBUSMENU-GTK\""
 
 libdbusmenu_gtk_la_LIBADD = \
diff --color -urp libdbusmenu-16.04.0.orig/tests/Makefile.am libdbusmenu-16.04.0/tests/Makefile.am
--- libdbusmenu-16.04.0.orig/tests/Makefile.am	2016-02-27 12:58:43.000000000 -0600
+++ libdbusmenu-16.04.0/tests/Makefile.am	2022-11-26 09:09:01.010428150 -0600
@@ -103,7 +103,7 @@ TESTS_ENVIRONMENT = env GI_TYPELIB_PATH=
 ############################################
 
 DBUSMENU_GLIB_TEST_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	-DG_DISABLE_DEPRECATED \
 	-I$(top_srcdir) \
 	$(DBUSMENUTESTS_CFLAGS) \
diff --color -urp libdbusmenu-16.04.0.orig/tests/Makefile.in libdbusmenu-16.04.0/tests/Makefile.in
--- libdbusmenu-16.04.0.orig/tests/Makefile.in	2016-02-27 13:00:14.000000000 -0600
+++ libdbusmenu-16.04.0/tests/Makefile.in	2022-11-26 09:09:10.014486843 -0600
@@ -979,7 +979,7 @@ TESTS_ENVIRONMENT = env GI_TYPELIB_PATH=
 # Shared vars for the dbusmenu-glib tests
 ############################################
 DBUSMENU_GLIB_TEST_CFLAGS = \
-	-Wall -Werror \
+	-Wall \
 	-DG_DISABLE_DEPRECATED \
 	-I$(top_srcdir) \
 	$(DBUSMENUTESTS_CFLAGS) \
diff --color -urp libdbusmenu-16.04.0.orig/tools/Makefile.am libdbusmenu-16.04.0/tools/Makefile.am
--- libdbusmenu-16.04.0.orig/tools/Makefile.am	2016-02-27 12:58:43.000000000 -0600
+++ libdbusmenu-16.04.0/tools/Makefile.am	2022-11-26 09:06:39.844488726 -0600
@@ -16,7 +16,7 @@ dbusmenu_dumper_CFLAGS = \
 	-I $(srcdir)/.. \
 	$(DBUSMENUGLIB_CFLAGS) \
 	$(DBUSMENUDUMPER_CFLAGS) \
-	-Wall -Werror
+	-Wall
 
 dbusmenu_dumper_LDADD = \
 	../libdbusmenu-glib/libdbusmenu-glib.la \
diff --color -urp libdbusmenu-16.04.0.orig/tools/Makefile.in libdbusmenu-16.04.0/tools/Makefile.in
--- libdbusmenu-16.04.0.orig/tools/Makefile.in	2016-02-27 13:00:14.000000000 -0600
+++ libdbusmenu-16.04.0/tools/Makefile.in	2022-11-26 09:06:56.877604179 -0600
@@ -454,7 +454,7 @@ dbusmenu_dumper_CFLAGS = \
 	-I $(srcdir)/.. \
 	$(DBUSMENUGLIB_CFLAGS) \
 	$(DBUSMENUDUMPER_CFLAGS) \
-	-Wall -Werror
+	-Wall
 
 dbusmenu_dumper_LDADD = \
 	../libdbusmenu-glib/libdbusmenu-glib.la \
diff --color -urp libdbusmenu-16.04.0.orig/tools/testapp/Makefile.am libdbusmenu-16.04.0/tools/testapp/Makefile.am
--- libdbusmenu-16.04.0.orig/tools/testapp/Makefile.am	2016-02-27 12:58:43.000000000 -0600
+++ libdbusmenu-16.04.0/tools/testapp/Makefile.am	2022-11-26 09:07:10.653697102 -0600
@@ -12,7 +12,7 @@ dbusmenu_testapp_CFLAGS = \
 	-I $(srcdir)/../.. \
 	$(DBUSMENUTESTS_CFLAGS) \
 	$(DBUSMENUGLIB_CFLAGS) \
-	-Wall -Werror
+	-Wall
 
 dbusmenu_testapp_LDADD = \
 	$(builddir)/../../libdbusmenu-glib/libdbusmenu-glib.la \
diff --color -urp libdbusmenu-16.04.0.orig/tools/testapp/Makefile.in libdbusmenu-16.04.0/tools/testapp/Makefile.in
--- libdbusmenu-16.04.0.orig/tools/testapp/Makefile.in	2016-02-27 13:00:14.000000000 -0600
+++ libdbusmenu-16.04.0/tools/testapp/Makefile.in	2022-11-26 09:07:21.954773031 -0600
@@ -377,7 +377,7 @@ dbusmenu_testapp_CFLAGS = \
 	-I $(srcdir)/../.. \
 	$(DBUSMENUTESTS_CFLAGS) \
 	$(DBUSMENUGLIB_CFLAGS) \
-	-Wall -Werror
+	-Wall
 
 dbusmenu_testapp_LDADD = \
 	$(builddir)/../../libdbusmenu-glib/libdbusmenu-glib.la \
