--- src/cut-n-paste-code/goffice/graph/gog-axis.c~	2005-09-18 15:54:03.000000000 +0200
+++ src/cut-n-paste-code/goffice/graph/gog-axis.c	2005-09-18 15:54:06.000000000 +0200
@@ -1677,8 +1677,8 @@
 	iface->dim_changed = gog_axis_dim_changed;
 }
 
-GSF_CLASS_FULL (GogAxis, gog_axis,
-		gog_axis_class_init, gog_axis_init,
+GSF_CLASS_FULL (GogAxis, gog_axis, NULL, NULL,
+		gog_axis_class_init, NULL, gog_axis_init,
 		GOG_STYLED_OBJECT_TYPE, 0,
 		GSF_INTERFACE (gog_axis_dataset_init, GOG_DATASET_TYPE))
 
--- src/cut-n-paste-code/goffice/graph/gog-error-bar.c~	2004-12-02 14:57:26.000000000 +0100
+++ src/cut-n-paste-code/goffice/graph/gog-error-bar.c	2005-09-18 15:57:43.000000000 +0200
@@ -487,8 +487,8 @@
 	iface->sax_save = gog_error_bar_persist_sax_save;
 }
 
-GSF_CLASS_FULL (GogErrorBar, gog_error_bar,
-		gog_error_bar_class_init, gog_error_bar_init,
+GSF_CLASS_FULL (GogErrorBar, gog_error_bar, NULL, NULL,
+		gog_error_bar_class_init, NULL, gog_error_bar_init,
 		G_TYPE_OBJECT, 0,
 		GSF_INTERFACE (gog_error_bar_persist_init, GOG_PERSIST_TYPE))
 
--- src/cut-n-paste-code/goffice/graph/gog-label.c~	2005-09-18 15:53:47.000000000 +0200
+++ src/cut-n-paste-code/goffice/graph/gog-label.c	2005-09-18 15:53:50.000000000 +0200
@@ -175,8 +175,8 @@
 	iface->dim_changed = gog_label_dim_changed;
 }
 
-GSF_CLASS_FULL (GogLabel, gog_label,
-		gog_label_class_init, NULL,
+GSF_CLASS_FULL (GogLabel, gog_label, NULL, NULL,
+		gog_label_class_init, NULL, NULL,
 		GOG_OUTLINED_OBJECT_TYPE, 0,
 		GSF_INTERFACE (gog_label_dataset_init, GOG_DATASET_TYPE))
 
--- src/cut-n-paste-code/goffice/graph/gog-series.c~	2005-01-05 19:05:53.000000000 +0100
+++ src/cut-n-paste-code/goffice/graph/gog-series.c	2005-09-18 15:56:10.000000000 +0200
@@ -588,8 +588,8 @@
 	iface->dim_changed = gog_series_dataset_dim_changed;
 }
 
-GSF_CLASS_FULL (GogSeries, gog_series,
-		gog_series_class_init, gog_series_init,
+GSF_CLASS_FULL (GogSeries, gog_series, NULL, NULL,
+		gog_series_class_init, NULL, gog_series_init,
 		GOG_STYLED_OBJECT_TYPE, 0,
 		GSF_INTERFACE (gog_series_dataset_init, GOG_DATASET_TYPE))
 
--- src/cut-n-paste-code/goffice/graph/gog-style.c~	2004-11-19 16:49:12.000000000 +0100
+++ src/cut-n-paste-code/goffice/graph/gog-style.c	2005-09-18 15:55:21.000000000 +0200
@@ -1686,8 +1686,8 @@
 	iface->sax_save = gog_style_persist_sax_save;
 }
 
-GSF_CLASS_FULL (GogStyle, gog_style,
-		gog_style_class_init, gog_style_init,
+GSF_CLASS_FULL (GogStyle, gog_style, NULL, NULL,
+		gog_style_class_init, NULL, gog_style_init,
 		G_TYPE_OBJECT, 0,
 		GSF_INTERFACE (gog_style_persist_init, GOG_PERSIST_TYPE))
 
--- src/widgets/gnumeric-expr-entry.c~	2004-09-07 21:28:41.000000000 +0200
+++ src/widgets/gnumeric-expr-entry.c	2005-09-18 16:17:07.000000000 +0200
@@ -502,8 +502,8 @@
 	iface->start_editing = gee_start_editing;
 }
 
-GSF_CLASS_FULL (GnmExprEntry, gnm_expr_entry,
-		gee_class_init, gee_init,
+GSF_CLASS_FULL (GnmExprEntry, gnm_expr_entry, NULL, NULL,
+		gee_class_init, NULL, gee_init,
 		GTK_TYPE_HBOX, 0,
 		GSF_INTERFACE (gee_cell_editable_init, GTK_TYPE_CELL_EDITABLE))
 
--- src/command-context-stderr.c~	2004-06-13 20:48:34.000000000 +0200
+++ src/command-context-stderr.c	2005-09-18 16:19:13.000000000 +0200
@@ -106,7 +106,7 @@
 	cc_class->error.error_info	= ccs_error_info;
 }
 
-GSF_CLASS_FULL (CmdContextStderr, cmd_context_stderr,
-		NULL, ccs_init,
+GSF_CLASS_FULL (CmdContextStderr, cmd_context_stderr, NULL, NULL,
+		NULL, NULL, ccs_init,
 		G_TYPE_OBJECT, 0,
 		GSF_INTERFACE (ccs_gnm_cmd_context_init, GNM_CMD_CONTEXT_TYPE))
--- src/io-context.c~	2004-07-20 02:37:13.000000000 +0200
+++ src/io-context.c	2005-09-18 16:20:05.000000000 +0200
@@ -116,8 +116,8 @@
 	klass->finalize = ioc_finalize;
 }
 
-GSF_CLASS_FULL (IOContext, io_context,
-		io_context_class_init, io_context_init,
+GSF_CLASS_FULL (IOContext, io_context, NULL, NULL,
+		io_context_class_init, NULL, io_context_init,
 		G_TYPE_OBJECT, 0,
 		GSF_INTERFACE (io_context_gnm_cmd_context_init, GNM_CMD_CONTEXT_TYPE))
 
--- src/sheet-filter.c~	2004-10-17 04:36:05.000000000 +0200
+++ src/sheet-filter.c	2005-09-18 16:25:33.000000000 +0200
@@ -562,8 +562,8 @@
 }
 typedef FooCanvasWidget		FilterFooView;
 typedef FooCanvasWidgetClass	FilterFooViewClass;
-static GSF_CLASS_FULL (FilterFooView, filter_foo_view,
-	NULL, NULL,
+static GSF_CLASS_FULL (FilterFooView, filter_foo_view, NULL, NULL,
+	NULL, NULL, NULL,
 	FOO_TYPE_CANVAS_WIDGET, 0,
 	GSF_INTERFACE (filter_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 
--- src/sheet-object-cell-comment.c~	2005-01-14 06:48:29.000000000 +0100
+++ src/sheet-object-cell-comment.c	2005-09-18 16:26:28.000000000 +0200
@@ -122,8 +122,8 @@
 }
 typedef FooCanvasPolygon	CommentFooView;
 typedef FooCanvasPolygonClass	CommentFooViewClass;
-static GSF_CLASS_FULL (CommentFooView, comment_foo_view,
-	NULL, NULL,
+static GSF_CLASS_FULL (CommentFooView, comment_foo_view, NULL, NULL,
+	NULL, NULL, NULL,
 	FOO_TYPE_CANVAS_POLYGON, 0,
 	GSF_INTERFACE (comment_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 
--- src/gnm-so-filled.c~	2005-09-18 16:27:52.000000000 +0200
+++ src/gnm-so-filled.c	2005-09-18 16:28:06.000000000 +0200
@@ -112,8 +112,8 @@
 }
 typedef FooCanvasGroup		FilledFooView;
 typedef FooCanvasGroupClass	FilledFooViewClass;
-static GSF_CLASS_FULL (FilledFooView, so_filled_foo_view,
-	NULL, NULL,
+static GSF_CLASS_FULL (FilledFooView, so_filled_foo_view, NULL, NULL,
+	NULL, NULL, NULL,
 	FOO_TYPE_CANVAS_GROUP, 0,
 	GSF_INTERFACE (so_filled_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 #endif /* WITH_GTK */
@@ -631,8 +631,8 @@
 }
 typedef FooCanvasPolygon	PolygonFooView;
 typedef FooCanvasPolygonClass	PolygonFooViewClass;
-static GSF_CLASS_FULL (PolygonFooView, so_polygon_foo_view,
-	NULL, NULL,
+static GSF_CLASS_FULL (PolygonFooView, so_polygon_foo_view, NULL, NULL,
+	NULL, NULL, NULL,
 	FOO_TYPE_CANVAS_POLYGON, 0,
 	GSF_INTERFACE (so_polygon_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 #endif /* WITH_GTK */
--- src/gnm-so-line.c~	2004-11-30 19:37:07.000000000 +0100
+++ src/gnm-so-line.c	2005-09-18 16:28:51.000000000 +0200
@@ -110,8 +110,8 @@
 }
 typedef FooCanvasLine		LineFooView;
 typedef FooCanvasLineClass	LineFooViewClass;
-static GSF_CLASS_FULL (LineFooView, so_line_foo_view,
-	NULL, NULL,
+static GSF_CLASS_FULL (LineFooView, so_line_foo_view, NULL, NULL,
+	NULL, NULL, NULL,
 	FOO_TYPE_CANVAS_LINE, 0,
 	GSF_INTERFACE (so_line_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 #endif /* WITH_GTK */
--- src/sheet-object-graph.c~	2004-10-29 06:13:13.000000000 +0200
+++ src/sheet-object-graph.c	2005-09-18 16:29:30.000000000 +0200
@@ -93,8 +93,8 @@
 }
 typedef GogControlFooCanvas		SOGraphFooView;
 typedef GogControlFooCanvasClass	SOGraphFooViewClass;
-static GSF_CLASS_FULL (SOGraphFooView, so_graph_foo_view,
-	NULL, NULL,
+static GSF_CLASS_FULL (SOGraphFooView, so_graph_foo_view, NULL, NULL,
+	NULL, NULL, NULL,
 	GOG_CONTROL_FOOCANVAS_TYPE, 0,
 	GSF_INTERFACE (so_graph_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 
--- src/sheet-object-image.c~	2004-12-18 05:48:37.000000000 +0100
+++ src/sheet-object-image.c	2005-09-18 16:30:39.000000000 +0200
@@ -89,8 +89,8 @@
 }
 typedef FooCanvasPixbuf		SOImageFooView;
 typedef FooCanvasPixbufClass	SOImageFooViewClass;
-static GSF_CLASS_FULL (SOImageFooView, so_image_foo_view,
-	NULL, NULL,
+static GSF_CLASS_FULL (SOImageFooView, so_image_foo_view, NULL, NULL,
+	NULL, NULL, NULL,
 	FOO_TYPE_CANVAS_PIXBUF, 0,
 	GSF_INTERFACE (so_image_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 
--- src/sheet-object-widget.c~	2005-02-09 22:54:29.000000000 +0100
+++ src/sheet-object-widget.c	2005-09-18 16:31:20.000000000 +0200
@@ -98,8 +98,8 @@
 }
 typedef FooCanvasWidget		SOWidgetFooView;
 typedef FooCanvasWidgetClass	SOWidgetFooViewClass;
-static GSF_CLASS_FULL (SOWidgetFooView, so_widget_foo_view,
-	NULL, NULL,
+static GSF_CLASS_FULL (SOWidgetFooView, so_widget_foo_view, NULL, NULL,
+	NULL, NULL, NULL,
 	FOO_TYPE_CANVAS_WIDGET, 0,
 	GSF_INTERFACE (so_widget_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 
--- src/workbook-control-gui.c~	2005-03-13 01:05:45.000000000 +0100
+++ src/workbook-control-gui.c	2005-09-18 16:33:18.000000000 +0200
@@ -2544,8 +2544,8 @@
 	wbcg->current_saver = NULL;
 }
 
-GSF_CLASS_FULL (WorkbookControlGUI, workbook_control_gui,
-		workbook_control_gui_class_init, workbook_control_gui_init,
+GSF_CLASS_FULL (WorkbookControlGUI, workbook_control_gui, NULL, NULL,
+		workbook_control_gui_class_init, NULL, workbook_control_gui_init,
 		WORKBOOK_CONTROL_TYPE, G_TYPE_FLAG_ABSTRACT,
 		GSF_INTERFACE (wbcg_go_plot_data_allocator_init, GOG_DATA_ALLOCATOR_TYPE);
 		GSF_INTERFACE (wbcg_gnm_cmd_context_init, GNM_CMD_CONTEXT_TYPE))
--- src/io-context-gtk.c~	2004-10-30 05:51:35.000000000 +0200
+++ src/io-context-gtk.c	2005-09-19 15:58:07.000000000 +0200
@@ -366,8 +366,8 @@
 	g_timer_start (icg->timer);
 }
 
-GSF_CLASS_FULL (IOContextGtk, io_context_gtk,
-		icg_class_init, icg_init,
+GSF_CLASS_FULL (IOContextGtk, io_context_gtk, NULL, NULL,
+		icg_class_init, NULL, icg_init,
 		TYPE_IO_CONTEXT, 0,
 		GSF_INTERFACE (icg_gnm_cmd_context_init, GNM_CMD_CONTEXT_TYPE))
 
--- plugins/corba/corba-workbook.c~	2004-09-24 22:35:15.000000000 +0200
+++ plugins/corba/corba-workbook.c	2005-09-19 16:18:27.000000000 +0200
@@ -312,8 +312,8 @@
 	CORBA_exception_free (&ev);
 }
 
-GSF_CLASS_FULL (WorkbookControlCORBA, workbook_control_corba,
-		wbcc_class_init, wbcc_init, 
+GSF_CLASS_FULL (WorkbookControlCORBA, workbook_control_corba, NULL, NULL,
+		wbcc_class_init, NULL, wbcc_init, 
 		WORKBOOK_CONTROL_TYPE, 0,
 		GSF_INTERFACE (wbcc_gnm_cmd_context_init,
 			       GNM_CMD_CONTEXT_TYPE))
