--- vile-9.8l+/filters/filters.h	2013-12-07 11:54:41.000000000 +0000
+++ vile-9.8m/filters/filters.h	2014-03-30 18:22:19.000000000 +0000
@@ -235,13 +235,15 @@
 extern FILE *yyget_out (void);
 extern char *yyget_text (void);
 extern int yyget_debug (void);
-extern int yyget_leng (void);
 extern int yyget_lineno (void);
 extern int yylex_destroy (void);
 extern void yyset_debug (int bdebug);
 extern void yyset_in (FILE * in_str);
 extern void yyset_lineno (int line_number);
 extern void yyset_out (FILE * out_str);
+#if !defined(YY_FLEX_SUBMINOR_VERSION) || (YY_FLEX_SUBMINOR_VERSION < 37)
+extern int yyget_leng (void);
+#endif
 /* there's also warnings for unused 'yyunput()', but I don't see a fix */
 /* flex's skeleton includes <unistd.h> - no particular reason apparent */
 #endif
