From 8bd714835159aa32a64343a8dc41684d7c36ab43 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston<jeremyhu@freedesktop.org>
Date: Fri, 28  Mar  2008  03:07:52  +0000
Subject: Build fix for file systems that are not case sensitive

---
diff --git a/Makefile.am b/Makefile.am
index a2931af..007ade8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -48,7 +48,7 @@ appdefaultdir = @appdefaultdir@
 
 
 APPDEFAULTFILES = \
-        XMore
+        app-defaults/XMore
 
 SUFFIXES = .ad
 
--
cgit v0.8.1-24-ge5fb
