From a85334f153e05119f237ff412a2ff15dce95878d Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston<jeremyhu@freedesktop.org>
Date: Fri, 28  Mar  2008  03:29:51  +0000
Subject: Added missing AM_PROG_CC_C_O

---
diff --git a/configure.ac b/configure.ac
index 5c233b6..c07e54c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -29,6 +29,7 @@ AM_MAINTAINER_MODE
 AM_CONFIG_HEADER(config.h)
 
 AC_PROG_CC
+AM_PROG_CC_C_O
 AC_PROG_INSTALL
 
 # Checks for pkg-config packages
--
cgit v0.8.1-24-ge5fb
