From 56ff9e213249bbdc0ecd3b5e7e4cffee60f46ef4 Mon Sep 17 00:00:00 2001
From: James Cloos<cloos@jhcloos.com>
Date: Wed, 20  Aug  2008  14:32:52  +0000
Subject: xaw8 is gone, use xaw7

---
diff --git a/configure.ac b/configure.ac
index c07e54c..b67765c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -33,7 +33,7 @@ AM_PROG_CC_C_O
 AC_PROG_INSTALL
 
 # Checks for pkg-config packages
-XAW_CHECK_XPRINT_SUPPORT(XMORE)
+PKG_CHECK_MODULES(XMORE, xaw7)
 if test "x$xaw_use_xprint" = "xyes" ; then
    PKG_CHECK_MODULES(XPRINT_UTIL, xprintutil xp)
 
--
cgit v0.8.1-24-ge5fb
