From 747f76737a8e7a7f50bd06fd111e5ada8305884d Mon Sep 17 00:00:00 2001
From: Julien Cristau<jcristau@debian.org>
Date: Sun, 16  Nov  2008  20:49:56  +0000
Subject: More xprint pruning

---
diff --git a/app-defaults/XMore.ad b/app-defaults/XMore.ad
index 98704d9..78b1e0a 100644
--- a/app-defaults/XMore.ad
+++ b/app-defaults/XMore.ad
@@ -13,7 +13,6 @@
         <Key>Return:    scroll-one-line-up()\n\
         <Key>J:         scroll-one-line-up()\n\
         <Key>E:         scroll-one-line-up()\n\
-        <Key>p:         print()\n\
         <Key>q:         quit()\n
 *text.baseTranslations: #override\n\
         <Key>space:     next-page()\n\
@@ -24,107 +23,9 @@
         <Key>Return:    scroll-one-line-up()\n\
         <Key>J:         scroll-one-line-up()\n\
         <Key>E:         scroll-one-line-up()\n\
-        <Key>p:         print()\n\
         <Key>q:         quit()\n
 
 ! tip messages
 *quit.tip:  Quit application
-*print.tip: Print file
-
-! Use "white" as background for printing
-*printshell*background:   white
-
-! Print dialog
-*printdialogshell*geometry:                       600x120
-*printdialogshell*title:                          Print
-*printdialogshell*main*ok.fromVert:               innerform
-*printdialogshell*main*ok.label:                  Print
-*printdialogshell*main*ok.tip:                    Print
-*printdialogshell*main*setup.fromHoriz:           ok
-*printdialogshell*main*setup.fromVert:            innerform
-*printdialogshell*main*setup.label:               Setup...
-*printdialogshell*main*setup.tip:                 Configure print job options (page size, orientation, etc.)
-*printdialogshell*main*cancel.fromHoriz:          setup
-*printdialogshell*main*cancel.fromVert:           innerform
-*printdialogshell*main*cancel.label:              Cancel
-*printdialogshell*main*cancel.tip:                Cancel printing
-*printdialogshell*main*desclabel.label:           Printer Description:
-*printdialogshell*main*desclabel.tip:             Short description of printer
-*printdialogshell*main*desc.fromHoriz:            desclabel
-*printdialogshell*main*desc.tip:                  Short description of printer
-*printdialogshell*main*info.fromHoriz:            desc
-*printdialogshell*main*info.label:                Printer info...
-*printdialogshell*main*info.tip:                  Display additional information about this printer
-*printdialogshell*main*namelabel.fromVert:        desclabel
-*printdialogshell*main*namelabel.label:           Printer Name:
-*printdialogshell*main*namelabel.tip:             Name of selected printer
-*printdialogshell*main*name.fromHoriz:            namelabel
-*printdialogshell*main*name.fromVert:             desclabel
-*printdialogshell*main*name.tip:                  Name of selected printer
-*printdialogshell*main*selectprinter.fromHoriz:   name
-*printdialogshell*main*selectprinter.fromVert:    desclabel
-*printdialogshell*main*selectprinter.label:       Select Printer...
-*printdialogshell*main*selectprinter.label:       Select Printer...
-*printdialogshell*main*selectprinter.tip:         Select a different printer
-*printdialogshell*main*filenamelabel.fromVert:    namelabel
-*printdialogshell*main*filenamelabel.label:       File Name:
-*printdialogshell*main*filenamelabel.tip:         File where the output should be stored
-*printdialogshell*main*filename.fromHoriz:        filenamelabel
-*printdialogshell*main*filename.fromVert:         namelabel
-*printdialogshell*main*filename.tip:              File where the output should be stored
-*printdialogshell*main*selectfile.fromHoriz:      filename
-*printdialogshell*main*selectfile.fromVert:       namelabel
-*printdialogshell*main*selectfile.label:          Select File...
-*printdialogshell*main*selectfile.tip:            Select file where the output should be stored
-*printdialogshell*main*printtoprinter.fromVert:   filenamelabel
-*printdialogshell*main*printtoprinter.label:      Print to Printer
-*printdialogshell*main*printtoprinter.tip:        Send print job to printer
-*printdialogshell*main*printtofile.fromVert:      filenamelabel
-*printdialogshell*main*printtofile.fromHoriz:     printtoprinter
-*printdialogshell*main*printtofile.label:         Print to File
-*printdialogshell*main*printtofile.tip:           Save print job in a file
-
-! Print job options dialog
-*printdialogshell*setup*geometry:                     600x400
-*printdialogshell*setup*title:                        Print: Print job options
-*printdialogshell*setup*ok.fromVert:                  list
-*printdialogshell*setup*ok.label:                     OK
-*printdialogshell*setup*ok.tip:                       Commit changes
-*printdialogshell*setup*cancel.fromHoriz:             ok
-*printdialogshell*setup*cancel.fromVert:              list
-*printdialogshell*setup*cancel.label:                 Cancel
-*printdialogshell*setup*cancel.tip:                   Cancel and reset to defaults
-*printdialogshell*setup*paperlist.tip:                Select paper size
-*printdialogshell*setup*resolutionlist.fromHoriz:     paperlist
-*printdialogshell*setup*resolutionlist.tip:           Select page resolution
-*printdialogshell*setup*orientationlist.fromHoriz:    resolutionlist
-*printdialogshell*setup*orientationlist.tip:          Select page orientation
-*printdialogshell*setup*plexlist.fromHoriz:           orientationlist
-*printdialogshell*setup*plexlist.tip:                 Select page plex mode (simplex, duplex, etc.)
-*printdialogshell*setup*colorspacelist.fromHoriz:     plexlist
-*printdialogshell*setup*colorspacelist.tip:           Select color space (color, grayscale, CYMK, etc.)
-*printdialogshell*setup*jobcopieslabel.fromVert:      paperlist
-*printdialogshell*setup*jobcopieslabel.tip:           Set number of job copies
-*printdialogshell*setup*jobcopieslabel.label:         Job Copies:
-*printdialogshell*setup*jobcopies.fromHoriz:          jobcopieslabel
-*printdialogshell*setup*jobcopies.fromVert:           paperlist
-*printdialogshell*setup*jobcopies.tip:                Set number of job copies
-
-! Printer selection
-*printdialogshell*printerselection*geometry:          400x150
-*printdialogshell*printerselection*title:             Print: Select printer
-*printdialogshell*printerselection*ok.fromVert:       list
-*printdialogshell*printerselection*ok.label:          OK
-*printdialogshell*printerselection*ok.tip:            Switch printer
-*printdialogshell*printerselection*cancel.fromHoriz:  ok
-*printdialogshell*printerselection*cancel.fromVert:   list
-*printdialogshell*printerselection*cancel.label:      Cancel
-*printdialogshell*printerselection*cancel.tip:        Cancel printer selection
-*printdialogshell*printerselection*list.tip:          Select printer name from list
-
-! Select job file
-*printdialogshell*selectfile*geometry:          400x80
-*printdialogshell*selectfile*title:             Print: Select job file
-*printdialogshell*selectfile*dialog.label:      Select Filename:
 
 ! EOF.
diff --git a/xmore.c b/xmore.c
index 4ed6e23..ff8c299 100644
--- a/xmore.c
+++ b/xmore.c
@@ -41,13 +41,6 @@ in this Software without prior written authorization from The Open Group.
 
 #include "xmore.h"
 
-#ifdef USE_XPRINT
-
-#include "printdialog.h"
-#include "print.h"
-
-#endif
-
 #include <stdlib.h>
 #include <stdio.h>
 #include <limits.h>
@@ -60,21 +53,14 @@ in this Software without prior written authorization from The Open Group.
 #define Log(x)   { if (userOptions.verbose) printf x; }
 
 /* Global vars */
-static Widget        printdialog_shell = NULL;
-static Widget        printdialog       = NULL;
 static Widget        toplevel          = NULL;
 static Widget        text              = NULL;
-static char          printJobNameBuffer[PATH_MAX+256];
 static const char   *ProgramName;  /* program name   (from argv[0]) */
 static const char   *viewFileName; /* file to browse (from argv[1]) */
 
 /* prototypes */
 static void quitAction(Widget w,  XEvent *event, String *params, Cardinal *num_params);
 static void quitXtProc(Widget w, XtPointer client_data, XtPointer callData);
-#ifdef USE_XPRINT
-static void printAction(Widget w, XEvent *event, String *params, Cardinal *num_params);
-static void printXtProc(Widget w, XtPointer client_data, XtPointer callData);
-#endif
 
 static XrmOptionDescRec options[] = {
 {
@@ -84,9 +70,6 @@ static XrmOptionDescRec options[] = {
 
 static XtActionsRec actions[] = {
     { "quit",          quitAction      },
-#ifdef USE_XPRINT
-    { "print",         printAction     }
-#endif
 };
 
 /* See xmore.h */
@@ -119,7 +102,6 @@ String fallback_resources[] = {
         "\t<Key>Return:    scroll-one-line-up()\\n\\"
         "\t<Key>J:         scroll-one-line-up()\\n\\"
         "\t<Key>E:         scroll-one-line-up()\\n\\"
-        "\t<Key>p:         print()\\n\\"
         "\t<Key>q:         quit()\\n",
     "*text.baseTranslations: #override \\n\\"
         "\t<Key>space:     next-page()\\n\\"
@@ -130,7 +112,6 @@ String fallback_resources[] = {
         "\t<Key>Return:    scroll-one-line-up()\\n\\"
         "\t<Key>J:         scroll-one-line-up()\\n\\"
         "\t<Key>E:         scroll-one-line-up()\\n\\"
-        "\t<Key>p:         print()\\n\\"
         "\t<Key>q:         quit()\\n",
     NULL,
 };
@@ -147,114 +128,10 @@ quitXtProc(Widget w, XtPointer client_data, XtPointer callData)
     XtCallActionProc(w, "quit", NULL, NULL, 0);
 }
 
-static void
-printshellDestroyXtProc(Widget w, XtPointer client_data, XtPointer callData)
-{
-    XawPrintDialogClosePrinterConnection(printdialog, False);
-}
-
-#ifdef USE_XPRINT
-
-static void
-printOKXtProc(Widget w, XtPointer client_data, XtPointer callData)
-{
-    XawPrintDialogCallbackStruct *pdcs = (XawPrintDialogCallbackStruct *)callData;
-    Cardinal                      n;
-    Arg                           args[2];
-    Widget                        textsource;
-
-    Log(("printOKXtProc: OK.\n"));
-    
-    /* Get TextSource object */
-    n = 0;
-    XtSetArg(args[n], XtNtextSource, &textsource); n++;
-    XtGetValues(text, args, n);
-    
-    Assertion(textsource != NULL, (("printOKXtProc: textsource == NULL.\n")));
-   
-    /* ||printJobNameBuffer| must live as long the print job prints
-     * because it is used for the job title AND the page headers... */
-    sprintf(printJobNameBuffer, "XMore print job %s", viewFileName);
-
-    DoPrintTextSource(ProgramName,
-                      textsource, toplevel,
-                      pdcs->pdpy, pdcs->pcontext, pdcs->colorspace,
-                      printshellDestroyXtProc,
-                      printJobNameBuffer,
-                      pdcs->printToFile?pdcs->printToFileName:NULL);
-
-    XtPopdown(printdialog_shell);
-}
-
-static void
-printCancelXtProc(Widget w, XtPointer client_data, XtPointer callData)
-{
-    Log(("printCancelXtProc: cancel.\n"));
-    XtPopdown(printdialog_shell);
-    
-    Log(("destroying print dialog shell...\n"));
-    XtDestroyWidget(printdialog_shell);
-    printdialog_shell = NULL;
-    printdialog       = NULL;
-    Log(("... done\n"));
-}
-
-static void
-printXtProc(Widget w, XtPointer client_data, XtPointer callData)
-{
-    XtCallActionProc(toplevel, "print", NULL, NULL, 0);
-}
-
-static void
-printAction(Widget w,  XEvent *event, String *params, Cardinal *num_params)
-{
-  Dimension   width, height;
-  Position    x, y;
-  Widget      parent = toplevel;
-  Log(("print!\n"));
-  
-  if (!printdialog) {
-    int n;
-    Arg args[20];
-
-    n = 0;
-    XtSetArg(args[n], XtNallowShellResize, True); n++;
-    printdialog_shell = XtCreatePopupShell("printdialogshell",
-                                           transientShellWidgetClass,
-                                           toplevel, args, n);
-    n = 0;
-    printdialog = XtCreateManagedWidget("printdialog", printDialogWidgetClass,
-                                        printdialog_shell, args, n);
-    XtAddCallback(printdialog, XawNOkCallback,     printOKXtProc,     NULL);
-    XtAddCallback(printdialog, XawNCancelCallback, printCancelXtProc, NULL);
-
-    XtRealizeWidget(printdialog_shell);
-  }
-
-  /* Center dialog */
-  XtVaGetValues(printdialog_shell,
-      XtNwidth,  &width,
-      XtNheight, &height,
-      NULL);
-
-  x = (Position)(XWidthOfScreen( XtScreen(parent)) - width)  / 2;
-  y = (Position)(XHeightOfScreen(XtScreen(parent)) - height) / 3;
-
-  XtVaSetValues(printdialog_shell,
-      XtNx, x,
-      XtNy, y,
-      NULL);
-        
-  XtPopup(printdialog_shell, XtGrabNonexclusive);
-}
-
-#endif
-
 int main( int argc, char *argv[] )
 {
   XtAppContext app;
   Widget       form;
-  Widget       printbutton;
   Widget       quitbutton;
   int          n;
   Arg          args[8];
@@ -292,25 +169,13 @@ int main( int argc, char *argv[] )
   XtSetArg(args[n], XtNfontSet,          userOptions.textfont);    n++;
   text = XtCreateManagedWidget("text", asciiTextWidgetClass, form, args, n);
 
-#ifdef USE_XPRINT
   n = 0;
-  XtSetArg(args[n], XtNfromHoriz,       NULL);            n++;
-  XtSetArg(args[n], XtNfromVert,        text);            n++;
-  XtSetArg(args[n], XtNlabel,           "Print...");      n++;
-  printbutton = XtCreateManagedWidget("print", commandWidgetClass, form, args, n);
-  XtAddCallback(printbutton, XtNcallback, printXtProc, 0);        
-#endif
-  
-  n = 0;
-  XtSetArg(args[n], XtNfromHoriz,       printbutton);            n++;
+  XtSetArg(args[n], XtNfromHoriz,       NULL);                   n++;
   XtSetArg(args[n], XtNfromVert,        text);                   n++;
   XtSetArg(args[n], XtNlabel,           "Quit");      n++;
   quitbutton = XtCreateManagedWidget("quit", commandWidgetClass, form, args, n);
   XtAddCallback(quitbutton, XtNcallback, quitXtProc, 0);
   
-  printdialog_shell = NULL;
-  printdialog       = NULL;
-
   XtRealizeWidget(toplevel);
   
   XtAppMainLoop(app);
diff --git a/xmore.man b/xmore.man
index 2936397..6475822 100644
--- a/xmore.man
+++ b/xmore.man
@@ -12,12 +12,5 @@ xmore \- plain text display program for the X Window System
 .ad b
 .SH DESCRIPTION
 xmore is a plain text browser.
-.SH ENVIRONMENT
-.TP 
-\fBXPSERVERLIST\fR 
-\fB${XPSERVERLIST}\fR must be set,
-identifying the available Xprint servers.
-See \fBXprint\fR(__miscmansuffix__)
-for more details.
 .SH "SEE ALSO"
 \fBX11\fR(__miscmansuffix__), \fBxman\fR(__appmansuffix__)
diff --git a/xmore.sgml b/xmore.sgml
index ec766c0..a92b317 100644
--- a/xmore.sgml
+++ b/xmore.sgml
@@ -38,24 +38,6 @@ manual volume numbers.
 
   </refsect1>
 
-  <refsect1>
-    <title>ENVIRONMENT</title>
-    <variablelist>
-      <varlistentry>
-        <term><envar>XPSERVERLIST</envar>
-        </term>
-        <listitem>
-	    <para>
-              <envar>${XPSERVERLIST}</envar> must be set,
-	      identifying the available Xprint servers.
-              See <citerefentry><refentrytitle>Xprint</refentrytitle><manvolnum>__miscmansuffix__</manvolnum></citerefentry>
-	      for more details.
-	  </para>
-	</listitem>
-      </varlistentry>
-    </variablelist>
-  </refsect1>
-
 <!-- not yet
   <refsect1>
     <title>KNOWN BUGS</title>
@@ -71,7 +53,6 @@ manual volume numbers.
         <!-- specific references -->
         <!-- none -->
         
-        <!-- Xprint general references -->
         <member><citerefentry><refentrytitle>X11</refentrytitle><manvolnum>__miscmansuffix__</manvolnum></citerefentry></member>
         <member><citerefentry><refentrytitle>xman</refentrytitle><manvolnum>__appmansuffix__</manvolnum></citerefentry></member>
       </simplelist>
--
cgit v0.8.1-24-ge5fb
