From d9b5448a77ff3ee06f6128faa69f43066cd16fe1 Mon Sep 17 00:00:00 2001
From: Julien Cristau<jcristau@debian.org>
Date: Wed, 20  Aug  2008  22:46:51  +0000
Subject: Fix build outside of source dir

---
diff --git a/Makefile.am b/Makefile.am
index d8aa027..3ba5868 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -42,6 +42,7 @@ APPDEFAULTFILES = \
 SUFFIXES = .ad
 
 .ad:
+	$(mkdir_p) app-defaults
 	cp $< $@
 
 appdefault_DATA = $(APPDEFAULTFILES)
--
cgit v0.8.1-24-ge5fb
