--- Makefile.orig	2004-08-31 19:51:06.000000000 +0200
+++ Makefile	2004-08-31 21:34:01.870761570 +0200
@@ -52,11 +52,7 @@
 
 
 
-include Makefile.jpeg
-include Makefile.vorbis
 include Makefile.ffmpeg
-include Makefile.lame
-include Makefile.dv
 include Makefile.encore50
 include Makefile.faad
 include Makefile.faac
@@ -71,7 +67,9 @@
 CFLAGS += -I. \
 	-I$(JPEG_DIR) \
 	-I$(MPEG3_DIR) \
-	-I$(LAME_DIR)/include
+	-I$(LAME_DIR)/include \
+	-I/usr/include/libmpeg3 \
+	-I/usr/include/lame
 
 
 #	$(OBJDIR)/div3.o \
@@ -109,7 +107,6 @@
 	$(OBJDIR)/hdlr.o \
 	$(OBJDIR)/ima4.o \
 	$(OBJDIR)/jpeg.o \
-	$(OBJDIR)/jpeg_old.o \
 	$(OBJDIR)/libdv.o \
 	$(OBJDIR)/libmjpeg.o \
 	$(OBJDIR)/matrix.o \
@@ -230,7 +227,7 @@
 	$(OBJDIR)/$(ENCORE_DIR)/mbtransquant/intel_mmx
 
 
-LIBS := $(OUTPUT) $(MPEG3_DIR)/$(OBJDIR)/libmpeg3.a -lpng -lz -lpthread -ldl -lm
+LIBS := $(OUTPUT) -lmpeg3 -lraw1394 -lmp3lame -logg -lvorbis -lvorbisenc -ldv -ljpeg -lpng -lz -lpthread -ldl -lm
 
 
 $(shell if ! test -d $(OBJDIR) \; then mkdir -p $(DIRS) \; fi )
@@ -242,12 +239,14 @@
 $(shell echo  $(OBJS) $(DV_OBJS) $(DVMMXOBJS) $(ENCOREMMX_OBJS) $(JPEG_NASM) $(JPEG_FAIL) $(FFMPEG_OBJS) $(FFMPEGIA32_OBJS) $(LAME_OBJS) $(FAAD_OBJS) $(FAAC_OBJS) > $(OBJDIR)/objs)
 #$(shell echo  $(OBJS) $(DV_OBJS) $(DVMMXOBJS) $(DECOREMMX_OBJS) $(ENCOREMMX_OBJS) $(JPEG_NASM) $(JPEG_FAIL) $(FFMPEG_OBJS) $(FFMPEGIA32_OBJS) $(LAME_OBJS) > $(OBJDIR)/objs)
 
-all: $(OBJDIR) $(OUTPUT)
+all: $(OBJDIR) $(OUTPUT) $(SHAREDOUTPUT) $(UTILS)
 
 #$(OUTPUT): $(DV_OBJS) $(DVMMXOBJ) $(OBJS) $(DVMMXOBJS) $(DECOREMMX_OBJS) $(ENCOREMMX_OBJS) $(JPEG_NASM) $(FFMPEG_OBJS) $(FFMPEGIA32_OBJS) $(LAME_OBJS) $(JPEG_FAIL) 
 $(OUTPUT): $(DV_OBJS) $(DVMMXOBJ) $(OBJS) $(DVMMXOBJS) $(ENCOREMMX_OBJS) $(JPEG_NASM) $(FFMPEG_OBJS) $(FFMPEGIA32_OBJS) $(LAME_OBJS) $(FAAD_OBJS) $(FAAC_OBJS) $(JPEG_FAIL) 
 	ar rcs $(OUTPUT) `cat $(OBJDIR)/objs`
 
+$(SHAREDOUTPUT): $(OUTPUT)
+	ld -shared -o $(SHAREDOUTPUT) `cat $(OBJDIR)/objs` $(LIBS)
 
 $(OBJDIR)/qtdump: $(OBJDIR)/dump.o $(OUTPUT)
 	$(CC) -o $(OBJDIR)/qtdump $(OBJDIR)/dump.o $(LIBS)
@@ -284,7 +283,13 @@
 	cd .. && \
 	tar zcvf quicktime.tar.gz quicktime
 
-install:
+install: $(OUTPUT) $(SHAREDOUTPUT) $(UTILS)
+	mkdir -p /usr/include/quicktime
+	mkdir -p /usr/include/quicktime/jpeg
+	cp quicktime.h qtprivate.h colormodels.h libmjpeg.h libdv.h /usr/include/quicktime
+	cp jpeg/jpeglib.h /usr/include/quicktime/jpeg
+	cp i686/libquicktime.* /usr/lib
+	ldconfig
 	cp $(UTILS) /usr/bin
 
 $(OBJDIR)/dump.o \
