diff -Naur ./src/display/canvas.h ../rtorrent-fixed/src/display/canvas.h
--- ./src/display/canvas.h	2011-04-07 14:36:10.000000000 +0200
+++ ../rtorrent-fixed/src/display/canvas.h	2011-09-20 16:06:02.520256895 +0200
@@ -48,7 +48,7 @@
 public:
   typedef std::vector<Attributes> attributes_list;
 
-  Canvas(int x = 0, int y = 0, int width = 0, int height = 0);
+  Canvas(int x = 0, int y = 0, int width = 1, int height = 1);
   ~Canvas() { delwin(m_window); }
 
   void                refresh()                                               { wnoutrefresh(m_window); }
