--- xcdroast-0.98alpha15/src/io.c	2003-10-26 15:24:48.000000000 +0100
+++ xcdroast-0.98alpha15-P/src/io.c	2006-09-25 20:07:33.000000000 +0200
@@ -9026,7 +9026,7 @@
 		strcat(tmp2,"-D ");
 	}
 	if (masterparam.opt[7]) {
-		strcat(tmp2,"-L ");
+		strcat(tmp2,"-allow-leading-dots ");
 	}
 	if (masterparam.opt[8]) {
 		strcat(tmp2,"-d ");
@@ -9160,7 +9160,7 @@
 	}	
 	if (masterparam.publisher) {
 		strcpy(tmptmp, masterparam.publisher);
-		g_snprintf(tmp5,MAXLINE,"-P \"%s\" ",
+		g_snprintf(tmp5,MAXLINE,"-publisher \"%s\" ",
 			convert_escape(tmptmp));
 		strcat(tmp4,tmp5);
 	}	

