From 4a5ca518ade0b84d74f6b6144a5aba7d790a2dae Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Fri, 4 Jun 2021 15:14:03 +0200
Subject: [PATCH] find recent re2

---
 configure.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.cmake b/configure.cmake
index 37215abb..92a2652c 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -12,6 +12,7 @@ else()
     find_package(BISON)
     find_package(FLEX)
     find_package(PkgConfig)
+    find_package(re2)
     find_package(Snappy)
     find_package(Nodejs 10.19)
 endif()
@@ -56,7 +57,7 @@ endif()
 qt_config_compile_test(re2
     LABEL "re2"
     LIBRARIES
-        PkgConfig::RE2
+        re2::re2
     CODE
 "
 #include \"re2/filtered_re2.h\"
-- 
2.33.1

