diff -ur synfigstudio-0.61.03/src/gtkmm/canvasview.cpp synfigstudio-0.61.03-gtkmm-2.8/src/gtkmm/canvasview.cpp
--- synfigstudio-0.61.03/src/gtkmm/canvasview.cpp	2005-11-06 05:51:58.000000000 +0100
+++ synfigstudio-0.61.03-gtkmm-2.8/src/gtkmm/canvasview.cpp	2005-12-13 12:59:33.000000000 +0100
@@ -1208,7 +1208,7 @@
 		sigc::hide_return(sigc::mem_fun(*this, &studio::CanvasView::image_import))
 	);
 	action_group->add( Gtk::Action::create("render", _("Render")),
-		sigc::mem_fun(render_settings,&studio::RenderSettings::present)
+		sigc::mem_fun0(render_settings,&studio::RenderSettings::present)
 	);
 	action_group->add( Gtk::Action::create("preview", _("Preview")),
 		sigc::mem_fun(*this,&CanvasView::on_preview_option)
@@ -1217,7 +1217,7 @@
 		sigc::mem_fun(*this,&CanvasView::on_audio_option)
 	);
 	action_group->add( Gtk::Action::create("options", _("Options")),
-		sigc::mem_fun(canvas_options,&studio::CanvasOptions::present)
+		sigc::mem_fun0(canvas_options,&studio::CanvasOptions::present)
 	);
 	action_group->add( Gtk::Action::create("close", Gtk::StockID("gtk-close")),
 		sigc::hide_return(sigc::mem_fun(*this,&studio::CanvasView::close))	
@@ -1248,7 +1248,7 @@
 	);
 
 	action_group->add( Gtk::Action::create("properties", Gtk::StockID("gtk-properties")),
-		sigc::mem_fun(canvas_properties,&studio::CanvasProperties::present)
+		sigc::mem_fun0(canvas_properties,&studio::CanvasProperties::present)
 	);
 
 	// Preview Quality Menu
@@ -1279,7 +1279,7 @@
 	);
 	
 	action_group->add( Gtk::Action::create("dialog-flipbook", _("Flipbook Dialog")),
-		sigc::mem_fun(*preview_dialog, &studio::Dialog_Preview::present)
+		sigc::mem_fun0(*preview_dialog, &studio::Dialog_Preview::present)
 	);
 
 	action_group->add( Gtk::Action::create("toggle-grid-show", _("Toggle Grid Show")),
diff -ur synfigstudio-0.61.03/src/gtkmm/dockdialog.cpp synfigstudio-0.61.03-gtkmm-2.8/src/gtkmm/dockdialog.cpp
--- synfigstudio-0.61.03/src/gtkmm/dockdialog.cpp	2005-11-06 05:51:58.000000000 +0100
+++ synfigstudio-0.61.03-gtkmm-2.8/src/gtkmm/dockdialog.cpp	2005-12-13 13:27:07.000000000 +0100
@@ -128,7 +128,7 @@
 */
 
 	add_accel_group(App::ui_manager()->get_accel_group());
-	App::signal_present_all().connect(sigc::mem_fun(*this,&DockDialog::present));
+	App::signal_present_all().connect(sigc::mem_fun0(*this,&DockDialog::present));
 
 }
 
diff -ur synfigstudio-0.61.03/src/gtkmm/toolbox.cpp synfigstudio-0.61.03-gtkmm-2.8/src/gtkmm/toolbox.cpp
--- synfigstudio-0.61.03/src/gtkmm/toolbox.cpp	2005-12-05 20:10:01.000000000 +0100
+++ synfigstudio-0.61.03-gtkmm-2.8/src/gtkmm/toolbox.cpp	2005-12-13 13:03:02.000000000 +0100
@@ -377,7 +377,7 @@
 
 	add_accel_group(App::ui_manager()->get_accel_group());
 	
-	App::signal_present_all().connect(sigc::mem_fun(*this,&Toolbox::present));
+	App::signal_present_all().connect(sigc::mem_fun0(*this,&Toolbox::present));
 }
 
 Toolbox::~Toolbox()
