From df1173155585281dc58d054c3d0905a6241a0531 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Sun, 7 Aug 2022 23:58:29 +0200
Subject: [PATCH] Fix function inlining with newer GCC

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 glib.h | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/glib.h b/glib.h
index b83cf5db3504..f3c36b0cc13c 100644
--- a/glib.h
+++ b/glib.h
@@ -202,20 +202,11 @@ extern "C" {
 #  endif
 #endif
 #ifndef G_INLINE_FUNC
-#  ifdef __GNUC__
-#    ifdef __OPTIMIZE__
-#      define G_INLINE_FUNC extern inline
-#    else
-#      undef G_CAN_INLINE
-#      define G_INLINE_FUNC extern
-#    endif
-#  else /* !__GNUC__ */
 #    ifdef G_CAN_INLINE
 #      define G_INLINE_FUNC static inline
 #    else
 #      define G_INLINE_FUNC extern
 #    endif
-#  endif /* !__GNUC__ */
 #endif /* !G_INLINE_FUNC */
 
 
-- 
2.37.1

