--- applets/applets.c	2007-11-10 02:40:53.000000000 +0100
+++ applets/applets.c-fixed	2008-01-18 01:25:01.000000000 +0100
@@ -10,17 +10,6 @@
 #include <assert.h>
 #include "busybox.h"
 
-/* Apparently uclibc defines __GLIBC__ (compat trick?). Oh well. */
-#if ENABLE_STATIC && defined(__GLIBC__) && !defined(__UCLIBC__)
-#warning Static linking against glibc produces buggy executables
-#warning (glibc does not cope well with ld --gc-sections).
-#warning See sources.redhat.com/bugzilla/show_bug.cgi?id=3400
-#warning Note that glibc is unsuitable for static linking anyway.
-#warning If you still want to do it, remove -Wl,--gc-sections
-#warning from top-level Makefile and remove this warning.
-#error Aborting compilation.
-#endif
-
 #if ENABLE_BUILD_LIBBUSYBOX
 int main(int argc, char **argv)
 {
