--- Makefile.in.orig	2005-06-27 09:11:34.000000000 +0400
+++ Makefile.in	2011-05-03 23:48:18.147684188 +0400
@@ -28,6 +28,7 @@
 BINDIR  = @bindir@
 SBINDIR = @sbindir@
 MANDIR  = @mandir@
+DATADIR = @datadir@
 
 INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@ $(AM_INSTALL_PROGRAM_FLAGS)
@@ -70,9 +71,9 @@
 install: all
 	$(INSTALL) -D -m0755 knockd $(DESTDIR)$(SBINDIR)/knockd
 	$(INSTALL) -D -m0755 knock $(DESTDIR)$(BINDIR)/knock
-	$(INSTALL) -D -m0644 $(MANSRC)knockd.1 $(DESTDIR)$(MANDIR)/man1/knockd.1
+	$(INSTALL) -D -m0644 $(MANSRC)knockd.1 $(DESTDIR)$(MANDIR)/man8/knockd.8
 	$(INSTALL) -D -m0644 $(MANSRC)knock.1 $(DESTDIR)$(MANDIR)/man1/knock.1
-	$(INSTALL) -D -m0644 knockd.conf $(DESTDIR)/etc/knockd.conf
+	$(INSTALL) -D -m0644 knockd.conf $(DESTDIR)/$(DATADIR)/knockd/knockd.conf
 
 clean:
 	rm -f *~ $(OBJDIR)*.o $(MANSRC)*.1
--- doc/knockd.1.in.orig	2005-06-29 22:45:17.000000000 +0400
+++ doc/knockd.1.in	2011-05-03 23:49:01.259684649 +0400
@@ -1,4 +1,4 @@
-.TH knockd 1 "June 26, 2005" "knockd #VERSION#" ""
+.TH knockd 8 "June 26, 2005" "knockd #VERSION#" ""
 .SH NAME
 knockd \- port-knock server
 .SH SYNOPSIS
