diff -Naur mozilla.orig/gfx/src/gtk/Makefile.in mozilla/gfx/src/gtk/Makefile.in
--- mozilla.orig/gfx/src/gtk/Makefile.in	2007-04-17 12:44:26.861000605 -0700
+++ mozilla/gfx/src/gtk/Makefile.in	2007-04-17 12:45:12.523034576 -0700
@@ -135,6 +135,8 @@
 SHARED_LIBRARY_LIBS += $(DIST)/lib/$(LIB_PREFIX)gfxft2_s.$(LIB_SUFFIX)
 endif
 
+EXTRA_DSO_LDOPTS = $(DIST)/lib/$(LIB_PREFIX)gfxshared_s.$(LIB_SUFFIX)
+
 include $(topsrcdir)/config/rules.mk
 
 ifdef MOZ_ENABLE_XINERAMA
@@ -145,7 +147,6 @@
 EXTRA_DSO_LDOPTS += \
 		$(LIBS_DIR) \
 		-lgkgfx \
-		-lgfxshared_s \
 		$(GFX_XINERAMA_LIBS) \
 		-lgtksuperwin \
 		$(MOZ_COMPONENT_LIBS) \
@@ -165,7 +166,6 @@
 ifdef MOZ_ENABLE_GTK2
 EXTRA_DSO_LDOPTS += $(MOZ_COMPONENT_LIBS) \
 		-lgkgfx \
-		-lgfxshared_s \
 		$(GFX_XINERAMA_LIBS) \
 		$(MOZ_JS_LIBS) \
 		$(MOZ_UNICHARUTIL_LIBS) \
