diff -Naur xfce4-session-4.2.3.orig/Makefile.in xfce4-session-4.2.3/Makefile.in
--- xfce4-session-4.2.3.orig/Makefile.in	2006-08-11 08:53:00.103774152 -0700
+++ xfce4-session-4.2.3/Makefile.in	2006-08-11 09:15:53.403000936 -0700
@@ -96,7 +96,7 @@
 CATOBJEXT = @CATOBJEXT@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS@ -I/usr/include/libgnome-2.0 -I/usr/include/bonobo-activation-2 -I/usr/include/libbonobo-2.0
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CXX = @CXX@
diff -Naur xfce4-session-4.2.3.orig/xfce4-session/Makefile.in xfce4-session-4.2.3/xfce4-session/Makefile.in
--- xfce4-session-4.2.3.orig/xfce4-session/Makefile.in	2006-08-11 08:53:00.118771872 -0700
+++ xfce4-session-4.2.3/xfce4-session/Makefile.in	2006-08-11 09:14:15.325910920 -0700
@@ -98,7 +98,7 @@
 CATOBJEXT = @CATOBJEXT@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS@ -I/usr/include/libgnome-2.0 -I/usr/include/bonobo-activation-2 -I/usr/include/libbonobo-2.0
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CXX = @CXX@
