diff -Naur screenlets.orig/Makefile screenlets/Makefile
--- screenlets.orig/Makefile	2008-02-07 18:46:21.000000000 -0800
+++ screenlets/Makefile	2008-02-26 19:09:09.674504084 -0800
@@ -11,10 +11,10 @@
 
 # install the Screenlets
 install:
-	python setup.py install
-	cp desktop-menu/screenlets.svg /usr/share/icons
-	cp desktop-menu/screenlets-manager.desktop /usr/share/applications
-	cp desktop-menu/screenlets-daemon.desktop $(HOME)/.config/autostart
+	python setup.py install --root="$(DEST_DIR)"
+	cp desktop-menu/screenlets.svg $(DEST_DIR)/usr/share/icons
+	cp desktop-menu/screenlets-manager.desktop $(DEST_DIR)/usr/share/applications
+	cp desktop-menu/screenlets-daemon.desktop $(DEST_DIR)/usr/share/applications/
 
 # uninstall the Screenlets (NOT FINISHED)
 uninstall:
diff -Naur screenlets.orig/desktop-menu/Makefile screenlets/desktop-menu/Makefile
--- screenlets.orig/desktop-menu/Makefile	2008-02-07 18:46:21.000000000 -0800
+++ screenlets/desktop-menu/Makefile	2008-02-26 19:41:16.885495964 -0800
@@ -7,10 +7,6 @@
 
 # install menu and desktop files
 install:
-	cp $(HOME)/.config/menus/applications.menu $(HOME)/.config/menus/applications.menu.bak
-	@echo "Makefile: Backed up old applications.menu in $(HOME)/.config/menus/"
-	cp applications.menu $(HOME)/.config/menus
 	cp applications/* $(XDG_DATA_DIRS)/applications
 	cp desktop-directories/* $(XDG_DATA_DIRS)/desktop-directories
 	@echo "Makefile: The Screenlets should now have their own submenu in your Applications-menu ..."
-
