diff --git a/konq-plugins/CMakeLists.txt b/konq-plugins/CMakeLists.txt
index 181c065..deeca20 100644
--- a/konq-plugins/CMakeLists.txt
+++ b/konq-plugins/CMakeLists.txt
@@ -24,8 +24,9 @@ add_subdirectory( minitools )
 add_subdirectory( autorefresh )
 add_subdirectory( fsview )
 add_subdirectory( searchbar )
+if(${QT_QT3SUPPORT_INCLUDE_DIR})
 add_subdirectory( rellinks )
-
+endif()
 add_subdirectory( adblock )
 
 add_subdirectory( akregator )

