diff --git a/util/cairo-trace/meson.build b/util/cairo-trace/meson.build
index 8d4ac131b..cf05352c5 100644
--- a/util/cairo-trace/meson.build
+++ b/util/cairo-trace/meson.build
@@ -11,7 +11,8 @@ shared_lib_ext = libcairo.full_path().split('.')[-1]
 libcairotrace = library('cairo-trace', cairo_trace_sources,
   include_directories: [incbase, incsrc],
   dependencies: deps,
-  c_args: ['-DSHARED_LIB_EXT="@0@"'.format(shared_lib_ext),] + pthread_c_args,
+  c_args: ['-DSHARED_LIB_EXT="@0@"'.format(shared_lib_ext),
+    '-DPACKAGE=cairo -DPACKAGE_VERSION="@0@"'.format(meson.project_version()),] + pthread_c_args,
   link_args: extra_link_args,
   gnu_symbol_visibility: 'hidden',
   install: true,
