--- configure~	2006-08-31 01:19:49.000000000 +0300
+++ configure	2006-08-31 01:20:03.000000000 +0300
@@ -22373,14 +22373,12 @@
 muse_enable_fluidlib="yes"
 synth_fluidsynth="fluidsynth"
 synth_fluid="fluid"
-synth_fluidsynth_build="yes"
 # Check whether --enable-fluidsynth or --disable-fluidsynth was given.
 if test "${enable_fluidsynth+set}" = set; then
   enableval="$enable_fluidsynth"
 
   case "$enableval" in
     "yes")
-      synth_fluidsynth_build="yes"
       synth_fluidsynth="fluidsynth"
       synth_fluid="fluid"
       ;;
@@ -22388,7 +22386,6 @@
       muse_enable_fluidlib="no"
       synth_fluidsynth=""
       synth_fluid=""
-      synth_fluidsynth_build="no"
 
       ;;
     *)
@@ -22410,7 +22407,6 @@
 fi
 
 
-if test "$muse_enable_fluidlib" = "yes"; then
 
 
 if test "x$ac_cv_env_PKG_CONFIG_set" != "xset"; then
@@ -22575,8 +22571,11 @@
 _ACEOF
 
 fi
-fi
 
+if test "$muse_enable_fluidlib" = "no"; then
+  synth_fluidsynth=""
+  synth_fluid=""
+fi
 
 
 
@@ -24985,7 +24984,7 @@
 
   Software synths
   ---------------
-  FluidSynth:        $synth_fluidsynth_build
+  FluidSynth:          $muse_enable_fluidlib
 
 " >&5
 echo "$as_me:
@@ -25014,7 +25013,7 @@
 
   Software synths
   ---------------
-  FluidSynth:        $synth_fluidsynth_build
+  FluidSynth:          $muse_enable_fluidlib
 
 " >&6;}
 
