--- a/synti/simpledrums/ssplugin.h	2004-12-13 22:41:01.000000000 +0200
+++ b/synti/simpledrums/ssplugin.h	2006-08-31 00:33:31.000000000 +0300
@@ -15,6 +15,7 @@
 
 //#include <ladspa.h>
 #include "muse/ladspa.h"
+#include "muse/fastlog.h"
 #include <math.h>
 
 //---------------------------------------------------------
@@ -133,27 +134,6 @@
 
    };
 
-
-static inline float fast_log2 (float val)
-      {
-      /* don't use reinterpret_cast<> because that prevents this
-         from being used by pure C code (for example, GnomeCanvasItems)
-      */
-      int* const exp_ptr = (int *)(&val);
-      int x              = *exp_ptr;
-      const int log_2    = ((x >> 23) & 255) - 128;
-      x &= ~(255 << 23);
-      x += 127 << 23;
-      *exp_ptr = x;
-      val = ((-1.0f/3) * val + 2) * val - 2.0f/3;   // (1)
-      return (val + log_2);
-      }
-
-static inline float fast_log10 (const float val)
-      {
-      return fast_log2(val) / 3.312500f;
-      }
-
 //---------------------------------------------------------
 //   PluginList
 //---------------------------------------------------------
--- a/synti/vam/vamgui.cpp	2006-01-01 23:11:05.000000000 +0200
+++ b/synti/vam/vamgui.cpp	2006-08-31 00:33:39.000000000 +0300
@@ -558,7 +558,7 @@
                                             "Select a preset");
 	if (fn.isEmpty())
 		return;
-	bool popenFlag;
+	//bool popenFlag;
 	FILE* f = fopen(fn.ascii(),"r");//fileOpen(this, fn, QString(".pre"), "r", popenFlag, true);
 	if (f == 0)
 		return;
@@ -605,9 +605,9 @@
 		}
 	}
 ende:
-	if (popenFlag)
-		pclose(f);
-	else
+	//if (popenFlag)
+	//	pclose(f);
+	//else
 		fclose(f);
 
 	if (presetFileName) delete presetFileName;
