--- src/libImaging/ImagingUtils.h.orig	2018-01-01 16:04:32.000000000 -0500
+++ src/libImaging/ImagingUtils.h	2018-02-02 20:47:03.489727256 -0500
@@ -34,7 +34,8 @@
    GCC generates code with partial dependency which is 3 times slower.
    See: http://stackoverflow.com/a/26588074/253146 */
 #if defined(__x86_64__) && defined(__SSE__) &&  ! defined(__NO_INLINE__) && \
-    ! defined(__clang__) && defined(GCC_VERSION) && (GCC_VERSION < 40900)
+    ! defined(__clang__) && defined(GCC_VERSION) && (GCC_VERSION < 40900) && \
+    (GCC_VERSION >= 40700)
 static float __attribute__((always_inline)) inline _i2f(int v) {
     float x;
     __asm__("xorps %0, %0; cvtsi2ss %1, %0" : "=X"(x) : "r"(v) );
