Index: programs/Xserver/render/mitri.c
===================================================================
RCS file: /cvs/xorg/xc/programs/Xserver/render/mitri.c,v
retrieving revision 1.3
diff -u -r1.3 mitri.c
--- programs/Xserver/render/mitri.c	6 Aug 2004 23:42:10 -0000	1.3
+++ programs/Xserver/render/mitri.c	30 Apr 2006 20:20:58 -0000
@@ -141,7 +141,7 @@
     if (npoint < 3)
 	return;
     ntri = npoint - 2;
-    tris = ALLOCATE_LOCAL (ntri & sizeof (xTriangle));
+    tris = ALLOCATE_LOCAL (ntri * sizeof (xTriangle));
     if (!tris)
 	return;
     for (tri = tris; npoint >= 3; npoint--, points++, tri++)
@@ -173,7 +173,7 @@
     if (npoint < 3)
 	return;
     ntri = npoint - 2;
-    tris = ALLOCATE_LOCAL (ntri & sizeof (xTriangle));
+    tris = ALLOCATE_LOCAL (ntri * sizeof (xTriangle));
     if (!tris)
 	return;
     first = points++;
