--- include/Motif-2.1/Xm/XmAll.h.orig	2009-02-06 22:42:24.000000000 +0000
+++ include/Motif-2.1/Xm/XmAll.h	2009-02-06 22:42:34.000000000 +0000
@@ -76,12 +76,10 @@
 #include <Xm/Manager.h>
 #include <Xm/Notebook.h>
 #include <Xm/Primitive.h>
-#include <Xm/Print.h>
 #include <Xm/SpinB.h>
 #include <Xm/Transfer.h>
 
 /* 2.1 stuff */
-#include <Xm/Print.h>
 #include <Xm/SSpinB.h>
 
 #endif /* _XM_XMALL_H */
--- include/Motif-2.1/Xm/Makefile.in.orig	2009-02-06 22:41:52.000000000 +0000
+++ include/Motif-2.1/Xm/Makefile.in	2009-02-06 22:42:11.000000000 +0000
@@ -329,7 +329,6 @@
 
 # Only in 2.1
 xm_21private_h = \
-        Print.h PrintSP.h \
 	SSpinB.h SSpinBP.h \
 	TransltnsP.h \
 	XmAll.h XmStrDefs.h 
--- include/Motif-2.1/Xm/Makefile.am.orig	2009-02-06 22:41:46.000000000 +0000
+++ include/Motif-2.1/Xm/Makefile.am	2009-02-06 22:42:08.000000000 +0000
@@ -76,7 +76,6 @@
 
 # Only in 2.1
 xm_21private_h = \
-        Print.h PrintSP.h \
 	SSpinB.h SSpinBP.h \
 	TransltnsP.h \
 	XmAll.h XmStrDefs.h 
