--- configure.ac.orig	2021-02-07 07:41:52.000000000 -0800
+++ configure.ac	2022-07-28 09:18:08.638682461 -0700
@@ -177,7 +177,7 @@
 AC_DEFUN([AX_CHECK_DOCBOOK], [
     XSLTPROC_FLAGS=--nonet
     DOCBOOK_ROOT=
-    for i in /usr/share/sgml/docbook/stylesheet/xsl/nwalsh /usr/share/xml/docbook/stylesheet/docbook-xs /usr/share/sgml/docbook/xsl-stylesheets/ /usr/local/share/xsl/docbook;
+    for i in /usr/share/sgml/docbook/stylesheet/xsl/nwalsh /usr/share/xml/docbook/stylesheet/docbook-xs /usr/share/sgml/docbook/xsl-stylesheets/ /usr/local/share/xsl/docbook /usr/share/xml/docbook/xsl-stylesheets-1.79;
     do
         if test -d "$i"; then
             DOCBOOK_ROOT=$i
