--- mcs/tools/gacutil/driver.cs.orig	2006-01-23 11:14:03.000000000 +0100
+++ mcs/tools/gacutil/driver.cs	2006-01-23 11:17:29.000000000 +0100
@@ -280,7 +280,8 @@
 				string ref_dir = Path.Combine (libdir, package);
 				string ref_path = Path.Combine (ref_dir, asmb_file);
 
-				if (File.Exists (ref_path))
+        Mono.Unix.Native.Stat ignored_stat;
+        if (Mono.Unix.Native.Syscall.lstat (ref_path, out ignored_stat) == 0)
 					File.Delete (ref_path);
 				try {
 					Directory.CreateDirectory (ref_dir);
--- mcs/tools/gacutil/Makefile~	2005-07-26 06:34:18.000000000 +0200
+++ mcs/tools/gacutil/Makefile	2006-01-23 11:17:49.000000000 +0100
@@ -2,7 +2,7 @@
 SUBDIRS = 
 include ../../build/rules.make
 
-LOCAL_MCS_FLAGS = -r:Mono.Security.dll 
+LOCAL_MCS_FLAGS = -r:Mono.Security.dll -r:Mono.Posix.dll
 
 ifeq (net_1_1_bootstrap, $(PROFILE))
 PROGRAM = $(topdir)/class/lib/$(PROFILE)/gacutil.exe
--- mcs/class/Makefile~	2006-01-05 12:30:49.000000000 +0100
+++ mcs/class/Makefile	2006-01-23 11:57:39.000000000 +0100
@@ -8,6 +8,7 @@
 	System					\
 	System.XML				\
 	Mono.Security				\
+	Mono.Posix				\
 	System.Security				\
 	System.Configuration			\
 	System					\
