From b168d2338cd89da7218c6666d0c3249034d95866 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 19:26:56 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 th-abstraction.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/th-abstraction.cabal b/th-abstraction.cabal
index 066331f..5bfe3e3 100644
--- a/th-abstraction.cabal
+++ b/th-abstraction.cabal
@@ -28,8 +28,8 @@ library
   other-modules:       Language.Haskell.TH.Datatype.Internal
   build-depends:       base             >=4.3   && <5,
                        ghc-prim,
-                       template-haskell >=2.5   && <2.14,
-                       containers       >=0.4   && <0.6
+                       template-haskell >=2.5,
+                       containers       >=0.4
   hs-source-dirs:      src
   default-language:    Haskell2010
 
-- 
2.19.1

