diff -Naur kdeedu-4.3.4.orig/marble/src/plugins/positionprovider/gpsd/GpsdConnection.cpp kdeedu-4.3.4/marble/src/plugins/positionprovider/gpsd/GpsdConnection.cpp
--- kdeedu-4.3.4.orig/marble/src/plugins/positionprovider/gpsd/GpsdConnection.cpp	2009-05-27 21:53:49.000000000 +0200
+++ kdeedu-4.3.4/marble/src/plugins/positionprovider/gpsd/GpsdConnection.cpp	2009-12-14 22:59:36.000000000 +0100
@@ -29,7 +29,7 @@
 }
 
 void GpsdConnection::update() {
-    gps_data_t* data = m_gpsd.query( "o" );
+    gps_data_t* data = m_gpsd.send( "o" );
     if ( data )
         emit gpsdInfo( *data );
 }
