From 0c58583b4d5cad37ee52fe5810c63497082e1a98 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Wed, 5 Aug 2015 08:26:00 +0200
Subject: [PATCH] depend on quickcheck < 2.9

---
 test-framework-quickcheck2.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test-framework-quickcheck2.cabal b/test-framework-quickcheck2.cabal
index 56242f6..020e4d7 100644
--- a/test-framework-quickcheck2.cabal
+++ b/test-framework-quickcheck2.cabal
@@ -25,7 +25,7 @@ Library
         Exposed-Modules:        Test.Framework.Providers.QuickCheck2
 
         Build-Depends:          test-framework >= 0.7.1,
-                                QuickCheck >= 2.4 && < 2.8,
+                                QuickCheck >= 2.4 && < 2.9,
                                 extensible-exceptions >= 0.1.1 && < 0.2.0
         if flag(base3)
                 Build-Depends:          base >= 3 && < 4, random >= 1
-- 
2.4.1

