diff -uNr kdelibs-4.3.75svn1048496/solid/solid/CMakeLists.txt kdelibs-4.3.75svn1048496.cp/solid/solid/CMakeLists.txt
--- kdelibs-4.3.75svn1048496/solid/solid/CMakeLists.txt	2009-09-23 21:24:04.000000000 +0200
+++ kdelibs-4.3.75svn1048496.cp/solid/solid/CMakeLists.txt	2009-11-14 15:54:25.000000000 +0100
@@ -107,6 +107,7 @@
 )
 
 if(NOT WIN32 AND NOT APPLE)
+ if(NOT DONT_USE_HAL)
    message(STATUS "-- Building Solid HAL backend." )
    set(solid_LIB_SRCS ${solid_LIB_SRCS}
 
@@ -135,6 +136,7 @@
    backends/hal/halsmartcardreader.cpp
 
    )
+ endif(NOT DONT_USE_HAL)
 endif(NOT WIN32 AND NOT APPLE)
 
 if(APPLE)
