--- kdelibs-4.0.68/kdecore/kernel/kglobal.cpp   2008/04/02 09:10:19     792807
+++ kdelibs-4.0.68/kdecore/kernel/kglobal.cpp   2008/04/04 08:28:47     793497
@@ -46,6 +46,7 @@
 #include <QtCore/QCoreApplication>
 #include <QtCore/QTextCodec>
 #include "kcmdlineargs.h"
+#include <unistd.h> // umask
 
 #ifndef NDEBUG
 #define MYASSERT(x) if (!x) \
@@ -61,6 +62,7 @@
 Q_CONSTRUCTOR_FUNCTION(qrand)
 
 typedef QSet<QString> KStringDict;
+mode_t s_umsk;
 
 class KGlobalPrivate
 {
@@ -72,8 +74,8 @@
         {
             // the umask is read here before any threads are created to avoid race conditions
             mode_t tmp = 0;
-            umsk = umask(tmp);
-            umask(umsk);
+            s_umsk = umask(tmp);
+            umask(s_umsk);
         }
 
         inline ~KGlobalPrivate()
@@ -91,7 +93,6 @@
         KStringDict *stringDict;
         KLocale *locale;
         KCharsets *charsets;
-        mode_t umsk;
 };
 
 K_GLOBAL_STATIC(KGlobalPrivate, globalData)
@@ -164,8 +165,8 @@
 
 mode_t KGlobal::umask()
 {
-    PRIVATE_DATA;
-    return d->umsk;
+    // Don't use PRIVATE_DATA here. This is called by ~KGlobalPrivate -> ~KConfig -> sync -> KSaveFile, so there's no KGlobalPrivate anymore.
+    return s_umsk;
 }
 
 KComponentData KGlobal::activeComponent()
--- kdelibs-4.0.68/kdecore/tests/kconfigtest.cpp        2008/04/04 08:26:12     793496
+++ kdelibs-4.0.68/kdecore/tests/kconfigtest.cpp        2008/04/04 08:28:47     793497
@@ -1021,3 +1021,12 @@
     QVERIFY(QFile::exists(file));
 }
 
+void KConfigTest::testSyncOnExit()
+{
+    // Often, the KGlobalPrivate global static's destructor ends up calling ~KConfig ->
+    // KConfig::sync ... and if that code triggers KGlobal code again then things could crash.
+    // So here's a test for modifying KGlobal::config() and not syncing, the process exit will sync.
+    KConfigGroup grp(KGlobal::config(), "syncOnExit");
+    grp.writeEntry("key", "value");
+}
+
--- kdelibs-4.0.68/kdecore/tests/kconfigtest.h  2008/04/04 08:26:12     793496
+++ kdelibs-4.0.68/kdecore/tests/kconfigtest.h  2008/04/04 08:28:47     793497
@@ -64,6 +64,9 @@
 
     // unrelated
     void testKAboutDataOrganizationDomain();
+
+    // should be last
+    void testSyncOnExit();
 private:
     void revertEntries();
     QList<QByteArray> readLines();

