$OpenBSD$
--- setup.py.orig	Sat May 11 16:08:37 2013
+++ setup.py	Thu May 16 18:44:04 2013
@@ -24,16 +24,16 @@ setuptools.setup(
     tests_require=['py.test', 'lxml', 'pygments'],
     data_files=[
         ('share/locale/ar/LC_MESSAGES', ['locale/ar/LC_MESSAGES/hatta.mo']),
-        ('share/locale/da/LC_MESSAGES', ['locale/cs/LC_MESSAGES/hatta.mo']),
+        ('share/locale/cs/LC_MESSAGES', ['locale/cs/LC_MESSAGES/hatta.mo']),
         ('share/locale/da/LC_MESSAGES', ['locale/da/LC_MESSAGES/hatta.mo']),
         ('share/locale/de/LC_MESSAGES', ['locale/de/LC_MESSAGES/hatta.mo']),
         ('share/locale/es/LC_MESSAGES', ['locale/es/LC_MESSAGES/hatta.mo']),
-        ('share/locale/es/LC_MESSAGES', ['locale/et/LC_MESSAGES/hatta.mo']),
+        ('share/locale/et/LC_MESSAGES', ['locale/et/LC_MESSAGES/hatta.mo']),
         ('share/locale/fr/LC_MESSAGES', ['locale/fr/LC_MESSAGES/hatta.mo']),
-        ('share/locale/ja/LC_MESSAGES', ['locale/hu/LC_MESSAGES/hatta.mo']),
+        ('share/locale/hu/LC_MESSAGES', ['locale/hu/LC_MESSAGES/hatta.mo']),
         ('share/locale/ja/LC_MESSAGES', ['locale/ja/LC_MESSAGES/hatta.mo']),
         ('share/locale/pl/LC_MESSAGES', ['locale/pl/LC_MESSAGES/hatta.mo']),
-        ('share/locale/pl/LC_MESSAGES', ['locale/ru/LC_MESSAGES/hatta.mo']),
+        ('share/locale/ru/LC_MESSAGES', ['locale/ru/LC_MESSAGES/hatta.mo']),
         ('share/locale/sv/LC_MESSAGES', ['locale/sv/LC_MESSAGES/hatta.mo']),
         ('share/doc/hatta/examples', [
             'examples/hatta.fcg',
