--- a/hatta/templates/base.html
+++ b/hatta/templates/base.html
@@ -38,7 +38,7 @@
         });
     </script>
     <script type="text/javascript"
-        src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"
+        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"
     ></script>
     {% endif %}
 {% endblock %}
diff --git a/hatta/wiki.py b/hatta/wiki.py
--- a/hatta/wiki.py
+++ b/hatta/wiki.py
@@ -104,7 +104,7 @@
         self.help_page = self.config.get('help_page', 'Help')
         self.math_url = self.config.get(
             'math_url',
-            'http://www.mathtran.org/cgi-bin/mathtran?tex=',
+            'mathjax',
         )
         self.pygments_style = self.config.get('pygments_style', 'tango')
         self.extension = self.config.get('extension')
