--- config.sub.old	2006-07-18 02:11:00.145083145 +0200
+++ config.sub	2006-07-18 02:15:42.020515400 +0200
@@ -153,7 +153,8 @@
 		| arme[lb] | pyramid | mn10200 | mn10300 | tron | a29k \
 		| 580 | i960 | h8300 | hppa | hppa1.0 | hppa1.1 | hppa2.0 \
 		| hppa2.0w \
-		| alpha | alphaev5 | alphaev56 | we32k | ns16k | clipper \
+		| alpha | alphaev5 | alphaev56 | alphaev6 | alphaev67 | alphaev7 \
+	        | we32k | ns16k | clipper \
 		| i370 | sh | powerpc | powerpcle | 1750a | dsp16xx | pdp11 \
 		| mips64 | mipsel | mips64el | mips64orion | mips64orionel \
 		| mipstx39 | mipstx39el | armv[34][lb] \
@@ -178,7 +179,8 @@
 	      | power-* | none-* | 580-* | cray2-* | h8300-* | i960-* \
 	      | xmp-* | ymp-* | hppa-* | hppa1.0-* | hppa1.1-* | hppa2.0-* \
 	      | hppa2.0w-* \
-	      | alpha-* | alphaev5-* | alphaev56-* | we32k-* | cydra-* \
+	      | alpha-* | alphaev5-* | alphaev56-* | alphaev6-* | alphaev67-* | alphaev7-* \
+	      | we32k-* | cydra-* \
 	      | ns16k-* | pn-* | np1-* | xps100-* | clipper-* | orion-* \
 	      | sparclite-* | pdp11-* | sh-* | powerpc-* | powerpcle-* \
 	      | sparc64-* | mips64-* | mipsel-* | armv[34][lb]-*\
@@ -501,6 +503,9 @@
 	pentiumii-* | pentium2-*)
 		basic_machine=i786-`echo $basic_machine | sed 's/^[^-]*-//'`
 		;;
+	x86_64-*)
+		basic_machine=x86_64-`echo $basic_machine | sed 's/^[^-]*-//'`
+		;;
 	pn)
 		basic_machine=pn-gould
 		;;
