diff -Naur xplab-0.3.1/src/Makefile xplab-0.3.1.dings/src/Makefile
--- xplab-0.3.1/src/Makefile	2006-01-20 20:52:49.000000000 +0100
+++ xplab-0.3.1.dings/src/Makefile	2007-02-12 21:49:36.000000000 +0100
@@ -8,7 +8,6 @@
 CFLAGS = -I/usr/X11R6/include      \
          -W                        \
          -Wall                     \
-         -Werror                   \
          -Wcast-qual               \
          -Wcast-align              \
          -Wstrict-prototypes       \
@@ -33,8 +32,6 @@
 
 EXENAME = xplab
 
-CC = @gcc -O2
-
 OBJS    = xplab_init_types.o xplab_gtk_types.o xplab_statistic_fe.o     \
           xplab_main_fe.o xplab_main_be.o xplab_preferences_be.o        \
           xplab_recall_fe.o xplab_recall_be.o xplab_statistic_be.o      \
