diff -Naur Farmerjoe_0.1.3.orig/Farmerjoe.pl Farmerjoe_0.1.3/Farmerjoe.pl
--- Farmerjoe_0.1.3.orig/Farmerjoe.pl	2008-03-09 14:52:53.810733090 -0700
+++ Farmerjoe_0.1.3/Farmerjoe.pl	2008-03-09 14:53:42.573457139 -0700
@@ -39,9 +39,7 @@
 
 #read config here
 my $util = FarmerTools->new();
-my $dir = '.';
-$dir  = dirname($0);
-my $conf = $util->load_conf("$dir/Farmerjoe.conf");
+my $conf = $util->load_conf("/etc/farmerjoe.conf");
 
 my $osconf = {master=>$conf->{master},port=>$conf->{port}};
 $osconf->{os} = $Config{osname};
