diff -Naur Farmerjoe_0.1.3.orig/bin/farmerjoe_submit.py Farmerjoe_0.1.3/bin/farmerjoe_submit.py
--- Farmerjoe_0.1.3.orig/bin/farmerjoe_submit.py	2008-03-09 15:00:31.310465423 -0700
+++ Farmerjoe_0.1.3/bin/farmerjoe_submit.py	2008-03-09 15:01:09.112456670 -0700
@@ -76,9 +76,9 @@
 'extensions'       :{ 'Targa':'.tga', 'Jpeg':'.jpg', 'PNG':'.png', 'BMP':'.bmp' },
 'filetypes'        :( 'Targa', 'Jpeg', 'PNG', 'BMP' ),
 'filetype_const'   :{ 'Targa':Scene.Render.TARGA, 'Jpeg':Scene.Render.JPEG, 'PNG':Scene.Render.PNG, 'BMP':Scene.Render.BMP },
-'linux_farmerjoe'  :'Farmerjoe.linux',
-'windows_farmerjoe'  :'Farmerjoe.exe',
-'osx_farmerjoe'  :'Farmerjoe.osx'
+'linux_farmerjoe'  :'farmerjoe',
+'windows_farmerjoe'  :'farmerjoe',
+'osx_farmerjoe'  :'farmerjoe'
 }
 #controls
 c_ = {
