diff -Nurp firehol-1.255.orig/firehol.sh firehol-1.255/firehol.sh
--- firehol-1.255.orig/firehol.sh	2007-07-24 05:03:26.000000000 +0500
+++ firehol-1.255/firehol.sh	2007-07-24 05:32:00.000000000 +0500
@@ -13,6 +13,11 @@
 # $Id: firehol.sh,v 1.255 2007/05/06 14:42:43 ktsaou Exp $
 #
 
+RUNLEVEL=3
+NEEDS="+network"
+
+. /etc/init.d/smgl_init
+
 # Make sure only root can run us.
 if [ ! "${UID}" = 0 ]
 then
@@ -86,7 +91,7 @@ require_cmd() {
 	# if one is found, return success
 	for x in $1
 	do
-		eval var=`echo ${x} | tr 'a-z' 'A-Z'`_CMD
+		eval var="echo ${x} | tr 'a-z' 'A-Z'"_CMD
 		eval val=\$\{${var}\}
 		if [ -z "${val}" ]
 		then
