diff -Nurp firehol-1.214.orig/firehol.sh firehol-1.214/firehol.sh
--- firehol-1.214.orig/firehol.sh	2004-11-27 01:57:37.000000000 +0100
+++ firehol-1.214/firehol.sh	2004-11-27 01:58:27.000000000 +0100
@@ -13,6 +13,11 @@
 # $Id: firehol.sh,v 1.214 2004/11/01 00:13:00 ktsaou Exp $
 #
 
+RUNLEVEL=3
+NEEDS="+network"
+
+. /etc/init.d/smgl_init
+
 # Remember who you are.
 FIREHOL_FILE="${0}"
 FIREHOL_DEFAULT_WORKING_DIRECTORY="${PWD}"
