From c0ce64be56c93e43343325ab54f30c89f3212721 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 4 Feb 2018 13:08:03 +0100
Subject: [PATCH] fix dependencies

---
 fclabels.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fclabels.cabal b/fclabels.cabal
index 80e1386..5e5e852 100644
--- a/fclabels.cabal
+++ b/fclabels.cabal
@@ -86,8 +86,8 @@ Library
 
   GHC-Options: -Wall
   Build-Depends:
-      base             >= 4.5 && < 4.10
-    , template-haskell >= 2.2 && < 2.12
+      base             >= 4.5 && < 4.11
+    , template-haskell >= 2.2 && < 2.13
     , mtl              >= 1.0 && < 2.3
     , transformers     >= 0.2 && < 0.6
 
-- 
2.16.1

