--- src/xine_interface.cxx.orig	2004-07-08 19:29:29.816800023 +0200
+++ src/xine_interface.cxx	2004-07-08 19:32:58.969142417 +0200
@@ -163,9 +163,9 @@
 			 xine_get_input_plugin_description (xine_data->xine,
 							    autoscan_plugins
 							    [i]));
-		(const char *const) auto_plugins[i].plugin_id =
+		auto_plugins[i].plugin_id = (char *)
 			autoscan_plugins[i];
-		(const char *) auto_plugins[i].plugin_desc =
+		auto_plugins[i].plugin_desc = (char *)
 			xine_get_input_plugin_description (xine_data->xine,
 							   autoscan_plugins
 							   [i]);
--- src/flxine.h~	2003-02-14 22:51:02.000000000 +0100
+++ src/flxine.h	2005-04-26 18:18:02.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef FL_XINE_H
 #define FL_XINE_H
 
+#include <unistd.h>
+
 #include "xine_interface.h"
 
 #define XINE_NAME_MAX 512
--- src/osd_defs.h~	2003-02-08 02:19:10.000000000 +0100
+++ src/osd_defs.h	2005-04-26 18:18:19.000000000 +0200
@@ -1,3 +1,5 @@
+#include <unistd.h>
+
 #include "xine_info.h"
 
 /************************************************************************************************
