Index: /libmusicbrainz/branches/xmlws/src/webservice.cpp
===================================================================
--- /libmusicbrainz/branches/xmlws/src/webservice.cpp (revision 9130)
+++ /libmusicbrainz/branches/xmlws/src/webservice.cpp (revision 9817)
@@ -25,4 +25,5 @@
 #include <map>
 #include <iostream>
+#include <stdlib.h>
 #include <string.h>
 #include <ne_session.h>
Index: /libmusicbrainz/branches/xmlws/src/utils_private.cpp
===================================================================
--- /libmusicbrainz/branches/xmlws/src/utils_private.cpp (revision 8466)
+++ /libmusicbrainz/branches/xmlws/src/utils_private.cpp (revision 9817)
@@ -21,4 +21,5 @@
  */
  
+#include <cstdlib>
 #include <cstring>
 #include <cstdarg>
