--- unix/configure.orig	2006-01-02 14:07:58.000000000 -0600
+++ unix/configure	2006-01-02 23:42:10.000000000 -0600
@@ -1946,7 +1946,7 @@
 	    # results, and the version is kept in special file).
 	
 	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-		system=MP-RAS-`awk '{print }' /etc/.relid'`
+		system=MP-RAS-`awk '{print }' /etc/.relid`
 	    fi
 	    if test "`uname -s`" = "AIX" ; then
 		system=AIX-`uname -v`.`uname -r`
