Date: Sat Jul 12 09:40:30 PDT 2008
Files: generic/tk.h
Bugid: 2010422

--- generic/tk.h.lastevent.old	2008-07-01 09:59:58.000000000 -0700
+++ generic/tk.h	2008-07-12 09:40:02.933155263 -0700
@@ -627,17 +627,15 @@
  *---------------------------------------------------------------------------
  */
 
-#define VirtualEvent	    (LASTEvent)
-#define ActivateNotify	    (LASTEvent + 1)
-#define DeactivateNotify    (LASTEvent + 2)
-#define MouseWheelEvent     (LASTEvent + 3)
-#define TK_LASTEVENT	    (LASTEvent + 4)
+#define VirtualEvent	    (MappingNotify + 1)
+#define ActivateNotify	    (MappingNotify + 2)
+#define DeactivateNotify    (MappingNotify + 3)
+#define MouseWheelEvent     (MappingNotify + 4)
+#define TK_LASTEVENT	    (MappingNotify + 5)
 
 #define MouseWheelMask	    (1L << 28)
-
 #define ActivateMask	    (1L << 29)
 #define VirtualEventMask    (1L << 30)
-#define TK_LASTEVENT	    (LASTEvent + 4)
 
 /*
  * A virtual event shares most of its fields with the XKeyEvent and
