diff -ur libunique-1.1.6/configure libunique-mod/configure
--- libunique-1.1.6/configure	2009-11-12 07:57:56.000000000 -0600
+++ libunique-mod/configure	2011-08-12 15:57:18.349054619 -0500
@@ -15398,7 +15398,7 @@
 if test "x$enable_maintainer_flags" = "xyes" &&  test "x$GCC" = "xyes"; then
 
 
-  list="-Werror -Wall -Wshadow -Wcast-align
+  list="-Wall -Wshadow -Wcast-align
                             -Wno-uninitialized -Wempty-body -Wformat-security
                             -Winit-self"
   flags_supported=""
diff -ur libunique-1.1.6/configure.ac libunique-mod/configure.ac
--- libunique-1.1.6/configure.ac	2009-11-12 07:57:39.000000000 -0600
+++ libunique-mod/configure.ac	2011-08-12 15:56:15.700054575 -0500
@@ -178,7 +178,7 @@
 AS_IF([test "x$enable_maintainer_flags" = "xyes" &&  test "x$GCC" = "xyes"],
       [
         AS_COMPILER_FLAGS([MAINTAINER_CFLAGS],
-                          ["-Werror -Wall -Wshadow -Wcast-align
+                          ["-Wall -Wshadow -Wcast-align
                             -Wno-uninitialized -Wempty-body -Wformat-security
                             -Winit-self"])
       ]
