diff -Naur qc-usb-0.6.4/qc-driver.c qc-usb-0.6.4/qc-driver.c
--- qc-usb-0.6.4/qc-driver.c	2006-05-05 22:51:50.000000000 -0400
+++ qc-usb-0.6.4/qc-driver.c	2006-10-09 01:04:50.116022392 -0400
@@ -867,7 +867,6 @@
 
 	out += sprintf(out, "\tGeneral driver status\n");
 	out += sprintf(out, "Driver version   : %s\n", VERSION);
-	out += sprintf(out, "Kernel version   : %s\n", UTS_RELEASE);
 	if (qc->dev!=NULL) {
 	out += sprintf(out, "Device Id        : %04X:%04X\n", (int)GET_VENDORID(qc), (int)GET_PRODUCTID(qc));
 	out += sprintf(out, "USB bus number   : %i\n", qc->dev->bus->busnum);
@@ -3266,8 +3265,8 @@
 
 	/* We found a QuickCam */
 	PRINTK(KERN_INFO,"QuickCam USB camera found (driver version %s)", VERSION);
-	PRINTK(KERN_INFO,"Kernel:%s bus:%i class:%02X subclass:%02X vendor:%04X product:%04X",
-		UTS_RELEASE, usbdev->bus->busnum, ifacedesc->bInterfaceClass, ifacedesc->bInterfaceSubClass,
+	PRINTK(KERN_INFO,"bus:%i class:%02X subclass:%02X vendor:%04X product:%04X",
+		usbdev->bus->busnum, ifacedesc->bInterfaceClass, ifacedesc->bInterfaceSubClass,
 		usbdev->descriptor.idVendor, usbdev->descriptor.idProduct);
 
 	/* The interface is claimed (bound) automatically to us when we return from this function (without error code) */
diff -Naur qc-usb-0.6.4/quickcam.h qc-usb-0.6.4/quickcam.h
--- qc-usb-0.6.4/quickcam.h	2006-05-05 23:00:26.000000000 -0400
+++ qc-usb-0.6.4/quickcam.h	2006-10-09 01:05:31.006577891 -0400
@@ -93,12 +93,13 @@
 #endif
 
 #include <linux/videodev.h>
+#include <media/v4l2-dev.h>
 #include <linux/usb.h>
 #include <asm/pgtable.h>		/* This is required for testing pte_offset_map */
 #include <linux/spinlock.h>
 
 /* {{{ [fold] SECTION: user configuration */
-#define VERSION		"QuickCam USB 0.6.4 $Date: 2006/05/06 03:00:26 $"
+#define VERSION		"QuickCam USB 0.6.4 $Date: 2006/05/02 13:05:20 $"
 #ifndef COMPRESS
 #define COMPRESS	1		/* 1=include compression support, 0=otherwise */
 #endif
