--- Makefile	2004-02-22 22:03:21.000000000 -0800
+++ Makefile	2004-02-22 22:04:58.000000000 -0800
@@ -21,10 +21,10 @@
 
 ####### Site-specific definitions #######
 # Destination directories and installation program for make install
-BINDIR = /usr/local/bin
-LIBDIR = /usr/local/lib
-MANDIR	= /usr/local/man/man1
-INSTALL = /usr/ucb/install
+BINDIR = /usr/bin
+LIBDIR = /usr/lib
+MANDIR	= /usr/man/man1
+INSTALL = /bin/install
 RM = rm -f
 # Location of help file
 HELP = \"$(LIBDIR)/viewfax.tif\"
@@ -43,8 +43,8 @@
 #LIBS =
 # sun solaris2 with gcc on supersparc:
 CC = gcc
-CFLAGS = -Wall $(OPT) -DHELPFILE=$(HELP) -msupersparc -I/usr/openwin/include
-LDFLAGS = $(OPT) -L/usr/openwin/lib -R/usr/openwin/lib
+CFLAGS = -Wall $(OPT) -DHELPFILE=$(HELP) ${MGETTY_CFLAGS}
+LDFLAGS = $(OPT)
 LIBS =
 
 ####### End of configurable definitions #######
