diff -uNr mgetty-1.1.35/Makefile mgetty-1.1.35.cp/Makefile
--- mgetty-1.1.35/Makefile	2006-02-22 14:58:33.000000000 +0100
+++ mgetty-1.1.35.cp/Makefile	2006-04-03 20:46:09.000000000 +0200
@@ -102,7 +102,7 @@
 #	    USTAT	  - ustat(), no statfs etc.
 #
 #CFLAGS=-Wall -O2 -pipe -DSECUREWARE -DUSE_POLL
-CFLAGS=-O2 -Wall -pipe
+CFLAGS=${MGETTY_CFLAGS} -DAUTO_PPP -Wall
 #CFLAGS=-O -DSVR4
 #CFLAGS=-O -DSVR4 -DSVR42
 #CFLAGS=-O -DUSE_POLL
@@ -172,7 +172,7 @@
 #
 # prefix, where most (all?) of the stuff lives, usually /usr/local or /usr
 #
-prefix=/usr/local
+prefix=/usr
 #
 # prefix for all the spool directories (usually /usr/spool or /var/spool)
 #
@@ -192,7 +192,7 @@
 #
 # where the configuration files (*.config, aliases, fax.allow/deny) go to
 #
-CONFDIR=$(prefix)/etc/mgetty+sendfax
+CONFDIR=/etc/mgetty+sendfax
 #CONFDIR=/etc/default/
 #
 #
@@ -216,24 +216,24 @@
 # (it's possible to run faxrunq(d) as root, but the FAX_OUT_USER 
 #  MUST NOT BE root or any other privileged account).
 #
-FAX_OUT_USER=fax
+FAX_OUT_USER=phone
 #
 #
 # Where section 1 manual pages should be placed
-MAN1DIR=$(prefix)/man/man1
+MAN1DIR=$(prefix)/share/man/man1
 #
 # Where section 4 manual pages (mgettydefs.4) should be placed
-MAN4DIR=$(prefix)/man/man4
+MAN4DIR=$(prefix)/share/man/man4
 #
 # Section 5 man pages (faxqueue.5)
-MAN5DIR=$(prefix)/man/man5
+MAN5DIR=$(prefix)/share/man/man5
 #
 # Section 8 man pages (sendfax.8)
-MAN8DIR=$(prefix)/man/man8
+MAN8DIR=$(prefix)/share/man/man8
 #
 # Where the GNU Info-Files are located
 #
-INFODIR=$(prefix)/info
+INFODIR=$(prefix)/share/info
 #
 #
 # A shell that understands bourne-shell syntax
diff -uNr mgetty-1.1.35/callback/Makefile mgetty-1.1.35.cp/callback/Makefile
--- mgetty-1.1.35/callback/Makefile	2000-08-14 21:33:08.000000000 +0200
+++ mgetty-1.1.35.cp/callback/Makefile	2006-04-03 15:04:44.000000000 +0200
@@ -5,7 +5,10 @@
 #
 
 CC=gcc
-CFLAGS=-O2 -I.. -g
+CFLAGS=-I.. -g ${MGETTY_CFLAGS}
+BINDIR=/usr/bin
+SBINDIR=/usr/sbin
+INSTALL=/bin/install
 
 zigzag:
 	cd .. ; $(MAKE) call-back
diff -uNr mgetty-1.1.35/frontends/X11/viewfax/Makefile mgetty-1.1.35.cp/frontends/X11/viewfax/Makefile
--- mgetty-1.1.35/frontends/X11/viewfax/Makefile	2004-11-14 21:38:24.000000000 +0100
+++ mgetty-1.1.35.cp/frontends/X11/viewfax/Makefile	2006-04-03 15:12:05.000000000 +0200
@@ -21,15 +21,15 @@
 
 ####### Site-specific definitions #######
 # Destination directories and installation program for make install
-BINDIR = /usr/local/bin
-LIBDIR = /usr/local/lib
-MANDIR	= /usr/local/man/man1
+BINDIR = /usr/bin
+LIBDIR = /usr/lib
+MANDIR	= /usr/share/man/man1
 INSTALL = /usr/bin/install
 RM = rm -f
 # Location of help file
 HELP = \"$(LIBDIR)/viewfax.tif\"
 # optimisation level, debug etc
-OPT = -g -O2 -Wno-uninitialized -ansi -pedantic
+OPT = -Wno-uninitialized -ansi -pedantic
 # C compiler and libraries
 # vanilla sysv
 #CC = cc
@@ -48,7 +48,7 @@
 #LIBS =
 # linux
 CC = gcc
-CFLAGS = $(OPT) -DHELPFILE=$(HELP)
+CFLAGS = $(OPT) -DHELPFILE=$(HELP) ${MGETTY_CFLAGS}
 LDFLAGS = $(OPT) -L/usr/X11R6/lib
 LIBS =
 
diff -uNr mgetty-1.1.35/frontends/dialog/Makefile mgetty-1.1.35.cp/frontends/dialog/Makefile
--- mgetty-1.1.35/frontends/dialog/Makefile	1998-03-26 10:05:10.000000000 +0100
+++ mgetty-1.1.35.cp/frontends/dialog/Makefile	2006-04-03 15:05:28.000000000 +0200
@@ -6,6 +6,8 @@
 
 #DIALOG_SCRIPTS=faxv listen xfaxq
 DIALOG_SCRIPTS=faxv xfaxq
+BINDIR=/usr/bin
+INSTALL=/bin/install
 
 all: $(DIALOG_SCRIPTS)
 
diff -uNr mgetty-1.1.35/tools/Makefile mgetty-1.1.35.cp/tools/Makefile
--- mgetty-1.1.35/tools/Makefile	2004-11-24 16:24:25.000000000 +0100
+++ mgetty-1.1.35.cp/tools/Makefile	2006-04-03 15:20:17.000000000 +0200
@@ -5,7 +5,7 @@
 #
 
 CC=gcc
-CFLAGS=-O2 -I..
+CFLAGS=${MGETTY_CFLAGS} -I..
 
 SCRIPTS=kvg cutbl
 PROGRAMS=ltest mid microcom
