--- policy.h	2002-01-10 15:13:05.000000000 -0700
+++ policy.h	2002-08-13 17:26:07.000000000 -0600
@@ -587,7 +587,7 @@
  * If you don't want this type of service, do not define it at all
  * Absolute path name has to be used here!
  */
-#define FAX_NOTIFY_PROGRAM "/usr/local/lib/mgetty+sendfax/new_fax"
+#define FAX_NOTIFY_PROGRAM "/etc/mgetty+sendfax/new_fax"
 
 /* default minimum space required on spooling partition for receiving a FAX
  * (in KILObytes)
