--- policy.h	2002-01-10 15:13:05.000000000 -0700
+++ policy.h	2002-08-13 17:13:52.000000000 -0600
@@ -176,7 +176,7 @@
  * current date and time, respectively.
  * override with "-p <prompt>" switch
  */
-#define LOGIN_PROMPT	"@!login: "
+#define LOGIN_PROMPT	"@!login: "
 
 /* On SVR4, maybe on other systems too, you can cause the 'login' program
  * to prompt with the same string as mgetty did, instead of the standard
@@ -206,7 +206,7 @@
  * (after that time a warning will be issued, after that, the call is
  * dropped). To disable that feature, do not define it.
  */
-#define MAX_LOGIN_TIME	240
+#define MAX_LOGIN_TIME	180
 
 /* nologin file
  *
