--- cmake/01-core.cmake.orig	2017-12-03 12:03:49.000000000 -0500
+++ cmake/01-core.cmake	2018-02-01 15:48:32.296902831 -0500
@@ -73,8 +73,8 @@
 
 if(ENABLE_CCACHE)
   querybin(ccache)
-  set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${BINPATH_ccache})
-  set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ${BINPATH_ccache})
+  set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
+  set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
 endif()
 
 # Install paths
