--- lib/xpp/CMakeLists.txt.orig	2020-12-12 04:16:10.000000000 -0800
+++ lib/xpp/CMakeLists.txt	2020-12-12 06:16:32.941122725 -0800
@@ -24,7 +24,7 @@
     pkg_get_variable(XCBPROTO_XCBINCLUDEDIR xcb-proto xcbincludedir)
 endif()
 
-find_package(PythonInterp 3.5 REQUIRED)
+find_package(PythonInterp 2.7 REQUIRED)
 find_package(XCB REQUIRED XCB ICCCM EWMH UTIL IMAGE)
 
 if(NOT PYTHON_EXECUTABLE)
--- lib/xpp/generators/cpp_client.py.orig	2020-12-12 04:16:10.000000000 -0800
+++ lib/xpp/generators/cpp_client.py	2020-12-12 06:19:43.304249600 -0800
@@ -1,6 +1,8 @@
 #!/usr/bin/env python
 # vim: set ts=4 sws=4 sw=4:
 
+from __future__ import print_function
+
 from xml.etree.ElementTree import *
 from os.path import basename
 from functools import reduce
