--- src/modules/xworkspaces.cpp.orig	2017-12-03 12:03:49.000000000 -0500
+++ src/modules/xworkspaces.cpp	2018-02-03 17:10:40.250293021 -0500
@@ -264,8 +264,10 @@
       output += module::get_output();
     }
 
-    m_builder->cmd(mousebtn::SCROLL_DOWN, string{EVENT_PREFIX} + string{EVENT_SCROLL_DOWN});
-    m_builder->cmd(mousebtn::SCROLL_UP, string{EVENT_PREFIX} + string{EVENT_SCROLL_UP});
+    if (m_scroll) {
+      m_builder->cmd(mousebtn::SCROLL_DOWN, string{EVENT_PREFIX} + string{EVENT_SCROLL_DOWN});
+      m_builder->cmd(mousebtn::SCROLL_UP, string{EVENT_PREFIX} + string{EVENT_SCROLL_UP});
+    }
 
     m_builder->append(output);
 
