diff -Nuar postfixmoreold/conf/postfix-files postfixold-2.2.9/conf/postfix-files
--- postfixmoreold/conf/postfix-files	2005-03-14 07:13:16.000000000 -0600
+++ postfixold-2.2.9/conf/postfix-files	2006-03-24 09:54:51.000000000 -0600
@@ -41,7 +41,6 @@
 # it is a good idea to "u" all the files that have special ownership or
 # permissions, so that running "make install" fixes any glitches.
 #
-$config_directory:d:root:-:755:u
 $daemon_directory:d:root:-:755:u
 $queue_directory:d:root:-:755:uc
 $sample_directory:d:root:-:755:o
@@ -100,31 +99,6 @@
 $sendmail_path:f:root:-:755
 $newaliases_path:l:$sendmail_path
 $mailq_path:l:$sendmail_path
-$config_directory/LICENSE:f:root:-:644
-$config_directory/TLS_LICENSE:f:root:-:644
-$config_directory/access:f:root:-:644:p
-$config_directory/aliases:f:root:-:644:p
-$config_directory/canonical:f:root:-:644:p
-$config_directory/cidr_table:f:root:-:644:o
-$config_directory/generics:f:root:-:644:o
-$config_directory/generic:f:root:-:644:p
-$config_directory/header_checks:f:root:-:644:p
-$config_directory/install.cf:f:root:-:644:o
-$config_directory/main.cf:f:root:-:644:p
-$config_directory/main.cf.default:f:root:-:644
-$config_directory/makedefs.out:f:root:-:644
-$config_directory/master.cf:f:root:-:644:p
-$config_directory/pcre_table:f:root:-:644:o
-$config_directory/postfix-files:f:root:-:644
-$config_directory/regexp_table:f:root:-:644:o
-$config_directory/relocated:f:root:-:644:p
-$config_directory/tcp_table:f:root:-:644:o
-$config_directory/transport:f:root:-:644:p
-$config_directory/virtual:f:root:-:644:p
-$config_directory/postfix-script:f:root:-:755
-$config_directory/postfix-script-sgid:f:root:-:755:o
-$config_directory/postfix-script-nosgid:f:root:-:755:o
-$config_directory/post-install:f:root:-:755
 $manpage_directory/man1/mailq.1:f:root:-:644
 $manpage_directory/man1/newaliases.1:f:root:-:644
 $manpage_directory/man1/postalias.1:f:root:-:644
diff -Nuar postfixmoreold/conf/post-install postfixold-2.2.9/conf/post-install
--- postfixmoreold/conf/post-install	2005-04-15 06:52:19.000000000 -0500
+++ postfixold-2.2.9/conf/post-install	2006-03-24 09:54:51.000000000 -0600
@@ -212,9 +212,8 @@
      create-missing) create=1;;
 	  set-perm*) create=1; set_perms=1;;
       upgrade-perm*) create=1; upgrade_perms=1;;
-      upgrade-conf*) upgrade_conf=1;;
-     upgrade-source) create=1; upgrade_conf=1; upgrade_perms=1;;
-    upgrade-package) create=1; upgrade_conf=1; set_perms=1;;
+     upgrade-source) create=1; upgrade_perms=1;;
+    upgrade-package) create=1; set_perms=1;;
      first-install*) first_install_reminder=1;;
 		  *) echo "$0: Error: $USAGE" 1>&2; exit 1;;
     esac
@@ -243,19 +242,6 @@
     exit 1
 }
 
-test -n "$config_directory" ||
-    config_directory=`$POSTCONF -d -h config_directory` || exit 1
-
-test -d "$config_directory" || {
-    echo $0: Error: $config_directory is not a directory. 1>&2
-    exit 1
-}
-
-test -f $config_directory/postfix-files || {
-    echo $0: Error: $config_directory/postfix-files is not a file. 1>&2
-    exit 1
-}
-
 # SunOS5 fmt(1) truncates lines > 1000 characters.
 
 fake_fmt() {
@@ -283,37 +269,6 @@
 # Extract parameter settings from the obsolete install.cf file, as
 # a transitional aid.
 
-grep setgid_group $config_directory/main.cf >/dev/null 2>&1 || {
-    test -f $config_directory/install.cf  && {
-        for name in sendmail_path newaliases_path mailq_path setgid manpages
-        do
-	eval junk=\$$name
-        case "$junk" in
-        "") eval unset $name;;
-        esac
-	   eval : \${$name="\`. $config_directory/install.cf; echo \$$name\`"} \
-		|| exit 1
-        done
-        : ${setgid_group=$setgid}
-        : ${manpage_directory=$manpages}
-    }
-}
-
-# Extract parameter settings from the installed main.cf file.
-
-test -f $config_directory/main.cf && {
-    for name in daemon_directory command_directory queue_directory mail_owner \
-        setgid_group sendmail_path newaliases_path mailq_path \
-	html_directory manpage_directory sample_directory readme_directory
-    do
-	eval junk=\$$name
-        case "$junk" in
-        "") eval unset $name;;
-        esac
-        eval : \${$name=\`$POSTCONF -c $config_directory -h $name\`} || exit 1
-    done
-}
-
 # Sanity checks
 
 case $manpage_directory in
@@ -380,36 +335,6 @@
 
 POSTCONF="$command_directory/postconf"
 
-# Save settings, allowing command line/environment override.
-
-override=
-for name in daemon_directory command_directory queue_directory mail_owner \
-    setgid_group sendmail_path newaliases_path mailq_path manpage_directory \
-    sample_directory readme_directory html_directory
-do
-    eval test \"\$$name\" = \"`$POSTCONF -c $config_directory -h $name`\" || {
-	override=1
-	break
-    }
-done
-
-test -n "$override" && {
-    $POSTCONF -c $config_directory -e \
-	"daemon_directory = $daemon_directory" \
-	"command_directory = $command_directory" \
-	"queue_directory = $queue_directory" \
-	"mail_owner = $mail_owner" \
-	"setgid_group = $setgid_group" \
-	"sendmail_path = $sendmail_path" \
-	"mailq_path = $mailq_path" \
-	"newaliases_path = $newaliases_path" \
-	"html_directory = $html_directory" \
-	"manpage_directory = $manpage_directory" \
-	"sample_directory = $sample_directory" \
-	"readme_directory = $readme_directory" \
-    || exit 1
-}
-
 # Use file/directory status information in $config_directory/postfix-files.
 
 test -n "$create" && {
diff -Nuar postfixmoreold/postfix-install postfixold-2.2.9/postfix-install
--- postfixmoreold/postfix-install	2005-02-02 15:39:36.000000000 -0600
+++ postfixold-2.2.9/postfix-install	2006-03-24 09:59:07.000000000 -0600
@@ -302,9 +302,6 @@
 tempdir_prompt="a directory for scratch files while installing
 Postfix.  You must have write permission in this directory."
 
-config_directory_prompt="the final destination directory for
-installed Postfix configuration files."
-
 daemon_directory_prompt="the final destination directory for
 installed Postfix daemon programs.  This directory should not be
 in the command search path of any users."
@@ -354,27 +351,28 @@
 
 : ${install_root=/}
 : ${tempdir=`pwd`}
-: ${config_directory=`bin/postconf -h -d config_directory`}
 
 # Find out the location of installed configuration files.
 
-test -z "$non_interactive" && for name in install_root tempdir config_directory
-do
-    while :
-    do
-	echo
-	eval echo Please specify \$${name}_prompt | ${FMT}
-	eval echo \$n "$name: [\$$name]\  \$c"
-	read ans
-	case $ans in
-	"") break;;
-	 *) case $ans in
-	    /*) eval $name=$ans; break;;
-	     *) echo; echo $0: Error: $name should be an absolute path name. 1>&2;;
-	    esac;;
-	esac
-    done
-done
+#test -z "$non_interactive" && for name in install_root tempdir
+#do
+#    while :
+#    do
+#	echo
+#	eval echo Please specify \$${name}_prompt | ${FMT}
+#	eval echo \$n "$name: [\$$name]\  \$c"
+#	read ans
+#	case $ans in
+#	"") break;;
+#	 *) case $ans in
+#	    /*) eval $name=$ans; break;;
+#	     *) echo; echo $0: Error: $name should be an absolute path name. 1>&2;;
+#	    esac;;
+#	esac
+#    done
+#done
+install_root=${INSTALL_ROOT}
+tempdir=${SOURCE_DIRECTORY}
 
 # In case some systems special-case pathnames beginning with //.
 
@@ -387,46 +385,6 @@
     exit 1
 }
 
-CONFIG_DIRECTORY=$install_root$config_directory
-
-# If a parameter is not set via the command line or environment,
-# try to use settings from installed configuration files.
-
-# Extract parameter settings from the obsolete install.cf file, as
-# a transitional aid.
-
-grep setgid_group $CONFIG_DIRECTORY/main.cf >/dev/null 2>&1 || {
-    test -f $CONFIG_DIRECTORY/install.cf && {
-	for name in sendmail_path newaliases_path mailq_path setgid manpages
-	do
-	    eval junk=\$$name
-	    case "$junk" in
-	    "") eval unset $name;;
-	    esac
-	    eval : \${$name="\`. $CONFIG_DIRECTORY/install.cf; echo \$$name\`"} \
-		|| exit 1
-	done
-	: ${setgid_group=$setgid}
-	: ${manpage_directory=$manpages}
-    }
-}
-
-# Extract parameter settings from the installed main.cf file.
-
-test -f $CONFIG_DIRECTORY/main.cf && {
-    for name in daemon_directory command_directory queue_directory mail_owner \
-	setgid_group sendmail_path newaliases_path mailq_path manpage_directory \
-	sample_directory html_directory readme_directory
-    do
-	eval junk=\$$name
-	case "$junk" in
-	"") eval unset $name;;
-	esac
-	eval : \${$name=\`bin/postconf -c $CONFIG_DIRECTORY -h $name\`} ||
-	    exit 1
-    done
-}
-
 # Use built-in defaults as the final source of parameter settings.
 
 for name in daemon_directory command_directory queue_directory mail_owner \
@@ -442,22 +400,33 @@
 
 # Override settings manually.
 
-test -z "$non_interactive" && for name in daemon_directory command_directory \
-    queue_directory sendmail_path newaliases_path mailq_path mail_owner \
-    setgid_group html_directory manpage_directory readme_directory
-do
-    while :
-    do
-	echo
-	eval echo Please specify \$${name}_prompt | ${FMT}
-	eval echo \$n "$name: [\$$name]\  \$c"
-	read ans
-	case $ans in
-	"") break;;
-	 *) eval $name=$ans; break;;
-	esac
-    done
-done
+#test -z "$non_interactive" && for name in daemon_directory command_directory \
+#    queue_directory sendmail_path newaliases_path mailq_path mail_owner \
+#    setgid_group html_directory manpage_directory readme_directory
+#do
+#    while :
+#    do
+#	echo
+#	eval echo Please specify \$${name}_prompt | ${FMT}
+#	eval echo \$n "$name: [\$$name]\  \$c"
+#	read ans
+#	case $ans in
+#	"") break;;
+#	 *) eval $name=$ans; break;;
+#	esac
+#    done
+#done
+daemon_directory=/usr/libexec/postfix
+command_directory=/usr/sbin
+queue_directory=/var/spool/postfix
+sendmail_path=/usr/sbin/sendmail
+newaliases_path=/usr/bin/newaliases
+mailq_path=/usr/bin/mailq
+mail_owner=postfix
+setgid_group=postdrop
+html_directory=/usr/share/doc/postfix/html
+manpage_directory=/usr/share/man
+readme_directory=/usr/share/doc/postfix
 
 # Sanity checks
 
@@ -504,7 +473,7 @@
     }
 done
 
-for path in command_directory config_directory daemon_directory \
+for path in command_directory daemon_directory \
     manpage_directory queue_directory html_directory readme_directory
 do
     eval test -f $install_root\$$path && {
@@ -571,12 +540,7 @@
 
 # Upgrade or first-time installation?
 
-if [ -f $CONFIG_DIRECTORY/main.cf ]
-then
-    post_install_options="upgrade-source"
-else
     post_install_options="first-install"
-fi
 
 # Install files, using information from the postfix-files file.
 
@@ -657,9 +621,6 @@
 	'$command_directory')
 	    compare_or_replace $mode "$owner" "$group" bin/$file \
 		$COMMAND_DIRECTORY/$file || exit 1;;
-	'$config_directory')
-	    compare_or_replace $mode "$owner" "$group" conf/$file \
-		$CONFIG_DIRECTORY/$file || exit 1;;
 	'$sendmail_path')
 	    check_parent $SENDMAIL_PATH || exit 1
 	    compare_or_replace $mode "$owner" "$group" bin/sendmail \
@@ -708,21 +669,6 @@
 # Postfix releases, and software should not suddenly be installed in
 # the wrong place when Postfix is being upgraded.
 
-bin/postconf -c $CONFIG_DIRECTORY -e \
-    "daemon_directory = $daemon_directory" \
-    "command_directory = $command_directory" \
-    "queue_directory = $queue_directory" \
-    "mail_owner = $mail_owner" \
-    "setgid_group = $setgid_group" \
-    "sendmail_path = $sendmail_path" \
-    "mailq_path = $mailq_path" \
-    "newaliases_path = $newaliases_path" \
-    "html_directory = $html_directory" \
-    "manpage_directory = $manpage_directory" \
-    "sample_directory = $sample_directory" \
-    "readme_directory = $readme_directory" \
-|| exit 1
-
 # If Postfix is being installed locally from source code, do the
 # post-install processing now.
 
