--- makedefs.orig	2012-11-30 03:53:34.000000000 +0400
+++ makedefs	2013-01-21 22:02:59.118469530 +0400
@@ -279,14 +279,14 @@
 		elif [ -f /usr/include/db/db.h ]
 		then
 		    CCARGS="$CCARGS -I/usr/include/db"
-		else
+		# else
 		    # No, we're not going to try db1 db2 db3 etc.
 		    # On a properly installed system, Postfix builds
 		    # by including <db.h> and by linking with -ldb
-		    echo "No <db.h> include file found." 1>&2
-		    echo "Install the appropriate db*-devel package first." 1>&2
-		    echo "See the RELEASE_NOTES file for more information." 1>&2
-		    exit 1
+		    # echo "No <db.h> include file found." 1>&2
+		    # echo "Install the appropriate db*-devel package first." 1>&2
+		    # echo "See the RELEASE_NOTES file for more information." 1>&2
+		    # exit 1
 		fi
 		# GDBM locks the DBM .pag file after open. This breaks postmap.
 		# if [ -f /usr/include/gdbm-ndbm.h ]
@@ -298,7 +298,7 @@
 		#     CCARGS="$CCARGS -DHAS_DBM -DPATH_NDBM_H='<gdbm/ndbm.h>'"
 		#     GDBM_LIBS=gdbm
 		# fi
-		SYSLIBS="-ldb"
+		# SYSLIBS="-ldb"
 		for name in nsl resolv $GDBM_LIBS
 		do
 		    for lib in /usr/lib64 /lib64 /usr/lib /lib
--- makedefs.orig	2012-11-30 03:53:34.000000000 +0400
+++ makedefs	2013-07-20 16:57:49.260069906 +0400
@@ -358,15 +358,15 @@
 		elif [ -f /usr/include/db/db.h ]
 		then
 		    CCARGS="$CCARGS -I/usr/include/db"
-		else
+		# else
 		    # On a properly installed system, Postfix builds
 		    # by including <db.h> and by linking with -ldb
-		    echo "No <db.h> include file found." 1>&2
-		    echo "Install the appropriate db*-devel package first." 1>&2
-		    echo "See the RELEASE_NOTES file for more information." 1>&2
-		    exit 1
+		    # echo "No <db.h> include file found." 1>&2
+		    # echo "Install the appropriate db*-devel package first." 1>&2
+		    # echo "See the RELEASE_NOTES file for more information." 1>&2
+		    # exit 1
 		fi
-		SYSLIBS="-ldb"
+		# SYSLIBS="-ldb"
 		for name in nsl resolv
 		do
 		    for lib in /usr/lib64 /lib64 /usr/lib /lib
