--- ../postfixmoreold/postfix-install	2005-02-02 15:39:36.000000000 -0600
+++ postfix-install	2006-03-24 10:34:39.000000000 -0600
@@ -358,23 +358,8 @@
 
 # Find out the location of installed configuration files.
 
-test -z "$non_interactive" && for name in install_root tempdir config_directory
-do
-    while :
-    do
-	echo
-	eval echo Please specify \$${name}_prompt | ${FMT}
-	eval echo \$n "$name: [\$$name]\  \$c"
-	read ans
-	case $ans in
-	"") break;;
-	 *) case $ans in
-	    /*) eval $name=$ans; break;;
-	     *) echo; echo $0: Error: $name should be an absolute path name. 1>&2;;
-	    esac;;
-	esac
-    done
-done
+install_root=${INSTALL_ROOT}
+tempdir=${SOURCE_DIRECTORY}
 
 # In case some systems special-case pathnames beginning with //.
 
@@ -442,22 +427,17 @@
 
 # Override settings manually.
 
-test -z "$non_interactive" && for name in daemon_directory command_directory \
-    queue_directory sendmail_path newaliases_path mailq_path mail_owner \
-    setgid_group html_directory manpage_directory readme_directory
-do
-    while :
-    do
-	echo
-	eval echo Please specify \$${name}_prompt | ${FMT}
-	eval echo \$n "$name: [\$$name]\  \$c"
-	read ans
-	case $ans in
-	"") break;;
-	 *) eval $name=$ans; break;;
-	esac
-    done
-done
+daemon_directory=/usr/libexec/postfix
+command_directory=/usr/sbin
+queue_directory=/var/spool/postfix
+sendmail_path=/usr/sbin/sendmail
+newaliases_path=/usr/bin/newaliases
+mailq_path=/usr/bin/mailq
+mail_owner=postfix
+setgid_group=postdrop
+html_directory=/usr/share/doc/postfix/html
+manpage_directory=/usr/share/man
+readme_directory=no
 
 # Sanity checks
 
