diff -Nur mjpegtools-1.9.0rc3-old/mpeg2enc/encodertypes.h mjpegtools-1.9.0rc3/mpeg2enc/encodertypes.h
--- mjpegtools-1.9.0rc3-old/mpeg2enc/encodertypes.h	2008-04-04 18:20:33.000000000 +0300
+++ mjpegtools-1.9.0rc3/mpeg2enc/encodertypes.h	2008-04-04 18:20:59.000000000 +0300
@@ -25,6 +25,8 @@
  */
 
 
+#include <cstdlib>
+
 class Parity
 {
 public:
diff -Nur mjpegtools-1.9.0rc3-old/mpeg2enc/macroblock.cc mjpegtools-1.9.0rc3/mpeg2enc/macroblock.cc
--- mjpegtools-1.9.0rc3-old/mpeg2enc/macroblock.cc	2008-04-04 18:20:33.000000000 +0300
+++ mjpegtools-1.9.0rc3/mpeg2enc/macroblock.cc	2008-04-04 18:20:49.000000000 +0300
@@ -21,6 +21,7 @@
 
 #include <stdio.h>
 #include <cassert>
+#include <climits>
 
 #include "macroblock.hh"
 #include "mpeg2syntaxcodes.h"
diff -Nur mjpegtools-1.9.0rc3-old/mpeg2enc/picturereader.cc mjpegtools-1.9.0rc3/mpeg2enc/picturereader.cc
--- mjpegtools-1.9.0rc3-old/mpeg2enc/picturereader.cc	2008-04-04 18:20:33.000000000 +0300
+++ mjpegtools-1.9.0rc3/mpeg2enc/picturereader.cc	2008-04-04 18:20:49.000000000 +0300
@@ -27,6 +27,7 @@
 //#include <stdlib.h>
 //#include <unistd.h>
 //#include <string.h>
+#include <climits>
 //#include <errno.h>
 //#include "simd.h"
 
diff -Nur mjpegtools-1.9.0rc3-old/mplex/inputstrm.cpp mjpegtools-1.9.0rc3/mplex/inputstrm.cpp
--- mjpegtools-1.9.0rc3-old/mplex/inputstrm.cpp	2008-04-04 18:20:33.000000000 +0300
+++ mjpegtools-1.9.0rc3/mplex/inputstrm.cpp	2008-04-04 18:20:49.000000000 +0300
@@ -23,6 +23,7 @@
 
 #include <config.h>
 #include <assert.h>
+#include <climits>
 
 #include "mjpeg_types.h"
 #include "inputstrm.hpp"
diff -Nur mjpegtools-1.9.0rc3-old/mplex/main.cpp mjpegtools-1.9.0rc3/mplex/main.cpp
--- mjpegtools-1.9.0rc3-old/mplex/main.cpp	2008-04-04 18:20:33.000000000 +0300
+++ mjpegtools-1.9.0rc3/mplex/main.cpp	2008-04-04 18:20:49.000000000 +0300
@@ -31,6 +31,7 @@
 #include <getopt.h>
 #endif
 #include <string>
+#include <string.h>
 #include <memory>
 #include <sys/stat.h>
 #ifndef _WIN32
diff -Nur mjpegtools-1.9.0rc3-old/mplex/multiplexor.cpp mjpegtools-1.9.0rc3/mplex/multiplexor.cpp
--- mjpegtools-1.9.0rc3-old/mplex/multiplexor.cpp	2008-04-04 18:20:33.000000000 +0300
+++ mjpegtools-1.9.0rc3/mplex/multiplexor.cpp	2008-04-04 18:20:49.000000000 +0300
@@ -22,6 +22,7 @@
 #include <config.h>
 #include <math.h>
 #include <stdlib.h>
+#include <string.h>
 
 #include <mjpeg_types.h>
 #include <mjpeg_logging.h>
diff -Nur mjpegtools-1.9.0rc3-old/mplex/padstrm.cpp mjpegtools-1.9.0rc3/mplex/padstrm.cpp
--- mjpegtools-1.9.0rc3-old/mplex/padstrm.cpp	2008-04-04 18:20:33.000000000 +0300
+++ mjpegtools-1.9.0rc3/mplex/padstrm.cpp	2008-04-04 18:20:49.000000000 +0300
@@ -24,6 +24,7 @@
 #include "config.h"
 #endif
 
+#include <string.h>
 #include "padstrm.hpp"
 
 
diff -Nur mjpegtools-1.9.0rc3-old/y4mdenoise/SkipList.hh mjpegtools-1.9.0rc3/y4mdenoise/SkipList.hh
--- mjpegtools-1.9.0rc3-old/y4mdenoise/SkipList.hh	2008-04-04 18:20:33.000000000 +0300
+++ mjpegtools-1.9.0rc3/y4mdenoise/SkipList.hh	2008-04-04 18:20:49.000000000 +0300
@@ -57,7 +57,7 @@
 		// Will give good sorting for up to e^10 items.
 	
 public:
-	typedef Allocator<Node,HEADERCHUNK> Allocator;
+        typedef ::Allocator<Node,HEADERCHUNK> Allocator;
 		// The type of node allocator to use.
 
 	static Allocator sm_oNodeAllocator;
