--- mjpegtools-1.9.0/mplex/lpcmstrm_in.cpp	2005-10-13 18:43:10.000000000 +0200
+++ mjpegtools-1.9.0.fixed/mplex/lpcmstrm_in.cpp	2010-04-15 11:16:46.103830391 +0200
@@ -53,7 +53,7 @@
 
 bool LPCMStream::Probe(IBitStream &bs )
 {
-    char *last_dot = strrchr( bs.StreamName(), '.' );
+    const char *last_dot = strrchr( bs.StreamName(), '.' );
     return 
         last_dot != NULL 
         && strcmp( last_dot+1, "lpcm") == 0;
