--- ../mjpegtools-1.8.0-nopng/configure.ac	2006-07-04 17:56:00.000000000 +0200
+++ configure.ac	2006-07-04 18:49:09.801873871 +0200
@@ -233,6 +233,10 @@
       have_jpeg=true ],,)
 fi
 
+dnl We need this so that skipping the quicktime test doesn't cause
+dnl png test to be skipped
+PKG_PROG_PKG_CONFIG()
+
 have_libquicktime=false
 if test x$with_libquicktime != xno ; then
   PKG_CHECK_MODULES(LIBQUICKTIME, [libquicktime >= 0.9.4],
