--- Makefile.in.old	2003-07-10 16:07:52.000000000 -0700
+++ Makefile.in	2003-07-10 16:07:58.000000000 -0700
@@ -67,7 +67,7 @@
 
 EXTRA_DIST = makedevs rmdevs cvsfs.spec.in
 
-SUBDIRS = cvsfs cvsfsd cvsmnt cvsmount cvsumount include init tools
+SUBDIRS = cvsfs cvsfsd cvsmnt cvsmount cvsumount include tools
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_CLEAN_FILES =  cvsfs.spec
