diff -urpN mpich-1.2.6.orig/configure mpich-1.2.6/configure
--- mpich-1.2.6.orig/configure	2004-08-02 20:00:32.000000000 +0000
+++ mpich-1.2.6/configure	2004-08-12 20:17:07.885556512 +0000
@@ -575,6 +575,9 @@ do
      -srcdir | --srcdir | --srcdi | --srcd | --src | --sr | --s)
 	next_srcdir=yes ;;
 
+     -sysconfdir=* | --sysconfdir=* )
+       sysconfdir=`echo $arg | sed 's/[-a-z_]*=//'` ;;
+
      -datadir=* | --datadir=* )
 	datadir=`echo $arg | sed 's/[-a-z_]*=//'` ;;
 
