Index: slicker/carddeskcore/Card.cpp
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/Card.cpp,v
retrieving revision 1.20
diff -u -r1.20 Card.cpp
--- slicker/carddeskcore/Card.cpp	19 Apr 2003 14:41:32 -0000	1.20
+++ slicker/carddeskcore/Card.cpp	10 Jul 2004 14:29:20 -0000
@@ -737,4 +737,4 @@
 	return false;
 }
 
-}; //namespace
+} //namespace
Index: slicker/carddeskcore/Card.h
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/Card.h,v
retrieving revision 1.12
diff -u -r1.12 Card.h
--- slicker/carddeskcore/Card.h	15 Apr 2003 15:46:37 -0000	1.12
+++ slicker/carddeskcore/Card.h	10 Jul 2004 14:29:21 -0000
@@ -411,7 +411,7 @@
 typedef QPtrListIterator<Card> CardListIterator;
 typedef QPtrVector<Card> CardVector;
 
-}; //namespace
+} //namespace
 
 #endif
 
Index: slicker/carddeskcore/CardApplet.cpp
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/CardApplet.cpp,v
retrieving revision 1.12
diff -u -r1.12 CardApplet.cpp
--- slicker/carddeskcore/CardApplet.cpp	12 Apr 2003 17:53:42 -0000	1.12
+++ slicker/carddeskcore/CardApplet.cpp	10 Jul 2004 14:29:21 -0000
@@ -177,4 +177,4 @@
 	_card->connect(this, SIGNAL(propertiesChanged()), SLOT(appletPropertiesChanged()));
 }
 
-}; //namespace
+} //namespace
Index: slicker/carddeskcore/CardApplet.h
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/CardApplet.h,v
retrieving revision 1.12
diff -u -r1.12 CardApplet.h
--- slicker/carddeskcore/CardApplet.h	13 Apr 2003 20:15:08 -0000	1.12
+++ slicker/carddeskcore/CardApplet.h	10 Jul 2004 14:29:22 -0000
@@ -316,6 +316,6 @@
 		QString _name;
 };
 
-}; //CardDesk namespace
+} //CardDesk namespace
 
 #endif
Index: slicker/carddeskcore/CardElements.cpp
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/CardElements.cpp,v
retrieving revision 1.30
diff -u -r1.30 CardElements.cpp
--- slicker/carddeskcore/CardElements.cpp	15 Sep 2003 22:51:59 -0000	1.30
+++ slicker/carddeskcore/CardElements.cpp	10 Jul 2004 14:29:27 -0000
@@ -1999,4 +1999,4 @@
 
 
 
-}; //namespace
+} //namespace
Index: slicker/carddeskcore/CardElements.h
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/CardElements.h,v
retrieving revision 1.9
diff -u -r1.9 CardElements.h
--- slicker/carddeskcore/CardElements.h	13 Apr 2003 20:15:08 -0000	1.9
+++ slicker/carddeskcore/CardElements.h	10 Jul 2004 14:29:28 -0000
@@ -605,7 +605,7 @@
 		int _originalContentsWidth, _newContentsWidth;
 };
 
-}; //namespace
+} //namespace
 
 #endif
 
Index: slicker/carddeskcore/CardManager.cpp
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/CardManager.cpp,v
retrieving revision 1.12
diff -u -r1.12 CardManager.cpp
--- slicker/carddeskcore/CardManager.cpp	19 Apr 2003 14:41:32 -0000	1.12
+++ slicker/carddeskcore/CardManager.cpp	10 Jul 2004 14:29:30 -0000
@@ -560,4 +560,4 @@
 {
 }
 
-}; //namespace
+} //namespace
Index: slicker/carddeskcore/CardManager.h
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/CardManager.h,v
retrieving revision 1.9
diff -u -r1.9 CardManager.h
--- slicker/carddeskcore/CardManager.h	19 Apr 2003 14:41:33 -0000	1.9
+++ slicker/carddeskcore/CardManager.h	10 Jul 2004 14:29:31 -0000
@@ -267,6 +267,6 @@
 		static Manager *_instance;
 };
 
-}; //namespace
+} //namespace
 
 #endif
Index: slicker/carddeskcore/Deck.cpp
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/Deck.cpp,v
retrieving revision 1.7
diff -u -r1.7 Deck.cpp
--- slicker/carddeskcore/Deck.cpp	12 Apr 2003 17:53:43 -0000	1.7
+++ slicker/carddeskcore/Deck.cpp	10 Jul 2004 14:29:32 -0000
@@ -554,7 +554,7 @@
 
 }
 
-}; //namespace
+} //namespace
 
 
 
Index: slicker/carddeskcore/Deck.h
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/Deck.h,v
retrieving revision 1.5
diff -u -r1.5 Deck.h
--- slicker/carddeskcore/Deck.h	12 Apr 2003 17:53:43 -0000	1.5
+++ slicker/carddeskcore/Deck.h	10 Jul 2004 14:29:33 -0000
@@ -209,7 +209,7 @@
 		bool _setWindowManager;
 };
 
-}; //namespace
+} //namespace
 
 #endif
 
Index: slicker/carddeskcore/LayoutSwitcher.cpp
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/LayoutSwitcher.cpp,v
retrieving revision 1.9
diff -u -r1.9 LayoutSwitcher.cpp
--- slicker/carddeskcore/LayoutSwitcher.cpp	2 May 2003 09:16:07 -0000	1.9
+++ slicker/carddeskcore/LayoutSwitcher.cpp	10 Jul 2004 14:29:33 -0000
@@ -218,4 +218,4 @@
 	setEdge(_edge);
 }
 
-}; //namespace
+} //namespace
Index: slicker/carddeskcore/LayoutSwitcher.h
===================================================================
RCS file: /cvsroot/slicker/slicker/slicker/carddeskcore/LayoutSwitcher.h,v
retrieving revision 1.5
diff -u -r1.5 LayoutSwitcher.h
--- slicker/carddeskcore/LayoutSwitcher.h	10 Apr 2003 11:20:43 -0000	1.5
+++ slicker/carddeskcore/LayoutSwitcher.h	10 Jul 2004 14:29:34 -0000
@@ -66,6 +66,6 @@
 		int _spacing;
 };
 
-}; //namespace
+} //namespace
 
 #endif
