diff -Naur a2ps-4.13.orig/contrib/emacs/Makefile.in a2ps-4.13/contrib/emacs/Makefile.in
--- a2ps-4.13.orig/contrib/emacs/Makefile.in	2005-06-24 14:24:07.024494280 -0700
+++ a2ps-4.13/contrib/emacs/Makefile.in	2005-06-24 14:24:15.678178720 -0700
@@ -31,7 +31,7 @@
 includedir = @includedir@
 oldincludedir = /usr/include
 
-DESTDIR =
+DESTDIR = @datadir@/@PACKAGE@
 
 pkgdatadir = $(datadir)/@PACKAGE@
 pkglibdir = $(libdir)/@PACKAGE@
