diff -uNr parted-1.8.3/include/parted/disk.h parted-1.8.3.cp/include/parted/disk.h
--- parted-1.8.3/include/parted/disk.h	2007-03-13 18:01:09.000000000 +0100
+++ parted-1.8.3.cp/include/parted/disk.h	2007-03-17 11:12:54.000000000 +0100
@@ -27,15 +27,6 @@
 #ifndef PED_DISK_H_INCLUDED
 #define PED_DISK_H_INCLUDED
 
-typedef enum _PedPartitionType          PedPartitionType;
-typedef enum _PedPartitionFlag          PedPartitionFlag;
-typedef enum _PedDiskTypeFeature        PedDiskTypeFeature;
-typedef struct _PedDisk                 PedDisk;
-typedef struct _PedPartition            PedPartition;
-typedef const struct _PedDiskOps        PedDiskOps;
-typedef struct _PedDiskType             PedDiskType;
-typedef const struct _PedDiskArchOps    PedDiskArchOps;
-
 #include <parted/device.h>
 #include <parted/filesys.h>
 #include <parted/natmath.h>
@@ -79,6 +70,16 @@
 #define PED_DISK_TYPE_FIRST_FEATURE    PED_DISK_TYPE_EXTENDED
 #define PED_DISK_TYPE_LAST_FEATURE     PED_DISK_TYPE_PARTITION_NAME
 
+typedef enum _PedPartitionType          PedPartitionType;
+typedef enum _PedPartitionFlag          PedPartitionFlag;
+typedef enum _PedDiskTypeFeature        PedDiskTypeFeature;
+typedef struct _PedDisk                 PedDisk;
+typedef struct _PedPartition            PedPartition;
+typedef const struct _PedDiskOps        PedDiskOps;
+typedef struct _PedDiskType             PedDiskType;
+typedef const struct _PedDiskArchOps    PedDiskArchOps;
+
+
 /** @} */
 
 /**
