From 50f6833653e3150226159474ec7bac02e1e01252 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 26 Jan 2018 21:12:58 +0100
Subject: [PATCH] fix dependency on entropy

---
 uuid.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/uuid.cabal b/uuid.cabal
index 2e46f6b..e95048d 100644
--- a/uuid.cabal
+++ b/uuid.cabal
@@ -34,7 +34,7 @@ Library
                        bytestring      >= 0.10     && < 0.11,
                        cryptohash-sha1 >= 0.11.100 && < 0.12,
                        cryptohash-md5  >= 0.11.100 && < 0.12,
-                       entropy         >= 0.3.7    && < 0.4,
+                       entropy         >= 0.3.7    && < 0.5,
                        network-info    == 0.2.*,
                        random          >= 1.0.1    && < 1.2,
                        time            >= 1.1      && < 1.9,
-- 
2.15.1

