From 0541e18b442d2f46abc0f0b09e0db950e1b657e5 Mon Sep 17 00:00:00 2001
From: Michael Drake <tlsa@netsurf-browser.org>
Date: Fri, 24 May 2024 21:19:20 +0100
Subject: [PATCH] select: computed: Squash -Wcalloc-transposed-args (gcc-14)

Upstream-Status: Backport [https://github.com/netsurf-browser/libcss/commit/0541e18b442d2f46abc0f0b09e0db950e1b657e5]
---
 src/select/computed.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/select/computed.c b/src/select/computed.c
index 78f3b80..590b8ba 100644
--- a/src/select/computed.c
+++ b/src/select/computed.c
@@ -73,7 +73,7 @@ css_error css__computed_style_create(css_computed_style **result)
 	if (result == NULL)
 		return CSS_BADPARM;
 
-	s = calloc(sizeof(css_computed_style), 1);
+	s = calloc(1, sizeof(css_computed_style));
 	if (s == NULL)
 		return CSS_NOMEM;
 
