--- mono/Makefile.in.orig	2008-11-18 12:43:36.000000000 -0600
+++ mono/Makefile.in	2008-11-18 12:44:24.000000000 -0600
@@ -488,11 +488,11 @@
 install-data-local:
 	@if test -n '$(TARGET)'; then									\
 	  if test -n '$(DESTDIR)'; then								\
-	    echo "$(GACUTIL) /i $(ASSEMBLY) /f /package $(PACKAGE_SHARP) /root $(DESTDIR)$(prefix)/lib";		\
-	    $(GACUTIL) /i $(ASSEMBLY) /f /package $(PACKAGE_SHARP) /root $(DESTDIR)$(prefix)/lib || exit 1;	\
+	    echo "$(GACUTIL) -bootstrap /i $(ASSEMBLY) /f /package $(PACKAGE_SHARP) /root $(DESTDIR)$(prefix)/lib";		\
+	    $(GACUTIL) -bootstrap /i $(ASSEMBLY) /f /package $(PACKAGE_SHARP) /root $(DESTDIR)$(prefix)/lib || exit 1;	\
 	  else												\
-	    echo "$(GACUTIL) /i $(ASSEMBLY) /f /package $(PACKAGE_SHARP) /gacdir $(prefix)/lib";			\
-	    $(GACUTIL) /i $(ASSEMBLY) /f /package $(PACKAGE_SHARP) /gacdir $(prefix)/lib || exit 1;		\
+	    echo "$(GACUTIL) -bootstrap /i $(ASSEMBLY) /f /package $(PACKAGE_SHARP) /gacdir $(prefix)/lib";			\
+	    $(GACUTIL) -bootstrap /i $(ASSEMBLY) /f /package $(PACKAGE_SHARP) /gacdir $(prefix)/lib || exit 1;		\
 	  fi;												\
 	fi
 
