diff -Naur xml-xalan.orig/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp xml-xalan/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp
--- xml-xalan.orig/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp	2008-04-26 23:26:57.562243805 -0700
+++ xml-xalan/c/src/xalanc/PlatformSupport/DirectoryEnumerator.hpp	2008-04-26 23:35:04.648219538 -0700
@@ -33,7 +33,7 @@
 #endif
 
 
-
+#include <cstring>
 #include <functional>
 #include <iterator>
 
diff -Naur xml-xalan.orig/c/src/xalanc/XalanDOM/XalanDOMString.cpp xml-xalan/c/src/xalanc/XalanDOM/XalanDOMString.cpp
--- xml-xalan.orig/c/src/xalanc/XalanDOM/XalanDOMString.cpp	2008-04-26 23:26:57.543246489 -0700
+++ xml-xalan/c/src/xalanc/XalanDOM/XalanDOMString.cpp	2008-04-26 23:29:52.102244630 -0700
@@ -19,7 +19,7 @@
 
 
 #include <cassert>
-
+#include <cstring>
 
 
 #include <xalanc/Include/XalanMemMngArrayAllocate.hpp>
diff -Naur xml-xalan.orig/c/src/xalanc/XalanExe/XalanExe.cpp xml-xalan/c/src/xalanc/XalanExe/XalanExe.cpp
--- xml-xalan.orig/c/src/xalanc/XalanExe/XalanExe.cpp	2008-04-26 23:26:57.539246609 -0700
+++ xml-xalan/c/src/xalanc/XalanExe/XalanExe.cpp	2008-04-26 23:39:40.217246885 -0700
@@ -19,6 +19,8 @@
 
 
 #include <cstdlib>
+#include <cstring>
+
 #if defined(XALAN_CLASSIC_IOSTREAMS)
 #include <iostream.h>
 #else
diff -Naur xml-xalan.orig/c/src/xalanc/XMLSupport/FormatterToHTML.cpp xml-xalan/c/src/xalanc/XMLSupport/FormatterToHTML.cpp
--- xml-xalan.orig/c/src/xalanc/XMLSupport/FormatterToHTML.cpp	2008-04-26 23:26:57.558245951 -0700
+++ xml-xalan/c/src/xalanc/XMLSupport/FormatterToHTML.cpp	2008-04-26 23:29:44.427218893 -0700
@@ -26,7 +26,7 @@
 
 
 #include <cassert>
-
+#include <cstring>
 
 
 #include <xercesc/sax/AttributeList.hpp>
diff -Naur xml-xalan.orig/c/src/xalanc/XSLT/ElemNumber.cpp xml-xalan/c/src/xalanc/XSLT/ElemNumber.cpp
--- xml-xalan.orig/c/src/xalanc/XSLT/ElemNumber.cpp	2008-04-26 23:26:57.549242642 -0700
+++ xml-xalan/c/src/xalanc/XSLT/ElemNumber.cpp	2008-04-26 23:32:11.155219240 -0700
@@ -15,7 +15,7 @@
  */
 #include "ElemNumber.hpp"
 
-
+#include <cstring>
 
 #include <xercesc/sax/AttributeList.hpp>
 
