--- src/eatmonkey.old	2010-03-08 23:05:25.868184630 -0500
+++ src/eatmonkey	2010-03-08 23:05:38.268059312 -0500
@@ -3,7 +3,7 @@
 # this wrapper script a little ridiculous.
 if test -x eat.rb; then
 	./eat.rb || exit 1
-elif test -x "/usr/local/share/eatmonkey"/eat.rb; then
-	cd "/usr/local/share/eatmonkey" && ./eat.rb || exit 1
+elif test -x "/usr/share/eatmonkey"/eat.rb; then
+	cd "/usr/share/eatmonkey" && ./eat.rb || exit 1
 fi
 exit 0
