--- libva-1.6.0/configure.ac	2015-07-01 09:52:44.000000000 +0200
+++ libva-1.6.0.fixed/configure.ac	2015-08-18 11:17:17.481068815 +0200
@@ -275,14 +275,15 @@
     if test "$USE_WAYLAND" = "yes"; then
         AC_DEFINE([HAVE_VA_WAYLAND], [1],
                   [Defined to 1 if VA/Wayland API is built])
+        m4_ifdef([WAYLAND_SCANNER_RULES],
+            [WAYLAND_SCANNER_RULES(['$(top_builddir)/va/wayland/protocol'])],
+            [wayland_scanner_rules=""; AC_SUBST(wayland_scanner_rules)])
+    else
+        AC_MSG_ERROR([Wayland selected but not found in system.])
     fi
 fi
 AM_CONDITIONAL(USE_WAYLAND, test "$USE_WAYLAND" = "yes")
 
-m4_ifdef([WAYLAND_SCANNER_RULES],
-    [WAYLAND_SCANNER_RULES(['$(top_builddir)/va/wayland/protocol'])],
-    [wayland_scanner_rules=""; AC_SUBST(wayland_scanner_rules)])
-
 pkgconfigdir=${libdir}/pkgconfig
 AC_SUBST(pkgconfigdir)
 
