diff -ur dirac-1.0.2/configure dirac-mod/configure
--- dirac-1.0.2/configure	2009-02-10 19:43:40.000000000 -0600
+++ dirac-mod/configure	2011-08-11 19:02:20.545001160 -0500
@@ -19910,7 +19910,7 @@
     else
 	:
     fi
-		TRY_CFLAGS="$OPT_CFLAGS -Werror"
+		TRY_CFLAGS="$OPT_CFLAGS"
 		{ $as_echo "$as_me:$LINENO: checking if $CXX supports $TRY_CFLAGS $CXXFLAGS flags" >&5
 $as_echo_n "checking if $CXX supports $TRY_CFLAGS $CXXFLAGS flags... " >&6; }
 	SAVE_CXXFLAGS="$CXXFLAGS"
@@ -20211,7 +20211,7 @@
 			CXXFLAGS=`echo $CXXFLAGS | sed 's/-O[0-9]*//g'`
 			CXXFLAGS="-DDIRAC_DEBUG -g -O0 $CXXFLAGS"
 		    # report only errors and warnings as errors
-		    CXXFLAGS="$CXXFLAGS -w1 -Werror -wr964 -wr963"
+		    CXXFLAGS="$CXXFLAGS -w1 -wr964 -wr963"
 		    ;;
 	    cl*|CL*)
 		    # -Od turn off optimisation, -Z7 turn on debugging info
diff -ur dirac-1.0.2/configure.ac dirac-mod/configure.ac
--- dirac-1.0.2/configure.ac	2009-02-09 18:16:37.000000000 -0600
+++ dirac-mod/configure.ac	2011-08-11 19:02:50.493000929 -0500
@@ -183,7 +183,7 @@
 		AC_TRY_CXXFLAGS([],[], [$TRY_CFLAGS $CXXFLAGS],[OPT_CFLAGS="$TRY_CFLAGS"])
 		TRY_CFLAGS="$OPT_CFLAGS -Wall"
 		AC_TRY_CXXFLAGS([],[], [$TRY_CFLAGS $CXXFLAGS],[OPT_CFLAGS="$TRY_CFLAGS"])
-		TRY_CFLAGS="$OPT_CFLAGS -Werror"
+		TRY_CFLAGS="$OPT_CFLAGS"
 		AC_TRY_CXXFLAGS([],[], [$TRY_CFLAGS $CXXFLAGS],[OPT_CFLAGS="$TRY_CFLAGS"])
 		TRY_CFLAGS="$OPT_CFLAGS -W"
 		AC_TRY_CXXFLAGS([],[], [$TRY_CFLAGS $CXXFLAGS],[OPT_CFLAGS="$TRY_CFLAGS"])
@@ -219,7 +219,7 @@
 			CXXFLAGS=`echo $CXXFLAGS | sed 's/-O[[0-9]]*//g'`
 			CXXFLAGS="-DDIRAC_DEBUG -g -O0 $CXXFLAGS"
 		    # report only errors and warnings as errors
-		    CXXFLAGS="$CXXFLAGS -w1 -Werror -wr964 -wr963" 
+		    CXXFLAGS="$CXXFLAGS -w1 -wr964 -wr963" 
 		    ;;
 	    cl*|CL*)
 		    # -Od turn off optimisation, -Z7 turn on debugging info
