From b8a1116b7911a5a3064e91c77d3ba537fb8b36b1 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Thu, 15 Sep 2016 19:03:19 +0200
Subject: [PATCH] fix install path

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index f6dea6c..f866037 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 abcde_version = abcde-2.7.2
 INSTALL = /usr/bin/install -c
 
-prefix = /usr/local
+prefix = /usr
 exec_prefix = $(prefix)
 bindir = $(exec_prefix)/bin
 sysconfdir = /etc
-- 
2.10.0

