--- bin/build-ooo.orig	2004-09-06 17:12:37.986713344 -0500
+++ bin/build-ooo	2004-09-06 17:13:26.048406856 -0500
@@ -49,7 +49,8 @@
 
 if test "z$ENABLE_JAVA" = "zyes"; then
 	# Many Java's can't cope with the NPTL on Linux.
-	LD_ASSUME_KERNEL=2.2.5 /bin/true 2> /dev/null || LD_ASSUME_KERNEL=2.4.10 /bin/true 2> /dev/null || LD_ASSUME_KERNEL=2.6.0 2> /dev/null || unset LD_ASSUME_KERNEL
+#	LD_ASSUME_KERNEL=2.2.5 /bin/true 2> /dev/null || LD_ASSUME_KERNEL=2.4.10 /bin/true 2> /dev/null || LD_ASSUME_KERNEL=2.6.0 2> /dev/null || unset LD_ASSUME_KERNEL
+	unset LD_ASSUME_KERNEL
 	export LD_ASSUME_KERNEL
 fi
 # Don't do a massive painful install
