--- installwatch-0.6.3/Makefile.ori	2004-09-08 16:05:24.475554253 +0200
+++ installwatch-0.6.3/Makefile	2004-09-08 16:06:55.238768317 +0200
@@ -16,10 +16,10 @@
 all: installwatch.so
 
 installwatch.so: installwatch.o
-	ld -shared -o installwatch.so installwatch.o -ldl -lc
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -nostdlib -o installwatch.so installwatch.o -ldl -lc
 
 installwatch.o: installwatch.c localdecls.h
-	gcc -Wall -c -D_GNU_SOURCE -DPIC -fPIC -D_REENTRANT -DVERSION=\"$(VERSION)\" installwatch.c
+	$(CC) $(CFLAGS) -c -D_GNU_SOURCE -DPIC -fPIC -D_REENTRANT -DVERSION=\"$(VERSION)\" installwatch.c
 
 localdecls.h:
 	./create-localdecls
@@ -33,7 +33,7 @@
 uninstall:
 	rm $(LIBDIR)/installwatch.so
 	rm $(BINDIR)/installwatch
-	
+
 clean:
 	rm -f *~ *.bak *.o installwatch.so core localdecls.h libctest test-installwatch
 
