diff -Naur installwatch.ori//create-localdecls installwatch/create-localdecls
--- installwatch.ori//create-localdecls	2011-02-05 12:46:19.000000000 +0100
+++ installwatch/create-localdecls	2011-02-05 12:50:10.000000000 +0100
@@ -105,6 +105,9 @@
 		12)
 			echo '#define GLIBC_MINOR 12' >> $OUTFILE
 			SUBVERSION='glibc-2.12' ;;
+		13)
+			echo '#define GLIBC_MINOR 13' >> $OUTFILE
+			SUBVERSION='glibc-2.13' ;;
 		*)
 			echo 'Treated as glibc >= 2.1 (finger crossed)'
 			echo '#define GLIBC_MINOR 1' >> $OUTFILE
