diff -urN installwatch-0.6.3/installwatch.c installwatch-0.6.4/installwatch.c
--- installwatch-0.6.3/installwatch.c	2004-11-10 17:34:11.817461389 +0100
+++ installwatch-0.6.4/installwatch.c	2004-11-10 17:00:42.614038577 +0100
@@ -164,7 +164,7 @@
 			syslog(LOGLEVEL, "Could not open `%s' to write `%s': %s\n", logname, buffer, strerror(errno));
 		
 	} else
-		syslog(LOGLEVEL, buffer);
+		syslog(LOGLEVEL, "%s", buffer);
 
 	/* restore errno */
 	errno = s_errno;
