--- Makefile.ori	2010-02-03 16:59:10.000000000 +0100
+++ Makefile	2010-02-03 18:02:11.000000000 +0100
@@ -27,11 +27,8 @@
 install: all
 	mkdir -p $(LIBDIR)
 	mkdir -p $(BINDIR)
-	if [ -r $(LIBDIR)/installwatch.so ]; then \
-		rm -f  $(LIBDIR)/installwatch.so; \
-	fi
-	install installwatch.so $(LIBDIR)
-	
+	cp installwatch.so $(LIBDIR)/installwatch.so.new
+	mv $(LIBDIR)/installwatch.so.new $(LIBDIR)/installwatch.so
 	sed -e "s|#PREFIX#|$(PREFIX)|" < installwatch > $(BINDIR)/installwatch
 	chmod 755 $(BINDIR)/installwatch
 
