diff --git a/installwatch-0.7.0beta4/Makefile b/installwatch-0.7.0beta4/Makefile
index 9f571fe..d64700e 100644
--- a/installwatch-0.7.0beta4/Makefile
+++ b/installwatch-0.7.0beta4/Makefile
@@ -16,10 +16,10 @@ LIBDIR=$(PREFIX)/lib
 all: installwatch.so
 
 installwatch.so: installwatch.o
-	ld -shared -o installwatch.so installwatch.o -ldl -lc
+	$(LD) $(LDFLAGS) -shared -o installwatch.so installwatch.o -ldl -lc
 
 installwatch.o: installwatch.c localdecls.h
-	gcc -Wall -c -D_GNU_SOURCE -DPIC -fPIC -D_REENTRANT -DVERSION=\"$(VERSION)\" installwatch.c
+	$(CC) $(CFLAGS) -c -D_GNU_SOURCE -DPIC -fPIC -D_REENTRANT -DVERSION=\"$(VERSION)\" installwatch.c
 
 localdecls.h:
 	./create-localdecls
