diff --git a/installwatch-0.7.0beta4/installwatch.c b/installwatch-0.7.0beta4/installwatch.c
index 7db45c9..3284760 100644
--- a/installwatch-0.7.0beta4/installwatch.c
+++ b/installwatch-0.7.0beta4/installwatch.c
@@ -1063,15 +1063,6 @@ #endif
 		if(NULL==__instw.root) return -1;
 		strcpy(__instw.root,wrkpath);
 
-		  /* this root path must exist */
-		if(__instw.root[0]=='\0' || true_stat(__instw.root,&info)) {
-			fprintf(stderr,
-				"Please check the INSTW_ROOTPATH and "
-				"be sure that it does exist please !\n"
-				"given value : %s\n", __instw.root);
-			return -1;	
-		}
-
 		if((pbackup=getenv("INSTW_BACKUP"))) {
 			if(	!strcmp(pbackup,"1") ||
 				!strcmp(pbackup,"yes") ||
