--- configure.orig	2005-10-10 19:59:58.000000000 -0500
+++ configure	2005-10-10 20:07:50.000000000 -0500
@@ -2266,23 +2266,23 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo "$as_me:$LINENO: checking for glade-sharp gnome-sharp gconf-sharp gtkhtml-sharp" >&5
-echo $ECHO_N "checking for glade-sharp gnome-sharp gconf-sharp gtkhtml-sharp... $ECHO_C" >&6
+        echo "$as_me:$LINENO: checking for glade-sharp-2.0 gnome-sharp-2.0 gconf-sharp-2.0 gtkhtml-sharp-2.0" >&5
+echo $ECHO_N "checking for glade-sharp-2.0 gnome-sharp-2.0 gconf-sharp-2.0 gtkhtml-sharp-2.0... $ECHO_C" >&6
 
-        if $PKG_CONFIG --exists "glade-sharp gnome-sharp gconf-sharp gtkhtml-sharp" ; then
+        if $PKG_CONFIG --exists "glade-sharp-2.0 gnome-sharp-2.0 gconf-sharp-2.0 gtkhtml-sharp-2.0" ; then
             echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
             succeeded=yes
 
             echo "$as_me:$LINENO: checking GTK_SHARP_CFLAGS" >&5
 echo $ECHO_N "checking GTK_SHARP_CFLAGS... $ECHO_C" >&6
-            GTK_SHARP_CFLAGS=`$PKG_CONFIG --cflags "glade-sharp gnome-sharp gconf-sharp gtkhtml-sharp"`
+            GTK_SHARP_CFLAGS=`$PKG_CONFIG --cflags "glade-sharp-2.0 gnome-sharp2.0 gconf-sharp-2.0 gtkhtml-sharp-2.0"`
             echo "$as_me:$LINENO: result: $GTK_SHARP_CFLAGS" >&5
 echo "${ECHO_T}$GTK_SHARP_CFLAGS" >&6
 
             echo "$as_me:$LINENO: checking GTK_SHARP_LIBS" >&5
 echo $ECHO_N "checking GTK_SHARP_LIBS... $ECHO_C" >&6
-            GTK_SHARP_LIBS=`$PKG_CONFIG --libs "glade-sharp gnome-sharp gconf-sharp gtkhtml-sharp"`
+            GTK_SHARP_LIBS=`$PKG_CONFIG --libs "glade-sharp-2.0 gnome-sharp-2.0 gconf-sharp-2.0 gtkhtml-sharp-2.0"`
             echo "$as_me:$LINENO: result: $GTK_SHARP_LIBS" >&5
 echo "${ECHO_T}$GTK_SHARP_LIBS" >&6
         else
@@ -2290,7 +2290,7 @@
             GTK_SHARP_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            GTK_SHARP_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "glade-sharp gnome-sharp gconf-sharp gtkhtml-sharp"`
+            GTK_SHARP_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "glade-sharp-2.0 gnome-sharp-2.0 gconf-sharp-2.0 gtkhtml-sharp-2.0"`
 
         fi
 
@@ -2370,23 +2370,23 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo "$as_me:$LINENO: checking for gecko-sharp = 0.6" >&5
+        echo "$as_me:$LINENO: checking for gecko-sharp-2.0 = 0.11" >&5
 echo $ECHO_N "checking for gecko-sharp = 0.6... $ECHO_C" >&6
 
-        if $PKG_CONFIG --exists "gecko-sharp = 0.6" ; then
+        if $PKG_CONFIG --exists "gecko-sharp-2.0 = 0.11" ; then
             echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
             succeeded=yes
 
             echo "$as_me:$LINENO: checking GECKO_SHARP_CFLAGS" >&5
 echo $ECHO_N "checking GECKO_SHARP_CFLAGS... $ECHO_C" >&6
-            GECKO_SHARP_CFLAGS=`$PKG_CONFIG --cflags "gecko-sharp = 0.6"`
+            GECKO_SHARP_CFLAGS=`$PKG_CONFIG --cflags "gecko-sharp-2.0 = 0.11"`
             echo "$as_me:$LINENO: result: $GECKO_SHARP_CFLAGS" >&5
 echo "${ECHO_T}$GECKO_SHARP_CFLAGS" >&6
 
             echo "$as_me:$LINENO: checking GECKO_SHARP_LIBS" >&5
 echo $ECHO_N "checking GECKO_SHARP_LIBS... $ECHO_C" >&6
-            GECKO_SHARP_LIBS=`$PKG_CONFIG --libs "gecko-sharp = 0.6"`
+            GECKO_SHARP_LIBS=`$PKG_CONFIG --libs "gecko-sharp-2.0 = 0.11"`
             echo "$as_me:$LINENO: result: $GECKO_SHARP_LIBS" >&5
 echo "${ECHO_T}$GECKO_SHARP_LIBS" >&6
         else
@@ -2394,7 +2394,7 @@
             GECKO_SHARP_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            GECKO_SHARP_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gecko-sharp = 0.6"`
+            GECKO_SHARP_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gecko-sharp-2.0 = 0.11"`
 
         fi
 
