diff -ur src.orig/tabe_tsidbint.c src/tabe_tsidbint.c
--- src.orig/tabe_tsidbint.c	2003-05-26 02:16:15.000000000 +0800
+++ src/tabe_tsidbint.c	2003-05-26 02:17:29.000000000 +0800
@@ -85,7 +85,7 @@
 #ifndef HAVE_DB3
       errno = db_open(db_name, DB_BTREE, DB_CREATE, 0644, NULL, NULL, &dbp);
 #else
-      errno = dbp->open(dbp, db_name, NULL, DB_BTREE, DB_CREATE, 0644);
+      errno = dbp->open(dbp, NULL, db_name, NULL, DB_BTREE, DB_CREATE, 0644);
 #endif
     }
   }
@@ -94,14 +94,14 @@
 #ifndef HAVE_DB3
       errno = db_open(db_name, DB_BTREE, DB_RDONLY, 0444, NULL, NULL, &dbp);
 #else
-      errno = dbp->open(dbp, db_name, NULL, DB_BTREE, DB_RDONLY, 0444);
+      errno = dbp->open(dbp, NULL, db_name, NULL, DB_BTREE, DB_RDONLY, 0444);
 #endif
     }
     else {
 #ifndef HAVE_DB3
       errno = db_open(db_name, DB_BTREE, 0, 0644, NULL, NULL, &dbp);
 #else
-      errno = dbp->open(dbp, db_name, NULL, DB_BTREE, 0, 0644);
+      errno = dbp->open(dbp, NULL, db_name, NULL, DB_BTREE, 0, 0644);
 #endif
     }
   }
diff -ur src.orig/tabe_tsiyindbint.c src/tabe_tsiyindbint.c
--- src.orig/tabe_tsiyindbint.c	2003-05-26 02:16:15.000000000 +0800
+++ src/tabe_tsiyindbint.c	2003-05-26 02:16:54.000000000 +0800
@@ -88,7 +88,7 @@
 #ifndef HAVE_DB3
       errno = db_open(db_name, DB_BTREE, DB_CREATE, 0644, NULL, NULL, &dbp);
 #else
-      errno = dbp->open(dbp, db_name, NULL, DB_BTREE, DB_CREATE, 0644);
+      errno = dbp->open(dbp, NULL, db_name, NULL, DB_BTREE, DB_CREATE, 0644);
 #endif
     }
   }
@@ -97,14 +97,14 @@
 #ifndef HAVE_DB3
       errno = db_open(db_name, DB_BTREE, DB_RDONLY, 0444, NULL, NULL, &dbp);
 #else
-      errno = dbp->open(dbp, db_name, NULL, DB_BTREE, DB_RDONLY, 0444);
+      errno = dbp->open(dbp, NULL, db_name, NULL, DB_BTREE, DB_RDONLY, 0444);
 #endif
     }
     else {
 #ifndef HAVE_DB3
       errno = db_open(db_name, DB_BTREE, 0, 0644, NULL, NULL, &dbp);
 #else
-      errno = dbp->open(dbp, db_name, NULL, DB_BTREE, 0, 0644);
+      errno = dbp->open(dbp, NULL, db_name, NULL, DB_BTREE, 0, 0644);
 #endif
     }
   }
