From 452d9985229d5676e2fc5b2f0f64b0845a2127eb Mon Sep 17 00:00:00 2001
From: llyyr <llyyr.public@gmail.com>
Date: Tue, 21 Nov 2023 23:55:19 +0530
Subject: [PATCH 02/15] meson: bump wlroots version after 0.17.0 release

Origin: Upstream
Upstream-Status: Backported [commit 4ad15a4015feef2ecfba311c13e64e3976cc611c]
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index c52d6629ab48..75654ad1df7c 100644
--- a/meson.build
+++ b/meson.build
@@ -37,7 +37,7 @@ if is_freebsd
 endif
 
 # Execute the wlroots subproject, if any
-wlroots_version = ['>=0.17.0', '<0.18.0']
+wlroots_version = ['>=0.18.0', '<0.19.0']
 subproject(
 	'wlroots',
 	default_options: ['examples=false'],
-- 
2.46.0

