From 4ad15a4015feef2ecfba311c13e64e3976cc611c Mon Sep 17 00:00:00 2001
From: llyyr <llyyr.public@gmail.com>
Date: Tue, 21 Nov 2023 23:55:19 +0530
Subject: [PATCH 2/15] meson: bump wlroots version after 0.17.0 release

Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index eeee39cde85b..adf49c196c19 100644
--- a/meson.build
+++ b/meson.build
@@ -37,7 +37,7 @@ if is_freebsd
 endif
 
 # Execute the wlroots subproject, if any
-wlroots_version = ['>=0.17.0', '<0.18.0']
+wlroots_version = ['>=0.18.0', '<0.19.0']
 subproject(
 	'wlroots',
 	default_options: ['examples=false'],
-- 
2.46.0

