From 21938477e5f0032419c3aad4d7f8c24e5ce162f9 Mon Sep 17 00:00:00 2001
From: Kirill Primak <vyivel@eclair.cafe>
Date: Thu, 25 Jan 2024 18:10:50 +0300
Subject: [PATCH 08/15] Chase wlroots!4443

Origin: Upstream
Upstream-Status: Backported [commit f202bc84d2c072ae39fc4e8a350a8b0a616f350b]
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 sway/server.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sway/server.c b/sway/server.c
index a49fa0d8a034..81df98a07727 100644
--- a/sway/server.c
+++ b/sway/server.c
@@ -132,7 +132,7 @@ bool server_init(struct sway_server *server) {
 
 	root = root_create(server->wl_display);
 
-	server->backend = wlr_backend_autocreate(server->wl_display, &server->session);
+	server->backend = wlr_backend_autocreate(server->wl_event_loop, &server->session);
 	if (!server->backend) {
 		sway_log(SWAY_ERROR, "Unable to create backend");
 		return false;
@@ -313,7 +313,7 @@ bool server_init(struct sway_server *server) {
 		return false;
 	}
 
-	server->headless_backend = wlr_headless_backend_create(server->wl_display);
+	server->headless_backend = wlr_headless_backend_create(server->wl_event_loop);
 	if (!server->headless_backend) {
 		sway_log(SWAY_ERROR, "Failed to create secondary headless backend");
 		wlr_backend_destroy(server->backend);
-- 
2.46.0

