From 287ffbeca8a2cf7b6e42542efa92adec2fb06018 Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Sun, 17 Mar 2024 13:27:34 -0400
Subject: [PATCH 12/15] Update for versioned wlroots files

References: https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4614
Origin: Upstream
Upstream-Status: Backported [commit 2e9139df664f1e2dbe14b5df4a9646411b924c66]
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 75654ad1df7c..af8ebca20580 100644
--- a/meson.build
+++ b/meson.build
@@ -44,7 +44,7 @@ subproject(
 	required: false,
 	version: wlroots_version,
 )
-wlroots = dependency('wlroots', version: wlroots_version)
+wlroots = dependency('wlroots-0.18', version: wlroots_version, fallback: 'wlroots')
 wlroots_features = {
 	'xwayland': false,
 	'libinput_backend': false,
-- 
2.46.0

