From c46fa5e2d3c55f4ca84c0d52189c26048e234930 Mon Sep 17 00:00:00 2001
From: Kirill Primak <vyivel@eclair.cafe>
Date: Thu, 27 Jun 2024 21:01:19 +0300
Subject: [PATCH 15/15] protocols: use tablet-v2 xml from stable/

Origin: Upstream
Upstream-Status: Backported [commit cdde0165dad94e4522f8f6f040e9d30145fbb14f]
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 protocols/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/protocols/meson.build b/protocols/meson.build
index 81edb5841b23..6eac8542c000 100644
--- a/protocols/meson.build
+++ b/protocols/meson.build
@@ -7,10 +7,10 @@ wayland_scanner = find_program(
 )
 
 protocols = [
+	wl_protocol_dir / 'stable/tablet/tablet-v2.xml',
 	wl_protocol_dir / 'stable/xdg-shell/xdg-shell.xml',
 	wl_protocol_dir / 'unstable/xdg-output/xdg-output-unstable-v1.xml',
 	wl_protocol_dir / 'unstable/pointer-constraints/pointer-constraints-unstable-v1.xml',
-	wl_protocol_dir / 'unstable/tablet/tablet-unstable-v2.xml',
 	wl_protocol_dir / 'unstable/linux-dmabuf/linux-dmabuf-unstable-v1.xml',
 	wl_protocol_dir / 'staging/content-type/content-type-v1.xml',
 	wl_protocol_dir / 'staging/cursor-shape/cursor-shape-v1.xml',
-- 
2.46.0

