--- libfreerdp/crypto/tls.c.orig	2020-04-09 09:00:51.000000000 -0700
+++ libfreerdp/crypto/tls.c	2020-04-10 10:34:39.099216963 -0700
@@ -661,7 +661,7 @@
 	SSL_CTX_set_mode(tls->ctx, SSL_MODE_ACCEPT_MOVING_WRITE_BUFFER | SSL_MODE_ENABLE_PARTIAL_WRITE);
 	SSL_CTX_set_options(tls->ctx, options);
 	SSL_CTX_set_read_ahead(tls->ctx, 1);
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)) || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER >= 0x2060000fL)
 	SSL_CTX_set_min_proto_version(tls->ctx, TLS1_VERSION); /* min version */
 	SSL_CTX_set_max_proto_version(tls->ctx, 0); /* highest supported version by library */
 #endif
