diff -Naur trousers-0.3.1.orig/configure trousers-0.3.1.fixed/configure
--- trousers-0.3.1.orig/configure	2007-11-02 22:59:49.000000000 +0100
+++ trousers-0.3.1.fixed/configure	2009-07-30 13:29:05.147050242 +0200
@@ -21770,7 +21770,7 @@
 
 
 
-CFLAGS="$CFLAGS -I../include -W -Wall -Werror -Wno-unused-parameter \
+CFLAGS="$CFLAGS -I../include -W -Wall -Wno-unused-parameter \
 	-DTCSD_DEFAULT_PORT=${TCSD_DEFAULT_PORT} -DTSS_VER_MAJOR=${TSS_VER_MAJOR} \
 	-DTSS_VER_MINOR=${TSS_VER_MINOR} -DTSS_SPEC_MAJOR=${TSS_SPEC_MAJOR} \
 	-DTSS_SPEC_MINOR=${TSS_SPEC_MINOR}"
diff -Naur trousers-0.3.1.orig/src/trspi/crypto/openssl/symmetric.c trousers-0.3.1.fixed/src/trspi/crypto/openssl/symmetric.c
--- trousers-0.3.1.orig/src/trspi/crypto/openssl/symmetric.c	2007-07-17 20:25:24.000000000 +0200
+++ trousers-0.3.1.fixed/src/trspi/crypto/openssl/symmetric.c	2009-07-30 13:10:52.654099250 +0200
@@ -16,6 +16,7 @@
  */
 
 #include <string.h>
+#include <limits.h>
 
 #include <openssl/evp.h>
 #include <openssl/err.h>
diff -Naur trousers-0.3.1.orig/src/tspi/tspi_aik.c trousers-0.3.1.fixed/src/tspi/tspi_aik.c
--- trousers-0.3.1.orig/src/tspi/tspi_aik.c	2007-09-25 20:57:58.000000000 +0200
+++ trousers-0.3.1.fixed/src/tspi/tspi_aik.c	2009-07-30 13:15:29.935097466 +0200
@@ -12,6 +12,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <inttypes.h>
+#include <limits.h>
 
 #include "trousers/tss.h"
 #include "trousers/trousers.h"
diff -Naur trousers-0.3.1.orig/src/tspi/tsp_ps.c trousers-0.3.1.fixed/src/tspi/tsp_ps.c
--- trousers-0.3.1.orig/src/tspi/tsp_ps.c	2007-07-19 20:00:05.000000000 +0200
+++ trousers-0.3.1.fixed/src/tspi/tsp_ps.c	2009-07-30 13:14:22.930119387 +0200
@@ -15,6 +15,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <stdio.h>
+#include <limits.h>
 
 #include "trousers/tss.h"
 #include "trousers/trousers.h"
