--- last.fm-1.4.1.57486/src/libFingerprint/libFingerprint.pro.orig	2007-12-22 13:08:59.000000000 +0100
+++ last.fm-1.4.1.57486/src/libFingerprint/libFingerprint.pro	2007-12-22 13:09:23.000000000 +0100
@@ -40,7 +40,7 @@
 
 unix:!mac {
     LIBPATH += $$BUILD_DIR/../fplib
-    LIBS += $$ROOT_DIR/build/fplib/libfplib$${EXT}.a -lsamplerate -lfftw3f
+    LIBS += $$ROOT_DIR/build/fplib/libfplib$${EXT}.a -lsamplerate -lfftw3f -lmad
 }
 
 win32 {
