--- src/compilers.c.orig	2022-07-22 20:08:24.000000000 -0700
+++ src/compilers.c	2022-07-22 21:07:09.847285953 -0700
@@ -465,7 +465,7 @@
 
 	switch ((enum compiler_warning_lvl)lvl) {
 	case compiler_warning_lvl_3:
-		argv[args.len] = "-Wpedantic";
+		argv[args.len] = "-Wall";
 		++args.len;
 	/* fallthrough */
 	case compiler_warning_lvl_2:
--- subprojects/bestline/bestline.c.orig	2022-07-22 20:08:24.000000000 -0700
+++ subprojects/bestline/bestline.c	2022-07-22 21:13:09.610868291 -0700
@@ -131,6 +131,8 @@
 ╚─────────────────────────────────────────────────────────────────────────────*/
 #include "bestline.h"
 
+#define _GNU_SOURCE
+
 #ifndef __COSMOPOLITAN__
 #define _POSIX_C_SOURCE  1 /* so GCC builds in ANSI mode */
 #define _XOPEN_SOURCE  700 /* so GCC builds in ANSI mode */
