--- src/cxxopts.hpp.orig	2017-08-14 14:13:46.000000000 -0400
+++ src/cxxopts.hpp	2017-08-14 14:50:39.388795666 -0400
@@ -35,7 +35,7 @@
 #include <iostream>
 #include <map>
 #include <memory>
-#include <regex>
+#include <boost/regex.hpp>
 #include <sstream>
 #include <string>
 #include <unordered_set>
@@ -854,10 +854,10 @@
     constexpr int OPTION_LONGEST = 30;
     constexpr int OPTION_DESC_GAP = 2;
 
-    std::basic_regex<char> option_matcher
+    boost::basic_regex<char> option_matcher
       ("--([[:alnum:]][-_[:alnum:]]+)(=(.*))?|-([[:alnum:]]+)");
 
-    std::basic_regex<char> option_specifier
+    boost::basic_regex<char> option_specifier
       ("(([[:alnum:]]),)?([[:alnum:]][-_[:alnum:]]+)");
 
     String
@@ -981,8 +981,8 @@
   std::string arg_help
 )
 {
-  std::match_results<const char*> result;
-  std::regex_match(opts.c_str(), result, option_specifier);
+  boost::match_results<const char*> result;
+  boost::regex_match(opts.c_str(), result, option_specifier);
 
   if (result.empty())
   {
@@ -1124,8 +1124,8 @@
       break;
     }
 
-    std::match_results<const char*> result;
-    std::regex_match(argv[current], result, option_matcher);
+    boost::match_results<const char*> result;
+    boost::regex_match(argv[current], result, option_matcher);
 
     if (result.empty())
     {
