From 3eb58af3579a17cce7b319b9baea7e98e7fc1d94 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Mon, 6 Aug 2018 22:50:06 +0200
Subject: [PATCH] fix build with gcc 8

---
 src/gui/entry/EditEntryWidget.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gui/entry/EditEntryWidget.cpp b/src/gui/entry/EditEntryWidget.cpp
index 6fd65c1..683ee77 100644
--- a/src/gui/entry/EditEntryWidget.cpp
+++ b/src/gui/entry/EditEntryWidget.cpp
@@ -32,6 +32,7 @@
 #include <QMimeData>
 #include <QEvent>
 #include <QColorDialog>
+#include <QButtonGroup>
 
 #include "autotype/AutoType.h"
 #include "core/Config.h"
-- 
2.18.0

