--- a/src/p11_front.c.orig	2017-08-05 14:17:47.255547585 +0200
+++ a/src/p11_front.c	2017-08-05 14:19:15.503800488 +0200
@@ -380,7 +380,7 @@
 
 void ERR_load_PKCS11_strings(void)
 {
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L  && !defined(LIBRESSL_VERSION_NUMBER)
 	OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CRYPTO_STRINGS, NULL);
 #endif
 	ERR_load_P11_strings();
--- a/src/eng_front.c.orig	2017-08-05 14:23:48.944584133 +0200
+++ 1/src/eng_front.c	2017-08-05 14:25:59.552960821 +0200
@@ -226,7 +226,7 @@
 #ifndef OPENSSL_NO_RSA
 			!ENGINE_set_RSA(e, PKCS11_get_rsa_method()) ||
 #endif
-#if OPENSSL_VERSION_NUMBER  >= 0x10100002L
+#if OPENSSL_VERSION_NUMBER  >= 0x10100002L   && !defined(LIBRESSL_VERSION_NUMBER)
 #ifndef OPENSSL_NO_EC
 			/* PKCS11_get_ec_key_method combines ECDSA and ECDH */
 			!ENGINE_set_EC(e, PKCS11_get_ec_key_method()) ||
