From 1ed0c0ad38c1308902c0d303971b0c1b08249db0 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 21:08:43 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 test-framework.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test-framework.cabal b/test-framework.cabal
index df3b918..618f6e8 100644
--- a/test-framework.cabal
+++ b/test-framework.cabal
@@ -48,7 +48,7 @@ Library
                               , ansi-terminal  >= 0.4.0  && < 0.9
                               , ansi-wl-pprint >= 0.5.1  && < 0.7
                               , random         >= 1.0    && < 1.2
-                              , containers     >= 0.1    && < 0.6
+                              , containers     >= 0.1
                               , regex-posix    >= 0.72   && < 0.96
                               , old-locale     >= 1.0    && < 1.1
                               , time           >= 1.1.2  && < 1.9
-- 
2.19.1

