From 28f7c93390c97a7af648d643fc96d344eeb0252a Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Mon, 7 Apr 2014 19:19:39 +0200
Subject: [PATCH] fix installation destination

---
 CMakeLists.txt     | 2 +-
 src/CMakeLists.txt | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a573528..5dea975 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ include_directories(include)
 find_package(OpenGL REQUIRED)
 
 SET(ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR})
-SET(INSTALL_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+SET(INSTALL_DIR ${CMAKE_INSTALL_PREFIX})
 SET(DEPENDENCY_DIR ${CMAKE_CURRENT_SOURCE_DIR}/dependencies)
 SET(GUI_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include/gui)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c882ded..14bb952 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -85,6 +85,6 @@ ENDIF()
 
 #${LIBTCOD_CPP_NAME} is empty on non-linux platforms
 INSTALL(TARGETS ${LIBTCOD_NAME} ${LIBTCOD_GUI_NAME} ${LIBTCOD_CPP_NAME}
-  RUNTIME DESTINATION ${INSTALL_DIR}
-  LIBRARY DESTINATION ${INSTALL_DIR}
+  RUNTIME DESTINATION ${INSTALL_DIR}/bin
+  LIBRARY DESTINATION ${INSTALL_DIR}/lib
 )
-- 
1.9.0

