--- libmpdclient-2.2/Makefile.am.orig	2010-09-05 22:34:23.000000000 +0200
+++ libmpdclient-2.2/Makefile.am	2010-09-05 22:34:14.000000000 +0200
@@ -141,7 +141,7 @@
 install-data-local: doc/api/html/index.html
 	$(mkinstalldirs) $(DESTDIR)$(docdir)/html
 	$(INSTALL_DATA) -c -m 644 doc/api/html/*.html doc/api/html/*.css \
-		doc/api/html/*.png doc/api/html/*.gif \
+		doc/api/html/*.png \
 		$(DESTDIR)$(docdir)/html
 
 uninstall-local:
