--- util/Makefile.in.orig	2012-07-19 16:53:28.000000000 +0400
+++ util/Makefile.in	2012-08-05 15:32:53.820922057 +0400
@@ -650,8 +650,6 @@
 install-exec-local:
 	$(MKDIR_P) $(DESTDIR)$(MOUNT_FUSE_PATH)
 	$(INSTALL_PROGRAM) $(builddir)/mount.fuse $(DESTDIR)$(MOUNT_FUSE_PATH)/mount.fuse
-	$(MKDIR_P) $(DESTDIR)$(INIT_D_PATH)
-	$(INSTALL_SCRIPT) $(srcdir)/init_script $(DESTDIR)$(INIT_D_PATH)/fuse
 	@if test -x /usr/sbin/update-rc.d; then \
 		echo "/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true"; \
 		/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true; \
