From 72441c32e66131d5a880f5a2804f2b3c075cd59d Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Fri, 26 Oct 2012 21:21:56 +0200
Subject: [PATCH] add missing header

---
 exrenvmap/blurImage.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/exrenvmap/blurImage.cpp b/exrenvmap/blurImage.cpp
index 04edb8d..9700914 100644
--- a/exrenvmap/blurImage.cpp
+++ b/exrenvmap/blurImage.cpp
@@ -45,6 +45,7 @@
 #include "Iex.h"
 #include <iostream>
 #include <algorithm>
+#include <string.h>
 
 using namespace std;
 using namespace Imf;
-- 
1.8.0

