Index: SConstruct
===================================================================
--- SConstruct	(Revision 1352)
+++ SConstruct	(Arbeitskopie)
@@ -535,6 +535,24 @@
 #
 env['top_srcdir'] = env.Dir( "." ).abspath
 
+# Be environmentally friendly...
+import os
+import SCons.Util
+
+if os.environ.has_key('CC'):
+        env['CC'] = os.environ['CC']
+if os.environ.has_key('CFLAGS'):
+        env['CCFLAGS'] += SCons.Util.CLVar(os.environ['CFLAGS'])
+if os.environ.has_key('CXX'):
+        env['CXX'] = os.environ['CXX']
+if os.environ.has_key('CXXFLAGS'):
+        env['CXXFLAGS'] += SCons.Util.CLVar(os.environ['CXXFLAGS'])
+if os.environ.has_key('LDFLAGS'):
+        env['LINKFLAGS'] += SCons.Util.CLVar(os.environ['LDFLAGS'])
+if os.environ.has_key('CPPFLAGS'):
+        env['CCFLAGS'] += SCons.Util.CLVar(os.environ['CPPFLAGS'])
+        env['CXXFLAGS'] += SCons.Util.CLVar(os.environ['CPPFLAGS'])
+
 #
 # Start building
 #
