Index: SConstruct
===================================================================
--- SConstruct	(Revision 1457)
+++ SConstruct	(Arbeitskopie)
@@ -290,7 +290,9 @@
 
 	# PyQT checks
 	build_mixer = False
-	if conf.CheckForApp( 'which pyuic4' ) and conf.CheckForPyModule( 'dbus' ) and conf.CheckForPyModule( 'PyQt4' ) and conf.CheckForPyModule( 'dbus.mainloop.qt' ):
+	print "NOT testing for dbus.mainloop.qt since there is some issue with the test, see http://subversion.ffado.org/ticket/171"
+	if conf.CheckForApp( 'which pyuic4' ) and conf.CheckForPyModule( 'dbus' ) and conf.CheckForPyModule( 'PyQt4' ):
+	# and conf.CheckForPyModule( 'dbus.mainloop.qt'):
 		env['PYUIC4'] = True
 		build_mixer = True
 	
