Addresses: https://github.com/hexchat/hexchat/issues/1990
--- src/fe-gtk/maingui.c.orig	2016-11-11 19:20:35.000000000 -0500
+++ src/fe-gtk/maingui.c	2017-04-08 22:52:19.421637506 -0400
@@ -1965,17 +1965,16 @@
 static GtkWidget *
 mg_create_flagbutton (char *tip, GtkWidget *box, char *face)
 {
-	GtkWidget *btn, *lbl;
-	char label_markup[16];
+	GtkWidget *btn, *lbl_tst;
+	GtkRequisition req;
 
-	g_snprintf (label_markup, sizeof(label_markup), "<tt>%s</tt>", face);
-	lbl = gtk_label_new (NULL);
-	gtk_label_set_markup (GTK_LABEL(lbl), label_markup);
+	lbl_tst = gtk_label_new("W");
+	gtk_widget_size_request(lbl_tst, &req);
+	gtk_widget_destroy (lbl_tst);
 
-	btn = gtk_toggle_button_new ();
-	gtk_widget_set_size_request (btn, -1, 0);
+	btn = gtk_toggle_button_new_with_label (face);
+	gtk_widget_set_size_request (btn, req.width + 10, 0);
 	gtk_widget_set_tooltip_text (btn, tip);
-	gtk_container_add (GTK_CONTAINER(btn), lbl);
 
 	gtk_box_pack_start (GTK_BOX (box), btn, 0, 0, 0);
 	g_signal_connect (G_OBJECT (btn), "toggled",
