diff --git a/include/libfq.h b/include/libfq.h
index 7e37e99..31a39ed 100644
--- a/include/libfq.h
+++ b/include/libfq.h
@@ -2,35 +2,12 @@
 #define LIBFQ_H
 
 #include <stdlib.h>
+#include <stdbool.h>
 #include <ibase.h>
 
 #ifndef C_H
 #define C_H
 
-
-#ifndef BOOL
-#define BOOL
-typedef char bool;
-
-
-#ifndef true
-#define true    ((bool) 1)
-#endif
-
-#ifndef false
-#define false   ((bool) 0)
-#endif
-
-typedef bool *BoolPtr;
-
-#ifndef TRUE
-#define TRUE    1
-#endif
-
-#ifndef FALSE
-#define FALSE   0
-#endif
-#endif
 /*
  * lengthof
  *              Number of elements in an array.
