From 8a8ec571f453d586daf9e836be55699b77a706bd Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sun, 20 Mar 2016 11:51:44 +0100
Subject: [PATCH] fix build error against ffmpeg 3

---
 examples/alffplay.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/examples/alffplay.c b/examples/alffplay.c
index ab67d21..a6ae725 100644
--- a/examples/alffplay.c
+++ b/examples/alffplay.c
@@ -942,7 +942,7 @@ static void update_picture(MovieState *movState, bool *first_update, SDL_Window
         void *pixels = NULL;
         int pitch = 0;
 
-        if(movState->video.st->codec->pix_fmt == PIX_FMT_YUV420P)
+        if(movState->video.st->codec->pix_fmt == AV_PIX_FMT_YUV420P)
             SDL_UpdateYUVTexture(vp->bmp, NULL,
                 frame->data[0], frame->linesize[0],
                 frame->data[1], frame->linesize[1],
@@ -960,7 +960,7 @@ static void update_picture(MovieState *movState, bool *first_update, SDL_Window
             if(!movState->video.swscale_ctx)
                 movState->video.swscale_ctx = sws_getContext(
                     w, h, movState->video.st->codec->pix_fmt,
-                    w, h, PIX_FMT_YUV420P, SWS_X, NULL, NULL, NULL
+                    w, h, AV_PIX_FMT_YUV420P, SWS_X, NULL, NULL, NULL
                 );
 
             /* point pict at the queue */
-- 
2.7.4

