--- openal-soft-1.14/CMakeLists.txt.orig	2012-04-25 21:19:57.838343605 +0200
+++ openal-soft-1.14/CMakeLists.txt	2012-04-25 21:20:08.572209185 +0200
@@ -762,7 +762,7 @@
 IF(EXAMPLES)
     IF(FFMPEG_FOUND)
         ADD_EXECUTABLE(alstream examples/alhelpers.c examples/alffmpeg.c examples/alstream.c)
-        TARGET_LINK_LIBRARIES(alstream ${FFMPEG_LIBRARIES} ${LIBNAME})
+        TARGET_LINK_LIBRARIES(alstream ${FFMPEG_LIBRARIES} ${LIBNAME} -lavutil)
         SET_TARGET_PROPERTIES(alstream PROPERTIES COMPILE_FLAGS "${FFMPEG_CFLAGS}")
         INSTALL(TARGETS alstream
                 RUNTIME DESTINATION bin
