Index: src/libassuan.m4
===================================================================
--- src/libassuan.m4    (revision 218)
+++ src/libassuan.m4    (working copy)
@@ -125,8 +125,8 @@
 AC_DEFUN([AM_PATH_LIBASSUAN_PTH],
 [ _AM_PATH_LIBASSUAN_COMMON($1,pth)
   if test $ok = yes; then
-    LIBASSUAN_PTH_CFLAGS=`$LIBASSUAN_CONFIG $libassuan_config_args --cflags`
-    LIBASSUAN_PTH_LIBS=`$LIBASSUAN_CONFIG $libassuan_config_args --libs`
+    LIBASSUAN_PTH_CFLAGS=`$LIBASSUAN_CONFIG $libassuan_config_args --thread=pth --cflags`
+    LIBASSUAN_PTH_LIBS=`$LIBASSUAN_CONFIG $libassuan_config_args --thread=pth --libs`
     ifelse([$2], , :, [$2])
   else
     LIBASSUAN_PTH_CFLAGS=""
@@ -146,8 +146,8 @@
 AC_DEFUN([AM_PATH_LIBASSUAN_PTHREAD],
 [ _AM_PATH_LIBASSUAN_COMMON($1,pth)
   if test $ok = yes; then
-    LIBASSUAN_PTHREAD_CFLAGS=`$LIBASSUAN_CONFIG $libassuan_config_args --cflags`
-    LIBASSUAN_PTHREAD_LIBS=`$LIBASSUAN_CONFIG $libassuan_config_args --libs`
+    LIBASSUAN_PTHREAD_CFLAGS=`$LIBASSUAN_CONFIG $libassuan_config_args --thread=pthread --cflags`
+    LIBASSUAN_PTHREAD_LIBS=`$LIBASSUAN_CONFIG $libassuan_config_args --thread=pthread --libs`
     ifelse([$2], , :, [$2])
   else
     LIBASSUAN_PTHREAD_CFLAGS=""
Index: configure.ac
