diff -urN ppp-2.4.3.org/pppd/Makefile.linux ppp-2.4.3/pppd/Makefile.linux
--- ppp-2.4.3.org/pppd/Makefile.linux	2004-11-13 12:02:22.000000000 +0000
+++ ppp-2.4.3/pppd/Makefile.linux	2005-04-22 16:52:01.000000000 +0000
@@ -123,6 +123,9 @@
 ifneq ($(wildcard /usr/lib/libcrypt.*),)
 LIBS	+= -lcrypt
 endif
+ifneq ($(wildcard /usr/lib64/libcrypt.*),)
+LIBS	+= -lcrypt
+endif
 
 ifdef NEEDDES
 ifndef USE_CRYPT
