--- 0.8.7e/sql.c	2009-06-28 21:45:02.000000000 -0400
+++ ../branches/0.8.7/sql.c	2009-08-18 20:51:30.000000000 -0400
@@ -186,6 +186,12 @@
 		die("FATAL: MySQL options unable to set timeout value");
 	}
 
+	my_bool reconnect = 1;
+	options_error = mysql_options(mysql, MYSQL_OPT_RECONNECT, &reconnect);
+	if (options_error < 0) {
+		die("FATAL: MySQL options unable to set reconnect option\n");
+	}
+
 	while (tries > 0) {
 		tries--;
 
