diff -Naur libICE-1.0.9.orig/src/ICElibint.h libICE-1.0.9/src/ICElibint.h
--- libICE-1.0.9.orig/src/ICElibint.h	2014-06-07 10:28:43.000000000 +0900
+++ libICE-1.0.9/src/ICElibint.h	2015-05-24 18:18:02.000000000 +0900
@@ -334,28 +334,28 @@
  * Extern declarations
  */
 
-extern IceConn     	_IceConnectionObjs[];
-extern char	    	*_IceConnectionStrings[];
-extern int     		_IceConnectionCount;
+extern IceConn     		_IceConnectionObjs[];
+extern char	    		*_IceConnectionStrings[];
+extern int     			_IceConnectionCount;
 
-extern _IceProtocol	_IceProtocols[];
-extern int         	_IceLastMajorOpcode;
+extern _IceProtocol		_IceProtocols[];
+extern int         		_IceLastMajorOpcode;
 
-extern int		_IceAuthCount;
-extern const char	*_IceAuthNames[];
-extern IcePoAuthProc	_IcePoAuthProcs[];
-extern IcePaAuthProc	_IcePaAuthProcs[];
+extern int			_IceAuthCount;
+extern const char		*_IceAuthNames[];
+extern IcePoAuthProc		_IcePoAuthProcs[];
+extern IcePaAuthProc		_IcePaAuthProcs[];
 
-extern int		_IceVersionCount;
-extern _IceVersion	_IceVersions[];
+extern const int		_IceVersionCount;
+extern const _IceVersion	_IceVersions[];
 
-extern _IceWatchProc	*_IceWatchProcs;
+extern _IceWatchProc		*_IceWatchProcs;
 
-extern IceErrorHandler   _IceErrorHandler;
-extern IceIOErrorHandler _IceIOErrorHandler;
+extern IceErrorHandler		_IceErrorHandler;
+extern IceIOErrorHandler	_IceIOErrorHandler;
 
-extern IceAuthDataEntry	 _IcePaAuthDataEntries[];
-extern int		 _IcePaAuthDataEntryCount;
+extern IceAuthDataEntry		_IcePaAuthDataEntries[];
+extern int			_IcePaAuthDataEntryCount;
 
 extern void _IceErrorBadMajor (
     IceConn		/* iceConn */,
diff -Naur libICE-1.0.9.orig/src/process.c libICE-1.0.9/src/process.c
--- libICE-1.0.9.orig/src/process.c	2014-06-07 10:28:43.000000000 +0900
+++ libICE-1.0.9/src/process.c	2015-05-24 18:20:23.000000000 +0900
@@ -856,7 +856,8 @@
 )
 {
     iceConnectionSetupMsg *message;
-    int  myVersionCount, hisVersionCount;
+    const int myVersionCount = _IceVersionCount;
+    int  hisVersionCount;
     int	 myVersionIndex, hisVersionIndex;
     int  hisMajorVersion, hisMinorVersion;
     int	 myAuthCount, hisAuthCount;
@@ -926,7 +927,6 @@
     }
 
     hisVersionCount = message->versionCount;
-    myVersionCount = _IceVersionCount;
 
     hisVersionIndex = myVersionIndex = found = 0;
 
@@ -2551,7 +2551,7 @@
 	*replyReadyRet = replyReady;
 }
 
-int		_IceVersionCount = 1;
-_IceVersion	_IceVersions[] = {
-		    {IceProtoMajor, IceProtoMinor, _IceProcessCoreMessage}};
+const int		_IceVersionCount = 1;
+const _IceVersion	_IceVersions[] = {
+			    {IceProtoMajor, IceProtoMinor, _IceProcessCoreMessage}};
 
