--- install.in	2005-06-23 04:06:12.000000000 +0000
+++ install.in.cp	2005-11-08 14:08:59.000000000 +0000
@@ -120,8 +120,6 @@
 
     # Add group and user and setup permissions
     if [ "$UID" == "0" ]; then
-	groupadd quasar >/dev/null 2>/dev/null
-	useradd -g quasar -d $TOP_DIR -s /bin/bash -c "Quasar" quasar 2>/dev/null
 	chgrp -R quasar $CONFIG_DIR
 	chgrp -R quasar $DATA_DIR
 
@@ -132,29 +130,6 @@
 	chgrp quasar /var/log/quasar
     fi
 
-    # Add inetd or xinetd entry for server
-    if [ -e /etc/init.d/xinetd ]; then
-	$INSTALL -m 0664 -D setup/quasar.xinetd $PREFIX/etc/xinetd.d/quasar
-	if [ "$UID" == "0" ]; then
-	    /etc/init.d/xinetd restart
-	    if [ -x /sbin/chkconfig ]; then
-		/sbin/chkconfig xinetd on
-	    elif [ -x /sbin/rc-update ]; then
-		/sbin/rc-update add xinetd default
-	    fi
-	fi
-    elif [ -e /etc/inetd.conf ]; then
-	if [ ! $(grep -q quasar /etc/services) ]; then
-	    echo -e "quasar\t\t3599/tcp" >>/etc/services
-	fi
-	if [ ! $(grep -q quasar /etc/inetd.conf) ]; then
-	    echo -e "quasar\t\tstream\ttcp\tnowait\tquasar.quasar\t/opt/quasar/bin/quasar_clientd" >>/etc/inetd.conf
-	    /etc/init.d/inetd restart
-	fi
-    else
-	echo "ERROR: can't install server process using xinetd or inetd"
-	echo "       you will need to setup quasard to run at boot time"
-    fi
 }
 
 function install_client ()
@@ -188,8 +163,6 @@
 
     # Add group and user
     if [ "$UID" == "0" ]; then
-	groupadd quasar >/dev/null 2>/dev/null
-	useradd -g quasar -d $TOP_DIR -s /bin/bash -c "Quasar" quasar 2>/dev/null
 	chgrp -R quasar $CONFIG_DIR
     fi
 
