diff --git a/GUI/xephem/versionmenu.c b/GUI/xephem/versionmenu.c
index 6c062b1a69c7..a5b3b465d2cf 100644
--- a/GUI/xephem/versionmenu.c
+++ b/GUI/xephem/versionmenu.c
@@ -50,7 +50,7 @@ static char *msg[] = {
 "Copyright (c) 2021 Elwood Charles Downey",
 "",
 "Permission is hereby granted, free of charge, to any person obtaining a copy",
-"of this software and associated documentation files (the "Software"), to deal",
+"of this software and associated documentation files (the \"Software\"), to deal",
 "in the Software without restriction, including without limitation the rights",
 "to use, copy, modify, merge, publish, distribute, sublicense, and/or sell",
 "copies of the Software, and to permit persons to whom the Software is",
@@ -59,7 +59,7 @@ static char *msg[] = {
 "The above copyright notice and this permission notice shall be included in all",
 "copies or substantial portions of the Software.",
 "",
-"THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR",
+"THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR",
 "IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,",
 "FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE",
 "AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER",
