diff -uNr qca-2.2.1/plugins/qca-ossl/ossl110-compat.h qca-2.2.1.fix/plugins/qca-ossl/ossl110-compat.h
--- qca-2.2.1/plugins/qca-ossl/ossl110-compat.h	2019-04-24 14:58:14.000000000 +0200
+++ qca-2.2.1.fix/plugins/qca-ossl/ossl110-compat.h	2019-04-26 18:43:23.371501310 +0200
@@ -25,7 +25,7 @@
 #include <openssl/rsa.h>
 #include <openssl/dsa.h>
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 #define RSA_F_RSA_METH_DUP 161
 
 static void DSA_SIG_get0(const DSA_SIG *sig, const BIGNUM **pr, const BIGNUM **ps)
diff -uNr qca-2.2.1/plugins/qca-ossl/qca-ossl.cpp qca-2.2.1.fix/plugins/qca-ossl/qca-ossl.cpp
--- qca-2.2.1/plugins/qca-ossl/qca-ossl.cpp	2019-04-24 14:58:14.000000000 +0200
+++ qca-2.2.1.fix/plugins/qca-ossl/qca-ossl.cpp	2019-04-26 18:42:11.179426042 +0200
@@ -57,7 +57,7 @@
 	((_STACK*) (1 ? p : (type*)0))
 #endif
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && ! defined(LIBRESSL_VERSION_NUMBER)
     #define OSSL_110
 #endif
 
