From 47520a991d727f0ecf66d8f9cec65883e0bf4856 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 18 Jan 2020 23:40:13 +0100
Subject: [PATCH] install to /usr

---
 src/Makefile | 30 +++++++++++++++---------------
 src/src.pro  |  2 +-
 2 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index b70edb4..f93aefa 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -285,29 +285,29 @@ RenderTexture.o: ../RenderTexture/RenderTexture.cpp ../RenderTexture/RenderTextu
 ####### Install
 
 install_target: first FORCE
-	@$(CHK_DIR_EXISTS) $(INSTALL_ROOT)/usr/local/lib/ || $(MKDIR) $(INSTALL_ROOT)/usr/local/lib/ 
-	-$(INSTALL_PROGRAM) "../lib/$(TARGET)" "$(INSTALL_ROOT)/usr/local/lib/$(TARGET)"
-	-$(STRIP) --strip-unneeded "$(INSTALL_ROOT)/usr/local/lib/$(TARGET)"
-	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/usr/local/lib/$(TARGET0)"
-	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/usr/local/lib/$(TARGET1)"
-	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/usr/local/lib/$(TARGET2)"
+	@$(CHK_DIR_EXISTS) $(INSTALL_ROOT)/usr/lib/ || $(MKDIR) $(INSTALL_ROOT)/usr/lib/ 
+	-$(INSTALL_PROGRAM) "../lib/$(TARGET)" "$(INSTALL_ROOT)/usr/lib/$(TARGET)"
+	-$(STRIP) --strip-unneeded "$(INSTALL_ROOT)/usr/lib/$(TARGET)"
+	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/usr/lib/$(TARGET0)"
+	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/usr/lib/$(TARGET1)"
+	-$(SYMLINK) "$(TARGET)" "$(INSTALL_ROOT)/usr/lib/$(TARGET2)"
 
 uninstall_target:  FORCE
-	-$(DEL_FILE) "$(INSTALL_ROOT)/usr/local/lib/$(TARGET)" 
-	 -$(DEL_FILE) "$(INSTALL_ROOT)/usr/local/lib/$(TARGET0)" 
-	 -$(DEL_FILE) "$(INSTALL_ROOT)/usr/local/lib/$(TARGET1)" 
-	 -$(DEL_FILE) "$(INSTALL_ROOT)/usr/local/lib/$(TARGET2)"
-	-$(DEL_DIR) $(INSTALL_ROOT)/usr/local/lib/ 
+	-$(DEL_FILE) "$(INSTALL_ROOT)/usr/lib/$(TARGET)" 
+	 -$(DEL_FILE) "$(INSTALL_ROOT)/usr/lib/$(TARGET0)" 
+	 -$(DEL_FILE) "$(INSTALL_ROOT)/usr/lib/$(TARGET1)" 
+	 -$(DEL_FILE) "$(INSTALL_ROOT)/usr/lib/$(TARGET2)"
+	-$(DEL_DIR) $(INSTALL_ROOT)/usr/lib/ 
 
 
 install_headers: first FORCE
-	@$(CHK_DIR_EXISTS) $(INSTALL_ROOT)/usr/local/include/ || $(MKDIR) $(INSTALL_ROOT)/usr/local/include/ 
-	-$(INSTALL_FILE) ../include/opencsg.h $(INSTALL_ROOT)/usr/local/include/
+	@$(CHK_DIR_EXISTS) $(INSTALL_ROOT)/usr/include/ || $(MKDIR) $(INSTALL_ROOT)/usr/include/ 
+	-$(INSTALL_FILE) ../include/opencsg.h $(INSTALL_ROOT)/usr/include/
 
 
 uninstall_headers:  FORCE
-	-$(DEL_FILE) -r $(INSTALL_ROOT)/usr/local/include/opencsg.h
-	-$(DEL_DIR) $(INSTALL_ROOT)/usr/local/include/ 
+	-$(DEL_FILE) -r $(INSTALL_ROOT)/usr/include/opencsg.h
+	-$(DEL_DIR) $(INSTALL_ROOT)/usr/include/ 
 
 
 install:  install_target install_headers  FORCE
diff --git a/src/src.pro b/src/src.pro
index a40b626..d654487 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -3,7 +3,7 @@ TARGET = opencsg
 VERSION = 1.4.2
 
 isEmpty(INSTALLDIR) {
-  INSTALLDIR = /usr/local
+  INSTALLDIR = /usr
 }
 
 CONFIG += opengl warn_on release
-- 
2.25.0

