From 84ac6804a8a695514dae3f62903f95c98cc23d57 Mon Sep 17 00:00:00 2001
From: Dongxiao Xu <dongxiao.xu@intel.com>
Date: Wed, 29 Jun 2011 10:44:00 +0000
Subject: Fix the getter for the path property in GypsyTime

---
(limited to 'gypsy/gypsy-time.c')

diff --git a/gypsy/gypsy-time.c b/gypsy/gypsy-time.c
index a7ce33d..9b4e101 100644
--- a/gypsy/gypsy-time.c
+++ b/gypsy/gypsy-time.c
@@ -160,6 +160,7 @@ get_property (GObject    *object,
 	priv = GET_PRIVATE (object);
 	switch (prop_id) {
 	case PROP_PATH:
+		g_value_set_string (value, priv->object_path);
 		break;
 
 	default:
--
cgit v0.9.0.2-2-gbebe
