*** stuff.h.old	2003-12-09 10:18:28.000000000 +0100
--- stuff.h	2003-12-09 10:18:50.000000000 +0100
***************
*** 1,4 ****
! #include <map>
! #include <vector>
  #include <fftw.h>
  #include <rfftw.h>
--- 1,4 ----
! #include <map.h>
! #include <vector.h>
  #include <fftw.h>
  #include <rfftw.h>
*** gtkplayer.h.old	2003-12-09 10:20:25.000000000 +0100
--- gtkplayer.h	2003-12-09 10:21:42.000000000 +0100
***************
*** 1,3 ****
! #include <set>
  
  #include <gtk--/main.h>
--- 1,3 ----
! #include <set.h>
  
  #include <gtk--/main.h>
***************
*** 15,19 ****
  #include "vis.h"
  
! #include <set>
  class GUIPlayer:public Player, public PlayParm, public Gtk::Main {
   public:
--- 15,19 ----
  #include "vis.h"
  
! #include <set.h>
  class GUIPlayer:public Player, public PlayParm, public Gtk::Main {
   public:
*** soundcard.cc.old	2003-12-09 10:33:04.000000000 +0100
--- soundcard.cc	2003-12-09 10:34:05.000000000 +0100
***************
*** 22,25 ****
--- 22,27 ----
  #include <errno.h>
  #include <assert.h>
+ #include <stdlib.h>
+ 
  
  #include <stdio.h>
