From ba7544e08b378ad014cfafc0745a0f97478562e9 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 22 May 2021 19:29:09 +0200
Subject: [PATCH] link against zlib if libxlsxwriter is detected

---
 src/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index e9a1454..5498911 100755
--- a/src/Makefile
+++ b/src/Makefile
@@ -129,8 +129,8 @@ ifneq (, $(shell which pkg-config))
 
   # NOTE: libxlsxwriter is required for xlsx file export support
   ifneq ($(shell pkg-config --exists xlsxwriter || echo 'no'),no)
-    CFLAGS += -DXLSX_EXPORT $(shell pkg-config --cflags xlsxwriter)
-    LDLIBS += $(shell pkg-config --libs xlsxwriter)
+    CFLAGS += -DXLSX_EXPORT $(shell pkg-config --cflags xlsxwriter) $(shell pkg-config --cflags zlib)
+    LDLIBS += $(shell pkg-config --libs xlsxwriter) $(shell pkg-config --libs zlib)
   endif
 
   # NOTE: lua support
-- 
2.31.1

