From affc72b043d5056c95721bec485dc1a1790ee536 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 16 Dec 2018 19:47:36 +0100
Subject: [PATCH] link against zlib if libxlsxwriter is detected

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index dd49467..518ed1f 100755
--- a/src/Makefile
+++ b/src/Makefile
@@ -75,7 +75,7 @@ endif
 # NOTE: libxlsxwriter is required for xlsx file export support
 ifneq (,$(wildcard /usr/include/xlsxwriter.h))
   CFLAGS += -DXLSX_EXPORT
-  LDLIBS += -lxlsxwriter
+  LDLIBS += -lxlsxwriter -lz
 endif
 ifneq (,$(wildcard /usr/local/include/xlsxwriter.h))
   CFLAGS += -DXLSX_EXPORT
-- 
2.20.1

