From 5d07b7fcd93e0684702484283f897664211b30f5 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Wed, 20 Mar 2013 18:20:34 +0100
Subject: [PATCH] add missing include

---
 ddxli.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ddxli.h b/ddxli.h
index 9122aa7..1dd7837 100644
--- a/ddxli.h
+++ b/ddxli.h
@@ -13,6 +13,7 @@
 #include <sys/ipc.h>
 #include <sys/shm.h>
 #include <X11/extensions/XShm.h>
+#include <string.h>
 
 #if defined(SYSV) || defined(VMS)
 #include <string.h>
-- 
1.8.1.3

