--- setup.py~	2005-03-09 23:27:24.000000000 +0100
+++ setup.py	2005-12-11 12:45:44.000000000 +0100
@@ -489,7 +489,7 @@
         #
         # http://www.sleepycat.com/update/index.html
 
-        max_db_ver = (4, 3)
+        max_db_ver = (4, 4)
         min_db_ver = (3, 2)
         db_setup_debug = False   # verbose debug prints from this script?
 
--- Modules/_bsddb.c~	2004-12-16 10:48:37.000000000 +0100
+++ Modules/_bsddb.c	2005-12-11 12:54:19.000000000 +0100
@@ -4101,8 +4101,10 @@
 #endif
     MAKE_ENTRY(nrequests);
     MAKE_ENTRY(nreleases);
+#if (DBVER < 44)
     MAKE_ENTRY(nnowaits);
     MAKE_ENTRY(nconflicts);
+#endif
     MAKE_ENTRY(ndeadlocks);
     MAKE_ENTRY(regsize);
     MAKE_ENTRY(region_wait);
@@ -4938,7 +4940,9 @@
     ADD_INT(d, DB_LOCK_IREAD);
     ADD_INT(d, DB_LOCK_IWR);
 #if (DBVER >= 33)
+#if (DBVER < 44)
     ADD_INT(d, DB_LOCK_DIRTY);
+#endif
     ADD_INT(d, DB_LOCK_WWRITE);
 #endif
 
