$OpenBSD: patch-Lib_ssl_py,v 1.2 2014/12/14 08:32:55 rpointel Exp $
--- Lib/ssl.py.orig	Wed Dec 10 16:59:40 2014
+++ Lib/ssl.py	Sat Dec 13 12:15:08 2014
@@ -106,7 +106,7 @@ from _ssl import CERT_NONE, CERT_OPTIONAL, CERT_REQUIR
 from _ssl import (VERIFY_DEFAULT, VERIFY_CRL_CHECK_LEAF, VERIFY_CRL_CHECK_CHAIN,
     VERIFY_X509_STRICT)
 from _ssl import txt2obj as _txt2obj, nid2obj as _nid2obj
-from _ssl import RAND_status, RAND_egd, RAND_add
+from _ssl import RAND_status, RAND_add
 
 def _import_symbols(prefix):
     for n in dir(_ssl):
$OpenBSD: patch-Modules__ssl_c,v 1.3 2014/12/14 08:32:55 rpointel Exp $
--- Modules/_ssl.c.orig	Wed Dec 10 16:59:53 2014
+++ Modules/_ssl.c	Sat Dec 13 12:16:17 2014
@@ -3301,32 +3301,6 @@ Returns 1 if the OpenSSL PRNG has been seeded with eno
 It is necessary to seed the PRNG with RAND_add() on some platforms before\n\
 using the ssl() function.");
 
-static PyObject *
-PySSL_RAND_egd(PyObject *self, PyObject *arg)
-{
-    int bytes;
-
-    if (!PyString_Check(arg))
-        return PyErr_Format(PyExc_TypeError,
-                            "RAND_egd() expected string, found %s",
-                            Py_TYPE(arg)->tp_name);
-    bytes = RAND_egd(PyString_AS_STRING(arg));
-    if (bytes == -1) {
-        PyErr_SetString(PySSLErrorObject,
-                        "EGD connection failed or EGD did not return "
-                        "enough data to seed the PRNG");
-        return NULL;
-    }
-    return PyInt_FromLong(bytes);
-}
-
-PyDoc_STRVAR(PySSL_RAND_egd_doc,
-"RAND_egd(path) -> bytes\n\
-\n\
-Queries the entropy gather daemon (EGD) on the socket named by 'path'.\n\
-Returns number of bytes read.  Raises SSLError if connection to EGD\n\
-fails or if it does not provide enough data to seed PRNG.");
-
 #endif /* HAVE_OPENSSL_RAND */
 
 
@@ -3720,8 +3694,6 @@ static PyMethodDef PySSL_methods[] = {
 #ifdef HAVE_OPENSSL_RAND
     {"RAND_add",            PySSL_RAND_add, METH_VARARGS,
      PySSL_RAND_add_doc},
-    {"RAND_egd",            PySSL_RAND_egd, METH_VARARGS,
-     PySSL_RAND_egd_doc},
     {"RAND_status",         (PyCFunction)PySSL_RAND_status, METH_NOARGS,
      PySSL_RAND_status_doc},
 #endif
$OpenBSD: patch-Lib_socket_py,v 1.1 2014/04/19 14:17:47 espie Exp $
--- Lib/socket.py.orig	Sat Apr 19 16:07:48 2014
+++ Lib/socket.py	Sat Apr 19 16:07:54 2014
@@ -67,7 +67,6 @@ else:
     from _ssl import SSLError as sslerror
     from _ssl import \
          RAND_add, \
-         RAND_egd, \
          RAND_status, \
          SSL_ERROR_ZERO_RETURN, \
          SSL_ERROR_WANT_READ, \
$OpenBSD: patch-Lib_test_test_ssl_py,v 1.2 2014/12/14 08:32:55 rpointel Exp $

No RAND_egd on OpenBSD.

--- Lib/test/test_ssl.py.orig	Wed Dec 10 16:59:47 2014
+++ Lib/test/test_ssl.py	Sat Dec 13 12:13:40 2014
@@ -169,8 +169,6 @@ class BasicSocketTests(unittest.TestCase):
             sys.stdout.write("\n RAND_status is %d (%s)\n"
                              % (v, (v and "sufficient randomness") or
                                 "insufficient randomness"))
-        self.assertRaises(TypeError, ssl.RAND_egd, 1)
-        self.assertRaises(TypeError, ssl.RAND_egd, 'foo', 1)
         ssl.RAND_add("this is a random string", 75.0)
 
     def test_parse_cert(self):
