From 5ca8302ade381f446a0f7c384918553460ad3b17 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 10 May 2020 14:48:04 +0200
Subject: [PATCH] Remove dead code that prevents successful configuration

---
 cmake/versions.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/versions.cmake b/cmake/versions.cmake
index 4c6c27f..dbd27d9 100644
--- a/cmake/versions.cmake
+++ b/cmake/versions.cmake
@@ -3,7 +3,6 @@ SET(VERSION "2.0")
 STRING(REGEX MATCHALL "[0-9]" VERSION_PARTS "${VERSION}")
 LIST(GET VERSION_PARTS 0 VERSION_MAJOR)
 LIST(GET VERSION_PARTS 1 VERSION_MINOR)
-LIST(GET VERSION_PARTS 2 VERSION_PATCH)
 SET(SOVERSION "${VERSION_MAJOR}.${VERSION_MINOR}")
 
 # Postfix of so's:
-- 
2.26.2

