diff -Naur a/src/modules/module_main.c b/src/modules/module_main.c
--- a/src/modules/module_main.c	2008-06-27 09:29:26.000000000 -0300
+++ b/src/modules/module_main.c	2010-03-28 10:41:00.504821585 -0300
@@ -21,9 +21,6 @@
  * $Id: module_main.c,v 1.16 2008-06-27 12:29:26 hanke Exp $
  */
 
-/* So that gcc doesn't comply */
-int getline(char**, size_t*, FILE*);
-
 #define PROCESS_CMD(command, function) \
 if (!strcmp(cmd_buf, #command"\n")){ \
  char *msg; \
diff -Naur a/src/modules/module_utils.h b/src/modules/module_utils.h
--- a/src/modules/module_utils.h	2008-07-30 05:53:07.000000000 -0300
+++ b/src/modules/module_utils.h	2010-03-28 10:40:37.127814226 -0300
@@ -419,10 +419,6 @@
 void module_report_event_pause(void);
 
 
-
-/* So that gcc doesn't complain */
-int getline(char**, size_t*, FILE*);
-
 pthread_mutex_t module_stdout_mutex;
 
 int module_utils_init(void);
