--- lxdvdrip/Makefile.orig	2010-08-17 15:22:25.000000000 +0200
+++ lxdvdrip/Makefile	2010-08-17 15:24:55.000000000 +0200
@@ -4,8 +4,8 @@
 HINWEIS2 = 'Comparez le fichier de config. en /etc avec le Changelog'
 
 all:
-	gcc -g -lm -ldvdread -lpthread -o lxdvdrip lxdvdrip.c streamanalyze.c ifo.c dvdinfo.c dvdbackup.c dvdcell.c systools.c vaporize.c dvdtools.c dvdcopy.c requant.c cputest.c tcmemcpy.c dvdformat.c badsect.c mpeg2dec.c
-	gcc -g -lm -o lxac3scan lxac3scan.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -lm -ldvdread -lpthread -o lxdvdrip lxdvdrip.c streamanalyze.c ifo.c dvdinfo.c dvdbackup.c dvdcell.c systools.c vaporize.c dvdtools.c dvdcopy.c requant.c cputest.c tcmemcpy.c dvdformat.c badsect.c mpeg2dec.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -lm -o lxac3scan lxac3scan.c
 	cd vamps && make && cd ..
 	cd dvdbackup && make && cd ..
 	cd requant && make && cd ..
@@ -21,8 +21,9 @@
 install:
 	cp lxdvdrip $(INSTALLDIR)/bin
 	cp lxac3scan $(INSTALLDIR)/bin
-	cp lxdvdrip.1 $(INSTALLDIR)/man/man1
-	cp lxdvdrip.wav $(INSTALLDIR)/share
+	cp lxdvdrip.1 $(INSTALLDIR)/share/man/man1
+	mkdir $(INSTALLDIR)/share/lxdvdrip
+	cp lxdvdrip.wav $(INSTALLDIR)/share/lxdvdrip
 	cd vamps && make install && cd ..
 	cd dvdbackup && make install && cd ..
 	cd requant && make install && cd ..
