From 470b88a364febcbb36b1bd5f701e023d623a8576 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sat, 15 Mar 2014 18:14:29 +0100
Subject: [PATCH] add missing includes

---
 processwrapper.cc                       | 2 ++
 qtsingleapplication/src/qtlocalpeer.cpp | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/processwrapper.cc b/processwrapper.cc
index f7f3f19..d6c7b09 100644
--- a/processwrapper.cc
+++ b/processwrapper.cc
@@ -93,6 +93,8 @@ unsigned int ProcessWrapper::findProcess(const char *name, unsigned int pid_skip
 }
 
 #else
+#include <sys/types.h>
+#include <unistd.h>
 
 unsigned int ProcessWrapper::currentProcessId()
 {
diff --git a/qtsingleapplication/src/qtlocalpeer.cpp b/qtsingleapplication/src/qtlocalpeer.cpp
index 382d182..0247eaf 100644
--- a/qtsingleapplication/src/qtlocalpeer.cpp
+++ b/qtsingleapplication/src/qtlocalpeer.cpp
@@ -50,6 +50,8 @@ static PProcessIdToSessionId pProcessIdToSessionId = 0;
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include <sys/types.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
-- 
1.9.0

