diff -NrU0 libmad-0.15.1b/configure.ac libmad-0.15.1b.patched/configure.ac
--- libmad-0.15.1b/configure.ac	2004-01-23 10:41:32.000000000 +0100
+++ libmad-0.15.1b.patched/configure.ac	2008-07-31 16:05:42.000000000 +0200
@@ -432 +432 @@
-	libmad.list])
+	libmad.list mad.pc])
diff -NrU0 libmad-0.15.1b/mad.pc.in libmad-0.15.1b.patched/mad.pc.in
--- libmad-0.15.1b/mad.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ libmad-0.15.1b.patched/mad.pc.in	2008-07-31 16:04:05.000000000 +0200
@@ -0,0 +1,14 @@
+# libmad pkg-config source file
+
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: mad
+Description: MPEG Audio Decoder
+Version: @VERSION@
+Requires:
+Conflicts:
+Libs: -L${libdir} -lmad -lm
+Cflags: -I${includedir}
diff -NrU0 libmad-0.15.1b/Makefile.am libmad-0.15.1b.patched/Makefile.am
--- libmad-0.15.1b/Makefile.am	2004-02-17 03:02:03.000000000 +0100
+++ libmad-0.15.1b.patched/Makefile.am	2008-07-31 16:05:25.000000000 +0200
@@ -26,0 +27,3 @@
+pkgconfigdir =         $(libdir)/pkgconfig
+pkgconfig_DATA =       mad.pc
+
@@ -37 +40,2 @@
-			CHANGES COPYRIGHT CREDITS README TODO VERSION
+			CHANGES COPYRIGHT CREDITS README TODO VERSION \
+			mad.pc.in
