Author: Daniel Baumann <mail@daniel-baumann.ch>
Description: Use embedded copy of gnu-efi.

diff -Naurp syslinux.orig/efi/clean-gnu-efi.sh syslinux/efi/clean-gnu-efi.sh
--- syslinux.orig/efi/clean-gnu-efi.sh
+++ syslinux/efi/clean-gnu-efi.sh
@@ -21,11 +21,6 @@ fi
 ARCH="$1"
 objdir=$(readlink -f "$2")
 
-(
-	cd ../..
-	git submodule update --init
-)
-
 if [ -d "$objdir/gnu-efi" ];then
 	cd "$objdir/gnu-efi"
 	EFIDIR="$(readlink -f "$objdir/../gnu-efi/gnu-efi-3.0")"
