From 0cf4d38609d0faab7e6bc74f4653bf7b1765faa0 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 4 Oct 2019 22:32:46 +0200
Subject: [PATCH] add missing header

---
 efi/wrapper.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/efi/wrapper.c b/efi/wrapper.c
index 73f191f..d8bdba5 100644
--- a/efi/wrapper.c
+++ b/efi/wrapper.c
@@ -17,6 +17,7 @@
 #include <linux/elf.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-- 
2.23.0

