diff -ur cpu-1.4.3/src/plugins/ldap/commandline.c cpu-1.4.3-mychanges/src/plugins/ldap/commandline.c
--- cpu-1.4.3/src/plugins/ldap/commandline.c	2004-01-11 22:31:02.000000000 -0700
+++ cpu-1.4.3-mychanges/src/plugins/ldap/commandline.c	2005-12-11 14:06:19.000000000 -0700
@@ -83,7 +83,7 @@
   if (passent == NULL)
     return -1;
   bzero (passent, sizeof (struct cpass));
-  (int) passent->sp_lstchg = passent->sp_min = passent->sp_max = -10;
+  passent->sp_lstchg = (int) (passent->sp_min = passent->sp_max = -10);
   passent->sp_warn = passent->sp_inact = passent->sp_expire = -10;
   passent->sp_flag = -10;
   passent->pw_gid = -10;
