--- cpu-1.4.3/src/plugins/ldap/ld.c.org	2006-05-31 17:21:56.429892571 +0200
+++ cpu-1.4.3/src/plugins/ldap/ld.c	2006-05-31 17:21:11.839892571 +0200
@@ -803,7 +803,9 @@ getlGid (LDAP * ld, char *groupn)
     return -10;
   if (gid != NULL)
     {
-      gid = ldap_get_values (ld, pos, gid)[0];
+	temp = ldap_get_values (ld, pos, gid);
+	gid = (char *)temp[0];
+//      gid = ldap_get_values (ld, pos, gid)[0];
       if (gid != NULL)
 	return atoi (gid);
       else
