diff -NrU5 kmess/crypt/sha1.cpp kmess/crypt/sha1.cpp
--- kmess/crypt/sha1.cpp	2008-06-28 17:01:11.000000000 +0200
+++ kmess/crypt/sha1.cpp	2008-06-28 16:44:42.000000000 +0200
@@ -36,10 +36,12 @@
 #include "bithelp.h"
 
 #include <stdio.h>
 #include <sys/stat.h>
 
+#include <cstring>
+
 Sha1::Sha1()
 {
 	context = new SHA1_CONTEXT;
 }
 
diff -NrU5 kmess-1.5.1.orig/kmess/notification/passivepopupview.h kmess-1.5.1/kmess/notification/passivepopupview.h
--- kmess/notification/passivepopupview.h	2008-06-28 17:02:28.000000000 +0200
+++ kmess/notification/passivepopupview.h	2008-06-28 16:44:43.000000000 +0200
@@ -41,11 +41,11 @@
 {
   Q_OBJECT
 
   public:  // Public methods
     // Class constructor
-    PassivePopupView(  QWidget *parent, QString icon, QString message, QString icon = QString::null );
+    PassivePopupView(  QWidget *parent, QString icon, QString message, QString status = QString::null );
     // Changes the popup icon
     void changeIcon( QString icon );
     // Changes the popup status icon
     void changeStatusIcon( QString status );
     // Changes the popup message
