From fd78590b182c1b91964826585fa8ed5d8539b6eb Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Thu, 1 Nov 2018 18:40:21 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 exceptions.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/exceptions.cabal b/exceptions.cabal
index e139c92..348a900 100644
--- a/exceptions.cabal
+++ b/exceptions.cabal
@@ -35,7 +35,7 @@ library
   build-depends:
     base                       >= 4.3      && < 5,
     stm                        >= 2.2      && < 3,
-    template-haskell           >= 2.2      && < 2.14,
+    template-haskell           >= 2.2,
     transformers               >= 0.2      && < 0.6,
     transformers-compat        >= 0.3      && < 0.7,
     mtl                        >= 2.0      && < 2.3
-- 
2.19.1

