From 00bb6f18d858949cec82fd9db8f79670bd5e03b0 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Sun, 24 Sep 2017 13:28:13 +0200
Subject: [PATCH] fix dependency on template-haskell

---
 exceptions.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/exceptions.cabal b/exceptions.cabal
index 402bd06..44cec9c 100644
--- a/exceptions.cabal
+++ b/exceptions.cabal
@@ -35,7 +35,7 @@ library
   build-depends:
     base                       >= 4.3      && < 5,
     stm                        >= 2.2      && < 3,
-    template-haskell           >= 2.2      && < 2.12,
+    template-haskell           >= 2.2      && < 2.13,
     transformers               >= 0.2      && < 0.6,
     transformers-compat        >= 0.3      && < 0.6,
     mtl                        >= 2.0      && < 2.3
-- 
2.14.1

