--- aboot-0.9b/Makefile	2002-09-18 19:48:21.000000000 +0000
+++ aboot-0.9b-ch/Makefile	2006-05-05 13:03:00.000000000 +0000
@@ -36,7 +36,7 @@
 TOP		= $(shell pwd)
 ifeq ($(TESTING),)
 CPPFLAGS	= $(CFGDEFS) -I$(TOP)/include -I$(KSRC)/include
-CFLAGS		= $(CPPFLAGS) -D__KERNEL__ -mcpu=ev4 -Os -Wall -fno-builtin -mno-fp-regs -ffixed-8
+CFLAGS		= $(CPPFLAGS) -D__KERNEL__ -Os -Wall -fno-builtin -mno-fp-regs -ffixed-8
 else
 CPPFLAGS	= -DTESTING $(CFGDEFS) -I$(TOP)/include -I$(KSRC)/include
 CFLAGS		= $(CPPFLAGS) -O -g3 -Wall -D__KERNEL__ -ffixed-8
