From a7d1e385caaf9fe831ad8733bfc480900abe9c65 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Tue, 30 Jan 2018 17:09:37 +0100
Subject: [PATCH] fix dependency on vector

---
 vector-algorithms.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vector-algorithms.cabal b/vector-algorithms.cabal
index e0e8af9..306129f 100644
--- a/vector-algorithms.cabal
+++ b/vector-algorithms.cabal
@@ -44,7 +44,7 @@ library
   hs-source-dirs: src
 
   build-depends: base >= 4.5 && < 5,
-                 vector >= 0.6 && < 0.12,
+                 vector >= 0.6 && < 0.13,
                  primitive >=0.3 && <0.7,
                  bytestring >= 0.9 && < 1.0
 
-- 
2.15.1

