From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Subject: [PATCH] Fix build against musl libc
Date: Tue,  3 Jun 2025 00:08:58 +0200

Origin: Source Mage
Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 netifaces.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/netifaces.c b/netifaces.c
index 839c42cb3001..8b4bd193334c 100644
--- a/netifaces.c
+++ b/netifaces.c
@@ -1805,9 +1805,6 @@ gateways (PyObject *self)
         sizeof(sanl_from),
         &iov,
         1,
-        NULL,
-        0,
-        0
       };
       int nllen;
 
