--- boost/asio/ssl/impl/context.ipp.orig	2015-03-23 15:24:12.000000000 -0500
+++ boost/asio/ssl/impl/context.ipp	2015-11-11 21:43:08.551733213 -0500
@@ -89,6 +89,8 @@
     handle_ = ::SSL_CTX_new(::SSLv2_server_method());
     break;
 #endif // defined(OPENSSL_NO_SSL2)
+#if defined(HAVE_SSLV3_METHOD) && defined(HAVE_SSLV3_SERVER_METHOD) && \
+    defined(HAVE_SSLV3_CLIENT_METHOD)
   case context::sslv3:
     handle_ = ::SSL_CTX_new(::SSLv3_method());
     break;
@@ -98,6 +100,7 @@
   case context::sslv3_server:
     handle_ = ::SSL_CTX_new(::SSLv3_server_method());
     break;
+#endif
   case context::tlsv1:
     handle_ = ::SSL_CTX_new(::TLSv1_method());
     break;
