--- trunk/boost/archive/polymorphic_iarchive.hpp (revision 43694)
+++ trunk/boost/archive/polymorphic_iarchive.hpp (revision 43958)
@@ -18,4 +18,5 @@
 
 #include <cstddef> // std::size_t
+#include <climits> // ULONG_MAX 
 #include <boost/config.hpp>
 
@@ -39,7 +40,14 @@
 // if there is no 64 bit int or if its the same as a long
 // we shouldn't define separate functions for int64 data types.
-#if defined(BOOST_NO_INT64_T) \
-    || (ULONG_MAX != 0xffffffff && ULONG_MAX == 18446744073709551615u) // 2**64 - 1
-#   define BOOST_NO_INTRINSIC_INT64_T
+#if defined(BOOST_NO_INT64_T)
+    #define BOOST_NO_INTRINSIC_INT64_T
+#else 
+    #if defined(ULONG_MAX)
+        #if(ULONG_MAX != 0xffffffff && ULONG_MAX == 18446744073709551615u) // 2**64 - 1
+            #define BOOST_NO_INTRINSIC_INT64_T
+        #endif
+    #else 
+        #define BOOST_NO_INTRINSIC_INT64_T
+    #endif
 #endif
 
--- trunk/boost/archive/polymorphic_oarchive.hpp (revision 43694)
+++ trunk/boost/archive/polymorphic_oarchive.hpp (revision 43958)
@@ -18,4 +18,5 @@
 
 #include <cstddef> // size_t
+#include <climits> // ULONG_MAX 
 #include <string>
 
@@ -38,7 +39,14 @@
 // if there is no 64 bit int or if its the same as a long
 // we shouldn't define separate functions for int64 data types.
-#if defined(BOOST_NO_INT64_T) \
-    || (ULONG_MAX != 0xffffffff && ULONG_MAX == 18446744073709551615u) // 2**64 - 1
-#   define BOOST_NO_INTRINSIC_INT64_T
+#if defined(BOOST_NO_INT64_T)
+    #define BOOST_NO_INTRINSIC_INT64_T
+#else 
+    #if defined(ULONG_MAX)
+        #if(ULONG_MAX != 0xffffffff && ULONG_MAX == 18446744073709551615u) // 2**64 - 1
+            #define BOOST_NO_INTRINSIC_INT64_T
+        #endif
+    #else 
+        #define BOOST_NO_INTRINSIC_INT64_T
+    #endif
 #endif
 
