--- 1.1/boost/numeric/conversion/converter_policies.hpp	2009-07-13 14:41:34 +01:00
+++ edited/boost/numeric/conversion/converter_policies.hpp	2011-01-14 12:06:18 +00:00
@@ -21,6 +21,8 @@
 #include "boost/mpl/if.hpp"
 #include "boost/mpl/integral_c.hpp"
 
+#include "boost/throw_exception.hpp"
+
 namespace boost { namespace numeric
 {
 
@@ -159,9 +161,9 @@
   void operator() ( range_check_result r ) // throw(negative_overflow,positive_overflow)
   {
     if ( r == cNegOverflow )
-      throw negative_overflow() ;
+      boost::throw_exception(negative_overflow()) ;
     else if ( r == cPosOverflow )
-           throw positive_overflow() ;
+           boost::throw_exception(positive_overflow()) ;
   }
 } ;
 

