diff -Naur squirrelmail-1.4.4.orig/functions/addressbook.php squirrelmail-1.4.4/functions/addressbook.php
--- squirrelmail-1.4.4.orig/functions/addressbook.php	2005-02-01 10:59:05.572877664 -0800
+++ squirrelmail-1.4.4/functions/addressbook.php	2005-02-01 10:59:12.050892856 -0800
@@ -544,7 +544,7 @@
             return -1;
         }
     }
-    return (strtolower($a['name']) > strtolower($b['name'])) ? 1 : -1;
+    return (strtolower($a['lastname']) > strtolower($b['lastname'])) ? 1 : -1;
 }
 
 
