From 00be753f0b2b5f71293b1b01602e059fd03982c6 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Sun, 6 Apr 2014 19:06:47 +0200
Subject: [PATCH] fix freetype include path

---
 src/xftglyphs.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/xftglyphs.c b/src/xftglyphs.c
index 2f3dc5a..602ef0f 100644
--- a/src/xftglyphs.c
+++ b/src/xftglyphs.c
@@ -21,10 +21,10 @@
  */
 
 #include "xftint.h"
-#include <freetype/ftoutln.h>
-#include <freetype/ftlcdfil.h>
+#include <freetype2/ftoutln.h>
+#include <freetype2/ftlcdfil.h>
 
-#include <freetype/ftsynth.h>
+#include <freetype2/ftsynth.h>
 
 /*
  * Validate the memory info for a font
-- 
1.9.0

