diff -upNr search/Beagle.Search.Tiles/Tile.cs search/Beagle.Search.Tiles/Tile.cs
--- search/Beagle.Search.Tiles/Tile.cs	2008-06-12 07:41:19.000000000 +0200
+++ search/Beagle.Search.Tiles/Tile.cs	2008-07-15 14:03:26.000000000 +0200
@@ -348,7 +348,7 @@ namespace Beagle.Search.Tiles {
 #else
 			MimeApplication app;
 			app = Mime.GetDefaultApplication (mimetype);
-			bool expect_uris = app.SupportUris ();
+			bool expect_uris = app.SupportsUris ();
 			path = hit.Path;

 			if (app == null) {
