--- tools/Settings.cs	2007-11-27 02:49:57.000000000 +0100
+++ tools/Settings.cs_new	2007-12-16 18:59:15.000000000 +0100
@@ -887,14 +887,14 @@
 			column.Title = Catalog.GetString ("Name");
 			CellRendererText renderer = new CellRendererText ();
 			column.PackStart (renderer, true);
-			column.SetCellDataFunc (renderer, NameCellFunc);
+			column.SetCellDataFunc (renderer, new TreeCellDataFunc (NameCellFunc));
 			AppendColumn (column);
 
                         column = new TreeViewColumn ();
                         column.Title = Catalog.GetString ("Address");
                         renderer = new CellRendererText ();
                         column.PackStart (renderer, true);
-                        column.SetCellDataFunc (renderer, AddressCellFunc);
+                        column.SetCellDataFunc (renderer, new TreeCellDataFunc (AddressCellFunc));
                         AppendColumn (column);
 		
 		}
