From 423622ae1b1b13de3d6df7a15cf302912ba14ef5 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <bwlf@bandrate.org>
Date: Fri, 2 Nov 2018 14:26:53 +0100
Subject: [PATCH] fix compatibility with ghc 8.6.1

---
 feed.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/feed.cabal b/feed.cabal
index 4dc9285..d0f43b0 100644
--- a/feed.cabal
+++ b/feed.cabal
@@ -71,7 +71,7 @@ library
     Data.Text.Util
     Data.XML.Compat
   build-depends:
-      base >= 4 && < 4.11
+      base >= 4
     , base-compat == 0.9.*
     , bytestring >= 0.9 && < 0.11
     , old-locale == 1.0.*
@@ -102,7 +102,7 @@ test-suite tests
     Text.RSS.Tests
     Text.RSS.Utils
   build-depends:
-      base >= 4 && < 4.11
+      base >= 4
     , HUnit >= 1.2 && < 1.7
     , base-compat == 0.9.*
     , feed
-- 
2.19.1

