From 1f76823c420ea9f8dc6a8a7dfc6b989145b206ce Mon Sep 17 00:00:00 2001
From: Treeve Jelbert <treeve@sourcemage.org>
Date: Tue, 30 Jan 2018 17:58:45 +0100
Subject: [PATCH] kjsembed optional

---
 CMakeLists.txt             | 8 +++++++-
 dataengines/CMakeLists.txt | 5 ++++-
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9977c37f..cad2eb76 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,7 +25,7 @@ include(ECMQtDeclareLoggingCategory)
 include(KDEPackageAppTemplates)
 
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
-                    Plasma DocTools Runner JsEmbed NotifyConfig Su NewStuff Wallet KCMUtils
+                    Plasma DocTools Runner NotifyConfig Su NewStuff Wallet KCMUtils
                     IdleTime Declarative TextWidgets KDELibs4Support Crash GlobalAccel
                     DBusAddons Wayland CoreAddons)
 find_package(KF5NetworkManagerQt ${KF5_MIN_VERSION})
@@ -45,6 +45,12 @@ set_package_properties(KF5Baloo PROPERTIES DESCRIPTION "File Searching"
                        PURPOSE "Needed for the File Search runner."
                       )
 
+find_package(KF5JsEmbed)
+set_package_properties(KF5JsEmbed PROPERTIES DESCRIPTION "scripting"
+                       TYPE OPTIONAL
+                       PURPOSE "Needed for sharing dataengine"
+                      )
+
 find_package(KF5TextEditor)
 find_package(KWinDBusInterface CONFIG REQUIRED)
 
diff --git a/dataengines/CMakeLists.txt b/dataengines/CMakeLists.txt
index 9fcdb57a..e730814f 100644
--- a/dataengines/CMakeLists.txt
+++ b/dataengines/CMakeLists.txt
@@ -25,7 +25,10 @@ add_subdirectory(soliddevice)
 add_subdirectory(time)
 add_subdirectory(weather)
 add_subdirectory(statusnotifieritem)
-add_subdirectory(share)
+
+if(KF5JsEmbed_FOUND)
+  add_subdirectory(share)
+endif()
 
 if(Boost_FOUND AND KdepimLibs_FOUND)
    if(Akonadi_FOUND)
-- 
2.16.0

