From 48fc738b16964ddd1fcc057c3558500694f4ac47 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.net>
Date: Sun, 14 Sep 2014 19:30:04 +0200
Subject: [PATCH] remove -fvisibility-inlines-hidden from CXXFLAGS

---
 build/aclocal/visibility.m4 | 2 +-
 configure                   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/aclocal/visibility.m4 b/build/aclocal/visibility.m4
index 05f9b80..61f2160 100644
--- a/build/aclocal/visibility.m4
+++ b/build/aclocal/visibility.m4
@@ -30,7 +30,7 @@ AC_DEFUN([WX_VISIBILITY],
   AC_REQUIRE([AC_PROG_CC])
   if test -n "$GCC"; then
     CFLAGS_VISIBILITY="-fvisibility=hidden"
-    CXXFLAGS_VISIBILITY="-fvisibility=hidden -fvisibility-inlines-hidden"
+    CXXFLAGS_VISIBILITY="-fvisibility=hidden"
     AC_MSG_CHECKING([for symbols visibility support])
     AC_CACHE_VAL(wx_cv_cc_visibility, [
       wx_save_CXXFLAGS="$CXXFLAGS"
diff --git a/configure b/configure
index c7291e2..5275d33 100755
--- a/configure
+++ b/configure
@@ -28904,7 +28904,7 @@ $as_echo "$wx_cv_version_script" >&6; }
 
   if test -n "$GCC"; then
     CFLAGS_VISIBILITY="-fvisibility=hidden"
-    CXXFLAGS_VISIBILITY="-fvisibility=hidden -fvisibility-inlines-hidden"
+    CXXFLAGS_VISIBILITY="-fvisibility=hidden"
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for symbols visibility support" >&5
 $as_echo_n "checking for symbols visibility support... " >&6; }
     if ${wx_cv_cc_visibility+:} false; then :
-- 
2.1.0

