--- Makefile.orig	2009-12-21 05:46:31.000000000 +0300
+++ Makefile	2010-01-06 17:26:14.924157579 +0300
@@ -23,7 +23,7 @@
 #CFLAGS = -I. -O2 -fomit-frame-pointer -Wall
 #CFLAGS = -I. -O3 -fomit-frame-pointer -Wall
 #CFLAGS = -I. -Os -fomit-frame-pointer -Wall
-CFLAGS = -I. -O3 -funroll-loops -fomit-frame-pointer -Wall -Wshadow
+#CFLAGS = -I. -O3 -funroll-loops -fomit-frame-pointer -Wall -Wshadow
 #CFLAGS = -I${ZINC} -I. -O3 -fomit-frame-pointer -Wall
 # [note that -Wall is a gcc-specific compilation flag ("all warnings on")]
 LDFLAGS =

