*** a/builtins/evalfile.c	2019-07-20 16:16:08.000000000 -0400
--- b/builtins/evalfile.c	2022-12-22 12:13:08.000000000 -0500
***************
*** 267,271 ****
  
    /* set the flags to be passed to parse_and_execute */
!   pflags = SEVAL_RESETLINE;
    pflags |= (flags & FEVAL_HISTORY) ? 0 : SEVAL_NOHIST;
  
--- 267,271 ----
  
    /* set the flags to be passed to parse_and_execute */
!   pflags = SEVAL_RESETLINE|SEVAL_NOOPTIMIZE;
    pflags |= (flags & FEVAL_HISTORY) ? 0 : SEVAL_NOHIST;
  
