From 3f8a54d95b72f37d0458d7d6a69ea6a90baad751 Mon Sep 17 00:00:00 2001
From: Emanuele Torre <torreemanuele6@gmail.com>
Date: Fri, 1 May 2020 03:24:25 +0200
Subject: [PATCH] Print X window IDs in uppercase for consistency with BSPWM,
 xlsw & xdo

---
 xqp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xqp.c b/xqp.c
index 48315c5..be97917 100644
--- a/xqp.c
+++ b/xqp.c
@@ -50,7 +50,7 @@ int main(int argc, char *argv[])
 				}
 			}
 		} else {
-			printf("0x%08x\n", qpr->child);
+			printf("0x%08X\n", qpr->child);
 		}
 	} else {
 		ret = EXIT_FAILURE;
