diff -urpN samba-3.0.8.orig/source/python/setup.py samba-3.0.8/source/python/setup.py
--- samba-3.0.8.orig/source/python/setup.py	2004-11-08 16:08:15.818787448 -0800
+++ samba-3.0.8/source/python/setup.py	2004-11-08 16:08:59.268182136 -0800
@@ -60,6 +60,8 @@ for lib in string.split(samba_libs):
     if lib[0:2] == "-W":
         # Skip linker flags
         continue
+    if lib in ("-s", "-z", "combreloc"):
+        continue
     print "Unknown entry '%s' in $LIBS variable passed to setup.py" % lib
     sys.exit(1)
 
