--- branches/SAMBA_3_0_25/source/python/setup.py	2007/02/05 15:28:11	21153
+++ branches/SAMBA_3_0_25/source/python/setup.py	2007/05/14 09:47:58	22840
@@ -67,6 +67,8 @@
         libraries.append(lib[2:])
     elif lib[0:8] == ("-pthread"):
         pass # Skip linker flags
+    elif lib[0:4] == ("-pie"):
+        pass # Skip linker flags
     elif lib[0:2] == "-L":
         library_dirs.append(lib[2:])
     elif lib[0:2] in ("-W","-s"):
