diff -Naur ./jwm-2.0.1/example.jwmrc jwm-2.0.1-fixed/example.jwmrc
--- ./jwm-2.0.1/example.jwmrc	2007-07-05 03:56:41.000000000 +0200
+++ jwm-2.0.1-fixed/example.jwmrc	2007-08-25 23:26:07.000000000 +0200
@@ -1,36 +1,8 @@
 <?xml version="1.0"?>
 
 <JWM>
-
-   <!-- The root menu, if this is undefined you will not get a menu. -->
-   <!-- Additional RootMenu attributes: onroot, labeled, label -->
-   <RootMenu height="32" onroot="123">
-      <Program icon="rxvt.png" label="Terminal">rxvt</Program>
-
-      <Menu icon="folder.png" label="Applications">
-         <Program icon="audacious.png" label="Audacious">audacious</Program>
-         <Program icon="dia.png" label="Dia">dia</Program>
-         <Program icon="firefox.png" label="Firefox">firefox</Program>
-         <Program icon="pidgin.png" label="Pidgin">pidgin</Program>
-         <Program icon="gimp.png" label="Gimp">gimp</Program>
-         <Program icon="gtk-gnutella.png" label="gtk-gnutella">
-            gtk-gnutella
-         </Program>
-         <Program icon="gxine.png" label="gxine">gxine</Program>
-         <Program icon="ooffice.png" label="Open Office">ooffice</Program>
-      </Menu>
-      <Menu icon="folder.png" label="Utilities">
-         <Program icon="xcalc.png">xcalc</Program>
-         <Program icon="xfontsel.png">xfontsel</Program>
-         <Program icon="xmag.png">xmag</Program>
-         <Program icon="xprop.png" label="xprop">
-            xprop | xmessage -file -
-         </Program>
-      </Menu>
-      <Separator/>
-      <Restart label="Restart" icon="restart.png"/>
-      <Exit label="Exit" confirm="true" icon="exit.png"/>
-   </RootMenu>
+   # include the Rootmenu
+   <Include>/etc/jwmrc-menu</Include>
 
    <Group>
       <Class>Pidgin</Class>
diff -Naur ./jwm-2.0.1/jwmrc-menu jwm-2.0.1-fixed/jwmrc-menu
--- ./jwm-2.0.1/jwmrc-menu	1970-01-01 01:00:00.000000000 +0100
+++ jwm-2.0.1-fixed/jwmrc-menu	2007-08-25 23:26:27.000000000 +0200
@@ -0,0 +1,31 @@
+<JWM>
+
+<!-- The root menu, if this is undefined you will not get a menu. -->
+<!-- Additional RootMenu attributes: onroot, labeled, label -->
+<RootMenu height="32" onroot="123">
+   <Program icon="rxvt.png" label="Terminal">rxvt</Program>
+   <Menu icon="folder.png" label="Applications">
+      <Program icon="audacious.png" label="Audacious">audacious</Program>
+      <Program icon="dia.png" label="Dia">dia</Program>
+      <Program icon="firefox.png" label="Firefox">firefox</Program>
+      <Program icon="pidgin.png" label="Pidgin">pidgin</Program>
+      <Program icon="gimp.png" label="Gimp">gimp</Program>
+      <Program icon="gtk-gnutella.png" label="gtk-gnutella">
+         gtk-gnutella
+      </Program>
+      <Program icon="gxine.png" label="gxine">gxine</Program>
+      <Program icon="ooffice.png" label="Open Office">ooffice</Program>
+   </Menu>
+   <Menu icon="folder.png" label="Utilities">
+      <Program icon="xcalc.png">xcalc</Program>
+      <Program icon="xfontsel.png">xfontsel</Program>
+      <Program icon="xmag.png">xmag</Program>
+      <Program icon="xprop.png" label="xprop">
+         xprop | xmessage -file -
+      </Program>
+   </Menu>
+   <Separator/>
+   <Restart label="Restart" icon="restart.png"/>
+   <Exit label="Exit" confirm="true" icon="exit.png"/>
+</RootMenu>
+</JWM>
