From 9a2e084b23fda0cac94de77e50168fb2e29bb5a5 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <Florian.Franzmann@bandrate.org>
Date: Sun, 2 May 2021 16:42:36 +0200
Subject: [PATCH] Fix missing include

---
 example/cxxopts.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/example/cxxopts.hpp b/example/cxxopts.hpp
index 6fd170d..c0fc1b7 100644
--- a/example/cxxopts.hpp
+++ b/example/cxxopts.hpp
@@ -29,6 +29,7 @@ THE SOFTWARE.
 #include <cctype>
 #include <exception>
 #include <iostream>
+#include <limits>
 #include <map>
 #include <memory>
 #include <regex>
-- 
2.31.1

